// @sap-ui/fx-components/dist/components/radiobutton/index.cjs
'use strict';

var react = require('react');
var tailwindMerge = require('tailwind-merge');
var reactI18next = require('react-i18next');
var jsxRuntime = require('react/jsx-runtime');

// src/components/radiobutton/RadioButton.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}

// src/types/radiobutton.ts
var RadioButtonWrappingType = /* @__PURE__ */ ((RadioButtonWrappingType2) => {
  RadioButtonWrappingType2["None"] = "None";
  RadioButtonWrappingType2["Normal"] = "Normal";
  return RadioButtonWrappingType2;
})(RadioButtonWrappingType || {});
var RadioButtonSize = /* @__PURE__ */ ((RadioButtonSize2) => {
  RadioButtonSize2["Small"] = "Small";
  RadioButtonSize2["Large"] = "Large";
  return RadioButtonSize2;
})(RadioButtonSize || {});
var RadioButtonValueState = /* @__PURE__ */ ((RadioButtonValueState2) => {
  RadioButtonValueState2["None"] = "None";
  RadioButtonValueState2["Positive"] = "Positive";
  RadioButtonValueState2["Critical"] = "Critical";
  RadioButtonValueState2["Negative"] = "Negative";
  RadioButtonValueState2["Information"] = "Information";
  return RadioButtonValueState2;
})(RadioButtonValueState || {});
var radioGroups = /* @__PURE__ */ new Map();
function getOrderedGroupEntries(name) {
  const entries = radioGroups.get(name) ?? [];
  const connectedEntries = entries.filter((entry) => entry.getNode()?.isConnected);
  if (connectedEntries.length !== entries.length) {
    if (connectedEntries.length === 0) {
      radioGroups.delete(name);
    } else {
      radioGroups.set(name, connectedEntries);
    }
  }
  return [...connectedEntries].sort((entryA, entryB) => {
    const nodeA = entryA.getNode();
    const nodeB = entryB.getNode();
    if (!nodeA || !nodeB || nodeA === nodeB) {
      return 0;
    }
    const position = nodeA.compareDocumentPosition(nodeB);
    if (position & Node.DOCUMENT_POSITION_FOLLOWING) {
      return -1;
    }
    if (position & Node.DOCUMENT_POSITION_PRECEDING) {
      return 1;
    }
    return 0;
  });
}
function updateGroupTabIndex(name) {
  const entries = getOrderedGroupEntries(name);
  if (entries.length === 0) {
    return;
  }
  const focusableEntries = entries.filter((entry) => !entry.isDisabled());
  const activeEntry = focusableEntries.find((entry) => entry.isChecked()) ?? focusableEntries[0] ?? null;
  entries.forEach((entry) => {
    entry.setTabIndex(!entry.isDisabled() && entry === activeEntry ? 0 : -1);
  });
}
function registerGroupEntry(name, entry) {
  const existingEntries = radioGroups.get(name) ?? [];
  const nextEntries = existingEntries.filter((existingEntry) => existingEntry.id !== entry.id);
  nextEntries.push(entry);
  radioGroups.set(name, nextEntries);
  updateGroupTabIndex(name);
}
function unregisterGroupEntry(name, id) {
  const existingEntries = radioGroups.get(name);
  if (!existingEntries) {
    return;
  }
  const nextEntries = existingEntries.filter((entry) => entry.id !== id);
  if (nextEntries.length === 0) {
    radioGroups.delete(name);
    return;
  }
  radioGroups.set(name, nextEntries);
  updateGroupTabIndex(name);
}
function getAdjacentGroupEntry(name, currentId, direction) {
  const focusableEntries = getOrderedGroupEntries(name).filter((entry) => !entry.isDisabled());
  if (focusableEntries.length <= 1) {
    return null;
  }
  const currentIndex = focusableEntries.findIndex((entry) => entry.id === currentId);
  const baseIndex = currentIndex >= 0 ? currentIndex : 0;
  const nextIndex = (baseIndex + direction + focusableEntries.length) % focusableEntries.length;
  return focusableEntries[nextIndex] ?? null;
}
function moveSelectionInGroup(name, currentId, direction, originalEvent) {
  const nextEntry = getAdjacentGroupEntry(name, currentId, direction);
  if (!nextEntry) {
    return;
  }
  nextEntry.focus();
  if (!nextEntry.isReadonly()) {
    nextEntry.select(originalEvent);
    return;
  }
  updateGroupTabIndex(name);
}
function uncheckOthersInGroup(groupName, currentId) {
  getOrderedGroupEntries(groupName).forEach((entry) => {
    if (entry.id !== currentId && entry.isChecked()) {
      entry.setChecked(false);
    }
  });
}
function getKeyNavigationDirection(key) {
  if (key === " " || key === "Enter") return "select";
  const isRTL = document.documentElement.dir === "rtl";
  if (key === "ArrowDown" || key === "Down" || key === "ArrowRight" && !isRTL || key === "Right" && !isRTL || key === "ArrowLeft" && isRTL || key === "Left" && isRTL) {
    return 1;
  }
  if (key === "ArrowUp" || key === "Up" || key === "ArrowLeft" && !isRTL || key === "Left" && !isRTL || key === "ArrowRight" && isRTL || key === "Right" && isRTL) {
    return -1;
  }
  return null;
}
function getEffectiveTabIndex(disabled, tabIndex, groupName, groupTabIndex) {
  if (disabled) return -1;
  if (tabIndex !== void 0) return tabIndex;
  if (groupName) return groupTabIndex ?? 0;
  return 0;
}
function getValueStateBorderClass(valueState) {
  const borderMap = {
    ["Positive" /* Positive */]: "border-sapphire-positive",
    ["Negative" /* Negative */]: "border-sapphire-negative",
    ["Critical" /* Critical */]: "border-sapphire-warning",
    ["Information" /* Information */]: "border-sapphire-info"
  };
  return borderMap[valueState] ?? "";
}
function getValueStateCheckedBorderClass(valueState) {
  const checkedBorderMap = {
    ["Positive" /* Positive */]: "border-sapphire-positive",
    ["Negative" /* Negative */]: "border-sapphire-negative",
    ["Critical" /* Critical */]: "border-sapphire-warning",
    ["Information" /* Information */]: "border-sapphire-info"
  };
  return checkedBorderMap[valueState] ?? "border-sapphire-border-secondary";
}
function getInnerDotColorClass(valueState) {
  const dotMap = {
    ["Positive" /* Positive */]: "bg-sapphire-positive",
    ["Negative" /* Negative */]: "bg-sapphire-negative",
    ["Critical" /* Critical */]: "bg-sapphire-warning",
    ["Information" /* Information */]: "bg-sapphire-info"
  };
  return dotMap[valueState] ?? "bg-sapphire-text-accent";
}
function getSizeClasses(isSmall, hasLabel) {
  if (isSmall) {
    return hasLabel ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2";
  }
  return hasLabel ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]";
}
function getFocusRingClasses(hasLabel, isSmall) {
  if (!hasLabel) return "after:inset-[5px] after:rounded-lg";
  if (isSmall) return "after:inset-x-0 after:inset-y-0";
  return "after:left-0 after:right-[2px] after:inset-y-[5px]";
}
var valueStateI18nKeys = {
  ["Negative" /* Negative */]: "RADIOBUTTON_VALUE_STATE_ERROR",
  ["Critical" /* Critical */]: "RADIOBUTTON_VALUE_STATE_WARNING",
  ["Positive" /* Positive */]: "RADIOBUTTON_VALUE_STATE_SUCCESS",
  ["Information" /* Information */]: "RADIOBUTTON_VALUE_STATE_INFORMATION"
};
function RadioCircle({
  isReadonly,
  isSmall,
  isChecked,
  hasWrappingLabel,
  isInteractive,
  valueState,
  svgR
}) {
  if (isReadonly) {
    return /* @__PURE__ */ jsxRuntime.jsxs(
      "span",
      {
        className: cn(
          "relative inline-flex items-center justify-center shrink-0",
          isSmall ? "size-4" : "size-[22px]",
          hasWrappingLabel && "self-start",
          isSmall && hasWrappingLabel && "mt-0.5"
        ),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            "svg",
            {
              className: "absolute inset-0 w-full h-full overflow-visible",
              focusable: "false",
              "aria-hidden": "true",
              children: /* @__PURE__ */ jsxRuntime.jsx(
                "circle",
                {
                  cx: "50%",
                  cy: "50%",
                  r: svgR,
                  fill: "var(--background-secondary)",
                  stroke: "var(--border-secondary)",
                  strokeWidth: "1",
                  strokeDasharray: "4, 2"
                }
              )
            }
          ),
          isChecked && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: cn(
                "relative rounded-full",
                isSmall ? "size-2" : "size-3",
                "bg-sapphire-text-secondary"
              )
            }
          )
        ]
      }
    );
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      className: cn(
        "relative inline-flex items-center justify-center shrink-0 rounded-full",
        isSmall ? "size-4" : "size-[22px]",
        hasWrappingLabel && "self-start",
        isSmall && hasWrappingLabel && "mt-0.5",
        "transition-colors",
        "border border-solid",
        !isChecked && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
        isChecked && getValueStateCheckedBorderClass(valueState),
        isChecked && "bg-sapphire-canvas-primary",
        !isChecked && getValueStateBorderClass(valueState),
        isInteractive && valueState === "None" /* None */ && !isChecked && "group-hover:border-sapphire-chrome-button-fg-selected group-hover:bg-sapphire-chrome-button-bg-selected",
        isInteractive && valueState === "None" /* None */ && isChecked && "group-hover:border-sapphire-border-accent group-hover:bg-sapphire-chrome-button-bg-selected"
      ),
      children: isChecked && /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: cn(
            "rounded-full",
            isSmall ? "size-2" : "size-3",
            getInnerDotColorClass(valueState)
          )
        }
      )
    }
  );
}
function RadioButton({
  checked: controlledChecked,
  defaultChecked = false,
  disabled = false,
  readonly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  valueState = "None" /* None */,
  valueStateMessage,
  name,
  value = "",
  wrappingType = "Normal" /* Normal */,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  ref
}) {
  const inputRef = react.useRef(null);
  const rootRef = react.useRef(null);
  const generatedId = react.useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = react.useState(defaultChecked);
  const [groupTabIndex, setGroupTabIndex] = react.useState(null);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const setChecked = react.useCallback(
    (checked) => {
      if (!isControlled) {
        setInternalChecked(checked);
      }
    },
    [isControlled]
  );
  const doSelect = react.useCallback(
    (originalEvent) => {
      if (disabled || readonly || isChecked) return;
      if (name) {
        uncheckOthersInGroup(name, id);
      }
      if (!isControlled) {
        setInternalChecked(true);
      }
      onChange?.({
        checked: true,
        value,
        originalEvent
      });
    },
    [disabled, readonly, isChecked, isControlled, name, id, onChange, value]
  );
  react.useEffect(() => {
    if (!name) {
      setGroupTabIndex(null);
      return;
    }
    const entry = {
      id,
      getNode: () => rootRef.current,
      isDisabled: () => disabled,
      isReadonly: () => readonly,
      isChecked: () => isChecked,
      setChecked,
      focus: () => rootRef.current?.focus(),
      select: doSelect,
      setTabIndex: setGroupTabIndex
    };
    registerGroupEntry(name, entry);
    return () => {
      unregisterGroupEntry(name, id);
    };
  }, [name, id, disabled, readonly, isChecked, setChecked, doSelect]);
  react.useEffect(() => {
    if (!name) {
      return;
    }
    updateGroupTabIndex(name);
  }, [name, isChecked, disabled]);
  react.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked]
  );
  const handleClick = react.useCallback(
    (e) => {
      if (disabled || readonly) {
        e.preventDefault();
        return;
      }
      doSelect(e);
    },
    [disabled, readonly, doSelect]
  );
  const handleMouseDown = react.useCallback(
    (e) => {
      if (disabled) {
        e.preventDefault();
      }
    },
    [disabled]
  );
  const handleKeyDown = react.useCallback(
    (e) => {
      if (disabled || readonly) return;
      const direction = getKeyNavigationDirection(e.key);
      if (name && (direction === 1 || direction === -1)) {
        e.preventDefault();
        moveSelectionInGroup(name, id, direction, e);
        return;
      }
      if (direction === "select") {
        e.preventDefault();
      }
    },
    [disabled, readonly, name, id]
  );
  const handleKeyUp = react.useCallback(
    (e) => {
      if (disabled || readonly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doSelect(e);
      }
    },
    [disabled, readonly, doSelect]
  );
  const labelContent = children ?? text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const effectiveTabIndex = getEffectiveTabIndex(disabled, tabIndex, name, groupTabIndex);
  const { t } = reactI18next.useTranslation("fx");
  const valueStateI18nKey = valueStateI18nKeys[valueState];
  const defaultValueStateText = valueStateI18nKey ? t(valueStateI18nKey) : void 0;
  const svgR = isSmall ? 7.5 : 10.5;
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        ref: inputRef,
        type: "radio",
        id,
        name,
        value,
        checked: isChecked,
        disabled,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: rootRef,
        role: "radio",
        "aria-checked": isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          accessibleDescriptionRef || null,
          valueState !== "None" /* None */ ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || void 0,
        "aria-invalid": valueState === "Negative" /* Negative */ || void 0,
        "aria-required": required || void 0,
        tabIndex: effectiveTabIndex,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        "data-radio-group": name,
        "data-radio-id": id,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          getSizeClasses(isSmall, !!labelContent),
          "text-sm",
          "rounded-lg",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          getFocusRingClasses(!!labelContent, isSmall),
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            RadioCircle,
            {
              isReadonly: readonly,
              isSmall,
              isChecked,
              hasWrappingLabel,
              isInteractive,
              valueState,
              svgR
            }
          ),
          labelContent && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" /* None */ && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage || defaultValueStateText }),
          accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}

exports.RadioButton = RadioButton;
exports.RadioButtonSize = RadioButtonSize;
exports.RadioButtonValueState = RadioButtonValueState;
exports.RadioButtonWrappingType = RadioButtonWrappingType;
