// @sap-ui/fx-components/dist/components/bar/index.cjs
'use strict';

var react = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

// src/components/bar/Bar.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var childStyles = [
  "[&>*]:mx-1 [&>*]:max-w-full [&>*]:box-border",
  "[&>:not(div)]:whitespace-nowrap [&>:not(div)]:overflow-hidden [&>:not(div)]:text-ellipsis"
].join(" ");
var barVariants = cva(
  [
    "flex items-center justify-between w-full overflow-hidden box-border"
  ],
  {
    variants: {
      design: {
        ["Header" /* Header */]: [
          "h-11",
          "bg-[var(--sapPageHeader_Background,var(--card))]",
          "shadow-[var(--sapContent_HeaderShadow,0_0_0.25rem_0_rgba(0,0,0,0.15))]"
        ],
        ["Subheader" /* Subheader */]: [
          "h-12 -mt-px",
          "bg-[var(--sapPageHeader_Background,var(--card))]",
          "shadow-[var(--sapContent_HeaderShadow,0_0_0.25rem_0_rgba(0,0,0,0.15))]"
        ],
        ["Footer" /* Footer */]: [
          "h-11 shadow-none",
          "bg-[var(--sapPageFooter_Background,var(--muted))]",
          "border-t border-[var(--sapPageFooter_BorderColor,var(--border))]"
        ],
        ["FloatingFooter" /* FloatingFooter */]: [
          "h-11 border-none",
          "bg-[var(--sapPageFooter_Background,var(--muted))]",
          "rounded-[var(--sapElement_BorderCornerRadius,var(--radius))]",
          "shadow-[var(--sapContent_Shadow1,0_0_1rem_0_rgba(0,0,0,0.15))]"
        ]
      }
    },
    defaultVariants: {
      design: "Header" /* Header */
    }
  }
);
function Bar({
  design = "Header" /* Header */,
  accessibleRole = "Toolbar" /* Toolbar */,
  accessibleName,
  accessibleNameRef,
  startContent,
  children,
  endContent,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const barRef = react.useRef(null);
  react.useImperativeHandle(
    ref,
    () => ({
      focus() {
        barRef.current?.focus();
      },
      blur() {
        barRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === barRef.current;
      },
      get nativeElement() {
        return barRef.current;
      }
    }),
    []
  );
  const isToolbar = accessibleRole !== "None";
  const effectiveRole = isToolbar ? "toolbar" : void 0;
  const effectiveAriaLabel = isToolbar ? accessibleName || (accessibleNameRef ? void 0 : design) : accessibleName;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: barRef,
      id,
      role: effectiveRole,
      "aria-label": accessibleNameRef ? void 0 : effectiveAriaLabel,
      "aria-labelledby": accessibleNameRef,
      className: cn(barVariants({ design }), className),
      style,
      "data-testid": dataTestId,
      "data-design": design,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(
              "flex items-center shrink basis-auto grow-0 ps-4 min-w-0",
              childStyles
            ),
            "data-part": "startContent",
            children: startContent
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(
              "flex items-center justify-center flex-auto px-2 min-w-0 overflow-hidden",
              childStyles
            ),
            "data-part": "midContent",
            children
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(
              "flex items-center shrink-0 grow-0 basis-auto pe-4 min-w-0",
              childStyles
            ),
            "data-part": "endContent",
            children: endContent
          }
        )
      ]
    }
  );
}

exports.Bar = Bar;
