// @sap-ui/fx-components/dist/components/link/index.cjs
'use strict';

var react = require('react');
var tailwindMerge = require('tailwind-merge');
var reactI18next = require('react-i18next');
var jsxRuntime = require('react/jsx-runtime');

// src/components/link/Link.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var SAFE_PROTOCOLS = ["http:", "https:", "mailto:", "tel:"];
var isSafeUrl = (url) => {
  try {
    const parsed = new URL(url, window.location.href);
    return SAFE_PROTOCOLS.includes(parsed.protocol);
  } catch {
    return false;
  }
};
var computeRel = (href, target, rel) => {
  if (rel) return rel;
  if (target === "_blank" && href) {
    try {
      const url = new URL(href, window.location.href);
      if (url.origin !== window.location.origin) {
        return "noreferrer noopener";
      }
    } catch {
    }
  }
  return void 0;
};
var linkVariants = cva(
  [
    // Layout
    "inline-flex items-center gap-0.5",
    // Typography — Figma: '72' Regular, 14px, line-height 20px
    "text-sm leading-5 font-normal cursor-pointer",
    // Transition
    "transition-colors",
    // Focus box: constant padding + transparent border to prevent layout shift.
    // Figma focus spec: border-focus, rounded 4px, px-4px py-2px.
    // Negative margins compensate for the padding+border so the link
    // sits flush in normal state.
    "px-1 py-0.5 -mx-1.5 -my-1",
    "border-2 border-transparent rounded-[4px]",
    "focus-visible:outline-none focus-visible:border-sapphire-border-focus"
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: [
          "text-sapphire-text-accent",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ],
        ["Subtle" /* Subtle */]: [
          "text-secondary-foreground",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ],
        ["Emphasized" /* Emphasized */]: [
          "text-sapphire-text-accent font-semibold",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ]
      },
      interactiveAreaSize: {
        ["Normal" /* Normal */]: "",
        ["Large" /* Large */]: "leading-6"
      },
      wrappingType: {
        ["None" /* None */]: "whitespace-nowrap max-w-full",
        ["Normal" /* Normal */]: "whitespace-normal break-words"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      interactiveAreaSize: "Normal" /* Normal */,
      wrappingType: "None" /* None */
    }
  }
);
function Link({
  design = "Default" /* Default */,
  interactiveAreaSize = "Normal" /* Normal */,
  wrappingType = "None" /* None */,
  href,
  target,
  rel,
  disabled = false,
  icon,
  endIcon,
  tooltip,
  children,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleRole = "Link" /* Link */,
  accessibilityAttributes,
  onClick,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  ref
}) {
  const linkRef = react.useRef(null);
  react.useImperativeHandle(
    ref,
    () => ({
      focus() {
        linkRef.current?.focus();
      },
      blur() {
        linkRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === linkRef.current;
      },
      get nativeElement() {
        return linkRef.current;
      }
    }),
    []
  );
  const { t } = reactI18next.useTranslation("fx");
  const linkTypeText = react.useMemo(() => {
    if (design === "Subtle" /* Subtle */) return t("LINK_SUBTLE");
    if (design === "Emphasized" /* Emphasized */) return t("LINK_EMPHASIZED");
    return void 0;
  }, [design, t]);
  const computedRel = react.useMemo(
    () => computeRel(href, target, rel),
    [href, target, rel]
  );
  const effectiveTabIndex = disabled ? -1 : tabIndex ?? 0;
  const handleClick = react.useCallback(
    (e) => {
      if (disabled) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        altKey: e.altKey,
        ctrlKey: e.ctrlKey,
        metaKey: e.metaKey,
        shiftKey: e.shiftKey,
        isKeyboard: false
      });
    },
    [disabled, onClick]
  );
  const handleMouseDown = react.useCallback(
    (_e) => {
      if (!disabled) {
        linkRef.current?.focus();
      }
    },
    [disabled]
  );
  const handleKeyDown = react.useCallback(
    (e) => {
      if (disabled) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " " && accessibleRole !== "Button" /* Button */) {
          return;
        }
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          altKey: e.altKey,
          ctrlKey: e.ctrlKey,
          metaKey: e.metaKey,
          shiftKey: e.shiftKey,
          isKeyboard: true
        });
        if (href && !disabled && isSafeUrl(href)) {
          window.location.href = href;
        }
      }
    },
    [disabled, onClick, href, accessibleRole]
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "a",
    {
      ref: linkRef,
      id,
      href: disabled ? void 0 : href,
      target,
      rel: computedRel,
      role: accessibleRole === "Button" /* Button */ ? "button" : void 0,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-describedby": accessibleDescription,
      "aria-disabled": disabled || void 0,
      "aria-expanded": accessibilityAttributes?.expanded,
      "aria-haspopup": accessibilityAttributes?.hasPopup,
      "aria-current": accessibilityAttributes?.current || void 0,
      title: tooltip,
      tabIndex: effectiveTabIndex,
      onClick: handleClick,
      onMouseDown: handleMouseDown,
      onKeyDown: handleKeyDown,
      className: cn(
        linkVariants({
          design,
          interactiveAreaSize,
          wrappingType
        }),
        disabled && "pointer-events-none cursor-default opacity-[var(--disabled-opacity)]",
        "group",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-shrink-0 size-3", "data-part": "icon", children: icon }),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            className: cn(
              wrappingType === "None" /* None */ && "overflow-hidden text-ellipsis min-w-0",
              design === "Subtle" /* Subtle */ && "underline decoration-dotted underline-offset-[0.25em]",
              (design === "Default" /* Default */ || design === "Emphasized" /* Emphasized */) && "group-hover:underline group-hover:decoration-dotted group-hover:underline-offset-[0.25em]"
            ),
            "data-part": "text",
            children
          }
        ),
        endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-shrink-0 size-3", "data-part": "end-icon", children: endIcon }),
        linkTypeText && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "sr-only", "data-part": "link-type", children: linkTypeText })
      ]
    }
  );
}

exports.Link = Link;
