// @sap-ui/fx-components/dist/components/progress-indicator/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/progress-indicator/ProgressIndicator.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData5,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData5 || pathData5.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData5.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M256 0q53 0 100 20t81.5 54.5T492 156t20 100-20 100-54.5 81.5T356 492t-100 20-100-20-81.5-54.5T20 356 0 256t20-100 54.5-81.5T156 20 256 0zm150 183q10-9 10-23 0-13-9.5-22.5T384 128t-22 9L186 308l-68-63q-9-9-22-9t-22.5 9.5T64 268q0 15 10 24l91 83q9 9 21 9 13 0 23-9z"
];
var SysEnter2Icon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SysEnter2Icon", iconName: "sys-enter-2" }
);
var pathData2 = [
  "M505 399q7 13 7 27 0 21-15.5 37.5T456 480H56q-25 0-40.5-16.5T0 426q0-14 7-27L208 59q17-27 48-27 14 0 27 6.5T304 59zM288 176q0-14-9-23t-23-9-23 9-9 23v96q0 14 9 23t23 9 23-9 9-23v-96zm-32 240q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9z"
];
var AlertIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "AlertIcon", iconName: "alert" }
);
var pathData3 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm45 256l74-73q9-11 9-23 0-13-9.5-22.5T352 128q-12 0-23 9l-73 74-73-74q-10-9-23-9t-22.5 9.5T128 160q0 12 9 23l74 73-74 73q-9 10-9 23t9.5 22.5T160 384t23-9l73-74 73 74q11 9 23 9 13 0 22.5-9.5T384 352t-9-23z"
];
var ErrorIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "ErrorIcon", iconName: "error" }
);
var pathData4 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm0 160q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9zm32 64q0-14-9-23t-23-9-23 9-9 23v160q0 14 9 23t23 9 23-9 9-23V224z"
];
var InformationIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "InformationIcon", iconName: "information" }
);
var trackBgClass = {
  ["None" /* None */]: "bg-sapphire-neutral-pressed-background-2",
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg",
  ["Information" /* Information */]: "bg-sapphire-info-bg"
};
var barColorClass = {
  ["None" /* None */]: "bg-sapphire-neutral-foreground-muted",
  ["Positive" /* Positive */]: "bg-sapphire-positive",
  ["Critical" /* Critical */]: "bg-sapphire-warning",
  ["Negative" /* Negative */]: "bg-sapphire-negative",
  ["Information" /* Information */]: "bg-sapphire-info"
};
var iconColorClass = {
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Information" /* Information */]: "text-sapphire-info"
};
var stateIconMap = {
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Critical" /* Critical */]: AlertIcon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Information" /* Information */]: InformationIcon
};
var stateLabel = {
  ["None" /* None */]: "",
  ["Positive" /* Positive */]: "Positive",
  ["Critical" /* Critical */]: "Critical",
  ["Negative" /* Negative */]: "Negative",
  ["Information" /* Information */]: "Information"
};
function ProgressIndicator({
  value = 0,
  displayValue,
  hideValue = false,
  disabled = false,
  valueState = "None" /* None */,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    []
  );
  const clampedValue = Math.min(100, Math.max(0, Number.isNaN(value) ? 0 : value));
  const valueText = displayValue || `${clampedValue}%`;
  const vs = valueState;
  const label = stateLabel[vs] || "";
  const ariaValueText = displayValue ? displayValue : label ? `${valueText} ${label}` : valueText;
  const StateIcon = stateIconMap[vs] || null;
  const hasIcon = vs !== "None" /* None */ && StateIcon;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      role: "progressbar",
      "aria-valuemin": 0,
      "aria-valuemax": 100,
      "aria-valuenow": clampedValue,
      "aria-valuetext": ariaValueText,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-disabled": disabled || void 0,
      tabIndex: -1,
      id,
      "data-testid": dataTestId,
      "data-part": "root",
      className: cn(
        "w-full",
        disabled && "opacity-40 pointer-events-none",
        className
      ),
      style,
      children: [
        !hideValue && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            "data-part": "value",
            className: "block text-sm text-sapphire-text-secondary leading-normal",
            children: valueText
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            "data-part": "indicator",
            className: cn("flex items-start w-full", hasIcon && "gap-1"),
            children: [
              /* @__PURE__ */ jsxRuntime.jsxs("div", { "data-part": "track-container", className: "relative h-4 flex-1", children: [
                /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    "data-part": "track",
                    className: cn(
                      "absolute left-0 right-0 top-1/2 -translate-y-1/2 h-1.5 rounded",
                      trackBgClass[vs] || trackBgClass["None" /* None */]
                    )
                  }
                ),
                /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    "data-part": "bar",
                    className: cn(
                      "absolute top-1/2 -translate-y-1/2 h-2.5 rounded-full transition-[width] duration-300 ease-linear",
                      barColorClass[vs] || barColorClass["None" /* None */],
                      clampedValue === 0 && "w-0"
                    ),
                    style: clampedValue > 0 ? { width: `${clampedValue}%` } : void 0
                  }
                ),
                /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    "data-part": "start-cap",
                    className: cn(
                      "absolute left-px top-1/2 -translate-y-1/2 size-1 rounded-full",
                      barColorClass[vs] || barColorClass["None" /* None */]
                    )
                  }
                ),
                /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    "data-part": "end-cap",
                    className: cn(
                      "absolute right-px top-1/2 -translate-y-1/2 size-1 rounded-full",
                      barColorClass[vs] || barColorClass["None" /* None */]
                    )
                  }
                )
              ] }),
              hasIcon && /* @__PURE__ */ jsxRuntime.jsx(
                StateIcon,
                {
                  "data-part": "icon",
                  className: cn("h-4 w-4 shrink-0", iconColorClass[vs])
                }
              )
            ]
          }
        )
      ]
    }
  );
}

exports.ProgressIndicator = ProgressIndicator;
