// @sap-ui/fx-components/dist/components/list/index.cjs
'use strict';

var React3 = require('react');
var tailwindMerge = require('tailwind-merge');
var lucideReact = require('lucide-react');
var jsxRuntime = require('react/jsx-runtime');
var reactI18next = require('react-i18next');

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var React3__namespace = /*#__PURE__*/_interopNamespace(React3);

// src/components/list/List.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var dotVariants = cva("rounded-full bg-current inline-block", {
  variants: {
    size: {
      ["S" /* S */]: "w-2 h-2",
      ["M" /* M */]: "w-4 h-4",
      ["L" /* L */]: "w-8 h-8"
    }
  },
  defaultVariants: { size: "M" /* M */ }
});
var dotGapClasses = {
  ["S" /* S */]: "gap-px",
  ["M" /* M */]: "gap-[0.1875rem]",
  ["L" /* L */]: "gap-1"
};
function BusyIndicator({
  size = "M" /* M */,
  textPlacement = "Bottom" /* Bottom */,
  active = false,
  delay = 1e3,
  text,
  children,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = React3.useRef(null);
  const overlayRef = React3.useRef(null);
  const [isBusy, setIsBusy] = React3.useState(false);
  const labelId = React3.useId();
  const hasChildren = React3__namespace.default.Children.count(children) > 0;
  const normalizedSize = size || "M" /* M */;
  const isTopPlacement = textPlacement === "Top";
  React3.useEffect(() => {
    if (!active) {
      setIsBusy(false);
      return;
    }
    if (delay <= 0) {
      setIsBusy(true);
      return;
    }
    const timer = setTimeout(() => {
      setIsBusy(true);
    }, delay);
    return () => clearTimeout(timer);
  }, [active, delay]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.focus();
        } else {
          rootRef.current?.focus();
        }
      },
      blur() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.blur();
        } else {
          rootRef.current?.blur();
        }
      },
      isFocused() {
        return document.activeElement === overlayRef.current || document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    [isBusy]
  );
  const textLabel = text ? /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      id: labelId,
      className: cn(
        "w-full text-center text-sapphire-text-tertiary text-sm font-medium",
        isTopPlacement ? "mb-2" : "mt-2"
      ),
      children: text
    }
  ) : null;
  const dots = /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "inline-flex items-center leading-none",
        dotGapClasses[normalizedSize]
      ),
      children: [0, 1, 2].map((i) => /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: dotVariants({ size: normalizedSize }),
          style: {
            animation: `busy-grow 1.6s cubic-bezier(0.32, 0.06, 0.85, 1.11) ${i * 200}ms infinite`
          }
        },
        i
      ))
    }
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      id,
      className: cn(
        "inline-block",
        hasChildren && "relative",
        isBusy && "text-ring",
        className
      ),
      style: { ...style, borderRadius: "inherit", height: "inherit" },
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("style", { children: `@keyframes busy-grow{0%,50%,100%{transform:scale(.5)}25%{transform:scale(1)}}` }),
        isBusy && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: overlayRef,
            role: "progressbar",
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            "aria-valuetext": "Busy",
            "aria-labelledby": text ? labelId : void 0,
            "aria-label": !text ? accessibleName ?? "Please wait" : void 0,
            title: !text ? accessibleName ?? "Please wait" : void 0,
            tabIndex: 0,
            "data-sap-focus-ref": true,
            className: cn(
              "flex flex-col items-center justify-center",
              "bg-inherit rounded-[inherit]",
              "focus-visible:outline focus-visible:outline-2 focus-visible:outline-ring focus-visible:outline-offset-[-2px]",
              hasChildren && "absolute inset-0 z-[99]"
            ),
            children: [
              isTopPlacement && textLabel,
              dots,
              !isTopPlacement && textLabel
            ]
          }
        ),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(isBusy && "opacity-40"),
            ...isBusy ? { inert: true } : {},
            children
          }
        )
      ]
    }
  );
}
var filledBrandStyles = [
  "bg-sapphire-brand-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-brand-hover-background",
  "active:bg-sapphire-brand-pressed-background [&.kb-active]:bg-sapphire-brand-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus"
];
var filledJouleStyles = [
  "bg-sapphire-joule-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-joule-hover-background",
  "active:bg-sapphire-joule-pressed-background [&.kb-active]:bg-sapphire-joule-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus"
];
var outlineBrandStyles = [
  "bg-transparent text-sapphire-brand-foreground border border-sapphire-brand-foreground",
  "hover:bg-sapphire-neutral-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-brand-pressed-foreground [&.kb-active]:text-sapphire-brand-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
];
var outlineJouleStyles = [
  "bg-transparent text-sapphire-joule-foreground border border-sapphire-joule-foreground",
  "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background hover:border-sapphire-joule-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-joule-pressed-foreground [&.kb-active]:text-sapphire-joule-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
];
var buttonVariants = cva(
  [
    // Base styles
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        // --- Primary (filled) ---
        ["Primary" /* Primary */]: filledBrandStyles,
        ["PrimaryJoule" /* PrimaryJoule */]: filledJouleStyles,
        // --- Secondary (outlined) ---
        ["Secondary" /* Secondary */]: outlineBrandStyles,
        ["SecondaryJoule" /* SecondaryJoule */]: outlineJouleStyles,
        // --- Tertiary (ghost, colored text) ---
        ["Tertiary" /* Tertiary */]: [
          "bg-transparent text-sapphire-brand-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-brand-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
        ],
        ["TertiaryJoule" /* TertiaryJoule */]: [
          "bg-transparent text-sapphire-joule-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-joule-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
        ],
        // --- Neutral (ghost, dark text) ---
        ["Neutral" /* Neutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-sapphire-neutral-foreground-muted",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-neutral-foreground-black [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- SecondaryNeutral (ghost, neutral-black text) ---
        ["SecondaryNeutral" /* SecondaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background [&.kb-active]:bg-sapphire-neutral-pressed-background",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- TertiaryNeutral (ghost, muted neutral text, no background) ---
        ["TertiaryNeutral" /* TertiaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-muted border border-transparent",
          "hover:text-sapphire-neutral-foreground-black",
          "active:text-sapphire-neutral-foreground-black [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ]
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var largeSizeClasses = {
  default: "h-10 px-4 py-3 text-base gap-2 rounded-lg",
  iconOnly: "h-10 w-10 p-0 text-base rounded-lg",
  icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
};
var mediumSizeClasses = {
  default: "h-8 px-3 py-2 text-sm gap-1.5 rounded",
  iconOnly: "h-8 w-8 p-0 text-sm rounded",
  icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
};
var smallSizeClasses = {
  default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
  iconOnly: "h-6 w-6 p-0 text-xs rounded",
  icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
};
var sizeClasses = {
  ["Large" /* Large */]: largeSizeClasses,
  ["Medium" /* Medium */]: mediumSizeClasses,
  ["Small" /* Small */]: smallSizeClasses
};
function getButtonType(type) {
  switch (type) {
    case "Submit" /* Submit */:
    case "Submit":
      return "submit";
    case "Reset" /* Reset */:
    case "Reset":
      return "reset";
    default:
      return "button";
  }
}
function Button({
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  type = "Button" /* Button */,
  disabled = false,
  loading = false,
  icon,
  endIcon,
  iconOnly = false,
  children,
  tooltip,
  states,
  state,
  stateTransition = "slide-up",
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Button" /* Button */,
  accessibilityAttributes,
  name,
  value,
  form,
  onClick,
  onFocus,
  onBlur,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  "data-growing-button": dataGrowingButton,
  ref
}) {
  const buttonRef = React3.useRef(null);
  const hiddenButtonRef = React3.useRef(null);
  const [animationPhase, setAnimationPhase] = React3.useState("idle");
  const [displayedState, setDisplayedState] = React3.useState(
    null
  );
  const [buttonWidth, setButtonWidth] = React3.useState(void 0);
  const currentStateObject = states?.find((s) => s.name === state) ?? states?.[0];
  const isMultiState = states && states.length > 0;
  React3.useEffect(() => {
    if (isMultiState && !displayedState) {
      setDisplayedState(currentStateObject ?? null);
    }
  }, [isMultiState, currentStateObject, displayedState]);
  React3.useEffect(() => {
    if (!isMultiState || !currentStateObject || animationPhase !== "idle") {
      return;
    }
    if (displayedState && displayedState.name !== currentStateObject.name) {
      if (buttonRef.current) {
        setButtonWidth(buttonRef.current.offsetWidth);
      }
      setAnimationPhase("fade-out");
    }
  }, [currentStateObject, displayedState, isMultiState, animationPhase]);
  React3.useEffect(() => {
    if (animationPhase === "fade-out") {
      const timer = setTimeout(() => {
        setAnimationPhase("fade-mid");
      }, 180);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-mid") {
      setDisplayedState(currentStateObject ?? null);
      if (hiddenButtonRef.current) {
        setButtonWidth(hiddenButtonRef.current.offsetWidth);
      }
      const timer = setTimeout(() => {
        setAnimationPhase("fade-in");
      }, 20);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-in") {
      const timer = setTimeout(() => {
        setAnimationPhase("idle");
        setButtonWidth(void 0);
      }, 160);
      return () => clearTimeout(timer);
    }
  }, [animationPhase, currentStateObject]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        buttonRef.current?.focus();
      },
      blur() {
        buttonRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === buttonRef.current;
      },
      get nativeElement() {
        return buttonRef.current;
      }
    }),
    []
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled || loading) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [disabled, loading, onClick]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " ") {
          e.preventDefault();
        }
        buttonRef.current?.classList.add("kb-active");
        if (e.key === " ") {
          onClick?.({
            originalEvent: e,
            isKeyboard: true
          });
        }
      }
    },
    [disabled, loading, onClick]
  );
  const handleKeyUp = React3.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        buttonRef.current?.classList.remove("kb-active");
      }
    },
    []
  );
  const ariaExpanded = accessibilityAttributes?.expanded;
  const ariaHasPopup = accessibilityAttributes?.hasPopup;
  const ariaControls = accessibilityAttributes?.controls;
  const ariaPressed = accessibilityAttributes?.pressed;
  const role = accessibleRole === "Link" /* Link */ || accessibleRole === "Link" ? "link" : void 0;
  const isDisabled = disabled;
  const effectiveIcon = isMultiState ? displayedState?.icon : icon;
  const effectiveEndIcon = isMultiState ? displayedState?.endIcon : endIcon;
  const effectiveChildren = isMultiState ? displayedState?.text : children;
  const effectiveIconOnly = isMultiState ? !displayedState?.text && !!displayedState?.icon : iconOnly;
  const effectivePulsing = isMultiState ? displayedState?.pulsing : false;
  const PulsingText = ({ children: text }) => /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "opacity-50", children: text }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "absolute inset-0 overflow-hidden",
        style: {
          background: "linear-gradient(90deg, transparent 0%, currentColor 50%, transparent 100%)",
          backgroundSize: "200% 100%",
          WebkitBackgroundClip: "text",
          backgroundClip: "text",
          WebkitTextFillColor: "transparent",
          animation: "pulse-sweep 1.5s ease-in-out infinite"
        },
        children: text
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: `
          @keyframes pulse-sweep {
            0% { background-position: 200% 0; }
            100% { background-position: -200% 0; }
          }
        ` })
  ] });
  const getTransitionStyles = (phase, transition) => {
    if (transition === "none" || phase === "idle") {
      return {};
    }
    const baseTransition = "all 100ms ease-out";
    switch (transition) {
      case "slide-up":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(-8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-down":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(-8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-left":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(-12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "slide-right":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(-12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "fade":
        if (phase === "fade-out") return { opacity: 0, transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0 };
        if (phase === "fade-in") return { opacity: 1, transition: baseTransition };
        break;
      case "scale":
        if (phase === "fade-out") return { opacity: 0, transform: "scale(0.75)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "scale(0.75)" };
        if (phase === "fade-in") return { opacity: 1, transform: "scale(1)", transition: baseTransition };
        break;
      case "flip":
        if (phase === "fade-out") return { opacity: 0, transform: "rotateX(90deg)", transition: baseTransition, transformStyle: "preserve-3d" };
        if (phase === "fade-mid") return { opacity: 0, transform: "rotateX(-90deg)", transformStyle: "preserve-3d" };
        if (phase === "fade-in") return { opacity: 1, transform: "rotateX(0)", transition: baseTransition, transformStyle: "preserve-3d" };
        break;
    }
    return {};
  };
  const contentAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  const iconAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: buttonRef,
        type: getButtonType(type),
        disabled: isDisabled,
        "aria-disabled": loading || void 0,
        id,
        name,
        value,
        form,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-expanded": ariaExpanded,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-pressed": ariaPressed,
        "aria-busy": loading || void 0,
        title: tooltip,
        tabIndex,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur: (e) => {
          buttonRef.current?.classList.remove("kb-active");
          onBlur?.(e);
        },
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][effectiveIconOnly ? "iconOnly" : "default"],
          isMultiState && !effectiveIconOnly && "transition-[width] duration-300 ease-out overflow-hidden",
          loading && "relative pointer-events-none",
          className
        ),
        style: {
          ...style,
          ...buttonWidth !== void 0 && !effectiveIconOnly ? { width: buttonWidth } : {}
        },
        "data-testid": dataTestId,
        "data-growing-button": dataGrowingButton,
        children: [
          loading && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "absolute inset-0 flex items-center justify-center z-[1] [&_*]:!text-inherit", inert: true, children: /* @__PURE__ */ jsxRuntime.jsx(BusyIndicator, { size: "S", active: true, delay: 0 }) }),
          effectiveIcon ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? iconAnimationStyles : void 0, children: effectiveIcon }) : null,
          !effectiveIconOnly && effectiveChildren && /* @__PURE__ */ jsxRuntime.jsx("span", { className: loading ? "opacity-40" : void 0, style: isMultiState ? contentAnimationStyles : void 0, children: effectivePulsing ? /* @__PURE__ */ jsxRuntime.jsx(PulsingText, { children: effectiveChildren }) : effectiveChildren }),
          effectiveEndIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? contentAnimationStyles : void 0, children: effectiveEndIcon })
        ]
      }
    ),
    isMultiState && /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: hiddenButtonRef,
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][!currentStateObject?.text && currentStateObject?.icon ? "iconOnly" : "default"],
          "absolute -top-[10000px] -left-[10000px] invisible w-fit"
        ),
        tabIndex: -1,
        "aria-hidden": "true",
        children: [
          currentStateObject?.icon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.icon }),
          currentStateObject?.text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.text }),
          currentStateObject?.endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.endIcon })
        ]
      }
    )
  ] });
}
var listContainerVariants = cva(
  "relative w-full",
  {
    variants: {
      bordered: {
        true: "border border-sapphire-border-primary rounded-lg overflow-hidden",
        false: ""
      }
    },
    defaultVariants: {
      bordered: false
    }
  }
);
var listVariants = cva(
  "w-full m-0 p-0 list-none",
  {
    variants: {
      indent: {
        true: "pl-4",
        false: ""
      },
      separators: {
        true: "[&>li:not(:last-child)]:border-b [&>li:not(:last-child)]:border-sapphire-border-primary",
        false: ""
      }
    },
    defaultVariants: {
      indent: false,
      separators: true
    }
  }
);
var ListContext = React3__namespace.createContext(null);
function useListContext() {
  const context = React3.useContext(ListContext);
  if (!context) {
    throw new Error("List item components must be used within a List");
  }
  return context;
}
function useOptionalListContext() {
  return React3.useContext(ListContext);
}
function useListSelection({
  mode,
  controlledKeys,
  defaultKeys = [],
  items,
  getItemKey,
  onSelectionChange
}) {
  const [internalKeys, setInternalKeys] = React3.useState(new Set(defaultKeys));
  const previousKeysRef = React3.useRef(new Set(defaultKeys));
  const isControlled = controlledKeys !== void 0;
  const selectedKeys = isControlled ? new Set(controlledKeys) : internalKeys;
  const getItemByKey = React3.useCallback((key) => {
    if (!items || !getItemKey) return void 0;
    return items.find((item, index) => getItemKey(item, index) === key);
  }, [items, getItemKey]);
  const getSelectedItems = React3.useCallback(() => {
    if (!items || !getItemKey) return [];
    return items.filter((item, index) => selectedKeys.has(getItemKey(item, index)));
  }, [items, getItemKey, selectedKeys]);
  const setSelected = React3.useCallback((key, selected) => {
    const modeStr = String(mode);
    if (modeStr === "None") return;
    const newKeys = new Set(selectedKeys);
    const previousKeys = new Set(selectedKeys);
    if (modeStr === "Single" || modeStr === "SingleStart" || modeStr === "SingleEnd") {
      if (selected) {
        newKeys.clear();
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    } else if (modeStr === "Multiple") {
      if (selected) {
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    }
    if (!isControlled) {
      setInternalKeys(newKeys);
    }
    const targetItem = getItemByKey(key);
    onSelectionChange?.({
      selectedItems: items?.filter((item, index) => getItemKey && newKeys.has(getItemKey(item, index))) ?? [],
      previouslySelectedItems: items?.filter((item, index) => getItemKey && previousKeys.has(getItemKey(item, index))) ?? [],
      selectionComponentPressed: false,
      targetItem,
      key,
      selectedKeys: Array.from(newKeys)
      // Always include the raw keys
    });
    previousKeysRef.current = newKeys;
  }, [mode, selectedKeys, isControlled, getItemByKey, items, getItemKey, onSelectionChange]);
  const toggleSelection = React3.useCallback((key) => {
    setSelected(key, !selectedKeys.has(key));
  }, [selectedKeys, setSelected]);
  const isSelected = React3.useCallback((key) => {
    return selectedKeys.has(key);
  }, [selectedKeys]);
  const selectAll = React3.useCallback(() => {
    if (!items || !getItemKey || String(mode) !== "Multiple") return;
    const allKeys = new Set(items.map(getItemKey));
    if (!isControlled) {
      setInternalKeys(allKeys);
    }
    onSelectionChange?.({
      selectedItems: items,
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: items[0]
    });
  }, [items, getItemKey, mode, isControlled, onSelectionChange, getSelectedItems]);
  const deselectAll = React3.useCallback(() => {
    if (!isControlled) {
      setInternalKeys(/* @__PURE__ */ new Set());
    }
    onSelectionChange?.({
      selectedItems: [],
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: void 0
    });
  }, [isControlled, onSelectionChange, getSelectedItems]);
  return {
    selectedKeys,
    setSelected,
    toggleSelection,
    isSelected,
    selectAll,
    deselectAll,
    getSelectedItems
  };
}
function useListNavigation({
  onFocusItem,
  onSelectItem,
  selectionMode,
  containerRef
}) {
  const [focusedIndex, setFocusedIndex] = React3.useState(-1);
  const getAllNavigable = React3.useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"], button[data-growing-button]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const getAllNavigableExcludingGrowingButton = React3.useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const focusItem = React3.useCallback((index) => {
    const allNavigable = getAllNavigable();
    if (index < 0 || index >= allNavigable.length) return;
    const element = allNavigable[index];
    if (!element) return;
    const isGrowingButton = element.hasAttribute("data-growing-button");
    if (!isGrowingButton) {
      setFocusedIndex(index);
    }
    element.focus();
    onFocusItem?.(index);
  }, [getAllNavigable, onFocusItem]);
  const handleKeyDown = React3.useCallback((e) => {
    const key = e.key;
    if (key === "Tab") {
      return;
    }
    const target = e.target;
    const isInteractiveElement2 = target.tagName === "INPUT" || target.tagName === "TEXTAREA" || target.tagName === "SELECT" || target.tagName === "BUTTON" || target.getAttribute("role") === "button" || target.getAttribute("contenteditable") === "true" || target.closest('button, input, textarea, select, [role="button"], [contenteditable="true"]') !== null;
    if (isInteractiveElement2) {
      if (key === " " || key === "Enter" || key === "Delete") {
        return;
      }
    }
    const allNavigable = getAllNavigable();
    if (allNavigable.length === 0) return;
    const activeElement = document.activeElement;
    let currentIndex = allNavigable.indexOf(activeElement);
    if (currentIndex === -1) {
      currentIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    }
    let newIndex = currentIndex;
    switch (key) {
      case "ArrowDown":
        e.preventDefault();
        newIndex = Math.min(currentIndex + 1, allNavigable.length - 1);
        break;
      case "ArrowUp":
        e.preventDefault();
        newIndex = Math.max(currentIndex - 1, 0);
        break;
      case "Home":
        e.preventDefault();
        const navigableExcludingButton = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButton.length > 0) {
          const firstElement = navigableExcludingButton[0];
          const indexInAll = allNavigable.indexOf(firstElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "End":
        e.preventDefault();
        const navigableExcludingButtonEnd = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButtonEnd.length > 0) {
          const lastElement = navigableExcludingButtonEnd[navigableExcludingButtonEnd.length - 1];
          const indexInAll = allNavigable.indexOf(lastElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "PageDown":
        e.preventDefault();
        const navigableExcludingButtonPageDown = getAllNavigableExcludingGrowingButton();
        const currentInFiltered = navigableExcludingButtonPageDown.indexOf(activeElement);
        if (currentInFiltered !== -1) {
          const newFilteredIndex = Math.min(currentInFiltered + 10, navigableExcludingButtonPageDown.length - 1);
          const targetElement = navigableExcludingButtonPageDown[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageDown.length > 0) {
            const lastElement = navigableExcludingButtonPageDown[navigableExcludingButtonPageDown.length - 1];
            newIndex = allNavigable.indexOf(lastElement);
          }
        }
        break;
      case "PageUp":
        e.preventDefault();
        const navigableExcludingButtonPageUp = getAllNavigableExcludingGrowingButton();
        const currentInFilteredPageUp = navigableExcludingButtonPageUp.indexOf(activeElement);
        if (currentInFilteredPageUp !== -1) {
          const newFilteredIndex = Math.max(currentInFilteredPageUp - 10, 0);
          const targetElement = navigableExcludingButtonPageUp[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageUp.length > 0) {
            const lastFilteredIndex = navigableExcludingButtonPageUp.length - 1;
            const targetIndex = Math.max(lastFilteredIndex - 10, 0);
            const targetElement = navigableExcludingButtonPageUp[targetIndex];
            newIndex = allNavigable.indexOf(targetElement);
          }
        }
        break;
      case " ":
        e.preventDefault();
        const spaceElement = allNavigable[currentIndex];
        if (spaceElement?.getAttribute("role") === "listitem") {
          if (String(selectionMode) !== "None" && String(selectionMode) !== "Delete") {
            onSelectItem?.(currentIndex);
          }
        }
        return;
      case "Delete":
        const deleteElement = allNavigable[currentIndex];
        if (deleteElement?.getAttribute("role") === "listitem" && String(selectionMode) === "Delete") {
          onSelectItem?.(currentIndex);
        }
        return;
      case "Enter":
        return;
      default:
        return;
    }
    if (newIndex !== currentIndex) {
      focusItem(newIndex);
    }
  }, [focusedIndex, getAllNavigable, getAllNavigableExcludingGrowingButton, selectionMode, onSelectItem, focusItem]);
  const registerItem = React3.useCallback((_index, _element) => {
  }, []);
  return {
    focusedIndex,
    setFocusedIndex,
    focusItem,
    handleKeyDown,
    registerItem
  };
}
function ListHeader({ text, children, sticky, className }) {
  if (!text && !children) return null;
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "flex items-center gap-2.5 px-[0.75rem] py-[0.25rem] text-base font-normal text-sapphire-text-primary bg-background",
        sticky && "sticky top-0 z-10 bg-background",
        className
      ),
      role: "heading",
      "aria-level": 2,
      children: children || text
    }
  );
}
function ListLoading({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20",
        className
      ),
      role: "progressbar",
      "aria-label": "Loading",
      children: /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-6 w-6 animate-spin text-sapphire-brand-foreground" })
    }
  );
}
function ListEmpty({ text, children, className }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "flex items-center justify-center px-4 py-12 text-sm text-sapphire-text-tertiary",
        className
      ),
      role: "status",
      children: children || text || "No data"
    }
  );
}
function ListGrowing({
  mode,
  buttonText = "More",
  onLoadMore,
  parentRef,
  loading,
  accessibilityAttributes,
  buttonRef
}) {
  const observerRef = React3.useRef(null);
  const triggerRef = React3.useRef(null);
  React3.useEffect(() => {
    if (String(mode) !== "Scroll" || !triggerRef.current) return;
    observerRef.current = new IntersectionObserver(
      (entries) => {
        const [entry] = entries;
        if (entry.isIntersecting && !loading) {
          onLoadMore?.();
        }
      },
      {
        root: parentRef?.current,
        rootMargin: "100px",
        threshold: 0
      }
    );
    observerRef.current.observe(triggerRef.current);
    return () => {
      observerRef.current?.disconnect();
    };
  }, [mode, onLoadMore, parentRef, loading]);
  if (String(mode) === "Button") {
    return /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center justify-center py-3 border-t border-sapphire-border-primary", children: /* @__PURE__ */ jsxRuntime.jsxs(
      Button,
      {
        ref: buttonRef,
        design: "Tertiary" /* Tertiary */,
        onClick: () => onLoadMore?.(),
        disabled: loading,
        accessibleName: accessibilityAttributes?.name,
        "data-growing-button": "true",
        tabIndex: 0,
        className: "w-full mx-4",
        children: [
          loading ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-4 w-4 animate-spin mr-2" }) : null,
          buttonText
        ]
      }
    ) });
  }
  if (String(mode) === "Scroll") {
    return /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        ref: triggerRef,
        className: "h-1",
        "aria-hidden": "true",
        "data-growing-trigger": true
      }
    );
  }
  return null;
}
function ListInner(props) {
  const {
    ref,
    // Header & Footer
    headerText,
    header,
    // Selection
    selectionMode = "None" /* None */,
    selectedKeys: controlledSelectedKeys,
    defaultSelectedKeys = [],
    onSelectionChange,
    // Data
    items,
    renderItem,
    getItemKey = (item, index) => String(item.id ?? item.key ?? index),
    children,
    // Empty & Loading
    noDataText = "No data available",
    noDataSlot,
    loading = false,
    loadingDelay = 1e3,
    // Growing
    growing = "None" /* None */,
    growingButtonText,
    hasMore = false,
    onLoadMore,
    // Visual
    separators = "All" /* All */,
    indent = false,
    stickyHeader = false,
    stickyFooter = false,
    // Events
    onItemClick,
    onItemDelete,
    onItemFocused,
    // Drag and Drop
    onMoveOver,
    onMove,
    // Accessibility
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    accessibleRole = "List" /* List */,
    accessibilityAttributes,
    // Standard
    className,
    style,
    id: providedId,
    "data-testid": dataTestId,
    "data-ai-context": dataAiContext
  } = props;
  const generatedId = React3.useId();
  const listId = providedId ?? generatedId;
  const containerRef = React3.useRef(null);
  const listRef = React3.useRef(null);
  const growingButtonRef = React3.useRef(null);
  const beforeSentinelRef = React3.useRef(null);
  const afterSentinelRef = React3.useRef(null);
  const [showLoading, setShowLoading] = React3.useState(false);
  const [mediaRange, setMediaRange] = React3.useState("M");
  const itemRegistry = React3.useRef(/* @__PURE__ */ new Map());
  const [previouslyFocusedItemKey, setPreviouslyFocusedItemKey] = React3.useState(null);
  const [draggedKey, setDraggedKey] = React3.useState(null);
  const [dropTargetKey, setDropTargetKey] = React3.useState(null);
  const [dropPlacement, setDropPlacement] = React3.useState(null);
  const selectionState = useListSelection({
    mode: selectionMode,
    controlledKeys: controlledSelectedKeys,
    defaultKeys: defaultSelectedKeys,
    items,
    getItemKey,
    onSelectionChange
  });
  const itemCount = React3.useMemo(() => {
    if (items) return items.length;
    return React3__namespace.Children.count(children);
  }, [items, children]);
  const navigationState = useListNavigation({
    onFocusItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (element?.getAttribute("role") === "listitem" && items) {
        const itemKey = element.getAttribute("data-item-key");
        if (itemKey) {
          const itemIndex = items.findIndex((item, idx) => getItemKey(item, idx) === itemKey);
          if (itemIndex >= 0) {
            onItemFocused?.({ item: items[itemIndex], index: itemIndex });
          }
        }
      }
    },
    onSelectItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (!element || element.getAttribute("role") !== "listitem") return;
      if (!items || !getItemKey) return;
      const itemKey = element.getAttribute("data-item-key");
      if (!itemKey) return;
      const itemIndex = items.findIndex((item2, idx) => getItemKey(item2, idx) === itemKey);
      if (itemIndex < 0) return;
      const item = items[itemIndex];
      const key = getItemKey(item, itemIndex);
      if (String(selectionMode) === "Delete") {
        onItemDelete?.({ item });
      } else {
        selectionState.toggleSelection(key);
      }
    },
    selectionMode,
    containerRef
  });
  React3.useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  React3.useEffect(() => {
    const updateMediaRange = () => {
      const width = window.innerWidth;
      if (width < 600) setMediaRange("S");
      else if (width < 1024) setMediaRange("M");
      else if (width < 1440) setMediaRange("L");
      else setMediaRange("XL");
    };
    updateMediaRange();
    window.addEventListener("resize", updateMediaRange);
    return () => window.removeEventListener("resize", updateMediaRange);
  }, []);
  const registerItem = React3.useCallback((key, index) => {
    itemRegistry.current.set(key, index);
  }, []);
  const unregisterItem = React3.useCallback((key) => {
    itemRegistry.current.delete(key);
  }, []);
  const handleDragStart = React3.useCallback((key, _e) => {
    setDraggedKey(key);
  }, []);
  const handleDragEnd = React3.useCallback((_key, _e) => {
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, []);
  const handleDragOver = React3.useCallback((key, e) => {
    if (draggedKey && key === draggedKey) return;
    setDropTargetKey(key);
    const rect = e.currentTarget.getBoundingClientRect();
    const y = e.clientY - rect.top;
    const height = rect.height;
    if (y < height * 0.33) {
      setDropPlacement("Before");
    } else if (y > height * 0.67) {
      setDropPlacement("After");
    } else {
      setDropPlacement("On");
    }
  }, [draggedKey]);
  const handleDrop = React3.useCallback((targetKey, e) => {
    const targetIndex = itemRegistry.current.get(targetKey);
    if (targetIndex === void 0) return;
    if (!items || !getItemKey) return;
    const targetItem = items[targetIndex];
    if (!targetItem) return;
    const droppedKey = draggedKey || e.dataTransfer.getData("text/plain");
    if (!droppedKey || targetKey === droppedKey) return;
    const placement = dropPlacement || "After";
    const sourceIndex = itemRegistry.current.get(droppedKey);
    const sourceItem = sourceIndex !== void 0 ? items[sourceIndex] : void 0;
    const isMultiple = String(selectionMode) === "Multiple";
    const draggedIsSelected = droppedKey ? selectionState.isSelected(droppedKey) : false;
    const selectedItemsList = isMultiple && draggedIsSelected ? selectionState.getSelectedItems() : void 0;
    const moveDetail = {
      source: {
        element: sourceItem,
        // May be undefined for cross-list drops
        elements: selectedItemsList,
        index: sourceIndex ?? -1
        // -1 indicates cross-list drop
      },
      destination: {
        element: targetItem,
        placement,
        index: targetIndex
      },
      originalEvent: e.nativeEvent
    };
    onMoveOver?.(moveDetail);
    onMove?.(moveDetail);
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, [draggedKey, dropPlacement, items, getItemKey, selectionMode, selectionState, onMoveOver, onMove]);
  const handleForwardTabAfter = React3.useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      const itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      );
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (String(growing) !== "None" && hasMore && growingButtonRef.current) {
      growingButtonRef.current.focus?.();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex + 1; i < allTabbable.length; i++) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (afterSentinelRef.current) {
          afterSentinelRef.current.focus();
        }
      }
    }
  }, [growing, hasMore, navigationState]);
  const handleForwardTabBefore = React3.useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    let itemElement = null;
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      ) || null;
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (itemElement && event.target !== itemElement) {
      itemElement.focus();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex - 1; i >= 0; i--) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (beforeSentinelRef.current) {
          beforeSentinelRef.current.focus();
        }
      }
    }
  }, [navigationState]);
  const handleBeforeSentinelFocus = React3.useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromAfterSentinel = relatedTarget === afterSentinelRef.current;
    const isFromListItem = relatedTarget?.closest('li[role="listitem"], li[role="option"]');
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    if (isFromAfterSentinel || isFromListItem || isFromGrowingButton) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const sentinelIndex = allTabbable.indexOf(beforeSentinelRef.current);
      if (sentinelIndex > 0) {
        allTabbable[sentinelIndex - 1].focus();
      }
    } else {
      e.preventDefault();
      if (containerRef.current) {
        const allNavigable = Array.from(
          containerRef.current.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
          )
        );
        if (allNavigable.length === 0) {
          if (afterSentinelRef.current) {
            afterSentinelRef.current.focus();
          }
          return;
        }
        if (previouslyFocusedItemKey) {
          const targetElement = allNavigable.find(
            (el) => el.getAttribute("data-item-key") === previouslyFocusedItemKey || el.id === previouslyFocusedItemKey
          );
          if (targetElement) {
            targetElement.focus();
            return;
          }
        }
        if (allNavigable[0]) {
          allNavigable[0].focus();
        }
      }
    }
  }, [previouslyFocusedItemKey]);
  const getAllTabbableElements = React3.useCallback(() => {
    return Array.from(
      document.querySelectorAll(
        'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
      )
    );
  }, []);
  const focusNextAfterList = React3.useCallback(() => {
    if (!listRef.current || !afterSentinelRef.current) return false;
    const allTabbable = getAllTabbableElements();
    const sentinelIndex = allTabbable.indexOf(afterSentinelRef.current);
    if (sentinelIndex >= 0) {
      for (let i = sentinelIndex + 1; i < allTabbable.length; i++) {
        if (!containerRef.current?.contains(allTabbable[i])) {
          allTabbable[i].focus();
          return true;
        }
      }
    }
    return false;
  }, [getAllTabbableElements]);
  const enterListAtFocusedItem = React3.useCallback(() => {
    if (!containerRef.current) return false;
    const allNavigable = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
      )
    );
    if (allNavigable.length === 0) {
      beforeSentinelRef.current?.focus();
      return true;
    }
    const focusedIndex = navigationState.focusedIndex;
    const targetIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    if (allNavigable[targetIndex]) {
      allNavigable[targetIndex].focus();
      return true;
    }
    return false;
  }, [navigationState.focusedIndex]);
  const handleAfterSentinelFocus = React3.useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    const isFromOutside = !relatedTarget || !containerRef.current?.contains(relatedTarget);
    if (isFromGrowingButton || isFromOutside) {
      enterListAtFocusedItem();
    } else {
      focusNextAfterList();
    }
  }, [focusNextAfterList, enterListAtFocusedItem]);
  const roleSupportsAriaSelected = React3.useMemo(() => {
    const roleStr = String(accessibleRole);
    return roleStr === "ListBox" /* ListBox */ || roleStr === "listbox" || roleStr === "Tree" /* Tree */ || roleStr === "tree";
  }, [accessibleRole]);
  const contextValue = React3.useMemo(() => ({
    selectionMode,
    selectedKeys: selectionState.selectedKeys,
    onSelect: selectionState.setSelected,
    onToggleSelect: selectionState.toggleSelection,
    isSelected: selectionState.isSelected,
    focusedIndex: navigationState.focusedIndex,
    onFocusItem: navigationState.setFocusedIndex,
    registerItem,
    unregisterItem,
    registerNavigationItem: navigationState.registerItem,
    // Tab navigation
    onForwardTabAfter: handleForwardTabAfter,
    onForwardTabBefore: handleForwardTabBefore,
    separators,
    indent,
    onItemClick: (_key, event) => {
      onItemClick?.({ item: event.currentTarget, originalEvent: event });
    },
    onItemDelete: (key) => {
      const index = itemRegistry.current.get(key);
      if (index !== void 0 && items) {
        onItemDelete?.({ item: items[index] });
        return;
      }
      if (items && getItemKey) {
        const item = items.find((item2, idx) => getItemKey(item2, idx) === key);
        if (item) {
          onItemDelete?.({ item });
        }
      }
    },
    // Drag and drop
    onDragStart: handleDragStart,
    onDragEnd: handleDragEnd,
    onDragOver: handleDragOver,
    onDrop: handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    mediaRange,
    totalItemCount: itemCount,
    getItemIndex: (key) => {
      return itemRegistry.current.get(key) ?? -1;
    },
    // Accessibility
    accessibleRole,
    supportsAriaSelected: roleSupportsAriaSelected
  }), [
    selectionMode,
    selectionState,
    navigationState,
    registerItem,
    unregisterItem,
    handleForwardTabAfter,
    handleForwardTabBefore,
    separators,
    indent,
    onItemClick,
    onItemDelete,
    items,
    getItemKey,
    mediaRange,
    itemCount,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    accessibleRole,
    roleSupportsAriaSelected
  ]);
  React3.useImperativeHandle(ref, () => ({
    // Focus
    focus: () => listRef.current?.focus(),
    focusItem: (index) => navigationState.focusItem(index),
    getFocusedItemIndex: () => navigationState.focusedIndex,
    // Selection
    getSelectedKeys: () => Array.from(selectionState.selectedKeys),
    getSelectedItems: () => selectionState.getSelectedItems(),
    isSelected: selectionState.isSelected,
    select: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, true));
    },
    deselect: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, false));
    },
    toggleSelection: selectionState.toggleSelection,
    selectAll: selectionState.selectAll,
    deselectAll: selectionState.deselectAll,
    // Data
    getItems: () => items ?? [],
    getItemCount: () => itemCount,
    getItemByKey: (key) => items?.find((item, index) => getItemKey(item, index) === key),
    // Scrolling
    scrollToItem: (index, align = "start") => {
      const element = listRef.current?.children[index];
      element?.scrollIntoView({ behavior: "smooth", block: align });
    },
    scrollToTop: () => {
      containerRef.current?.scrollTo({ top: 0, behavior: "smooth" });
    },
    scrollToBottom: () => {
      containerRef.current?.scrollTo({
        top: containerRef.current.scrollHeight,
        behavior: "smooth"
      });
    },
    // Elements
    get listElement() {
      return listRef.current;
    },
    get scrollContainer() {
      return containerRef.current;
    }
  }), [selectionState, navigationState, items, itemCount, getItemKey]);
  const handleLoadMore = React3.useCallback(() => {
    onLoadMore?.({ currentCount: itemCount });
  }, [onLoadMore, itemCount]);
  const getRole = () => {
    const roleStr = String(accessibleRole);
    switch (roleStr) {
      case "Menu":
        return "menu";
      case "Tree":
        return "tree";
      case "ListBox":
        return "listbox";
      default:
        return "list";
    }
  };
  const isEmpty = itemCount === 0 && !loading;
  const renderedItems = React3.useMemo(() => {
    itemRegistry.current.clear();
    if (items && renderItem) {
      return items.map((item, index) => {
        const key = getItemKey(item, index);
        itemRegistry.current.set(key, index);
        return /* @__PURE__ */ jsxRuntime.jsx(React3__namespace.Fragment, { children: renderItem(item, index) }, key);
      });
    }
    if (children) {
      let index = 0;
      React3__namespace.Children.forEach(children, (child) => {
        if (React3__namespace.isValidElement(child) && child.props && typeof child.props.itemKey === "string") {
          itemRegistry.current.set(child.props.itemKey, index);
          index++;
        }
      });
    }
    return children;
  }, [items, renderItem, getItemKey, children]);
  return /* @__PURE__ */ jsxRuntime.jsx(ListContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: containerRef,
      className: cn(
        listContainerVariants({ bordered: false }),
        // Make scrollable when growing mode or sticky header is used
        (String(growing) !== "None" || stickyHeader || stickyFooter) && "overflow-auto",
        // Always prevent horizontal scroll
        "overflow-x-hidden",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-ai-context": dataAiContext,
      onKeyDown: navigationState.handleKeyDown,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          ListHeader,
          {
            text: headerText,
            sticky: stickyHeader,
            children: header
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            ref: beforeSentinelRef,
            tabIndex: 0,
            onFocus: handleBeforeSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "ul",
          {
            ref: listRef,
            id: listId,
            role: getRole(),
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-describedby": accessibleDescriptionRef,
            "aria-busy": loading || void 0,
            "aria-multiselectable": roleSupportsAriaSelected && String(selectionMode) === "Multiple" ? true : void 0,
            className: cn(listVariants({ indent, separators: String(separators) !== "None" })),
            children: !isEmpty && renderedItems
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            ref: afterSentinelRef,
            tabIndex: 0,
            onFocus: handleAfterSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        isEmpty && /* @__PURE__ */ jsxRuntime.jsx(ListEmpty, { text: noDataText, children: noDataSlot }),
        String(growing) !== "None" && hasMore && !loading && /* @__PURE__ */ jsxRuntime.jsx(
          ListGrowing,
          {
            mode: growing,
            buttonText: growingButtonText,
            onLoadMore: handleLoadMore,
            parentRef: containerRef,
            loading,
            accessibilityAttributes: accessibilityAttributes?.growingButton,
            buttonRef: growingButtonRef
          }
        ),
        showLoading && /* @__PURE__ */ jsxRuntime.jsx(ListLoading, {})
      ]
    }
  ) });
}
var List = React3__namespace.memo(ListInner);
List.displayName = "List";
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData8,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React3.useRef(null);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData8 || pathData8.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData8.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M64 192q26 0 45 19t19 45-19 45-45 19-45-19-19-45 19-45 45-19zm192 0q26 0 45 19t19 45-19 45-45 19-45-19-19-45 19-45 45-19zm192 0q26 0 45 19t19 45-19 45-45 19-45-19-19-45 19-45 45-19z"
];
var OverflowIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "OverflowIcon", iconName: "overflow" }
);

// src/components/list/utils/tabbable.ts
var TABBABLE_SELECTOR = [
  "a[href]",
  "area[href]",
  'input:not([disabled]):not([type="hidden"])',
  "select:not([disabled])",
  "textarea:not([disabled])",
  "button:not([disabled])",
  "iframe",
  "object",
  "embed",
  '[tabindex]:not([tabindex="-1"])',
  '[contenteditable]:not([contenteditable="false"])'
].join(",");
function getTabbableElements(container) {
  if (!container) return [];
  const elements = Array.from(
    container.querySelectorAll(TABBABLE_SELECTOR)
  );
  return elements.filter((el) => {
    const tabindex = el.getAttribute("tabindex");
    if (tabindex === "-1") return false;
    const style = window.getComputedStyle(el);
    if (style.display === "none" || style.visibility === "hidden") return false;
    let current = el;
    while (current && current !== container) {
      if (current.getAttribute("aria-hidden") === "true") return false;
      current = current.parentElement;
    }
    return true;
  });
}
function getActiveElement() {
  let activeElement = document.activeElement;
  while (activeElement?.shadowRoot?.activeElement) {
    activeElement = activeElement.shadowRoot.activeElement;
  }
  return activeElement;
}

// src/components/list/hooks/useListItemSelection.ts
var isInteractiveElement = (target, container) => {
  let el = target;
  while (el && el !== container) {
    const tag = el.tagName.toLowerCase();
    if (tag === "input" || tag === "textarea" || tag === "select" || tag === "button" || tag === "a" && el.hasAttribute("href") || el.getAttribute("role") === "button" || el.isContentEditable) {
      return true;
    }
    el = el.parentElement;
  }
  return false;
};
function useListItemSelection(options) {
  const { id, disabled, type, selectionMode, onClick, itemRef } = options;
  const listContext = useOptionalListContext();
  const isSelected = listContext ? listContext.isSelected(id) : false;
  const shouldForwardTabAfter = React3.useCallback(() => {
    if (!itemRef?.current) return true;
    const tabbables = getTabbableElements(itemRef.current);
    if (tabbables.length === 0) return true;
    const activeElement = getActiveElement();
    const lastTabbable = tabbables[tabbables.length - 1];
    return activeElement === lastTabbable;
  }, [itemRef]);
  const shouldForwardTabBefore = React3.useCallback((target) => {
    if (!itemRef?.current) return false;
    if (target === itemRef.current) return true;
    const tabbables = getTabbableElements(itemRef.current);
    if (tabbables.length === 0) return false;
    return target === tabbables[0];
  }, [itemRef]);
  const handleClick = React3.useCallback((e) => {
    if (disabled) return;
    const container = itemRef?.current ?? null;
    const clickTarget = e.target;
    if (clickTarget !== container && isInteractiveElement(clickTarget, container)) {
      return;
    }
    const typeStr = String(type);
    if (typeStr === "Inactive" || typeStr === "Detail") {
      return;
    }
    const modeStr = String(selectionMode);
    if (modeStr !== "None" && modeStr !== "Delete") {
      listContext?.onToggleSelect(id);
    }
    onClick?.(e);
    listContext?.onItemClick?.(id, e);
  }, [disabled, type, selectionMode, listContext, id, onClick, itemRef]);
  const handleKeyDown = React3.useCallback((e) => {
    if (disabled) return;
    const modeStr = String(selectionMode);
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      const target = e.target;
      const tabbables = getTabbableElements(itemRef?.current || null);
      if (e.shiftKey) {
        if (shouldForwardTabBefore(target)) {
          e.preventDefault();
          listContext?.onForwardTabBefore?.(id, e);
        }
      } else {
        if (target === itemRef?.current && tabbables.length > 0) {
          e.preventDefault();
          tabbables[0].focus();
          return;
        }
        if (shouldForwardTabAfter()) {
          e.preventDefault();
          listContext?.onForwardTabAfter?.(id, e);
          return;
        }
      }
      return;
    }
    if (e.key === "Enter" || e.key === " ") {
      const keyTarget = e.target;
      const keyTag = keyTarget.tagName.toLowerCase();
      if (keyTag === "input" || keyTag === "textarea" || keyTag === "select" || keyTarget.isContentEditable || keyTarget.getAttribute("role") === "button" || keyTag === "a" && keyTarget.hasAttribute("href")) {
        return;
      }
      e.preventDefault();
      const typeStr = String(type);
      if (typeStr === "Inactive" || typeStr === "Detail") {
        return;
      }
      if (modeStr !== "None" && modeStr !== "Delete") {
        listContext?.onToggleSelect(id);
      }
      onClick?.(e);
      listContext?.onItemClick?.(id, e);
    }
  }, [disabled, type, selectionMode, listContext, id, onClick, shouldForwardTabAfter, shouldForwardTabBefore, itemRef]);
  const handleSelectionChange = React3.useCallback((checked) => {
    if (disabled) return;
    listContext?.onSelect(id, checked);
  }, [disabled, listContext, id]);
  return {
    isSelected,
    handleClick,
    handleKeyDown,
    handleSelectionChange
  };
}
function useListItemFocus(options) {
  const { itemRef, disabled, selectionMode, accessibleRole, controlledSelected } = options;
  const listContext = useOptionalListContext();
  const [isFocused, setIsFocused] = React3.useState(false);
  const [isFocusedWithin, setIsFocusedWithin] = React3.useState(false);
  const [isActive, setIsActive] = React3.useState(false);
  const getItemIndex = React3.useCallback(() => {
    let rootList = itemRef.current?.parentElement;
    while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
      rootList = rootList.parentElement;
    }
    if (rootList && itemRef.current) {
      const allNavigable = Array.from(
        rootList.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
        )
      );
      return allNavigable.indexOf(itemRef.current);
    }
    return -1;
  }, [itemRef]);
  const handleFocus = React3.useCallback((e) => {
    const isItemItself = e.target === itemRef.current;
    setIsFocused(isItemItself);
    setIsFocusedWithin(true);
    if (listContext) {
      const itemIndex2 = getItemIndex();
      if (itemIndex2 >= 0) {
        listContext.onFocusItem(itemIndex2);
      }
    }
  }, [listContext, getItemIndex, itemRef]);
  const handleBlur = React3.useCallback((e) => {
    const isLeavingItem = !itemRef.current?.contains(e.relatedTarget);
    if (isLeavingItem) {
      setIsFocused(false);
      setIsFocusedWithin(false);
    } else {
      setIsFocused(false);
    }
  }, [itemRef]);
  const handleMouseDown = React3.useCallback(() => {
    setIsActive(true);
  }, []);
  const handleMouseUp = React3.useCallback(() => {
    setIsActive(false);
  }, []);
  const handleMouseDownCapture = React3.useCallback(() => {
    setIsFocusedWithin(true);
  }, []);
  const focus = React3.useCallback(() => {
    itemRef.current?.focus();
  }, [itemRef]);
  const blur = React3.useCallback(() => {
    itemRef.current?.blur();
  }, [itemRef]);
  const modeStr = String(selectionMode);
  const isSelected = controlledSelected ?? (listContext?.isSelected(itemRef.current?.getAttribute("data-item-key") || "") || false);
  const roleStr = String(accessibleRole);
  const supportsAriaSelected = listContext?.supportsAriaSelected ?? false;
  const hasSelection = modeStr !== "None" && modeStr !== "Delete";
  const ariaSelected = roleStr === "Option" ? isSelected : supportsAriaSelected && hasSelection ? isSelected : void 0;
  const itemIndex = getItemIndex();
  const isFocusedItem = listContext?.focusedIndex === itemIndex;
  const isFirstItem = itemIndex === 0;
  const isTabbable = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem;
  const tabbableIndex = isTabbable ? 0 : -1;
  const tabIndex = disabled ? -1 : tabbableIndex;
  return {
    isFocused,
    isFocusedWithin,
    isActive,
    tabIndex,
    ariaSelected,
    handleFocus,
    handleBlur,
    handleMouseDown,
    handleMouseUp,
    handleMouseDownCapture,
    focus,
    blur,
    getItemIndex
  };
}
function useListItemDragDrop(options) {
  const { id, movable, disabled, onDragStart: onDragStartProp } = options;
  const listContext = useOptionalListContext();
  const isDragging = listContext?.draggedKey === id;
  const isDropTarget = listContext?.dropTargetKey === id;
  const dropPlacement = isDropTarget ? listContext?.dropPlacement : null;
  const handleDragStart = React3.useCallback((e) => {
    if (!movable || disabled) {
      e.preventDefault();
      return;
    }
    e.dataTransfer.effectAllowed = "move";
    e.dataTransfer.dropEffect = "move";
    e.dataTransfer.setData("text/plain", id);
    listContext?.onDragStart?.(id, e);
    onDragStartProp?.();
  }, [movable, disabled, id, listContext, onDragStartProp]);
  const handleDragEnd = React3.useCallback((e) => {
    listContext?.onDragEnd?.(id, e);
  }, [id, listContext]);
  const handleDragOver = React3.useCallback((e) => {
    e.preventDefault();
    e.dataTransfer.dropEffect = "move";
    listContext?.onDragOver?.(id, e);
  }, [id, listContext]);
  const handleDrop = React3.useCallback((e) => {
    e.preventDefault();
    listContext?.onDrop?.(id, e);
  }, [id, listContext]);
  const handleDragEnter = React3.useCallback((e) => {
    e.preventDefault();
  }, []);
  const handleDragLeave = React3.useCallback((_e) => {
  }, []);
  return {
    isDragging,
    isDropTarget,
    dropPlacement: dropPlacement || null,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDragEnter,
    handleDragLeave,
    handleDrop
  };
}
function useListItemRegistration(options) {
  const { itemRef, itemKey } = options;
  const listContext = useOptionalListContext();
  const itemIndexRef = React3.useRef(-1);
  React3.useEffect(() => {
    if (listContext && itemKey) {
      let rootList = itemRef.current?.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(itemRef.current);
        if (index >= 0) {
          itemIndexRef.current = index;
          listContext.registerItem(itemKey, index);
          listContext.registerNavigationItem(index, itemRef.current);
        }
      }
    }
    return () => {
      if (listContext && itemKey && itemIndexRef.current >= 0) {
        listContext.unregisterItem(itemKey);
        listContext.registerNavigationItem(itemIndexRef.current, null);
        itemIndexRef.current = -1;
      }
    };
  }, [listContext, itemKey, itemRef]);
  return {};
}

// src/components/list/useListItem.ts
function useListItem(options = {}) {
  const {
    itemKey,
    disabled = false,
    movable = false,
    type = "Active",
    accessibleRole = "ListItem",
    selected: controlledSelected,
    onClick,
    onDragStart
  } = options;
  const itemRef = React3.useRef(null);
  const generatedId = React3.useId();
  const id = itemKey ?? generatedId;
  const listContext = useOptionalListContext();
  const selectionMode = listContext?.selectionMode ?? "None" /* None */;
  useListItemRegistration({ itemRef, itemKey });
  const selection = useListItemSelection({
    id,
    disabled,
    type,
    selectionMode,
    onClick,
    itemRef
  });
  const focus = useListItemFocus({
    itemRef,
    disabled,
    selectionMode,
    accessibleRole,
    controlledSelected
  });
  const handleMouseDown = React3.useCallback(() => {
    const typeStr = String(type);
    if (typeStr === "Active" || typeStr === "Navigation") {
      focus.handleMouseDown();
    }
  }, [type, focus]);
  const handleMouseUp = React3.useCallback(() => {
    focus.handleMouseUp();
  }, [focus]);
  const dragDrop = useListItemDragDrop({
    id,
    movable,
    disabled,
    onDragStart
  });
  return {
    // Refs
    itemRef,
    id,
    // State from sub-hooks (controlledSelected overrides context-based selection)
    isSelected: controlledSelected ?? selection.isSelected,
    isFocused: focus.isFocused,
    isFocusedWithin: focus.isFocusedWithin,
    isActive: focus.isActive,
    isDragging: dragDrop.isDragging,
    isDropTarget: dragDrop.isDropTarget,
    dropPlacement: dragDrop.dropPlacement,
    // Context values
    selectionMode,
    // Computed values
    tabIndex: focus.tabIndex,
    ariaSelected: focus.ariaSelected,
    // Event handlers (composed from sub-hooks)
    handleClick: selection.handleClick,
    handleKeyDown: selection.handleKeyDown,
    handleMouseDown,
    handleMouseUp,
    handleMouseDownCapture: focus.handleMouseDownCapture,
    handleFocus: focus.handleFocus,
    handleBlur: focus.handleBlur,
    handleDragStart: dragDrop.handleDragStart,
    handleDragEnd: dragDrop.handleDragEnd,
    handleDragOver: dragDrop.handleDragOver,
    handleDragEnter: dragDrop.handleDragEnter,
    handleDragLeave: dragDrop.handleDragLeave,
    handleDrop: dragDrop.handleDrop,
    // Selection handlers
    handleSelectionChange: selection.handleSelectionChange,
    // Helper functions
    focus: focus.focus,
    blur: focus.blur,
    getItemIndex: focus.getItemIndex
  };
}
var pathData2 = [
  "M438 224q11 0 18.5 7.5T464 250v140q0 38-26 64t-64 26H106q-38 0-64-26t-26-64V122q0-38 26-64t64-26h237q11 0 18 7.5t7 18.5-7 18-18 7H106q-16 0-27.5 11.5T67 122v268q0 16 11.5 27.5T106 429h268q16 0 27.5-11.5T413 390V250q0-11 7-18.5t18-7.5zm32-192q11 0 18.5 7.5T496 58q0 10-7 17L257 312q-6 8-18 8-10 0-18-8l-70-71q-7-7-7-18t7.5-18 18.5-7 18 7l51 53L452 40q8-8 18-8z"
];
var CompleteIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "CompleteIcon", iconName: "complete" }
);
var pathData3 = [
  "M390 480H122q-38 0-64-26t-26-64V122q0-38 26-64t64-26h268q38 0 64 26t26 64v268q0 38-26 64t-64 26zM122 83q-17 0-28 11t-11 28v268q0 17 11 28t28 11h268q17 0 28-11t11-28V122q0-17-11-28t-28-11H122z"
];
var BorderIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "BorderIcon", iconName: "border" }
);
var pathData4 = [
  "M390 32q38 0 64 26t26 64v268q0 38-26 64t-64 26H122q-38 0-64-26t-26-64V122q0-38 26-64t64-26h268zm39 90q0-17-11-28t-28-11H122q-17 0-28 11t-11 28v268q0 17 11 28t28 11h268q17 0 28-11t11-28V122zm-77 38v192H160V160h192z"
];
var TriStateIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "TriStateIcon", iconName: "tri-state" }
);
function CheckBox({
  checked: controlledChecked,
  defaultChecked = false,
  indeterminate = false,
  disabled = false,
  readonly = false,
  displayOnly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  wrappingType = "Normal" /* Normal */,
  valueState = "None",
  valueStateMessage,
  name,
  value = "on",
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  accessibilityAttributes,
  ref
}) {
  const inputRef = React3.useRef(null);
  const rootRef = React3.useRef(null);
  const generatedId = React3.useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = React3.useState(defaultChecked);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const isActive = isChecked || indeterminate;
  const doToggle = React3.useCallback(
    (originalEvent) => {
      if (disabled || readonly || displayOnly) return;
      const newChecked = !isChecked;
      if (!isControlled) {
        setInternalChecked(newChecked);
      }
      onChange?.({
        checked: newChecked,
        indeterminate: false,
        originalEvent
      });
    },
    [disabled, readonly, displayOnly, isChecked, isControlled, onChange]
  );
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      toggle() {
        doToggle();
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked, doToggle]
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) {
        e.preventDefault();
        return;
      }
      doToggle(e);
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const handleMouseDown = React3.useCallback(
    (e) => {
      if (disabled || displayOnly) {
        e.preventDefault();
      }
    },
    [disabled, displayOnly]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
      }
    },
    [disabled, readonly, displayOnly]
  );
  const handleKeyUp = React3.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doToggle(e);
      }
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const labelContent = children || text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly && !displayOnly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const getValueStateBorderClass2 = () => {
    if (isActive) return "";
    const borderMap = {
      Positive: "border-sapphire-positive",
      Negative: "border-sapphire-negative",
      Critical: "border-sapphire-warning",
      Information: "border-sapphire-info"
    };
    return borderMap[valueState] ?? "";
  };
  const getValueStateCheckedClass = () => {
    if (!isActive) return "";
    const checkedMap = {
      Positive: "bg-sapphire-positive border-sapphire-positive",
      Negative: "bg-sapphire-negative border-sapphire-negative",
      Critical: "bg-sapphire-warning border-sapphire-warning",
      Information: "bg-sapphire-info border-sapphire-info"
    };
    return checkedMap[valueState] ?? "bg-sapphire-canvas-primary border-sapphire-border-secondary";
  };
  const renderDisplayOnlyIcon = () => {
    const cls = "size-4 text-foreground";
    if (indeterminate) return /* @__PURE__ */ jsxRuntime.jsx(TriStateIcon, { className: cls });
    if (isChecked) return /* @__PURE__ */ jsxRuntime.jsx(CompleteIcon, { className: cls });
    return /* @__PURE__ */ jsxRuntime.jsx(BorderIcon, { className: cls });
  };
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        ref: inputRef,
        type: "checkbox",
        name,
        value,
        checked: isChecked,
        disabled: disabled || displayOnly,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: rootRef,
        id,
        role: "checkbox",
        "aria-checked": indeterminate ? "mixed" : isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          valueState !== "None" && valueStateMessage ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null,
          accessibilityAttributes?.describedBy
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || displayOnly || void 0,
        "aria-readonly": readonly || void 0,
        "aria-invalid": valueState === "Negative" || void 0,
        "aria-required": required || void 0,
        tabIndex: tabIndex !== void 0 ? tabIndex : disabled || displayOnly ? -1 : 0,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          // Touch area: Cozy = 44px, Compact = 32px
          isSmall ? labelContent ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2" : labelContent ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]",
          "text-sm",
          "rounded-lg",
          // Focus ring via pseudo-element for asymmetric insets
          "outline-none",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          labelContent ? isSmall ? "after:inset-x-0 after:inset-y-0" : "after:left-0 after:right-[2px] after:inset-y-[5px]" : "after:inset-[5px] after:rounded-lg",
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          displayOnly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          displayOnly ? renderDisplayOnlyIcon() : /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: cn(
                "relative inline-flex items-center justify-center",
                "shrink-0",
                hasWrappingLabel && "self-start",
                isSmall ? hasWrappingLabel ? "size-4 rounded p-[2px] mt-0.5" : "size-4 rounded p-[2px]" : hasWrappingLabel ? "size-[22px] rounded-md p-[3px] mt-0.5" : "size-[22px] rounded-md p-[3px]",
                "transition-colors",
                // Border style: dashed for readonly, solid otherwise
                readonly ? "border border-dashed" : "border border-solid",
                // --- Unchecked state ---
                !isActive && !readonly && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
                // --- Checked / indeterminate state (not readonly) ---
                // None value state: blue icon on white bg; value states: white icon on colored bg
                isActive && !readonly && valueState === "None" && "text-sapphire-shell-button-fg-selected",
                isActive && !readonly && valueState !== "None" && "text-sapphire-neutral-foreground-white",
                !readonly && getValueStateCheckedClass(),
                !readonly && getValueStateBorderClass2(),
                // --- Hover (unchecked, not readonly/disabled, no value state) ---
                !isActive && isInteractive && valueState === "None" && "group-hover:border-primary group-hover:bg-sapphire-shell-button-bg-selected",
                // --- Hover (checked, not readonly/disabled, no value state) ---
                isActive && isInteractive && valueState === "None" && "group-hover:bg-sapphire-shell-button-bg-selected group-hover:border-primary",
                // --- Read-only state: gray bg + dashed secondary border ---
                readonly && "bg-sapphire-background-secondary border-sapphire-border-secondary",
                // --- Read-only icon color: secondary-foreground (#353C4A) ---
                readonly && isActive && "text-secondary-foreground"
              ),
              children: indeterminate ? (
                // Solid square for tristate per Figma: 12px large, 8px small
                /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn(
                  "block bg-current",
                  isSmall ? "size-2" : "size-3"
                ) })
              ) : isChecked ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Check, { className: isSmall ? "size-3" : "size-4", strokeWidth: 3 }) : null
            }
          ),
          labelContent && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" && valueStateMessage && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage }),
          accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}
var radioGroups = /* @__PURE__ */ new Map();
function getOrderedGroupEntries(name) {
  const entries = radioGroups.get(name) ?? [];
  const connectedEntries = entries.filter((entry) => entry.getNode()?.isConnected);
  if (connectedEntries.length !== entries.length) {
    if (connectedEntries.length === 0) {
      radioGroups.delete(name);
    } else {
      radioGroups.set(name, connectedEntries);
    }
  }
  return [...connectedEntries].sort((entryA, entryB) => {
    const nodeA = entryA.getNode();
    const nodeB = entryB.getNode();
    if (!nodeA || !nodeB || nodeA === nodeB) {
      return 0;
    }
    const position = nodeA.compareDocumentPosition(nodeB);
    if (position & Node.DOCUMENT_POSITION_FOLLOWING) {
      return -1;
    }
    if (position & Node.DOCUMENT_POSITION_PRECEDING) {
      return 1;
    }
    return 0;
  });
}
function updateGroupTabIndex(name) {
  const entries = getOrderedGroupEntries(name);
  if (entries.length === 0) {
    return;
  }
  const focusableEntries = entries.filter((entry) => !entry.isDisabled());
  const activeEntry = focusableEntries.find((entry) => entry.isChecked()) ?? focusableEntries[0] ?? null;
  entries.forEach((entry) => {
    entry.setTabIndex(!entry.isDisabled() && entry === activeEntry ? 0 : -1);
  });
}
function registerGroupEntry(name, entry) {
  const existingEntries = radioGroups.get(name) ?? [];
  const nextEntries = existingEntries.filter((existingEntry) => existingEntry.id !== entry.id);
  nextEntries.push(entry);
  radioGroups.set(name, nextEntries);
  updateGroupTabIndex(name);
}
function unregisterGroupEntry(name, id) {
  const existingEntries = radioGroups.get(name);
  if (!existingEntries) {
    return;
  }
  const nextEntries = existingEntries.filter((entry) => entry.id !== id);
  if (nextEntries.length === 0) {
    radioGroups.delete(name);
    return;
  }
  radioGroups.set(name, nextEntries);
  updateGroupTabIndex(name);
}
function getAdjacentGroupEntry(name, currentId, direction) {
  const focusableEntries = getOrderedGroupEntries(name).filter((entry) => !entry.isDisabled());
  if (focusableEntries.length <= 1) {
    return null;
  }
  const currentIndex = focusableEntries.findIndex((entry) => entry.id === currentId);
  const baseIndex = currentIndex >= 0 ? currentIndex : 0;
  const nextIndex = (baseIndex + direction + focusableEntries.length) % focusableEntries.length;
  return focusableEntries[nextIndex] ?? null;
}
function moveSelectionInGroup(name, currentId, direction, originalEvent) {
  const nextEntry = getAdjacentGroupEntry(name, currentId, direction);
  if (!nextEntry) {
    return;
  }
  nextEntry.focus();
  if (!nextEntry.isReadonly()) {
    nextEntry.select(originalEvent);
    return;
  }
  updateGroupTabIndex(name);
}
function uncheckOthersInGroup(groupName, currentId) {
  getOrderedGroupEntries(groupName).forEach((entry) => {
    if (entry.id !== currentId && entry.isChecked()) {
      entry.setChecked(false);
    }
  });
}
function getKeyNavigationDirection(key) {
  if (key === " " || key === "Enter") return "select";
  const isRTL = document.documentElement.dir === "rtl";
  if (key === "ArrowDown" || key === "Down" || key === "ArrowRight" && !isRTL || key === "Right" && !isRTL || key === "ArrowLeft" && isRTL || key === "Left" && isRTL) {
    return 1;
  }
  if (key === "ArrowUp" || key === "Up" || key === "ArrowLeft" && !isRTL || key === "Left" && !isRTL || key === "ArrowRight" && isRTL || key === "Right" && isRTL) {
    return -1;
  }
  return null;
}
function getEffectiveTabIndex(disabled, tabIndex, groupName, groupTabIndex) {
  if (disabled) return -1;
  if (tabIndex !== void 0) return tabIndex;
  if (groupName) return groupTabIndex ?? 0;
  return 0;
}
function getValueStateBorderClass(valueState) {
  const borderMap = {
    ["Positive" /* Positive */]: "border-sapphire-positive",
    ["Negative" /* Negative */]: "border-sapphire-negative",
    ["Critical" /* Critical */]: "border-sapphire-warning",
    ["Information" /* Information */]: "border-sapphire-info"
  };
  return borderMap[valueState] ?? "";
}
function getValueStateCheckedBorderClass(valueState) {
  const checkedBorderMap = {
    ["Positive" /* Positive */]: "border-sapphire-positive",
    ["Negative" /* Negative */]: "border-sapphire-negative",
    ["Critical" /* Critical */]: "border-sapphire-warning",
    ["Information" /* Information */]: "border-sapphire-info"
  };
  return checkedBorderMap[valueState] ?? "border-sapphire-border-secondary";
}
function getInnerDotColorClass(valueState) {
  const dotMap = {
    ["Positive" /* Positive */]: "bg-sapphire-positive",
    ["Negative" /* Negative */]: "bg-sapphire-negative",
    ["Critical" /* Critical */]: "bg-sapphire-warning",
    ["Information" /* Information */]: "bg-sapphire-info"
  };
  return dotMap[valueState] ?? "bg-sapphire-text-accent";
}
function getSizeClasses(isSmall, hasLabel) {
  if (isSmall) {
    return hasLabel ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2";
  }
  return hasLabel ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]";
}
function getFocusRingClasses(hasLabel, isSmall) {
  if (!hasLabel) return "after:inset-[5px] after:rounded-lg";
  if (isSmall) return "after:inset-x-0 after:inset-y-0";
  return "after:left-0 after:right-[2px] after:inset-y-[5px]";
}
var valueStateI18nKeys = {
  ["Negative" /* Negative */]: "RADIOBUTTON_VALUE_STATE_ERROR",
  ["Critical" /* Critical */]: "RADIOBUTTON_VALUE_STATE_WARNING",
  ["Positive" /* Positive */]: "RADIOBUTTON_VALUE_STATE_SUCCESS",
  ["Information" /* Information */]: "RADIOBUTTON_VALUE_STATE_INFORMATION"
};
function RadioCircle({
  isReadonly,
  isSmall,
  isChecked,
  hasWrappingLabel,
  isInteractive,
  valueState,
  svgR
}) {
  if (isReadonly) {
    return /* @__PURE__ */ jsxRuntime.jsxs(
      "span",
      {
        className: cn(
          "relative inline-flex items-center justify-center shrink-0",
          isSmall ? "size-4" : "size-[22px]",
          hasWrappingLabel && "self-start",
          isSmall && hasWrappingLabel && "mt-0.5"
        ),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            "svg",
            {
              className: "absolute inset-0 w-full h-full overflow-visible",
              focusable: "false",
              "aria-hidden": "true",
              children: /* @__PURE__ */ jsxRuntime.jsx(
                "circle",
                {
                  cx: "50%",
                  cy: "50%",
                  r: svgR,
                  fill: "var(--background-secondary)",
                  stroke: "var(--border-secondary)",
                  strokeWidth: "1",
                  strokeDasharray: "4, 2"
                }
              )
            }
          ),
          isChecked && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: cn(
                "relative rounded-full",
                isSmall ? "size-2" : "size-3",
                "bg-sapphire-text-secondary"
              )
            }
          )
        ]
      }
    );
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      className: cn(
        "relative inline-flex items-center justify-center shrink-0 rounded-full",
        isSmall ? "size-4" : "size-[22px]",
        hasWrappingLabel && "self-start",
        isSmall && hasWrappingLabel && "mt-0.5",
        "transition-colors",
        "border border-solid",
        !isChecked && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
        isChecked && getValueStateCheckedBorderClass(valueState),
        isChecked && "bg-sapphire-canvas-primary",
        !isChecked && getValueStateBorderClass(valueState),
        isInteractive && valueState === "None" /* None */ && !isChecked && "group-hover:border-sapphire-chrome-button-fg-selected group-hover:bg-sapphire-chrome-button-bg-selected",
        isInteractive && valueState === "None" /* None */ && isChecked && "group-hover:border-sapphire-border-accent group-hover:bg-sapphire-chrome-button-bg-selected"
      ),
      children: isChecked && /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: cn(
            "rounded-full",
            isSmall ? "size-2" : "size-3",
            getInnerDotColorClass(valueState)
          )
        }
      )
    }
  );
}
function RadioButton({
  checked: controlledChecked,
  defaultChecked = false,
  disabled = false,
  readonly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  valueState = "None" /* None */,
  valueStateMessage,
  name,
  value = "",
  wrappingType = "Normal" /* Normal */,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  ref
}) {
  const inputRef = React3.useRef(null);
  const rootRef = React3.useRef(null);
  const generatedId = React3.useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = React3.useState(defaultChecked);
  const [groupTabIndex, setGroupTabIndex] = React3.useState(null);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const setChecked = React3.useCallback(
    (checked) => {
      if (!isControlled) {
        setInternalChecked(checked);
      }
    },
    [isControlled]
  );
  const doSelect = React3.useCallback(
    (originalEvent) => {
      if (disabled || readonly || isChecked) return;
      if (name) {
        uncheckOthersInGroup(name, id);
      }
      if (!isControlled) {
        setInternalChecked(true);
      }
      onChange?.({
        checked: true,
        value,
        originalEvent
      });
    },
    [disabled, readonly, isChecked, isControlled, name, id, onChange, value]
  );
  React3.useEffect(() => {
    if (!name) {
      setGroupTabIndex(null);
      return;
    }
    const entry = {
      id,
      getNode: () => rootRef.current,
      isDisabled: () => disabled,
      isReadonly: () => readonly,
      isChecked: () => isChecked,
      setChecked,
      focus: () => rootRef.current?.focus(),
      select: doSelect,
      setTabIndex: setGroupTabIndex
    };
    registerGroupEntry(name, entry);
    return () => {
      unregisterGroupEntry(name, id);
    };
  }, [name, id, disabled, readonly, isChecked, setChecked, doSelect]);
  React3.useEffect(() => {
    if (!name) {
      return;
    }
    updateGroupTabIndex(name);
  }, [name, isChecked, disabled]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked]
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled || readonly) {
        e.preventDefault();
        return;
      }
      doSelect(e);
    },
    [disabled, readonly, doSelect]
  );
  const handleMouseDown = React3.useCallback(
    (e) => {
      if (disabled) {
        e.preventDefault();
      }
    },
    [disabled]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled || readonly) return;
      const direction = getKeyNavigationDirection(e.key);
      if (name && (direction === 1 || direction === -1)) {
        e.preventDefault();
        moveSelectionInGroup(name, id, direction, e);
        return;
      }
      if (direction === "select") {
        e.preventDefault();
      }
    },
    [disabled, readonly, name, id]
  );
  const handleKeyUp = React3.useCallback(
    (e) => {
      if (disabled || readonly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doSelect(e);
      }
    },
    [disabled, readonly, doSelect]
  );
  const labelContent = children ?? text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const effectiveTabIndex = getEffectiveTabIndex(disabled, tabIndex, name, groupTabIndex);
  const { t } = reactI18next.useTranslation("fx");
  const valueStateI18nKey = valueStateI18nKeys[valueState];
  const defaultValueStateText = valueStateI18nKey ? t(valueStateI18nKey) : void 0;
  const svgR = isSmall ? 7.5 : 10.5;
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        ref: inputRef,
        type: "radio",
        id,
        name,
        value,
        checked: isChecked,
        disabled,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: rootRef,
        role: "radio",
        "aria-checked": isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          accessibleDescriptionRef || null,
          valueState !== "None" /* None */ ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || void 0,
        "aria-invalid": valueState === "Negative" /* Negative */ || void 0,
        "aria-required": required || void 0,
        tabIndex: effectiveTabIndex,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        "data-radio-group": name,
        "data-radio-id": id,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          getSizeClasses(isSmall, !!labelContent),
          "text-sm",
          "rounded-lg",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          getFocusRingClasses(!!labelContent, isSmall),
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            RadioCircle,
            {
              isReadonly: readonly,
              isSmall,
              isChecked,
              hasWrappingLabel,
              isInteractive,
              valueState,
              svgR
            }
          ),
          labelContent && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" /* None */ && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage || defaultValueStateText }),
          accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}
var pathData5 = [
  "M454 109q11 0 18.5 7t7.5 18-7.5 18.5T454 160h-19v294q0 24-17 41t-41 17H135q-24 0-41-17t-17-41V160H58q-11 0-18.5-7.5T32 134t7.5-18 18.5-7h70V58q0-24 17-41t41-17h140q24 0 41 17t17 41v51h70zm-275 0h154V58q0-7-7-7H186q-7 0-7 7v51zm205 51H128v294q0 7 7 7h242q7 0 7-7V160zm-186 64q11 0 18.5 7.5T224 250v140q0 11-7.5 18.5T198 416t-18-7.5-7-18.5V250q0-11 7-18.5t18-7.5zm116 0q11 0 18 7.5t7 18.5v140q0 11-7 18.5t-18 7.5-18.5-7.5T288 390V250q0-11 7.5-18.5T314 224z"
];
var DeleteIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData5, ...props })
  ),
  { displayName: "DeleteIcon", iconName: "delete" }
);
function ListSelectionControl({
  selectionMode,
  position,
  isSelected,
  disabled = false,
  itemText = "item",
  deleteButton,
  onSelectionChange,
  onDeleteClick,
  ref
}) {
  const modeStr = String(selectionMode);
  const showSelectionBefore = position === "before" && (modeStr === "Multiple" || modeStr === "SingleStart");
  const showSelectionAfter = position === "after" && (modeStr === "SingleEnd" || modeStr === "Delete");
  const showRadio = modeStr === "SingleStart" || modeStr === "SingleEnd" || modeStr === "Single";
  const showCheckbox = modeStr === "Multiple";
  const showDelete = modeStr === "Delete";
  if (!showSelectionBefore && !showSelectionAfter) {
    return null;
  }
  const handleDeleteClick = (detail) => {
    detail.originalEvent.stopPropagation();
    onDeleteClick?.();
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "span",
    {
      ref,
      className: "shrink-0 pl-3 pr-1 flex items-center",
      onClick: (e) => e.stopPropagation(),
      children: [
        showCheckbox && /* @__PURE__ */ jsxRuntime.jsx(
          CheckBox,
          {
            checked: isSelected,
            onChange: (e) => onSelectionChange(e.checked),
            disabled,
            accessibleName: `Select ${itemText}`,
            tabIndex: -1
          }
        ),
        showRadio && !showDelete && /* @__PURE__ */ jsxRuntime.jsx(
          RadioButton,
          {
            checked: isSelected,
            onChange: (e) => onSelectionChange(e.checked),
            disabled,
            accessibleName: `Select ${itemText}`,
            tabIndex: -1
          }
        ),
        showDelete && (deleteButton ? React3__namespace.cloneElement(deleteButton, {
          onClick: handleDeleteClick
        }) : /* @__PURE__ */ jsxRuntime.jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsxRuntime.jsx(DeleteIcon, { className: "h-4 w-4" }),
            onClick: handleDeleteClick,
            accessibleName: "Delete item",
            tooltip: "Delete",
            tabIndex: -1
          }
        ))
      ]
    }
  );
}
function DropIndicator({ visible, placement, orientation = "vertical", className, ref }) {
  if (!visible) return null;
  const isVertical = orientation === "vertical";
  const isBefore = placement === "Before";
  const isOn = placement === "On";
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      className: cn(
        "absolute pointer-events-none z-20",
        isVertical ? cn(
          "left-0 right-0 h-0.5 bg-primary",
          isBefore ? "-top-px" : isOn ? "inset-0 h-full bg-primary/10 border-2 border-primary border-dashed" : "-bottom-px"
        ) : cn(
          "top-0 bottom-0 w-0.5 bg-primary",
          isBefore ? "-left-px" : isOn ? "inset-0 w-full bg-primary/10 border-2 border-primary border-dashed" : "-right-px"
        ),
        className
      ),
      "aria-hidden": "true",
      "data-drop-indicator": true,
      "data-placement": placement
    }
  );
}

// src/components/list/shared-styles.ts
var LIST_ITEM_BASE_CLASSES = [
  "relative",
  "flex",
  "items-center",
  "w-full",
  "min-h-[2.75rem]",
  "transition-colors",
  "outline-none"
];
var LIST_ITEM_FOCUS_CLASSES = "ring-2 ring-inset ring-sapphire-border-focus z-10";
function ListItemBase(props) {
  const {
    ref,
    itemKey,
    type = "Active" /* Active */,
    disabled = false,
    movable = false,
    accessibleRole = "ListItem" /* ListItem */,
    accessibleNameRef,
    selected: controlledSelected,
    focused: externalFocused,
    onClick,
    onDragStart,
    onDelete,
    deleteButton,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const listContext = useOptionalListContext();
  const effectiveAccessibleRole = React3.useMemo(() => {
    if (accessibleRole !== void 0 && accessibleRole !== "ListItem" /* ListItem */) {
      return accessibleRole;
    }
    const parentRole = String(listContext?.accessibleRole || "List");
    if (accessibleRole === "ListItem" /* ListItem */ && parentRole === "List") {
      return "ListItem" /* ListItem */;
    }
    switch (parentRole) {
      case "ListBox":
        return "Option" /* Option */;
      case "Menu":
        return "MenuItem" /* MenuItem */;
      case "Tree":
        return "TreeItem" /* TreeItem */;
      default:
        return "ListItem" /* ListItem */;
    }
  }, [accessibleRole, listContext?.accessibleRole]);
  const listItem = useListItem({
    itemKey,
    disabled,
    movable,
    type,
    accessibleRole: effectiveAccessibleRole,
    selected: controlledSelected,
    onClick,
    onDragStart
  });
  const handleDeleteClick = React3.useCallback(() => {
    if (itemKey) {
      listContext?.onItemDelete?.(itemKey);
    }
    onDelete?.();
  }, [itemKey, listContext, onDelete]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        listItem.focus();
      },
      blur() {
        listItem.blur();
      },
      isFocused() {
        return document.activeElement === listItem.itemRef.current;
      },
      get nativeElement() {
        return listItem.itemRef.current;
      }
    }),
    [listItem]
  );
  const renderState = {
    isSelected: listItem.isSelected,
    isFocused: listItem.isFocused,
    isFocusedWithin: listItem.isFocusedWithin,
    isActive: listItem.isActive,
    isDragging: listItem.isDragging,
    isDropTarget: listItem.isDropTarget,
    dropPlacement: listItem.dropPlacement,
    selectionMode: String(listItem.selectionMode)
  };
  const modeStr = String(listItem.selectionMode);
  const position = modeStr === "SingleEnd" || modeStr === "Delete" ? "after" : "before";
  const shouldShowControl = modeStr !== "None";
  const content = typeof children === "function" ? children(renderState) : children;
  const resolvedRole = (() => {
    const roleStr = String(effectiveAccessibleRole);
    switch (roleStr) {
      case "Option":
        return "option";
      case "MenuItem":
        return "menuitem";
      case "TreeItem":
        return "treeitem";
      case "Group":
        return "group";
      case "None":
        return void 0;
      default:
        return "listitem";
    }
  })();
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "li",
    {
      ref: listItem.itemRef,
      id: listItem.id,
      role: resolvedRole,
      "aria-labelledby": accessibleNameRef,
      "aria-selected": listItem.ariaSelected,
      "aria-disabled": disabled || void 0,
      "aria-setsize": listContext?.totalItemCount,
      "aria-posinset": itemKey && listContext ? listContext.getItemIndex(itemKey) + 1 || void 0 : void 0,
      tabIndex: listItem.tabIndex,
      draggable: movable,
      onClick: listItem.handleClick,
      onKeyDown: listItem.handleKeyDown,
      onMouseDownCapture: listItem.handleMouseDownCapture,
      onMouseDown: listItem.handleMouseDown,
      onMouseUp: listItem.handleMouseUp,
      onMouseLeave: listItem.handleMouseUp,
      onFocus: listItem.handleFocus,
      onBlur: listItem.handleBlur,
      onDragStart: listItem.handleDragStart,
      onDragEnd: listItem.handleDragEnd,
      onDragOver: listItem.handleDragOver,
      onDragEnter: listItem.handleDragEnter,
      onDragLeave: listItem.handleDragLeave,
      onDrop: listItem.handleDrop,
      className: cn(
        // Base item styles - container layout (shared with group headers)
        LIST_ITEM_BASE_CLASSES,
        "text-sm text-sapphire-text-primary",
        // State classes
        listItem.isSelected && "!bg-sapphire-brand-selected-background",
        listItem.isSelected && !(externalFocused || listItem.isFocused) && "!border-sapphire-border-accent",
        (externalFocused || listItem.isFocused) && LIST_ITEM_FOCUS_CLASSES,
        listItem.isActive && "bg-muted/80",
        listItem.isDragging && "opacity-50",
        movable && "cursor-grab",
        listItem.isDragging && "cursor-grabbing",
        disabled && "opacity-50 pointer-events-none cursor-not-allowed",
        // Interactive states
        (type === "Active" /* Active */ || type === "Navigation" /* Navigation */) && "cursor-pointer hover:bg-[var(--table-row-hover)] active:bg-[var(--neutral-pressed-background-2)]",
        type === "Inactive" /* Inactive */ && "cursor-default",
        // User's custom classes
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-notification-item": props["data-notification-item"] || void 0,
      "data-notification-group": props["data-notification-group"] || void 0,
      "data-item-key": itemKey,
      "data-dragging": listItem.isDragging || void 0,
      children: [
        listItem.dropPlacement && /* @__PURE__ */ jsxRuntime.jsx(DropIndicator, { visible: true, placement: listItem.dropPlacement }),
        shouldShowControl && position === "before" && /* @__PURE__ */ jsxRuntime.jsx(
          ListSelectionControl,
          {
            selectionMode: listItem.selectionMode,
            position: "before",
            isSelected: listItem.isSelected,
            disabled,
            deleteButton,
            onSelectionChange: listItem.handleSelectionChange,
            onDeleteClick: handleDeleteClick
          }
        ),
        content,
        shouldShowControl && position === "after" && /* @__PURE__ */ jsxRuntime.jsx(
          ListSelectionControl,
          {
            selectionMode: listItem.selectionMode,
            position: "after",
            isSelected: listItem.isSelected,
            disabled,
            deleteButton,
            onSelectionChange: listItem.handleSelectionChange,
            onDeleteClick: handleDeleteClick
          }
        )
      ]
    }
  );
}
ListItemBase.displayName = "ListItemBase";
function NavArrowIcon({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", className, children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M5.20493 3.23532C5.48927 2.93424 5.96333 2.92076 6.2645 3.20504L10.7645 7.45211C10.9144 7.59365 10.9997 7.79085 10.9998 7.99704C10.9999 8.20338 10.9145 8.40121 10.7645 8.54293L6.2645 12.7949C5.96343 13.0792 5.48937 13.0656 5.20493 12.7646C4.92058 12.4635 4.93419 11.9895 5.2352 11.705L9.15707 7.99704L5.2352 4.29489C4.93413 4.01054 4.92064 3.53648 5.20493 3.23532Z", fill: "currentColor" }) });
}
var highlightBarVariants = cva(
  "absolute left-0 top-0 bottom-0 w-[3px]",
  {
    variants: {
      highlight: {
        ["None" /* None */]: "bg-transparent",
        ["Positive" /* Positive */]: "bg-sapphire-positive",
        ["Critical" /* Critical */]: "bg-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info",
        ["Negative" /* Negative */]: "bg-sapphire-negative"
      }
    },
    defaultVariants: {
      highlight: "None" /* None */
    }
  }
);
var additionalTextVariants = cva(
  "text-xs shrink-0",
  {
    variants: {
      state: {
        ["None" /* None */]: "text-sapphire-text-tertiary",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Negative" /* Negative */]: "text-sapphire-negative"
      }
    },
    defaultVariants: {
      state: "None" /* None */
    }
  }
);
function ListItem(props) {
  const {
    ref,
    // Content
    text,
    description,
    icon,
    iconEnd = false,
    additionalText,
    additionalTextState = "None" /* None */,
    image,
    action,
    children,
    // Behavior
    type = "Active" /* Active */,
    highlight = "None" /* None */,
    navigated = false,
    wrappingType = "None" /* None */,
    tooltip,
    // Events
    onDetailClick,
    onDelete: _onDelete,
    // Custom elements
    deleteButton,
    // Rest passed to ListItemBase
    ...baseProps
  } = props;
  const shouldWrap = String(wrappingType) === "Normal";
  const displayIcon = icon && !iconEnd;
  const displayIconEnd = icon && iconEnd;
  const handleDetailButtonClick = React3__namespace.useCallback((detail) => {
    onDetailClick?.(detail.originalEvent);
  }, [onDetailClick]);
  const showNavArrow = String(type) === "Navigation" || navigated;
  return /* @__PURE__ */ jsxRuntime.jsx(
    ListItemBase,
    {
      ref,
      type,
      deleteButton,
      ...baseProps,
      children: () => /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
        String(highlight) !== "None" && /* @__PURE__ */ jsxRuntime.jsx("span", { className: highlightBarVariants({ highlight }) }),
        /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "flex-1 flex items-center px-[0.75rem] py-[0.5rem] gap-[0.75rem] min-w-0", children: [
          image && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0 size-8 max-w-8 max-h-8 overflow-hidden flex items-center justify-center", children: image }),
          displayIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0 text-sapphire-text-tertiary", children: icon }),
          /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "flex-1 flex flex-col justify-center min-w-0 gap-0.5", children: [
            /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex items-center gap-2", children: /* @__PURE__ */ jsxRuntime.jsx(
              "span",
              {
                className: cn(
                  "text-sm font-semibold text-sapphire-text-primary",
                  !shouldWrap && "truncate",
                  shouldWrap && "break-words"
                ),
                title: tooltip,
                children: children || text
              }
            ) }),
            description && /* @__PURE__ */ jsxRuntime.jsx(
              "span",
              {
                className: cn(
                  "text-xs text-sapphire-text-tertiary",
                  !shouldWrap && "truncate",
                  shouldWrap && "break-words"
                ),
                children: description
              }
            )
          ] }),
          additionalText && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn(
            additionalTextVariants({ state: additionalTextState }),
            "shrink-0"
          ), children: additionalText }),
          displayIconEnd && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0 text-sapphire-text-tertiary", children: icon }),
          String(type) === "Detail" && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0", children: /* @__PURE__ */ jsxRuntime.jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              iconOnly: true,
              icon: /* @__PURE__ */ jsxRuntime.jsx(OverflowIcon, { className: "h-4 w-4" }),
              onClick: handleDetailButtonClick,
              accessibleName: "Show details",
              tooltip: "Details"
            }
          ) }),
          (action || showNavArrow) && /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "shrink-0 flex items-center gap-[0.5rem]", children: [
            action && /* @__PURE__ */ jsxRuntime.jsx(
              "span",
              {
                onClick: (e) => e.stopPropagation(),
                onKeyDown: (e) => e.stopPropagation(),
                children: action
              }
            ),
            showNavArrow && /* @__PURE__ */ jsxRuntime.jsx(NavArrowIcon, { className: "text-sapphire-text-tertiary" })
          ] })
        ] })
      ] })
    }
  );
}
ListItem.displayName = "ListItem";
var highlightBarVariants2 = cva(
  "absolute left-0 top-0 bottom-0 w-[3px]",
  {
    variants: {
      highlight: {
        ["None" /* None */]: "bg-transparent",
        ["Positive" /* Positive */]: "bg-sapphire-positive",
        ["Critical" /* Critical */]: "bg-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info",
        ["Negative" /* Negative */]: "bg-sapphire-negative"
      }
    },
    defaultVariants: {
      highlight: "None" /* None */
    }
  }
);
function ListItemCustom(props) {
  const {
    ref,
    // Content
    children,
    // Behavior
    type = "Active" /* Active */,
    highlight = "None" /* None */,
    // Rest passed to ListItemBase
    ...baseProps
  } = props;
  return /* @__PURE__ */ jsxRuntime.jsx(
    ListItemBase,
    {
      ref,
      type,
      ...baseProps,
      children: () => /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
        String(highlight) !== "None" && /* @__PURE__ */ jsxRuntime.jsx("span", { className: highlightBarVariants2({ highlight }) }),
        /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-1 px-3 py-2 min-w-0", children })
      ] })
    }
  );
}
ListItemCustom.displayName = "ListItemCustom";
function ConversationListItem({
  itemKey,
  content,
  selected = false,
  bold = false,
  onClick,
  actions,
  className,
  ref
}) {
  const isStringContent = typeof content === "string";
  return /* @__PURE__ */ jsxRuntime.jsx(
    ListItemBase,
    {
      ref,
      itemKey,
      type: "Active" /* Active */,
      selected,
      onClick,
      className: cn(
        "group px-4 py-0",
        selected && "!bg-sapphire-brand-background/10",
        className
      ),
      children: ({ isFocused, isFocusedWithin }) => /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center gap-2 w-full min-w-0", children: [
        /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex-1 min-w-0", children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: cn(
          "text-sm",
          isStringContent && "truncate",
          selected ? "font-semibold text-sapphire-brand-foreground" : bold ? "font-bold text-sapphire-text-primary group-hover:text-sapphire-brand-foreground" : "font-normal text-sapphire-text-primary group-hover:text-sapphire-brand-foreground"
        ), children: content }) }),
        actions && /* @__PURE__ */ jsxRuntime.jsx("div", { className: cn(
          "shrink-0 flex items-center gap-1 transition-opacity",
          selected || isFocused || isFocusedWithin ? "visible opacity-100 relative" : "invisible opacity-0 absolute group-hover:visible group-hover:opacity-100 group-hover:relative"
        ), children: actions })
      ] })
    }
  );
}
var pathData6 = [
  "M410.53 200.75C418.95 210.98 417.48 226.11 407.25 234.53L271.34 346.53C262.48 353.83 249.68 353.83 240.81 346.53L104.75 234.53C94.52 226.11 93.05 210.98 101.47 200.75C109.89 190.52 125.02 189.05 135.25 197.47L256.06 296.91L376.75 197.47C386.98 189.05 402.11 190.52 410.53 200.75Z"
];
var SlimArrowDownIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData6, ...props })
  ),
  { displayName: "SlimArrowDownIcon", iconName: "slim-arrow-down" }
);
var pathData7 = [
  "M186 416q-11 0-18.5-7.5T160 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T186 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L203 409q-7 7-17 7z"
];
var SlimArrowRightIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData7, ...props })
  ),
  { displayName: "SlimArrowRightIcon", iconName: "slim-arrow-right" }
);
function ListItemGroup(props) {
  const {
    ref,
    // Content
    headerText,
    headerAccessibleName,
    header,
    children,
    // Behavior
    wrappingType = "None" /* None */,
    stickyHeader = false,
    collapsible = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    onToggle,
    // Standard
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const groupRef = React3.useRef(null);
  const headerRef = React3.useRef(null);
  const generatedId = React3.useId();
  const headerId = `${generatedId}-header`;
  const contentId = `${generatedId}-content`;
  const listContext = useOptionalListContext();
  const [internalCollapsed, setInternalCollapsed] = React3.useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const [isFocused, setIsFocused] = React3.useState(false);
  const headerIndexRef = React3.useRef(-1);
  React3.useEffect(() => {
    if (listContext && headerRef.current) {
      let rootList = groupRef.current?.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(headerRef.current);
        if (index >= 0) {
          headerIndexRef.current = index;
          listContext.registerNavigationItem?.(index, headerRef.current);
        }
      }
    }
    return () => {
      if (listContext && headerIndexRef.current >= 0) {
        listContext.registerNavigationItem?.(headerIndexRef.current, null);
        headerIndexRef.current = -1;
      }
    };
  }, [listContext]);
  const headerIndex = headerIndexRef.current;
  const isFocusedItem = listContext?.focusedIndex === headerIndex;
  const isFirstItem = headerIndex === 0;
  const computedTabIndex = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem ? 0 : -1;
  const handleToggle = React3.useCallback(() => {
    if (!collapsible) return;
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.(newCollapsed);
  }, [collapsible, isCollapsed, isControlled, onToggle]);
  const handleKeyDown = React3.useCallback((e) => {
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      return;
    }
    if (collapsible && (e.key === "Enter" || e.key === " ")) {
      e.preventDefault();
      handleToggle();
    }
    if (collapsible) {
      if (e.key === "ArrowLeft" && !isCollapsed) {
        e.preventDefault();
        handleToggle();
      } else if (e.key === "ArrowRight" && isCollapsed) {
        e.preventDefault();
        handleToggle();
      }
    }
  }, [collapsible, handleToggle, isCollapsed]);
  const shouldWrap = String(wrappingType) === "Normal";
  const hasHeader = headerText || header;
  const separatorClass = listContext?.separators !== "None" ? "border-b border-sapphire-border-primary last:border-b-0" : "";
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "li",
    {
      ref: (node) => {
        groupRef.current = node;
        if (typeof ref === "function") {
          ref(node);
        } else if (ref) {
          ref.current = node;
        }
      },
      role: "group",
      "aria-labelledby": hasHeader ? headerId : void 0,
      className: cn(
        "relative list-none",
        separatorClass,
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        hasHeader && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: headerRef,
            id: headerId,
            role: collapsible ? "button" : "heading",
            "aria-level": 3,
            "aria-expanded": collapsible ? !isCollapsed : void 0,
            "aria-controls": collapsible ? contentId : void 0,
            "aria-label": headerAccessibleName,
            tabIndex: computedTabIndex,
            onClick: collapsible ? handleToggle : void 0,
            onKeyDown: handleKeyDown,
            onFocus: (e) => {
              const isHeaderItself = e.target === headerRef.current;
              setIsFocused(isHeaderItself);
              if (listContext && headerIndex >= 0) {
                listContext.onFocusItem(headerIndex);
              }
            },
            onBlur: (e) => {
              const isLeavingHeader = !headerRef.current?.contains(e.relatedTarget);
              if (isLeavingHeader) {
                setIsFocused(false);
              }
            },
            "data-group-header": "true",
            className: cn(
              LIST_ITEM_BASE_CLASSES,
              "gap-2 px-4 py-2",
              "bg-muted/50 text-sm font-semibold text-sapphire-text-tertiary",
              "select-none",
              stickyHeader && "sticky top-0 z-10",
              collapsible && "cursor-pointer hover:bg-muted/70",
              isFocused && LIST_ITEM_FOCUS_CLASSES
            ),
            children: [
              collapsible && !header && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0 text-sapphire-text-tertiary", children: isCollapsed ? /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: "h-4 w-4" }) }),
              header || /* @__PURE__ */ jsxRuntime.jsx(
                "span",
                {
                  className: cn(
                    "flex-1",
                    !shouldWrap && "truncate",
                    shouldWrap && "break-words"
                  ),
                  children: headerText
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "ul",
          {
            id: contentId,
            role: "group",
            className: cn(
              "list-none m-0 p-0 w-full",
              collapsible && isCollapsed && "hidden"
            ),
            children
          }
        )
      ]
    }
  );
}
ListItemGroup.displayName = "ListItemGroup";
function ConversationGroupListItem({
  headerText,
  collapsible = true,
  defaultCollapsed = false,
  collapsed,
  onToggle,
  stickyHeader = false,
  onSeeAll,
  children,
  className,
  "data-testid": dataTestId,
  ref
}) {
  const [internalCollapsed, setInternalCollapsed] = React3__namespace.useState(defaultCollapsed);
  const isControlled = collapsed !== void 0;
  const isCollapsed = isControlled ? collapsed : internalCollapsed;
  const content = onSeeAll ? /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    children,
    /* @__PURE__ */ jsxRuntime.jsx("li", { className: "list-none px-4 py-2", children: /* @__PURE__ */ jsxRuntime.jsx(
      Button,
      {
        design: "Tertiary" /* Tertiary */,
        onClick: onSeeAll,
        className: "text-sm font-semibold text-sapphire-text hover:text-sapphire-brand-foreground",
        children: "See All"
      }
    ) })
  ] }) : children;
  const customHeader = collapsible ? /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center justify-between w-full gap-2", children: [
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-1 truncate font-normal text-sm text-sapphire-text-tertiary", children: headerText }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "shrink-0 flex items-center justify-center w-8 h-8 text-sapphire-text-tertiary",
        "aria-hidden": "true",
        children: isCollapsed ? /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "h-4 w-4 pointer-events-none" }) : /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: "h-4 w-4 pointer-events-none" })
      }
    )
  ] }) : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "font-normal text-sm text-sapphire-text-tertiary", children: headerText });
  return /* @__PURE__ */ jsxRuntime.jsx(
    ListItemGroup,
    {
      ref,
      header: customHeader,
      collapsible,
      defaultCollapsed,
      collapsed,
      onToggle: (newCollapsed) => {
        if (!isControlled) {
          setInternalCollapsed(newCollapsed);
        }
        onToggle?.(newCollapsed);
      },
      stickyHeader,
      wrappingType: "None" /* None */,
      className: cn(
        "[&>[data-group-header]]:!bg-transparent",
        "[&>[data-group-header]]:pl-2",
        "[&>[data-group-header]]:pr-0",
        "[&>[data-group-header]]:py-1.5",
        "[&>ul>li]:border-b-0",
        className
      ),
      "data-testid": dataTestId,
      children: content
    }
  );
}
function ListItemSeparator(props) {
  const {
    ref,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  return /* @__PURE__ */ jsxRuntime.jsx(
    "li",
    {
      ref,
      role: "separator",
      "aria-orientation": "horizontal",
      className: cn(
        "relative",
        "w-full",
        "h-px",
        "bg-sapphire-border-primary",
        "my-2",
        "pointer-events-none",
        className
      ),
      style,
      "data-testid": dataTestId
    }
  );
}
var ListItemGroupContext = React3.createContext(null);
function useListItemGroup() {
  return React3.useContext(ListItemGroupContext);
}

// src/components/list/useListGroupHeader.ts
function useListGroupHeader(options = {}) {
  const {
    collapsible = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    onToggle
  } = options;
  const headerRef = React3.useRef(null);
  const generatedId = React3.useId();
  const headerId = `${generatedId}-header`;
  const groupContext = useListItemGroup();
  const contentId = groupContext?.contentId || `${generatedId}-content`;
  const listContext = useOptionalListContext();
  const [internalCollapsed, setInternalCollapsed] = React3.useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const [isFocused, setIsFocused] = React3.useState(false);
  const headerIndexRef = React3.useRef(-1);
  React3.useEffect(() => {
    if (listContext && headerRef.current) {
      let rootList = headerRef.current.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(headerRef.current);
        if (index >= 0) {
          headerIndexRef.current = index;
          listContext.registerNavigationItem?.(index, headerRef.current);
        }
      }
    }
    return () => {
      if (listContext && headerIndexRef.current >= 0) {
        listContext.registerNavigationItem?.(headerIndexRef.current, null);
        headerIndexRef.current = -1;
      }
    };
  }, [listContext]);
  const handleToggle = React3.useCallback(() => {
    if (!collapsible) return;
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.(newCollapsed);
  }, [collapsible, isCollapsed, isControlled, onToggle]);
  const handleKeyDown = React3.useCallback((e) => {
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      return;
    }
    if (collapsible && (e.key === "Enter" || e.key === " ")) {
      e.preventDefault();
      handleToggle();
    }
  }, [collapsible, handleToggle]);
  const handleFocus = React3.useCallback((e) => {
    const isHeaderItself = e.target === headerRef.current;
    setIsFocused(isHeaderItself);
    if (listContext && headerIndexRef.current >= 0) {
      listContext.onFocusItem(headerIndexRef.current);
    }
  }, [listContext]);
  const handleBlur = React3.useCallback((e) => {
    const isLeavingHeader = !headerRef.current?.contains(e.relatedTarget);
    if (isLeavingHeader) {
      setIsFocused(false);
    }
  }, []);
  const headerIndex = headerIndexRef.current;
  const isFocusedItem = listContext?.focusedIndex === headerIndex;
  const isFirstItem = headerIndex === 0;
  const tabIndex = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem ? 0 : -1;
  const focus = React3.useCallback(() => {
    headerRef.current?.focus();
  }, []);
  const blur = React3.useCallback(() => {
    headerRef.current?.blur();
  }, []);
  return {
    // Refs
    headerRef,
    headerId,
    contentId,
    // State
    isCollapsed,
    isFocused,
    // Computed values
    tabIndex,
    // Event handlers
    handleToggle,
    handleKeyDown,
    handleFocus,
    handleBlur,
    // Helper functions
    focus,
    blur
  };
}
function ListGroupHeaderBase(props) {
  const {
    ref,
    collapsible = false,
    collapsed,
    defaultCollapsed = false,
    onToggle,
    accessibleName,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const groupHeader = useListGroupHeader({
    collapsible,
    collapsed,
    defaultCollapsed,
    onToggle
  });
  const renderState = {
    isCollapsed: groupHeader.isCollapsed,
    isFocused: groupHeader.isFocused,
    collapsible
  };
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref: (node) => {
        groupHeader.headerRef.current = node;
        if (typeof ref === "function") {
          ref(node);
        } else if (ref) {
          ref.current = node;
        }
      },
      id: groupHeader.headerId,
      role: collapsible ? "button" : "heading",
      "aria-level": !collapsible ? 3 : void 0,
      "aria-expanded": collapsible ? !groupHeader.isCollapsed : void 0,
      "aria-controls": collapsible ? groupHeader.contentId : void 0,
      "aria-label": accessibleName,
      tabIndex: groupHeader.tabIndex,
      onClick: collapsible ? groupHeader.handleToggle : void 0,
      onKeyDown: groupHeader.handleKeyDown,
      onFocus: groupHeader.handleFocus,
      onBlur: groupHeader.handleBlur,
      "data-group-header": "true",
      className: cn(
        LIST_ITEM_BASE_CLASSES,
        groupHeader.isFocused && LIST_ITEM_FOCUS_CLASSES,
        className
      ),
      style,
      "data-testid": dataTestId,
      children: typeof children === "function" ? children(renderState) : children
    }
  );
}
function ListItemGroupBase(props) {
  const {
    ref,
    header,
    headerAccessibleName,
    collapsible = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    onToggle,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const generatedId = React3.useId();
  const contentId = `${generatedId}-content`;
  const [internalCollapsed, setInternalCollapsed] = React3.useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const handleToggle = React3.useCallback((collapsed) => {
    if (!isControlled) {
      setInternalCollapsed(collapsed);
    }
    onToggle?.(collapsed);
  }, [isControlled, onToggle]);
  const toggle = React3.useCallback(() => {
    handleToggle(!isCollapsed);
  }, [handleToggle, isCollapsed]);
  const contextValue = React3.useMemo(() => ({
    isCollapsed,
    collapsible,
    toggle,
    contentId
  }), [isCollapsed, collapsible, toggle, contentId]);
  const headerContent = typeof header === "function" ? header({ isCollapsed, isFocused: false }) : header;
  return /* @__PURE__ */ jsxRuntime.jsxs(ListItemGroupContext.Provider, { value: contextValue, children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      ListGroupHeaderBase,
      {
        ref,
        collapsible,
        collapsed: isCollapsed,
        onToggle: handleToggle,
        accessibleName: headerAccessibleName,
        className,
        style,
        "data-testid": dataTestId,
        children: headerContent
      }
    ),
    (!collapsible || !isCollapsed) && children
  ] });
}

exports.ConversationGroupListItem = ConversationGroupListItem;
exports.ConversationListItem = ConversationListItem;
exports.DropIndicator = DropIndicator;
exports.LIST_ITEM_BASE_CLASSES = LIST_ITEM_BASE_CLASSES;
exports.LIST_ITEM_FOCUS_CLASSES = LIST_ITEM_FOCUS_CLASSES;
exports.List = List;
exports.ListContext = ListContext;
exports.ListGroupHeaderBase = ListGroupHeaderBase;
exports.ListItem = ListItem;
exports.ListItemBase = ListItemBase;
exports.ListItemCustom = ListItemCustom;
exports.ListItemGroup = ListItemGroup;
exports.ListItemGroupBase = ListItemGroupBase;
exports.ListItemSeparator = ListItemSeparator;
exports.ListSelectionControl = ListSelectionControl;
exports.useListContext = useListContext;
exports.useListGroupHeader = useListGroupHeader;
exports.useListItem = useListItem;
exports.useListItemDragDrop = useListItemDragDrop;
exports.useListItemFocus = useListItemFocus;
exports.useListItemGroup = useListItemGroup;
exports.useListItemRegistration = useListItemRegistration;
exports.useListItemSelection = useListItemSelection;
exports.useOptionalListContext = useOptionalListContext;
