// @sap-ui/fx-components/dist/components/title/index.cjs
'use strict';

var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

// src/lib/utils.ts
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var titleVariants = cva(
  ["text-foreground"],
  {
    variants: {
      level: {
        ["H1" /* H1 */]: "text-[3.5rem] leading-[4rem] font-light",
        ["H2" /* H2 */]: "text-[2.5rem] leading-[3rem] font-normal",
        ["H3" /* H3 */]: "text-[2rem] leading-[2.5rem] font-normal",
        ["H4" /* H4 */]: "text-xl leading-7 font-normal",
        ["H5" /* H5 */]: "text-base leading-[1.375rem] font-semibold",
        ["H6" /* H6 */]: "text-base leading-[1.375rem] font-normal"
      },
      wrappingType: {
        ["None" /* None */]: "truncate",
        ["Normal" /* Normal */]: ""
      }
    },
    defaultVariants: {
      level: "H2" /* H2 */,
      wrappingType: "None" /* None */
    }
  }
);
var levelToElement = {
  ["H1" /* H1 */]: "h1",
  ["H2" /* H2 */]: "h2",
  ["H3" /* H3 */]: "h3",
  ["H4" /* H4 */]: "h4",
  ["H5" /* H5 */]: "h5",
  ["H6" /* H6 */]: "h6"
};
function Title({
  level = "H2" /* H2 */,
  wrappingType = "None" /* None */,
  children,
  className,
  style,
  id,
  ref
}) {
  const Tag = levelToElement[level] || "h2";
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tag,
    {
      ref,
      id,
      className: cn(
        titleVariants({
          level,
          wrappingType
        }),
        className
      ),
      style,
      children
    }
  );
}

exports.Title = Title;
