// @sap-ui/fx-components/dist/components/combobox/index.cjs
'use strict';

var React11 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');
var lucideReact = require('lucide-react');
var reactDom = require('react-dom');
var reactI18next = require('react-i18next');

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var React11__namespace = /*#__PURE__*/_interopNamespace(React11);

var __defProp = Object.defineProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}

// src/types/value-state.ts
var ValueState = /* @__PURE__ */ ((ValueState2) => {
  ValueState2["None"] = "None";
  ValueState2["Positive"] = "Positive";
  ValueState2["Negative"] = "Negative";
  ValueState2["Critical"] = "Critical";
  ValueState2["Information"] = "Information";
  return ValueState2;
})(ValueState || {});

// src/types/combobox.ts
var ComboBoxSize = /* @__PURE__ */ ((ComboBoxSize2) => {
  ComboBoxSize2["Large"] = "Large";
  ComboBoxSize2["Medium"] = "Medium";
  return ComboBoxSize2;
})(ComboBoxSize || {});
var ComboBoxFilter = /* @__PURE__ */ ((ComboBoxFilter2) => {
  ComboBoxFilter2["StartsWithPerTerm"] = "StartsWithPerTerm";
  ComboBoxFilter2["StartsWith"] = "StartsWith";
  ComboBoxFilter2["Contains"] = "Contains";
  ComboBoxFilter2["None"] = "None";
  return ComboBoxFilter2;
})(ComboBoxFilter || {});
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData13,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React11.useRef(null);
  React11.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick2 = React11.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React11.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData13 || pathData13.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick2,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData13.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M282 416q-11 0-18.5-7.5T256 390t7-18l109-109q7-7 18-7t18.5 7.5T416 282t-7 18L300 409q-7 7-18 7zm-160 0q-11 0-18.5-7.5T96 390t7-18l269-269q7-7 18-7t18.5 7.5T416 122t-7 18L140 409q-7 7-18 7z"
];
var ResizeCornerIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "ResizeCornerIcon", iconName: "resize-corner" }
);
var titleVariants = cva(
  ["text-foreground"],
  {
    variants: {
      level: {
        ["H1" /* H1 */]: "text-[3.5rem] leading-[4rem] font-light",
        ["H2" /* H2 */]: "text-[2.5rem] leading-[3rem] font-normal",
        ["H3" /* H3 */]: "text-[2rem] leading-[2.5rem] font-normal",
        ["H4" /* H4 */]: "text-xl leading-7 font-normal",
        ["H5" /* H5 */]: "text-base leading-[1.375rem] font-semibold",
        ["H6" /* H6 */]: "text-base leading-[1.375rem] font-normal"
      },
      wrappingType: {
        ["None" /* None */]: "truncate",
        ["Normal" /* Normal */]: ""
      }
    },
    defaultVariants: {
      level: "H2" /* H2 */,
      wrappingType: "None" /* None */
    }
  }
);
var levelToElement = {
  ["H1" /* H1 */]: "h1",
  ["H2" /* H2 */]: "h2",
  ["H3" /* H3 */]: "h3",
  ["H4" /* H4 */]: "h4",
  ["H5" /* H5 */]: "h5",
  ["H6" /* H6 */]: "h6"
};
function Title({
  level = "H2" /* H2 */,
  wrappingType = "None" /* None */,
  children,
  className,
  style,
  id,
  ref
}) {
  const Tag = levelToElement[level] || "h2";
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tag,
    {
      ref,
      id,
      className: cn(
        titleVariants({
          level,
          wrappingType
        }),
        className
      ),
      style,
      children
    }
  );
}

// src/lib/popupRegistry.ts
var openedPopups = [];
function attachListeners() {
  document.addEventListener("keydown", handleKeydown);
  document.addEventListener("mousedown", handleClick, true);
}
function detachListeners() {
  document.removeEventListener("keydown", handleKeydown);
  document.removeEventListener("mousedown", handleClick, true);
}
function handleKeydown(e) {
  if (e.key !== "Escape" || openedPopups.length === 0) return;
  e.preventDefault();
  e.stopImmediatePropagation();
  const topmost = openedPopups[openedPopups.length - 1];
  topmost.close(true);
}
function handleClick(e) {
  if (openedPopups.length === 0) return;
  const target = e.target;
  for (let i = openedPopups.length - 1; i >= 0; i--) {
    const entry = openedPopups[i];
    if (entry.isModal || entry.type === "dialog") return;
    if (entry.getOpenerEl) {
      const openerEl = entry.getOpenerEl();
      if (openerEl?.contains(target)) return;
    }
    if (entry.element.contains(target)) return;
    if (entry.isInsidePopup?.(target)) return;
    entry.close();
  }
}
function addOpenedPopup(entry) {
  openedPopups.push(entry);
  if (openedPopups.length === 1) {
    attachListeners();
  }
}
function removeOpenedPopup(element) {
  const idx = openedPopups.findIndex((p) => p.element === element);
  if (idx !== -1) openedPopups.splice(idx, 1);
  if (openedPopups.length === 0) {
    detachListeners();
  }
}

// src/lib/Device.ts
var isSSR = typeof document === "undefined";
function getUserAgent() {
  return isSSR ? "" : navigator.userAgent;
}
function hasTouch() {
  if (isSSR) return false;
  return "ontouchstart" in window || navigator.maxTouchPoints > 0;
}
function isIPad() {
  if (isSSR) return false;
  const ua = getUserAgent();
  return /ipad/i.test(ua) || /Macintosh/i.test(ua) && "ontouchend" in document;
}
function isWindowsPlatform() {
  if (isSSR) return false;
  return navigator.platform.indexOf("Win") !== -1;
}
function isWindows8OrAbove() {
  if (!isWindowsPlatform()) return false;
  const matches = getUserAgent().match(/Windows NT (\d+)/);
  const winVer = matches ? parseFloat(matches[1]) : 0;
  return winVer >= 8;
}
function isAndroidUA() {
  if (isSSR) return false;
  return !isWindowsPlatform() && /Android/.test(getUserAgent());
}
function isChromeUA() {
  if (isSSR) return false;
  return /(Chrome|CriOS)/.test(getUserAgent());
}
function isWebkitUA() {
  if (isSSR) return false;
  return /webkit/.test(getUserAgent());
}
function isTablet() {
  if (isSSR) return false;
  const touch = hasTouch();
  const ua = getUserAgent();
  if (isIPad()) return true;
  if (touch) {
    if (isWindows8OrAbove()) return true;
    if (isChromeUA() && isAndroidUA()) {
      return !/Mobile Safari\/[.0-9]+/.test(ua);
    }
    let densityFactor = window.devicePixelRatio || 1;
    if (isAndroidUA() && isWebkitUA()) {
      const wkMatches = ua.match(/(webkit)[ /]([\w.]+)/);
      const wkVer = wkMatches ? parseFloat(wkMatches[2]) : 0;
      if (wkVer >= 537.1) densityFactor = 1;
    }
    return Math.min(
      window.screen.width / densityFactor,
      window.screen.height / densityFactor
    ) >= 600;
  }
  return ua.indexOf("Touch") !== -1 || isAndroidUA() && !/(?=android)(?=.*mobile)/i.test(ua);
}
function isPhone() {
  if (isSSR) return false;
  return hasTouch() && !isTablet();
}
function isDesktop() {
  if (isSSR) return false;
  return !isTablet() && !isPhone() || isWindows8OrAbove();
}
var ARROW_DEPTH = 8;
var ARROW_BASE = 16;
var ARROW_SIZE = ARROW_DEPTH;
var VIEWPORT_MARGIN = 10;
function clamp(val, min, max) {
  return Math.min(Math.max(val, min), max);
}
var FOCUSABLE = 'a[href], button:not([disabled]), textarea:not([disabled]), input:not([disabled]), select:not([disabled]), [tabindex]:not([tabindex="-1"])';
function getFirstFocusable(container) {
  for (const el of container.querySelectorAll(FOCUSABLE)) {
    if (el.dataset.focusTrap) continue;
    return el;
  }
  return null;
}
function getLastFocusable(container) {
  const list = container.querySelectorAll(FOCUSABLE);
  for (let i = list.length - 1; i >= 0; i--) {
    if (list[i].dataset.focusTrap) continue;
    return list[i];
  }
  return null;
}
function resolveOpener(opener) {
  if (!opener) return null;
  if (opener instanceof HTMLElement) return opener;
  if ("current" in opener) {
    const val = opener.current;
    if (!val) return null;
    if (val instanceof HTMLElement) return val;
    if (typeof val === "object" && "nativeElement" in val && val.nativeElement instanceof HTMLElement) {
      return val.nativeElement;
    }
  }
  return null;
}
function useIsRtl(ref) {
  const [isRtl, setIsRtl] = React11.useState(false);
  React11.useEffect(() => {
    const el = ref?.current ?? document.documentElement;
    const dir = el.closest("[dir]")?.getAttribute("dir") ?? document.documentElement.getAttribute("dir");
    setIsRtl(dir === "rtl");
  });
  return isRtl;
}
function isOpenerClippedByAncestor(opener) {
  let parent = opener.parentElement;
  while (parent && parent !== document.documentElement) {
    if (parent.hasAttribute("popover")) {
      try {
        if (parent.matches(":popover-open")) return false;
      } catch {
      }
    }
    const { overflow, overflowX, overflowY } = window.getComputedStyle(parent);
    const clips = [overflow, overflowX, overflowY].some(
      (v) => v === "hidden" || v === "scroll" || v === "auto"
    );
    if (clips) {
      const parentRect = parent.getBoundingClientRect();
      const openerRect = opener.getBoundingClientRect();
      if (openerRect.bottom <= parentRect.top || openerRect.top >= parentRect.bottom || openerRect.right <= parentRect.left || openerRect.left >= parentRect.right) {
        return true;
      }
    }
    parent = parent.parentElement;
  }
  return false;
}
function getActualHorizontalAlign(align, isRtl) {
  switch (align) {
    case "Start" /* Start */:
      return isRtl ? "Right" /* Right */ : "Left" /* Left */;
    case "End" /* End */:
      return isRtl ? "Left" /* Left */ : "Right" /* Right */;
    case "Stretch" /* Stretch */:
      return "Stretch" /* Stretch */;
    default:
      return "Center" /* Center */;
  }
}
function fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) {
  if (targetRect.left > popoverSize.width) return "Left" /* Left */;
  if (clientWidth - targetRect.right > targetRect.left) return "Right" /* Right */;
  if (clientHeight - targetRect.bottom > popoverSize.height) return "Bottom" /* Bottom */;
  if (clientHeight - targetRect.bottom < targetRect.top) return "Top" /* Top */;
  return void 0;
}
function getActualPlacement(desiredPlacement, isRtl, hideArrow, targetRect, popoverSize) {
  let actual;
  switch (desiredPlacement) {
    case "Start" /* Start */:
      actual = isRtl ? "Right" /* Right */ : "Left" /* Left */;
      break;
    case "End" /* End */:
      actual = isRtl ? "Left" /* Left */ : "Right" /* Right */;
      break;
    case "Top" /* Top */:
      actual = "Top" /* Top */;
      break;
    case "Bottom" /* Bottom */:
      actual = "Bottom" /* Bottom */;
      break;
    default:
      actual = isRtl ? "Left" /* Left */ : "Right" /* Right */;
  }
  const clientWidth = document.documentElement.clientWidth;
  let clientHeight = document.documentElement.clientHeight;
  const isVertical = desiredPlacement === "Top" /* Top */ || desiredPlacement === "Bottom" /* Bottom */;
  let popoverHeight = popoverSize.height;
  if (isVertical) {
    popoverHeight += hideArrow ? 0 : ARROW_SIZE;
    clientHeight -= VIEWPORT_MARGIN;
  }
  switch (actual) {
    case "Top" /* Top */:
      if (targetRect.top < popoverHeight && targetRect.top < clientHeight - targetRect.bottom)
        actual = "Bottom" /* Bottom */;
      break;
    case "Bottom" /* Bottom */:
      if (clientHeight - targetRect.bottom < popoverHeight && clientHeight - targetRect.bottom < targetRect.top)
        actual = "Top" /* Top */;
      break;
    case "Left" /* Left */:
      if (targetRect.left < popoverSize.width)
        actual = fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) || actual;
      break;
    case "Right" /* Right */:
      if (clientWidth - targetRect.right < popoverSize.width)
        actual = fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) || actual;
      break;
  }
  return actual;
}
function calcPlacement(targetRect, popoverSize, desiredPlacement, horizontalAlign, verticalAlign, isRtl, hideArrow, allowTargetOverlap, borderRadius, offset) {
  const clientWidth = document.documentElement.clientWidth;
  const clientHeight = document.documentElement.clientHeight;
  let maxHeight = clientHeight;
  let maxWidth = clientWidth;
  let stretchWidth;
  let stretchHeight;
  const actualPlacement = getActualPlacement(
    desiredPlacement,
    isRtl,
    hideArrow,
    targetRect,
    popoverSize
  );
  const isVertical = actualPlacement === "Top" /* Top */ || actualPlacement === "Bottom" /* Bottom */;
  if (horizontalAlign === "Stretch" /* Stretch */ && isVertical) {
    popoverSize = { ...popoverSize, width: targetRect.width };
    stretchWidth = `${targetRect.width}px`;
  } else if (verticalAlign === "Stretch" /* Stretch */ && !isVertical) {
    popoverSize = { ...popoverSize, height: targetRect.height };
    stretchHeight = `${targetRect.height}px`;
  }
  const arrowOffset = (hideArrow ? 0 : ARROW_SIZE) + offset;
  const actualHAlign = getActualHorizontalAlign(horizontalAlign, isRtl);
  const getVerticalLeft = () => {
    switch (actualHAlign) {
      case "Center" /* Center */:
      case "Stretch" /* Stretch */:
        return targetRect.left - (popoverSize.width - targetRect.width) / 2;
      case "Left" /* Left */:
        return targetRect.left;
      case "Right" /* Right */:
        return targetRect.right - popoverSize.width;
      default:
        return VIEWPORT_MARGIN;
    }
  };
  const getHorizontalTop = () => {
    switch (verticalAlign) {
      case "Center" /* Center */:
      case "Stretch" /* Stretch */:
        return targetRect.top - (popoverSize.height - targetRect.height) / 2;
      case "Top" /* Top */:
        return targetRect.top;
      case "Bottom" /* Bottom */:
        return targetRect.bottom - popoverSize.height;
      default:
        return 0;
    }
  };
  let left = VIEWPORT_MARGIN;
  let top = 0;
  switch (actualPlacement) {
    case "Top" /* Top */:
      left = getVerticalLeft();
      top = Math.max(targetRect.top - popoverSize.height - arrowOffset, 0);
      if (!allowTargetOverlap) maxHeight = targetRect.top - arrowOffset;
      break;
    case "Bottom" /* Bottom */:
      left = getVerticalLeft();
      top = targetRect.bottom + arrowOffset;
      if (allowTargetOverlap) {
        top = Math.max(Math.min(top, clientHeight - popoverSize.height), 0);
      } else {
        maxHeight = clientHeight - targetRect.bottom - arrowOffset;
      }
      break;
    case "Left" /* Left */:
      left = Math.max(targetRect.left - popoverSize.width - arrowOffset, 0);
      top = getHorizontalTop();
      if (!allowTargetOverlap) maxWidth = targetRect.left - arrowOffset;
      break;
    case "Right" /* Right */:
      left = targetRect.left + targetRect.width + arrowOffset;
      top = getHorizontalTop();
      if (allowTargetOverlap) {
        left = Math.max(Math.min(left, clientWidth - popoverSize.width), 0);
      } else {
        maxWidth = clientWidth - targetRect.right - arrowOffset;
      }
      break;
  }
  if (isVertical) {
    if (popoverSize.width > clientWidth || left < VIEWPORT_MARGIN) {
      left = VIEWPORT_MARGIN;
    } else if (left + popoverSize.width > clientWidth - VIEWPORT_MARGIN) {
      left = clientWidth - VIEWPORT_MARGIN - popoverSize.width;
    }
  } else {
    if (popoverSize.height > clientHeight || top < VIEWPORT_MARGIN) {
      top = VIEWPORT_MARGIN;
    } else if (top + popoverSize.height > clientHeight - VIEWPORT_MARGIN) {
      top = clientHeight - VIEWPORT_MARGIN - popoverSize.height;
    }
  }
  left = Math.round(left);
  top = Math.round(top);
  maxHeight = Math.round(maxHeight - VIEWPORT_MARGIN);
  maxWidth = Math.round(maxWidth - VIEWPORT_MARGIN);
  let arrowXCentered = actualHAlign === "Center" /* Center */ || actualHAlign === "Stretch" /* Stretch */;
  if (actualHAlign === "Right" /* Right */ && left <= targetRect.left)
    arrowXCentered = true;
  if (actualHAlign === "Left" /* Left */ && left + popoverSize.width >= targetRect.left + targetRect.width)
    arrowXCentered = true;
  let arrowTranslateX = 0;
  if (isVertical && arrowXCentered) {
    arrowTranslateX = targetRect.left + targetRect.width / 2 - left - popoverSize.width / 2;
  }
  let arrowTranslateY = 0;
  if (!isVertical) {
    arrowTranslateY = targetRect.top + targetRect.height / 2 - top - popoverSize.height / 2;
  }
  const safeY = popoverSize.height / 2 - borderRadius - ARROW_BASE / 2 - 2;
  arrowTranslateY = clamp(arrowTranslateY, -safeY, safeY);
  const safeX = popoverSize.width / 2 - borderRadius - ARROW_BASE / 2 - 2;
  arrowTranslateX = clamp(arrowTranslateX, -safeX, safeX);
  return {
    arrow: { x: Math.round(arrowTranslateX), y: Math.round(arrowTranslateY) },
    top,
    left,
    maxHeight,
    maxWidth,
    actualPlacement,
    stretchWidth,
    stretchHeight
  };
}
function getResizeHandleCorner(popoverRect, openerRect, side, verticalAlign, actualHAlign, isRtl) {
  const offset = 2;
  let openerCX = Math.floor(openerRect.x + openerRect.width / 2);
  const openerCY = Math.floor(openerRect.y + openerRect.height / 2);
  let popoverCX = Math.floor(popoverRect.x + popoverRect.width / 2);
  const popoverCY = Math.floor(popoverRect.y + popoverRect.height / 2);
  const isPopoverWider = popoverRect.width > openerRect.width;
  const isPopoverTaller = popoverRect.height > openerRect.height;
  if (isRtl) {
    openerCX = -openerCX;
    popoverCX = -popoverCX;
  }
  switch (side) {
    case "Left" /* Left */:
      if (isPopoverTaller) {
        return popoverCY > openerCY + offset ? "bottom-left" : "top-left";
      }
      return verticalAlign === "Top" /* Top */ ? "bottom-left" : "top-left";
    case "Right" /* Right */:
      if (isPopoverTaller) {
        return popoverCY + offset < openerCY ? "top-right" : "bottom-right";
      }
      return verticalAlign === "Bottom" /* Bottom */ ? "top-right" : "bottom-right";
    case "Bottom" /* Bottom */:
      if (isPopoverWider) {
        if (popoverCX + offset < openerCX) {
          return isRtl ? "bottom-right" : "bottom-left";
        }
        return isRtl ? "bottom-left" : "bottom-right";
      }
      if (isRtl) {
        return actualHAlign === "Left" /* Left */ ? "bottom-right" : "bottom-left";
      }
      return actualHAlign === "Right" /* Right */ ? "bottom-left" : "bottom-right";
    case "Top" /* Top */:
    default:
      if (isPopoverWider) {
        if (popoverCX + offset < openerCX) {
          return isRtl ? "top-right" : "top-left";
        }
        return isRtl ? "top-left" : "top-right";
      }
      if (isRtl) {
        return actualHAlign === "Left" /* Left */ ? "top-right" : "top-left";
      }
      return actualHAlign === "Right" /* Right */ ? "top-left" : "top-right";
  }
}
function Popover({
  headerText,
  placement = "End" /* End */,
  horizontalAlign = "Center" /* Center */,
  verticalAlign = "Center" /* Center */,
  hideArrow = false,
  offset: offsetProp,
  allowTargetOverlap = false,
  resizable = false,
  noPadding = false,
  opener,
  open = false,
  initialFocus,
  preventFocusRestore = false,
  preventInitialFocus = false,
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Dialog" /* Dialog */,
  accessibleDescription,
  header,
  footer,
  children,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  onClose,
  isInsidePopup,
  className,
  style,
  id,
  ref
}) {
  const popoverRef = React11.useRef(null);
  const arrowRef = React11.useRef(null);
  const focusedBeforeOpen = React11.useRef(null);
  const [isOpen, setIsOpen] = React11.useState(false);
  const [pos, setPos] = React11.useState(null);
  const [resizeSize, setResizeSize] = React11.useState(null);
  const resizeState = React11.useRef(null);
  const [isDesktop2] = React11.useState(() => isDesktop());
  const containerRef = React11.useRef(null);
  const isRtl = useIsRtl(containerRef);
  const getOpenerEl = React11.useCallback(() => resolveOpener(opener), [opener]);
  const doCloseRef = React11.useRef(() => {
  });
  const reposition = React11.useCallback(() => {
    const openerEl = getOpenerEl();
    const popEl = popoverRef.current;
    if (!openerEl || !popEl) return;
    const targetRect = openerEl.getBoundingClientRect();
    if (targetRect.bottom < 0 || targetRect.top > window.innerHeight || targetRect.right < 0 || targetRect.left > window.innerWidth) {
      doCloseRef.current();
      return;
    }
    if (targetRect.top === 0 && targetRect.bottom === 0 && targetRect.left === 0 && targetRect.right === 0) {
      doCloseRef.current();
      return;
    }
    if (isOpenerClippedByAncestor(openerEl)) {
      doCloseRef.current();
      return;
    }
    const rect = popEl.getBoundingClientRect();
    const popoverSize = { width: rect.width, height: rect.height };
    if (popoverSize.width === 0 || popoverSize.height === 0) return;
    const borderRadius = Number.parseInt(
      window.getComputedStyle(popEl).getPropertyValue("border-radius") || "0"
    );
    const result = calcPlacement(
      targetRect,
      popoverSize,
      placement,
      horizontalAlign,
      verticalAlign,
      isRtl,
      hideArrow,
      allowTargetOverlap,
      borderRadius,
      offsetProp ?? 0
    );
    setPos(result);
  }, [getOpenerEl, placement, horizontalAlign, verticalAlign, isRtl, hideArrow, allowTargetOverlap, offsetProp]);
  React11.useEffect(() => {
    if (!isOpen) return;
    const raf = requestAnimationFrame(() => {
      reposition();
      requestAnimationFrame(reposition);
    });
    const onScrollResize = () => reposition();
    window.addEventListener("scroll", onScrollResize, true);
    window.addEventListener("resize", onScrollResize);
    const openerEl = getOpenerEl();
    let ro = null;
    if (openerEl instanceof Element) {
      ro = new ResizeObserver(reposition);
      ro.observe(openerEl);
    }
    let roPopover = null;
    if (popoverRef.current instanceof Element) {
      roPopover = new ResizeObserver(reposition);
      roPopover.observe(popoverRef.current);
    }
    return () => {
      cancelAnimationFrame(raf);
      window.removeEventListener("scroll", onScrollResize, true);
      window.removeEventListener("resize", onScrollResize);
      ro?.disconnect();
      roPopover?.disconnect();
    };
  }, [isOpen, reposition, getOpenerEl]);
  const doOpen = React11.useCallback(() => {
    if (isOpen) return;
    if (onBeforeOpen?.() === false) return;
    focusedBeforeOpen.current = document.activeElement;
    setIsOpen(true);
  }, [isOpen, onBeforeOpen]);
  const doClose = React11.useCallback(
    (escPressed = false) => {
      if (!isOpen) return;
      if (onBeforeClose?.({ escPressed }) === false) return;
      const el = popoverRef.current;
      if (el && el.matches(":popover-open")) {
        el.hidePopover();
      }
      setIsOpen(false);
      setResizeSize(null);
      setPos(null);
      if (!preventFocusRestore && focusedBeforeOpen.current) {
        focusedBeforeOpen.current.focus();
        focusedBeforeOpen.current = null;
      }
      onClose?.();
    },
    [isOpen, onBeforeClose, onClose, preventFocusRestore]
  );
  doCloseRef.current = doClose;
  React11.useEffect(() => {
    if (open) doOpen();
    else if (isOpen) doClose();
  }, [open]);
  React11.useEffect(() => {
    const el = popoverRef.current;
    if (!el) return;
    if (isOpen && !el.matches(":popover-open")) {
      el.showPopover();
    }
    return () => {
      if (el.matches(":popover-open")) {
        el.hidePopover();
      }
    };
  }, [isOpen]);
  React11.useEffect(() => {
    if (!isOpen) return;
    requestAnimationFrame(() => {
      if (preventInitialFocus || !popoverRef.current) return;
      if (initialFocus) {
        const el = document.getElementById(initialFocus);
        if (el) {
          el.focus();
          onOpen?.();
          return;
        }
      }
      const first = getFirstFocusable(popoverRef.current);
      if (first) first.focus();
      else popoverRef.current.focus();
      onOpen?.();
    });
  }, [isOpen]);
  React11.useEffect(() => {
    if (!isOpen || !popoverRef.current) return;
    const el = popoverRef.current;
    addOpenedPopup({
      element: el,
      close: (escPressed) => doCloseRef.current(escPressed),
      type: "popover",
      isModal: false,
      getOpenerEl,
      isInsidePopup
    });
    return () => removeOpenedPopup(el);
  }, [isOpen, getOpenerEl, isInsidePopup]);
  const focusFirst = React11.useCallback(() => {
    if (!popoverRef.current) return;
    const el = getFirstFocusable(popoverRef.current);
    (el || popoverRef.current).focus();
  }, []);
  const focusLast = React11.useCallback(() => {
    if (!popoverRef.current) return;
    const el = getLastFocusable(popoverRef.current);
    (el || popoverRef.current).focus();
  }, []);
  const handleResizeMouseDown = React11.useCallback(
    (e) => {
      if (!resizable || !popoverRef.current) return;
      e.preventDefault();
      const rect = popoverRef.current.getBoundingClientRect();
      const openerEl = getOpenerEl();
      const opRect = openerEl?.getBoundingClientRect();
      const corner = opRect && pos ? getResizeHandleCorner(
        rect,
        opRect,
        pos.actualPlacement,
        verticalAlign,
        getActualHorizontalAlign(horizontalAlign, isRtl),
        isRtl
      ) : "bottom-right";
      resizeState.current = {
        startX: e.clientX,
        startY: e.clientY,
        startWidth: rect.width,
        startHeight: rect.height,
        corner
      };
      const onMove = (ev) => {
        if (!resizeState.current) return;
        const { startX, startY, startWidth, startHeight, corner: c } = resizeState.current;
        const dx = ev.clientX - startX;
        const dy = ev.clientY - startY;
        const isL = c === "top-left" || c === "bottom-left";
        const isT = c === "top-left" || c === "top-right";
        const w = clamp(startWidth + (isL ? -dx : dx), 150, window.innerWidth - VIEWPORT_MARGIN * 2);
        const h = clamp(startHeight + (isT ? -dy : dy), 100, window.innerHeight - VIEWPORT_MARGIN * 2);
        setResizeSize({ width: w, height: h });
      };
      const onUp = () => {
        resizeState.current = null;
        window.removeEventListener("mousemove", onMove);
        window.removeEventListener("mouseup", onUp);
      };
      window.addEventListener("mousemove", onMove);
      window.addEventListener("mouseup", onUp);
    },
    [resizable, getOpenerEl, pos]
  );
  React11.useImperativeHandle(ref, () => ({
    open() {
      doOpen();
    },
    close() {
      doClose();
    },
    async applyFocus() {
      if (!popoverRef.current) return;
      const el = getFirstFocusable(popoverRef.current);
      (el || popoverRef.current).focus();
    },
    isOpen() {
      return isOpen;
    },
    get nativeElement() {
      return popoverRef.current;
    }
  }), [isOpen, doOpen, doClose]);
  const resizeHandleCorner = React11.useMemo(() => {
    if (!resizable || !isDesktop2 || !popoverRef.current || !pos) return "bottom-right";
    const openerEl = getOpenerEl();
    if (!openerEl) return "bottom-right";
    return getResizeHandleCorner(
      popoverRef.current.getBoundingClientRect(),
      openerEl.getBoundingClientRect(),
      pos.actualPlacement,
      verticalAlign,
      getActualHorizontalAlign(horizontalAlign, isRtl),
      isRtl
    );
  }, [resizable, isDesktop2, getOpenerEl, pos, verticalAlign, horizontalAlign, isRtl]);
  const role = accessibleRole === "None" /* None */ ? void 0 : accessibleRole === "AlertDialog" /* AlertDialog */ ? "alertdialog" : "dialog";
  const ariaModal = accessibleRole === "None" /* None */ ? void 0 : "true";
  const displayHeader = !!(header || headerText);
  const actualPlacement = pos?.actualPlacement ?? "Right" /* Right */;
  const resizeHandleClasses = {
    "top-left": "-top-2 -left-2 cursor-nw-resize",
    "top-right": "-top-2 -right-2 cursor-ne-resize",
    "bottom-left": "-bottom-2 -left-2 cursor-ne-resize",
    "bottom-right": "-bottom-2 -right-2 cursor-nw-resize"
  };
  const resizeIconRotation = {
    "top-left": "180deg",
    "top-right": "270deg",
    "bottom-left": "90deg",
    "bottom-right": "0deg"
  };
  if (!isOpen) {
    return /* @__PURE__ */ jsxRuntime.jsx("div", { ref: containerRef, className: "hidden" });
  }
  const popoverStyle = {
    position: "fixed",
    top: pos ? pos.top : -1e4,
    left: pos ? pos.left : -1e4,
    maxHeight: pos ? pos.maxHeight : void 0,
    maxWidth: pos ? pos.maxWidth : void 0,
    boxShadow: "0px 10px 15px -5px rgba(10, 10, 10, 0.15), 0px 5px 5px -2px rgba(10, 10, 10, 0.05)",
    ...pos?.stretchWidth ? { width: pos.stretchWidth } : {},
    ...pos?.stretchHeight ? { height: pos.stretchHeight } : {},
    ...resizeSize ? { width: resizeSize.width, height: resizeSize.height } : {},
    ...style
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsx("div", { ref: containerRef, className: "hidden" }),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: popoverRef,
        id,
        popover: "manual",
        role,
        "aria-modal": ariaModal,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef || (!accessibleName && headerText ? `${id || "popover"}-title` : void 0),
        "aria-describedby": accessibleDescription ? `${id || "popover"}-desc` : void 0,
        tabIndex: -1,
        className: cn(
          "rounded-lg border border-border bg-popover text-popover-foreground outline-none",
          className
        ),
        style: popoverStyle,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", { tabIndex: 0, onFocus: focusLast, "data-focus-trap": "true", className: "sr-only", role: "none" }),
          !hideArrow && pos && (() => {
            const arrowBorderColor = "var(--border)";
            const arrowFill = "var(--popover)";
            let svgWidth;
            let svgHeight;
            let fillPoints;
            let strokePath;
            const wrapperStyle = { position: "absolute", lineHeight: 0 };
            switch (actualPlacement) {
              case "Bottom" /* Bottom */:
                svgWidth = ARROW_BASE;
                svgHeight = ARROW_DEPTH;
                fillPoints = `0,${ARROW_DEPTH} ${ARROW_BASE / 2},0 ${ARROW_BASE},${ARROW_DEPTH}`;
                strokePath = `M0,${ARROW_DEPTH} L${ARROW_BASE / 2},0 L${ARROW_BASE},${ARROW_DEPTH}`;
                wrapperStyle.left = "50%";
                wrapperStyle.top = -ARROW_DEPTH;
                wrapperStyle.transform = `translateX(calc(-50% + ${pos.arrow.x}px))`;
                break;
              case "Top" /* Top */:
                svgWidth = ARROW_BASE;
                svgHeight = ARROW_DEPTH;
                fillPoints = `0,0 ${ARROW_BASE / 2},${ARROW_DEPTH} ${ARROW_BASE},0`;
                strokePath = `M0,0 L${ARROW_BASE / 2},${ARROW_DEPTH} L${ARROW_BASE},0`;
                wrapperStyle.left = "50%";
                wrapperStyle.bottom = -ARROW_DEPTH;
                wrapperStyle.transform = `translateX(calc(-50% + ${pos.arrow.x}px))`;
                break;
              case "Right" /* Right */:
                svgWidth = ARROW_DEPTH;
                svgHeight = ARROW_BASE;
                fillPoints = `${ARROW_DEPTH},0 0,${ARROW_BASE / 2} ${ARROW_DEPTH},${ARROW_BASE}`;
                strokePath = `M${ARROW_DEPTH},0 L0,${ARROW_BASE / 2} L${ARROW_DEPTH},${ARROW_BASE}`;
                wrapperStyle.top = "50%";
                wrapperStyle.left = -ARROW_DEPTH;
                wrapperStyle.transform = `translateY(calc(-50% + ${pos.arrow.y}px))`;
                break;
              case "Left" /* Left */:
              default:
                svgWidth = ARROW_DEPTH;
                svgHeight = ARROW_BASE;
                fillPoints = `0,0 ${ARROW_DEPTH},${ARROW_BASE / 2} 0,${ARROW_BASE}`;
                strokePath = `M0,0 L${ARROW_DEPTH},${ARROW_BASE / 2} L0,${ARROW_BASE}`;
                wrapperStyle.top = "50%";
                wrapperStyle.right = -ARROW_DEPTH;
                wrapperStyle.transform = `translateY(calc(-50% + ${pos.arrow.y}px))`;
                break;
            }
            return /* @__PURE__ */ jsxRuntime.jsx("div", { ref: arrowRef, style: wrapperStyle, children: /* @__PURE__ */ jsxRuntime.jsxs("svg", { width: svgWidth, height: svgHeight, viewBox: `0 0 ${svgWidth} ${svgHeight}`, style: { display: "block" }, children: [
              /* @__PURE__ */ jsxRuntime.jsx("polygon", { points: fillPoints, fill: arrowFill, stroke: "none" }),
              /* @__PURE__ */ jsxRuntime.jsx("path", { d: strokePath, fill: "none", stroke: arrowBorderColor, strokeWidth: 1 })
            ] }) });
          })(),
          /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "isolate z-1 flex flex-col gap-sapphire-2xs overflow-hidden rounded-lg", style: { maxHeight: "inherit" }, children: [
            displayHeader && /* @__PURE__ */ jsxRuntime.jsx(
              "header",
              {
                className: cn(
                  "shrink-0",
                  // When custom header is provided, let it control its own styling.
                  // When using headerText, apply default popover header styling.
                  !header && "px-sapphire-s py-sapphire-xs border-b border-sapphire-border-primary"
                ),
                children: header || /* @__PURE__ */ jsxRuntime.jsx(Title, { level: "H1", id: `${id || "popover"}-title`, className: "text-base font-bold", children: headerText })
              }
            ),
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: cn("overflow-y-auto overflow-x-hidden flex-1 min-h-0", !noPadding && "px-sapphire-s py-sapphire-xs"), children }),
            accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id || "popover"}-desc`, className: "sr-only", children: accessibleDescription }),
            footer && /* @__PURE__ */ jsxRuntime.jsx("footer", { className: "px-sapphire-s py-sapphire-xs border-t border-sapphire-border-primary shrink-0", children: footer })
          ] }),
          resizable && isDesktop2 && /* @__PURE__ */ jsxRuntime.jsx(
            "div",
            {
              className: cn(
                "absolute flex items-center justify-center w-6 h-6 rounded-full z-[1]",
                resizeHandleClasses[resizeHandleCorner]
              ),
              onMouseDown: handleResizeMouseDown,
              children: /* @__PURE__ */ jsxRuntime.jsx(
                ResizeCornerIcon,
                {
                  className: "h-4 w-4 text-sapphire-text-tertiary",
                  style: { transform: `rotate(${resizeIconRotation[resizeHandleCorner]})` }
                }
              )
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsx("span", { tabIndex: 0, onFocus: focusFirst, "data-focus-trap": "true", className: "sr-only", role: "none" })
        ]
      }
    )
  ] });
}
var pathData2 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm0 461q42 0 79.5-16t65.5-44 44-65.5 16-79.5-16-79.5-44-65.5-65.5-44T256 51t-79.5 16-65.5 44-44 65.5T51 256t16 79.5 44 65.5 65.5 44 79.5 16zm70-301q11 0 18.5 7.5T352 186t-7 18l-53 52 53 52q7 7 7 18t-7.5 18.5T326 352q-10 0-18-8l-52-52-52 52q-8 8-18 8-11 0-18.5-7.5T160 326q0-10 8-18l52-52-52-52q-8-8-8-18 0-11 7.5-18.5T186 160t18 7l52 53 52-53q7-7 18-7z"
];
var MessageErrorIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "MessageErrorIcon", iconName: "message-error" }
);
var pathData3 = [
  "M256 512q-53 0-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0t99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20zm0-461q-42 0-79.5 16T111 111t-44 65.5T51 256t16 79.5 44 65.5 65.5 44 79.5 16 79.5-16 65.5-44 44-65.5 16-79.5-16-79.5-44-65.5-65.5-44T256 51zm-43 301q-10 0-17-7l-71-63q-9-8-9-19t7.5-18.5T142 237q10 0 17 7l53 47 128-131q8-8 18-8 11 0 18.5 7.5T384 178q0 10-7 17L231 344q-8 8-18 8z"
];
var MessageSuccessIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "MessageSuccessIcon", iconName: "message-success" }
);
var pathData4 = [
  "M436 480H76q-31 0-53.5-21T0 407q0-21 10-37L191 68q10-17 27.5-26.5T256 32t37.5 9.5T321 68l181 302q10 16 10 37 0 31-22.5 52T436 480zM256 83q-15 0-21 12L54 396q-3 6-3 11 0 9 7.5 15.5T76 429h360q10 0 17.5-6.5T461 407q0-5-3-11L277 95q-6-12-21-12zm0 213q-11 0-18.5-7.5T230 270v-85q0-11 7.5-18.5T256 159t18.5 7.5T282 185v85q0 11-7.5 18.5T256 296zm0 105q-13 0-22-9t-9-23q0-13 9-22t22-9 22 9 9 22q0 14-9 23t-22 9z"
];
var MessageWarningIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "MessageWarningIcon", iconName: "message-warning" }
);
var pathData5 = [
  "M256 0q53 0 100 20t81.5 54.5T492 156t20 100-20 100-54.5 81.5T356 492t-100 20-100-20-81.5-54.5T20 356 0 256t20-100 54.5-81.5T156 20 256 0zm0 461q43 0 80.5-16t65-44 43.5-65 16-80-16-80.5-43.5-65-65-43.5T256 51t-80 16-65 43.5-44 65T51 256t16 80 44 65 65 44 80 16zm0-333q13 0 22.5 9.5T288 160t-9.5 22.5T256 192t-22.5-9.5T224 160t9.5-22.5T256 128zm0 115q11 0 18 7.5t7 18.5v102q0 11-7 18.5t-18 7.5-18.5-7.5T230 371V269q0-11 7.5-18.5T256 243z"
];
var MessageInformationIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData5, ...props })
  ),
  { displayName: "MessageInformationIcon", iconName: "message-information" }
);
var pathData6 = [
  "M181 103q7-7 17-7 11 0 18.5 7.5T224 122q0 10-8 18l-94 90h332q11 0 18.5 7.5T480 256t-7.5 18.5T454 282H122l94 90q8 8 8 18 0 11-7.5 18.5T198 416q-10 0-17-7L40 274q-8-6-8-18 0-11 8-19z"
];
var ArrowLeftIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData6, ...props })
  ),
  { displayName: "ArrowLeftIcon", iconName: "arrow-left" }
);

// src/lib/dialogStack.ts
var DialogStackManager = class {
  constructor() {
    __publicField(this, "stack", []);
    __publicField(this, "baseZIndex", 50);
  }
  createBackdrop(onBackdropClick) {
    const el = document.createElement("div");
    el.setAttribute("aria-hidden", "true");
    el.setAttribute("data-dialog-backdrop", "true");
    el.setAttribute("popover", "manual");
    el.style.cssText = "position:fixed;inset:0;pointer-events:all;background:transparent;width:100vw;height:100vh;";
    const stop = (e) => e.stopPropagation();
    el.addEventListener("pointerdown", stop);
    el.addEventListener("pointerup", stop);
    const handleClick2 = (e) => {
      e.stopPropagation();
      onBackdropClick?.();
    };
    el.addEventListener("click", handleClick2);
    document.body.appendChild(el);
    el.showPopover();
    const cleanup = () => {
      el.removeEventListener("pointerdown", stop);
      el.removeEventListener("pointerup", stop);
      el.removeEventListener("click", handleClick2);
    };
    return { el, cleanup };
  }
  destroyBackdrop(item) {
    item.cleanupBackdrop();
    try {
      item.backdropEl.hidePopover();
    } catch {
    }
    item.backdropEl.remove();
  }
  /**
   * Register a new dialog on the stack
   */
  push(id, onBackdropClick) {
    const { el: backdropEl, cleanup: cleanupBackdrop } = this.createBackdrop(onBackdropClick);
    this.stack.push({ id, backdropEl, cleanupBackdrop });
    return this.baseZIndex + this.stack.length;
  }
  /**
   * Remove a dialog from the stack
   */
  pop(id) {
    const index = this.stack.findIndex((item2) => item2.id === id);
    if (index === -1) return;
    const item = this.stack[index];
    this.stack.splice(index, 1);
    this.destroyBackdrop(item);
  }
  /**
   * Get z-index for a specific dialog
   */
  getZIndex(id) {
    const index = this.stack.findIndex((item) => item.id === id);
    if (index === -1) return this.baseZIndex;
    return this.baseZIndex + index + 1;
  }
  /**
   * Check if dialog is the topmost
   */
  isTop(id) {
    if (this.stack.length === 0) return false;
    return this.stack[this.stack.length - 1].id === id;
  }
  /**
   * Get the number of open dialogs
   */
  getCount() {
    return this.stack.length;
  }
};
var dialogStack = new DialogStackManager();

// src/lib/focusUtils.ts
function isElementHidden(element) {
  if (element.hidden) {
    return true;
  }
  const style = window.getComputedStyle(element);
  if (style.display === "none" || style.visibility === "hidden") {
    return true;
  }
  return false;
}
function isScrollable(element) {
  const style = window.getComputedStyle(element);
  const overflowX = style.overflowX;
  const overflowY = style.overflowY;
  return (overflowX === "scroll" || overflowX === "auto" || overflowY === "scroll" || overflowY === "auto") && (element.scrollHeight > element.clientHeight || element.scrollWidth > element.clientWidth);
}
function isElementClickable(element) {
  if (element.disabled) {
    return false;
  }
  const tabindex = element.getAttribute("tabindex");
  if (tabindex && parseInt(tabindex, 10) < 0) {
    return false;
  }
  const clickableTags = ["BUTTON", "A", "INPUT", "SELECT", "TEXTAREA"];
  if (clickableTags.includes(element.tagName)) {
    return true;
  }
  if (element.onclick || element.hasAttribute("onclick")) {
    return true;
  }
  if (tabindex && parseInt(tabindex, 10) >= 0) {
    return true;
  }
  return false;
}
function isElemFocusable(element) {
  if (isElementHidden(element)) {
    return false;
  }
  return true;
}
function getChildrenForTraversal(element) {
  const children = [];
  if (element.shadowRoot) {
    children.push(...Array.from(element.shadowRoot.children));
  }
  if (element instanceof HTMLSlotElement) {
    const assigned = element.assignedElements({ flatten: true });
    children.push(...assigned);
  } else {
    children.push(...Array.from(element.children));
  }
  return children.filter((child) => child.nodeType === 1);
}
function findFocusableElementRecursive(container, forward = true, startFromContainer = false) {
  if (startFromContainer && isElemFocusable(container) && isElementClickable(container)) {
    return container;
  }
  const children = getChildrenForTraversal(container);
  const traverseOrder = forward ? children : children.reverse();
  for (const child of traverseOrder) {
    if (isElemFocusable(child) && isElementClickable(child)) {
      const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
      if (isSafari && isScrollable(child) && !isElementClickable(child)) {
        continue;
      }
      return child;
    }
    const found = findFocusableElementRecursive(child, forward, false);
    if (found) {
      return found;
    }
  }
  return null;
}
function findFirstFocusableElement(container, initialFocusId) {
  if (!container) return null;
  if (initialFocusId) {
    const targetElement = document.getElementById(initialFocusId);
    if (targetElement && container.contains(targetElement) && isElemFocusable(targetElement) && isElementClickable(targetElement)) {
      return targetElement;
    }
  }
  return findFocusableElementRecursive(container, true, false);
}
var dotVariants = cva("rounded-full bg-current inline-block", {
  variants: {
    size: {
      ["S" /* S */]: "w-2 h-2",
      ["M" /* M */]: "w-4 h-4",
      ["L" /* L */]: "w-8 h-8"
    }
  },
  defaultVariants: { size: "M" /* M */ }
});
var dotGapClasses = {
  ["S" /* S */]: "gap-px",
  ["M" /* M */]: "gap-[0.1875rem]",
  ["L" /* L */]: "gap-1"
};
function BusyIndicator({
  size = "M" /* M */,
  textPlacement = "Bottom" /* Bottom */,
  active = false,
  delay = 1e3,
  text,
  children,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = React11.useRef(null);
  const overlayRef = React11.useRef(null);
  const [isBusy, setIsBusy] = React11.useState(false);
  const labelId = React11.useId();
  const hasChildren = React11__namespace.default.Children.count(children) > 0;
  const normalizedSize = size || "M" /* M */;
  const isTopPlacement = textPlacement === "Top";
  React11.useEffect(() => {
    if (!active) {
      setIsBusy(false);
      return;
    }
    if (delay <= 0) {
      setIsBusy(true);
      return;
    }
    const timer = setTimeout(() => {
      setIsBusy(true);
    }, delay);
    return () => clearTimeout(timer);
  }, [active, delay]);
  React11.useImperativeHandle(
    ref,
    () => ({
      focus() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.focus();
        } else {
          rootRef.current?.focus();
        }
      },
      blur() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.blur();
        } else {
          rootRef.current?.blur();
        }
      },
      isFocused() {
        return document.activeElement === overlayRef.current || document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    [isBusy]
  );
  const textLabel = text ? /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      id: labelId,
      className: cn(
        "w-full text-center text-sapphire-text-tertiary text-sm font-medium",
        isTopPlacement ? "mb-2" : "mt-2"
      ),
      children: text
    }
  ) : null;
  const dots = /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "inline-flex items-center leading-none",
        dotGapClasses[normalizedSize]
      ),
      children: [0, 1, 2].map((i) => /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: dotVariants({ size: normalizedSize }),
          style: {
            animation: `busy-grow 1.6s cubic-bezier(0.32, 0.06, 0.85, 1.11) ${i * 200}ms infinite`
          }
        },
        i
      ))
    }
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      id,
      className: cn(
        "inline-block",
        hasChildren && "relative",
        isBusy && "text-ring",
        className
      ),
      style: { ...style, borderRadius: "inherit", height: "inherit" },
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("style", { children: `@keyframes busy-grow{0%,50%,100%{transform:scale(.5)}25%{transform:scale(1)}}` }),
        isBusy && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: overlayRef,
            role: "progressbar",
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            "aria-valuetext": "Busy",
            "aria-labelledby": text ? labelId : void 0,
            "aria-label": !text ? accessibleName ?? "Please wait" : void 0,
            title: !text ? accessibleName ?? "Please wait" : void 0,
            tabIndex: 0,
            "data-sap-focus-ref": true,
            className: cn(
              "flex flex-col items-center justify-center",
              "bg-inherit rounded-[inherit]",
              "focus-visible:outline focus-visible:outline-2 focus-visible:outline-ring focus-visible:outline-offset-[-2px]",
              hasChildren && "absolute inset-0 z-[99]"
            ),
            children: [
              isTopPlacement && textLabel,
              dots,
              !isTopPlacement && textLabel
            ]
          }
        ),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(isBusy && "opacity-40"),
            ...isBusy ? { inert: true } : {},
            children
          }
        )
      ]
    }
  );
}
var filledBrandStyles = [
  "bg-sapphire-brand-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-brand-hover-background",
  "active:bg-sapphire-brand-pressed-background [&.kb-active]:bg-sapphire-brand-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus"
];
var filledJouleStyles = [
  "bg-sapphire-joule-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-joule-hover-background",
  "active:bg-sapphire-joule-pressed-background [&.kb-active]:bg-sapphire-joule-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus"
];
var outlineBrandStyles = [
  "bg-transparent text-sapphire-brand-foreground border border-sapphire-brand-foreground",
  "hover:bg-sapphire-neutral-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-brand-pressed-foreground [&.kb-active]:text-sapphire-brand-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
];
var outlineJouleStyles = [
  "bg-transparent text-sapphire-joule-foreground border border-sapphire-joule-foreground",
  "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background hover:border-sapphire-joule-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-joule-pressed-foreground [&.kb-active]:text-sapphire-joule-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
];
var buttonVariants = cva(
  [
    // Base styles
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        // --- Primary (filled) ---
        ["Primary" /* Primary */]: filledBrandStyles,
        ["PrimaryJoule" /* PrimaryJoule */]: filledJouleStyles,
        // --- Secondary (outlined) ---
        ["Secondary" /* Secondary */]: outlineBrandStyles,
        ["SecondaryJoule" /* SecondaryJoule */]: outlineJouleStyles,
        // --- Tertiary (ghost, colored text) ---
        ["Tertiary" /* Tertiary */]: [
          "bg-transparent text-sapphire-brand-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-brand-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
        ],
        ["TertiaryJoule" /* TertiaryJoule */]: [
          "bg-transparent text-sapphire-joule-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-joule-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
        ],
        // --- Neutral (ghost, dark text) ---
        ["Neutral" /* Neutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-sapphire-neutral-foreground-muted",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-neutral-foreground-black [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- SecondaryNeutral (ghost, neutral-black text) ---
        ["SecondaryNeutral" /* SecondaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background [&.kb-active]:bg-sapphire-neutral-pressed-background",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- TertiaryNeutral (ghost, muted neutral text, no background) ---
        ["TertiaryNeutral" /* TertiaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-muted border border-transparent",
          "hover:text-sapphire-neutral-foreground-black",
          "active:text-sapphire-neutral-foreground-black [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ]
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var largeSizeClasses = {
  default: "h-10 px-4 py-3 text-base gap-2 rounded-lg",
  iconOnly: "h-10 w-10 p-0 text-base rounded-lg",
  icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
};
var mediumSizeClasses = {
  default: "h-8 px-3 py-2 text-sm gap-1.5 rounded",
  iconOnly: "h-8 w-8 p-0 text-sm rounded",
  icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
};
var smallSizeClasses = {
  default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
  iconOnly: "h-6 w-6 p-0 text-xs rounded",
  icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
};
var sizeClasses = {
  ["Large" /* Large */]: largeSizeClasses,
  ["Medium" /* Medium */]: mediumSizeClasses,
  ["Small" /* Small */]: smallSizeClasses
};
function getButtonType(type) {
  switch (type) {
    case "Submit" /* Submit */:
    case "Submit":
      return "submit";
    case "Reset" /* Reset */:
    case "Reset":
      return "reset";
    default:
      return "button";
  }
}
function Button({
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  type = "Button" /* Button */,
  disabled = false,
  loading = false,
  icon,
  endIcon,
  iconOnly = false,
  children,
  tooltip,
  states,
  state,
  stateTransition = "slide-up",
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Button" /* Button */,
  accessibilityAttributes,
  name,
  value,
  form,
  onClick,
  onFocus,
  onBlur,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  "data-growing-button": dataGrowingButton,
  ref
}) {
  const buttonRef = React11.useRef(null);
  const hiddenButtonRef = React11.useRef(null);
  const [animationPhase, setAnimationPhase] = React11.useState("idle");
  const [displayedState, setDisplayedState] = React11.useState(
    null
  );
  const [buttonWidth, setButtonWidth] = React11.useState(void 0);
  const currentStateObject = states?.find((s) => s.name === state) ?? states?.[0];
  const isMultiState = states && states.length > 0;
  React11.useEffect(() => {
    if (isMultiState && !displayedState) {
      setDisplayedState(currentStateObject ?? null);
    }
  }, [isMultiState, currentStateObject, displayedState]);
  React11.useEffect(() => {
    if (!isMultiState || !currentStateObject || animationPhase !== "idle") {
      return;
    }
    if (displayedState && displayedState.name !== currentStateObject.name) {
      if (buttonRef.current) {
        setButtonWidth(buttonRef.current.offsetWidth);
      }
      setAnimationPhase("fade-out");
    }
  }, [currentStateObject, displayedState, isMultiState, animationPhase]);
  React11.useEffect(() => {
    if (animationPhase === "fade-out") {
      const timer = setTimeout(() => {
        setAnimationPhase("fade-mid");
      }, 180);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-mid") {
      setDisplayedState(currentStateObject ?? null);
      if (hiddenButtonRef.current) {
        setButtonWidth(hiddenButtonRef.current.offsetWidth);
      }
      const timer = setTimeout(() => {
        setAnimationPhase("fade-in");
      }, 20);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-in") {
      const timer = setTimeout(() => {
        setAnimationPhase("idle");
        setButtonWidth(void 0);
      }, 160);
      return () => clearTimeout(timer);
    }
  }, [animationPhase, currentStateObject]);
  React11.useImperativeHandle(
    ref,
    () => ({
      focus() {
        buttonRef.current?.focus();
      },
      blur() {
        buttonRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === buttonRef.current;
      },
      get nativeElement() {
        return buttonRef.current;
      }
    }),
    []
  );
  const handleClick2 = React11.useCallback(
    (e) => {
      if (disabled || loading) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [disabled, loading, onClick]
  );
  const handleKeyDown = React11.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " ") {
          e.preventDefault();
        }
        buttonRef.current?.classList.add("kb-active");
        if (e.key === " ") {
          onClick?.({
            originalEvent: e,
            isKeyboard: true
          });
        }
      }
    },
    [disabled, loading, onClick]
  );
  const handleKeyUp = React11.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        buttonRef.current?.classList.remove("kb-active");
      }
    },
    []
  );
  const ariaExpanded = accessibilityAttributes?.expanded;
  const ariaHasPopup = accessibilityAttributes?.hasPopup;
  const ariaControls = accessibilityAttributes?.controls;
  const ariaPressed = accessibilityAttributes?.pressed;
  const role = accessibleRole === "Link" /* Link */ || accessibleRole === "Link" ? "link" : void 0;
  const isDisabled = disabled;
  const effectiveIcon = isMultiState ? displayedState?.icon : icon;
  const effectiveEndIcon = isMultiState ? displayedState?.endIcon : endIcon;
  const effectiveChildren = isMultiState ? displayedState?.text : children;
  const effectiveIconOnly = isMultiState ? !displayedState?.text && !!displayedState?.icon : iconOnly;
  const effectivePulsing = isMultiState ? displayedState?.pulsing : false;
  const PulsingText = ({ children: text }) => /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "opacity-50", children: text }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "absolute inset-0 overflow-hidden",
        style: {
          background: "linear-gradient(90deg, transparent 0%, currentColor 50%, transparent 100%)",
          backgroundSize: "200% 100%",
          WebkitBackgroundClip: "text",
          backgroundClip: "text",
          WebkitTextFillColor: "transparent",
          animation: "pulse-sweep 1.5s ease-in-out infinite"
        },
        children: text
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: `
          @keyframes pulse-sweep {
            0% { background-position: 200% 0; }
            100% { background-position: -200% 0; }
          }
        ` })
  ] });
  const getTransitionStyles = (phase, transition) => {
    if (transition === "none" || phase === "idle") {
      return {};
    }
    const baseTransition = "all 100ms ease-out";
    switch (transition) {
      case "slide-up":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(-8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-down":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(-8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-left":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(-12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "slide-right":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(-12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "fade":
        if (phase === "fade-out") return { opacity: 0, transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0 };
        if (phase === "fade-in") return { opacity: 1, transition: baseTransition };
        break;
      case "scale":
        if (phase === "fade-out") return { opacity: 0, transform: "scale(0.75)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "scale(0.75)" };
        if (phase === "fade-in") return { opacity: 1, transform: "scale(1)", transition: baseTransition };
        break;
      case "flip":
        if (phase === "fade-out") return { opacity: 0, transform: "rotateX(90deg)", transition: baseTransition, transformStyle: "preserve-3d" };
        if (phase === "fade-mid") return { opacity: 0, transform: "rotateX(-90deg)", transformStyle: "preserve-3d" };
        if (phase === "fade-in") return { opacity: 1, transform: "rotateX(0)", transition: baseTransition, transformStyle: "preserve-3d" };
        break;
    }
    return {};
  };
  const contentAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  const iconAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: buttonRef,
        type: getButtonType(type),
        disabled: isDisabled,
        "aria-disabled": loading || void 0,
        id,
        name,
        value,
        form,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-expanded": ariaExpanded,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-pressed": ariaPressed,
        "aria-busy": loading || void 0,
        title: tooltip,
        tabIndex,
        onClick: handleClick2,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur: (e) => {
          buttonRef.current?.classList.remove("kb-active");
          onBlur?.(e);
        },
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][effectiveIconOnly ? "iconOnly" : "default"],
          isMultiState && !effectiveIconOnly && "transition-[width] duration-300 ease-out overflow-hidden",
          loading && "relative pointer-events-none",
          className
        ),
        style: {
          ...style,
          ...buttonWidth !== void 0 && !effectiveIconOnly ? { width: buttonWidth } : {}
        },
        "data-testid": dataTestId,
        "data-growing-button": dataGrowingButton,
        children: [
          loading && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "absolute inset-0 flex items-center justify-center z-[1] [&_*]:!text-inherit", inert: true, children: /* @__PURE__ */ jsxRuntime.jsx(BusyIndicator, { size: "S", active: true, delay: 0 }) }),
          effectiveIcon ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? iconAnimationStyles : void 0, children: effectiveIcon }) : null,
          !effectiveIconOnly && effectiveChildren && /* @__PURE__ */ jsxRuntime.jsx("span", { className: loading ? "opacity-40" : void 0, style: isMultiState ? contentAnimationStyles : void 0, children: effectivePulsing ? /* @__PURE__ */ jsxRuntime.jsx(PulsingText, { children: effectiveChildren }) : effectiveChildren }),
          effectiveEndIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? contentAnimationStyles : void 0, children: effectiveEndIcon })
        ]
      }
    ),
    isMultiState && /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: hiddenButtonRef,
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][!currentStateObject?.text && currentStateObject?.icon ? "iconOnly" : "default"],
          "absolute -top-[10000px] -left-[10000px] invisible w-fit"
        ),
        tabIndex: -1,
        "aria-hidden": "true",
        children: [
          currentStateObject?.icon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.icon }),
          currentStateObject?.text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.text }),
          currentStateObject?.endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.endIcon })
        ]
      }
    )
  ] });
}
function useIsPhone() {
  const [phone, setPhone] = React11__namespace.useState(() => isPhone());
  React11__namespace.useEffect(() => {
    const handler = () => setPhone(isPhone());
    window.addEventListener("resize", handler);
    window.addEventListener("orientationchange", handler);
    return () => {
      window.removeEventListener("resize", handler);
      window.removeEventListener("orientationchange", handler);
    };
  }, []);
  return phone;
}
var dialogVariants = cva(
  [
    "relative flex flex-col",
    "bg-sapphire-card-bg-primary rounded-2xl",
    "shadow-[0_20px_25px_-5px_rgba(10,10,10,0.15),0_10px_15px_-5px_rgba(10,10,10,0.05)]",
    "min-w-[320px] max-w-[554px]",
    "max-h-[90vh]",
    "outline-none",
    "p-0 m-0",
    "border border-border"
  ],
  {
    variants: {
      state: {
        ["None" /* None */]: "",
        ["Error" /* Error */]: "",
        ["Warning" /* Warning */]: "",
        ["Success" /* Success */]: "",
        ["Information" /* Information */]: ""
      },
      stretch: {
        // Desktop stretch: ~90% viewport, no rounding. Phone full-screen handled via JS (isPhone()).
        true: "max-w-[95vw] h-[90vh] w-[90%]",
        false: ""
      },
      resizable: {
        true: "overflow-hidden max-w-[95vw] max-h-[95vh]",
        false: ""
      }
    },
    defaultVariants: {
      state: "None" /* None */,
      stretch: false,
      resizable: false
    }
  }
);
var headerVariants = cva(
  [
    "flex items-center justify-between",
    "px-6 py-5",
    "shrink-0",
    "focus:outline-none focus:rounded-t-2xl focus:[box-shadow:inset_0_0_0_4px_white,inset_0_0_0_6px_var(--ring)]"
  ],
  {
    variants: {
      state: {
        ["None" /* None */]: "border-sapphire-border-primary",
        ["Error" /* Error */]: "bg-sapphire-negative-bg border-sapphire-negative",
        ["Warning" /* Warning */]: "bg-sapphire-warning-bg border-sapphire-warning",
        ["Success" /* Success */]: "bg-sapphire-positive-bg border-sapphire-positive",
        ["Information" /* Information */]: "bg-sapphire-info-bg border-sapphire-info"
      },
      draggable: {
        true: "cursor-move select-none",
        false: ""
      }
    },
    defaultVariants: {
      state: "None" /* None */,
      draggable: false
    }
  }
);
function Dialog({
  className,
  open: controlledOpen,
  defaultOpen = false,
  headerText,
  subHeaderText,
  headerEndContent,
  showBackButton,
  onBackButtonClick,
  hideHeaderBorder,
  header,
  children,
  footer,
  state = "None" /* None */,
  stretch = false,
  draggable = false,
  resizable = false,
  preventInitialFocus = false,
  enableBackdropClick = false,
  noPadding = false,
  initialFocus,
  accessibleName,
  accessibleNameRef,
  accessibleDescribedBy,
  accessibleRole,
  ariaRoleDescription,
  onBeforeOpen,
  onAfterOpen,
  onBeforeClose,
  onAfterClose,
  onEscapePress,
  onOpenChange,
  id,
  ref
}) {
  const [internalOpen, setInternalOpen] = React11__namespace.useState(defaultOpen);
  const isOpen = controlledOpen ?? internalOpen;
  const onPhone = useIsPhone();
  const [isPopoverActuallyOpen, setIsPopoverActuallyOpen] = React11__namespace.useState(false);
  const dialogId = React11__namespace.useId();
  const stableId = id || dialogId;
  const dialogRef = React11__namespace.useRef(null);
  const headerRef = React11__namespace.useRef(null);
  const contentRef = React11__namespace.useRef(null);
  const previousActiveElementRef = React11__namespace.useRef(null);
  const returnValueRef = React11__namespace.useRef(void 0);
  const isClosingRef = React11__namespace.useRef(false);
  const didPopRef = React11__namespace.useRef(false);
  const [isDragging, setIsDragging] = React11__namespace.useState(false);
  const [position, setPosition] = React11__namespace.useState(null);
  const positionRef = React11__namespace.useRef(position);
  React11__namespace.useLayoutEffect(() => {
    positionRef.current = position;
  }, [position]);
  const dragStartRef = React11__namespace.useRef({ x: 0, y: 0, dialogX: 0, dialogY: 0 });
  const effectiveRole = React11__namespace.useMemo(() => {
    if (state === "Error" /* Error */ || state === "Warning" /* Warning */) {
      return "alertdialog";
    }
    return accessibleRole || "dialog";
  }, [state, accessibleRole]);
  const effectiveAriaDescribedBy = React11__namespace.useMemo(() => {
    const parts = [];
    if (accessibleDescribedBy) {
      parts.push(accessibleDescribedBy);
    }
    if (draggable || resizable) {
      parts.push(`${stableId}-movement-descr`);
    }
    return parts.length > 0 ? parts.join(" ") : void 0;
  }, [accessibleDescribedBy, draggable, resizable, stableId]);
  const getMovementDescription = () => {
    if (draggable && resizable) {
      return "This dialog can be moved and resized. Use Tab to focus the header, then arrow keys to move. Use Shift+Arrow keys to resize.";
    }
    if (draggable) {
      return "This dialog can be moved. Use Tab to focus the header, then arrow keys to reposition.";
    }
    if (resizable) {
      return "This dialog can be resized. Use Tab to focus the header, then Shift+Arrow keys to resize.";
    }
    return "";
  };
  const ensureDirectPositioning = React11__namespace.useCallback(() => {
    const dialog = dialogRef.current;
    if (!dialog || positionRef.current) return positionRef.current;
    const rect = dialog.getBoundingClientRect();
    dialog.style.width = `${rect.width}px`;
    const pos = { x: rect.left, y: rect.top };
    setPosition(pos);
    return pos;
  }, []);
  const clampPosition = React11__namespace.useCallback(
    (x, y) => {
      const dialog = dialogRef.current;
      if (!dialog) return { x, y };
      const w = dialog.offsetWidth;
      const h = dialog.offsetHeight;
      const vw = window.innerWidth;
      const vh = window.innerHeight;
      if (!w || !h || !vw || !vh) return { x, y };
      return {
        x: Math.max(0, Math.min(vw - w, x)),
        y: Math.max(0, Math.min(vh - h, y))
      };
    },
    []
  );
  const handleClose = React11__namespace.useCallback((escPressed) => {
    const dialog = dialogRef.current;
    if (!dialog) return;
    if (isClosingRef.current) return;
    isClosingRef.current = true;
    const beforeCloseDetail = {
      escPressed,
      returnValue: returnValueRef.current
    };
    const shouldClose = onBeforeClose?.(beforeCloseDetail);
    if (shouldClose === false) {
      isClosingRef.current = false;
      if (controlledOpen === void 0) {
        setInternalOpen(true);
      } else {
        onOpenChange?.(true);
      }
      return;
    }
    if (dialog.matches(":popover-open")) {
      dialog.hidePopover();
      if (!didPopRef.current) {
        didPopRef.current = true;
        dialogStack.pop(stableId);
      }
      setIsPopoverActuallyOpen(false);
    }
    setPosition(null);
    dialog.style.width = "";
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    onOpenChange?.(false);
    const afterCloseDetail = {
      returnValue: returnValueRef.current
    };
    setTimeout(() => {
      onAfterClose?.(afterCloseDetail);
      previousActiveElementRef.current?.focus();
      returnValueRef.current = void 0;
      isClosingRef.current = false;
      didPopRef.current = false;
    }, 200);
  }, [controlledOpen, onBeforeClose, onAfterClose, onOpenChange, stableId]);
  React11__namespace.useEffect(() => {
    if (!isPopoverActuallyOpen) return;
    const dialog = dialogRef.current;
    if (!dialog) return;
    const resizeObserver = new ResizeObserver(() => {
      if (!dialogRef.current) return;
      if (resizable || draggable) {
        const pos = positionRef.current ?? ensureDirectPositioning();
        if (!pos) return;
        const dialogRect = dialogRef.current.getBoundingClientRect();
        const vw = window.innerWidth;
        const vh = window.innerHeight;
        const newX = Math.max(0, Math.min(vw - dialogRect.width, pos.x));
        const newY = Math.max(0, Math.min(vh - dialogRect.height, pos.y));
        if (newX !== pos.x || newY !== pos.y) {
          setPosition({ x: newX, y: newY });
        }
      }
    });
    resizeObserver.observe(dialog);
    return () => {
      resizeObserver.disconnect();
    };
  }, [isPopoverActuallyOpen, draggable, resizable, ensureDirectPositioning]);
  React11__namespace.useEffect(() => {
    if (!isPopoverActuallyOpen) return;
    const handleResize = () => {
      const dialog = dialogRef.current;
      if (dialog) {
        dialog.style.width = "";
      }
      setPosition(null);
    };
    window.addEventListener("resize", handleResize);
    return () => window.removeEventListener("resize", handleResize);
  }, [isPopoverActuallyOpen]);
  const isPopoverActuallyOpenRef = React11__namespace.useRef(isPopoverActuallyOpen);
  React11__namespace.useEffect(() => {
    isPopoverActuallyOpenRef.current = isPopoverActuallyOpen;
  }, [isPopoverActuallyOpen]);
  const onAfterCloseRef = React11__namespace.useRef(onAfterClose);
  React11__namespace.useEffect(() => {
    onAfterCloseRef.current = onAfterClose;
  }, [onAfterClose]);
  const handleCloseRef = React11__namespace.useRef(handleClose);
  React11__namespace.useEffect(() => {
    handleCloseRef.current = handleClose;
  }, [handleClose]);
  React11__namespace.useEffect(() => {
    const dialog = dialogRef.current;
    const id2 = stableId;
    return () => {
      const isOpen2 = dialog?.matches(":popover-open");
      if (isOpen2) {
        try {
          dialog.hidePopover();
        } catch {
        }
      }
      if ((isOpen2 || isPopoverActuallyOpenRef.current) && !didPopRef.current) {
        didPopRef.current = true;
        dialogStack.pop(id2);
        onAfterCloseRef.current?.({ returnValue: returnValueRef.current });
      }
    };
  }, [stableId]);
  React11__namespace.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog) return;
    const isPopoverOpen = dialog.matches(":popover-open");
    if (isOpen && !isPopoverOpen) {
      isClosingRef.current = false;
      didPopRef.current = false;
      previousActiveElementRef.current = document.activeElement;
      const beforeOpenDetail = {};
      const shouldOpen = onBeforeOpen?.(beforeOpenDetail);
      if (shouldOpen === false) {
        if (controlledOpen === void 0) {
          setInternalOpen(false);
        } else {
          onOpenChange?.(false);
        }
        return;
      }
      dialogStack.push(stableId, enableBackdropClick ? () => handleCloseRef.current(false) : void 0);
      dialog.showPopover();
      setIsPopoverActuallyOpen(true);
      requestAnimationFrame(() => {
        requestAnimationFrame(() => {
          if (!preventInitialFocus) {
            const firstFocusable = findFirstFocusableElement(dialog, initialFocus);
            if (firstFocusable) {
              firstFocusable.focus();
            } else {
              dialog.focus();
            }
          }
        });
      });
      const timer = setTimeout(() => {
        onAfterOpen?.();
      }, 200);
      return () => clearTimeout(timer);
    } else if (!isOpen && isPopoverOpen) {
      handleClose(false);
    }
    if (!isPopoverOpen && isPopoverActuallyOpen) {
      setIsPopoverActuallyOpen(false);
    }
  }, [
    isOpen,
    handleClose,
    initialFocus,
    preventInitialFocus,
    onBeforeOpen,
    onAfterOpen,
    isPopoverActuallyOpen,
    controlledOpen,
    onOpenChange,
    stableId,
    enableBackdropClick
  ]);
  React11__namespace.useEffect(() => {
    const dialog = dialogRef.current;
    if (!isPopoverActuallyOpen || !dialog) return;
    addOpenedPopup({
      element: dialog,
      close: (escPressed) => {
        if (escPressed) onEscapePress?.();
        handleClose(!!escPressed);
      },
      type: "dialog",
      isModal: true
    });
    return () => removeOpenedPopup(dialog);
  }, [isPopoverActuallyOpen, handleClose, onEscapePress]);
  React11__namespace.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog || !isPopoverActuallyOpen) return;
    const handleKeyDown = (e) => {
      if (e.key === "Escape") return;
      if (e.key === " " || ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"].includes(e.key)) {
        const activeElement = document.activeElement;
        if (activeElement === dialog || activeElement === document.body) {
          e.preventDefault();
        }
      }
      if (e.key === "Tab") {
        const focusableElements = dialog?.querySelectorAll(
          'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
        );
        if (!focusableElements || focusableElements.length === 0) return;
        const firstElement = focusableElements[0];
        const lastElement = focusableElements[focusableElements.length - 1];
        if (e.shiftKey) {
          if (document.activeElement === firstElement) {
            e.preventDefault();
            lastElement.focus();
          }
        } else {
          if (document.activeElement === lastElement) {
            e.preventDefault();
            firstElement.focus();
          }
        }
      }
      if (["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"].includes(e.key)) {
        const activeElement = document.activeElement;
        const isInput = activeElement instanceof HTMLInputElement || activeElement instanceof HTMLTextAreaElement || activeElement?.getAttribute("contenteditable") === "true";
        if (isInput) return;
        const isHeaderFocused = headerRef.current?.contains(activeElement);
        const shouldResize = e.shiftKey && resizable && isHeaderFocused;
        const shouldDrag = !e.shiftKey && draggable && isHeaderFocused;
        if (!shouldDrag && !shouldResize) {
          return;
        }
        e.preventDefault();
        const step = 10;
        let deltaX = 0;
        let deltaY = 0;
        switch (e.key) {
          case "ArrowUp":
            deltaY = -step;
            break;
          case "ArrowDown":
            deltaY = step;
            break;
          case "ArrowLeft":
            deltaX = -step;
            break;
          case "ArrowRight":
            deltaX = step;
            break;
        }
        if (shouldDrag) {
          const base = positionRef.current ?? ensureDirectPositioning() ?? { x: 0, y: 0 };
          setPosition(clampPosition(base.x + deltaX, base.y + deltaY));
        } else if (shouldResize && dialog) {
          const currentWidth = dialog.offsetWidth;
          const currentHeight = dialog.offsetHeight;
          const isRTL = document.documentElement.dir === "rtl";
          const base = positionRef.current ?? ensureDirectPositioning() ?? { x: 0, y: 0 };
          const effectiveDeltaX = isRTL ? -deltaX : deltaX;
          const newWidth = Math.max(200, currentWidth + effectiveDeltaX * 2);
          const newHeight = Math.max(150, currentHeight + deltaY * 2);
          dialog.style.width = `${newWidth}px`;
          dialog.style.height = `${newHeight}px`;
          const widthDiff = newWidth - currentWidth;
          const newX = isRTL ? base.x - widthDiff : base.x;
          setPosition(clampPosition(newX, base.y));
        }
      }
    };
    dialog.addEventListener("keydown", handleKeyDown);
    return () => {
      dialog.removeEventListener("keydown", handleKeyDown);
    };
  }, [isPopoverActuallyOpen, draggable, resizable, clampPosition, ensureDirectPositioning]);
  React11__namespace.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog || !isPopoverActuallyOpen) return;
    const handleFocusOut = (e) => {
      const newFocusTarget = e.relatedTarget;
      if (!newFocusTarget || !dialog.contains(newFocusTarget)) {
        requestAnimationFrame(() => {
          const currentFocus = document.activeElement;
          if (!dialog.contains(currentFocus)) {
            dialog.focus();
          }
        });
      }
    };
    dialog.addEventListener("focusout", handleFocusOut);
    return () => {
      dialog.removeEventListener("focusout", handleFocusOut);
    };
  }, [isPopoverActuallyOpen]);
  const handleMouseDown = React11__namespace.useCallback(
    (e) => {
      if (!draggable) return;
      if (stretch && onPhone) return;
      if (!headerRef.current?.contains(e.target)) return;
      const pos = position ?? ensureDirectPositioning();
      setIsDragging(true);
      dragStartRef.current = {
        x: e.clientX,
        y: e.clientY,
        dialogX: pos?.x ?? 0,
        dialogY: pos?.y ?? 0
      };
    },
    [draggable, stretch, onPhone, position, ensureDirectPositioning]
  );
  const handleMouseMove = React11__namespace.useCallback(
    (e) => {
      if (!isDragging) return;
      const deltaX = e.clientX - dragStartRef.current.x;
      const deltaY = e.clientY - dragStartRef.current.y;
      const rawX = dragStartRef.current.dialogX + deltaX;
      const rawY = dragStartRef.current.dialogY + deltaY;
      setPosition(clampPosition(rawX, rawY));
    },
    [isDragging, clampPosition]
  );
  const handleMouseUp = React11__namespace.useCallback(() => {
    setIsDragging(false);
  }, []);
  const resizeState = React11__namespace.useRef(null);
  const handleResizeMouseDown = React11__namespace.useCallback(
    (e) => {
      if (!resizable) return;
      if (stretch && onPhone) return;
      e.preventDefault();
      e.stopPropagation();
      const dialog = dialogRef.current;
      if (!dialog) return;
      const pos = positionRef.current ?? ensureDirectPositioning();
      const rect = dialog.getBoundingClientRect();
      const rtl = document.documentElement.dir === "rtl";
      resizeState.current = {
        startX: e.clientX,
        startY: e.clientY,
        startWidth: rect.width,
        startHeight: rect.height,
        startLeft: pos?.x ?? rect.left,
        isRTL: rtl
      };
      const onMove = (ev) => {
        if (!resizeState.current || !dialogRef.current) return;
        const { startX, startY, startWidth, startHeight, startLeft, isRTL } = resizeState.current;
        const dx = ev.clientX - startX;
        const dy = ev.clientY - startY;
        let newWidth;
        let newLeft;
        if (isRTL) {
          newWidth = Math.max(200, Math.min(startWidth - dx, window.innerWidth * 0.95));
          newLeft = startLeft + dx;
          if (newLeft < 0) {
            newWidth = startWidth + startLeft;
            newLeft = 0;
          }
        } else {
          newWidth = Math.max(200, Math.min(startWidth + dx, window.innerWidth * 0.95));
        }
        const newHeight = Math.max(150, Math.min(startHeight + dy, window.innerHeight * 0.95));
        dialogRef.current.style.width = `${newWidth}px`;
        dialogRef.current.style.height = `${newHeight}px`;
        if (isRTL && newLeft !== void 0) {
          setPosition(clampPosition(newLeft, positionRef.current?.y ?? startLeft));
        }
      };
      const onUp = () => {
        resizeState.current = null;
        window.removeEventListener("mousemove", onMove);
        window.removeEventListener("mouseup", onUp);
      };
      window.addEventListener("mousemove", onMove);
      window.addEventListener("mouseup", onUp);
    },
    [resizable, stretch, onPhone, ensureDirectPositioning, clampPosition]
  );
  React11__namespace.useEffect(() => {
    if (isDragging) {
      document.addEventListener("mousemove", handleMouseMove);
      document.addEventListener("mouseup", handleMouseUp);
      return () => {
        document.removeEventListener("mousemove", handleMouseMove);
        document.removeEventListener("mouseup", handleMouseUp);
      };
    }
  }, [isDragging, handleMouseMove, handleMouseUp]);
  React11__namespace.useImperativeHandle(
    ref,
    () => ({
      show: () => {
        if (controlledOpen === void 0) {
          setInternalOpen(true);
        }
        onOpenChange?.(true);
      },
      showModal: () => {
        if (controlledOpen === void 0) {
          setInternalOpen(true);
        }
        onOpenChange?.(true);
      },
      close: (returnValue) => {
        returnValueRef.current = returnValue;
        handleClose(false);
      },
      focus: () => {
        dialogRef.current?.focus();
      }
    }),
    [controlledOpen, onOpenChange, handleClose]
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "dialog",
    {
      ref: dialogRef,
      id: stableId,
      popover: "manual",
      role: effectiveRole,
      "aria-labelledby": accessibleNameRef || (!accessibleName && headerText ? `${stableId}-title` : void 0),
      "aria-describedby": effectiveAriaDescribedBy,
      "aria-label": accessibleName || (!headerText && !accessibleNameRef ? "Dialog" : void 0),
      "aria-roledescription": ariaRoleDescription,
      tabIndex: -1,
      className: cn(
        dialogVariants({ state, stretch, resizable }),
        stretch && onPhone && "w-screen max-w-none h-[100dvh] max-h-none rounded-none min-w-0",
        className
      ),
      style: {
        ...stretch && onPhone ? {
          left: 0,
          top: 0,
          right: "auto",
          transform: "none"
        } : position ? {
          left: `${position.x}px`,
          top: `${position.y}px`,
          right: "auto",
          transform: "none"
        } : {
          left: "50%",
          top: "50%",
          right: "auto",
          transform: "translate(-50%, -50%)"
        }
      },
      onClick: (e) => e.stopPropagation(),
      onMouseDown: handleMouseDown,
      children: [
        (draggable || resizable) && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            id: `${stableId}-movement-descr`,
            className: "absolute w-px h-px p-0 -m-px overflow-hidden whitespace-nowrap border-0",
            style: { clip: "rect(0, 0, 0, 0)" },
            children: getMovementDescription()
          }
        ),
        (header || headerText) && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            ref: headerRef,
            className: cn(
              headerVariants({ state, draggable }),
              !hideHeaderBorder && state !== "Error" /* Error */ && state !== "Success" /* Success */ && state !== "Information" /* Information */ && state !== "Warning" /* Warning */ && "border-b"
            ),
            tabIndex: draggable || resizable ? 0 : void 0,
            "aria-roledescription": draggable && ariaRoleDescription ? ariaRoleDescription : void 0,
            children: header || /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center gap-3 w-full", children: [
              /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center gap-3 min-w-0", children: [
                state === "Error" && /* @__PURE__ */ jsxRuntime.jsx(MessageErrorIcon, { className: "h-5 w-5 text-sapphire-negative shrink-0" }),
                state === "Success" && /* @__PURE__ */ jsxRuntime.jsx(MessageSuccessIcon, { className: "h-5 w-5 text-sapphire-positive shrink-0" }),
                state === "Warning" && /* @__PURE__ */ jsxRuntime.jsx(MessageWarningIcon, { className: "h-5 w-5 text-sapphire-warning shrink-0" }),
                state === "Information" && /* @__PURE__ */ jsxRuntime.jsx(MessageInformationIcon, { className: "h-5 w-5 text-sapphire-info shrink-0" }),
                showBackButton && state === "None" && /* @__PURE__ */ jsxRuntime.jsx(
                  Button,
                  {
                    design: "SecondaryNeutral",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsxRuntime.jsx(ArrowLeftIcon, {}),
                    onClick: onBackButtonClick
                  }
                ),
                /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex flex-col min-w-0", children: [
                  /* @__PURE__ */ jsxRuntime.jsx(
                    Title,
                    {
                      level: "H1",
                      id: `${stableId}-title`,
                      className: "text-xl font-semibold text-foreground truncate",
                      children: headerText
                    }
                  ),
                  subHeaderText && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "text-sm font-normal text-sapphire-text-tertiary truncate", children: subHeaderText })
                ] })
              ] }),
              headerEndContent && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center gap-2 shrink-0 max-w-[50%] ml-auto", children: headerEndContent })
            ] })
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            ref: contentRef,
            className: cn("flex-auto min-h-0 overflow-y-auto", !noPadding && "px-6 py-4"),
            tabIndex: -1,
            children
          }
        ),
        footer && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center justify-end gap-2 px-6 py-5 shrink-0", children: footer }),
        resizable && !onPhone && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: "dialog-resize-handle absolute bottom-0 w-6 h-6 rounded-full flex items-center justify-center z-[1] end-0",
            onMouseDown: handleResizeMouseDown,
            "data-resize-handle": "true",
            children: /* @__PURE__ */ jsxRuntime.jsx(
              ResizeCornerIcon,
              {
                className: "h-4 w-4 text-sapphire-text-tertiary rtl:rotate-90"
              }
            )
          }
        )
      ]
    }
  );
}
function useIsPhone2() {
  const [phone] = React11.useState(() => isPhone());
  return phone;
}
function ResponsivePopover({
  headerText,
  placement,
  horizontalAlign,
  verticalAlign,
  hideArrow,
  offset,
  allowTargetOverlap,
  opener,
  open = false,
  initialFocus,
  preventFocusRestore,
  preventInitialFocus,
  accessibleName,
  accessibleNameRef,
  accessibleRole,
  accessibleDescription,
  header,
  footer,
  children,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  onClose,
  contentOnlyOnDesktop = false,
  noPadding = false,
  showCloseButton = true,
  className,
  style,
  id,
  ref
}) {
  const isPhone2 = useIsPhone2();
  const popoverRef = React11.useRef(null);
  const dialogRef = React11.useRef(null);
  const handleMobileClose = React11.useCallback(() => {
    onBeforeClose?.({ escPressed: false });
    onClose?.();
  }, [onBeforeClose, onClose]);
  React11.useImperativeHandle(
    ref,
    () => ({
      open() {
        if (isPhone2) {
          dialogRef.current?.show();
        } else {
          popoverRef.current?.open();
        }
      },
      close() {
        if (isPhone2) {
          dialogRef.current?.close();
        } else {
          popoverRef.current?.close();
        }
      },
      async applyFocus() {
        if (isPhone2) {
          dialogRef.current?.focus();
        } else {
          await popoverRef.current?.applyFocus();
        }
      },
      isOpen() {
        if (isPhone2) {
          return open;
        }
        return popoverRef.current?.isOpen() ?? false;
      },
      get nativeElement() {
        if (isPhone2) {
          return null;
        }
        return popoverRef.current?.nativeElement ?? null;
      }
    }),
    [isPhone2, open]
  );
  if (isPhone2) {
    const titleId = `${id || "responsive-popover"}-title`;
    const mobileHeader = header || headerText || showCloseButton ? /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center justify-between w-full", children: [
      header || headerText && /* @__PURE__ */ jsxRuntime.jsx(Title, { level: "H1", id: titleId, className: "text-base font-semibold", children: headerText }),
      showCloseButton && /* @__PURE__ */ jsxRuntime.jsx(
        Button,
        {
          design: "Tertiary",
          iconOnly: true,
          accessibleName: "Close",
          onClick: handleMobileClose,
          className: "ml-auto",
          icon: /* @__PURE__ */ jsxRuntime.jsxs(
            "svg",
            {
              xmlns: "http://www.w3.org/2000/svg",
              width: "16",
              height: "16",
              viewBox: "0 0 24 24",
              fill: "none",
              stroke: "currentColor",
              strokeWidth: "2",
              strokeLinecap: "round",
              strokeLinejoin: "round",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("line", { x1: "18", y1: "6", x2: "6", y2: "18" }),
                /* @__PURE__ */ jsxRuntime.jsx("line", { x1: "6", y1: "6", x2: "18", y2: "18" })
              ]
            }
          )
        }
      )
    ] }) : void 0;
    return /* @__PURE__ */ jsxRuntime.jsx(
      Dialog,
      {
        ref: dialogRef,
        open,
        onOpenChange: (isOpen) => {
          if (!isOpen) onClose?.();
        },
        headerText: !header && !showCloseButton ? headerText : void 0,
        header: mobileHeader,
        footer,
        stretch: true,
        noPadding,
        initialFocus,
        preventInitialFocus,
        accessibleName,
        accessibleNameRef: accessibleNameRef || (!accessibleName && headerText && !header && showCloseButton ? titleId : void 0),
        accessibleRole: accessibleRole === "AlertDialog" ? "alertdialog" : accessibleRole === "None" ? void 0 : "dialog",
        onBeforeOpen: () => onBeforeOpen?.(),
        onAfterOpen: () => onOpen?.(),
        onBeforeClose: (detail) => onBeforeClose?.({ escPressed: !!detail.escPressed }),
        onAfterClose: () => onClose?.(),
        className,
        id,
        children
      }
    );
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    Popover,
    {
      ref: popoverRef,
      open,
      opener,
      headerText: contentOnlyOnDesktop ? void 0 : headerText,
      header: contentOnlyOnDesktop ? void 0 : header,
      footer: contentOnlyOnDesktop ? void 0 : footer,
      placement,
      horizontalAlign,
      verticalAlign,
      hideArrow,
      offset,
      allowTargetOverlap,
      noPadding,
      initialFocus,
      preventFocusRestore,
      preventInitialFocus,
      accessibleName,
      accessibleNameRef,
      accessibleRole,
      accessibleDescription,
      onBeforeOpen,
      onOpen,
      onBeforeClose,
      onClose,
      className,
      style,
      id,
      children
    }
  );
}
var listContainerVariants = cva(
  "relative w-full",
  {
    variants: {
      bordered: {
        true: "border border-sapphire-border-primary rounded-lg overflow-hidden",
        false: ""
      }
    },
    defaultVariants: {
      bordered: false
    }
  }
);
var listVariants = cva(
  "w-full m-0 p-0 list-none",
  {
    variants: {
      indent: {
        true: "pl-4",
        false: ""
      },
      separators: {
        true: "[&>li:not(:last-child)]:border-b [&>li:not(:last-child)]:border-sapphire-border-primary",
        false: ""
      }
    },
    defaultVariants: {
      indent: false,
      separators: true
    }
  }
);
var ListContext = React11__namespace.createContext(null);
function useListSelection({
  mode,
  controlledKeys,
  defaultKeys = [],
  items,
  getItemKey,
  onSelectionChange
}) {
  const [internalKeys, setInternalKeys] = React11.useState(new Set(defaultKeys));
  const previousKeysRef = React11.useRef(new Set(defaultKeys));
  const isControlled = controlledKeys !== void 0;
  const selectedKeys = isControlled ? new Set(controlledKeys) : internalKeys;
  const getItemByKey = React11.useCallback((key) => {
    if (!items || !getItemKey) return void 0;
    return items.find((item, index) => getItemKey(item, index) === key);
  }, [items, getItemKey]);
  const getSelectedItems = React11.useCallback(() => {
    if (!items || !getItemKey) return [];
    return items.filter((item, index) => selectedKeys.has(getItemKey(item, index)));
  }, [items, getItemKey, selectedKeys]);
  const setSelected = React11.useCallback((key, selected) => {
    const modeStr = String(mode);
    if (modeStr === "None") return;
    const newKeys = new Set(selectedKeys);
    const previousKeys = new Set(selectedKeys);
    if (modeStr === "Single" || modeStr === "SingleStart" || modeStr === "SingleEnd") {
      if (selected) {
        newKeys.clear();
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    } else if (modeStr === "Multiple") {
      if (selected) {
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    }
    if (!isControlled) {
      setInternalKeys(newKeys);
    }
    const targetItem = getItemByKey(key);
    onSelectionChange?.({
      selectedItems: items?.filter((item, index) => getItemKey && newKeys.has(getItemKey(item, index))) ?? [],
      previouslySelectedItems: items?.filter((item, index) => getItemKey && previousKeys.has(getItemKey(item, index))) ?? [],
      selectionComponentPressed: false,
      targetItem,
      key,
      selectedKeys: Array.from(newKeys)
      // Always include the raw keys
    });
    previousKeysRef.current = newKeys;
  }, [mode, selectedKeys, isControlled, getItemByKey, items, getItemKey, onSelectionChange]);
  const toggleSelection = React11.useCallback((key) => {
    setSelected(key, !selectedKeys.has(key));
  }, [selectedKeys, setSelected]);
  const isSelected = React11.useCallback((key) => {
    return selectedKeys.has(key);
  }, [selectedKeys]);
  const selectAll = React11.useCallback(() => {
    if (!items || !getItemKey || String(mode) !== "Multiple") return;
    const allKeys = new Set(items.map(getItemKey));
    if (!isControlled) {
      setInternalKeys(allKeys);
    }
    onSelectionChange?.({
      selectedItems: items,
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: items[0]
    });
  }, [items, getItemKey, mode, isControlled, onSelectionChange, getSelectedItems]);
  const deselectAll = React11.useCallback(() => {
    if (!isControlled) {
      setInternalKeys(/* @__PURE__ */ new Set());
    }
    onSelectionChange?.({
      selectedItems: [],
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: void 0
    });
  }, [isControlled, onSelectionChange, getSelectedItems]);
  return {
    selectedKeys,
    setSelected,
    toggleSelection,
    isSelected,
    selectAll,
    deselectAll,
    getSelectedItems
  };
}
function useListNavigation({
  onFocusItem,
  onSelectItem,
  selectionMode,
  containerRef
}) {
  const [focusedIndex, setFocusedIndex] = React11.useState(-1);
  const getAllNavigable = React11.useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"], button[data-growing-button]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const getAllNavigableExcludingGrowingButton = React11.useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const focusItem = React11.useCallback((index) => {
    const allNavigable = getAllNavigable();
    if (index < 0 || index >= allNavigable.length) return;
    const element = allNavigable[index];
    if (!element) return;
    const isGrowingButton = element.hasAttribute("data-growing-button");
    if (!isGrowingButton) {
      setFocusedIndex(index);
    }
    element.focus();
    onFocusItem?.(index);
  }, [getAllNavigable, onFocusItem]);
  const handleKeyDown = React11.useCallback((e) => {
    const key = e.key;
    if (key === "Tab") {
      return;
    }
    const target = e.target;
    const isInteractiveElement = target.tagName === "INPUT" || target.tagName === "TEXTAREA" || target.tagName === "SELECT" || target.tagName === "BUTTON" || target.getAttribute("role") === "button" || target.getAttribute("contenteditable") === "true" || target.closest('button, input, textarea, select, [role="button"], [contenteditable="true"]') !== null;
    if (isInteractiveElement) {
      if (key === " " || key === "Enter" || key === "Delete") {
        return;
      }
    }
    const allNavigable = getAllNavigable();
    if (allNavigable.length === 0) return;
    const activeElement = document.activeElement;
    let currentIndex = allNavigable.indexOf(activeElement);
    if (currentIndex === -1) {
      currentIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    }
    let newIndex = currentIndex;
    switch (key) {
      case "ArrowDown":
        e.preventDefault();
        newIndex = Math.min(currentIndex + 1, allNavigable.length - 1);
        break;
      case "ArrowUp":
        e.preventDefault();
        newIndex = Math.max(currentIndex - 1, 0);
        break;
      case "Home":
        e.preventDefault();
        const navigableExcludingButton = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButton.length > 0) {
          const firstElement = navigableExcludingButton[0];
          const indexInAll = allNavigable.indexOf(firstElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "End":
        e.preventDefault();
        const navigableExcludingButtonEnd = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButtonEnd.length > 0) {
          const lastElement = navigableExcludingButtonEnd[navigableExcludingButtonEnd.length - 1];
          const indexInAll = allNavigable.indexOf(lastElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "PageDown":
        e.preventDefault();
        const navigableExcludingButtonPageDown = getAllNavigableExcludingGrowingButton();
        const currentInFiltered = navigableExcludingButtonPageDown.indexOf(activeElement);
        if (currentInFiltered !== -1) {
          const newFilteredIndex = Math.min(currentInFiltered + 10, navigableExcludingButtonPageDown.length - 1);
          const targetElement = navigableExcludingButtonPageDown[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageDown.length > 0) {
            const lastElement = navigableExcludingButtonPageDown[navigableExcludingButtonPageDown.length - 1];
            newIndex = allNavigable.indexOf(lastElement);
          }
        }
        break;
      case "PageUp":
        e.preventDefault();
        const navigableExcludingButtonPageUp = getAllNavigableExcludingGrowingButton();
        const currentInFilteredPageUp = navigableExcludingButtonPageUp.indexOf(activeElement);
        if (currentInFilteredPageUp !== -1) {
          const newFilteredIndex = Math.max(currentInFilteredPageUp - 10, 0);
          const targetElement = navigableExcludingButtonPageUp[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageUp.length > 0) {
            const lastFilteredIndex = navigableExcludingButtonPageUp.length - 1;
            const targetIndex = Math.max(lastFilteredIndex - 10, 0);
            const targetElement = navigableExcludingButtonPageUp[targetIndex];
            newIndex = allNavigable.indexOf(targetElement);
          }
        }
        break;
      case " ":
        e.preventDefault();
        const spaceElement = allNavigable[currentIndex];
        if (spaceElement?.getAttribute("role") === "listitem") {
          if (String(selectionMode) !== "None" && String(selectionMode) !== "Delete") {
            onSelectItem?.(currentIndex);
          }
        }
        return;
      case "Delete":
        const deleteElement = allNavigable[currentIndex];
        if (deleteElement?.getAttribute("role") === "listitem" && String(selectionMode) === "Delete") {
          onSelectItem?.(currentIndex);
        }
        return;
      case "Enter":
        return;
      default:
        return;
    }
    if (newIndex !== currentIndex) {
      focusItem(newIndex);
    }
  }, [focusedIndex, getAllNavigable, getAllNavigableExcludingGrowingButton, selectionMode, onSelectItem, focusItem]);
  const registerItem = React11.useCallback((_index, _element) => {
  }, []);
  return {
    focusedIndex,
    setFocusedIndex,
    focusItem,
    handleKeyDown,
    registerItem
  };
}
function ListHeader({ text, children, sticky, className }) {
  if (!text && !children) return null;
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "flex items-center gap-2.5 px-[0.75rem] py-[0.25rem] text-base font-normal text-sapphire-text-primary bg-background",
        sticky && "sticky top-0 z-10 bg-background",
        className
      ),
      role: "heading",
      "aria-level": 2,
      children: children || text
    }
  );
}
function ListLoading({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20",
        className
      ),
      role: "progressbar",
      "aria-label": "Loading",
      children: /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-6 w-6 animate-spin text-sapphire-brand-foreground" })
    }
  );
}
function ListEmpty({ text, children, className }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "flex items-center justify-center px-4 py-12 text-sm text-sapphire-text-tertiary",
        className
      ),
      role: "status",
      children: children || text || "No data"
    }
  );
}
function ListGrowing({
  mode,
  buttonText = "More",
  onLoadMore,
  parentRef,
  loading,
  accessibilityAttributes,
  buttonRef
}) {
  const observerRef = React11.useRef(null);
  const triggerRef = React11.useRef(null);
  React11.useEffect(() => {
    if (String(mode) !== "Scroll" || !triggerRef.current) return;
    observerRef.current = new IntersectionObserver(
      (entries) => {
        const [entry] = entries;
        if (entry.isIntersecting && !loading) {
          onLoadMore?.();
        }
      },
      {
        root: parentRef?.current,
        rootMargin: "100px",
        threshold: 0
      }
    );
    observerRef.current.observe(triggerRef.current);
    return () => {
      observerRef.current?.disconnect();
    };
  }, [mode, onLoadMore, parentRef, loading]);
  if (String(mode) === "Button") {
    return /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center justify-center py-3 border-t border-sapphire-border-primary", children: /* @__PURE__ */ jsxRuntime.jsxs(
      Button,
      {
        ref: buttonRef,
        design: "Tertiary" /* Tertiary */,
        onClick: () => onLoadMore?.(),
        disabled: loading,
        accessibleName: accessibilityAttributes?.name,
        "data-growing-button": "true",
        tabIndex: 0,
        className: "w-full mx-4",
        children: [
          loading ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-4 w-4 animate-spin mr-2" }) : null,
          buttonText
        ]
      }
    ) });
  }
  if (String(mode) === "Scroll") {
    return /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        ref: triggerRef,
        className: "h-1",
        "aria-hidden": "true",
        "data-growing-trigger": true
      }
    );
  }
  return null;
}
function ListInner(props) {
  const {
    ref,
    // Header & Footer
    headerText,
    header,
    // Selection
    selectionMode = "None" /* None */,
    selectedKeys: controlledSelectedKeys,
    defaultSelectedKeys = [],
    onSelectionChange,
    // Data
    items,
    renderItem,
    getItemKey = (item, index) => String(item.id ?? item.key ?? index),
    children,
    // Empty & Loading
    noDataText = "No data available",
    noDataSlot,
    loading = false,
    loadingDelay = 1e3,
    // Growing
    growing = "None" /* None */,
    growingButtonText,
    hasMore = false,
    onLoadMore,
    // Visual
    separators = "All" /* All */,
    indent = false,
    stickyHeader = false,
    stickyFooter = false,
    // Events
    onItemClick,
    onItemDelete,
    onItemFocused,
    // Drag and Drop
    onMoveOver,
    onMove,
    // Accessibility
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    accessibleRole = "List" /* List */,
    accessibilityAttributes,
    // Standard
    className,
    style,
    id: providedId,
    "data-testid": dataTestId,
    "data-ai-context": dataAiContext
  } = props;
  const generatedId = React11.useId();
  const listId = providedId ?? generatedId;
  const containerRef = React11.useRef(null);
  const listRef = React11.useRef(null);
  const growingButtonRef = React11.useRef(null);
  const beforeSentinelRef = React11.useRef(null);
  const afterSentinelRef = React11.useRef(null);
  const [showLoading, setShowLoading] = React11.useState(false);
  const [mediaRange, setMediaRange] = React11.useState("M");
  const itemRegistry = React11.useRef(/* @__PURE__ */ new Map());
  const [previouslyFocusedItemKey, setPreviouslyFocusedItemKey] = React11.useState(null);
  const [draggedKey, setDraggedKey] = React11.useState(null);
  const [dropTargetKey, setDropTargetKey] = React11.useState(null);
  const [dropPlacement, setDropPlacement] = React11.useState(null);
  const selectionState = useListSelection({
    mode: selectionMode,
    controlledKeys: controlledSelectedKeys,
    defaultKeys: defaultSelectedKeys,
    items,
    getItemKey,
    onSelectionChange
  });
  const itemCount = React11.useMemo(() => {
    if (items) return items.length;
    return React11__namespace.Children.count(children);
  }, [items, children]);
  const navigationState = useListNavigation({
    onFocusItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (element?.getAttribute("role") === "listitem" && items) {
        const itemKey = element.getAttribute("data-item-key");
        if (itemKey) {
          const itemIndex = items.findIndex((item, idx) => getItemKey(item, idx) === itemKey);
          if (itemIndex >= 0) {
            onItemFocused?.({ item: items[itemIndex], index: itemIndex });
          }
        }
      }
    },
    onSelectItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (!element || element.getAttribute("role") !== "listitem") return;
      if (!items || !getItemKey) return;
      const itemKey = element.getAttribute("data-item-key");
      if (!itemKey) return;
      const itemIndex = items.findIndex((item2, idx) => getItemKey(item2, idx) === itemKey);
      if (itemIndex < 0) return;
      const item = items[itemIndex];
      const key = getItemKey(item, itemIndex);
      if (String(selectionMode) === "Delete") {
        onItemDelete?.({ item });
      } else {
        selectionState.toggleSelection(key);
      }
    },
    selectionMode,
    containerRef
  });
  React11.useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  React11.useEffect(() => {
    const updateMediaRange = () => {
      const width = window.innerWidth;
      if (width < 600) setMediaRange("S");
      else if (width < 1024) setMediaRange("M");
      else if (width < 1440) setMediaRange("L");
      else setMediaRange("XL");
    };
    updateMediaRange();
    window.addEventListener("resize", updateMediaRange);
    return () => window.removeEventListener("resize", updateMediaRange);
  }, []);
  const registerItem = React11.useCallback((key, index) => {
    itemRegistry.current.set(key, index);
  }, []);
  const unregisterItem = React11.useCallback((key) => {
    itemRegistry.current.delete(key);
  }, []);
  const handleDragStart = React11.useCallback((key, _e) => {
    setDraggedKey(key);
  }, []);
  const handleDragEnd = React11.useCallback((_key, _e) => {
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, []);
  const handleDragOver = React11.useCallback((key, e) => {
    if (draggedKey && key === draggedKey) return;
    setDropTargetKey(key);
    const rect = e.currentTarget.getBoundingClientRect();
    const y = e.clientY - rect.top;
    const height = rect.height;
    if (y < height * 0.33) {
      setDropPlacement("Before");
    } else if (y > height * 0.67) {
      setDropPlacement("After");
    } else {
      setDropPlacement("On");
    }
  }, [draggedKey]);
  const handleDrop = React11.useCallback((targetKey, e) => {
    const targetIndex = itemRegistry.current.get(targetKey);
    if (targetIndex === void 0) return;
    if (!items || !getItemKey) return;
    const targetItem = items[targetIndex];
    if (!targetItem) return;
    const droppedKey = draggedKey || e.dataTransfer.getData("text/plain");
    if (!droppedKey || targetKey === droppedKey) return;
    const placement = dropPlacement || "After";
    const sourceIndex = itemRegistry.current.get(droppedKey);
    const sourceItem = sourceIndex !== void 0 ? items[sourceIndex] : void 0;
    const isMultiple = String(selectionMode) === "Multiple";
    const draggedIsSelected = droppedKey ? selectionState.isSelected(droppedKey) : false;
    const selectedItemsList = isMultiple && draggedIsSelected ? selectionState.getSelectedItems() : void 0;
    const moveDetail = {
      source: {
        element: sourceItem,
        // May be undefined for cross-list drops
        elements: selectedItemsList,
        index: sourceIndex ?? -1
        // -1 indicates cross-list drop
      },
      destination: {
        element: targetItem,
        placement,
        index: targetIndex
      },
      originalEvent: e.nativeEvent
    };
    onMoveOver?.(moveDetail);
    onMove?.(moveDetail);
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, [draggedKey, dropPlacement, items, getItemKey, selectionMode, selectionState, onMoveOver, onMove]);
  const handleForwardTabAfter = React11.useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      const itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      );
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (String(growing) !== "None" && hasMore && growingButtonRef.current) {
      growingButtonRef.current.focus?.();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex + 1; i < allTabbable.length; i++) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (afterSentinelRef.current) {
          afterSentinelRef.current.focus();
        }
      }
    }
  }, [growing, hasMore, navigationState]);
  const handleForwardTabBefore = React11.useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    let itemElement = null;
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      ) || null;
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (itemElement && event.target !== itemElement) {
      itemElement.focus();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex - 1; i >= 0; i--) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (beforeSentinelRef.current) {
          beforeSentinelRef.current.focus();
        }
      }
    }
  }, [navigationState]);
  const handleBeforeSentinelFocus = React11.useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromAfterSentinel = relatedTarget === afterSentinelRef.current;
    const isFromListItem = relatedTarget?.closest('li[role="listitem"], li[role="option"]');
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    if (isFromAfterSentinel || isFromListItem || isFromGrowingButton) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const sentinelIndex = allTabbable.indexOf(beforeSentinelRef.current);
      if (sentinelIndex > 0) {
        allTabbable[sentinelIndex - 1].focus();
      }
    } else {
      e.preventDefault();
      if (containerRef.current) {
        const allNavigable = Array.from(
          containerRef.current.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
          )
        );
        if (allNavigable.length === 0) {
          if (afterSentinelRef.current) {
            afterSentinelRef.current.focus();
          }
          return;
        }
        if (previouslyFocusedItemKey) {
          const targetElement = allNavigable.find(
            (el) => el.getAttribute("data-item-key") === previouslyFocusedItemKey || el.id === previouslyFocusedItemKey
          );
          if (targetElement) {
            targetElement.focus();
            return;
          }
        }
        if (allNavigable[0]) {
          allNavigable[0].focus();
        }
      }
    }
  }, [previouslyFocusedItemKey]);
  const getAllTabbableElements = React11.useCallback(() => {
    return Array.from(
      document.querySelectorAll(
        'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
      )
    );
  }, []);
  const focusNextAfterList = React11.useCallback(() => {
    if (!listRef.current || !afterSentinelRef.current) return false;
    const allTabbable = getAllTabbableElements();
    const sentinelIndex = allTabbable.indexOf(afterSentinelRef.current);
    if (sentinelIndex >= 0) {
      for (let i = sentinelIndex + 1; i < allTabbable.length; i++) {
        if (!containerRef.current?.contains(allTabbable[i])) {
          allTabbable[i].focus();
          return true;
        }
      }
    }
    return false;
  }, [getAllTabbableElements]);
  const enterListAtFocusedItem = React11.useCallback(() => {
    if (!containerRef.current) return false;
    const allNavigable = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
      )
    );
    if (allNavigable.length === 0) {
      beforeSentinelRef.current?.focus();
      return true;
    }
    const focusedIndex = navigationState.focusedIndex;
    const targetIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    if (allNavigable[targetIndex]) {
      allNavigable[targetIndex].focus();
      return true;
    }
    return false;
  }, [navigationState.focusedIndex]);
  const handleAfterSentinelFocus = React11.useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    const isFromOutside = !relatedTarget || !containerRef.current?.contains(relatedTarget);
    if (isFromGrowingButton || isFromOutside) {
      enterListAtFocusedItem();
    } else {
      focusNextAfterList();
    }
  }, [focusNextAfterList, enterListAtFocusedItem]);
  const roleSupportsAriaSelected = React11.useMemo(() => {
    const roleStr = String(accessibleRole);
    return roleStr === "ListBox" /* ListBox */ || roleStr === "listbox" || roleStr === "Tree" /* Tree */ || roleStr === "tree";
  }, [accessibleRole]);
  const contextValue = React11.useMemo(() => ({
    selectionMode,
    selectedKeys: selectionState.selectedKeys,
    onSelect: selectionState.setSelected,
    onToggleSelect: selectionState.toggleSelection,
    isSelected: selectionState.isSelected,
    focusedIndex: navigationState.focusedIndex,
    onFocusItem: navigationState.setFocusedIndex,
    registerItem,
    unregisterItem,
    registerNavigationItem: navigationState.registerItem,
    // Tab navigation
    onForwardTabAfter: handleForwardTabAfter,
    onForwardTabBefore: handleForwardTabBefore,
    separators,
    indent,
    onItemClick: (_key, event) => {
      onItemClick?.({ item: event.currentTarget, originalEvent: event });
    },
    onItemDelete: (key) => {
      const index = itemRegistry.current.get(key);
      if (index !== void 0 && items) {
        onItemDelete?.({ item: items[index] });
        return;
      }
      if (items && getItemKey) {
        const item = items.find((item2, idx) => getItemKey(item2, idx) === key);
        if (item) {
          onItemDelete?.({ item });
        }
      }
    },
    // Drag and drop
    onDragStart: handleDragStart,
    onDragEnd: handleDragEnd,
    onDragOver: handleDragOver,
    onDrop: handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    mediaRange,
    totalItemCount: itemCount,
    getItemIndex: (key) => {
      return itemRegistry.current.get(key) ?? -1;
    },
    // Accessibility
    accessibleRole,
    supportsAriaSelected: roleSupportsAriaSelected
  }), [
    selectionMode,
    selectionState,
    navigationState,
    registerItem,
    unregisterItem,
    handleForwardTabAfter,
    handleForwardTabBefore,
    separators,
    indent,
    onItemClick,
    onItemDelete,
    items,
    getItemKey,
    mediaRange,
    itemCount,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    accessibleRole,
    roleSupportsAriaSelected
  ]);
  React11.useImperativeHandle(ref, () => ({
    // Focus
    focus: () => listRef.current?.focus(),
    focusItem: (index) => navigationState.focusItem(index),
    getFocusedItemIndex: () => navigationState.focusedIndex,
    // Selection
    getSelectedKeys: () => Array.from(selectionState.selectedKeys),
    getSelectedItems: () => selectionState.getSelectedItems(),
    isSelected: selectionState.isSelected,
    select: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, true));
    },
    deselect: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, false));
    },
    toggleSelection: selectionState.toggleSelection,
    selectAll: selectionState.selectAll,
    deselectAll: selectionState.deselectAll,
    // Data
    getItems: () => items ?? [],
    getItemCount: () => itemCount,
    getItemByKey: (key) => items?.find((item, index) => getItemKey(item, index) === key),
    // Scrolling
    scrollToItem: (index, align = "start") => {
      const element = listRef.current?.children[index];
      element?.scrollIntoView({ behavior: "smooth", block: align });
    },
    scrollToTop: () => {
      containerRef.current?.scrollTo({ top: 0, behavior: "smooth" });
    },
    scrollToBottom: () => {
      containerRef.current?.scrollTo({
        top: containerRef.current.scrollHeight,
        behavior: "smooth"
      });
    },
    // Elements
    get listElement() {
      return listRef.current;
    },
    get scrollContainer() {
      return containerRef.current;
    }
  }), [selectionState, navigationState, items, itemCount, getItemKey]);
  const handleLoadMore = React11.useCallback(() => {
    onLoadMore?.({ currentCount: itemCount });
  }, [onLoadMore, itemCount]);
  const getRole = () => {
    const roleStr = String(accessibleRole);
    switch (roleStr) {
      case "Menu":
        return "menu";
      case "Tree":
        return "tree";
      case "ListBox":
        return "listbox";
      default:
        return "list";
    }
  };
  const isEmpty = itemCount === 0 && !loading;
  const renderedItems = React11.useMemo(() => {
    itemRegistry.current.clear();
    if (items && renderItem) {
      return items.map((item, index) => {
        const key = getItemKey(item, index);
        itemRegistry.current.set(key, index);
        return /* @__PURE__ */ jsxRuntime.jsx(React11__namespace.Fragment, { children: renderItem(item, index) }, key);
      });
    }
    if (children) {
      let index = 0;
      React11__namespace.Children.forEach(children, (child) => {
        if (React11__namespace.isValidElement(child) && child.props && typeof child.props.itemKey === "string") {
          itemRegistry.current.set(child.props.itemKey, index);
          index++;
        }
      });
    }
    return children;
  }, [items, renderItem, getItemKey, children]);
  return /* @__PURE__ */ jsxRuntime.jsx(ListContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: containerRef,
      className: cn(
        listContainerVariants({ bordered: false }),
        // Make scrollable when growing mode or sticky header is used
        (String(growing) !== "None" || stickyHeader || stickyFooter) && "overflow-auto",
        // Always prevent horizontal scroll
        "overflow-x-hidden",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-ai-context": dataAiContext,
      onKeyDown: navigationState.handleKeyDown,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          ListHeader,
          {
            text: headerText,
            sticky: stickyHeader,
            children: header
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            ref: beforeSentinelRef,
            tabIndex: 0,
            onFocus: handleBeforeSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "ul",
          {
            ref: listRef,
            id: listId,
            role: getRole(),
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-describedby": accessibleDescriptionRef,
            "aria-busy": loading || void 0,
            "aria-multiselectable": roleSupportsAriaSelected && String(selectionMode) === "Multiple" ? true : void 0,
            className: cn(listVariants({ indent, separators: String(separators) !== "None" })),
            children: !isEmpty && renderedItems
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            ref: afterSentinelRef,
            tabIndex: 0,
            onFocus: handleAfterSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        isEmpty && /* @__PURE__ */ jsxRuntime.jsx(ListEmpty, { text: noDataText, children: noDataSlot }),
        String(growing) !== "None" && hasMore && !loading && /* @__PURE__ */ jsxRuntime.jsx(
          ListGrowing,
          {
            mode: growing,
            buttonText: growingButtonText,
            onLoadMore: handleLoadMore,
            parentRef: containerRef,
            loading,
            accessibilityAttributes: accessibilityAttributes?.growingButton,
            buttonRef: growingButtonRef
          }
        ),
        showLoading && /* @__PURE__ */ jsxRuntime.jsx(ListLoading, {})
      ]
    }
  ) });
}
var List = React11__namespace.memo(ListInner);
List.displayName = "List";
function ComboBoxPopover({
  open,
  triggerRef,
  hasValueState = false,
  valueStatePortalRef,
  headerContent,
  loading = false,
  mobileTitle = "Suggestions",
  onBeforeClose,
  onClose,
  children,
  className,
  ref
}) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    ResponsivePopover,
    {
      ref,
      open,
      opener: triggerRef,
      placement: "Bottom",
      horizontalAlign: "Stretch",
      hideArrow: true,
      preventInitialFocus: true,
      preventFocusRestore: true,
      noPadding: true,
      headerText: mobileTitle,
      contentOnlyOnDesktop: true,
      onBeforeClose,
      onClose,
      className,
      children: /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex flex-col", children: [
        hasValueState && /* @__PURE__ */ jsxRuntime.jsx("div", { ref: valueStatePortalRef }),
        loading ? /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center justify-center py-8", children: [
          /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-5 w-5 animate-spin text-sapphire-text-tertiary" }),
          /* @__PURE__ */ jsxRuntime.jsx("span", { className: "ml-2 text-sm text-sapphire-text-tertiary", children: "Loading..." })
        ] }) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
          headerContent,
          /* @__PURE__ */ jsxRuntime.jsx(
            List,
            {
              accessibleRole: "ListBox" /* ListBox */,
              separators: "None" /* None */,
              children
            }
          )
        ] })
      ] })
    }
  );
}
var pathData7 = [
  "M256 0q53 0 100 20t81.5 54.5T492 156t20 100-20 100-54.5 81.5T356 492t-100 20-100-20-81.5-54.5T20 356 0 256t20-100 54.5-81.5T156 20 256 0zm150 183q10-9 10-23 0-13-9.5-22.5T384 128t-22 9L186 308l-68-63q-9-9-22-9t-22.5 9.5T64 268q0 15 10 24l91 83q9 9 21 9 13 0 23-9z"
];
var SysEnter2Icon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData7, ...props })
  ),
  { displayName: "SysEnter2Icon", iconName: "sys-enter-2" }
);
var pathData8 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm45 256l74-73q9-11 9-23 0-13-9.5-22.5T352 128q-12 0-23 9l-73 74-73-74q-10-9-23-9t-22.5 9.5T128 160q0 12 9 23l74 73-74 73q-9 10-9 23t9.5 22.5T160 384t23-9l73-74 73 74q11 9 23 9 13 0 22.5-9.5T384 352t-9-23z"
];
var ErrorIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData8, ...props })
  ),
  { displayName: "ErrorIcon", iconName: "error" }
);
var pathData9 = [
  "M505 399q7 13 7 27 0 21-15.5 37.5T456 480H56q-25 0-40.5-16.5T0 426q0-14 7-27L208 59q17-27 48-27 14 0 27 6.5T304 59zM288 176q0-14-9-23t-23-9-23 9-9 23v96q0 14 9 23t23 9 23-9 9-23v-96zm-32 240q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9z"
];
var WarningIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData9, ...props })
  ),
  { displayName: "WarningIcon", iconName: "warning" }
);
var pathData10 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm0 160q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9zm32 64q0-14-9-23t-23-9-23 9-9 23v160q0 14 9 23t23 9 23-9 9-23V224z"
];
var InformationIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData10, ...props })
  ),
  { displayName: "InformationIcon", iconName: "information" }
);

// src/lib/value-state-utils.ts
var valueStateIcons = {
  ["None" /* None */]: null,
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Critical" /* Critical */]: WarningIcon,
  ["Information" /* Information */]: InformationIcon
};
var valueStateMessageColors = {
  ["None" /* None */]: "text-sapphire-text-tertiary",
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Information" /* Information */]: "text-sapphire-info"
};
var valueStateLabelKeys = {
  ["None" /* None */]: "",
  ["Positive" /* Positive */]: "VALUE_STATE_SUCCESS",
  ["Negative" /* Negative */]: "VALUE_STATE_ERROR",
  ["Critical" /* Critical */]: "VALUE_STATE_WARNING",
  ["Information" /* Information */]: "VALUE_STATE_INFORMATION"
};
var valueStateVariants = cva(
  "flex items-center gap-2 px-3 py-2 text-sm border-b",
  {
    variants: {
      state: {
        ["None" /* None */]: "hidden",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info border-sapphire-info"
      }
    },
    defaultVariants: {
      state: "None" /* None */
    }
  }
);
var pathData11 = [
  "M292 256l117 116q7 7 7 18 0 12-7.5 19t-18.5 7q-10 0-18-8L256 292 140 408q-8 8-18 8-11 0-18.5-7.5T96 390q0-10 8-18l116-116-116-116q-8-8-8-18 0-11 7.5-18.5T122 96t18 7l116 117 116-117q7-7 18-7t18.5 7.5T416 122t-7 18z"
];
var DeclineIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData11, ...props })
  ),
  { displayName: "DeclineIcon", iconName: "decline" }
);
var pathData12 = [
  "M410.53 200.75C418.95 210.98 417.48 226.11 407.25 234.53L271.34 346.53C262.48 353.83 249.68 353.83 240.81 346.53L104.75 234.53C94.52 226.11 93.05 210.98 101.47 200.75C109.89 190.52 125.02 189.05 135.25 197.47L256.06 296.91L376.75 197.47C386.98 189.05 402.11 190.52 410.53 200.75Z"
];
var SlimArrowDownIcon = Object.assign(
  React11__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData12, ...props })
  ),
  { displayName: "SlimArrowDownIcon", iconName: "slim-arrow-down" }
);
var valueStateInputVariants = cva(
  [
    "flex items-center w-full border bg-sapphire-background-primary",
    "overflow-clip transition-colors relative"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: "rounded-lg",
        ["Medium" /* Medium */]: "rounded"
      },
      valueState: {
        ["None" /* None */]: [
          "border-sapphire-border-active",
          "hover:border-sapphire-border-accent",
          "focus-within:border-sapphire-border-accent"
        ],
        ["Positive" /* Positive */]: [
          "border-sapphire-positive bg-sapphire-positive-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-positive focus-within:bg-sapphire-background-primary"
        ],
        ["Negative" /* Negative */]: [
          "border-sapphire-negative bg-sapphire-negative-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-negative focus-within:bg-sapphire-background-primary"
        ],
        ["Critical" /* Critical */]: [
          "border-sapphire-warning bg-sapphire-warning-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-warning focus-within:bg-sapphire-background-primary"
        ],
        ["Information" /* Information */]: [
          "border-sapphire-info bg-sapphire-info-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-info focus-within:bg-sapphire-background-primary"
        ]
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      readonly: {
        true: "bg-sapphire-background-secondary border-sapphire-border-primary hover:border-sapphire-border-primary focus-within:!border-sapphire-border-accent",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
function filterItems(text, filterValue, filterMode) {
  if (!filterValue || filterMode === "None" /* None */) {
    return true;
  }
  const normalizedText = text.toLowerCase();
  const normalizedFilter = filterValue.toLowerCase();
  switch (filterMode) {
    case "StartsWith" /* StartsWith */:
      return normalizedText.startsWith(normalizedFilter);
    case "Contains" /* Contains */:
      return normalizedText.includes(normalizedFilter);
    case "StartsWithPerTerm" /* StartsWithPerTerm */:
    default:
      if (normalizedText.startsWith(normalizedFilter)) return true;
      const wordBoundaries = [...normalizedText.matchAll(/\s+/g)];
      return wordBoundaries.some(
        (m) => normalizedText.substring(m.index + m[0].length).startsWith(normalizedFilter)
      );
  }
}
function ValueStateMessage({
  isOpen,
  hasValueState,
  portalTarget,
  messageHeight,
  messageRef,
  messageId,
  normalizedValueState,
  valueStateMessage
}) {
  const { t } = reactI18next.useTranslation("fx");
  const isNegativeState = normalizedValueState === "Negative" /* Negative */;
  const ValueStateIcon = valueStateIcons[normalizedValueState];
  if (isOpen && hasValueState && portalTarget) {
    return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      /* @__PURE__ */ jsxRuntime.jsx("div", { style: { height: messageHeight }, className: "mt-2" }),
      reactDom.createPortal(
        /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            id: messageId,
            className: cn(
              valueStateVariants({ state: normalizedValueState }),
              "border-b-0 items-start gap-1 px-4 py-1.5 shrink-0 rounded-t-lg"
            ),
            children: [
              ValueStateIcon && /* @__PURE__ */ jsxRuntime.jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
              /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("text-sm", valueStateMessageColors[normalizedValueState]), children: valueStateMessage })
            ]
          }
        ),
        portalTarget
      )
    ] });
  }
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: messageRef,
      id: messageId,
      role: isNegativeState ? "alert" : void 0,
      "aria-live": isNegativeState ? "assertive" : "polite",
      className: cn(
        "mt-2 text-sm flex items-start gap-1",
        valueStateMessageColors[normalizedValueState]
      ),
      children: [
        normalizedValueState !== "None" /* None */ && valueStateLabelKeys[normalizedValueState] && /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "sr-only", children: [
          t(valueStateLabelKeys[normalizedValueState]),
          ":",
          " "
        ] }),
        ValueStateIcon && /* @__PURE__ */ jsxRuntime.jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
        /* @__PURE__ */ jsxRuntime.jsx("span", { children: valueStateMessage })
      ]
    }
  );
}
function ClearButton({ onClick, accessibleName = "Clear value", comboBoxSize = "Large" /* Large */ }) {
  const isMedium = comboBoxSize === "Medium" /* Medium */;
  return /* @__PURE__ */ jsxRuntime.jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsxRuntime.jsx(
    Button,
    {
      design: "SecondaryNeutral" /* SecondaryNeutral */,
      size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
      iconOnly: true,
      icon: /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsxRuntime.jsx(DeclineIcon, { className: "h-full w-full" }) }),
      tabIndex: -1,
      onClick,
      accessibleName,
      className: "shrink-0"
    }
  ) });
}
function DropdownArrowButton({ onClick, disabled = false, comboBoxSize = "Large" /* Large */ }) {
  const isMedium = comboBoxSize === "Medium" /* Medium */;
  return /* @__PURE__ */ jsxRuntime.jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsxRuntime.jsx(
    Button,
    {
      design: "SecondaryNeutral" /* SecondaryNeutral */,
      size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
      iconOnly: true,
      icon: /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: "h-full w-full" }) }),
      tabIndex: -1,
      onClick,
      accessibleName: "Open suggestions",
      className: "shrink-0",
      disabled
    }
  ) });
}
function computeVisibleItems(flattenedItems, filterValue, filter, getDisplayText) {
  if (!filterValue || filter === "None" /* None */) {
    return flattenedItems.map((item, index) => ({ ...item, originalIndex: index, isVisible: true }));
  }
  return flattenedItems.map((item, index) => {
    const displayText = getDisplayText(item.element);
    const isVisible = filterItems(displayText, filterValue, filter);
    return { ...item, originalIndex: index, isVisible };
  });
}
function computeNavigableItems(visibleItems) {
  return visibleItems.filter((item) => item.isVisible && !item.element.props.disabled);
}
function useResultAnnouncement(filterValue, filteredCount) {
  const { t } = reactI18next.useTranslation("fx");
  const [resultAnnouncement, setResultAnnouncement] = React11.useState("");
  React11.useEffect(() => {
    if (!filterValue) {
      setResultAnnouncement("");
      return;
    }
    const timer = setTimeout(() => {
      if (filteredCount === 0) {
        setResultAnnouncement(t("COMBOBOX_NO_RESULTS"));
      } else if (filteredCount === 1) {
        setResultAnnouncement(t("COMBOBOX_ONE_RESULT"));
      } else {
        setResultAnnouncement(
          t("COMBOBOX_RESULTS_AVAILABLE", { count: filteredCount })
        );
      }
    }, 300);
    return () => clearTimeout(timer);
  }, [filterValue, filteredCount, t]);
  return { resultAnnouncement, t };
}
function useMessageHeight(messageRef, valueStateMessage) {
  const [messageHeight, setMessageHeight] = React11.useState(0);
  React11.useLayoutEffect(() => {
    if (messageRef.current) {
      setMessageHeight(messageRef.current.offsetHeight);
    }
  }, [valueStateMessage]);
  return messageHeight;
}
function usePopoverControls({
  disabled,
  readonly,
  controlledOpen,
  isOpen,
  setInternalOpen,
  setFocusedIndex,
  onOpen,
  onClose
}) {
  const openPopover = React11.useCallback(() => {
    if (disabled || readonly) return;
    if (controlledOpen === void 0) {
      setInternalOpen(true);
    }
    onOpen?.();
  }, [disabled, readonly, controlledOpen, onOpen, setInternalOpen]);
  const closePopover = React11.useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    setFocusedIndex(-1);
  }, [controlledOpen, setInternalOpen, setFocusedIndex]);
  const handlePopoverClose = React11.useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    setFocusedIndex(-1);
    onClose?.();
  }, [controlledOpen, onClose, setInternalOpen, setFocusedIndex]);
  const togglePopover = React11.useCallback(() => {
    if (isOpen) {
      closePopover();
    } else {
      openPopover();
    }
  }, [isOpen, openPopover, closePopover]);
  return { openPopover, closePopover, handlePopoverClose, togglePopover };
}
function LiveRegion({ message, mode = "polite" }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      "aria-live": mode,
      "aria-atomic": true,
      className: "sr-only",
      children: message
    }
  );
}
var inputContainerVariants = (opts) => {
  const isLarge = opts.size === "Large" /* Large */;
  return cn(
    valueStateInputVariants(opts),
    isLarge ? "h-10 pl-3.5 pr-1 py-1 gap-1" : "h-8 pl-3 pr-1 py-1 gap-1",
    isLarge ? "focus-within:border-2 focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]" : "focus-within:border-2 focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]",
    opts.readonly && (isLarge ? "py-2.5" : "py-1.5")
  );
};
function getItemData(element) {
  const { text, value, children, additionalText, icon, disabled } = element.props;
  const displayText = text ?? (typeof children === "string" ? children : "");
  const itemValue = value ?? displayText;
  return { displayText, itemValue, additionalText, icon, disabled };
}
function buildItemData(element) {
  const { displayText, itemValue, additionalText, icon, disabled } = getItemData(element);
  return { text: displayText, value: itemValue, additionalText, icon, disabled };
}
function ComboBox({
  value: controlledValue,
  defaultValue = "",
  selectedValue: controlledSelectedValue,
  defaultSelectedValue,
  size = "Large" /* Large */,
  placeholder = "",
  disabled = false,
  readonly = false,
  required = false,
  name,
  filter = "StartsWithPerTerm" /* StartsWithPerTerm */,
  noTypeahead = false,
  valueState = "None" /* None */,
  valueStateMessage,
  showClearIcon = false,
  loading = false,
  open: controlledOpen,
  defaultOpen = false,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onInput,
  onChange,
  onSelectionChange,
  onOpen,
  onClose,
  children,
  className,
  icon,
  accessibilityAttributes,
  ref
}) {
  const isLargeSize = size === "Large" /* Large */;
  const containerRef = React11.useRef(null);
  const inputRef = React11.useRef(null);
  const messageRef = React11.useRef(null);
  const [portalTarget, setPortalTarget] = React11.useState(null);
  const openedByTypingRef = React11.useRef(false);
  const linkNavigationRef = React11.useRef(false);
  const [internalValue, setInternalValue] = React11.useState(defaultValue);
  const [internalOpen, setInternalOpen] = React11.useState(defaultOpen);
  const [filterValue, setFilterValue] = React11.useState("");
  const [focusedIndex, setFocusedIndex] = React11.useState(-1);
  const [lastCommittedValue, setLastCommittedValue] = React11.useState(defaultValue);
  const [internalSelectedValue, setInternalSelectedValue] = React11.useState(defaultSelectedValue);
  const [, setSelectedItem] = React11.useState(null);
  const value = controlledValue ?? internalValue;
  const isOpen = controlledOpen ?? internalOpen;
  const selectedValue = controlledSelectedValue ?? internalSelectedValue;
  const updateSelectedValue = React11.useCallback((val) => {
    if (controlledSelectedValue === void 0) {
      setInternalSelectedValue(val);
    }
  }, [controlledSelectedValue]);
  const prevIsOpenRef = React11.useRef(isOpen);
  const normalizedValueState = valueState || "None" /* None */;
  const hasValueState = normalizedValueState !== "None" /* None */ && !!valueStateMessage;
  const isNegativeState = normalizedValueState === "Negative" /* Negative */;
  const baseId = React11.useMemo(() => `combobox-${Math.random().toString(36).slice(2, 9)}`, []);
  const popoverId = `${baseId}-popover`;
  const inputId = `${baseId}-input`;
  const messageId = `${baseId}-message`;
  const accessibleDescSpanId = `${baseId}-accessible-desc`;
  const linkShortcutHintId = `${baseId}-link-hint`;
  const flattenedItems = React11.useMemo(() => {
    const items = [];
    React11.Children.forEach(children, (child, groupIdx) => {
      if (!React11.isValidElement(child)) return;
      if (child.type?.displayName === "ComboBoxItemGroup") {
        const groupProps = child.props;
        React11.Children.forEach(groupProps.children, (groupChild) => {
          if (React11.isValidElement(groupChild)) {
            items.push({
              element: groupChild,
              groupIndex: groupIdx
            });
          }
        });
      } else {
        items.push({
          element: child
        });
      }
    });
    return items;
  }, [children]);
  const visibleItems = React11.useMemo(
    () => computeVisibleItems(flattenedItems, filterValue, filter, (el) => getItemData(el).displayText),
    [flattenedItems, filterValue, filter]
  );
  const navigableItems = React11.useMemo(() => computeNavigableItems(visibleItems), [visibleItems]);
  const filteredCount = React11.useMemo(
    () => visibleItems.filter((item) => item.isVisible).length,
    [visibleItems]
  );
  const { resultAnnouncement, t } = useResultAnnouncement(filterValue, filteredCount);
  const messageHeight = useMessageHeight(messageRef, valueStateMessage);
  const syncInputToFocusedItem = React11.useCallback(
    (index) => {
      if (index < 0 || index >= navigableItems.length) return;
      const item = navigableItems[index];
      const { displayText } = getItemData(item.element);
      if (controlledValue === void 0) {
        setInternalValue(displayText);
      }
      requestAnimationFrame(() => {
        if (inputRef.current) {
          inputRef.current.setSelectionRange(0, displayText.length);
        }
      });
    },
    [navigableItems, controlledValue]
  );
  React11.useEffect(() => {
    if (!isOpen || !portalTarget) return;
    const handleLinkKeyDown = (e) => {
      if (!linkNavigationRef.current) return;
      const links = Array.from(portalTarget.querySelectorAll("a"));
      if (links.length === 0) return;
      const currentIndex = links.indexOf(e.target);
      if (currentIndex === -1) return;
      const focusInput = () => {
        linkNavigationRef.current = false;
        inputRef.current?.focus();
      };
      switch (e.key) {
        case "Tab":
          e.preventDefault();
          if (e.shiftKey) {
            if (currentIndex <= 0) {
              focusInput();
            } else {
              links[currentIndex - 1].focus();
            }
          } else {
            if (currentIndex >= links.length - 1) {
              focusInput();
            } else {
              links[currentIndex + 1].focus();
            }
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          focusInput();
          break;
        case "ArrowDown":
          e.preventDefault();
          focusInput();
          setFocusedIndex(0);
          syncInputToFocusedItem(0);
          break;
        case "Escape":
          e.preventDefault();
          e.stopPropagation();
          focusInput();
          break;
      }
    };
    portalTarget.addEventListener("keydown", handleLinkKeyDown);
    return () => portalTarget.removeEventListener("keydown", handleLinkKeyDown);
  }, [isOpen, portalTarget, syncInputToFocusedItem]);
  const navigateItemClosed = React11.useCallback(
    (itemElement) => {
      const { displayText, itemValue } = getItemData(itemElement);
      if (controlledValue === void 0) {
        setInternalValue(displayText);
      }
      setFilterValue("");
      updateSelectedValue(itemValue);
      const itemData = buildItemData(itemElement);
      setSelectedItem(itemData);
      onInput?.(displayText);
      onSelectionChange?.({ item: itemData, value: displayText });
    },
    [controlledValue, onInput, onSelectionChange, updateSelectedValue]
  );
  const { openPopover, closePopover, handlePopoverClose, togglePopover } = usePopoverControls({
    disabled,
    readonly,
    controlledOpen,
    isOpen,
    setInternalOpen,
    setFocusedIndex,
    onOpen,
    onClose
  });
  React11.useEffect(() => {
    if (isOpen && filterValue && visibleItems.every((item) => !item.isVisible)) {
      closePopover();
    }
  }, [isOpen, filterValue, visibleItems, closePopover]);
  React11.useEffect(() => {
    const wasOpen = prevIsOpenRef.current;
    prevIsOpenRef.current = isOpen;
    if (!isOpen || wasOpen) return;
    if (navigableItems.length === 0) return;
    let initialIndex = -1;
    if (value) {
      const selectedIdx = navigableItems.findIndex((item) => {
        const { displayText } = getItemData(item.element);
        return displayText === value;
      });
      if (selectedIdx >= 0) {
        initialIndex = selectedIdx;
      }
    }
    setFocusedIndex(initialIndex);
    if (!openedByTypingRef.current && initialIndex >= 0) {
      syncInputToFocusedItem(initialIndex);
    }
    openedByTypingRef.current = false;
  }, [isOpen]);
  const selectItem = React11.useCallback(
    (itemElement) => {
      const { displayText, itemValue } = getItemData(itemElement);
      if (controlledValue === void 0) {
        setInternalValue(displayText);
      }
      setFilterValue("");
      setLastCommittedValue(displayText);
      updateSelectedValue(itemValue);
      const itemData = buildItemData(itemElement);
      setSelectedItem(itemData);
      onInput?.(displayText);
      onChange?.(displayText);
      onSelectionChange?.({ item: itemData, value: displayText });
      closePopover();
      inputRef.current?.focus();
    },
    [controlledValue, onInput, onChange, onSelectionChange, closePopover, updateSelectedValue]
  );
  const handleInputChange = React11.useCallback(
    (e) => {
      const newValue = e.target.value;
      if (controlledValue === void 0) {
        setInternalValue(newValue);
      }
      setFilterValue(newValue);
      onInput?.(newValue);
      if (!isOpen && newValue) {
        openedByTypingRef.current = true;
        openPopover();
      }
      const nativeInputType = e.nativeEvent.inputType ?? "";
      const isDeletion = nativeInputType.startsWith("delete") || nativeInputType === "historyUndo";
      if (isDeletion) {
        handleDeletionInput(newValue);
        return;
      }
      if (!noTypeahead && newValue) {
        handleTypeahead(newValue);
      }
    },
    [controlledValue, isOpen, noTypeahead, navigableItems, onInput, onSelectionChange, openPopover, updateSelectedValue, selectedValue, internalSelectedValue]
  );
  const handleDeletionInput = React11.useCallback(
    (newValue) => {
      if (selectedValue !== void 0 || internalSelectedValue !== void 0) {
        updateSelectedValue(void 0);
        onSelectionChange?.({ item: null, value: newValue });
      }
      setFocusedIndex(-1);
    },
    [selectedValue, internalSelectedValue, updateSelectedValue, onSelectionChange]
  );
  const handleTypeahead = React11.useCallback(
    (newValue) => {
      const matchingItem = navigableItems.find((item) => {
        const { displayText } = getItemData(item.element);
        return displayText.toLowerCase().startsWith(newValue.toLowerCase());
      });
      if (matchingItem && inputRef.current) {
        const { displayText, itemValue } = getItemData(matchingItem.element);
        if (controlledValue === void 0) {
          setInternalValue(displayText);
        }
        updateSelectedValue(itemValue);
        const itemData = buildItemData(matchingItem.element);
        onSelectionChange?.({ item: itemData, value: displayText });
        const startsWithInput = displayText.toLowerCase().startsWith(newValue.toLowerCase());
        const selectionStart = startsWithInput ? newValue.length : 0;
        requestAnimationFrame(() => {
          inputRef.current?.setSelectionRange(selectionStart, displayText.length);
        });
      } else if (selectedValue !== void 0 || internalSelectedValue !== void 0) {
        updateSelectedValue(void 0);
        onSelectionChange?.({ item: null, value: newValue });
      }
    },
    [controlledValue, navigableItems, updateSelectedValue, onSelectionChange, selectedValue, internalSelectedValue]
  );
  const handleF4 = React11.useCallback(() => {
    if (!isOpen) {
      openPopover();
      if (value) {
        const matchIdx = navigableItems.findIndex((item) => {
          const { displayText } = getItemData(item.element);
          return displayText === value;
        });
        if (matchIdx >= 0) {
          setFocusedIndex(matchIdx);
          syncInputToFocusedItem(matchIdx);
        } else {
          setFocusedIndex(-1);
        }
      } else if (navigableItems.length > 0) {
        setFocusedIndex(0);
        syncInputToFocusedItem(0);
      }
    } else {
      closePopover();
    }
  }, [isOpen, value, navigableItems, openPopover, closePopover, syncInputToFocusedItem]);
  const handleClear = React11.useCallback(() => {
    if (controlledValue === void 0) {
      setInternalValue("");
    }
    setFilterValue("");
    setSelectedItem(null);
    updateSelectedValue(void 0);
    onInput?.("");
    onChange?.("");
    onSelectionChange?.({ item: null, value: "" });
    inputRef.current?.focus();
  }, [controlledValue, onInput, onChange, onSelectionChange, updateSelectedValue]);
  const handleKeyDown = React11.useCallback(
    (e) => {
      if (disabled || readonly) return;
      const findCurrentIndex = () => navigableItems.findIndex((item) => {
        const { displayText } = getItemData(item.element);
        return displayText === value;
      });
      const navigateClosed = (index) => {
        if (index >= 0 && index < navigableItems.length) {
          navigateItemClosed(navigableItems[index].element);
        }
      };
      const focusOpenItem = (index) => {
        setFocusedIndex(index);
        syncInputToFocusedItem(index);
      };
      const navigateToIndex = (index) => {
        if (isOpen) {
          focusOpenItem(index);
        } else {
          navigateClosed(index);
        }
      };
      if (e.key === "F8" && (e.ctrlKey || e.metaKey) && e.altKey && isOpen && portalTarget) {
        e.preventDefault();
        const links = portalTarget.querySelectorAll("a");
        if (links.length > 0) {
          linkNavigationRef.current = true;
          setFocusedIndex(-1);
          links[0].focus();
        }
        return;
      }
      switch (e.key) {
        case "F4":
          e.preventDefault();
          handleF4();
          break;
        case "ArrowDown":
          e.preventDefault();
          if (e.altKey) {
            togglePopover();
          } else if (isOpen) {
            const nextDown = focusedIndex + 1;
            if (nextDown < navigableItems.length) {
              focusOpenItem(nextDown);
            }
          } else {
            navigateClosed(findCurrentIndex() + 1);
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          if (e.altKey) {
            togglePopover();
          } else if (isOpen) {
            if (focusedIndex <= 0) {
              setFocusedIndex(-1);
              if (controlledValue === void 0) {
                setInternalValue(filterValue);
              }
            } else {
              focusOpenItem(focusedIndex - 1);
            }
          } else {
            const currentIndex = findCurrentIndex();
            if (currentIndex > 0) {
              navigateClosed(currentIndex - 1);
            }
          }
          break;
        case "PageDown":
          e.preventDefault();
          navigateToIndex(
            isOpen ? Math.min(focusedIndex + 10, navigableItems.length - 1) : Math.min(findCurrentIndex() + 10, navigableItems.length - 1)
          );
          break;
        case "PageUp":
          e.preventDefault();
          navigateToIndex(
            isOpen ? Math.max(focusedIndex - 10, 0) : Math.max(findCurrentIndex() - 10, 0)
          );
          break;
        case "Home":
          e.preventDefault();
          navigateToIndex(0);
          break;
        case "End":
          e.preventDefault();
          navigateToIndex(navigableItems.length - 1);
          break;
        case "Enter":
          if (isOpen && focusedIndex >= 0 && focusedIndex < navigableItems.length) {
            e.preventDefault();
            selectItem(navigableItems[focusedIndex].element);
          } else if (!isOpen && value !== lastCommittedValue) {
            setLastCommittedValue(value);
            onChange?.(value);
          }
          break;
        case "Escape":
          e.preventDefault();
          if (isOpen) {
            closePopover();
          } else if (showClearIcon && value && !disabled && !readonly) {
            handleClear();
          } else {
            if (controlledValue === void 0) {
              setInternalValue(lastCommittedValue);
            }
            setFilterValue("");
          }
          break;
        case "Tab":
          if (isOpen) {
            closePopover();
          }
          break;
      }
    },
    [
      disabled,
      readonly,
      isOpen,
      focusedIndex,
      navigableItems,
      value,
      filterValue,
      lastCommittedValue,
      controlledValue,
      togglePopover,
      closePopover,
      selectItem,
      navigateItemClosed,
      syncInputToFocusedItem,
      onChange,
      portalTarget,
      handleF4,
      showClearIcon,
      handleClear
    ]
  );
  const handleBlur = React11.useCallback(
    (e) => {
      const container = containerRef.current;
      if (container?.contains(e.relatedTarget)) {
        return;
      }
      if (linkNavigationRef.current) {
        return;
      }
      if (value !== lastCommittedValue) {
        setLastCommittedValue(value);
        onChange?.(value);
      }
    },
    [value, lastCommittedValue, onChange]
  );
  const renderChildren = React11.useCallback(() => {
    let itemIndex = 0;
    const isItemSelected = (element) => {
      const { displayText, itemValue } = getItemData(element);
      if (selectedValue !== void 0) {
        return itemValue === selectedValue;
      }
      return displayText === value;
    };
    const getItemProps = (element) => {
      const currentItemIndex = itemIndex++;
      const visibleItem = visibleItems.find((vi) => vi.originalIndex === currentItemIndex);
      const isVisible = visibleItem?.isVisible ?? true;
      const isSelected = isItemSelected(element);
      const isFocused = navigableItems.findIndex((ni) => ni.originalIndex === currentItemIndex) === focusedIndex;
      const showSelected = isOpen && focusedIndex >= 0 ? isFocused : isSelected;
      return { isVisible, selected: showSelected, focused: isFocused, onClick: () => selectItem(element) };
    };
    return React11.Children.map(children, (child) => {
      if (!React11.isValidElement(child)) return child;
      if (child.type?.displayName === "ComboBoxItemGroup") {
        const groupProps = child.props;
        let hasVisibleChild = false;
        const groupChildren = React11.Children.map(groupProps.children, (groupChild) => {
          if (!React11.isValidElement(groupChild)) return groupChild;
          const props = getItemProps(groupChild);
          if (props.isVisible) hasVisibleChild = true;
          return React11.cloneElement(groupChild, props);
        });
        return React11.cloneElement(child, {
          isVisible: hasVisibleChild,
          children: groupChildren
        });
      }
      return React11.cloneElement(
        child,
        getItemProps(child)
      );
    });
  }, [children, visibleItems, navigableItems, focusedIndex, value, selectedValue, isOpen, selectItem]);
  const showClear = showClearIcon && value && !disabled && !readonly;
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { ref, className: cn("relative w-full", className), children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: containerRef,
        className: cn(
          inputContainerVariants({
            size,
            valueState,
            disabled,
            readonly
          }),
          isOpen && focusedIndex >= 0 && (isLargeSize ? "!border !pl-3.5 !pr-1 !py-1" : "!border !pl-3 !pr-1 !py-1")
        ),
        children: [
          icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex items-center justify-center shrink-0 text-sapphire-text-tertiary", children: icon }),
          /* @__PURE__ */ jsxRuntime.jsx(
            "input",
            {
              ref: inputRef,
              id: inputId,
              type: "text",
              role: "combobox",
              "aria-expanded": isOpen,
              "aria-haspopup": "listbox",
              "aria-autocomplete": "both",
              "aria-controls": [isOpen ? popoverId : null, accessibilityAttributes?.controls].filter(Boolean).join(" ") || void 0,
              "aria-activedescendant": focusedIndex >= 0 && focusedIndex < navigableItems.length ? (() => {
                const item = navigableItems[focusedIndex];
                if (!item) return void 0;
                const { displayText, itemValue } = getItemData(item.element);
                return itemValue ?? displayText;
              })() : void 0,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-required": required,
              "aria-invalid": isNegativeState || void 0,
              "aria-busy": loading || void 0,
              "aria-describedby": [
                valueStateMessage ? messageId : null,
                hasValueState ? linkShortcutHintId : null,
                accessibleDescriptionRef,
                accessibleDescription ? accessibleDescSpanId : null
              ].filter(Boolean).join(" ") || void 0,
              name,
              value,
              placeholder,
              disabled,
              readOnly: readonly,
              onChange: handleInputChange,
              onKeyDown: handleKeyDown,
              onBlur: handleBlur,
              className: cn(
                "flex-1 h-full min-w-0 bg-transparent text-sm text-sapphire-text-primary",
                "outline-none",
                "placeholder:text-sapphire-text-tertiary placeholder:italic",
                "disabled:cursor-not-allowed",
                readonly && "text-sapphire-text-tertiary"
              )
            }
          ),
          showClear && /* @__PURE__ */ jsxRuntime.jsx(ClearButton, { onClick: handleClear, comboBoxSize: size }),
          !readonly && /* @__PURE__ */ jsxRuntime.jsx(
            DropdownArrowButton,
            {
              onClick: () => {
                togglePopover();
                inputRef.current?.focus();
              },
              disabled,
              comboBoxSize: size
            }
          )
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx(
      ComboBoxPopover,
      {
        open: isOpen,
        triggerRef: containerRef,
        hasValueState,
        valueStatePortalRef: setPortalTarget,
        loading,
        onClose: handlePopoverClose,
        children: renderChildren()
      }
    ),
    valueStateMessage && /* @__PURE__ */ jsxRuntime.jsx(
      ValueStateMessage,
      {
        isOpen,
        hasValueState,
        portalTarget,
        messageHeight,
        messageRef,
        messageId,
        normalizedValueState,
        valueStateMessage
      }
    ),
    accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: accessibleDescSpanId, className: "sr-only", children: accessibleDescription }),
    hasValueState && /* @__PURE__ */ jsxRuntime.jsx("span", { id: linkShortcutHintId, className: "sr-only", children: t("COMBOBOX_VALUE_STATE_LINK_SHORTCUT") }),
    /* @__PURE__ */ jsxRuntime.jsx(LiveRegion, { message: resultAnnouncement })
  ] });
}
function ComboBoxItem({
  text,
  additionalText,
  value,
  icon: _icon,
  disabled = false,
  selected = false,
  focused = false,
  isVisible = true,
  onClick,
  children,
  className,
  ref
}) {
  if (!isVisible) {
    return null;
  }
  const displayText = text ?? (typeof children === "string" ? children : "");
  const itemValue = value ?? displayText;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "li",
    {
      ref,
      role: "option",
      "aria-selected": selected,
      "aria-disabled": disabled || void 0,
      id: itemValue,
      "data-value": itemValue,
      onClick: onClick && !disabled ? () => onClick() : void 0,
      className: cn(
        "flex items-center min-h-8 px-4 py-1.5 gap-4 text-sm cursor-pointer",
        "bg-sapphire-background-primary text-sapphire-text-primary",
        "transition-colors outline-none relative",
        !disabled && "hover:bg-sapphire-neutral-hover-background dark:hover:bg-neutral-800",
        focused && !selected && "bg-sapphire-neutral-hover-background dark:bg-neutral-800",
        focused && "after:absolute after:inset-0.5 after:border-2 after:border-ring after:rounded-sm after:pointer-events-none",
        selected && "bg-sapphire-brand-selected-background",
        disabled && "opacity-50 pointer-events-none cursor-not-allowed",
        className
      ),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-1", children: children || displayText }),
        additionalText && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0 text-sapphire-text-tertiary text-sm", children: additionalText })
      ]
    }
  );
}
ComboBoxItem.displayName = "ComboBoxItem";
function ComboBoxItemGroup({ headerText, isVisible = true, children, className, ref }) {
  if (!isVisible) {
    return null;
  }
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref,
      role: "group",
      "aria-label": headerText,
      className: cn("py-1", className),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(
              "px-3 py-1.5 text-xs font-medium text-sapphire-text-tertiary",
              "uppercase tracking-wide select-none"
            ),
            "aria-hidden": "true",
            children: headerText
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx("div", { role: "presentation", children })
      ]
    }
  );
}
ComboBoxItemGroup.displayName = "ComboBoxItemGroup";

exports.ComboBox = ComboBox;
exports.ComboBoxFilter = ComboBoxFilter;
exports.ComboBoxItem = ComboBoxItem;
exports.ComboBoxItemGroup = ComboBoxItemGroup;
exports.ComboBoxPopover = ComboBoxPopover;
exports.ComboBoxSize = ComboBoxSize;
exports.ValueState = ValueState;
