// @sap-ui/fx-components/dist/components/textarea/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');
var reactI18next = require('react-i18next');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/textarea/Textarea.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData5,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData5 || pathData5.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData5.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M256 0q53 0 100 20t81.5 54.5T492 156t20 100-20 100-54.5 81.5T356 492t-100 20-100-20-81.5-54.5T20 356 0 256t20-100 54.5-81.5T156 20 256 0zm150 183q10-9 10-23 0-13-9.5-22.5T384 128t-22 9L186 308l-68-63q-9-9-22-9t-22.5 9.5T64 268q0 15 10 24l91 83q9 9 21 9 13 0 23-9z"
];
var SysEnter2Icon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SysEnter2Icon", iconName: "sys-enter-2" }
);
var pathData2 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm45 256l74-73q9-11 9-23 0-13-9.5-22.5T352 128q-12 0-23 9l-73 74-73-74q-10-9-23-9t-22.5 9.5T128 160q0 12 9 23l74 73-74 73q-9 10-9 23t9.5 22.5T160 384t23-9l73-74 73 74q11 9 23 9 13 0 22.5-9.5T384 352t-9-23z"
];
var ErrorIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "ErrorIcon", iconName: "error" }
);
var pathData3 = [
  "M505 399q7 13 7 27 0 21-15.5 37.5T456 480H56q-25 0-40.5-16.5T0 426q0-14 7-27L208 59q17-27 48-27 14 0 27 6.5T304 59zM288 176q0-14-9-23t-23-9-23 9-9 23v96q0 14 9 23t23 9 23-9 9-23v-96zm-32 240q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9z"
];
var WarningIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "WarningIcon", iconName: "warning" }
);
var pathData4 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm0 160q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9zm32 64q0-14-9-23t-23-9-23 9-9 23v160q0 14 9 23t23 9 23-9 9-23V224z"
];
var InformationIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "InformationIcon", iconName: "information" }
);

// src/lib/value-state-utils.ts
var valueStateIcons = {
  ["None" /* None */]: null,
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Critical" /* Critical */]: WarningIcon,
  ["Information" /* Information */]: InformationIcon
};
var valueStateMessageColors = {
  ["None" /* None */]: "text-sapphire-text-tertiary",
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Information" /* Information */]: "text-sapphire-info"
};
var valueStateLabelKeys = {
  ["None" /* None */]: "",
  ["Positive" /* Positive */]: "VALUE_STATE_SUCCESS",
  ["Negative" /* Negative */]: "VALUE_STATE_ERROR",
  ["Critical" /* Critical */]: "VALUE_STATE_WARNING",
  ["Information" /* Information */]: "VALUE_STATE_INFORMATION"
};
var textareaContainerVariants = cva(
  [
    "flex flex-col w-full rounded border bg-sapphire-canvas-primary",
    "overflow-clip transition-colors relative",
    "p-1",
    "focus-within:border-2 focus-within:p-[3px]"
  ],
  {
    variants: {
      valueState: {
        ["None" /* None */]: "border-sapphire-border-secondary hover:border-sapphire-border-accent focus-within:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-positive focus-within:border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-negative focus-within:border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-warning focus-within:border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-info focus-within:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      readonly: {
        true: "bg-sapphire-background-secondary border-sapphire-border-primary hover:border-sapphire-border-primary focus-within:!border-sapphire-border-accent",
        false: ""
      }
    },
    defaultVariants: {
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
function Textarea(props) {
  const {
    value: controlledValue,
    defaultValue = "",
    disabled = false,
    readonly = false,
    required = false,
    autocomplete,
    valueState = "None" /* None */,
    valueStateMessage,
    maxLength,
    showExceededText = false,
    minLength,
    rows = 0,
    growing = false,
    growingMaxRows,
    placeholder,
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    name,
    id: providedId,
    onInput,
    onChange,
    onFocus,
    onBlur,
    onKeyDown,
    onKeyUp,
    onSelect,
    onScroll,
    className,
    style,
    "data-testid": dataTestId,
    "data-ai-field": dataAiField,
    ref
  } = props;
  const { t } = reactI18next.useTranslation("fx");
  const [internalValue, setInternalValue] = React2.useState(defaultValue);
  const [focused, setFocused] = React2.useState(false);
  const [previousValue, setPreviousValue] = React2.useState(defaultValue);
  const textareaRef = React2.useRef(null);
  const isComposing = React2.useRef(false);
  const isControlled = controlledValue !== void 0;
  const currentValue = isControlled ? controlledValue : internalValue;
  const generatedId = React2.useId();
  const textareaId = providedId || generatedId;
  const valueStateId = `${textareaId}-valuestate`;
  const characterCount = currentValue.length;
  const exceededCharacters = maxLength ? Math.max(0, characterCount - maxLength) : 0;
  const hasExceededMax = exceededCharacters > 0;
  React2.useEffect(() => {
    if (growing && textareaRef.current) {
      const textarea = textareaRef.current;
      textarea.style.height = "auto";
      const newHeight = textarea.scrollHeight;
      if (growingMaxRows) {
        const lineHeight = parseInt(getComputedStyle(textarea).lineHeight);
        const maxHeight = lineHeight * growingMaxRows;
        textarea.style.height = `${Math.min(newHeight, maxHeight)}px`;
      } else {
        textarea.style.height = `${newHeight}px`;
      }
    }
  }, [currentValue, growing, growingMaxRows]);
  const handleInternalValueChange = React2.useCallback(
    (newValue) => {
      if (!isControlled) {
        setInternalValue(newValue);
      }
    },
    [isControlled]
  );
  const handleInput = React2.useCallback(
    (e) => {
      if (isComposing.current) return;
      const newValue = e.currentTarget.value;
      handleInternalValueChange(newValue);
      onInput?.({
        value: newValue,
        escapePressed: false
      });
    },
    [handleInternalValueChange, onInput]
  );
  const handleChange = React2.useCallback(() => {
    if (currentValue !== previousValue) {
      onChange?.({
        value: currentValue,
        previousValue,
        source: "user"
      });
      setPreviousValue(currentValue);
    }
  }, [currentValue, previousValue, onChange]);
  const handleFocus = React2.useCallback(
    (e) => {
      setFocused(true);
      onFocus?.(e);
    },
    [onFocus]
  );
  const handleBlur = React2.useCallback(
    (e) => {
      setFocused(false);
      handleChange();
      onBlur?.(e);
    },
    [onBlur, handleChange]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (e.key === "Escape") {
        onInput?.({
          value: currentValue,
          escapePressed: true
        });
      }
      onKeyDown?.(e);
    },
    [onKeyDown, onInput, currentValue]
  );
  const handleSelect = React2.useCallback(
    () => {
      if (!textareaRef.current) return;
      const { selectionStart, selectionEnd, value } = textareaRef.current;
      onSelect?.({
        start: selectionStart,
        end: selectionEnd,
        selectedText: value.substring(selectionStart, selectionEnd)
      });
    },
    [onSelect]
  );
  const handleCompositionStart = React2.useCallback(() => {
    isComposing.current = true;
  }, []);
  const handleCompositionEnd = React2.useCallback(
    (e) => {
      isComposing.current = false;
      const newValue = e.currentTarget.value;
      handleInternalValueChange(newValue);
      onInput?.({
        value: newValue,
        escapePressed: false
      });
    },
    [handleInternalValueChange, onInput]
  );
  React2.useImperativeHandle(
    ref,
    () => ({
      focus: () => textareaRef.current?.focus(),
      blur: () => textareaRef.current?.blur(),
      select: () => textareaRef.current?.select(),
      setSelectionRange: (start, end) => {
        textareaRef.current?.setSelectionRange(start, end);
      },
      getValue: () => currentValue,
      setValue: (value, options) => {
        handleInternalValueChange(value);
        if (!options?.silent) {
          onChange?.({
            value,
            previousValue: currentValue,
            source: "programmatic"
          });
        }
        setPreviousValue(value);
      },
      clear: () => {
        handleInternalValueChange("");
        onChange?.({
          value: "",
          previousValue: currentValue,
          source: "programmatic"
        });
        setPreviousValue("");
      },
      getState: () => ({
        value: currentValue,
        focused,
        valid: textareaRef.current?.checkValidity() ?? true,
        valueState,
        characterCount,
        exceededCharacters
      }),
      checkValidity: () => textareaRef.current?.checkValidity() ?? true,
      reportValidity: () => textareaRef.current?.reportValidity() ?? true,
      setCustomValidity: (message) => {
        textareaRef.current?.setCustomValidity(message);
      },
      scrollTo: (x, y) => {
        textareaRef.current?.scrollTo(x, y);
      },
      get nativeElement() {
        return textareaRef.current;
      }
    }),
    [
      currentValue,
      focused,
      valueState,
      characterCount,
      exceededCharacters,
      handleInternalValueChange,
      onChange
    ]
  );
  const effectiveValueState = valueState;
  const isNegativeState = effectiveValueState === "Negative" /* Negative */;
  const ValueStateIcon = valueStateIcons[effectiveValueState];
  const showValueStateIcon = ValueStateIcon && effectiveValueState !== "None" /* None */;
  const ariaLabel = accessibleName;
  const ariaLabelledBy = accessibleNameRef;
  const ariaDescribedBy = [
    accessibleDescriptionRef,
    valueStateMessage ? valueStateId : null
  ].filter(Boolean).join(" ") || void 0;
  const ariaInvalid = effectiveValueState === "Negative" /* Negative */ ? true : void 0;
  const ariaRequired = required || void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: cn("w-full", className), style, children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        className: cn(textareaContainerVariants({
          valueState: effectiveValueState,
          disabled,
          readonly
        })),
        "data-part": "container",
        children: /* @__PURE__ */ jsxRuntime.jsx(
          "textarea",
          {
            ref: textareaRef,
            id: textareaId,
            name,
            value: currentValue,
            disabled,
            readOnly: readonly,
            required,
            placeholder,
            maxLength: maxLength && !showExceededText ? maxLength : void 0,
            minLength,
            rows: rows || void 0,
            autoComplete: autocomplete,
            "aria-label": ariaLabel,
            "aria-labelledby": ariaLabelledBy,
            "aria-describedby": ariaDescribedBy,
            "aria-invalid": ariaInvalid,
            "aria-required": ariaRequired,
            "data-testid": dataTestId,
            "data-ai-field": dataAiField,
            "data-part": "textarea",
            onInput: handleInput,
            onFocus: handleFocus,
            onBlur: handleBlur,
            onKeyDown: handleKeyDown,
            onKeyUp,
            onSelect: handleSelect,
            onScroll,
            onCompositionStart: handleCompositionStart,
            onCompositionEnd: handleCompositionEnd,
            className: cn(
              "w-full resize-none bg-transparent px-2 py-[3px]",
              "text-sm leading-5 text-sapphire-text-primary outline-none",
              "placeholder:text-sapphire-text-tertiary placeholder:italic",
              "disabled:cursor-not-allowed",
              readonly && "text-sapphire-text-secondary",
              growing && "overflow-hidden"
            )
          }
        )
      }
    ),
    (maxLength || showExceededText) && /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        className: cn(
          "flex items-center justify-end gap-2 text-xs mt-1",
          hasExceededMax ? "text-sapphire-negative" : "text-sapphire-text-tertiary"
        ),
        "data-part": "counter",
        children: [
          showExceededText && hasExceededMax && /* @__PURE__ */ jsxRuntime.jsxs("span", { children: [
            "+",
            exceededCharacters
          ] }),
          maxLength && /* @__PURE__ */ jsxRuntime.jsxs("span", { children: [
            characterCount,
            " / ",
            maxLength
          ] })
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        id: valueStateId,
        role: isNegativeState ? "alert" : "status",
        className: cn(
          "text-sm flex items-start gap-1",
          valueStateMessage ? "mt-2" : "hidden",
          valueStateMessageColors[effectiveValueState]
        ),
        "data-part": "value-state-message",
        children: valueStateMessage && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
          effectiveValueState !== "None" /* None */ && valueStateLabelKeys[effectiveValueState] && /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "sr-only", children: [
            t(valueStateLabelKeys[effectiveValueState]),
            ":",
            " "
          ] }),
          showValueStateIcon && ValueStateIcon && /* @__PURE__ */ jsxRuntime.jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
          /* @__PURE__ */ jsxRuntime.jsx("span", { children: valueStateMessage })
        ] })
      }
    )
  ] });
}

exports.Textarea = Textarea;
