// @sap-ui/fx-components/dist/components/badge/index.js
import { cn } from '../../chunk-2F2PZMJ5.js';
import '../../chunk-V6TY7KAL.js';
import { useRef, useId, useImperativeHandle, useCallback } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

var dotColorClass = {
  ["Neutral" /* Neutral */]: "bg-sapphire-border-active",
  ["Information" /* Information */]: "bg-sapphire-info",
  ["Positive" /* Positive */]: "bg-sapphire-positive",
  ["Negative" /* Negative */]: "bg-sapphire-negative",
  ["Critical" /* Critical */]: "bg-sapphire-warning"
};
var outlineBorderClass = {
  ["Neutral" /* Neutral */]: "border-sapphire-border-active",
  ["Information" /* Information */]: "border-sapphire-info",
  ["Positive" /* Positive */]: "border-sapphire-positive",
  ["Negative" /* Negative */]: "border-sapphire-negative",
  ["Critical" /* Critical */]: "border-sapphire-warning"
};
var filledDesignClass = {
  ["Neutral" /* Neutral */]: "bg-muted text-foreground",
  ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info",
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning"
};
var tintedDesignClass = {
  ["Neutral" /* Neutral */]: "bg-muted text-foreground border-sapphire-border-active",
  ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info border-sapphire-info",
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive border-sapphire-positive",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative border-sapphire-negative",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning border-sapphire-warning"
};
var sizeClasses = {
  ["Filled" /* Filled */]: {
    ["S" /* S */]: "text-sm px-2 py-1 gap-1.5",
    ["L" /* L */]: "text-sm px-2 py-1 gap-1.5"
  },
  ["Minimal" /* Minimal */]: {
    ["S" /* S */]: "text-xs leading-[18px] px-2 py-[3px] gap-1.5",
    ["L" /* L */]: "text-base leading-[18px] px-3 py-[3px] gap-2"
  },
  ["Outline" /* Outline */]: {
    ["S" /* S */]: "text-xs leading-[18px] px-2 py-[3px] gap-1.5",
    ["L" /* L */]: "text-base leading-[18px] px-3 py-[3px] gap-2"
  },
  ["Tinted" /* Tinted */]: {
    ["S" /* S */]: "text-xs leading-[18px] px-2 py-[3px] gap-1.5",
    ["L" /* L */]: "text-base leading-[18px] px-3 py-[3px] gap-2"
  }
};
function getDotIcon(design, size) {
  const dotSize = size === "L" /* L */ ? "w-2.5 h-2.5" : "w-2 h-2";
  return /* @__PURE__ */ jsx(
    "span",
    {
      className: cn("rounded-full flex-shrink-0 inline-block", dotSize, dotColorClass[design]),
      "aria-hidden": "true"
    }
  );
}
function Badge({
  variant = "Filled" /* Filled */,
  design = "Neutral" /* Neutral */,
  size = "S" /* S */,
  interactive = false,
  wrappingType = "Normal" /* Normal */,
  hideStateIcon = false,
  showBorder = true,
  icon,
  children,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const badgeRef = useRef(null);
  const generatedId = useId();
  const v = variant;
  const d = design;
  const s = size;
  const descriptionId = `${id ?? generatedId}-desc`;
  const ariaDescribedBy = [accessibleDescriptionRef, accessibleDescription ? descriptionId : null].filter(Boolean).join(" ") || void 0;
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        badgeRef.current?.focus();
      },
      blur() {
        badgeRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === badgeRef.current;
      },
      get nativeElement() {
        return badgeRef.current;
      }
    }),
    []
  );
  const handleClick = useCallback(
    (e) => {
      if (!interactive) return;
      onClick?.({ originalEvent: e, isKeyboard: false });
    },
    [interactive, onClick]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (!interactive) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({ originalEvent: e, isKeyboard: true });
      }
    },
    [interactive, onClick]
  );
  let displayIcon = null;
  if (icon) {
    displayIcon = icon;
  } else if (!hideStateIcon && (v === "Minimal" /* Minimal */ || v === "Outline" /* Outline */ || v === "Tinted" /* Tinted */)) {
    displayIcon = getDotIcon(d, s);
  }
  const variantClasses = (() => {
    switch (v) {
      case "Filled" /* Filled */:
        return cn("font-semibold", filledDesignClass[d]);
      case "Minimal" /* Minimal */:
        return cn("font-normal text-sapphire-text-secondary border border-sapphire-border-primary");
      case "Outline" /* Outline */:
        return cn("font-normal text-sapphire-text-secondary border", outlineBorderClass[d]);
      case "Tinted" /* Tinted */:
        return showBorder ? cn("font-normal border", tintedDesignClass[d]) : cn("font-normal", filledDesignClass[d]);
      default:
        return filledDesignClass[d];
    }
  })();
  const wrappingClasses = wrappingType === "None" /* None */ ? "whitespace-nowrap overflow-hidden text-ellipsis max-w-full" : "whitespace-normal break-words";
  const interactiveClasses = interactive ? "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-1" : "";
  return /* @__PURE__ */ jsxs(
    "span",
    {
      ref: badgeRef,
      id,
      role: interactive ? "button" : void 0,
      tabIndex: interactive ? 0 : void 0,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-describedby": ariaDescribedBy,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        "inline-flex items-center rounded transition-colors",
        variantClasses,
        sizeClasses[v]?.[s] ?? sizeClasses["Filled" /* Filled */]["S" /* S */],
        wrappingClasses,
        interactiveClasses,
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        displayIcon && /* @__PURE__ */ jsx("span", { className: "flex-shrink-0", "data-part": "icon", children: displayIcon }),
        children && /* @__PURE__ */ jsx("span", { "data-part": "text", children }),
        accessibleDescription && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription })
      ]
    }
  );
}

export { Badge };
