// @sap-ui/fx-components/dist/components/notification/index.cjs
'use strict';

var React3 = require('react');
var tailwindMerge = require('tailwind-merge');
var lucideReact = require('lucide-react');
var jsxRuntime = require('react/jsx-runtime');
var reactI18next = require('react-i18next');

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var React3__namespace = /*#__PURE__*/_interopNamespace(React3);

// src/components/notification/NotificationList.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var dotVariants = cva("rounded-full bg-current inline-block", {
  variants: {
    size: {
      ["S" /* S */]: "w-2 h-2",
      ["M" /* M */]: "w-4 h-4",
      ["L" /* L */]: "w-8 h-8"
    }
  },
  defaultVariants: { size: "M" /* M */ }
});
var dotGapClasses = {
  ["S" /* S */]: "gap-px",
  ["M" /* M */]: "gap-[0.1875rem]",
  ["L" /* L */]: "gap-1"
};
function BusyIndicator({
  size = "M" /* M */,
  textPlacement = "Bottom" /* Bottom */,
  active = false,
  delay = 1e3,
  text,
  children,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = React3.useRef(null);
  const overlayRef = React3.useRef(null);
  const [isBusy, setIsBusy] = React3.useState(false);
  const labelId = React3.useId();
  const hasChildren = React3__namespace.default.Children.count(children) > 0;
  const normalizedSize = size || "M" /* M */;
  const isTopPlacement = textPlacement === "Top";
  React3.useEffect(() => {
    if (!active) {
      setIsBusy(false);
      return;
    }
    if (delay <= 0) {
      setIsBusy(true);
      return;
    }
    const timer = setTimeout(() => {
      setIsBusy(true);
    }, delay);
    return () => clearTimeout(timer);
  }, [active, delay]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.focus();
        } else {
          rootRef.current?.focus();
        }
      },
      blur() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.blur();
        } else {
          rootRef.current?.blur();
        }
      },
      isFocused() {
        return document.activeElement === overlayRef.current || document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    [isBusy]
  );
  const textLabel = text ? /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      id: labelId,
      className: cn(
        "w-full text-center text-sapphire-text-tertiary text-sm font-medium",
        isTopPlacement ? "mb-2" : "mt-2"
      ),
      children: text
    }
  ) : null;
  const dots = /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "inline-flex items-center leading-none",
        dotGapClasses[normalizedSize]
      ),
      children: [0, 1, 2].map((i) => /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: dotVariants({ size: normalizedSize }),
          style: {
            animation: `busy-grow 1.6s cubic-bezier(0.32, 0.06, 0.85, 1.11) ${i * 200}ms infinite`
          }
        },
        i
      ))
    }
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      id,
      className: cn(
        "inline-block",
        hasChildren && "relative",
        isBusy && "text-ring",
        className
      ),
      style: { ...style, borderRadius: "inherit", height: "inherit" },
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("style", { children: `@keyframes busy-grow{0%,50%,100%{transform:scale(.5)}25%{transform:scale(1)}}` }),
        isBusy && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: overlayRef,
            role: "progressbar",
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            "aria-valuetext": "Busy",
            "aria-labelledby": text ? labelId : void 0,
            "aria-label": !text ? accessibleName ?? "Please wait" : void 0,
            title: !text ? accessibleName ?? "Please wait" : void 0,
            tabIndex: 0,
            "data-sap-focus-ref": true,
            className: cn(
              "flex flex-col items-center justify-center",
              "bg-inherit rounded-[inherit]",
              "focus-visible:outline focus-visible:outline-2 focus-visible:outline-ring focus-visible:outline-offset-[-2px]",
              hasChildren && "absolute inset-0 z-[99]"
            ),
            children: [
              isTopPlacement && textLabel,
              dots,
              !isTopPlacement && textLabel
            ]
          }
        ),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(isBusy && "opacity-40"),
            ...isBusy ? { inert: true } : {},
            children
          }
        )
      ]
    }
  );
}
var filledBrandStyles = [
  "bg-sapphire-brand-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-brand-hover-background",
  "active:bg-sapphire-brand-pressed-background [&.kb-active]:bg-sapphire-brand-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus"
];
var filledJouleStyles = [
  "bg-sapphire-joule-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-joule-hover-background",
  "active:bg-sapphire-joule-pressed-background [&.kb-active]:bg-sapphire-joule-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus"
];
var outlineBrandStyles = [
  "bg-transparent text-sapphire-brand-foreground border border-sapphire-brand-foreground",
  "hover:bg-sapphire-neutral-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-brand-pressed-foreground [&.kb-active]:text-sapphire-brand-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
];
var outlineJouleStyles = [
  "bg-transparent text-sapphire-joule-foreground border border-sapphire-joule-foreground",
  "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background hover:border-sapphire-joule-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-joule-pressed-foreground [&.kb-active]:text-sapphire-joule-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
];
var buttonVariants = cva(
  [
    // Base styles
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        // --- Primary (filled) ---
        ["Primary" /* Primary */]: filledBrandStyles,
        ["PrimaryJoule" /* PrimaryJoule */]: filledJouleStyles,
        // --- Secondary (outlined) ---
        ["Secondary" /* Secondary */]: outlineBrandStyles,
        ["SecondaryJoule" /* SecondaryJoule */]: outlineJouleStyles,
        // --- Tertiary (ghost, colored text) ---
        ["Tertiary" /* Tertiary */]: [
          "bg-transparent text-sapphire-brand-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-brand-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
        ],
        ["TertiaryJoule" /* TertiaryJoule */]: [
          "bg-transparent text-sapphire-joule-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-joule-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
        ],
        // --- Neutral (ghost, dark text) ---
        ["Neutral" /* Neutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-sapphire-neutral-foreground-muted",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-neutral-foreground-black [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- SecondaryNeutral (ghost, neutral-black text) ---
        ["SecondaryNeutral" /* SecondaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background [&.kb-active]:bg-sapphire-neutral-pressed-background",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- TertiaryNeutral (ghost, muted neutral text, no background) ---
        ["TertiaryNeutral" /* TertiaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-muted border border-transparent",
          "hover:text-sapphire-neutral-foreground-black",
          "active:text-sapphire-neutral-foreground-black [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ]
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var largeSizeClasses = {
  default: "h-10 px-4 py-3 text-base gap-2 rounded-lg",
  iconOnly: "h-10 w-10 p-0 text-base rounded-lg",
  icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
};
var mediumSizeClasses = {
  default: "h-8 px-3 py-2 text-sm gap-1.5 rounded",
  iconOnly: "h-8 w-8 p-0 text-sm rounded",
  icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
};
var smallSizeClasses = {
  default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
  iconOnly: "h-6 w-6 p-0 text-xs rounded",
  icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
};
var sizeClasses = {
  ["Large" /* Large */]: largeSizeClasses,
  ["Medium" /* Medium */]: mediumSizeClasses,
  ["Small" /* Small */]: smallSizeClasses
};
function getButtonType(type) {
  switch (type) {
    case "Submit" /* Submit */:
    case "Submit":
      return "submit";
    case "Reset" /* Reset */:
    case "Reset":
      return "reset";
    default:
      return "button";
  }
}
function Button({
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  type = "Button" /* Button */,
  disabled = false,
  loading = false,
  icon,
  endIcon,
  iconOnly = false,
  children,
  tooltip,
  states,
  state,
  stateTransition = "slide-up",
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Button" /* Button */,
  accessibilityAttributes,
  name,
  value,
  form,
  onClick,
  onFocus,
  onBlur,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  "data-growing-button": dataGrowingButton,
  ref
}) {
  const buttonRef = React3.useRef(null);
  const hiddenButtonRef = React3.useRef(null);
  const [animationPhase, setAnimationPhase] = React3.useState("idle");
  const [displayedState, setDisplayedState] = React3.useState(
    null
  );
  const [buttonWidth, setButtonWidth] = React3.useState(void 0);
  const currentStateObject = states?.find((s) => s.name === state) ?? states?.[0];
  const isMultiState = states && states.length > 0;
  React3.useEffect(() => {
    if (isMultiState && !displayedState) {
      setDisplayedState(currentStateObject ?? null);
    }
  }, [isMultiState, currentStateObject, displayedState]);
  React3.useEffect(() => {
    if (!isMultiState || !currentStateObject || animationPhase !== "idle") {
      return;
    }
    if (displayedState && displayedState.name !== currentStateObject.name) {
      if (buttonRef.current) {
        setButtonWidth(buttonRef.current.offsetWidth);
      }
      setAnimationPhase("fade-out");
    }
  }, [currentStateObject, displayedState, isMultiState, animationPhase]);
  React3.useEffect(() => {
    if (animationPhase === "fade-out") {
      const timer = setTimeout(() => {
        setAnimationPhase("fade-mid");
      }, 180);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-mid") {
      setDisplayedState(currentStateObject ?? null);
      if (hiddenButtonRef.current) {
        setButtonWidth(hiddenButtonRef.current.offsetWidth);
      }
      const timer = setTimeout(() => {
        setAnimationPhase("fade-in");
      }, 20);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-in") {
      const timer = setTimeout(() => {
        setAnimationPhase("idle");
        setButtonWidth(void 0);
      }, 160);
      return () => clearTimeout(timer);
    }
  }, [animationPhase, currentStateObject]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        buttonRef.current?.focus();
      },
      blur() {
        buttonRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === buttonRef.current;
      },
      get nativeElement() {
        return buttonRef.current;
      }
    }),
    []
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled || loading) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [disabled, loading, onClick]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " ") {
          e.preventDefault();
        }
        buttonRef.current?.classList.add("kb-active");
        if (e.key === " ") {
          onClick?.({
            originalEvent: e,
            isKeyboard: true
          });
        }
      }
    },
    [disabled, loading, onClick]
  );
  const handleKeyUp = React3.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        buttonRef.current?.classList.remove("kb-active");
      }
    },
    []
  );
  const ariaExpanded = accessibilityAttributes?.expanded;
  const ariaHasPopup = accessibilityAttributes?.hasPopup;
  const ariaControls = accessibilityAttributes?.controls;
  const ariaPressed = accessibilityAttributes?.pressed;
  const role = accessibleRole === "Link" /* Link */ || accessibleRole === "Link" ? "link" : void 0;
  const isDisabled = disabled;
  const effectiveIcon = isMultiState ? displayedState?.icon : icon;
  const effectiveEndIcon = isMultiState ? displayedState?.endIcon : endIcon;
  const effectiveChildren = isMultiState ? displayedState?.text : children;
  const effectiveIconOnly = isMultiState ? !displayedState?.text && !!displayedState?.icon : iconOnly;
  const effectivePulsing = isMultiState ? displayedState?.pulsing : false;
  const PulsingText = ({ children: text }) => /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "opacity-50", children: text }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "absolute inset-0 overflow-hidden",
        style: {
          background: "linear-gradient(90deg, transparent 0%, currentColor 50%, transparent 100%)",
          backgroundSize: "200% 100%",
          WebkitBackgroundClip: "text",
          backgroundClip: "text",
          WebkitTextFillColor: "transparent",
          animation: "pulse-sweep 1.5s ease-in-out infinite"
        },
        children: text
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: `
          @keyframes pulse-sweep {
            0% { background-position: 200% 0; }
            100% { background-position: -200% 0; }
          }
        ` })
  ] });
  const getTransitionStyles = (phase, transition) => {
    if (transition === "none" || phase === "idle") {
      return {};
    }
    const baseTransition = "all 100ms ease-out";
    switch (transition) {
      case "slide-up":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(-8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-down":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(-8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-left":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(-12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "slide-right":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(-12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "fade":
        if (phase === "fade-out") return { opacity: 0, transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0 };
        if (phase === "fade-in") return { opacity: 1, transition: baseTransition };
        break;
      case "scale":
        if (phase === "fade-out") return { opacity: 0, transform: "scale(0.75)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "scale(0.75)" };
        if (phase === "fade-in") return { opacity: 1, transform: "scale(1)", transition: baseTransition };
        break;
      case "flip":
        if (phase === "fade-out") return { opacity: 0, transform: "rotateX(90deg)", transition: baseTransition, transformStyle: "preserve-3d" };
        if (phase === "fade-mid") return { opacity: 0, transform: "rotateX(-90deg)", transformStyle: "preserve-3d" };
        if (phase === "fade-in") return { opacity: 1, transform: "rotateX(0)", transition: baseTransition, transformStyle: "preserve-3d" };
        break;
    }
    return {};
  };
  const contentAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  const iconAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: buttonRef,
        type: getButtonType(type),
        disabled: isDisabled,
        "aria-disabled": loading || void 0,
        id,
        name,
        value,
        form,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-expanded": ariaExpanded,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-pressed": ariaPressed,
        "aria-busy": loading || void 0,
        title: tooltip,
        tabIndex,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur: (e) => {
          buttonRef.current?.classList.remove("kb-active");
          onBlur?.(e);
        },
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][effectiveIconOnly ? "iconOnly" : "default"],
          isMultiState && !effectiveIconOnly && "transition-[width] duration-300 ease-out overflow-hidden",
          loading && "relative pointer-events-none",
          className
        ),
        style: {
          ...style,
          ...buttonWidth !== void 0 && !effectiveIconOnly ? { width: buttonWidth } : {}
        },
        "data-testid": dataTestId,
        "data-growing-button": dataGrowingButton,
        children: [
          loading && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "absolute inset-0 flex items-center justify-center z-[1] [&_*]:!text-inherit", inert: true, children: /* @__PURE__ */ jsxRuntime.jsx(BusyIndicator, { size: "S", active: true, delay: 0 }) }),
          effectiveIcon ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? iconAnimationStyles : void 0, children: effectiveIcon }) : null,
          !effectiveIconOnly && effectiveChildren && /* @__PURE__ */ jsxRuntime.jsx("span", { className: loading ? "opacity-40" : void 0, style: isMultiState ? contentAnimationStyles : void 0, children: effectivePulsing ? /* @__PURE__ */ jsxRuntime.jsx(PulsingText, { children: effectiveChildren }) : effectiveChildren }),
          effectiveEndIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? contentAnimationStyles : void 0, children: effectiveEndIcon })
        ]
      }
    ),
    isMultiState && /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: hiddenButtonRef,
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][!currentStateObject?.text && currentStateObject?.icon ? "iconOnly" : "default"],
          "absolute -top-[10000px] -left-[10000px] invisible w-fit"
        ),
        tabIndex: -1,
        "aria-hidden": "true",
        children: [
          currentStateObject?.icon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.icon }),
          currentStateObject?.text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.text }),
          currentStateObject?.endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.endIcon })
        ]
      }
    )
  ] });
}
var listContainerVariants = cva(
  "relative w-full",
  {
    variants: {
      bordered: {
        true: "border border-sapphire-border-primary rounded-lg overflow-hidden",
        false: ""
      }
    },
    defaultVariants: {
      bordered: false
    }
  }
);
var listVariants = cva(
  "w-full m-0 p-0 list-none",
  {
    variants: {
      indent: {
        true: "pl-4",
        false: ""
      },
      separators: {
        true: "[&>li:not(:last-child)]:border-b [&>li:not(:last-child)]:border-sapphire-border-primary",
        false: ""
      }
    },
    defaultVariants: {
      indent: false,
      separators: true
    }
  }
);
var ListContext = React3__namespace.createContext(null);
function useOptionalListContext() {
  return React3.useContext(ListContext);
}
function useListSelection({
  mode,
  controlledKeys,
  defaultKeys = [],
  items,
  getItemKey,
  onSelectionChange
}) {
  const [internalKeys, setInternalKeys] = React3.useState(new Set(defaultKeys));
  const previousKeysRef = React3.useRef(new Set(defaultKeys));
  const isControlled = controlledKeys !== void 0;
  const selectedKeys = isControlled ? new Set(controlledKeys) : internalKeys;
  const getItemByKey = React3.useCallback((key) => {
    if (!items || !getItemKey) return void 0;
    return items.find((item, index) => getItemKey(item, index) === key);
  }, [items, getItemKey]);
  const getSelectedItems = React3.useCallback(() => {
    if (!items || !getItemKey) return [];
    return items.filter((item, index) => selectedKeys.has(getItemKey(item, index)));
  }, [items, getItemKey, selectedKeys]);
  const setSelected = React3.useCallback((key, selected) => {
    const modeStr = String(mode);
    if (modeStr === "None") return;
    const newKeys = new Set(selectedKeys);
    const previousKeys = new Set(selectedKeys);
    if (modeStr === "Single" || modeStr === "SingleStart" || modeStr === "SingleEnd") {
      if (selected) {
        newKeys.clear();
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    } else if (modeStr === "Multiple") {
      if (selected) {
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    }
    if (!isControlled) {
      setInternalKeys(newKeys);
    }
    const targetItem = getItemByKey(key);
    onSelectionChange?.({
      selectedItems: items?.filter((item, index) => getItemKey && newKeys.has(getItemKey(item, index))) ?? [],
      previouslySelectedItems: items?.filter((item, index) => getItemKey && previousKeys.has(getItemKey(item, index))) ?? [],
      selectionComponentPressed: false,
      targetItem,
      key,
      selectedKeys: Array.from(newKeys)
      // Always include the raw keys
    });
    previousKeysRef.current = newKeys;
  }, [mode, selectedKeys, isControlled, getItemByKey, items, getItemKey, onSelectionChange]);
  const toggleSelection = React3.useCallback((key) => {
    setSelected(key, !selectedKeys.has(key));
  }, [selectedKeys, setSelected]);
  const isSelected = React3.useCallback((key) => {
    return selectedKeys.has(key);
  }, [selectedKeys]);
  const selectAll = React3.useCallback(() => {
    if (!items || !getItemKey || String(mode) !== "Multiple") return;
    const allKeys = new Set(items.map(getItemKey));
    if (!isControlled) {
      setInternalKeys(allKeys);
    }
    onSelectionChange?.({
      selectedItems: items,
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: items[0]
    });
  }, [items, getItemKey, mode, isControlled, onSelectionChange, getSelectedItems]);
  const deselectAll = React3.useCallback(() => {
    if (!isControlled) {
      setInternalKeys(/* @__PURE__ */ new Set());
    }
    onSelectionChange?.({
      selectedItems: [],
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: void 0
    });
  }, [isControlled, onSelectionChange, getSelectedItems]);
  return {
    selectedKeys,
    setSelected,
    toggleSelection,
    isSelected,
    selectAll,
    deselectAll,
    getSelectedItems
  };
}
function useListNavigation({
  onFocusItem,
  onSelectItem,
  selectionMode,
  containerRef
}) {
  const [focusedIndex, setFocusedIndex] = React3.useState(-1);
  const getAllNavigable = React3.useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"], button[data-growing-button]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const getAllNavigableExcludingGrowingButton = React3.useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const focusItem = React3.useCallback((index) => {
    const allNavigable = getAllNavigable();
    if (index < 0 || index >= allNavigable.length) return;
    const element = allNavigable[index];
    if (!element) return;
    const isGrowingButton = element.hasAttribute("data-growing-button");
    if (!isGrowingButton) {
      setFocusedIndex(index);
    }
    element.focus();
    onFocusItem?.(index);
  }, [getAllNavigable, onFocusItem]);
  const handleKeyDown = React3.useCallback((e) => {
    const key = e.key;
    if (key === "Tab") {
      return;
    }
    const target = e.target;
    const isInteractiveElement2 = target.tagName === "INPUT" || target.tagName === "TEXTAREA" || target.tagName === "SELECT" || target.tagName === "BUTTON" || target.getAttribute("role") === "button" || target.getAttribute("contenteditable") === "true" || target.closest('button, input, textarea, select, [role="button"], [contenteditable="true"]') !== null;
    if (isInteractiveElement2) {
      if (key === " " || key === "Enter" || key === "Delete") {
        return;
      }
    }
    const allNavigable = getAllNavigable();
    if (allNavigable.length === 0) return;
    const activeElement = document.activeElement;
    let currentIndex = allNavigable.indexOf(activeElement);
    if (currentIndex === -1) {
      currentIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    }
    let newIndex = currentIndex;
    switch (key) {
      case "ArrowDown":
        e.preventDefault();
        newIndex = Math.min(currentIndex + 1, allNavigable.length - 1);
        break;
      case "ArrowUp":
        e.preventDefault();
        newIndex = Math.max(currentIndex - 1, 0);
        break;
      case "Home":
        e.preventDefault();
        const navigableExcludingButton = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButton.length > 0) {
          const firstElement = navigableExcludingButton[0];
          const indexInAll = allNavigable.indexOf(firstElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "End":
        e.preventDefault();
        const navigableExcludingButtonEnd = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButtonEnd.length > 0) {
          const lastElement = navigableExcludingButtonEnd[navigableExcludingButtonEnd.length - 1];
          const indexInAll = allNavigable.indexOf(lastElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "PageDown":
        e.preventDefault();
        const navigableExcludingButtonPageDown = getAllNavigableExcludingGrowingButton();
        const currentInFiltered = navigableExcludingButtonPageDown.indexOf(activeElement);
        if (currentInFiltered !== -1) {
          const newFilteredIndex = Math.min(currentInFiltered + 10, navigableExcludingButtonPageDown.length - 1);
          const targetElement = navigableExcludingButtonPageDown[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageDown.length > 0) {
            const lastElement = navigableExcludingButtonPageDown[navigableExcludingButtonPageDown.length - 1];
            newIndex = allNavigable.indexOf(lastElement);
          }
        }
        break;
      case "PageUp":
        e.preventDefault();
        const navigableExcludingButtonPageUp = getAllNavigableExcludingGrowingButton();
        const currentInFilteredPageUp = navigableExcludingButtonPageUp.indexOf(activeElement);
        if (currentInFilteredPageUp !== -1) {
          const newFilteredIndex = Math.max(currentInFilteredPageUp - 10, 0);
          const targetElement = navigableExcludingButtonPageUp[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageUp.length > 0) {
            const lastFilteredIndex = navigableExcludingButtonPageUp.length - 1;
            const targetIndex = Math.max(lastFilteredIndex - 10, 0);
            const targetElement = navigableExcludingButtonPageUp[targetIndex];
            newIndex = allNavigable.indexOf(targetElement);
          }
        }
        break;
      case " ":
        e.preventDefault();
        const spaceElement = allNavigable[currentIndex];
        if (spaceElement?.getAttribute("role") === "listitem") {
          if (String(selectionMode) !== "None" && String(selectionMode) !== "Delete") {
            onSelectItem?.(currentIndex);
          }
        }
        return;
      case "Delete":
        const deleteElement = allNavigable[currentIndex];
        if (deleteElement?.getAttribute("role") === "listitem" && String(selectionMode) === "Delete") {
          onSelectItem?.(currentIndex);
        }
        return;
      case "Enter":
        return;
      default:
        return;
    }
    if (newIndex !== currentIndex) {
      focusItem(newIndex);
    }
  }, [focusedIndex, getAllNavigable, getAllNavigableExcludingGrowingButton, selectionMode, onSelectItem, focusItem]);
  const registerItem = React3.useCallback((_index, _element) => {
  }, []);
  return {
    focusedIndex,
    setFocusedIndex,
    focusItem,
    handleKeyDown,
    registerItem
  };
}
function ListHeader({ text, children, sticky, className }) {
  if (!text && !children) return null;
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "flex items-center gap-2.5 px-[0.75rem] py-[0.25rem] text-base font-normal text-sapphire-text-primary bg-background",
        sticky && "sticky top-0 z-10 bg-background",
        className
      ),
      role: "heading",
      "aria-level": 2,
      children: children || text
    }
  );
}
function ListLoading({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20",
        className
      ),
      role: "progressbar",
      "aria-label": "Loading",
      children: /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-6 w-6 animate-spin text-sapphire-brand-foreground" })
    }
  );
}
function ListEmpty({ text, children, className }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "flex items-center justify-center px-4 py-12 text-sm text-sapphire-text-tertiary",
        className
      ),
      role: "status",
      children: children || text || "No data"
    }
  );
}
function ListGrowing({
  mode,
  buttonText = "More",
  onLoadMore,
  parentRef,
  loading,
  accessibilityAttributes,
  buttonRef
}) {
  const observerRef = React3.useRef(null);
  const triggerRef = React3.useRef(null);
  React3.useEffect(() => {
    if (String(mode) !== "Scroll" || !triggerRef.current) return;
    observerRef.current = new IntersectionObserver(
      (entries) => {
        const [entry] = entries;
        if (entry.isIntersecting && !loading) {
          onLoadMore?.();
        }
      },
      {
        root: parentRef?.current,
        rootMargin: "100px",
        threshold: 0
      }
    );
    observerRef.current.observe(triggerRef.current);
    return () => {
      observerRef.current?.disconnect();
    };
  }, [mode, onLoadMore, parentRef, loading]);
  if (String(mode) === "Button") {
    return /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center justify-center py-3 border-t border-sapphire-border-primary", children: /* @__PURE__ */ jsxRuntime.jsxs(
      Button,
      {
        ref: buttonRef,
        design: "Tertiary" /* Tertiary */,
        onClick: () => onLoadMore?.(),
        disabled: loading,
        accessibleName: accessibilityAttributes?.name,
        "data-growing-button": "true",
        tabIndex: 0,
        className: "w-full mx-4",
        children: [
          loading ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-4 w-4 animate-spin mr-2" }) : null,
          buttonText
        ]
      }
    ) });
  }
  if (String(mode) === "Scroll") {
    return /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        ref: triggerRef,
        className: "h-1",
        "aria-hidden": "true",
        "data-growing-trigger": true
      }
    );
  }
  return null;
}
function ListInner(props) {
  const {
    ref,
    // Header & Footer
    headerText,
    header,
    // Selection
    selectionMode = "None" /* None */,
    selectedKeys: controlledSelectedKeys,
    defaultSelectedKeys = [],
    onSelectionChange,
    // Data
    items,
    renderItem,
    getItemKey = (item, index) => String(item.id ?? item.key ?? index),
    children,
    // Empty & Loading
    noDataText = "No data available",
    noDataSlot,
    loading = false,
    loadingDelay = 1e3,
    // Growing
    growing = "None" /* None */,
    growingButtonText,
    hasMore = false,
    onLoadMore,
    // Visual
    separators = "All" /* All */,
    indent = false,
    stickyHeader = false,
    stickyFooter = false,
    // Events
    onItemClick,
    onItemDelete,
    onItemFocused,
    // Drag and Drop
    onMoveOver,
    onMove,
    // Accessibility
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    accessibleRole = "List" /* List */,
    accessibilityAttributes,
    // Standard
    className,
    style,
    id: providedId,
    "data-testid": dataTestId,
    "data-ai-context": dataAiContext
  } = props;
  const generatedId = React3.useId();
  const listId = providedId ?? generatedId;
  const containerRef = React3.useRef(null);
  const listRef = React3.useRef(null);
  const growingButtonRef = React3.useRef(null);
  const beforeSentinelRef = React3.useRef(null);
  const afterSentinelRef = React3.useRef(null);
  const [showLoading, setShowLoading] = React3.useState(false);
  const [mediaRange, setMediaRange] = React3.useState("M");
  const itemRegistry = React3.useRef(/* @__PURE__ */ new Map());
  const [previouslyFocusedItemKey, setPreviouslyFocusedItemKey] = React3.useState(null);
  const [draggedKey, setDraggedKey] = React3.useState(null);
  const [dropTargetKey, setDropTargetKey] = React3.useState(null);
  const [dropPlacement, setDropPlacement] = React3.useState(null);
  const selectionState = useListSelection({
    mode: selectionMode,
    controlledKeys: controlledSelectedKeys,
    defaultKeys: defaultSelectedKeys,
    items,
    getItemKey,
    onSelectionChange
  });
  const itemCount = React3.useMemo(() => {
    if (items) return items.length;
    return React3__namespace.Children.count(children);
  }, [items, children]);
  const navigationState = useListNavigation({
    onFocusItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (element?.getAttribute("role") === "listitem" && items) {
        const itemKey = element.getAttribute("data-item-key");
        if (itemKey) {
          const itemIndex = items.findIndex((item, idx) => getItemKey(item, idx) === itemKey);
          if (itemIndex >= 0) {
            onItemFocused?.({ item: items[itemIndex], index: itemIndex });
          }
        }
      }
    },
    onSelectItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (!element || element.getAttribute("role") !== "listitem") return;
      if (!items || !getItemKey) return;
      const itemKey = element.getAttribute("data-item-key");
      if (!itemKey) return;
      const itemIndex = items.findIndex((item2, idx) => getItemKey(item2, idx) === itemKey);
      if (itemIndex < 0) return;
      const item = items[itemIndex];
      const key = getItemKey(item, itemIndex);
      if (String(selectionMode) === "Delete") {
        onItemDelete?.({ item });
      } else {
        selectionState.toggleSelection(key);
      }
    },
    selectionMode,
    containerRef
  });
  React3.useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  React3.useEffect(() => {
    const updateMediaRange = () => {
      const width = window.innerWidth;
      if (width < 600) setMediaRange("S");
      else if (width < 1024) setMediaRange("M");
      else if (width < 1440) setMediaRange("L");
      else setMediaRange("XL");
    };
    updateMediaRange();
    window.addEventListener("resize", updateMediaRange);
    return () => window.removeEventListener("resize", updateMediaRange);
  }, []);
  const registerItem = React3.useCallback((key, index) => {
    itemRegistry.current.set(key, index);
  }, []);
  const unregisterItem = React3.useCallback((key) => {
    itemRegistry.current.delete(key);
  }, []);
  const handleDragStart = React3.useCallback((key, _e) => {
    setDraggedKey(key);
  }, []);
  const handleDragEnd = React3.useCallback((_key, _e) => {
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, []);
  const handleDragOver = React3.useCallback((key, e) => {
    if (draggedKey && key === draggedKey) return;
    setDropTargetKey(key);
    const rect = e.currentTarget.getBoundingClientRect();
    const y = e.clientY - rect.top;
    const height = rect.height;
    if (y < height * 0.33) {
      setDropPlacement("Before");
    } else if (y > height * 0.67) {
      setDropPlacement("After");
    } else {
      setDropPlacement("On");
    }
  }, [draggedKey]);
  const handleDrop = React3.useCallback((targetKey, e) => {
    const targetIndex = itemRegistry.current.get(targetKey);
    if (targetIndex === void 0) return;
    if (!items || !getItemKey) return;
    const targetItem = items[targetIndex];
    if (!targetItem) return;
    const droppedKey = draggedKey || e.dataTransfer.getData("text/plain");
    if (!droppedKey || targetKey === droppedKey) return;
    const placement = dropPlacement || "After";
    const sourceIndex = itemRegistry.current.get(droppedKey);
    const sourceItem = sourceIndex !== void 0 ? items[sourceIndex] : void 0;
    const isMultiple = String(selectionMode) === "Multiple";
    const draggedIsSelected = droppedKey ? selectionState.isSelected(droppedKey) : false;
    const selectedItemsList = isMultiple && draggedIsSelected ? selectionState.getSelectedItems() : void 0;
    const moveDetail = {
      source: {
        element: sourceItem,
        // May be undefined for cross-list drops
        elements: selectedItemsList,
        index: sourceIndex ?? -1
        // -1 indicates cross-list drop
      },
      destination: {
        element: targetItem,
        placement,
        index: targetIndex
      },
      originalEvent: e.nativeEvent
    };
    onMoveOver?.(moveDetail);
    onMove?.(moveDetail);
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, [draggedKey, dropPlacement, items, getItemKey, selectionMode, selectionState, onMoveOver, onMove]);
  const handleForwardTabAfter = React3.useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      const itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      );
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (String(growing) !== "None" && hasMore && growingButtonRef.current) {
      growingButtonRef.current.focus?.();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex + 1; i < allTabbable.length; i++) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (afterSentinelRef.current) {
          afterSentinelRef.current.focus();
        }
      }
    }
  }, [growing, hasMore, navigationState]);
  const handleForwardTabBefore = React3.useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    let itemElement = null;
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      ) || null;
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (itemElement && event.target !== itemElement) {
      itemElement.focus();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex - 1; i >= 0; i--) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (beforeSentinelRef.current) {
          beforeSentinelRef.current.focus();
        }
      }
    }
  }, [navigationState]);
  const handleBeforeSentinelFocus = React3.useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromAfterSentinel = relatedTarget === afterSentinelRef.current;
    const isFromListItem = relatedTarget?.closest('li[role="listitem"], li[role="option"]');
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    if (isFromAfterSentinel || isFromListItem || isFromGrowingButton) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const sentinelIndex = allTabbable.indexOf(beforeSentinelRef.current);
      if (sentinelIndex > 0) {
        allTabbable[sentinelIndex - 1].focus();
      }
    } else {
      e.preventDefault();
      if (containerRef.current) {
        const allNavigable = Array.from(
          containerRef.current.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
          )
        );
        if (allNavigable.length === 0) {
          if (afterSentinelRef.current) {
            afterSentinelRef.current.focus();
          }
          return;
        }
        if (previouslyFocusedItemKey) {
          const targetElement = allNavigable.find(
            (el) => el.getAttribute("data-item-key") === previouslyFocusedItemKey || el.id === previouslyFocusedItemKey
          );
          if (targetElement) {
            targetElement.focus();
            return;
          }
        }
        if (allNavigable[0]) {
          allNavigable[0].focus();
        }
      }
    }
  }, [previouslyFocusedItemKey]);
  const getAllTabbableElements = React3.useCallback(() => {
    return Array.from(
      document.querySelectorAll(
        'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
      )
    );
  }, []);
  const focusNextAfterList = React3.useCallback(() => {
    if (!listRef.current || !afterSentinelRef.current) return false;
    const allTabbable = getAllTabbableElements();
    const sentinelIndex = allTabbable.indexOf(afterSentinelRef.current);
    if (sentinelIndex >= 0) {
      for (let i = sentinelIndex + 1; i < allTabbable.length; i++) {
        if (!containerRef.current?.contains(allTabbable[i])) {
          allTabbable[i].focus();
          return true;
        }
      }
    }
    return false;
  }, [getAllTabbableElements]);
  const enterListAtFocusedItem = React3.useCallback(() => {
    if (!containerRef.current) return false;
    const allNavigable = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
      )
    );
    if (allNavigable.length === 0) {
      beforeSentinelRef.current?.focus();
      return true;
    }
    const focusedIndex = navigationState.focusedIndex;
    const targetIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    if (allNavigable[targetIndex]) {
      allNavigable[targetIndex].focus();
      return true;
    }
    return false;
  }, [navigationState.focusedIndex]);
  const handleAfterSentinelFocus = React3.useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    const isFromOutside = !relatedTarget || !containerRef.current?.contains(relatedTarget);
    if (isFromGrowingButton || isFromOutside) {
      enterListAtFocusedItem();
    } else {
      focusNextAfterList();
    }
  }, [focusNextAfterList, enterListAtFocusedItem]);
  const roleSupportsAriaSelected = React3.useMemo(() => {
    const roleStr = String(accessibleRole);
    return roleStr === "ListBox" /* ListBox */ || roleStr === "listbox" || roleStr === "Tree" /* Tree */ || roleStr === "tree";
  }, [accessibleRole]);
  const contextValue = React3.useMemo(() => ({
    selectionMode,
    selectedKeys: selectionState.selectedKeys,
    onSelect: selectionState.setSelected,
    onToggleSelect: selectionState.toggleSelection,
    isSelected: selectionState.isSelected,
    focusedIndex: navigationState.focusedIndex,
    onFocusItem: navigationState.setFocusedIndex,
    registerItem,
    unregisterItem,
    registerNavigationItem: navigationState.registerItem,
    // Tab navigation
    onForwardTabAfter: handleForwardTabAfter,
    onForwardTabBefore: handleForwardTabBefore,
    separators,
    indent,
    onItemClick: (_key, event) => {
      onItemClick?.({ item: event.currentTarget, originalEvent: event });
    },
    onItemDelete: (key) => {
      const index = itemRegistry.current.get(key);
      if (index !== void 0 && items) {
        onItemDelete?.({ item: items[index] });
        return;
      }
      if (items && getItemKey) {
        const item = items.find((item2, idx) => getItemKey(item2, idx) === key);
        if (item) {
          onItemDelete?.({ item });
        }
      }
    },
    // Drag and drop
    onDragStart: handleDragStart,
    onDragEnd: handleDragEnd,
    onDragOver: handleDragOver,
    onDrop: handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    mediaRange,
    totalItemCount: itemCount,
    getItemIndex: (key) => {
      return itemRegistry.current.get(key) ?? -1;
    },
    // Accessibility
    accessibleRole,
    supportsAriaSelected: roleSupportsAriaSelected
  }), [
    selectionMode,
    selectionState,
    navigationState,
    registerItem,
    unregisterItem,
    handleForwardTabAfter,
    handleForwardTabBefore,
    separators,
    indent,
    onItemClick,
    onItemDelete,
    items,
    getItemKey,
    mediaRange,
    itemCount,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    accessibleRole,
    roleSupportsAriaSelected
  ]);
  React3.useImperativeHandle(ref, () => ({
    // Focus
    focus: () => listRef.current?.focus(),
    focusItem: (index) => navigationState.focusItem(index),
    getFocusedItemIndex: () => navigationState.focusedIndex,
    // Selection
    getSelectedKeys: () => Array.from(selectionState.selectedKeys),
    getSelectedItems: () => selectionState.getSelectedItems(),
    isSelected: selectionState.isSelected,
    select: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, true));
    },
    deselect: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, false));
    },
    toggleSelection: selectionState.toggleSelection,
    selectAll: selectionState.selectAll,
    deselectAll: selectionState.deselectAll,
    // Data
    getItems: () => items ?? [],
    getItemCount: () => itemCount,
    getItemByKey: (key) => items?.find((item, index) => getItemKey(item, index) === key),
    // Scrolling
    scrollToItem: (index, align = "start") => {
      const element = listRef.current?.children[index];
      element?.scrollIntoView({ behavior: "smooth", block: align });
    },
    scrollToTop: () => {
      containerRef.current?.scrollTo({ top: 0, behavior: "smooth" });
    },
    scrollToBottom: () => {
      containerRef.current?.scrollTo({
        top: containerRef.current.scrollHeight,
        behavior: "smooth"
      });
    },
    // Elements
    get listElement() {
      return listRef.current;
    },
    get scrollContainer() {
      return containerRef.current;
    }
  }), [selectionState, navigationState, items, itemCount, getItemKey]);
  const handleLoadMore = React3.useCallback(() => {
    onLoadMore?.({ currentCount: itemCount });
  }, [onLoadMore, itemCount]);
  const getRole = () => {
    const roleStr = String(accessibleRole);
    switch (roleStr) {
      case "Menu":
        return "menu";
      case "Tree":
        return "tree";
      case "ListBox":
        return "listbox";
      default:
        return "list";
    }
  };
  const isEmpty = itemCount === 0 && !loading;
  const renderedItems = React3.useMemo(() => {
    itemRegistry.current.clear();
    if (items && renderItem) {
      return items.map((item, index) => {
        const key = getItemKey(item, index);
        itemRegistry.current.set(key, index);
        return /* @__PURE__ */ jsxRuntime.jsx(React3__namespace.Fragment, { children: renderItem(item, index) }, key);
      });
    }
    if (children) {
      let index = 0;
      React3__namespace.Children.forEach(children, (child) => {
        if (React3__namespace.isValidElement(child) && child.props && typeof child.props.itemKey === "string") {
          itemRegistry.current.set(child.props.itemKey, index);
          index++;
        }
      });
    }
    return children;
  }, [items, renderItem, getItemKey, children]);
  return /* @__PURE__ */ jsxRuntime.jsx(ListContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: containerRef,
      className: cn(
        listContainerVariants({ bordered: false }),
        // Make scrollable when growing mode or sticky header is used
        (String(growing) !== "None" || stickyHeader || stickyFooter) && "overflow-auto",
        // Always prevent horizontal scroll
        "overflow-x-hidden",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-ai-context": dataAiContext,
      onKeyDown: navigationState.handleKeyDown,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          ListHeader,
          {
            text: headerText,
            sticky: stickyHeader,
            children: header
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            ref: beforeSentinelRef,
            tabIndex: 0,
            onFocus: handleBeforeSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "ul",
          {
            ref: listRef,
            id: listId,
            role: getRole(),
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-describedby": accessibleDescriptionRef,
            "aria-busy": loading || void 0,
            "aria-multiselectable": roleSupportsAriaSelected && String(selectionMode) === "Multiple" ? true : void 0,
            className: cn(listVariants({ indent, separators: String(separators) !== "None" })),
            children: !isEmpty && renderedItems
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            ref: afterSentinelRef,
            tabIndex: 0,
            onFocus: handleAfterSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        isEmpty && /* @__PURE__ */ jsxRuntime.jsx(ListEmpty, { text: noDataText, children: noDataSlot }),
        String(growing) !== "None" && hasMore && !loading && /* @__PURE__ */ jsxRuntime.jsx(
          ListGrowing,
          {
            mode: growing,
            buttonText: growingButtonText,
            onLoadMore: handleLoadMore,
            parentRef: containerRef,
            loading,
            accessibilityAttributes: accessibilityAttributes?.growingButton,
            buttonRef: growingButtonRef
          }
        ),
        showLoading && /* @__PURE__ */ jsxRuntime.jsx(ListLoading, {})
      ]
    }
  ) });
}
var List = React3__namespace.memo(ListInner);
List.displayName = "List";
var NotificationListContext = React3__namespace.createContext(null);
function useNotificationListContext() {
  const context = React3.useContext(NotificationListContext);
  if (!context) {
    throw new Error(
      "Notification item components must be used within a NotificationList"
    );
  }
  return context;
}
function useOptionalNotificationListContext() {
  return React3.useContext(NotificationListContext);
}
function NotificationList(props) {
  const {
    children,
    noDataText = "No notifications",
    // Events
    onItemClick,
    onItemClose,
    onItemToggle,
    // Accessibility
    accessibleName = "Notifications",
    accessibleNameRef,
    // Standard
    className,
    style,
    id,
    "data-testid": dataTestId,
    ref
  } = props;
  const listRef = React3.useRef(null);
  const allNavigationItemsRef = React3.useRef([]);
  const updateNavigationItems = React3.useCallback(() => {
    const container = listRef.current?.scrollContainer;
    if (!container) {
      return;
    }
    const items = [];
    const mainList = container.querySelector(":scope > ul");
    if (!mainList) return;
    const topLevelItems = Array.from(mainList.children);
    topLevelItems.forEach((item) => {
      if (!item.hasAttribute("data-notification-item")) {
        return;
      }
      const isGroup = item.hasAttribute("data-notification-group");
      if (isGroup) {
        const groupButton = item.querySelector('[role="button"]');
        if (groupButton) {
          items.push(groupButton);
        }
        const isCollapsed = groupButton?.getAttribute("aria-expanded") === "false";
        if (!isCollapsed) {
          const childList = item.querySelector("ul[id]");
          if (childList) {
            const childItems = childList.querySelectorAll("li[data-notification-item]");
            childItems.forEach((child) => items.push(child));
          }
          const growingBtn = item.querySelector("[data-growing-button]");
          if (growingBtn) {
            items.push(growingBtn);
          }
        }
      } else {
        items.push(item);
      }
    });
    allNavigationItemsRef.current = items;
    items.forEach((item, index) => {
      const newTabIndex = index === 0 ? "0" : "-1";
      item.setAttribute("tabindex", newTabIndex);
    });
  }, []);
  React3.useLayoutEffect(() => {
    updateNavigationItems();
  }, [children, updateNavigationItems]);
  React3.useEffect(() => {
    const container = listRef.current?.scrollContainer;
    if (!container) return;
    updateNavigationItems();
    const observer = new MutationObserver(() => {
      const items = allNavigationItemsRef.current;
      items.forEach((item, index) => {
        const expectedTabIndex = index === 0 ? "0" : "-1";
        const currentTabIndex = item.getAttribute("tabindex");
        if (currentTabIndex !== expectedTabIndex) {
          item.setAttribute("tabindex", expectedTabIndex);
        }
      });
    });
    observer.observe(container, {
      childList: true,
      subtree: true,
      attributes: true,
      attributeFilter: ["tabindex", "aria-expanded"]
    });
    return () => observer.disconnect();
  }, [updateNavigationItems]);
  const findGroupStart = React3.useCallback((items, currentIndex) => {
    let groupStartIndex = 0;
    for (let i = currentIndex - 1; i >= 0; i--) {
      const item = items[i];
      const isGroupBtn = item.getAttribute("role") === "button" && item.closest("[data-notification-group]");
      if (isGroupBtn) {
        groupStartIndex = i + 1;
        break;
      }
    }
    return groupStartIndex;
  }, []);
  const findGroupEnd = React3.useCallback((items, currentIndex) => {
    let groupEndIndex = items.length - 1;
    for (let i = currentIndex + 1; i < items.length; i++) {
      const item = items[i];
      const isGroupBtn = item.getAttribute("role") === "button" && item.closest("[data-notification-group]");
      if (isGroupBtn) {
        groupEndIndex = i - 1;
        break;
      }
    }
    return groupEndIndex;
  }, []);
  const getNextNavigationIndex = React3.useCallback((key, currentIndex, items, isGroupButton, isExpanded) => {
    if (key === "ArrowDown" || key === "ArrowRight") {
      if (key === "ArrowRight" && isGroupButton && isExpanded) {
        return currentIndex + 1;
      }
      return Math.min(items.length - 1, currentIndex + 1);
    }
    if (key === "ArrowUp" || key === "ArrowLeft") {
      return Math.max(0, currentIndex - 1);
    }
    if (key === "Home") {
      return findGroupStart(items, currentIndex);
    }
    if (key === "End") {
      return findGroupEnd(items, currentIndex);
    }
    return currentIndex;
  }, [findGroupStart, findGroupEnd]);
  const handleKeyDown = React3.useCallback((e) => {
    const target = e.target;
    const isGroupButton = target.getAttribute("role") === "button" && target.closest("[data-notification-group]");
    const isNotificationItem = target.hasAttribute("data-notification-item");
    const isGrowingButton = target.hasAttribute("data-growing-button");
    if (!isGroupButton && !isNotificationItem && !isGrowingButton) {
      return;
    }
    const items = allNavigationItemsRef.current;
    const currentIndex = items.indexOf(target);
    if (currentIndex === -1) return;
    const navigableKeys = ["ArrowDown", "ArrowRight", "ArrowUp", "ArrowLeft", "Home", "End"];
    if (!navigableKeys.includes(e.key)) return;
    e.preventDefault();
    const isExpanded = target.getAttribute?.("aria-expanded") === "true";
    const nextIndex = getNextNavigationIndex(e.key, currentIndex, items, isGroupButton, isExpanded);
    if (nextIndex !== currentIndex) {
      const nextItem = items[nextIndex];
      if (nextItem) {
        items.forEach((item, idx) => {
          item.setAttribute("tabindex", idx === nextIndex ? "0" : "-1");
        });
        nextItem.focus();
      }
    }
  }, [getNextNavigationIndex]);
  const contextValue = React3.useMemo(
    () => ({
      onItemClick,
      onItemClose,
      onItemToggle: (detail) => {
        onItemToggle?.(detail);
        setTimeout(updateNavigationItems, 0);
      },
      updateNavigationItems
    }),
    [onItemClick, onItemClose, onItemToggle, updateNavigationItems]
  );
  React3.useImperativeHandle(
    ref,
    () => ({
      focus: () => {
        listRef.current?.focus();
      },
      blur: () => {
        document.activeElement?.blur();
      },
      isFocused: () => {
        return listRef.current?.scrollContainer?.contains(document.activeElement) ?? false;
      },
      get nativeElement() {
        return listRef.current?.scrollContainer ?? null;
      }
    }),
    []
  );
  return /* @__PURE__ */ jsxRuntime.jsx(NotificationListContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsx("div", { onKeyDown: handleKeyDown, className: "[&_ul]:flex [&_ul]:flex-col [&_ul]:gap-2", children: /* @__PURE__ */ jsxRuntime.jsx(
    List,
    {
      ref: listRef,
      noDataText,
      accessibleName,
      accessibleNameRef,
      className,
      style,
      id,
      "data-testid": dataTestId,
      children
    }
  ) }) });
}

// src/components/list/utils/tabbable.ts
var TABBABLE_SELECTOR = [
  "a[href]",
  "area[href]",
  'input:not([disabled]):not([type="hidden"])',
  "select:not([disabled])",
  "textarea:not([disabled])",
  "button:not([disabled])",
  "iframe",
  "object",
  "embed",
  '[tabindex]:not([tabindex="-1"])',
  '[contenteditable]:not([contenteditable="false"])'
].join(",");
function getTabbableElements(container) {
  if (!container) return [];
  const elements = Array.from(
    container.querySelectorAll(TABBABLE_SELECTOR)
  );
  return elements.filter((el) => {
    const tabindex = el.getAttribute("tabindex");
    if (tabindex === "-1") return false;
    const style = window.getComputedStyle(el);
    if (style.display === "none" || style.visibility === "hidden") return false;
    let current = el;
    while (current && current !== container) {
      if (current.getAttribute("aria-hidden") === "true") return false;
      current = current.parentElement;
    }
    return true;
  });
}
function getActiveElement() {
  let activeElement = document.activeElement;
  while (activeElement?.shadowRoot?.activeElement) {
    activeElement = activeElement.shadowRoot.activeElement;
  }
  return activeElement;
}

// src/components/list/hooks/useListItemSelection.ts
var isInteractiveElement = (target, container) => {
  let el = target;
  while (el && el !== container) {
    const tag = el.tagName.toLowerCase();
    if (tag === "input" || tag === "textarea" || tag === "select" || tag === "button" || tag === "a" && el.hasAttribute("href") || el.getAttribute("role") === "button" || el.isContentEditable) {
      return true;
    }
    el = el.parentElement;
  }
  return false;
};
function useListItemSelection(options) {
  const { id, disabled, type, selectionMode, onClick, itemRef } = options;
  const listContext = useOptionalListContext();
  const isSelected = listContext ? listContext.isSelected(id) : false;
  const shouldForwardTabAfter = React3.useCallback(() => {
    if (!itemRef?.current) return true;
    const tabbables = getTabbableElements(itemRef.current);
    if (tabbables.length === 0) return true;
    const activeElement = getActiveElement();
    const lastTabbable = tabbables[tabbables.length - 1];
    return activeElement === lastTabbable;
  }, [itemRef]);
  const shouldForwardTabBefore = React3.useCallback((target) => {
    if (!itemRef?.current) return false;
    if (target === itemRef.current) return true;
    const tabbables = getTabbableElements(itemRef.current);
    if (tabbables.length === 0) return false;
    return target === tabbables[0];
  }, [itemRef]);
  const handleClick = React3.useCallback((e) => {
    if (disabled) return;
    const container = itemRef?.current ?? null;
    const clickTarget = e.target;
    if (clickTarget !== container && isInteractiveElement(clickTarget, container)) {
      return;
    }
    const typeStr = String(type);
    if (typeStr === "Inactive" || typeStr === "Detail") {
      return;
    }
    const modeStr = String(selectionMode);
    if (modeStr !== "None" && modeStr !== "Delete") {
      listContext?.onToggleSelect(id);
    }
    onClick?.(e);
    listContext?.onItemClick?.(id, e);
  }, [disabled, type, selectionMode, listContext, id, onClick, itemRef]);
  const handleKeyDown = React3.useCallback((e) => {
    if (disabled) return;
    const modeStr = String(selectionMode);
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      const target = e.target;
      const tabbables = getTabbableElements(itemRef?.current || null);
      if (e.shiftKey) {
        if (shouldForwardTabBefore(target)) {
          e.preventDefault();
          listContext?.onForwardTabBefore?.(id, e);
        }
      } else {
        if (target === itemRef?.current && tabbables.length > 0) {
          e.preventDefault();
          tabbables[0].focus();
          return;
        }
        if (shouldForwardTabAfter()) {
          e.preventDefault();
          listContext?.onForwardTabAfter?.(id, e);
          return;
        }
      }
      return;
    }
    if (e.key === "Enter" || e.key === " ") {
      const keyTarget = e.target;
      const keyTag = keyTarget.tagName.toLowerCase();
      if (keyTag === "input" || keyTag === "textarea" || keyTag === "select" || keyTarget.isContentEditable || keyTarget.getAttribute("role") === "button" || keyTag === "a" && keyTarget.hasAttribute("href")) {
        return;
      }
      e.preventDefault();
      const typeStr = String(type);
      if (typeStr === "Inactive" || typeStr === "Detail") {
        return;
      }
      if (modeStr !== "None" && modeStr !== "Delete") {
        listContext?.onToggleSelect(id);
      }
      onClick?.(e);
      listContext?.onItemClick?.(id, e);
    }
  }, [disabled, type, selectionMode, listContext, id, onClick, shouldForwardTabAfter, shouldForwardTabBefore, itemRef]);
  const handleSelectionChange = React3.useCallback((checked) => {
    if (disabled) return;
    listContext?.onSelect(id, checked);
  }, [disabled, listContext, id]);
  return {
    isSelected,
    handleClick,
    handleKeyDown,
    handleSelectionChange
  };
}
function useListItemFocus(options) {
  const { itemRef, disabled, selectionMode, accessibleRole, controlledSelected } = options;
  const listContext = useOptionalListContext();
  const [isFocused, setIsFocused] = React3.useState(false);
  const [isFocusedWithin, setIsFocusedWithin] = React3.useState(false);
  const [isActive, setIsActive] = React3.useState(false);
  const getItemIndex = React3.useCallback(() => {
    let rootList = itemRef.current?.parentElement;
    while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
      rootList = rootList.parentElement;
    }
    if (rootList && itemRef.current) {
      const allNavigable = Array.from(
        rootList.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
        )
      );
      return allNavigable.indexOf(itemRef.current);
    }
    return -1;
  }, [itemRef]);
  const handleFocus = React3.useCallback((e) => {
    const isItemItself = e.target === itemRef.current;
    setIsFocused(isItemItself);
    setIsFocusedWithin(true);
    if (listContext) {
      const itemIndex2 = getItemIndex();
      if (itemIndex2 >= 0) {
        listContext.onFocusItem(itemIndex2);
      }
    }
  }, [listContext, getItemIndex, itemRef]);
  const handleBlur = React3.useCallback((e) => {
    const isLeavingItem = !itemRef.current?.contains(e.relatedTarget);
    if (isLeavingItem) {
      setIsFocused(false);
      setIsFocusedWithin(false);
    } else {
      setIsFocused(false);
    }
  }, [itemRef]);
  const handleMouseDown = React3.useCallback(() => {
    setIsActive(true);
  }, []);
  const handleMouseUp = React3.useCallback(() => {
    setIsActive(false);
  }, []);
  const handleMouseDownCapture = React3.useCallback(() => {
    setIsFocusedWithin(true);
  }, []);
  const focus = React3.useCallback(() => {
    itemRef.current?.focus();
  }, [itemRef]);
  const blur = React3.useCallback(() => {
    itemRef.current?.blur();
  }, [itemRef]);
  const modeStr = String(selectionMode);
  const isSelected = controlledSelected ?? (listContext?.isSelected(itemRef.current?.getAttribute("data-item-key") || "") || false);
  const roleStr = String(accessibleRole);
  const supportsAriaSelected = listContext?.supportsAriaSelected ?? false;
  const hasSelection = modeStr !== "None" && modeStr !== "Delete";
  const ariaSelected = roleStr === "Option" ? isSelected : supportsAriaSelected && hasSelection ? isSelected : void 0;
  const itemIndex = getItemIndex();
  const isFocusedItem = listContext?.focusedIndex === itemIndex;
  const isFirstItem = itemIndex === 0;
  const isTabbable = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem;
  const tabbableIndex = isTabbable ? 0 : -1;
  const tabIndex = disabled ? -1 : tabbableIndex;
  return {
    isFocused,
    isFocusedWithin,
    isActive,
    tabIndex,
    ariaSelected,
    handleFocus,
    handleBlur,
    handleMouseDown,
    handleMouseUp,
    handleMouseDownCapture,
    focus,
    blur,
    getItemIndex
  };
}
function useListItemDragDrop(options) {
  const { id, movable, disabled, onDragStart: onDragStartProp } = options;
  const listContext = useOptionalListContext();
  const isDragging = listContext?.draggedKey === id;
  const isDropTarget = listContext?.dropTargetKey === id;
  const dropPlacement = isDropTarget ? listContext?.dropPlacement : null;
  const handleDragStart = React3.useCallback((e) => {
    if (!movable || disabled) {
      e.preventDefault();
      return;
    }
    e.dataTransfer.effectAllowed = "move";
    e.dataTransfer.dropEffect = "move";
    e.dataTransfer.setData("text/plain", id);
    listContext?.onDragStart?.(id, e);
    onDragStartProp?.();
  }, [movable, disabled, id, listContext, onDragStartProp]);
  const handleDragEnd = React3.useCallback((e) => {
    listContext?.onDragEnd?.(id, e);
  }, [id, listContext]);
  const handleDragOver = React3.useCallback((e) => {
    e.preventDefault();
    e.dataTransfer.dropEffect = "move";
    listContext?.onDragOver?.(id, e);
  }, [id, listContext]);
  const handleDrop = React3.useCallback((e) => {
    e.preventDefault();
    listContext?.onDrop?.(id, e);
  }, [id, listContext]);
  const handleDragEnter = React3.useCallback((e) => {
    e.preventDefault();
  }, []);
  const handleDragLeave = React3.useCallback((_e) => {
  }, []);
  return {
    isDragging,
    isDropTarget,
    dropPlacement: dropPlacement || null,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDragEnter,
    handleDragLeave,
    handleDrop
  };
}
function useListItemRegistration(options) {
  const { itemRef, itemKey } = options;
  const listContext = useOptionalListContext();
  const itemIndexRef = React3.useRef(-1);
  React3.useEffect(() => {
    if (listContext && itemKey) {
      let rootList = itemRef.current?.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(itemRef.current);
        if (index >= 0) {
          itemIndexRef.current = index;
          listContext.registerItem(itemKey, index);
          listContext.registerNavigationItem(index, itemRef.current);
        }
      }
    }
    return () => {
      if (listContext && itemKey && itemIndexRef.current >= 0) {
        listContext.unregisterItem(itemKey);
        listContext.registerNavigationItem(itemIndexRef.current, null);
        itemIndexRef.current = -1;
      }
    };
  }, [listContext, itemKey, itemRef]);
  return {};
}

// src/components/list/useListItem.ts
function useListItem(options = {}) {
  const {
    itemKey,
    disabled = false,
    movable = false,
    type = "Active",
    accessibleRole = "ListItem",
    selected: controlledSelected,
    onClick,
    onDragStart
  } = options;
  const itemRef = React3.useRef(null);
  const generatedId = React3.useId();
  const id = itemKey ?? generatedId;
  const listContext = useOptionalListContext();
  const selectionMode = listContext?.selectionMode ?? "None" /* None */;
  useListItemRegistration({ itemRef, itemKey });
  const selection = useListItemSelection({
    id,
    disabled,
    type,
    selectionMode,
    onClick,
    itemRef
  });
  const focus = useListItemFocus({
    itemRef,
    disabled,
    selectionMode,
    accessibleRole,
    controlledSelected
  });
  const handleMouseDown = React3.useCallback(() => {
    const typeStr = String(type);
    if (typeStr === "Active" || typeStr === "Navigation") {
      focus.handleMouseDown();
    }
  }, [type, focus]);
  const handleMouseUp = React3.useCallback(() => {
    focus.handleMouseUp();
  }, [focus]);
  const dragDrop = useListItemDragDrop({
    id,
    movable,
    disabled,
    onDragStart
  });
  return {
    // Refs
    itemRef,
    id,
    // State from sub-hooks (controlledSelected overrides context-based selection)
    isSelected: controlledSelected ?? selection.isSelected,
    isFocused: focus.isFocused,
    isFocusedWithin: focus.isFocusedWithin,
    isActive: focus.isActive,
    isDragging: dragDrop.isDragging,
    isDropTarget: dragDrop.isDropTarget,
    dropPlacement: dragDrop.dropPlacement,
    // Context values
    selectionMode,
    // Computed values
    tabIndex: focus.tabIndex,
    ariaSelected: focus.ariaSelected,
    // Event handlers (composed from sub-hooks)
    handleClick: selection.handleClick,
    handleKeyDown: selection.handleKeyDown,
    handleMouseDown,
    handleMouseUp,
    handleMouseDownCapture: focus.handleMouseDownCapture,
    handleFocus: focus.handleFocus,
    handleBlur: focus.handleBlur,
    handleDragStart: dragDrop.handleDragStart,
    handleDragEnd: dragDrop.handleDragEnd,
    handleDragOver: dragDrop.handleDragOver,
    handleDragEnter: dragDrop.handleDragEnter,
    handleDragLeave: dragDrop.handleDragLeave,
    handleDrop: dragDrop.handleDrop,
    // Selection handlers
    handleSelectionChange: selection.handleSelectionChange,
    // Helper functions
    focus: focus.focus,
    blur: focus.blur,
    getItemIndex: focus.getItemIndex
  };
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData9,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React3.useRef(null);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData9 || pathData9.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData9.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M438 224q11 0 18.5 7.5T464 250v140q0 38-26 64t-64 26H106q-38 0-64-26t-26-64V122q0-38 26-64t64-26h237q11 0 18 7.5t7 18.5-7 18-18 7H106q-16 0-27.5 11.5T67 122v268q0 16 11.5 27.5T106 429h268q16 0 27.5-11.5T413 390V250q0-11 7-18.5t18-7.5zm32-192q11 0 18.5 7.5T496 58q0 10-7 17L257 312q-6 8-18 8-10 0-18-8l-70-71q-7-7-7-18t7.5-18 18.5-7 18 7l51 53L452 40q8-8 18-8z"
];
var CompleteIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "CompleteIcon", iconName: "complete" }
);
var pathData2 = [
  "M390 480H122q-38 0-64-26t-26-64V122q0-38 26-64t64-26h268q38 0 64 26t26 64v268q0 38-26 64t-64 26zM122 83q-17 0-28 11t-11 28v268q0 17 11 28t28 11h268q17 0 28-11t11-28V122q0-17-11-28t-28-11H122z"
];
var BorderIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "BorderIcon", iconName: "border" }
);
var pathData3 = [
  "M390 32q38 0 64 26t26 64v268q0 38-26 64t-64 26H122q-38 0-64-26t-26-64V122q0-38 26-64t64-26h268zm39 90q0-17-11-28t-28-11H122q-17 0-28 11t-11 28v268q0 17 11 28t28 11h268q17 0 28-11t11-28V122zm-77 38v192H160V160h192z"
];
var TriStateIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "TriStateIcon", iconName: "tri-state" }
);
function CheckBox({
  checked: controlledChecked,
  defaultChecked = false,
  indeterminate = false,
  disabled = false,
  readonly = false,
  displayOnly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  wrappingType = "Normal" /* Normal */,
  valueState = "None",
  valueStateMessage,
  name,
  value = "on",
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  accessibilityAttributes,
  ref
}) {
  const inputRef = React3.useRef(null);
  const rootRef = React3.useRef(null);
  const generatedId = React3.useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = React3.useState(defaultChecked);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const isActive = isChecked || indeterminate;
  const doToggle = React3.useCallback(
    (originalEvent) => {
      if (disabled || readonly || displayOnly) return;
      const newChecked = !isChecked;
      if (!isControlled) {
        setInternalChecked(newChecked);
      }
      onChange?.({
        checked: newChecked,
        indeterminate: false,
        originalEvent
      });
    },
    [disabled, readonly, displayOnly, isChecked, isControlled, onChange]
  );
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      toggle() {
        doToggle();
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked, doToggle]
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) {
        e.preventDefault();
        return;
      }
      doToggle(e);
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const handleMouseDown = React3.useCallback(
    (e) => {
      if (disabled || displayOnly) {
        e.preventDefault();
      }
    },
    [disabled, displayOnly]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
      }
    },
    [disabled, readonly, displayOnly]
  );
  const handleKeyUp = React3.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doToggle(e);
      }
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const labelContent = children || text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly && !displayOnly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const getValueStateBorderClass2 = () => {
    if (isActive) return "";
    const borderMap = {
      Positive: "border-sapphire-positive",
      Negative: "border-sapphire-negative",
      Critical: "border-sapphire-warning",
      Information: "border-sapphire-info"
    };
    return borderMap[valueState] ?? "";
  };
  const getValueStateCheckedClass = () => {
    if (!isActive) return "";
    const checkedMap = {
      Positive: "bg-sapphire-positive border-sapphire-positive",
      Negative: "bg-sapphire-negative border-sapphire-negative",
      Critical: "bg-sapphire-warning border-sapphire-warning",
      Information: "bg-sapphire-info border-sapphire-info"
    };
    return checkedMap[valueState] ?? "bg-sapphire-canvas-primary border-sapphire-border-secondary";
  };
  const renderDisplayOnlyIcon = () => {
    const cls = "size-4 text-foreground";
    if (indeterminate) return /* @__PURE__ */ jsxRuntime.jsx(TriStateIcon, { className: cls });
    if (isChecked) return /* @__PURE__ */ jsxRuntime.jsx(CompleteIcon, { className: cls });
    return /* @__PURE__ */ jsxRuntime.jsx(BorderIcon, { className: cls });
  };
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        ref: inputRef,
        type: "checkbox",
        name,
        value,
        checked: isChecked,
        disabled: disabled || displayOnly,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: rootRef,
        id,
        role: "checkbox",
        "aria-checked": indeterminate ? "mixed" : isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          valueState !== "None" && valueStateMessage ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null,
          accessibilityAttributes?.describedBy
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || displayOnly || void 0,
        "aria-readonly": readonly || void 0,
        "aria-invalid": valueState === "Negative" || void 0,
        "aria-required": required || void 0,
        tabIndex: tabIndex !== void 0 ? tabIndex : disabled || displayOnly ? -1 : 0,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          // Touch area: Cozy = 44px, Compact = 32px
          isSmall ? labelContent ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2" : labelContent ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]",
          "text-sm",
          "rounded-lg",
          // Focus ring via pseudo-element for asymmetric insets
          "outline-none",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          labelContent ? isSmall ? "after:inset-x-0 after:inset-y-0" : "after:left-0 after:right-[2px] after:inset-y-[5px]" : "after:inset-[5px] after:rounded-lg",
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          displayOnly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          displayOnly ? renderDisplayOnlyIcon() : /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: cn(
                "relative inline-flex items-center justify-center",
                "shrink-0",
                hasWrappingLabel && "self-start",
                isSmall ? hasWrappingLabel ? "size-4 rounded p-[2px] mt-0.5" : "size-4 rounded p-[2px]" : hasWrappingLabel ? "size-[22px] rounded-md p-[3px] mt-0.5" : "size-[22px] rounded-md p-[3px]",
                "transition-colors",
                // Border style: dashed for readonly, solid otherwise
                readonly ? "border border-dashed" : "border border-solid",
                // --- Unchecked state ---
                !isActive && !readonly && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
                // --- Checked / indeterminate state (not readonly) ---
                // None value state: blue icon on white bg; value states: white icon on colored bg
                isActive && !readonly && valueState === "None" && "text-sapphire-shell-button-fg-selected",
                isActive && !readonly && valueState !== "None" && "text-sapphire-neutral-foreground-white",
                !readonly && getValueStateCheckedClass(),
                !readonly && getValueStateBorderClass2(),
                // --- Hover (unchecked, not readonly/disabled, no value state) ---
                !isActive && isInteractive && valueState === "None" && "group-hover:border-primary group-hover:bg-sapphire-shell-button-bg-selected",
                // --- Hover (checked, not readonly/disabled, no value state) ---
                isActive && isInteractive && valueState === "None" && "group-hover:bg-sapphire-shell-button-bg-selected group-hover:border-primary",
                // --- Read-only state: gray bg + dashed secondary border ---
                readonly && "bg-sapphire-background-secondary border-sapphire-border-secondary",
                // --- Read-only icon color: secondary-foreground (#353C4A) ---
                readonly && isActive && "text-secondary-foreground"
              ),
              children: indeterminate ? (
                // Solid square for tristate per Figma: 12px large, 8px small
                /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn(
                  "block bg-current",
                  isSmall ? "size-2" : "size-3"
                ) })
              ) : isChecked ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Check, { className: isSmall ? "size-3" : "size-4", strokeWidth: 3 }) : null
            }
          ),
          labelContent && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" && valueStateMessage && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage }),
          accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}
var radioGroups = /* @__PURE__ */ new Map();
function getOrderedGroupEntries(name) {
  const entries = radioGroups.get(name) ?? [];
  const connectedEntries = entries.filter((entry) => entry.getNode()?.isConnected);
  if (connectedEntries.length !== entries.length) {
    if (connectedEntries.length === 0) {
      radioGroups.delete(name);
    } else {
      radioGroups.set(name, connectedEntries);
    }
  }
  return [...connectedEntries].sort((entryA, entryB) => {
    const nodeA = entryA.getNode();
    const nodeB = entryB.getNode();
    if (!nodeA || !nodeB || nodeA === nodeB) {
      return 0;
    }
    const position = nodeA.compareDocumentPosition(nodeB);
    if (position & Node.DOCUMENT_POSITION_FOLLOWING) {
      return -1;
    }
    if (position & Node.DOCUMENT_POSITION_PRECEDING) {
      return 1;
    }
    return 0;
  });
}
function updateGroupTabIndex(name) {
  const entries = getOrderedGroupEntries(name);
  if (entries.length === 0) {
    return;
  }
  const focusableEntries = entries.filter((entry) => !entry.isDisabled());
  const activeEntry = focusableEntries.find((entry) => entry.isChecked()) ?? focusableEntries[0] ?? null;
  entries.forEach((entry) => {
    entry.setTabIndex(!entry.isDisabled() && entry === activeEntry ? 0 : -1);
  });
}
function registerGroupEntry(name, entry) {
  const existingEntries = radioGroups.get(name) ?? [];
  const nextEntries = existingEntries.filter((existingEntry) => existingEntry.id !== entry.id);
  nextEntries.push(entry);
  radioGroups.set(name, nextEntries);
  updateGroupTabIndex(name);
}
function unregisterGroupEntry(name, id) {
  const existingEntries = radioGroups.get(name);
  if (!existingEntries) {
    return;
  }
  const nextEntries = existingEntries.filter((entry) => entry.id !== id);
  if (nextEntries.length === 0) {
    radioGroups.delete(name);
    return;
  }
  radioGroups.set(name, nextEntries);
  updateGroupTabIndex(name);
}
function getAdjacentGroupEntry(name, currentId, direction) {
  const focusableEntries = getOrderedGroupEntries(name).filter((entry) => !entry.isDisabled());
  if (focusableEntries.length <= 1) {
    return null;
  }
  const currentIndex = focusableEntries.findIndex((entry) => entry.id === currentId);
  const baseIndex = currentIndex >= 0 ? currentIndex : 0;
  const nextIndex = (baseIndex + direction + focusableEntries.length) % focusableEntries.length;
  return focusableEntries[nextIndex] ?? null;
}
function moveSelectionInGroup(name, currentId, direction, originalEvent) {
  const nextEntry = getAdjacentGroupEntry(name, currentId, direction);
  if (!nextEntry) {
    return;
  }
  nextEntry.focus();
  if (!nextEntry.isReadonly()) {
    nextEntry.select(originalEvent);
    return;
  }
  updateGroupTabIndex(name);
}
function uncheckOthersInGroup(groupName, currentId) {
  getOrderedGroupEntries(groupName).forEach((entry) => {
    if (entry.id !== currentId && entry.isChecked()) {
      entry.setChecked(false);
    }
  });
}
function getKeyNavigationDirection(key) {
  if (key === " " || key === "Enter") return "select";
  const isRTL = document.documentElement.dir === "rtl";
  if (key === "ArrowDown" || key === "Down" || key === "ArrowRight" && !isRTL || key === "Right" && !isRTL || key === "ArrowLeft" && isRTL || key === "Left" && isRTL) {
    return 1;
  }
  if (key === "ArrowUp" || key === "Up" || key === "ArrowLeft" && !isRTL || key === "Left" && !isRTL || key === "ArrowRight" && isRTL || key === "Right" && isRTL) {
    return -1;
  }
  return null;
}
function getEffectiveTabIndex(disabled, tabIndex, groupName, groupTabIndex) {
  if (disabled) return -1;
  if (tabIndex !== void 0) return tabIndex;
  if (groupName) return groupTabIndex ?? 0;
  return 0;
}
function getValueStateBorderClass(valueState) {
  const borderMap = {
    ["Positive" /* Positive */]: "border-sapphire-positive",
    ["Negative" /* Negative */]: "border-sapphire-negative",
    ["Critical" /* Critical */]: "border-sapphire-warning",
    ["Information" /* Information */]: "border-sapphire-info"
  };
  return borderMap[valueState] ?? "";
}
function getValueStateCheckedBorderClass(valueState) {
  const checkedBorderMap = {
    ["Positive" /* Positive */]: "border-sapphire-positive",
    ["Negative" /* Negative */]: "border-sapphire-negative",
    ["Critical" /* Critical */]: "border-sapphire-warning",
    ["Information" /* Information */]: "border-sapphire-info"
  };
  return checkedBorderMap[valueState] ?? "border-sapphire-border-secondary";
}
function getInnerDotColorClass(valueState) {
  const dotMap = {
    ["Positive" /* Positive */]: "bg-sapphire-positive",
    ["Negative" /* Negative */]: "bg-sapphire-negative",
    ["Critical" /* Critical */]: "bg-sapphire-warning",
    ["Information" /* Information */]: "bg-sapphire-info"
  };
  return dotMap[valueState] ?? "bg-sapphire-text-accent";
}
function getSizeClasses(isSmall, hasLabel) {
  if (isSmall) {
    return hasLabel ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2";
  }
  return hasLabel ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]";
}
function getFocusRingClasses(hasLabel, isSmall) {
  if (!hasLabel) return "after:inset-[5px] after:rounded-lg";
  if (isSmall) return "after:inset-x-0 after:inset-y-0";
  return "after:left-0 after:right-[2px] after:inset-y-[5px]";
}
var valueStateI18nKeys = {
  ["Negative" /* Negative */]: "RADIOBUTTON_VALUE_STATE_ERROR",
  ["Critical" /* Critical */]: "RADIOBUTTON_VALUE_STATE_WARNING",
  ["Positive" /* Positive */]: "RADIOBUTTON_VALUE_STATE_SUCCESS",
  ["Information" /* Information */]: "RADIOBUTTON_VALUE_STATE_INFORMATION"
};
function RadioCircle({
  isReadonly,
  isSmall,
  isChecked,
  hasWrappingLabel,
  isInteractive,
  valueState,
  svgR
}) {
  if (isReadonly) {
    return /* @__PURE__ */ jsxRuntime.jsxs(
      "span",
      {
        className: cn(
          "relative inline-flex items-center justify-center shrink-0",
          isSmall ? "size-4" : "size-[22px]",
          hasWrappingLabel && "self-start",
          isSmall && hasWrappingLabel && "mt-0.5"
        ),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            "svg",
            {
              className: "absolute inset-0 w-full h-full overflow-visible",
              focusable: "false",
              "aria-hidden": "true",
              children: /* @__PURE__ */ jsxRuntime.jsx(
                "circle",
                {
                  cx: "50%",
                  cy: "50%",
                  r: svgR,
                  fill: "var(--background-secondary)",
                  stroke: "var(--border-secondary)",
                  strokeWidth: "1",
                  strokeDasharray: "4, 2"
                }
              )
            }
          ),
          isChecked && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: cn(
                "relative rounded-full",
                isSmall ? "size-2" : "size-3",
                "bg-sapphire-text-secondary"
              )
            }
          )
        ]
      }
    );
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      className: cn(
        "relative inline-flex items-center justify-center shrink-0 rounded-full",
        isSmall ? "size-4" : "size-[22px]",
        hasWrappingLabel && "self-start",
        isSmall && hasWrappingLabel && "mt-0.5",
        "transition-colors",
        "border border-solid",
        !isChecked && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
        isChecked && getValueStateCheckedBorderClass(valueState),
        isChecked && "bg-sapphire-canvas-primary",
        !isChecked && getValueStateBorderClass(valueState),
        isInteractive && valueState === "None" /* None */ && !isChecked && "group-hover:border-sapphire-chrome-button-fg-selected group-hover:bg-sapphire-chrome-button-bg-selected",
        isInteractive && valueState === "None" /* None */ && isChecked && "group-hover:border-sapphire-border-accent group-hover:bg-sapphire-chrome-button-bg-selected"
      ),
      children: isChecked && /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: cn(
            "rounded-full",
            isSmall ? "size-2" : "size-3",
            getInnerDotColorClass(valueState)
          )
        }
      )
    }
  );
}
function RadioButton({
  checked: controlledChecked,
  defaultChecked = false,
  disabled = false,
  readonly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  valueState = "None" /* None */,
  valueStateMessage,
  name,
  value = "",
  wrappingType = "Normal" /* Normal */,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  ref
}) {
  const inputRef = React3.useRef(null);
  const rootRef = React3.useRef(null);
  const generatedId = React3.useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = React3.useState(defaultChecked);
  const [groupTabIndex, setGroupTabIndex] = React3.useState(null);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const setChecked = React3.useCallback(
    (checked) => {
      if (!isControlled) {
        setInternalChecked(checked);
      }
    },
    [isControlled]
  );
  const doSelect = React3.useCallback(
    (originalEvent) => {
      if (disabled || readonly || isChecked) return;
      if (name) {
        uncheckOthersInGroup(name, id);
      }
      if (!isControlled) {
        setInternalChecked(true);
      }
      onChange?.({
        checked: true,
        value,
        originalEvent
      });
    },
    [disabled, readonly, isChecked, isControlled, name, id, onChange, value]
  );
  React3.useEffect(() => {
    if (!name) {
      setGroupTabIndex(null);
      return;
    }
    const entry = {
      id,
      getNode: () => rootRef.current,
      isDisabled: () => disabled,
      isReadonly: () => readonly,
      isChecked: () => isChecked,
      setChecked,
      focus: () => rootRef.current?.focus(),
      select: doSelect,
      setTabIndex: setGroupTabIndex
    };
    registerGroupEntry(name, entry);
    return () => {
      unregisterGroupEntry(name, id);
    };
  }, [name, id, disabled, readonly, isChecked, setChecked, doSelect]);
  React3.useEffect(() => {
    if (!name) {
      return;
    }
    updateGroupTabIndex(name);
  }, [name, isChecked, disabled]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked]
  );
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled || readonly) {
        e.preventDefault();
        return;
      }
      doSelect(e);
    },
    [disabled, readonly, doSelect]
  );
  const handleMouseDown = React3.useCallback(
    (e) => {
      if (disabled) {
        e.preventDefault();
      }
    },
    [disabled]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled || readonly) return;
      const direction = getKeyNavigationDirection(e.key);
      if (name && (direction === 1 || direction === -1)) {
        e.preventDefault();
        moveSelectionInGroup(name, id, direction, e);
        return;
      }
      if (direction === "select") {
        e.preventDefault();
      }
    },
    [disabled, readonly, name, id]
  );
  const handleKeyUp = React3.useCallback(
    (e) => {
      if (disabled || readonly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doSelect(e);
      }
    },
    [disabled, readonly, doSelect]
  );
  const labelContent = children ?? text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const effectiveTabIndex = getEffectiveTabIndex(disabled, tabIndex, name, groupTabIndex);
  const { t } = reactI18next.useTranslation("fx");
  const valueStateI18nKey = valueStateI18nKeys[valueState];
  const defaultValueStateText = valueStateI18nKey ? t(valueStateI18nKey) : void 0;
  const svgR = isSmall ? 7.5 : 10.5;
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        ref: inputRef,
        type: "radio",
        id,
        name,
        value,
        checked: isChecked,
        disabled,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: rootRef,
        role: "radio",
        "aria-checked": isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          accessibleDescriptionRef || null,
          valueState !== "None" /* None */ ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || void 0,
        "aria-invalid": valueState === "Negative" /* Negative */ || void 0,
        "aria-required": required || void 0,
        tabIndex: effectiveTabIndex,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        "data-radio-group": name,
        "data-radio-id": id,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          getSizeClasses(isSmall, !!labelContent),
          "text-sm",
          "rounded-lg",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          getFocusRingClasses(!!labelContent, isSmall),
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            RadioCircle,
            {
              isReadonly: readonly,
              isSmall,
              isChecked,
              hasWrappingLabel,
              isInteractive,
              valueState,
              svgR
            }
          ),
          labelContent && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" /* None */ && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage || defaultValueStateText }),
          accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}
var pathData4 = [
  "M454 109q11 0 18.5 7t7.5 18-7.5 18.5T454 160h-19v294q0 24-17 41t-41 17H135q-24 0-41-17t-17-41V160H58q-11 0-18.5-7.5T32 134t7.5-18 18.5-7h70V58q0-24 17-41t41-17h140q24 0 41 17t17 41v51h70zm-275 0h154V58q0-7-7-7H186q-7 0-7 7v51zm205 51H128v294q0 7 7 7h242q7 0 7-7V160zm-186 64q11 0 18.5 7.5T224 250v140q0 11-7.5 18.5T198 416t-18-7.5-7-18.5V250q0-11 7-18.5t18-7.5zm116 0q11 0 18 7.5t7 18.5v140q0 11-7 18.5t-18 7.5-18.5-7.5T288 390V250q0-11 7.5-18.5T314 224z"
];
var DeleteIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "DeleteIcon", iconName: "delete" }
);
function ListSelectionControl({
  selectionMode,
  position,
  isSelected,
  disabled = false,
  itemText = "item",
  deleteButton,
  onSelectionChange,
  onDeleteClick,
  ref
}) {
  const modeStr = String(selectionMode);
  const showSelectionBefore = position === "before" && (modeStr === "Multiple" || modeStr === "SingleStart");
  const showSelectionAfter = position === "after" && (modeStr === "SingleEnd" || modeStr === "Delete");
  const showRadio = modeStr === "SingleStart" || modeStr === "SingleEnd" || modeStr === "Single";
  const showCheckbox = modeStr === "Multiple";
  const showDelete = modeStr === "Delete";
  if (!showSelectionBefore && !showSelectionAfter) {
    return null;
  }
  const handleDeleteClick = (detail) => {
    detail.originalEvent.stopPropagation();
    onDeleteClick?.();
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "span",
    {
      ref,
      className: "shrink-0 pl-3 pr-1 flex items-center",
      onClick: (e) => e.stopPropagation(),
      children: [
        showCheckbox && /* @__PURE__ */ jsxRuntime.jsx(
          CheckBox,
          {
            checked: isSelected,
            onChange: (e) => onSelectionChange(e.checked),
            disabled,
            accessibleName: `Select ${itemText}`,
            tabIndex: -1
          }
        ),
        showRadio && !showDelete && /* @__PURE__ */ jsxRuntime.jsx(
          RadioButton,
          {
            checked: isSelected,
            onChange: (e) => onSelectionChange(e.checked),
            disabled,
            accessibleName: `Select ${itemText}`,
            tabIndex: -1
          }
        ),
        showDelete && (deleteButton ? React3__namespace.cloneElement(deleteButton, {
          onClick: handleDeleteClick
        }) : /* @__PURE__ */ jsxRuntime.jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsxRuntime.jsx(DeleteIcon, { className: "h-4 w-4" }),
            onClick: handleDeleteClick,
            accessibleName: "Delete item",
            tooltip: "Delete",
            tabIndex: -1
          }
        ))
      ]
    }
  );
}
function DropIndicator({ visible, placement, orientation = "vertical", className, ref }) {
  if (!visible) return null;
  const isVertical = orientation === "vertical";
  const isBefore = placement === "Before";
  const isOn = placement === "On";
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      className: cn(
        "absolute pointer-events-none z-20",
        isVertical ? cn(
          "left-0 right-0 h-0.5 bg-primary",
          isBefore ? "-top-px" : isOn ? "inset-0 h-full bg-primary/10 border-2 border-primary border-dashed" : "-bottom-px"
        ) : cn(
          "top-0 bottom-0 w-0.5 bg-primary",
          isBefore ? "-left-px" : isOn ? "inset-0 w-full bg-primary/10 border-2 border-primary border-dashed" : "-right-px"
        ),
        className
      ),
      "aria-hidden": "true",
      "data-drop-indicator": true,
      "data-placement": placement
    }
  );
}

// src/components/list/shared-styles.ts
var LIST_ITEM_BASE_CLASSES = [
  "relative",
  "flex",
  "items-center",
  "w-full",
  "min-h-[2.75rem]",
  "transition-colors",
  "outline-none"
];
var LIST_ITEM_FOCUS_CLASSES = "ring-2 ring-inset ring-sapphire-border-focus z-10";
function ListItemBase(props) {
  const {
    ref,
    itemKey,
    type = "Active" /* Active */,
    disabled = false,
    movable = false,
    accessibleRole = "ListItem" /* ListItem */,
    accessibleNameRef,
    selected: controlledSelected,
    focused: externalFocused,
    onClick,
    onDragStart,
    onDelete,
    deleteButton,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const listContext = useOptionalListContext();
  const effectiveAccessibleRole = React3.useMemo(() => {
    if (accessibleRole !== void 0 && accessibleRole !== "ListItem" /* ListItem */) {
      return accessibleRole;
    }
    const parentRole = String(listContext?.accessibleRole || "List");
    if (accessibleRole === "ListItem" /* ListItem */ && parentRole === "List") {
      return "ListItem" /* ListItem */;
    }
    switch (parentRole) {
      case "ListBox":
        return "Option" /* Option */;
      case "Menu":
        return "MenuItem" /* MenuItem */;
      case "Tree":
        return "TreeItem" /* TreeItem */;
      default:
        return "ListItem" /* ListItem */;
    }
  }, [accessibleRole, listContext?.accessibleRole]);
  const listItem = useListItem({
    itemKey,
    disabled,
    movable,
    type,
    accessibleRole: effectiveAccessibleRole,
    selected: controlledSelected,
    onClick,
    onDragStart
  });
  const handleDeleteClick = React3.useCallback(() => {
    if (itemKey) {
      listContext?.onItemDelete?.(itemKey);
    }
    onDelete?.();
  }, [itemKey, listContext, onDelete]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        listItem.focus();
      },
      blur() {
        listItem.blur();
      },
      isFocused() {
        return document.activeElement === listItem.itemRef.current;
      },
      get nativeElement() {
        return listItem.itemRef.current;
      }
    }),
    [listItem]
  );
  const renderState = {
    isSelected: listItem.isSelected,
    isFocused: listItem.isFocused,
    isFocusedWithin: listItem.isFocusedWithin,
    isActive: listItem.isActive,
    isDragging: listItem.isDragging,
    isDropTarget: listItem.isDropTarget,
    dropPlacement: listItem.dropPlacement,
    selectionMode: String(listItem.selectionMode)
  };
  const modeStr = String(listItem.selectionMode);
  const position = modeStr === "SingleEnd" || modeStr === "Delete" ? "after" : "before";
  const shouldShowControl = modeStr !== "None";
  const content = typeof children === "function" ? children(renderState) : children;
  const resolvedRole = (() => {
    const roleStr = String(effectiveAccessibleRole);
    switch (roleStr) {
      case "Option":
        return "option";
      case "MenuItem":
        return "menuitem";
      case "TreeItem":
        return "treeitem";
      case "Group":
        return "group";
      case "None":
        return void 0;
      default:
        return "listitem";
    }
  })();
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "li",
    {
      ref: listItem.itemRef,
      id: listItem.id,
      role: resolvedRole,
      "aria-labelledby": accessibleNameRef,
      "aria-selected": listItem.ariaSelected,
      "aria-disabled": disabled || void 0,
      "aria-setsize": listContext?.totalItemCount,
      "aria-posinset": itemKey && listContext ? listContext.getItemIndex(itemKey) + 1 || void 0 : void 0,
      tabIndex: listItem.tabIndex,
      draggable: movable,
      onClick: listItem.handleClick,
      onKeyDown: listItem.handleKeyDown,
      onMouseDownCapture: listItem.handleMouseDownCapture,
      onMouseDown: listItem.handleMouseDown,
      onMouseUp: listItem.handleMouseUp,
      onMouseLeave: listItem.handleMouseUp,
      onFocus: listItem.handleFocus,
      onBlur: listItem.handleBlur,
      onDragStart: listItem.handleDragStart,
      onDragEnd: listItem.handleDragEnd,
      onDragOver: listItem.handleDragOver,
      onDragEnter: listItem.handleDragEnter,
      onDragLeave: listItem.handleDragLeave,
      onDrop: listItem.handleDrop,
      className: cn(
        // Base item styles - container layout (shared with group headers)
        LIST_ITEM_BASE_CLASSES,
        "text-sm text-sapphire-text-primary",
        // State classes
        listItem.isSelected && "!bg-sapphire-brand-selected-background",
        listItem.isSelected && !(externalFocused || listItem.isFocused) && "!border-sapphire-border-accent",
        (externalFocused || listItem.isFocused) && LIST_ITEM_FOCUS_CLASSES,
        listItem.isActive && "bg-muted/80",
        listItem.isDragging && "opacity-50",
        movable && "cursor-grab",
        listItem.isDragging && "cursor-grabbing",
        disabled && "opacity-50 pointer-events-none cursor-not-allowed",
        // Interactive states
        (type === "Active" /* Active */ || type === "Navigation" /* Navigation */) && "cursor-pointer hover:bg-[var(--table-row-hover)] active:bg-[var(--neutral-pressed-background-2)]",
        type === "Inactive" /* Inactive */ && "cursor-default",
        // User's custom classes
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-notification-item": props["data-notification-item"] || void 0,
      "data-notification-group": props["data-notification-group"] || void 0,
      "data-item-key": itemKey,
      "data-dragging": listItem.isDragging || void 0,
      children: [
        listItem.dropPlacement && /* @__PURE__ */ jsxRuntime.jsx(DropIndicator, { visible: true, placement: listItem.dropPlacement }),
        shouldShowControl && position === "before" && /* @__PURE__ */ jsxRuntime.jsx(
          ListSelectionControl,
          {
            selectionMode: listItem.selectionMode,
            position: "before",
            isSelected: listItem.isSelected,
            disabled,
            deleteButton,
            onSelectionChange: listItem.handleSelectionChange,
            onDeleteClick: handleDeleteClick
          }
        ),
        content,
        shouldShowControl && position === "after" && /* @__PURE__ */ jsxRuntime.jsx(
          ListSelectionControl,
          {
            selectionMode: listItem.selectionMode,
            position: "after",
            isSelected: listItem.isSelected,
            disabled,
            deleteButton,
            onSelectionChange: listItem.handleSelectionChange,
            onDeleteClick: handleDeleteClick
          }
        )
      ]
    }
  );
}
ListItemBase.displayName = "ListItemBase";
var SAFE_PROTOCOLS = ["http:", "https:", "mailto:", "tel:"];
var isSafeUrl = (url) => {
  try {
    const parsed = new URL(url, window.location.href);
    return SAFE_PROTOCOLS.includes(parsed.protocol);
  } catch {
    return false;
  }
};
var computeRel = (href, target, rel) => {
  if (rel) return rel;
  if (target === "_blank" && href) {
    try {
      const url = new URL(href, window.location.href);
      if (url.origin !== window.location.origin) {
        return "noreferrer noopener";
      }
    } catch {
    }
  }
  return void 0;
};
var linkVariants = cva(
  [
    // Layout
    "inline-flex items-center gap-0.5",
    // Typography — Figma: '72' Regular, 14px, line-height 20px
    "text-sm leading-5 font-normal cursor-pointer",
    // Transition
    "transition-colors",
    // Focus box: constant padding + transparent border to prevent layout shift.
    // Figma focus spec: border-focus, rounded 4px, px-4px py-2px.
    // Negative margins compensate for the padding+border so the link
    // sits flush in normal state.
    "px-1 py-0.5 -mx-1.5 -my-1",
    "border-2 border-transparent rounded-[4px]",
    "focus-visible:outline-none focus-visible:border-sapphire-border-focus"
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: [
          "text-sapphire-text-accent",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ],
        ["Subtle" /* Subtle */]: [
          "text-secondary-foreground",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ],
        ["Emphasized" /* Emphasized */]: [
          "text-sapphire-text-accent font-semibold",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ]
      },
      interactiveAreaSize: {
        ["Normal" /* Normal */]: "",
        ["Large" /* Large */]: "leading-6"
      },
      wrappingType: {
        ["None" /* None */]: "whitespace-nowrap max-w-full",
        ["Normal" /* Normal */]: "whitespace-normal break-words"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      interactiveAreaSize: "Normal" /* Normal */,
      wrappingType: "None" /* None */
    }
  }
);
function Link({
  design = "Default" /* Default */,
  interactiveAreaSize = "Normal" /* Normal */,
  wrappingType = "None" /* None */,
  href,
  target,
  rel,
  disabled = false,
  icon,
  endIcon,
  tooltip,
  children,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleRole = "Link" /* Link */,
  accessibilityAttributes,
  onClick,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  ref
}) {
  const linkRef = React3.useRef(null);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        linkRef.current?.focus();
      },
      blur() {
        linkRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === linkRef.current;
      },
      get nativeElement() {
        return linkRef.current;
      }
    }),
    []
  );
  const { t } = reactI18next.useTranslation("fx");
  const linkTypeText = React3.useMemo(() => {
    if (design === "Subtle" /* Subtle */) return t("LINK_SUBTLE");
    if (design === "Emphasized" /* Emphasized */) return t("LINK_EMPHASIZED");
    return void 0;
  }, [design, t]);
  const computedRel = React3.useMemo(
    () => computeRel(href, target, rel),
    [href, target, rel]
  );
  const effectiveTabIndex = disabled ? -1 : tabIndex ?? 0;
  const handleClick = React3.useCallback(
    (e) => {
      if (disabled) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        altKey: e.altKey,
        ctrlKey: e.ctrlKey,
        metaKey: e.metaKey,
        shiftKey: e.shiftKey,
        isKeyboard: false
      });
    },
    [disabled, onClick]
  );
  const handleMouseDown = React3.useCallback(
    (_e) => {
      if (!disabled) {
        linkRef.current?.focus();
      }
    },
    [disabled]
  );
  const handleKeyDown = React3.useCallback(
    (e) => {
      if (disabled) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " " && accessibleRole !== "Button" /* Button */) {
          return;
        }
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          altKey: e.altKey,
          ctrlKey: e.ctrlKey,
          metaKey: e.metaKey,
          shiftKey: e.shiftKey,
          isKeyboard: true
        });
        if (href && !disabled && isSafeUrl(href)) {
          window.location.href = href;
        }
      }
    },
    [disabled, onClick, href, accessibleRole]
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "a",
    {
      ref: linkRef,
      id,
      href: disabled ? void 0 : href,
      target,
      rel: computedRel,
      role: accessibleRole === "Button" /* Button */ ? "button" : void 0,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-describedby": accessibleDescription,
      "aria-disabled": disabled || void 0,
      "aria-expanded": accessibilityAttributes?.expanded,
      "aria-haspopup": accessibilityAttributes?.hasPopup,
      "aria-current": accessibilityAttributes?.current || void 0,
      title: tooltip,
      tabIndex: effectiveTabIndex,
      onClick: handleClick,
      onMouseDown: handleMouseDown,
      onKeyDown: handleKeyDown,
      className: cn(
        linkVariants({
          design,
          interactiveAreaSize,
          wrappingType
        }),
        disabled && "pointer-events-none cursor-default opacity-[var(--disabled-opacity)]",
        "group",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-shrink-0 size-3", "data-part": "icon", children: icon }),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            className: cn(
              wrappingType === "None" /* None */ && "overflow-hidden text-ellipsis min-w-0",
              design === "Subtle" /* Subtle */ && "underline decoration-dotted underline-offset-[0.25em]",
              (design === "Default" /* Default */ || design === "Emphasized" /* Emphasized */) && "group-hover:underline group-hover:decoration-dotted group-hover:underline-offset-[0.25em]"
            ),
            "data-part": "text",
            children
          }
        ),
        endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-shrink-0 size-3", "data-part": "end-icon", children: endIcon }),
        linkTypeText && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "sr-only", "data-part": "link-type", children: linkTypeText })
      ]
    }
  );
}
var designClasses = {
  // Semantic
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning",
  ["None" /* None */]: "bg-sapphire-background-quaternary text-sapphire-text-secondary",
  ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info",
  // Jobs
  ["Draft" /* Draft */]: "bg-sapphire-background-quaternary text-sapphire-text-primary",
  ["Active" /* Active */]: "bg-sapphire-status-tag-active-bg text-sapphire-status-tag-active-text",
  ["Waiting" /* Waiting */]: "bg-transparent border-sapphire-joule-foreground text-sapphire-joule-foreground",
  ["Paused" /* Paused */]: "bg-transparent border-sapphire-border-active text-sapphire-text-primary"
};
var borderedDesigns = /* @__PURE__ */ new Set(["Waiting" /* Waiting */, "Paused" /* Paused */]);
function IconPositive({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M6 0C9.31371 0 12 2.68629 12 6C12 9.31371 9.31371 12 6 12C2.68629 12 0 9.31371 0 6C0 2.68629 2.68629 0 6 0ZM6 1.125C3.30761 1.125 1.125 3.30761 1.125 6C1.125 8.69239 3.30761 10.875 6 10.875C8.69239 10.875 10.875 8.69239 10.875 6C10.875 3.30761 8.69239 1.125 6 1.125ZM7.2605 4.32129C7.46269 4.08553 7.81788 4.05836 8.05371 4.2605C8.28947 4.46269 8.31664 4.81788 8.1145 5.05371L5.8645 7.67871C5.65457 7.92349 5.26776 7.93817 5.03979 7.71021L3.91479 6.58521C3.69513 6.36554 3.69513 6.00946 3.91479 5.78979C4.13446 5.57013 4.49054 5.57013 4.71021 5.78979L5.40527 6.48486L7.2605 4.32129Z", fill: "currentColor" }) });
}
function IconNegative({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M6 0C9.31371 0 12 2.68629 12 6C12 9.31371 9.31371 12 6 12C2.68629 12 0 9.31371 0 6C0 2.68629 2.68629 0 6 0ZM6 1.125C3.30761 1.125 1.125 3.30761 1.125 6C1.125 8.69239 3.30761 10.875 6 10.875C8.69239 10.875 10.875 8.69239 10.875 6C10.875 3.30761 8.69239 1.125 6 1.125ZM7.28979 3.91479C7.50948 3.69531 7.8656 3.69519 8.08521 3.91479C8.30463 4.13442 8.30463 4.49058 8.08521 4.71021L6.79541 6L8.08521 7.28979C8.30463 7.50942 8.30463 7.86558 8.08521 8.08521C7.8656 8.30481 7.50948 8.30469 7.28979 8.08521L6 6.79541L4.71021 8.08521C4.4906 8.30481 4.13448 8.30469 3.91479 8.08521C3.69513 7.86554 3.69513 7.50946 3.91479 7.28979L5.20459 6L3.91479 4.71021C3.69513 4.49054 3.69513 4.13446 3.91479 3.91479C4.13448 3.69531 4.4906 3.69519 4.71021 3.91479L6 5.20459L7.28979 3.91479Z", fill: "currentColor" }) });
}
function IconCritical({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M4.86848 1.39197C5.37707 0.536388 6.61611 0.535876 7.12507 1.39124L11.8118 9.26917C12.3322 10.144 11.7019 11.2526 10.6839 11.2526H1.31403C0.296556 11.2523 -0.333519 10.1439 0.186105 9.26917L4.86848 1.39197ZM6.15827 1.96619C6.08557 1.84406 5.90795 1.84406 5.83528 1.96619L1.1529 9.84412C1.07905 9.96896 1.16895 10.1273 1.31403 10.1276H10.6839C10.8293 10.1276 10.9193 9.96909 10.845 9.84412L6.15827 1.96619ZM5.99568 7.86951C6.40981 7.86951 6.74555 8.2054 6.74568 8.61951C6.74568 9.03372 6.40989 9.36951 5.99568 9.36951C5.5816 9.36935 5.24568 9.03362 5.24568 8.61951C5.2458 8.2055 5.58168 7.86967 5.99568 7.86951ZM5.99568 3.74451C6.30626 3.74451 6.55805 3.99646 6.55818 4.30701V6.55701C6.55818 6.86767 6.30634 7.11951 5.99568 7.11951C5.68515 7.11935 5.43318 6.86757 5.43318 6.55701V4.30701C5.4333 3.99656 5.68523 3.74467 5.99568 3.74451Z", fill: "currentColor" }) });
}
function IconNone({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M9.2644 0.00533307C9.57207 0.047783 9.78719 0.331953 9.74487 0.639611L9.41895 3.00021H11.4375C11.7481 3.00021 11.9999 3.25215 12 3.56271C12 3.87337 11.7482 4.12521 11.4375 4.12521H9.2644L8.74658 7.87521H11.4375C11.7481 7.87521 11.9999 8.12715 12 8.43771C12 8.74837 11.7482 9.00021 11.4375 9.00021H8.59204L8.24487 11.5146C8.20241 11.8224 7.91833 12.0375 7.6106 11.9951C7.30288 11.9526 7.08768 11.6685 7.13013 11.3608L7.45605 9.00021H3.71704L3.36987 11.5146C3.32741 11.8224 3.04333 12.0375 2.7356 11.9951C2.42788 11.9526 2.21268 11.6685 2.25513 11.3608L2.58105 9.00021H0.5625C0.25184 9.00021 0 8.74837 0 8.43771C0.000115934 8.12715 0.251911 7.87521 0.5625 7.87521H2.7356L3.25342 4.12521H0.5625C0.25184 4.12521 0 3.87337 0 3.56271C0.000115741 3.25215 0.251911 3.00021 0.5625 3.00021H3.40796L3.75513 0.485802C3.79769 0.178191 4.08175 -0.0370864 4.3894 0.00533307C4.69707 0.047783 4.91219 0.331953 4.86987 0.639611L4.54395 3.00021H8.28296L8.63013 0.485802C8.67269 0.178191 8.95675 -0.0370864 9.2644 0.00533307ZM4.3894 4.12521L3.87158 7.87521H7.6106L8.12842 4.12521H4.3894Z", fill: "currentColor" }) });
}
function IconInformation({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M6 0C9.31371 0 12 2.68629 12 6C12 9.31371 9.31371 12 6 12C2.68629 12 0 9.31371 0 6C0 2.68629 2.68629 0 6 0ZM6 1.125C3.30761 1.125 1.125 3.30761 1.125 6C1.125 8.69239 3.30761 10.875 6 10.875C8.69239 10.875 10.875 8.69239 10.875 6C10.875 3.30761 8.69239 1.125 6 1.125ZM6 5.625C6.31066 5.625 6.5625 5.87684 6.5625 6.1875V8.8125C6.5625 9.12316 6.31066 9.375 6 9.375C5.68934 9.375 5.4375 9.12316 5.4375 8.8125V6.1875C5.4375 5.87684 5.68934 5.625 6 5.625ZM6 3C6.41421 3 6.75 3.33579 6.75 3.75C6.75 4.16421 6.41421 4.5 6 4.5C5.58579 4.5 5.25 4.16421 5.25 3.75C5.25 3.33579 5.58579 3 6 3Z", fill: "currentColor" }) });
}
function IconDraft({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M6.95035 10.7802C7.25505 10.72 7.55071 10.9186 7.61099 11.2233C7.67115 11.528 7.47326 11.8236 7.16861 11.8839C6.41066 12.0339 5.58574 12.0339 4.82779 11.8839C4.52314 11.8236 4.32526 11.528 4.38541 11.2233C4.44569 10.9186 4.74135 10.72 5.04605 10.7802C5.66097 10.9018 6.33543 10.9018 6.95035 10.7802ZM1.16348 8.55216C1.42163 8.37933 1.77142 8.44855 1.94425 8.7067C2.30023 9.23833 2.75807 9.69618 3.28971 10.0522C3.54785 10.225 3.61707 10.5748 3.44425 10.8329C3.27141 11.0908 2.92228 11.1601 2.66422 10.9875C2.01015 10.5496 1.44683 9.98625 1.00894 9.33219C0.836303 9.07412 0.905558 8.72499 1.16348 8.55216ZM10.0522 8.7067C10.225 8.44855 10.5748 8.37933 10.8329 8.55216C11.0908 8.72499 11.1601 9.07412 10.9875 9.33219C10.5496 9.98625 9.98625 10.5496 9.33218 10.9875C9.07412 11.1601 8.72499 11.0908 8.55216 10.8329C8.37933 10.5748 8.44855 10.225 8.7067 10.0522C9.23833 9.69618 9.69618 9.23833 10.0522 8.7067ZM5.8107 2.2482C6.12136 2.2482 6.3732 2.50004 6.3732 2.8107V5.71036L8.39981 7.23966C8.64749 7.4268 8.69659 7.77913 8.50968 8.02701C8.32256 8.27499 7.96957 8.32473 7.72159 8.13761L5.47159 6.43985C5.33084 6.33359 5.24826 6.16723 5.2482 5.99088V2.8107C5.2482 2.50004 5.50004 2.2482 5.8107 2.2482ZM0.112459 4.82779C0.172763 4.52314 0.468445 4.32526 0.773104 4.38541C1.07781 4.44569 1.27641 4.74135 1.21622 5.04605C1.09457 5.66097 1.09457 6.33543 1.21622 6.95035C1.27641 7.25505 1.07781 7.55072 0.773104 7.61099C0.468445 7.67115 0.172763 7.47326 0.112459 7.16861C-0.0374864 6.41066 -0.0374864 5.58574 0.112459 4.82779ZM11.2233 4.38541C11.528 4.32526 11.8236 4.52314 11.8839 4.82779C12.0339 5.58574 12.0339 6.41066 11.8839 7.16861C11.8236 7.47326 11.528 7.67115 11.2233 7.61099C10.9186 7.55072 10.72 7.25505 10.7802 6.95035C10.9018 6.33543 10.9018 5.66097 10.7802 5.04605C10.72 4.74135 10.9186 4.44569 11.2233 4.38541ZM2.66422 1.00894C2.92228 0.836303 3.27141 0.905558 3.44425 1.16348C3.61707 1.42163 3.54785 1.77142 3.28971 1.94425C2.75807 2.30023 2.30023 2.75807 1.94425 3.28971C1.77142 3.54785 1.42163 3.61707 1.16348 3.44425C0.905558 3.27141 0.836303 2.92228 1.00894 2.66422C1.44683 2.01015 2.01015 1.44683 2.66422 1.00894ZM8.55216 1.16348C8.72499 0.905558 9.07412 0.836303 9.33218 1.00894C9.98625 1.44683 10.5496 2.01015 10.9875 2.66422C11.1601 2.92228 11.0908 3.27141 10.8329 3.44425C10.5748 3.61707 10.225 3.54785 10.0522 3.28971C9.69618 2.75807 9.23833 2.30023 8.7067 1.94425C8.44855 1.77142 8.37933 1.42163 8.55216 1.16348ZM7.16861 0.112459C7.47326 0.172763 7.67115 0.468445 7.61099 0.773104C7.55071 1.07781 7.25505 1.27641 6.95035 1.21622C6.33543 1.09457 5.66097 1.09457 5.04605 1.21622C4.74135 1.27641 4.44569 1.07781 4.38541 0.773104C4.32526 0.468445 4.52314 0.172763 4.82779 0.112459C5.58574 -0.0374864 6.41066 -0.0374864 7.16861 0.112459Z", fill: "currentColor" }) });
}
function IconActive({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M3.1875 7.5C3.49815 7.5 3.74998 7.75185 3.75 8.0625C3.75 8.37316 3.49816 8.625 3.1875 8.625H1.58863C2.64516 9.99216 4.32371 10.875 6 10.875C8.06547 10.875 10.0091 9.71735 10.9314 7.81714C11.067 7.53778 11.4034 7.42097 11.6829 7.5564C11.9623 7.69199 12.0791 8.02843 11.9436 8.30786C10.8259 10.6109 8.47945 12 6 12C4.17405 12 2.38418 11.155 1.125 9.81226V10.6875C1.125 10.9982 0.873163 11.25 0.562502 11.25C0.251842 11.25 2.3614e-06 10.9982 2.33724e-06 10.6875V8.0625C2.0437e-05 7.75185 0.251853 7.5 0.562502 7.5H3.1875ZM6 0C7.85123 0 9.62811 0.77453 10.875 2.12549V1.3125C10.875 1.00184 11.1268 0.750006 11.4375 0.75C11.7482 0.75 12 1.00184 12 1.3125V3.9375C12 4.24816 11.7482 4.5 11.4375 4.5H8.8125C8.50185 4.49999 8.25 4.24816 8.25 3.9375C8.25 3.62684 8.50185 3.37501 8.8125 3.375H10.4465C9.41906 1.95995 7.75372 1.125 6 1.125C4.00756 1.12501 2.01177 2.37221 1.06128 4.19751C0.917727 4.47287 0.577954 4.57972 0.302493 4.43628C0.0271292 4.29273 -0.0797061 3.95295 0.063723 3.67749C1.19827 1.49894 3.56754 8.85687e-06 6 0Z", fill: "currentColor" }) });
}
function IconWaiting({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M2.0625 0C2.37316 0 2.625 0.25184 2.625 0.5625V0.98584C3.64695 0.5891 4.80205 0.69103 5.76855 1.32568C6.97313 2.12873 8.41746 2.40325 9.67456 1.56519C9.84892 1.47343 10.0596 1.47946 10.2283 1.5813C10.3968 1.68316 10.5 1.86554 10.5 2.0625V7.65015C10.4999 8.03865 10.1148 8.22234 9.82104 8.38477L9.82178 8.3855C8.51084 9.13071 6.87227 9.28435 5.5166 8.36133C5.40391 8.28602 5.29061 8.21148 5.17822 8.13574C4.43437 7.63393 3.4632 7.71249 2.625 8.26172V11.4375C2.625 11.7482 2.37316 12 2.0625 12C1.75184 12 1.5 11.7482 1.5 11.4375V0.5625C1.5 0.25184 1.75184 0 2.0625 0ZM5.17822 2.28223C4.39017 1.7506 3.41821 1.74435 2.625 2.23901V6.98877C3.6321 6.55035 4.80315 6.52518 5.80737 7.20264L6.14575 7.42896L6.15015 7.43188C7.13228 8.10042 8.4041 7.94545 9.375 7.34033V2.96338C7.94167 3.52223 6.41144 3.11409 5.17822 2.28223Z", fill: "currentColor" }) });
}
function IconPaused({ className }) {
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { className, viewBox: "0 0 12 12", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M6 0C9.31371 0 12 2.68629 12 6C12 9.31371 9.31371 12 6 12C2.68629 12 0 9.31371 0 6C0 2.68629 2.68629 0 6 0ZM6 1.125C3.30761 1.125 1.125 3.30761 1.125 6C1.125 8.69239 3.30761 10.875 6 10.875C8.69239 10.875 10.875 8.69239 10.875 6C10.875 3.30761 8.69239 1.125 6 1.125ZM4.6875 3.75C4.99816 3.75 5.25 4.00184 5.25 4.3125V7.6875C5.25 7.99816 4.99816 8.25 4.6875 8.25C4.37684 8.25 4.125 7.99816 4.125 7.6875V4.3125C4.125 4.00184 4.37684 3.75 4.6875 3.75ZM7.3125 3.75C7.62316 3.75 7.875 4.00184 7.875 4.3125V7.6875C7.875 7.99816 7.62316 8.25 7.3125 8.25C7.00184 8.25 6.75 7.99816 6.75 7.6875V4.3125C6.75 4.00184 7.00184 3.75 7.3125 3.75Z", fill: "currentColor" }) });
}
var defaultIconMap = {
  ["Positive" /* Positive */]: IconPositive,
  ["Negative" /* Negative */]: IconNegative,
  ["Critical" /* Critical */]: IconCritical,
  ["None" /* None */]: IconNone,
  ["Information" /* Information */]: IconInformation,
  ["Draft" /* Draft */]: IconDraft,
  ["Active" /* Active */]: IconActive,
  ["Waiting" /* Waiting */]: IconWaiting,
  ["Paused" /* Paused */]: IconPaused
};
function Tag({
  design = "None" /* None */,
  children,
  icon,
  hideIcon = false,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const tagRef = React3.useRef(null);
  const d = design;
  React3.useImperativeHandle(
    ref,
    () => ({
      focus() {
        tagRef.current?.focus();
      },
      blur() {
        tagRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === tagRef.current;
      },
      get nativeElement() {
        return tagRef.current;
      }
    }),
    []
  );
  let displayIcon = null;
  if (icon) {
    displayIcon = icon;
  } else if (!hideIcon) {
    const DefaultIcon = defaultIconMap[d];
    if (DefaultIcon) {
      displayIcon = /* @__PURE__ */ jsxRuntime.jsx(DefaultIcon, { className: "h-3 w-3" });
    }
  }
  const hasBorder = borderedDesigns.has(d);
  const screenReaderText = "Tag";
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "span",
    {
      ref: tagRef,
      id,
      className: cn(
        // Layout: h-6 (24px), px-2 (8px), py-1 (4px), gap-1, rounded-sm (4px)
        "inline-flex items-center h-6 px-2 py-1 gap-1 rounded-sm",
        // Typography: semibold 12px/16px
        "text-xs leading-4 font-semibold whitespace-nowrap",
        // Border: always reserve space, only show for bordered designs
        hasBorder ? "border" : "border border-transparent",
        // Design-specific colors
        designClasses[d] ?? designClasses["None" /* None */],
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("span", { className: "sr-only", children: screenReaderText }),
        displayIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-flex items-center justify-center flex-shrink-0 size-3", "data-part": "icon", children: displayIcon }),
        children && /* @__PURE__ */ jsxRuntime.jsx("span", { "data-part": "text", children })
      ]
    }
  );
}
var pathData5 = [
  "M292 256l117 116q7 7 7 18 0 12-7.5 19t-18.5 7q-10 0-18-8L256 292 140 408q-8 8-18 8-11 0-18.5-7.5T96 390q0-10 8-18l116-116-116-116q-8-8-8-18 0-11 7.5-18.5T122 96t18 7l116 117 116-117q7-7 18-7t18.5 7.5T416 122t-7 18z"
];
var DeclineIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData5, ...props })
  ),
  { displayName: "DeclineIcon", iconName: "decline" }
);
var pathData6 = [
  "M64 192q26 0 45 19t19 45-19 45-45 19-45-19-19-45 19-45 45-19zm192 0q26 0 45 19t19 45-19 45-45 19-45-19-19-45 19-45 45-19zm192 0q26 0 45 19t19 45-19 45-45 19-45-19-19-45 19-45 45-19z"
];
var OverflowIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData6, ...props })
  ),
  { displayName: "OverflowIcon", iconName: "overflow" }
);
var titleVariants = cva("text-[13px] leading-4", {
  variants: {
    read: {
      true: "font-normal",
      false: "font-bold"
    },
    wrapping: {
      ["None" /* None */]: "line-clamp-2",
      ["Normal" /* Normal */]: "break-words"
    }
  },
  defaultVariants: {
    read: false,
    wrapping: "None" /* None */
  }
});
var stateIconVariants = cva("h-4 w-4 shrink-0", {
  variants: {
    state: {
      ["None" /* None */]: "hidden",
      ["Positive" /* Positive */]: "text-sapphire-positive",
      ["Critical" /* Critical */]: "text-sapphire-warning",
      ["Negative" /* Negative */]: "text-sapphire-negative",
      ["Information" /* Information */]: "text-sapphire-info"
    }
  },
  defaultVariants: {
    state: "None" /* None */
  }
});
function getStateText(state) {
  const normalized = state;
  switch (normalized) {
    case "Positive" /* Positive */:
      return "Positive";
    case "Critical" /* Critical */:
      return "Critical";
    case "Negative" /* Negative */:
      return "Negative";
    case "Information" /* Information */:
      return "Information";
    default:
      return "";
  }
}
function StateIcon({ state }) {
  const normalized = state;
  const baseClassName = stateIconVariants({ state: normalized });
  switch (normalized) {
    case "Positive" /* Positive */:
      return /* @__PURE__ */ jsxRuntime.jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsxRuntime.jsx("circle", { cx: "8", cy: "8", r: "8", fill: "currentColor" }),
        /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M11.3333 5.5L6.75 10.0833L4.66667 8", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })
      ] });
    case "Critical" /* Critical */:
      return /* @__PURE__ */ jsxRuntime.jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M6.86078 2.75995L1.21078 12.2499C1.09697 12.4547 1.03736 12.6863 1.03772 12.9218C1.03809 13.1574 1.09843 13.3888 1.21288 13.5933C1.32733 13.7977 1.49214 13.9681 1.69198 14.0881C1.89182 14.2081 2.11982 14.2736 2.35245 14.2783H13.6524C13.8851 14.2736 14.1131 14.2081 14.3129 14.0881C14.5128 13.9681 14.6776 13.7977 14.792 13.5933C14.9065 13.3888 14.9668 13.1574 14.9672 12.9218C14.9675 12.6863 14.9079 12.4547 14.7941 12.2499L9.14412 2.75995C9.02679 2.56103 8.86051 2.39698 8.66099 2.28264C8.46148 2.1683 8.23542 2.10767 8.00495 2.10767C7.77448 2.10767 7.54842 2.1683 7.34891 2.28264C7.14939 2.39698 6.98311 2.56103 6.86578 2.75995H6.86078Z", fill: "currentColor" }),
        /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M8 6V9", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" }),
        /* @__PURE__ */ jsxRuntime.jsx("circle", { cx: "8", cy: "11.5", r: "0.75", fill: "white" })
      ] });
    case "Negative" /* Negative */:
      return /* @__PURE__ */ jsxRuntime.jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsxRuntime.jsx("circle", { cx: "8", cy: "8", r: "8", fill: "currentColor" }),
        /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M10 6L6 10M6 6L10 10", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })
      ] });
    case "Information" /* Information */:
      return /* @__PURE__ */ jsxRuntime.jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsxRuntime.jsx("circle", { cx: "8", cy: "8", r: "8", fill: "currentColor" }),
        /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M8 11V8M8 5H8.00667", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })
      ] });
    default:
      return null;
  }
}
function ImportanceBadge({
  importance
}) {
  if (String(importance) !== "Important") return null;
  return /* @__PURE__ */ jsxRuntime.jsx(Tag, { design: "Negative" /* Negative */, children: "Important" });
}
function NotificationListItem(props) {
  const {
    // Content
    titleText,
    children,
    avatar,
    menu,
    footnotes,
    // Visual
    wrappingType = "None" /* None */,
    state = "None" /* None */,
    importance = "Standard" /* Standard */,
    read = false,
    // Behavior
    showClose = false,
    loading = false,
    loadingDelay = 1e3,
    // Events
    onClose,
    onClick,
    // Standard
    className,
    style,
    id,
    "data-testid": dataTestId,
    ref
  } = props;
  const baseItemRef = React3.useRef(null);
  const descriptionRef = React3.useRef(null);
  const menuBtnRef = React3.useRef(null);
  const listContext = useOptionalNotificationListContext();
  const itemId = id || baseItemRef.current?.nativeElement?.id;
  const ariaLabelledBy = React3__namespace.useMemo(() => {
    if (!itemId) return void 0;
    const ids = [];
    if (String(importance) === "Important") {
      ids.push(`${itemId}-importance`);
    }
    if (String(state) !== "None") {
      ids.push(`${itemId}-state`);
    }
    ids.push(`${itemId}-title`);
    ids.push(`${itemId}-read-status`);
    if (React3__namespace.Children.count(children) > 0) {
      ids.push(`${itemId}-description`);
    }
    if (footnotes && footnotes.length > 0) {
      ids.push(`${itemId}-footnotes`);
    }
    return ids.join(" ");
  }, [itemId, importance, state, children, footnotes]);
  const [showLoading, setShowLoading] = React3.useState(false);
  React3.useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  const [expanded, setExpanded] = React3.useState(false);
  const [hasOverflow, setHasOverflow] = React3.useState(false);
  const hasOverflowRef = React3.useRef(false);
  const expandedRef = React3.useRef(false);
  React3.useEffect(() => {
    expandedRef.current = expanded;
  }, [expanded]);
  React3.useEffect(() => {
    const el = descriptionRef.current;
    if (!el || String(wrappingType) === "Normal") return;
    const observer = new ResizeObserver(() => {
      if (!expandedRef.current) {
        const isOverflowing = el.scrollHeight > el.clientHeight;
        hasOverflowRef.current = isOverflowing;
        setHasOverflow(isOverflowing);
      }
    });
    observer.observe(el);
    return () => observer.disconnect();
  }, [wrappingType]);
  const toggleExpanded = React3.useCallback(() => {
    setExpanded((prev) => !prev);
  }, []);
  const [menuOpen, setMenuOpen] = React3.useState(false);
  const handleClick = React3.useCallback(
    (e) => {
      onClick?.(e);
      const el = baseItemRef.current?.nativeElement;
      if (el) {
        listContext?.onItemClick?.({ item: el, originalEvent: e });
      }
    },
    [listContext, onClick]
  );
  const handleClose = React3.useCallback(
    () => {
      onClose?.();
      const el = baseItemRef.current?.nativeElement;
      if (el) {
        listContext?.onItemClose?.({ item: el });
      }
    },
    [onClose, listContext]
  );
  React3.useImperativeHandle(
    ref,
    () => ({
      focus: () => baseItemRef.current?.focus?.(),
      blur: () => baseItemRef.current?.blur?.(),
      isFocused: () => baseItemRef.current?.isFocused?.() ?? false,
      get nativeElement() {
        return baseItemRef.current?.nativeElement ?? null;
      }
    }),
    []
  );
  const normalizedState = state;
  const normalizedWrapping = wrappingType;
  const hasDescription = React3__namespace.Children.count(children) > 0;
  const hasFootnotes = footnotes && footnotes.length > 0;
  const showStateIcon = String(state) !== "None";
  return /* @__PURE__ */ jsxRuntime.jsxs(
    ListItemBase,
    {
      ref: baseItemRef,
      itemKey: id,
      type: "Active" /* Active */,
      onClick: handleClick,
      accessibleNameRef: ariaLabelledBy,
      className: cn(
        "relative border border-border rounded",
        "transition-colors duration-150",
        "hover:bg-[var(--table-row-hover)]",
        "active:bg-[var(--neutral-pressed-background-2)]",
        "bg-[var(--canvas-primary)]",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-notification-item": true,
      children: [
        showLoading && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20 rounded", children: /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-5 w-5 animate-spin text-sapphire-brand-foreground" }) }),
        /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            className: cn(
              "flex gap-3 w-full min-w-0 p-4",
              showLoading && "opacity-60 pointer-events-none"
            ),
            children: [
              String(importance) === "Important" && itemId && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${itemId}-importance`, className: "sr-only", children: "Important" }),
              showStateIcon && itemId && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${itemId}-state`, className: "sr-only", children: getStateText(state) }),
              itemId && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${itemId}-read-status`, className: "sr-only", children: read ? "Read" : "Unread" }),
              avatar && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "shrink-0", children: avatar }),
              /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex-1 min-w-0", children: [
                /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-start gap-1 pr-16", children: [
                  /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex-1 min-w-0 flex items-start gap-1", children: [
                    showStateIcon && /* @__PURE__ */ jsxRuntime.jsx(StateIcon, { state: normalizedState }),
                    /* @__PURE__ */ jsxRuntime.jsx(
                      "span",
                      {
                        id: itemId ? `${itemId}-title` : void 0,
                        className: cn(
                          titleVariants({ read, wrapping: normalizedWrapping }),
                          "flex-1 min-w-0"
                        ),
                        children: titleText
                      }
                    )
                  ] }),
                  /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "absolute top-2 right-2 flex items-center gap-0.5 shrink-0", children: [
                    menu && /* @__PURE__ */ jsxRuntime.jsxs("span", { onClick: (e) => e.stopPropagation(), children: [
                      /* @__PURE__ */ jsxRuntime.jsx(
                        Button,
                        {
                          ref: menuBtnRef,
                          design: "Tertiary" /* Tertiary */,
                          size: "Medium" /* Medium */,
                          icon: /* @__PURE__ */ jsxRuntime.jsx(OverflowIcon, { className: "h-4 w-4" }),
                          onClick: (detail) => {
                            detail.originalEvent.stopPropagation();
                            setMenuOpen(!menuOpen);
                          },
                          accessibleName: "More actions",
                          accessibilityAttributes: { hasPopup: "menu" }
                        }
                      ),
                      React3__namespace.isValidElement(menu) && React3__namespace.cloneElement(
                        menu,
                        {
                          open: menuOpen,
                          opener: menuBtnRef.current?.nativeElement ?? null,
                          onClose: () => setMenuOpen(false)
                        }
                      )
                    ] }),
                    showClose && /* @__PURE__ */ jsxRuntime.jsx(
                      Button,
                      {
                        design: "Tertiary" /* Tertiary */,
                        size: "Medium" /* Medium */,
                        icon: /* @__PURE__ */ jsxRuntime.jsx(DeclineIcon, { className: "h-4 w-4" }),
                        onClick: (detail) => {
                          detail.originalEvent.stopPropagation();
                          handleClose();
                        },
                        accessibleName: "Close notification"
                      }
                    )
                  ] })
                ] }),
                hasDescription && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "mt-2", children: /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    id: itemId ? `${itemId}-description` : void 0,
                    ref: descriptionRef,
                    className: cn(
                      "text-sm leading-normal text-[var(--text/text-secondary,#353c4a)]",
                      String(wrappingType) !== "Normal" && !expanded && "line-clamp-2",
                      String(wrappingType) === "Normal" && "break-words"
                    ),
                    children
                  }
                ) }),
                (hasFootnotes || hasOverflow || expanded) && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center gap-2 mt-2 text-sm text-[var(--text\\/text-tertiary,#616d85)]", children: [
                  hasFootnotes && /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children: footnotes.map((footnote, index) => /* @__PURE__ */ jsxRuntime.jsxs(React3__namespace.Fragment, { children: [
                    index > 0 && /* @__PURE__ */ jsxRuntime.jsx("span", { children: "\u2022" }),
                    /* @__PURE__ */ jsxRuntime.jsx(
                      "span",
                      {
                        id: index === 0 && itemId ? `${itemId}-footnotes` : void 0,
                        children: footnote
                      }
                    )
                  ] }, index)) }),
                  (hasOverflow || expanded) && /* @__PURE__ */ jsxRuntime.jsx(
                    Link,
                    {
                      onClick: (detail) => {
                        detail.originalEvent.stopPropagation();
                        toggleExpanded();
                      },
                      children: expanded ? "Less" : "More"
                    }
                  )
                ] }),
                String(importance) === "Important" && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "mt-2", children: /* @__PURE__ */ jsxRuntime.jsx(ImportanceBadge, { importance }) })
              ] })
            ]
          }
        )
      ]
    }
  );
}
var pathData7 = [
  "M410.53 200.75C418.95 210.98 417.48 226.11 407.25 234.53L271.34 346.53C262.48 353.83 249.68 353.83 240.81 346.53L104.75 234.53C94.52 226.11 93.05 210.98 101.47 200.75C109.89 190.52 125.02 189.05 135.25 197.47L256.06 296.91L376.75 197.47C386.98 189.05 402.11 190.52 410.53 200.75Z"
];
var SlimArrowDownIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData7, ...props })
  ),
  { displayName: "SlimArrowDownIcon", iconName: "slim-arrow-down" }
);
var pathData8 = [
  "M186 416q-11 0-18.5-7.5T160 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T186 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L203 409q-7 7-17 7z"
];
var SlimArrowRightIcon = Object.assign(
  React3__namespace.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData8, ...props })
  ),
  { displayName: "SlimArrowRightIcon", iconName: "slim-arrow-right" }
);
function NotificationListGroupItem(props) {
  const {
    // Content
    titleText,
    children,
    // Behavior
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    growing = "None" /* None */,
    loading = false,
    loadingDelay = 1e3,
    // Events
    onToggle,
    onLoadMore,
    // Standard
    className,
    style,
    accessibleName,
    "data-testid": dataTestId,
    ref
  } = props;
  const groupRef = React3.useRef(null);
  const listContext = useOptionalNotificationListContext();
  const contentId = React3.useId();
  const [internalCollapsed, setInternalCollapsed] = React3.useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const [showLoading, setShowLoading] = React3.useState(false);
  React3.useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  const handleToggle = React3.useCallback(() => {
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.(newCollapsed);
    listContext?.onItemToggle?.({
      item: groupRef.current,
      collapsed: newCollapsed
    });
  }, [isCollapsed, isControlled, onToggle, listContext]);
  const handleHeaderKeyDown = React3.useCallback((e) => {
    const shouldToggle = e.key === "Enter" || e.key === " " || e.key === "+" && isCollapsed || e.key === "-" && !isCollapsed || e.key === "ArrowRight" && isCollapsed || e.key === "ArrowLeft" && !isCollapsed;
    if (!shouldToggle) return;
    e.preventDefault();
    if (e.key === "ArrowRight" || e.key === "ArrowLeft" || e.key === "+" || e.key === "-") {
      e.stopPropagation();
    }
    handleToggle();
  }, [handleToggle, isCollapsed]);
  React3.useImperativeHandle(
    ref,
    () => ({
      focus: () => groupRef.current?.focus(),
      blur: () => groupRef.current?.blur(),
      isFocused: () => document.activeElement === groupRef.current,
      toggle: handleToggle,
      get nativeElement() {
        return groupRef.current;
      }
    }),
    [handleToggle]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(
    "li",
    {
      ref: groupRef,
      role: "listitem",
      "aria-level": 1,
      "aria-labelledby": accessibleName ? void 0 : `${contentId}-title`,
      "aria-label": accessibleName || void 0,
      "data-notification-item": true,
      "data-notification-group": true,
      className: cn(
        "mb-2",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "w-full", children: [
        /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            role: "button",
            "aria-expanded": !isCollapsed,
            "aria-controls": !isCollapsed ? contentId : void 0,
            "aria-label": accessibleName || titleText,
            tabIndex: 0,
            onClick: handleToggle,
            onKeyDown: handleHeaderKeyDown,
            className: cn(
              "flex items-center gap-1 w-full pl-1 pr-2 py-1 cursor-pointer rounded",
              "hover:bg-sapphire-canvas-tertiary focus:outline-none transition-colors",
              "focus:shadow-[inset_0_0_0_2px_var(--border-focus)]"
            ),
            children: [
              /* @__PURE__ */ jsxRuntime.jsx("span", { className: "shrink-0 h-8 w-8 flex items-center justify-center text-sapphire-text-primary", children: isCollapsed ? /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: "h-4 w-4" }) }),
              /* @__PURE__ */ jsxRuntime.jsx(
                "span",
                {
                  id: `${contentId}-title`,
                  className: cn(
                    "flex-1 text-base leading-[22px] truncate font-semibold text-[var(--text/text-primary,#0b0c0f)]"
                  ),
                  children: titleText
                }
              )
            ]
          }
        ),
        !isCollapsed && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative", children: [
          showLoading && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20", children: /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-5 w-5 animate-spin text-sapphire-brand-foreground" }) }),
          /* @__PURE__ */ jsxRuntime.jsx(
            "ul",
            {
              id: contentId,
              className: cn(
                "m-0 p-0 list-none flex flex-col gap-2",
                showLoading && "opacity-60 pointer-events-none"
              ),
              children
            }
          ),
          String(growing) === "Button" /* Button */ && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center justify-center py-2 border-t border-border", children: /* @__PURE__ */ jsxRuntime.jsxs(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              onClick: (detail) => {
                detail.originalEvent.stopPropagation();
                onLoadMore?.();
              },
              disabled: showLoading,
              className: "w-full mx-4",
              "data-growing-button": "true",
              children: [
                showLoading ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-4 w-4 animate-spin mr-2" }) : null,
                "More"
              ]
            }
          ) })
        ] })
      ] })
    }
  );
}

exports.NotificationList = NotificationList;
exports.NotificationListContext = NotificationListContext;
exports.NotificationListGroupItem = NotificationListGroupItem;
exports.NotificationListItem = NotificationListItem;
exports.useNotificationListContext = useNotificationListContext;
exports.useOptionalNotificationListContext = useOptionalNotificationListContext;
