// @sap-ui/fx-components/dist/components/tag-exploration/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/tag-exploration/TagExploration.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData7,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData7 || pathData7.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData7.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M292 256l117 116q7 7 7 18 0 12-7.5 19t-18.5 7q-10 0-18-8L256 292 140 408q-8 8-18 8-11 0-18.5-7.5T96 390q0-10 8-18l116-116-116-116q-8-8-8-18 0-11 7.5-18.5T122 96t18 7l116 117 116-117q7-7 18-7t18.5 7.5T416 122t-7 18z"
];
var DeclineIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "DeclineIcon", iconName: "decline" }
);
var pathData2 = [
  "M256 0q53 0 100 20t81.5 54.5T492 156t20 100-20 100-54.5 81.5T356 492t-100 20-100-20-81.5-54.5T20 356 0 256t20-100 54.5-81.5T156 20 256 0zm150 183q10-9 10-23 0-13-9.5-22.5T384 128t-22 9L186 308l-68-63q-9-9-22-9t-22.5 9.5T64 268q0 15 10 24l91 83q9 9 21 9 13 0 23-9z"
];
var SysEnter2Icon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "SysEnter2Icon", iconName: "sys-enter-2" }
);
var pathData3 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm45 256l74-73q9-11 9-23 0-13-9.5-22.5T352 128q-12 0-23 9l-73 74-73-74q-10-9-23-9t-22.5 9.5T128 160q0 12 9 23l74 73-74 73q-9 10-9 23t9.5 22.5T160 384t23-9l73-74 73 74q11 9 23 9 13 0 22.5-9.5T384 352t-9-23z"
];
var ErrorIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "ErrorIcon", iconName: "error" }
);
var pathData4 = [
  "M505 399q7 13 7 27 0 21-15.5 37.5T456 480H56q-25 0-40.5-16.5T0 426q0-14 7-27L208 59q17-27 48-27 14 0 27 6.5T304 59zM288 176q0-14-9-23t-23-9-23 9-9 23v96q0 14 9 23t23 9 23-9 9-23v-96zm-32 240q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9z"
];
var AlertIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "AlertIcon", iconName: "alert" }
);
var pathData5 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm0 160q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9zm32 64q0-14-9-23t-23-9-23 9-9 23v160q0 14 9 23t23 9 23-9 9-23V224z"
];
var InformationIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData5, ...props })
  ),
  { displayName: "InformationIcon", iconName: "information" }
);
var pathData6 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm32 289q31-10 50.5-36.5T358 192q0-43-29.5-72.5T256 90q-21 0-39.5 8T184 118.5 162 148t-8 35q0 20 9 30.5t23 10.5q13 0 22.5-9t9.5-23q0-16 11-27t27-11 27 11 11 27-11 27-27 11q-14 0-23 9.5t-9 22.5v26q0 14 9 23t23 9q13 0 22-9t10-22zm-32 127q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9z"
];
var SysHelp2Icon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData6, ...props })
  ),
  { displayName: "SysHelp2Icon", iconName: "sys-help-2" }
);
var semanticDesignClass = {
  ["Neutral" /* Neutral */]: "bg-background text-sapphire-text-accent-2 border-sapphire-tag-border",
  ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info border-transparent",
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive border-transparent",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative border-transparent",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning border-transparent"
};
var stateIconMap = {
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Critical" /* Critical */]: AlertIcon,
  ["Information" /* Information */]: InformationIcon,
  ["Neutral" /* Neutral */]: SysHelp2Icon
};
var set1Classes = {
  ["1" /* CS1 */]: "bg-purple-600 text-sapphire-neutral-foreground-white border-transparent",
  ["2" /* CS2 */]: "bg-blue-600 text-sapphire-neutral-foreground-white border-transparent",
  ["3" /* CS3 */]: "bg-neutral-600 text-sapphire-neutral-foreground-white border-transparent",
  ["4" /* CS4 */]: "bg-purple-700 text-sapphire-neutral-foreground-white border-transparent",
  ["5" /* CS5 */]: "bg-blue-700 text-sapphire-neutral-foreground-white border-transparent",
  ["6" /* CS6 */]: "bg-neutral-700 text-sapphire-neutral-foreground-white border-transparent",
  ["7" /* CS7 */]: "bg-purple-800 text-sapphire-neutral-foreground-white border-transparent",
  ["8" /* CS8 */]: "bg-blue-800 text-sapphire-neutral-foreground-white border-transparent",
  ["9" /* CS9 */]: "bg-neutral-800 text-sapphire-neutral-foreground-white border-transparent",
  ["10" /* CS10 */]: "bg-purple-500 text-sapphire-neutral-foreground-white border-transparent"
};
var set2Classes = {
  ["1" /* CS1 */]: "bg-purple-100 text-purple-700 border-transparent",
  ["2" /* CS2 */]: "bg-blue-100 text-blue-700 border-transparent",
  ["3" /* CS3 */]: "bg-neutral-100 text-neutral-700 border-transparent",
  ["4" /* CS4 */]: "bg-purple-200 text-purple-800 border-transparent",
  ["5" /* CS5 */]: "bg-blue-200 text-blue-800 border-transparent",
  ["6" /* CS6 */]: "bg-neutral-200 text-neutral-800 border-transparent",
  ["7" /* CS7 */]: "bg-purple-50 text-purple-600 border-transparent",
  ["8" /* CS8 */]: "bg-blue-50 text-blue-600 border-transparent",
  ["9" /* CS9 */]: "bg-neutral-50 text-neutral-600 border-transparent",
  ["10" /* CS10 */]: "bg-purple-100 text-purple-600 border-transparent"
};
function getDesignClasses(design, colorScheme, accent) {
  if (design === "Set1" /* Set1 */) {
    return set1Classes[colorScheme] ?? set1Classes["1" /* CS1 */];
  }
  if (design === "Set2" /* Set2 */) {
    return set2Classes[colorScheme] ?? set2Classes["1" /* CS1 */];
  }
  if (design === "Neutral" /* Neutral */ && accent) {
    return "bg-sapphire-tag-accent-bg text-sapphire-text-accent-2 border-sapphire-tag-accent-bg";
  }
  return semanticDesignClass[design] ?? semanticDesignClass["Neutral" /* Neutral */];
}
function TagExploration({
  design = "Neutral" /* Neutral */,
  size = "S" /* S */,
  colorScheme = "1" /* CS1 */,
  children,
  icon,
  interactive = false,
  readOnly = false,
  accent = false,
  wrappingType = "None" /* None */,
  hideStateIcon = false,
  onClose,
  onClick,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const tagRef = React2.useRef(null);
  const d = design;
  const s = size;
  const cs = colorScheme;
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        tagRef.current?.focus();
      },
      blur() {
        tagRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === tagRef.current;
      },
      get nativeElement() {
        return tagRef.current;
      }
    }),
    [tagRef]
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (!interactive) return;
      onClick?.({ originalEvent: e, isKeyboard: false });
    },
    [interactive, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (!interactive) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({ originalEvent: e, isKeyboard: true });
      }
    },
    [interactive, onClick]
  );
  const handleCloseClick = React2.useCallback(
    (e) => {
      e.stopPropagation();
      onClose?.({ originalEvent: e, isKeyboard: false });
    },
    [onClose]
  );
  const handleCloseKeyDown = React2.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        e.stopPropagation();
        onClose?.({ originalEvent: e, isKeyboard: true });
      }
    },
    [onClose]
  );
  const isSemanticDesign = d !== "Set1" /* Set1 */ && d !== "Set2" /* Set2 */;
  let displayIcon = null;
  if (icon) {
    displayIcon = icon;
  } else if (isSemanticDesign && !hideStateIcon) {
    const StateIcon = stateIconMap[d];
    if (StateIcon) {
      const iconSizeClass = s === "L" /* L */ ? "h-4 w-4" : "h-3 w-3";
      displayIcon = /* @__PURE__ */ jsxRuntime.jsx(StateIcon, { className: iconSizeClass });
    }
  }
  const sizeClass = s === "L" /* L */ ? "text-sm px-[9px] py-0.5 rounded-full gap-1" : "text-xs px-2 py-1 rounded-full gap-1";
  const closeBtnSizeClass = s === "L" /* L */ ? "h-4 w-4" : "h-3 w-3";
  const wrappingClasses = wrappingType === "None" /* None */ ? "max-w-full" : "";
  const textWrappingClasses = wrappingType === "None" /* None */ ? "overflow-hidden text-ellipsis whitespace-nowrap" : "whitespace-normal break-words";
  const focusPaddingClass = s === "L" /* L */ ? "focus-visible:px-[8px] focus-visible:py-px" : "focus-visible:px-[7px] focus-visible:py-[3px]";
  const interactiveClasses = interactive && !readOnly ? [
    "cursor-pointer",
    "hover:bg-sapphire-tag-hover-bg hover:border-transparent",
    "active:bg-sapphire-tag-active-bg active:border-transparent active:rounded-[20px]",
    `focus-visible:outline-none focus-visible:bg-sapphire-tag-hover-bg focus-visible:border-2 focus-visible:border-sapphire-tag-focus-border ${focusPaddingClass}`
  ].join(" ") : "";
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "span",
    {
      ref: tagRef,
      id,
      role: interactive && !readOnly ? "button" : void 0,
      tabIndex: interactive && !readOnly ? 0 : void 0,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        "inline-flex items-center border transition-colors font-normal leading-5 tracking-[0.25px]",
        sizeClass,
        getDesignClasses(d, cs, accent),
        wrappingClasses,
        interactiveClasses,
        readOnly && "opacity-40 pointer-events-none",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        displayIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-shrink-0", "data-part": "icon", children: displayIcon }),
        children && /* @__PURE__ */ jsxRuntime.jsx("span", { "data-part": "text", className: textWrappingClasses, children }),
        onClose && !readOnly && /* @__PURE__ */ jsxRuntime.jsx(
          "button",
          {
            type: "button",
            onClick: handleCloseClick,
            onKeyDown: handleCloseKeyDown,
            "aria-label": "Remove",
            "data-part": "close",
            className: cn(
              "flex-shrink-0 inline-flex items-center justify-center rounded-full",
              "hover:opacity-70 focus:outline-none focus:ring-1 focus:ring-ring",
              "cursor-pointer"
            ),
            tabIndex: 0,
            children: /* @__PURE__ */ jsxRuntime.jsx(DeclineIcon, { className: closeBtnSizeClass })
          }
        )
      ]
    }
  );
}

exports.TagExploration = TagExploration;
