// @sap-ui/fx-components/dist/components/toast/index.cjs
'use strict';

var React = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React__default = /*#__PURE__*/_interopDefault(React);

// src/components/toast/Toast.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var MIN_DURATION = 500;
var MAX_TRANSITION = 1e3;
var placementStyles = {
  ["TopStart" /* TopStart */]: { top: "3rem", left: "2rem" },
  ["TopCenter" /* TopCenter */]: { top: "3rem", left: "50%", transform: "translateX(-50%)" },
  ["TopEnd" /* TopEnd */]: { top: "3rem", right: "2rem" },
  ["MiddleStart" /* MiddleStart */]: { top: "50%", left: "2rem", transform: "translateY(-50%)" },
  ["MiddleCenter" /* MiddleCenter */]: { top: "50%", left: "50%", transform: "translate(-50%, -50%)" },
  ["MiddleEnd" /* MiddleEnd */]: { top: "50%", right: "2rem", transform: "translateY(-50%)" },
  ["BottomStart" /* BottomStart */]: { bottom: "3rem", left: "2rem" },
  ["BottomCenter" /* BottomCenter */]: { bottom: "3rem", left: "50%", transform: "translateX(-50%)" },
  ["BottomEnd" /* BottomEnd */]: { bottom: "3rem", right: "2rem" }
};
function Toast({
  open = false,
  duration = 3e3,
  placement = "BottomCenter" /* BottomCenter */,
  children,
  onClose,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const toastRef = React.useRef(null);
  const [hovered, setHovered] = React.useState(false);
  const [focused, setFocused] = React.useState(false);
  const [fading, setFading] = React.useState(false);
  React__default.default.useImperativeHandle(ref, () => toastRef.current, []);
  const effectiveDuration = Math.max(duration, MIN_DURATION);
  const transitionDuration = Math.min(effectiveDuration / 3, MAX_TRANSITION);
  React.useEffect(() => {
    const el = toastRef.current;
    if (!el) return;
    if (open) {
      if (!el.matches(":popover-open")) {
        el.showPopover();
      }
      requestAnimationFrame(() => {
        setFading(true);
      });
    } else {
      setFading(false);
      if (el.matches(":popover-open")) {
        el.hidePopover();
      }
    }
  }, [open]);
  const handleTransitionEnd = React.useCallback(() => {
    if (hovered || focused) return;
    setFading(false);
    onClose?.();
  }, [hovered, focused, onClose]);
  const paused = hovered || focused;
  React.useEffect(() => {
    if (!open) return;
    const handleKeydown = (e) => {
      const isCtrl = e.metaKey || !navigator.userAgent.includes("Mac") && e.ctrlKey;
      if (isCtrl && e.shiftKey && e.key.toLowerCase() === "m") {
        e.preventDefault();
        if (focused) {
          setFocused(false);
        } else {
          toastRef.current?.focus();
        }
      }
    };
    document.addEventListener("keydown", handleKeydown);
    return () => document.removeEventListener("keydown", handleKeydown);
  }, [open, focused]);
  const handleKeyDown = React.useCallback((e) => {
    if (e.key === "Escape") {
      setFocused(false);
    }
  }, []);
  const posStyles = placementStyles[placement] || placementStyles["BottomCenter" /* BottomCenter */];
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref: toastRef,
      id,
      popover: "manual",
      role: "alert",
      tabIndex: focused ? 0 : -1,
      "data-testid": dataTestId,
      className: cn(
        // Visual style
        "max-w-60 rounded p-4",
        "bg-foreground dark:bg-neutral-300 text-background text-sm text-center",
        "shadow-lg",
        "break-words whitespace-pre-line",
        // Focus ring
        focused && "outline-2 outline-offset-[-3px] outline-[var(--ring)]",
        className
      ),
      style: {
        // Reset popover UA defaults — must be inline to win specificity
        position: "fixed",
        inset: "unset",
        margin: 0,
        padding: void 0,
        border: "none",
        background: void 0,
        // Placement positioning
        ...posStyles,
        // Fade transition
        transitionProperty: "opacity",
        transitionDuration: fading ? `${transitionDuration}ms` : "0ms",
        transitionDelay: fading ? `${effectiveDuration - transitionDuration}ms` : "0ms",
        opacity: fading && !paused ? 0 : 1,
        ...style
      },
      onTransitionEnd: handleTransitionEnd,
      onMouseEnter: () => setHovered(true),
      onMouseLeave: () => setHovered(false),
      onFocusCapture: () => setFocused(true),
      onBlurCapture: () => setFocused(false),
      onKeyDown: handleKeyDown,
      children
    }
  );
}
Toast.displayName = "Toast";

exports.Toast = Toast;
