// @sap-ui/fx-components/dist/components/checkbox/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var lucideReact = require('lucide-react');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/checkbox/CheckBox.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData4,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData4 || pathData4.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData4.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M438 224q11 0 18.5 7.5T464 250v140q0 38-26 64t-64 26H106q-38 0-64-26t-26-64V122q0-38 26-64t64-26h237q11 0 18 7.5t7 18.5-7 18-18 7H106q-16 0-27.5 11.5T67 122v268q0 16 11.5 27.5T106 429h268q16 0 27.5-11.5T413 390V250q0-11 7-18.5t18-7.5zm32-192q11 0 18.5 7.5T496 58q0 10-7 17L257 312q-6 8-18 8-10 0-18-8l-70-71q-7-7-7-18t7.5-18 18.5-7 18 7l51 53L452 40q8-8 18-8z"
];
var CompleteIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "CompleteIcon", iconName: "complete" }
);
var pathData2 = [
  "M390 480H122q-38 0-64-26t-26-64V122q0-38 26-64t64-26h268q38 0 64 26t26 64v268q0 38-26 64t-64 26zM122 83q-17 0-28 11t-11 28v268q0 17 11 28t28 11h268q17 0 28-11t11-28V122q0-17-11-28t-28-11H122z"
];
var BorderIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "BorderIcon", iconName: "border" }
);
var pathData3 = [
  "M390 32q38 0 64 26t26 64v268q0 38-26 64t-64 26H122q-38 0-64-26t-26-64V122q0-38 26-64t64-26h268zm39 90q0-17-11-28t-28-11H122q-17 0-28 11t-11 28v268q0 17 11 28t28 11h268q17 0 28-11t11-28V122zm-77 38v192H160V160h192z"
];
var TriStateIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "TriStateIcon", iconName: "tri-state" }
);
function CheckBox({
  checked: controlledChecked,
  defaultChecked = false,
  indeterminate = false,
  disabled = false,
  readonly = false,
  displayOnly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  wrappingType = "Normal" /* Normal */,
  valueState = "None",
  valueStateMessage,
  name,
  value = "on",
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  accessibilityAttributes,
  ref
}) {
  const inputRef = React2.useRef(null);
  const rootRef = React2.useRef(null);
  const generatedId = React2.useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = React2.useState(defaultChecked);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const isActive = isChecked || indeterminate;
  const doToggle = React2.useCallback(
    (originalEvent) => {
      if (disabled || readonly || displayOnly) return;
      const newChecked = !isChecked;
      if (!isControlled) {
        setInternalChecked(newChecked);
      }
      onChange?.({
        checked: newChecked,
        indeterminate: false,
        originalEvent
      });
    },
    [disabled, readonly, displayOnly, isChecked, isControlled, onChange]
  );
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      toggle() {
        doToggle();
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked, doToggle]
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) {
        e.preventDefault();
        return;
      }
      doToggle(e);
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const handleMouseDown = React2.useCallback(
    (e) => {
      if (disabled || displayOnly) {
        e.preventDefault();
      }
    },
    [disabled, displayOnly]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
      }
    },
    [disabled, readonly, displayOnly]
  );
  const handleKeyUp = React2.useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doToggle(e);
      }
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const labelContent = children || text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly && !displayOnly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const getValueStateBorderClass = () => {
    if (isActive) return "";
    const borderMap = {
      Positive: "border-sapphire-positive",
      Negative: "border-sapphire-negative",
      Critical: "border-sapphire-warning",
      Information: "border-sapphire-info"
    };
    return borderMap[valueState] ?? "";
  };
  const getValueStateCheckedClass = () => {
    if (!isActive) return "";
    const checkedMap = {
      Positive: "bg-sapphire-positive border-sapphire-positive",
      Negative: "bg-sapphire-negative border-sapphire-negative",
      Critical: "bg-sapphire-warning border-sapphire-warning",
      Information: "bg-sapphire-info border-sapphire-info"
    };
    return checkedMap[valueState] ?? "bg-sapphire-canvas-primary border-sapphire-border-secondary";
  };
  const renderDisplayOnlyIcon = () => {
    const cls = "size-4 text-foreground";
    if (indeterminate) return /* @__PURE__ */ jsxRuntime.jsx(TriStateIcon, { className: cls });
    if (isChecked) return /* @__PURE__ */ jsxRuntime.jsx(CompleteIcon, { className: cls });
    return /* @__PURE__ */ jsxRuntime.jsx(BorderIcon, { className: cls });
  };
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        ref: inputRef,
        type: "checkbox",
        name,
        value,
        checked: isChecked,
        disabled: disabled || displayOnly,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: rootRef,
        id,
        role: "checkbox",
        "aria-checked": indeterminate ? "mixed" : isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          valueState !== "None" && valueStateMessage ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null,
          accessibilityAttributes?.describedBy
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || displayOnly || void 0,
        "aria-readonly": readonly || void 0,
        "aria-invalid": valueState === "Negative" || void 0,
        "aria-required": required || void 0,
        tabIndex: tabIndex !== void 0 ? tabIndex : disabled || displayOnly ? -1 : 0,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          // Touch area: Cozy = 44px, Compact = 32px
          isSmall ? labelContent ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2" : labelContent ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]",
          "text-sm",
          "rounded-lg",
          // Focus ring via pseudo-element for asymmetric insets
          "outline-none",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          labelContent ? isSmall ? "after:inset-x-0 after:inset-y-0" : "after:left-0 after:right-[2px] after:inset-y-[5px]" : "after:inset-[5px] after:rounded-lg",
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          displayOnly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          displayOnly ? renderDisplayOnlyIcon() : /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: cn(
                "relative inline-flex items-center justify-center",
                "shrink-0",
                hasWrappingLabel && "self-start",
                isSmall ? hasWrappingLabel ? "size-4 rounded p-[2px] mt-0.5" : "size-4 rounded p-[2px]" : hasWrappingLabel ? "size-[22px] rounded-md p-[3px] mt-0.5" : "size-[22px] rounded-md p-[3px]",
                "transition-colors",
                // Border style: dashed for readonly, solid otherwise
                readonly ? "border border-dashed" : "border border-solid",
                // --- Unchecked state ---
                !isActive && !readonly && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
                // --- Checked / indeterminate state (not readonly) ---
                // None value state: blue icon on white bg; value states: white icon on colored bg
                isActive && !readonly && valueState === "None" && "text-sapphire-shell-button-fg-selected",
                isActive && !readonly && valueState !== "None" && "text-sapphire-neutral-foreground-white",
                !readonly && getValueStateCheckedClass(),
                !readonly && getValueStateBorderClass(),
                // --- Hover (unchecked, not readonly/disabled, no value state) ---
                !isActive && isInteractive && valueState === "None" && "group-hover:border-primary group-hover:bg-sapphire-shell-button-bg-selected",
                // --- Hover (checked, not readonly/disabled, no value state) ---
                isActive && isInteractive && valueState === "None" && "group-hover:bg-sapphire-shell-button-bg-selected group-hover:border-primary",
                // --- Read-only state: gray bg + dashed secondary border ---
                readonly && "bg-sapphire-background-secondary border-sapphire-border-secondary",
                // --- Read-only icon color: secondary-foreground (#353C4A) ---
                readonly && isActive && "text-secondary-foreground"
              ),
              children: indeterminate ? (
                // Solid square for tristate per Figma: 12px large, 8px small
                /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn(
                  "block bg-current",
                  isSmall ? "size-2" : "size-3"
                ) })
              ) : isChecked ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Check, { className: isSmall ? "size-3" : "size-4", strokeWidth: 3 }) : null
            }
          ),
          labelContent && /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" && valueStateMessage && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage }),
          accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}

exports.CheckBox = CheckBox;
