// @sap-ui/fx-components/dist/components/card/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/card/Card.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData2,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData2 || pathData2.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData2.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M26 192q-11 0-18.5-7.5T0 166q0-2 1-6t5-12q17-33 42.5-60.5t58-47 70-30T254 0q60 0 113.5 23.5T461 90V58q0-11 7-18.5t18-7.5 18.5 7.5T512 58v108q0 11-7.5 18.5T486 192H378q-11 0-18.5-7.5T352 166t7.5-18 18.5-7h59q-33-43-81-66.5T254 51q-66 0-121 34t-84 93q-8 14-23 14zm460 128q11 0 18.5 7.5T512 346q0 2-1 6t-5 12q-16 33-42.5 60.5t-59 47-70 30T258 512q-60 0-113.5-23.5T51 422v32q0 11-7 18.5T26 480t-18.5-7.5T0 454V346q0-11 7.5-18.5T26 320h108q11 0 18.5 7.5T160 346t-7.5 18-18.5 7H75q33 43 81 66.5T258 461q66 0 121-34t84-93q8-14 23-14z"
];
var SynchronizeIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SynchronizeIcon", iconName: "synchronize" }
);
function Card({
  children,
  header,
  footer,
  toolbar,
  toolbarVisible,
  toolbarClassName,
  loading = false,
  loadingDelay = 1e3,
  accessibleName,
  accessibleNameRef,
  accessibleDescriptionRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  onMouseEnter,
  onMouseLeave,
  onMouseDown,
  onMouseUp,
  onClick,
  onFocus,
  onBlur,
  ref
}) {
  const cardRef = React2.useRef(null);
  const [showLoading, setShowLoading] = React2.useState(false);
  const isHeaderInteractive = React2.isValidElement(header) && header.props.interactive === true;
  React2.useEffect(() => {
    if (!loading) {
      setShowLoading(false);
      return;
    }
    if (loadingDelay <= 0) {
      setShowLoading(true);
      return;
    }
    const timer = setTimeout(() => {
      setShowLoading(true);
    }, loadingDelay);
    return () => clearTimeout(timer);
  }, [loading, loadingDelay]);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        cardRef.current?.focus();
      },
      blur() {
        cardRef.current?.blur();
      },
      get nativeElement() {
        return cardRef.current;
      }
    }),
    []
  );
  const hasContent = children != null && children !== false && children !== "" && React2.Children.count(children) > 0;
  const card = /* @__PURE__ */ jsxRuntime.jsxs(
    "article",
    {
      ref: cardRef,
      id,
      role: "region",
      "aria-roledescription": "Card",
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-describedby": accessibleDescriptionRef,
      "aria-busy": loading || void 0,
      className: cn(
        "relative rounded-[8px] border border-sapphire-border-primary bg-sapphire-card-bg-primary text-sapphire-text-primary",
        "flex flex-col",
        isHeaderInteractive && [
          // Only transition border and shadow, not outline (focus should be instant)
          "transition-[border-color,box-shadow] duration-200",
          // Focus ring on the card when header is focused (using outline to avoid layout shift)
          "focus-within:outline focus-within:outline-2 focus-within:outline-sapphire-border-focus focus-within:outline-offset-[-2px]",
          // Hover state
          "hover:border-sapphire-border-active hover:shadow-[0px_1px_2px_0px_rgba(10,10,10,0.25)]"
        ],
        !hasContent && "card--nocontent",
        // When no content/footer, make header corners fully rounded
        !hasContent && !footer && "[&>*:first-child]:rounded-[8px]",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      onMouseEnter,
      onMouseLeave,
      onMouseDown,
      onMouseUp,
      onClick,
      onFocus,
      onBlur,
      children: [
        header,
        hasContent && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            className: "flex-1 relative overflow-hidden",
            role: "group",
            "aria-label": "Card Content",
            "data-part": "content",
            children: [
              children,
              showLoading && /* @__PURE__ */ jsxRuntime.jsx(
                "div",
                {
                  className: cn(
                    "absolute inset-0",
                    "bg-sapphire-background-primary/80 backdrop-blur-sm",
                    "flex items-center justify-center",
                    "z-10"
                  ),
                  "aria-hidden": "true",
                  children: /* @__PURE__ */ jsxRuntime.jsx(SynchronizeIcon, { className: "h-8 w-8 animate-spin text-sapphire-text-tertiary" })
                }
              )
            ]
          }
        ),
        footer
      ]
    }
  );
  if (toolbar) {
    return /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative group/card", "data-part": "card-toolbar-wrapper", children: [
      /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          className: cn(
            "absolute bottom-full left-0 right-0 z-10 translate-y-3",
            !toolbarVisible && "invisible group-hover/card:visible"
          ),
          "data-part": "toolbar",
          children: /* @__PURE__ */ jsxRuntime.jsx(
            "div",
            {
              className: cn(
                "flex items-center justify-center gap-px p-1",
                "bg-sapphire-card-bg-primary border border-brand-purple rounded-[8px]",
                "shadow-[0px_4px_10px_0px_rgba(10,10,10,0.15),0px_2px_5px_0px_rgba(10,10,10,0.05)]",
                toolbarClassName
              ),
              children: toolbar
            }
          )
        }
      ),
      card
    ] });
  }
  return card;
}
var statusVariants = cva("text-sm", {
  variants: {
    status: {
      ["None" /* None */]: "text-sapphire-text-tertiary",
      ["Positive" /* Positive */]: "text-sapphire-positive",
      ["Negative" /* Negative */]: "text-sapphire-negative",
      ["Critical" /* Critical */]: "text-sapphire-warning",
      ["Information" /* Information */]: "text-sapphire-info"
    }
  },
  defaultVariants: {
    status: "None" /* None */
  }
});
var CardHeader = React2__default.default.forwardRef(
  ({
    titleText,
    subtitleText,
    additionalText,
    interactive = false,
    avatar,
    action,
    status = "None" /* None */,
    ariaLevel = 3,
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    onClick,
    className,
    style,
    id,
    "data-testid": dataTestId
  }, ref) => {
    const headerRef = React2.useRef(null);
    const interactiveRef = React2.useRef(null);
    const [isPressed, setIsPressed] = React2.useState(false);
    React2.useImperativeHandle(
      ref,
      () => {
        const target = interactive ? interactiveRef.current : headerRef.current;
        return {
          focus() {
            target?.focus();
          },
          blur() {
            target?.blur();
          },
          isFocused() {
            if (interactive) {
              return document.activeElement === interactiveRef.current || headerRef.current?.contains(document.activeElement) === true;
            }
            return document.activeElement === headerRef.current;
          },
          get nativeElement() {
            return headerRef.current;
          }
        };
      },
      [interactive]
    );
    const handleClick = React2.useCallback(
      (e) => {
        if (!interactive) return;
        onClick?.({ originalEvent: e, isKeyboard: false });
      },
      [interactive, onClick]
    );
    const handleKeyDown = React2.useCallback(
      (e) => {
        if (!interactive) return;
        if (e.key === "Enter" || e.key === " ") {
          e.preventDefault();
          setIsPressed(true);
          onClick?.({ originalEvent: e, isKeyboard: true });
        }
      },
      [interactive, onClick]
    );
    const handleKeyUp = React2.useCallback((e) => {
      if (e.key === "Enter" || e.key === " ") {
        setIsPressed(false);
      }
    }, []);
    const renderTitle = (suppressHeading) => {
      if (!titleText) return null;
      return /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          role: suppressHeading ? void 0 : "heading",
          "aria-level": suppressHeading ? void 0 : ariaLevel,
          className: "text-base font-semibold text-sapphire-text-primary leading-[22px] line-clamp-3 flex-1 min-w-0",
          "data-part": "title",
          children: titleText
        }
      );
    };
    const renderContent = () => /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex-1 min-w-0", children: [
      (titleText || additionalText) && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center gap-2", children: [
        renderTitle(interactive),
        additionalText && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            className: cn("flex-shrink-0", statusVariants({ status })),
            "data-part": "additional-text",
            children: additionalText
          }
        )
      ] }),
      subtitleText && /* @__PURE__ */ jsxRuntime.jsx("p", { className: "text-sm leading-5 text-sapphire-text-secondary line-clamp-2 mt-1", "data-part": "subtitle", children: subtitleText })
    ] });
    return /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: headerRef,
        id,
        className: cn(
          "flex items-center gap-4 p-4 border-b border-sapphire-border-primary rounded-t-[8px]",
          interactive && [
            "hover:bg-sapphire-background-secondary/50 transition-colors",
            isPressed && "bg-sapphire-background-secondary/70"
            // No focus ring on header when interactive - focus shows on card border
          ],
          className
        ),
        style,
        "data-testid": dataTestId,
        "data-part": "root",
        children: [
          /* @__PURE__ */ jsxRuntime.jsxs(
            "div",
            {
              ref: interactive ? interactiveRef : void 0,
              role: interactive ? "button" : void 0,
              tabIndex: interactive ? 0 : void 0,
              "aria-roledescription": interactive ? "Interactive Card Header" : void 0,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-describedby": accessibleDescriptionRef,
              onClick: interactive ? handleClick : void 0,
              onKeyDown: interactive ? handleKeyDown : void 0,
              onKeyUp: interactive ? handleKeyUp : void 0,
              onMouseDown: interactive ? () => setIsPressed(true) : void 0,
              onMouseUp: interactive ? () => setIsPressed(false) : void 0,
              onMouseLeave: interactive && isPressed ? () => setIsPressed(false) : void 0,
              className: cn(
                "flex items-center gap-3 flex-1 min-w-0",
                interactive && "cursor-pointer focus:outline-none rounded-[4px]"
              ),
              "data-part": interactive ? "interactive-area" : "content-area",
              children: [
                avatar && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex-shrink-0", "data-part": "avatar", children: avatar }),
                renderContent()
              ]
            }
          ),
          action && /* @__PURE__ */ jsxRuntime.jsx(
            "div",
            {
              className: "flex-shrink-0 z-10",
              "data-part": "action",
              children: action
            }
          )
        ]
      }
    );
  }
);
CardHeader.displayName = "CardHeader";
function CardFooter({ children, className, style, "data-testid": dataTestId, ref }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      className: cn(
        "flex items-center justify-end gap-4 p-4 pt-3 border-t border-sapphire-border-primary",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "footer",
      children
    }
  );
}
function CardContent({ children, className, style, "data-testid": dataTestId, ref }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      className: cn("p-4", className),
      style,
      "data-testid": dataTestId,
      "data-part": "content",
      children
    }
  );
}

exports.Card = Card;
exports.CardContent = CardContent;
exports.CardFooter = CardFooter;
exports.CardHeader = CardHeader;
