// @sap-ui/fx-components/dist/components/label/index.cjs
'use strict';

var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

// src/lib/utils.ts
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}

// src/types/label.ts
var LabelWrappingType = /* @__PURE__ */ ((LabelWrappingType2) => {
  LabelWrappingType2["None"] = "None";
  LabelWrappingType2["Normal"] = "Normal";
  return LabelWrappingType2;
})(LabelWrappingType || {});
function Label({
  id,
  for: forProp,
  htmlFor,
  showColon = false,
  required = false,
  wrappingType = "Normal" /* Normal */,
  children,
  text,
  className,
  style,
  "data-testid": dataTestId,
  ref
}) {
  const labelFor = forProp ?? htmlFor;
  const labelContent = children ?? text;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "label",
    {
      ref,
      id,
      htmlFor: labelFor,
      className: cn(
        "text-sm font-normal text-sapphire-text-tertiary",
        wrappingType === "None" /* None */ && "truncate",
        wrappingType === "Normal" /* Normal */ && "break-words",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        labelContent,
        showColon && ":",
        required && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "text-destructive ml-0.5", "aria-hidden": "true", children: "*" }),
        required && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "sr-only", children: "(required)" })
      ]
    }
  );
}

exports.Label = Label;
exports.LabelWrappingType = LabelWrappingType;
