// @sap-ui/fx-components/dist/components/split-button/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');
var lucideReact = require('lucide-react');
var reactI18next = require('react-i18next');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/split-button/SplitButton.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData2,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData2 || pathData2.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData2.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M410.53 200.75C418.95 210.98 417.48 226.11 407.25 234.53L271.34 346.53C262.48 353.83 249.68 353.83 240.81 346.53L104.75 234.53C94.52 226.11 93.05 210.98 101.47 200.75C109.89 190.52 125.02 189.05 135.25 197.47L256.06 296.91L376.75 197.47C386.98 189.05 402.11 190.52 410.53 200.75Z"
];
var SlimArrowDownIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SlimArrowDownIcon", iconName: "slim-arrow-down" }
);
function isSpace(e) {
  return e.key === " " || e.key === "Spacebar";
}
function isEnter(e) {
  return e.key === "Enter";
}
function isDown(e) {
  return e.key === "ArrowDown" && !e.altKey;
}
function isUp(e) {
  return e.key === "ArrowUp" && !e.altKey;
}
function isDownAlt(e) {
  return e.key === "ArrowDown" && e.altKey;
}
function isUpAlt(e) {
  return e.key === "ArrowUp" && e.altKey;
}
function isF4(e) {
  return e.key === "F4";
}
function isShift(e) {
  return e.key === "Shift";
}
function isEscape(e) {
  return e.key === "Escape";
}
function isTab(e) {
  return e.key === "Tab";
}
function isArrowKeyAction(e) {
  return isDown(e) || isUp(e) || isDownAlt(e) || isUpAlt(e) || isF4(e);
}
function isDefaultAction(e) {
  return isSpace(e) || isEnter(e);
}
var containerVariants = cva(
  [
    "inline-flex relative group",
    "border"
  ],
  {
    variants: {
      design: {
        ["Secondary" /* Secondary */]: "bg-transparent border-sapphire-brand-foreground",
        ["Primary" /* Primary */]: "bg-sapphire-brand-background border-sapphire-brand-background",
        ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-joule-background border-sapphire-joule-background",
        ["SecondaryJoule" /* SecondaryJoule */]: "bg-transparent border-sapphire-joule-foreground",
        ["Neutral" /* Neutral */]: "bg-transparent border-sapphire-neutral-foreground-muted",
        ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-transparent border-transparent",
        ["TertiaryNeutral" /* TertiaryNeutral */]: "bg-transparent border-transparent",
        ["Tertiary" /* Tertiary */]: "bg-transparent border-transparent",
        ["TertiaryJoule" /* TertiaryJoule */]: "bg-transparent border-transparent"
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var textColorClass = {
  ["Secondary" /* Secondary */]: "text-sapphire-brand-foreground",
  ["Primary" /* Primary */]: "text-sapphire-neutral-foreground-white",
  ["PrimaryJoule" /* PrimaryJoule */]: "text-sapphire-neutral-foreground-white",
  ["SecondaryJoule" /* SecondaryJoule */]: "text-sapphire-joule-foreground",
  ["Neutral" /* Neutral */]: "text-sapphire-neutral-foreground-black",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "text-sapphire-neutral-foreground-black",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "text-sapphire-neutral-foreground-muted",
  ["Tertiary" /* Tertiary */]: "text-sapphire-brand-foreground",
  ["TertiaryJoule" /* TertiaryJoule */]: "text-sapphire-joule-foreground"
};
var textBtnHoverClass = {
  ["Secondary" /* Secondary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Primary" /* Primary */]: "hover:bg-sapphire-brand-hover-background hover:z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "hover:bg-sapphire-joule-hover-background hover:z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Neutral" /* Neutral */]: "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["Tertiary" /* Tertiary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10"
};
var textBtnActiveClass = {
  ["Secondary" /* Secondary */]: "bg-sapphire-neutral-pressed-background text-sapphire-brand-pressed-foreground z-10",
  ["Primary" /* Primary */]: "bg-sapphire-brand-pressed-background text-sapphire-neutral-foreground-white z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-joule-pressed-background text-sapphire-neutral-foreground-white z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "bg-sapphire-neutral-pressed-background text-sapphire-joule-pressed-foreground z-10",
  ["Neutral" /* Neutral */]: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "text-sapphire-neutral-foreground-black z-10",
  ["Tertiary" /* Tertiary */]: "bg-sapphire-neutral-pressed-background text-sapphire-brand-pressed-foreground z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "bg-sapphire-neutral-pressed-background text-sapphire-joule-pressed-foreground z-10"
};
var arrowBtnHoverClass = {
  ["Secondary" /* Secondary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Primary" /* Primary */]: "hover:bg-sapphire-brand-hover-background hover:z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "hover:bg-sapphire-joule-hover-background hover:z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Neutral" /* Neutral */]: "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["Tertiary" /* Tertiary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10"
};
var arrowBtnActiveClass = {
  ["Secondary" /* Secondary */]: "bg-sapphire-brand-toggle-background text-sapphire-brand-foreground z-10",
  ["Primary" /* Primary */]: "bg-sapphire-brand-hover-background text-sapphire-neutral-foreground-white z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-joule-hover-background text-sapphire-neutral-foreground-white z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "bg-sapphire-brand-toggle-background text-sapphire-joule-foreground z-10",
  ["Neutral" /* Neutral */]: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-sapphire-brand-toggle-background text-sapphire-neutral-foreground-black z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "text-sapphire-neutral-foreground-black z-10",
  ["Tertiary" /* Tertiary */]: "bg-sapphire-brand-toggle-background text-sapphire-brand-foreground z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "bg-sapphire-brand-toggle-background text-sapphire-joule-foreground z-10"
};
var arrowBtnActiveHoverClass = {
  ["Secondary" /* Secondary */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["Primary" /* Primary */]: "hover:bg-blue-800 hover:z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "hover:bg-purple-800 hover:z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["Neutral" /* Neutral */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["Tertiary" /* Tertiary */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10"
};
var separatorColor = {
  ["Secondary" /* Secondary */]: "bg-sapphire-brand-foreground",
  ["Primary" /* Primary */]: "bg-sapphire-neutral-foreground-white",
  ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-neutral-foreground-white",
  ["SecondaryJoule" /* SecondaryJoule */]: "bg-sapphire-joule-foreground",
  ["Neutral" /* Neutral */]: "bg-sapphire-neutral-foreground-black",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-sapphire-neutral-foreground-black",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "bg-sapphire-neutral-foreground-muted",
  ["Tertiary" /* Tertiary */]: "bg-sapphire-brand-foreground",
  ["TertiaryJoule" /* TertiaryJoule */]: "bg-sapphire-joule-foreground"
};
var separatorDesigns = /* @__PURE__ */ new Set([
  "Secondary" /* Secondary */,
  "SecondaryJoule" /* SecondaryJoule */,
  "Neutral" /* Neutral */
]);
var borderlessDesigns = /* @__PURE__ */ new Set([
  "Tertiary" /* Tertiary */,
  "TertiaryJoule" /* TertiaryJoule */,
  "TertiaryNeutral" /* TertiaryNeutral */,
  "SecondaryNeutral" /* SecondaryNeutral */
]);
var focusSizeClass = {
  Large: "split-btn-focus-lg",
  Medium: "split-btn-focus-md",
  Small: "split-btn-focus-sm"
};
var innerSizeClassMap = {
  Large: "split-btn-inner-lg",
  Medium: "split-btn-inner-md",
  Small: "split-btn-inner-sm"
};
var hoverRoundingMap = {
  Large: "hover:rounded-lg",
  Medium: "hover:rounded",
  Small: "hover:rounded"
};
var activeRoundingMap = {
  Large: "rounded-lg",
  Medium: "rounded",
  Small: "rounded"
};
function SplitButton({
  text,
  icon,
  endIcon,
  design = "Secondary" /* Secondary */,
  size = "Large",
  disabled = false,
  loading = false,
  activeArrowButton: activeArrowButtonProp = false,
  accessibleName,
  accessibilityAttributes,
  onClick,
  onArrowClick,
  className,
  style,
  id: idProp,
  ref
}) {
  const autoId = React2.useId();
  const id = idProp || autoId;
  const { t } = reactI18next.useTranslation("fx");
  const containerRef = React2.useRef(null);
  const textButtonRef = React2.useRef(null);
  const arrowButtonRef = React2.useRef(null);
  const [textButtonActive, setTextButtonActive] = React2.useState(false);
  const [internalArrowActive, setInternalArrowActive] = React2.useState(false);
  const [shiftOrEscapeDuringSpace, setShiftOrEscapeDuringSpace] = React2.useState(false);
  const [tabIndex, setTabIndex] = React2.useState(disabled ? -1 : 0);
  const effectiveActiveArrow = activeArrowButtonProp || internalArrowActive;
  const d = design;
  const sizeClasses = {
    Large: { container: "h-10 rounded-lg", arrow: "w-10", text: "text-base px-4 gap-2 font-semibold", icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]", chevron: "h-4 w-4" },
    Medium: { container: "h-8 rounded", arrow: "w-8", text: "text-sm px-3 gap-1.5 font-semibold", icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5", chevron: "h-3.5 w-3.5" },
    Small: { container: "h-6 rounded", arrow: "w-6", text: "text-xs px-2 gap-1 font-normal", icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]", chevron: "h-3 w-3" }
  }[size];
  React2.useImperativeHandle(ref, () => ({
    focus() {
      containerRef.current?.focus();
    },
    get nativeElement() {
      return containerRef.current;
    },
    get textButton() {
      return textButtonRef.current;
    },
    get arrowButton() {
      return arrowButtonRef.current;
    }
  }));
  const fireClick = React2.useCallback(() => {
    onClick?.();
  }, [onClick]);
  const fireArrowClick = React2.useCallback(() => {
    onArrowClick?.();
  }, [onArrowClick]);
  const resetActionStates = React2.useCallback(() => {
    setInternalArrowActive(false);
    setTextButtonActive(false);
    setShiftOrEscapeDuringSpace(false);
  }, []);
  const setTabIndexValue = React2.useCallback(
    (innerButtonPressed) => {
      let val = disabled ? -1 : 0;
      if (val === -1 && innerButtonPressed) {
        val = 0;
      }
      setTabIndex(val);
    },
    [disabled]
  );
  const handleFocusOut = React2.useCallback(
    (e) => {
      if (disabled) return;
      const related = e.relatedTarget;
      if (containerRef.current?.contains(related)) return;
      resetActionStates();
      setTabIndexValue();
    },
    [disabled, resetActionStates, setTabIndexValue]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (isArrowKeyAction(e)) {
        e.preventDefault();
        fireArrowClick();
        setInternalArrowActive(true);
        return;
      }
      if (isDefaultAction(e)) {
        e.preventDefault();
        const target = e.target;
        if (isEnter(e)) {
          if (arrowButtonRef.current && target === arrowButtonRef.current) {
            setInternalArrowActive(true);
            onArrowClick?.();
            return;
          }
          setTextButtonActive(true);
          fireClick();
          return;
        }
        if (isSpace(e)) {
          setTextButtonActive(true);
        }
        if (isTab(e)) {
          resetActionStates();
        }
        return;
      }
      if ((isShift(e) || isEscape(e)) && textButtonActive) {
        e.preventDefault();
        setShiftOrEscapeDuringSpace(true);
      }
      if (isEscape(e) && !textButtonActive) {
        resetActionStates();
      }
      setTabIndex(-1);
    },
    [disabled, loading, fireArrowClick, fireClick, onArrowClick, textButtonActive, resetActionStates]
  );
  const handleKeyUp = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (isArrowKeyAction(e)) {
        e.preventDefault();
        setInternalArrowActive(false);
        return;
      }
      if (isSpace(e)) {
        e.preventDefault();
        e.stopPropagation();
        setTextButtonActive(false);
        if (!shiftOrEscapeDuringSpace) {
          const target = e.target;
          if (arrowButtonRef.current && target === arrowButtonRef.current) {
            onArrowClick?.();
          } else {
            fireClick();
          }
        }
        setShiftOrEscapeDuringSpace(false);
        return;
      }
      if (isEnter(e) || isShift(e) && textButtonActive) {
        setTextButtonActive(false);
      }
    },
    [disabled, loading, shiftOrEscapeDuringSpace, fireClick, onArrowClick, textButtonActive]
  );
  const handleTextMouseDown = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      e.stopPropagation();
      setTextButtonActive(true);
      setTabIndex(-1);
    },
    [disabled, loading]
  );
  const handleTextMouseUp = React2.useCallback(() => {
    setTextButtonActive(false);
    setTabIndex(-1);
  }, []);
  const handleTextClick = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      e.stopPropagation();
      fireClick();
    },
    [disabled, loading, fireClick]
  );
  const handleArrowMouseDown = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      e.stopPropagation();
      setTabIndex(-1);
    },
    [disabled, loading]
  );
  const handleArrowClick = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      e.preventDefault();
      e.stopPropagation();
      fireArrowClick();
    },
    [disabled, loading, fireArrowClick]
  );
  const handleInnerFocusIn = React2.useCallback(
    (e) => {
      e.stopPropagation();
      setTabIndexValue(true);
      e.target.focus();
    },
    [setTabIndexValue]
  );
  const arrowHasPopup = accessibilityAttributes?.arrowButton?.hasPopup || "menu";
  const arrowExpanded = accessibilityAttributes?.arrowButton?.expanded ?? effectiveActiveArrow;
  const arrowTitle = accessibilityAttributes?.arrowButton?.title || t("SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP");
  const rootHasPopup = accessibilityAttributes?.root?.hasPopup;
  const rootRoleDescription = accessibilityAttributes?.root?.roleDescription || t("SPLIT_BUTTON_DESCRIPTION");
  const rootTitle = accessibilityAttributes?.root?.title;
  const rootAriaKeyShortcuts = accessibilityAttributes?.root?.ariaKeyShortcuts;
  const isDisabled = disabled || loading;
  const isBorderless = borderlessDesigns.has(d);
  const hoverRounding = isBorderless ? hoverRoundingMap[size] : "";
  const activeRounding = isBorderless ? activeRoundingMap[size] : "";
  const isFilled = d === "Primary" /* Primary */ || d === "PrimaryJoule" /* PrimaryJoule */;
  const containerFocusVariant = {
    ["Primary" /* Primary */]: "split-btn-focus split-btn-focus-filled-brand",
    ["PrimaryJoule" /* PrimaryJoule */]: "split-btn-focus split-btn-focus-filled-joule",
    ["Secondary" /* Secondary */]: "split-btn-focus split-btn-focus-brand",
    ["SecondaryJoule" /* SecondaryJoule */]: "split-btn-focus split-btn-focus-joule",
    ["Tertiary" /* Tertiary */]: "split-btn-focus split-btn-focus-brand",
    ["TertiaryJoule" /* TertiaryJoule */]: "split-btn-focus split-btn-focus-joule",
    ["Neutral" /* Neutral */]: "split-btn-focus split-btn-focus-neutral",
    ["SecondaryNeutral" /* SecondaryNeutral */]: "split-btn-focus split-btn-focus-neutral",
    ["TertiaryNeutral" /* TertiaryNeutral */]: "split-btn-focus split-btn-focus-neutral"
  };
  const containerFocusSizeClass = focusSizeClass[size];
  const containerFocusOutlineClass = cn(containerFocusVariant[d], containerFocusSizeClass);
  const innerFocusColorClass = {
    ["Primary" /* Primary */]: "split-btn-inner-white",
    ["PrimaryJoule" /* PrimaryJoule */]: "split-btn-inner-white",
    ["Secondary" /* Secondary */]: "split-btn-inner-brand",
    ["SecondaryJoule" /* SecondaryJoule */]: "split-btn-inner-joule",
    ["Tertiary" /* Tertiary */]: "split-btn-inner-brand",
    ["TertiaryJoule" /* TertiaryJoule */]: "split-btn-inner-joule",
    ["Neutral" /* Neutral */]: "split-btn-inner-neutral",
    ["SecondaryNeutral" /* SecondaryNeutral */]: "split-btn-inner-neutral",
    ["TertiaryNeutral" /* TertiaryNeutral */]: "split-btn-inner-neutral"
  };
  const innerSizeClass = innerSizeClassMap[size];
  const innerFocusClass = cn(
    "split-btn-inner",
    innerSizeClass,
    innerFocusColorClass[d]
  );
  const textFocusRing = innerFocusClass;
  const arrowFocusRing = innerFocusClass;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: containerRef,
      id,
      role: "group",
      tabIndex: isDisabled ? -1 : tabIndex,
      "aria-labelledby": `${id}-invisibleTextDefault ${id}-invisibleText`,
      "aria-haspopup": rootHasPopup,
      "aria-roledescription": rootRoleDescription,
      "aria-keyshortcuts": rootAriaKeyShortcuts,
      className: cn(
        containerVariants({ design: d }),
        sizeClasses.container,
        "shrink-0 w-fit",
        isDisabled && "pointer-events-none opacity-50",
        "outline-none",
        containerFocusOutlineClass,
        className
      ),
      style,
      onBlur: handleFocusOut,
      onKeyDown: handleKeyDown,
      onKeyUp: handleKeyUp,
      children: [
        /* @__PURE__ */ jsxRuntime.jsxs(
          "button",
          {
            ref: textButtonRef,
            type: "button",
            tabIndex: -1,
            disabled: isDisabled,
            title: rootTitle,
            onClick: handleTextClick,
            onMouseDown: handleTextMouseDown,
            onMouseUp: handleTextMouseUp,
            onFocus: handleInnerFocusIn,
            className: cn(
              "inline-flex items-center justify-start cursor-pointer relative",
              sizeClasses.text,
              "flex-grow",
              "rounded-s-[inherit] rounded-e-none",
              "bg-transparent",
              isFilled && "border-r border-r-sapphire-neutral-foreground-white",
              textColorClass[d],
              "whitespace-nowrap",
              "outline-none transition-colors",
              "disabled:pointer-events-none",
              textFocusRing,
              !textButtonActive && textBtnHoverClass[d],
              !textButtonActive && hoverRounding,
              textButtonActive && textBtnActiveClass[d],
              textButtonActive && activeRounding
            ),
            children: [
              loading ? /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: cn(sizeClasses.icon, "animate-spin") }) : icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses.icon), children: icon }),
              text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: text }),
              endIcon && !loading && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-flex items-center shrink-0", children: endIcon })
            ]
          }
        ),
        separatorDesigns.has(d) && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            "aria-hidden": "true",
            className: cn(
              "w-px shrink-0 self-stretch z-[1] pointer-events-none",
              separatorColor[d]
            )
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "button",
          {
            ref: arrowButtonRef,
            type: "button",
            tabIndex: -1,
            disabled: isDisabled,
            "aria-haspopup": arrowHasPopup,
            "aria-expanded": arrowExpanded,
            title: arrowTitle,
            onClick: handleArrowClick,
            onMouseDown: handleArrowMouseDown,
            onFocus: handleInnerFocusIn,
            className: cn(
              "inline-flex items-center justify-center cursor-pointer",
              sizeClasses.arrow,
              "shrink-0",
              "rounded-e-[inherit] rounded-s-none",
              "bg-transparent",
              textColorClass[d],
              "outline-none transition-colors overflow-visible relative",
              "disabled:pointer-events-none",
              arrowFocusRing,
              !effectiveActiveArrow && arrowBtnHoverClass[d],
              !effectiveActiveArrow && hoverRounding,
              effectiveActiveArrow && arrowBtnActiveClass[d],
              effectiveActiveArrow && arrowBtnActiveHoverClass[d],
              effectiveActiveArrow && activeRounding
            ),
            children: /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: sizeClasses.chevron })
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs("span", { id: `${id}-invisibleText`, className: "sr-only", children: [
          t("SPLIT_BUTTON_KEYBOARD_HINT"),
          accessibleName ? ` ${accessibleName}` : ""
        ] }),
        /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id}-invisibleTextDefault`, className: "sr-only", children: text })
      ]
    }
  );
}

exports.SplitButton = SplitButton;
