// @sap-ui/fx-components/dist/components/menu/index.cjs
'use strict';

var React2 = require('react');
var reactDom = require('react-dom');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');
var lucideReact = require('lucide-react');
var reactI18next = require('react-i18next');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/menu/Menu.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData4,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick2 = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData4 || pathData4.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick2,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData4.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M186 416q-11 0-18.5-7.5T160 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T186 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L203 409q-7 7-17 7z"
];
var SlimArrowRightIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SlimArrowRightIcon", iconName: "slim-arrow-right" }
);
var pathData2 = [
  "M187 416q-12 0-20-9L71 299q-7-7-7-17 0-11 7.5-18.5T90 256q12 0 19 9l77 87 217-247q8-9 19-9t18.5 7.5T448 122q0 10-6 16L206 407q-7 9-19 9z"
];
var AcceptIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "AcceptIcon", iconName: "accept" }
);
var pathData3 = [
  "M282 416q-11 0-18.5-7.5T256 390t7-18l109-109q7-7 18-7t18.5 7.5T416 282t-7 18L300 409q-7 7-18 7zm-160 0q-11 0-18.5-7.5T96 390t7-18l269-269q7-7 18-7t18.5 7.5T416 122t-7 18L140 409q-7 7-18 7z"
];
var ResizeCornerIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "ResizeCornerIcon", iconName: "resize-corner" }
);
var titleVariants = cva(
  ["text-foreground"],
  {
    variants: {
      level: {
        ["H1" /* H1 */]: "text-[3.5rem] leading-[4rem] font-light",
        ["H2" /* H2 */]: "text-[2.5rem] leading-[3rem] font-normal",
        ["H3" /* H3 */]: "text-[2rem] leading-[2.5rem] font-normal",
        ["H4" /* H4 */]: "text-xl leading-7 font-normal",
        ["H5" /* H5 */]: "text-base leading-[1.375rem] font-semibold",
        ["H6" /* H6 */]: "text-base leading-[1.375rem] font-normal"
      },
      wrappingType: {
        ["None" /* None */]: "truncate",
        ["Normal" /* Normal */]: ""
      }
    },
    defaultVariants: {
      level: "H2" /* H2 */,
      wrappingType: "None" /* None */
    }
  }
);
var levelToElement = {
  ["H1" /* H1 */]: "h1",
  ["H2" /* H2 */]: "h2",
  ["H3" /* H3 */]: "h3",
  ["H4" /* H4 */]: "h4",
  ["H5" /* H5 */]: "h5",
  ["H6" /* H6 */]: "h6"
};
function Title({
  level = "H2" /* H2 */,
  wrappingType = "None" /* None */,
  children,
  className,
  style,
  id,
  ref
}) {
  const Tag = levelToElement[level] || "h2";
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tag,
    {
      ref,
      id,
      className: cn(
        titleVariants({
          level,
          wrappingType
        }),
        className
      ),
      style,
      children
    }
  );
}

// src/lib/popupRegistry.ts
var openedPopups = [];
function attachListeners() {
  document.addEventListener("keydown", handleKeydown);
  document.addEventListener("mousedown", handleClick, true);
}
function detachListeners() {
  document.removeEventListener("keydown", handleKeydown);
  document.removeEventListener("mousedown", handleClick, true);
}
function handleKeydown(e) {
  if (e.key !== "Escape" || openedPopups.length === 0) return;
  e.preventDefault();
  e.stopImmediatePropagation();
  const topmost = openedPopups[openedPopups.length - 1];
  topmost.close(true);
}
function handleClick(e) {
  if (openedPopups.length === 0) return;
  const target = e.target;
  for (let i = openedPopups.length - 1; i >= 0; i--) {
    const entry = openedPopups[i];
    if (entry.isModal || entry.type === "dialog") return;
    if (entry.getOpenerEl) {
      const openerEl = entry.getOpenerEl();
      if (openerEl?.contains(target)) return;
    }
    if (entry.element.contains(target)) return;
    if (entry.isInsidePopup?.(target)) return;
    entry.close();
  }
}
function addOpenedPopup(entry) {
  openedPopups.push(entry);
  if (openedPopups.length === 1) {
    attachListeners();
  }
}
function removeOpenedPopup(element) {
  const idx = openedPopups.findIndex((p) => p.element === element);
  if (idx !== -1) openedPopups.splice(idx, 1);
  if (openedPopups.length === 0) {
    detachListeners();
  }
}

// src/lib/Device.ts
var isSSR = typeof document === "undefined";
function getUserAgent() {
  return isSSR ? "" : navigator.userAgent;
}
function hasTouch() {
  if (isSSR) return false;
  return "ontouchstart" in window || navigator.maxTouchPoints > 0;
}
function isIPad() {
  if (isSSR) return false;
  const ua = getUserAgent();
  return /ipad/i.test(ua) || /Macintosh/i.test(ua) && "ontouchend" in document;
}
function isWindowsPlatform() {
  if (isSSR) return false;
  return navigator.platform.indexOf("Win") !== -1;
}
function isWindows8OrAbove() {
  if (!isWindowsPlatform()) return false;
  const matches = getUserAgent().match(/Windows NT (\d+)/);
  const winVer = matches ? parseFloat(matches[1]) : 0;
  return winVer >= 8;
}
function isAndroidUA() {
  if (isSSR) return false;
  return !isWindowsPlatform() && /Android/.test(getUserAgent());
}
function isChromeUA() {
  if (isSSR) return false;
  return /(Chrome|CriOS)/.test(getUserAgent());
}
function isWebkitUA() {
  if (isSSR) return false;
  return /webkit/.test(getUserAgent());
}
function isTablet() {
  if (isSSR) return false;
  const touch = hasTouch();
  const ua = getUserAgent();
  if (isIPad()) return true;
  if (touch) {
    if (isWindows8OrAbove()) return true;
    if (isChromeUA() && isAndroidUA()) {
      return !/Mobile Safari\/[.0-9]+/.test(ua);
    }
    let densityFactor = window.devicePixelRatio || 1;
    if (isAndroidUA() && isWebkitUA()) {
      const wkMatches = ua.match(/(webkit)[ /]([\w.]+)/);
      const wkVer = wkMatches ? parseFloat(wkMatches[2]) : 0;
      if (wkVer >= 537.1) densityFactor = 1;
    }
    return Math.min(
      window.screen.width / densityFactor,
      window.screen.height / densityFactor
    ) >= 600;
  }
  return ua.indexOf("Touch") !== -1 || isAndroidUA() && !/(?=android)(?=.*mobile)/i.test(ua);
}
function isPhone() {
  if (isSSR) return false;
  return hasTouch() && !isTablet();
}
function isDesktop() {
  if (isSSR) return false;
  return !isTablet() && !isPhone() || isWindows8OrAbove();
}
var ARROW_DEPTH = 8;
var ARROW_BASE = 16;
var ARROW_SIZE = ARROW_DEPTH;
var VIEWPORT_MARGIN = 10;
function clamp(val, min, max) {
  return Math.min(Math.max(val, min), max);
}
var FOCUSABLE = 'a[href], button:not([disabled]), textarea:not([disabled]), input:not([disabled]), select:not([disabled]), [tabindex]:not([tabindex="-1"])';
function getFirstFocusable(container) {
  for (const el of container.querySelectorAll(FOCUSABLE)) {
    if (el.dataset.focusTrap) continue;
    return el;
  }
  return null;
}
function getLastFocusable(container) {
  const list = container.querySelectorAll(FOCUSABLE);
  for (let i = list.length - 1; i >= 0; i--) {
    if (list[i].dataset.focusTrap) continue;
    return list[i];
  }
  return null;
}
function resolveOpener(opener) {
  if (!opener) return null;
  if (opener instanceof HTMLElement) return opener;
  if ("current" in opener) {
    const val = opener.current;
    if (!val) return null;
    if (val instanceof HTMLElement) return val;
    if (typeof val === "object" && "nativeElement" in val && val.nativeElement instanceof HTMLElement) {
      return val.nativeElement;
    }
  }
  return null;
}
function useIsRtl(ref) {
  const [isRtl, setIsRtl] = React2.useState(false);
  React2.useEffect(() => {
    const el = ref?.current ?? document.documentElement;
    const dir = el.closest("[dir]")?.getAttribute("dir") ?? document.documentElement.getAttribute("dir");
    setIsRtl(dir === "rtl");
  });
  return isRtl;
}
function isOpenerClippedByAncestor(opener) {
  let parent = opener.parentElement;
  while (parent && parent !== document.documentElement) {
    if (parent.hasAttribute("popover")) {
      try {
        if (parent.matches(":popover-open")) return false;
      } catch {
      }
    }
    const { overflow, overflowX, overflowY } = window.getComputedStyle(parent);
    const clips = [overflow, overflowX, overflowY].some(
      (v) => v === "hidden" || v === "scroll" || v === "auto"
    );
    if (clips) {
      const parentRect = parent.getBoundingClientRect();
      const openerRect = opener.getBoundingClientRect();
      if (openerRect.bottom <= parentRect.top || openerRect.top >= parentRect.bottom || openerRect.right <= parentRect.left || openerRect.left >= parentRect.right) {
        return true;
      }
    }
    parent = parent.parentElement;
  }
  return false;
}
function getActualHorizontalAlign(align, isRtl) {
  switch (align) {
    case "Start" /* Start */:
      return isRtl ? "Right" /* Right */ : "Left" /* Left */;
    case "End" /* End */:
      return isRtl ? "Left" /* Left */ : "Right" /* Right */;
    case "Stretch" /* Stretch */:
      return "Stretch" /* Stretch */;
    default:
      return "Center" /* Center */;
  }
}
function fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) {
  if (targetRect.left > popoverSize.width) return "Left" /* Left */;
  if (clientWidth - targetRect.right > targetRect.left) return "Right" /* Right */;
  if (clientHeight - targetRect.bottom > popoverSize.height) return "Bottom" /* Bottom */;
  if (clientHeight - targetRect.bottom < targetRect.top) return "Top" /* Top */;
  return void 0;
}
function getActualPlacement(desiredPlacement, isRtl, hideArrow, targetRect, popoverSize) {
  let actual;
  switch (desiredPlacement) {
    case "Start" /* Start */:
      actual = isRtl ? "Right" /* Right */ : "Left" /* Left */;
      break;
    case "End" /* End */:
      actual = isRtl ? "Left" /* Left */ : "Right" /* Right */;
      break;
    case "Top" /* Top */:
      actual = "Top" /* Top */;
      break;
    case "Bottom" /* Bottom */:
      actual = "Bottom" /* Bottom */;
      break;
    default:
      actual = isRtl ? "Left" /* Left */ : "Right" /* Right */;
  }
  const clientWidth = document.documentElement.clientWidth;
  let clientHeight = document.documentElement.clientHeight;
  const isVertical = desiredPlacement === "Top" /* Top */ || desiredPlacement === "Bottom" /* Bottom */;
  let popoverHeight = popoverSize.height;
  if (isVertical) {
    popoverHeight += hideArrow ? 0 : ARROW_SIZE;
    clientHeight -= VIEWPORT_MARGIN;
  }
  switch (actual) {
    case "Top" /* Top */:
      if (targetRect.top < popoverHeight && targetRect.top < clientHeight - targetRect.bottom)
        actual = "Bottom" /* Bottom */;
      break;
    case "Bottom" /* Bottom */:
      if (clientHeight - targetRect.bottom < popoverHeight && clientHeight - targetRect.bottom < targetRect.top)
        actual = "Top" /* Top */;
      break;
    case "Left" /* Left */:
      if (targetRect.left < popoverSize.width)
        actual = fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) || actual;
      break;
    case "Right" /* Right */:
      if (clientWidth - targetRect.right < popoverSize.width)
        actual = fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) || actual;
      break;
  }
  return actual;
}
function calcPlacement(targetRect, popoverSize, desiredPlacement, horizontalAlign, verticalAlign, isRtl, hideArrow, allowTargetOverlap, borderRadius, offset) {
  const clientWidth = document.documentElement.clientWidth;
  const clientHeight = document.documentElement.clientHeight;
  let maxHeight = clientHeight;
  let maxWidth = clientWidth;
  let stretchWidth;
  let stretchHeight;
  const actualPlacement = getActualPlacement(
    desiredPlacement,
    isRtl,
    hideArrow,
    targetRect,
    popoverSize
  );
  const isVertical = actualPlacement === "Top" /* Top */ || actualPlacement === "Bottom" /* Bottom */;
  if (horizontalAlign === "Stretch" /* Stretch */ && isVertical) {
    popoverSize = { ...popoverSize, width: targetRect.width };
    stretchWidth = `${targetRect.width}px`;
  } else if (verticalAlign === "Stretch" /* Stretch */ && !isVertical) {
    popoverSize = { ...popoverSize, height: targetRect.height };
    stretchHeight = `${targetRect.height}px`;
  }
  const arrowOffset = (hideArrow ? 0 : ARROW_SIZE) + offset;
  const actualHAlign = getActualHorizontalAlign(horizontalAlign, isRtl);
  const getVerticalLeft = () => {
    switch (actualHAlign) {
      case "Center" /* Center */:
      case "Stretch" /* Stretch */:
        return targetRect.left - (popoverSize.width - targetRect.width) / 2;
      case "Left" /* Left */:
        return targetRect.left;
      case "Right" /* Right */:
        return targetRect.right - popoverSize.width;
      default:
        return VIEWPORT_MARGIN;
    }
  };
  const getHorizontalTop = () => {
    switch (verticalAlign) {
      case "Center" /* Center */:
      case "Stretch" /* Stretch */:
        return targetRect.top - (popoverSize.height - targetRect.height) / 2;
      case "Top" /* Top */:
        return targetRect.top;
      case "Bottom" /* Bottom */:
        return targetRect.bottom - popoverSize.height;
      default:
        return 0;
    }
  };
  let left = VIEWPORT_MARGIN;
  let top = 0;
  switch (actualPlacement) {
    case "Top" /* Top */:
      left = getVerticalLeft();
      top = Math.max(targetRect.top - popoverSize.height - arrowOffset, 0);
      if (!allowTargetOverlap) maxHeight = targetRect.top - arrowOffset;
      break;
    case "Bottom" /* Bottom */:
      left = getVerticalLeft();
      top = targetRect.bottom + arrowOffset;
      if (allowTargetOverlap) {
        top = Math.max(Math.min(top, clientHeight - popoverSize.height), 0);
      } else {
        maxHeight = clientHeight - targetRect.bottom - arrowOffset;
      }
      break;
    case "Left" /* Left */:
      left = Math.max(targetRect.left - popoverSize.width - arrowOffset, 0);
      top = getHorizontalTop();
      if (!allowTargetOverlap) maxWidth = targetRect.left - arrowOffset;
      break;
    case "Right" /* Right */:
      left = targetRect.left + targetRect.width + arrowOffset;
      top = getHorizontalTop();
      if (allowTargetOverlap) {
        left = Math.max(Math.min(left, clientWidth - popoverSize.width), 0);
      } else {
        maxWidth = clientWidth - targetRect.right - arrowOffset;
      }
      break;
  }
  if (isVertical) {
    if (popoverSize.width > clientWidth || left < VIEWPORT_MARGIN) {
      left = VIEWPORT_MARGIN;
    } else if (left + popoverSize.width > clientWidth - VIEWPORT_MARGIN) {
      left = clientWidth - VIEWPORT_MARGIN - popoverSize.width;
    }
  } else {
    if (popoverSize.height > clientHeight || top < VIEWPORT_MARGIN) {
      top = VIEWPORT_MARGIN;
    } else if (top + popoverSize.height > clientHeight - VIEWPORT_MARGIN) {
      top = clientHeight - VIEWPORT_MARGIN - popoverSize.height;
    }
  }
  left = Math.round(left);
  top = Math.round(top);
  maxHeight = Math.round(maxHeight - VIEWPORT_MARGIN);
  maxWidth = Math.round(maxWidth - VIEWPORT_MARGIN);
  let arrowXCentered = actualHAlign === "Center" /* Center */ || actualHAlign === "Stretch" /* Stretch */;
  if (actualHAlign === "Right" /* Right */ && left <= targetRect.left)
    arrowXCentered = true;
  if (actualHAlign === "Left" /* Left */ && left + popoverSize.width >= targetRect.left + targetRect.width)
    arrowXCentered = true;
  let arrowTranslateX = 0;
  if (isVertical && arrowXCentered) {
    arrowTranslateX = targetRect.left + targetRect.width / 2 - left - popoverSize.width / 2;
  }
  let arrowTranslateY = 0;
  if (!isVertical) {
    arrowTranslateY = targetRect.top + targetRect.height / 2 - top - popoverSize.height / 2;
  }
  const safeY = popoverSize.height / 2 - borderRadius - ARROW_BASE / 2 - 2;
  arrowTranslateY = clamp(arrowTranslateY, -safeY, safeY);
  const safeX = popoverSize.width / 2 - borderRadius - ARROW_BASE / 2 - 2;
  arrowTranslateX = clamp(arrowTranslateX, -safeX, safeX);
  return {
    arrow: { x: Math.round(arrowTranslateX), y: Math.round(arrowTranslateY) },
    top,
    left,
    maxHeight,
    maxWidth,
    actualPlacement,
    stretchWidth,
    stretchHeight
  };
}
function getResizeHandleCorner(popoverRect, openerRect, side, verticalAlign, actualHAlign, isRtl) {
  const offset = 2;
  let openerCX = Math.floor(openerRect.x + openerRect.width / 2);
  const openerCY = Math.floor(openerRect.y + openerRect.height / 2);
  let popoverCX = Math.floor(popoverRect.x + popoverRect.width / 2);
  const popoverCY = Math.floor(popoverRect.y + popoverRect.height / 2);
  const isPopoverWider = popoverRect.width > openerRect.width;
  const isPopoverTaller = popoverRect.height > openerRect.height;
  if (isRtl) {
    openerCX = -openerCX;
    popoverCX = -popoverCX;
  }
  switch (side) {
    case "Left" /* Left */:
      if (isPopoverTaller) {
        return popoverCY > openerCY + offset ? "bottom-left" : "top-left";
      }
      return verticalAlign === "Top" /* Top */ ? "bottom-left" : "top-left";
    case "Right" /* Right */:
      if (isPopoverTaller) {
        return popoverCY + offset < openerCY ? "top-right" : "bottom-right";
      }
      return verticalAlign === "Bottom" /* Bottom */ ? "top-right" : "bottom-right";
    case "Bottom" /* Bottom */:
      if (isPopoverWider) {
        if (popoverCX + offset < openerCX) {
          return isRtl ? "bottom-right" : "bottom-left";
        }
        return isRtl ? "bottom-left" : "bottom-right";
      }
      if (isRtl) {
        return actualHAlign === "Left" /* Left */ ? "bottom-right" : "bottom-left";
      }
      return actualHAlign === "Right" /* Right */ ? "bottom-left" : "bottom-right";
    case "Top" /* Top */:
    default:
      if (isPopoverWider) {
        if (popoverCX + offset < openerCX) {
          return isRtl ? "top-right" : "top-left";
        }
        return isRtl ? "top-left" : "top-right";
      }
      if (isRtl) {
        return actualHAlign === "Left" /* Left */ ? "top-right" : "top-left";
      }
      return actualHAlign === "Right" /* Right */ ? "top-left" : "top-right";
  }
}
function Popover({
  headerText,
  placement = "End" /* End */,
  horizontalAlign = "Center" /* Center */,
  verticalAlign = "Center" /* Center */,
  hideArrow = false,
  offset: offsetProp,
  allowTargetOverlap = false,
  resizable = false,
  noPadding = false,
  opener,
  open = false,
  initialFocus,
  preventFocusRestore = false,
  preventInitialFocus = false,
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Dialog" /* Dialog */,
  accessibleDescription,
  header,
  footer,
  children,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  onClose,
  isInsidePopup,
  className,
  style,
  id,
  ref
}) {
  const popoverRef = React2.useRef(null);
  const arrowRef = React2.useRef(null);
  const focusedBeforeOpen = React2.useRef(null);
  const [isOpen, setIsOpen] = React2.useState(false);
  const [pos, setPos] = React2.useState(null);
  const [resizeSize, setResizeSize] = React2.useState(null);
  const resizeState = React2.useRef(null);
  const [isDesktop2] = React2.useState(() => isDesktop());
  const containerRef = React2.useRef(null);
  const isRtl = useIsRtl(containerRef);
  const getOpenerEl = React2.useCallback(() => resolveOpener(opener), [opener]);
  const doCloseRef = React2.useRef(() => {
  });
  const reposition = React2.useCallback(() => {
    const openerEl = getOpenerEl();
    const popEl = popoverRef.current;
    if (!openerEl || !popEl) return;
    const targetRect = openerEl.getBoundingClientRect();
    if (targetRect.bottom < 0 || targetRect.top > window.innerHeight || targetRect.right < 0 || targetRect.left > window.innerWidth) {
      doCloseRef.current();
      return;
    }
    if (targetRect.top === 0 && targetRect.bottom === 0 && targetRect.left === 0 && targetRect.right === 0) {
      doCloseRef.current();
      return;
    }
    if (isOpenerClippedByAncestor(openerEl)) {
      doCloseRef.current();
      return;
    }
    const rect = popEl.getBoundingClientRect();
    const popoverSize = { width: rect.width, height: rect.height };
    if (popoverSize.width === 0 || popoverSize.height === 0) return;
    const borderRadius = Number.parseInt(
      window.getComputedStyle(popEl).getPropertyValue("border-radius") || "0"
    );
    const result = calcPlacement(
      targetRect,
      popoverSize,
      placement,
      horizontalAlign,
      verticalAlign,
      isRtl,
      hideArrow,
      allowTargetOverlap,
      borderRadius,
      offsetProp ?? 0
    );
    setPos(result);
  }, [getOpenerEl, placement, horizontalAlign, verticalAlign, isRtl, hideArrow, allowTargetOverlap, offsetProp]);
  React2.useEffect(() => {
    if (!isOpen) return;
    const raf = requestAnimationFrame(() => {
      reposition();
      requestAnimationFrame(reposition);
    });
    const onScrollResize = () => reposition();
    window.addEventListener("scroll", onScrollResize, true);
    window.addEventListener("resize", onScrollResize);
    const openerEl = getOpenerEl();
    let ro = null;
    if (openerEl instanceof Element) {
      ro = new ResizeObserver(reposition);
      ro.observe(openerEl);
    }
    let roPopover = null;
    if (popoverRef.current instanceof Element) {
      roPopover = new ResizeObserver(reposition);
      roPopover.observe(popoverRef.current);
    }
    return () => {
      cancelAnimationFrame(raf);
      window.removeEventListener("scroll", onScrollResize, true);
      window.removeEventListener("resize", onScrollResize);
      ro?.disconnect();
      roPopover?.disconnect();
    };
  }, [isOpen, reposition, getOpenerEl]);
  const doOpen = React2.useCallback(() => {
    if (isOpen) return;
    if (onBeforeOpen?.() === false) return;
    focusedBeforeOpen.current = document.activeElement;
    setIsOpen(true);
  }, [isOpen, onBeforeOpen]);
  const doClose = React2.useCallback(
    (escPressed = false) => {
      if (!isOpen) return;
      if (onBeforeClose?.({ escPressed }) === false) return;
      const el = popoverRef.current;
      if (el && el.matches(":popover-open")) {
        el.hidePopover();
      }
      setIsOpen(false);
      setResizeSize(null);
      setPos(null);
      if (!preventFocusRestore && focusedBeforeOpen.current) {
        focusedBeforeOpen.current.focus();
        focusedBeforeOpen.current = null;
      }
      onClose?.();
    },
    [isOpen, onBeforeClose, onClose, preventFocusRestore]
  );
  doCloseRef.current = doClose;
  React2.useEffect(() => {
    if (open) doOpen();
    else if (isOpen) doClose();
  }, [open]);
  React2.useEffect(() => {
    const el = popoverRef.current;
    if (!el) return;
    if (isOpen && !el.matches(":popover-open")) {
      el.showPopover();
    }
    return () => {
      if (el.matches(":popover-open")) {
        el.hidePopover();
      }
    };
  }, [isOpen]);
  React2.useEffect(() => {
    if (!isOpen) return;
    requestAnimationFrame(() => {
      if (preventInitialFocus || !popoverRef.current) return;
      if (initialFocus) {
        const el = document.getElementById(initialFocus);
        if (el) {
          el.focus();
          onOpen?.();
          return;
        }
      }
      const first = getFirstFocusable(popoverRef.current);
      if (first) first.focus();
      else popoverRef.current.focus();
      onOpen?.();
    });
  }, [isOpen]);
  React2.useEffect(() => {
    if (!isOpen || !popoverRef.current) return;
    const el = popoverRef.current;
    addOpenedPopup({
      element: el,
      close: (escPressed) => doCloseRef.current(escPressed),
      type: "popover",
      isModal: false,
      getOpenerEl,
      isInsidePopup
    });
    return () => removeOpenedPopup(el);
  }, [isOpen, getOpenerEl, isInsidePopup]);
  const focusFirst = React2.useCallback(() => {
    if (!popoverRef.current) return;
    const el = getFirstFocusable(popoverRef.current);
    (el || popoverRef.current).focus();
  }, []);
  const focusLast = React2.useCallback(() => {
    if (!popoverRef.current) return;
    const el = getLastFocusable(popoverRef.current);
    (el || popoverRef.current).focus();
  }, []);
  const handleResizeMouseDown = React2.useCallback(
    (e) => {
      if (!resizable || !popoverRef.current) return;
      e.preventDefault();
      const rect = popoverRef.current.getBoundingClientRect();
      const openerEl = getOpenerEl();
      const opRect = openerEl?.getBoundingClientRect();
      const corner = opRect && pos ? getResizeHandleCorner(
        rect,
        opRect,
        pos.actualPlacement,
        verticalAlign,
        getActualHorizontalAlign(horizontalAlign, isRtl),
        isRtl
      ) : "bottom-right";
      resizeState.current = {
        startX: e.clientX,
        startY: e.clientY,
        startWidth: rect.width,
        startHeight: rect.height,
        corner
      };
      const onMove = (ev) => {
        if (!resizeState.current) return;
        const { startX, startY, startWidth, startHeight, corner: c } = resizeState.current;
        const dx = ev.clientX - startX;
        const dy = ev.clientY - startY;
        const isL = c === "top-left" || c === "bottom-left";
        const isT = c === "top-left" || c === "top-right";
        const w = clamp(startWidth + (isL ? -dx : dx), 150, window.innerWidth - VIEWPORT_MARGIN * 2);
        const h = clamp(startHeight + (isT ? -dy : dy), 100, window.innerHeight - VIEWPORT_MARGIN * 2);
        setResizeSize({ width: w, height: h });
      };
      const onUp = () => {
        resizeState.current = null;
        window.removeEventListener("mousemove", onMove);
        window.removeEventListener("mouseup", onUp);
      };
      window.addEventListener("mousemove", onMove);
      window.addEventListener("mouseup", onUp);
    },
    [resizable, getOpenerEl, pos]
  );
  React2.useImperativeHandle(ref, () => ({
    open() {
      doOpen();
    },
    close() {
      doClose();
    },
    async applyFocus() {
      if (!popoverRef.current) return;
      const el = getFirstFocusable(popoverRef.current);
      (el || popoverRef.current).focus();
    },
    isOpen() {
      return isOpen;
    },
    get nativeElement() {
      return popoverRef.current;
    }
  }), [isOpen, doOpen, doClose]);
  const resizeHandleCorner = React2.useMemo(() => {
    if (!resizable || !isDesktop2 || !popoverRef.current || !pos) return "bottom-right";
    const openerEl = getOpenerEl();
    if (!openerEl) return "bottom-right";
    return getResizeHandleCorner(
      popoverRef.current.getBoundingClientRect(),
      openerEl.getBoundingClientRect(),
      pos.actualPlacement,
      verticalAlign,
      getActualHorizontalAlign(horizontalAlign, isRtl),
      isRtl
    );
  }, [resizable, isDesktop2, getOpenerEl, pos, verticalAlign, horizontalAlign, isRtl]);
  const role = accessibleRole === "None" /* None */ ? void 0 : accessibleRole === "AlertDialog" /* AlertDialog */ ? "alertdialog" : "dialog";
  const ariaModal = accessibleRole === "None" /* None */ ? void 0 : "true";
  const displayHeader = !!(header || headerText);
  const actualPlacement = pos?.actualPlacement ?? "Right" /* Right */;
  const resizeHandleClasses = {
    "top-left": "-top-2 -left-2 cursor-nw-resize",
    "top-right": "-top-2 -right-2 cursor-ne-resize",
    "bottom-left": "-bottom-2 -left-2 cursor-ne-resize",
    "bottom-right": "-bottom-2 -right-2 cursor-nw-resize"
  };
  const resizeIconRotation = {
    "top-left": "180deg",
    "top-right": "270deg",
    "bottom-left": "90deg",
    "bottom-right": "0deg"
  };
  if (!isOpen) {
    return /* @__PURE__ */ jsxRuntime.jsx("div", { ref: containerRef, className: "hidden" });
  }
  const popoverStyle = {
    position: "fixed",
    top: pos ? pos.top : -1e4,
    left: pos ? pos.left : -1e4,
    maxHeight: pos ? pos.maxHeight : void 0,
    maxWidth: pos ? pos.maxWidth : void 0,
    boxShadow: "0px 10px 15px -5px rgba(10, 10, 10, 0.15), 0px 5px 5px -2px rgba(10, 10, 10, 0.05)",
    ...pos?.stretchWidth ? { width: pos.stretchWidth } : {},
    ...pos?.stretchHeight ? { height: pos.stretchHeight } : {},
    ...resizeSize ? { width: resizeSize.width, height: resizeSize.height } : {},
    ...style
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsx("div", { ref: containerRef, className: "hidden" }),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: popoverRef,
        id,
        popover: "manual",
        role,
        "aria-modal": ariaModal,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef || (!accessibleName && headerText ? `${id || "popover"}-title` : void 0),
        "aria-describedby": accessibleDescription ? `${id || "popover"}-desc` : void 0,
        tabIndex: -1,
        className: cn(
          "rounded-lg border border-border bg-popover text-popover-foreground outline-none",
          className
        ),
        style: popoverStyle,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", { tabIndex: 0, onFocus: focusLast, "data-focus-trap": "true", className: "sr-only", role: "none" }),
          !hideArrow && pos && (() => {
            const arrowBorderColor = "var(--border)";
            const arrowFill = "var(--popover)";
            let svgWidth;
            let svgHeight;
            let fillPoints;
            let strokePath;
            const wrapperStyle = { position: "absolute", lineHeight: 0 };
            switch (actualPlacement) {
              case "Bottom" /* Bottom */:
                svgWidth = ARROW_BASE;
                svgHeight = ARROW_DEPTH;
                fillPoints = `0,${ARROW_DEPTH} ${ARROW_BASE / 2},0 ${ARROW_BASE},${ARROW_DEPTH}`;
                strokePath = `M0,${ARROW_DEPTH} L${ARROW_BASE / 2},0 L${ARROW_BASE},${ARROW_DEPTH}`;
                wrapperStyle.left = "50%";
                wrapperStyle.top = -ARROW_DEPTH;
                wrapperStyle.transform = `translateX(calc(-50% + ${pos.arrow.x}px))`;
                break;
              case "Top" /* Top */:
                svgWidth = ARROW_BASE;
                svgHeight = ARROW_DEPTH;
                fillPoints = `0,0 ${ARROW_BASE / 2},${ARROW_DEPTH} ${ARROW_BASE},0`;
                strokePath = `M0,0 L${ARROW_BASE / 2},${ARROW_DEPTH} L${ARROW_BASE},0`;
                wrapperStyle.left = "50%";
                wrapperStyle.bottom = -ARROW_DEPTH;
                wrapperStyle.transform = `translateX(calc(-50% + ${pos.arrow.x}px))`;
                break;
              case "Right" /* Right */:
                svgWidth = ARROW_DEPTH;
                svgHeight = ARROW_BASE;
                fillPoints = `${ARROW_DEPTH},0 0,${ARROW_BASE / 2} ${ARROW_DEPTH},${ARROW_BASE}`;
                strokePath = `M${ARROW_DEPTH},0 L0,${ARROW_BASE / 2} L${ARROW_DEPTH},${ARROW_BASE}`;
                wrapperStyle.top = "50%";
                wrapperStyle.left = -ARROW_DEPTH;
                wrapperStyle.transform = `translateY(calc(-50% + ${pos.arrow.y}px))`;
                break;
              case "Left" /* Left */:
              default:
                svgWidth = ARROW_DEPTH;
                svgHeight = ARROW_BASE;
                fillPoints = `0,0 ${ARROW_DEPTH},${ARROW_BASE / 2} 0,${ARROW_BASE}`;
                strokePath = `M0,0 L${ARROW_DEPTH},${ARROW_BASE / 2} L0,${ARROW_BASE}`;
                wrapperStyle.top = "50%";
                wrapperStyle.right = -ARROW_DEPTH;
                wrapperStyle.transform = `translateY(calc(-50% + ${pos.arrow.y}px))`;
                break;
            }
            return /* @__PURE__ */ jsxRuntime.jsx("div", { ref: arrowRef, style: wrapperStyle, children: /* @__PURE__ */ jsxRuntime.jsxs("svg", { width: svgWidth, height: svgHeight, viewBox: `0 0 ${svgWidth} ${svgHeight}`, style: { display: "block" }, children: [
              /* @__PURE__ */ jsxRuntime.jsx("polygon", { points: fillPoints, fill: arrowFill, stroke: "none" }),
              /* @__PURE__ */ jsxRuntime.jsx("path", { d: strokePath, fill: "none", stroke: arrowBorderColor, strokeWidth: 1 })
            ] }) });
          })(),
          /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "isolate z-1 flex flex-col gap-sapphire-2xs overflow-hidden rounded-lg", style: { maxHeight: "inherit" }, children: [
            displayHeader && /* @__PURE__ */ jsxRuntime.jsx(
              "header",
              {
                className: cn(
                  "shrink-0",
                  // When custom header is provided, let it control its own styling.
                  // When using headerText, apply default popover header styling.
                  !header && "px-sapphire-s py-sapphire-xs border-b border-sapphire-border-primary"
                ),
                children: header || /* @__PURE__ */ jsxRuntime.jsx(Title, { level: "H1", id: `${id || "popover"}-title`, className: "text-base font-bold", children: headerText })
              }
            ),
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: cn("overflow-y-auto overflow-x-hidden flex-1 min-h-0", !noPadding && "px-sapphire-s py-sapphire-xs"), children }),
            accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: `${id || "popover"}-desc`, className: "sr-only", children: accessibleDescription }),
            footer && /* @__PURE__ */ jsxRuntime.jsx("footer", { className: "px-sapphire-s py-sapphire-xs border-t border-sapphire-border-primary shrink-0", children: footer })
          ] }),
          resizable && isDesktop2 && /* @__PURE__ */ jsxRuntime.jsx(
            "div",
            {
              className: cn(
                "absolute flex items-center justify-center w-6 h-6 rounded-full z-[1]",
                resizeHandleClasses[resizeHandleCorner]
              ),
              onMouseDown: handleResizeMouseDown,
              children: /* @__PURE__ */ jsxRuntime.jsx(
                ResizeCornerIcon,
                {
                  className: "h-4 w-4 text-sapphire-text-tertiary",
                  style: { transform: `rotate(${resizeIconRotation[resizeHandleCorner]})` }
                }
              )
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsx("span", { tabIndex: 0, onFocus: focusFirst, "data-focus-trap": "true", className: "sr-only", role: "none" })
        ]
      }
    )
  ] });
}
var MenuContext = React2.createContext({
  closeMenu: () => {
  },
  closeAllSubmenus: () => {
  },
  registerSubmenu: () => {
  },
  unregisterSubmenu: () => {
  },
  submenuDepth: 0
});
var MenuItemGroupContext = React2.createContext(null);
function calculateSubmenuPosition(parentItem, submenu) {
  const parentRect = parentItem.getBoundingClientRect();
  const submenuRect = submenu.getBoundingClientRect();
  const viewportWidth = document.documentElement.clientWidth;
  const viewportHeight = document.documentElement.clientHeight;
  const gap = 4;
  const margin = 8;
  const spaceRight = viewportWidth - parentRect.right - gap;
  const spaceLeft = parentRect.left - gap;
  let top = parentRect.top;
  let left;
  if (spaceRight >= submenuRect.width) {
    left = parentRect.right + gap;
  } else if (spaceLeft >= submenuRect.width) {
    left = parentRect.left - submenuRect.width - gap;
  } else {
    const parentMenu = parentItem.closest('[role="menu"], [data-submenu="true"]');
    if (parentMenu) {
      const parentMenuRect = parentMenu.getBoundingClientRect();
      left = parentMenuRect.left;
    } else {
      left = parentRect.left;
    }
    left = Math.max(Math.min(left, viewportWidth - submenuRect.width - margin), margin);
  }
  if (top + submenuRect.height > viewportHeight - margin) {
    top = viewportHeight - submenuRect.height - margin;
  }
  if (top < margin) {
    top = margin;
  }
  return { top, left };
}
function getFocusableItems(container) {
  const all = Array.from(
    container.querySelectorAll(
      '[role^="menuitem"]:not([aria-disabled="true"])'
    )
  );
  return all.filter((item) => {
    let parent = item.parentElement;
    while (parent && parent !== container) {
      if (parent.getAttribute("role") === "menu" || parent.getAttribute("data-submenu") === "true") {
        return false;
      }
      parent = parent.parentElement;
    }
    return true;
  });
}
function useMenuNavigation(containerRef, options) {
  const typeAheadBufferRef = React2.useRef("");
  const typeAheadTimeoutRef = React2.useRef(null);
  const handleKeyDown = React2.useCallback(
    (e) => {
      const container = containerRef.current;
      if (!container) return;
      const items = getFocusableItems(container);
      if (items.length === 0) return;
      const currentIndex = items.indexOf(
        document.activeElement
      );
      switch (e.key) {
        case "ArrowDown": {
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex < items.length - 1) {
            items[currentIndex + 1]?.focus();
          }
          break;
        }
        case "ArrowUp": {
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex > 0) {
            items[currentIndex - 1]?.focus();
          }
          break;
        }
        case "Home": {
          e.preventDefault();
          e.stopPropagation();
          items[0]?.focus();
          break;
        }
        case "End": {
          e.preventDefault();
          e.stopPropagation();
          items[items.length - 1]?.focus();
          break;
        }
        case "ArrowLeft": {
          if (options?.isSubmenu) {
            e.preventDefault();
            e.stopPropagation();
            options.onClose?.();
            options.parentItemRef?.current?.focus();
          }
          break;
        }
        case "Escape": {
          if (options?.isSubmenu) {
            e.preventDefault();
            e.stopPropagation();
            e.nativeEvent.stopImmediatePropagation();
            options.onClose?.();
            options.parentItemRef?.current?.focus();
          }
          break;
        }
        case "Tab": {
          e.preventDefault();
          if (options?.isSubmenu) {
            e.stopPropagation();
            e.nativeEvent.stopImmediatePropagation();
            options.onClose?.();
            options.parentItemRef?.current?.focus();
          } else {
            options?.onCloseRoot?.();
          }
          break;
        }
        default: {
          if (e.key.length === 1 && !e.ctrlKey && !e.metaKey && !e.altKey) {
            e.stopPropagation();
            if (typeAheadTimeoutRef.current) {
              clearTimeout(typeAheadTimeoutRef.current);
            }
            typeAheadBufferRef.current += e.key.toLowerCase();
            typeAheadTimeoutRef.current = setTimeout(() => {
              typeAheadBufferRef.current = "";
            }, 500);
            const buffer = typeAheadBufferRef.current;
            const match = items.find((item) => {
              const text = item.textContent?.toLowerCase() || "";
              return text.startsWith(buffer);
            });
            if (match) {
              match.focus();
            }
          }
          break;
        }
      }
    },
    [containerRef, options]
  );
  React2.useEffect(() => {
    return () => {
      if (typeAheadTimeoutRef.current) {
        clearTimeout(typeAheadTimeoutRef.current);
      }
    };
  }, []);
  return handleKeyDown;
}
function LoadingSpinner({ delay = 50 }) {
  const [show, setShow] = React2.useState(delay <= 0);
  React2.useEffect(() => {
    if (delay <= 0) return;
    const timer = setTimeout(() => setShow(true), delay);
    return () => clearTimeout(timer);
  }, [delay]);
  if (!show) return null;
  return /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex items-center justify-center py-4 px-8", children: /* @__PURE__ */ jsxRuntime.jsx(lucideReact.Loader2, { className: "h-5 w-5 animate-spin text-sapphire-text-tertiary" }) });
}
function mapMenuPlacementToPopover(menuPlacement = "bottom-start") {
  switch (menuPlacement) {
    case "bottom-start":
      return { placement: "Bottom" /* Bottom */, horizontalAlign: "Start" /* Start */ };
    case "bottom-end":
      return { placement: "Bottom" /* Bottom */, horizontalAlign: "End" /* End */ };
    case "top-start":
      return { placement: "Top" /* Top */, horizontalAlign: "Start" /* Start */ };
    case "top-end":
      return { placement: "Top" /* Top */, horizontalAlign: "End" /* End */ };
    case "right-start":
      return { placement: "End" /* End */, verticalAlign: "Top" /* Top */ };
    case "right-end":
      return { placement: "End" /* End */, verticalAlign: "Bottom" /* Bottom */ };
    case "left-start":
      return { placement: "Start" /* Start */, verticalAlign: "Top" /* Top */ };
    case "left-end":
      return { placement: "Start" /* Start */, verticalAlign: "Bottom" /* Bottom */ };
    default:
      return { placement: "Bottom" /* Bottom */, horizontalAlign: "Start" /* Start */ };
  }
}
function Menu({
  open = false,
  opener,
  children,
  placement = "bottom-start",
  onClose,
  onItemClick,
  onCheck,
  className,
  style,
  headerText,
  loading,
  loadingDelay = 50,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  ref
}) {
  const { t } = reactI18next.useTranslation("fx");
  const menuRef = React2.useRef(null);
  const submenuClosersRef = React2.useRef(/* @__PURE__ */ new Map());
  const closingRef = React2.useRef(false);
  const onBeforeCloseRef = React2.useRef(onBeforeClose);
  const onCloseRef = React2.useRef(onClose);
  const onItemClickRef = React2.useRef(onItemClick);
  const onCheckRef = React2.useRef(onCheck);
  React2.useEffect(() => {
    onBeforeCloseRef.current = onBeforeClose;
    onCloseRef.current = onClose;
    onItemClickRef.current = onItemClick;
    onCheckRef.current = onCheck;
  });
  const resolvedOpener = typeof opener === "string" ? document.getElementById(opener) : opener;
  const openerRef = React2.useRef(null);
  React2.useEffect(() => {
    openerRef.current = resolvedOpener ?? null;
  }, [resolvedOpener]);
  const closeMenu = React2.useCallback((escPressed) => {
    closingRef.current = true;
    onBeforeCloseRef.current?.({ escPressed: !!escPressed });
    onCloseRef.current?.();
  }, []);
  React2.useImperativeHandle(ref, () => ({
    close: () => closeMenu(),
    isOpen: () => open
  }), [closeMenu, open]);
  React2.useEffect(() => {
    if (open) {
      closingRef.current = false;
    }
  }, [open]);
  const closeAllSubmenus = React2.useCallback(() => {
    submenuClosersRef.current.forEach((close) => close());
  }, []);
  const registerSubmenu = React2.useCallback((id, close) => {
    submenuClosersRef.current.set(id, close);
  }, []);
  const unregisterSubmenu = React2.useCallback((id) => {
    submenuClosersRef.current.delete(id);
  }, []);
  const handleKeyDown = useMenuNavigation(menuRef, { onCloseRoot: closeMenu });
  const stableOnItemClick = React2.useCallback(
    (detail) => onItemClickRef.current?.(detail),
    []
  );
  const stableOnCheck = React2.useCallback(
    (detail) => onCheckRef.current?.(detail),
    []
  );
  const prevOpenLoadingRef = React2.useRef({ open, loading });
  React2.useEffect(() => {
    const prev = prevOpenLoadingRef.current;
    prevOpenLoadingRef.current = { open, loading };
    if (prev.open && open && prev.loading && !loading && menuRef.current) {
      requestAnimationFrame(() => {
        if (menuRef.current) {
          const items = getFocusableItems(menuRef.current);
          const selected = items.find((el) => el.hasAttribute("data-menu-selected") || el.getAttribute("aria-checked") === "true");
          (selected ?? items[0])?.focus();
        }
      });
    }
  }, [loading, open]);
  const menuContextValue = React2.useMemo(
    () => ({ closeMenu, onItemClick: stableOnItemClick, onCheck: stableOnCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, submenuDepth: 0 }),
    [closeMenu, stableOnItemClick, stableOnCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu]
  );
  const popoverPlacement = React2.useMemo(
    () => mapMenuPlacementToPopover(placement),
    [placement]
  );
  const handleOpen = React2.useCallback(() => {
    onOpen?.();
    if (!menuRef.current) return;
    const items = getFocusableItems(menuRef.current);
    const selected = items.find((el) => el.hasAttribute("data-menu-selected") || el.getAttribute("aria-checked") === "true");
    (selected ?? items[0])?.focus();
  }, [onOpen]);
  const isInsidePopup = React2.useCallback((target) => {
    return !!target.closest?.('[data-submenu="true"]');
  }, []);
  const handlePopoverBeforeClose = React2.useCallback(({ escPressed }) => {
    if (closingRef.current) return;
    onBeforeCloseRef.current?.({ escPressed });
  }, []);
  const handlePopoverClose = React2.useCallback(() => {
    const wasClosing = closingRef.current;
    closingRef.current = false;
    if (wasClosing) return;
    onCloseRef.current?.();
  }, []);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Popover,
    {
      opener: openerRef,
      open,
      onBeforeOpen,
      onOpen: handleOpen,
      onBeforeClose: handlePopoverBeforeClose,
      onClose: handlePopoverClose,
      hideArrow: true,
      noPadding: true,
      offset: 4,
      placement: popoverPlacement.placement,
      horizontalAlign: popoverPlacement.horizontalAlign,
      verticalAlign: popoverPlacement.verticalAlign,
      accessibleRole: "Dialog" /* Dialog */,
      accessibleName: t("MENU_POPOVER_ACCESSIBLE_NAME"),
      isInsidePopup,
      className: "p-0 border border-sapphire-border-primary shadow-lg [&>div]:overflow-visible",
      style: {
        filter: "none",
        ...style
      },
      children: /* @__PURE__ */ jsxRuntime.jsx(MenuContext.Provider, { value: menuContextValue, children: /* @__PURE__ */ jsxRuntime.jsxs(
        "div",
        {
          ref: menuRef,
          role: "menu",
          tabIndex: -1,
          className: cn("min-w-[18.75rem] py-sapphire-xs outline-none", className),
          onKeyDown: handleKeyDown,
          children: [
            headerText && /* @__PURE__ */ jsxRuntime.jsx(MenuHeader, { text: headerText }),
            loading ? /* @__PURE__ */ jsxRuntime.jsx(LoadingSpinner, { delay: loadingDelay }) : children
          ]
        }
      ) })
    }
  );
}
function SubmenuScope({ closeMenu, onItemClick, onCheck, parentDepth, children }) {
  const submenuClosersRef = React2.useRef(/* @__PURE__ */ new Map());
  const closeAllSubmenus = React2.useCallback(() => {
    submenuClosersRef.current.forEach((close) => close());
  }, []);
  const registerSubmenu = React2.useCallback((id, close) => {
    submenuClosersRef.current.set(id, close);
  }, []);
  const unregisterSubmenu = React2.useCallback((id) => {
    submenuClosersRef.current.delete(id);
  }, []);
  const depth = parentDepth + 1;
  const contextValue = React2.useMemo(
    () => ({ closeMenu, onItemClick, onCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, submenuDepth: depth }),
    [closeMenu, onItemClick, onCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, depth]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(MenuContext.Provider, { value: contextValue, children });
}
function MenuItem({
  text,
  icon,
  endContent,
  disabled = false,
  startsSection = false,
  children,
  onClick,
  className,
  data,
  checked,
  additionalText,
  loading: itemLoading,
  loadingDelay = 50,
  tooltip,
  accessibleName,
  accessibilityAttributes,
  ref
}) {
  const { closeMenu, onItemClick, onCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, submenuDepth } = React2.useContext(MenuContext);
  const groupContext = React2.useContext(MenuItemGroupContext);
  const itemRef = React2.useRef(null);
  const submenuRef = React2.useRef(null);
  const [submenuOpen, setSubmenuOpen] = React2.useState(false);
  const [portalContainer, setPortalContainer] = React2.useState(null);
  const hoverTimeoutRef = React2.useRef(null);
  const submenuId = React2.useId();
  const submenuChildren = [];
  let hasSubmenu = false;
  React2.Children.forEach(children, (child) => {
    if (React2.isValidElement(child)) {
      const displayName = child.type?.displayName;
      if (displayName === "MenuItem" || displayName === "MenuSeparator" || displayName === "MenuHeader" || displayName === "MenuItemGroup") {
        hasSubmenu = true;
        submenuChildren.push(child);
      }
    }
  });
  if (itemLoading && !hasSubmenu) {
    hasSubmenu = true;
  }
  const checkMode = groupContext?.checkMode ?? "None" /* None */;
  const isInCheckGroup = checkMode !== "None" /* None */;
  let role = "menuitem";
  if (checkMode === "Single" /* Single */) {
    role = "menuitemradio";
  } else if (checkMode === "Multiple" /* Multiple */) {
    role = "menuitemcheckbox";
  }
  if (accessibilityAttributes?.role) {
    role = accessibilityAttributes.role;
  }
  const openSubmenu = React2.useCallback(() => {
    if (itemRef.current) {
      const popoverEl = itemRef.current.closest("[popover]");
      setPortalContainer(popoverEl ?? document.body);
    }
    setSubmenuOpen(true);
  }, []);
  React2.useEffect(() => {
    if (hasSubmenu) {
      registerSubmenu(submenuId, () => setSubmenuOpen(false));
      return () => unregisterSubmenu(submenuId);
    }
  }, [hasSubmenu, registerSubmenu, unregisterSubmenu]);
  const updateSubmenuPosition = React2.useCallback(() => {
    if (!itemRef.current || !submenuRef.current) return;
    const pos = calculateSubmenuPosition(itemRef.current, submenuRef.current);
    submenuRef.current.style.top = `${pos.top}px`;
    submenuRef.current.style.left = `${pos.left}px`;
  }, []);
  const setSubmenuRef = React2.useCallback((node) => {
    submenuRef.current = node;
    if (node) {
      updateSubmenuPosition();
    }
  }, [updateSubmenuPosition]);
  React2.useEffect(() => {
    if (!submenuOpen) return;
    window.addEventListener("scroll", updateSubmenuPosition, true);
    window.addEventListener("resize", updateSubmenuPosition);
    return () => {
      window.removeEventListener("scroll", updateSubmenuPosition, true);
      window.removeEventListener("resize", updateSubmenuPosition);
    };
  }, [submenuOpen, updateSubmenuPosition]);
  const prevSubmenuLoadingRef = React2.useRef({ submenuOpen, itemLoading });
  React2.useEffect(() => {
    const prev = prevSubmenuLoadingRef.current;
    prevSubmenuLoadingRef.current = { submenuOpen, itemLoading };
    if (prev.submenuOpen && submenuOpen && prev.itemLoading && !itemLoading && submenuRef.current) {
      requestAnimationFrame(() => {
        if (submenuRef.current) {
          const items = getFocusableItems(submenuRef.current);
          items[0]?.focus();
        }
      });
    }
  }, [itemLoading, submenuOpen]);
  const openedByKeyboardRef = React2.useRef(false);
  React2.useEffect(() => {
    if (submenuOpen && submenuRef.current && openedByKeyboardRef.current) {
      openedByKeyboardRef.current = false;
      requestAnimationFrame(() => {
        if (submenuRef.current) {
          const items = getFocusableItems(submenuRef.current);
          if (items.length > 0) {
            items[0].focus();
          }
        }
      });
    }
  }, [submenuOpen]);
  const activateItem = (e) => {
    onClick?.(e);
    const result = onItemClick?.({
      item: { text, icon, endContent, disabled, startsSection, hasSubmenu, data, checked },
      text: text || ""
    });
    if (result !== false) {
      closeMenu();
    }
  };
  const handleClick2 = (e) => {
    if (disabled) return;
    if (hasSubmenu) {
      e.stopPropagation();
      if (submenuOpen) {
        setSubmenuOpen(false);
      } else {
        openSubmenu();
      }
      return;
    }
    if (isInCheckGroup) {
      const detail = {
        item: { text, icon, endContent, disabled, startsSection, hasSubmenu, data, checked },
        text: text || "",
        checked: !checked
      };
      onCheck?.(detail);
      onClick?.(e);
      onItemClick?.(detail);
      if (!e.shiftKey) {
        closeMenu();
      }
      return;
    }
    activateItem(e);
  };
  const handleMouseEnter = () => {
    if (disabled) return;
    if (hoverTimeoutRef.current) {
      clearTimeout(hoverTimeoutRef.current);
      hoverTimeoutRef.current = null;
    }
    if (hasSubmenu) {
      closeAllSubmenus();
      openSubmenu();
    }
  };
  const handleMouseLeave = () => {
    if (hasSubmenu) {
      hoverTimeoutRef.current = setTimeout(() => {
        setSubmenuOpen(false);
      }, 150);
    }
  };
  const handleSubmenuMouseEnter = () => {
    if (hoverTimeoutRef.current) {
      clearTimeout(hoverTimeoutRef.current);
      hoverTimeoutRef.current = null;
    }
  };
  const handleSubmenuMouseLeave = () => {
    hoverTimeoutRef.current = setTimeout(() => {
      setSubmenuOpen(false);
    }, 150);
  };
  const handleKeyDown = (e) => {
    if (disabled) return;
    if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      if (isInCheckGroup) {
        const detail = {
          item: { text, icon, endContent, disabled, startsSection, hasSubmenu, data, checked },
          text: text || "",
          checked: !checked
        };
        onCheck?.(detail);
        onItemClick?.(detail);
        if (!e.shiftKey) {
          closeMenu();
        }
        return;
      }
      if (hasSubmenu) {
        e.stopPropagation();
        openedByKeyboardRef.current = true;
        openSubmenu();
        return;
      }
      activateItem(e);
    } else if (e.key === "ArrowRight" && hasSubmenu) {
      e.preventDefault();
      e.stopPropagation();
      openedByKeyboardRef.current = true;
      openSubmenu();
    }
  };
  const closeSubmenu = React2.useCallback(() => setSubmenuOpen(false), []);
  const submenuKeyDown = useMenuNavigation(submenuRef, {
    isSubmenu: true,
    onClose: closeSubmenu,
    parentItemRef: itemRef
  });
  React2.useEffect(() => {
    return () => {
      if (hoverTimeoutRef.current) {
        clearTimeout(hoverTimeoutRef.current);
      }
    };
  }, []);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    startsSection && /* @__PURE__ */ jsxRuntime.jsx(MenuSeparator, {}),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: (node) => {
          itemRef.current = node;
          if (typeof ref === "function") {
            ref(node);
          } else if (ref) {
            ref.current = node;
          }
        },
        role,
        tabIndex: disabled ? -1 : 0,
        "aria-disabled": disabled,
        "aria-haspopup": hasSubmenu ? "menu" : void 0,
        "aria-expanded": hasSubmenu ? submenuOpen : void 0,
        "aria-checked": isInCheckGroup ? !!checked : void 0,
        "data-menu-selected": checked && !isInCheckGroup ? "" : void 0,
        "aria-label": accessibleName,
        "aria-keyshortcuts": accessibilityAttributes?.ariaKeyShortcuts,
        title: tooltip,
        onClick: handleClick2,
        onMouseEnter: handleMouseEnter,
        onMouseLeave: handleMouseLeave,
        onKeyDown: handleKeyDown,
        className: cn(
          "group/item relative flex min-h-[54px] cursor-pointer select-none items-center gap-4 px-4 py-2 text-base font-semibold text-sapphire-text-secondary outline outline-2 outline-transparent outline-offset-[-4px]",
          "hover:bg-sapphire-neutral-hover-background-2 hover:text-foreground",
          "active:bg-sapphire-brand-selected-background active:text-foreground",
          "focus:outline-ring focus:z-10",
          checked && !isInCheckGroup && "bg-sapphire-brand-selected-background text-foreground border border-sapphire-brand-foreground focus:border-transparent",
          checked && !isInCheckGroup && "hover:bg-sapphire-brand-toggle-background",
          disabled && "pointer-events-none opacity-50",
          className
        ),
        children: [
          isInCheckGroup && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex size-5 shrink-0 items-center justify-center", children: checked && /* @__PURE__ */ jsxRuntime.jsx(AcceptIcon, { className: "size-5" }) }),
          icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("flex size-5 shrink-0 items-center justify-center [&>svg]:size-5 group-hover/item:text-sapphire-brand-foreground group-active/item:text-sapphire-brand-foreground", checked && !isInCheckGroup && "text-sapphire-brand-foreground"), children: icon }),
          /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-1 truncate", children: text }),
          hasSubmenu ? /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "ml-auto size-5 shrink-0 text-sapphire-text-tertiary" }) : endContent ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: "ml-auto", children: endContent }) : additionalText ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: "ml-4 shrink-0 text-sm text-sapphire-text-tertiary", children: additionalText }) : null
        ]
      }
    ),
    hasSubmenu && submenuOpen && reactDom.createPortal(
      /* @__PURE__ */ jsxRuntime.jsx(SubmenuScope, { closeMenu, onItemClick, onCheck, parentDepth: submenuDepth, children: /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          ref: setSubmenuRef,
          role: "menu",
          "data-submenu": "true",
          className: "min-w-[18.75rem] overflow-hidden rounded-lg border border-sapphire-border-primary bg-sapphire-card-bg-primary py-3.5 text-popover-foreground shadow-lg",
          style: {
            position: "fixed",
            // submenuDepth is the *parent's* depth from context.
            // +1 yields the z-index for this child submenu, so nested
            // submenus stack correctly: depth 0→z61, depth 1→z62, etc.
            zIndex: 60 + submenuDepth + 1
          },
          onMouseEnter: handleSubmenuMouseEnter,
          onMouseLeave: handleSubmenuMouseLeave,
          onKeyDown: submenuKeyDown,
          children: itemLoading ? /* @__PURE__ */ jsxRuntime.jsx(LoadingSpinner, { delay: loadingDelay }) : submenuChildren
        }
      ) }),
      portalContainer ?? document.body
    )
  ] });
}
MenuItem.displayName = "MenuItem";
function MenuItemGroup({ checkMode = "None" /* None */, children, accessibleName, className, ref }) {
  const mode = checkMode;
  let defaultLabel;
  if (mode === "Single" /* Single */) {
    defaultLabel = "Menu Item Group, Single Selection";
  } else if (mode === "Multiple" /* Multiple */) {
    defaultLabel = "Menu Item Group, Multi Selection";
  }
  const groupContextValue = React2.useMemo(() => ({ checkMode: mode }), [mode]);
  return /* @__PURE__ */ jsxRuntime.jsx(MenuItemGroupContext.Provider, { value: groupContextValue, children: /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      role: "group",
      "aria-label": accessibleName || defaultLabel,
      className,
      children
    }
  ) });
}
MenuItemGroup.displayName = "MenuItemGroup";
function MenuSeparator({ className, ref }) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      role: "separator",
      className: cn("my-2 h-px bg-border mx-4", className)
    }
  );
}
MenuSeparator.displayName = "MenuSeparator";
function MenuHeader({ text, children, endContent, className, ref }) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref,
      className: cn(
        "px-4 py-1.5 text-sm font-semibold text-sapphire-text-tertiary",
        endContent && "flex items-center gap-2",
        className
      ),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("span", { className: endContent ? "flex-1" : void 0, children: text || children }),
        endContent && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex shrink-0 items-center gap-1", children: endContent })
      ]
    }
  );
}
MenuHeader.displayName = "MenuHeader";

exports.Menu = Menu;
exports.MenuHeader = MenuHeader;
exports.MenuItem = MenuItem;
exports.MenuItemGroup = MenuItemGroup;
exports.MenuSeparator = MenuSeparator;
