// @sap-ui/fx-components/dist/components/segmented-button/index.cjs
'use strict';

var react = require('react');
var tailwindMerge = require('tailwind-merge');
var reactI18next = require('react-i18next');
var jsxRuntime = require('react/jsx-runtime');

// src/components/segmented-button/SegmentedButton.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var SegmentedButtonContext = react.createContext({
  selectionMode: "Single",
  size: "Large",
  onItemSelect: () => {
  }
});
function SegmentedButton({
  selectedId,
  selectedIds,
  selectionMode = "Single",
  size = "Large",
  children,
  onSelectionChange,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  ref
}) {
  const { t } = reactI18next.useTranslation("fx");
  const handleItemSelect = react.useCallback(
    (id, item) => {
      if (selectionMode === "Multiple") {
        const current = selectedIds ?? [];
        const newIds = current.includes(id) ? current.filter((i) => i !== id) : [...current, id];
        onSelectionChange?.({
          selectedItem: item,
          selectedId: id,
          selectedIds: newIds
        });
      } else {
        onSelectionChange?.({
          selectedItem: item,
          selectedId: id,
          selectedIds: [id]
        });
      }
    },
    [onSelectionChange, selectionMode, selectedIds]
  );
  const innerRef = react.useRef(null);
  const handleKeyDown = react.useCallback(
    (e) => {
      const container = innerRef.current;
      if (!container) return;
      if (e.key === "ArrowLeft" || e.key === "ArrowRight") {
        const items = Array.from(
          container.querySelectorAll(
            'li[role="option"]:not([aria-disabled="true"])'
          )
        );
        if (items.length === 0) return;
        const currentIndex = items.indexOf(
          document.activeElement?.closest?.("li")
        );
        if (currentIndex === -1) return;
        e.preventDefault();
        const next = e.key === "ArrowRight" ? (currentIndex + 1) % items.length : (currentIndex - 1 + items.length) % items.length;
        items[next].focus();
      } else if (e.key === "Enter" || e.key === " ") {
        const focused = document.activeElement?.closest?.(
          'li[role="option"]'
        );
        if (focused) {
          e.preventDefault();
          focused.click();
        }
      }
    },
    []
  );
  const isLarge = size === "Large";
  return /* @__PURE__ */ jsxRuntime.jsx(
    SegmentedButtonContext.Provider,
    {
      value: {
        selectedId,
        selectedIds,
        selectionMode,
        size,
        onItemSelect: handleItemSelect
      },
      children: /* @__PURE__ */ jsxRuntime.jsx(
        "ul",
        {
          ref: (node) => {
            innerRef.current = node;
            if (typeof ref === "function") ref(node);
            else if (ref) ref.current = node;
          },
          role: "listbox",
          "aria-label": accessibleName || t("SEGMENTEDBUTTON_ARIA_DESCRIPTION"),
          "aria-labelledby": accessibleNameRef,
          "aria-multiselectable": selectionMode === "Multiple" || void 0,
          onKeyDown: handleKeyDown,
          className: cn(
            "inline-flex list-none m-0 p-0 outline outline-1 -outline-offset-1 outline-sapphire-border-primary bg-background",
            isLarge ? "rounded-lg" : "rounded",
            className
          ),
          style,
          children
        }
      )
    }
  );
}
function SegmentedButtonItem({
  id,
  text,
  icon,
  selected: controlledSelected,
  disabled = false,
  tooltip,
  className,
  ref
}) {
  const { selectedId, selectedIds, selectionMode, size, onItemSelect } = react.useContext(SegmentedButtonContext);
  const { t } = reactI18next.useTranslation("fx");
  const isSelected = controlledSelected ?? (selectionMode === "Multiple" ? selectedIds?.includes(id) ?? false : selectedId === id);
  const itemRef = react.useRef(null);
  const handleClick = () => {
    if (disabled) return;
    itemRef.current?.focus();
    onItemSelect(id, { id, text, icon, selected: true, disabled, tooltip });
  };
  const isIconOnly = icon && !text;
  const isLarge = size === "Large";
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "li",
    {
      ref: (node) => {
        itemRef.current = node;
        if (typeof ref === "function") ref(node);
        else if (ref) ref.current = node;
      },
      role: "option",
      "aria-selected": isSelected,
      "aria-disabled": disabled || void 0,
      "aria-roledescription": t("SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION"),
      tabIndex: isSelected ? 0 : -1,
      title: tooltip,
      onClick: handleClick,
      className: cn(
        "inline-flex items-center justify-center gap-sapphire-2xs transition-[color,background-color,border-color] cursor-pointer relative",
        isLarge ? "rounded-lg h-10 min-w-[2.5rem] px-4 text-base gap-2 font-semibold" : size === "Medium" ? "rounded h-8 min-w-[2.25rem] px-3 text-sm gap-1.5 font-semibold" : "rounded h-6 min-w-[1.75rem] px-2 text-xs gap-1 font-normal",
        "focus:z-20",
        "border",
        isSelected ? "z-10 border-sapphire-brand-foreground text-sapphire-brand-foreground focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring" : "border-transparent text-sapphire-neutral-foreground-black hover:bg-sapphire-neutral-hover-background focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring",
        disabled && "pointer-events-none text-sapphire-button-fg-disabled",
        isIconOnly && (isLarge ? "px-2" : size === "Medium" ? "px-2" : "px-1.5"),
        className
      ),
      children: [
        icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center", isLarge ? "[&>svg]:h-5 [&>svg]:w-5" : size === "Medium" ? "[&>svg]:h-4 [&>svg]:w-4" : "[&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"), children: icon }),
        text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: text })
      ]
    }
  );
}

exports.SegmentedButton = SegmentedButton;
exports.SegmentedButtonItem = SegmentedButtonItem;
