// @sap-ui/fx-components/dist/components/switch/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/switch/Switch.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}

// src/types/switch.ts
var SwitchDesign = /* @__PURE__ */ ((SwitchDesign2) => {
  SwitchDesign2["Textual"] = "Textual";
  SwitchDesign2["Graphical"] = "Graphical";
  return SwitchDesign2;
})(SwitchDesign || {});
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData3,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData3 || pathData3.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData3.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M187 416q-12 0-20-9L71 299q-7-7-7-17 0-11 7.5-18.5T90 256q12 0 19 9l77 87 217-247q8-9 19-9t18.5 7.5T448 122q0 10-6 16L206 407q-7 9-19 9z"
];
var AcceptIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "AcceptIcon", iconName: "accept" }
);
var pathData2 = [
  "M454 275H58q-11 0-18.5-7T32 250t7.5-18.5T58 224h396q11 0 18.5 7.5T480 250t-7.5 18-18.5 7z"
];
var LessIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "LessIcon", iconName: "less" }
);
var switchContainerBase = "relative inline-flex items-center rounded-full transition-colors duration-200 ease-in-out cursor-pointer focus:outline-none box-border";
var switchHandleBase = "absolute rounded-full transition-transform duration-200 ease-in-out flex items-center justify-center box-border";
function Switch({
  checked: controlledChecked,
  defaultChecked = false,
  disabled = false,
  required = false,
  design = "Textual" /* Textual */,
  textOn,
  textOff,
  iconOn,
  iconOff,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  tooltip,
  name,
  value = "on",
  id,
  onChange,
  className,
  style,
  "data-testid": dataTestId,
  accessibilityAttributes,
  ref
}) {
  const [internalChecked, setInternalChecked] = React2.useState(defaultChecked);
  const textOnRef = React2.useRef(null);
  const textOffRef = React2.useRef(null);
  const [textWidths, setTextWidths] = React2.useState({ on: 0, off: 0 });
  const generatedId = React2.useId();
  const isControlled = controlledChecked !== void 0;
  const checked = isControlled ? controlledChecked : internalChecked;
  const descriptionId = `${id ?? generatedId}-desc`;
  const ariaDescribedBy = [accessibleDescriptionRef, accessibleDescription ? descriptionId : null, accessibilityAttributes?.describedBy].filter(Boolean).join(" ") || void 0;
  const isGraphical = design === "Graphical" /* Graphical */;
  const hasText = !isGraphical && (textOn || textOff);
  const measureTextWidths = React2.useCallback(() => {
    const onWidth = textOnRef.current?.offsetWidth || 0;
    const offWidth = textOffRef.current?.offsetWidth || 0;
    if (onWidth !== textWidths.on || offWidth !== textWidths.off) {
      setTextWidths({ on: onWidth, off: offWidth });
    }
  }, [textWidths.on, textWidths.off]);
  const textOnCallbackRef = React2.useCallback((node) => {
    textOnRef.current = node;
    if (node) measureTextWidths();
  }, [measureTextWidths]);
  const textOffCallbackRef = React2.useCallback((node) => {
    textOffRef.current = node;
    if (node) measureTextWidths();
  }, [measureTextWidths]);
  const handlePadding = 12;
  const borderWidth = 2;
  const handleWidthOn = hasText ? textWidths.on + handlePadding : 14;
  const handleWidthOff = hasText ? textWidths.off + handlePadding : 14;
  const maxHandleWidth = Math.max(handleWidthOn, handleWidthOff);
  const containerWidth = hasText ? Math.ceil(maxHandleWidth * 1.6) : 32;
  const currentHandleWidth = checked ? handleWidthOn : handleWidthOff;
  const handleTranslateX = hasText ? checked ? containerWidth - currentHandleWidth - borderWidth * 2 : 0 : checked ? 10 : 0;
  const handleToggle = React2.useCallback(() => {
    if (disabled) return;
    const newChecked = !checked;
    if (!isControlled) {
      setInternalChecked(newChecked);
    }
    onChange?.({ checked: newChecked });
  }, [checked, disabled, isControlled, onChange]);
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        handleToggle();
      }
    },
    [handleToggle]
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref,
        id,
        type: "button",
        role: "switch",
        "aria-checked": checked,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-describedby": ariaDescribedBy,
        "aria-required": required,
        "aria-disabled": disabled,
        title: tooltip,
        disabled,
        onClick: handleToggle,
        onKeyDown: handleKeyDown,
        "data-testid": dataTestId,
        className: cn(
          switchContainerBase,
          "fx-switch",
          className
        ),
        style: {
          ...style,
          backgroundColor: "var(--_switch-bg)",
          width: hasText ? `${containerWidth}px` : "32px",
          height: "20px"
        },
        children: [
          hasText && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
            /* @__PURE__ */ jsxRuntime.jsx(
              "span",
              {
                ref: textOnCallbackRef,
                className: "absolute invisible text-xs font-medium px-1.5",
                "aria-hidden": "true",
                children: textOn
              }
            ),
            /* @__PURE__ */ jsxRuntime.jsx(
              "span",
              {
                ref: textOffCallbackRef,
                className: "absolute invisible text-xs font-medium px-1.5",
                "aria-hidden": "true",
                children: textOff
              }
            )
          ] }),
          /* @__PURE__ */ jsxRuntime.jsx(
            "span",
            {
              className: switchHandleBase,
              style: {
                transform: `translateX(${handleTranslateX}px)`,
                width: "14px",
                height: "14px",
                top: "3px",
                left: "4px",
                backgroundColor: "var(--_switch-handle-bg)",
                boxShadow: "0 0 0 1px var(--_switch-handle-border)"
              },
              "aria-hidden": "true",
              children: isGraphical ? /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children: checked ? iconOn || /* @__PURE__ */ jsxRuntime.jsx(AcceptIcon, { className: "h-3 w-3", style: { color: "var(--_switch-bg)" } }) : iconOff || /* @__PURE__ */ jsxRuntime.jsx(LessIcon, { className: "h-3 w-3" }) }) : hasText ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: "whitespace-nowrap", children: checked ? textOn : textOff }) : null
            }
          )
        ]
      }
    ),
    name && /* @__PURE__ */ jsxRuntime.jsx(
      "input",
      {
        type: "checkbox",
        name,
        value,
        checked,
        onChange: () => {
        },
        required,
        disabled,
        className: "sr-only",
        tabIndex: -1,
        "aria-hidden": "true"
      }
    ),
    accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription })
  ] });
}

exports.Switch = Switch;
exports.SwitchDesign = SwitchDesign;
