// @sap-ui/fx-components/dist/components/calendar/index.cjs
'use strict';

var React6 = require('react');
var tailwindMerge = require('tailwind-merge');
var getCachedLocaleDataInstance4 = require('@ui5/webcomponents-localization/dist/getCachedLocaleDataInstance.js');
var Locale2 = require('@ui5/webcomponents-base/dist/locale/Locale.js');
var jsxRuntime = require('react/jsx-runtime');
require('@ui5/webcomponents-localization/dist/Assets.js');
require('@ui5/webcomponents-localization/dist/sap/ui/core/date/Gregorian.js');
var LocaleData_js = require('@ui5/webcomponents-base/dist/asset-registries/LocaleData.js');
var reactI18next = require('react-i18next');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React6__default = /*#__PURE__*/_interopDefault(React6);
var getCachedLocaleDataInstance4__default = /*#__PURE__*/_interopDefault(getCachedLocaleDataInstance4);
var Locale2__default = /*#__PURE__*/_interopDefault(Locale2);

// src/components/calendar/Calendar.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}

// src/types/calendar.ts
var CalendarSelectionMode = /* @__PURE__ */ ((CalendarSelectionMode2) => {
  CalendarSelectionMode2["Single"] = "Single";
  CalendarSelectionMode2["Multiple"] = "Multiple";
  CalendarSelectionMode2["Range"] = "Range";
  return CalendarSelectionMode2;
})(CalendarSelectionMode || {});
var CalendarWeekNumbering = /* @__PURE__ */ ((CalendarWeekNumbering2) => {
  CalendarWeekNumbering2["Default"] = "Default";
  CalendarWeekNumbering2["ISO_8601"] = "ISO_8601";
  CalendarWeekNumbering2["MiddleEastern"] = "MiddleEastern";
  CalendarWeekNumbering2["WesternTraditional"] = "WesternTraditional";
  return CalendarWeekNumbering2;
})(CalendarWeekNumbering || {});
var CalendarPickersMode = /* @__PURE__ */ ((CalendarPickersMode2) => {
  CalendarPickersMode2["DAY_MONTH_YEAR"] = "DAY_MONTH_YEAR";
  CalendarPickersMode2["MONTH_YEAR"] = "MONTH_YEAR";
  CalendarPickersMode2["YEAR"] = "YEAR";
  return CalendarPickersMode2;
})(CalendarPickersMode || {});
var CalendarLegendItemType = /* @__PURE__ */ ((CalendarLegendItemType2) => {
  CalendarLegendItemType2["None"] = "None";
  CalendarLegendItemType2["Today"] = "Today";
  CalendarLegendItemType2["Selected"] = "Selected";
  CalendarLegendItemType2["Working"] = "Working";
  CalendarLegendItemType2["NonWorking"] = "NonWorking";
  CalendarLegendItemType2["Type01"] = "Type01";
  CalendarLegendItemType2["Type02"] = "Type02";
  CalendarLegendItemType2["Type03"] = "Type03";
  CalendarLegendItemType2["Type04"] = "Type04";
  CalendarLegendItemType2["Type05"] = "Type05";
  CalendarLegendItemType2["Type06"] = "Type06";
  CalendarLegendItemType2["Type07"] = "Type07";
  CalendarLegendItemType2["Type08"] = "Type08";
  CalendarLegendItemType2["Type09"] = "Type09";
  CalendarLegendItemType2["Type10"] = "Type10";
  CalendarLegendItemType2["Type11"] = "Type11";
  CalendarLegendItemType2["Type12"] = "Type12";
  CalendarLegendItemType2["Type13"] = "Type13";
  CalendarLegendItemType2["Type14"] = "Type14";
  CalendarLegendItemType2["Type15"] = "Type15";
  CalendarLegendItemType2["Type16"] = "Type16";
  CalendarLegendItemType2["Type17"] = "Type17";
  CalendarLegendItemType2["Type18"] = "Type18";
  CalendarLegendItemType2["Type19"] = "Type19";
  CalendarLegendItemType2["Type20"] = "Type20";
  return CalendarLegendItemType2;
})(CalendarLegendItemType || {});
var CalendarPickerView = /* @__PURE__ */ ((CalendarPickerView2) => {
  CalendarPickerView2["Day"] = "day";
  CalendarPickerView2["Month"] = "month";
  CalendarPickerView2["Year"] = "year";
  return CalendarPickerView2;
})(CalendarPickerView || {});
function getLocale(localeCode) {
  return new Locale2__default.default(localeCode || "en-US");
}
function getWeekStartDay(localeCode) {
  const localeData = getCachedLocaleDataInstance4__default.default(getLocale(localeCode));
  return localeData.getFirstDayOfWeek();
}

// src/lib/date-utils.ts
function format(date, token) {
  switch (token) {
    case "d":
      return String(date.getDate());
    case "dd":
      return String(date.getDate()).padStart(2, "0");
    case "M":
      return String(date.getMonth() + 1);
    case "MM":
      return String(date.getMonth() + 1).padStart(2, "0");
    case "yy":
      return String(date.getFullYear()).slice(-2);
    case "yyyy":
      return String(date.getFullYear());
    default:
      return String(date.getDate());
  }
}
function parseDate(value) {
  if (value instanceof Date) return value;
  if (typeof value === "number") return new Date(value * 1e3);
  const isoMatch = value.match(/^(\d{4})-(\d{2})-(\d{2})$/);
  if (isoMatch) {
    return new Date(+isoMatch[1], +isoMatch[2] - 1, +isoMatch[3]);
  }
  const d = new Date(value);
  if (!isNaN(d.getTime())) return d;
  throw new Error(`Invalid date value: ${value}`);
}
function formatDateToISO(date) {
  const y = date.getFullYear();
  const m = String(date.getMonth() + 1).padStart(2, "0");
  const d = String(date.getDate()).padStart(2, "0");
  return `${y}-${m}-${d}`;
}
function getWeekStart(date, weekStartsOn = 1) {
  const d = new Date(date);
  const day = d.getDay();
  const diff = (day - weekStartsOn + 7) % 7;
  d.setDate(d.getDate() - diff);
  d.setHours(0, 0, 0, 0);
  return d;
}
function getWeekEnd(date, weekStartsOn = 1) {
  const start = getWeekStart(date, weekStartsOn);
  const d = new Date(start);
  d.setDate(d.getDate() + 6);
  d.setHours(23, 59, 59, 999);
  return d;
}
function getMonthDays(date, weekStartsOn = 1) {
  const monthStart = new Date(date.getFullYear(), date.getMonth(), 1);
  const monthEnd = new Date(date.getFullYear(), date.getMonth() + 1, 0);
  const gridStart = getWeekStart(monthStart, weekStartsOn);
  const gridEnd = getWeekEnd(monthEnd, weekStartsOn);
  const days = [];
  const cur = new Date(gridStart);
  cur.setHours(0, 0, 0, 0);
  while (cur <= gridEnd) {
    days.push(new Date(cur));
    cur.setDate(cur.getDate() + 1);
  }
  return days;
}
function isoWeekNumber(date) {
  const d = new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()));
  d.setUTCDate(d.getUTCDate() + 4 - (d.getUTCDay() || 7));
  const yearStart = new Date(Date.UTC(d.getUTCFullYear(), 0, 1));
  return Math.ceil(((d.getTime() - yearStart.getTime()) / 864e5 + 1) / 7);
}
function westernWeekNumber(date) {
  const yearStart = new Date(date.getFullYear(), 0, 1);
  const startDow = yearStart.getDay();
  const dayOfYear = Math.floor((date.getTime() - yearStart.getTime()) / 864e5);
  return Math.floor((dayOfYear + startDow) / 7) + 1;
}
function middleEasternWeekNumber(date) {
  const yearStart = new Date(date.getFullYear(), 0, 1);
  const startDow = (yearStart.getDay() + 1) % 7;
  const dayOfYear = Math.floor((date.getTime() - yearStart.getTime()) / 864e5);
  return Math.floor((dayOfYear + startDow) / 7) + 1;
}
function getWeekNumber(date, numbering = "ISO_8601" /* ISO_8601 */) {
  switch (numbering) {
    case "WesternTraditional" /* WesternTraditional */:
      return westernWeekNumber(date);
    case "MiddleEastern" /* MiddleEastern */:
      return middleEasternWeekNumber(date);
    case "ISO_8601" /* ISO_8601 */:
    case "Default" /* Default */:
    default:
      return isoWeekNumber(date);
  }
}
function isSameDayUtil(date1, date2) {
  return date1.getFullYear() === date2.getFullYear() && date1.getMonth() === date2.getMonth() && date1.getDate() === date2.getDate();
}
function isSameMonthUtil(date, compareDate) {
  return date.getFullYear() === compareDate.getFullYear() && date.getMonth() === compareDate.getMonth();
}
function isDateInRange(date, start, end) {
  if (!start || !end) return false;
  const t = date.getTime();
  return t >= start.getTime() && t <= end.getTime();
}
function isDateBefore(date, compareDate) {
  return date.getTime() < compareDate.getTime();
}
function isDateAfter(date, compareDate) {
  return date.getTime() > compareDate.getTime();
}
function addMonthsToDate(date, amount) {
  const d = new Date(date);
  d.setMonth(d.getMonth() + amount);
  return d;
}
function getToday() {
  const today = /* @__PURE__ */ new Date();
  today.setHours(0, 0, 0, 0);
  return today;
}
function isToday(date) {
  return isSameDayUtil(date, getToday());
}
function getMonthName(date, formatStr = "long", localeCode) {
  const localeData = getCachedLocaleDataInstance4__default.default(getLocale(localeCode));
  const width = formatStr === "long" ? "wide" : "abbreviated";
  return localeData.getMonthsStandAlone(width)[date.getMonth()];
}
function getYear(date) {
  return date.getFullYear();
}
function getMonth(date) {
  return date.getMonth();
}
function createDate(year, month, day = 1) {
  return new Date(year, month, day);
}
function useCalendarSelection(mode, initialDates, rangeStart, rangeEnd, minDate, maxDate, disabledDates) {
  const [selectedDates, setSelectedDates] = React6.useState(() => {
    if (mode === "Range") return [];
    return initialDates?.map((d) => parseDate(d)) || [];
  });
  const [rangeStartDate, setRangeStartDate] = React6.useState(
    () => rangeStart ? parseDate(rangeStart) : null
  );
  const [rangeEndDate, setRangeEndDate] = React6.useState(
    () => rangeEnd ? parseDate(rangeEnd) : null
  );
  const [hoveredDate, setHoveredDate] = React6.useState(null);
  const minDateObj = minDate ? parseDate(minDate) : null;
  const maxDateObj = maxDate ? parseDate(maxDate) : null;
  const isDateDisabled = React6.useCallback(
    (date) => {
      if (minDateObj && isDateBefore(date, minDateObj)) return true;
      if (maxDateObj && isDateAfter(date, maxDateObj)) return true;
      if (disabledDates) {
        for (const range of disabledDates) {
          const start = range.startDate ? parseDate(range.startDate) : null;
          const end = range.endDate ? parseDate(range.endDate) : null;
          if (start && end) {
            if (isDateInRange(date, start, end)) return true;
          } else if (start) {
            if (isSameDayUtil(date, start) || isDateAfter(date, start)) return true;
          } else if (end) {
            if (isSameDayUtil(date, end) || isDateBefore(date, end)) return true;
          }
        }
      }
      return false;
    },
    [minDateObj, maxDateObj, disabledDates]
  );
  const isDateSelected = React6.useCallback(
    (date) => {
      if (mode === "Range") {
        if (!rangeStartDate) return false;
        if (!rangeEndDate) return isSameDayUtil(date, rangeStartDate);
        return isDateInRange(date, rangeStartDate, rangeEndDate);
      }
      return selectedDates.some((d) => isSameDayUtil(d, date));
    },
    [mode, selectedDates, rangeStartDate, rangeEndDate]
  );
  const isRangeStart = React6.useCallback(
    (date) => {
      if (mode !== "Range" || !rangeStartDate) return false;
      if (!rangeEndDate && hoveredDate) {
        if (isDateBefore(hoveredDate, rangeStartDate)) {
          return isSameDayUtil(date, hoveredDate);
        }
      }
      return isSameDayUtil(date, rangeStartDate);
    },
    [mode, rangeStartDate, rangeEndDate, hoveredDate]
  );
  const isRangeEnd = React6.useCallback(
    (date) => {
      if (mode !== "Range") return false;
      if (rangeEndDate) return isSameDayUtil(date, rangeEndDate);
      if (rangeStartDate && hoveredDate) {
        if (isDateBefore(hoveredDate, rangeStartDate)) {
          return isSameDayUtil(date, rangeStartDate);
        }
        return isSameDayUtil(date, hoveredDate);
      }
      return false;
    },
    [mode, rangeStartDate, rangeEndDate, hoveredDate]
  );
  const isDateBetweenRange = React6.useCallback(
    (date) => {
      if (mode !== "Range" || !rangeStartDate) return false;
      if (rangeEndDate) {
        return isDateInRange(date, rangeStartDate, rangeEndDate) && !isSameDayUtil(date, rangeStartDate) && !isSameDayUtil(date, rangeEndDate);
      }
      if (hoveredDate) {
        const previewStart = isDateBefore(hoveredDate, rangeStartDate) ? hoveredDate : rangeStartDate;
        const previewEnd = isDateBefore(hoveredDate, rangeStartDate) ? rangeStartDate : hoveredDate;
        return isDateInRange(date, previewStart, previewEnd) && !isSameDayUtil(date, previewStart) && !isSameDayUtil(date, previewEnd);
      }
      return false;
    },
    [mode, rangeStartDate, rangeEndDate, hoveredDate]
  );
  const selectDate = React6.useCallback(
    (date) => {
      if (isDateDisabled(date)) return null;
      let newSelectedValues = [];
      let newSelectedTimestamps = [];
      if (mode === "Single") {
        setSelectedDates([date]);
        newSelectedValues = [formatDateToISO(date)];
        newSelectedTimestamps = [Math.floor(date.getTime() / 1e3)];
      } else if (mode === "Multiple") {
        let newDates = [];
        setSelectedDates((prev) => {
          const exists2 = prev.some((d) => isSameDayUtil(d, date));
          if (exists2) {
            newDates = prev.filter((d) => !isSameDayUtil(d, date));
          } else {
            newDates = [...prev, date];
          }
          return newDates;
        });
        const exists = selectedDates.some((d) => isSameDayUtil(d, date));
        if (exists) {
          newDates = selectedDates.filter((d) => !isSameDayUtil(d, date));
        } else {
          newDates = [...selectedDates, date];
        }
        newSelectedValues = newDates.map(formatDateToISO);
        newSelectedTimestamps = newDates.map((d) => Math.floor(d.getTime() / 1e3));
      } else if (mode === "Range") {
        if (!rangeStartDate || rangeStartDate && rangeEndDate) {
          setRangeStartDate(date);
          setRangeEndDate(null);
          setHoveredDate(null);
          newSelectedValues = [formatDateToISO(date)];
          newSelectedTimestamps = [Math.floor(date.getTime() / 1e3)];
        } else {
          let newStart;
          let newEnd;
          if (isDateBefore(date, rangeStartDate)) {
            newStart = date;
            newEnd = rangeStartDate;
            setRangeStartDate(date);
            setRangeEndDate(rangeStartDate);
          } else {
            newStart = rangeStartDate;
            newEnd = date;
            setRangeEndDate(date);
          }
          setHoveredDate(null);
          newSelectedValues = [formatDateToISO(newStart), formatDateToISO(newEnd)];
          newSelectedTimestamps = [
            Math.floor(newStart.getTime() / 1e3),
            Math.floor(newEnd.getTime() / 1e3)
          ];
        }
      }
      return { selectedValues: newSelectedValues, selectedTimestamps: newSelectedTimestamps };
    },
    [mode, isDateDisabled, rangeStartDate, rangeEndDate, selectedDates]
  );
  const getSelectedValuesAsISO = React6.useCallback(() => {
    if (mode === "Range") {
      const values = [];
      if (rangeStartDate) values.push(formatDateToISO(rangeStartDate));
      if (rangeEndDate) values.push(formatDateToISO(rangeEndDate));
      return values;
    }
    return selectedDates.map(formatDateToISO);
  }, [mode, selectedDates, rangeStartDate, rangeEndDate]);
  const getSelectedTimestamps = React6.useCallback(() => {
    if (mode === "Range") {
      const timestamps = [];
      if (rangeStartDate) timestamps.push(Math.floor(rangeStartDate.getTime() / 1e3));
      if (rangeEndDate) timestamps.push(Math.floor(rangeEndDate.getTime() / 1e3));
      return timestamps;
    }
    return selectedDates.map((d) => Math.floor(d.getTime() / 1e3));
  }, [mode, selectedDates, rangeStartDate, rangeEndDate]);
  const isRangeSelectionInProgress = mode === "Range" && rangeStartDate !== null && rangeEndDate === null;
  return {
    selectedDates,
    rangeStartDate,
    rangeEndDate,
    hoveredDate,
    setHoveredDate,
    isRangeSelectionInProgress,
    isDateDisabled,
    isDateSelected,
    isRangeStart,
    isRangeEnd,
    isDateBetweenRange,
    selectDate,
    getSelectedValuesAsISO,
    getSelectedTimestamps
  };
}
function useSpecialDates(specialDates) {
  const getSpecialDate = React6.useCallback(
    (date) => {
      if (!specialDates) return null;
      const found = specialDates.find((sd) => {
        const specialDate = parseDate(sd.date);
        return isSameDayUtil(specialDate, date);
      });
      return found ? { type: found.type, tooltip: found.tooltip } : null;
    },
    [specialDates]
  );
  return { getSpecialDate };
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData3,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React6.useRef(null);
  React6.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React6.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React6.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData3 || pathData3.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData3.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M326 96q11 0 18.5 7.5T352 122q0 10-8 18L223 256l121 116q8 8 8 18 0 11-7.5 18.5T326 416q-10 0-17-7L168 274q-8-6-8-18 0-11 8-19l141-134q7-7 17-7z"
];
var SlimArrowLeftIcon = Object.assign(
  React6__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SlimArrowLeftIcon", iconName: "slim-arrow-left" }
);
var pathData2 = [
  "M186 416q-11 0-18.5-7.5T160 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T186 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L203 409q-7 7-17 7z"
];
var SlimArrowRightIcon = Object.assign(
  React6__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "SlimArrowRightIcon", iconName: "slim-arrow-right" }
);
var dotVariants = cva("rounded-full bg-current inline-block", {
  variants: {
    size: {
      ["S" /* S */]: "w-2 h-2",
      ["M" /* M */]: "w-4 h-4",
      ["L" /* L */]: "w-8 h-8"
    }
  },
  defaultVariants: { size: "M" /* M */ }
});
var dotGapClasses = {
  ["S" /* S */]: "gap-px",
  ["M" /* M */]: "gap-[0.1875rem]",
  ["L" /* L */]: "gap-1"
};
function BusyIndicator({
  size = "M" /* M */,
  textPlacement = "Bottom" /* Bottom */,
  active = false,
  delay = 1e3,
  text,
  children,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = React6.useRef(null);
  const overlayRef = React6.useRef(null);
  const [isBusy, setIsBusy] = React6.useState(false);
  const labelId = React6.useId();
  const hasChildren = React6__default.default.Children.count(children) > 0;
  const normalizedSize = size || "M" /* M */;
  const isTopPlacement = textPlacement === "Top";
  React6.useEffect(() => {
    if (!active) {
      setIsBusy(false);
      return;
    }
    if (delay <= 0) {
      setIsBusy(true);
      return;
    }
    const timer = setTimeout(() => {
      setIsBusy(true);
    }, delay);
    return () => clearTimeout(timer);
  }, [active, delay]);
  React6.useImperativeHandle(
    ref,
    () => ({
      focus() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.focus();
        } else {
          rootRef.current?.focus();
        }
      },
      blur() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.blur();
        } else {
          rootRef.current?.blur();
        }
      },
      isFocused() {
        return document.activeElement === overlayRef.current || document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    [isBusy]
  );
  const textLabel = text ? /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      id: labelId,
      className: cn(
        "w-full text-center text-sapphire-text-tertiary text-sm font-medium",
        isTopPlacement ? "mb-2" : "mt-2"
      ),
      children: text
    }
  ) : null;
  const dots = /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "inline-flex items-center leading-none",
        dotGapClasses[normalizedSize]
      ),
      children: [0, 1, 2].map((i) => /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: dotVariants({ size: normalizedSize }),
          style: {
            animation: `busy-grow 1.6s cubic-bezier(0.32, 0.06, 0.85, 1.11) ${i * 200}ms infinite`
          }
        },
        i
      ))
    }
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      id,
      className: cn(
        "inline-block",
        hasChildren && "relative",
        isBusy && "text-ring",
        className
      ),
      style: { ...style, borderRadius: "inherit", height: "inherit" },
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("style", { children: `@keyframes busy-grow{0%,50%,100%{transform:scale(.5)}25%{transform:scale(1)}}` }),
        isBusy && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: overlayRef,
            role: "progressbar",
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            "aria-valuetext": "Busy",
            "aria-labelledby": text ? labelId : void 0,
            "aria-label": !text ? accessibleName ?? "Please wait" : void 0,
            title: !text ? accessibleName ?? "Please wait" : void 0,
            tabIndex: 0,
            "data-sap-focus-ref": true,
            className: cn(
              "flex flex-col items-center justify-center",
              "bg-inherit rounded-[inherit]",
              "focus-visible:outline focus-visible:outline-2 focus-visible:outline-ring focus-visible:outline-offset-[-2px]",
              hasChildren && "absolute inset-0 z-[99]"
            ),
            children: [
              isTopPlacement && textLabel,
              dots,
              !isTopPlacement && textLabel
            ]
          }
        ),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(isBusy && "opacity-40"),
            ...isBusy ? { inert: true } : {},
            children
          }
        )
      ]
    }
  );
}
var filledBrandStyles = [
  "bg-sapphire-brand-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-brand-hover-background",
  "active:bg-sapphire-brand-pressed-background [&.kb-active]:bg-sapphire-brand-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus"
];
var filledJouleStyles = [
  "bg-sapphire-joule-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-joule-hover-background",
  "active:bg-sapphire-joule-pressed-background [&.kb-active]:bg-sapphire-joule-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus"
];
var outlineBrandStyles = [
  "bg-transparent text-sapphire-brand-foreground border border-sapphire-brand-foreground",
  "hover:bg-sapphire-neutral-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-brand-pressed-foreground [&.kb-active]:text-sapphire-brand-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
];
var outlineJouleStyles = [
  "bg-transparent text-sapphire-joule-foreground border border-sapphire-joule-foreground",
  "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background hover:border-sapphire-joule-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-joule-pressed-foreground [&.kb-active]:text-sapphire-joule-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
];
var buttonVariants = cva(
  [
    // Base styles
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        // --- Primary (filled) ---
        ["Primary" /* Primary */]: filledBrandStyles,
        ["PrimaryJoule" /* PrimaryJoule */]: filledJouleStyles,
        // --- Secondary (outlined) ---
        ["Secondary" /* Secondary */]: outlineBrandStyles,
        ["SecondaryJoule" /* SecondaryJoule */]: outlineJouleStyles,
        // --- Tertiary (ghost, colored text) ---
        ["Tertiary" /* Tertiary */]: [
          "bg-transparent text-sapphire-brand-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-brand-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
        ],
        ["TertiaryJoule" /* TertiaryJoule */]: [
          "bg-transparent text-sapphire-joule-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-joule-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
        ],
        // --- Neutral (ghost, dark text) ---
        ["Neutral" /* Neutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-sapphire-neutral-foreground-muted",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-neutral-foreground-black [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- SecondaryNeutral (ghost, neutral-black text) ---
        ["SecondaryNeutral" /* SecondaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background [&.kb-active]:bg-sapphire-neutral-pressed-background",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- TertiaryNeutral (ghost, muted neutral text, no background) ---
        ["TertiaryNeutral" /* TertiaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-muted border border-transparent",
          "hover:text-sapphire-neutral-foreground-black",
          "active:text-sapphire-neutral-foreground-black [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ]
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var largeSizeClasses = {
  default: "h-10 px-4 py-3 text-base gap-2 rounded-lg",
  iconOnly: "h-10 w-10 p-0 text-base rounded-lg",
  icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
};
var mediumSizeClasses = {
  default: "h-8 px-3 py-2 text-sm gap-1.5 rounded",
  iconOnly: "h-8 w-8 p-0 text-sm rounded",
  icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
};
var smallSizeClasses = {
  default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
  iconOnly: "h-6 w-6 p-0 text-xs rounded",
  icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
};
var sizeClasses = {
  ["Large" /* Large */]: largeSizeClasses,
  ["Medium" /* Medium */]: mediumSizeClasses,
  ["Small" /* Small */]: smallSizeClasses
};
function getButtonType(type) {
  switch (type) {
    case "Submit" /* Submit */:
    case "Submit":
      return "submit";
    case "Reset" /* Reset */:
    case "Reset":
      return "reset";
    default:
      return "button";
  }
}
function Button({
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  type = "Button" /* Button */,
  disabled = false,
  loading = false,
  icon,
  endIcon,
  iconOnly = false,
  children,
  tooltip,
  states,
  state,
  stateTransition = "slide-up",
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Button" /* Button */,
  accessibilityAttributes,
  name,
  value,
  form,
  onClick,
  onFocus,
  onBlur,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  "data-growing-button": dataGrowingButton,
  ref
}) {
  const buttonRef = React6.useRef(null);
  const hiddenButtonRef = React6.useRef(null);
  const [animationPhase, setAnimationPhase] = React6.useState("idle");
  const [displayedState, setDisplayedState] = React6.useState(
    null
  );
  const [buttonWidth, setButtonWidth] = React6.useState(void 0);
  const currentStateObject = states?.find((s) => s.name === state) ?? states?.[0];
  const isMultiState = states && states.length > 0;
  React6.useEffect(() => {
    if (isMultiState && !displayedState) {
      setDisplayedState(currentStateObject ?? null);
    }
  }, [isMultiState, currentStateObject, displayedState]);
  React6.useEffect(() => {
    if (!isMultiState || !currentStateObject || animationPhase !== "idle") {
      return;
    }
    if (displayedState && displayedState.name !== currentStateObject.name) {
      if (buttonRef.current) {
        setButtonWidth(buttonRef.current.offsetWidth);
      }
      setAnimationPhase("fade-out");
    }
  }, [currentStateObject, displayedState, isMultiState, animationPhase]);
  React6.useEffect(() => {
    if (animationPhase === "fade-out") {
      const timer = setTimeout(() => {
        setAnimationPhase("fade-mid");
      }, 180);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-mid") {
      setDisplayedState(currentStateObject ?? null);
      if (hiddenButtonRef.current) {
        setButtonWidth(hiddenButtonRef.current.offsetWidth);
      }
      const timer = setTimeout(() => {
        setAnimationPhase("fade-in");
      }, 20);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-in") {
      const timer = setTimeout(() => {
        setAnimationPhase("idle");
        setButtonWidth(void 0);
      }, 160);
      return () => clearTimeout(timer);
    }
  }, [animationPhase, currentStateObject]);
  React6.useImperativeHandle(
    ref,
    () => ({
      focus() {
        buttonRef.current?.focus();
      },
      blur() {
        buttonRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === buttonRef.current;
      },
      get nativeElement() {
        return buttonRef.current;
      }
    }),
    []
  );
  const handleClick = React6.useCallback(
    (e) => {
      if (disabled || loading) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [disabled, loading, onClick]
  );
  const handleKeyDown = React6.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " ") {
          e.preventDefault();
        }
        buttonRef.current?.classList.add("kb-active");
        if (e.key === " ") {
          onClick?.({
            originalEvent: e,
            isKeyboard: true
          });
        }
      }
    },
    [disabled, loading, onClick]
  );
  const handleKeyUp = React6.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        buttonRef.current?.classList.remove("kb-active");
      }
    },
    []
  );
  const ariaExpanded = accessibilityAttributes?.expanded;
  const ariaHasPopup = accessibilityAttributes?.hasPopup;
  const ariaControls = accessibilityAttributes?.controls;
  const ariaPressed = accessibilityAttributes?.pressed;
  const role = accessibleRole === "Link" /* Link */ || accessibleRole === "Link" ? "link" : void 0;
  const isDisabled = disabled;
  const effectiveIcon = isMultiState ? displayedState?.icon : icon;
  const effectiveEndIcon = isMultiState ? displayedState?.endIcon : endIcon;
  const effectiveChildren = isMultiState ? displayedState?.text : children;
  const effectiveIconOnly = isMultiState ? !displayedState?.text && !!displayedState?.icon : iconOnly;
  const effectivePulsing = isMultiState ? displayedState?.pulsing : false;
  const PulsingText = ({ children: text }) => /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "opacity-50", children: text }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "absolute inset-0 overflow-hidden",
        style: {
          background: "linear-gradient(90deg, transparent 0%, currentColor 50%, transparent 100%)",
          backgroundSize: "200% 100%",
          WebkitBackgroundClip: "text",
          backgroundClip: "text",
          WebkitTextFillColor: "transparent",
          animation: "pulse-sweep 1.5s ease-in-out infinite"
        },
        children: text
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: `
          @keyframes pulse-sweep {
            0% { background-position: 200% 0; }
            100% { background-position: -200% 0; }
          }
        ` })
  ] });
  const getTransitionStyles = (phase, transition) => {
    if (transition === "none" || phase === "idle") {
      return {};
    }
    const baseTransition = "all 100ms ease-out";
    switch (transition) {
      case "slide-up":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(-8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-down":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(-8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-left":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(-12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "slide-right":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(-12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "fade":
        if (phase === "fade-out") return { opacity: 0, transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0 };
        if (phase === "fade-in") return { opacity: 1, transition: baseTransition };
        break;
      case "scale":
        if (phase === "fade-out") return { opacity: 0, transform: "scale(0.75)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "scale(0.75)" };
        if (phase === "fade-in") return { opacity: 1, transform: "scale(1)", transition: baseTransition };
        break;
      case "flip":
        if (phase === "fade-out") return { opacity: 0, transform: "rotateX(90deg)", transition: baseTransition, transformStyle: "preserve-3d" };
        if (phase === "fade-mid") return { opacity: 0, transform: "rotateX(-90deg)", transformStyle: "preserve-3d" };
        if (phase === "fade-in") return { opacity: 1, transform: "rotateX(0)", transition: baseTransition, transformStyle: "preserve-3d" };
        break;
    }
    return {};
  };
  const contentAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  const iconAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: buttonRef,
        type: getButtonType(type),
        disabled: isDisabled,
        "aria-disabled": loading || void 0,
        id,
        name,
        value,
        form,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-expanded": ariaExpanded,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-pressed": ariaPressed,
        "aria-busy": loading || void 0,
        title: tooltip,
        tabIndex,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur: (e) => {
          buttonRef.current?.classList.remove("kb-active");
          onBlur?.(e);
        },
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][effectiveIconOnly ? "iconOnly" : "default"],
          isMultiState && !effectiveIconOnly && "transition-[width] duration-300 ease-out overflow-hidden",
          loading && "relative pointer-events-none",
          className
        ),
        style: {
          ...style,
          ...buttonWidth !== void 0 && !effectiveIconOnly ? { width: buttonWidth } : {}
        },
        "data-testid": dataTestId,
        "data-growing-button": dataGrowingButton,
        children: [
          loading && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "absolute inset-0 flex items-center justify-center z-[1] [&_*]:!text-inherit", inert: true, children: /* @__PURE__ */ jsxRuntime.jsx(BusyIndicator, { size: "S", active: true, delay: 0 }) }),
          effectiveIcon ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? iconAnimationStyles : void 0, children: effectiveIcon }) : null,
          !effectiveIconOnly && effectiveChildren && /* @__PURE__ */ jsxRuntime.jsx("span", { className: loading ? "opacity-40" : void 0, style: isMultiState ? contentAnimationStyles : void 0, children: effectivePulsing ? /* @__PURE__ */ jsxRuntime.jsx(PulsingText, { children: effectiveChildren }) : effectiveChildren }),
          effectiveEndIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? contentAnimationStyles : void 0, children: effectiveEndIcon })
        ]
      }
    ),
    isMultiState && /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: hiddenButtonRef,
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][!currentStateObject?.text && currentStateObject?.icon ? "iconOnly" : "default"],
          "absolute -top-[10000px] -left-[10000px] invisible w-fit"
        ),
        tabIndex: -1,
        "aria-hidden": "true",
        children: [
          currentStateObject?.icon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.icon }),
          currentStateObject?.text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.text }),
          currentStateObject?.endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.endIcon })
        ]
      }
    )
  ] });
}
function CalendarHeader({
  monthText,
  yearText,
  cldrReady = true,
  isPreviousDisabled,
  isNextDisabled,
  onPrevious,
  onNext,
  onMonthClick,
  onYearClick,
  previousTooltip = "Previous",
  nextTooltip = "Next",
  className,
  ref
}) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref,
      className: cn(
        "flex items-center gap-2 px-2 py-[3px]",
        className
      ),
      role: "toolbar",
      "aria-label": "Calendar navigation",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            size: "Medium" /* Medium */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsxRuntime.jsx(SlimArrowLeftIcon, { className: "h-4 w-4 rtl:rotate-180" }),
            accessibleName: previousTooltip,
            tooltip: previousTooltip,
            disabled: isPreviousDisabled,
            onClick: onPrevious
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex items-center gap-2 flex-1 justify-center", role: "group", "aria-label": "Current date", children: [
          monthText && /* @__PURE__ */ jsxRuntime.jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              size: "Medium" /* Medium */,
              accessibleName: `${monthText}, Select month`,
              onClick: onMonthClick,
              className: "text-sapphire-text-accent",
              children: cldrReady ? monthText : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-4 w-16 rounded bg-muted animate-pulse" })
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              size: "Medium" /* Medium */,
              accessibleName: `${yearText}, Select year`,
              onClick: onYearClick,
              className: "text-sapphire-text-accent",
              children: cldrReady ? yearText : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-4 w-10 rounded bg-muted animate-pulse" })
            }
          )
        ] }),
        /* @__PURE__ */ jsxRuntime.jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            size: "Medium" /* Medium */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "h-4 w-4 rtl:rotate-180" }),
            accessibleName: nextTooltip,
            tooltip: nextTooltip,
            disabled: isNextDisabled,
            onClick: onNext
          }
        )
      ]
    }
  );
}
var cache = /* @__PURE__ */ new Map();
function ensureCldr(localeStr) {
  let p = cache.get(localeStr);
  if (!p) {
    const loc = new Locale2__default.default(localeStr);
    p = LocaleData_js.fetchCldr(loc.getLanguage(), loc.getRegion(), loc.getScript()).catch(
      (err) => {
        console.warn(
          `[fx-components] CLDR load failed for "${localeStr}":`,
          err
        );
        cache.delete(localeStr);
      }
    );
    cache.set(localeStr, p);
  }
  return p;
}
function isCldrLoaded(localeStr) {
  try {
    const localeData = getCachedLocaleDataInstance4__default.default(new Locale2__default.default(localeStr));
    localeData.getDatePattern("medium");
    return true;
  } catch {
    return false;
  }
}
ensureCldr(typeof navigator !== "undefined" ? navigator.language : "en");

// src/i18n/useEnsureCldr.ts
function useEnsureCldr(locale = "en-US") {
  const [readyLocale, setReadyLocale] = React6.useState(
    () => isCldrLoaded(locale) ? locale : null
  );
  React6.useEffect(() => {
    if (isCldrLoaded(locale)) {
      setReadyLocale(locale);
      return;
    }
    let cancelled = false;
    setReadyLocale(null);
    ensureCldr(locale).then(() => {
      if (!cancelled) setReadyLocale(locale);
    });
    return () => {
      cancelled = true;
    };
  }, [locale]);
  return readyLocale === locale || isCldrLoaded(locale);
}
var dayCellVariants = cva(
  "relative z-[1] w-8 h-8 rounded text-sm font-normal transition-colors focus:outline-none focus:[box-shadow:inset_0_0_0_2px_var(--border-focus)] flex items-center justify-center",
  {
    variants: {
      variant: {
        default: "text-foreground",
        selected: "",
        rangeStart: "",
        rangeEnd: "",
        rangeBetween: "",
        today: "",
        outside: "text-sapphire-text-tertiary",
        disabled: "text-sapphire-text-tertiary opacity-40 cursor-not-allowed"
      },
      specialType: {
        none: "",
        Type01: "border-b-2 border-b-blue-500",
        Type02: "border-b-2 border-b-green-500",
        Type03: "border-b-2 border-b-yellow-500",
        Type04: "border-b-2 border-b-red-500",
        Type05: "border-b-2 border-b-purple-500",
        Type06: "border-b-2 border-b-pink-500",
        Type07: "border-b-2 border-b-indigo-500",
        Type08: "border-b-2 border-b-cyan-500",
        Type09: "border-b-2 border-b-teal-500",
        Type10: "border-b-2 border-b-orange-500",
        Type11: "border-b-2 border-b-lime-500",
        Type12: "border-b-2 border-b-emerald-500",
        Type13: "border-b-2 border-b-sky-500",
        Type14: "border-b-2 border-b-violet-500",
        Type15: "border-b-2 border-b-fuchsia-500",
        Type16: "border-b-2 border-b-rose-500",
        Type17: "border-b-2 border-b-amber-500",
        Type18: "border-b-2 border-b-slate-500",
        Type19: "border-b-2 border-b-zinc-500",
        Type20: "border-b-2 border-b-neutral-500",
        Today: "font-bold",
        Selected: "",
        Working: "bg-sapphire-positive-bg",
        NonWorking: "bg-sapphire-negative-bg"
      }
    },
    defaultVariants: {
      variant: "default",
      specialType: "none"
    }
  }
);
var dayCellInnerVariants = cva(
  "w-6 h-6 rounded flex items-center justify-center transition-colors",
  {
    variants: {
      variant: {
        default: "hover:bg-sapphire-bg-secondary",
        selected: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background hover:text-sapphire-text-on-surface",
        rangeStart: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background",
        rangeEnd: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background",
        rangeBetween: "border border-sapphire-border-accent bg-sapphire-brand-toggle-background text-sapphire-text hover:bg-sapphire-brand-toggle-background",
        today: "rounded-sm border border-sapphire-border-accent text-sapphire-text-accent font-semibold",
        outside: "",
        disabled: ""
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
var FALLBACK_WIDE_DAYS = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
var FALLBACK_FIRST_DAY_OF_WEEK = 1;
var FALLBACK_WIDE_MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
function DayPicker({
  currentMonth,
  focusedDate,
  autoFocus = false,
  isDateSelected,
  isDateDisabled,
  isRangeStart,
  isRangeEnd,
  isDateBetweenRange,
  isRangeSelectionInProgress = false,
  getSpecialDate,
  onSelectDate,
  onFocusChange,
  onHoverDate,
  showWeekNumbers = false,
  weekNumbering,
  firstDayOfWeek,
  locale,
  className,
  ref
}) {
  const cldrReady = useEnsureCldr(locale);
  const effectiveFirstDay = React6__default.default.useMemo(() => {
    if (firstDayOfWeek !== void 0) return firstDayOfWeek;
    if (cldrReady) return getWeekStartDay(locale);
    return FALLBACK_FIRST_DAY_OF_WEEK;
  }, [firstDayOfWeek, cldrReady, locale]);
  const days = getMonthDays(currentMonth, effectiveFirstDay);
  const weekDays = React6__default.default.useMemo(() => {
    if (!cldrReady) return null;
    const localeData = getCachedLocaleDataInstance4__default.default(getLocale(locale));
    const allDays = localeData.getDaysStandAlone("abbreviated");
    const start = (effectiveFirstDay % 7 + 7) % 7;
    return [...allDays.slice(start), ...allDays.slice(0, start)];
  }, [locale, cldrReady, effectiveFirstDay]);
  const { wideDays, wideMonths } = React6__default.default.useMemo(() => {
    if (!cldrReady) return {
      wideDays: FALLBACK_WIDE_DAYS,
      wideMonths: FALLBACK_WIDE_MONTHS
    };
    const localeData = getCachedLocaleDataInstance4__default.default(getLocale(locale));
    return {
      wideDays: localeData.getDaysStandAlone("wide"),
      wideMonths: localeData.getMonthsStandAlone("wide")
    };
  }, [locale, cldrReady]);
  const gridRef = React6.useRef(null);
  const focusedCellRef = React6.useRef(null);
  const isMountedRef = React6.useRef(false);
  const setFocusedCellRef = React6.useCallback((node) => {
    focusedCellRef.current = node;
    if (node && isMountedRef.current) {
      node.focus({ preventScroll: true });
    }
  }, []);
  React6.useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, [focusedDate]);
  React6.useEffect(() => {
    isMountedRef.current = true;
  }, []);
  React6.useEffect(() => {
    if (!autoFocus) return;
    const rafId = requestAnimationFrame(() => {
      focusedCellRef.current?.focus({ preventScroll: true });
    });
    return () => cancelAnimationFrame(rafId);
  }, []);
  const handleKeyDown = React6.useCallback(
    (e, date) => {
      let newDate = null;
      const currentDate = new Date(date);
      switch (e.key) {
        case "ArrowLeft":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() - 1);
          newDate = currentDate;
          break;
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() + 1);
          newDate = currentDate;
          break;
        case "ArrowUp":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() - 7);
          newDate = currentDate;
          break;
        case "ArrowDown":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() + 7);
          newDate = currentDate;
          break;
        case "Home":
          e.preventDefault();
          if (e.ctrlKey) {
            newDate = new Date(
              currentDate.getFullYear(),
              currentDate.getMonth(),
              1
            );
          } else {
            const dayOfWeek = currentDate.getDay();
            const diff = (dayOfWeek - effectiveFirstDay + 7) % 7;
            currentDate.setDate(currentDate.getDate() - diff);
            newDate = currentDate;
          }
          break;
        case "End":
          e.preventDefault();
          if (e.ctrlKey) {
            newDate = new Date(
              currentDate.getFullYear(),
              currentDate.getMonth() + 1,
              0
            );
          } else {
            const dayOfWeek = currentDate.getDay();
            const diff = (effectiveFirstDay + 6 - dayOfWeek + 7) % 7;
            currentDate.setDate(currentDate.getDate() + diff);
            newDate = currentDate;
          }
          break;
        case "PageUp":
          e.preventDefault();
          if (e.shiftKey && e.ctrlKey) {
            currentDate.setFullYear(currentDate.getFullYear() - 10);
          } else if (e.shiftKey) {
            currentDate.setFullYear(currentDate.getFullYear() - 1);
          } else {
            currentDate.setMonth(currentDate.getMonth() - 1);
          }
          newDate = currentDate;
          break;
        case "PageDown":
          e.preventDefault();
          if (e.shiftKey && e.ctrlKey) {
            currentDate.setFullYear(currentDate.getFullYear() + 10);
          } else if (e.shiftKey) {
            currentDate.setFullYear(currentDate.getFullYear() + 1);
          } else {
            currentDate.setMonth(currentDate.getMonth() + 1);
          }
          newDate = currentDate;
          break;
        case "Enter":
        case " ":
          e.preventDefault();
          onSelectDate(date);
          return;
      }
      if (newDate && !isDateDisabled(newDate)) {
        onFocusChange(newDate);
        if (isRangeSelectionInProgress && onHoverDate) {
          onHoverDate(newDate);
        }
      }
    },
    [isDateDisabled, onSelectDate, onFocusChange, isRangeSelectionInProgress, onHoverDate, effectiveFirstDay]
  );
  const handleMouseOver = React6.useCallback(
    (date) => {
      if (isRangeSelectionInProgress && onHoverDate) {
        onHoverDate(date);
      }
    },
    [isRangeSelectionInProgress, onHoverDate]
  );
  const handleMouseLeave = React6.useCallback(() => {
    if (isRangeSelectionInProgress && onHoverDate) {
      onHoverDate(null);
    }
  }, [isRangeSelectionInProgress, onHoverDate]);
  const getDayCellVariant = (date) => {
    if (isDateDisabled(date)) return "disabled";
    const isStart = isRangeStart(date);
    const isEnd = isRangeEnd(date);
    if (isStart && isEnd) return "selected";
    if (isStart) return "rangeStart";
    if (isEnd) return "rangeEnd";
    if (isDateBetweenRange(date)) return "rangeBetween";
    if (isDateSelected(date)) return "selected";
    if (!isSameMonthUtil(date, currentMonth)) return "outside";
    if (isToday(date)) return "today";
    return "default";
  };
  const getDayCellSpecialType = (date) => {
    const special = getSpecialDate(date);
    return special?.type || "None" /* None */;
  };
  const getWeekNumbers = () => {
    if (!showWeekNumbers) return [];
    const weeks = [];
    for (let i = 0; i < days.length; i += 7) {
      weeks.push(getWeekNumber(days[i], weekNumbering));
    }
    return weeks;
  };
  const getRangeStripClass = (date, idx) => {
    const isStart = isRangeStart(date);
    const isEnd = isRangeEnd(date);
    const isBetween = isDateBetweenRange(date);
    if (!isStart && !isEnd && !isBetween) return null;
    const colIdx = idx % 7;
    const isFirstCol = colIdx === 0;
    const isLastCol = colIdx === 6;
    if (isStart && isEnd) {
      return null;
    }
    if (isStart) {
      return isLastCol ? null : "range-strip range-strip-right";
    }
    if (isEnd) {
      return isFirstCol ? null : "range-strip range-strip-left";
    }
    if (isFirstCol) return "range-strip range-strip-right";
    if (isLastCol) return "range-strip range-strip-left";
    return "range-strip range-strip-both";
  };
  const weekNumbers = getWeekNumbers();
  const gridCols = showWeekNumbers ? "grid-cols-[repeat(8,2rem)]" : "grid-cols-[repeat(7,2rem)]";
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { ref, className: cn("p-2 flex flex-col items-center", className), children: [
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: `
          .range-strip {
            position: relative;
          }
          .range-strip::before {
            content: "";
            position: absolute;
            top: 0;
            bottom: 0;
            z-index: 0;
            background: transparent;
            border-radius: 0;
            pointer-events: none;
          }
          .range-strip-right::before {
            left: 0;
            right: 0;
            border-radius: 0;
          }
          .range-strip-left::before {
            left: 0;
            right: 0;
            border-radius: 0;
          }
          .range-strip-both::before {
            left: 0;
            right: 0;
            border-radius: 0;
          }
        ` }),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        ref: gridRef,
        className: "flex flex-col",
        role: "grid",
        "aria-label": "Calendar days",
        onMouseLeave: handleMouseLeave,
        children: [
          /* @__PURE__ */ jsxRuntime.jsxs("div", { role: "row", className: cn("grid", gridCols), children: [
            showWeekNumbers && /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                role: "columnheader",
                className: "h-8 w-full flex items-center justify-center text-xs font-medium text-sapphire-text-tertiary"
              }
            ),
            weekDays ? weekDays.map((day) => /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                role: "columnheader",
                className: "h-8 w-full flex items-center justify-center text-xs font-medium text-sapphire-text-tertiary",
                children: day
              },
              day
            )) : Array.from({ length: 7 }, (_, i) => /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                role: "columnheader",
                className: "h-8 w-full flex items-center justify-center",
                children: /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-3 w-6 rounded bg-muted animate-pulse" })
              },
              i
            ))
          ] }),
          Array.from({ length: Math.ceil(days.length / 7) }, (_, weekIndex) => {
            const weekStart = weekIndex * 7;
            const weekDates = days.slice(weekStart, weekStart + 7);
            const weekKey = weekDates.length > 0 ? `week-${formatDateToISO(weekDates[0])}` : `week-${weekIndex}`;
            return /* @__PURE__ */ jsxRuntime.jsxs("div", { role: "row", className: cn("grid items-center", gridCols), children: [
              showWeekNumbers && /* @__PURE__ */ jsxRuntime.jsx(
                "div",
                {
                  role: "rowheader",
                  className: "h-8 w-full flex items-center justify-center text-xs text-sapphire-text-tertiary",
                  children: cldrReady ? weekNumbers[weekIndex] : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-3 w-4 rounded bg-muted animate-pulse" })
                }
              ),
              weekDates.map((date, dayIndex) => {
                const idx = weekStart + dayIndex;
                const variant = getDayCellVariant(date);
                const specialType = getDayCellSpecialType(date);
                const special = getSpecialDate(date);
                const isFocused = focusedDate && date.getTime() === focusedDate.getTime();
                const isCurrentMonth = isSameMonthUtil(date, currentMonth);
                const isTodayDate = isToday(date);
                const isDisabled = isDateDisabled(date);
                const rangeStripClass = getRangeStripClass(date, idx);
                const dayName = wideDays[date.getDay()];
                const monthName = wideMonths[date.getMonth()];
                let ariaLabel = `${dayName}, ${monthName} ${date.getDate()}, ${date.getFullYear()}`;
                if (isTodayDate) ariaLabel += ", Today";
                if (!isCurrentMonth) ariaLabel += ", Outside current month";
                if (isDisabled) ariaLabel += ", Not available";
                if (special?.tooltip) ariaLabel += `, ${special.tooltip}`;
                const isWeekend = date.getDay() === 0 || date.getDay() === 6;
                const gridcell = /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    ref: isFocused ? setFocusedCellRef : void 0,
                    role: "gridcell",
                    "aria-label": ariaLabel,
                    "aria-selected": isDateSelected(date),
                    "aria-current": isTodayDate ? "date" : void 0,
                    "aria-disabled": isDisabled,
                    title: special?.tooltip,
                    tabIndex: isFocused ? 0 : -1,
                    className: cn(
                      dayCellVariants({
                        variant,
                        specialType
                      }),
                      "cursor-pointer",
                      isWeekend && variant === "default" && "bg-sapphire-background-quaternary border border-sapphire-background-primary rounded"
                    ),
                    onClick: () => {
                      if (!isDateDisabled(date)) {
                        onFocusChange(date);
                        onSelectDate(date);
                      }
                    },
                    onMouseOver: () => handleMouseOver(date),
                    onKeyDown: (e) => handleKeyDown(e, date),
                    children: /* @__PURE__ */ jsxRuntime.jsx("span", { className: dayCellInnerVariants({ variant }), children: cldrReady ? format(date, "d") : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-3 w-4 rounded bg-muted animate-pulse" }) })
                  }
                );
                return rangeStripClass ? /* @__PURE__ */ jsxRuntime.jsx("div", { className: rangeStripClass, children: gridcell }, formatDateToISO(date)) : /* @__PURE__ */ jsxRuntime.jsx(React6__default.default.Fragment, { children: gridcell }, formatDateToISO(date));
              })
            ] }, weekKey);
          })
        ]
      }
    )
  ] });
}
var monthCellVariants = cva(
  "min-h-[2.75rem] w-full rounded-lg text-sm font-normal transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "hover:bg-sapphire-bg-secondary hover:text-sapphire-text",
        selected: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function MonthPicker({
  currentYear,
  focusedMonth,
  selectedMonth,
  onSelectMonth,
  onFocusChange,
  onPreviousYear,
  onNextYear,
  locale,
  className,
  ref
}) {
  const months = Array.from({ length: 12 }, (_, i) => i);
  const focusedCellRef = React6.useRef(null);
  const cldrReady = useEnsureCldr(locale);
  React6.useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, [focusedMonth]);
  React6.useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, []);
  const handleKeyDown = React6.useCallback(
    (e, month) => {
      let newMonth = null;
      switch (e.key) {
        case "ArrowLeft":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month === 0 ? 11 : month - 1;
          break;
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month === 11 ? 0 : month + 1;
          break;
        case "ArrowUp":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month < 3 ? month + 9 : month - 3;
          break;
        case "ArrowDown":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month > 8 ? month - 9 : month + 3;
          break;
        case "Home":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newMonth = 0;
          } else {
            newMonth = Math.floor(month / 3) * 3;
          }
          break;
        case "End":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newMonth = 11;
          } else {
            newMonth = Math.min(Math.floor(month / 3) * 3 + 2, 11);
          }
          break;
        case "PageUp":
          e.preventDefault();
          e.stopPropagation();
          onPreviousYear?.();
          return;
        case "PageDown":
          e.preventDefault();
          e.stopPropagation();
          onNextYear?.();
          return;
        case "Enter":
        case " ":
          e.preventDefault();
          e.stopPropagation();
          onSelectMonth(month);
          return;
      }
      if (newMonth !== null) {
        onFocusChange(newMonth);
      }
    },
    [onSelectMonth, onFocusChange, onPreviousYear, onNextYear]
  );
  return /* @__PURE__ */ jsxRuntime.jsx("div", { ref, className: cn("p-3 w-full", className), children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex flex-col gap-2 w-full", role: "grid", "aria-label": "Select month", children: Array.from({ length: 4 }, (_, rowIndex) => {
    const rowMonths = months.slice(rowIndex * 3, (rowIndex + 1) * 3);
    return /* @__PURE__ */ jsxRuntime.jsx("div", { role: "row", className: "grid grid-cols-3 gap-2", children: rowMonths.map((month) => {
      const date = createDate(currentYear, month);
      const monthName = cldrReady ? getMonthName(date, "short", locale) : "";
      const isFocused = month === focusedMonth;
      const isSelected = month === selectedMonth;
      return /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          ref: isFocused ? focusedCellRef : void 0,
          role: "gridcell",
          "aria-label": `${monthName} ${currentYear}`,
          "aria-selected": isSelected,
          tabIndex: isFocused ? 0 : -1,
          className: cn(
            monthCellVariants({
              variant: isSelected ? "selected" : "default"
            }),
            "flex items-center justify-center cursor-pointer"
          ),
          onClick: () => onSelectMonth(month),
          onKeyDown: (e) => handleKeyDown(e, month),
          children: cldrReady ? monthName : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-3.5 w-8 rounded bg-muted animate-pulse" })
        },
        month
      );
    }) }, rowIndex);
  }) }) });
}
var yearCellVariants = cva(
  "min-h-[2.75rem] w-full rounded-lg text-sm font-normal transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "hover:bg-sapphire-bg-secondary hover:text-sapphire-text",
        selected: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function YearPicker({
  startYear,
  endYear,
  focusedYear,
  selectedYear,
  onSelectYear,
  onFocusChange,
  onPreviousRange,
  onNextRange,
  className,
  ref
}) {
  const years = Array.from(
    { length: endYear - startYear + 1 },
    (_, i) => startYear + i
  );
  const focusedCellRef = React6.useRef(null);
  React6.useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, [focusedYear]);
  React6.useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, []);
  const handleKeyDown = React6.useCallback(
    (e, year) => {
      const currentIndex = years.indexOf(year);
      let newYear = null;
      switch (e.key) {
        case "ArrowLeft":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex > 0) {
            newYear = years[currentIndex - 1];
          } else {
            onPreviousRange();
          }
          break;
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex < years.length - 1) {
            newYear = years[currentIndex + 1];
          } else {
            onNextRange();
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex >= 4) {
            newYear = years[currentIndex - 4];
          } else {
            onPreviousRange();
          }
          break;
        case "ArrowDown":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex < years.length - 4) {
            newYear = years[currentIndex + 4];
          } else {
            onNextRange();
          }
          break;
        case "Home":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newYear = years[0];
          } else {
            newYear = years[Math.floor(currentIndex / 4) * 4];
          }
          break;
        case "End":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newYear = years[years.length - 1];
          } else {
            newYear = years[Math.min(Math.floor(currentIndex / 4) * 4 + 3, years.length - 1)];
          }
          break;
        case "PageUp":
          e.preventDefault();
          e.stopPropagation();
          onPreviousRange();
          return;
        case "PageDown":
          e.preventDefault();
          e.stopPropagation();
          onNextRange();
          return;
        case "Enter":
        case " ":
          e.preventDefault();
          e.stopPropagation();
          onSelectYear(year);
          return;
      }
      if (newYear !== null) {
        onFocusChange(newYear);
      }
    },
    [years, onSelectYear, onFocusChange, onPreviousRange, onNextRange]
  );
  return /* @__PURE__ */ jsxRuntime.jsx("div", { ref, className: cn("p-3 w-full", className), children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex flex-col gap-2 w-full", role: "grid", "aria-label": "Select year", children: Array.from({ length: 5 }, (_, rowIndex) => {
    const rowYears = years.slice(rowIndex * 4, (rowIndex + 1) * 4);
    return /* @__PURE__ */ jsxRuntime.jsx("div", { role: "row", className: "grid grid-cols-4 gap-2", children: rowYears.map((year) => {
      const isFocused = year === focusedYear;
      const isSelected = year === selectedYear;
      return /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          ref: isFocused ? focusedCellRef : void 0,
          role: "gridcell",
          "aria-label": `Year ${year}`,
          "aria-selected": isSelected,
          tabIndex: isFocused ? 0 : -1,
          className: cn(
            yearCellVariants({
              variant: isSelected ? "selected" : "default"
            }),
            "flex items-center justify-center cursor-pointer"
          ),
          onClick: () => onSelectYear(year),
          onKeyDown: (e) => handleKeyDown(e, year),
          children: year
        },
        year
      );
    }) }, rowIndex);
  }) }) });
}
var legendItemBoxVariants = cva(
  "h-4 w-4 border border-sapphire-border flex items-center justify-center rounded-sm",
  {
    variants: {
      type: {
        ["None" /* None */]: "",
        ["Today" /* Today */]: "border-2 border-sapphire-border-accent bg-transparent",
        ["Selected" /* Selected */]: "bg-sapphire-brand-hover-background",
        ["Working" /* Working */]: "bg-sapphire-positive-bg",
        ["NonWorking" /* NonWorking */]: "bg-sapphire-negative-bg",
        ["Type01" /* Type01 */]: "bg-blue-500",
        ["Type02" /* Type02 */]: "bg-green-500",
        ["Type03" /* Type03 */]: "bg-yellow-500",
        ["Type04" /* Type04 */]: "bg-red-500",
        ["Type05" /* Type05 */]: "bg-purple-500",
        ["Type06" /* Type06 */]: "bg-pink-500",
        ["Type07" /* Type07 */]: "bg-indigo-500",
        ["Type08" /* Type08 */]: "bg-cyan-500",
        ["Type09" /* Type09 */]: "bg-teal-500",
        ["Type10" /* Type10 */]: "bg-orange-500",
        ["Type11" /* Type11 */]: "bg-lime-500",
        ["Type12" /* Type12 */]: "bg-emerald-500",
        ["Type13" /* Type13 */]: "bg-sky-500",
        ["Type14" /* Type14 */]: "bg-violet-500",
        ["Type15" /* Type15 */]: "bg-fuchsia-500",
        ["Type16" /* Type16 */]: "bg-rose-500",
        ["Type17" /* Type17 */]: "bg-amber-500",
        ["Type18" /* Type18 */]: "bg-slate-500",
        ["Type19" /* Type19 */]: "bg-zinc-500",
        ["Type20" /* Type20 */]: "bg-neutral-500"
      }
    },
    defaultVariants: {
      type: "None" /* None */
    }
  }
);
function CalendarLegendItem({ type, text, onClick, className, ref }) {
  const isSelected = type === "Selected" /* Selected */;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref,
      role: "listitem",
      tabIndex: -1,
      onClick,
      className: cn(
        "flex items-center gap-2 cursor-pointer",
        "hover:bg-sapphire-bg-secondary rounded-sm p-1 transition-colors",
        "focus:outline-none focus:ring-2 focus:ring-sapphire-ring focus:ring-offset-1",
        className
      ),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("div", { className: cn(legendItemBoxVariants({ type })), children: isSelected && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "h-1.5 w-1.5 rounded-full bg-sapphire-text-on-surface" }) }),
        /* @__PURE__ */ jsxRuntime.jsx("span", { className: "text-sm text-sapphire-text truncate", children: text })
      ]
    }
  );
}
var getDefaultLegendItemText = (type, _locale) => {
  const labels = {
    ["Today" /* Today */]: "Today",
    ["Selected" /* Selected */]: "Selected Day",
    ["Working" /* Working */]: "Working Day",
    ["NonWorking" /* NonWorking */]: "Non-Working Day"
  };
  return labels[type] || type;
};
function CalendarLegend({
  hideToday = false,
  hideSelectedDay = false,
  hideNonWorkingDay = false,
  hideWorkingDay = false,
  items = [],
  onLegendItemSelect,
  locale,
  className,
  ref
}) {
  const defaultItems = [
    {
      type: "Today" /* Today */,
      text: getDefaultLegendItemText("Today" /* Today */),
      hidden: hideToday
    },
    {
      type: "Selected" /* Selected */,
      text: getDefaultLegendItemText("Selected" /* Selected */),
      hidden: hideSelectedDay
    },
    {
      type: "Working" /* Working */,
      text: getDefaultLegendItemText("Working" /* Working */),
      hidden: hideWorkingDay
    },
    {
      type: "NonWorking" /* NonWorking */,
      text: getDefaultLegendItemText("NonWorking" /* NonWorking */),
      hidden: hideNonWorkingDay
    }
  ].filter((item) => !item.hidden);
  const allItems = [...defaultItems, ...items];
  if (allItems.length === 0) {
    return null;
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref,
      role: "list",
      "aria-label": "Calendar legend",
      className: cn(
        "p-3 bg-sapphire-card-bg border-t border-sapphire-border",
        "columns-2 sm:columns-3 gap-2",
        "w-full",
        className
      ),
      children: allItems.map((item, index) => /* @__PURE__ */ jsxRuntime.jsx(
        CalendarLegendItem,
        {
          type: item.type,
          text: item.text,
          onClick: onLegendItemSelect ? () => onLegendItemSelect({ type: item.type, text: item.text }) : void 0
        },
        `${item.type}-${index}`
      ))
    }
  );
}
function getInitialView(pickersMode) {
  switch (pickersMode) {
    case "YEAR" /* YEAR */:
      return "year" /* Year */;
    case "MONTH_YEAR" /* MONTH_YEAR */:
      return "month" /* Month */;
    default:
      return "day" /* Day */;
  }
}
function Calendar({
  selectionMode = "Single" /* Single */,
  hideWeekNumbers = false,
  calendarWeekNumbering,
  pickersMode = "DAY_MONTH_YEAR" /* DAY_MONTH_YEAR */,
  minDate,
  maxDate,
  selectedDates,
  rangeStartDate,
  rangeEndDate,
  specialDates,
  disabledDates,
  focusedDate,
  locale: localeProp,
  firstDayOfWeek,
  showLegend = false,
  hideLegendToday = false,
  hideLegendSelectedDay = false,
  hideLegendWorkingDay = false,
  hideLegendNonWorkingDay = false,
  legendItems = [],
  autoFocus = false,
  onSelectionChange,
  onShowMonthView,
  onShowYearView,
  onLegendItemSelect,
  className,
  ref
}) {
  const { i18n } = reactI18next.useTranslation("fx");
  const locale = localeProp || i18n.language;
  const cldrReady = useEnsureCldr(locale);
  const [currentView, setCurrentView] = React6.useState(
    () => getInitialView(pickersMode)
  );
  const [displayedDate, setDisplayedDate] = React6.useState(() => {
    if (focusedDate) return parseDate(focusedDate);
    if (selectedDates && selectedDates.length > 0) return parseDate(selectedDates[0]);
    if (rangeStartDate) return parseDate(rangeStartDate);
    return getToday();
  });
  const [focusDate, setFocusDate] = React6.useState(displayedDate);
  const [yearRangeStart, setYearRangeStart] = React6.useState(() => {
    const year = getYear(displayedDate);
    return Math.floor(year / 20) * 20;
  });
  const minDateObj = React6.useMemo(() => minDate ? parseDate(minDate) : void 0, [minDate]);
  const maxDateObj = React6.useMemo(() => maxDate ? parseDate(maxDate) : void 0, [maxDate]);
  const isPreviousDisabled = React6.useMemo(() => {
    if (!minDateObj) return false;
    if (currentView === "day" /* Day */) {
      const prevMonth = addMonthsToDate(displayedDate, -1);
      const lastDayOfPrevMonth = new Date(getYear(prevMonth), getMonth(prevMonth) + 1, 0);
      return lastDayOfPrevMonth < minDateObj;
    } else if (currentView === "month" /* Month */) {
      return getYear(displayedDate) - 1 < getYear(minDateObj);
    } else if (currentView === "year" /* Year */) {
      return yearRangeStart - 1 < getYear(minDateObj);
    }
    return false;
  }, [minDateObj, currentView, displayedDate, yearRangeStart]);
  const isNextDisabled = React6.useMemo(() => {
    if (!maxDateObj) return false;
    if (currentView === "day" /* Day */) {
      const nextMonth = addMonthsToDate(displayedDate, 1);
      const firstDayOfNextMonth = createDate(getYear(nextMonth), getMonth(nextMonth), 1);
      return firstDayOfNextMonth > maxDateObj;
    } else if (currentView === "month" /* Month */) {
      return getYear(displayedDate) + 1 > getYear(maxDateObj);
    } else if (currentView === "year" /* Year */) {
      return yearRangeStart + 20 > getYear(maxDateObj);
    }
    return false;
  }, [maxDateObj, currentView, displayedDate, yearRangeStart]);
  const selection = useCalendarSelection(
    selectionMode,
    selectedDates,
    rangeStartDate,
    rangeEndDate,
    minDate,
    maxDate,
    disabledDates
  );
  const { getSpecialDate } = useSpecialDates(specialDates);
  const handleDateSelect = React6.useCallback(
    (date) => {
      const result = selection.selectDate(date);
      if (onSelectionChange && result) {
        onSelectionChange({
          selectedValues: result.selectedValues,
          selectedDates: result.selectedTimestamps,
          timestamp: Math.floor(date.getTime() / 1e3)
        });
      }
    },
    [selection, onSelectionChange]
  );
  const handleMonthSelect = React6.useCallback((month) => {
    const newDate = createDate(getYear(displayedDate), month);
    setDisplayedDate(newDate);
    setFocusDate(newDate);
    if (pickersMode === "MONTH_YEAR" /* MONTH_YEAR */) {
      if (onSelectionChange) {
        const isoValue = `${getYear(newDate)}-${String(month + 1).padStart(2, "0")}`;
        onSelectionChange({
          selectedValues: [isoValue],
          selectedDates: [Math.floor(newDate.getTime() / 1e3)],
          timestamp: Math.floor(newDate.getTime() / 1e3)
        });
      }
    } else {
      setCurrentView("day" /* Day */);
    }
  }, [displayedDate, pickersMode, onSelectionChange]);
  const handleYearSelect = React6.useCallback((year) => {
    const newDate = createDate(year, getMonth(displayedDate));
    setDisplayedDate(newDate);
    setFocusDate(newDate);
    if (pickersMode === "YEAR" /* YEAR */) {
      if (onSelectionChange) {
        onSelectionChange({
          selectedValues: [String(year)],
          selectedDates: [Math.floor(newDate.getTime() / 1e3)],
          timestamp: Math.floor(newDate.getTime() / 1e3)
        });
      }
    } else {
      setCurrentView("month" /* Month */);
    }
  }, [displayedDate, pickersMode, onSelectionChange]);
  const handlePrevious = React6.useCallback(() => {
    if (currentView === "day" /* Day */) {
      setDisplayedDate((prev) => addMonthsToDate(prev, -1));
    } else if (currentView === "month" /* Month */) {
      setDisplayedDate((prev) => createDate(getYear(prev) - 1, getMonth(prev)));
    } else if (currentView === "year" /* Year */) {
      setYearRangeStart((prev) => prev - 20);
    }
  }, [currentView]);
  const handleNext = React6.useCallback(() => {
    if (currentView === "day" /* Day */) {
      setDisplayedDate((prev) => addMonthsToDate(prev, 1));
    } else if (currentView === "month" /* Month */) {
      setDisplayedDate((prev) => createDate(getYear(prev) + 1, getMonth(prev)));
    } else if (currentView === "year" /* Year */) {
      setYearRangeStart((prev) => prev + 20);
    }
  }, [currentView]);
  const handleShowMonthView = React6.useCallback(() => {
    if (pickersMode === "YEAR" /* YEAR */) return;
    setCurrentView("month" /* Month */);
    onShowMonthView?.();
  }, [onShowMonthView, pickersMode]);
  const handleShowYearView = React6.useCallback(() => {
    setCurrentView("year" /* Year */);
    onShowYearView?.();
  }, [onShowYearView]);
  const getHeaderText = React6.useCallback(() => {
    const year = getYear(displayedDate);
    const FALLBACK_LONG_MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
    const month = cldrReady ? getMonthName(displayedDate, "long", locale) : FALLBACK_LONG_MONTHS[displayedDate.getMonth()];
    if (currentView === "day" /* Day */) {
      return { monthText: month, yearText: String(year) };
    } else if (currentView === "month" /* Month */) {
      return { monthText: "", yearText: String(year) };
    } else if (currentView === "year" /* Year */) {
      return { monthText: "", yearText: `${yearRangeStart} - ${yearRangeStart + 19}` };
    }
    return { monthText: "", yearText: "" };
  }, [currentView, displayedDate, yearRangeStart, locale, cldrReady]);
  const handleHoverDate = React6.useCallback((date) => {
    selection.setHoveredDate(date);
  }, [selection]);
  const handleFocusChange = React6.useCallback((date) => {
    setFocusDate(date);
    if (getMonth(date) !== getMonth(displayedDate)) {
      setDisplayedDate(date);
    }
  }, [displayedDate]);
  const handleContainerKeyDown = React6.useCallback((e) => {
    if (e.key === "F4") {
      e.preventDefault();
      if (e.shiftKey) {
        handleShowYearView();
      } else {
        handleShowMonthView();
      }
    }
  }, [handleShowMonthView, handleShowYearView]);
  const getTooltipText = React6.useCallback(() => {
    if (currentView === "day" /* Day */) {
      return { previous: "Previous month", next: "Next month" };
    } else if (currentView === "month" /* Month */) {
      return { previous: "Previous year", next: "Next year" };
    } else if (currentView === "year" /* Year */) {
      return { previous: "Previous year range", next: "Next year range" };
    }
    return { previous: "Previous", next: "Next" };
  }, [currentView]);
  const headerText = getHeaderText();
  const tooltipText = getTooltipText();
  const showMonthButton = pickersMode !== "YEAR" /* YEAR */ && pickersMode !== "MONTH_YEAR" /* MONTH_YEAR */ && currentView === "day" /* Day */;
  const rootRef = React6.useRef(null);
  React6.useImperativeHandle(ref, () => ({
    switchToDayPicker() {
      if (pickersMode === "YEAR" /* YEAR */ || pickersMode === "MONTH_YEAR" /* MONTH_YEAR */) return;
      setCurrentView("day" /* Day */);
    },
    switchToMonthPicker() {
      if (pickersMode === "YEAR" /* YEAR */) return;
      setCurrentView("month" /* Month */);
    },
    switchToYearPicker() {
      setCurrentView("year" /* Year */);
    },
    get nativeElement() {
      return rootRef.current;
    }
  }), [pickersMode]);
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      "data-testid": "calendar",
      onKeyDown: handleContainerKeyDown,
      className: cn(
        "flex flex-col rounded-lg border border-sapphire-border-primary bg-sapphire-card-bg-primary text-foreground shadow-[0_4px_10px_-2px_rgba(10,10,10,0.15),0_2px_5px_-2px_rgba(10,10,10,0.05)]",
        "w-fit min-w-[17rem]",
        className
      ),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          CalendarHeader,
          {
            monthText: showMonthButton ? headerText.monthText : "",
            yearText: headerText.yearText,
            cldrReady,
            isPreviousDisabled,
            isNextDisabled,
            previousTooltip: tooltipText.previous,
            nextTooltip: tooltipText.next,
            onPrevious: handlePrevious,
            onNext: handleNext,
            onMonthClick: handleShowMonthView,
            onYearClick: handleShowYearView
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "relative w-full", children: [
          currentView === "day" /* Day */ && /* @__PURE__ */ jsxRuntime.jsx(
            DayPicker,
            {
              currentMonth: displayedDate,
              focusedDate: focusDate,
              autoFocus,
              isDateSelected: selection.isDateSelected,
              isDateDisabled: selection.isDateDisabled,
              isRangeStart: selection.isRangeStart,
              isRangeEnd: selection.isRangeEnd,
              isDateBetweenRange: selection.isDateBetweenRange,
              isRangeSelectionInProgress: selection.isRangeSelectionInProgress,
              getSpecialDate,
              onSelectDate: handleDateSelect,
              onFocusChange: handleFocusChange,
              onHoverDate: handleHoverDate,
              showWeekNumbers: !hideWeekNumbers,
              weekNumbering: calendarWeekNumbering,
              firstDayOfWeek,
              locale
            }
          ),
          currentView === "month" /* Month */ && /* @__PURE__ */ jsxRuntime.jsx(
            MonthPicker,
            {
              currentYear: getYear(displayedDate),
              focusedMonth: getMonth(focusDate),
              selectedMonth: getMonth(displayedDate),
              locale,
              onSelectMonth: handleMonthSelect,
              onFocusChange: (month) => {
                const newDate = createDate(getYear(displayedDate), month);
                setFocusDate(newDate);
              },
              onPreviousYear: () => {
                setDisplayedDate((prev) => createDate(getYear(prev) - 1, getMonth(prev)));
              },
              onNextYear: () => {
                setDisplayedDate((prev) => createDate(getYear(prev) + 1, getMonth(prev)));
              }
            }
          ),
          currentView === "year" /* Year */ && (cldrReady ? /* @__PURE__ */ jsxRuntime.jsx(
            YearPicker,
            {
              startYear: yearRangeStart,
              endYear: yearRangeStart + 19,
              focusedYear: getYear(focusDate),
              selectedYear: getYear(displayedDate),
              onSelectYear: handleYearSelect,
              onFocusChange: (year) => {
                const newDate = createDate(year, getMonth(displayedDate));
                setFocusDate(newDate);
              },
              onPreviousRange: () => setYearRangeStart((prev) => prev - 20),
              onNextRange: () => setYearRangeStart((prev) => prev + 20)
            }
          ) : /* @__PURE__ */ jsxRuntime.jsx("div", { className: "p-3 w-full", children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex flex-col gap-2 w-full", children: Array.from({ length: 5 }, (_, row) => /* @__PURE__ */ jsxRuntime.jsx("div", { className: "grid grid-cols-4 gap-2", children: Array.from({ length: 4 }, (_2, col) => /* @__PURE__ */ jsxRuntime.jsx("div", { className: "min-h-[2.75rem] w-full flex items-center justify-center", children: /* @__PURE__ */ jsxRuntime.jsx("span", { className: "inline-block h-3.5 w-8 rounded bg-muted animate-pulse" }) }, col)) }, row)) }) }))
        ] }),
        showLegend && /* @__PURE__ */ jsxRuntime.jsx(
          CalendarLegend,
          {
            hideToday: hideLegendToday,
            hideSelectedDay: hideLegendSelectedDay,
            hideWorkingDay: hideLegendWorkingDay,
            hideNonWorkingDay: hideLegendNonWorkingDay,
            items: legendItems,
            onLegendItemSelect,
            locale
          }
        )
      ]
    }
  );
}

exports.Calendar = Calendar;
exports.CalendarHeader = CalendarHeader;
exports.CalendarLegend = CalendarLegend;
exports.CalendarLegendItem = CalendarLegendItem;
exports.CalendarLegendItemType = CalendarLegendItemType;
exports.CalendarPickerView = CalendarPickerView;
exports.CalendarPickersMode = CalendarPickersMode;
exports.CalendarSelectionMode = CalendarSelectionMode;
exports.CalendarWeekNumbering = CalendarWeekNumbering;
exports.DayPicker = DayPicker;
exports.MonthPicker = MonthPicker;
exports.YearPicker = YearPicker;
