// @sap-ui/fx-components/dist/components/panel/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/panel/Panel.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData4,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData4 || pathData4.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData4.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M256 224q-12 0-18-8L103 75q-7-7-7-17 0-11 7.5-18.5T122 32q10 0 18 8l116 121L372 40q8-8 18-8 11 0 18.5 7.5T416 58q0 10-7 17L275 216q-8 8-19 8zm134 256q-10 0-18-8L256 351 140 472q-8 8-18 8-11 0-18.5-7.5T96 454q0-10 7-17l135-141q6-8 18-8 11 0 19 8l134 141q7 7 7 17 0 11-7.5 18.5T390 480z"
];
var CollapseIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "CollapseIcon", iconName: "collapse" }
);
var pathData2 = [
  "M256 32q11 0 19 8l134 141q7 7 7 17 0 11-7.5 18.5T390 224q-10 0-18-8L256 95 140 216q-8 8-18 8-11 0-18.5-7.5T96 198q0-10 7-17L238 40q6-8 18-8zm134 256q11 0 18.5 7.5T416 314q0 10-7 17L275 472q-8 8-19 8-12 0-18-8L103 331q-7-7-7-17 0-11 7.5-18.5T122 288q10 0 18 8l116 121 116-121q8-8 18-8z"
];
var ExpandIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "ExpandIcon", iconName: "expand" }
);
var pathData3 = [
  "M186 416q-11 0-18.5-7.5T160 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T186 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L203 409q-7 7-17 7z"
];
var NavigationRightArrowIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "NavigationRightArrowIcon", iconName: "navigation-right-arrow" }
);
var roleMap = {
  ["Complementary" /* Complementary */]: "complementary",
  ["Form" /* Form */]: "form",
  ["Region" /* Region */]: "region"
};
var levelToTag = {
  ["H1" /* H1 */]: "h1",
  ["H2" /* H2 */]: "h2",
  ["H3" /* H3 */]: "h3",
  ["H4" /* H4 */]: "h4",
  ["H5" /* H5 */]: "h5",
  ["H6" /* H6 */]: "h6"
};
var rootDesignClasses = {
  ["Primary" /* Primary */]: "bg-sapphire-card-bg-primary border border-sapphire-border-primary",
  ["Secondary" /* Secondary */]: "bg-sapphire-background-secondary border border-sapphire-border-primary",
  ["Child" /* Child */]: ""
};
var headerDesignClasses = {
  ["Primary" /* Primary */]: "pl-3 pr-4 py-3 gap-3",
  ["Secondary" /* Secondary */]: "pl-3 pr-4 py-3 gap-3",
  ["Child" /* Child */]: "pl-1 pr-2 py-1 gap-1"
};
var headerTitleClasses = {
  ["Primary" /* Primary */]: "text-base font-normal",
  ["Secondary" /* Secondary */]: "text-base font-normal",
  ["Child" /* Child */]: "text-sm font-normal"
};
var contentDesignClasses = {
  ["Primary" /* Primary */]: "p-4",
  ["Secondary" /* Secondary */]: "p-4",
  ["Child" /* Child */]: "p-3"
};
var headerBgClasses = {
  ["Primary" /* Primary */]: "bg-sapphire-card-bg-primary",
  ["Secondary" /* Secondary */]: "bg-sapphire-background-secondary",
  ["Child" /* Child */]: "bg-background"
};
var footerDesignClasses = {
  ["Primary" /* Primary */]: "px-4 py-4",
  ["Secondary" /* Secondary */]: "px-4 py-4 bg-sapphire-card-bg-secondary",
  ["Child" /* Child */]: ""
};
function Panel(props) {
  const {
    // Visual
    design = "Primary" /* Primary */,
    // Content
    children,
    headerText,
    header,
    subtitle,
    endSlot,
    footer,
    // Behavior
    fixed = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    noAnimation = false,
    stickyHeader = false,
    noPadding = false,
    // Accessibility
    accessibleRole = "Form" /* Form */,
    headerLevel = "H2" /* H2 */,
    accessibleName,
    // Events
    onToggle,
    // Standard
    className,
    style,
    id: providedId,
    "data-testid": dataTestId,
    ref
  } = props;
  const panelRef = React2.useRef(null);
  const headerRef = React2.useRef(null);
  const contentWrapperRef = React2.useRef(null);
  const contentRef = React2.useRef(null);
  const generatedId = React2.useId();
  const panelId = providedId || generatedId;
  const headerId = `${panelId}-header`;
  const contentId = `${panelId}-content`;
  const [internalCollapsed, setInternalCollapsed] = React2.useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = fixed ? false : isControlled ? controlledCollapsed : internalCollapsed;
  const animationRef = React2.useRef(null);
  const prevCollapsedRef = React2.useRef(isCollapsed);
  const [contentExpanded, setContentExpanded] = React2.useState(!isCollapsed);
  React2.useImperativeHandle(ref, () => ({
    focus() {
      headerRef.current?.focus();
    },
    blur() {
      headerRef.current?.blur();
    },
    isFocused() {
      return document.activeElement === headerRef.current;
    },
    get nativeElement() {
      return panelRef.current;
    }
  }), []);
  const handleToggle = React2.useCallback(() => {
    if (fixed) return;
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.({ collapsed: newCollapsed });
  }, [fixed, isCollapsed, isControlled, onToggle]);
  const pendingToggleRef = React2.useRef(false);
  const handleKeyDown = React2.useCallback((e) => {
    if (fixed) return;
    if (e.key === "Enter") {
      e.preventDefault();
      handleToggle();
    }
    if (e.key === " ") {
      e.preventDefault();
      pendingToggleRef.current = true;
    }
    if (e.key === "Escape" && pendingToggleRef.current) {
      e.preventDefault();
      pendingToggleRef.current = false;
    }
  }, [fixed, handleToggle]);
  const handleKeyUp = React2.useCallback((e) => {
    if (fixed) return;
    if (e.key === " ") {
      if (pendingToggleRef.current) {
        handleToggle();
      }
      pendingToggleRef.current = false;
    }
  }, [fixed, handleToggle]);
  React2.useEffect(() => {
    const wrapper = contentWrapperRef.current;
    if (!wrapper) {
      prevCollapsedRef.current = isCollapsed;
      return;
    }
    const wasCollapsed = prevCollapsedRef.current;
    prevCollapsedRef.current = isCollapsed;
    if (wasCollapsed === isCollapsed) return;
    if (noAnimation || fixed) {
      if (isCollapsed) {
        wrapper.style.display = "none";
        setContentExpanded(false);
      } else {
        wrapper.style.display = "";
        setContentExpanded(true);
      }
      return;
    }
    if (animationRef.current !== null) {
      cancelAnimationFrame(animationRef.current);
      animationRef.current = null;
    }
    const DURATION = 400;
    if (isCollapsed) {
      const computed = getComputedStyle(wrapper);
      const startHeight = parseFloat(computed.height);
      const startPaddingTop = parseFloat(computed.paddingTop);
      const startPaddingBottom = parseFloat(computed.paddingBottom);
      wrapper.style.overflow = "hidden";
      let start = null;
      const step = (timestamp) => {
        if (start === null) start = timestamp;
        const elapsed = timestamp - start;
        const progress = Math.min(elapsed / DURATION, 1);
        wrapper.style.height = `${startHeight * (1 - progress)}px`;
        wrapper.style.paddingTop = `${startPaddingTop * (1 - progress)}px`;
        wrapper.style.paddingBottom = `${startPaddingBottom * (1 - progress)}px`;
        if (progress < 1) {
          animationRef.current = requestAnimationFrame(step);
        } else {
          wrapper.style.display = "none";
          wrapper.style.overflow = "";
          wrapper.style.height = "";
          wrapper.style.paddingTop = "";
          wrapper.style.paddingBottom = "";
          animationRef.current = null;
          setContentExpanded(false);
        }
      };
      animationRef.current = requestAnimationFrame(step);
    } else {
      wrapper.style.display = "block";
      setContentExpanded(true);
      const computed = getComputedStyle(wrapper);
      const targetHeight = parseFloat(computed.height);
      const targetPaddingTop = parseFloat(computed.paddingTop);
      const targetPaddingBottom = parseFloat(computed.paddingBottom);
      wrapper.style.overflow = "hidden";
      wrapper.style.height = "0px";
      wrapper.style.paddingTop = "0px";
      wrapper.style.paddingBottom = "0px";
      let start = null;
      const step = (timestamp) => {
        if (start === null) start = timestamp;
        const elapsed = timestamp - start;
        const progress = Math.min(elapsed / DURATION, 1);
        wrapper.style.display = "block";
        wrapper.style.height = `${targetHeight * progress}px`;
        wrapper.style.paddingTop = `${targetPaddingTop * progress}px`;
        wrapper.style.paddingBottom = `${targetPaddingBottom * progress}px`;
        if (progress < 1) {
          animationRef.current = requestAnimationFrame(step);
        } else {
          wrapper.style.overflow = "";
          wrapper.style.height = "";
          wrapper.style.paddingTop = "";
          wrapper.style.paddingBottom = "";
          animationRef.current = null;
        }
      };
      animationRef.current = requestAnimationFrame(step);
    }
    return () => {
      if (animationRef.current !== null) {
        cancelAnimationFrame(animationRef.current);
        animationRef.current = null;
      }
    };
  }, [isCollapsed, noAnimation, fixed]);
  const resolvedDesign = design in rootDesignClasses ? design : "Primary" /* Primary */;
  const ariaRole = roleMap[accessibleRole] || "form";
  const HeadingTag = levelToTag[headerLevel] || "h2";
  const hasHeader = headerText || header;
  const isInteractive = !fixed;
  const isChild = resolvedDesign === "Child" /* Child */;
  const toggleIcon = isChild ? /* @__PURE__ */ jsxRuntime.jsx(
    NavigationRightArrowIcon,
    {
      className: cn(
        "h-4 w-4 shrink-0 text-sapphire-text-tertiary",
        !noAnimation && "transition-transform duration-200",
        !isCollapsed && "rotate-90"
      ),
      "aria-hidden": "true"
    }
  ) : isCollapsed ? /* @__PURE__ */ jsxRuntime.jsx(ExpandIcon, { className: "h-4 w-4 shrink-0 text-sapphire-text-tertiary", "aria-hidden": "true" }) : /* @__PURE__ */ jsxRuntime.jsx(CollapseIcon, { className: "h-4 w-4 shrink-0 text-sapphire-text-tertiary", "aria-hidden": "true" });
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: panelRef,
      id: panelId,
      role: ariaRole,
      "aria-label": accessibleName,
      "aria-labelledby": !accessibleName && hasHeader ? headerId : void 0,
      className: cn(
        "rounded-lg overflow-clip text-sapphire-text-primary",
        rootDesignClasses[resolvedDesign],
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        (hasHeader || isInteractive) && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: headerRef,
            id: headerId,
            role: isInteractive ? "button" : void 0,
            tabIndex: isInteractive ? 0 : void 0,
            "aria-expanded": isInteractive ? !isCollapsed : void 0,
            "aria-controls": isInteractive ? contentId : void 0,
            onClick: isInteractive ? handleToggle : void 0,
            onKeyDown: isInteractive ? handleKeyDown : void 0,
            onKeyUp: isInteractive ? handleKeyUp : void 0,
            className: cn(
              "flex items-center min-h-11",
              headerDesignClasses[resolvedDesign],
              !stickyHeader && headerBgClasses[resolvedDesign],
              stickyHeader && cn("sticky top-0 z-10", headerBgClasses[resolvedDesign]),
              isInteractive && "cursor-pointer hover:bg-accent transition-colors",
              isInteractive && "focus:outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
              "select-none",
              isChild ? "rounded-lg" : "rounded-t-lg",
              isCollapsed && !isChild && "rounded-b-lg"
            ),
            "data-part": "header",
            children: [
              isInteractive && toggleIcon,
              header ? /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex-1 min-w-0", "data-part": "header-content", children: header }) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
                (headerText || subtitle) && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex-1 min-w-0", children: [
                  headerText && /* @__PURE__ */ jsxRuntime.jsx(
                    HeadingTag,
                    {
                      className: cn(
                        "text-sapphire-text-primary truncate m-0",
                        headerTitleClasses[resolvedDesign]
                      ),
                      "data-part": "header-text",
                      children: headerText
                    }
                  ),
                  subtitle && /* @__PURE__ */ jsxRuntime.jsx(
                    "p",
                    {
                      className: "text-sm text-sapphire-text-tertiary truncate m-0",
                      "data-part": "header-subtitle",
                      children: subtitle
                    }
                  )
                ] }),
                endSlot && /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    className: "flex-shrink-0",
                    "data-part": "header-end-slot",
                    onClick: (e) => e.stopPropagation(),
                    onKeyDown: (e) => e.stopPropagation(),
                    onKeyUp: (e) => e.stopPropagation(),
                    children: endSlot
                  }
                )
              ] })
            ]
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: contentWrapperRef,
            id: contentId,
            role: "region",
            "aria-labelledby": hasHeader ? headerId : void 0,
            style: !contentExpanded ? { display: "none" } : void 0,
            "data-part": "content",
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(
                "div",
                {
                  ref: contentRef,
                  className: cn(
                    noPadding ? "" : contentDesignClasses[resolvedDesign]
                  ),
                  children
                }
              ),
              footer && !isChild && /* @__PURE__ */ jsxRuntime.jsx(
                "div",
                {
                  className: cn(
                    "flex items-center gap-2",
                    footerDesignClasses[resolvedDesign]
                  ),
                  "data-part": "footer",
                  children: footer
                }
              )
            ]
          }
        )
      ]
    }
  );
}

exports.Panel = Panel;
