// @sap-ui/fx-components/dist/components/sidenav/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/sidenav/SideNavigation.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData3,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData3 || pathData3.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData3.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M410.53 200.75C418.95 210.98 417.48 226.11 407.25 234.53L271.34 346.53C262.48 353.83 249.68 353.83 240.81 346.53L104.75 234.53C94.52 226.11 93.05 210.98 101.47 200.75C109.89 190.52 125.02 189.05 135.25 197.47L256.06 296.91L376.75 197.47C386.98 189.05 402.11 190.52 410.53 200.75Z"
];
var SlimArrowDownIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "SlimArrowDownIcon", iconName: "slim-arrow-down" }
);
var pathData2 = [
  "M186 416q-11 0-18.5-7.5T160 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T186 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L203 409q-7 7-17 7z"
];
var SlimArrowRightIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "SlimArrowRightIcon", iconName: "slim-arrow-right" }
);
var SideNavigationContext = React2.createContext({
  collapsed: false,
  onItemSelect: () => {
  }
});
function SideNavigation({
  collapsed = false,
  selectedKey,
  children,
  fixedItems,
  header,
  onSelectionChange,
  className,
  style,
  ref
}) {
  const navRef = React2.useRef(null);
  React2.useImperativeHandle(ref, () => ({
    get nativeElement() {
      return navRef.current;
    },
    getNativeElement: () => navRef.current
  }));
  const handleItemSelect = React2.useCallback(
    (key, item) => {
      onSelectionChange?.({ item: { ...item, itemKey: key } });
    },
    [onSelectionChange]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(
    SideNavigationContext.Provider,
    {
      value: {
        collapsed,
        selectedKey,
        onItemSelect: handleItemSelect
      },
      children: /* @__PURE__ */ jsxRuntime.jsxs(
        "nav",
        {
          ref: navRef,
          role: "navigation",
          "aria-label": "Side navigation",
          className: cn(
            "flex flex-col h-full bg-card",
            collapsed ? "w-14" : "w-60",
            "transition-[width] duration-200",
            className
          ),
          style,
          children: [
            header && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex-shrink-0 p-3 border-b border-border", children: header }),
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex-1 overflow-y-auto py-2", children: /* @__PURE__ */ jsxRuntime.jsx("ul", { role: "tree", className: "space-y-1 px-2", children }) }),
            fixedItems && /* @__PURE__ */ jsxRuntime.jsx("div", { className: "flex-shrink-0 border-t border-border py-2", children: /* @__PURE__ */ jsxRuntime.jsx("ul", { className: "space-y-1 px-2", children: fixedItems }) })
          ]
        }
      )
    }
  );
}
function SideNavigationItem({
  itemKey,
  text,
  icon,
  expanded: controlledExpanded,
  selected: controlledSelected,
  disabled = false,
  badge,
  href,
  target,
  children,
  onClick,
  className,
  wholeItemToggleable = false,
  ref
}) {
  const { collapsed, selectedKey, onItemSelect } = React2.useContext(SideNavigationContext);
  const [internalExpanded, setInternalExpanded] = React2.useState(false);
  const hasChildren = React2__default.default.Children.count(children) > 0;
  const isExpanded = controlledExpanded ?? internalExpanded;
  const isSelected = controlledSelected ?? (itemKey ? selectedKey === itemKey : false);
  const handleClick = (e) => {
    if (disabled) return;
    if (hasChildren && !collapsed) {
      setInternalExpanded(!isExpanded);
      if (!wholeItemToggleable) {
        return;
      }
    }
    if (itemKey) {
      onItemSelect(itemKey, {
        itemKey,
        text,
        icon,
        expanded: isExpanded,
        selected: true,
        disabled,
        badge,
        href,
        target
      });
    }
    onClick?.(e);
  };
  const handleKeyDown = (e) => {
    if (disabled) return;
    if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      handleClick(e);
    }
    if (hasChildren) {
      if (e.key === "ArrowRight" && !isExpanded) {
        setInternalExpanded(true);
      } else if (e.key === "ArrowLeft" && isExpanded) {
        setInternalExpanded(false);
      }
    }
  };
  const content = /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "h-5 w-5 shrink-0 flex items-center justify-center", children: icon }),
    !collapsed && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-1 truncate", children: text }),
      badge && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "ml-auto", children: badge }),
      hasChildren && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "ml-1", children: isExpanded ? /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "h-4 w-4" }) })
    ] })
  ] });
  const itemClasses = cn(
    "flex items-center gap-3 rounded-lg px-3 py-2 text-sm transition-colors",
    "hover:bg-accent hover:text-accent-foreground",
    "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
    isSelected && "bg-primary/10 text-primary font-medium",
    disabled && "pointer-events-none opacity-50",
    collapsed && "justify-center px-0",
    className
  );
  return /* @__PURE__ */ jsxRuntime.jsxs("li", { ref, role: "treeitem", "aria-selected": isSelected, "aria-expanded": hasChildren ? isExpanded : void 0, children: [
    href ? /* @__PURE__ */ jsxRuntime.jsx(
      "a",
      {
        href,
        target,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        className: itemClasses,
        tabIndex: disabled ? -1 : 0,
        children: content
      }
    ) : /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        role: "button",
        tabIndex: disabled ? -1 : 0,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        className: cn(itemClasses, "cursor-pointer"),
        children: content
      }
    ),
    hasChildren && isExpanded && !collapsed && /* @__PURE__ */ jsxRuntime.jsx("ul", { role: "group", className: "mt-1 ml-4 space-y-1", children })
  ] });
}
function SideNavigationSubItem({
  itemKey,
  text,
  icon,
  selected: controlledSelected,
  disabled = false,
  href,
  target,
  onClick,
  className,
  ref
}) {
  const { selectedKey, onItemSelect } = React2.useContext(SideNavigationContext);
  const isSelected = controlledSelected ?? (itemKey ? selectedKey === itemKey : false);
  const handleClick = (e) => {
    if (disabled) return;
    if (itemKey) {
      onItemSelect(itemKey, {
        itemKey,
        text,
        icon,
        selected: true,
        disabled,
        href,
        target
      });
    }
    onClick?.(e);
  };
  const handleKeyDown = (e) => {
    if (disabled) return;
    if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      handleClick(e);
    }
  };
  const itemClasses = cn(
    "flex items-center gap-2 rounded-md px-3 py-1.5 text-sm transition-colors",
    "hover:bg-accent hover:text-accent-foreground",
    "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
    isSelected && "bg-primary/10 text-primary font-medium",
    disabled && "pointer-events-none opacity-50",
    className
  );
  const content = /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    icon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "h-4 w-4 shrink-0", children: icon }),
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "truncate", children: text })
  ] });
  return /* @__PURE__ */ jsxRuntime.jsx("li", { ref, role: "treeitem", "aria-selected": isSelected, children: href ? /* @__PURE__ */ jsxRuntime.jsx(
    "a",
    {
      href,
      target,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: itemClasses,
      tabIndex: disabled ? -1 : 0,
      children: content
    }
  ) : /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      role: "button",
      tabIndex: disabled ? -1 : 0,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(itemClasses, "cursor-pointer"),
      children: content
    }
  ) });
}
function SideNavigationGroup({ text, expanded: controlledExpanded = true, children, className, ref }) {
  const [internalExpanded, setInternalExpanded] = React2.useState(true);
  const { collapsed } = React2.useContext(SideNavigationContext);
  const isExpanded = controlledExpanded ?? internalExpanded;
  if (collapsed) {
    return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children });
  }
  return /* @__PURE__ */ jsxRuntime.jsxs("li", { ref, className: cn("pt-4 first:pt-0", className), children: [
    text && /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        type: "button",
        onClick: () => setInternalExpanded(!isExpanded),
        className: "flex items-center gap-2 px-3 py-1 text-xs font-semibold text-sapphire-text-tertiary uppercase tracking-wider w-full hover:text-foreground",
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", { className: "flex-1 text-left", children: text }),
          isExpanded ? /* @__PURE__ */ jsxRuntime.jsx(SlimArrowDownIcon, { className: "h-3 w-3" }) : /* @__PURE__ */ jsxRuntime.jsx(SlimArrowRightIcon, { className: "h-3 w-3" })
        ]
      }
    ),
    isExpanded && /* @__PURE__ */ jsxRuntime.jsx("ul", { className: "mt-1 space-y-1", children })
  ] });
}

exports.SideNavigation = SideNavigation;
exports.SideNavigationGroup = SideNavigationGroup;
exports.SideNavigationItem = SideNavigationItem;
exports.SideNavigationSubItem = SideNavigationSubItem;
