// @sap-ui/fx-components/dist/chunk-CTYK4HEK.js
import { getTabbableElements, getActiveElement } from './chunk-O5ILLEDP.js';
import { RadioButton } from './chunk-2MAGJSMZ.js';
import { CheckBox } from './chunk-TNSICZD6.js';
import { useOptionalListContext } from './chunk-CJXZUSUP.js';
import { DeleteIcon } from './chunk-MJFYIHUV.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import * as React from 'react';
import { useCallback, useState, useRef, useEffect, useId, useMemo, useImperativeHandle } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

var isInteractiveElement = (target, container) => {
  let el = target;
  while (el && el !== container) {
    const tag = el.tagName.toLowerCase();
    if (tag === "input" || tag === "textarea" || tag === "select" || tag === "button" || tag === "a" && el.hasAttribute("href") || el.getAttribute("role") === "button" || el.isContentEditable) {
      return true;
    }
    el = el.parentElement;
  }
  return false;
};
function useListItemSelection(options) {
  const { id, disabled, type, selectionMode, onClick, itemRef } = options;
  const listContext = useOptionalListContext();
  const isSelected = listContext ? listContext.isSelected(id) : false;
  const shouldForwardTabAfter = useCallback(() => {
    if (!itemRef?.current) return true;
    const tabbables = getTabbableElements(itemRef.current);
    if (tabbables.length === 0) return true;
    const activeElement = getActiveElement();
    const lastTabbable = tabbables[tabbables.length - 1];
    return activeElement === lastTabbable;
  }, [itemRef]);
  const shouldForwardTabBefore = useCallback((target) => {
    if (!itemRef?.current) return false;
    if (target === itemRef.current) return true;
    const tabbables = getTabbableElements(itemRef.current);
    if (tabbables.length === 0) return false;
    return target === tabbables[0];
  }, [itemRef]);
  const handleClick = useCallback((e) => {
    if (disabled) return;
    const container = itemRef?.current ?? null;
    const clickTarget = e.target;
    if (clickTarget !== container && isInteractiveElement(clickTarget, container)) {
      return;
    }
    const typeStr = String(type);
    if (typeStr === "Inactive" || typeStr === "Detail") {
      return;
    }
    const modeStr = String(selectionMode);
    if (modeStr !== "None" && modeStr !== "Delete") {
      listContext?.onToggleSelect(id);
    }
    onClick?.(e);
    listContext?.onItemClick?.(id, e);
  }, [disabled, type, selectionMode, listContext, id, onClick, itemRef]);
  const handleKeyDown = useCallback((e) => {
    if (disabled) return;
    const modeStr = String(selectionMode);
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      const target = e.target;
      const tabbables = getTabbableElements(itemRef?.current || null);
      if (e.shiftKey) {
        if (shouldForwardTabBefore(target)) {
          e.preventDefault();
          listContext?.onForwardTabBefore?.(id, e);
        }
      } else {
        if (target === itemRef?.current && tabbables.length > 0) {
          e.preventDefault();
          tabbables[0].focus();
          return;
        }
        if (shouldForwardTabAfter()) {
          e.preventDefault();
          listContext?.onForwardTabAfter?.(id, e);
          return;
        }
      }
      return;
    }
    if (e.key === "Enter" || e.key === " ") {
      const keyTarget = e.target;
      const keyTag = keyTarget.tagName.toLowerCase();
      if (keyTag === "input" || keyTag === "textarea" || keyTag === "select" || keyTarget.isContentEditable || keyTarget.getAttribute("role") === "button" || keyTag === "a" && keyTarget.hasAttribute("href")) {
        return;
      }
      e.preventDefault();
      const typeStr = String(type);
      if (typeStr === "Inactive" || typeStr === "Detail") {
        return;
      }
      if (modeStr !== "None" && modeStr !== "Delete") {
        listContext?.onToggleSelect(id);
      }
      onClick?.(e);
      listContext?.onItemClick?.(id, e);
    }
  }, [disabled, type, selectionMode, listContext, id, onClick, shouldForwardTabAfter, shouldForwardTabBefore, itemRef]);
  const handleSelectionChange = useCallback((checked) => {
    if (disabled) return;
    listContext?.onSelect(id, checked);
  }, [disabled, listContext, id]);
  return {
    isSelected,
    handleClick,
    handleKeyDown,
    handleSelectionChange
  };
}
function useListItemFocus(options) {
  const { itemRef, disabled, selectionMode, accessibleRole, controlledSelected } = options;
  const listContext = useOptionalListContext();
  const [isFocused, setIsFocused] = useState(false);
  const [isFocusedWithin, setIsFocusedWithin] = useState(false);
  const [isActive, setIsActive] = useState(false);
  const getItemIndex = useCallback(() => {
    let rootList = itemRef.current?.parentElement;
    while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
      rootList = rootList.parentElement;
    }
    if (rootList && itemRef.current) {
      const allNavigable = Array.from(
        rootList.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
        )
      );
      return allNavigable.indexOf(itemRef.current);
    }
    return -1;
  }, [itemRef]);
  const handleFocus = useCallback((e) => {
    const isItemItself = e.target === itemRef.current;
    setIsFocused(isItemItself);
    setIsFocusedWithin(true);
    if (listContext) {
      const itemIndex2 = getItemIndex();
      if (itemIndex2 >= 0) {
        listContext.onFocusItem(itemIndex2);
      }
    }
  }, [listContext, getItemIndex, itemRef]);
  const handleBlur = useCallback((e) => {
    const isLeavingItem = !itemRef.current?.contains(e.relatedTarget);
    if (isLeavingItem) {
      setIsFocused(false);
      setIsFocusedWithin(false);
    } else {
      setIsFocused(false);
    }
  }, [itemRef]);
  const handleMouseDown = useCallback(() => {
    setIsActive(true);
  }, []);
  const handleMouseUp = useCallback(() => {
    setIsActive(false);
  }, []);
  const handleMouseDownCapture = useCallback(() => {
    setIsFocusedWithin(true);
  }, []);
  const focus = useCallback(() => {
    itemRef.current?.focus();
  }, [itemRef]);
  const blur = useCallback(() => {
    itemRef.current?.blur();
  }, [itemRef]);
  const modeStr = String(selectionMode);
  const isSelected = controlledSelected ?? (listContext?.isSelected(itemRef.current?.getAttribute("data-item-key") || "") || false);
  const roleStr = String(accessibleRole);
  const supportsAriaSelected = listContext?.supportsAriaSelected ?? false;
  const hasSelection = modeStr !== "None" && modeStr !== "Delete";
  const ariaSelected = roleStr === "Option" ? isSelected : supportsAriaSelected && hasSelection ? isSelected : void 0;
  const itemIndex = getItemIndex();
  const isFocusedItem = listContext?.focusedIndex === itemIndex;
  const isFirstItem = itemIndex === 0;
  const isTabbable = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem;
  const tabbableIndex = isTabbable ? 0 : -1;
  const tabIndex = disabled ? -1 : tabbableIndex;
  return {
    isFocused,
    isFocusedWithin,
    isActive,
    tabIndex,
    ariaSelected,
    handleFocus,
    handleBlur,
    handleMouseDown,
    handleMouseUp,
    handleMouseDownCapture,
    focus,
    blur,
    getItemIndex
  };
}
function useListItemDragDrop(options) {
  const { id, movable, disabled, onDragStart: onDragStartProp } = options;
  const listContext = useOptionalListContext();
  const isDragging = listContext?.draggedKey === id;
  const isDropTarget = listContext?.dropTargetKey === id;
  const dropPlacement = isDropTarget ? listContext?.dropPlacement : null;
  const handleDragStart = useCallback((e) => {
    if (!movable || disabled) {
      e.preventDefault();
      return;
    }
    e.dataTransfer.effectAllowed = "move";
    e.dataTransfer.dropEffect = "move";
    e.dataTransfer.setData("text/plain", id);
    listContext?.onDragStart?.(id, e);
    onDragStartProp?.();
  }, [movable, disabled, id, listContext, onDragStartProp]);
  const handleDragEnd = useCallback((e) => {
    listContext?.onDragEnd?.(id, e);
  }, [id, listContext]);
  const handleDragOver = useCallback((e) => {
    e.preventDefault();
    e.dataTransfer.dropEffect = "move";
    listContext?.onDragOver?.(id, e);
  }, [id, listContext]);
  const handleDrop = useCallback((e) => {
    e.preventDefault();
    listContext?.onDrop?.(id, e);
  }, [id, listContext]);
  const handleDragEnter = useCallback((e) => {
    e.preventDefault();
  }, []);
  const handleDragLeave = useCallback((_e) => {
  }, []);
  return {
    isDragging,
    isDropTarget,
    dropPlacement: dropPlacement || null,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDragEnter,
    handleDragLeave,
    handleDrop
  };
}
function useListItemRegistration(options) {
  const { itemRef, itemKey } = options;
  const listContext = useOptionalListContext();
  const itemIndexRef = useRef(-1);
  useEffect(() => {
    if (listContext && itemKey) {
      let rootList = itemRef.current?.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(itemRef.current);
        if (index >= 0) {
          itemIndexRef.current = index;
          listContext.registerItem(itemKey, index);
          listContext.registerNavigationItem(index, itemRef.current);
        }
      }
    }
    return () => {
      if (listContext && itemKey && itemIndexRef.current >= 0) {
        listContext.unregisterItem(itemKey);
        listContext.registerNavigationItem(itemIndexRef.current, null);
        itemIndexRef.current = -1;
      }
    };
  }, [listContext, itemKey, itemRef]);
  return {};
}
function useListItem(options = {}) {
  const {
    itemKey,
    disabled = false,
    movable = false,
    type = "Active",
    accessibleRole = "ListItem",
    selected: controlledSelected,
    onClick,
    onDragStart
  } = options;
  const itemRef = useRef(null);
  const generatedId = useId();
  const id = itemKey ?? generatedId;
  const listContext = useOptionalListContext();
  const selectionMode = listContext?.selectionMode ?? "None" /* None */;
  useListItemRegistration({ itemRef, itemKey });
  const selection = useListItemSelection({
    id,
    disabled,
    type,
    selectionMode,
    onClick,
    itemRef
  });
  const focus = useListItemFocus({
    itemRef,
    disabled,
    selectionMode,
    accessibleRole,
    controlledSelected
  });
  const handleMouseDown = useCallback(() => {
    const typeStr = String(type);
    if (typeStr === "Active" || typeStr === "Navigation") {
      focus.handleMouseDown();
    }
  }, [type, focus]);
  const handleMouseUp = useCallback(() => {
    focus.handleMouseUp();
  }, [focus]);
  const dragDrop = useListItemDragDrop({
    id,
    movable,
    disabled,
    onDragStart
  });
  return {
    // Refs
    itemRef,
    id,
    // State from sub-hooks (controlledSelected overrides context-based selection)
    isSelected: controlledSelected ?? selection.isSelected,
    isFocused: focus.isFocused,
    isFocusedWithin: focus.isFocusedWithin,
    isActive: focus.isActive,
    isDragging: dragDrop.isDragging,
    isDropTarget: dragDrop.isDropTarget,
    dropPlacement: dragDrop.dropPlacement,
    // Context values
    selectionMode,
    // Computed values
    tabIndex: focus.tabIndex,
    ariaSelected: focus.ariaSelected,
    // Event handlers (composed from sub-hooks)
    handleClick: selection.handleClick,
    handleKeyDown: selection.handleKeyDown,
    handleMouseDown,
    handleMouseUp,
    handleMouseDownCapture: focus.handleMouseDownCapture,
    handleFocus: focus.handleFocus,
    handleBlur: focus.handleBlur,
    handleDragStart: dragDrop.handleDragStart,
    handleDragEnd: dragDrop.handleDragEnd,
    handleDragOver: dragDrop.handleDragOver,
    handleDragEnter: dragDrop.handleDragEnter,
    handleDragLeave: dragDrop.handleDragLeave,
    handleDrop: dragDrop.handleDrop,
    // Selection handlers
    handleSelectionChange: selection.handleSelectionChange,
    // Helper functions
    focus: focus.focus,
    blur: focus.blur,
    getItemIndex: focus.getItemIndex
  };
}
function ListSelectionControl({
  selectionMode,
  position,
  isSelected,
  disabled = false,
  itemText = "item",
  deleteButton,
  onSelectionChange,
  onDeleteClick,
  ref
}) {
  const modeStr = String(selectionMode);
  const showSelectionBefore = position === "before" && (modeStr === "Multiple" || modeStr === "SingleStart");
  const showSelectionAfter = position === "after" && (modeStr === "SingleEnd" || modeStr === "Delete");
  const showRadio = modeStr === "SingleStart" || modeStr === "SingleEnd" || modeStr === "Single";
  const showCheckbox = modeStr === "Multiple";
  const showDelete = modeStr === "Delete";
  if (!showSelectionBefore && !showSelectionAfter) {
    return null;
  }
  const handleDeleteClick = (detail) => {
    detail.originalEvent.stopPropagation();
    onDeleteClick?.();
  };
  return /* @__PURE__ */ jsxs(
    "span",
    {
      ref,
      className: "shrink-0 pl-3 pr-1 flex items-center",
      onClick: (e) => e.stopPropagation(),
      children: [
        showCheckbox && /* @__PURE__ */ jsx(
          CheckBox,
          {
            checked: isSelected,
            onChange: (e) => onSelectionChange(e.checked),
            disabled,
            accessibleName: `Select ${itemText}`,
            tabIndex: -1
          }
        ),
        showRadio && !showDelete && /* @__PURE__ */ jsx(
          RadioButton,
          {
            checked: isSelected,
            onChange: (e) => onSelectionChange(e.checked),
            disabled,
            accessibleName: `Select ${itemText}`,
            tabIndex: -1
          }
        ),
        showDelete && (deleteButton ? React.cloneElement(deleteButton, {
          onClick: handleDeleteClick
        }) : /* @__PURE__ */ jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsx(DeleteIcon, { className: "h-4 w-4" }),
            onClick: handleDeleteClick,
            accessibleName: "Delete item",
            tooltip: "Delete",
            tabIndex: -1
          }
        ))
      ]
    }
  );
}
function DropIndicator({ visible, placement, orientation = "vertical", className, ref }) {
  if (!visible) return null;
  const isVertical = orientation === "vertical";
  const isBefore = placement === "Before";
  const isOn = placement === "On";
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn(
        "absolute pointer-events-none z-20",
        isVertical ? cn(
          "left-0 right-0 h-0.5 bg-primary",
          isBefore ? "-top-px" : isOn ? "inset-0 h-full bg-primary/10 border-2 border-primary border-dashed" : "-bottom-px"
        ) : cn(
          "top-0 bottom-0 w-0.5 bg-primary",
          isBefore ? "-left-px" : isOn ? "inset-0 w-full bg-primary/10 border-2 border-primary border-dashed" : "-right-px"
        ),
        className
      ),
      "aria-hidden": "true",
      "data-drop-indicator": true,
      "data-placement": placement
    }
  );
}

// src/components/list/shared-styles.ts
var LIST_ITEM_BASE_CLASSES = [
  "relative",
  "flex",
  "items-center",
  "w-full",
  "min-h-[2.75rem]",
  "transition-colors",
  "outline-none"
];
var LIST_ITEM_FOCUS_CLASSES = "ring-2 ring-inset ring-sapphire-border-focus z-10";
function ListItemBase(props) {
  const {
    ref,
    itemKey,
    type = "Active" /* Active */,
    disabled = false,
    movable = false,
    accessibleRole = "ListItem" /* ListItem */,
    accessibleNameRef,
    selected: controlledSelected,
    focused: externalFocused,
    onClick,
    onDragStart,
    onDelete,
    deleteButton,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const listContext = useOptionalListContext();
  const effectiveAccessibleRole = useMemo(() => {
    if (accessibleRole !== void 0 && accessibleRole !== "ListItem" /* ListItem */) {
      return accessibleRole;
    }
    const parentRole = String(listContext?.accessibleRole || "List");
    if (accessibleRole === "ListItem" /* ListItem */ && parentRole === "List") {
      return "ListItem" /* ListItem */;
    }
    switch (parentRole) {
      case "ListBox":
        return "Option" /* Option */;
      case "Menu":
        return "MenuItem" /* MenuItem */;
      case "Tree":
        return "TreeItem" /* TreeItem */;
      default:
        return "ListItem" /* ListItem */;
    }
  }, [accessibleRole, listContext?.accessibleRole]);
  const listItem = useListItem({
    itemKey,
    disabled,
    movable,
    type,
    accessibleRole: effectiveAccessibleRole,
    selected: controlledSelected,
    onClick,
    onDragStart
  });
  const handleDeleteClick = useCallback(() => {
    if (itemKey) {
      listContext?.onItemDelete?.(itemKey);
    }
    onDelete?.();
  }, [itemKey, listContext, onDelete]);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        listItem.focus();
      },
      blur() {
        listItem.blur();
      },
      isFocused() {
        return document.activeElement === listItem.itemRef.current;
      },
      get nativeElement() {
        return listItem.itemRef.current;
      }
    }),
    [listItem]
  );
  const renderState = {
    isSelected: listItem.isSelected,
    isFocused: listItem.isFocused,
    isFocusedWithin: listItem.isFocusedWithin,
    isActive: listItem.isActive,
    isDragging: listItem.isDragging,
    isDropTarget: listItem.isDropTarget,
    dropPlacement: listItem.dropPlacement,
    selectionMode: String(listItem.selectionMode)
  };
  const modeStr = String(listItem.selectionMode);
  const position = modeStr === "SingleEnd" || modeStr === "Delete" ? "after" : "before";
  const shouldShowControl = modeStr !== "None";
  const content = typeof children === "function" ? children(renderState) : children;
  const resolvedRole = (() => {
    const roleStr = String(effectiveAccessibleRole);
    switch (roleStr) {
      case "Option":
        return "option";
      case "MenuItem":
        return "menuitem";
      case "TreeItem":
        return "treeitem";
      case "Group":
        return "group";
      case "None":
        return void 0;
      default:
        return "listitem";
    }
  })();
  return /* @__PURE__ */ jsxs(
    "li",
    {
      ref: listItem.itemRef,
      id: listItem.id,
      role: resolvedRole,
      "aria-labelledby": accessibleNameRef,
      "aria-selected": listItem.ariaSelected,
      "aria-disabled": disabled || void 0,
      "aria-setsize": listContext?.totalItemCount,
      "aria-posinset": itemKey && listContext ? listContext.getItemIndex(itemKey) + 1 || void 0 : void 0,
      tabIndex: listItem.tabIndex,
      draggable: movable,
      onClick: listItem.handleClick,
      onKeyDown: listItem.handleKeyDown,
      onMouseDownCapture: listItem.handleMouseDownCapture,
      onMouseDown: listItem.handleMouseDown,
      onMouseUp: listItem.handleMouseUp,
      onMouseLeave: listItem.handleMouseUp,
      onFocus: listItem.handleFocus,
      onBlur: listItem.handleBlur,
      onDragStart: listItem.handleDragStart,
      onDragEnd: listItem.handleDragEnd,
      onDragOver: listItem.handleDragOver,
      onDragEnter: listItem.handleDragEnter,
      onDragLeave: listItem.handleDragLeave,
      onDrop: listItem.handleDrop,
      className: cn(
        // Base item styles - container layout (shared with group headers)
        LIST_ITEM_BASE_CLASSES,
        "text-sm text-sapphire-text-primary",
        // State classes
        listItem.isSelected && "!bg-sapphire-brand-selected-background",
        listItem.isSelected && !(externalFocused || listItem.isFocused) && "!border-sapphire-border-accent",
        (externalFocused || listItem.isFocused) && LIST_ITEM_FOCUS_CLASSES,
        listItem.isActive && "bg-muted/80",
        listItem.isDragging && "opacity-50",
        movable && "cursor-grab",
        listItem.isDragging && "cursor-grabbing",
        disabled && "opacity-50 pointer-events-none cursor-not-allowed",
        // Interactive states
        (type === "Active" /* Active */ || type === "Navigation" /* Navigation */) && "cursor-pointer hover:bg-[var(--table-row-hover)] active:bg-[var(--neutral-pressed-background-2)]",
        type === "Inactive" /* Inactive */ && "cursor-default",
        // User's custom classes
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-notification-item": props["data-notification-item"] || void 0,
      "data-notification-group": props["data-notification-group"] || void 0,
      "data-item-key": itemKey,
      "data-dragging": listItem.isDragging || void 0,
      children: [
        listItem.dropPlacement && /* @__PURE__ */ jsx(DropIndicator, { visible: true, placement: listItem.dropPlacement }),
        shouldShowControl && position === "before" && /* @__PURE__ */ jsx(
          ListSelectionControl,
          {
            selectionMode: listItem.selectionMode,
            position: "before",
            isSelected: listItem.isSelected,
            disabled,
            deleteButton,
            onSelectionChange: listItem.handleSelectionChange,
            onDeleteClick: handleDeleteClick
          }
        ),
        content,
        shouldShowControl && position === "after" && /* @__PURE__ */ jsx(
          ListSelectionControl,
          {
            selectionMode: listItem.selectionMode,
            position: "after",
            isSelected: listItem.isSelected,
            disabled,
            deleteButton,
            onSelectionChange: listItem.handleSelectionChange,
            onDeleteClick: handleDeleteClick
          }
        )
      ]
    }
  );
}
ListItemBase.displayName = "ListItemBase";

export { DropIndicator, LIST_ITEM_BASE_CLASSES, LIST_ITEM_FOCUS_CLASSES, ListItemBase, ListSelectionControl, useListItem, useListItemDragDrop, useListItemFocus, useListItemRegistration, useListItemSelection };
