// @sap-ui/fx-components/dist/chunk-TQOJRVTN.js
import { createIllustration } from './chunk-UHU7LLWX.js';

// src/illustrations/tnt/Avatar/Avatar.tsx
var Avatar = createIllustration({
  name: "TntAvatar",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2MDPTB3V.js'),
  lazySmall: () => import('./Small-36WJC3FG.js'),
  lazyMedium: () => import('./Medium-3QZ4CAQB.js'),
  lazyLarge: () => import('./Large-YK2UVSWR.js')
});

// src/illustrations/tnt/Calculator/Calculator.tsx
var Calculator = createIllustration({
  name: "TntCalculator",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-IX5BHZ6L.js'),
  lazySmall: () => import('./Small-HNJEQNWD.js'),
  lazyMedium: () => import('./Medium-QZV2PRVW.js'),
  lazyLarge: () => import('./Large-MHGW2SM2.js')
});

// src/illustrations/tnt/ChartArea/ChartArea.tsx
var ChartArea = createIllustration({
  name: "TntChartArea",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-EKXJF6CC.js'),
  lazySmall: () => import('./Small-H4D5INCN.js'),
  lazyMedium: () => import('./Medium-NYGYORLD.js'),
  lazyLarge: () => import('./Large-RLICU6X6.js')
});

// src/illustrations/tnt/ChartArea2/ChartArea2.tsx
var ChartArea2 = createIllustration({
  name: "TntChartArea2",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-5FODIANL.js'),
  lazySmall: () => import('./Small-F6W4KGJL.js'),
  lazyMedium: () => import('./Medium-PBDAXDU3.js'),
  lazyLarge: () => import('./Large-JBZ2HS2V.js')
});

// src/illustrations/tnt/ChartBPMNFlow/ChartBPMNFlow.tsx
var ChartBPMNFlow = createIllustration({
  name: "TntChartBPMNFlow",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-F5VNXSNL.js'),
  lazySmall: () => import('./Small-7BSHRL3Q.js'),
  lazyMedium: () => import('./Medium-XSFR6NZB.js'),
  lazyLarge: () => import('./Large-QUM6JR27.js')
});

// src/illustrations/tnt/ChartBar/ChartBar.tsx
var ChartBar = createIllustration({
  name: "TntChartBar",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-BTQPTITZ.js'),
  lazySmall: () => import('./Small-SQDVFJIU.js'),
  lazyMedium: () => import('./Medium-ILXVHZ46.js'),
  lazyLarge: () => import('./Large-Y3WNQ75V.js')
});

// src/illustrations/tnt/ChartBullet/ChartBullet.tsx
var ChartBullet = createIllustration({
  name: "TntChartBullet",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-KJFNTOFG.js'),
  lazySmall: () => import('./Small-VRYNBTYH.js'),
  lazyMedium: () => import('./Medium-ND5LWGMC.js'),
  lazyLarge: () => import('./Large-52ND5UYV.js')
});

// src/illustrations/tnt/ChartDoughnut/ChartDoughnut.tsx
var ChartDoughnut = createIllustration({
  name: "TntChartDoughnut",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-QHCKJQZI.js'),
  lazySmall: () => import('./Small-HJDTM3FB.js'),
  lazyMedium: () => import('./Medium-OMOOXQBC.js'),
  lazyLarge: () => import('./Large-Y3UXYOAD.js')
});

// src/illustrations/tnt/ChartFlow/ChartFlow.tsx
var ChartFlow = createIllustration({
  name: "TntChartFlow",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-3L6MANV2.js'),
  lazySmall: () => import('./Small-D3HO75SH.js'),
  lazyMedium: () => import('./Medium-3N5464NR.js'),
  lazyLarge: () => import('./Large-JTOYXRZT.js')
});

// src/illustrations/tnt/ChartGantt/ChartGantt.tsx
var ChartGantt = createIllustration({
  name: "TntChartGantt",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-3J3SKMBA.js'),
  lazySmall: () => import('./Small-AOE6PO43.js'),
  lazyMedium: () => import('./Medium-46YPQQ2T.js'),
  lazyLarge: () => import('./Large-DHBA4PY6.js')
});

// src/illustrations/tnt/ChartOrg/ChartOrg.tsx
var ChartOrg = createIllustration({
  name: "TntChartOrg",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-6VSHUD2J.js'),
  lazySmall: () => import('./Small-NYCDN7SM.js'),
  lazyMedium: () => import('./Medium-GKATL2ES.js'),
  lazyLarge: () => import('./Large-ID2E3FPS.js')
});

// src/illustrations/tnt/ChartPie/ChartPie.tsx
var ChartPie = createIllustration({
  name: "TntChartPie",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-IY5GU6F7.js'),
  lazySmall: () => import('./Small-LMFUVRIM.js'),
  lazyMedium: () => import('./Medium-BQAOTBA6.js'),
  lazyLarge: () => import('./Large-BWGYDMCS.js')
});

// src/illustrations/tnt/CodePlaceholder/CodePlaceholder.tsx
var CodePlaceholder = createIllustration({
  name: "TntCodePlaceholder",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-XKLBANFY.js'),
  lazySmall: () => import('./Small-4NGGLJWU.js'),
  lazyMedium: () => import('./Medium-VR57P2VE.js'),
  lazyLarge: () => import('./Large-NWVR3NJZ.js')
});

// src/illustrations/tnt/Company/Company.tsx
var Company = createIllustration({
  name: "TntCompany",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-CEJSSXVT.js'),
  lazySmall: () => import('./Small-X5V6B476.js'),
  lazyMedium: () => import('./Medium-22L7GGUW.js'),
  lazyLarge: () => import('./Large-GQL4BUOL.js')
});

// src/illustrations/tnt/Compass/Compass.tsx
var Compass = createIllustration({
  name: "TntCompass",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-4KIJ5AH5.js'),
  lazySmall: () => import('./Small-56DAKIGE.js'),
  lazyMedium: () => import('./Medium-SCOAARIG.js'),
  lazyLarge: () => import('./Large-QIGMV2LD.js')
});

// src/illustrations/tnt/Components/Components.tsx
var Components = createIllustration({
  name: "TntComponents",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2WPM3DKC.js'),
  lazySmall: () => import('./Small-KV5T4YZD.js'),
  lazyMedium: () => import('./Medium-UWTRFUNA.js'),
  lazyLarge: () => import('./Large-ZAO5ZLE7.js')
});

// src/illustrations/tnt/Dialog/Dialog.tsx
var Dialog = createIllustration({
  name: "TntDialog",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-WHYI5YCF.js'),
  lazySmall: () => import('./Small-B7MHWV5X.js'),
  lazyMedium: () => import('./Medium-QLHX2PT7.js'),
  lazyLarge: () => import('./Large-KHJT3EET.js')
});

// src/illustrations/tnt/EmptyContentPane/EmptyContentPane.tsx
var EmptyContentPane = createIllustration({
  name: "TntEmptyContentPane",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-5SUJYQ5I.js'),
  lazySmall: () => import('./Small-5TVC6SID.js'),
  lazyMedium: () => import('./Medium-ILYHYMS7.js'),
  lazyLarge: () => import('./Large-4MV45DAW.js')
});

// src/illustrations/tnt/ExternalLink/ExternalLink.tsx
var ExternalLink = createIllustration({
  name: "TntExternalLink",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-UTNDBUDF.js'),
  lazySmall: () => import('./Small-KQ5JPMZ2.js'),
  lazyMedium: () => import('./Medium-HOR7GY7M.js'),
  lazyLarge: () => import('./Large-XX5PXXXP.js')
});

// src/illustrations/tnt/FaceID/FaceID.tsx
var FaceID = createIllustration({
  name: "TntFaceID",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-C6BTJQZ6.js'),
  lazySmall: () => import('./Small-FH7H74T4.js'),
  lazyMedium: () => import('./Medium-6TYYOSUW.js'),
  lazyLarge: () => import('./Large-A2DF7GK6.js')
});

// src/illustrations/tnt/Fingerprint/Fingerprint.tsx
var Fingerprint = createIllustration({
  name: "TntFingerprint",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2FLOYNCL.js'),
  lazySmall: () => import('./Small-76S3ARYV.js'),
  lazyMedium: () => import('./Medium-D5TEY4VW.js'),
  lazyLarge: () => import('./Large-WVJNLL6Y.js')
});

// src/illustrations/tnt/Handshake/Handshake.tsx
var Handshake = createIllustration({
  name: "TntHandshake",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-MLIUX27X.js'),
  lazySmall: () => import('./Small-DZM2LYDY.js'),
  lazyMedium: () => import('./Medium-45IHLGVO.js'),
  lazyLarge: () => import('./Large-DE4NFNBR.js')
});

// src/illustrations/tnt/Help/Help.tsx
var Help = createIllustration({
  name: "TntHelp",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2NR5RXXM.js'),
  lazySmall: () => import('./Small-OM54SBL6.js'),
  lazyMedium: () => import('./Medium-EWRJH62T.js'),
  lazyLarge: () => import('./Large-A26R6JKN.js')
});

// src/illustrations/tnt/Lock/Lock.tsx
var Lock = createIllustration({
  name: "TntLock",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-VSBSTA5Y.js'),
  lazySmall: () => import('./Small-UDBCACYM.js'),
  lazyMedium: () => import('./Medium-5TQ6K3P6.js'),
  lazyLarge: () => import('./Large-2J7B53LZ.js')
});

// src/illustrations/tnt/Mission/Mission.tsx
var Mission = createIllustration({
  name: "TntMission",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-IPBMDQLR.js'),
  lazySmall: () => import('./Small-WL4GH65K.js'),
  lazyMedium: () => import('./Medium-NQ5BRZKE.js'),
  lazyLarge: () => import('./Large-XPZNCZEG.js')
});

// src/illustrations/tnt/MissionFailed/MissionFailed.tsx
var MissionFailed = createIllustration({
  name: "TntMissionFailed",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-F5TJ43HP.js'),
  lazySmall: () => import('./Small-N7NL5VLB.js'),
  lazyMedium: () => import('./Medium-VGDY2ADW.js'),
  lazyLarge: () => import('./Large-EWLNQMJI.js')
});

// src/illustrations/tnt/NoApplications/NoApplications.tsx
var NoApplications = createIllustration({
  name: "TntNoApplications",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-UBQGS3H7.js'),
  lazySmall: () => import('./Small-5PV6MUXL.js'),
  lazyMedium: () => import('./Medium-DXMG6MML.js'),
  lazyLarge: () => import('./Large-DYTBDIVU.js')
});

// src/illustrations/tnt/NoFlows/NoFlows.tsx
var NoFlows = createIllustration({
  name: "TntNoFlows",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-5Y6SMZUF.js'),
  lazySmall: () => import('./Small-EUPWRVGZ.js'),
  lazyMedium: () => import('./Medium-QQL5LPCF.js'),
  lazyLarge: () => import('./Large-5EX5MGGV.js')
});

// src/illustrations/tnt/NoUsers/NoUsers.tsx
var NoUsers = createIllustration({
  name: "TntNoUsers",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-VRU5I2OP.js'),
  lazySmall: () => import('./Small-UGSF5G2M.js'),
  lazyMedium: () => import('./Medium-QQIXZAOO.js'),
  lazyLarge: () => import('./Large-BDHYIY2K.js')
});

// src/illustrations/tnt/Radar/Radar.tsx
var Radar = createIllustration({
  name: "TntRadar",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-S3MT7ZK6.js'),
  lazySmall: () => import('./Small-IA7L2QS2.js'),
  lazyMedium: () => import('./Medium-DRQXAEKU.js'),
  lazyLarge: () => import('./Large-35MO7H46.js')
});

// src/illustrations/tnt/RoadMap/RoadMap.tsx
var RoadMap = createIllustration({
  name: "TntRoadMap",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-FKQQMFQJ.js'),
  lazySmall: () => import('./Small-7ITT7MLK.js'),
  lazyMedium: () => import('./Medium-7KMRM5J6.js'),
  lazyLarge: () => import('./Large-DGTJ5GKZ.js')
});

// src/illustrations/tnt/Secrets/Secrets.tsx
var Secrets = createIllustration({
  name: "TntSecrets",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-XHY2PTA5.js'),
  lazySmall: () => import('./Small-L5YTNKCM.js'),
  lazyMedium: () => import('./Medium-Z3PYMTP6.js'),
  lazyLarge: () => import('./Large-KEPTPC4Z.js')
});

// src/illustrations/tnt/Services/Services.tsx
var Services = createIllustration({
  name: "TntServices",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2JBVDLU3.js'),
  lazySmall: () => import('./Small-2TAQTNLA.js'),
  lazyMedium: () => import('./Medium-EETKPGTP.js'),
  lazyLarge: () => import('./Large-6T6MZ7SJ.js')
});

// src/illustrations/tnt/SessionExpired/SessionExpired.tsx
var SessionExpired = createIllustration({
  name: "TntSessionExpired",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-M2ERH62B.js'),
  lazySmall: () => import('./Small-4A3H6F7M.js'),
  lazyMedium: () => import('./Medium-XJTESZQ3.js'),
  lazyLarge: () => import('./Large-5FDZTOWR.js')
});

// src/illustrations/tnt/SessionExpiring/SessionExpiring.tsx
var SessionExpiring = createIllustration({
  name: "TntSessionExpiring",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-CD4NG73S.js'),
  lazySmall: () => import('./Small-P3IFPIBU.js'),
  lazyMedium: () => import('./Medium-ZYZIIYHZ.js'),
  lazyLarge: () => import('./Large-FR2DKOD3.js')
});

// src/illustrations/tnt/Settings/Settings.tsx
var Settings = createIllustration({
  name: "TntSettings",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-MHNIREZL.js'),
  lazySmall: () => import('./Small-M6VNBJFV.js'),
  lazyMedium: () => import('./Medium-INFFOUFE.js'),
  lazyLarge: () => import('./Large-X6XYJCNY.js')
});

// src/illustrations/tnt/Success/Success.tsx
var Success = createIllustration({
  name: "TntSuccess",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-PS6W26UV.js'),
  lazySmall: () => import('./Small-IYARXDHY.js'),
  lazyMedium: () => import('./Medium-ZMBZSQH6.js'),
  lazyLarge: () => import('./Large-VODFCLC6.js')
});

// src/illustrations/tnt/SuccessfulAuth/SuccessfulAuth.tsx
var SuccessfulAuth = createIllustration({
  name: "TntSuccessfulAuth",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-HXJLYZL2.js'),
  lazySmall: () => import('./Small-Y6JXV6PB.js'),
  lazyMedium: () => import('./Medium-ZQNWZXYK.js'),
  lazyLarge: () => import('./Large-IVQSWKK5.js')
});

// src/illustrations/tnt/Systems/Systems.tsx
var Systems = createIllustration({
  name: "TntSystems",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-VSLNZT6N.js'),
  lazySmall: () => import('./Small-S47P3LIH.js'),
  lazyMedium: () => import('./Medium-YWEE7EL3.js'),
  lazyLarge: () => import('./Large-TTQXN6YZ.js')
});

// src/illustrations/tnt/Teams/Teams.tsx
var Teams = createIllustration({
  name: "TntTeams",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-AUFUMASJ.js'),
  lazySmall: () => import('./Small-6GQWTPSF.js'),
  lazyMedium: () => import('./Medium-I3Z7V2UJ.js'),
  lazyLarge: () => import('./Large-CV7ZOE3T.js')
});

// src/illustrations/tnt/Tools/Tools.tsx
var Tools = createIllustration({
  name: "TntTools",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-H4GQKDFE.js'),
  lazySmall: () => import('./Small-VU4XDRBW.js'),
  lazyMedium: () => import('./Medium-RF3SGGSQ.js'),
  lazyLarge: () => import('./Large-O2KZVOPR.js')
});

// src/illustrations/tnt/Tutorials/Tutorials.tsx
var Tutorials = createIllustration({
  name: "TntTutorials",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-7WGUXVEN.js'),
  lazySmall: () => import('./Small-Z2FEGJEQ.js'),
  lazyMedium: () => import('./Medium-T5WK2ICT.js'),
  lazyLarge: () => import('./Large-QYF6LPOB.js')
});

// src/illustrations/tnt/UnableToLoad/UnableToLoad.tsx
var UnableToLoad = createIllustration({
  name: "TntUnableToLoad",
  title: "Unable to load data",
  subtitle: "Check your internet connection. And if that's not it, try reloading. If that still doesn't help, check with your administrator.",
  lazyExtraSmall: () => import('./ExtraSmall-HPD55E6V.js'),
  lazySmall: () => import('./Small-WTHQYHIZ.js'),
  lazyMedium: () => import('./Medium-DXLKPKDV.js'),
  lazyLarge: () => import('./Large-KCQIWGRG.js')
});

// src/illustrations/tnt/Unlock/Unlock.tsx
var Unlock = createIllustration({
  name: "TntUnlock",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-C4BJ2RIW.js'),
  lazySmall: () => import('./Small-VQUCLOZE.js'),
  lazyMedium: () => import('./Medium-IWKDD533.js'),
  lazyLarge: () => import('./Large-WTR275QM.js')
});

// src/illustrations/tnt/UnsuccessfulAuth/UnsuccessfulAuth.tsx
var UnsuccessfulAuth = createIllustration({
  name: "TntUnsuccessfulAuth",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-43GGAGBI.js'),
  lazySmall: () => import('./Small-DE7W5HOB.js'),
  lazyMedium: () => import('./Medium-KHU4DYVZ.js'),
  lazyLarge: () => import('./Large-FTIUC6VS.js')
});

// src/illustrations/tnt/User2/User2.tsx
var User2 = createIllustration({
  name: "TntUser2",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-DXGRDNHS.js'),
  lazySmall: () => import('./Small-UO6LCE45.js'),
  lazyMedium: () => import('./Medium-G2YIRQQ4.js'),
  lazyLarge: () => import('./Large-OCHG6NMF.js')
});

export { Avatar, Calculator, ChartArea, ChartArea2, ChartBPMNFlow, ChartBar, ChartBullet, ChartDoughnut, ChartFlow, ChartGantt, ChartOrg, ChartPie, CodePlaceholder, Company, Compass, Components, Dialog, EmptyContentPane, ExternalLink, FaceID, Fingerprint, Handshake, Help, Lock, Mission, MissionFailed, NoApplications, NoFlows, NoUsers, Radar, RoadMap, Secrets, Services, SessionExpired, SessionExpiring, Settings, Success, SuccessfulAuth, Systems, Teams, Tools, Tutorials, UnableToLoad, Unlock, UnsuccessfulAuth, User2 };
