// @sap-ui/fx-components/dist/chunk-ZDBE5GTH.js
import { isDesktop, addOpenedPopup, removeOpenedPopup } from './chunk-YSPWI6FA.js';
import { ResizeCornerIcon } from './chunk-GCJAV5G5.js';
import { Title } from './chunk-BM2JFIU4.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useCallback, useEffect, useImperativeHandle, useMemo } from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

// src/types/popover.ts
var PopoverPlacement = /* @__PURE__ */ ((PopoverPlacement2) => {
  PopoverPlacement2["Start"] = "Start";
  PopoverPlacement2["End"] = "End";
  PopoverPlacement2["Top"] = "Top";
  PopoverPlacement2["Bottom"] = "Bottom";
  return PopoverPlacement2;
})(PopoverPlacement || {});
var PopoverHorizontalAlign = /* @__PURE__ */ ((PopoverHorizontalAlign2) => {
  PopoverHorizontalAlign2["Center"] = "Center";
  PopoverHorizontalAlign2["Start"] = "Start";
  PopoverHorizontalAlign2["End"] = "End";
  PopoverHorizontalAlign2["Stretch"] = "Stretch";
  return PopoverHorizontalAlign2;
})(PopoverHorizontalAlign || {});
var PopoverVerticalAlign = /* @__PURE__ */ ((PopoverVerticalAlign2) => {
  PopoverVerticalAlign2["Center"] = "Center";
  PopoverVerticalAlign2["Top"] = "Top";
  PopoverVerticalAlign2["Bottom"] = "Bottom";
  PopoverVerticalAlign2["Stretch"] = "Stretch";
  return PopoverVerticalAlign2;
})(PopoverVerticalAlign || {});
var PopupAccessibleRole = /* @__PURE__ */ ((PopupAccessibleRole2) => {
  PopupAccessibleRole2["Dialog"] = "Dialog";
  PopupAccessibleRole2["AlertDialog"] = "AlertDialog";
  PopupAccessibleRole2["None"] = "None";
  return PopupAccessibleRole2;
})(PopupAccessibleRole || {});
var ARROW_DEPTH = 8;
var ARROW_BASE = 16;
var ARROW_SIZE = ARROW_DEPTH;
var VIEWPORT_MARGIN = 10;
function clamp(val, min, max) {
  return Math.min(Math.max(val, min), max);
}
var FOCUSABLE = 'a[href], button:not([disabled]), textarea:not([disabled]), input:not([disabled]), select:not([disabled]), [tabindex]:not([tabindex="-1"])';
function getFirstFocusable(container) {
  for (const el of container.querySelectorAll(FOCUSABLE)) {
    if (el.dataset.focusTrap) continue;
    return el;
  }
  return null;
}
function getLastFocusable(container) {
  const list = container.querySelectorAll(FOCUSABLE);
  for (let i = list.length - 1; i >= 0; i--) {
    if (list[i].dataset.focusTrap) continue;
    return list[i];
  }
  return null;
}
function resolveOpener(opener) {
  if (!opener) return null;
  if (opener instanceof HTMLElement) return opener;
  if ("current" in opener) {
    const val = opener.current;
    if (!val) return null;
    if (val instanceof HTMLElement) return val;
    if (typeof val === "object" && "nativeElement" in val && val.nativeElement instanceof HTMLElement) {
      return val.nativeElement;
    }
  }
  return null;
}
function useIsRtl(ref) {
  const [isRtl, setIsRtl] = useState(false);
  useEffect(() => {
    const el = ref?.current ?? document.documentElement;
    const dir = el.closest("[dir]")?.getAttribute("dir") ?? document.documentElement.getAttribute("dir");
    setIsRtl(dir === "rtl");
  });
  return isRtl;
}
function isOpenerClippedByAncestor(opener) {
  let parent = opener.parentElement;
  while (parent && parent !== document.documentElement) {
    if (parent.hasAttribute("popover")) {
      try {
        if (parent.matches(":popover-open")) return false;
      } catch {
      }
    }
    const { overflow, overflowX, overflowY } = window.getComputedStyle(parent);
    const clips = [overflow, overflowX, overflowY].some(
      (v) => v === "hidden" || v === "scroll" || v === "auto"
    );
    if (clips) {
      const parentRect = parent.getBoundingClientRect();
      const openerRect = opener.getBoundingClientRect();
      if (openerRect.bottom <= parentRect.top || openerRect.top >= parentRect.bottom || openerRect.right <= parentRect.left || openerRect.left >= parentRect.right) {
        return true;
      }
    }
    parent = parent.parentElement;
  }
  return false;
}
function getActualHorizontalAlign(align, isRtl) {
  switch (align) {
    case "Start" /* Start */:
      return isRtl ? "Right" /* Right */ : "Left" /* Left */;
    case "End" /* End */:
      return isRtl ? "Left" /* Left */ : "Right" /* Right */;
    case "Stretch" /* Stretch */:
      return "Stretch" /* Stretch */;
    default:
      return "Center" /* Center */;
  }
}
function fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) {
  if (targetRect.left > popoverSize.width) return "Left" /* Left */;
  if (clientWidth - targetRect.right > targetRect.left) return "Right" /* Right */;
  if (clientHeight - targetRect.bottom > popoverSize.height) return "Bottom" /* Bottom */;
  if (clientHeight - targetRect.bottom < targetRect.top) return "Top" /* Top */;
  return void 0;
}
function getActualPlacement(desiredPlacement, isRtl, hideArrow, targetRect, popoverSize) {
  let actual;
  switch (desiredPlacement) {
    case "Start" /* Start */:
      actual = isRtl ? "Right" /* Right */ : "Left" /* Left */;
      break;
    case "End" /* End */:
      actual = isRtl ? "Left" /* Left */ : "Right" /* Right */;
      break;
    case "Top" /* Top */:
      actual = "Top" /* Top */;
      break;
    case "Bottom" /* Bottom */:
      actual = "Bottom" /* Bottom */;
      break;
    default:
      actual = isRtl ? "Left" /* Left */ : "Right" /* Right */;
  }
  const clientWidth = document.documentElement.clientWidth;
  let clientHeight = document.documentElement.clientHeight;
  const isVertical = desiredPlacement === "Top" /* Top */ || desiredPlacement === "Bottom" /* Bottom */;
  let popoverHeight = popoverSize.height;
  if (isVertical) {
    popoverHeight += hideArrow ? 0 : ARROW_SIZE;
    clientHeight -= VIEWPORT_MARGIN;
  }
  switch (actual) {
    case "Top" /* Top */:
      if (targetRect.top < popoverHeight && targetRect.top < clientHeight - targetRect.bottom)
        actual = "Bottom" /* Bottom */;
      break;
    case "Bottom" /* Bottom */:
      if (clientHeight - targetRect.bottom < popoverHeight && clientHeight - targetRect.bottom < targetRect.top)
        actual = "Top" /* Top */;
      break;
    case "Left" /* Left */:
      if (targetRect.left < popoverSize.width)
        actual = fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) || actual;
      break;
    case "Right" /* Right */:
      if (clientWidth - targetRect.right < popoverSize.width)
        actual = fallbackPlacement(clientWidth, clientHeight, targetRect, popoverSize) || actual;
      break;
  }
  return actual;
}
function calcPlacement(targetRect, popoverSize, desiredPlacement, horizontalAlign, verticalAlign, isRtl, hideArrow, allowTargetOverlap, borderRadius, offset) {
  const clientWidth = document.documentElement.clientWidth;
  const clientHeight = document.documentElement.clientHeight;
  let maxHeight = clientHeight;
  let maxWidth = clientWidth;
  let stretchWidth;
  let stretchHeight;
  const actualPlacement = getActualPlacement(
    desiredPlacement,
    isRtl,
    hideArrow,
    targetRect,
    popoverSize
  );
  const isVertical = actualPlacement === "Top" /* Top */ || actualPlacement === "Bottom" /* Bottom */;
  if (horizontalAlign === "Stretch" /* Stretch */ && isVertical) {
    popoverSize = { ...popoverSize, width: targetRect.width };
    stretchWidth = `${targetRect.width}px`;
  } else if (verticalAlign === "Stretch" /* Stretch */ && !isVertical) {
    popoverSize = { ...popoverSize, height: targetRect.height };
    stretchHeight = `${targetRect.height}px`;
  }
  const arrowOffset = (hideArrow ? 0 : ARROW_SIZE) + offset;
  const actualHAlign = getActualHorizontalAlign(horizontalAlign, isRtl);
  const getVerticalLeft = () => {
    switch (actualHAlign) {
      case "Center" /* Center */:
      case "Stretch" /* Stretch */:
        return targetRect.left - (popoverSize.width - targetRect.width) / 2;
      case "Left" /* Left */:
        return targetRect.left;
      case "Right" /* Right */:
        return targetRect.right - popoverSize.width;
      default:
        return VIEWPORT_MARGIN;
    }
  };
  const getHorizontalTop = () => {
    switch (verticalAlign) {
      case "Center" /* Center */:
      case "Stretch" /* Stretch */:
        return targetRect.top - (popoverSize.height - targetRect.height) / 2;
      case "Top" /* Top */:
        return targetRect.top;
      case "Bottom" /* Bottom */:
        return targetRect.bottom - popoverSize.height;
      default:
        return 0;
    }
  };
  let left = VIEWPORT_MARGIN;
  let top = 0;
  switch (actualPlacement) {
    case "Top" /* Top */:
      left = getVerticalLeft();
      top = Math.max(targetRect.top - popoverSize.height - arrowOffset, 0);
      if (!allowTargetOverlap) maxHeight = targetRect.top - arrowOffset;
      break;
    case "Bottom" /* Bottom */:
      left = getVerticalLeft();
      top = targetRect.bottom + arrowOffset;
      if (allowTargetOverlap) {
        top = Math.max(Math.min(top, clientHeight - popoverSize.height), 0);
      } else {
        maxHeight = clientHeight - targetRect.bottom - arrowOffset;
      }
      break;
    case "Left" /* Left */:
      left = Math.max(targetRect.left - popoverSize.width - arrowOffset, 0);
      top = getHorizontalTop();
      if (!allowTargetOverlap) maxWidth = targetRect.left - arrowOffset;
      break;
    case "Right" /* Right */:
      left = targetRect.left + targetRect.width + arrowOffset;
      top = getHorizontalTop();
      if (allowTargetOverlap) {
        left = Math.max(Math.min(left, clientWidth - popoverSize.width), 0);
      } else {
        maxWidth = clientWidth - targetRect.right - arrowOffset;
      }
      break;
  }
  if (isVertical) {
    if (popoverSize.width > clientWidth || left < VIEWPORT_MARGIN) {
      left = VIEWPORT_MARGIN;
    } else if (left + popoverSize.width > clientWidth - VIEWPORT_MARGIN) {
      left = clientWidth - VIEWPORT_MARGIN - popoverSize.width;
    }
  } else {
    if (popoverSize.height > clientHeight || top < VIEWPORT_MARGIN) {
      top = VIEWPORT_MARGIN;
    } else if (top + popoverSize.height > clientHeight - VIEWPORT_MARGIN) {
      top = clientHeight - VIEWPORT_MARGIN - popoverSize.height;
    }
  }
  left = Math.round(left);
  top = Math.round(top);
  maxHeight = Math.round(maxHeight - VIEWPORT_MARGIN);
  maxWidth = Math.round(maxWidth - VIEWPORT_MARGIN);
  let arrowXCentered = actualHAlign === "Center" /* Center */ || actualHAlign === "Stretch" /* Stretch */;
  if (actualHAlign === "Right" /* Right */ && left <= targetRect.left)
    arrowXCentered = true;
  if (actualHAlign === "Left" /* Left */ && left + popoverSize.width >= targetRect.left + targetRect.width)
    arrowXCentered = true;
  let arrowTranslateX = 0;
  if (isVertical && arrowXCentered) {
    arrowTranslateX = targetRect.left + targetRect.width / 2 - left - popoverSize.width / 2;
  }
  let arrowTranslateY = 0;
  if (!isVertical) {
    arrowTranslateY = targetRect.top + targetRect.height / 2 - top - popoverSize.height / 2;
  }
  const safeY = popoverSize.height / 2 - borderRadius - ARROW_BASE / 2 - 2;
  arrowTranslateY = clamp(arrowTranslateY, -safeY, safeY);
  const safeX = popoverSize.width / 2 - borderRadius - ARROW_BASE / 2 - 2;
  arrowTranslateX = clamp(arrowTranslateX, -safeX, safeX);
  return {
    arrow: { x: Math.round(arrowTranslateX), y: Math.round(arrowTranslateY) },
    top,
    left,
    maxHeight,
    maxWidth,
    actualPlacement,
    stretchWidth,
    stretchHeight
  };
}
function getResizeHandleCorner(popoverRect, openerRect, side, verticalAlign, actualHAlign, isRtl) {
  const offset = 2;
  let openerCX = Math.floor(openerRect.x + openerRect.width / 2);
  const openerCY = Math.floor(openerRect.y + openerRect.height / 2);
  let popoverCX = Math.floor(popoverRect.x + popoverRect.width / 2);
  const popoverCY = Math.floor(popoverRect.y + popoverRect.height / 2);
  const isPopoverWider = popoverRect.width > openerRect.width;
  const isPopoverTaller = popoverRect.height > openerRect.height;
  if (isRtl) {
    openerCX = -openerCX;
    popoverCX = -popoverCX;
  }
  switch (side) {
    case "Left" /* Left */:
      if (isPopoverTaller) {
        return popoverCY > openerCY + offset ? "bottom-left" : "top-left";
      }
      return verticalAlign === "Top" /* Top */ ? "bottom-left" : "top-left";
    case "Right" /* Right */:
      if (isPopoverTaller) {
        return popoverCY + offset < openerCY ? "top-right" : "bottom-right";
      }
      return verticalAlign === "Bottom" /* Bottom */ ? "top-right" : "bottom-right";
    case "Bottom" /* Bottom */:
      if (isPopoverWider) {
        if (popoverCX + offset < openerCX) {
          return isRtl ? "bottom-right" : "bottom-left";
        }
        return isRtl ? "bottom-left" : "bottom-right";
      }
      if (isRtl) {
        return actualHAlign === "Left" /* Left */ ? "bottom-right" : "bottom-left";
      }
      return actualHAlign === "Right" /* Right */ ? "bottom-left" : "bottom-right";
    case "Top" /* Top */:
    default:
      if (isPopoverWider) {
        if (popoverCX + offset < openerCX) {
          return isRtl ? "top-right" : "top-left";
        }
        return isRtl ? "top-left" : "top-right";
      }
      if (isRtl) {
        return actualHAlign === "Left" /* Left */ ? "top-right" : "top-left";
      }
      return actualHAlign === "Right" /* Right */ ? "top-left" : "top-right";
  }
}
function Popover({
  headerText,
  placement = "End" /* End */,
  horizontalAlign = "Center" /* Center */,
  verticalAlign = "Center" /* Center */,
  hideArrow = false,
  offset: offsetProp,
  allowTargetOverlap = false,
  resizable = false,
  noPadding = false,
  opener,
  open = false,
  initialFocus,
  preventFocusRestore = false,
  preventInitialFocus = false,
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Dialog" /* Dialog */,
  accessibleDescription,
  header,
  footer,
  children,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  onClose,
  isInsidePopup,
  className,
  style,
  id,
  ref
}) {
  const popoverRef = useRef(null);
  const arrowRef = useRef(null);
  const focusedBeforeOpen = useRef(null);
  const [isOpen, setIsOpen] = useState(false);
  const [pos, setPos] = useState(null);
  const [resizeSize, setResizeSize] = useState(null);
  const resizeState = useRef(null);
  const [isDesktop2] = useState(() => isDesktop());
  const containerRef = useRef(null);
  const isRtl = useIsRtl(containerRef);
  const getOpenerEl = useCallback(() => resolveOpener(opener), [opener]);
  const doCloseRef = useRef(() => {
  });
  const reposition = useCallback(() => {
    const openerEl = getOpenerEl();
    const popEl = popoverRef.current;
    if (!openerEl || !popEl) return;
    const targetRect = openerEl.getBoundingClientRect();
    if (targetRect.bottom < 0 || targetRect.top > window.innerHeight || targetRect.right < 0 || targetRect.left > window.innerWidth) {
      doCloseRef.current();
      return;
    }
    if (targetRect.top === 0 && targetRect.bottom === 0 && targetRect.left === 0 && targetRect.right === 0) {
      doCloseRef.current();
      return;
    }
    if (isOpenerClippedByAncestor(openerEl)) {
      doCloseRef.current();
      return;
    }
    const rect = popEl.getBoundingClientRect();
    const popoverSize = { width: rect.width, height: rect.height };
    if (popoverSize.width === 0 || popoverSize.height === 0) return;
    const borderRadius = Number.parseInt(
      window.getComputedStyle(popEl).getPropertyValue("border-radius") || "0"
    );
    const result = calcPlacement(
      targetRect,
      popoverSize,
      placement,
      horizontalAlign,
      verticalAlign,
      isRtl,
      hideArrow,
      allowTargetOverlap,
      borderRadius,
      offsetProp ?? 0
    );
    setPos(result);
  }, [getOpenerEl, placement, horizontalAlign, verticalAlign, isRtl, hideArrow, allowTargetOverlap, offsetProp]);
  useEffect(() => {
    if (!isOpen) return;
    const raf = requestAnimationFrame(() => {
      reposition();
      requestAnimationFrame(reposition);
    });
    const onScrollResize = () => reposition();
    window.addEventListener("scroll", onScrollResize, true);
    window.addEventListener("resize", onScrollResize);
    const openerEl = getOpenerEl();
    let ro = null;
    if (openerEl instanceof Element) {
      ro = new ResizeObserver(reposition);
      ro.observe(openerEl);
    }
    let roPopover = null;
    if (popoverRef.current instanceof Element) {
      roPopover = new ResizeObserver(reposition);
      roPopover.observe(popoverRef.current);
    }
    return () => {
      cancelAnimationFrame(raf);
      window.removeEventListener("scroll", onScrollResize, true);
      window.removeEventListener("resize", onScrollResize);
      ro?.disconnect();
      roPopover?.disconnect();
    };
  }, [isOpen, reposition, getOpenerEl]);
  const doOpen = useCallback(() => {
    if (isOpen) return;
    if (onBeforeOpen?.() === false) return;
    focusedBeforeOpen.current = document.activeElement;
    setIsOpen(true);
  }, [isOpen, onBeforeOpen]);
  const doClose = useCallback(
    (escPressed = false) => {
      if (!isOpen) return;
      if (onBeforeClose?.({ escPressed }) === false) return;
      const el = popoverRef.current;
      if (el && el.matches(":popover-open")) {
        el.hidePopover();
      }
      setIsOpen(false);
      setResizeSize(null);
      setPos(null);
      if (!preventFocusRestore && focusedBeforeOpen.current) {
        focusedBeforeOpen.current.focus();
        focusedBeforeOpen.current = null;
      }
      onClose?.();
    },
    [isOpen, onBeforeClose, onClose, preventFocusRestore]
  );
  doCloseRef.current = doClose;
  useEffect(() => {
    if (open) doOpen();
    else if (isOpen) doClose();
  }, [open]);
  useEffect(() => {
    const el = popoverRef.current;
    if (!el) return;
    if (isOpen && !el.matches(":popover-open")) {
      el.showPopover();
    }
    return () => {
      if (el.matches(":popover-open")) {
        el.hidePopover();
      }
    };
  }, [isOpen]);
  useEffect(() => {
    if (!isOpen) return;
    requestAnimationFrame(() => {
      if (preventInitialFocus || !popoverRef.current) return;
      if (initialFocus) {
        const el = document.getElementById(initialFocus);
        if (el) {
          el.focus();
          onOpen?.();
          return;
        }
      }
      const first = getFirstFocusable(popoverRef.current);
      if (first) first.focus();
      else popoverRef.current.focus();
      onOpen?.();
    });
  }, [isOpen]);
  useEffect(() => {
    if (!isOpen || !popoverRef.current) return;
    const el = popoverRef.current;
    addOpenedPopup({
      element: el,
      close: (escPressed) => doCloseRef.current(escPressed),
      type: "popover",
      isModal: false,
      getOpenerEl,
      isInsidePopup
    });
    return () => removeOpenedPopup(el);
  }, [isOpen, getOpenerEl, isInsidePopup]);
  const focusFirst = useCallback(() => {
    if (!popoverRef.current) return;
    const el = getFirstFocusable(popoverRef.current);
    (el || popoverRef.current).focus();
  }, []);
  const focusLast = useCallback(() => {
    if (!popoverRef.current) return;
    const el = getLastFocusable(popoverRef.current);
    (el || popoverRef.current).focus();
  }, []);
  const handleResizeMouseDown = useCallback(
    (e) => {
      if (!resizable || !popoverRef.current) return;
      e.preventDefault();
      const rect = popoverRef.current.getBoundingClientRect();
      const openerEl = getOpenerEl();
      const opRect = openerEl?.getBoundingClientRect();
      const corner = opRect && pos ? getResizeHandleCorner(
        rect,
        opRect,
        pos.actualPlacement,
        verticalAlign,
        getActualHorizontalAlign(horizontalAlign, isRtl),
        isRtl
      ) : "bottom-right";
      resizeState.current = {
        startX: e.clientX,
        startY: e.clientY,
        startWidth: rect.width,
        startHeight: rect.height,
        corner
      };
      const onMove = (ev) => {
        if (!resizeState.current) return;
        const { startX, startY, startWidth, startHeight, corner: c } = resizeState.current;
        const dx = ev.clientX - startX;
        const dy = ev.clientY - startY;
        const isL = c === "top-left" || c === "bottom-left";
        const isT = c === "top-left" || c === "top-right";
        const w = clamp(startWidth + (isL ? -dx : dx), 150, window.innerWidth - VIEWPORT_MARGIN * 2);
        const h = clamp(startHeight + (isT ? -dy : dy), 100, window.innerHeight - VIEWPORT_MARGIN * 2);
        setResizeSize({ width: w, height: h });
      };
      const onUp = () => {
        resizeState.current = null;
        window.removeEventListener("mousemove", onMove);
        window.removeEventListener("mouseup", onUp);
      };
      window.addEventListener("mousemove", onMove);
      window.addEventListener("mouseup", onUp);
    },
    [resizable, getOpenerEl, pos]
  );
  useImperativeHandle(ref, () => ({
    open() {
      doOpen();
    },
    close() {
      doClose();
    },
    async applyFocus() {
      if (!popoverRef.current) return;
      const el = getFirstFocusable(popoverRef.current);
      (el || popoverRef.current).focus();
    },
    isOpen() {
      return isOpen;
    },
    get nativeElement() {
      return popoverRef.current;
    }
  }), [isOpen, doOpen, doClose]);
  const resizeHandleCorner = useMemo(() => {
    if (!resizable || !isDesktop2 || !popoverRef.current || !pos) return "bottom-right";
    const openerEl = getOpenerEl();
    if (!openerEl) return "bottom-right";
    return getResizeHandleCorner(
      popoverRef.current.getBoundingClientRect(),
      openerEl.getBoundingClientRect(),
      pos.actualPlacement,
      verticalAlign,
      getActualHorizontalAlign(horizontalAlign, isRtl),
      isRtl
    );
  }, [resizable, isDesktop2, getOpenerEl, pos, verticalAlign, horizontalAlign, isRtl]);
  const role = accessibleRole === "None" /* None */ ? void 0 : accessibleRole === "AlertDialog" /* AlertDialog */ ? "alertdialog" : "dialog";
  const ariaModal = accessibleRole === "None" /* None */ ? void 0 : "true";
  const displayHeader = !!(header || headerText);
  const actualPlacement = pos?.actualPlacement ?? "Right" /* Right */;
  const resizeHandleClasses = {
    "top-left": "-top-2 -left-2 cursor-nw-resize",
    "top-right": "-top-2 -right-2 cursor-ne-resize",
    "bottom-left": "-bottom-2 -left-2 cursor-ne-resize",
    "bottom-right": "-bottom-2 -right-2 cursor-nw-resize"
  };
  const resizeIconRotation = {
    "top-left": "180deg",
    "top-right": "270deg",
    "bottom-left": "90deg",
    "bottom-right": "0deg"
  };
  if (!isOpen) {
    return /* @__PURE__ */ jsx("div", { ref: containerRef, className: "hidden" });
  }
  const popoverStyle = {
    position: "fixed",
    top: pos ? pos.top : -1e4,
    left: pos ? pos.left : -1e4,
    maxHeight: pos ? pos.maxHeight : void 0,
    maxWidth: pos ? pos.maxWidth : void 0,
    boxShadow: "0px 10px 15px -5px rgba(10, 10, 10, 0.15), 0px 5px 5px -2px rgba(10, 10, 10, 0.05)",
    ...pos?.stretchWidth ? { width: pos.stretchWidth } : {},
    ...pos?.stretchHeight ? { height: pos.stretchHeight } : {},
    ...resizeSize ? { width: resizeSize.width, height: resizeSize.height } : {},
    ...style
  };
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx("div", { ref: containerRef, className: "hidden" }),
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: popoverRef,
        id,
        popover: "manual",
        role,
        "aria-modal": ariaModal,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef || (!accessibleName && headerText ? `${id || "popover"}-title` : void 0),
        "aria-describedby": accessibleDescription ? `${id || "popover"}-desc` : void 0,
        tabIndex: -1,
        className: cn(
          "rounded-lg border border-border bg-popover text-popover-foreground outline-none",
          className
        ),
        style: popoverStyle,
        children: [
          /* @__PURE__ */ jsx("span", { tabIndex: 0, onFocus: focusLast, "data-focus-trap": "true", className: "sr-only", role: "none" }),
          !hideArrow && pos && (() => {
            const arrowBorderColor = "var(--border)";
            const arrowFill = "var(--popover)";
            let svgWidth;
            let svgHeight;
            let fillPoints;
            let strokePath;
            const wrapperStyle = { position: "absolute", lineHeight: 0 };
            switch (actualPlacement) {
              case "Bottom" /* Bottom */:
                svgWidth = ARROW_BASE;
                svgHeight = ARROW_DEPTH;
                fillPoints = `0,${ARROW_DEPTH} ${ARROW_BASE / 2},0 ${ARROW_BASE},${ARROW_DEPTH}`;
                strokePath = `M0,${ARROW_DEPTH} L${ARROW_BASE / 2},0 L${ARROW_BASE},${ARROW_DEPTH}`;
                wrapperStyle.left = "50%";
                wrapperStyle.top = -ARROW_DEPTH;
                wrapperStyle.transform = `translateX(calc(-50% + ${pos.arrow.x}px))`;
                break;
              case "Top" /* Top */:
                svgWidth = ARROW_BASE;
                svgHeight = ARROW_DEPTH;
                fillPoints = `0,0 ${ARROW_BASE / 2},${ARROW_DEPTH} ${ARROW_BASE},0`;
                strokePath = `M0,0 L${ARROW_BASE / 2},${ARROW_DEPTH} L${ARROW_BASE},0`;
                wrapperStyle.left = "50%";
                wrapperStyle.bottom = -ARROW_DEPTH;
                wrapperStyle.transform = `translateX(calc(-50% + ${pos.arrow.x}px))`;
                break;
              case "Right" /* Right */:
                svgWidth = ARROW_DEPTH;
                svgHeight = ARROW_BASE;
                fillPoints = `${ARROW_DEPTH},0 0,${ARROW_BASE / 2} ${ARROW_DEPTH},${ARROW_BASE}`;
                strokePath = `M${ARROW_DEPTH},0 L0,${ARROW_BASE / 2} L${ARROW_DEPTH},${ARROW_BASE}`;
                wrapperStyle.top = "50%";
                wrapperStyle.left = -ARROW_DEPTH;
                wrapperStyle.transform = `translateY(calc(-50% + ${pos.arrow.y}px))`;
                break;
              case "Left" /* Left */:
              default:
                svgWidth = ARROW_DEPTH;
                svgHeight = ARROW_BASE;
                fillPoints = `0,0 ${ARROW_DEPTH},${ARROW_BASE / 2} 0,${ARROW_BASE}`;
                strokePath = `M0,0 L${ARROW_DEPTH},${ARROW_BASE / 2} L0,${ARROW_BASE}`;
                wrapperStyle.top = "50%";
                wrapperStyle.right = -ARROW_DEPTH;
                wrapperStyle.transform = `translateY(calc(-50% + ${pos.arrow.y}px))`;
                break;
            }
            return /* @__PURE__ */ jsx("div", { ref: arrowRef, style: wrapperStyle, children: /* @__PURE__ */ jsxs("svg", { width: svgWidth, height: svgHeight, viewBox: `0 0 ${svgWidth} ${svgHeight}`, style: { display: "block" }, children: [
              /* @__PURE__ */ jsx("polygon", { points: fillPoints, fill: arrowFill, stroke: "none" }),
              /* @__PURE__ */ jsx("path", { d: strokePath, fill: "none", stroke: arrowBorderColor, strokeWidth: 1 })
            ] }) });
          })(),
          /* @__PURE__ */ jsxs("div", { className: "isolate z-1 flex flex-col gap-sapphire-2xs overflow-hidden rounded-lg", style: { maxHeight: "inherit" }, children: [
            displayHeader && /* @__PURE__ */ jsx(
              "header",
              {
                className: cn(
                  "shrink-0",
                  // When custom header is provided, let it control its own styling.
                  // When using headerText, apply default popover header styling.
                  !header && "px-sapphire-s py-sapphire-xs border-b border-sapphire-border-primary"
                ),
                children: header || /* @__PURE__ */ jsx(Title, { level: "H1", id: `${id || "popover"}-title`, className: "text-base font-bold", children: headerText })
              }
            ),
            /* @__PURE__ */ jsx("div", { className: cn("overflow-y-auto overflow-x-hidden flex-1 min-h-0", !noPadding && "px-sapphire-s py-sapphire-xs"), children }),
            accessibleDescription && /* @__PURE__ */ jsx("span", { id: `${id || "popover"}-desc`, className: "sr-only", children: accessibleDescription }),
            footer && /* @__PURE__ */ jsx("footer", { className: "px-sapphire-s py-sapphire-xs border-t border-sapphire-border-primary shrink-0", children: footer })
          ] }),
          resizable && isDesktop2 && /* @__PURE__ */ jsx(
            "div",
            {
              className: cn(
                "absolute flex items-center justify-center w-6 h-6 rounded-full z-[1]",
                resizeHandleClasses[resizeHandleCorner]
              ),
              onMouseDown: handleResizeMouseDown,
              children: /* @__PURE__ */ jsx(
                ResizeCornerIcon,
                {
                  className: "h-4 w-4 text-sapphire-text-tertiary",
                  style: { transform: `rotate(${resizeIconRotation[resizeHandleCorner]})` }
                }
              )
            }
          ),
          /* @__PURE__ */ jsx("span", { tabIndex: 0, onFocus: focusFirst, "data-focus-trap": "true", className: "sr-only", role: "none" })
        ]
      }
    )
  ] });
}

export { Popover, PopoverHorizontalAlign, PopoverPlacement, PopoverVerticalAlign, PopupAccessibleRole };
