// @sap-ui/fx-components/dist/chunk-DJBESHY7.js
import { valueStateIcons, valueStateLabelKeys, valueStateMessageColors } from './chunk-WWAPKHIM.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useState, useRef, useId, useEffect, useCallback, useImperativeHandle } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';

var textareaContainerVariants = cva(
  [
    "flex flex-col w-full rounded border bg-sapphire-canvas-primary",
    "overflow-clip transition-colors relative",
    "p-1",
    "focus-within:border-2 focus-within:p-[3px]"
  ],
  {
    variants: {
      valueState: {
        ["None" /* None */]: "border-sapphire-border-secondary hover:border-sapphire-border-accent focus-within:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-positive focus-within:border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-negative focus-within:border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-warning focus-within:border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-info focus-within:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      readonly: {
        true: "bg-sapphire-background-secondary border-sapphire-border-primary hover:border-sapphire-border-primary focus-within:!border-sapphire-border-accent",
        false: ""
      }
    },
    defaultVariants: {
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
function Textarea(props) {
  const {
    value: controlledValue,
    defaultValue = "",
    disabled = false,
    readonly = false,
    required = false,
    autocomplete,
    valueState = "None" /* None */,
    valueStateMessage,
    maxLength,
    showExceededText = false,
    minLength,
    rows = 0,
    growing = false,
    growingMaxRows,
    placeholder,
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    name,
    id: providedId,
    onInput,
    onChange,
    onFocus,
    onBlur,
    onKeyDown,
    onKeyUp,
    onSelect,
    onScroll,
    className,
    style,
    "data-testid": dataTestId,
    "data-ai-field": dataAiField,
    ref
  } = props;
  const { t } = useTranslation("fx");
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [focused, setFocused] = useState(false);
  const [previousValue, setPreviousValue] = useState(defaultValue);
  const textareaRef = useRef(null);
  const isComposing = useRef(false);
  const isControlled = controlledValue !== void 0;
  const currentValue = isControlled ? controlledValue : internalValue;
  const generatedId = useId();
  const textareaId = providedId || generatedId;
  const valueStateId = `${textareaId}-valuestate`;
  const characterCount = currentValue.length;
  const exceededCharacters = maxLength ? Math.max(0, characterCount - maxLength) : 0;
  const hasExceededMax = exceededCharacters > 0;
  useEffect(() => {
    if (growing && textareaRef.current) {
      const textarea = textareaRef.current;
      textarea.style.height = "auto";
      const newHeight = textarea.scrollHeight;
      if (growingMaxRows) {
        const lineHeight = parseInt(getComputedStyle(textarea).lineHeight);
        const maxHeight = lineHeight * growingMaxRows;
        textarea.style.height = `${Math.min(newHeight, maxHeight)}px`;
      } else {
        textarea.style.height = `${newHeight}px`;
      }
    }
  }, [currentValue, growing, growingMaxRows]);
  const handleInternalValueChange = useCallback(
    (newValue) => {
      if (!isControlled) {
        setInternalValue(newValue);
      }
    },
    [isControlled]
  );
  const handleInput = useCallback(
    (e) => {
      if (isComposing.current) return;
      const newValue = e.currentTarget.value;
      handleInternalValueChange(newValue);
      onInput?.({
        value: newValue,
        escapePressed: false
      });
    },
    [handleInternalValueChange, onInput]
  );
  const handleChange = useCallback(() => {
    if (currentValue !== previousValue) {
      onChange?.({
        value: currentValue,
        previousValue,
        source: "user"
      });
      setPreviousValue(currentValue);
    }
  }, [currentValue, previousValue, onChange]);
  const handleFocus = useCallback(
    (e) => {
      setFocused(true);
      onFocus?.(e);
    },
    [onFocus]
  );
  const handleBlur = useCallback(
    (e) => {
      setFocused(false);
      handleChange();
      onBlur?.(e);
    },
    [onBlur, handleChange]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === "Escape") {
        onInput?.({
          value: currentValue,
          escapePressed: true
        });
      }
      onKeyDown?.(e);
    },
    [onKeyDown, onInput, currentValue]
  );
  const handleSelect = useCallback(
    () => {
      if (!textareaRef.current) return;
      const { selectionStart, selectionEnd, value } = textareaRef.current;
      onSelect?.({
        start: selectionStart,
        end: selectionEnd,
        selectedText: value.substring(selectionStart, selectionEnd)
      });
    },
    [onSelect]
  );
  const handleCompositionStart = useCallback(() => {
    isComposing.current = true;
  }, []);
  const handleCompositionEnd = useCallback(
    (e) => {
      isComposing.current = false;
      const newValue = e.currentTarget.value;
      handleInternalValueChange(newValue);
      onInput?.({
        value: newValue,
        escapePressed: false
      });
    },
    [handleInternalValueChange, onInput]
  );
  useImperativeHandle(
    ref,
    () => ({
      focus: () => textareaRef.current?.focus(),
      blur: () => textareaRef.current?.blur(),
      select: () => textareaRef.current?.select(),
      setSelectionRange: (start, end) => {
        textareaRef.current?.setSelectionRange(start, end);
      },
      getValue: () => currentValue,
      setValue: (value, options) => {
        handleInternalValueChange(value);
        if (!options?.silent) {
          onChange?.({
            value,
            previousValue: currentValue,
            source: "programmatic"
          });
        }
        setPreviousValue(value);
      },
      clear: () => {
        handleInternalValueChange("");
        onChange?.({
          value: "",
          previousValue: currentValue,
          source: "programmatic"
        });
        setPreviousValue("");
      },
      getState: () => ({
        value: currentValue,
        focused,
        valid: textareaRef.current?.checkValidity() ?? true,
        valueState,
        characterCount,
        exceededCharacters
      }),
      checkValidity: () => textareaRef.current?.checkValidity() ?? true,
      reportValidity: () => textareaRef.current?.reportValidity() ?? true,
      setCustomValidity: (message) => {
        textareaRef.current?.setCustomValidity(message);
      },
      scrollTo: (x, y) => {
        textareaRef.current?.scrollTo(x, y);
      },
      get nativeElement() {
        return textareaRef.current;
      }
    }),
    [
      currentValue,
      focused,
      valueState,
      characterCount,
      exceededCharacters,
      handleInternalValueChange,
      onChange
    ]
  );
  const effectiveValueState = valueState;
  const isNegativeState = effectiveValueState === "Negative" /* Negative */;
  const ValueStateIcon = valueStateIcons[effectiveValueState];
  const showValueStateIcon = ValueStateIcon && effectiveValueState !== "None" /* None */;
  const ariaLabel = accessibleName;
  const ariaLabelledBy = accessibleNameRef;
  const ariaDescribedBy = [
    accessibleDescriptionRef,
    valueStateMessage ? valueStateId : null
  ].filter(Boolean).join(" ") || void 0;
  const ariaInvalid = effectiveValueState === "Negative" /* Negative */ ? true : void 0;
  const ariaRequired = required || void 0;
  return /* @__PURE__ */ jsxs("div", { className: cn("w-full", className), style, children: [
    /* @__PURE__ */ jsx(
      "div",
      {
        className: cn(textareaContainerVariants({
          valueState: effectiveValueState,
          disabled,
          readonly
        })),
        "data-part": "container",
        children: /* @__PURE__ */ jsx(
          "textarea",
          {
            ref: textareaRef,
            id: textareaId,
            name,
            value: currentValue,
            disabled,
            readOnly: readonly,
            required,
            placeholder,
            maxLength: maxLength && !showExceededText ? maxLength : void 0,
            minLength,
            rows: rows || void 0,
            autoComplete: autocomplete,
            "aria-label": ariaLabel,
            "aria-labelledby": ariaLabelledBy,
            "aria-describedby": ariaDescribedBy,
            "aria-invalid": ariaInvalid,
            "aria-required": ariaRequired,
            "data-testid": dataTestId,
            "data-ai-field": dataAiField,
            "data-part": "textarea",
            onInput: handleInput,
            onFocus: handleFocus,
            onBlur: handleBlur,
            onKeyDown: handleKeyDown,
            onKeyUp,
            onSelect: handleSelect,
            onScroll,
            onCompositionStart: handleCompositionStart,
            onCompositionEnd: handleCompositionEnd,
            className: cn(
              "w-full resize-none bg-transparent px-2 py-[3px]",
              "text-sm leading-5 text-sapphire-text-primary outline-none",
              "placeholder:text-sapphire-text-tertiary placeholder:italic",
              "disabled:cursor-not-allowed",
              readonly && "text-sapphire-text-secondary",
              growing && "overflow-hidden"
            )
          }
        )
      }
    ),
    (maxLength || showExceededText) && /* @__PURE__ */ jsxs(
      "div",
      {
        className: cn(
          "flex items-center justify-end gap-2 text-xs mt-1",
          hasExceededMax ? "text-sapphire-negative" : "text-sapphire-text-tertiary"
        ),
        "data-part": "counter",
        children: [
          showExceededText && hasExceededMax && /* @__PURE__ */ jsxs("span", { children: [
            "+",
            exceededCharacters
          ] }),
          maxLength && /* @__PURE__ */ jsxs("span", { children: [
            characterCount,
            " / ",
            maxLength
          ] })
        ]
      }
    ),
    /* @__PURE__ */ jsx(
      "div",
      {
        id: valueStateId,
        role: isNegativeState ? "alert" : "status",
        className: cn(
          "text-sm flex items-start gap-1",
          valueStateMessage ? "mt-2" : "hidden",
          valueStateMessageColors[effectiveValueState]
        ),
        "data-part": "value-state-message",
        children: valueStateMessage && /* @__PURE__ */ jsxs(Fragment, { children: [
          effectiveValueState !== "None" /* None */ && valueStateLabelKeys[effectiveValueState] && /* @__PURE__ */ jsxs("span", { className: "sr-only", children: [
            t(valueStateLabelKeys[effectiveValueState]),
            ":",
            " "
          ] }),
          showValueStateIcon && ValueStateIcon && /* @__PURE__ */ jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
          /* @__PURE__ */ jsx("span", { children: valueStateMessage })
        ] })
      }
    )
  ] });
}

export { Textarea };
