// @sap-ui/fx-components/dist/chunk-3OK4EB4T.js
import { SlimArrowRightIcon } from './chunk-ZQSLIXR7.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { cn } from './chunk-2F2PZMJ5.js';
import React, { createContext, useRef, useImperativeHandle, useCallback, useContext, useState } from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

var SideNavigationContext = createContext({
  collapsed: false,
  onItemSelect: () => {
  }
});
function SideNavigation({
  collapsed = false,
  selectedKey,
  children,
  fixedItems,
  header,
  onSelectionChange,
  className,
  style,
  ref
}) {
  const navRef = useRef(null);
  useImperativeHandle(ref, () => ({
    get nativeElement() {
      return navRef.current;
    },
    getNativeElement: () => navRef.current
  }));
  const handleItemSelect = useCallback(
    (key, item) => {
      onSelectionChange?.({ item: { ...item, itemKey: key } });
    },
    [onSelectionChange]
  );
  return /* @__PURE__ */ jsx(
    SideNavigationContext.Provider,
    {
      value: {
        collapsed,
        selectedKey,
        onItemSelect: handleItemSelect
      },
      children: /* @__PURE__ */ jsxs(
        "nav",
        {
          ref: navRef,
          role: "navigation",
          "aria-label": "Side navigation",
          className: cn(
            "flex flex-col h-full bg-card",
            collapsed ? "w-14" : "w-60",
            "transition-[width] duration-200",
            className
          ),
          style,
          children: [
            header && /* @__PURE__ */ jsx("div", { className: "flex-shrink-0 p-3 border-b border-border", children: header }),
            /* @__PURE__ */ jsx("div", { className: "flex-1 overflow-y-auto py-2", children: /* @__PURE__ */ jsx("ul", { role: "tree", className: "space-y-1 px-2", children }) }),
            fixedItems && /* @__PURE__ */ jsx("div", { className: "flex-shrink-0 border-t border-border py-2", children: /* @__PURE__ */ jsx("ul", { className: "space-y-1 px-2", children: fixedItems }) })
          ]
        }
      )
    }
  );
}
function SideNavigationItem({
  itemKey,
  text,
  icon,
  expanded: controlledExpanded,
  selected: controlledSelected,
  disabled = false,
  badge,
  href,
  target,
  children,
  onClick,
  className,
  wholeItemToggleable = false,
  ref
}) {
  const { collapsed, selectedKey, onItemSelect } = useContext(SideNavigationContext);
  const [internalExpanded, setInternalExpanded] = useState(false);
  const hasChildren = React.Children.count(children) > 0;
  const isExpanded = controlledExpanded ?? internalExpanded;
  const isSelected = controlledSelected ?? (itemKey ? selectedKey === itemKey : false);
  const handleClick = (e) => {
    if (disabled) return;
    if (hasChildren && !collapsed) {
      setInternalExpanded(!isExpanded);
      if (!wholeItemToggleable) {
        return;
      }
    }
    if (itemKey) {
      onItemSelect(itemKey, {
        itemKey,
        text,
        icon,
        expanded: isExpanded,
        selected: true,
        disabled,
        badge,
        href,
        target
      });
    }
    onClick?.(e);
  };
  const handleKeyDown = (e) => {
    if (disabled) return;
    if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      handleClick(e);
    }
    if (hasChildren) {
      if (e.key === "ArrowRight" && !isExpanded) {
        setInternalExpanded(true);
      } else if (e.key === "ArrowLeft" && isExpanded) {
        setInternalExpanded(false);
      }
    }
  };
  const content = /* @__PURE__ */ jsxs(Fragment, { children: [
    icon && /* @__PURE__ */ jsx("span", { className: "h-5 w-5 shrink-0 flex items-center justify-center", children: icon }),
    !collapsed && /* @__PURE__ */ jsxs(Fragment, { children: [
      /* @__PURE__ */ jsx("span", { className: "flex-1 truncate", children: text }),
      badge && /* @__PURE__ */ jsx("span", { className: "ml-auto", children: badge }),
      hasChildren && /* @__PURE__ */ jsx("span", { className: "ml-1", children: isExpanded ? /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "h-4 w-4" }) })
    ] })
  ] });
  const itemClasses = cn(
    "flex items-center gap-3 rounded-lg px-3 py-2 text-sm transition-colors",
    "hover:bg-accent hover:text-accent-foreground",
    "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
    isSelected && "bg-primary/10 text-primary font-medium",
    disabled && "pointer-events-none opacity-50",
    collapsed && "justify-center px-0",
    className
  );
  return /* @__PURE__ */ jsxs("li", { ref, role: "treeitem", "aria-selected": isSelected, "aria-expanded": hasChildren ? isExpanded : void 0, children: [
    href ? /* @__PURE__ */ jsx(
      "a",
      {
        href,
        target,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        className: itemClasses,
        tabIndex: disabled ? -1 : 0,
        children: content
      }
    ) : /* @__PURE__ */ jsx(
      "div",
      {
        role: "button",
        tabIndex: disabled ? -1 : 0,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        className: cn(itemClasses, "cursor-pointer"),
        children: content
      }
    ),
    hasChildren && isExpanded && !collapsed && /* @__PURE__ */ jsx("ul", { role: "group", className: "mt-1 ml-4 space-y-1", children })
  ] });
}
function SideNavigationSubItem({
  itemKey,
  text,
  icon,
  selected: controlledSelected,
  disabled = false,
  href,
  target,
  onClick,
  className,
  ref
}) {
  const { selectedKey, onItemSelect } = useContext(SideNavigationContext);
  const isSelected = controlledSelected ?? (itemKey ? selectedKey === itemKey : false);
  const handleClick = (e) => {
    if (disabled) return;
    if (itemKey) {
      onItemSelect(itemKey, {
        itemKey,
        text,
        icon,
        selected: true,
        disabled,
        href,
        target
      });
    }
    onClick?.(e);
  };
  const handleKeyDown = (e) => {
    if (disabled) return;
    if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      handleClick(e);
    }
  };
  const itemClasses = cn(
    "flex items-center gap-2 rounded-md px-3 py-1.5 text-sm transition-colors",
    "hover:bg-accent hover:text-accent-foreground",
    "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
    isSelected && "bg-primary/10 text-primary font-medium",
    disabled && "pointer-events-none opacity-50",
    className
  );
  const content = /* @__PURE__ */ jsxs(Fragment, { children: [
    icon && /* @__PURE__ */ jsx("span", { className: "h-4 w-4 shrink-0", children: icon }),
    /* @__PURE__ */ jsx("span", { className: "truncate", children: text })
  ] });
  return /* @__PURE__ */ jsx("li", { ref, role: "treeitem", "aria-selected": isSelected, children: href ? /* @__PURE__ */ jsx(
    "a",
    {
      href,
      target,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: itemClasses,
      tabIndex: disabled ? -1 : 0,
      children: content
    }
  ) : /* @__PURE__ */ jsx(
    "div",
    {
      role: "button",
      tabIndex: disabled ? -1 : 0,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(itemClasses, "cursor-pointer"),
      children: content
    }
  ) });
}
function SideNavigationGroup({ text, expanded: controlledExpanded = true, children, className, ref }) {
  const [internalExpanded, setInternalExpanded] = useState(true);
  const { collapsed } = useContext(SideNavigationContext);
  const isExpanded = controlledExpanded ?? internalExpanded;
  if (collapsed) {
    return /* @__PURE__ */ jsx(Fragment, { children });
  }
  return /* @__PURE__ */ jsxs("li", { ref, className: cn("pt-4 first:pt-0", className), children: [
    text && /* @__PURE__ */ jsxs(
      "button",
      {
        type: "button",
        onClick: () => setInternalExpanded(!isExpanded),
        className: "flex items-center gap-2 px-3 py-1 text-xs font-semibold text-sapphire-text-tertiary uppercase tracking-wider w-full hover:text-foreground",
        children: [
          /* @__PURE__ */ jsx("span", { className: "flex-1 text-left", children: text }),
          isExpanded ? /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-3 w-3" }) : /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "h-3 w-3" })
        ]
      }
    ),
    isExpanded && /* @__PURE__ */ jsx("ul", { className: "mt-1 space-y-1", children })
  ] });
}

export { SideNavigation, SideNavigationGroup, SideNavigationItem, SideNavigationSubItem };
