// @sap-ui/fx-components/dist/chunk-76CYWX53.js
import { SearchIcon } from './chunk-Y5TA5SIT.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useId, useState, useImperativeHandle, useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { jsx, jsxs } from 'react/jsx-runtime';

// src/types/search-field.ts
var SearchFieldSize = /* @__PURE__ */ ((SearchFieldSize2) => {
  SearchFieldSize2["Large"] = "Large";
  SearchFieldSize2["Medium"] = "Medium";
  return SearchFieldSize2;
})(SearchFieldSize || {});
var containerVariants = cva(
  [
    "flex items-center w-full border bg-sapphire-canvas-primary",
    "overflow-clip transition-colors relative",
    "border-sapphire-border-active hover:border-sapphire-border-accent"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: [
          "h-10 rounded-lg",
          "pl-3.5 pr-1 py-1 gap-1",
          "focus-within:border-2 focus-within:border-sapphire-border-accent focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]"
        ],
        ["Medium" /* Medium */]: [
          "h-8 rounded",
          "pl-3 pr-1 py-1 gap-1",
          "focus-within:border-2 focus-within:border-sapphire-border-accent focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]"
        ]
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      disabled: false
    }
  }
);
function SearchFieldBase(props) {
  const {
    ref,
    size = "Large" /* Large */,
    value: controlledValue,
    defaultValue = "",
    placeholder,
    disabled = false,
    loading = false,
    showClearIcon = false,
    accessibleName,
    accessibleNameRef,
    accessibleDescription,
    accessibleDescriptionRef,
    name,
    id,
    onInput,
    onSearch,
    onChange,
    onFocus,
    onBlur,
    onKeyDown,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const { t } = useTranslation("fx");
  const inputRef = useRef(null);
  const descId = useId();
  const generatedId = useId();
  const inputId = id || generatedId;
  const isControlled = controlledValue !== void 0;
  const [internalValue, setInternalValue] = useState(defaultValue);
  const currentValue = isControlled ? controlledValue : internalValue;
  useImperativeHandle(
    ref,
    () => ({
      focus: () => inputRef.current?.focus(),
      blur: () => inputRef.current?.blur(),
      getValue: () => inputRef.current?.value ?? "",
      setValue: (val) => {
        if (!isControlled) {
          setInternalValue(val);
        }
        onInput?.(val);
      },
      clear: () => {
        if (!isControlled) {
          setInternalValue("");
        }
        onInput?.("");
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isControlled, onInput]
  );
  const handleNativeChange = useCallback(
    (e) => {
      const val = e.target.value;
      if (!isControlled) {
        setInternalValue(val);
      }
      onInput?.(val);
    },
    [isControlled, onInput]
  );
  const handleNativeBlur = useCallback(
    (e) => {
      onChange?.(e.target.value);
      onBlur?.(e);
    },
    [onChange, onBlur]
  );
  const handleKeyDown = useCallback(
    (e) => {
      onKeyDown?.(e);
      if (e.key === "Enter" && !e.defaultPrevented && loading) {
        e.preventDefault();
        onSearch?.({
          value: inputRef.current?.value ?? "",
          clearButtonPressed: false
        });
      } else if (e.key === "Escape" && !e.defaultPrevented) {
        const currentVal = inputRef.current?.value ?? "";
        if (currentVal.length > 0) {
          e.preventDefault();
          if (!isControlled) {
            setInternalValue("");
          }
          onInput?.("");
          onSearch?.({
            value: "",
            clearButtonPressed: true
          });
        }
      }
    },
    [onKeyDown, isControlled, onInput, onSearch, loading]
  );
  const handleSubmit = useCallback((e) => {
    e.preventDefault();
    onSearch?.({
      value: inputRef.current?.value ?? "",
      clearButtonPressed: false
    });
    inputRef.current?.focus();
  }, [onSearch]);
  const handleClear = useCallback(() => {
    if (!isControlled) {
      setInternalValue("");
    }
    onInput?.("");
    onSearch?.({
      value: "",
      clearButtonPressed: true
    });
    inputRef.current?.focus();
  }, [isControlled, onInput, onSearch]);
  const hasValue = currentValue.length > 0;
  const isMedium = size === "Medium" /* Medium */;
  const hasCustomDescription = accessibleDescription && !accessibleDescriptionRef;
  const ariaDescribedBy = accessibleDescriptionRef ? accessibleDescriptionRef : descId;
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: cn("inline-flex w-full", className),
      style,
      "data-testid": dataTestId,
      children: /* @__PURE__ */ jsxs(
        "form",
        {
          onSubmit: handleSubmit,
          className: containerVariants({ size, disabled: !!disabled }),
          children: [
            !accessibleDescriptionRef && /* @__PURE__ */ jsx("span", { id: descId, className: "sr-only", children: hasCustomDescription ? accessibleDescription : t("SEARCHFIELD_ARIA_DESCRIPTION") }),
            /* @__PURE__ */ jsx(
              "input",
              {
                ref: inputRef,
                type: "search",
                id: inputId,
                name,
                disabled,
                role: "searchbox",
                enterKeyHint: "search",
                "aria-label": accessibleNameRef ? void 0 : accessibleName || t("SEARCHFIELD_SEARCH"),
                "aria-labelledby": accessibleNameRef,
                "aria-describedby": ariaDescribedBy,
                "aria-roledescription": t("SEARCHFIELD_ROLE_DESCRIPTION"),
                "aria-disabled": disabled || void 0,
                value: currentValue,
                placeholder,
                className: cn(
                  "flex-1 min-w-0 bg-transparent outline-none text-sm text-foreground",
                  "placeholder:text-sapphire-text-tertiary placeholder:italic",
                  "[&::-webkit-search-cancel-button]:appearance-none",
                  "[&::-webkit-search-decoration]:appearance-none",
                  disabled && "cursor-not-allowed"
                ),
                onChange: handleNativeChange,
                onFocus,
                onBlur: handleNativeBlur,
                onKeyDown: handleKeyDown
              }
            ),
            showClearIcon && hasValue && !disabled && /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
              Button,
              {
                design: "SecondaryNeutral" /* SecondaryNeutral */,
                size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
                iconOnly: true,
                icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-full w-full" }) }),
                tabIndex: -1,
                onClick: handleClear,
                accessibleName: t("SEARCHFIELD_CLEAR"),
                className: "flex-none"
              }
            ) }),
            /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
              Button,
              {
                design: "SecondaryNeutral" /* SecondaryNeutral */,
                size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
                iconOnly: true,
                type: "Submit" /* Submit */,
                icon: loading ? /* @__PURE__ */ jsx(
                  "span",
                  {
                    className: cn(
                      "block rounded-full border-2 border-current border-t-transparent animate-spin",
                      isMedium ? "h-3 w-3" : "h-4 w-4"
                    ),
                    role: "status",
                    "aria-label": t("LOADING")
                  }
                ) : /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(SearchIcon, { className: "h-full w-full" }) }),
                tabIndex: -1,
                disabled: disabled || loading,
                accessibleName: t("SEARCHFIELD_SEARCH"),
                className: "flex-none"
              }
            ) })
          ]
        }
      )
    }
  );
}
var SearchField = React.memo(SearchFieldBase);
SearchField.displayName = "SearchField";

export { SearchField, SearchFieldSize };
