// @sap-ui/fx-components/dist/chunk-MIIXFVME.js
import { Toolbar } from './chunk-3GZW6BV2.js';
import { Menu, MenuItem } from './chunk-EHO24TXD.js';
import { SortAscendingIcon, SortDescendingIcon, AiIcon } from './chunk-H4UBXTRR.js';
import { OverflowIcon } from './chunk-YHUSMQDT.js';
import { SlimArrowRightIcon } from './chunk-ZQSLIXR7.js';
import { getTabbableElements } from './chunk-O5ILLEDP.js';
import { RadioButton } from './chunk-2MAGJSMZ.js';
import { CheckBox } from './chunk-TNSICZD6.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import React, { createContext, useId, useRef, useMemo, useState, useEffect, useCallback, useImperativeHandle, Children, isValidElement, useLayoutEffect, useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { Loader2 } from 'lucide-react';

var defaultContext = {
  tableId: "",
  columnCount: 0,
  columnsMeta: [],
  gridTemplateColumns: "",
  selectionMode: "None",
  selectionBehavior: "RowSelector",
  headerSelector: "SelectAll",
  isSelected: () => false,
  toggleSelection: () => {
  },
  selectRange: () => {
  },
  applyKeyboardRange: () => {
  },
  clearRangeSession: () => {
  },
  selectedKeys: /* @__PURE__ */ new Set(),
  areAllSelected: () => false,
  selectAll: () => {
  },
  deselectAll: () => {
  },
  getLastSelectedKey: () => null,
  focusedRowIndex: -1,
  focusedCellIndex: null,
  focusedCell: null,
  navigationMode: "Row",
  setFocusedRow: () => {
  },
  setFocusedCell: () => {
  },
  rowActionCount: 0,
  hasSelectionColumn: false,
  hasActionsColumn: false,
  overflowMode: "Scroll",
  rowKeys: [],
  getRowIndexByKey: () => -1,
  visibleColumnIndices: [],
  popinColumnIndices: [],
  isPopinActive: false,
  popinColumnIndex: 0,
  setMouseNavigating: () => {
  },
  navigateToRow: () => {
  },
  onRowClick: void 0,
  onRowActionClick: void 0,
  dataRowCount: 0,
  alternateRowColors: false,
  hasNavigatedColumn: false,
  hasGrowingButton: false,
  hasToolbar: false
};
var TableContext = createContext(defaultContext);
TableContext.displayName = "TableContext";
function useTableContext() {
  return useContext(TableContext);
}
function scanTableChildren(children) {
  let toolbar = null;
  let headerRow = null;
  const dataRows = [];
  let selectionFeature = null;
  let growingFeature = null;
  React.Children.forEach(children, (child) => {
    if (!React.isValidElement(child)) return;
    const role = child.type?._tableRole;
    switch (role) {
      case "Toolbar":
        toolbar = child;
        break;
      case "HeaderRow":
        headerRow = child;
        break;
      case "Row":
        dataRows.push(child);
        break;
      case "SelectionMulti":
      case "SelectionSingle":
        selectionFeature = child;
        break;
      case "Growing":
        growingFeature = child;
        break;
    }
  });
  return { toolbar, headerRow, dataRows, selectionFeature, growingFeature };
}
function extractTextContent(node) {
  if (node == null || typeof node === "boolean") return "";
  if (typeof node === "string" || typeof node === "number") return String(node);
  if (Array.isArray(node)) return node.map(extractTextContent).join("");
  if (React.isValidElement(node)) {
    return extractTextContent(node.props.children);
  }
  return "";
}
function extractColumnsMeta(headerRow) {
  if (!headerRow) return [];
  const metas = [];
  React.Children.forEach(headerRow.props.children, (child) => {
    if (!React.isValidElement(child)) return;
    const role = child.type?._tableRole;
    if (role === "HeaderCell") {
      const p = child.props;
      const headerText = extractTextContent(p.children).trim() || void 0;
      metas.push({
        width: p.width,
        minWidth: p.minWidth,
        importance: p.importance ?? 0,
        popinText: p.popinText,
        headerText,
        popinHidden: p.popinHidden,
        horizontalAlign: p.horizontalAlign,
        sortIndicator: p.sortIndicator
      });
    }
  });
  return metas;
}
function extractRowKeys(dataRows) {
  return dataRows.map((row) => {
    const p = row.props;
    return p.rowKey ?? "";
  });
}
function scanRowChildren(children) {
  const cells = [];
  const actions = [];
  let navigationAction = null;
  React.Children.forEach(children, (child) => {
    if (!React.isValidElement(child)) return;
    const role = child.type?._tableRole;
    switch (role) {
      case "Cell":
        cells.push(child);
        break;
      case "RowAction":
        actions.push(child);
        break;
      case "RowActionNavigation":
        navigationAction = child;
        break;
      default:
        cells.push(child);
        break;
    }
  });
  return { cells, actions, navigationAction };
}
function findVerticalScrollContainer(element) {
  let current = element.parentElement;
  while (current) {
    const { overflowY } = getComputedStyle(current);
    if (overflowY === "auto" || overflowY === "scroll") {
      return current;
    }
    current = current.parentElement;
  }
  return null;
}
function getAlignmentClass(align) {
  switch (align) {
    case "Left":
    case "Start":
      return "text-start justify-start";
    case "Right":
    case "End":
      return "text-end justify-end";
    case "Center":
      return "text-center justify-center";
    default:
      return "";
  }
}
function buildGridTemplateColumns(columnsMeta, visibleIndices, hasSelectionColumn, hasActionsColumn, hasNavigatedColumn) {
  const parts = [];
  if (hasSelectionColumn) {
    parts.push("40px");
  }
  const isOnlyColumn = visibleIndices.length === 1;
  visibleIndices.forEach((idx) => {
    const col = columnsMeta[idx];
    if (isOnlyColumn) {
      const min = col?.width || col?.minWidth || "150px";
      parts.push(`minmax(${min}, 1fr)`);
    } else if (col?.width) {
      parts.push(col.width);
    } else if (col?.minWidth) {
      parts.push(`minmax(${col.minWidth}, 1fr)`);
    } else {
      parts.push("minmax(150px, 1fr)");
    }
  });
  if (hasActionsColumn) {
    parts.push("auto");
  }
  if (hasNavigatedColumn) {
    parts.push("4px");
  }
  return parts.join(" ");
}
function parseSelected(value) {
  if (!value) return /* @__PURE__ */ new Set();
  return new Set(value.split(" ").filter(Boolean));
}
function useTableSelection(options) {
  const { mode, selected: controlledSelected, defaultSelected, allRowKeys, onChange } = options;
  const isControlled = controlledSelected !== void 0;
  const [internalSelected, setInternalSelected] = useState(
    () => parseSelected(defaultSelected)
  );
  const lastSelectedRef = useRef(null);
  const rangeSessionRef = useRef(null);
  const selectedKeys = isControlled ? parseSelected(controlledSelected) : internalSelected;
  const fireChange = useCallback(
    (newKeys, prevKeys) => {
      onChange?.({ selectedKeys: newKeys, previousSelectedKeys: prevKeys });
    },
    [onChange]
  );
  const updateSelection = useCallback(
    (newKeys) => {
      const prev = new Set(selectedKeys);
      if (!isControlled) {
        setInternalSelected(newKeys);
      }
      fireChange(newKeys, prev);
    },
    [selectedKeys, isControlled, fireChange]
  );
  const isSelected = useCallback(
    (rowKey) => selectedKeys.has(rowKey),
    [selectedKeys]
  );
  const toggleSelection = useCallback(
    (rowKey) => {
      if (mode === "None") return;
      const newKeys = new Set(selectedKeys);
      if (mode === "Single") {
        if (newKeys.has(rowKey)) {
          newKeys.clear();
        } else {
          newKeys.clear();
          newKeys.add(rowKey);
        }
      } else {
        if (newKeys.has(rowKey)) {
          newKeys.delete(rowKey);
        } else {
          newKeys.add(rowKey);
        }
      }
      rangeSessionRef.current = null;
      lastSelectedRef.current = rowKey;
      updateSelection(newKeys);
    },
    [mode, selectedKeys, updateSelection]
  );
  const setSelected = useCallback(
    (rowKey, value) => {
      if (mode === "None") return;
      const newKeys = new Set(selectedKeys);
      if (mode === "Single") {
        newKeys.clear();
        if (value) newKeys.add(rowKey);
      } else {
        if (value) {
          newKeys.add(rowKey);
        } else {
          newKeys.delete(rowKey);
        }
      }
      rangeSessionRef.current = null;
      if (value) {
        lastSelectedRef.current = rowKey;
      }
      updateSelection(newKeys);
    },
    [mode, selectedKeys, updateSelection]
  );
  const selectAll = useCallback(() => {
    if (mode !== "Multi") return;
    rangeSessionRef.current = null;
    const newKeys = new Set(allRowKeys);
    updateSelection(newKeys);
  }, [mode, allRowKeys, updateSelection]);
  const deselectAll = useCallback(() => {
    rangeSessionRef.current = null;
    updateSelection(/* @__PURE__ */ new Set());
  }, [updateSelection]);
  const areAllSelected = useCallback(
    () => allRowKeys.length > 0 && allRowKeys.every((key) => selectedKeys.has(key)),
    [allRowKeys, selectedKeys]
  );
  const selectRange = useCallback(
    (anchorKey, endKey) => {
      if (mode !== "Multi") return;
      const anchorIdx = allRowKeys.indexOf(anchorKey);
      const endIdx = allRowKeys.indexOf(endKey);
      if (anchorIdx === -1 || endIdx === -1) return;
      const start = Math.min(anchorIdx, endIdx);
      const end = Math.max(anchorIdx, endIdx);
      const activeSession = rangeSessionRef.current;
      const keepSessionBase = activeSession?.type === "select" && activeSession.anchorKey === anchorKey;
      const baseKeys = keepSessionBase ? activeSession.baseKeys : selectedKeys;
      const newKeys = new Set(baseKeys);
      for (let i = start; i <= end; i++) {
        newKeys.add(allRowKeys[i]);
      }
      updateSelection(newKeys);
      lastSelectedRef.current = anchorKey;
      rangeSessionRef.current = {
        type: "select",
        anchorKey,
        baseKeys
      };
    },
    [mode, allRowKeys, selectedKeys, updateSelection]
  );
  const deselectRange = useCallback(
    (startKey, endKey) => {
      if (mode !== "Multi") return;
      const startIdx = allRowKeys.indexOf(startKey);
      const endIdx = allRowKeys.indexOf(endKey);
      if (startIdx === -1 || endIdx === -1) return;
      const minIdx = Math.min(startIdx, endIdx);
      const maxIdx = Math.max(startIdx, endIdx);
      const activeSession = rangeSessionRef.current;
      const keepSessionBase = activeSession?.type === "deselect" && activeSession.anchorKey === startKey;
      const baseKeys = keepSessionBase ? activeSession.baseKeys : selectedKeys;
      const rangeKeys = new Set(allRowKeys.slice(minIdx, maxIdx + 1));
      const newKeys = new Set([...baseKeys].filter((key) => !rangeKeys.has(key)));
      updateSelection(newKeys);
      lastSelectedRef.current = startKey;
      rangeSessionRef.current = {
        type: "deselect",
        anchorKey: startKey,
        baseKeys
      };
    },
    [mode, allRowKeys, selectedKeys, updateSelection]
  );
  const applyKeyboardRange = useCallback(
    (currentKey, endKey) => {
      if (mode !== "Multi") return;
      const shouldSelect = isSelected(currentKey);
      const anchorKey = shouldSelect ? lastSelectedRef.current ?? currentKey : currentKey;
      if (!anchorKey) return;
      if (shouldSelect) {
        selectRange(anchorKey, endKey);
      } else {
        deselectRange(anchorKey, endKey);
      }
    },
    [deselectRange, isSelected, mode, selectRange]
  );
  const getLastSelectedKey = useCallback(
    () => lastSelectedRef.current,
    []
  );
  const clearRangeSession = useCallback((options2) => {
    rangeSessionRef.current = null;
    if (options2?.clearAnchor) {
      lastSelectedRef.current = null;
    }
  }, []);
  const setSelectedAsSet = useCallback(
    (newKeys) => {
      rangeSessionRef.current = null;
      updateSelection(newKeys);
    },
    [updateSelection]
  );
  return {
    selectedKeys,
    isSelected,
    toggleSelection,
    setSelected,
    setSelectedAsSet,
    selectAll,
    deselectAll,
    areAllSelected,
    selectRange,
    deselectRange,
    applyKeyboardRange,
    getLastSelectedKey,
    clearRangeSession
  };
}
function handleTableKeyboardNavigation(opts) {
  if (!opts.isCellMode) {
    return handleRowModeKey(opts);
  }
  return handleCellModeKey(opts);
}
function handleRowArrowDown(event, currentRow, dataLength, hasGrowingButton, focusRowAt, onRowShiftRange) {
  const nextRow = currentRow < 0 ? 0 : Math.min(currentRow + 1, dataLength - 1);
  if (event.shiftKey && onRowShiftRange && dataLength > 0) {
    event.preventDefault();
    onRowShiftRange(currentRow, nextRow);
    focusRowAt(currentRow >= dataLength - 1 && hasGrowingButton ? dataLength : nextRow);
    return;
  }
  event.preventDefault();
  if (currentRow < 0) {
    if (dataLength > 0) focusRowAt(0);
  } else if (currentRow >= dataLength - 1) {
    if (hasGrowingButton) focusRowAt(dataLength);
  } else {
    focusRowAt(currentRow + 1);
  }
}
function handleRowArrowUp(event, currentRow, dataLength, focusRowAt, onRowShiftRange) {
  const nextRow = currentRow === dataLength ? dataLength - 1 : Math.max(currentRow - 1, 0);
  if (event.shiftKey && onRowShiftRange && dataLength > 0) {
    event.preventDefault();
    onRowShiftRange(currentRow, nextRow);
    focusRowAt(currentRow === dataLength ? dataLength - 1 : nextRow);
    return;
  }
  event.preventDefault();
  if (currentRow === dataLength) {
    focusRowAt(dataLength - 1);
  } else if (currentRow <= 0) {
    focusRowAt(-1);
  } else {
    focusRowAt(currentRow - 1);
  }
}
function handleRowPageNavigation(key, event, currentRow, dataLength, hasGrowingButton, focusRowAt, pageSize) {
  event.preventDefault();
  if (key === "PageUp") {
    focusRowAt(currentRow <= 0 ? -1 : Math.max(currentRow - pageSize, 0));
  } else {
    if (currentRow >= dataLength - 1) {
      if (hasGrowingButton) focusRowAt(dataLength);
    } else {
      focusRowAt(Math.min(currentRow + pageSize, dataLength - 1));
    }
  }
  return true;
}
function handleRowModeKey(opts) {
  const { event, currentRow, dataLength, hasGrowingButton, horizontalForwardKey, focusRowAt, focusCellAt, onRowShiftRange, pageSize = 10 } = opts;
  switch (event.key) {
    case "ArrowDown":
      handleRowArrowDown(event, currentRow, dataLength, hasGrowingButton, focusRowAt, onRowShiftRange);
      return true;
    case "ArrowUp":
      handleRowArrowUp(event, currentRow, dataLength, focusRowAt, onRowShiftRange);
      return true;
    case "ArrowRight":
    case "ArrowLeft": {
      if (event.key !== horizontalForwardKey) return true;
      if (currentRow <= dataLength - 1) {
        event.preventDefault();
        focusCellAt(currentRow, 0);
      }
      return true;
    }
    case "Home": {
      event.preventDefault();
      focusRowAt(currentRow === 0 ? -1 : 0);
      return true;
    }
    case "End": {
      event.preventDefault();
      if (dataLength === 0) {
        focusRowAt(-1);
        return true;
      }
      focusRowAt(currentRow === dataLength - 1 && hasGrowingButton ? dataLength : dataLength - 1);
      return true;
    }
    case "PageUp":
    case "PageDown":
      return handleRowPageNavigation(event.key, event, currentRow, dataLength, hasGrowingButton, focusRowAt, pageSize);
    default:
      return false;
  }
}
function handleCellVerticalArrow(direction, event, currentRow, currentCol, dataLength, hasGrowingButton, focusRowAt, focusCellAt, onCellShiftRange) {
  event.preventDefault();
  const destinationRow = currentRow + direction;
  if (event.shiftKey && onCellShiftRange && currentRow >= 0 && currentRow < dataLength && destinationRow >= 0 && destinationRow < dataLength) {
    onCellShiftRange(currentRow, destinationRow);
  }
  if (direction === 1) {
    if (currentRow < dataLength - 1) {
      focusCellAt(currentRow + 1, currentCol);
    } else if (hasGrowingButton) {
      focusRowAt(dataLength);
    }
  } else {
    focusCellAt(currentRow <= -1 ? -1 : currentRow - 1, currentCol);
  }
  return true;
}
function handleCellPageNavigation(key, event, currentRow, currentCol, dataLength, hasGrowingButton, focusRowAt, focusCellAt, pageSize) {
  event.preventDefault();
  if (key === "PageUp") {
    focusCellAt(currentRow <= 0 ? -1 : Math.max(currentRow - pageSize, 0), currentCol);
  } else {
    if (currentRow >= dataLength - 1) {
      if (hasGrowingButton) focusRowAt(dataLength);
    } else {
      focusCellAt(Math.min(currentRow + pageSize, dataLength - 1), currentCol);
    }
  }
  return true;
}
function handleCellModeKey(opts) {
  const {
    event,
    currentRow,
    currentCol,
    dataLength,
    hasGrowingButton,
    horizontalForwardKey,
    horizontalBackwardKey,
    columnIndices,
    focusRowAt,
    focusCellAt,
    onCellShiftRange,
    pageSize = 10
  } = opts;
  const indices = columnIndices.length > 0 ? columnIndices : [currentCol];
  const firstColumn = indices[0] ?? 0;
  const lastColumn = indices[indices.length - 1] ?? firstColumn;
  const currentColPosition = indices.findIndex((index) => index === currentCol);
  switch (event.key) {
    case "ArrowLeft":
    case "ArrowRight": {
      if (event.key === horizontalBackwardKey) {
        event.preventDefault();
        if (currentColPosition <= 0) {
          focusRowAt(currentRow);
        } else {
          focusCellAt(currentRow, indices[currentColPosition - 1] ?? firstColumn);
        }
        return true;
      }
      if (event.key === horizontalForwardKey) {
        event.preventDefault();
        if (currentColPosition !== -1 && currentColPosition < indices.length - 1) {
          focusCellAt(currentRow, indices[currentColPosition + 1] ?? lastColumn);
        }
      }
      return true;
    }
    case "ArrowUp":
      return handleCellVerticalArrow(-1, event, currentRow, currentCol, dataLength, hasGrowingButton, focusRowAt, focusCellAt, onCellShiftRange);
    case "ArrowDown":
      return handleCellVerticalArrow(1, event, currentRow, currentCol, dataLength, hasGrowingButton, focusRowAt, focusCellAt, onCellShiftRange);
    case "Home": {
      event.preventDefault();
      if (currentCol === firstColumn) {
        focusRowAt(currentRow);
      } else {
        focusCellAt(currentRow, firstColumn);
      }
      return true;
    }
    case "End": {
      event.preventDefault();
      if (currentCol === lastColumn) {
        focusRowAt(currentRow);
      } else {
        focusCellAt(currentRow, lastColumn);
      }
      return true;
    }
    case "PageUp":
    case "PageDown":
      return handleCellPageNavigation(event.key, event, currentRow, currentCol, dataLength, hasGrowingButton, focusRowAt, focusCellAt, pageSize);
    default:
      return false;
  }
}
function useTableNavigation(options) {
  const {
    rowCount,
    columnCount,
    hasHeaderRow = true,
    hasGrowingButton = false,
    onNavigate,
    onFocusHeader,
    onFocusGrowing
  } = options;
  const [focusedRowIndex, setFocusedRowIndex] = useState(-1);
  const [focusedCell, setFocusedCell] = useState(null);
  const [isCellContentFocused, setIsCellContentFocused] = useState(false);
  const focusMemoryRef = useRef(null);
  const mouseNavigatingRef = useRef(false);
  const navigateTo = useCallback(
    (rowIndex, columnIndex) => {
      const minRow = hasHeaderRow ? -1 : 0;
      const maxRow = hasGrowingButton ? rowCount : rowCount - 1;
      let targetRow = Math.max(minRow, Math.min(rowIndex, maxRow));
      if (targetRow === -1 && hasHeaderRow) {
        if (columnIndex !== void 0) {
          const col = Math.max(0, Math.min(columnIndex, columnCount - 1));
          setFocusedRowIndex(-1);
          setFocusedCell({ rowIndex: -1, columnIndex: col });
          setIsCellContentFocused(false);
          onNavigate?.(-1, col);
          return;
        }
        setFocusedRowIndex(-1);
        setFocusedCell(null);
        setIsCellContentFocused(false);
        onFocusHeader?.();
        return;
      }
      if (targetRow === rowCount && hasGrowingButton) {
        setFocusedRowIndex(rowCount);
        setFocusedCell(null);
        setIsCellContentFocused(false);
        onFocusGrowing?.();
        return;
      }
      targetRow = Math.max(0, Math.min(targetRow, rowCount - 1));
      if (columnIndex !== void 0) {
        const col = Math.max(0, Math.min(columnIndex, columnCount - 1));
        setFocusedRowIndex(targetRow);
        setFocusedCell({ rowIndex: targetRow, columnIndex: col });
        setIsCellContentFocused(false);
        onNavigate?.(targetRow, col);
      } else {
        setFocusedRowIndex(targetRow);
        setFocusedCell(null);
        setIsCellContentFocused(false);
        onNavigate?.(targetRow);
      }
    },
    [rowCount, columnCount, hasHeaderRow, hasGrowingButton, onNavigate, onFocusHeader, onFocusGrowing]
  );
  const enterCellMode = useCallback(
    (rowIndex, columnIndex = 0) => {
      const col = Math.max(0, Math.min(columnIndex, columnCount - 1));
      setFocusedCell({ rowIndex, columnIndex: col });
      setIsCellContentFocused(false);
      onNavigate?.(rowIndex, col);
    },
    [columnCount, onNavigate]
  );
  const exitCellMode = useCallback(
    (rowIndex) => {
      setFocusedCell(null);
      setIsCellContentFocused(false);
      onNavigate?.(rowIndex);
    },
    [onNavigate]
  );
  return {
    focusedRowIndex,
    focusedCell,
    isCellContentFocused,
    setIsCellContentFocused,
    navigateTo,
    enterCellMode,
    exitCellMode,
    setFocusedRowIndex,
    focusMemoryRef,
    mouseNavigatingRef
  };
}
function isTextInputLike(el) {
  if (!el) return false;
  if (el.isContentEditable) return true;
  const tag = el.tagName.toLowerCase();
  return tag === "textarea" || tag === "input";
}
function useTableDomNavigation({
  tableContainerRef,
  tableRef,
  beforeSentinelRef,
  afterSentinelRef,
  mouseNavigatingRef,
  dataLength,
  hasGrowingButton,
  hasSelectionColumn,
  selectionMode,
  navigationState,
  selectionState,
  getRowKey,
  pageSize = 10
}) {
  const lastFocusedRowRef = useRef(null);
  const lastFocusedCellRef = useRef(null);
  const skipSentinelRedirectRef = useRef(null);
  const rowInteractiveMemoryRef = useRef(0);
  const cellInteractiveMemoryRef = useRef(0);
  const getHeaderRowElement = useCallback(() => {
    return tableRef.current?.querySelector('[data-header-row="true"]') ?? null;
  }, [tableRef]);
  const getBodyRowElement = useCallback((rowIndex) => {
    return tableRef.current?.querySelector(`[data-row-focusable="true"][data-row-index="${rowIndex}"]`) ?? null;
  }, [tableRef]);
  const getGrowingButtonElement = useCallback(() => {
    return tableContainerRef.current?.querySelector("[data-growing-button]") ?? null;
  }, [tableContainerRef]);
  const getRowElement = useCallback((rowIndex) => {
    if (rowIndex === -1) return getHeaderRowElement();
    if (rowIndex >= 0 && rowIndex < dataLength) return getBodyRowElement(rowIndex);
    return null;
  }, [dataLength, getBodyRowElement, getHeaderRowElement]);
  const getFocusableCellsInRow = useCallback((rowEl) => {
    if (!rowEl) return [];
    return Array.from(rowEl.querySelectorAll('[data-cell-focusable="true"]'));
  }, []);
  const getColumnIndicesForRow = useCallback((rowIndex) => {
    const rowEl = getRowElement(rowIndex);
    const cells = getFocusableCellsInRow(rowEl);
    return cells.map((cell) => Number(cell.dataset.columnIndex)).filter((value) => !Number.isNaN(value)).sort((a, b) => a - b);
  }, [getFocusableCellsInRow, getRowElement]);
  const getCellElement = useCallback((rowIndex, columnIndex) => {
    const rowEl = getRowElement(rowIndex);
    const cells = getFocusableCellsInRow(rowEl);
    if (cells.length === 0) return null;
    const exact = cells.find((cell) => Number(cell.dataset.columnIndex) === columnIndex);
    if (exact) return exact;
    return cells[0] ?? null;
  }, [getFocusableCellsInRow, getRowElement]);
  const getInteractiveElements = useCallback((container) => {
    return getTabbableElements(container).filter(
      (el) => !el.matches('[data-row-focusable="true"], [data-cell-focusable="true"]')
    );
  }, []);
  const resolveRowIndexFromElement = useCallback((element) => {
    if (!element) return null;
    if (element.hasAttribute("data-growing-button")) return dataLength;
    const rowEl = element.closest('[data-row-index][data-row-focusable="true"], [data-header-row="true"]');
    if (!rowEl) return null;
    if (rowEl.dataset.headerRow === "true") return -1;
    const rowIndex = Number(rowEl.dataset.rowIndex);
    return Number.isNaN(rowIndex) ? null : rowIndex;
  }, [dataLength]);
  const resolveColumnIndexFromElement = useCallback((element) => {
    if (!element) return null;
    const cellEl = element.closest('[data-cell-focusable="true"]');
    if (!cellEl) return null;
    const colIndex = Number(cellEl.dataset.columnIndex);
    return Number.isNaN(colIndex) ? null : colIndex;
  }, []);
  const focusRowAt = useCallback(
    (rowIndex, options) => {
      const { syncNavigation = true, remember = true, preventScroll = false } = options ?? {};
      if (rowIndex === dataLength && hasGrowingButton) {
        const growingButton = getGrowingButtonElement();
        if (syncNavigation) navigationState.navigateTo(rowIndex);
        if (growingButton && growingButton !== document.activeElement) {
          growingButton.focus({ preventScroll });
        }
        if (remember) lastFocusedCellRef.current = null;
        return Boolean(growingButton);
      }
      const rowEl = getRowElement(rowIndex);
      if (!rowEl) return false;
      if (syncNavigation) navigationState.navigateTo(rowIndex);
      if (rowEl !== document.activeElement) rowEl.focus({ preventScroll });
      const cellEls = rowEl.querySelectorAll('[data-cell-focusable="true"][tabindex]');
      cellEls.forEach((el) => el.removeAttribute("tabindex"));
      if (remember && rowIndex >= -1 && rowIndex < dataLength) {
        lastFocusedRowRef.current = rowIndex;
        lastFocusedCellRef.current = null;
      }
      return true;
    },
    [dataLength, getGrowingButtonElement, getRowElement, hasGrowingButton, navigationState]
  );
  const focusCellAt = useCallback(
    (rowIndex, columnIndex, options) => {
      const { syncNavigation = true, remember = true, preventScroll = false, forceCellFocus = false } = options ?? {};
      const cellEl = getCellElement(rowIndex, columnIndex);
      if (!cellEl) return false;
      const resolvedColumnIndex = Number(cellEl.dataset.columnIndex ?? columnIndex);
      if (syncNavigation) navigationState.navigateTo(rowIndex, resolvedColumnIndex);
      const activeEl = document.activeElement;
      const activeIsInteractiveInsideTarget = Boolean(activeEl) && activeEl !== cellEl && cellEl.contains(activeEl);
      if ((forceCellFocus || !activeIsInteractiveInsideTarget) && cellEl !== activeEl) {
        if (!cellEl.hasAttribute("tabindex")) {
          cellEl.setAttribute("tabindex", "-1");
        }
        cellEl.focus({ preventScroll });
      }
      if (remember && rowIndex >= -1 && rowIndex < dataLength) {
        lastFocusedRowRef.current = rowIndex;
        lastFocusedCellRef.current = { rowIndex, columnIndex: resolvedColumnIndex };
      }
      return true;
    },
    [dataLength, getCellElement, navigationState]
  );
  const focusRememberedOrFirstRow = useCallback(() => {
    const remembered = lastFocusedRowRef.current;
    if (typeof remembered === "number" && remembered >= -1 && remembered < dataLength) {
      if (focusRowAt(remembered)) return;
    }
    if (dataLength > 0) {
      focusRowAt(0);
      return;
    }
    focusRowAt(-1);
  }, [dataLength, focusRowAt]);
  const focusInteractiveInRow = useCallback((rowIndex, preferredIndex) => {
    const rowEl = getRowElement(rowIndex);
    const interactive = getInteractiveElements(rowEl);
    if (interactive.length === 0) return false;
    const targetIndex = Math.max(0, Math.min(preferredIndex, interactive.length - 1));
    rowInteractiveMemoryRef.current = targetIndex;
    interactive[targetIndex]?.focus();
    return true;
  }, [getInteractiveElements, getRowElement]);
  const focusInteractiveInCell = useCallback((rowIndex, columnIndex, preferredIndex) => {
    const cellEl = getCellElement(rowIndex, columnIndex);
    const interactive = getInteractiveElements(cellEl);
    if (interactive.length === 0) return false;
    const targetIndex = Math.max(0, Math.min(preferredIndex, interactive.length - 1));
    cellInteractiveMemoryRef.current = targetIndex;
    interactive[targetIndex]?.focus();
    return true;
  }, [getCellElement, getInteractiveElements]);
  const focusMatchingInteractiveInCell = useCallback(
    (rowIndex, columnIndex, sourceElement) => {
      const cellEl = getCellElement(rowIndex, columnIndex);
      if (!cellEl) return false;
      const sourceRole = sourceElement.getAttribute("role");
      if (sourceRole) {
        const roleMatch = cellEl.querySelector(`[role="${sourceRole}"]`);
        if (roleMatch) {
          roleMatch.focus();
          return true;
        }
      }
      const tagMatch = cellEl.querySelector(sourceElement.tagName.toLowerCase());
      if (tagMatch) {
        tagMatch.focus();
        return true;
      }
      return false;
    },
    [getCellElement]
  );
  const rememberInteractivePosition = useCallback((target) => {
    const rowEl = target.closest('[data-row-index][data-row-focusable="true"], [data-header-row="true"]');
    if (rowEl) {
      const rowInteractive = getInteractiveElements(rowEl);
      const rowIdx = rowInteractive.findIndex((el) => el === target || el.contains(target));
      if (rowIdx >= 0) rowInteractiveMemoryRef.current = rowIdx;
    }
    const cellEl = target.closest('[data-cell-focusable="true"]');
    if (cellEl) {
      const cellInteractive = getInteractiveElements(cellEl);
      const cellIdx = cellInteractive.findIndex((el) => el === target || el.contains(target));
      if (cellIdx >= 0) cellInteractiveMemoryRef.current = cellIdx;
    }
  }, [getInteractiveElements]);
  const forwardFocusOutOfTable = useCallback((shiftKey) => {
    const currentRow = navigationState.focusedCell?.rowIndex ?? navigationState.focusedRowIndex;
    if (currentRow >= -1 && currentRow < dataLength) {
      lastFocusedRowRef.current = currentRow;
    }
    if (navigationState.focusedCell) {
      lastFocusedCellRef.current = navigationState.focusedCell;
    }
    if (shiftKey) {
      skipSentinelRedirectRef.current = "before";
      beforeSentinelRef.current?.focus();
      return;
    }
    skipSentinelRedirectRef.current = "after";
    afterSentinelRef.current?.focus();
  }, [dataLength, navigationState.focusedCell, navigationState.focusedRowIndex, beforeSentinelRef, afterSentinelRef]);
  const handleBeforeSentinelFocus = useCallback(() => {
    if (skipSentinelRedirectRef.current === "before") {
      skipSentinelRedirectRef.current = null;
      return true;
    }
    focusRememberedOrFirstRow();
    return false;
  }, [focusRememberedOrFirstRow]);
  const handleAfterSentinelFocus = useCallback(() => {
    if (skipSentinelRedirectRef.current === "after") {
      skipSentinelRedirectRef.current = null;
      return true;
    }
    focusRememberedOrFirstRow();
    return false;
  }, [focusRememberedOrFirstRow]);
  const applyKeyboardRangeBetweenRows = useCallback((sourceRow, destinationRow) => {
    if (selectionMode !== "Multi") return;
    if (sourceRow < 0 || sourceRow >= dataLength) return;
    if (destinationRow < 0 || destinationRow >= dataLength) return;
    const sourceKey = getRowKey(sourceRow);
    const destinationKey = getRowKey(destinationRow);
    if (sourceKey && destinationKey) {
      selectionState.applyKeyboardRange(sourceKey, destinationKey);
    }
  }, [dataLength, getRowKey, selectionMode, selectionState]);
  const applyRowModeShiftRangeSelection = useCallback((sourceRow, destinationRow) => {
    if (selectionMode !== "Multi" || dataLength === 0) return;
    if (destinationRow < 0 || destinationRow >= dataLength) return;
    let normalizedSourceRow;
    if (sourceRow === dataLength) {
      normalizedSourceRow = dataLength - 1;
    } else if (sourceRow < 0) {
      normalizedSourceRow = destinationRow;
    } else {
      normalizedSourceRow = sourceRow;
    }
    const anchorKey = selectionState.getLastSelectedKey() ?? getRowKey(normalizedSourceRow);
    const endKey = getRowKey(destinationRow);
    if (anchorKey && endKey) {
      selectionState.selectRange(anchorKey, endKey);
    }
  }, [dataLength, getRowKey, selectionMode, selectionState]);
  const handleInteractiveInCellArrow = useCallback((e, target, currentRow, currentCol) => {
    if (isTextInputLike(target) && !(e.ctrlKey || e.metaKey)) return false;
    e.preventDefault();
    rememberInteractivePosition(target);
    const direction = e.key === "ArrowDown" ? 1 : -1;
    const destinationRow = currentRow + direction;
    if (e.shiftKey && selectionMode === "Multi" && currentRow >= 0 && currentRow < dataLength && destinationRow >= 0 && destinationRow < dataLength) {
      applyKeyboardRangeBetweenRows(currentRow, destinationRow);
    }
    if (destinationRow < -1) {
      focusCellAt(-1, currentCol);
      return true;
    }
    if (destinationRow > dataLength - 1) {
      if (hasGrowingButton) focusRowAt(dataLength);
      return true;
    }
    if (!focusInteractiveInCell(destinationRow, currentCol, cellInteractiveMemoryRef.current) && !focusMatchingInteractiveInCell(destinationRow, currentCol, target)) {
      focusCellAt(destinationRow, currentCol);
    }
    return true;
  }, [
    applyKeyboardRangeBetweenRows,
    dataLength,
    focusCellAt,
    focusInteractiveInCell,
    focusMatchingInteractiveInCell,
    focusRowAt,
    hasGrowingButton,
    rememberInteractivePosition,
    selectionMode
  ]);
  const handleF2Key = useCallback((e, currentRow, currentCol, targetIsInteractiveInCell, targetIsInteractiveInRow, targetIsCell, targetIsRow) => {
    if (targetIsInteractiveInCell) {
      e.preventDefault();
      focusCellAt(currentRow, currentCol, { forceCellFocus: true });
      return;
    }
    if (targetIsInteractiveInRow) {
      e.preventDefault();
      focusRowAt(currentRow);
      return;
    }
    if (targetIsCell) {
      e.preventDefault();
      focusInteractiveInCell(currentRow, currentCol, 0);
      return;
    }
    if (targetIsRow && currentRow >= 0 && currentRow < dataLength) {
      e.preventDefault();
      focusInteractiveInRow(currentRow, 0);
    }
  }, [dataLength, focusCellAt, focusInteractiveInCell, focusInteractiveInRow, focusRowAt]);
  const handleF7Key = useCallback((e, target, currentRow, currentCol, targetIsInteractiveInCell, targetIsInteractiveInRow, targetIsCell, targetIsRow) => {
    if (targetIsInteractiveInCell || targetIsInteractiveInRow) {
      e.preventDefault();
      rememberInteractivePosition(target);
      focusRowAt(currentRow);
      return;
    }
    if (targetIsCell) {
      e.preventDefault();
      focusInteractiveInCell(currentRow, currentCol, cellInteractiveMemoryRef.current);
      return;
    }
    if (targetIsRow && currentRow >= 0 && currentRow < dataLength) {
      e.preventDefault();
      focusInteractiveInRow(currentRow, rowInteractiveMemoryRef.current);
    }
  }, [dataLength, focusInteractiveInCell, focusInteractiveInRow, focusRowAt, rememberInteractivePosition]);
  const handleSelectAllToggle = useCallback(() => {
    if (selectionState.areAllSelected()) {
      selectionState.deselectAll();
    } else {
      selectionState.selectAll();
    }
  }, [selectionState]);
  const handleSelectionKey = useCallback((e, isCellMode, currentRow) => {
    if ((e.key === "a" || e.key === "A") && (e.ctrlKey || e.metaKey) && selectionMode === "Multi") {
      e.preventDefault();
      handleSelectAllToggle();
      return true;
    }
    if (e.key === " " && !isCellMode && selectionMode !== "None") {
      if (currentRow === -1) {
        e.preventDefault();
        handleSelectAllToggle();
        return true;
      }
      if (currentRow >= 0 && currentRow < dataLength) {
        e.preventDefault();
        const key = getRowKey(currentRow);
        if (key) selectionState.toggleSelection(key);
        return true;
      }
    }
    return false;
  }, [dataLength, getRowKey, handleSelectAllToggle, selectionMode, selectionState]);
  const handleFocusCapture = useCallback((e) => {
    const target = e.target;
    if (!tableContainerRef.current?.contains(target)) return;
    const rowIndex = resolveRowIndexFromElement(target);
    if (rowIndex !== null && rowIndex >= -1 && rowIndex < dataLength) {
      lastFocusedRowRef.current = rowIndex;
    }
    if (target.matches('[data-row-focusable="true"]')) {
      if (rowIndex !== null) navigationState.navigateTo(rowIndex);
      return;
    }
    if (target.matches('[data-cell-focusable="true"]')) {
      const colIndex = resolveColumnIndexFromElement(target);
      if (rowIndex !== null && colIndex !== null) {
        navigationState.navigateTo(rowIndex, colIndex);
        if (rowIndex >= -1 && rowIndex < dataLength) {
          lastFocusedCellRef.current = { rowIndex, columnIndex: colIndex };
        }
      }
      return;
    }
    if (target.hasAttribute("data-growing-button")) {
      navigationState.navigateTo(dataLength);
      return;
    }
    rememberInteractivePosition(target);
  }, [
    dataLength,
    navigationState,
    rememberInteractivePosition,
    resolveColumnIndexFromElement,
    resolveRowIndexFromElement,
    tableContainerRef
  ]);
  const resolveKeydownContext = useCallback((e) => {
    const target = e.target ?? document.activeElement;
    if (!target || !tableContainerRef.current?.contains(target)) return null;
    if (target.closest("[role='status']")) return null;
    const rowEl = target.closest('[data-row-index][data-row-focusable="true"], [data-header-row="true"]');
    const cellEl = target.closest('[data-cell-focusable="true"]');
    const targetIsRow = target.matches('[data-row-focusable="true"]') || target.hasAttribute("data-growing-button");
    const targetIsCell = target.matches('[data-cell-focusable="true"]');
    const targetIsInteractiveInRow = Boolean(rowEl) && !targetIsRow && !targetIsCell;
    const targetIsInteractiveInCell = Boolean(cellEl) && !targetIsCell;
    const rowIndexFromDom = resolveRowIndexFromElement(target);
    const colIndexFromDom = resolveColumnIndexFromElement(target);
    const stateRow = navigationState.focusedCell?.rowIndex ?? navigationState.focusedRowIndex;
    const stateCol = navigationState.focusedCell?.columnIndex ?? 0;
    return {
      target,
      currentRow: rowIndexFromDom ?? stateRow,
      currentCol: colIndexFromDom ?? stateCol,
      isCellMode: targetIsCell || !targetIsRow && !targetIsInteractiveInRow && navigationState.focusedCell !== null,
      targetIsRow,
      targetIsCell,
      targetIsInteractiveInRow,
      targetIsInteractiveInCell
    };
  }, [navigationState.focusedCell, navigationState.focusedRowIndex, resolveColumnIndexFromElement, resolveRowIndexFromElement, tableContainerRef]);
  const handleSelectionCellKey = useCallback((e, currentRow) => {
    if (selectionMode !== "None" && currentRow >= 0 && currentRow < dataLength) {
      e.preventDefault();
      const key = getRowKey(currentRow);
      if (key) selectionState.toggleSelection(key);
      return true;
    }
    if (currentRow === -1 && selectionMode === "Multi") {
      e.preventDefault();
      handleSelectAllToggle();
      return true;
    }
    return false;
  }, [dataLength, getRowKey, handleSelectAllToggle, selectionMode, selectionState]);
  const handleEscapeKey = useCallback((e, currentRow, currentCol, targetIsInteractiveInCell, targetIsCell) => {
    if (targetIsInteractiveInCell) {
      e.preventDefault();
      focusCellAt(currentRow, currentCol, { forceCellFocus: true });
      return true;
    }
    if (targetIsCell) {
      e.preventDefault();
      focusRowAt(currentRow);
      return true;
    }
    return false;
  }, [focusCellAt, focusRowAt]);
  const handleKeyDown = useCallback((e) => {
    if (e.defaultPrevented) return;
    const ctx = resolveKeydownContext(e);
    if (!ctx) return;
    const { target, currentRow, currentCol, isCellMode, targetIsRow, targetIsCell, targetIsInteractiveInRow, targetIsInteractiveInCell } = ctx;
    const isRtl = tableRef.current ? window.getComputedStyle(tableRef.current).direction === "rtl" : document?.documentElement?.dir === "rtl";
    const horizontalForwardKey = isRtl ? "ArrowLeft" : "ArrowRight";
    const horizontalBackwardKey = isRtl ? "ArrowRight" : "ArrowLeft";
    if ((targetIsRow || targetIsCell) && e.key === "Tab") {
      forwardFocusOutOfTable(e.shiftKey);
      return;
    }
    if (targetIsInteractiveInCell && (e.key === "ArrowDown" || e.key === "ArrowUp")) {
      handleInteractiveInCellArrow(e, target, currentRow, currentCol);
      return;
    }
    if (e.key === "F2") {
      handleF2Key(e, currentRow, currentCol, targetIsInteractiveInCell, targetIsInteractiveInRow, targetIsCell, targetIsRow);
      return;
    }
    if (e.key === "F7") {
      handleF7Key(e, target, currentRow, currentCol, targetIsInteractiveInCell, targetIsInteractiveInRow, targetIsCell, targetIsRow);
      return;
    }
    if ((e.key === " " || e.key === "Enter") && targetIsCell && hasSelectionColumn && currentCol === 0) {
      if (handleSelectionCellKey(e, currentRow)) return;
    }
    if (e.key === "Enter" && targetIsCell) {
      e.preventDefault();
      focusInteractiveInCell(currentRow, currentCol, 0);
      return;
    }
    if (e.key === "Escape") {
      if (handleEscapeKey(e, currentRow, currentCol, targetIsInteractiveInCell, targetIsCell)) return;
    }
    if (targetIsInteractiveInRow) return;
    if (handleSelectionKey(e, isCellMode, currentRow)) return;
    handleTableKeyboardNavigation({
      event: e,
      isCellMode,
      currentRow,
      currentCol,
      dataLength,
      hasGrowingButton,
      horizontalForwardKey,
      horizontalBackwardKey,
      columnIndices: isCellMode ? getColumnIndicesForRow(currentRow) : [],
      focusRowAt,
      focusCellAt,
      onRowShiftRange: selectionMode === "Multi" ? applyRowModeShiftRangeSelection : void 0,
      onCellShiftRange: selectionMode === "Multi" ? applyKeyboardRangeBetweenRows : void 0,
      pageSize
    });
  }, [
    applyKeyboardRangeBetweenRows,
    applyRowModeShiftRangeSelection,
    dataLength,
    focusCellAt,
    focusInteractiveInCell,
    focusRowAt,
    forwardFocusOutOfTable,
    getColumnIndicesForRow,
    handleF2Key,
    handleF7Key,
    handleEscapeKey,
    handleInteractiveInCellArrow,
    handleSelectionCellKey,
    handleSelectionKey,
    hasGrowingButton,
    hasSelectionColumn,
    pageSize,
    resolveKeydownContext,
    selectionMode,
    tableRef
  ]);
  const handleKeyUp = useCallback((e) => {
    if (e.key === "Shift") {
      selectionState.clearRangeSession({ clearAnchor: true });
    }
  }, [selectionState]);
  const prevFocusSnapshotRef = useRef(`row:${navigationState.focusedRowIndex}`);
  useEffect(() => {
    const cell = navigationState.focusedCell;
    const snapshot = cell ? `cell:${cell.rowIndex}:${cell.columnIndex}` : `row:${navigationState.focusedRowIndex}`;
    if (snapshot === prevFocusSnapshotRef.current) return;
    prevFocusSnapshotRef.current = snapshot;
    if (mouseNavigatingRef.current) {
      mouseNavigatingRef.current = false;
      return;
    }
    if (cell) {
      focusCellAt(cell.rowIndex, cell.columnIndex, { syncNavigation: false, remember: false });
      return;
    }
    focusRowAt(navigationState.focusedRowIndex, { syncNavigation: false, remember: false });
  }, [navigationState.focusedCell, navigationState.focusedRowIndex, focusCellAt, focusRowAt, mouseNavigatingRef]);
  return {
    focusRowAt,
    focusCellAt,
    focusRememberedOrFirstRow,
    handleKeyDown,
    handleKeyUp,
    handleFocusCapture,
    handleBeforeSentinelFocus,
    handleAfterSentinelFocus,
    lastFocusedCellRef
  };
}
function parseWidth(width) {
  if (!width) return 150;
  const num = parseFloat(width);
  return isNaN(num) ? 150 : num;
}
function calculatePopin(columnsMeta, availableWidth, reservedWidth) {
  if (columnsMeta.length === 0) return { visible: [], popin: [] };
  const spaceForColumns = availableWidth - reservedWidth;
  if (spaceForColumns <= 0) {
    const sorted = columnsMeta.map((col, idx) => ({ col, idx })).sort((a, b) => b.col.importance - a.col.importance);
    return {
      visible: [sorted[0].idx],
      popin: sorted.slice(1).filter(({ col }) => !col.popinHidden).map(({ idx }) => idx)
    };
  }
  const indexed = columnsMeta.map((col, idx) => ({ col, idx }));
  const byImportance = [...indexed].sort((a, b) => b.col.importance - a.col.importance);
  let usedWidth = 0;
  const visible = [];
  const popin = [];
  let overflowed = false;
  for (const { col, idx } of byImportance) {
    const colWidth = parseWidth(col.width || col.minWidth);
    if (!overflowed && usedWidth + colWidth <= spaceForColumns) {
      visible.push(idx);
      usedWidth += colWidth;
    } else {
      overflowed = true;
      if (!col.popinHidden) {
        popin.push(idx);
      }
    }
  }
  visible.sort((a, b) => a - b);
  popin.sort((a, b) => a - b);
  if (visible.length === 0 && columnsMeta.length > 0) {
    const mostImportant = byImportance[0].idx;
    visible.push(mostImportant);
    const popinIdx = popin.indexOf(mostImportant);
    if (popinIdx !== -1) popin.splice(popinIdx, 1);
  }
  return { visible, popin };
}
function useTablePopin(options) {
  const { columnsMeta, enabled, containerRef, reservedWidth = 0 } = options;
  const [containerWidth, setContainerWidth] = useState(0);
  const observerRef = useRef(null);
  const allIndices = columnsMeta.map((_col, idx) => idx);
  const recalculate = useCallback(() => {
    if (containerRef.current) {
      setContainerWidth(containerRef.current.clientWidth);
    }
  }, [containerRef]);
  useLayoutEffect(() => {
    if (enabled && containerRef.current) {
      setContainerWidth(containerRef.current.clientWidth);
    }
  }, [enabled, containerRef]);
  useEffect(() => {
    if (!enabled || !containerRef.current) return;
    const observer = new ResizeObserver((entries) => {
      const entry = entries[0];
      if (entry) {
        setContainerWidth(entry.contentRect.width);
      }
    });
    observer.observe(containerRef.current);
    observerRef.current = observer;
    return () => {
      observer.disconnect();
      observerRef.current = null;
    };
  }, [enabled, containerRef]);
  if (!enabled || containerWidth === 0) {
    return {
      visibleIndices: allIndices,
      popinIndices: [],
      isPopinActive: false,
      containerWidth,
      recalculate
    };
  }
  const { visible, popin } = calculatePopin(columnsMeta, containerWidth, reservedWidth);
  return {
    visibleIndices: visible,
    popinIndices: popin,
    isPopinActive: popin.length > 0,
    containerWidth,
    recalculate
  };
}
function TableInner({
  children,
  accessibleName,
  accessibleNameRef,
  noDataText,
  noData,
  overflowMode = "Popin",
  scrollHeight,
  loading = false,
  loadingDelay = 1e3,
  rowActionCount = 0,
  alternateRowColors = false,
  stickyTop = "0px",
  onRowClick,
  onMoveOver: _onMoveOver,
  onMove: _onMove,
  onRowActionClick,
  className,
  style,
  id,
  "data-testid": dataTestId
}, ref) {
  const generatedId = useId();
  const tableId = id ?? generatedId;
  const gridRef = useRef(null);
  const tableWrapperRef = useRef(null);
  const beforeSentinelRef = useRef(null);
  const afterSentinelRef = useRef(null);
  const loadingRef = useRef(null);
  const noDataRef = useRef(null);
  const skipLoadingRedirectRef = useRef(false);
  const skipNoDataRedirectRef = useRef(false);
  const loadingHadFocusRef = useRef(false);
  const lastFocusWasNoDataRef = useRef(null);
  const prevHadGrowingRef = useRef(false);
  const prevRowCountRef = useRef(0);
  const scanned = useMemo(() => scanTableChildren(children), [children]);
  const { toolbar, headerRow, dataRows, selectionFeature, growingFeature } = scanned;
  const columnsMeta = useMemo(() => extractColumnsMeta(headerRow), [headerRow]);
  const rowKeys = useMemo(() => extractRowKeys(dataRows), [dataRows]);
  const selectionMode = useMemo(() => {
    if (!selectionFeature) return "None";
    const role = selectionFeature.type?._tableRole;
    return role === "SelectionMulti" ? "Multi" : "Single";
  }, [selectionFeature]);
  const selectionProps = useMemo(() => {
    if (!selectionFeature) return {};
    return selectionFeature.props;
  }, [selectionFeature]);
  const selectionBehavior = selectionProps.behavior ?? "RowSelector";
  const headerSelector = selectionProps.headerSelector ?? "SelectAll";
  const hasSelectionColumn = selectionMode !== "None" && selectionBehavior === "RowSelector";
  const hasActionsColumn = rowActionCount > 0;
  const hasNavigatedColumn = useMemo(
    () => dataRows.some((row) => row.props.navigated === true),
    [dataRows]
  );
  const selection = useTableSelection({
    mode: selectionMode,
    selected: selectionProps.selected,
    defaultSelected: selectionProps.defaultSelected,
    allRowKeys: rowKeys,
    onChange: selectionProps.onChange
  });
  const { t } = useTranslation("fx");
  const [selectionAnnouncement, setSelectionAnnouncement] = useState("");
  const selectionCountRef = useRef(selection.selectedKeys.size);
  useEffect(() => {
    const count = selection.selectedKeys.size;
    if (count !== selectionCountRef.current) {
      selectionCountRef.current = count;
      if (selectionMode !== "None" && count > 0) {
        setSelectionAnnouncement(t("TABLE_ROWS_SELECTED", { count }));
      } else {
        setSelectionAnnouncement("");
      }
    }
  }, [selection.selectedKeys, selectionMode, t]);
  const actionsReserved = hasActionsColumn ? rowActionCount * 32 + Math.max(0, rowActionCount - 1) * 4 + 16 : 0;
  const popin = useTablePopin({
    columnsMeta,
    enabled: overflowMode === "Popin",
    containerRef: tableWrapperRef,
    reservedWidth: (hasSelectionColumn ? 40 : 0) + actionsReserved + (hasNavigatedColumn ? 4 : 0)
  });
  const totalGridCols = popin.visibleIndices.length + (hasSelectionColumn ? 1 : 0) + (hasActionsColumn ? 1 : 0);
  const popinColumnIndex = totalGridCols;
  const navColumnCount = totalGridCols + (popin.isPopinActive ? 1 : 0);
  const gridTemplateColumns = useMemo(
    () => buildGridTemplateColumns(columnsMeta, popin.visibleIndices, hasSelectionColumn, hasActionsColumn, hasNavigatedColumn),
    [columnsMeta, popin.visibleIndices, hasSelectionColumn, hasActionsColumn, hasNavigatedColumn]
  );
  const getRowIndexByKey = useCallback(
    (rowKey) => rowKeys.indexOf(rowKey),
    [rowKeys]
  );
  const getRowKey = useCallback(
    (rowIndex) => rowKeys[rowIndex] ?? null,
    [rowKeys]
  );
  const hasGrowingButton = !!growingFeature;
  const isGridScrollContainer = overflowMode === "Scroll" && !!scrollHeight;
  const hasToolbar = !!toolbar;
  const isToolbarSticky = hasToolbar && !!toolbar.props?.sticky;
  const isHeaderSticky = !!headerRow && !!headerRow.props?.sticky;
  const TOOLBAR_HEIGHT = 48;
  const toolbarTop = stickyTop;
  const headerRowTopWhenNotScrollContainer = isToolbarSticky && isHeaderSticky ? `calc(${stickyTop} + ${TOOLBAR_HEIGHT}px)` : stickyTop;
  const headerRowTop = isGridScrollContainer ? "0px" : headerRowTopWhenNotScrollContainer;
  const navigationState = useTableNavigation({
    rowCount: dataRows.length,
    columnCount: navColumnCount,
    hasHeaderRow: !!headerRow,
    hasGrowingButton
  });
  const domNavigation = useTableDomNavigation({
    tableContainerRef: tableWrapperRef,
    tableRef: gridRef,
    beforeSentinelRef,
    afterSentinelRef,
    mouseNavigatingRef: navigationState.mouseNavigatingRef,
    dataLength: dataRows.length,
    hasGrowingButton,
    hasSelectionColumn,
    selectionMode,
    navigationState,
    selectionState: {
      toggleSelection: selection.toggleSelection,
      selectAll: selection.selectAll,
      deselectAll: selection.deselectAll,
      areAllSelected: selection.areAllSelected,
      applyKeyboardRange: selection.applyKeyboardRange,
      selectRange: selection.selectRange,
      getLastSelectedKey: selection.getLastSelectedKey,
      clearRangeSession: selection.clearRangeSession
    },
    getRowKey,
    pageSize: 10
  });
  const ariaRowCount = dataRows.length + 1;
  const ariaColCount = navColumnCount;
  const [showLoading, setShowLoading] = useState(loading && loadingDelay === 0);
  useEffect(() => {
    if (!loading) {
      setShowLoading(false);
      return;
    }
    if (loadingDelay === 0) {
      setShowLoading(true);
      return;
    }
    const timer = setTimeout(() => setShowLoading(true), loadingDelay);
    return () => clearTimeout(timer);
  }, [loading, loadingDelay]);
  const isEmpty = dataRows.length === 0 && !loading;
  const handleBeforeSentinelFocus = useCallback(() => {
    if (skipLoadingRedirectRef.current) {
      skipLoadingRedirectRef.current = false;
      return;
    }
    if (skipNoDataRedirectRef.current) {
      skipNoDataRedirectRef.current = false;
      return;
    }
    if (domNavigation.handleBeforeSentinelFocus()) return;
    if (showLoading) {
      loadingRef.current?.focus();
      return;
    }
    if (isEmpty && lastFocusWasNoDataRef.current !== false) {
      noDataRef.current?.focus();
    }
  }, [showLoading, isEmpty, domNavigation]);
  const handleAfterSentinelFocus = useCallback(() => {
    if (skipLoadingRedirectRef.current) {
      skipLoadingRedirectRef.current = false;
      return;
    }
    if (skipNoDataRedirectRef.current) {
      skipNoDataRedirectRef.current = false;
      return;
    }
    if (domNavigation.handleAfterSentinelFocus()) return;
    if (showLoading) {
      loadingRef.current?.focus();
      return;
    }
    if (isEmpty && lastFocusWasNoDataRef.current !== false) {
      noDataRef.current?.focus();
    }
  }, [showLoading, isEmpty, domNavigation]);
  const handleLoadingKeyDown = useCallback((e) => {
    if (e.key !== "Tab") return;
    skipLoadingRedirectRef.current = true;
    if (e.shiftKey) {
      beforeSentinelRef.current?.focus();
    } else {
      afterSentinelRef.current?.focus();
    }
  }, []);
  const handleNoDataKeyDown = useCallback((e) => {
    if (e.key === "Tab") {
      skipNoDataRedirectRef.current = true;
      if (e.shiftKey) {
        beforeSentinelRef.current?.focus();
      } else {
        afterSentinelRef.current?.focus();
      }
      return;
    }
    if (e.key === "ArrowUp") {
      e.preventDefault();
      lastFocusWasNoDataRef.current = false;
      domNavigation.focusRowAt(-1);
    }
  }, [domNavigation]);
  useEffect(() => {
    if (showLoading) {
      if (tableWrapperRef.current?.contains(document.activeElement)) {
        loadingRef.current?.focus();
      }
    } else {
      if (loadingHadFocusRef.current) {
        loadingHadFocusRef.current = false;
        if (isEmpty) {
          noDataRef.current?.focus();
        } else {
          domNavigation.focusRememberedOrFirstRow();
        }
      }
    }
  }, [showLoading, isEmpty, domNavigation]);
  useEffect(() => {
    const prevHadGrowing = prevHadGrowingRef.current;
    const prevCount = prevRowCountRef.current;
    prevHadGrowingRef.current = hasGrowingButton;
    prevRowCountRef.current = dataRows.length;
    if (prevHadGrowing && !hasGrowingButton && dataRows.length > prevCount && prevCount > 0) {
      const activeEl = document.activeElement;
      if (!activeEl || activeEl === document.body || !tableWrapperRef.current?.contains(activeEl)) {
        domNavigation.focusRowAt(prevCount);
      }
    }
  }, [hasGrowingButton, dataRows.length, domNavigation]);
  useImperativeHandle(ref, () => ({
    get tableElement() {
      return gridRef.current;
    },
    get scrollContainer() {
      return tableWrapperRef.current;
    },
    focus() {
      if (showLoading && loadingRef.current) {
        loadingRef.current.focus();
        return;
      }
      if (noDataRef.current && !gridRef.current?.querySelector("[data-row-key]")) {
        noDataRef.current.focus();
        return;
      }
      domNavigation.focusRememberedOrFirstRow();
    }
  }), [showLoading, domNavigation]);
  const selectionRef = selectionProps.ref;
  useImperativeHandle(selectionRef ?? null, () => {
    if (selectionMode === "None") return {};
    if (selectionMode === "Multi") {
      return {
        getSelectedRows: () => {
          const grid = gridRef.current;
          if (!grid) return [];
          return [...selection.selectedKeys].map((key) => grid.querySelector(`[data-row-key="${key}"]`)).filter(Boolean);
        },
        areAllRowsSelected: () => selection.areAllSelected(),
        getSelectedAsSet: () => new Set(selection.selectedKeys),
        setSelectedAsSet: (set) => {
          selection.setSelectedAsSet(set);
        },
        isSelected: (rowKey) => selection.isSelected(rowKey),
        setSelected: (rowKey, selected) => selection.setSelected(rowKey, selected)
      };
    }
    return {
      getSelectedRow: () => {
        const grid = gridRef.current;
        if (!grid) return void 0;
        const key = [...selection.selectedKeys][0];
        if (!key) return void 0;
        return grid.querySelector(`[data-row-key="${key}"]`);
      },
      isSelected: (rowKey) => selection.isSelected(rowKey),
      setSelected: (rowKey, selected) => selection.setSelected(rowKey, selected)
    };
  }, [selectionMode, selection]);
  const contextValue = useMemo(() => ({
    tableId,
    columnCount: columnsMeta.length,
    columnsMeta,
    gridTemplateColumns,
    selectionMode,
    selectionBehavior,
    headerSelector,
    isSelected: selection.isSelected,
    toggleSelection: selection.toggleSelection,
    selectRange: selection.selectRange,
    applyKeyboardRange: selection.applyKeyboardRange,
    clearRangeSession: selection.clearRangeSession,
    selectedKeys: selection.selectedKeys,
    areAllSelected: selection.areAllSelected,
    selectAll: selection.selectAll,
    deselectAll: selection.deselectAll,
    getLastSelectedKey: selection.getLastSelectedKey,
    focusedRowIndex: navigationState.focusedRowIndex,
    focusedCellIndex: navigationState.focusedCell?.columnIndex ?? null,
    focusedCell: navigationState.focusedCell,
    navigationMode: navigationState.focusedCell ? "Cell" : "Row",
    setFocusedRow: (index) => {
      domNavigation.focusRowAt(index);
    },
    setFocusedCell: (rowIndex, cellIndex) => {
      if (cellIndex !== null) {
        domNavigation.focusCellAt(rowIndex, cellIndex);
      } else {
        domNavigation.focusRowAt(rowIndex);
      }
    },
    rowActionCount,
    hasSelectionColumn,
    hasActionsColumn,
    overflowMode,
    rowKeys,
    getRowIndexByKey,
    visibleColumnIndices: popin.visibleIndices,
    popinColumnIndices: popin.popinIndices,
    isPopinActive: popin.isPopinActive,
    popinColumnIndex,
    onRowClick,
    onRowActionClick,
    setMouseNavigating: () => {
      navigationState.mouseNavigatingRef.current = true;
    },
    navigateToRow: (rowIndex) => {
      navigationState.navigateTo(rowIndex);
    },
    dataRowCount: dataRows.length,
    alternateRowColors,
    hasNavigatedColumn,
    hasGrowingButton,
    hasToolbar
  }), [
    tableId,
    columnsMeta,
    gridTemplateColumns,
    selectionMode,
    selectionBehavior,
    headerSelector,
    selection,
    navigationState,
    rowActionCount,
    hasSelectionColumn,
    hasActionsColumn,
    overflowMode,
    rowKeys,
    getRowIndexByKey,
    popin,
    popinColumnIndex,
    onRowClick,
    onRowActionClick,
    domNavigation,
    dataRows.length,
    alternateRowColors,
    hasNavigatedColumn,
    hasGrowingButton,
    hasToolbar
  ]);
  const handleKeyDownCapture = useCallback((e) => {
    if (isEmpty && e.key === "ArrowDown") {
      const target = e.target;
      if (target.closest("[data-header-row]")) {
        e.preventDefault();
        noDataRef.current?.focus();
        return;
      }
    }
    domNavigation.handleKeyDown(e);
  }, [isEmpty, domNavigation]);
  const selectionDescription = selectionMode === "Multi" ? t("TABLE_MULTI_SELECTABLE") : selectionMode === "Single" ? t("TABLE_SINGLE_SELECTABLE") : void 0;
  return /* @__PURE__ */ jsx(TableContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxs(
    "div",
    {
      ref: tableWrapperRef,
      className: cn("relative overflow-x-clip flex flex-col border border-border rounded-lg", className),
      style,
      "data-testid": dataTestId,
      onKeyDownCapture: handleKeyDownCapture,
      onKeyUpCapture: domNavigation.handleKeyUp,
      onFocusCapture: domNavigation.handleFocusCapture,
      children: [
        toolbar && React.cloneElement(toolbar, {
          style: {
            ...toolbar.props?.style,
            ...isToolbarSticky ? { top: toolbarTop } : {}
          }
        }),
        /* @__PURE__ */ jsx(
          "div",
          {
            ref: beforeSentinelRef,
            tabIndex: 0,
            onFocus: handleBeforeSentinelFocus,
            "aria-hidden": "true",
            className: "absolute w-0 h-0 overflow-hidden"
          }
        ),
        /* @__PURE__ */ jsxs(
          "div",
          {
            ref: gridRef,
            id: tableId,
            role: "grid",
            "aria-rowcount": ariaRowCount,
            "aria-colcount": ariaColCount,
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-busy": loading || void 0,
            "aria-multiselectable": selectionMode === "Multi" || void 0,
            "aria-description": selectionDescription,
            onMouseDown: () => {
              navigationState.mouseNavigatingRef.current = true;
            },
            style: {
              display: "grid",
              gridTemplateColumns,
              ...overflowMode === "Scroll" && scrollHeight ? { height: scrollHeight } : {}
            },
            className: cn(
              "w-full text-sm [&>[data-row-key]:last-child]:border-b-0 content-start",
              isGridScrollContainer && "[&>[data-row-key]:has(+:not([data-row-key]))]:border-b-0",
              !hasGrowingButton && "[&>[data-row-key]:last-child]:rounded-b-lg",
              overflowMode === "Scroll" && "overflow-x-auto",
              overflowMode === "Scroll" && scrollHeight && "overflow-y-auto"
            ),
            children: [
              headerRow && React.cloneElement(headerRow, {
                style: {
                  ...headerRow.props?.style,
                  ...headerRow.props?.sticky ? { top: headerRowTop } : {}
                }
              }),
              dataRows.map((row) => row),
              isGridScrollContainer && growingFeature
            ]
          }
        ),
        isEmpty && /* @__PURE__ */ jsx(
          "div",
          {
            ref: noDataRef,
            tabIndex: 0,
            onKeyDown: handleNoDataKeyDown,
            onFocus: () => {
              lastFocusWasNoDataRef.current = true;
            },
            className: "flex items-center justify-center py-12 text-sapphire-text-tertiary text-sm outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
            role: "status",
            children: noData ?? noDataText ?? "No data available"
          }
        ),
        !isGridScrollContainer && growingFeature,
        showLoading && /* @__PURE__ */ jsx(
          "div",
          {
            ref: loadingRef,
            tabIndex: 0,
            onKeyDown: handleLoadingKeyDown,
            onFocus: () => {
              loadingHadFocusRef.current = true;
            },
            onBlur: () => {
              loadingHadFocusRef.current = false;
            },
            className: "absolute inset-0 flex items-center justify-center bg-background/60 z-20 outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
            role: "status",
            "aria-label": "Loading",
            children: /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 text-sm text-sapphire-text-tertiary", children: [
              /* @__PURE__ */ jsxs("svg", { className: "animate-spin h-5 w-5", viewBox: "0 0 24 24", fill: "none", children: [
                /* @__PURE__ */ jsx("circle", { className: "opacity-25", cx: "12", cy: "12", r: "10", stroke: "currentColor", strokeWidth: "4" }),
                /* @__PURE__ */ jsx("path", { className: "opacity-75", fill: "currentColor", d: "M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4z" })
              ] }),
              "Loading..."
            ] })
          }
        ),
        /* @__PURE__ */ jsx(
          "div",
          {
            ref: afterSentinelRef,
            tabIndex: 0,
            onFocus: handleAfterSentinelFocus,
            "aria-hidden": "true",
            className: "absolute w-0 h-0 overflow-hidden"
          }
        ),
        selectionMode !== "None" && /* @__PURE__ */ jsx("div", { "aria-live": "polite", "aria-atomic": "true", className: "sr-only", children: selectionAnnouncement })
      ]
    }
  ) });
}
var Table = React.forwardRef(TableInner);
Table.displayName = "Table";
function TableHeaderRowInner({ children, sticky, className, style }, ref) {
  const ctx = useTableContext();
  const { t } = useTranslation("fx");
  const { headerSelector, deselectAll, areAllSelected, selectAll } = ctx;
  const handleHeaderCheckbox = useCallback(() => {
    if (headerSelector === "ClearAll") {
      deselectAll();
    } else if (areAllSelected()) {
      deselectAll();
    } else {
      selectAll();
    }
  }, [headerSelector, deselectAll, areAllSelected, selectAll]);
  const isInteractiveTarget = useCallback(
    (target, rowElement) => {
      if (target.closest('input, button, a, [role="checkbox"], [role="radio"]')) return true;
      const tabbableElements = getTabbableElements(rowElement);
      return tabbableElements.some((el) => el === target || el.contains(target));
    },
    []
  );
  const handleMouseDown = useCallback(
    (e) => {
      const target = e.target;
      const rowElement = e.currentTarget;
      if (isInteractiveTarget(target, rowElement)) return;
      ctx.setMouseNavigating();
      ctx.navigateToRow(-1);
      setTimeout(() => {
        rowElement.focus({ preventScroll: true });
      }, 0);
    },
    [ctx, isInteractiveTarget]
  );
  const allSelected = ctx.selectionMode === "Multi" && ctx.areAllSelected() && ctx.rowKeys.length > 0;
  const isFocusedRow = ctx.focusedRowIndex === -1 && ctx.focusedCell == null;
  const isCellMode = ctx.navigationMode === "Cell";
  const visibleDataCount = ctx.visibleColumnIndices.length > 0 ? ctx.visibleColumnIndices.length : ctx.columnCount;
  const actionsAriaColIndex = (ctx.hasSelectionColumn ? 1 : 0) + visibleDataCount + 1;
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      role: "row",
      "aria-rowindex": 1,
      tabIndex: isFocusedRow ? 0 : -1,
      "data-header-row": "true",
      "data-row-focusable": "true",
      onMouseDown: handleMouseDown,
      className: cn(
        "group/row grid col-span-full border-b border-border bg-sapphire-canvas-primary",
        ctx.hasToolbar && "border-t",
        !ctx.hasToolbar && "rounded-t-lg",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        sticky && "sticky z-10",
        className
      ),
      style: {
        gridTemplateColumns: "subgrid",
        gridColumn: "1 / -1",
        ...style
      },
      children: [
        ctx.hasSelectionColumn && /* @__PURE__ */ jsx(
          "div",
          {
            role: "columnheader",
            "aria-colindex": 1,
            "aria-label": ctx.selectionMode === "Multi" ? allSelected ? t("TABLE_DESELECT_ALL_ROWS") : t("TABLE_SELECT_ALL_ROWS") : t("TABLE_SELECTION"),
            "data-cell-focusable": "true",
            "data-column-index": 0,
            tabIndex: isCellMode ? -1 : void 0,
            className: cn(
              "flex items-center justify-center min-h-[40px]",
              "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
              !ctx.hasToolbar && "rounded-tl-lg",
              ctx.overflowMode === "Scroll" && "sticky left-0 z-[2] bg-sapphire-canvas-primary",
              ctx.overflowMode === "Scroll" && (ctx.hasToolbar ? "group-focus/row:[clip-path:inset(2px)]" : "group-focus/row:[clip-path:inset(2px_round_8px_0_0_0)]")
            ),
            children: ctx.selectionMode === "Multi" && /* @__PURE__ */ jsx(
              CheckBox,
              {
                checked: allSelected,
                onChange: handleHeaderCheckbox,
                size: "Small",
                tabIndex: -1,
                "aria-label": allSelected ? t("TABLE_DESELECT_ALL_ROWS") : t("TABLE_SELECT_ALL_ROWS")
              }
            )
          }
        ),
        React.Children.toArray(children).filter(React.isValidElement).map((child, origIdx) => ({ child, origIdx })).filter(
          ({ origIdx }) => ctx.visibleColumnIndices.length === 0 || ctx.visibleColumnIndices.includes(origIdx)
        ).map(
          ({ child, origIdx }, idx) => React.cloneElement(child, {
            key: `col-${origIdx}`,
            "aria-colindex": (ctx.hasSelectionColumn ? 2 : 1) + idx,
            _columnIndex: origIdx,
            _navPosition: idx
          })
        ),
        ctx.hasActionsColumn && /* @__PURE__ */ jsx(
          "div",
          {
            role: "columnheader",
            "aria-colindex": actionsAriaColIndex,
            "aria-label": t("TABLE_ROW_ACTIONS"),
            "data-cell-focusable": "true",
            "data-column-index": (ctx.hasSelectionColumn ? 1 : 0) + visibleDataCount,
            tabIndex: isCellMode ? -1 : void 0,
            className: cn(
              "flex items-center px-2 py-1 min-h-[40px]",
              "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
              !ctx.hasToolbar && "rounded-tr-lg",
              ctx.overflowMode === "Scroll" && "sticky right-0 z-[2] bg-sapphire-canvas-primary",
              ctx.overflowMode === "Scroll" && (ctx.hasToolbar ? "group-focus/row:[clip-path:inset(2px)]" : "group-focus/row:[clip-path:inset(2px_round_0_8px_0_0)]")
            )
          }
        ),
        ctx.hasNavigatedColumn && /* @__PURE__ */ jsx("div", { "aria-hidden": "true" })
      ]
    }
  );
}
var TableHeaderRow = React.forwardRef(TableHeaderRowInner);
TableHeaderRow.displayName = "TableHeaderRow";
TableHeaderRow._tableRole = "HeaderRow";
function TableHeaderCellInner({
  children,
  width: _width,
  minWidth: _minWidth,
  importance: _importance,
  popinText: _popinText,
  sortIndicator,
  popinHidden: _popinHidden,
  horizontalAlign,
  action,
  className,
  _columnIndex,
  _navPosition,
  ...rest
}, ref) {
  const ctx = useTableContext();
  const isFirstColumn = _columnIndex != null && _columnIndex === ctx.visibleColumnIndices[0];
  const isLastColumn = _columnIndex != null && _columnIndex === ctx.visibleColumnIndices[ctx.visibleColumnIndices.length - 1];
  const needsExtraLeftPad = isFirstColumn && !ctx.hasSelectionColumn;
  const needsExtraRightPad = isLastColumn && !ctx.hasActionsColumn;
  const selectionOffset = ctx.hasSelectionColumn ? 1 : 0;
  const navColumnIndex = _navPosition != null ? _navPosition + selectionOffset : void 0;
  const ariaSortMap = { Ascending: "ascending", Descending: "descending" };
  const ariaSortValue = sortIndicator ? ariaSortMap[sortIndicator] ?? "none" : void 0;
  const isCellMode = ctx.navigationMode === "Cell";
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      role: "columnheader",
      "aria-sort": ariaSortValue,
      tabIndex: isCellMode ? -1 : void 0,
      "data-cell-focusable": "true",
      "data-column-index": navColumnIndex,
      className: cn(
        "flex items-center gap-2 px-2 py-1 font-semibold text-foreground text-sm",
        "min-h-[40px]",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        needsExtraLeftPad && "pl-4",
        needsExtraLeftPad && !ctx.hasToolbar && "rounded-tl-lg",
        needsExtraRightPad && "pr-4",
        needsExtraRightPad && !ctx.hasToolbar && "rounded-tr-lg",
        getAlignmentClass(horizontalAlign),
        className
      ),
      ...rest,
      children: [
        action,
        /* @__PURE__ */ jsx("span", { className: "truncate", children }),
        sortIndicator && sortIndicator !== "None" && (sortIndicator === "Ascending" ? /* @__PURE__ */ jsx(SortAscendingIcon, { className: "h-4 w-4", "aria-hidden": "true" }) : /* @__PURE__ */ jsx(SortDescendingIcon, { className: "h-4 w-4", "aria-hidden": "true" }))
      ]
    }
  );
}
var TableHeaderCell = React.forwardRef(TableHeaderCellInner);
TableHeaderCell.displayName = "TableHeaderCell";
TableHeaderCell._tableRole = "HeaderCell";
function TableToolbar({ sticky, alignContent = "Start" /* Start */, className, style, children, ref, ...rest }) {
  const innerRef = useRef(null);
  const { t } = useTranslation("fx");
  const mergedRef = React.useCallback(
    (instance) => {
      innerRef.current = instance;
      instance?.nativeElement?.setAttribute("aria-roledescription", t("TABLE_TOOLBAR_ROLEDESCRIPTION"));
      if (typeof ref === "function") ref(instance);
      else if (ref) ref.current = instance;
    },
    [ref, t]
  );
  return /* @__PURE__ */ jsx(
    Toolbar,
    {
      ref: mergedRef,
      design: "Transparent" /* Transparent */,
      accessibleName: t("TABLE_TOOLBAR"),
      alignContent,
      className: cn(
        "pl-4 min-h-[48px] rounded-t-lg",
        sticky && "sticky z-20 bg-sapphire-canvas-primary",
        className
      ),
      style,
      ...rest,
      children
    }
  );
}
TableToolbar.displayName = "TableToolbar";
TableToolbar._tableRole = "Toolbar";
function stickyScrollClasses({ isScroll, isAlternate, isInteractive, isSelected, enterActive }, isBottomEdge, side) {
  if (!isScroll) return "";
  const stickyBase = side === "left" ? "sticky left-0 z-[1] bg-sapphire-canvas-primary" : "sticky right-0 z-[1] bg-sapphire-canvas-primary";
  const clipNormal = "group-focus/row:[clip-path:inset(2px)]";
  const clipBottom = side === "left" ? "group-focus/row:[clip-path:inset(2px_round_0_0_0_8px)]" : "group-focus/row:[clip-path:inset(2px_round_0_0_8px_0)]";
  return cn(
    stickyBase,
    !isBottomEdge && clipNormal,
    isBottomEdge && clipBottom,
    isAlternate && "bg-accent/60",
    isInteractive && "group-hover/row:bg-sapphire-table-row-hover",
    isInteractive && "group-focus/row:group-active/row:bg-sapphire-neutral-pressed-background",
    isSelected && !enterActive && "bg-sapphire-brand-selected-background",
    isSelected && !enterActive && isInteractive && "group-hover/row:bg-sapphire-brand-toggle-background",
    isInteractive && isSelected && "group-focus/row:group-active/row:bg-sapphire-neutral-pressed-background",
    enterActive && "bg-sapphire-neutral-pressed-background"
  );
}
function SelectionCell({
  ctx,
  rowKey,
  isSelected,
  isCellMode,
  isBottomEdge,
  stickyState,
  handleCheckboxChange,
  t
}) {
  return /* @__PURE__ */ jsxs(
    "div",
    {
      role: "gridcell",
      "aria-colindex": 1,
      "aria-label": t("TABLE_SELECTION"),
      "data-cell-focusable": "true",
      "data-column-index": 0,
      tabIndex: isCellMode ? -1 : void 0,
      className: cn(
        "flex items-center justify-center min-h-[40px] cursor-pointer",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        isBottomEdge && "rounded-bl-lg",
        stickyScrollClasses(stickyState, isBottomEdge, "left")
      ),
      onClick: (e) => e.stopPropagation(),
      children: [
        ctx.selectionMode === "Multi" && /* @__PURE__ */ jsx(
          CheckBox,
          {
            checked: isSelected,
            onChange: handleCheckboxChange,
            size: "Small",
            tabIndex: -1,
            "aria-label": `Select row ${rowKey}`
          }
        ),
        ctx.selectionMode === "Single" && /* @__PURE__ */ jsx(
          RadioButton,
          {
            checked: isSelected,
            onChange: handleCheckboxChange,
            size: "Small",
            tabIndex: -1,
            name: `${ctx.tableId}-selection`,
            "aria-label": `Select row ${rowKey}`
          }
        )
      ]
    }
  );
}
function ActionsCell({
  actionsColumnIndex,
  isCellMode,
  isBottomEdge,
  stickyState,
  flexibleVisible,
  overflowActions,
  fixedVisible,
  hasOverflow,
  overflowMenuOpen,
  setOverflowMenuOpen,
  overflowOpener,
  setOverflowOpener,
  overflowButtonRef,
  t
}) {
  return /* @__PURE__ */ jsxs(
    "div",
    {
      role: "gridcell",
      "aria-colindex": actionsColumnIndex + 1,
      "aria-label": t("TABLE_ROW_ACTIONS"),
      "data-cell-focusable": "true",
      "data-column-index": actionsColumnIndex,
      tabIndex: isCellMode ? -1 : void 0,
      className: cn(
        "flex items-center gap-1 px-2 py-1 min-h-[40px]",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        isBottomEdge && "rounded-br-lg",
        stickyScrollClasses(stickyState, isBottomEdge, "right")
      ),
      children: [
        flexibleVisible,
        hasOverflow && /* @__PURE__ */ jsxs(Fragment, { children: [
          /* @__PURE__ */ jsx(
            Button,
            {
              ref: overflowButtonRef,
              design: "SecondaryNeutral",
              size: "Medium",
              icon: /* @__PURE__ */ jsx(OverflowIcon, {}),
              iconOnly: true,
              accessibleName: "More actions",
              onClick: (e) => {
                e.originalEvent?.stopPropagation?.();
                setOverflowOpener(overflowButtonRef.current?.nativeElement ?? null);
                setOverflowMenuOpen((prev) => !prev);
              }
            }
          ),
          /* @__PURE__ */ jsx(
            Menu,
            {
              open: overflowMenuOpen,
              opener: overflowOpener,
              onClose: () => setOverflowMenuOpen(false),
              children: overflowActions.map((action, actionIdx) => {
                const props = action.props;
                return /* @__PURE__ */ jsx(
                  MenuItem,
                  {
                    icon: props.icon,
                    text: props.text ?? "",
                    onClick: () => props.onClick?.()
                  },
                  props.text ?? `overflow-${actionIdx}`
                );
              })
            }
          )
        ] }),
        fixedVisible
      ]
    }
  );
}
function PopinArea({
  ctx,
  popinCells,
  rowIndex,
  popinTabIndex,
  isInteractive,
  enterActive,
  isSelected
}) {
  const start = ctx.hasSelectionColumn ? "2" : "1";
  const trailingCols = (ctx.hasActionsColumn ? 1 : 0) + (ctx.hasNavigatedColumn ? 1 : 0);
  const end = trailingCols > 0 ? `${-(trailingCols + 1)}` : "-1";
  return /* @__PURE__ */ jsx(
    "div",
    {
      role: "gridcell",
      "aria-colindex": ctx.popinColumnIndex + 1,
      "data-cell-focusable": "true",
      "data-column-index": ctx.popinColumnIndex,
      "data-row-index": rowIndex,
      "data-popin-row": "true",
      tabIndex: popinTabIndex,
      className: cn(
        "py-1 bg-clip-content",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        ctx.hasSelectionColumn ? "px-2" : "pl-4 pr-2",
        !enterActive && isSelected ? "bg-sapphire-brand-selected-background" : !enterActive && "bg-muted/20",
        !enterActive && isSelected && isInteractive && "group-hover/row:bg-sapphire-brand-toggle-background",
        !enterActive && !isSelected && isInteractive && "group-hover/row:bg-sapphire-table-row-hover",
        isInteractive && "group-focus/row:group-active/row:bg-sapphire-neutral-pressed-background",
        enterActive && "bg-sapphire-neutral-pressed-background"
      ),
      style: { gridColumn: `${start} / ${end}` },
      children: popinCells.map((cell, idx) => {
        const colIdx = ctx.popinColumnIndices[idx];
        const meta = ctx.columnsMeta[colIdx];
        const label = meta?.popinText ?? meta?.headerText;
        return /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 py-1 text-sm", children: [
          label && /* @__PURE__ */ jsxs("span", { className: "text-sapphire-text-tertiary font-medium", children: [
            label,
            ":"
          ] }),
          /* @__PURE__ */ jsx("span", { children: cell.props.children })
        ] }, colIdx);
      })
    }
  );
}
function TableRowInner({ children, rowKey, interactive, navigated, movable: _movable, className }, ref) {
  const ctx = useTableContext();
  const { t } = useTranslation("fx");
  const rowIndex = ctx.getRowIndexByKey(rowKey);
  const isSelected = ctx.isSelected(rowKey);
  const isAlternate = ctx.alternateRowColors && rowIndex % 2 === 1;
  const isLastRow = rowIndex === ctx.dataRowCount - 1;
  const isBottomEdge = isLastRow && !ctx.hasGrowingButton;
  const isInteractive = !!(interactive || ctx.selectionBehavior === "RowOnly");
  const { cells, actions, navigationAction } = useMemo(() => scanRowChildren(children), [children]);
  const isFocusedRow = ctx.focusedRowIndex === rowIndex && ctx.focusedCell == null;
  const isCellMode = ctx.navigationMode === "Cell";
  const [enterActive, setEnterActive] = useState(false);
  const handleClick = useCallback(
    (e) => {
      const target = e.target;
      const rowElement = e.currentTarget;
      const tabbableElements = getTabbableElements(rowElement);
      if (tabbableElements.some((el) => el === target || el.contains(target))) return;
      if (e.shiftKey && ctx.selectionMode === "Multi") {
        const anchor = ctx.getLastSelectedKey();
        if (anchor) {
          ctx.selectRange(anchor, rowKey);
          return;
        }
      }
      if (ctx.selectionMode !== "None" && ctx.selectionBehavior === "RowOnly") {
        ctx.toggleSelection(rowKey);
        return;
      }
      if (interactive && ctx.onRowClick) {
        ctx.onRowClick({ row: e.currentTarget, rowKey });
      }
    },
    [ctx, rowKey, interactive]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key !== "Enter" || e.repeat) return;
      const target = e.target;
      if (!target.matches('[data-row-focusable="true"]')) return;
      if (!isInteractive) return;
      e.preventDefault();
      setEnterActive(true);
    },
    [isInteractive]
  );
  const handleKeyUp = useCallback(
    (e) => {
      if (e.key !== "Enter") return;
      const target = e.target;
      if (!target.matches('[data-row-focusable="true"]') || !enterActive) return;
      setEnterActive(false);
      if (ctx.selectionMode !== "None" && ctx.selectionBehavior === "RowOnly") {
        ctx.toggleSelection(rowKey);
        return;
      }
      if (interactive && ctx.onRowClick) {
        ctx.onRowClick({ row: e.currentTarget, rowKey });
      }
    },
    [ctx, rowKey, interactive, enterActive]
  );
  const handleCheckboxChange = useCallback(
    (detail) => {
      if (ctx.selectionMode === "Multi") {
        const nativeEvent = detail.originalEvent?.nativeEvent;
        if (nativeEvent?.shiftKey) {
          const lastKey = ctx.getLastSelectedKey();
          if (lastKey) {
            ctx.selectRange(lastKey, rowKey);
            return;
          }
        }
      }
      ctx.toggleSelection(rowKey);
    },
    [ctx, rowKey]
  );
  const handleMouseDown = useCallback(
    (e) => {
      const target = e.target;
      const rowElement = e.currentTarget;
      const tabbableElements = getTabbableElements(rowElement);
      if (tabbableElements.some((el) => el === target || el.contains(target))) return;
      ctx.setMouseNavigating();
      ctx.navigateToRow(rowIndex);
      setTimeout(() => {
        rowElement.focus({ preventScroll: true });
      }, 0);
    },
    [ctx, rowIndex]
  );
  const visibleCells = ctx.visibleColumnIndices.length > 0 ? cells.filter((_cell, idx) => ctx.visibleColumnIndices.includes(idx)) : cells;
  const popinCells = ctx.isPopinActive ? cells.filter((_cell, idx) => ctx.popinColumnIndices.includes(idx)) : [];
  const [overflowMenuOpen, setOverflowMenuOpen] = useState(false);
  const [overflowOpener, setOverflowOpener] = useState(null);
  const overflowButtonRef = useRef(null);
  const { flexibleVisible, overflowActions, fixedVisible, hasOverflow } = useMemo(() => {
    const fixedActions = navigationAction ? [navigationAction] : [];
    const fixedCount = fixedActions.length;
    const totalCount = ctx.rowActionCount || actions.length + fixedCount;
    const visibleFlexible = actions.filter((a) => !a.props.invisible);
    let maxFlexible = totalCount - fixedCount;
    let flexVisible;
    let overflow;
    if (maxFlexible < 1) {
      flexVisible = [];
      overflow = visibleFlexible;
    } else if (visibleFlexible.length <= maxFlexible) {
      flexVisible = actions;
      overflow = [];
    } else {
      maxFlexible -= 1;
      flexVisible = visibleFlexible.slice(0, maxFlexible);
      overflow = visibleFlexible.slice(maxFlexible);
    }
    const hasOvf = overflow.length > 0;
    const fixedVis = hasOvf ? fixedActions.slice(0, Math.max(0, totalCount - 1)) : fixedActions;
    return { flexibleVisible: flexVisible, overflowActions: overflow, fixedVisible: fixedVis, hasOverflow: hasOvf };
  }, [actions, navigationAction, ctx.rowActionCount]);
  const actionsColumnIndex = (ctx.hasSelectionColumn ? 1 : 0) + (ctx.visibleColumnIndices.length > 0 ? ctx.visibleColumnIndices.length : cells.length);
  const isFocusedPopinCell = ctx.focusedCell?.rowIndex === rowIndex && ctx.focusedCell?.columnIndex === ctx.popinColumnIndex;
  const popinTabIndex = isCellMode ? isFocusedPopinCell ? 0 : -1 : void 0;
  const stickyState = {
    isScroll: ctx.overflowMode === "Scroll",
    isAlternate,
    isInteractive,
    isSelected,
    enterActive
  };
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      role: "row",
      "aria-rowindex": rowIndex + 2,
      "aria-selected": ctx.selectionMode !== "None" ? isSelected : void 0,
      "aria-current": navigated ? "true" : void 0,
      tabIndex: isFocusedRow ? 0 : -1,
      "data-row-focusable": "true",
      "data-row-index": rowIndex,
      "data-row-key": rowKey,
      onMouseDown: handleMouseDown,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      onKeyUp: handleKeyUp,
      className: cn(
        "group/row grid col-span-full border-b border-border bg-sapphire-canvas-primary",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        ctx.hasNavigatedColumn && "relative",
        className,
        isAlternate && "bg-accent/60",
        isInteractive && "cursor-pointer",
        isInteractive && "hover:bg-sapphire-table-row-hover",
        isInteractive && "focus:active:bg-sapphire-neutral-pressed-background",
        isSelected && !enterActive && "bg-sapphire-brand-selected-background",
        isSelected && !enterActive && isInteractive && "hover:bg-sapphire-brand-toggle-background",
        isSelected && isInteractive && "focus:active:bg-sapphire-neutral-pressed-background",
        enterActive && "bg-sapphire-neutral-pressed-background"
      ),
      style: { gridTemplateColumns: "subgrid", gridColumn: "1 / -1" },
      children: [
        ctx.hasSelectionColumn && /* @__PURE__ */ jsx(
          SelectionCell,
          {
            ctx,
            rowKey,
            isSelected,
            isCellMode,
            isBottomEdge,
            stickyState,
            handleCheckboxChange,
            t
          }
        ),
        visibleCells.map((cell, idx) => {
          const colIdx = ctx.visibleColumnIndices.length > 0 ? ctx.visibleColumnIndices[idx] : idx;
          return React.cloneElement(cell, {
            key: `cell-${colIdx}`,
            "aria-colindex": (ctx.hasSelectionColumn ? 2 : 1) + idx,
            _columnIndex: colIdx,
            _navPosition: idx,
            _isLastRow: isBottomEdge
          });
        }),
        ctx.hasActionsColumn && /* @__PURE__ */ jsx(
          ActionsCell,
          {
            actionsColumnIndex,
            isCellMode,
            isBottomEdge,
            stickyState,
            flexibleVisible,
            overflowActions,
            fixedVisible,
            hasOverflow,
            overflowMenuOpen,
            setOverflowMenuOpen,
            overflowOpener,
            setOverflowOpener,
            overflowButtonRef,
            t
          }
        ),
        popinCells.length > 0 && /* @__PURE__ */ jsx(
          PopinArea,
          {
            ctx,
            popinCells,
            rowIndex,
            popinTabIndex,
            isInteractive,
            enterActive,
            isSelected
          }
        ),
        ctx.hasNavigatedColumn && /* @__PURE__ */ jsx("div", { "aria-hidden": "true" }),
        ctx.hasNavigatedColumn && navigated && /* @__PURE__ */ jsx(
          "div",
          {
            "aria-hidden": "true",
            className: "absolute inset-y-0 end-0 w-[3px] bg-primary pointer-events-none group-focus/row:inset-y-1 group-focus/row:me-1"
          }
        )
      ]
    }
  );
}
var TableRow = React.forwardRef(TableRowInner);
TableRow.displayName = "TableRow";
TableRow._tableRole = "Row";
function TableCellInner({ children, merged, rowHeader, horizontalAlign, className, _columnIndex, _navPosition, _isLastRow, ...rest }, ref) {
  const ctx = useTableContext();
  const effectiveAlign = horizontalAlign ?? (_columnIndex != null ? ctx.columnsMeta[_columnIndex]?.horizontalAlign : void 0);
  const isFirstColumn = _columnIndex != null && _columnIndex === ctx.visibleColumnIndices[0];
  const isLastColumn = _columnIndex != null && _columnIndex === ctx.visibleColumnIndices[ctx.visibleColumnIndices.length - 1];
  const needsExtraLeftPad = isFirstColumn && !ctx.hasSelectionColumn;
  const needsExtraRightPad = isLastColumn && !ctx.hasActionsColumn;
  const selectionOffset = ctx.hasSelectionColumn ? 1 : 0;
  const navColumnIndex = _navPosition != null ? _navPosition + selectionOffset : void 0;
  const isCellMode = ctx.navigationMode === "Cell";
  const isTextOnly = Children.toArray(children).every((child) => !isValidElement(child));
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      role: rowHeader ? "rowheader" : "gridcell",
      tabIndex: isCellMode ? -1 : void 0,
      "data-cell-focusable": "true",
      "data-column-index": navColumnIndex,
      className: cn(
        "flex items-center px-2 py-1 text-sm font-normal text-foreground",
        "min-h-[40px]",
        "outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
        needsExtraLeftPad && "pl-4",
        needsExtraRightPad && "pr-4",
        _isLastRow && needsExtraLeftPad && "rounded-bl-lg",
        _isLastRow && needsExtraRightPad && "rounded-br-lg",
        merged && "border-t-0",
        getAlignmentClass(effectiveAlign),
        className
      ),
      ...rest,
      children: isTextOnly ? /* @__PURE__ */ jsx("span", { className: cn("truncate", effectiveAlign ? void 0 : "text-start"), children }) : children
    }
  );
}
var TableCell = React.forwardRef(TableCellInner);
TableCell.displayName = "TableCell";
TableCell._tableRole = "Cell";
function TableSelectionMultiInner(_props, ref) {
  useImperativeHandle(ref, () => ({
    getSelectedRows: () => [],
    areAllRowsSelected: () => false,
    getSelectedAsSet: () => /* @__PURE__ */ new Set(),
    setSelectedAsSet: () => {
    },
    isSelected: () => false,
    setSelected: () => {
    }
  }), []);
  return null;
}
var TableSelectionMulti = React.forwardRef(TableSelectionMultiInner);
TableSelectionMulti.displayName = "TableSelectionMulti";
TableSelectionMulti._tableRole = "SelectionMulti";
function TableSelectionSingleInner(_props, ref) {
  useImperativeHandle(ref, () => ({
    getSelectedRow: () => void 0,
    isSelected: () => false,
    setSelected: () => {
    }
  }), []);
  return null;
}
var TableSelectionSingle = React.forwardRef(TableSelectionSingleInner);
TableSelectionSingle.displayName = "TableSelectionSingle";
TableSelectionSingle._tableRole = "SelectionSingle";
function TableGrowingInner({ mode = "Button", text, subtext, onLoadMore }, ref) {
  const ctx = useTableContext();
  const [isLoading, setIsLoading] = useState(false);
  const [activeState, setActiveState] = useState(false);
  const isLoadingRef = useRef(false);
  const onLoadMoreRef = useRef(onLoadMore);
  onLoadMoreRef.current = onLoadMore;
  const prevDataRowCountRef = useRef(ctx.dataRowCount);
  const endRowRef = useRef(null);
  const hadFocusRef = useRef(false);
  const { t } = useTranslation("fx");
  const [showButtonFallback, setShowButtonFallback] = useState(mode === "Scroll");
  const handleLoadMore = useCallback(async () => {
    if (isLoadingRef.current || !onLoadMoreRef.current) return;
    const activeEl = document.activeElement;
    hadFocusRef.current = !!activeEl?.hasAttribute("data-growing-button");
    isLoadingRef.current = true;
    setIsLoading(true);
    try {
      await Promise.resolve(onLoadMoreRef.current({}));
    } finally {
      isLoadingRef.current = false;
      setIsLoading(false);
    }
  }, []);
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        setActiveState(true);
        handleLoadMore();
      }
    },
    [handleLoadMore]
  );
  const handleKeyUp = useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        setActiveState(false);
      }
    },
    []
  );
  const handleFocusOut = useCallback(() => setActiveState(false), []);
  const { dataRowCount, setFocusedRow } = ctx;
  useEffect(() => {
    const prevCount = prevDataRowCountRef.current;
    prevDataRowCountRef.current = dataRowCount;
    if (mode !== "Button" && !showButtonFallback) return;
    if (dataRowCount <= prevCount || prevCount === 0) return;
    if (hadFocusRef.current) {
      hadFocusRef.current = false;
      setFocusedRow(prevCount);
    }
  }, [dataRowCount, mode, showButtonFallback, setFocusedRow]);
  useEffect(() => {
    if (mode !== "Scroll") return;
    const endRow = endRowRef.current;
    if (!endRow) return;
    const scrollContainer = findVerticalScrollContainer(endRow);
    if (!scrollContainer) return;
    const isScrollable = scrollContainer.scrollHeight > scrollContainer.clientHeight;
    if (!isScrollable) {
      setShowButtonFallback(true);
      return;
    }
    setShowButtonFallback(false);
    const observer = new IntersectionObserver(
      (entries) => {
        if (entries.some((e) => e.isIntersecting) && !isLoadingRef.current) {
          handleLoadMore();
        }
      },
      {
        root: scrollContainer,
        rootMargin: "0px 0px 5px 0px"
      }
    );
    observer.observe(endRow);
    return () => observer.disconnect();
  }, [mode, handleLoadMore, dataRowCount]);
  const buttonText = text ?? t("TABLE_MORE");
  const renderButton = () => /* @__PURE__ */ jsxs(
    "button",
    {
      type: "button",
      "data-growing-button": "true",
      "aria-label": [buttonText, subtext].filter(Boolean).join(", "),
      "aria-busy": isLoading || void 0,
      onClick: handleLoadMore,
      onKeyDown: handleKeyDown,
      onKeyUp: handleKeyUp,
      onBlur: handleFocusOut,
      className: cn(
        "flex items-center flex-col cursor-pointer w-full",
        "border-t border-border bg-background rounded-b-lg",
        "outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-inset",
        "hover:bg-sapphire-table-row-hover",
        activeState && "bg-sapphire-brand-selected-background",
        isLoading && "opacity-50 pointer-events-none"
      ),
      children: [
        /* @__PURE__ */ jsx("span", { className: cn("pt-3 text-sm font-bold text-sapphire-brand-foreground", !subtext && "pb-3"), children: isLoading ? /* @__PURE__ */ jsx(Loader2, { className: "h-4 w-4 animate-spin" }) : buttonText }),
        subtext && /* @__PURE__ */ jsx("span", { className: "pt-3 pb-3 text-sm text-sapphire-brand-foreground", children: subtext }),
        /* @__PURE__ */ jsx("span", { className: "sr-only", children: t("TABLE_MORE_DESCRIPTION") })
      ]
    }
  );
  if (mode === "Scroll") {
    return /* @__PURE__ */ jsxs("div", { ref, style: { gridColumn: "1 / -1" }, children: [
      /* @__PURE__ */ jsx("div", { ref: endRowRef, "aria-hidden": "true", "data-growing-sentinel": "true", className: "h-px" }),
      showButtonFallback ? renderButton() : isLoading && /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-center py-4 text-sm text-sapphire-text-tertiary", children: [
        /* @__PURE__ */ jsx(Loader2, { className: "h-4 w-4 animate-spin mr-2" }),
        t("TABLE_LOADING_MORE")
      ] })
    ] });
  }
  return /* @__PURE__ */ jsx("div", { ref, style: { gridColumn: "1 / -1" }, children: renderButton() });
}
var TableGrowing = React.forwardRef(TableGrowingInner);
TableGrowing.displayName = "TableGrowing";
TableGrowing._tableRole = "Growing";
function TableRowActionInner({ icon, text, invisible = false, onClick }, _ref) {
  if (invisible) {
    return /* @__PURE__ */ jsx("span", { className: "w-8", "aria-hidden": "true" });
  }
  return /* @__PURE__ */ jsx(
    Button,
    {
      design: "SecondaryNeutral",
      size: "Medium",
      icon,
      iconOnly: !!icon,
      onClick,
      accessibleName: text,
      tooltip: icon ? text : void 0,
      children: !icon && text ? text : void 0
    }
  );
}
var TableRowAction = React.forwardRef(TableRowActionInner);
TableRowAction.displayName = "TableRowAction";
TableRowAction._tableRole = "RowAction";
function TableRowActionNavigationInner({ invisible = false, onClick }, ref) {
  const { t } = useTranslation("fx");
  if (invisible) {
    return /* @__PURE__ */ jsx("span", { className: "w-8", "aria-hidden": "true" });
  }
  return /* @__PURE__ */ jsx(
    Button,
    {
      ref,
      design: "TertiaryNeutral",
      size: "Small",
      icon: /* @__PURE__ */ jsx(SlimArrowRightIcon, {}),
      iconOnly: true,
      onClick,
      accessibleName: t("TABLE_NAVIGATION"),
      tooltip: t("TABLE_NAVIGATION")
    }
  );
}
var TableRowActionNavigation = React.forwardRef(TableRowActionNavigationInner);
TableRowActionNavigation.displayName = "TableRowActionNavigation";
TableRowActionNavigation._tableRole = "RowActionNavigation";
function TableHeaderCellActionAI({ onClick }) {
  const iconRef = useRef(null);
  const { t } = useTranslation("fx");
  const handleClick = useCallback(
    (detail) => {
      const el = iconRef.current?.nativeElement;
      if (onClick && el) {
        detail.originalEvent.stopPropagation();
        onClick({ targetRef: el });
      }
    },
    [onClick]
  );
  return /* @__PURE__ */ jsx(
    AiIcon,
    {
      ref: iconRef,
      mode: "Interactive" /* Interactive */,
      accessibleName: t("TABLE_AI_GENERATED"),
      showTooltip: true,
      onClick: handleClick,
      className: "h-4 w-4 text-sapphire-brand-foreground"
    }
  );
}
TableHeaderCellActionAI.displayName = "TableHeaderCellActionAI";

export { Table, TableCell, TableContext, TableGrowing, TableHeaderCell, TableHeaderCellActionAI, TableHeaderRow, TableRow, TableRowAction, TableRowActionNavigation, TableSelectionMulti, TableSelectionSingle, TableToolbar, useTableContext };
