// @sap-ui/fx-components/dist/chunk-CJXZUSUP.js
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import * as React from 'react';
import { useId, useRef, useState, useMemo, useEffect, useCallback, useImperativeHandle, useContext } from 'react';
import { Loader2 } from 'lucide-react';
import { jsx, jsxs } from 'react/jsx-runtime';

// src/types/list.ts
var ListSelectionMode = /* @__PURE__ */ ((ListSelectionMode2) => {
  ListSelectionMode2["None"] = "None";
  ListSelectionMode2["SingleStart"] = "SingleStart";
  ListSelectionMode2["SingleEnd"] = "SingleEnd";
  ListSelectionMode2["Single"] = "Single";
  ListSelectionMode2["Multiple"] = "Multiple";
  ListSelectionMode2["Delete"] = "Delete";
  return ListSelectionMode2;
})(ListSelectionMode || {});
var ListGrowingMode = /* @__PURE__ */ ((ListGrowingMode2) => {
  ListGrowingMode2["None"] = "None";
  ListGrowingMode2["Button"] = "Button";
  ListGrowingMode2["Scroll"] = "Scroll";
  return ListGrowingMode2;
})(ListGrowingMode || {});
var ListSeparator = /* @__PURE__ */ ((ListSeparator2) => {
  ListSeparator2["All"] = "All";
  ListSeparator2["Inner"] = "Inner";
  ListSeparator2["None"] = "None";
  return ListSeparator2;
})(ListSeparator || {});
var ListAccessibleRole = /* @__PURE__ */ ((ListAccessibleRole2) => {
  ListAccessibleRole2["List"] = "List";
  ListAccessibleRole2["Menu"] = "Menu";
  ListAccessibleRole2["Tree"] = "Tree";
  ListAccessibleRole2["ListBox"] = "ListBox";
  return ListAccessibleRole2;
})(ListAccessibleRole || {});
var ListItemAccessibleRole = /* @__PURE__ */ ((ListItemAccessibleRole2) => {
  ListItemAccessibleRole2["ListItem"] = "ListItem";
  ListItemAccessibleRole2["MenuItem"] = "MenuItem";
  ListItemAccessibleRole2["TreeItem"] = "TreeItem";
  ListItemAccessibleRole2["Option"] = "Option";
  ListItemAccessibleRole2["Group"] = "Group";
  ListItemAccessibleRole2["None"] = "None";
  return ListItemAccessibleRole2;
})(ListItemAccessibleRole || {});
var ListItemType = /* @__PURE__ */ ((ListItemType2) => {
  ListItemType2["Active"] = "Active";
  ListItemType2["Inactive"] = "Inactive";
  ListItemType2["Detail"] = "Detail";
  ListItemType2["Navigation"] = "Navigation";
  return ListItemType2;
})(ListItemType || {});
var ListItemHighlight = /* @__PURE__ */ ((ListItemHighlight2) => {
  ListItemHighlight2["None"] = "None";
  ListItemHighlight2["Positive"] = "Positive";
  ListItemHighlight2["Critical"] = "Critical";
  ListItemHighlight2["Information"] = "Information";
  ListItemHighlight2["Negative"] = "Negative";
  return ListItemHighlight2;
})(ListItemHighlight || {});
var ListItemValueState = /* @__PURE__ */ ((ListItemValueState2) => {
  ListItemValueState2["None"] = "None";
  ListItemValueState2["Positive"] = "Positive";
  ListItemValueState2["Critical"] = "Critical";
  ListItemValueState2["Information"] = "Information";
  ListItemValueState2["Negative"] = "Negative";
  return ListItemValueState2;
})(ListItemValueState || {});
var ListItemWrappingType = /* @__PURE__ */ ((ListItemWrappingType2) => {
  ListItemWrappingType2["None"] = "None";
  ListItemWrappingType2["Normal"] = "Normal";
  return ListItemWrappingType2;
})(ListItemWrappingType || {});
var listContainerVariants = cva(
  "relative w-full",
  {
    variants: {
      bordered: {
        true: "border border-sapphire-border-primary rounded-lg overflow-hidden",
        false: ""
      }
    },
    defaultVariants: {
      bordered: false
    }
  }
);
var listVariants = cva(
  "w-full m-0 p-0 list-none",
  {
    variants: {
      indent: {
        true: "pl-4",
        false: ""
      },
      separators: {
        true: "[&>li:not(:last-child)]:border-b [&>li:not(:last-child)]:border-sapphire-border-primary",
        false: ""
      }
    },
    defaultVariants: {
      indent: false,
      separators: true
    }
  }
);
var ListContext = React.createContext(null);
function useListContext() {
  const context = useContext(ListContext);
  if (!context) {
    throw new Error("List item components must be used within a List");
  }
  return context;
}
function useOptionalListContext() {
  return useContext(ListContext);
}
function useListSelection({
  mode,
  controlledKeys,
  defaultKeys = [],
  items,
  getItemKey,
  onSelectionChange
}) {
  const [internalKeys, setInternalKeys] = useState(new Set(defaultKeys));
  const previousKeysRef = useRef(new Set(defaultKeys));
  const isControlled = controlledKeys !== void 0;
  const selectedKeys = isControlled ? new Set(controlledKeys) : internalKeys;
  const getItemByKey = useCallback((key) => {
    if (!items || !getItemKey) return void 0;
    return items.find((item, index) => getItemKey(item, index) === key);
  }, [items, getItemKey]);
  const getSelectedItems = useCallback(() => {
    if (!items || !getItemKey) return [];
    return items.filter((item, index) => selectedKeys.has(getItemKey(item, index)));
  }, [items, getItemKey, selectedKeys]);
  const setSelected = useCallback((key, selected) => {
    const modeStr = String(mode);
    if (modeStr === "None") return;
    const newKeys = new Set(selectedKeys);
    const previousKeys = new Set(selectedKeys);
    if (modeStr === "Single" || modeStr === "SingleStart" || modeStr === "SingleEnd") {
      if (selected) {
        newKeys.clear();
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    } else if (modeStr === "Multiple") {
      if (selected) {
        newKeys.add(key);
      } else {
        newKeys.delete(key);
      }
    }
    if (!isControlled) {
      setInternalKeys(newKeys);
    }
    const targetItem = getItemByKey(key);
    onSelectionChange?.({
      selectedItems: items?.filter((item, index) => getItemKey && newKeys.has(getItemKey(item, index))) ?? [],
      previouslySelectedItems: items?.filter((item, index) => getItemKey && previousKeys.has(getItemKey(item, index))) ?? [],
      selectionComponentPressed: false,
      targetItem,
      key,
      selectedKeys: Array.from(newKeys)
      // Always include the raw keys
    });
    previousKeysRef.current = newKeys;
  }, [mode, selectedKeys, isControlled, getItemByKey, items, getItemKey, onSelectionChange]);
  const toggleSelection = useCallback((key) => {
    setSelected(key, !selectedKeys.has(key));
  }, [selectedKeys, setSelected]);
  const isSelected = useCallback((key) => {
    return selectedKeys.has(key);
  }, [selectedKeys]);
  const selectAll = useCallback(() => {
    if (!items || !getItemKey || String(mode) !== "Multiple") return;
    const allKeys = new Set(items.map(getItemKey));
    if (!isControlled) {
      setInternalKeys(allKeys);
    }
    onSelectionChange?.({
      selectedItems: items,
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: items[0]
    });
  }, [items, getItemKey, mode, isControlled, onSelectionChange, getSelectedItems]);
  const deselectAll = useCallback(() => {
    if (!isControlled) {
      setInternalKeys(/* @__PURE__ */ new Set());
    }
    onSelectionChange?.({
      selectedItems: [],
      previouslySelectedItems: getSelectedItems(),
      selectionComponentPressed: false,
      targetItem: void 0
    });
  }, [isControlled, onSelectionChange, getSelectedItems]);
  return {
    selectedKeys,
    setSelected,
    toggleSelection,
    isSelected,
    selectAll,
    deselectAll,
    getSelectedItems
  };
}
function useListNavigation({
  onFocusItem,
  onSelectItem,
  selectionMode,
  containerRef
}) {
  const [focusedIndex, setFocusedIndex] = useState(-1);
  const getAllNavigable = useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"], button[data-growing-button]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const getAllNavigableExcludingGrowingButton = useCallback(() => {
    if (!containerRef.current) return [];
    const elements = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"], li[role="menuitem"], li[role="treeitem"]'
      )
    );
    return elements.filter((el) => {
      const groupContent = el.closest('ul[role="group"]');
      if (groupContent && groupContent.classList.contains("hidden")) return false;
      return true;
    });
  }, [containerRef]);
  const focusItem = useCallback((index) => {
    const allNavigable = getAllNavigable();
    if (index < 0 || index >= allNavigable.length) return;
    const element = allNavigable[index];
    if (!element) return;
    const isGrowingButton = element.hasAttribute("data-growing-button");
    if (!isGrowingButton) {
      setFocusedIndex(index);
    }
    element.focus();
    onFocusItem?.(index);
  }, [getAllNavigable, onFocusItem]);
  const handleKeyDown = useCallback((e) => {
    const key = e.key;
    if (key === "Tab") {
      return;
    }
    const target = e.target;
    const isInteractiveElement = target.tagName === "INPUT" || target.tagName === "TEXTAREA" || target.tagName === "SELECT" || target.tagName === "BUTTON" || target.getAttribute("role") === "button" || target.getAttribute("contenteditable") === "true" || target.closest('button, input, textarea, select, [role="button"], [contenteditable="true"]') !== null;
    if (isInteractiveElement) {
      if (key === " " || key === "Enter" || key === "Delete") {
        return;
      }
    }
    const allNavigable = getAllNavigable();
    if (allNavigable.length === 0) return;
    const activeElement = document.activeElement;
    let currentIndex = allNavigable.indexOf(activeElement);
    if (currentIndex === -1) {
      currentIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    }
    let newIndex = currentIndex;
    switch (key) {
      case "ArrowDown":
        e.preventDefault();
        newIndex = Math.min(currentIndex + 1, allNavigable.length - 1);
        break;
      case "ArrowUp":
        e.preventDefault();
        newIndex = Math.max(currentIndex - 1, 0);
        break;
      case "Home":
        e.preventDefault();
        const navigableExcludingButton = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButton.length > 0) {
          const firstElement = navigableExcludingButton[0];
          const indexInAll = allNavigable.indexOf(firstElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "End":
        e.preventDefault();
        const navigableExcludingButtonEnd = getAllNavigableExcludingGrowingButton();
        if (navigableExcludingButtonEnd.length > 0) {
          const lastElement = navigableExcludingButtonEnd[navigableExcludingButtonEnd.length - 1];
          const indexInAll = allNavigable.indexOf(lastElement);
          if (indexInAll !== -1) {
            newIndex = indexInAll;
          }
        }
        break;
      case "PageDown":
        e.preventDefault();
        const navigableExcludingButtonPageDown = getAllNavigableExcludingGrowingButton();
        const currentInFiltered = navigableExcludingButtonPageDown.indexOf(activeElement);
        if (currentInFiltered !== -1) {
          const newFilteredIndex = Math.min(currentInFiltered + 10, navigableExcludingButtonPageDown.length - 1);
          const targetElement = navigableExcludingButtonPageDown[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageDown.length > 0) {
            const lastElement = navigableExcludingButtonPageDown[navigableExcludingButtonPageDown.length - 1];
            newIndex = allNavigable.indexOf(lastElement);
          }
        }
        break;
      case "PageUp":
        e.preventDefault();
        const navigableExcludingButtonPageUp = getAllNavigableExcludingGrowingButton();
        const currentInFilteredPageUp = navigableExcludingButtonPageUp.indexOf(activeElement);
        if (currentInFilteredPageUp !== -1) {
          const newFilteredIndex = Math.max(currentInFilteredPageUp - 10, 0);
          const targetElement = navigableExcludingButtonPageUp[newFilteredIndex];
          newIndex = allNavigable.indexOf(targetElement);
        } else {
          if (navigableExcludingButtonPageUp.length > 0) {
            const lastFilteredIndex = navigableExcludingButtonPageUp.length - 1;
            const targetIndex = Math.max(lastFilteredIndex - 10, 0);
            const targetElement = navigableExcludingButtonPageUp[targetIndex];
            newIndex = allNavigable.indexOf(targetElement);
          }
        }
        break;
      case " ":
        e.preventDefault();
        const spaceElement = allNavigable[currentIndex];
        if (spaceElement?.getAttribute("role") === "listitem") {
          if (String(selectionMode) !== "None" && String(selectionMode) !== "Delete") {
            onSelectItem?.(currentIndex);
          }
        }
        return;
      case "Delete":
        const deleteElement = allNavigable[currentIndex];
        if (deleteElement?.getAttribute("role") === "listitem" && String(selectionMode) === "Delete") {
          onSelectItem?.(currentIndex);
        }
        return;
      case "Enter":
        return;
      default:
        return;
    }
    if (newIndex !== currentIndex) {
      focusItem(newIndex);
    }
  }, [focusedIndex, getAllNavigable, getAllNavigableExcludingGrowingButton, selectionMode, onSelectItem, focusItem]);
  const registerItem = useCallback((_index, _element) => {
  }, []);
  return {
    focusedIndex,
    setFocusedIndex,
    focusItem,
    handleKeyDown,
    registerItem
  };
}
function ListHeader({ text, children, sticky, className }) {
  if (!text && !children) return null;
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: cn(
        "flex items-center gap-2.5 px-[0.75rem] py-[0.25rem] text-base font-normal text-sapphire-text-primary bg-background",
        sticky && "sticky top-0 z-10 bg-background",
        className
      ),
      role: "heading",
      "aria-level": 2,
      children: children || text
    }
  );
}
function ListLoading({ className }) {
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: cn(
        "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20",
        className
      ),
      role: "progressbar",
      "aria-label": "Loading",
      children: /* @__PURE__ */ jsx(Loader2, { className: "h-6 w-6 animate-spin text-sapphire-brand-foreground" })
    }
  );
}
function ListEmpty({ text, children, className }) {
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: cn(
        "flex items-center justify-center px-4 py-12 text-sm text-sapphire-text-tertiary",
        className
      ),
      role: "status",
      children: children || text || "No data"
    }
  );
}
function ListGrowing({
  mode,
  buttonText = "More",
  onLoadMore,
  parentRef,
  loading,
  accessibilityAttributes,
  buttonRef
}) {
  const observerRef = useRef(null);
  const triggerRef = useRef(null);
  useEffect(() => {
    if (String(mode) !== "Scroll" || !triggerRef.current) return;
    observerRef.current = new IntersectionObserver(
      (entries) => {
        const [entry] = entries;
        if (entry.isIntersecting && !loading) {
          onLoadMore?.();
        }
      },
      {
        root: parentRef?.current,
        rootMargin: "100px",
        threshold: 0
      }
    );
    observerRef.current.observe(triggerRef.current);
    return () => {
      observerRef.current?.disconnect();
    };
  }, [mode, onLoadMore, parentRef, loading]);
  if (String(mode) === "Button") {
    return /* @__PURE__ */ jsx("div", { className: "flex items-center justify-center py-3 border-t border-sapphire-border-primary", children: /* @__PURE__ */ jsxs(
      Button,
      {
        ref: buttonRef,
        design: "Tertiary" /* Tertiary */,
        onClick: () => onLoadMore?.(),
        disabled: loading,
        accessibleName: accessibilityAttributes?.name,
        "data-growing-button": "true",
        tabIndex: 0,
        className: "w-full mx-4",
        children: [
          loading ? /* @__PURE__ */ jsx(Loader2, { className: "h-4 w-4 animate-spin mr-2" }) : null,
          buttonText
        ]
      }
    ) });
  }
  if (String(mode) === "Scroll") {
    return /* @__PURE__ */ jsx(
      "div",
      {
        ref: triggerRef,
        className: "h-1",
        "aria-hidden": "true",
        "data-growing-trigger": true
      }
    );
  }
  return null;
}
function ListInner(props) {
  const {
    ref,
    // Header & Footer
    headerText,
    header,
    // Selection
    selectionMode = "None" /* None */,
    selectedKeys: controlledSelectedKeys,
    defaultSelectedKeys = [],
    onSelectionChange,
    // Data
    items,
    renderItem,
    getItemKey = (item, index) => String(item.id ?? item.key ?? index),
    children,
    // Empty & Loading
    noDataText = "No data available",
    noDataSlot,
    loading = false,
    loadingDelay = 1e3,
    // Growing
    growing = "None" /* None */,
    growingButtonText,
    hasMore = false,
    onLoadMore,
    // Visual
    separators = "All" /* All */,
    indent = false,
    stickyHeader = false,
    stickyFooter = false,
    // Events
    onItemClick,
    onItemDelete,
    onItemFocused,
    // Drag and Drop
    onMoveOver,
    onMove,
    // Accessibility
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    accessibleRole = "List" /* List */,
    accessibilityAttributes,
    // Standard
    className,
    style,
    id: providedId,
    "data-testid": dataTestId,
    "data-ai-context": dataAiContext
  } = props;
  const generatedId = useId();
  const listId = providedId ?? generatedId;
  const containerRef = useRef(null);
  const listRef = useRef(null);
  const growingButtonRef = useRef(null);
  const beforeSentinelRef = useRef(null);
  const afterSentinelRef = useRef(null);
  const [showLoading, setShowLoading] = useState(false);
  const [mediaRange, setMediaRange] = useState("M");
  const itemRegistry = useRef(/* @__PURE__ */ new Map());
  const [previouslyFocusedItemKey, setPreviouslyFocusedItemKey] = useState(null);
  const [draggedKey, setDraggedKey] = useState(null);
  const [dropTargetKey, setDropTargetKey] = useState(null);
  const [dropPlacement, setDropPlacement] = useState(null);
  const selectionState = useListSelection({
    mode: selectionMode,
    controlledKeys: controlledSelectedKeys,
    defaultKeys: defaultSelectedKeys,
    items,
    getItemKey,
    onSelectionChange
  });
  const itemCount = useMemo(() => {
    if (items) return items.length;
    return React.Children.count(children);
  }, [items, children]);
  const navigationState = useListNavigation({
    onFocusItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (element?.getAttribute("role") === "listitem" && items) {
        const itemKey = element.getAttribute("data-item-key");
        if (itemKey) {
          const itemIndex = items.findIndex((item, idx) => getItemKey(item, idx) === itemKey);
          if (itemIndex >= 0) {
            onItemFocused?.({ item: items[itemIndex], index: itemIndex });
          }
        }
      }
    },
    onSelectItem: (index) => {
      const allNavigable = Array.from(
        containerRef.current?.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"]'
        ) || []
      );
      const element = allNavigable[index];
      if (!element || element.getAttribute("role") !== "listitem") return;
      if (!items || !getItemKey) return;
      const itemKey = element.getAttribute("data-item-key");
      if (!itemKey) return;
      const itemIndex = items.findIndex((item2, idx) => getItemKey(item2, idx) === itemKey);
      if (itemIndex < 0) return;
      const item = items[itemIndex];
      const key = getItemKey(item, itemIndex);
      if (String(selectionMode) === "Delete") {
        onItemDelete?.({ item });
      } else {
        selectionState.toggleSelection(key);
      }
    },
    selectionMode,
    containerRef
  });
  useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  useEffect(() => {
    const updateMediaRange = () => {
      const width = window.innerWidth;
      if (width < 600) setMediaRange("S");
      else if (width < 1024) setMediaRange("M");
      else if (width < 1440) setMediaRange("L");
      else setMediaRange("XL");
    };
    updateMediaRange();
    window.addEventListener("resize", updateMediaRange);
    return () => window.removeEventListener("resize", updateMediaRange);
  }, []);
  const registerItem = useCallback((key, index) => {
    itemRegistry.current.set(key, index);
  }, []);
  const unregisterItem = useCallback((key) => {
    itemRegistry.current.delete(key);
  }, []);
  const handleDragStart = useCallback((key, _e) => {
    setDraggedKey(key);
  }, []);
  const handleDragEnd = useCallback((_key, _e) => {
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, []);
  const handleDragOver = useCallback((key, e) => {
    if (draggedKey && key === draggedKey) return;
    setDropTargetKey(key);
    const rect = e.currentTarget.getBoundingClientRect();
    const y = e.clientY - rect.top;
    const height = rect.height;
    if (y < height * 0.33) {
      setDropPlacement("Before");
    } else if (y > height * 0.67) {
      setDropPlacement("After");
    } else {
      setDropPlacement("On");
    }
  }, [draggedKey]);
  const handleDrop = useCallback((targetKey, e) => {
    const targetIndex = itemRegistry.current.get(targetKey);
    if (targetIndex === void 0) return;
    if (!items || !getItemKey) return;
    const targetItem = items[targetIndex];
    if (!targetItem) return;
    const droppedKey = draggedKey || e.dataTransfer.getData("text/plain");
    if (!droppedKey || targetKey === droppedKey) return;
    const placement = dropPlacement || "After";
    const sourceIndex = itemRegistry.current.get(droppedKey);
    const sourceItem = sourceIndex !== void 0 ? items[sourceIndex] : void 0;
    const isMultiple = String(selectionMode) === "Multiple";
    const draggedIsSelected = droppedKey ? selectionState.isSelected(droppedKey) : false;
    const selectedItemsList = isMultiple && draggedIsSelected ? selectionState.getSelectedItems() : void 0;
    const moveDetail = {
      source: {
        element: sourceItem,
        // May be undefined for cross-list drops
        elements: selectedItemsList,
        index: sourceIndex ?? -1
        // -1 indicates cross-list drop
      },
      destination: {
        element: targetItem,
        placement,
        index: targetIndex
      },
      originalEvent: e.nativeEvent
    };
    onMoveOver?.(moveDetail);
    onMove?.(moveDetail);
    setDraggedKey(null);
    setDropTargetKey(null);
    setDropPlacement(null);
  }, [draggedKey, dropPlacement, items, getItemKey, selectionMode, selectionState, onMoveOver, onMove]);
  const handleForwardTabAfter = useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      const itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      );
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (String(growing) !== "None" && hasMore && growingButtonRef.current) {
      growingButtonRef.current.focus?.();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex + 1; i < allTabbable.length; i++) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (afterSentinelRef.current) {
          afterSentinelRef.current.focus();
        }
      }
    }
  }, [growing, hasMore, navigationState]);
  const handleForwardTabBefore = useCallback((key, event) => {
    setPreviouslyFocusedItemKey(key);
    let itemElement = null;
    if (containerRef.current) {
      const allNavigable = Array.from(
        containerRef.current.querySelectorAll(
          'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
        )
      );
      itemElement = allNavigable.find(
        (el) => el.getAttribute("data-item-key") === key || el.id === key
      ) || null;
      if (itemElement) {
        const itemIndex = allNavigable.indexOf(itemElement);
        navigationState.setFocusedIndex(itemIndex);
      }
    }
    if (itemElement && event.target !== itemElement) {
      itemElement.focus();
      return;
    }
    if (listRef.current) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const target = event.target;
      const currentIndex = allTabbable.indexOf(target);
      if (currentIndex >= 0) {
        for (let i = currentIndex - 1; i >= 0; i--) {
          if (!listRef.current.contains(allTabbable[i])) {
            allTabbable[i].focus();
            return;
          }
        }
      } else {
        if (beforeSentinelRef.current) {
          beforeSentinelRef.current.focus();
        }
      }
    }
  }, [navigationState]);
  const handleBeforeSentinelFocus = useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromAfterSentinel = relatedTarget === afterSentinelRef.current;
    const isFromListItem = relatedTarget?.closest('li[role="listitem"], li[role="option"]');
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    if (isFromAfterSentinel || isFromListItem || isFromGrowingButton) {
      const allTabbable = Array.from(
        document.querySelectorAll(
          'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
        )
      );
      const sentinelIndex = allTabbable.indexOf(beforeSentinelRef.current);
      if (sentinelIndex > 0) {
        allTabbable[sentinelIndex - 1].focus();
      }
    } else {
      e.preventDefault();
      if (containerRef.current) {
        const allNavigable = Array.from(
          containerRef.current.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"], li[role="option"], button[data-growing-button]'
          )
        );
        if (allNavigable.length === 0) {
          if (afterSentinelRef.current) {
            afterSentinelRef.current.focus();
          }
          return;
        }
        if (previouslyFocusedItemKey) {
          const targetElement = allNavigable.find(
            (el) => el.getAttribute("data-item-key") === previouslyFocusedItemKey || el.id === previouslyFocusedItemKey
          );
          if (targetElement) {
            targetElement.focus();
            return;
          }
        }
        if (allNavigable[0]) {
          allNavigable[0].focus();
        }
      }
    }
  }, [previouslyFocusedItemKey]);
  const getAllTabbableElements = useCallback(() => {
    return Array.from(
      document.querySelectorAll(
        'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
      )
    );
  }, []);
  const focusNextAfterList = useCallback(() => {
    if (!listRef.current || !afterSentinelRef.current) return false;
    const allTabbable = getAllTabbableElements();
    const sentinelIndex = allTabbable.indexOf(afterSentinelRef.current);
    if (sentinelIndex >= 0) {
      for (let i = sentinelIndex + 1; i < allTabbable.length; i++) {
        if (!containerRef.current?.contains(allTabbable[i])) {
          allTabbable[i].focus();
          return true;
        }
      }
    }
    return false;
  }, [getAllTabbableElements]);
  const enterListAtFocusedItem = useCallback(() => {
    if (!containerRef.current) return false;
    const allNavigable = Array.from(
      containerRef.current.querySelectorAll(
        'div[data-group-header="true"], li[role="listitem"], li[role="option"]'
      )
    );
    if (allNavigable.length === 0) {
      beforeSentinelRef.current?.focus();
      return true;
    }
    const focusedIndex = navigationState.focusedIndex;
    const targetIndex = focusedIndex >= 0 && focusedIndex < allNavigable.length ? focusedIndex : 0;
    if (allNavigable[targetIndex]) {
      allNavigable[targetIndex].focus();
      return true;
    }
    return false;
  }, [navigationState.focusedIndex]);
  const handleAfterSentinelFocus = useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const isFromGrowingButton = relatedTarget?.hasAttribute("data-growing-button");
    const isFromOutside = !relatedTarget || !containerRef.current?.contains(relatedTarget);
    if (isFromGrowingButton || isFromOutside) {
      enterListAtFocusedItem();
    } else {
      focusNextAfterList();
    }
  }, [focusNextAfterList, enterListAtFocusedItem]);
  const roleSupportsAriaSelected = useMemo(() => {
    const roleStr = String(accessibleRole);
    return roleStr === "ListBox" /* ListBox */ || roleStr === "listbox" || roleStr === "Tree" /* Tree */ || roleStr === "tree";
  }, [accessibleRole]);
  const contextValue = useMemo(() => ({
    selectionMode,
    selectedKeys: selectionState.selectedKeys,
    onSelect: selectionState.setSelected,
    onToggleSelect: selectionState.toggleSelection,
    isSelected: selectionState.isSelected,
    focusedIndex: navigationState.focusedIndex,
    onFocusItem: navigationState.setFocusedIndex,
    registerItem,
    unregisterItem,
    registerNavigationItem: navigationState.registerItem,
    // Tab navigation
    onForwardTabAfter: handleForwardTabAfter,
    onForwardTabBefore: handleForwardTabBefore,
    separators,
    indent,
    onItemClick: (_key, event) => {
      onItemClick?.({ item: event.currentTarget, originalEvent: event });
    },
    onItemDelete: (key) => {
      const index = itemRegistry.current.get(key);
      if (index !== void 0 && items) {
        onItemDelete?.({ item: items[index] });
        return;
      }
      if (items && getItemKey) {
        const item = items.find((item2, idx) => getItemKey(item2, idx) === key);
        if (item) {
          onItemDelete?.({ item });
        }
      }
    },
    // Drag and drop
    onDragStart: handleDragStart,
    onDragEnd: handleDragEnd,
    onDragOver: handleDragOver,
    onDrop: handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    mediaRange,
    totalItemCount: itemCount,
    getItemIndex: (key) => {
      return itemRegistry.current.get(key) ?? -1;
    },
    // Accessibility
    accessibleRole,
    supportsAriaSelected: roleSupportsAriaSelected
  }), [
    selectionMode,
    selectionState,
    navigationState,
    registerItem,
    unregisterItem,
    handleForwardTabAfter,
    handleForwardTabBefore,
    separators,
    indent,
    onItemClick,
    onItemDelete,
    items,
    getItemKey,
    mediaRange,
    itemCount,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDrop,
    draggedKey,
    dropTargetKey,
    dropPlacement,
    accessibleRole,
    roleSupportsAriaSelected
  ]);
  useImperativeHandle(ref, () => ({
    // Focus
    focus: () => listRef.current?.focus(),
    focusItem: (index) => navigationState.focusItem(index),
    getFocusedItemIndex: () => navigationState.focusedIndex,
    // Selection
    getSelectedKeys: () => Array.from(selectionState.selectedKeys),
    getSelectedItems: () => selectionState.getSelectedItems(),
    isSelected: selectionState.isSelected,
    select: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, true));
    },
    deselect: (keys) => {
      const keyArray = Array.isArray(keys) ? keys : [keys];
      keyArray.forEach((key) => selectionState.setSelected(key, false));
    },
    toggleSelection: selectionState.toggleSelection,
    selectAll: selectionState.selectAll,
    deselectAll: selectionState.deselectAll,
    // Data
    getItems: () => items ?? [],
    getItemCount: () => itemCount,
    getItemByKey: (key) => items?.find((item, index) => getItemKey(item, index) === key),
    // Scrolling
    scrollToItem: (index, align = "start") => {
      const element = listRef.current?.children[index];
      element?.scrollIntoView({ behavior: "smooth", block: align });
    },
    scrollToTop: () => {
      containerRef.current?.scrollTo({ top: 0, behavior: "smooth" });
    },
    scrollToBottom: () => {
      containerRef.current?.scrollTo({
        top: containerRef.current.scrollHeight,
        behavior: "smooth"
      });
    },
    // Elements
    get listElement() {
      return listRef.current;
    },
    get scrollContainer() {
      return containerRef.current;
    }
  }), [selectionState, navigationState, items, itemCount, getItemKey]);
  const handleLoadMore = useCallback(() => {
    onLoadMore?.({ currentCount: itemCount });
  }, [onLoadMore, itemCount]);
  const getRole = () => {
    const roleStr = String(accessibleRole);
    switch (roleStr) {
      case "Menu":
        return "menu";
      case "Tree":
        return "tree";
      case "ListBox":
        return "listbox";
      default:
        return "list";
    }
  };
  const isEmpty = itemCount === 0 && !loading;
  const renderedItems = useMemo(() => {
    itemRegistry.current.clear();
    if (items && renderItem) {
      return items.map((item, index) => {
        const key = getItemKey(item, index);
        itemRegistry.current.set(key, index);
        return /* @__PURE__ */ jsx(React.Fragment, { children: renderItem(item, index) }, key);
      });
    }
    if (children) {
      let index = 0;
      React.Children.forEach(children, (child) => {
        if (React.isValidElement(child) && child.props && typeof child.props.itemKey === "string") {
          itemRegistry.current.set(child.props.itemKey, index);
          index++;
        }
      });
    }
    return children;
  }, [items, renderItem, getItemKey, children]);
  return /* @__PURE__ */ jsx(ListContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxs(
    "div",
    {
      ref: containerRef,
      className: cn(
        listContainerVariants({ bordered: false }),
        // Make scrollable when growing mode or sticky header is used
        (String(growing) !== "None" || stickyHeader || stickyFooter) && "overflow-auto",
        // Always prevent horizontal scroll
        "overflow-x-hidden",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-ai-context": dataAiContext,
      onKeyDown: navigationState.handleKeyDown,
      children: [
        /* @__PURE__ */ jsx(
          ListHeader,
          {
            text: headerText,
            sticky: stickyHeader,
            children: header
          }
        ),
        /* @__PURE__ */ jsx(
          "span",
          {
            ref: beforeSentinelRef,
            tabIndex: 0,
            onFocus: handleBeforeSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        /* @__PURE__ */ jsx(
          "ul",
          {
            ref: listRef,
            id: listId,
            role: getRole(),
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-describedby": accessibleDescriptionRef,
            "aria-busy": loading || void 0,
            "aria-multiselectable": roleSupportsAriaSelected && String(selectionMode) === "Multiple" ? true : void 0,
            className: cn(listVariants({ indent, separators: String(separators) !== "None" })),
            children: !isEmpty && renderedItems
          }
        ),
        /* @__PURE__ */ jsx(
          "span",
          {
            ref: afterSentinelRef,
            tabIndex: 0,
            onFocus: handleAfterSentinelFocus,
            style: {
              position: "absolute",
              width: "1px",
              height: "1px",
              padding: 0,
              margin: "-1px",
              overflow: "hidden",
              clip: "rect(0, 0, 0, 0)",
              whiteSpace: "nowrap",
              border: 0
            }
          }
        ),
        isEmpty && /* @__PURE__ */ jsx(ListEmpty, { text: noDataText, children: noDataSlot }),
        String(growing) !== "None" && hasMore && !loading && /* @__PURE__ */ jsx(
          ListGrowing,
          {
            mode: growing,
            buttonText: growingButtonText,
            onLoadMore: handleLoadMore,
            parentRef: containerRef,
            loading,
            accessibilityAttributes: accessibilityAttributes?.growingButton,
            buttonRef: growingButtonRef
          }
        ),
        showLoading && /* @__PURE__ */ jsx(ListLoading, {})
      ]
    }
  ) });
}
var List = React.memo(ListInner);
List.displayName = "List";

export { List, ListAccessibleRole, ListContext, ListGrowingMode, ListItemAccessibleRole, ListItemHighlight, ListItemType, ListItemValueState, ListItemWrappingType, ListSelectionMode, ListSeparator, useListContext, useOptionalListContext };
