// @sap-ui/fx-components/dist/chunk-MCJ7D5KG.js
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useRef, useImperativeHandle, useMemo, useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/link.ts
var LinkDesign = /* @__PURE__ */ ((LinkDesign2) => {
  LinkDesign2["Default"] = "Default";
  LinkDesign2["Subtle"] = "Subtle";
  LinkDesign2["Emphasized"] = "Emphasized";
  return LinkDesign2;
})(LinkDesign || {});
var LinkInteractiveAreaSize = /* @__PURE__ */ ((LinkInteractiveAreaSize2) => {
  LinkInteractiveAreaSize2["Normal"] = "Normal";
  LinkInteractiveAreaSize2["Large"] = "Large";
  return LinkInteractiveAreaSize2;
})(LinkInteractiveAreaSize || {});
var LinkWrappingType = /* @__PURE__ */ ((LinkWrappingType2) => {
  LinkWrappingType2["None"] = "None";
  LinkWrappingType2["Normal"] = "Normal";
  return LinkWrappingType2;
})(LinkWrappingType || {});
var LinkAccessibleRole = /* @__PURE__ */ ((LinkAccessibleRole2) => {
  LinkAccessibleRole2["Link"] = "Link";
  LinkAccessibleRole2["Button"] = "Button";
  return LinkAccessibleRole2;
})(LinkAccessibleRole || {});
var SAFE_PROTOCOLS = ["http:", "https:", "mailto:", "tel:"];
var isSafeUrl = (url) => {
  try {
    const parsed = new URL(url, window.location.href);
    return SAFE_PROTOCOLS.includes(parsed.protocol);
  } catch {
    return false;
  }
};
var computeRel = (href, target, rel) => {
  if (rel) return rel;
  if (target === "_blank" && href) {
    try {
      const url = new URL(href, window.location.href);
      if (url.origin !== window.location.origin) {
        return "noreferrer noopener";
      }
    } catch {
    }
  }
  return void 0;
};
var linkVariants = cva(
  [
    // Layout
    "inline-flex items-center gap-0.5",
    // Typography — Figma: '72' Regular, 14px, line-height 20px
    "text-sm leading-5 font-normal cursor-pointer",
    // Transition
    "transition-colors",
    // Focus box: constant padding + transparent border to prevent layout shift.
    // Figma focus spec: border-focus, rounded 4px, px-4px py-2px.
    // Negative margins compensate for the padding+border so the link
    // sits flush in normal state.
    "px-1 py-0.5 -mx-1.5 -my-1",
    "border-2 border-transparent rounded-[4px]",
    "focus-visible:outline-none focus-visible:border-sapphire-border-focus"
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: [
          "text-sapphire-text-accent",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ],
        ["Subtle" /* Subtle */]: [
          "text-secondary-foreground",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ],
        ["Emphasized" /* Emphasized */]: [
          "text-sapphire-text-accent font-semibold",
          "no-underline",
          "hover:text-sapphire-brand-hover-background",
          "active:text-sapphire-brand-pressed-foreground"
        ]
      },
      interactiveAreaSize: {
        ["Normal" /* Normal */]: "",
        ["Large" /* Large */]: "leading-6"
      },
      wrappingType: {
        ["None" /* None */]: "whitespace-nowrap max-w-full",
        ["Normal" /* Normal */]: "whitespace-normal break-words"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      interactiveAreaSize: "Normal" /* Normal */,
      wrappingType: "None" /* None */
    }
  }
);
function Link({
  design = "Default" /* Default */,
  interactiveAreaSize = "Normal" /* Normal */,
  wrappingType = "None" /* None */,
  href,
  target,
  rel,
  disabled = false,
  icon,
  endIcon,
  tooltip,
  children,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleRole = "Link" /* Link */,
  accessibilityAttributes,
  onClick,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  ref
}) {
  const linkRef = useRef(null);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        linkRef.current?.focus();
      },
      blur() {
        linkRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === linkRef.current;
      },
      get nativeElement() {
        return linkRef.current;
      }
    }),
    []
  );
  const { t } = useTranslation("fx");
  const linkTypeText = useMemo(() => {
    if (design === "Subtle" /* Subtle */) return t("LINK_SUBTLE");
    if (design === "Emphasized" /* Emphasized */) return t("LINK_EMPHASIZED");
    return void 0;
  }, [design, t]);
  const computedRel = useMemo(
    () => computeRel(href, target, rel),
    [href, target, rel]
  );
  const effectiveTabIndex = disabled ? -1 : tabIndex ?? 0;
  const handleClick = useCallback(
    (e) => {
      if (disabled) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        altKey: e.altKey,
        ctrlKey: e.ctrlKey,
        metaKey: e.metaKey,
        shiftKey: e.shiftKey,
        isKeyboard: false
      });
    },
    [disabled, onClick]
  );
  const handleMouseDown = useCallback(
    (_e) => {
      if (!disabled) {
        linkRef.current?.focus();
      }
    },
    [disabled]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " " && accessibleRole !== "Button" /* Button */) {
          return;
        }
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          altKey: e.altKey,
          ctrlKey: e.ctrlKey,
          metaKey: e.metaKey,
          shiftKey: e.shiftKey,
          isKeyboard: true
        });
        if (href && !disabled && isSafeUrl(href)) {
          window.location.href = href;
        }
      }
    },
    [disabled, onClick, href, accessibleRole]
  );
  return /* @__PURE__ */ jsxs(
    "a",
    {
      ref: linkRef,
      id,
      href: disabled ? void 0 : href,
      target,
      rel: computedRel,
      role: accessibleRole === "Button" /* Button */ ? "button" : void 0,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-describedby": accessibleDescription,
      "aria-disabled": disabled || void 0,
      "aria-expanded": accessibilityAttributes?.expanded,
      "aria-haspopup": accessibilityAttributes?.hasPopup,
      "aria-current": accessibilityAttributes?.current || void 0,
      title: tooltip,
      tabIndex: effectiveTabIndex,
      onClick: handleClick,
      onMouseDown: handleMouseDown,
      onKeyDown: handleKeyDown,
      className: cn(
        linkVariants({
          design,
          interactiveAreaSize,
          wrappingType
        }),
        disabled && "pointer-events-none cursor-default opacity-[var(--disabled-opacity)]",
        "group",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        icon && /* @__PURE__ */ jsx("span", { className: "flex-shrink-0 size-3", "data-part": "icon", children: icon }),
        children && /* @__PURE__ */ jsx(
          "span",
          {
            className: cn(
              wrappingType === "None" /* None */ && "overflow-hidden text-ellipsis min-w-0",
              design === "Subtle" /* Subtle */ && "underline decoration-dotted underline-offset-[0.25em]",
              (design === "Default" /* Default */ || design === "Emphasized" /* Emphasized */) && "group-hover:underline group-hover:decoration-dotted group-hover:underline-offset-[0.25em]"
            ),
            "data-part": "text",
            children
          }
        ),
        endIcon && /* @__PURE__ */ jsx("span", { className: "flex-shrink-0 size-3", "data-part": "end-icon", children: endIcon }),
        linkTypeText && /* @__PURE__ */ jsx("span", { className: "sr-only", "data-part": "link-type", children: linkTypeText })
      ]
    }
  );
}

export { Link, LinkAccessibleRole, LinkDesign, LinkInteractiveAreaSize, LinkWrappingType };
