// @sap-ui/fx-components/dist/chunk-OXMWPC4U.js
import { PersonPlaceholderIcon } from './chunk-LCV3JX36.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useImperativeHandle } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/avatar.ts
var AvatarShape = /* @__PURE__ */ ((AvatarShape2) => {
  AvatarShape2["Circle"] = "Circle";
  AvatarShape2["Square"] = "Square";
  return AvatarShape2;
})(AvatarShape || {});
var AvatarSize = /* @__PURE__ */ ((AvatarSize2) => {
  AvatarSize2["XS"] = "XS";
  AvatarSize2["S"] = "S";
  AvatarSize2["M"] = "M";
  AvatarSize2["L"] = "L";
  AvatarSize2["XL"] = "XL";
  return AvatarSize2;
})(AvatarSize || {});
var AvatarColorScheme = /* @__PURE__ */ ((AvatarColorScheme2) => {
  AvatarColorScheme2["Accent1"] = "Accent1";
  AvatarColorScheme2["Accent2"] = "Accent2";
  AvatarColorScheme2["Accent3"] = "Accent3";
  AvatarColorScheme2["Accent4"] = "Accent4";
  AvatarColorScheme2["Accent5"] = "Accent5";
  AvatarColorScheme2["Accent6"] = "Accent6";
  AvatarColorScheme2["Accent7"] = "Accent7";
  AvatarColorScheme2["Accent8"] = "Accent8";
  AvatarColorScheme2["Accent9"] = "Accent9";
  AvatarColorScheme2["Accent10"] = "Accent10";
  AvatarColorScheme2["Placeholder"] = "Placeholder";
  return AvatarColorScheme2;
})(AvatarColorScheme || {});
var avatarVariants = cva(
  [
    "inline-flex items-center justify-center",
    "font-medium overflow-hidden",
    "select-none"
  ],
  {
    variants: {
      shape: {
        ["Circle" /* Circle */]: "rounded-full",
        ["Square" /* Square */]: ""
      },
      size: {
        ["XS" /* XS */]: "h-[2rem] w-[2rem] text-[1rem]",
        ["S" /* S */]: "h-[3rem] w-[3rem] text-[1.125rem]",
        ["M" /* M */]: "h-[4rem] w-[4rem] text-[1.5rem]",
        ["L" /* L */]: "h-[5rem] w-[5rem] text-[2.25rem]",
        ["XL" /* XL */]: "h-[7rem] w-[7rem] text-[3rem]"
      },
      colorScheme: {
        ["Accent1" /* Accent1 */]: "bg-[var(--avatar-accent-bg-1)] text-[var(--avatar-accent-color-1)]",
        ["Accent2" /* Accent2 */]: "bg-[var(--avatar-accent-bg-2)] text-[var(--avatar-accent-color-2)]",
        ["Accent3" /* Accent3 */]: "bg-[var(--avatar-accent-bg-3)] text-[var(--avatar-accent-color-3)]",
        ["Accent4" /* Accent4 */]: "bg-[var(--avatar-accent-bg-4)] text-[var(--avatar-accent-color-4)]",
        ["Accent5" /* Accent5 */]: "bg-[var(--avatar-accent-bg-5)] text-[var(--avatar-accent-color-5)]",
        ["Accent6" /* Accent6 */]: "bg-[var(--avatar-accent-bg-6)] text-[var(--avatar-accent-color-6)]",
        ["Accent7" /* Accent7 */]: "bg-[var(--avatar-accent-bg-7)] text-[var(--avatar-accent-color-7)]",
        ["Accent8" /* Accent8 */]: "bg-[var(--avatar-accent-bg-8)] text-[var(--avatar-accent-color-8)]",
        ["Accent9" /* Accent9 */]: "bg-[var(--avatar-accent-bg-9)] text-[var(--avatar-accent-color-9)]",
        ["Accent10" /* Accent10 */]: "bg-[var(--avatar-accent-bg-10)] text-[var(--avatar-accent-color-10)]",
        ["Placeholder" /* Placeholder */]: "bg-muted text-sapphire-text-tertiary"
      },
      interactive: {
        true: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
        false: ""
      },
      disabled: {
        true: "opacity-50 cursor-not-allowed",
        false: ""
      }
    },
    compoundVariants: [
      {
        shape: "Square" /* Square */,
        size: "XS" /* XS */,
        className: "rounded-sapphire-s"
        // 4px — Figma token --radius-s
      },
      {
        shape: "Square" /* Square */,
        size: "S" /* S */,
        className: "rounded-[6px]"
        // 6px — hardcoded in Figma
      },
      {
        shape: "Square" /* Square */,
        size: "M" /* M */,
        className: "rounded-sapphire-m"
        // 8px — Figma token --radius-m
      },
      {
        shape: "Square" /* Square */,
        size: "L" /* L */,
        className: "rounded-[12px]"
        // 12px — hardcoded in Figma
      },
      {
        shape: "Square" /* Square */,
        size: "XL" /* XL */,
        className: "rounded-sapphire-l"
        // 16px — Figma token --radius-l
      },
      // Neutralize interactive hover/focus/cursor when disabled
      {
        interactive: true,
        disabled: true,
        className: "cursor-not-allowed hover:opacity-50 focus:ring-0 focus:ring-offset-0"
      }
    ],
    defaultVariants: {
      shape: "Circle" /* Circle */,
      size: "M" /* M */,
      colorScheme: "Accent6" /* Accent6 */,
      interactive: false,
      disabled: false
    }
  }
);
var iconSizeMap = {
  ["XS" /* XS */]: "h-[1rem] w-[1rem]",
  ["S" /* S */]: "h-[1.125rem] w-[1.125rem]",
  ["M" /* M */]: "h-[1.5rem] w-[1.5rem]",
  ["L" /* L */]: "h-[2.25rem] w-[2.25rem]",
  ["XL" /* XL */]: "h-[3rem] w-[3rem]"
};
function Avatar({
  shape = "Circle" /* Circle */,
  size = "M" /* M */,
  colorScheme = "Accent6" /* Accent6 */,
  icon,
  image,
  fallbackImage,
  initials,
  interactive = false,
  disabled = false,
  accessibleName = "Avatar",
  badge,
  onClick,
  className,
  style,
  ref
}) {
  const avatarRef = useRef(null);
  const [imageError, setImageError] = useState(false);
  const [fallbackError, setFallbackError] = useState(false);
  useImperativeHandle(ref, () => ({
    focus: () => avatarRef.current?.focus(),
    isFocused: () => document.activeElement === avatarRef.current,
    get nativeElement() {
      return avatarRef.current;
    },
    getNativeElement: () => avatarRef.current
  }));
  const handleImageError = () => {
    if (!imageError) {
      setImageError(true);
    } else if (fallbackImage && !fallbackError) {
      setFallbackError(true);
    }
  };
  const handleClick = (e) => {
    if (!disabled && interactive && onClick) {
      onClick(e);
    }
  };
  const handleKeyDown = (e) => {
    if (!disabled && interactive && (e.key === "Enter" || e.key === " ")) {
      e.preventDefault();
      onClick?.(e);
    }
  };
  const showImage = image && !imageError;
  const showFallbackImage = !showImage && fallbackImage && !fallbackError;
  const showInitials = !showImage && !showFallbackImage && initials;
  const showIcon = !showImage && !showFallbackImage && !showInitials;
  const iconSize = iconSizeMap[size] || iconSizeMap["M" /* M */];
  return /* @__PURE__ */ jsxs("div", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: avatarRef,
        role: interactive ? "button" : "img",
        tabIndex: interactive && !disabled ? 0 : void 0,
        "aria-label": showInitials ? `${accessibleName} ${initials}` : accessibleName,
        "aria-disabled": disabled,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        className: cn(
          avatarVariants({
            shape,
            size,
            colorScheme,
            interactive,
            disabled
          }),
          className
        ),
        style,
        children: [
          showImage && /* @__PURE__ */ jsx(
            "img",
            {
              src: image,
              alt: accessibleName || "",
              className: "h-full w-full object-cover",
              onError: handleImageError
            }
          ),
          showFallbackImage && /* @__PURE__ */ jsx(
            "img",
            {
              src: fallbackImage,
              alt: accessibleName || "",
              className: "h-full w-full object-cover",
              onError: handleImageError
            }
          ),
          showInitials && /* @__PURE__ */ jsx("span", { className: "uppercase", children: initials.slice(0, 3) }),
          showIcon && (icon ? /* @__PURE__ */ jsx("span", { className: cn(iconSize, "flex items-center justify-center"), children: icon }) : /* @__PURE__ */ jsx(PersonPlaceholderIcon, { className: iconSize }))
        ]
      }
    ),
    badge && /* @__PURE__ */ jsx("div", { className: "absolute top-0 right-0", children: badge })
  ] });
}

export { Avatar, AvatarColorScheme, AvatarShape, AvatarSize };
