// @sap-ui/fx-components/dist/chunk-DTDQJNHA.js
import { cva, cn } from './chunk-2F2PZMJ5.js';
import * as React from 'react';
import { createContext, useContext, useRef, useState, useEffect } from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

// src/types/illustrated-message.ts
var IllustrationDesign = /* @__PURE__ */ ((IllustrationDesign2) => {
  IllustrationDesign2["Auto"] = "Auto";
  IllustrationDesign2["Base"] = "Base";
  IllustrationDesign2["ExtraSmall"] = "ExtraSmall";
  IllustrationDesign2["Small"] = "Small";
  IllustrationDesign2["Medium"] = "Medium";
  IllustrationDesign2["Large"] = "Large";
  return IllustrationDesign2;
})(IllustrationDesign || {});
var IllustrationSize = /* @__PURE__ */ ((IllustrationSize2) => {
  IllustrationSize2["Base"] = "Base";
  IllustrationSize2["ExtraSmall"] = "ExtraSmall";
  IllustrationSize2["Small"] = "Small";
  IllustrationSize2["Medium"] = "Medium";
  IllustrationSize2["Large"] = "Large";
  return IllustrationSize2;
})(IllustrationSize || {});
var IllustrationSizeContext = createContext(
  "Medium" /* Medium */
);
function useIllustrationSizeContext() {
  return useContext(IllustrationSizeContext);
}
function useIllustrationSize(containerRef, design) {
  const isAuto = design === "Auto" /* Auto */ || design === "Auto";
  const [resolvedSize, setResolvedSize] = useState(
    isAuto ? "Medium" /* Medium */ : design
  );
  useEffect(() => {
    if (!isAuto) {
      setResolvedSize(design);
      return;
    }
    const el = containerRef.current;
    if (!el || typeof ResizeObserver === "undefined") return;
    const ro = new ResizeObserver(([entry]) => {
      const width = entry.contentBoxSize?.[0]?.inlineSize ?? entry.contentRect.width;
      let newSize;
      if (width <= 160) {
        newSize = "Base" /* Base */;
      } else if (width <= 260) {
        newSize = "ExtraSmall" /* ExtraSmall */;
      } else if (width <= 360) {
        newSize = "Small" /* Small */;
      } else if (width <= 681) {
        newSize = "Medium" /* Medium */;
      } else {
        newSize = "Large" /* Large */;
      }
      setResolvedSize((prev) => prev !== newSize ? newSize : prev);
    });
    ro.observe(el);
    return () => ro.disconnect();
  }, [design, isAuto, containerRef]);
  return resolvedSize;
}
var rootVariants = cva(
  ["flex w-full h-full box-border"],
  {
    variants: {
      size: {
        ["Large" /* Large */]: "flex-col items-center justify-center text-center gap-4 p-4",
        ["Medium" /* Medium */]: "flex-col items-center justify-center text-center gap-3 p-4",
        ["Small" /* Small */]: "flex-col items-center justify-center text-center gap-2 p-2",
        ["ExtraSmall" /* ExtraSmall */]: "flex-row items-center justify-center gap-3 p-2",
        ["Base" /* Base */]: "flex-col items-center justify-center text-center gap-1 p-1"
      }
    },
    defaultVariants: {
      size: "Large" /* Large */
    }
  }
);
var titleVariants = cva("text-foreground", {
  variants: {
    size: {
      ["Large" /* Large */]: "text-[32px] font-normal leading-[40px] max-w-[61.9375rem]",
      ["Medium" /* Medium */]: "text-xl font-semibold leading-[32px] max-w-[40.5625rem]",
      ["Small" /* Small */]: "text-xl font-semibold leading-[32px] max-w-[21.5rem]",
      ["ExtraSmall" /* ExtraSmall */]: "text-base font-semibold leading-[22px]",
      ["Base" /* Base */]: "text-sm font-semibold leading-[22px] max-w-[10rem]"
    }
  }
});
var subtitleVariants = cva("text-secondary-foreground", {
  variants: {
    size: {
      ["Large" /* Large */]: "text-base leading-[22px] max-w-[61.9375rem]",
      ["Medium" /* Medium */]: "text-sm leading-[20px] max-w-[40.5625rem]",
      ["Small" /* Small */]: "text-sm leading-[20px] max-w-[21.5rem]",
      ["ExtraSmall" /* ExtraSmall */]: "text-xs leading-[20px]",
      ["Base" /* Base */]: "text-xs leading-[20px] max-w-[10rem]"
    }
  }
});
var illustrationSizeClasses = {
  ["Large" /* Large */]: "w-80 h-60",
  ["Medium" /* Medium */]: "w-40 h-40",
  ["Small" /* Small */]: "w-32 h-32",
  ["ExtraSmall" /* ExtraSmall */]: "w-[2.8125rem] h-[2.8125rem] flex-shrink-0",
  ["Base" /* Base */]: "hidden"
};
function IllustratedMessage({
  illustration,
  design = "Auto" /* Auto */,
  titleText,
  subtitleText,
  title,
  subtitle,
  children,
  decorative = false,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const containerRef = useRef(null);
  const resolvedSize = useIllustrationSize(containerRef, design);
  React.useImperativeHandle(ref, () => ({
    get nativeElement() {
      return containerRef.current;
    }
  }), []);
  const hasTitle = !!(title || titleText);
  const hasSubtitle = !!(subtitle || subtitleText);
  const showIllustration = resolvedSize !== "Base" /* Base */;
  const showActions = resolvedSize !== "Base" /* Base */;
  const effectiveAccessibleName = accessibleName || "Illustration";
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref: containerRef,
      id,
      className: cn(rootVariants({ size: resolvedSize }), className),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: /* @__PURE__ */ jsxs(IllustrationSizeContext.Provider, { value: resolvedSize, children: [
        showIllustration && illustration && /* @__PURE__ */ jsx(
          "div",
          {
            "data-part": "illustration",
            role: decorative ? "presentation" : "img",
            "aria-hidden": decorative || void 0,
            "aria-label": !decorative ? effectiveAccessibleName : void 0,
            className: illustrationSizeClasses[resolvedSize],
            children: illustration
          }
        ),
        resolvedSize === "ExtraSmall" /* ExtraSmall */ ? /* @__PURE__ */ jsxs("div", { className: "flex flex-col items-center gap-2", children: [
          (hasTitle || hasSubtitle) && /* @__PURE__ */ jsxs("div", { "data-part": "content", className: "text-center", children: [
            hasTitle && /* @__PURE__ */ jsx(
              "div",
              {
                "data-part": "title",
                className: titleVariants({ size: resolvedSize }),
                children: title || titleText
              }
            ),
            hasSubtitle && /* @__PURE__ */ jsx(
              "div",
              {
                "data-part": "subtitle",
                className: subtitleVariants({ size: resolvedSize }),
                children: subtitle || subtitleText
              }
            )
          ] }),
          showActions && children && /* @__PURE__ */ jsx("div", { "data-part": "actions", className: "flex gap-2", children })
        ] }) : /* @__PURE__ */ jsxs(Fragment, { children: [
          (hasTitle || hasSubtitle) && /* @__PURE__ */ jsxs("div", { "data-part": "content", children: [
            hasTitle && /* @__PURE__ */ jsx(
              "div",
              {
                "data-part": "title",
                className: titleVariants({ size: resolvedSize }),
                children: title || titleText
              }
            ),
            hasSubtitle && /* @__PURE__ */ jsx(
              "div",
              {
                "data-part": "subtitle",
                className: subtitleVariants({ size: resolvedSize }),
                children: subtitle || subtitleText
              }
            )
          ] }),
          showActions && children && /* @__PURE__ */ jsx("div", { "data-part": "actions", className: "flex gap-2 mt-2", children })
        ] })
      ] })
    }
  );
}

export { IllustratedMessage, IllustrationDesign, IllustrationSize, IllustrationSizeContext, useIllustrationSizeContext };
