// @sap-ui/fx-components/dist/chunk-KQMKCSMK.js
import { valueStateIcons, valueStateMessageColors, valueStateLabelKeys } from './chunk-WWAPKHIM.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { useState, useEffect, useLayoutEffect, useCallback, useRef } from 'react';
import { createPortal } from 'react-dom';
import { useTranslation } from 'react-i18next';

// src/types/combobox.ts
var ComboBoxSize = /* @__PURE__ */ ((ComboBoxSize2) => {
  ComboBoxSize2["Large"] = "Large";
  ComboBoxSize2["Medium"] = "Medium";
  return ComboBoxSize2;
})(ComboBoxSize || {});
var ComboBoxFilter = /* @__PURE__ */ ((ComboBoxFilter2) => {
  ComboBoxFilter2["StartsWithPerTerm"] = "StartsWithPerTerm";
  ComboBoxFilter2["StartsWith"] = "StartsWith";
  ComboBoxFilter2["Contains"] = "Contains";
  ComboBoxFilter2["None"] = "None";
  return ComboBoxFilter2;
})(ComboBoxFilter || {});
var valueStateVariants = cva(
  "flex items-center gap-2 px-3 py-2 text-sm border-b",
  {
    variants: {
      state: {
        ["None" /* None */]: "hidden",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info border-sapphire-info"
      }
    },
    defaultVariants: {
      state: "None" /* None */
    }
  }
);
function SelectPopover({
  open,
  triggerRef,
  valueState: _valueState,
  valueStateMessage: _valueStateMessage,
  hasValueState = false,
  valueStatePortalRef,
  mobileTitle = "Select option",
  onBeforeClose,
  onClose,
  children,
  className,
  ref
}) {
  return /* @__PURE__ */ jsx(
    ResponsivePopover,
    {
      ref,
      open,
      opener: triggerRef,
      placement: "Bottom",
      horizontalAlign: "Stretch",
      hideArrow: true,
      preventInitialFocus: true,
      preventFocusRestore: true,
      noPadding: true,
      headerText: mobileTitle,
      contentOnlyOnDesktop: true,
      onBeforeClose,
      onClose,
      className,
      children: /* @__PURE__ */ jsxs("div", { className: "flex flex-col", children: [
        hasValueState && /* @__PURE__ */ jsx("div", { ref: valueStatePortalRef }),
        /* @__PURE__ */ jsx(
          List,
          {
            accessibleRole: "ListBox" /* ListBox */,
            separators: "None" /* None */,
            children
          }
        )
      ] })
    }
  );
}
var valueStateInputVariants = cva(
  [
    "flex items-center w-full border bg-sapphire-background-primary",
    "overflow-clip transition-colors relative"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: "rounded-lg",
        ["Medium" /* Medium */]: "rounded"
      },
      valueState: {
        ["None" /* None */]: [
          "border-sapphire-border-active",
          "hover:border-sapphire-border-accent",
          "focus-within:border-sapphire-border-accent"
        ],
        ["Positive" /* Positive */]: [
          "border-sapphire-positive bg-sapphire-positive-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-positive focus-within:bg-sapphire-background-primary"
        ],
        ["Negative" /* Negative */]: [
          "border-sapphire-negative bg-sapphire-negative-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-negative focus-within:bg-sapphire-background-primary"
        ],
        ["Critical" /* Critical */]: [
          "border-sapphire-warning bg-sapphire-warning-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-warning focus-within:bg-sapphire-background-primary"
        ],
        ["Information" /* Information */]: [
          "border-sapphire-info bg-sapphire-info-bg",
          "hover:bg-sapphire-background-primary",
          "focus-within:border-sapphire-info focus-within:bg-sapphire-background-primary"
        ]
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      readonly: {
        true: "bg-sapphire-background-secondary border-sapphire-border-primary hover:border-sapphire-border-primary focus-within:!border-sapphire-border-accent",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
function filterItems(text, filterValue, filterMode) {
  if (!filterValue || filterMode === "None" /* None */) {
    return true;
  }
  const normalizedText = text.toLowerCase();
  const normalizedFilter = filterValue.toLowerCase();
  switch (filterMode) {
    case "StartsWith" /* StartsWith */:
      return normalizedText.startsWith(normalizedFilter);
    case "Contains" /* Contains */:
      return normalizedText.includes(normalizedFilter);
    case "StartsWithPerTerm" /* StartsWithPerTerm */:
    default:
      if (normalizedText.startsWith(normalizedFilter)) return true;
      const wordBoundaries = [...normalizedText.matchAll(/\s+/g)];
      return wordBoundaries.some(
        (m) => normalizedText.substring(m.index + m[0].length).startsWith(normalizedFilter)
      );
  }
}
function ValueStateMessage({
  isOpen,
  hasValueState,
  portalTarget,
  messageHeight,
  messageRef,
  messageId,
  normalizedValueState,
  valueStateMessage
}) {
  const { t } = useTranslation("fx");
  const isNegativeState = normalizedValueState === "Negative" /* Negative */;
  const ValueStateIcon = valueStateIcons[normalizedValueState];
  if (isOpen && hasValueState && portalTarget) {
    return /* @__PURE__ */ jsxs(Fragment, { children: [
      /* @__PURE__ */ jsx("div", { style: { height: messageHeight }, className: "mt-2" }),
      createPortal(
        /* @__PURE__ */ jsxs(
          "div",
          {
            id: messageId,
            className: cn(
              valueStateVariants({ state: normalizedValueState }),
              "border-b-0 items-start gap-1 px-4 py-1.5 shrink-0 rounded-t-lg"
            ),
            children: [
              ValueStateIcon && /* @__PURE__ */ jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
              /* @__PURE__ */ jsx("span", { className: cn("text-sm", valueStateMessageColors[normalizedValueState]), children: valueStateMessage })
            ]
          }
        ),
        portalTarget
      )
    ] });
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: messageRef,
      id: messageId,
      role: isNegativeState ? "alert" : void 0,
      "aria-live": isNegativeState ? "assertive" : "polite",
      className: cn(
        "mt-2 text-sm flex items-start gap-1",
        valueStateMessageColors[normalizedValueState]
      ),
      children: [
        normalizedValueState !== "None" /* None */ && valueStateLabelKeys[normalizedValueState] && /* @__PURE__ */ jsxs("span", { className: "sr-only", children: [
          t(valueStateLabelKeys[normalizedValueState]),
          ":",
          " "
        ] }),
        ValueStateIcon && /* @__PURE__ */ jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
        /* @__PURE__ */ jsx("span", { children: valueStateMessage })
      ]
    }
  );
}
function ClearButton({ onClick, accessibleName = "Clear value", comboBoxSize = "Large" /* Large */ }) {
  const isMedium = comboBoxSize === "Medium" /* Medium */;
  return /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
    Button,
    {
      design: "SecondaryNeutral" /* SecondaryNeutral */,
      size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
      iconOnly: true,
      icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-full w-full" }) }),
      tabIndex: -1,
      onClick,
      accessibleName,
      className: "shrink-0"
    }
  ) });
}
function DropdownArrowButton({ onClick, disabled = false, comboBoxSize = "Large" /* Large */ }) {
  const isMedium = comboBoxSize === "Medium" /* Medium */;
  return /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
    Button,
    {
      design: "SecondaryNeutral" /* SecondaryNeutral */,
      size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
      iconOnly: true,
      icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-full w-full" }) }),
      tabIndex: -1,
      onClick,
      accessibleName: "Open suggestions",
      className: "shrink-0",
      disabled
    }
  ) });
}
function computeVisibleItems(flattenedItems, filterValue, filter, getDisplayText) {
  if (!filterValue || filter === "None" /* None */) {
    return flattenedItems.map((item, index) => ({ ...item, originalIndex: index, isVisible: true }));
  }
  return flattenedItems.map((item, index) => {
    const displayText = getDisplayText(item.element);
    const isVisible = filterItems(displayText, filterValue, filter);
    return { ...item, originalIndex: index, isVisible };
  });
}
function computeNavigableItems(visibleItems) {
  return visibleItems.filter((item) => item.isVisible && !item.element.props.disabled);
}
function useResultAnnouncement(filterValue, filteredCount) {
  const { t } = useTranslation("fx");
  const [resultAnnouncement, setResultAnnouncement] = useState("");
  useEffect(() => {
    if (!filterValue) {
      setResultAnnouncement("");
      return;
    }
    const timer = setTimeout(() => {
      if (filteredCount === 0) {
        setResultAnnouncement(t("COMBOBOX_NO_RESULTS"));
      } else if (filteredCount === 1) {
        setResultAnnouncement(t("COMBOBOX_ONE_RESULT"));
      } else {
        setResultAnnouncement(
          t("COMBOBOX_RESULTS_AVAILABLE", { count: filteredCount })
        );
      }
    }, 300);
    return () => clearTimeout(timer);
  }, [filterValue, filteredCount, t]);
  return { resultAnnouncement, t };
}
function useMessageHeight(messageRef, valueStateMessage) {
  const [messageHeight, setMessageHeight] = useState(0);
  useLayoutEffect(() => {
    if (messageRef.current) {
      setMessageHeight(messageRef.current.offsetHeight);
    }
  }, [valueStateMessage]);
  return messageHeight;
}
function usePopoverControls({
  disabled,
  readonly,
  controlledOpen,
  isOpen,
  setInternalOpen,
  setFocusedIndex,
  onOpen,
  onClose
}) {
  const openPopover = useCallback(() => {
    if (disabled || readonly) return;
    if (controlledOpen === void 0) {
      setInternalOpen(true);
    }
    onOpen?.();
  }, [disabled, readonly, controlledOpen, onOpen, setInternalOpen]);
  const closePopover = useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    setFocusedIndex(-1);
  }, [controlledOpen, setInternalOpen, setFocusedIndex]);
  const handlePopoverClose = useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    setFocusedIndex(-1);
    onClose?.();
  }, [controlledOpen, onClose, setInternalOpen, setFocusedIndex]);
  const togglePopover = useCallback(() => {
    if (isOpen) {
      closePopover();
    } else {
      openPopover();
    }
  }, [isOpen, openPopover, closePopover]);
  return { openPopover, closePopover, handlePopoverClose, togglePopover };
}
function useAnnounce() {
  const [announcement, setAnnouncement] = useState({ message: "", mode: "polite" });
  const timeoutRef = useRef(void 0);
  const announce = useCallback((text, mode = "polite") => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current);
    }
    setAnnouncement({ message: text, mode });
    timeoutRef.current = setTimeout(() => {
      setAnnouncement((prev) => ({ ...prev, message: "" }));
    }, 3e3);
  }, []);
  useEffect(() => {
    return () => {
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
    };
  }, []);
  const liveRegionProps = {
    "aria-live": announcement.mode,
    "aria-atomic": true,
    className: "sr-only",
    children: announcement.message
  };
  return { announce, message: announcement.message, liveRegionProps };
}
function LiveRegion({ message, mode = "polite" }) {
  return /* @__PURE__ */ jsx(
    "div",
    {
      "aria-live": mode,
      "aria-atomic": true,
      className: "sr-only",
      children: message
    }
  );
}

export { ClearButton, ComboBoxFilter, ComboBoxSize, DropdownArrowButton, LiveRegion, SelectPopover, ValueStateMessage, computeNavigableItems, computeVisibleItems, useAnnounce, useMessageHeight, usePopoverControls, useResultAnnouncement, valueStateInputVariants };
