// @sap-ui/fx-components/dist/chunk-YSPWI6FA.js
// src/lib/popupRegistry.ts
var openedPopups = [];
function attachListeners() {
  document.addEventListener("keydown", handleKeydown);
  document.addEventListener("mousedown", handleClick, true);
}
function detachListeners() {
  document.removeEventListener("keydown", handleKeydown);
  document.removeEventListener("mousedown", handleClick, true);
}
function handleKeydown(e) {
  if (e.key !== "Escape" || openedPopups.length === 0) return;
  e.preventDefault();
  e.stopImmediatePropagation();
  const topmost = openedPopups[openedPopups.length - 1];
  topmost.close(true);
}
function handleClick(e) {
  if (openedPopups.length === 0) return;
  const target = e.target;
  for (let i = openedPopups.length - 1; i >= 0; i--) {
    const entry = openedPopups[i];
    if (entry.isModal || entry.type === "dialog") return;
    if (entry.getOpenerEl) {
      const openerEl = entry.getOpenerEl();
      if (openerEl?.contains(target)) return;
    }
    if (entry.element.contains(target)) return;
    if (entry.isInsidePopup?.(target)) return;
    entry.close();
  }
}
function addOpenedPopup(entry) {
  openedPopups.push(entry);
  if (openedPopups.length === 1) {
    attachListeners();
  }
}
function removeOpenedPopup(element) {
  const idx = openedPopups.findIndex((p) => p.element === element);
  if (idx !== -1) openedPopups.splice(idx, 1);
  if (openedPopups.length === 0) {
    detachListeners();
  }
}

// src/lib/Device.ts
var isSSR = typeof document === "undefined";
function getUserAgent() {
  return isSSR ? "" : navigator.userAgent;
}
function hasTouch() {
  if (isSSR) return false;
  return "ontouchstart" in window || navigator.maxTouchPoints > 0;
}
function isIPad() {
  if (isSSR) return false;
  const ua = getUserAgent();
  return /ipad/i.test(ua) || /Macintosh/i.test(ua) && "ontouchend" in document;
}
function isWindowsPlatform() {
  if (isSSR) return false;
  return navigator.platform.indexOf("Win") !== -1;
}
function isWindows8OrAbove() {
  if (!isWindowsPlatform()) return false;
  const matches = getUserAgent().match(/Windows NT (\d+)/);
  const winVer = matches ? parseFloat(matches[1]) : 0;
  return winVer >= 8;
}
function isAndroidUA() {
  if (isSSR) return false;
  return !isWindowsPlatform() && /Android/.test(getUserAgent());
}
function isChromeUA() {
  if (isSSR) return false;
  return /(Chrome|CriOS)/.test(getUserAgent());
}
function isWebkitUA() {
  if (isSSR) return false;
  return /webkit/.test(getUserAgent());
}
function isTablet() {
  if (isSSR) return false;
  const touch = hasTouch();
  const ua = getUserAgent();
  if (isIPad()) return true;
  if (touch) {
    if (isWindows8OrAbove()) return true;
    if (isChromeUA() && isAndroidUA()) {
      return !/Mobile Safari\/[.0-9]+/.test(ua);
    }
    let densityFactor = window.devicePixelRatio || 1;
    if (isAndroidUA() && isWebkitUA()) {
      const wkMatches = ua.match(/(webkit)[ /]([\w.]+)/);
      const wkVer = wkMatches ? parseFloat(wkMatches[2]) : 0;
      if (wkVer >= 537.1) densityFactor = 1;
    }
    return Math.min(
      window.screen.width / densityFactor,
      window.screen.height / densityFactor
    ) >= 600;
  }
  return ua.indexOf("Touch") !== -1 || isAndroidUA() && !/(?=android)(?=.*mobile)/i.test(ua);
}
function isPhone() {
  if (isSSR) return false;
  return hasTouch() && !isTablet();
}
function isDesktop() {
  if (isSSR) return false;
  return !isTablet() && !isPhone() || isWindows8OrAbove();
}

export { addOpenedPopup, isDesktop, isPhone, removeOpenedPopup };
