// @sap-ui/fx-components/dist/chunk-FSSKJCND.js
import { ListItem } from './chunk-E4NK6IRO.js';
import { NavigationDownArrowIcon } from './chunk-OGRUTROU.js';
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { Button } from './chunk-T7M6DGUN.js';
import React, { useState, useRef, useMemo, useCallback, useEffect } from 'react';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';

function useTabbarOverflow({
  enabled = false,
  tabs,
  selectedTabId,
  tabListRef
}) {
  const [visibleTabIds, setVisibleTabIds] = useState(tabs.map((t) => t.id));
  const [overflowTabIds, setOverflowTabIds] = useState([]);
  const [isReady, setIsReady] = useState(false);
  const tabWidthsRef = useRef(/* @__PURE__ */ new Map());
  const hasMeasuredRef = useRef(false);
  const rafId = useRef(null);
  const retryTimeoutRef = useRef(null);
  const tabIds = useMemo(() => tabs.map((t) => t.id), [tabs]);
  const measureTabs = useCallback(() => {
    if (!tabListRef.current) return false;
    const container = tabListRef.current;
    const tabElements = container.querySelectorAll("[data-tab-id]");
    let measuredCount = 0;
    tabElements.forEach((element) => {
      const tabId = element.getAttribute("data-tab-id");
      if (tabId && tabIds.includes(tabId)) {
        const htmlElement = element;
        const rect = htmlElement.getBoundingClientRect();
        if (rect.width > 0) {
          const style = window.getComputedStyle(htmlElement);
          const width = rect.width + parseFloat(style.marginLeft || "0") + parseFloat(style.marginRight || "0");
          tabWidthsRef.current.set(tabId, width);
          measuredCount++;
        } else if (tabWidthsRef.current.has(tabId)) {
          measuredCount++;
        }
      }
    });
    const allMeasured = measuredCount === tabIds.length;
    if (allMeasured) {
      hasMeasuredRef.current = true;
    }
    return allMeasured;
  }, [tabIds, tabListRef]);
  const calculateOverflow = useCallback(() => {
    if (!enabled || !tabListRef.current || tabIds.length === 0) {
      setVisibleTabIds(tabIds);
      setOverflowTabIds([]);
      setIsReady(true);
      return;
    }
    const container = tabListRef.current;
    const containerWidth = container.clientWidth;
    if (containerWidth === 0) {
      return;
    }
    if (!hasMeasuredRef.current || tabWidthsRef.current.size < tabIds.length) {
      if (!measureTabs()) {
        return;
      }
    }
    const tabWidths = tabWidthsRef.current;
    const gap = parseFloat(getComputedStyle(container).columnGap || "0");
    let totalWidth = 0;
    for (const tabId of tabIds) {
      totalWidth += tabWidths.get(tabId) || 0;
    }
    totalWidth += (tabIds.length - 1) * gap;
    const overflowButtonWidth = 80;
    if (totalWidth <= containerWidth) {
      setVisibleTabIds(tabIds);
      setOverflowTabIds([]);
      setIsReady(true);
      return;
    }
    const availableWidth = containerWidth - overflowButtonWidth - gap;
    const visible = [];
    const overflow = [];
    let usedWidth = 0;
    for (let i = 0; i < tabIds.length; i++) {
      const tabId = tabIds[i];
      const width = tabWidths.get(tabId) || 0;
      const neededGap = visible.length > 0 ? gap : 0;
      if (usedWidth + neededGap + width <= availableWidth) {
        visible.push(tabId);
        usedWidth += neededGap + width;
      } else {
        overflow.push(tabId);
      }
    }
    if (overflow.includes(selectedTabId)) {
      const selectedWidth = tabWidths.get(selectedTabId) || 0;
      while (visible.length > 0) {
        const lastVisible = visible[visible.length - 1];
        if (lastVisible === selectedTabId) break;
        const lastWidth = tabWidths.get(lastVisible) || 0;
        usedWidth -= lastWidth + (visible.length > 1 ? gap : 0);
        visible.pop();
        const overflowIdx = overflow.indexOf(selectedTabId);
        overflow.splice(overflowIdx, 0, lastVisible);
        const neededGap = visible.length > 0 ? gap : 0;
        if (usedWidth + neededGap + selectedWidth <= availableWidth) {
          const selIdx = overflow.indexOf(selectedTabId);
          overflow.splice(selIdx, 1);
          visible.push(selectedTabId);
          usedWidth += neededGap + selectedWidth;
          break;
        }
      }
    }
    setVisibleTabIds(visible);
    setOverflowTabIds(overflow);
    setIsReady(true);
  }, [enabled, tabIds, selectedTabId, tabListRef, measureTabs]);
  const scheduleCalculation = useCallback(() => {
    if (rafId.current) {
      cancelAnimationFrame(rafId.current);
    }
    rafId.current = requestAnimationFrame(() => {
      calculateOverflow();
      rafId.current = null;
    });
  }, [calculateOverflow]);
  useEffect(() => {
    if (!enabled) {
      setVisibleTabIds(tabIds);
      setIsReady(true);
      return;
    }
    if (tabWidthsRef.current.size !== tabIds.length) {
      hasMeasuredRef.current = false;
    }
    let retryCount = 0;
    const maxRetries = 20;
    const tryCalculate = () => {
      if (measureTabs()) {
        calculateOverflow();
      } else if (retryCount < maxRetries) {
        retryCount++;
        retryTimeoutRef.current = setTimeout(tryCalculate, 16);
      }
    };
    tryCalculate();
    return () => {
      if (retryTimeoutRef.current) {
        clearTimeout(retryTimeoutRef.current);
        retryTimeoutRef.current = null;
      }
    };
  }, [enabled, tabIds, measureTabs, calculateOverflow]);
  useEffect(() => {
    if (!enabled || !isReady) return;
    calculateOverflow();
  }, [selectedTabId, enabled, isReady, calculateOverflow]);
  useEffect(() => {
    if (!enabled || !tabListRef.current) return;
    let lastWidth = tabListRef.current.clientWidth;
    const observer = new ResizeObserver((entries) => {
      const newWidth = entries[0]?.contentRect?.width ?? 0;
      if (Math.abs(newWidth - lastWidth) > 1) {
        lastWidth = newWidth;
        scheduleCalculation();
      }
    });
    observer.observe(tabListRef.current);
    return () => {
      observer.disconnect();
      if (rafId.current) {
        cancelAnimationFrame(rafId.current);
      }
    };
  }, [enabled, tabListRef, scheduleCalculation]);
  return {
    visibleTabIds,
    overflowTabIds,
    hasOverflow: overflowTabIds.length > 0,
    isReady
  };
}
function TabbarOverflowButton({ tabs, selectedTabId, onSelect, isFocused, onFocus, overflowId, onKeyDown, className, ref }) {
  const [isOpen, setIsOpen] = useState(false);
  const buttonRef = useRef(null);
  React.useImperativeHandle(ref, () => buttonRef.current, []);
  const hasSelectedInOverflow = tabs.some((tab) => tab.id === selectedTabId);
  const handleClick = useCallback(() => {
    setIsOpen((prev) => !prev);
  }, []);
  const handleKeyDown = useCallback((e) => {
    const isRTL = document.documentElement.dir === "rtl";
    const openKey = isRTL ? "ArrowLeft" : "ArrowRight";
    if (e.key === "ArrowDown" || e.key === openKey) {
      e.preventDefault();
      e.stopPropagation();
      setIsOpen(true);
      return;
    }
    onKeyDown?.(e);
  }, [onKeyDown]);
  if (tabs.length === 0) return null;
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx("div", { className: "fx-tabbar-item", children: /* @__PURE__ */ jsxs(
      "button",
      {
        ref: buttonRef,
        type: "button",
        tabIndex: isFocused ? 0 : -1,
        onClick: handleClick,
        onFocus,
        onKeyDown: handleKeyDown,
        "data-tab-id": overflowId,
        "aria-expanded": isOpen,
        "aria-haspopup": "menu",
        "aria-label": "More tabs",
        className: `flex h-full items-center p-2 cursor-pointer bg-transparent border-0 border-b-2 transition-colors gap-1 focus:outline-none border-transparent ${hasSelectedInOverflow || false ? "font-semibold" : "font-normal hover:font-semibold"} ${className || ""}`,
        children: [
          /* @__PURE__ */ jsx(
            "span",
            {
              className: "text-sm text-center whitespace-nowrap text-foreground after:content-[attr(data-text)] after:font-semibold after:block after:h-0 after:overflow-hidden after:invisible",
              "data-text": "More",
              children: "More"
            }
          ),
          /* @__PURE__ */ jsx(NavigationDownArrowIcon, { className: "h-5 w-5 text-foreground" })
        ]
      }
    ) }),
    /* @__PURE__ */ jsx(
      ResponsivePopover,
      {
        open: isOpen,
        opener: buttonRef,
        placement: "Bottom" /* Bottom */,
        horizontalAlign: "End" /* End */,
        hideArrow: true,
        noPadding: true,
        showCloseButton: false,
        contentOnlyOnDesktop: true,
        footer: /* @__PURE__ */ jsx("div", { className: "flex justify-end px-4 py-2", children: /* @__PURE__ */ jsx(Button, { design: "Tertiary", onClick: () => setIsOpen(false), children: "Cancel" }) }),
        onClose: () => setIsOpen(false),
        accessibleRole: "None",
        children: /* @__PURE__ */ jsx("div", { className: "min-w-[200px] max-h-[300px] overflow-auto", children: /* @__PURE__ */ jsx(List, { selectionMode: "None", separators: "None", children: tabs.map((tab) => {
          const isSelected = selectedTabId === tab.id;
          return /* @__PURE__ */ jsx(
            ListItem,
            {
              itemKey: tab.id,
              text: tab.label,
              selected: isSelected,
              onClick: () => {
                onSelect(tab.id);
                setIsOpen(false);
              }
            },
            tab.id
          );
        }) }) })
      }
    )
  ] });
}
var OVERFLOW_BUTTON_ID = "__overflow__";
var Tabbar = ({
  className,
  items,
  selectedTab: controlledSelectedTab,
  defaultSelectedTab,
  onTabSelect,
  enableOverflow = false,
  noUnderline = false,
  accessibleName
}) => {
  const tabItems = items;
  const resolvedDefault = defaultSelectedTab ?? tabItems[0]?.id ?? "";
  const [internalSelectedTab, setInternalSelectedTab] = useState(resolvedDefault);
  const [focusedId, setFocusedId] = useState(tabItems[0]?.id ?? "");
  const tabListRef = useRef(null);
  const tabRefs = useRef(/* @__PURE__ */ new Map());
  const overflowButtonRef = useRef(null);
  const selectedTab = controlledSelectedTab ?? internalSelectedTab;
  const { visibleTabIds, overflowTabIds, hasOverflow, isReady } = useTabbarOverflow({
    enabled: enableOverflow,
    tabs: tabItems.map((t) => ({ id: t.id, label: t.label })),
    selectedTabId: selectedTab,
    tabListRef
  });
  const visibleTabs = enableOverflow && isReady ? tabItems.filter((t) => visibleTabIds.includes(t.id)) : [...tabItems];
  useEffect(() => {
    if (focusedId === OVERFLOW_BUTTON_ID) return;
    const stillVisible = visibleTabs.some((t) => t.id === focusedId);
    if (!stillVisible) {
      setFocusedId(visibleTabs[0]?.id ?? tabItems[0]?.id ?? "");
    }
  }, [visibleTabs.map((t) => t.id).join(",")]);
  const handleTabSelect = useCallback(
    (tab) => {
      const previousTab = selectedTab;
      if (controlledSelectedTab === void 0) {
        setInternalSelectedTab(tab);
      }
      if (onTabSelect) {
        onTabSelect({ selectedTab: tab, previousTab });
      }
    },
    [selectedTab, controlledSelectedTab, onTabSelect]
  );
  const focusById = useCallback((id) => {
    setFocusedId(id);
    if (id === OVERFLOW_BUTTON_ID) {
      overflowButtonRef.current?.focus();
    } else {
      tabRefs.current.get(id)?.focus();
    }
  }, []);
  const handleKeyDown = useCallback((e) => {
    const items2 = [
      ...visibleTabs.map((t) => t.id),
      ...hasOverflow ? [OVERFLOW_BUTTON_ID] : []
    ];
    const activeId = (() => {
      const active = document.activeElement;
      if (!active) return focusedId;
      const tabId = active.dataset.tabId;
      if (tabId && items2.includes(tabId)) return tabId;
      if (overflowButtonRef.current === active) return OVERFLOW_BUTTON_ID;
      return focusedId;
    })();
    const currentIndex = items2.indexOf(activeId);
    if (currentIndex === -1) return;
    const isRTL = document.documentElement.dir === "rtl";
    switch (e.key) {
      case "ArrowRight": {
        e.preventDefault();
        e.stopPropagation();
        const nextIndex = currentIndex + (isRTL ? -1 : 1);
        if (nextIndex >= 0 && nextIndex < items2.length) focusById(items2[nextIndex]);
        break;
      }
      case "ArrowLeft": {
        e.preventDefault();
        e.stopPropagation();
        const nextIndex = currentIndex + (isRTL ? 1 : -1);
        if (nextIndex >= 0 && nextIndex < items2.length) focusById(items2[nextIndex]);
        break;
      }
      case "Home": {
        e.preventDefault();
        e.stopPropagation();
        focusById(items2[0]);
        break;
      }
      case "End": {
        e.preventDefault();
        e.stopPropagation();
        focusById(items2[items2.length - 1]);
        break;
      }
      case "Enter": {
        if (activeId === OVERFLOW_BUTTON_ID) break;
        e.preventDefault();
        handleTabSelect(activeId);
        break;
      }
      case " ": {
        if (activeId === OVERFLOW_BUTTON_ID) break;
        e.preventDefault();
        break;
      }
    }
  }, [focusedId, visibleTabs, hasOverflow, focusById, handleTabSelect]);
  const handleKeyUp = useCallback((e) => {
    if (e.key === " ") {
      const items2 = [
        ...visibleTabs.map((t) => t.id),
        ...hasOverflow ? [OVERFLOW_BUTTON_ID] : []
      ];
      const active = document.activeElement;
      const activeId = (() => {
        if (!active) return focusedId;
        const tabId = active.dataset.tabId;
        if (tabId && items2.includes(tabId)) return tabId;
        if (overflowButtonRef.current === active) return OVERFLOW_BUTTON_ID;
        return focusedId;
      })();
      if (activeId !== OVERFLOW_BUTTON_ID) {
        e.preventDefault();
        handleTabSelect(activeId);
      }
    }
  }, [focusedId, visibleTabs, hasOverflow, handleTabSelect]);
  const renderTab = (tabId, shouldHide, posinset, setsize) => {
    const isSelected = selectedTab === tabId;
    const isFocused = !shouldHide && focusedId === tabId;
    const tabItem = tabItems.find((t) => t.id === tabId);
    return /* @__PURE__ */ jsx("div", { className: "fx-tabbar-item", style: shouldHide ? { display: "none" } : void 0, children: /* @__PURE__ */ jsx(
      "button",
      {
        ref: (el) => {
          if (el) {
            tabRefs.current.set(tabId, el);
          } else {
            tabRefs.current.delete(tabId);
          }
        },
        role: "tab",
        "aria-selected": isSelected,
        "aria-posinset": posinset,
        "aria-setsize": setsize,
        tabIndex: isFocused ? 0 : -1,
        "data-tab-id": tabId,
        onClick: () => {
          setFocusedId(tabId);
          handleTabSelect(tabId);
        },
        onFocus: () => setFocusedId(tabId),
        "aria-hidden": shouldHide ? true : void 0,
        className: `relative z-10 flex h-full items-center p-2 cursor-pointer bg-transparent border-0 border-b-2 transition-colors ${isSelected ? "border-sapphire-text-accent border-solid" : "border-transparent hover:border-sapphire-text-accent/30"}`,
        children: /* @__PURE__ */ jsx(
          "span",
          {
            "data-text": tabItem.label,
            className: `text-sm text-center whitespace-nowrap after:content-[attr(data-text)] after:font-semibold after:block after:h-0 after:overflow-hidden after:invisible ${isSelected ? "font-semibold text-sapphire-text-accent" : "font-normal text-foreground"}`,
            children: tabItem.label
          }
        )
      }
    ) }, tabId);
  };
  return /* @__PURE__ */ jsxs("div", { className: className || "relative flex flex-col h-12", children: [
    !noUnderline && /* @__PURE__ */ jsx("div", { className: "absolute bottom-0 left-0 right-0 h-px bg-border" }),
    /* @__PURE__ */ jsxs("div", { className: "flex flex-1 min-h-0 min-w-0 w-full items-stretch", children: [
      /* @__PURE__ */ jsx(
        "div",
        {
          ref: tabListRef,
          role: "tablist",
          "aria-label": accessibleName,
          onKeyDown: handleKeyDown,
          onKeyUp: handleKeyUp,
          className: "flex gap-4 h-full items-stretch flex-1 min-w-0 overflow-hidden",
          children: tabItems.map((tab, index) => {
            const shouldHide = enableOverflow && isReady && !visibleTabIds.includes(tab.id);
            const visibleIndex = visibleTabs.findIndex((t) => t.id === tab.id);
            const posinset = shouldHide ? index + 1 : visibleIndex + 1;
            const setsize = enableOverflow && isReady ? visibleTabs.length : tabItems.length;
            return renderTab(tab.id, shouldHide, posinset, setsize);
          })
        }
      ),
      enableOverflow && isReady && hasOverflow && /* @__PURE__ */ jsx(
        TabbarOverflowButton,
        {
          ref: overflowButtonRef,
          tabs: overflowTabIds.map((id) => {
            const tab = tabItems.find((t) => t.id === id);
            return { id, label: tab.label };
          }),
          selectedTabId: selectedTab,
          onSelect: (tabId) => handleTabSelect(tabId),
          isFocused: focusedId === OVERFLOW_BUTTON_ID,
          onFocus: () => setFocusedId(OVERFLOW_BUTTON_ID),
          overflowId: OVERFLOW_BUTTON_ID,
          onKeyDown: handleKeyDown
        }
      )
    ] })
  ] });
};

export { Tabbar, TabbarOverflowButton, useTabbarOverflow };
