// @sap-ui/fx-components/dist/chunk-E2WLAC4U.js
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useState, useId, useEffect, useImperativeHandle } from 'react';
import { jsx, jsxs } from 'react/jsx-runtime';

// src/types/busy-indicator.ts
var BusyIndicatorSize = /* @__PURE__ */ ((BusyIndicatorSize2) => {
  BusyIndicatorSize2["S"] = "S";
  BusyIndicatorSize2["M"] = "M";
  BusyIndicatorSize2["L"] = "L";
  return BusyIndicatorSize2;
})(BusyIndicatorSize || {});
var BusyIndicatorTextPlacement = /* @__PURE__ */ ((BusyIndicatorTextPlacement2) => {
  BusyIndicatorTextPlacement2["Top"] = "Top";
  BusyIndicatorTextPlacement2["Bottom"] = "Bottom";
  return BusyIndicatorTextPlacement2;
})(BusyIndicatorTextPlacement || {});
var dotVariants = cva("rounded-full bg-current inline-block", {
  variants: {
    size: {
      ["S" /* S */]: "w-2 h-2",
      ["M" /* M */]: "w-4 h-4",
      ["L" /* L */]: "w-8 h-8"
    }
  },
  defaultVariants: { size: "M" /* M */ }
});
var dotGapClasses = {
  ["S" /* S */]: "gap-px",
  ["M" /* M */]: "gap-[0.1875rem]",
  ["L" /* L */]: "gap-1"
};
function BusyIndicator({
  size = "M" /* M */,
  textPlacement = "Bottom" /* Bottom */,
  active = false,
  delay = 1e3,
  text,
  children,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = useRef(null);
  const overlayRef = useRef(null);
  const [isBusy, setIsBusy] = useState(false);
  const labelId = useId();
  const hasChildren = React.Children.count(children) > 0;
  const normalizedSize = size || "M" /* M */;
  const isTopPlacement = textPlacement === "Top";
  useEffect(() => {
    if (!active) {
      setIsBusy(false);
      return;
    }
    if (delay <= 0) {
      setIsBusy(true);
      return;
    }
    const timer = setTimeout(() => {
      setIsBusy(true);
    }, delay);
    return () => clearTimeout(timer);
  }, [active, delay]);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.focus();
        } else {
          rootRef.current?.focus();
        }
      },
      blur() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.blur();
        } else {
          rootRef.current?.blur();
        }
      },
      isFocused() {
        return document.activeElement === overlayRef.current || document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    [isBusy]
  );
  const textLabel = text ? /* @__PURE__ */ jsx(
    "span",
    {
      id: labelId,
      className: cn(
        "w-full text-center text-sapphire-text-tertiary text-sm font-medium",
        isTopPlacement ? "mb-2" : "mt-2"
      ),
      children: text
    }
  ) : null;
  const dots = /* @__PURE__ */ jsx(
    "div",
    {
      className: cn(
        "inline-flex items-center leading-none",
        dotGapClasses[normalizedSize]
      ),
      children: [0, 1, 2].map((i) => /* @__PURE__ */ jsx(
        "span",
        {
          className: dotVariants({ size: normalizedSize }),
          style: {
            animation: `busy-grow 1.6s cubic-bezier(0.32, 0.06, 0.85, 1.11) ${i * 200}ms infinite`
          }
        },
        i
      ))
    }
  );
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: rootRef,
      id,
      className: cn(
        "inline-block",
        hasChildren && "relative",
        isBusy && "text-ring",
        className
      ),
      style: { ...style, borderRadius: "inherit", height: "inherit" },
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsx("style", { children: `@keyframes busy-grow{0%,50%,100%{transform:scale(.5)}25%{transform:scale(1)}}` }),
        isBusy && /* @__PURE__ */ jsxs(
          "div",
          {
            ref: overlayRef,
            role: "progressbar",
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            "aria-valuetext": "Busy",
            "aria-labelledby": text ? labelId : void 0,
            "aria-label": !text ? accessibleName ?? "Please wait" : void 0,
            title: !text ? accessibleName ?? "Please wait" : void 0,
            tabIndex: 0,
            "data-sap-focus-ref": true,
            className: cn(
              "flex flex-col items-center justify-center",
              "bg-inherit rounded-[inherit]",
              "focus-visible:outline focus-visible:outline-2 focus-visible:outline-ring focus-visible:outline-offset-[-2px]",
              hasChildren && "absolute inset-0 z-[99]"
            ),
            children: [
              isTopPlacement && textLabel,
              dots,
              !isTopPlacement && textLabel
            ]
          }
        ),
        children && /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(isBusy && "opacity-40"),
            ...isBusy ? { inert: true } : {},
            children
          }
        )
      ]
    }
  );
}

export { BusyIndicator, BusyIndicatorSize, BusyIndicatorTextPlacement };
