// @sap-ui/fx-components/dist/chunk-QDXU3YWA.js
import { useAnnounce, useMessageHeight, SelectPopover, ValueStateMessage } from './chunk-KQMKCSMK.js';
import { ListItem } from './chunk-E4NK6IRO.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useMemo, Children, isValidElement, useId, useCallback, cloneElement, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';

// src/types/select.ts
var SelectSize = /* @__PURE__ */ ((SelectSize2) => {
  SelectSize2["Large"] = "Large";
  SelectSize2["Medium"] = "Medium";
  return SelectSize2;
})(SelectSize || {});
var TextSeparator = /* @__PURE__ */ ((TextSeparator2) => {
  TextSeparator2["Dash"] = "Dash";
  TextSeparator2["Bullet"] = "Bullet";
  TextSeparator2["VerticalLine"] = "VerticalLine";
  return TextSeparator2;
})(TextSeparator || {});
var selectContainerVariants = cva(
  [
    "flex items-center border bg-card cursor-pointer",
    "overflow-clip"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: [
          "h-10 rounded-lg min-w-[2.75rem]",
          "pl-3.5 pr-1 py-1 gap-2",
          "focus-within:border-2 focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]"
        ],
        ["Medium" /* Medium */]: [
          "h-8 rounded min-w-[2rem]",
          "pl-3 pr-1 py-1 gap-2",
          "focus-within:border-2 focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]"
        ]
      },
      valueState: {
        ["None" /* None */]: "border-sapphire-border-active hover:border-sapphire-border-accent focus-within:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "border-sapphire-positive hover:bg-sapphire-canvas-primary focus-within:border-sapphire-positive",
        ["Negative" /* Negative */]: "border-sapphire-negative hover:bg-sapphire-canvas-primary focus-within:border-sapphire-negative",
        ["Critical" /* Critical */]: "border-sapphire-warning hover:bg-sapphire-canvas-primary focus-within:border-sapphire-warning",
        ["Information" /* Information */]: "border-sapphire-info hover:bg-sapphire-canvas-primary focus-within:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      readonly: {
        true: "bg-muted cursor-default hover:border-sapphire-border-secondary focus-within:!border-sapphire-border-accent",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
function getSeparator(separator) {
  switch (separator) {
    case "Bullet" /* Bullet */:
      return " \xB7 ";
    case "VerticalLine" /* VerticalLine */:
      return " | ";
    case "Dash" /* Dash */:
    default:
      return " \u2013 ";
  }
}
function getOptionData(element) {
  const props = element.props;
  const isCustom = element.type?.displayName === "OptionCustom";
  if (isCustom) {
    const customProps = props;
    return {
      value: customProps.value,
      text: customProps.displayText ?? "",
      tooltip: customProps.tooltip,
      disabled: customProps.disabled
    };
  }
  const optionProps = props;
  const text = typeof optionProps.children === "string" ? optionProps.children : "";
  return {
    value: optionProps.value,
    text,
    additionalText: optionProps.additionalText,
    icon: optionProps.icon,
    tooltip: optionProps.tooltip,
    disabled: optionProps.disabled
  };
}
function useTypeAhead(options, onSelect) {
  const [, setTypedChars] = useState("");
  const timeoutRef = useRef(void 0);
  const handleKeyPress = useCallback(
    (char) => {
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
      setTypedChars((prev) => {
        const newChars = prev + char.toLowerCase();
        let searchStr = newChars;
        if (newChars.split("").every((c) => c === newChars[0])) {
          searchStr = newChars[0];
        }
        const matchIndex = options.findIndex(
          (opt) => !opt.disabled && opt.text.toLowerCase().startsWith(searchStr)
        );
        if (matchIndex >= 0) {
          onSelect(matchIndex);
        }
        return newChars;
      });
      timeoutRef.current = setTimeout(() => {
        setTypedChars("");
      }, 1e3);
    },
    [options, onSelect]
  );
  useEffect(() => {
    return () => {
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
    };
  }, []);
  return { handleKeyPress, reset: () => setTypedChars("") };
}
function Select({
  value: controlledValue,
  defaultValue,
  size = "Large" /* Large */,
  disabled = false,
  readonly = false,
  required = false,
  name,
  valueState = "None" /* None */,
  valueStateMessage,
  icon,
  tooltip,
  textSeparator = "Dash" /* Dash */,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  open: controlledOpen,
  defaultOpen = false,
  onChange,
  onLiveChange,
  onOpen,
  onClose,
  children,
  className,
  label,
  accessibilityAttributes,
  ref
}) {
  const triggerRef = useRef(null);
  const messageRef = useRef(null);
  const [portalTarget, setPortalTarget] = useState(null);
  const { announce, liveRegionProps } = useAnnounce();
  const { t } = useTranslation("fx");
  const options = useMemo(() => {
    const opts = [];
    Children.forEach(children, (child, index) => {
      if (isValidElement(child)) {
        const data = getOptionData(child);
        opts.push({ element: child, data, index });
      }
    });
    return opts;
  }, [children]);
  const initialSelectedIndex = useMemo(() => {
    const valueToFind = controlledValue ?? defaultValue;
    if (valueToFind !== void 0) {
      const byValue = options.findIndex((opt) => opt.data.value === valueToFind);
      if (byValue >= 0) return byValue;
      const byText = options.findIndex((opt) => opt.data.text === valueToFind);
      if (byText >= 0) return byText;
    }
    const bySelectedProp = options.findIndex((opt) => {
      const props = opt.element.props;
      return props.selected === true;
    });
    if (bySelectedProp >= 0) return bySelectedProp;
    const firstEnabled = options.findIndex((opt) => !opt.data.disabled);
    return firstEnabled >= 0 ? firstEnabled : 0;
  }, [options, controlledValue, defaultValue]);
  const [internalSelectedIndex, setInternalSelectedIndex] = useState(initialSelectedIndex);
  const [internalOpen, setInternalOpen] = useState(defaultOpen);
  const [focusedIndex, setFocusedIndex] = useState(-1);
  const [previousSelectedIndex, setPreviousSelectedIndex] = useState(initialSelectedIndex);
  const isOpen = controlledOpen ?? internalOpen;
  const normalizedValueState = valueState || "None" /* None */;
  const hasValueState = normalizedValueState !== "None" /* None */ && !!valueStateMessage;
  const baseId = useId();
  const messageId = `${baseId}-message`;
  const messageHeight = useMessageHeight(messageRef, valueStateMessage);
  const selectedOption = useMemo(() => {
    if (controlledValue !== void 0) {
      const opt = options.find(
        (o) => o.data.value === controlledValue || o.data.text === controlledValue
      );
      return opt?.data ?? null;
    }
    return options[internalSelectedIndex]?.data ?? null;
  }, [options, controlledValue, internalSelectedIndex]);
  const popoverId = `${baseId}-popover`;
  const descriptionId = `${baseId}-description`;
  const { handleKeyPress } = useTypeAhead(
    options.map((o) => o.data),
    (index) => {
      if (!isOpen) {
        setInternalSelectedIndex(index);
        const data = options[index]?.data;
        if (data) {
          onChange?.({ selectedOption: data });
        }
      } else {
        setFocusedIndex(index);
        onLiveChange?.({ selectedOption: options[index]?.data ?? null });
      }
    }
  );
  const openPopover = useCallback(() => {
    if (disabled || readonly) return;
    setPreviousSelectedIndex(internalSelectedIndex);
    setFocusedIndex(internalSelectedIndex);
    if (controlledOpen === void 0) {
      setInternalOpen(true);
    }
    onOpen?.();
  }, [disabled, readonly, internalSelectedIndex, controlledOpen, onOpen]);
  const closePopover = useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    setFocusedIndex(-1);
    onClose?.();
  }, [controlledOpen, onClose]);
  const handleBeforeClose = useCallback(
    (detail) => {
      if (detail.escPressed) {
        setInternalSelectedIndex(previousSelectedIndex);
      }
    },
    [previousSelectedIndex]
  );
  const togglePopover = useCallback(() => {
    if (isOpen) {
      closePopover();
    } else {
      openPopover();
    }
  }, [isOpen, openPopover, closePopover]);
  const selectOption = useCallback(
    (index, closeAfter = true) => {
      if (index < 0 || index >= options.length) return;
      if (options[index].data.disabled) return;
      setInternalSelectedIndex(index);
      const data = options[index].data;
      onChange?.({ selectedOption: data });
      if (closeAfter) {
        closePopover();
        triggerRef.current?.focus();
      }
    },
    [options, onChange, closePopover]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled) return;
      switch (e.key) {
        case "F4":
          e.preventDefault();
          togglePopover();
          break;
        case " ":
        case "Enter":
          e.preventDefault();
          if (isOpen) {
            if (focusedIndex >= 0) {
              selectOption(focusedIndex);
            }
          } else {
            openPopover();
          }
          break;
        case "ArrowDown":
          e.preventDefault();
          if (e.altKey) {
            openPopover();
          } else if (isOpen) {
            setFocusedIndex((prev) => {
              const nextIdx = prev + 1;
              const next = nextIdx >= options.length ? 0 : nextIdx;
              if (options[next]?.data.disabled) {
                const nextNonDisabled = options.findIndex(
                  (o, i) => i > next && !o.data.disabled
                );
                return nextNonDisabled >= 0 ? nextNonDisabled : prev;
              }
              onLiveChange?.({ selectedOption: options[next]?.data ?? null });
              return next;
            });
          } else {
            const currentIdx = options.findIndex((o) => o.data === selectedOption);
            const nextIdx = currentIdx + 1;
            if (nextIdx < options.length && !options[nextIdx].data.disabled) {
              setInternalSelectedIndex(nextIdx);
              onChange?.({ selectedOption: options[nextIdx].data });
              const optionText = options[nextIdx].data.text;
              announce(t("SELECT_OPTION_SELECTED", { text: optionText }));
            }
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          if (e.altKey) {
            closePopover();
          } else if (isOpen) {
            setFocusedIndex((prev) => {
              const nextIdx = prev - 1;
              const next = nextIdx < 0 ? options.length - 1 : nextIdx;
              if (options[next]?.data.disabled) {
                const prevNonDisabled = options.slice(0, next).reverse().findIndex((o) => !o.data.disabled);
                return prevNonDisabled >= 0 ? next - 1 - prevNonDisabled : prev;
              }
              onLiveChange?.({ selectedOption: options[next]?.data ?? null });
              return next;
            });
          } else {
            const currentIdx = options.findIndex((o) => o.data === selectedOption);
            const prevIdx = currentIdx - 1;
            if (prevIdx >= 0 && !options[prevIdx].data.disabled) {
              setInternalSelectedIndex(prevIdx);
              onChange?.({ selectedOption: options[prevIdx].data });
              const optionText = options[prevIdx].data.text;
              announce(t("SELECT_OPTION_SELECTED", { text: optionText }));
            }
          }
          break;
        case "Home":
          e.preventDefault();
          if (isOpen) {
            const firstNonDisabled = options.findIndex((o) => !o.data.disabled);
            if (firstNonDisabled >= 0) {
              setFocusedIndex(firstNonDisabled);
              onLiveChange?.({ selectedOption: options[firstNonDisabled]?.data ?? null });
            }
          }
          break;
        case "End":
          e.preventDefault();
          if (isOpen) {
            const lastNonDisabled = options.length - 1 - [...options].reverse().findIndex((o) => !o.data.disabled);
            if (lastNonDisabled >= 0 && lastNonDisabled < options.length) {
              setFocusedIndex(lastNonDisabled);
              onLiveChange?.({ selectedOption: options[lastNonDisabled]?.data ?? null });
            }
          }
          break;
        case "Escape":
          e.preventDefault();
          if (isOpen) {
            closePopover();
          }
          break;
        case "Tab":
          if (isOpen) {
            selectOption(focusedIndex);
            closePopover();
          }
          break;
      }
      if (e.key.length === 1 && !e.ctrlKey && !e.altKey && !e.metaKey) {
        handleKeyPress(e.key);
      }
    },
    [
      disabled,
      isOpen,
      focusedIndex,
      options,
      selectedOption,
      handleKeyPress,
      togglePopover,
      openPopover,
      closePopover,
      selectOption,
      onChange,
      onLiveChange,
      announce
    ]
  );
  const handleClick = useCallback(() => {
    if (!disabled && !readonly) {
      togglePopover();
    }
  }, [disabled, readonly, togglePopover]);
  const renderDisplayText = () => {
    if (label) {
      return label;
    }
    if (!selectedOption) {
      return /* @__PURE__ */ jsx("span", { className: "text-sapphire-text-tertiary", children: "Select..." });
    }
    if (readonly && selectedOption.additionalText) {
      return /* @__PURE__ */ jsxs("span", { children: [
        selectedOption.text,
        getSeparator(textSeparator),
        selectedOption.additionalText
      ] });
    }
    return /* @__PURE__ */ jsx("span", { children: selectedOption.text });
  };
  const renderChildren = useCallback(() => {
    return Children.map(children, (child, index) => {
      if (!isValidElement(child)) return child;
      const data = getOptionData(child);
      const isSelected = data.value != null && data.value === selectedOption?.value || data.text === selectedOption?.text;
      const isFocused = index === focusedIndex;
      return cloneElement(child, {
        selected: isSelected,
        focused: isFocused,
        onClick: () => selectOption(index)
      });
    });
  }, [children, selectedOption, focusedIndex, selectOption]);
  const isIconOnly = !!icon && !label;
  return /* @__PURE__ */ jsxs("div", { ref, className: cn("relative w-full", isIconOnly && "w-auto", className), children: [
    name && /* @__PURE__ */ jsx(
      "input",
      {
        type: "hidden",
        name,
        value: selectedOption?.value ?? selectedOption?.text ?? ""
      }
    ),
    /* @__PURE__ */ jsx(
      "div",
      {
        ref: triggerRef,
        role: "combobox",
        "aria-expanded": isOpen,
        "aria-haspopup": "listbox",
        "aria-roledescription": t("SELECT_ROLE_DESCRIPTION"),
        "aria-controls": [isOpen ? popoverId : null, accessibilityAttributes?.controls].filter(Boolean).join(" ") || void 0,
        "aria-activedescendant": focusedIndex >= 0 && options[focusedIndex] ? options[focusedIndex].data.value ?? options[focusedIndex].data.text : void 0,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-describedby": [
          valueStateMessage ? messageId : null,
          valueState !== "None" /* None */ ? descriptionId : null,
          accessibleDescriptionRef
        ].filter(Boolean).join(" ") || void 0,
        "aria-required": required,
        "aria-disabled": disabled,
        "aria-readonly": readonly,
        "aria-invalid": valueState === "Negative" /* Negative */,
        tabIndex: disabled ? -1 : 0,
        title: tooltip,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        className: cn(
          selectContainerVariants({
            size,
            valueState,
            disabled,
            readonly
          }),
          isIconOnly && "p-2"
        ),
        children: isIconOnly ? /* @__PURE__ */ jsx("span", { className: "flex items-center justify-center text-sapphire-text-tertiary", children: icon }) : /* @__PURE__ */ jsxs(Fragment, { children: [
          /* @__PURE__ */ jsx("div", { className: "flex-1 py-2 text-sm truncate min-w-0", children: renderDisplayText() }),
          !readonly && /* @__PURE__ */ jsx("span", { className: cn(
            "flex items-center justify-center shrink-0 text-sapphire-neutral-foreground-black",
            size === "Medium" /* Medium */ ? "h-6 w-6" : "h-8 w-8"
          ), children: /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: size === "Medium" /* Medium */ ? "h-3 w-3" : "h-4 w-4" }) })
        ] })
      }
    ),
    valueState !== "None" /* None */ && valueStateMessage && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage }),
    accessibleDescription && /* @__PURE__ */ jsx("span", { id: `${baseId}-desc-text`, className: "sr-only", children: accessibleDescription }),
    /* @__PURE__ */ jsx(
      SelectPopover,
      {
        open: isOpen,
        triggerRef,
        valueState,
        valueStateMessage,
        hasValueState,
        valueStatePortalRef: setPortalTarget,
        onBeforeClose: handleBeforeClose,
        onClose: closePopover,
        children: renderChildren()
      }
    ),
    valueStateMessage && /* @__PURE__ */ jsx(
      ValueStateMessage,
      {
        isOpen,
        hasValueState,
        portalTarget,
        messageHeight,
        messageRef,
        messageId,
        normalizedValueState,
        valueStateMessage
      }
    ),
    /* @__PURE__ */ jsx("div", { ...liveRegionProps })
  ] });
}
function Option({
  value,
  icon,
  additionalText,
  tooltip,
  selected = false,
  disabled = false,
  focused = false,
  onClick,
  children,
  className
}) {
  const displayText = typeof children === "string" ? children : "";
  return /* @__PURE__ */ jsx(
    ListItem,
    {
      accessibleRole: "Option" /* Option */,
      text: displayText,
      icon: icon ? /* @__PURE__ */ jsx("span", { className: "text-current", children: icon }) : void 0,
      additionalText,
      tooltip,
      selected,
      disabled,
      focused,
      onClick: onClick ? () => onClick() : void 0,
      itemKey: value ?? displayText,
      className: cn(
        "!min-h-8 !h-8 [&>span]:!px-4 [&>span]:!py-0 [&_span]:!font-normal",
        // Override the default ring focus style with an inset ::after pseudo-element
        // so the focus ring is not clipped by the popover's rounded corners
        "!ring-0",
        focused && "after:absolute after:inset-0.5 after:border-2 after:border-ring after:rounded-sm after:pointer-events-none",
        selected && "!bg-sapphire-brand-selected-background",
        className
      ),
      children: typeof children !== "string" ? children : void 0
    }
  );
}
Option.displayName = "Option";

export { Option, Select, SelectSize, TextSeparator };
