// @sap-ui/fx-components/dist/chunk-LM6CSHOP.js
import { ComboBoxPopover } from './chunk-EEGGRFXR.js';
import { computeVisibleItems, computeNavigableItems, useResultAnnouncement, useMessageHeight, usePopoverControls, ClearButton, DropdownArrowButton, ValueStateMessage, LiveRegion, valueStateInputVariants } from './chunk-KQMKCSMK.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useCallback, useMemo, Children, isValidElement, useEffect, cloneElement } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

var inputContainerVariants = (opts) => {
  const isLarge = opts.size === "Large" /* Large */;
  return cn(
    valueStateInputVariants(opts),
    isLarge ? "h-10 pl-3.5 pr-1 py-1 gap-1" : "h-8 pl-3 pr-1 py-1 gap-1",
    isLarge ? "focus-within:border-2 focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]" : "focus-within:border-2 focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]",
    opts.readonly && (isLarge ? "py-2.5" : "py-1.5")
  );
};
function getItemData(element) {
  const { text, value, children, additionalText, icon, disabled } = element.props;
  const displayText = text ?? (typeof children === "string" ? children : "");
  const itemValue = value ?? displayText;
  return { displayText, itemValue, additionalText, icon, disabled };
}
function buildItemData(element) {
  const { displayText, itemValue, additionalText, icon, disabled } = getItemData(element);
  return { text: displayText, value: itemValue, additionalText, icon, disabled };
}
function ComboBox({
  value: controlledValue,
  defaultValue = "",
  selectedValue: controlledSelectedValue,
  defaultSelectedValue,
  size = "Large" /* Large */,
  placeholder = "",
  disabled = false,
  readonly = false,
  required = false,
  name,
  filter = "StartsWithPerTerm" /* StartsWithPerTerm */,
  noTypeahead = false,
  valueState = "None" /* None */,
  valueStateMessage,
  showClearIcon = false,
  loading = false,
  open: controlledOpen,
  defaultOpen = false,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onInput,
  onChange,
  onSelectionChange,
  onOpen,
  onClose,
  children,
  className,
  icon,
  accessibilityAttributes,
  ref
}) {
  const isLargeSize = size === "Large" /* Large */;
  const containerRef = useRef(null);
  const inputRef = useRef(null);
  const messageRef = useRef(null);
  const [portalTarget, setPortalTarget] = useState(null);
  const openedByTypingRef = useRef(false);
  const linkNavigationRef = useRef(false);
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [internalOpen, setInternalOpen] = useState(defaultOpen);
  const [filterValue, setFilterValue] = useState("");
  const [focusedIndex, setFocusedIndex] = useState(-1);
  const [lastCommittedValue, setLastCommittedValue] = useState(defaultValue);
  const [internalSelectedValue, setInternalSelectedValue] = useState(defaultSelectedValue);
  const [, setSelectedItem] = useState(null);
  const value = controlledValue ?? internalValue;
  const isOpen = controlledOpen ?? internalOpen;
  const selectedValue = controlledSelectedValue ?? internalSelectedValue;
  const updateSelectedValue = useCallback((val) => {
    if (controlledSelectedValue === void 0) {
      setInternalSelectedValue(val);
    }
  }, [controlledSelectedValue]);
  const prevIsOpenRef = useRef(isOpen);
  const normalizedValueState = valueState || "None" /* None */;
  const hasValueState = normalizedValueState !== "None" /* None */ && !!valueStateMessage;
  const isNegativeState = normalizedValueState === "Negative" /* Negative */;
  const baseId = useMemo(() => `combobox-${Math.random().toString(36).slice(2, 9)}`, []);
  const popoverId = `${baseId}-popover`;
  const inputId = `${baseId}-input`;
  const messageId = `${baseId}-message`;
  const accessibleDescSpanId = `${baseId}-accessible-desc`;
  const linkShortcutHintId = `${baseId}-link-hint`;
  const flattenedItems = useMemo(() => {
    const items = [];
    Children.forEach(children, (child, groupIdx) => {
      if (!isValidElement(child)) return;
      if (child.type?.displayName === "ComboBoxItemGroup") {
        const groupProps = child.props;
        Children.forEach(groupProps.children, (groupChild) => {
          if (isValidElement(groupChild)) {
            items.push({
              element: groupChild,
              groupIndex: groupIdx
            });
          }
        });
      } else {
        items.push({
          element: child
        });
      }
    });
    return items;
  }, [children]);
  const visibleItems = useMemo(
    () => computeVisibleItems(flattenedItems, filterValue, filter, (el) => getItemData(el).displayText),
    [flattenedItems, filterValue, filter]
  );
  const navigableItems = useMemo(() => computeNavigableItems(visibleItems), [visibleItems]);
  const filteredCount = useMemo(
    () => visibleItems.filter((item) => item.isVisible).length,
    [visibleItems]
  );
  const { resultAnnouncement, t } = useResultAnnouncement(filterValue, filteredCount);
  const messageHeight = useMessageHeight(messageRef, valueStateMessage);
  const syncInputToFocusedItem = useCallback(
    (index) => {
      if (index < 0 || index >= navigableItems.length) return;
      const item = navigableItems[index];
      const { displayText } = getItemData(item.element);
      if (controlledValue === void 0) {
        setInternalValue(displayText);
      }
      requestAnimationFrame(() => {
        if (inputRef.current) {
          inputRef.current.setSelectionRange(0, displayText.length);
        }
      });
    },
    [navigableItems, controlledValue]
  );
  useEffect(() => {
    if (!isOpen || !portalTarget) return;
    const handleLinkKeyDown = (e) => {
      if (!linkNavigationRef.current) return;
      const links = Array.from(portalTarget.querySelectorAll("a"));
      if (links.length === 0) return;
      const currentIndex = links.indexOf(e.target);
      if (currentIndex === -1) return;
      const focusInput = () => {
        linkNavigationRef.current = false;
        inputRef.current?.focus();
      };
      switch (e.key) {
        case "Tab":
          e.preventDefault();
          if (e.shiftKey) {
            if (currentIndex <= 0) {
              focusInput();
            } else {
              links[currentIndex - 1].focus();
            }
          } else {
            if (currentIndex >= links.length - 1) {
              focusInput();
            } else {
              links[currentIndex + 1].focus();
            }
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          focusInput();
          break;
        case "ArrowDown":
          e.preventDefault();
          focusInput();
          setFocusedIndex(0);
          syncInputToFocusedItem(0);
          break;
        case "Escape":
          e.preventDefault();
          e.stopPropagation();
          focusInput();
          break;
      }
    };
    portalTarget.addEventListener("keydown", handleLinkKeyDown);
    return () => portalTarget.removeEventListener("keydown", handleLinkKeyDown);
  }, [isOpen, portalTarget, syncInputToFocusedItem]);
  const navigateItemClosed = useCallback(
    (itemElement) => {
      const { displayText, itemValue } = getItemData(itemElement);
      if (controlledValue === void 0) {
        setInternalValue(displayText);
      }
      setFilterValue("");
      updateSelectedValue(itemValue);
      const itemData = buildItemData(itemElement);
      setSelectedItem(itemData);
      onInput?.(displayText);
      onSelectionChange?.({ item: itemData, value: displayText });
    },
    [controlledValue, onInput, onSelectionChange, updateSelectedValue]
  );
  const { openPopover, closePopover, handlePopoverClose, togglePopover } = usePopoverControls({
    disabled,
    readonly,
    controlledOpen,
    isOpen,
    setInternalOpen,
    setFocusedIndex,
    onOpen,
    onClose
  });
  useEffect(() => {
    if (isOpen && filterValue && visibleItems.every((item) => !item.isVisible)) {
      closePopover();
    }
  }, [isOpen, filterValue, visibleItems, closePopover]);
  useEffect(() => {
    const wasOpen = prevIsOpenRef.current;
    prevIsOpenRef.current = isOpen;
    if (!isOpen || wasOpen) return;
    if (navigableItems.length === 0) return;
    let initialIndex = -1;
    if (value) {
      const selectedIdx = navigableItems.findIndex((item) => {
        const { displayText } = getItemData(item.element);
        return displayText === value;
      });
      if (selectedIdx >= 0) {
        initialIndex = selectedIdx;
      }
    }
    setFocusedIndex(initialIndex);
    if (!openedByTypingRef.current && initialIndex >= 0) {
      syncInputToFocusedItem(initialIndex);
    }
    openedByTypingRef.current = false;
  }, [isOpen]);
  const selectItem = useCallback(
    (itemElement) => {
      const { displayText, itemValue } = getItemData(itemElement);
      if (controlledValue === void 0) {
        setInternalValue(displayText);
      }
      setFilterValue("");
      setLastCommittedValue(displayText);
      updateSelectedValue(itemValue);
      const itemData = buildItemData(itemElement);
      setSelectedItem(itemData);
      onInput?.(displayText);
      onChange?.(displayText);
      onSelectionChange?.({ item: itemData, value: displayText });
      closePopover();
      inputRef.current?.focus();
    },
    [controlledValue, onInput, onChange, onSelectionChange, closePopover, updateSelectedValue]
  );
  const handleInputChange = useCallback(
    (e) => {
      const newValue = e.target.value;
      if (controlledValue === void 0) {
        setInternalValue(newValue);
      }
      setFilterValue(newValue);
      onInput?.(newValue);
      if (!isOpen && newValue) {
        openedByTypingRef.current = true;
        openPopover();
      }
      const nativeInputType = e.nativeEvent.inputType ?? "";
      const isDeletion = nativeInputType.startsWith("delete") || nativeInputType === "historyUndo";
      if (isDeletion) {
        handleDeletionInput(newValue);
        return;
      }
      if (!noTypeahead && newValue) {
        handleTypeahead(newValue);
      }
    },
    [controlledValue, isOpen, noTypeahead, navigableItems, onInput, onSelectionChange, openPopover, updateSelectedValue, selectedValue, internalSelectedValue]
  );
  const handleDeletionInput = useCallback(
    (newValue) => {
      if (selectedValue !== void 0 || internalSelectedValue !== void 0) {
        updateSelectedValue(void 0);
        onSelectionChange?.({ item: null, value: newValue });
      }
      setFocusedIndex(-1);
    },
    [selectedValue, internalSelectedValue, updateSelectedValue, onSelectionChange]
  );
  const handleTypeahead = useCallback(
    (newValue) => {
      const matchingItem = navigableItems.find((item) => {
        const { displayText } = getItemData(item.element);
        return displayText.toLowerCase().startsWith(newValue.toLowerCase());
      });
      if (matchingItem && inputRef.current) {
        const { displayText, itemValue } = getItemData(matchingItem.element);
        if (controlledValue === void 0) {
          setInternalValue(displayText);
        }
        updateSelectedValue(itemValue);
        const itemData = buildItemData(matchingItem.element);
        onSelectionChange?.({ item: itemData, value: displayText });
        const startsWithInput = displayText.toLowerCase().startsWith(newValue.toLowerCase());
        const selectionStart = startsWithInput ? newValue.length : 0;
        requestAnimationFrame(() => {
          inputRef.current?.setSelectionRange(selectionStart, displayText.length);
        });
      } else if (selectedValue !== void 0 || internalSelectedValue !== void 0) {
        updateSelectedValue(void 0);
        onSelectionChange?.({ item: null, value: newValue });
      }
    },
    [controlledValue, navigableItems, updateSelectedValue, onSelectionChange, selectedValue, internalSelectedValue]
  );
  const handleF4 = useCallback(() => {
    if (!isOpen) {
      openPopover();
      if (value) {
        const matchIdx = navigableItems.findIndex((item) => {
          const { displayText } = getItemData(item.element);
          return displayText === value;
        });
        if (matchIdx >= 0) {
          setFocusedIndex(matchIdx);
          syncInputToFocusedItem(matchIdx);
        } else {
          setFocusedIndex(-1);
        }
      } else if (navigableItems.length > 0) {
        setFocusedIndex(0);
        syncInputToFocusedItem(0);
      }
    } else {
      closePopover();
    }
  }, [isOpen, value, navigableItems, openPopover, closePopover, syncInputToFocusedItem]);
  const handleClear = useCallback(() => {
    if (controlledValue === void 0) {
      setInternalValue("");
    }
    setFilterValue("");
    setSelectedItem(null);
    updateSelectedValue(void 0);
    onInput?.("");
    onChange?.("");
    onSelectionChange?.({ item: null, value: "" });
    inputRef.current?.focus();
  }, [controlledValue, onInput, onChange, onSelectionChange, updateSelectedValue]);
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled || readonly) return;
      const findCurrentIndex = () => navigableItems.findIndex((item) => {
        const { displayText } = getItemData(item.element);
        return displayText === value;
      });
      const navigateClosed = (index) => {
        if (index >= 0 && index < navigableItems.length) {
          navigateItemClosed(navigableItems[index].element);
        }
      };
      const focusOpenItem = (index) => {
        setFocusedIndex(index);
        syncInputToFocusedItem(index);
      };
      const navigateToIndex = (index) => {
        if (isOpen) {
          focusOpenItem(index);
        } else {
          navigateClosed(index);
        }
      };
      if (e.key === "F8" && (e.ctrlKey || e.metaKey) && e.altKey && isOpen && portalTarget) {
        e.preventDefault();
        const links = portalTarget.querySelectorAll("a");
        if (links.length > 0) {
          linkNavigationRef.current = true;
          setFocusedIndex(-1);
          links[0].focus();
        }
        return;
      }
      switch (e.key) {
        case "F4":
          e.preventDefault();
          handleF4();
          break;
        case "ArrowDown":
          e.preventDefault();
          if (e.altKey) {
            togglePopover();
          } else if (isOpen) {
            const nextDown = focusedIndex + 1;
            if (nextDown < navigableItems.length) {
              focusOpenItem(nextDown);
            }
          } else {
            navigateClosed(findCurrentIndex() + 1);
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          if (e.altKey) {
            togglePopover();
          } else if (isOpen) {
            if (focusedIndex <= 0) {
              setFocusedIndex(-1);
              if (controlledValue === void 0) {
                setInternalValue(filterValue);
              }
            } else {
              focusOpenItem(focusedIndex - 1);
            }
          } else {
            const currentIndex = findCurrentIndex();
            if (currentIndex > 0) {
              navigateClosed(currentIndex - 1);
            }
          }
          break;
        case "PageDown":
          e.preventDefault();
          navigateToIndex(
            isOpen ? Math.min(focusedIndex + 10, navigableItems.length - 1) : Math.min(findCurrentIndex() + 10, navigableItems.length - 1)
          );
          break;
        case "PageUp":
          e.preventDefault();
          navigateToIndex(
            isOpen ? Math.max(focusedIndex - 10, 0) : Math.max(findCurrentIndex() - 10, 0)
          );
          break;
        case "Home":
          e.preventDefault();
          navigateToIndex(0);
          break;
        case "End":
          e.preventDefault();
          navigateToIndex(navigableItems.length - 1);
          break;
        case "Enter":
          if (isOpen && focusedIndex >= 0 && focusedIndex < navigableItems.length) {
            e.preventDefault();
            selectItem(navigableItems[focusedIndex].element);
          } else if (!isOpen && value !== lastCommittedValue) {
            setLastCommittedValue(value);
            onChange?.(value);
          }
          break;
        case "Escape":
          e.preventDefault();
          if (isOpen) {
            closePopover();
          } else if (showClearIcon && value && !disabled && !readonly) {
            handleClear();
          } else {
            if (controlledValue === void 0) {
              setInternalValue(lastCommittedValue);
            }
            setFilterValue("");
          }
          break;
        case "Tab":
          if (isOpen) {
            closePopover();
          }
          break;
      }
    },
    [
      disabled,
      readonly,
      isOpen,
      focusedIndex,
      navigableItems,
      value,
      filterValue,
      lastCommittedValue,
      controlledValue,
      togglePopover,
      closePopover,
      selectItem,
      navigateItemClosed,
      syncInputToFocusedItem,
      onChange,
      portalTarget,
      handleF4,
      showClearIcon,
      handleClear
    ]
  );
  const handleBlur = useCallback(
    (e) => {
      const container = containerRef.current;
      if (container?.contains(e.relatedTarget)) {
        return;
      }
      if (linkNavigationRef.current) {
        return;
      }
      if (value !== lastCommittedValue) {
        setLastCommittedValue(value);
        onChange?.(value);
      }
    },
    [value, lastCommittedValue, onChange]
  );
  const renderChildren = useCallback(() => {
    let itemIndex = 0;
    const isItemSelected = (element) => {
      const { displayText, itemValue } = getItemData(element);
      if (selectedValue !== void 0) {
        return itemValue === selectedValue;
      }
      return displayText === value;
    };
    const getItemProps = (element) => {
      const currentItemIndex = itemIndex++;
      const visibleItem = visibleItems.find((vi) => vi.originalIndex === currentItemIndex);
      const isVisible = visibleItem?.isVisible ?? true;
      const isSelected = isItemSelected(element);
      const isFocused = navigableItems.findIndex((ni) => ni.originalIndex === currentItemIndex) === focusedIndex;
      const showSelected = isOpen && focusedIndex >= 0 ? isFocused : isSelected;
      return { isVisible, selected: showSelected, focused: isFocused, onClick: () => selectItem(element) };
    };
    return Children.map(children, (child) => {
      if (!isValidElement(child)) return child;
      if (child.type?.displayName === "ComboBoxItemGroup") {
        const groupProps = child.props;
        let hasVisibleChild = false;
        const groupChildren = Children.map(groupProps.children, (groupChild) => {
          if (!isValidElement(groupChild)) return groupChild;
          const props = getItemProps(groupChild);
          if (props.isVisible) hasVisibleChild = true;
          return cloneElement(groupChild, props);
        });
        return cloneElement(child, {
          isVisible: hasVisibleChild,
          children: groupChildren
        });
      }
      return cloneElement(
        child,
        getItemProps(child)
      );
    });
  }, [children, visibleItems, navigableItems, focusedIndex, value, selectedValue, isOpen, selectItem]);
  const showClear = showClearIcon && value && !disabled && !readonly;
  return /* @__PURE__ */ jsxs("div", { ref, className: cn("relative w-full", className), children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: containerRef,
        className: cn(
          inputContainerVariants({
            size,
            valueState,
            disabled,
            readonly
          }),
          isOpen && focusedIndex >= 0 && (isLargeSize ? "!border !pl-3.5 !pr-1 !py-1" : "!border !pl-3 !pr-1 !py-1")
        ),
        children: [
          icon && /* @__PURE__ */ jsx("span", { className: "flex items-center justify-center shrink-0 text-sapphire-text-tertiary", children: icon }),
          /* @__PURE__ */ jsx(
            "input",
            {
              ref: inputRef,
              id: inputId,
              type: "text",
              role: "combobox",
              "aria-expanded": isOpen,
              "aria-haspopup": "listbox",
              "aria-autocomplete": "both",
              "aria-controls": [isOpen ? popoverId : null, accessibilityAttributes?.controls].filter(Boolean).join(" ") || void 0,
              "aria-activedescendant": focusedIndex >= 0 && focusedIndex < navigableItems.length ? (() => {
                const item = navigableItems[focusedIndex];
                if (!item) return void 0;
                const { displayText, itemValue } = getItemData(item.element);
                return itemValue ?? displayText;
              })() : void 0,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-required": required,
              "aria-invalid": isNegativeState || void 0,
              "aria-busy": loading || void 0,
              "aria-describedby": [
                valueStateMessage ? messageId : null,
                hasValueState ? linkShortcutHintId : null,
                accessibleDescriptionRef,
                accessibleDescription ? accessibleDescSpanId : null
              ].filter(Boolean).join(" ") || void 0,
              name,
              value,
              placeholder,
              disabled,
              readOnly: readonly,
              onChange: handleInputChange,
              onKeyDown: handleKeyDown,
              onBlur: handleBlur,
              className: cn(
                "flex-1 h-full min-w-0 bg-transparent text-sm text-sapphire-text-primary",
                "outline-none",
                "placeholder:text-sapphire-text-tertiary placeholder:italic",
                "disabled:cursor-not-allowed",
                readonly && "text-sapphire-text-tertiary"
              )
            }
          ),
          showClear && /* @__PURE__ */ jsx(ClearButton, { onClick: handleClear, comboBoxSize: size }),
          !readonly && /* @__PURE__ */ jsx(
            DropdownArrowButton,
            {
              onClick: () => {
                togglePopover();
                inputRef.current?.focus();
              },
              disabled,
              comboBoxSize: size
            }
          )
        ]
      }
    ),
    /* @__PURE__ */ jsx(
      ComboBoxPopover,
      {
        open: isOpen,
        triggerRef: containerRef,
        hasValueState,
        valueStatePortalRef: setPortalTarget,
        loading,
        onClose: handlePopoverClose,
        children: renderChildren()
      }
    ),
    valueStateMessage && /* @__PURE__ */ jsx(
      ValueStateMessage,
      {
        isOpen,
        hasValueState,
        portalTarget,
        messageHeight,
        messageRef,
        messageId,
        normalizedValueState,
        valueStateMessage
      }
    ),
    accessibleDescription && /* @__PURE__ */ jsx("span", { id: accessibleDescSpanId, className: "sr-only", children: accessibleDescription }),
    hasValueState && /* @__PURE__ */ jsx("span", { id: linkShortcutHintId, className: "sr-only", children: t("COMBOBOX_VALUE_STATE_LINK_SHORTCUT") }),
    /* @__PURE__ */ jsx(LiveRegion, { message: resultAnnouncement })
  ] });
}
function ComboBoxItem({
  text,
  additionalText,
  value,
  icon: _icon,
  disabled = false,
  selected = false,
  focused = false,
  isVisible = true,
  onClick,
  children,
  className,
  ref
}) {
  if (!isVisible) {
    return null;
  }
  const displayText = text ?? (typeof children === "string" ? children : "");
  const itemValue = value ?? displayText;
  return /* @__PURE__ */ jsxs(
    "li",
    {
      ref,
      role: "option",
      "aria-selected": selected,
      "aria-disabled": disabled || void 0,
      id: itemValue,
      "data-value": itemValue,
      onClick: onClick && !disabled ? () => onClick() : void 0,
      className: cn(
        "flex items-center min-h-8 px-4 py-1.5 gap-4 text-sm cursor-pointer",
        "bg-sapphire-background-primary text-sapphire-text-primary",
        "transition-colors outline-none relative",
        !disabled && "hover:bg-sapphire-neutral-hover-background dark:hover:bg-neutral-800",
        focused && !selected && "bg-sapphire-neutral-hover-background dark:bg-neutral-800",
        focused && "after:absolute after:inset-0.5 after:border-2 after:border-ring after:rounded-sm after:pointer-events-none",
        selected && "bg-sapphire-brand-selected-background",
        disabled && "opacity-50 pointer-events-none cursor-not-allowed",
        className
      ),
      children: [
        /* @__PURE__ */ jsx("span", { className: "flex-1", children: children || displayText }),
        additionalText && /* @__PURE__ */ jsx("span", { className: "shrink-0 text-sapphire-text-tertiary text-sm", children: additionalText })
      ]
    }
  );
}
ComboBoxItem.displayName = "ComboBoxItem";
function ComboBoxItemGroup({ headerText, isVisible = true, children, className, ref }) {
  if (!isVisible) {
    return null;
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      role: "group",
      "aria-label": headerText,
      className: cn("py-1", className),
      children: [
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "px-3 py-1.5 text-xs font-medium text-sapphire-text-tertiary",
              "uppercase tracking-wide select-none"
            ),
            "aria-hidden": "true",
            children: headerText
          }
        ),
        /* @__PURE__ */ jsx("div", { role: "presentation", children })
      ]
    }
  );
}
ComboBoxItemGroup.displayName = "ComboBoxItemGroup";

export { ComboBox, ComboBoxItem, ComboBoxItemGroup };
