// @sap-ui/fx-components/dist/chunk-P6PNCHF2.js
import { Tokenizer } from './chunk-T77SNGYR.js';
import { ComboBoxPopover } from './chunk-EEGGRFXR.js';
import { computeVisibleItems, computeNavigableItems, useResultAnnouncement, useMessageHeight, usePopoverControls, ClearButton, DropdownArrowButton, ValueStateMessage, LiveRegion, valueStateInputVariants } from './chunk-KQMKCSMK.js';
import { AcceptIcon } from './chunk-VNDUTC7E.js';
import { Token } from './chunk-A3BV5G37.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useMemo, Children, isValidElement, useEffect, useCallback, cloneElement } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

var inputContainerVariants = (opts) => {
  const isLarge = opts.size === "Large" /* Large */;
  return cn(
    valueStateInputVariants(opts),
    isLarge ? "h-10 pr-1 gap-1" : "h-8 pr-1 gap-1",
    opts.hasTokens ? isLarge ? "pl-1 py-0" : "pl-0.5 py-0" : isLarge ? "pl-3.5 py-1" : "pl-3 py-1",
    opts.hasTokens ? isLarge ? "focus-within:border-2 focus-within:pl-[3px] focus-within:pr-[3px] focus-within:py-0" : "focus-within:border-2 focus-within:pl-px focus-within:pr-[3px] focus-within:py-0" : isLarge ? "focus-within:border-2 focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]" : "focus-within:border-2 focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]"
  );
};
function getItemData(element) {
  const { text, value, children, additionalText, icon, disabled } = element.props;
  const displayText = text ?? (typeof children === "string" ? children : "");
  const itemValue = value ?? displayText;
  return { displayText, itemValue, additionalText, icon, disabled };
}
function buildItemData(element) {
  const { displayText, itemValue, additionalText, icon, disabled } = getItemData(element);
  return { text: displayText, value: itemValue, additionalText, icon, disabled };
}
function MultiComboBox({
  value: controlledValue,
  defaultValue = "",
  size = "Large" /* Large */,
  selectedValues: controlledSelectedValues,
  defaultSelectedValues = [],
  placeholder = "",
  disabled = false,
  readonly = false,
  required = false,
  name,
  filter = "StartsWithPerTerm" /* StartsWithPerTerm */,
  noTypeahead: _noTypeahead = false,
  noValidation: _noValidation = false,
  valueState = "None" /* None */,
  valueStateMessage,
  showClearIcon = false,
  showSelectAll = false,
  loading = false,
  open: controlledOpen,
  defaultOpen = false,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onInput,
  onChange,
  onSelectionChange,
  onOpen,
  onClose,
  children,
  className,
  icon,
  accessibilityAttributes,
  ref
}) {
  const containerRef = useRef(null);
  const inputRef = useRef(null);
  const tokenizerRef = useRef(null);
  const listRef = useRef(null);
  const selectAllRef = useRef(null);
  const selectAllFocusedRef = useRef(false);
  const messageRef = useRef(null);
  const [portalTarget, setPortalTarget] = useState(null);
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [internalOpen, setInternalOpen] = useState(defaultOpen);
  const [filterValue, setFilterValue] = useState("");
  const [focusedIndex, setFocusedIndex] = useState(-1);
  const [selectedOnTop, setSelectedOnTop] = useState(false);
  const [inputFocused, setInputFocused] = useState(false);
  const [internalSelectedValues, setInternalSelectedValues] = useState(
    () => new Set(defaultSelectedValues)
  );
  const inputValue = controlledValue ?? internalValue;
  const isOpen = controlledOpen ?? internalOpen;
  const selectedValuesSet = useMemo(() => {
    if (controlledSelectedValues !== void 0) {
      return new Set(controlledSelectedValues);
    }
    return internalSelectedValues;
  }, [controlledSelectedValues, internalSelectedValues]);
  const normalizedValueState = valueState || "None" /* None */;
  const hasValueState = normalizedValueState !== "None" /* None */ && !!valueStateMessage;
  const isNegativeState = normalizedValueState === "Negative" /* Negative */;
  const tokensExpanded = inputFocused || isOpen;
  const baseId = useMemo(() => `multicombobox-${Math.random().toString(36).slice(2, 9)}`, []);
  const popoverId = `${baseId}-popover`;
  const inputId = `${baseId}-input`;
  const messageId = `${baseId}-message`;
  const accessibleDescSpanId = `${baseId}-accessible-desc`;
  const flattenedItems = useMemo(() => {
    const items = [];
    Children.forEach(children, (child, groupIdx) => {
      if (!isValidElement(child)) return;
      if (child.type?.displayName === "MultiComboBoxItemGroup") {
        const groupProps = child.props;
        Children.forEach(groupProps.children, (groupChild) => {
          if (isValidElement(groupChild)) {
            items.push({
              element: groupChild,
              groupIndex: groupIdx
            });
          }
        });
      } else {
        items.push({
          element: child
        });
      }
    });
    return items;
  }, [children]);
  const visibleItems = useMemo(
    () => computeVisibleItems(flattenedItems, filterValue, filter, (el) => getItemData(el).displayText),
    [flattenedItems, filterValue, filter]
  );
  const navigableItems = useMemo(() => {
    const base = computeNavigableItems(visibleItems);
    if (!selectedOnTop) return base;
    return base.filter(({ element }) => {
      const { itemValue } = getItemData(element);
      return selectedValuesSet.has(itemValue);
    });
  }, [visibleItems, selectedOnTop, selectedValuesSet]);
  const selectedItems = useMemo(() => {
    const items = [];
    for (const { element } of flattenedItems) {
      const { displayText, itemValue } = getItemData(element);
      if (selectedValuesSet.has(itemValue)) {
        items.push({ text: displayText, value: itemValue });
      }
    }
    return items;
  }, [flattenedItems, selectedValuesSet]);
  const filteredCount = useMemo(
    () => visibleItems.filter((item) => item.isVisible).length,
    [visibleItems]
  );
  const { resultAnnouncement, t } = useResultAnnouncement(filterValue, filteredCount);
  useEffect(() => {
    if (!isOpen || focusedIndex < 0) return;
    if (navigableItems.length === 0) {
      setFocusedIndex(-1);
    } else if (focusedIndex >= navigableItems.length) {
      setFocusedIndex(navigableItems.length - 1);
    }
  }, [isOpen, focusedIndex, navigableItems.length]);
  useEffect(() => {
    if (!isOpen) return;
    if (focusedIndex >= 0) {
      requestAnimationFrame(() => {
        const popoverEl = containerRef.current?.parentElement?.querySelector?.("[popover]") ?? document.querySelector(`[data-popover-open]`);
        if (!popoverEl) return;
        const options = popoverEl.querySelectorAll('li[role="option"]:not([data-select-all])');
        let visibleIdx = 0;
        for (let i = 0; i < options.length; i++) {
          if (options[i].offsetParent !== null || options[i].offsetHeight > 0) {
            if (visibleIdx === focusedIndex) {
              options[i].focus();
              return;
            }
            visibleIdx++;
          }
        }
      });
    } else if (focusedIndex === -1) {
      if (selectAllFocusedRef.current) {
        requestAnimationFrame(() => {
          selectAllRef.current?.focus();
        });
        selectAllFocusedRef.current = false;
      } else {
        inputRef.current?.focus();
      }
    }
  }, [focusedIndex, isOpen, navigableItems.length]);
  const messageHeight = useMessageHeight(messageRef, valueStateMessage);
  const { openPopover, closePopover: baseClosePopover, handlePopoverClose: baseHandlePopoverClose, togglePopover } = usePopoverControls({
    disabled,
    readonly,
    controlledOpen,
    isOpen,
    setInternalOpen,
    setFocusedIndex,
    onOpen,
    onClose
  });
  const closePopover = useCallback(() => {
    setSelectedOnTop(false);
    baseClosePopover();
  }, [baseClosePopover]);
  const handlePopoverClose = useCallback(() => {
    setSelectedOnTop(false);
    baseHandlePopoverClose();
  }, [baseHandlePopoverClose]);
  const buildSelectedItemsList = useCallback(
    (valuesSet) => {
      const items = [];
      for (const { element } of flattenedItems) {
        const data = buildItemData(element);
        if (valuesSet.has(data.value ?? data.text)) {
          items.push(data);
        }
      }
      return items;
    },
    [flattenedItems]
  );
  const toggleItem = useCallback(
    (itemElement) => {
      const { itemValue } = getItemData(itemElement);
      const newSet = new Set(selectedValuesSet);
      const wasSelected = newSet.has(itemValue);
      if (wasSelected) {
        newSet.delete(itemValue);
      } else {
        newSet.add(itemValue);
      }
      if (controlledSelectedValues === void 0) {
        setInternalSelectedValues(newSet);
      }
      const changedItem = buildItemData(itemElement);
      onSelectionChange?.({
        items: buildSelectedItemsList(newSet),
        changedItem,
        selected: !wasSelected
      });
    },
    [selectedValuesSet, controlledSelectedValues, onSelectionChange, buildSelectedItemsList]
  );
  const handleTokenDelete = useCallback(
    (detail) => {
      const newSet = new Set(selectedValuesSet);
      const deletedItems = [];
      for (const idx of detail.tokens) {
        if (idx >= 0 && idx < selectedItems.length) {
          const item = selectedItems[idx];
          newSet.delete(item.value);
          const flatItem = flattenedItems.find((fi) => {
            const { itemValue } = getItemData(fi.element);
            return itemValue === item.value;
          });
          if (flatItem) {
            deletedItems.push(buildItemData(flatItem.element));
          }
        }
      }
      if (controlledSelectedValues === void 0) {
        setInternalSelectedValues(newSet);
      }
      for (const changedItem of deletedItems) {
        onSelectionChange?.({
          items: buildSelectedItemsList(newSet),
          changedItem,
          selected: false
        });
      }
      inputRef.current?.focus();
    },
    [selectedValuesSet, selectedItems, flattenedItems, controlledSelectedValues, onSelectionChange, buildSelectedItemsList]
  );
  const handleSelectAll = useCallback(() => {
    const visibleNonDisabled = navigableItems;
    const allVisibleValues = visibleNonDisabled.map((item) => getItemData(item.element).itemValue);
    const allSelected = allVisibleValues.every((v) => selectedValuesSet.has(v));
    const newSet = new Set(selectedValuesSet);
    if (allSelected) {
      for (const v of allVisibleValues) {
        newSet.delete(v);
      }
    } else {
      for (const v of allVisibleValues) {
        newSet.add(v);
      }
    }
    if (controlledSelectedValues === void 0) {
      setInternalSelectedValues(newSet);
    }
    const dummyItem = { text: allSelected ? "Deselect All" : "Select All" };
    onSelectionChange?.({
      items: buildSelectedItemsList(newSet),
      changedItem: dummyItem,
      selected: !allSelected
    });
  }, [navigableItems, selectedValuesSet, controlledSelectedValues, onSelectionChange, buildSelectedItemsList]);
  const selectAllState = useMemo(() => {
    if (!showSelectAll) return "unchecked";
    const visibleValues = navigableItems.map((item) => getItemData(item.element).itemValue);
    if (visibleValues.length === 0) return "unchecked";
    const selectedCount = visibleValues.filter((v) => selectedValuesSet.has(v)).length;
    if (selectedCount === 0) return "unchecked";
    if (selectedCount === visibleValues.length) return "checked";
    return "indeterminate";
  }, [showSelectAll, navigableItems, selectedValuesSet]);
  const handleInputChange = useCallback(
    (e) => {
      const newValue = e.target.value;
      if (controlledValue === void 0) {
        setInternalValue(newValue);
      }
      setFilterValue(newValue);
      setSelectedOnTop(false);
      onInput?.(newValue);
      if (!isOpen && newValue) {
        openPopover();
      }
      setFocusedIndex(-1);
    },
    [controlledValue, isOpen, onInput, openPopover]
  );
  const handleClear = useCallback(() => {
    if (controlledValue === void 0) {
      setInternalValue("");
    }
    setFilterValue("");
    const newSet = /* @__PURE__ */ new Set();
    if (controlledSelectedValues === void 0) {
      setInternalSelectedValues(newSet);
    }
    onInput?.("");
    onChange?.("");
    onSelectionChange?.({
      items: [],
      changedItem: { text: "Clear All" },
      selected: false
    });
    inputRef.current?.focus();
  }, [controlledValue, controlledSelectedValues, onInput, onChange, onSelectionChange]);
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled) return;
      if (readonly && e.key !== "Tab") return;
      switch (e.key) {
        case "F4":
          e.preventDefault();
          togglePopover();
          break;
        case "ArrowDown":
          e.preventDefault();
          if (e.altKey) {
            togglePopover();
          } else if (isOpen) {
            if (showSelectAll && navigableItems.length > 0) {
              selectAllRef.current?.focus();
            } else if (navigableItems.length > 0) {
              setFocusedIndex(0);
            }
          } else {
            openPopover();
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          if (e.altKey) {
            togglePopover();
          }
          break;
        case "ArrowLeft": {
          const input = inputRef.current;
          if (input && input.selectionStart === 0 && input.selectionEnd === 0 && selectedItems.length > 0) {
            e.preventDefault();
            tokenizerRef.current?.focusLast();
          }
          break;
        }
        case "Backspace": {
          const inp = inputRef.current;
          if (inp && !inputValue && selectedItems.length > 0) {
            e.preventDefault();
            tokenizerRef.current?.focusLast();
          }
          break;
        }
        case "Enter":
          if (!isOpen && inputValue) {
            onChange?.(inputValue);
          }
          break;
        case "Escape":
          e.preventDefault();
          if (isOpen) {
            closePopover();
          } else if (showClearIcon && (selectedItems.length > 0 || inputValue) && !disabled && !readonly) {
            handleClear();
          }
          break;
        case "Tab":
          if (isOpen) {
            closePopover();
          }
          break;
        case "a":
        case "A":
          if ((e.ctrlKey || e.metaKey) && showSelectAll && isOpen) {
            e.preventDefault();
            handleSelectAll();
          }
          break;
      }
    },
    [
      disabled,
      readonly,
      isOpen,
      navigableItems,
      inputValue,
      selectedItems.length,
      togglePopover,
      openPopover,
      closePopover,
      onChange,
      showSelectAll,
      handleSelectAll,
      showClearIcon,
      handleClear
    ]
  );
  const handleBlur = useCallback(
    (e) => {
      const container = containerRef.current;
      if (container?.contains(e.relatedTarget)) {
        return;
      }
      setInputFocused(false);
    },
    []
  );
  const renderChildren = useCallback(() => {
    let itemIndex = 0;
    const getItemProps = (element) => {
      const currentItemIndex = itemIndex++;
      const visibleItem = visibleItems.find((vi) => vi.originalIndex === currentItemIndex);
      const isVisible = visibleItem?.isVisible ?? true;
      const { itemValue } = getItemData(element);
      const isSelected = selectedValuesSet.has(itemValue);
      const navIndex = navigableItems.findIndex((ni) => ni.originalIndex === currentItemIndex);
      const isFocused = navIndex === focusedIndex;
      return {
        isVisible,
        selected: isSelected,
        focused: isFocused,
        tabIndex: isFocused ? 0 : -1,
        onClick: () => toggleItem(element)
      };
    };
    if (selectedOnTop) {
      const result = [];
      Children.forEach(children, (child) => {
        if (!isValidElement(child)) return;
        if (child.type?.displayName === "MultiComboBoxItemGroup") {
          const groupProps = child.props;
          const selectedGroupChildren = [];
          Children.forEach(groupProps.children, (groupChild) => {
            if (!isValidElement(groupChild)) return;
            const props = getItemProps(groupChild);
            if (props.selected) {
              selectedGroupChildren.push(
                cloneElement(groupChild, props)
              );
            }
          });
          if (selectedGroupChildren.length > 0) {
            result.push(
              cloneElement(child, {
                isVisible: true,
                children: selectedGroupChildren
              })
            );
          }
        } else {
          const props = getItemProps(child);
          if (props.selected) {
            result.push(
              cloneElement(child, props)
            );
          }
        }
      });
      return result;
    }
    return Children.map(children, (child) => {
      if (!isValidElement(child)) return child;
      if (child.type?.displayName === "MultiComboBoxItemGroup") {
        const groupProps = child.props;
        let hasVisibleChild = false;
        const groupChildren = Children.map(groupProps.children, (groupChild) => {
          if (!isValidElement(groupChild)) return groupChild;
          const props = getItemProps(groupChild);
          if (props.isVisible) hasVisibleChild = true;
          return cloneElement(groupChild, props);
        });
        return cloneElement(child, {
          isVisible: hasVisibleChild,
          children: groupChildren
        });
      }
      return cloneElement(
        child,
        getItemProps(child)
      );
    });
  }, [children, visibleItems, navigableItems, focusedIndex, selectedValuesSet, toggleItem, selectedOnTop]);
  const handleListKeyDown = useCallback(
    (e) => {
      const isOnSelectAll = e.target.hasAttribute("data-select-all");
      switch (e.key) {
        case "ArrowDown":
          e.preventDefault();
          if (isOnSelectAll) {
            if (navigableItems.length > 0) {
              setFocusedIndex(0);
            }
          } else if (focusedIndex < navigableItems.length - 1) {
            setFocusedIndex(focusedIndex + 1);
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          if (isOnSelectAll) {
            setFocusedIndex(-1);
            inputRef.current?.focus();
          } else if (focusedIndex > 0) {
            setFocusedIndex(focusedIndex - 1);
          } else if (showSelectAll) {
            selectAllFocusedRef.current = true;
            setFocusedIndex(-1);
          } else {
            setFocusedIndex(-1);
          }
          break;
        case "Enter":
        case " ":
          e.preventDefault();
          if (isOnSelectAll) {
            handleSelectAll();
          } else if (focusedIndex >= 0 && focusedIndex < navigableItems.length) {
            toggleItem(navigableItems[focusedIndex].element);
          }
          break;
        case "Escape":
          e.preventDefault();
          closePopover();
          inputRef.current?.focus();
          break;
        case "Tab":
          closePopover();
          break;
        case "Home":
          e.preventDefault();
          if (showSelectAll) {
            selectAllFocusedRef.current = true;
            setFocusedIndex(-1);
          } else if (navigableItems.length > 0) {
            setFocusedIndex(0);
          }
          break;
        case "End":
          e.preventDefault();
          if (navigableItems.length > 0) {
            setFocusedIndex(navigableItems.length - 1);
          }
          break;
        case "F4":
          e.preventDefault();
          closePopover();
          inputRef.current?.focus();
          break;
      }
    },
    [focusedIndex, navigableItems, toggleItem, closePopover, showSelectAll, handleSelectAll]
  );
  const showClear = showClearIcon && (selectedItems.length > 0 || inputValue) && !disabled && !readonly;
  const showPlaceholder = selectedItems.length === 0;
  return /* @__PURE__ */ jsxs("div", { ref, className: cn("relative w-full", className), children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: containerRef,
        className: cn(
          inputContainerVariants({
            size,
            valueState,
            disabled,
            readonly,
            hasTokens: selectedItems.length > 0
          })
        ),
        onClick: () => {
          if (!disabled && !readonly) {
            inputRef.current?.focus();
          }
        },
        children: [
          icon && /* @__PURE__ */ jsx("span", { className: "flex items-center justify-center shrink-0 text-sapphire-text-tertiary", children: icon }),
          selectedItems.length > 0 && /* @__PURE__ */ jsx(
            Tokenizer,
            {
              ref: tokenizerRef,
              readonly,
              disabled,
              multiLine: tokensExpanded,
              onTokenDelete: handleTokenDelete,
              onNMoreClick: () => {
                setSelectedOnTop(true);
                openPopover();
                inputRef.current?.focus();
              },
              className: cn(
                "shrink min-w-0 [&_[data-part='content']]:gap-1 !h-full [&_[data-part='content']]:h-full",
                tokensExpanded ? "!flex-nowrap !items-center overflow-x-auto overflow-y-hidden scrollbar-none [&_[data-part='content']]:!flex-nowrap" : ""
              ),
              children: selectedItems.map((item, idx) => /* @__PURE__ */ jsx(
                Token,
                {
                  text: item.text,
                  readonly,
                  "data-testid": `token-${idx}`
                },
                item.value
              ))
            }
          ),
          !readonly && /* @__PURE__ */ jsx(
            "input",
            {
              ref: inputRef,
              id: inputId,
              type: "text",
              role: "combobox",
              "aria-expanded": isOpen,
              "aria-haspopup": "listbox",
              "aria-autocomplete": "list",
              "aria-controls": [isOpen ? popoverId : null, accessibilityAttributes?.controls].filter(Boolean).join(" ") || void 0,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-required": required,
              "aria-invalid": isNegativeState || void 0,
              "aria-busy": loading || void 0,
              "aria-describedby": [
                valueStateMessage ? messageId : null,
                accessibleDescriptionRef,
                accessibleDescription ? accessibleDescSpanId : null
              ].filter(Boolean).join(" ") || void 0,
              name: name ? `${name}_input` : void 0,
              value: inputValue,
              placeholder: showPlaceholder ? placeholder : void 0,
              disabled,
              onChange: handleInputChange,
              onKeyDown: handleKeyDown,
              onFocus: () => setInputFocused(true),
              onBlur: handleBlur,
              className: cn(
                "flex-1 h-6 bg-transparent text-sm text-sapphire-text-primary",
                "outline-none",
                "placeholder:text-sapphire-text-tertiary placeholder:italic",
                "disabled:cursor-not-allowed",
                // When tokens are present, allow input to shrink to near-zero when not typing
                selectedItems.length > 0 ? "min-w-[40px]" : "min-w-[60px]"
              )
            }
          ),
          showClear && /* @__PURE__ */ jsx(ClearButton, { onClick: handleClear, accessibleName: "Clear selection", comboBoxSize: size }),
          !readonly && /* @__PURE__ */ jsx(
            DropdownArrowButton,
            {
              onClick: () => {
                togglePopover();
                inputRef.current?.focus();
              },
              disabled,
              comboBoxSize: size
            }
          )
        ]
      }
    ),
    name && selectedItems.map((item, i) => /* @__PURE__ */ jsx("input", { type: "hidden", name, value: item.value }, i)),
    /* @__PURE__ */ jsx(
      ComboBoxPopover,
      {
        open: isOpen,
        triggerRef: containerRef,
        hasValueState,
        valueStatePortalRef: setPortalTarget,
        loading,
        onClose: handlePopoverClose,
        headerContent: showSelectAll && navigableItems.length > 0 ? (
          // eslint-disable-next-line jsx-a11y/no-static-element-interactions
          /* @__PURE__ */ jsxs(
            "div",
            {
              ref: selectAllRef,
              role: "checkbox",
              "data-select-all": true,
              tabIndex: -1,
              "aria-checked": selectAllState === "checked" ? true : selectAllState === "indeterminate" ? "mixed" : false,
              "aria-label": t("MULTICOMBOBOX_SELECT_ALL"),
              onClick: handleSelectAll,
              onKeyDown: handleListKeyDown,
              className: cn(
                "flex items-center min-h-8 px-4 py-1.5 gap-4 text-sm cursor-pointer",
                "text-sapphire-text-primary bg-sapphire-background-primary",
                "hover:bg-sapphire-neutral-hover-background",
                "focus:bg-sapphire-neutral-hover-background",
                "transition-colors outline-none border-b border-sapphire-border-primary relative",
                "focus:after:absolute focus:after:inset-0.5 focus:after:border-2 focus:after:border-ring focus:after:rounded-sm focus:after:pointer-events-none"
              ),
              children: [
                /* @__PURE__ */ jsxs(
                  "span",
                  {
                    className: cn(
                      "flex items-center justify-center shrink-0 size-4 rounded border border-solid transition-colors",
                      selectAllState === "checked" ? "bg-sapphire-brand-background border-sapphire-brand-background text-white" : selectAllState === "indeterminate" ? "bg-sapphire-brand-background border-sapphire-brand-background text-white" : "bg-sapphire-canvas-primary border-sapphire-border-secondary"
                    ),
                    "aria-hidden": "true",
                    children: [
                      selectAllState === "checked" && /* @__PURE__ */ jsx(AcceptIcon, { className: "size-3" }),
                      selectAllState === "indeterminate" && /* @__PURE__ */ jsx("span", { className: "block w-2 h-0.5 bg-white rounded-sm" })
                    ]
                  }
                ),
                /* @__PURE__ */ jsx("span", { className: "flex-1 font-medium", children: t("MULTICOMBOBOX_SELECT_ALL") })
              ]
            }
          )
        ) : void 0,
        children: /* @__PURE__ */ jsx("div", { ref: listRef, onKeyDown: handleListKeyDown, children: renderChildren() })
      }
    ),
    valueStateMessage && /* @__PURE__ */ jsx(
      ValueStateMessage,
      {
        isOpen,
        hasValueState,
        portalTarget,
        messageHeight,
        messageRef,
        messageId,
        normalizedValueState,
        valueStateMessage
      }
    ),
    accessibleDescription && /* @__PURE__ */ jsx("span", { id: accessibleDescSpanId, className: "sr-only", children: accessibleDescription }),
    /* @__PURE__ */ jsx(LiveRegion, { message: resultAnnouncement })
  ] });
}
function MultiComboBoxItem({
  text,
  additionalText,
  value,
  icon: _icon,
  disabled = false,
  selected = false,
  focused = false,
  isVisible = true,
  onClick,
  children,
  className,
  tabIndex,
  ref
}) {
  if (!isVisible) {
    return null;
  }
  const displayText = text ?? (typeof children === "string" ? children : "");
  const itemValue = value ?? displayText;
  return /* @__PURE__ */ jsxs(
    "li",
    {
      ref,
      role: "option",
      "aria-selected": selected,
      "aria-disabled": disabled || void 0,
      id: itemValue,
      "data-value": itemValue,
      onClick: onClick && !disabled ? () => onClick() : void 0,
      tabIndex,
      className: cn(
        "flex items-center h-8 px-4 gap-2 text-sm cursor-pointer",
        "text-sapphire-text-primary",
        "transition-colors outline-none relative",
        !disabled && !selected && "bg-sapphire-background-primary",
        !disabled && !selected && "hover:bg-sapphire-neutral-hover-background dark:hover:bg-neutral-800",
        focused && !selected && "bg-sapphire-neutral-hover-background dark:bg-neutral-800",
        focused && "after:absolute after:inset-0.5 after:border-2 after:border-ring after:rounded-sm after:pointer-events-none",
        selected && "bg-sapphire-brand-selected-background",
        disabled && "opacity-50 pointer-events-none cursor-not-allowed",
        className
      ),
      children: [
        /* @__PURE__ */ jsx(
          "span",
          {
            className: cn(
              "relative inline-flex items-center justify-center shrink-0",
              "size-4 rounded p-[2px] border border-solid transition-colors",
              selected ? "bg-sapphire-canvas-primary border-primary text-primary" : "bg-sapphire-canvas-primary border-sapphire-border-secondary"
            ),
            "aria-hidden": "true",
            children: selected && /* @__PURE__ */ jsx(AcceptIcon, { className: "size-3" })
          }
        ),
        /* @__PURE__ */ jsx("span", { className: "flex-1 truncate", children: children || displayText }),
        additionalText && /* @__PURE__ */ jsx("span", { className: "shrink-0 text-sapphire-text-tertiary text-sm", children: additionalText })
      ]
    }
  );
}
MultiComboBoxItem.displayName = "MultiComboBoxItem";
function MultiComboBoxItemGroup({ headerText, isVisible = true, children, className, ref }) {
  if (!isVisible) {
    return null;
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      role: "group",
      "aria-label": headerText,
      className: cn("py-1", className),
      children: [
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "px-3 py-1.5 text-xs font-medium text-sapphire-text-tertiary",
              "uppercase tracking-wide select-none"
            ),
            "aria-hidden": "true",
            children: headerText
          }
        ),
        /* @__PURE__ */ jsx("div", { role: "presentation", children })
      ]
    }
  );
}
MultiComboBoxItemGroup.displayName = "MultiComboBoxItemGroup";

export { MultiComboBox, MultiComboBoxItem, MultiComboBoxItemGroup };
