// @sap-ui/fx-components/dist/chunk-2DPLFZUY.js
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useImperativeHandle } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

var toggleButtonVariants = cva(
  [
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        ["Secondary" /* Secondary */]: "border focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring",
        ["Primary" /* Primary */]: "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus",
        ["PrimaryJoule" /* PrimaryJoule */]: "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus",
        ["SecondaryJoule" /* SecondaryJoule */]: "border focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule",
        ["Neutral" /* Neutral */]: "border focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground",
        ["SecondaryNeutral" /* SecondaryNeutral */]: "border border-transparent focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground",
        ["TertiaryNeutral" /* TertiaryNeutral */]: "border border-transparent focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground",
        ["Tertiary" /* Tertiary */]: "border border-transparent focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring",
        ["TertiaryJoule" /* TertiaryJoule */]: "border border-transparent focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
      },
      pressed: {
        true: "",
        false: ""
      }
    },
    compoundVariants: [
      // Secondary
      {
        design: "Secondary" /* Secondary */,
        pressed: false,
        className: "bg-transparent text-sapphire-brand-foreground border-sapphire-brand-foreground hover:bg-sapphire-neutral-hover-background active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground"
      },
      {
        design: "Secondary" /* Secondary */,
        pressed: true,
        className: "bg-sapphire-brand-selected-background text-sapphire-brand-foreground border-transparent hover:bg-sapphire-brand-toggle-background hover:border-sapphire-border-toggle"
      },
      // Primary
      {
        design: "Primary" /* Primary */,
        pressed: false,
        className: "bg-sapphire-brand-background/20 text-sapphire-brand-foreground hover:bg-sapphire-brand-background/30"
      },
      {
        design: "Primary" /* Primary */,
        pressed: true,
        className: "bg-sapphire-brand-hover-background text-sapphire-neutral-foreground-white hover:bg-blue-800"
      },
      // PrimaryJoule
      {
        design: "PrimaryJoule" /* PrimaryJoule */,
        pressed: false,
        className: "bg-sapphire-joule-background/20 text-sapphire-joule-foreground hover:bg-sapphire-joule-background/30"
      },
      {
        design: "PrimaryJoule" /* PrimaryJoule */,
        pressed: true,
        className: "bg-sapphire-joule-hover-background text-sapphire-neutral-foreground-white hover:bg-purple-800"
      },
      // SecondaryJoule
      {
        design: "SecondaryJoule" /* SecondaryJoule */,
        pressed: false,
        className: "bg-transparent text-sapphire-joule-foreground border-sapphire-joule-foreground hover:bg-sapphire-neutral-hover-background active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground"
      },
      {
        design: "SecondaryJoule" /* SecondaryJoule */,
        pressed: true,
        className: "bg-sapphire-joule-background-area text-sapphire-joule-foreground border-transparent hover:bg-sapphire-joule-background-light hover:border-sapphire-border-toggle"
      },
      // Neutral
      {
        design: "Neutral" /* Neutral */,
        pressed: false,
        className: "bg-transparent text-sapphire-neutral-foreground-black border-sapphire-neutral-foreground-muted hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black"
      },
      {
        design: "Neutral" /* Neutral */,
        pressed: true,
        className: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black border-sapphire-neutral-foreground-muted hover:bg-sapphire-neutral-hover-background"
      },
      // SecondaryNeutral
      {
        design: "SecondaryNeutral" /* SecondaryNeutral */,
        pressed: false,
        className: "bg-transparent text-sapphire-neutral-foreground-black border-transparent hover:bg-sapphire-neutral-hover-background active:bg-sapphire-neutral-pressed-background"
      },
      {
        design: "SecondaryNeutral" /* SecondaryNeutral */,
        pressed: true,
        className: "bg-sapphire-brand-selected-background text-sapphire-brand-foreground border-sapphire-border-toggle hover:bg-sapphire-brand-toggle-background hover:border-sapphire-border-toggle"
      },
      // TertiaryNeutral
      {
        design: "TertiaryNeutral" /* TertiaryNeutral */,
        pressed: false,
        className: "bg-transparent text-sapphire-neutral-foreground-muted border-transparent hover:text-sapphire-neutral-foreground-black active:text-sapphire-neutral-foreground-black"
      },
      {
        design: "TertiaryNeutral" /* TertiaryNeutral */,
        pressed: true,
        className: "bg-sapphire-brand-selected-background text-sapphire-brand-foreground border-sapphire-border-toggle hover:bg-sapphire-brand-toggle-background hover:border-sapphire-border-toggle"
      },
      // Tertiary
      {
        design: "Tertiary" /* Tertiary */,
        pressed: false,
        className: "bg-transparent text-sapphire-brand-foreground hover:bg-sapphire-neutral-hover-background active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground"
      },
      {
        design: "Tertiary" /* Tertiary */,
        pressed: true,
        className: "bg-sapphire-brand-selected-background text-sapphire-brand-foreground border-transparent hover:bg-sapphire-brand-toggle-background hover:border-sapphire-border-toggle"
      },
      // TertiaryJoule
      {
        design: "TertiaryJoule" /* TertiaryJoule */,
        pressed: false,
        className: "bg-transparent text-sapphire-joule-foreground hover:bg-sapphire-neutral-hover-background active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground"
      },
      {
        design: "TertiaryJoule" /* TertiaryJoule */,
        pressed: true,
        className: "bg-sapphire-joule-background-area text-sapphire-joule-foreground border-transparent hover:bg-sapphire-joule-background-light hover:border-sapphire-border-toggle"
      }
    ],
    defaultVariants: {
      design: "Secondary" /* Secondary */,
      pressed: false
    }
  }
);
var sizeClasses = {
  ["Large" /* Large */]: {
    default: "h-10 px-4 py-2 rounded-lg",
    iconOnly: "h-10 w-10 p-0 rounded-lg",
    icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
  },
  ["Medium" /* Medium */]: {
    default: "h-8 px-3 py-2 text-sm rounded",
    iconOnly: "h-8 w-8 p-0 text-sm rounded",
    icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
  },
  ["Small" /* Small */]: {
    default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
    iconOnly: "h-6 w-6 p-0 text-xs rounded",
    icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
  }
};
function ToggleButton({
  pressed: controlledPressed,
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  icon,
  endIcon,
  children,
  disabled = false,
  iconOnly = false,
  tooltip,
  accessibleName,
  onChange,
  onClick,
  tabIndex,
  className,
  style,
  ref
}) {
  const buttonRef = useRef(null);
  const [internalPressed, setInternalPressed] = useState(false);
  const isControlled = controlledPressed !== void 0;
  const pressed = isControlled ? controlledPressed : internalPressed;
  useImperativeHandle(ref, () => ({
    focus: () => buttonRef.current?.focus(),
    isFocused: () => document.activeElement === buttonRef.current,
    get nativeElement() {
      return buttonRef.current;
    },
    getNativeElement: () => buttonRef.current
  }));
  const handleClick = (e) => {
    if (disabled) return;
    const newPressed = !pressed;
    if (!isControlled) {
      setInternalPressed(newPressed);
    }
    onChange?.({ pressed: newPressed });
    onClick?.(e);
  };
  const isIconOnly = iconOnly || icon && !children;
  return /* @__PURE__ */ jsxs(
    "button",
    {
      ref: buttonRef,
      type: "button",
      disabled,
      tabIndex,
      title: tooltip,
      "aria-label": accessibleName,
      "aria-pressed": pressed,
      onClick: handleClick,
      className: cn(
        toggleButtonVariants({
          design,
          pressed
        }),
        isIconOnly ? sizeClasses[size].iconOnly : sizeClasses[size].default,
        className
      ),
      style,
      children: [
        icon && /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon), children: icon }),
        !isIconOnly && children,
        endIcon && /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon), children: endIcon })
      ]
    }
  );
}

export { ToggleButton };
