// @sap-ui/fx-components/dist/chunk-CPOUYZCH.js
// src/types/button.ts
var ButtonDesign = /* @__PURE__ */ ((ButtonDesign2) => {
  ButtonDesign2["Primary"] = "Primary";
  ButtonDesign2["PrimaryJoule"] = "PrimaryJoule";
  ButtonDesign2["Secondary"] = "Secondary";
  ButtonDesign2["SecondaryJoule"] = "SecondaryJoule";
  ButtonDesign2["Tertiary"] = "Tertiary";
  ButtonDesign2["TertiaryJoule"] = "TertiaryJoule";
  ButtonDesign2["Neutral"] = "Neutral";
  ButtonDesign2["SecondaryNeutral"] = "SecondaryNeutral";
  ButtonDesign2["TertiaryNeutral"] = "TertiaryNeutral";
  return ButtonDesign2;
})(ButtonDesign || {});
var ButtonSize = /* @__PURE__ */ ((ButtonSize2) => {
  ButtonSize2["Large"] = "Large";
  ButtonSize2["Medium"] = "Medium";
  ButtonSize2["Small"] = "Small";
  return ButtonSize2;
})(ButtonSize || {});
var ButtonType = /* @__PURE__ */ ((ButtonType2) => {
  ButtonType2["Button"] = "Button";
  ButtonType2["Submit"] = "Submit";
  ButtonType2["Reset"] = "Reset";
  return ButtonType2;
})(ButtonType || {});
var ButtonAccessibleRole = /* @__PURE__ */ ((ButtonAccessibleRole2) => {
  ButtonAccessibleRole2["Button"] = "Button";
  ButtonAccessibleRole2["Link"] = "Link";
  return ButtonAccessibleRole2;
})(ButtonAccessibleRole || {});

export { ButtonAccessibleRole, ButtonDesign, ButtonSize, ButtonType };
