// @sap-ui/fx-components/dist/chunk-NCI4MJ5T.js
import { SysHelp2Icon } from './chunk-4Y6CARK4.js';
import { AlertIcon } from './chunk-4IU3KSGT.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { InformationIcon, SysEnter2Icon } from './chunk-ZK4METPA.js';
import { ErrorIcon } from './chunk-FL4UPYZT.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useImperativeHandle, useCallback } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/tag-exploration.ts
var TagExplorationDesign = /* @__PURE__ */ ((TagExplorationDesign2) => {
  TagExplorationDesign2["Neutral"] = "Neutral";
  TagExplorationDesign2["Information"] = "Information";
  TagExplorationDesign2["Positive"] = "Positive";
  TagExplorationDesign2["Negative"] = "Negative";
  TagExplorationDesign2["Critical"] = "Critical";
  TagExplorationDesign2["Set1"] = "Set1";
  TagExplorationDesign2["Set2"] = "Set2";
  return TagExplorationDesign2;
})(TagExplorationDesign || {});
var TagExplorationSize = /* @__PURE__ */ ((TagExplorationSize2) => {
  TagExplorationSize2["S"] = "S";
  TagExplorationSize2["L"] = "L";
  return TagExplorationSize2;
})(TagExplorationSize || {});
var TagExplorationWrappingType = /* @__PURE__ */ ((TagExplorationWrappingType2) => {
  TagExplorationWrappingType2["None"] = "None";
  TagExplorationWrappingType2["Normal"] = "Normal";
  return TagExplorationWrappingType2;
})(TagExplorationWrappingType || {});
var TagExplorationColorScheme = /* @__PURE__ */ ((TagExplorationColorScheme2) => {
  TagExplorationColorScheme2["CS1"] = "1";
  TagExplorationColorScheme2["CS2"] = "2";
  TagExplorationColorScheme2["CS3"] = "3";
  TagExplorationColorScheme2["CS4"] = "4";
  TagExplorationColorScheme2["CS5"] = "5";
  TagExplorationColorScheme2["CS6"] = "6";
  TagExplorationColorScheme2["CS7"] = "7";
  TagExplorationColorScheme2["CS8"] = "8";
  TagExplorationColorScheme2["CS9"] = "9";
  TagExplorationColorScheme2["CS10"] = "10";
  return TagExplorationColorScheme2;
})(TagExplorationColorScheme || {});
var semanticDesignClass = {
  ["Neutral" /* Neutral */]: "bg-background text-sapphire-text-accent-2 border-sapphire-tag-border",
  ["Information" /* Information */]: "bg-sapphire-info-bg text-sapphire-info border-transparent",
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg text-sapphire-positive border-transparent",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg text-sapphire-negative border-transparent",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg text-sapphire-warning border-transparent"
};
var stateIconMap = {
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Critical" /* Critical */]: AlertIcon,
  ["Information" /* Information */]: InformationIcon,
  ["Neutral" /* Neutral */]: SysHelp2Icon
};
var set1Classes = {
  ["1" /* CS1 */]: "bg-purple-600 text-sapphire-neutral-foreground-white border-transparent",
  ["2" /* CS2 */]: "bg-blue-600 text-sapphire-neutral-foreground-white border-transparent",
  ["3" /* CS3 */]: "bg-neutral-600 text-sapphire-neutral-foreground-white border-transparent",
  ["4" /* CS4 */]: "bg-purple-700 text-sapphire-neutral-foreground-white border-transparent",
  ["5" /* CS5 */]: "bg-blue-700 text-sapphire-neutral-foreground-white border-transparent",
  ["6" /* CS6 */]: "bg-neutral-700 text-sapphire-neutral-foreground-white border-transparent",
  ["7" /* CS7 */]: "bg-purple-800 text-sapphire-neutral-foreground-white border-transparent",
  ["8" /* CS8 */]: "bg-blue-800 text-sapphire-neutral-foreground-white border-transparent",
  ["9" /* CS9 */]: "bg-neutral-800 text-sapphire-neutral-foreground-white border-transparent",
  ["10" /* CS10 */]: "bg-purple-500 text-sapphire-neutral-foreground-white border-transparent"
};
var set2Classes = {
  ["1" /* CS1 */]: "bg-purple-100 text-purple-700 border-transparent",
  ["2" /* CS2 */]: "bg-blue-100 text-blue-700 border-transparent",
  ["3" /* CS3 */]: "bg-neutral-100 text-neutral-700 border-transparent",
  ["4" /* CS4 */]: "bg-purple-200 text-purple-800 border-transparent",
  ["5" /* CS5 */]: "bg-blue-200 text-blue-800 border-transparent",
  ["6" /* CS6 */]: "bg-neutral-200 text-neutral-800 border-transparent",
  ["7" /* CS7 */]: "bg-purple-50 text-purple-600 border-transparent",
  ["8" /* CS8 */]: "bg-blue-50 text-blue-600 border-transparent",
  ["9" /* CS9 */]: "bg-neutral-50 text-neutral-600 border-transparent",
  ["10" /* CS10 */]: "bg-purple-100 text-purple-600 border-transparent"
};
function getDesignClasses(design, colorScheme, accent) {
  if (design === "Set1" /* Set1 */) {
    return set1Classes[colorScheme] ?? set1Classes["1" /* CS1 */];
  }
  if (design === "Set2" /* Set2 */) {
    return set2Classes[colorScheme] ?? set2Classes["1" /* CS1 */];
  }
  if (design === "Neutral" /* Neutral */ && accent) {
    return "bg-sapphire-tag-accent-bg text-sapphire-text-accent-2 border-sapphire-tag-accent-bg";
  }
  return semanticDesignClass[design] ?? semanticDesignClass["Neutral" /* Neutral */];
}
function TagExploration({
  design = "Neutral" /* Neutral */,
  size = "S" /* S */,
  colorScheme = "1" /* CS1 */,
  children,
  icon,
  interactive = false,
  readOnly = false,
  accent = false,
  wrappingType = "None" /* None */,
  hideStateIcon = false,
  onClose,
  onClick,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const tagRef = useRef(null);
  const d = design;
  const s = size;
  const cs = colorScheme;
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        tagRef.current?.focus();
      },
      blur() {
        tagRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === tagRef.current;
      },
      get nativeElement() {
        return tagRef.current;
      }
    }),
    [tagRef]
  );
  const handleClick = useCallback(
    (e) => {
      if (!interactive) return;
      onClick?.({ originalEvent: e, isKeyboard: false });
    },
    [interactive, onClick]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (!interactive) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({ originalEvent: e, isKeyboard: true });
      }
    },
    [interactive, onClick]
  );
  const handleCloseClick = useCallback(
    (e) => {
      e.stopPropagation();
      onClose?.({ originalEvent: e, isKeyboard: false });
    },
    [onClose]
  );
  const handleCloseKeyDown = useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        e.stopPropagation();
        onClose?.({ originalEvent: e, isKeyboard: true });
      }
    },
    [onClose]
  );
  const isSemanticDesign = d !== "Set1" /* Set1 */ && d !== "Set2" /* Set2 */;
  let displayIcon = null;
  if (icon) {
    displayIcon = icon;
  } else if (isSemanticDesign && !hideStateIcon) {
    const StateIcon = stateIconMap[d];
    if (StateIcon) {
      const iconSizeClass = s === "L" /* L */ ? "h-4 w-4" : "h-3 w-3";
      displayIcon = /* @__PURE__ */ jsx(StateIcon, { className: iconSizeClass });
    }
  }
  const sizeClass = s === "L" /* L */ ? "text-sm px-[9px] py-0.5 rounded-full gap-1" : "text-xs px-2 py-1 rounded-full gap-1";
  const closeBtnSizeClass = s === "L" /* L */ ? "h-4 w-4" : "h-3 w-3";
  const wrappingClasses = wrappingType === "None" /* None */ ? "max-w-full" : "";
  const textWrappingClasses = wrappingType === "None" /* None */ ? "overflow-hidden text-ellipsis whitespace-nowrap" : "whitespace-normal break-words";
  const focusPaddingClass = s === "L" /* L */ ? "focus-visible:px-[8px] focus-visible:py-px" : "focus-visible:px-[7px] focus-visible:py-[3px]";
  const interactiveClasses = interactive && !readOnly ? [
    "cursor-pointer",
    "hover:bg-sapphire-tag-hover-bg hover:border-transparent",
    "active:bg-sapphire-tag-active-bg active:border-transparent active:rounded-[20px]",
    `focus-visible:outline-none focus-visible:bg-sapphire-tag-hover-bg focus-visible:border-2 focus-visible:border-sapphire-tag-focus-border ${focusPaddingClass}`
  ].join(" ") : "";
  return /* @__PURE__ */ jsxs(
    "span",
    {
      ref: tagRef,
      id,
      role: interactive && !readOnly ? "button" : void 0,
      tabIndex: interactive && !readOnly ? 0 : void 0,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        "inline-flex items-center border transition-colors font-normal leading-5 tracking-[0.25px]",
        sizeClass,
        getDesignClasses(d, cs, accent),
        wrappingClasses,
        interactiveClasses,
        readOnly && "opacity-40 pointer-events-none",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        displayIcon && /* @__PURE__ */ jsx("span", { className: "flex-shrink-0", "data-part": "icon", children: displayIcon }),
        children && /* @__PURE__ */ jsx("span", { "data-part": "text", className: textWrappingClasses, children }),
        onClose && !readOnly && /* @__PURE__ */ jsx(
          "button",
          {
            type: "button",
            onClick: handleCloseClick,
            onKeyDown: handleCloseKeyDown,
            "aria-label": "Remove",
            "data-part": "close",
            className: cn(
              "flex-shrink-0 inline-flex items-center justify-center rounded-full",
              "hover:opacity-70 focus:outline-none focus:ring-1 focus:ring-ring",
              "cursor-pointer"
            ),
            tabIndex: 0,
            children: /* @__PURE__ */ jsx(DeclineIcon, { className: closeBtnSizeClass })
          }
        )
      ]
    }
  );
}

export { TagExploration, TagExplorationColorScheme, TagExplorationDesign, TagExplorationSize, TagExplorationWrappingType };
