// @sap-ui/fx-components/dist/chunk-TNSICZD6.js
import { TriStateIcon, CompleteIcon, BorderIcon } from './chunk-56AYC3ZC.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useId, useState, useCallback, useImperativeHandle } from 'react';
import { Check } from 'lucide-react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/checkbox.ts
var WrappingType = /* @__PURE__ */ ((WrappingType2) => {
  WrappingType2["None"] = "None";
  WrappingType2["Normal"] = "Normal";
  return WrappingType2;
})(WrappingType || {});
var CheckBoxSize = /* @__PURE__ */ ((CheckBoxSize2) => {
  CheckBoxSize2["Large"] = "Large";
  CheckBoxSize2["Small"] = "Small";
  return CheckBoxSize2;
})(CheckBoxSize || {});
function CheckBox({
  checked: controlledChecked,
  defaultChecked = false,
  indeterminate = false,
  disabled = false,
  readonly = false,
  displayOnly = false,
  required = false,
  size = "Large" /* Large */,
  text,
  children,
  wrappingType = "Normal" /* Normal */,
  valueState = "None",
  valueStateMessage,
  name,
  value = "on",
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  onChange,
  onFocus,
  onBlur,
  className,
  style,
  id: providedId,
  "data-testid": dataTestId,
  tabIndex,
  accessibilityAttributes,
  ref
}) {
  const inputRef = useRef(null);
  const rootRef = useRef(null);
  const generatedId = useId();
  const id = providedId || generatedId;
  const descriptionId = `${id}-description`;
  const [internalChecked, setInternalChecked] = useState(defaultChecked);
  const isControlled = controlledChecked !== void 0;
  const isChecked = isControlled ? controlledChecked : internalChecked;
  const isActive = isChecked || indeterminate;
  const doToggle = useCallback(
    (originalEvent) => {
      if (disabled || readonly || displayOnly) return;
      const newChecked = !isChecked;
      if (!isControlled) {
        setInternalChecked(newChecked);
      }
      onChange?.({
        checked: newChecked,
        indeterminate: false,
        originalEvent
      });
    },
    [disabled, readonly, displayOnly, isChecked, isControlled, onChange]
  );
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      isChecked() {
        return isChecked;
      },
      toggle() {
        doToggle();
      },
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [isChecked, doToggle]
  );
  const handleClick = useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) {
        e.preventDefault();
        return;
      }
      doToggle(e);
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const handleMouseDown = useCallback(
    (e) => {
      if (disabled || displayOnly) {
        e.preventDefault();
      }
    },
    [disabled, displayOnly]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
      }
    },
    [disabled, readonly, displayOnly]
  );
  const handleKeyUp = useCallback(
    (e) => {
      if (disabled || readonly || displayOnly) return;
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        doToggle(e);
      }
    },
    [disabled, readonly, displayOnly, doToggle]
  );
  const labelContent = children || text;
  const isSmall = size === "Small" /* Small */;
  const isInteractive = !disabled && !readonly && !displayOnly;
  const hasWrappingLabel = wrappingType !== "None" /* None */ && !!labelContent;
  const getValueStateBorderClass = () => {
    if (isActive) return "";
    const borderMap = {
      Positive: "border-sapphire-positive",
      Negative: "border-sapphire-negative",
      Critical: "border-sapphire-warning",
      Information: "border-sapphire-info"
    };
    return borderMap[valueState] ?? "";
  };
  const getValueStateCheckedClass = () => {
    if (!isActive) return "";
    const checkedMap = {
      Positive: "bg-sapphire-positive border-sapphire-positive",
      Negative: "bg-sapphire-negative border-sapphire-negative",
      Critical: "bg-sapphire-warning border-sapphire-warning",
      Information: "bg-sapphire-info border-sapphire-info"
    };
    return checkedMap[valueState] ?? "bg-sapphire-canvas-primary border-sapphire-border-secondary";
  };
  const renderDisplayOnlyIcon = () => {
    const cls = "size-4 text-foreground";
    if (indeterminate) return /* @__PURE__ */ jsx(TriStateIcon, { className: cls });
    if (isChecked) return /* @__PURE__ */ jsx(CompleteIcon, { className: cls });
    return /* @__PURE__ */ jsx(BorderIcon, { className: cls });
  };
  return /* @__PURE__ */ jsxs("div", { className: "relative inline-block max-w-full align-top", children: [
    /* @__PURE__ */ jsx(
      "input",
      {
        ref: inputRef,
        type: "checkbox",
        name,
        value,
        checked: isChecked,
        disabled: disabled || displayOnly,
        readOnly: readonly,
        required,
        tabIndex: -1,
        "aria-hidden": "true",
        onChange: () => {
        },
        className: "absolute size-0 opacity-0 pointer-events-none"
      }
    ),
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: rootRef,
        id,
        role: "checkbox",
        "aria-checked": indeterminate ? "mixed" : isChecked,
        "aria-label": accessibleName,
        "aria-labelledby": !accessibleName ? accessibleNameRef || (labelContent ? `${id}-label` : void 0) : void 0,
        "aria-describedby": [
          valueState !== "None" && valueStateMessage ? descriptionId : null,
          accessibleDescription ? `${id}-desc-text` : null,
          accessibilityAttributes?.describedBy
        ].filter(Boolean).join(" ") || void 0,
        "aria-disabled": disabled || displayOnly || void 0,
        "aria-readonly": readonly || void 0,
        "aria-invalid": valueState === "Negative" || void 0,
        "aria-required": required || void 0,
        tabIndex: tabIndex !== void 0 ? tabIndex : disabled || displayOnly ? -1 : 0,
        onMouseDown: handleMouseDown,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur,
        className: cn(
          "group relative inline-flex select-none max-w-full",
          "outline-none",
          labelContent ? "items-center" : "items-center justify-center",
          // Touch area: Cozy = 44px, Compact = 32px
          isSmall ? labelContent ? "min-h-8 py-1.5 ps-1 pe-1 gap-2" : "size-8 p-2" : labelContent ? "min-h-[44px] py-[11px] ps-1 pe-2 gap-[11px]" : "size-[44px] p-[11px]",
          "text-sm",
          "rounded-lg",
          // Focus ring via pseudo-element for asymmetric insets
          "outline-none",
          "after:content-[''] after:absolute after:pointer-events-none after:border-2 after:border-transparent after:rounded-md",
          labelContent ? isSmall ? "after:inset-x-0 after:inset-y-0" : "after:left-0 after:right-[2px] after:inset-y-[5px]" : "after:inset-[5px] after:rounded-lg",
          "focus:after:border-[var(--color-ring)]",
          disabled && "opacity-40 cursor-not-allowed",
          readonly && "cursor-default",
          displayOnly && "cursor-default",
          isInteractive && "cursor-pointer",
          className
        ),
        style,
        "data-testid": dataTestId,
        children: [
          displayOnly ? renderDisplayOnlyIcon() : /* @__PURE__ */ jsx(
            "span",
            {
              className: cn(
                "relative inline-flex items-center justify-center",
                "shrink-0",
                hasWrappingLabel && "self-start",
                isSmall ? hasWrappingLabel ? "size-4 rounded p-[2px] mt-0.5" : "size-4 rounded p-[2px]" : hasWrappingLabel ? "size-[22px] rounded-md p-[3px] mt-0.5" : "size-[22px] rounded-md p-[3px]",
                "transition-colors",
                // Border style: dashed for readonly, solid otherwise
                readonly ? "border border-dashed" : "border border-solid",
                // --- Unchecked state ---
                !isActive && !readonly && "bg-sapphire-canvas-primary border-sapphire-border-secondary",
                // --- Checked / indeterminate state (not readonly) ---
                // None value state: blue icon on white bg; value states: white icon on colored bg
                isActive && !readonly && valueState === "None" && "text-sapphire-shell-button-fg-selected",
                isActive && !readonly && valueState !== "None" && "text-sapphire-neutral-foreground-white",
                !readonly && getValueStateCheckedClass(),
                !readonly && getValueStateBorderClass(),
                // --- Hover (unchecked, not readonly/disabled, no value state) ---
                !isActive && isInteractive && valueState === "None" && "group-hover:border-primary group-hover:bg-sapphire-shell-button-bg-selected",
                // --- Hover (checked, not readonly/disabled, no value state) ---
                isActive && isInteractive && valueState === "None" && "group-hover:bg-sapphire-shell-button-bg-selected group-hover:border-primary",
                // --- Read-only state: gray bg + dashed secondary border ---
                readonly && "bg-sapphire-background-secondary border-sapphire-border-secondary",
                // --- Read-only icon color: secondary-foreground (#353C4A) ---
                readonly && isActive && "text-secondary-foreground"
              ),
              children: indeterminate ? (
                // Solid square for tristate per Figma: 12px large, 8px small
                /* @__PURE__ */ jsx("span", { className: cn(
                  "block bg-current",
                  isSmall ? "size-2" : "size-3"
                ) })
              ) : isChecked ? /* @__PURE__ */ jsx(Check, { className: isSmall ? "size-3" : "size-4", strokeWidth: 3 }) : null
            }
          ),
          labelContent && /* @__PURE__ */ jsx(
            "span",
            {
              id: `${id}-label`,
              className: cn(
                "text-sm min-w-0 flex-1 text-foreground",
                wrappingType === "None" /* None */ && "truncate",
                wrappingType === "Normal" /* Normal */ && "break-words",
                required && "after:content-['*'] after:ml-0.5 after:text-sapphire-negative"
              ),
              children: labelContent
            }
          ),
          valueState !== "None" && valueStateMessage && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: valueStateMessage }),
          accessibleDescription && /* @__PURE__ */ jsx("span", { id: `${id}-desc-text`, className: "sr-only", children: accessibleDescription })
        ]
      }
    )
  ] });
}

export { CheckBox, CheckBoxSize, WrappingType };
