// @sap-ui/fx-components/dist/chunk-KIJTYAE6.js
import { createIllustration } from './chunk-UHU7LLWX.js';

// src/illustrations/fiori/Achievement/Achievement.tsx
var Achievement = createIllustration({
  name: "Achievement",
  title: "Great job!",
  subtitle: "Congratulations on your outstanding performance and dedication!",
  lazyExtraSmall: () => import('./ExtraSmall-CSMF2NJU.js'),
  lazySmall: () => import('./Small-MKTJTH36.js'),
  lazyMedium: () => import('./Medium-TKHUW6RC.js'),
  lazyLarge: () => import('./Large-CQHSBFF4.js')
});

// src/illustrations/fiori/AddColumn/AddColumn.tsx
var AddColumn = createIllustration({
  name: "AddColumn",
  title: "Looks like there's free space",
  subtitle: "You can add more columns in the table settings.",
  lazyExtraSmall: () => import('./ExtraSmall-IST2SGKF.js'),
  lazySmall: () => import('./Small-MU3CZTNB.js'),
  lazyMedium: () => import('./Medium-CFCD7K7L.js'),
  lazyLarge: () => import('./Large-22I52POS.js')
});

// src/illustrations/fiori/AddDimensions/AddDimensions.tsx
var AddDimensions = createIllustration({
  name: "AddDimensions",
  title: "Some dimensions are missing",
  subtitle: "Add more dimensions to complete your chart.",
  lazyExtraSmall: () => import('./ExtraSmall-AMGMOABJ.js'),
  lazySmall: () => import('./Small-XSDGXTR4.js'),
  lazyMedium: () => import('./Medium-LURKZNYE.js'),
  lazyLarge: () => import('./Large-YBUCKLED.js')
});

// src/illustrations/fiori/AddPeople/AddPeople.tsx
var AddPeople = createIllustration({
  name: "AddPeople",
  title: "You've not added anyone to the calendar yet",
  subtitle: "Do you want to add someone now?",
  lazyExtraSmall: () => import('./ExtraSmall-GA3THNJP.js'),
  lazySmall: () => import('./Small-W7XFRLB7.js'),
  lazyMedium: () => import('./Medium-75OYHW6D.js'),
  lazyLarge: () => import('./Large-PXATATZN.js')
});

// src/illustrations/fiori/AddPeopleToCalendar/AddPeopleToCalendar.tsx
var AddPeopleToCalendar = createIllustration({
  name: "AddPeopleToCalendar",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2GPVXEOV.js'),
  lazySmall: () => import('./Small-CL4ORA4R.js'),
  lazyMedium: () => import('./Medium-REY3ZJCM.js'),
  lazyLarge: () => import('./Large-4T6WBSTM.js')
});

// src/illustrations/fiori/AddingColumns/AddingColumns.tsx
var AddingColumns = createIllustration({
  name: "AddingColumns",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-OM42UQAE.js'),
  lazySmall: () => import('./Small-GOPVU5GI.js'),
  lazyMedium: () => import('./Medium-6PKCFCWR.js'),
  lazyLarge: () => import('./Large-IDIEQAJI.js')
});

// src/illustrations/fiori/BalloonSky/BalloonSky.tsx
var BalloonSky = createIllustration({
  name: "BalloonSky",
  title: "You've been appreciated!",
  subtitle: "Keep up the great work!",
  lazyExtraSmall: () => import('./ExtraSmall-4KPSWQGF.js'),
  lazySmall: () => import('./Small-D7HW3C6N.js'),
  lazyMedium: () => import('./Medium-ZODSKJGN.js'),
  lazyLarge: () => import('./Large-YHLFBLNT.js')
});

// src/illustrations/fiori/BeforeSearch/BeforeSearch.tsx
var BeforeSearch = createIllustration({
  name: "BeforeSearch",
  title: "Let's get some results",
  subtitle: "Start by providing your search criteria.",
  lazyExtraSmall: () => import('./ExtraSmall-5D5EDA7G.js'),
  lazySmall: () => import('./Small-YXRKIC5C.js'),
  lazyMedium: () => import('./Medium-QDVWZYSZ.js'),
  lazyLarge: () => import('./Large-LTCRUCFF.js')
});

// src/illustrations/fiori/Connection/Connection.tsx
var Connection = createIllustration({
  name: "Connection",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-KCOAHPY6.js'),
  lazySmall: () => import('./Small-DQZ5EIBD.js'),
  lazyMedium: () => import('./Medium-47IO7KQ4.js'),
  lazyLarge: () => import('./Large-COCBMUFH.js')
});

// src/illustrations/fiori/DragFilesToUpload/DragFilesToUpload.tsx
var DragFilesToUpload = createIllustration({
  name: "DragFilesToUpload",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-SC6XCPJR.js'),
  lazySmall: () => import('./Small-KBBFHKKG.js'),
  lazyMedium: () => import('./Medium-VLWAZ7N2.js'),
  lazyLarge: () => import('./Large-ZR7R6NUK.js')
});

// src/illustrations/fiori/EmptyCalendar/EmptyCalendar.tsx
var EmptyCalendar = createIllustration({
  name: "EmptyCalendar",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2EZNS373.js'),
  lazySmall: () => import('./Small-NRTSKCYG.js'),
  lazyMedium: () => import('./Medium-XNSFGAHM.js'),
  lazyLarge: () => import('./Large-E5SAQVEA.js')
});

// src/illustrations/fiori/EmptyList/EmptyList.tsx
var EmptyList = createIllustration({
  name: "EmptyList",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-565BEIYZ.js'),
  lazySmall: () => import('./Small-6B5HF4R5.js'),
  lazyMedium: () => import('./Medium-ZYAUPOZ7.js'),
  lazyLarge: () => import('./Large-NBJYNPOD.js')
});

// src/illustrations/fiori/EmptyPlanningCalendar/EmptyPlanningCalendar.tsx
var EmptyPlanningCalendar = createIllustration({
  name: "EmptyPlanningCalendar",
  title: "Nothing planned yet",
  subtitle: "There are no activities in this time frame.",
  lazyExtraSmall: () => import('./ExtraSmall-JHQ2RJPM.js'),
  lazySmall: () => import('./Small-EJVYFKZ3.js'),
  lazyMedium: () => import('./Medium-MXNSOSCS.js'),
  lazyLarge: () => import('./Large-RT3KXDEY.js')
});

// src/illustrations/fiori/ErrorScreen/ErrorScreen.tsx
var ErrorScreen = createIllustration({
  name: "ErrorScreen",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-ZZ7IX3NK.js'),
  lazySmall: () => import('./Small-XZ6T7MGY.js'),
  lazyMedium: () => import('./Medium-6N5KOWAA.js'),
  lazyLarge: () => import('./Large-7AVXOJUL.js')
});

// src/illustrations/fiori/FilterTable/FilterTable.tsx
var FilterTable = createIllustration({
  name: "FilterTable",
  title: "Filter options are available",
  subtitle: "Filters help you focus on what's most relevant for you.",
  lazyExtraSmall: () => import('./ExtraSmall-YBSOJ3JH.js'),
  lazySmall: () => import('./Small-7B5VLSDJ.js'),
  lazyMedium: () => import('./Medium-3PTN4MGZ.js'),
  lazyLarge: () => import('./Large-7ZYGGSWA.js')
});

// src/illustrations/fiori/FilteringColumns/FilteringColumns.tsx
var FilteringColumns = createIllustration({
  name: "FilteringColumns",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-IQAYAT2M.js'),
  lazySmall: () => import('./Small-XKMGGUJY.js'),
  lazyMedium: () => import('./Medium-GCIQU7EZ.js'),
  lazyLarge: () => import('./Large-BSTBE4LR.js')
});

// src/illustrations/fiori/GroupTable/GroupTable.tsx
var GroupTable = createIllustration({
  name: "GroupTable",
  title: "Try grouping items for a better overview",
  subtitle: "You can choose grouping categories in the group settings.",
  lazyExtraSmall: () => import('./ExtraSmall-B5Z76N5U.js'),
  lazySmall: () => import('./Small-BPXHL44P.js'),
  lazyMedium: () => import('./Medium-APVOVZWS.js'),
  lazyLarge: () => import('./Large-TYU7SRPG.js')
});

// src/illustrations/fiori/GroupingColumns/GroupingColumns.tsx
var GroupingColumns = createIllustration({
  name: "GroupingColumns",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-U2HNBYEV.js'),
  lazySmall: () => import('./Small-OGTLPIYA.js'),
  lazyMedium: () => import('./Medium-QG2OTKFS.js'),
  lazyLarge: () => import('./Large-SLJDRHQT.js')
});

// src/illustrations/fiori/KeyTask/KeyTask.tsx
var KeyTask = createIllustration({
  name: "KeyTask",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-UIG7CCZM.js'),
  lazySmall: () => import('./Small-O7HAL5E4.js'),
  lazyMedium: () => import('./Medium-JM37AF3H.js'),
  lazyLarge: () => import('./Large-DJBE3J6X.js')
});

// src/illustrations/fiori/NewMail/NewMail.tsx
var NewMail = createIllustration({
  name: "NewMail",
  title: "New mail",
  subtitle: "You have new mail in your inbox.",
  lazyExtraSmall: () => import('./ExtraSmall-EWPU6YNJ.js'),
  lazySmall: () => import('./Small-RZBJKWOB.js'),
  lazyMedium: () => import('./Medium-NIEXD3VI.js'),
  lazyLarge: () => import('./Large-DNYQZWOT.js')
});

// src/illustrations/fiori/NoActivities/NoActivities.tsx
var NoActivities = createIllustration({
  name: "NoActivities",
  title: "You've not added any activities yet",
  subtitle: "Would you like to add one now?",
  lazyExtraSmall: () => import('./ExtraSmall-XQTZUSSW.js'),
  lazySmall: () => import('./Small-GMBPODKS.js'),
  lazyMedium: () => import('./Medium-HXPLPAJ2.js'),
  lazyLarge: () => import('./Large-R4BI2Y4O.js')
});

// src/illustrations/fiori/NoChartData/NoChartData.tsx
var NoChartData = createIllustration({
  name: "NoChartData",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-ODTAZUGW.js'),
  lazySmall: () => import('./Small-QSCK6LKG.js'),
  lazyMedium: () => import('./Medium-74ZQO6NN.js'),
  lazyLarge: () => import('./Large-SGJDZMV3.js')
});

// src/illustrations/fiori/NoColumnsSet/NoColumnsSet.tsx
var NoColumnsSet = createIllustration({
  name: "NoColumnsSet",
  title: "Add columns to see the content",
  subtitle: "Select the columns you need in the table settings.",
  lazyExtraSmall: () => import('./ExtraSmall-E5LMT3S3.js'),
  lazySmall: () => import('./Small-BAYKBUH4.js'),
  lazyMedium: () => import('./Medium-WIZRN7HA.js'),
  lazyLarge: () => import('./Large-QVB7HW3T.js')
});

// src/illustrations/fiori/NoData/NoData.tsx
var NoData = createIllustration({
  name: "NoData",
  title: "There's no data yet.",
  subtitle: "When there is, you'll see it here.",
  lazyExtraSmall: () => import('./ExtraSmall-QAIEMLEY.js'),
  lazySmall: () => import('./Small-MHY3COQL.js'),
  lazyMedium: () => import('./Medium-AICIR6SF.js'),
  lazyLarge: () => import('./Large-OAOSDLMJ.js')
});

// src/illustrations/fiori/NoDimensionsSet/NoDimensionsSet.tsx
var NoDimensionsSet = createIllustration({
  name: "NoDimensionsSet",
  title: "No chart data",
  subtitle: "Try adjusting your chart settings.",
  lazyExtraSmall: () => import('./ExtraSmall-NQ2XY72E.js'),
  lazySmall: () => import('./Small-IESF653F.js'),
  lazyMedium: () => import('./Medium-E7OB6SOR.js'),
  lazyLarge: () => import('./Large-RXP5OTUT.js')
});

// src/illustrations/fiori/NoEntries/NoEntries.tsx
var NoEntries = createIllustration({
  name: "NoEntries",
  title: "There are no entries yet",
  subtitle: "When there are, you'll see them here.",
  lazyExtraSmall: () => import('./ExtraSmall-QGOPNUIF.js'),
  lazySmall: () => import('./Small-XCOUHZZF.js'),
  lazyMedium: () => import('./Medium-TJFXAOI7.js'),
  lazyLarge: () => import('./Large-K3EYAWGW.js')
});

// src/illustrations/fiori/NoFilterResults/NoFilterResults.tsx
var NoFilterResults = createIllustration({
  name: "NoFilterResults",
  title: "No results found",
  subtitle: "Try adjusting your filter criteria.",
  lazyExtraSmall: () => import('./ExtraSmall-52XM657H.js'),
  lazySmall: () => import('./Small-QESDBUL4.js'),
  lazyMedium: () => import('./Medium-U52XWPQH.js'),
  lazyLarge: () => import('./Large-5Z2TOSCE.js')
});

// src/illustrations/fiori/NoMail/NoMail.tsx
var NoMail = createIllustration({
  name: "NoMail",
  title: "No mail",
  subtitle: "Check back again later.",
  lazyExtraSmall: () => import('./ExtraSmall-D3EHMOP3.js'),
  lazySmall: () => import('./Small-7Y4VGD53.js'),
  lazyMedium: () => import('./Medium-TEUX4AHZ.js'),
  lazyLarge: () => import('./Large-OUGUEX2I.js')
});

// src/illustrations/fiori/NoMail_v1/NoMail_v1.tsx
var NoMail_v1 = createIllustration({
  name: "NoMail_v1",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-33DII4T3.js'),
  lazySmall: () => import('./Small-FBZOTCGA.js'),
  lazyMedium: () => import('./Medium-BB7D63K5.js'),
  lazyLarge: () => import('./Large-Y6XPWMYA.js')
});

// src/illustrations/fiori/NoNotifications/NoNotifications.tsx
var NoNotifications = createIllustration({
  name: "NoNotifications",
  title: "You've no notifications",
  subtitle: "Check back again later.",
  lazyExtraSmall: () => import('./ExtraSmall-2Z7OMQGJ.js'),
  lazySmall: () => import('./Small-GJF6DPH2.js'),
  lazyMedium: () => import('./Medium-3LKSNLDY.js'),
  lazyLarge: () => import('./Large-Z42XHLEQ.js')
});

// src/illustrations/fiori/NoSavedItems/NoSavedItems.tsx
var NoSavedItems = createIllustration({
  name: "NoSavedItems",
  title: "You've no pins",
  subtitle: "Would you like to add one now?",
  lazyExtraSmall: () => import('./ExtraSmall-OONNBJRH.js'),
  lazySmall: () => import('./Small-TLJDSO7E.js'),
  lazyMedium: () => import('./Medium-ZXCHPQ6U.js'),
  lazyLarge: () => import('./Large-7PURMVRL.js')
});

// src/illustrations/fiori/NoSavedItems_v1/NoSavedItems_v1.tsx
var NoSavedItems_v1 = createIllustration({
  name: "NoSavedItems_v1",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-I6G2EHCS.js'),
  lazySmall: () => import('./Small-5QXBRQGJ.js'),
  lazyMedium: () => import('./Medium-LO3NWFGN.js'),
  lazyLarge: () => import('./Large-ZNXQZB2D.js')
});

// src/illustrations/fiori/NoSearchResults/NoSearchResults.tsx
var NoSearchResults = createIllustration({
  name: "NoSearchResults",
  title: "We could not find this.",
  subtitle: "Try adjusting your search.",
  lazyExtraSmall: () => import('./ExtraSmall-EYNQCGX2.js'),
  lazySmall: () => import('./Small-QWTD3BZL.js'),
  lazyMedium: () => import('./Medium-M3P7ZW5C.js'),
  lazyLarge: () => import('./Large-FMTNGSAN.js')
});

// src/illustrations/fiori/NoTasks/NoTasks.tsx
var NoTasks = createIllustration({
  name: "NoTasks",
  title: "You've no tasks",
  subtitle: "When you do, you'll see them here.",
  lazyExtraSmall: () => import('./ExtraSmall-YXMFJYXN.js'),
  lazySmall: () => import('./Small-66D5HLNB.js'),
  lazyMedium: () => import('./Medium-QLMJHMYU.js'),
  lazyLarge: () => import('./Large-CMJ4XSBP.js')
});

// src/illustrations/fiori/NoTasks_v1/NoTasks_v1.tsx
var NoTasks_v1 = createIllustration({
  name: "NoTasks_v1",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-OVMW7VEY.js'),
  lazySmall: () => import('./Small-GKJBJGB6.js'),
  lazyMedium: () => import('./Medium-5GYYW77R.js'),
  lazyLarge: () => import('./Large-IUJLKWAT.js')
});

// src/illustrations/fiori/PageNotFound/PageNotFound.tsx
var PageNotFound = createIllustration({
  name: "PageNotFound",
  title: "Sorry, we can't find this page",
  subtitle: "Please check the URL you are using to call the app.",
  lazyExtraSmall: () => import('./ExtraSmall-NI2X3RGL.js'),
  lazySmall: () => import('./Small-GDARE5BU.js'),
  lazyMedium: () => import('./Medium-UVHNQC7V.js'),
  lazyLarge: () => import('./Large-OS5NZE4D.js')
});

// src/illustrations/fiori/ReceiveAppreciation/ReceiveAppreciation.tsx
var ReceiveAppreciation = createIllustration({
  name: "ReceiveAppreciation",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-7VPBYSMP.js'),
  lazySmall: () => import('./Small-N3CEYZXY.js'),
  lazyMedium: () => import('./Medium-CONM3C6C.js'),
  lazyLarge: () => import('./Large-AILE4FDG.js')
});

// src/illustrations/fiori/ReloadScreen/ReloadScreen.tsx
var ReloadScreen = createIllustration({
  name: "ReloadScreen",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-VXTLKVRS.js'),
  lazySmall: () => import('./Small-3ON4B7IM.js'),
  lazyMedium: () => import('./Medium-IULNGXLZ.js'),
  lazyLarge: () => import('./Large-RG7IR3PN.js')
});

// src/illustrations/fiori/ResizeColumn/ResizeColumn.tsx
var ResizeColumn = createIllustration({
  name: "ResizeColumn",
  title: "Choose your own column width",
  subtitle: "You can resize columns by dragging the column borders.",
  lazyExtraSmall: () => import('./ExtraSmall-XHIRNISS.js'),
  lazySmall: () => import('./Small-2KTW7662.js'),
  lazyMedium: () => import('./Medium-7OVKV3Y3.js'),
  lazyLarge: () => import('./Large-ZSLW7H6D.js')
});

// src/illustrations/fiori/ResizingColumns/ResizingColumns.tsx
var ResizingColumns = createIllustration({
  name: "ResizingColumns",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-H5EZLLV5.js'),
  lazySmall: () => import('./Small-2LEIQTK7.js'),
  lazyMedium: () => import('./Medium-HYXIPNOU.js'),
  lazyLarge: () => import('./Large-BWJ3SWNW.js')
});

// src/illustrations/fiori/SearchEarth/SearchEarth.tsx
var SearchEarth = createIllustration({
  name: "SearchEarth",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-XL2R4MTB.js'),
  lazySmall: () => import('./Small-FXKDMLUY.js'),
  lazyMedium: () => import('./Medium-UW2FMJXB.js'),
  lazyLarge: () => import('./Large-YUFRKAJS.js')
});

// src/illustrations/fiori/SearchFolder/SearchFolder.tsx
var SearchFolder = createIllustration({
  name: "SearchFolder",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-AR3CKHBM.js'),
  lazySmall: () => import('./Small-EEFXFMUI.js'),
  lazyMedium: () => import('./Medium-PJZ5OF7P.js'),
  lazyLarge: () => import('./Large-GAUBK4FY.js')
});

// src/illustrations/fiori/SignOut/SignOut.tsx
var SignOut = createIllustration({
  name: "SignOut",
  title: "You've been signed out",
  subtitle: "You can now close this window.",
  lazyExtraSmall: () => import('./ExtraSmall-OVDEOCOH.js'),
  lazySmall: () => import('./Small-4SYF3ONS.js'),
  lazyMedium: () => import('./Medium-HUVGNLCW.js'),
  lazyLarge: () => import('./Large-G63CIXQA.js')
});

// src/illustrations/fiori/SimpleBalloon/SimpleBalloon.tsx
var SimpleBalloon = createIllustration({
  name: "SimpleBalloon",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-ETGQNTJH.js'),
  lazySmall: () => import('./Small-T5KUJ3V5.js'),
  lazyMedium: () => import('./Medium-OBDY53VA.js'),
  lazyLarge: () => import('./Large-GMD6NIBZ.js')
});

// src/illustrations/fiori/SimpleBell/SimpleBell.tsx
var SimpleBell = createIllustration({
  name: "SimpleBell",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-O5POZBA4.js'),
  lazySmall: () => import('./Small-WRV2YSMN.js'),
  lazyMedium: () => import('./Medium-FD224MXU.js'),
  lazyLarge: () => import('./Large-WDHJUL3V.js')
});

// src/illustrations/fiori/SimpleCalendar/SimpleCalendar.tsx
var SimpleCalendar = createIllustration({
  name: "SimpleCalendar",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-WPQ5AUR6.js'),
  lazySmall: () => import('./Small-I46ARMZK.js'),
  lazyMedium: () => import('./Medium-XJKQH2LG.js'),
  lazyLarge: () => import('./Large-5Z73LQQ2.js')
});

// src/illustrations/fiori/SimpleCheckMark/SimpleCheckMark.tsx
var SimpleCheckMark = createIllustration({
  name: "SimpleCheckMark",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-63Q5DOOE.js'),
  lazySmall: () => import('./Small-ECCRET4G.js'),
  lazyMedium: () => import('./Medium-U3WAXIBO.js'),
  lazyLarge: () => import('./Large-DOL2MUQJ.js')
});

// src/illustrations/fiori/SimpleConnection/SimpleConnection.tsx
var SimpleConnection = createIllustration({
  name: "SimpleConnection",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-JWJBKV5M.js'),
  lazySmall: () => import('./Small-BEUICZ2D.js'),
  lazyMedium: () => import('./Medium-5PNB6EZJ.js'),
  lazyLarge: () => import('./Large-LMDAQR3M.js')
});

// src/illustrations/fiori/SimpleEmptyDoc/SimpleEmptyDoc.tsx
var SimpleEmptyDoc = createIllustration({
  name: "SimpleEmptyDoc",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-XOXXLSFZ.js'),
  lazySmall: () => import('./Small-K3ZTMNIQ.js'),
  lazyMedium: () => import('./Medium-P6XAZSK2.js'),
  lazyLarge: () => import('./Large-M7TBQXOM.js')
});

// src/illustrations/fiori/SimpleEmptyList/SimpleEmptyList.tsx
var SimpleEmptyList = createIllustration({
  name: "SimpleEmptyList",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-P3IEA3SI.js'),
  lazySmall: () => import('./Small-ZG75NXUT.js'),
  lazyMedium: () => import('./Medium-CEDJBVQ2.js'),
  lazyLarge: () => import('./Large-QLPYACSU.js')
});

// src/illustrations/fiori/SimpleError/SimpleError.tsx
var SimpleError = createIllustration({
  name: "SimpleError",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-BZSUQCVG.js'),
  lazySmall: () => import('./Small-A2DVJ5A2.js'),
  lazyMedium: () => import('./Medium-OHTZY6WY.js'),
  lazyLarge: () => import('./Large-ELQCR7XQ.js')
});

// src/illustrations/fiori/SimpleMagnifier/SimpleMagnifier.tsx
var SimpleMagnifier = createIllustration({
  name: "SimpleMagnifier",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-ZPFAVNTF.js'),
  lazySmall: () => import('./Small-5HTK2RNT.js'),
  lazyMedium: () => import('./Medium-YA2QSBCF.js'),
  lazyLarge: () => import('./Large-22PEBSBL.js')
});

// src/illustrations/fiori/SimpleMail/SimpleMail.tsx
var SimpleMail = createIllustration({
  name: "SimpleMail",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-JGBFH5P4.js'),
  lazySmall: () => import('./Small-CTT2X42U.js'),
  lazyMedium: () => import('./Medium-LJTEC72G.js'),
  lazyLarge: () => import('./Large-KKVKPRHR.js')
});

// src/illustrations/fiori/SimpleNoSavedItems/SimpleNoSavedItems.tsx
var SimpleNoSavedItems = createIllustration({
  name: "SimpleNoSavedItems",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-M4T7WVSM.js'),
  lazySmall: () => import('./Small-XWZ3ULU2.js'),
  lazyMedium: () => import('./Medium-6WWJNQ35.js'),
  lazyLarge: () => import('./Large-RXGB26TB.js')
});

// src/illustrations/fiori/SimpleNotFoundMagnifier/SimpleNotFoundMagnifier.tsx
var SimpleNotFoundMagnifier = createIllustration({
  name: "SimpleNotFoundMagnifier",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-2J5QYMPL.js'),
  lazySmall: () => import('./Small-IIBZLL6X.js'),
  lazyMedium: () => import('./Medium-4KQFQUMD.js'),
  lazyLarge: () => import('./Large-267TTIFX.js')
});

// src/illustrations/fiori/SimpleReload/SimpleReload.tsx
var SimpleReload = createIllustration({
  name: "SimpleReload",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-NAIW4WVG.js'),
  lazySmall: () => import('./Small-CFQS5TYK.js'),
  lazyMedium: () => import('./Medium-X2EGFU4V.js'),
  lazyLarge: () => import('./Large-ARC6Q54P.js')
});

// src/illustrations/fiori/SimpleTask/SimpleTask.tsx
var SimpleTask = createIllustration({
  name: "SimpleTask",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-ZBS2MBY4.js'),
  lazySmall: () => import('./Small-IHTBBCDB.js'),
  lazyMedium: () => import('./Medium-JHBGZ5MI.js'),
  lazyLarge: () => import('./Large-Z2GVNYNQ.js')
});

// src/illustrations/fiori/SleepingBell/SleepingBell.tsx
var SleepingBell = createIllustration({
  name: "SleepingBell",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-7RJH7MYZ.js'),
  lazySmall: () => import('./Small-GMKWLXOV.js'),
  lazyMedium: () => import('./Medium-O4TOQ4QY.js'),
  lazyLarge: () => import('./Large-6FAHN6YL.js')
});

// src/illustrations/fiori/SortColumn/SortColumn.tsx
var SortColumn = createIllustration({
  name: "SortColumn",
  title: "Not seeing the most important items first?",
  subtitle: "Choose the sort criteria in the sort settings.",
  lazyExtraSmall: () => import('./ExtraSmall-GVQES4YN.js'),
  lazySmall: () => import('./Small-PVSFHK22.js'),
  lazyMedium: () => import('./Medium-WM77KWFX.js'),
  lazyLarge: () => import('./Large-ZAAZ4IVL.js')
});

// src/illustrations/fiori/SortingColumns/SortingColumns.tsx
var SortingColumns = createIllustration({
  name: "SortingColumns",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-SBKUMOCM.js'),
  lazySmall: () => import('./Small-5YD2X7EF.js'),
  lazyMedium: () => import('./Medium-QVNOM2CB.js'),
  lazyLarge: () => import('./Large-ILLSC5SU.js')
});

// src/illustrations/fiori/SuccessBalloon/SuccessBalloon.tsx
var SuccessBalloon = createIllustration({
  name: "SuccessBalloon",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-R4JUPNBH.js'),
  lazySmall: () => import('./Small-PZU3RRYU.js'),
  lazyMedium: () => import('./Medium-G3HUWRUN.js'),
  lazyLarge: () => import('./Large-ZFKN4PNG.js')
});

// src/illustrations/fiori/SuccessCheckMark/SuccessCheckMark.tsx
var SuccessCheckMark = createIllustration({
  name: "SuccessCheckMark",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-PJDM2BRZ.js'),
  lazySmall: () => import('./Small-LAYEUFHU.js'),
  lazyMedium: () => import('./Medium-SLOTYO5D.js'),
  lazyLarge: () => import('./Large-OTN2D5NI.js')
});

// src/illustrations/fiori/SuccessHighFive/SuccessHighFive.tsx
var SuccessHighFive = createIllustration({
  name: "SuccessHighFive",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-WUJQNQJP.js'),
  lazySmall: () => import('./Small-J7OASRJP.js'),
  lazyMedium: () => import('./Medium-JQ5X56ZM.js'),
  lazyLarge: () => import('./Large-XIZD5AXC.js')
});

// src/illustrations/fiori/SuccessScreen/SuccessScreen.tsx
var SuccessScreen = createIllustration({
  name: "SuccessScreen",
  title: "Nicely done!",
  subtitle: "You completed all your learning assignments.",
  lazyExtraSmall: () => import('./ExtraSmall-VB3RL4J7.js'),
  lazySmall: () => import('./Small-KOEFGN3P.js'),
  lazyMedium: () => import('./Medium-OJMFCDBH.js'),
  lazyLarge: () => import('./Large-MSVYLVRE.js')
});

// src/illustrations/fiori/Survey/Survey.tsx
var Survey = createIllustration({
  name: "Survey",
  title: "Your Opinion Matters",
  subtitle: "We want to hear what you think about SAP software. Share your feedback with us by taking our short survey.",
  lazyExtraSmall: () => import('./ExtraSmall-AOKN4T7A.js'),
  lazySmall: () => import('./Small-VOYBG5BW.js'),
  lazyMedium: () => import('./Medium-67L7EZKI.js'),
  lazyLarge: () => import('./Large-IPT3QOUC.js')
});

// src/illustrations/fiori/Tent/Tent.tsx
var Tent = createIllustration({
  name: "Tent",
  title: "",
  subtitle: "",
  lazyExtraSmall: () => import('./ExtraSmall-O7VDTOYH.js'),
  lazySmall: () => import('./Small-PBRMBZLM.js'),
  lazyMedium: () => import('./Medium-BV6IUIPQ.js'),
  lazyLarge: () => import('./Large-KB5CCXWO.js')
});

// src/illustrations/fiori/UnableToLoad/UnableToLoad.tsx
var UnableToLoad = createIllustration({
  name: "UnableToLoad",
  title: "Unable to load data",
  subtitle: "Check your internet connection. And if that's not it, try reloading. If that still doesn't help, check with your administrator.",
  lazyExtraSmall: () => import('./ExtraSmall-IREUN3MF.js'),
  lazySmall: () => import('./Small-SQO2Y2DI.js'),
  lazyMedium: () => import('./Medium-QRR7UPXL.js'),
  lazyLarge: () => import('./Large-RAT6DW3F.js')
});

// src/illustrations/fiori/UnableToLoadImage/UnableToLoadImage.tsx
var UnableToLoadImage = createIllustration({
  name: "UnableToLoadImage",
  title: "Unable to load image",
  subtitle: "We couldn't find the image at the specified location, or the server isn't responding.",
  lazyExtraSmall: () => import('./ExtraSmall-7PHOSUT3.js'),
  lazySmall: () => import('./Small-7Z46D7U7.js'),
  lazyMedium: () => import('./Medium-EIZTU3ZK.js'),
  lazyLarge: () => import('./Large-OK435IBF.js')
});

// src/illustrations/fiori/UnableToUpload/UnableToUpload.tsx
var UnableToUpload = createIllustration({
  name: "UnableToUpload",
  title: "Unable to upload data",
  subtitle: "Check your Internet connection. If that doesn\u2019t help, check the file format and file size. Otherwise contact your administrator.",
  lazyExtraSmall: () => import('./ExtraSmall-COKMXUEU.js'),
  lazySmall: () => import('./Small-EUMXA72J.js'),
  lazyMedium: () => import('./Medium-R5RZQA3P.js'),
  lazyLarge: () => import('./Large-SYJKFCF6.js')
});

// src/illustrations/fiori/UploadCollection/UploadCollection.tsx
var UploadCollection = createIllustration({
  name: "UploadCollection",
  title: "Drop files here",
  subtitle: "You can also upload several files all at once.",
  lazyExtraSmall: () => import('./ExtraSmall-YY3WVKVH.js'),
  lazySmall: () => import('./Small-HB5VUIIJ.js'),
  lazyMedium: () => import('./Medium-6WCGAAWI.js'),
  lazyLarge: () => import('./Large-VOTGI4YE.js')
});

// src/illustrations/fiori/UploadToCloud/UploadToCloud.tsx
var UploadToCloud = createIllustration({
  name: "UploadToCloud",
  title: "Migrate to SAP Integration Suite",
  subtitle: "Move your integration content from the existing Process Integration subscription to the Cloud Integration capability.",
  lazyExtraSmall: () => import('./ExtraSmall-LORJ7V6F.js'),
  lazySmall: () => import('./Small-DCFCDVRQ.js'),
  lazyMedium: () => import('./Medium-PKBUF3IU.js'),
  lazyLarge: () => import('./Large-ALRASRUH.js')
});

// src/illustrations/fiori/UserHasSignedUp/UserHasSignedUp.tsx
var UserHasSignedUp = createIllustration({
  name: "UserHasSignedUp",
  title: "Success!",
  subtitle: "You set up your account. Want to take a quick tour?",
  lazyExtraSmall: () => import('./ExtraSmall-QSJIAOIL.js'),
  lazySmall: () => import('./Small-GHT23EF3.js'),
  lazyMedium: () => import('./Medium-5N5WN2MY.js'),
  lazyLarge: () => import('./Large-RS4ZGIWM.js')
});

export { Achievement, AddColumn, AddDimensions, AddPeople, AddPeopleToCalendar, AddingColumns, BalloonSky, BeforeSearch, Connection, DragFilesToUpload, EmptyCalendar, EmptyList, EmptyPlanningCalendar, ErrorScreen, FilterTable, FilteringColumns, GroupTable, GroupingColumns, KeyTask, NewMail, NoActivities, NoChartData, NoColumnsSet, NoData, NoDimensionsSet, NoEntries, NoFilterResults, NoMail, NoMail_v1, NoNotifications, NoSavedItems, NoSavedItems_v1, NoSearchResults, NoTasks, NoTasks_v1, PageNotFound, ReceiveAppreciation, ReloadScreen, ResizeColumn, ResizingColumns, SearchEarth, SearchFolder, SignOut, SimpleBalloon, SimpleBell, SimpleCalendar, SimpleCheckMark, SimpleConnection, SimpleEmptyDoc, SimpleEmptyList, SimpleError, SimpleMagnifier, SimpleMail, SimpleNoSavedItems, SimpleNotFoundMagnifier, SimpleReload, SimpleTask, SleepingBell, SortColumn, SortingColumns, SuccessBalloon, SuccessCheckMark, SuccessHighFive, SuccessScreen, Survey, Tent, UnableToLoad, UnableToLoadImage, UnableToUpload, UploadCollection, UploadToCloud, UserHasSignedUp };
