// @sap-ui/fx-components/dist/chunk-EHO24TXD.js
import { AcceptIcon } from './chunk-VNDUTC7E.js';
import { SlimArrowRightIcon } from './chunk-ZQSLIXR7.js';
import { Popover } from './chunk-ZDBE5GTH.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { createContext, useRef, useEffect, useCallback, useImperativeHandle, useMemo, useContext, useState, useId, Children, isValidElement } from 'react';
import { createPortal } from 'react-dom';
import { Loader2 } from 'lucide-react';
import { useTranslation } from 'react-i18next';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

// src/types/menu.ts
var MenuItemGroupCheckMode = /* @__PURE__ */ ((MenuItemGroupCheckMode2) => {
  MenuItemGroupCheckMode2["None"] = "None";
  MenuItemGroupCheckMode2["Single"] = "Single";
  MenuItemGroupCheckMode2["Multiple"] = "Multiple";
  return MenuItemGroupCheckMode2;
})(MenuItemGroupCheckMode || {});
var MenuContext = createContext({
  closeMenu: () => {
  },
  closeAllSubmenus: () => {
  },
  registerSubmenu: () => {
  },
  unregisterSubmenu: () => {
  },
  submenuDepth: 0
});
var MenuItemGroupContext = createContext(null);
function calculateSubmenuPosition(parentItem, submenu) {
  const parentRect = parentItem.getBoundingClientRect();
  const submenuRect = submenu.getBoundingClientRect();
  const viewportWidth = document.documentElement.clientWidth;
  const viewportHeight = document.documentElement.clientHeight;
  const gap = 4;
  const margin = 8;
  const spaceRight = viewportWidth - parentRect.right - gap;
  const spaceLeft = parentRect.left - gap;
  let top = parentRect.top;
  let left;
  if (spaceRight >= submenuRect.width) {
    left = parentRect.right + gap;
  } else if (spaceLeft >= submenuRect.width) {
    left = parentRect.left - submenuRect.width - gap;
  } else {
    const parentMenu = parentItem.closest('[role="menu"], [data-submenu="true"]');
    if (parentMenu) {
      const parentMenuRect = parentMenu.getBoundingClientRect();
      left = parentMenuRect.left;
    } else {
      left = parentRect.left;
    }
    left = Math.max(Math.min(left, viewportWidth - submenuRect.width - margin), margin);
  }
  if (top + submenuRect.height > viewportHeight - margin) {
    top = viewportHeight - submenuRect.height - margin;
  }
  if (top < margin) {
    top = margin;
  }
  return { top, left };
}
function getFocusableItems(container) {
  const all = Array.from(
    container.querySelectorAll(
      '[role^="menuitem"]:not([aria-disabled="true"])'
    )
  );
  return all.filter((item) => {
    let parent = item.parentElement;
    while (parent && parent !== container) {
      if (parent.getAttribute("role") === "menu" || parent.getAttribute("data-submenu") === "true") {
        return false;
      }
      parent = parent.parentElement;
    }
    return true;
  });
}
function useMenuNavigation(containerRef, options) {
  const typeAheadBufferRef = useRef("");
  const typeAheadTimeoutRef = useRef(null);
  const handleKeyDown = useCallback(
    (e) => {
      const container = containerRef.current;
      if (!container) return;
      const items = getFocusableItems(container);
      if (items.length === 0) return;
      const currentIndex = items.indexOf(
        document.activeElement
      );
      switch (e.key) {
        case "ArrowDown": {
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex < items.length - 1) {
            items[currentIndex + 1]?.focus();
          }
          break;
        }
        case "ArrowUp": {
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex > 0) {
            items[currentIndex - 1]?.focus();
          }
          break;
        }
        case "Home": {
          e.preventDefault();
          e.stopPropagation();
          items[0]?.focus();
          break;
        }
        case "End": {
          e.preventDefault();
          e.stopPropagation();
          items[items.length - 1]?.focus();
          break;
        }
        case "ArrowLeft": {
          if (options?.isSubmenu) {
            e.preventDefault();
            e.stopPropagation();
            options.onClose?.();
            options.parentItemRef?.current?.focus();
          }
          break;
        }
        case "Escape": {
          if (options?.isSubmenu) {
            e.preventDefault();
            e.stopPropagation();
            e.nativeEvent.stopImmediatePropagation();
            options.onClose?.();
            options.parentItemRef?.current?.focus();
          }
          break;
        }
        case "Tab": {
          e.preventDefault();
          if (options?.isSubmenu) {
            e.stopPropagation();
            e.nativeEvent.stopImmediatePropagation();
            options.onClose?.();
            options.parentItemRef?.current?.focus();
          } else {
            options?.onCloseRoot?.();
          }
          break;
        }
        default: {
          if (e.key.length === 1 && !e.ctrlKey && !e.metaKey && !e.altKey) {
            e.stopPropagation();
            if (typeAheadTimeoutRef.current) {
              clearTimeout(typeAheadTimeoutRef.current);
            }
            typeAheadBufferRef.current += e.key.toLowerCase();
            typeAheadTimeoutRef.current = setTimeout(() => {
              typeAheadBufferRef.current = "";
            }, 500);
            const buffer = typeAheadBufferRef.current;
            const match = items.find((item) => {
              const text = item.textContent?.toLowerCase() || "";
              return text.startsWith(buffer);
            });
            if (match) {
              match.focus();
            }
          }
          break;
        }
      }
    },
    [containerRef, options]
  );
  useEffect(() => {
    return () => {
      if (typeAheadTimeoutRef.current) {
        clearTimeout(typeAheadTimeoutRef.current);
      }
    };
  }, []);
  return handleKeyDown;
}
function LoadingSpinner({ delay = 50 }) {
  const [show, setShow] = useState(delay <= 0);
  useEffect(() => {
    if (delay <= 0) return;
    const timer = setTimeout(() => setShow(true), delay);
    return () => clearTimeout(timer);
  }, [delay]);
  if (!show) return null;
  return /* @__PURE__ */ jsx("div", { className: "flex items-center justify-center py-4 px-8", children: /* @__PURE__ */ jsx(Loader2, { className: "h-5 w-5 animate-spin text-sapphire-text-tertiary" }) });
}
function mapMenuPlacementToPopover(menuPlacement = "bottom-start") {
  switch (menuPlacement) {
    case "bottom-start":
      return { placement: "Bottom" /* Bottom */, horizontalAlign: "Start" /* Start */ };
    case "bottom-end":
      return { placement: "Bottom" /* Bottom */, horizontalAlign: "End" /* End */ };
    case "top-start":
      return { placement: "Top" /* Top */, horizontalAlign: "Start" /* Start */ };
    case "top-end":
      return { placement: "Top" /* Top */, horizontalAlign: "End" /* End */ };
    case "right-start":
      return { placement: "End" /* End */, verticalAlign: "Top" /* Top */ };
    case "right-end":
      return { placement: "End" /* End */, verticalAlign: "Bottom" /* Bottom */ };
    case "left-start":
      return { placement: "Start" /* Start */, verticalAlign: "Top" /* Top */ };
    case "left-end":
      return { placement: "Start" /* Start */, verticalAlign: "Bottom" /* Bottom */ };
    default:
      return { placement: "Bottom" /* Bottom */, horizontalAlign: "Start" /* Start */ };
  }
}
function Menu({
  open = false,
  opener,
  children,
  placement = "bottom-start",
  onClose,
  onItemClick,
  onCheck,
  className,
  style,
  headerText,
  loading,
  loadingDelay = 50,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  ref
}) {
  const { t } = useTranslation("fx");
  const menuRef = useRef(null);
  const submenuClosersRef = useRef(/* @__PURE__ */ new Map());
  const closingRef = useRef(false);
  const onBeforeCloseRef = useRef(onBeforeClose);
  const onCloseRef = useRef(onClose);
  const onItemClickRef = useRef(onItemClick);
  const onCheckRef = useRef(onCheck);
  useEffect(() => {
    onBeforeCloseRef.current = onBeforeClose;
    onCloseRef.current = onClose;
    onItemClickRef.current = onItemClick;
    onCheckRef.current = onCheck;
  });
  const resolvedOpener = typeof opener === "string" ? document.getElementById(opener) : opener;
  const openerRef = useRef(null);
  useEffect(() => {
    openerRef.current = resolvedOpener ?? null;
  }, [resolvedOpener]);
  const closeMenu = useCallback((escPressed) => {
    closingRef.current = true;
    onBeforeCloseRef.current?.({ escPressed: !!escPressed });
    onCloseRef.current?.();
  }, []);
  useImperativeHandle(ref, () => ({
    close: () => closeMenu(),
    isOpen: () => open
  }), [closeMenu, open]);
  useEffect(() => {
    if (open) {
      closingRef.current = false;
    }
  }, [open]);
  const closeAllSubmenus = useCallback(() => {
    submenuClosersRef.current.forEach((close) => close());
  }, []);
  const registerSubmenu = useCallback((id, close) => {
    submenuClosersRef.current.set(id, close);
  }, []);
  const unregisterSubmenu = useCallback((id) => {
    submenuClosersRef.current.delete(id);
  }, []);
  const handleKeyDown = useMenuNavigation(menuRef, { onCloseRoot: closeMenu });
  const stableOnItemClick = useCallback(
    (detail) => onItemClickRef.current?.(detail),
    []
  );
  const stableOnCheck = useCallback(
    (detail) => onCheckRef.current?.(detail),
    []
  );
  const prevOpenLoadingRef = useRef({ open, loading });
  useEffect(() => {
    const prev = prevOpenLoadingRef.current;
    prevOpenLoadingRef.current = { open, loading };
    if (prev.open && open && prev.loading && !loading && menuRef.current) {
      requestAnimationFrame(() => {
        if (menuRef.current) {
          const items = getFocusableItems(menuRef.current);
          const selected = items.find((el) => el.hasAttribute("data-menu-selected") || el.getAttribute("aria-checked") === "true");
          (selected ?? items[0])?.focus();
        }
      });
    }
  }, [loading, open]);
  const menuContextValue = useMemo(
    () => ({ closeMenu, onItemClick: stableOnItemClick, onCheck: stableOnCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, submenuDepth: 0 }),
    [closeMenu, stableOnItemClick, stableOnCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu]
  );
  const popoverPlacement = useMemo(
    () => mapMenuPlacementToPopover(placement),
    [placement]
  );
  const handleOpen = useCallback(() => {
    onOpen?.();
    if (!menuRef.current) return;
    const items = getFocusableItems(menuRef.current);
    const selected = items.find((el) => el.hasAttribute("data-menu-selected") || el.getAttribute("aria-checked") === "true");
    (selected ?? items[0])?.focus();
  }, [onOpen]);
  const isInsidePopup = useCallback((target) => {
    return !!target.closest?.('[data-submenu="true"]');
  }, []);
  const handlePopoverBeforeClose = useCallback(({ escPressed }) => {
    if (closingRef.current) return;
    onBeforeCloseRef.current?.({ escPressed });
  }, []);
  const handlePopoverClose = useCallback(() => {
    const wasClosing = closingRef.current;
    closingRef.current = false;
    if (wasClosing) return;
    onCloseRef.current?.();
  }, []);
  return /* @__PURE__ */ jsx(
    Popover,
    {
      opener: openerRef,
      open,
      onBeforeOpen,
      onOpen: handleOpen,
      onBeforeClose: handlePopoverBeforeClose,
      onClose: handlePopoverClose,
      hideArrow: true,
      noPadding: true,
      offset: 4,
      placement: popoverPlacement.placement,
      horizontalAlign: popoverPlacement.horizontalAlign,
      verticalAlign: popoverPlacement.verticalAlign,
      accessibleRole: "Dialog" /* Dialog */,
      accessibleName: t("MENU_POPOVER_ACCESSIBLE_NAME"),
      isInsidePopup,
      className: "p-0 border border-sapphire-border-primary shadow-lg [&>div]:overflow-visible",
      style: {
        filter: "none",
        ...style
      },
      children: /* @__PURE__ */ jsx(MenuContext.Provider, { value: menuContextValue, children: /* @__PURE__ */ jsxs(
        "div",
        {
          ref: menuRef,
          role: "menu",
          tabIndex: -1,
          className: cn("min-w-[18.75rem] py-sapphire-xs outline-none", className),
          onKeyDown: handleKeyDown,
          children: [
            headerText && /* @__PURE__ */ jsx(MenuHeader, { text: headerText }),
            loading ? /* @__PURE__ */ jsx(LoadingSpinner, { delay: loadingDelay }) : children
          ]
        }
      ) })
    }
  );
}
function SubmenuScope({ closeMenu, onItemClick, onCheck, parentDepth, children }) {
  const submenuClosersRef = useRef(/* @__PURE__ */ new Map());
  const closeAllSubmenus = useCallback(() => {
    submenuClosersRef.current.forEach((close) => close());
  }, []);
  const registerSubmenu = useCallback((id, close) => {
    submenuClosersRef.current.set(id, close);
  }, []);
  const unregisterSubmenu = useCallback((id) => {
    submenuClosersRef.current.delete(id);
  }, []);
  const depth = parentDepth + 1;
  const contextValue = useMemo(
    () => ({ closeMenu, onItemClick, onCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, submenuDepth: depth }),
    [closeMenu, onItemClick, onCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, depth]
  );
  return /* @__PURE__ */ jsx(MenuContext.Provider, { value: contextValue, children });
}
function MenuItem({
  text,
  icon,
  endContent,
  disabled = false,
  startsSection = false,
  children,
  onClick,
  className,
  data,
  checked,
  additionalText,
  loading: itemLoading,
  loadingDelay = 50,
  tooltip,
  accessibleName,
  accessibilityAttributes,
  ref
}) {
  const { closeMenu, onItemClick, onCheck, closeAllSubmenus, registerSubmenu, unregisterSubmenu, submenuDepth } = useContext(MenuContext);
  const groupContext = useContext(MenuItemGroupContext);
  const itemRef = useRef(null);
  const submenuRef = useRef(null);
  const [submenuOpen, setSubmenuOpen] = useState(false);
  const [portalContainer, setPortalContainer] = useState(null);
  const hoverTimeoutRef = useRef(null);
  const submenuId = useId();
  const submenuChildren = [];
  let hasSubmenu = false;
  Children.forEach(children, (child) => {
    if (isValidElement(child)) {
      const displayName = child.type?.displayName;
      if (displayName === "MenuItem" || displayName === "MenuSeparator" || displayName === "MenuHeader" || displayName === "MenuItemGroup") {
        hasSubmenu = true;
        submenuChildren.push(child);
      }
    }
  });
  if (itemLoading && !hasSubmenu) {
    hasSubmenu = true;
  }
  const checkMode = groupContext?.checkMode ?? "None" /* None */;
  const isInCheckGroup = checkMode !== "None" /* None */;
  let role = "menuitem";
  if (checkMode === "Single" /* Single */) {
    role = "menuitemradio";
  } else if (checkMode === "Multiple" /* Multiple */) {
    role = "menuitemcheckbox";
  }
  if (accessibilityAttributes?.role) {
    role = accessibilityAttributes.role;
  }
  const openSubmenu = useCallback(() => {
    if (itemRef.current) {
      const popoverEl = itemRef.current.closest("[popover]");
      setPortalContainer(popoverEl ?? document.body);
    }
    setSubmenuOpen(true);
  }, []);
  useEffect(() => {
    if (hasSubmenu) {
      registerSubmenu(submenuId, () => setSubmenuOpen(false));
      return () => unregisterSubmenu(submenuId);
    }
  }, [hasSubmenu, registerSubmenu, unregisterSubmenu]);
  const updateSubmenuPosition = useCallback(() => {
    if (!itemRef.current || !submenuRef.current) return;
    const pos = calculateSubmenuPosition(itemRef.current, submenuRef.current);
    submenuRef.current.style.top = `${pos.top}px`;
    submenuRef.current.style.left = `${pos.left}px`;
  }, []);
  const setSubmenuRef = useCallback((node) => {
    submenuRef.current = node;
    if (node) {
      updateSubmenuPosition();
    }
  }, [updateSubmenuPosition]);
  useEffect(() => {
    if (!submenuOpen) return;
    window.addEventListener("scroll", updateSubmenuPosition, true);
    window.addEventListener("resize", updateSubmenuPosition);
    return () => {
      window.removeEventListener("scroll", updateSubmenuPosition, true);
      window.removeEventListener("resize", updateSubmenuPosition);
    };
  }, [submenuOpen, updateSubmenuPosition]);
  const prevSubmenuLoadingRef = useRef({ submenuOpen, itemLoading });
  useEffect(() => {
    const prev = prevSubmenuLoadingRef.current;
    prevSubmenuLoadingRef.current = { submenuOpen, itemLoading };
    if (prev.submenuOpen && submenuOpen && prev.itemLoading && !itemLoading && submenuRef.current) {
      requestAnimationFrame(() => {
        if (submenuRef.current) {
          const items = getFocusableItems(submenuRef.current);
          items[0]?.focus();
        }
      });
    }
  }, [itemLoading, submenuOpen]);
  const openedByKeyboardRef = useRef(false);
  useEffect(() => {
    if (submenuOpen && submenuRef.current && openedByKeyboardRef.current) {
      openedByKeyboardRef.current = false;
      requestAnimationFrame(() => {
        if (submenuRef.current) {
          const items = getFocusableItems(submenuRef.current);
          if (items.length > 0) {
            items[0].focus();
          }
        }
      });
    }
  }, [submenuOpen]);
  const activateItem = (e) => {
    onClick?.(e);
    const result = onItemClick?.({
      item: { text, icon, endContent, disabled, startsSection, hasSubmenu, data, checked },
      text: text || ""
    });
    if (result !== false) {
      closeMenu();
    }
  };
  const handleClick = (e) => {
    if (disabled) return;
    if (hasSubmenu) {
      e.stopPropagation();
      if (submenuOpen) {
        setSubmenuOpen(false);
      } else {
        openSubmenu();
      }
      return;
    }
    if (isInCheckGroup) {
      const detail = {
        item: { text, icon, endContent, disabled, startsSection, hasSubmenu, data, checked },
        text: text || "",
        checked: !checked
      };
      onCheck?.(detail);
      onClick?.(e);
      onItemClick?.(detail);
      if (!e.shiftKey) {
        closeMenu();
      }
      return;
    }
    activateItem(e);
  };
  const handleMouseEnter = () => {
    if (disabled) return;
    if (hoverTimeoutRef.current) {
      clearTimeout(hoverTimeoutRef.current);
      hoverTimeoutRef.current = null;
    }
    if (hasSubmenu) {
      closeAllSubmenus();
      openSubmenu();
    }
  };
  const handleMouseLeave = () => {
    if (hasSubmenu) {
      hoverTimeoutRef.current = setTimeout(() => {
        setSubmenuOpen(false);
      }, 150);
    }
  };
  const handleSubmenuMouseEnter = () => {
    if (hoverTimeoutRef.current) {
      clearTimeout(hoverTimeoutRef.current);
      hoverTimeoutRef.current = null;
    }
  };
  const handleSubmenuMouseLeave = () => {
    hoverTimeoutRef.current = setTimeout(() => {
      setSubmenuOpen(false);
    }, 150);
  };
  const handleKeyDown = (e) => {
    if (disabled) return;
    if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      if (isInCheckGroup) {
        const detail = {
          item: { text, icon, endContent, disabled, startsSection, hasSubmenu, data, checked },
          text: text || "",
          checked: !checked
        };
        onCheck?.(detail);
        onItemClick?.(detail);
        if (!e.shiftKey) {
          closeMenu();
        }
        return;
      }
      if (hasSubmenu) {
        e.stopPropagation();
        openedByKeyboardRef.current = true;
        openSubmenu();
        return;
      }
      activateItem(e);
    } else if (e.key === "ArrowRight" && hasSubmenu) {
      e.preventDefault();
      e.stopPropagation();
      openedByKeyboardRef.current = true;
      openSubmenu();
    }
  };
  const closeSubmenu = useCallback(() => setSubmenuOpen(false), []);
  const submenuKeyDown = useMenuNavigation(submenuRef, {
    isSubmenu: true,
    onClose: closeSubmenu,
    parentItemRef: itemRef
  });
  useEffect(() => {
    return () => {
      if (hoverTimeoutRef.current) {
        clearTimeout(hoverTimeoutRef.current);
      }
    };
  }, []);
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    startsSection && /* @__PURE__ */ jsx(MenuSeparator, {}),
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: (node) => {
          itemRef.current = node;
          if (typeof ref === "function") {
            ref(node);
          } else if (ref) {
            ref.current = node;
          }
        },
        role,
        tabIndex: disabled ? -1 : 0,
        "aria-disabled": disabled,
        "aria-haspopup": hasSubmenu ? "menu" : void 0,
        "aria-expanded": hasSubmenu ? submenuOpen : void 0,
        "aria-checked": isInCheckGroup ? !!checked : void 0,
        "data-menu-selected": checked && !isInCheckGroup ? "" : void 0,
        "aria-label": accessibleName,
        "aria-keyshortcuts": accessibilityAttributes?.ariaKeyShortcuts,
        title: tooltip,
        onClick: handleClick,
        onMouseEnter: handleMouseEnter,
        onMouseLeave: handleMouseLeave,
        onKeyDown: handleKeyDown,
        className: cn(
          "group/item relative flex min-h-[54px] cursor-pointer select-none items-center gap-4 px-4 py-2 text-base font-semibold text-sapphire-text-secondary outline outline-2 outline-transparent outline-offset-[-4px]",
          "hover:bg-sapphire-neutral-hover-background-2 hover:text-foreground",
          "active:bg-sapphire-brand-selected-background active:text-foreground",
          "focus:outline-ring focus:z-10",
          checked && !isInCheckGroup && "bg-sapphire-brand-selected-background text-foreground border border-sapphire-brand-foreground focus:border-transparent",
          checked && !isInCheckGroup && "hover:bg-sapphire-brand-toggle-background",
          disabled && "pointer-events-none opacity-50",
          className
        ),
        children: [
          isInCheckGroup && /* @__PURE__ */ jsx("span", { className: "flex size-5 shrink-0 items-center justify-center", children: checked && /* @__PURE__ */ jsx(AcceptIcon, { className: "size-5" }) }),
          icon && /* @__PURE__ */ jsx("span", { className: cn("flex size-5 shrink-0 items-center justify-center [&>svg]:size-5 group-hover/item:text-sapphire-brand-foreground group-active/item:text-sapphire-brand-foreground", checked && !isInCheckGroup && "text-sapphire-brand-foreground"), children: icon }),
          /* @__PURE__ */ jsx("span", { className: "flex-1 truncate", children: text }),
          hasSubmenu ? /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "ml-auto size-5 shrink-0 text-sapphire-text-tertiary" }) : endContent ? /* @__PURE__ */ jsx("span", { className: "ml-auto", children: endContent }) : additionalText ? /* @__PURE__ */ jsx("span", { className: "ml-4 shrink-0 text-sm text-sapphire-text-tertiary", children: additionalText }) : null
        ]
      }
    ),
    hasSubmenu && submenuOpen && createPortal(
      /* @__PURE__ */ jsx(SubmenuScope, { closeMenu, onItemClick, onCheck, parentDepth: submenuDepth, children: /* @__PURE__ */ jsx(
        "div",
        {
          ref: setSubmenuRef,
          role: "menu",
          "data-submenu": "true",
          className: "min-w-[18.75rem] overflow-hidden rounded-lg border border-sapphire-border-primary bg-sapphire-card-bg-primary py-3.5 text-popover-foreground shadow-lg",
          style: {
            position: "fixed",
            // submenuDepth is the *parent's* depth from context.
            // +1 yields the z-index for this child submenu, so nested
            // submenus stack correctly: depth 0→z61, depth 1→z62, etc.
            zIndex: 60 + submenuDepth + 1
          },
          onMouseEnter: handleSubmenuMouseEnter,
          onMouseLeave: handleSubmenuMouseLeave,
          onKeyDown: submenuKeyDown,
          children: itemLoading ? /* @__PURE__ */ jsx(LoadingSpinner, { delay: loadingDelay }) : submenuChildren
        }
      ) }),
      portalContainer ?? document.body
    )
  ] });
}
MenuItem.displayName = "MenuItem";
function MenuItemGroup({ checkMode = "None" /* None */, children, accessibleName, className, ref }) {
  const mode = checkMode;
  let defaultLabel;
  if (mode === "Single" /* Single */) {
    defaultLabel = "Menu Item Group, Single Selection";
  } else if (mode === "Multiple" /* Multiple */) {
    defaultLabel = "Menu Item Group, Multi Selection";
  }
  const groupContextValue = useMemo(() => ({ checkMode: mode }), [mode]);
  return /* @__PURE__ */ jsx(MenuItemGroupContext.Provider, { value: groupContextValue, children: /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      role: "group",
      "aria-label": accessibleName || defaultLabel,
      className,
      children
    }
  ) });
}
MenuItemGroup.displayName = "MenuItemGroup";
function MenuSeparator({ className, ref }) {
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      role: "separator",
      className: cn("my-2 h-px bg-border mx-4", className)
    }
  );
}
MenuSeparator.displayName = "MenuSeparator";
function MenuHeader({ text, children, endContent, className, ref }) {
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      className: cn(
        "px-4 py-1.5 text-sm font-semibold text-sapphire-text-tertiary",
        endContent && "flex items-center gap-2",
        className
      ),
      children: [
        /* @__PURE__ */ jsx("span", { className: endContent ? "flex-1" : void 0, children: text || children }),
        endContent && /* @__PURE__ */ jsx("span", { className: "flex shrink-0 items-center gap-1", children: endContent })
      ]
    }
  );
}
MenuHeader.displayName = "MenuHeader";

export { Menu, MenuHeader, MenuItem, MenuItemGroup, MenuItemGroupCheckMode, MenuSeparator };
