// @sap-ui/fx-components/dist/chunk-S77RYWD7.js
import { AlertIcon } from './chunk-4IU3KSGT.js';
import { InformationIcon, SysEnter2Icon } from './chunk-ZK4METPA.js';
import { ErrorIcon } from './chunk-FL4UPYZT.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useImperativeHandle } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/progress-indicator.ts
var ProgressIndicatorValueState = /* @__PURE__ */ ((ProgressIndicatorValueState2) => {
  ProgressIndicatorValueState2["None"] = "None";
  ProgressIndicatorValueState2["Positive"] = "Positive";
  ProgressIndicatorValueState2["Critical"] = "Critical";
  ProgressIndicatorValueState2["Negative"] = "Negative";
  ProgressIndicatorValueState2["Information"] = "Information";
  return ProgressIndicatorValueState2;
})(ProgressIndicatorValueState || {});
var trackBgClass = {
  ["None" /* None */]: "bg-sapphire-neutral-pressed-background-2",
  ["Positive" /* Positive */]: "bg-sapphire-positive-bg",
  ["Critical" /* Critical */]: "bg-sapphire-warning-bg",
  ["Negative" /* Negative */]: "bg-sapphire-negative-bg",
  ["Information" /* Information */]: "bg-sapphire-info-bg"
};
var barColorClass = {
  ["None" /* None */]: "bg-sapphire-neutral-foreground-muted",
  ["Positive" /* Positive */]: "bg-sapphire-positive",
  ["Critical" /* Critical */]: "bg-sapphire-warning",
  ["Negative" /* Negative */]: "bg-sapphire-negative",
  ["Information" /* Information */]: "bg-sapphire-info"
};
var iconColorClass = {
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Information" /* Information */]: "text-sapphire-info"
};
var stateIconMap = {
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Critical" /* Critical */]: AlertIcon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Information" /* Information */]: InformationIcon
};
var stateLabel = {
  ["None" /* None */]: "",
  ["Positive" /* Positive */]: "Positive",
  ["Critical" /* Critical */]: "Critical",
  ["Negative" /* Negative */]: "Negative",
  ["Information" /* Information */]: "Information"
};
function ProgressIndicator({
  value = 0,
  displayValue,
  hideValue = false,
  disabled = false,
  valueState = "None" /* None */,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = useRef(null);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        rootRef.current?.focus();
      },
      blur() {
        rootRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    []
  );
  const clampedValue = Math.min(100, Math.max(0, Number.isNaN(value) ? 0 : value));
  const valueText = displayValue || `${clampedValue}%`;
  const vs = valueState;
  const label = stateLabel[vs] || "";
  const ariaValueText = displayValue ? displayValue : label ? `${valueText} ${label}` : valueText;
  const StateIcon = stateIconMap[vs] || null;
  const hasIcon = vs !== "None" /* None */ && StateIcon;
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: rootRef,
      role: "progressbar",
      "aria-valuemin": 0,
      "aria-valuemax": 100,
      "aria-valuenow": clampedValue,
      "aria-valuetext": ariaValueText,
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-disabled": disabled || void 0,
      tabIndex: -1,
      id,
      "data-testid": dataTestId,
      "data-part": "root",
      className: cn(
        "w-full",
        disabled && "opacity-40 pointer-events-none",
        className
      ),
      style,
      children: [
        !hideValue && /* @__PURE__ */ jsx(
          "span",
          {
            "data-part": "value",
            className: "block text-sm text-sapphire-text-secondary leading-normal",
            children: valueText
          }
        ),
        /* @__PURE__ */ jsxs(
          "div",
          {
            "data-part": "indicator",
            className: cn("flex items-start w-full", hasIcon && "gap-1"),
            children: [
              /* @__PURE__ */ jsxs("div", { "data-part": "track-container", className: "relative h-4 flex-1", children: [
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    "data-part": "track",
                    className: cn(
                      "absolute left-0 right-0 top-1/2 -translate-y-1/2 h-1.5 rounded",
                      trackBgClass[vs] || trackBgClass["None" /* None */]
                    )
                  }
                ),
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    "data-part": "bar",
                    className: cn(
                      "absolute top-1/2 -translate-y-1/2 h-2.5 rounded-full transition-[width] duration-300 ease-linear",
                      barColorClass[vs] || barColorClass["None" /* None */],
                      clampedValue === 0 && "w-0"
                    ),
                    style: clampedValue > 0 ? { width: `${clampedValue}%` } : void 0
                  }
                ),
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    "data-part": "start-cap",
                    className: cn(
                      "absolute left-px top-1/2 -translate-y-1/2 size-1 rounded-full",
                      barColorClass[vs] || barColorClass["None" /* None */]
                    )
                  }
                ),
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    "data-part": "end-cap",
                    className: cn(
                      "absolute right-px top-1/2 -translate-y-1/2 size-1 rounded-full",
                      barColorClass[vs] || barColorClass["None" /* None */]
                    )
                  }
                )
              ] }),
              hasIcon && /* @__PURE__ */ jsx(
                StateIcon,
                {
                  "data-part": "icon",
                  className: cn("h-4 w-4 shrink-0", iconColorClass[vs])
                }
              )
            ]
          }
        )
      ]
    }
  );
}

export { ProgressIndicator, ProgressIndicatorValueState };
