// @sap-ui/fx-components/dist/chunk-XU2XVLPF.js
import { LessIcon } from './chunk-YWNUKDTJ.js';
import { AcceptIcon } from './chunk-VNDUTC7E.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useState, useRef, useId, useCallback } from 'react';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';

// src/types/switch.ts
var SwitchDesign = /* @__PURE__ */ ((SwitchDesign2) => {
  SwitchDesign2["Textual"] = "Textual";
  SwitchDesign2["Graphical"] = "Graphical";
  return SwitchDesign2;
})(SwitchDesign || {});
var switchContainerBase = "relative inline-flex items-center rounded-full transition-colors duration-200 ease-in-out cursor-pointer focus:outline-none box-border";
var switchHandleBase = "absolute rounded-full transition-transform duration-200 ease-in-out flex items-center justify-center box-border";
function Switch({
  checked: controlledChecked,
  defaultChecked = false,
  disabled = false,
  required = false,
  design = "Textual" /* Textual */,
  textOn,
  textOff,
  iconOn,
  iconOff,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  tooltip,
  name,
  value = "on",
  id,
  onChange,
  className,
  style,
  "data-testid": dataTestId,
  accessibilityAttributes,
  ref
}) {
  const [internalChecked, setInternalChecked] = useState(defaultChecked);
  const textOnRef = useRef(null);
  const textOffRef = useRef(null);
  const [textWidths, setTextWidths] = useState({ on: 0, off: 0 });
  const generatedId = useId();
  const isControlled = controlledChecked !== void 0;
  const checked = isControlled ? controlledChecked : internalChecked;
  const descriptionId = `${id ?? generatedId}-desc`;
  const ariaDescribedBy = [accessibleDescriptionRef, accessibleDescription ? descriptionId : null, accessibilityAttributes?.describedBy].filter(Boolean).join(" ") || void 0;
  const isGraphical = design === "Graphical" /* Graphical */;
  const hasText = !isGraphical && (textOn || textOff);
  const measureTextWidths = useCallback(() => {
    const onWidth = textOnRef.current?.offsetWidth || 0;
    const offWidth = textOffRef.current?.offsetWidth || 0;
    if (onWidth !== textWidths.on || offWidth !== textWidths.off) {
      setTextWidths({ on: onWidth, off: offWidth });
    }
  }, [textWidths.on, textWidths.off]);
  const textOnCallbackRef = useCallback((node) => {
    textOnRef.current = node;
    if (node) measureTextWidths();
  }, [measureTextWidths]);
  const textOffCallbackRef = useCallback((node) => {
    textOffRef.current = node;
    if (node) measureTextWidths();
  }, [measureTextWidths]);
  const handlePadding = 12;
  const borderWidth = 2;
  const handleWidthOn = hasText ? textWidths.on + handlePadding : 14;
  const handleWidthOff = hasText ? textWidths.off + handlePadding : 14;
  const maxHandleWidth = Math.max(handleWidthOn, handleWidthOff);
  const containerWidth = hasText ? Math.ceil(maxHandleWidth * 1.6) : 32;
  const currentHandleWidth = checked ? handleWidthOn : handleWidthOff;
  const handleTranslateX = hasText ? checked ? containerWidth - currentHandleWidth - borderWidth * 2 : 0 : checked ? 10 : 0;
  const handleToggle = useCallback(() => {
    if (disabled) return;
    const newChecked = !checked;
    if (!isControlled) {
      setInternalChecked(newChecked);
    }
    onChange?.({ checked: newChecked });
  }, [checked, disabled, isControlled, onChange]);
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === " " || e.key === "Enter") {
        e.preventDefault();
        handleToggle();
      }
    },
    [handleToggle]
  );
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsxs(
      "button",
      {
        ref,
        id,
        type: "button",
        role: "switch",
        "aria-checked": checked,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-describedby": ariaDescribedBy,
        "aria-required": required,
        "aria-disabled": disabled,
        title: tooltip,
        disabled,
        onClick: handleToggle,
        onKeyDown: handleKeyDown,
        "data-testid": dataTestId,
        className: cn(
          switchContainerBase,
          "fx-switch",
          className
        ),
        style: {
          ...style,
          backgroundColor: "var(--_switch-bg)",
          width: hasText ? `${containerWidth}px` : "32px",
          height: "20px"
        },
        children: [
          hasText && /* @__PURE__ */ jsxs(Fragment, { children: [
            /* @__PURE__ */ jsx(
              "span",
              {
                ref: textOnCallbackRef,
                className: "absolute invisible text-xs font-medium px-1.5",
                "aria-hidden": "true",
                children: textOn
              }
            ),
            /* @__PURE__ */ jsx(
              "span",
              {
                ref: textOffCallbackRef,
                className: "absolute invisible text-xs font-medium px-1.5",
                "aria-hidden": "true",
                children: textOff
              }
            )
          ] }),
          /* @__PURE__ */ jsx(
            "span",
            {
              className: switchHandleBase,
              style: {
                transform: `translateX(${handleTranslateX}px)`,
                width: "14px",
                height: "14px",
                top: "3px",
                left: "4px",
                backgroundColor: "var(--_switch-handle-bg)",
                boxShadow: "0 0 0 1px var(--_switch-handle-border)"
              },
              "aria-hidden": "true",
              children: isGraphical ? /* @__PURE__ */ jsx(Fragment, { children: checked ? iconOn || /* @__PURE__ */ jsx(AcceptIcon, { className: "h-3 w-3", style: { color: "var(--_switch-bg)" } }) : iconOff || /* @__PURE__ */ jsx(LessIcon, { className: "h-3 w-3" }) }) : hasText ? /* @__PURE__ */ jsx("span", { className: "whitespace-nowrap", children: checked ? textOn : textOff }) : null
            }
          )
        ]
      }
    ),
    name && /* @__PURE__ */ jsx(
      "input",
      {
        type: "checkbox",
        name,
        value,
        checked,
        onChange: () => {
        },
        required,
        disabled,
        className: "sr-only",
        tabIndex: -1,
        "aria-hidden": "true"
      }
    ),
    accessibleDescription && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription })
  ] });
}

export { Switch, SwitchDesign };
