// @sap-ui/fx-components/dist/chunk-TNAICDBU.js
import { valueStateIcons, valueStateLabelKeys, valueStateMessageColors } from './chunk-WWAPKHIM.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useId, useState, useImperativeHandle, useEffect, useCallback, useMemo, useLayoutEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';

// src/types/input.ts
var InputSize = /* @__PURE__ */ ((InputSize2) => {
  InputSize2["Large"] = "Large";
  InputSize2["Small"] = "Small";
  return InputSize2;
})(InputSize || {});
var InputType = /* @__PURE__ */ ((InputType2) => {
  InputType2["Text"] = "Text";
  InputType2["Email"] = "Email";
  InputType2["Number"] = "Number";
  InputType2["Password"] = "Password";
  InputType2["Tel"] = "Tel";
  InputType2["URL"] = "URL";
  InputType2["Search"] = "Search";
  return InputType2;
})(InputType || {});
var InputFilter = /* @__PURE__ */ ((InputFilter2) => {
  InputFilter2["StartsWith"] = "StartsWith";
  InputFilter2["StartsWithPerTerm"] = "StartsWithPerTerm";
  InputFilter2["Contains"] = "Contains";
  InputFilter2["None"] = "None";
  return InputFilter2;
})(InputFilter || {});
var inputContainerVariants = cva(
  [
    "flex items-center w-full border bg-sapphire-canvas-primary",
    "overflow-clip transition-colors relative"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: [
          "h-10 rounded-lg min-w-[2.75rem]",
          "pl-3.5 pr-1 py-1 gap-2",
          "focus-within:border-2 focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]"
        ],
        ["Small" /* Small */]: [
          "h-8 rounded min-w-[2rem]",
          "pl-3 pr-1 py-1 gap-2",
          "focus-within:border-2 focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]"
        ]
      },
      valueState: {
        ["None" /* None */]: "border-sapphire-border-active hover:border-sapphire-border-accent focus-within:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-positive focus-within:border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-negative focus-within:border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-warning focus-within:border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg hover:bg-sapphire-canvas-primary focus-within:bg-sapphire-canvas-primary border-sapphire-info focus-within:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      readonly: {
        true: "bg-sapphire-background-secondary border-sapphire-border-primary hover:border-sapphire-border-primary focus-within:!border-sapphire-border-accent",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
function mapInputType(type) {
  switch (type) {
    case "Text" /* Text */:
      return "text";
    case "Email" /* Email */:
      return "email";
    case "Number" /* Number */:
      return "number";
    case "Password" /* Password */:
      return "password";
    case "Tel" /* Tel */:
      return "tel";
    case "URL" /* URL */:
      return "url";
    case "Search" /* Search */:
      return "search";
    default:
      return "text";
  }
}
function InputBase({
  // Value
  value: controlledValue,
  defaultValue = "",
  // Type
  type = "Text" /* Text */,
  // Size
  size = "Large" /* Large */,
  // Behavior
  disabled = false,
  readonly = false,
  required = false,
  autocomplete,
  // Validation
  valueState = "None" /* None */,
  valueStateMessage,
  showValueStateIcon = false,
  // Constraints
  maxLength,
  minLength,
  pattern,
  min,
  max,
  step,
  // Display
  placeholder,
  showClearIcon = false,
  hideStepButtons = false,
  // Icons
  icon,
  iconPosition = "start",
  // Accessibility
  accessibleName,
  accessibleNameRef,
  accessibleDescriptionRef,
  // Form
  name,
  id: providedId,
  // Events
  onInput,
  onChange,
  onFocus,
  onBlur,
  onKeyDown,
  onKeyUp,
  // Standard
  className,
  style,
  "data-testid": dataTestId,
  "data-ai-field": dataAiField,
  accessibilityAttributes,
  ref
}) {
  const { t } = useTranslation("fx");
  const inputRef = useRef(null);
  const generatedId = useId();
  const inputId = providedId ?? generatedId;
  const messageId = `${inputId}-message`;
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [previousValue, setPreviousValue] = useState(defaultValue);
  const [valueOnFocus, setValueOnFocus] = useState(defaultValue);
  const [isFocused, setIsFocused] = useState(false);
  const [isComposing, setIsComposing] = useState(false);
  const value = controlledValue ?? internalValue;
  const isControlled = controlledValue !== void 0;
  const isNumberType = String(type) === "Number";
  const normalizedValueState = valueState;
  const isNegativeState = normalizedValueState === "Negative" /* Negative */;
  useImperativeHandle(
    ref,
    () => ({
      focus: () => inputRef.current?.focus(),
      blur: () => inputRef.current?.blur(),
      select: () => inputRef.current?.select(),
      setSelectionRange: (start, end) => inputRef.current?.setSelectionRange(start, end),
      getValue: () => value,
      getValueAsNumber: () => inputRef.current?.valueAsNumber ?? parseFloat(value) ?? NaN,
      setValue: (newValue, options) => {
        if (!isControlled) {
          setInternalValue(newValue);
        }
        if (!options?.silent) {
          onInput?.(newValue);
          onChange?.({
            value: newValue,
            previousValue: value,
            source: "programmatic"
          });
        }
      },
      clear: () => {
        if (!isControlled) {
          setInternalValue("");
        }
        onInput?.("");
        onChange?.({
          value: "",
          previousValue: value,
          source: "programmatic"
        });
      },
      getState: () => ({
        value,
        focused: isFocused,
        valid: inputRef.current?.checkValidity() ?? true,
        valueState
      }),
      checkValidity: () => inputRef.current?.checkValidity() ?? true,
      reportValidity: () => inputRef.current?.reportValidity() ?? true,
      setCustomValidity: (message) => inputRef.current?.setCustomValidity(message),
      get nativeElement() {
        return inputRef.current;
      }
    }),
    [value, isFocused, valueState, isControlled, onInput, onChange]
  );
  useEffect(() => {
    if (isControlled) {
      setInternalValue(controlledValue);
    }
  }, [controlledValue, isControlled]);
  const handleChange = useCallback(
    (e) => {
      const newValue = e.target.value;
      if (!isControlled) {
        setInternalValue(newValue);
      }
      onInput?.(newValue);
    },
    [isControlled, onInput]
  );
  const handleFocus = useCallback(
    (e) => {
      setIsFocused(true);
      setPreviousValue(value);
      setValueOnFocus(value);
      onFocus?.(e);
    },
    [value, onFocus]
  );
  const handleBlur = useCallback(
    (e) => {
      setIsFocused(false);
      const currentValue = e.target.value;
      if (currentValue !== valueOnFocus) {
        onChange?.({
          value: currentValue,
          previousValue: valueOnFocus,
          source: "user"
        });
        setPreviousValue(currentValue);
      }
      onBlur?.(e);
    },
    [valueOnFocus, onChange, onBlur]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (isComposing) {
        onKeyDown?.(e);
        return;
      }
      if (e.key === "Escape") {
        if (showClearIcon && value && !disabled && !readonly) {
          e.preventDefault();
          const oldValue = value;
          if (!isControlled) {
            setInternalValue("");
          }
          onInput?.("");
          onChange?.({
            value: "",
            previousValue: oldValue,
            source: "user"
          });
        } else {
          if (!isControlled) {
            setInternalValue(valueOnFocus);
          }
          onInput?.(valueOnFocus);
          setPreviousValue(valueOnFocus);
          inputRef.current?.blur();
        }
      }
      if (e.key === "Enter") {
        inputRef.current?.blur();
      }
      onKeyDown?.(e);
    },
    [value, previousValue, valueOnFocus, isControlled, isComposing, showClearIcon, disabled, readonly, onChange, onInput, onKeyDown]
  );
  const handleCompositionStart = useCallback(() => {
    setIsComposing(true);
  }, []);
  const handleCompositionEnd = useCallback(
    (e) => {
      setIsComposing(false);
      const newValue = e.currentTarget.value;
      if (!isControlled) {
        setInternalValue(newValue);
      }
      onInput?.(newValue);
    },
    [isControlled, onInput]
  );
  const handleClear = useCallback(
    (detail) => {
      detail.originalEvent.preventDefault();
      detail.originalEvent.stopPropagation();
      const oldValue = value;
      if (!isControlled) {
        setInternalValue("");
      }
      onInput?.("");
      onChange?.({
        value: "",
        previousValue: oldValue,
        source: "user"
      });
      inputRef.current?.focus();
    },
    [value, isControlled, onInput, onChange]
  );
  const clearVisible = showClearIcon && !!value && !disabled && !readonly;
  const clearReserveSpace = showClearIcon && !disabled && !readonly && isFocused && !value;
  const showClear = clearVisible || clearReserveSpace;
  const ValueStateIcon = valueStateIcons[valueState];
  const showStateIcon = showValueStateIcon && ValueStateIcon;
  const ariaDescribedBy = useMemo(() => {
    const ids = [];
    if (valueStateMessage) ids.push(messageId);
    if (accessibleDescriptionRef) ids.push(accessibleDescriptionRef);
    return ids.length > 0 ? ids.join(" ") : void 0;
  }, [valueStateMessage, accessibleDescriptionRef, messageId]);
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn("w-full", className),
      style,
      "data-testid": dataTestId,
      "data-ai-field": dataAiField,
      children: [
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: cn(
              inputContainerVariants({
                size,
                valueState,
                disabled,
                readonly
              }),
              hideStepButtons && "[&_input::-webkit-inner-spin-button]:appearance-none [&_input::-webkit-outer-spin-button]:appearance-none [&_input]:[-moz-appearance:textfield]"
            ),
            children: [
              icon && iconPosition === "start" && /* @__PURE__ */ jsx("span", { className: "flex items-center justify-center shrink-0 text-sapphire-text-tertiary", children: icon }),
              /* @__PURE__ */ jsx(
                "input",
                {
                  ref: inputRef,
                  id: inputId,
                  name,
                  type: mapInputType(type),
                  value,
                  placeholder,
                  disabled,
                  readOnly: readonly,
                  required,
                  autoComplete: autocomplete,
                  maxLength: isNumberType ? void 0 : maxLength,
                  minLength,
                  pattern,
                  min: isNumberType ? min : void 0,
                  max: isNumberType ? max : void 0,
                  step: isNumberType ? step : void 0,
                  role: accessibilityAttributes?.role,
                  "aria-label": accessibleName,
                  "aria-labelledby": accessibleNameRef,
                  "aria-describedby": ariaDescribedBy,
                  "aria-invalid": isNegativeState || void 0,
                  "aria-required": required || void 0,
                  "aria-disabled": disabled || void 0,
                  "aria-readonly": readonly || void 0,
                  "aria-controls": accessibilityAttributes?.controls,
                  "aria-haspopup": accessibilityAttributes?.hasPopup || void 0,
                  "aria-expanded": accessibilityAttributes?.expanded,
                  "aria-autocomplete": accessibilityAttributes?.autoComplete,
                  "aria-activedescendant": accessibilityAttributes?.activeDescendant,
                  onChange: handleChange,
                  onFocus: handleFocus,
                  onBlur: handleBlur,
                  onKeyDown: handleKeyDown,
                  onKeyUp,
                  onCompositionStart: handleCompositionStart,
                  onCompositionEnd: handleCompositionEnd,
                  className: cn(
                    "flex-1 h-full bg-transparent text-sm text-sapphire-text-primary",
                    "outline-none min-w-0",
                    "placeholder:text-sapphire-text-tertiary placeholder:italic",
                    "disabled:cursor-not-allowed",
                    readonly && "text-sapphire-text-tertiary",
                    "file:border-0 file:bg-transparent file:text-sm file:font-medium"
                  )
                }
              ),
              showClear && /* @__PURE__ */ jsx(
                "span",
                {
                  onMouseDown: (e) => e.preventDefault(),
                  className: cn(!clearVisible && "invisible"),
                  children: /* @__PURE__ */ jsx(
                    Button,
                    {
                      design: "SecondaryNeutral" /* SecondaryNeutral */,
                      size: size === "Small" /* Small */ ? "Small" /* Small */ : "Medium" /* Medium */,
                      iconOnly: true,
                      icon: /* @__PURE__ */ jsx(DeclineIcon, { className: size === "Small" /* Small */ ? "h-3 w-3" : "h-4 w-4" }),
                      tabIndex: -1,
                      onClick: handleClear,
                      accessibleName: t("INPUT_CLEAR"),
                      className: "shrink-0"
                    }
                  )
                }
              ),
              icon && iconPosition === "end" && /* @__PURE__ */ jsx("span", { className: "flex items-center justify-center shrink-0 text-sapphire-text-tertiary", children: icon })
            ]
          }
        ),
        valueStateMessage && /* @__PURE__ */ jsxs(
          "div",
          {
            id: messageId,
            role: isNegativeState ? "alert" : void 0,
            "aria-live": isNegativeState ? "assertive" : "polite",
            className: cn(
              "mt-2 text-sm flex items-start gap-1",
              valueStateMessageColors[valueState]
            ),
            children: [
              normalizedValueState !== "None" /* None */ && valueStateLabelKeys[normalizedValueState] && /* @__PURE__ */ jsxs("span", { className: "sr-only", children: [
                t(valueStateLabelKeys[normalizedValueState]),
                ":",
                " "
              ] }),
              showStateIcon && ValueStateIcon && /* @__PURE__ */ jsx(ValueStateIcon, { className: "h-4 w-4 shrink-0 mt-0.5" }),
              /* @__PURE__ */ jsx("span", { children: valueStateMessage })
            ]
          }
        )
      ]
    }
  );
}
var Input = React.memo(InputBase);
Input.displayName = "Input";
function normalizeItem(item) {
  if (typeof item === "string") {
    return { text: item, value: item };
  }
  return { ...item, value: item.value ?? item.text };
}
function isGroup(item) {
  return typeof item === "object" && "headerText" in item && "items" in item;
}
function filterItem(item, filterValue, filterMode) {
  if (!filterValue || filterMode === "None" /* None */) {
    return true;
  }
  const normalizedText = item.text.toLowerCase();
  const normalizedFilter = filterValue.toLowerCase();
  switch (filterMode) {
    case "StartsWith" /* StartsWith */:
      return normalizedText.startsWith(normalizedFilter);
    case "Contains" /* Contains */:
      return normalizedText.includes(normalizedFilter);
    case "StartsWithPerTerm" /* StartsWithPerTerm */:
    default: {
      const words = normalizedText.split(/\s+/);
      return words.some((word) => word.startsWith(normalizedFilter));
    }
  }
}
function highlightText(text, match, highlightClass = "font-semibold bg-sapphire-warning-bg") {
  if (!match) return text;
  const lowerText = text.toLowerCase();
  const lowerMatch = match.toLowerCase();
  const index = lowerText.indexOf(lowerMatch);
  if (index === -1) return text;
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    text.slice(0, index),
    /* @__PURE__ */ jsx("span", { className: highlightClass, children: text.slice(index, index + match.length) }),
    text.slice(index + match.length)
  ] });
}
function InputSuggestions({
  suggestions = [],
  filter = "StartsWith" /* StartsWith */,
  noTypeahead = false,
  highlightMatch = false,
  suggestionRenderer,
  onSuggestionSelect,
  onSuggestionsOpen,
  onSuggestionsClose,
  showNoSuggestionsMessage = true,
  noSuggestionsMessage,
  // Pass through to Input
  value: controlledValue,
  defaultValue = "",
  onInput,
  onChange,
  onFocus,
  onBlur,
  onKeyDown,
  disabled,
  readonly,
  ref,
  ...inputProps
}) {
  const { t } = useTranslation("fx");
  const inputRef = useRef(null);
  const containerRef = useRef(null);
  const popoverContentRef = useRef(null);
  const baseId = useId();
  const listboxId = `${baseId}-listbox`;
  const optionIdPrefix = `${baseId}-option-`;
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [userTypedValue, setUserTypedValue] = useState(defaultValue);
  const [highlightedIndex, setHighlightedIndex] = useState(-1);
  const [hasFocus, setHasFocus] = useState(false);
  const [hasTyped, setHasTyped] = useState(false);
  const [dismissed, setDismissed] = useState(false);
  const [manualOpen, setManualOpen] = useState(false);
  const shouldAutocompleteRef = useRef(true);
  const value = controlledValue ?? internalValue;
  const isControlled = controlledValue !== void 0;
  const filterValue = userTypedValue;
  const flattenedItems = useMemo(() => {
    const result = [];
    for (const suggestion of suggestions) {
      if (isGroup(suggestion)) {
        for (const item of suggestion.items) {
          result.push({
            item: normalizeItem(item),
            groupHeader: suggestion.headerText
          });
        }
      } else {
        result.push({ item: normalizeItem(suggestion) });
      }
    }
    return result;
  }, [suggestions]);
  const filteredItems = useMemo(() => {
    return flattenedItems.filter(
      ({ item }) => filterItem(item, filterValue, filter) && !item.disabled
    );
  }, [flattenedItems, filterValue, filter]);
  const itemsWithHeaders = useMemo(() => {
    const result = [];
    let lastGroup;
    let itemIndex = 0;
    for (const { item, groupHeader } of filteredItems) {
      if (groupHeader && groupHeader !== lastGroup) {
        result.push({ type: "header", content: groupHeader });
        lastGroup = groupHeader;
      }
      result.push({ type: "item", content: item, index: itemIndex });
      itemIndex++;
    }
    return result;
  }, [filteredItems]);
  const shouldShowSuggestions = hasFocus && hasTyped && !disabled && !readonly && (filteredItems.length > 0 || showNoSuggestionsMessage);
  const isOpen = manualOpen || shouldShowSuggestions && !dismissed;
  useImperativeHandle(
    ref,
    () => ({
      ...inputRef.current,
      openSuggestions: () => {
        setManualOpen(true);
        onSuggestionsOpen?.();
      },
      closeSuggestions: () => {
        setManualOpen(false);
        setDismissed(true);
        setHighlightedIndex(-1);
        onSuggestionsClose?.();
      },
      isSuggestionsOpen: () => isOpen
    }),
    [isOpen, onSuggestionsOpen, onSuggestionsClose]
  );
  const prevOpenRef = useRef(false);
  useEffect(() => {
    if (isOpen && !prevOpenRef.current) {
      onSuggestionsOpen?.();
    } else if (!isOpen && prevOpenRef.current) {
      onSuggestionsClose?.();
    }
    prevOpenRef.current = isOpen;
  }, [isOpen, onSuggestionsOpen, onSuggestionsClose]);
  const pendingSelectionRef = useRef(null);
  const previewSuggestion = useCallback(
    (item) => {
      const lowerTyped = filterValue.toLowerCase();
      const lowerMatch = item.text.toLowerCase();
      if (lowerMatch.startsWith(lowerTyped) && lowerMatch !== lowerTyped) {
        const completedValue = filterValue + item.text.substring(filterValue.length);
        if (!isControlled) {
          setInternalValue(completedValue);
        }
        pendingSelectionRef.current = { start: filterValue.length, end: completedValue.length };
      }
    },
    [filterValue, isControlled]
  );
  const restoreTypedValue = useCallback(() => {
    if (!isControlled) {
      setInternalValue(filterValue);
    }
    pendingSelectionRef.current = null;
  }, [filterValue, isControlled]);
  useEffect(() => {
    if (noTypeahead || !shouldAutocompleteRef.current || !hasFocus || filteredItems.length === 0 || !filterValue) return;
    const firstMatch = filteredItems[0].item;
    const lowerTyped = filterValue.toLowerCase();
    const lowerMatch = firstMatch.text.toLowerCase();
    if (lowerMatch.startsWith(lowerTyped) && lowerMatch !== lowerTyped) {
      const completedValue = filterValue + firstMatch.text.substring(filterValue.length);
      if (!isControlled) {
        setInternalValue(completedValue);
      }
      pendingSelectionRef.current = { start: filterValue.length, end: completedValue.length };
    }
    shouldAutocompleteRef.current = false;
  }, [filterValue, filteredItems, noTypeahead, hasFocus, isControlled]);
  useLayoutEffect(() => {
    const pending = pendingSelectionRef.current;
    if (!pending) return;
    pendingSelectionRef.current = null;
    const nativeInput = inputRef.current?.nativeElement;
    if (nativeInput) {
      nativeInput.setSelectionRange(pending.start, pending.end);
    }
  });
  useEffect(() => {
    if (highlightedIndex >= 0 && popoverContentRef.current) {
      const items = popoverContentRef.current.querySelectorAll("[data-suggestion-item]");
      const item = items[highlightedIndex];
      item?.scrollIntoView({ block: "nearest" });
    }
  }, [highlightedIndex]);
  const handleInput = useCallback(
    (newValue) => {
      setUserTypedValue(newValue);
      setHasTyped(true);
      setDismissed(false);
      if (!isControlled) {
        setInternalValue(newValue);
      }
      setHighlightedIndex(-1);
      onInput?.(newValue);
    },
    [isControlled, onInput]
  );
  const handleFocus = useCallback(
    (e) => {
      setHasFocus(true);
      setHasTyped(false);
      setDismissed(false);
      onFocus?.(e);
    },
    [onFocus]
  );
  const handleBlur = useCallback(
    (e) => {
      if (!isControlled && internalValue !== userTypedValue) {
        setInternalValue(userTypedValue);
      }
      setTimeout(() => {
        setHasFocus(false);
        setHasTyped(false);
        setHighlightedIndex(-1);
      }, 150);
      onBlur?.(e);
    },
    [onBlur, isControlled, internalValue, userTypedValue]
  );
  const selectItem = useCallback(
    (item) => {
      const newValue = item.value ?? item.text;
      setUserTypedValue(newValue);
      if (!isControlled) {
        setInternalValue(newValue);
      }
      setDismissed(true);
      setManualOpen(false);
      setHighlightedIndex(-1);
      onInput?.(newValue);
      onSuggestionSelect?.({ item, value: newValue });
      inputRef.current?.focus();
    },
    [isControlled, onInput, onSuggestionSelect]
  );
  const tryAcceptTypeahead = useCallback(() => {
    if (noTypeahead || filteredItems.length === 0) return false;
    const firstMatch = filteredItems[0].item;
    if (firstMatch.text.toLowerCase().startsWith(filterValue.toLowerCase())) {
      selectItem(firstMatch);
      return true;
    }
    return false;
  }, [noTypeahead, filteredItems, filterValue, selectItem]);
  const dismissSuggestions = useCallback(() => {
    setDismissed(true);
    setManualOpen(false);
    setHighlightedIndex(-1);
  }, []);
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === "F4") {
        e.preventDefault();
        if (isOpen) {
          dismissSuggestions();
          onSuggestionsClose?.();
        } else {
          setDismissed(false);
          setHasTyped(true);
        }
        onKeyDown?.(e);
        return;
      }
      shouldAutocompleteRef.current = !noTypeahead && e.key !== "Backspace" && e.key !== "Delete" && e.key !== "Escape";
      if (!isOpen) {
        onKeyDown?.(e);
        return;
      }
      switch (e.key) {
        case "ArrowDown": {
          e.preventDefault();
          const nextIndex = highlightedIndex < filteredItems.length - 1 ? highlightedIndex + 1 : highlightedIndex;
          setHighlightedIndex(nextIndex);
          if (nextIndex >= 0 && filteredItems[nextIndex]) {
            previewSuggestion(filteredItems[nextIndex].item);
          }
          break;
        }
        case "ArrowUp": {
          e.preventDefault();
          const nextIndex = highlightedIndex > 0 ? highlightedIndex - 1 : -1;
          setHighlightedIndex(nextIndex);
          if (nextIndex >= 0 && filteredItems[nextIndex]) {
            previewSuggestion(filteredItems[nextIndex].item);
          } else {
            restoreTypedValue();
          }
          break;
        }
        case "ArrowRight": {
          const nativeInput = inputRef.current?.nativeElement;
          const hasTypeaheadSelection = nativeInput && nativeInput.selectionStart !== nativeInput.selectionEnd && nativeInput.selectionStart === filterValue.length;
          if (hasTypeaheadSelection && tryAcceptTypeahead()) {
            e.preventDefault();
          }
          break;
        }
        case "Enter":
          if (highlightedIndex >= 0 && filteredItems[highlightedIndex]) {
            e.preventDefault();
            selectItem(filteredItems[highlightedIndex].item);
          } else if (tryAcceptTypeahead()) {
            e.preventDefault();
          }
          break;
        case "Escape":
          e.preventDefault();
          dismissSuggestions();
          onSuggestionsClose?.();
          break;
        case "Tab":
          dismissSuggestions();
          break;
      }
      onKeyDown?.(e);
    },
    [isOpen, highlightedIndex, filteredItems, selectItem, onKeyDown, onSuggestionsClose, noTypeahead, filterValue, previewSuggestion, restoreTypedValue, tryAcceptTypeahead, dismissSuggestions]
  );
  const handleItemClick = useCallback(
    (item) => (e) => {
      e.preventDefault();
      selectItem(item);
    },
    [selectItem]
  );
  const handlePopoverClose = useCallback(() => {
    dismissSuggestions();
  }, [dismissSuggestions]);
  const srAnnouncement = useMemo(() => {
    if (!isOpen) return "";
    if (filteredItems.length === 0) return t("INPUT_SUGGESTIONS_NONE");
    if (filteredItems.length === 1) return t("INPUT_SUGGESTIONS_ONE");
    return t("INPUT_SUGGESTIONS_AVAILABLE", { count: filteredItems.length });
  }, [isOpen, filteredItems.length, t]);
  const activeDescendantId = isOpen && highlightedIndex >= 0 ? `${optionIdPrefix}${highlightedIndex}` : void 0;
  const renderItem = (item, index) => {
    if (suggestionRenderer) {
      return suggestionRenderer(item, index);
    }
    return /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 w-full", children: [
      item.icon && /* @__PURE__ */ jsx("span", { className: "flex-shrink-0", children: item.icon }),
      /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0", children: [
        /* @__PURE__ */ jsx("div", { className: "truncate", children: highlightMatch ? highlightText(item.text, filterValue) : item.text }),
        item.additionalText && /* @__PURE__ */ jsx("div", { className: "text-xs text-sapphire-text-tertiary truncate", children: item.additionalText })
      ] })
    ] });
  };
  const noSuggestionsText = noSuggestionsMessage ?? t("INPUT_SUGGESTIONS_NONE");
  return /* @__PURE__ */ jsxs("div", { ref: containerRef, className: "relative w-full", children: [
    /* @__PURE__ */ jsx(
      Input,
      {
        ref: inputRef,
        value,
        onInput: handleInput,
        onChange,
        onFocus: handleFocus,
        onBlur: handleBlur,
        onKeyDown: handleKeyDown,
        disabled,
        readonly,
        accessibilityAttributes: {
          expanded: isOpen || void 0,
          hasPopup: "listbox",
          autoComplete: "list",
          controls: isOpen ? listboxId : void 0,
          activeDescendant: activeDescendantId
        },
        ...inputProps
      }
    ),
    /* @__PURE__ */ jsx("div", { className: "sr-only", "aria-live": "polite", "aria-atomic": "true", children: srAnnouncement }),
    isOpen && /* @__PURE__ */ jsx(
      ResponsivePopover,
      {
        open: isOpen,
        opener: containerRef,
        placement: "Bottom",
        horizontalAlign: "Stretch",
        hideArrow: true,
        preventInitialFocus: true,
        preventFocusRestore: true,
        noPadding: true,
        contentOnlyOnDesktop: true,
        onClose: handlePopoverClose,
        children: /* @__PURE__ */ jsx("div", { ref: popoverContentRef, children: filteredItems.length === 0 ? /* @__PURE__ */ jsx("div", { className: "px-4 py-2 text-sm text-sapphire-text-tertiary", children: noSuggestionsText }) : /* @__PURE__ */ jsx(
          List,
          {
            id: listboxId,
            accessibleRole: "ListBox" /* ListBox */,
            separators: "None" /* None */,
            children: itemsWithHeaders.map((entry, idx) => {
              if (entry.type === "header") {
                return /* @__PURE__ */ jsx(
                  "li",
                  {
                    className: "px-4 py-1.5 text-xs font-semibold text-sapphire-text-tertiary uppercase tracking-wide",
                    "aria-hidden": "true",
                    children: entry.content
                  },
                  `header-${idx}`
                );
              }
              const item = entry.content;
              const itemIndex = entry.index;
              const isHighlighted = itemIndex === highlightedIndex;
              return /* @__PURE__ */ jsx(
                "li",
                {
                  id: `${optionIdPrefix}${itemIndex}`,
                  "data-suggestion-item": true,
                  role: "option",
                  "aria-selected": isHighlighted,
                  className: cn(
                    "flex items-center min-h-[2.5rem] px-4 text-sm cursor-pointer",
                    "bg-sapphire-canvas-primary text-sapphire-text-primary",
                    "transition-colors",
                    "hover:bg-sapphire-background-tertiary",
                    isHighlighted && "bg-sapphire-background-tertiary"
                  ),
                  onClick: handleItemClick(item),
                  onMouseEnter: () => setHighlightedIndex(itemIndex),
                  children: renderItem(item, itemIndex)
                },
                `item-${itemIndex}`
              );
            })
          }
        ) })
      }
    )
  ] });
}

export { Input, InputFilter, InputSize, InputSuggestions, InputType };
