// @sap-ui/fx-components/dist/chunk-4PDXVETU.js
import { cn } from './chunk-2F2PZMJ5.js';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/label.ts
var LabelWrappingType = /* @__PURE__ */ ((LabelWrappingType2) => {
  LabelWrappingType2["None"] = "None";
  LabelWrappingType2["Normal"] = "Normal";
  return LabelWrappingType2;
})(LabelWrappingType || {});
function Label({
  id,
  for: forProp,
  htmlFor,
  showColon = false,
  required = false,
  wrappingType = "Normal" /* Normal */,
  children,
  text,
  className,
  style,
  "data-testid": dataTestId,
  ref
}) {
  const labelFor = forProp ?? htmlFor;
  const labelContent = children ?? text;
  return /* @__PURE__ */ jsxs(
    "label",
    {
      ref,
      id,
      htmlFor: labelFor,
      className: cn(
        "text-sm font-normal text-sapphire-text-tertiary",
        wrappingType === "None" /* None */ && "truncate",
        wrappingType === "Normal" /* Normal */ && "break-words",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        labelContent,
        showColon && ":",
        required && /* @__PURE__ */ jsx("span", { className: "text-destructive ml-0.5", "aria-hidden": "true", children: "*" }),
        required && /* @__PURE__ */ jsx("span", { className: "sr-only", children: "(required)" })
      ]
    }
  );
}

export { Label, LabelWrappingType };
