// @sap-ui/fx-components/dist/chunk-EEGGRFXR.js
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { Loader2 } from 'lucide-react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

function ComboBoxPopover({
  open,
  triggerRef,
  hasValueState = false,
  valueStatePortalRef,
  headerContent,
  loading = false,
  mobileTitle = "Suggestions",
  onBeforeClose,
  onClose,
  children,
  className,
  ref
}) {
  return /* @__PURE__ */ jsx(
    ResponsivePopover,
    {
      ref,
      open,
      opener: triggerRef,
      placement: "Bottom",
      horizontalAlign: "Stretch",
      hideArrow: true,
      preventInitialFocus: true,
      preventFocusRestore: true,
      noPadding: true,
      headerText: mobileTitle,
      contentOnlyOnDesktop: true,
      onBeforeClose,
      onClose,
      className,
      children: /* @__PURE__ */ jsxs("div", { className: "flex flex-col", children: [
        hasValueState && /* @__PURE__ */ jsx("div", { ref: valueStatePortalRef }),
        loading ? /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-center py-8", children: [
          /* @__PURE__ */ jsx(Loader2, { className: "h-5 w-5 animate-spin text-sapphire-text-tertiary" }),
          /* @__PURE__ */ jsx("span", { className: "ml-2 text-sm text-sapphire-text-tertiary", children: "Loading..." })
        ] }) : /* @__PURE__ */ jsxs(Fragment, { children: [
          headerContent,
          /* @__PURE__ */ jsx(
            List,
            {
              accessibleRole: "ListBox" /* ListBox */,
              separators: "None" /* None */,
              children
            }
          )
        ] })
      ] })
    }
  );
}

export { ComboBoxPopover };
