// @sap-ui/fx-components/dist/chunk-C4DRGF7N.js
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useImperativeHandle } from 'react';
import { jsx } from 'react/jsx-runtime';

// src/types/text.ts
var TextEmptyIndicatorMode = /* @__PURE__ */ ((TextEmptyIndicatorMode2) => {
  TextEmptyIndicatorMode2["Off"] = "Off";
  TextEmptyIndicatorMode2["On"] = "On";
  return TextEmptyIndicatorMode2;
})(TextEmptyIndicatorMode || {});
function isEmpty(children) {
  if (children == null) return true;
  if (typeof children === "string" && children.trim() === "") return true;
  if (Array.isArray(children) && children.length === 0) return true;
  return false;
}
function Text({
  maxLines = 0,
  emptyIndicatorMode = "Off" /* Off */,
  children,
  className,
  style,
  ref
}) {
  const innerRef = useRef(null);
  useImperativeHandle(
    ref,
    () => ({
      focus: () => innerRef.current?.focus(),
      get nativeElement() {
        return innerRef.current;
      }
    }),
    []
  );
  const showEmptyIndicator = emptyIndicatorMode === "On" /* On */ && isEmpty(children);
  const clampStyles = maxLines === 1 ? { maxWidth: "100%" } : maxLines > 1 ? {
    display: "-webkit-inline-box",
    WebkitLineClamp: maxLines,
    lineClamp: maxLines,
    WebkitBoxOrient: "vertical"
  } : {};
  return /* @__PURE__ */ jsx(
    "span",
    {
      ref: innerRef,
      className: cn(
        "text-foreground font-normal text-sm",
        maxLines === 1 && "inline-block overflow-hidden text-ellipsis whitespace-nowrap",
        maxLines > 1 && "overflow-hidden",
        className
      ),
      style: { ...clampStyles, ...style },
      children: showEmptyIndicator ? /* @__PURE__ */ jsx("span", { className: "text-sapphire-text-tertiary", children: "\u2013" }) : children
    }
  );
}

export { Text, TextEmptyIndicatorMode };
