// @sap-ui/fx-components/dist/chunk-ZH374RLZ.js
import { cn } from './chunk-2F2PZMJ5.js';
import { createContext, useCallback, useRef, useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { jsx, jsxs } from 'react/jsx-runtime';

var SegmentedButtonContext = createContext({
  selectionMode: "Single",
  size: "Large",
  onItemSelect: () => {
  }
});
function SegmentedButton({
  selectedId,
  selectedIds,
  selectionMode = "Single",
  size = "Large",
  children,
  onSelectionChange,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  ref
}) {
  const { t } = useTranslation("fx");
  const handleItemSelect = useCallback(
    (id, item) => {
      if (selectionMode === "Multiple") {
        const current = selectedIds ?? [];
        const newIds = current.includes(id) ? current.filter((i) => i !== id) : [...current, id];
        onSelectionChange?.({
          selectedItem: item,
          selectedId: id,
          selectedIds: newIds
        });
      } else {
        onSelectionChange?.({
          selectedItem: item,
          selectedId: id,
          selectedIds: [id]
        });
      }
    },
    [onSelectionChange, selectionMode, selectedIds]
  );
  const innerRef = useRef(null);
  const handleKeyDown = useCallback(
    (e) => {
      const container = innerRef.current;
      if (!container) return;
      if (e.key === "ArrowLeft" || e.key === "ArrowRight") {
        const items = Array.from(
          container.querySelectorAll(
            'li[role="option"]:not([aria-disabled="true"])'
          )
        );
        if (items.length === 0) return;
        const currentIndex = items.indexOf(
          document.activeElement?.closest?.("li")
        );
        if (currentIndex === -1) return;
        e.preventDefault();
        const next = e.key === "ArrowRight" ? (currentIndex + 1) % items.length : (currentIndex - 1 + items.length) % items.length;
        items[next].focus();
      } else if (e.key === "Enter" || e.key === " ") {
        const focused = document.activeElement?.closest?.(
          'li[role="option"]'
        );
        if (focused) {
          e.preventDefault();
          focused.click();
        }
      }
    },
    []
  );
  const isLarge = size === "Large";
  return /* @__PURE__ */ jsx(
    SegmentedButtonContext.Provider,
    {
      value: {
        selectedId,
        selectedIds,
        selectionMode,
        size,
        onItemSelect: handleItemSelect
      },
      children: /* @__PURE__ */ jsx(
        "ul",
        {
          ref: (node) => {
            innerRef.current = node;
            if (typeof ref === "function") ref(node);
            else if (ref) ref.current = node;
          },
          role: "listbox",
          "aria-label": accessibleName || t("SEGMENTEDBUTTON_ARIA_DESCRIPTION"),
          "aria-labelledby": accessibleNameRef,
          "aria-multiselectable": selectionMode === "Multiple" || void 0,
          onKeyDown: handleKeyDown,
          className: cn(
            "inline-flex list-none m-0 p-0 outline outline-1 -outline-offset-1 outline-sapphire-border-primary bg-background",
            isLarge ? "rounded-lg" : "rounded",
            className
          ),
          style,
          children
        }
      )
    }
  );
}
function SegmentedButtonItem({
  id,
  text,
  icon,
  selected: controlledSelected,
  disabled = false,
  tooltip,
  className,
  ref
}) {
  const { selectedId, selectedIds, selectionMode, size, onItemSelect } = useContext(SegmentedButtonContext);
  const { t } = useTranslation("fx");
  const isSelected = controlledSelected ?? (selectionMode === "Multiple" ? selectedIds?.includes(id) ?? false : selectedId === id);
  const itemRef = useRef(null);
  const handleClick = () => {
    if (disabled) return;
    itemRef.current?.focus();
    onItemSelect(id, { id, text, icon, selected: true, disabled, tooltip });
  };
  const isIconOnly = icon && !text;
  const isLarge = size === "Large";
  return /* @__PURE__ */ jsxs(
    "li",
    {
      ref: (node) => {
        itemRef.current = node;
        if (typeof ref === "function") ref(node);
        else if (ref) ref.current = node;
      },
      role: "option",
      "aria-selected": isSelected,
      "aria-disabled": disabled || void 0,
      "aria-roledescription": t("SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION"),
      tabIndex: isSelected ? 0 : -1,
      title: tooltip,
      onClick: handleClick,
      className: cn(
        "inline-flex items-center justify-center gap-sapphire-2xs transition-[color,background-color,border-color] cursor-pointer relative",
        isLarge ? "rounded-lg h-10 min-w-[2.5rem] px-4 text-base gap-2 font-semibold" : size === "Medium" ? "rounded h-8 min-w-[2.25rem] px-3 text-sm gap-1.5 font-semibold" : "rounded h-6 min-w-[1.75rem] px-2 text-xs gap-1 font-normal",
        "focus:z-20",
        "border",
        isSelected ? "z-10 border-sapphire-brand-foreground text-sapphire-brand-foreground focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring" : "border-transparent text-sapphire-neutral-foreground-black hover:bg-sapphire-neutral-hover-background focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring",
        disabled && "pointer-events-none text-sapphire-button-fg-disabled",
        isIconOnly && (isLarge ? "px-2" : size === "Medium" ? "px-2" : "px-1.5"),
        className
      ),
      children: [
        icon && /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center", isLarge ? "[&>svg]:h-5 [&>svg]:w-5" : size === "Medium" ? "[&>svg]:h-4 [&>svg]:w-4" : "[&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"), children: icon }),
        text && /* @__PURE__ */ jsx("span", { children: text })
      ]
    }
  );
}

export { SegmentedButton, SegmentedButtonItem };
