// @sap-ui/fx-components/dist/chunk-ELIHCCZQ.js
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useId, useRef, useState, useImperativeHandle, useCallback } from 'react';
import { Loader2 } from 'lucide-react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx } from 'react/jsx-runtime';

function isSpace(e) {
  return e.key === " " || e.key === "Spacebar";
}
function isEnter(e) {
  return e.key === "Enter";
}
function isDown(e) {
  return e.key === "ArrowDown" && !e.altKey;
}
function isUp(e) {
  return e.key === "ArrowUp" && !e.altKey;
}
function isDownAlt(e) {
  return e.key === "ArrowDown" && e.altKey;
}
function isUpAlt(e) {
  return e.key === "ArrowUp" && e.altKey;
}
function isF4(e) {
  return e.key === "F4";
}
function isShift(e) {
  return e.key === "Shift";
}
function isEscape(e) {
  return e.key === "Escape";
}
function isTab(e) {
  return e.key === "Tab";
}
function isArrowKeyAction(e) {
  return isDown(e) || isUp(e) || isDownAlt(e) || isUpAlt(e) || isF4(e);
}
function isDefaultAction(e) {
  return isSpace(e) || isEnter(e);
}
var containerVariants = cva(
  [
    "inline-flex relative group",
    "border"
  ],
  {
    variants: {
      design: {
        ["Secondary" /* Secondary */]: "bg-transparent border-sapphire-brand-foreground",
        ["Primary" /* Primary */]: "bg-sapphire-brand-background border-sapphire-brand-background",
        ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-joule-background border-sapphire-joule-background",
        ["SecondaryJoule" /* SecondaryJoule */]: "bg-transparent border-sapphire-joule-foreground",
        ["Neutral" /* Neutral */]: "bg-transparent border-sapphire-neutral-foreground-muted",
        ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-transparent border-transparent",
        ["TertiaryNeutral" /* TertiaryNeutral */]: "bg-transparent border-transparent",
        ["Tertiary" /* Tertiary */]: "bg-transparent border-transparent",
        ["TertiaryJoule" /* TertiaryJoule */]: "bg-transparent border-transparent"
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var textColorClass = {
  ["Secondary" /* Secondary */]: "text-sapphire-brand-foreground",
  ["Primary" /* Primary */]: "text-sapphire-neutral-foreground-white",
  ["PrimaryJoule" /* PrimaryJoule */]: "text-sapphire-neutral-foreground-white",
  ["SecondaryJoule" /* SecondaryJoule */]: "text-sapphire-joule-foreground",
  ["Neutral" /* Neutral */]: "text-sapphire-neutral-foreground-black",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "text-sapphire-neutral-foreground-black",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "text-sapphire-neutral-foreground-muted",
  ["Tertiary" /* Tertiary */]: "text-sapphire-brand-foreground",
  ["TertiaryJoule" /* TertiaryJoule */]: "text-sapphire-joule-foreground"
};
var textBtnHoverClass = {
  ["Secondary" /* Secondary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Primary" /* Primary */]: "hover:bg-sapphire-brand-hover-background hover:z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "hover:bg-sapphire-joule-hover-background hover:z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Neutral" /* Neutral */]: "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["Tertiary" /* Tertiary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10"
};
var textBtnActiveClass = {
  ["Secondary" /* Secondary */]: "bg-sapphire-neutral-pressed-background text-sapphire-brand-pressed-foreground z-10",
  ["Primary" /* Primary */]: "bg-sapphire-brand-pressed-background text-sapphire-neutral-foreground-white z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-joule-pressed-background text-sapphire-neutral-foreground-white z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "bg-sapphire-neutral-pressed-background text-sapphire-joule-pressed-foreground z-10",
  ["Neutral" /* Neutral */]: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "text-sapphire-neutral-foreground-black z-10",
  ["Tertiary" /* Tertiary */]: "bg-sapphire-neutral-pressed-background text-sapphire-brand-pressed-foreground z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "bg-sapphire-neutral-pressed-background text-sapphire-joule-pressed-foreground z-10"
};
var arrowBtnHoverClass = {
  ["Secondary" /* Secondary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Primary" /* Primary */]: "hover:bg-sapphire-brand-hover-background hover:z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "hover:bg-sapphire-joule-hover-background hover:z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["Neutral" /* Neutral */]: "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["Tertiary" /* Tertiary */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "hover:bg-sapphire-neutral-hover-background hover:z-10"
};
var arrowBtnActiveClass = {
  ["Secondary" /* Secondary */]: "bg-sapphire-brand-toggle-background text-sapphire-brand-foreground z-10",
  ["Primary" /* Primary */]: "bg-sapphire-brand-hover-background text-sapphire-neutral-foreground-white z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-joule-hover-background text-sapphire-neutral-foreground-white z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "bg-sapphire-brand-toggle-background text-sapphire-joule-foreground z-10",
  ["Neutral" /* Neutral */]: "bg-sapphire-neutral-pressed-background text-sapphire-neutral-foreground-black z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-sapphire-brand-toggle-background text-sapphire-neutral-foreground-black z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "text-sapphire-neutral-foreground-black z-10",
  ["Tertiary" /* Tertiary */]: "bg-sapphire-brand-toggle-background text-sapphire-brand-foreground z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "bg-sapphire-brand-toggle-background text-sapphire-joule-foreground z-10"
};
var arrowBtnActiveHoverClass = {
  ["Secondary" /* Secondary */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["Primary" /* Primary */]: "hover:bg-blue-800 hover:z-10",
  ["PrimaryJoule" /* PrimaryJoule */]: "hover:bg-purple-800 hover:z-10",
  ["SecondaryJoule" /* SecondaryJoule */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["Neutral" /* Neutral */]: "hover:bg-sapphire-neutral-hover-background hover:z-10",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "hover:text-sapphire-neutral-foreground-black hover:z-10",
  ["Tertiary" /* Tertiary */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10",
  ["TertiaryJoule" /* TertiaryJoule */]: "hover:bg-sapphire-brand-toggle-hover-background hover:z-10"
};
var separatorColor = {
  ["Secondary" /* Secondary */]: "bg-sapphire-brand-foreground",
  ["Primary" /* Primary */]: "bg-sapphire-neutral-foreground-white",
  ["PrimaryJoule" /* PrimaryJoule */]: "bg-sapphire-neutral-foreground-white",
  ["SecondaryJoule" /* SecondaryJoule */]: "bg-sapphire-joule-foreground",
  ["Neutral" /* Neutral */]: "bg-sapphire-neutral-foreground-black",
  ["SecondaryNeutral" /* SecondaryNeutral */]: "bg-sapphire-neutral-foreground-black",
  ["TertiaryNeutral" /* TertiaryNeutral */]: "bg-sapphire-neutral-foreground-muted",
  ["Tertiary" /* Tertiary */]: "bg-sapphire-brand-foreground",
  ["TertiaryJoule" /* TertiaryJoule */]: "bg-sapphire-joule-foreground"
};
var separatorDesigns = /* @__PURE__ */ new Set([
  "Secondary" /* Secondary */,
  "SecondaryJoule" /* SecondaryJoule */,
  "Neutral" /* Neutral */
]);
var borderlessDesigns = /* @__PURE__ */ new Set([
  "Tertiary" /* Tertiary */,
  "TertiaryJoule" /* TertiaryJoule */,
  "TertiaryNeutral" /* TertiaryNeutral */,
  "SecondaryNeutral" /* SecondaryNeutral */
]);
var focusSizeClass = {
  Large: "split-btn-focus-lg",
  Medium: "split-btn-focus-md",
  Small: "split-btn-focus-sm"
};
var innerSizeClassMap = {
  Large: "split-btn-inner-lg",
  Medium: "split-btn-inner-md",
  Small: "split-btn-inner-sm"
};
var hoverRoundingMap = {
  Large: "hover:rounded-lg",
  Medium: "hover:rounded",
  Small: "hover:rounded"
};
var activeRoundingMap = {
  Large: "rounded-lg",
  Medium: "rounded",
  Small: "rounded"
};
function SplitButton({
  text,
  icon,
  endIcon,
  design = "Secondary" /* Secondary */,
  size = "Large",
  disabled = false,
  loading = false,
  activeArrowButton: activeArrowButtonProp = false,
  accessibleName,
  accessibilityAttributes,
  onClick,
  onArrowClick,
  className,
  style,
  id: idProp,
  ref
}) {
  const autoId = useId();
  const id = idProp || autoId;
  const { t } = useTranslation("fx");
  const containerRef = useRef(null);
  const textButtonRef = useRef(null);
  const arrowButtonRef = useRef(null);
  const [textButtonActive, setTextButtonActive] = useState(false);
  const [internalArrowActive, setInternalArrowActive] = useState(false);
  const [shiftOrEscapeDuringSpace, setShiftOrEscapeDuringSpace] = useState(false);
  const [tabIndex, setTabIndex] = useState(disabled ? -1 : 0);
  const effectiveActiveArrow = activeArrowButtonProp || internalArrowActive;
  const d = design;
  const sizeClasses = {
    Large: { container: "h-10 rounded-lg", arrow: "w-10", text: "text-base px-4 gap-2 font-semibold", icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]", chevron: "h-4 w-4" },
    Medium: { container: "h-8 rounded", arrow: "w-8", text: "text-sm px-3 gap-1.5 font-semibold", icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5", chevron: "h-3.5 w-3.5" },
    Small: { container: "h-6 rounded", arrow: "w-6", text: "text-xs px-2 gap-1 font-normal", icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]", chevron: "h-3 w-3" }
  }[size];
  useImperativeHandle(ref, () => ({
    focus() {
      containerRef.current?.focus();
    },
    get nativeElement() {
      return containerRef.current;
    },
    get textButton() {
      return textButtonRef.current;
    },
    get arrowButton() {
      return arrowButtonRef.current;
    }
  }));
  const fireClick = useCallback(() => {
    onClick?.();
  }, [onClick]);
  const fireArrowClick = useCallback(() => {
    onArrowClick?.();
  }, [onArrowClick]);
  const resetActionStates = useCallback(() => {
    setInternalArrowActive(false);
    setTextButtonActive(false);
    setShiftOrEscapeDuringSpace(false);
  }, []);
  const setTabIndexValue = useCallback(
    (innerButtonPressed) => {
      let val = disabled ? -1 : 0;
      if (val === -1 && innerButtonPressed) {
        val = 0;
      }
      setTabIndex(val);
    },
    [disabled]
  );
  const handleFocusOut = useCallback(
    (e) => {
      if (disabled) return;
      const related = e.relatedTarget;
      if (containerRef.current?.contains(related)) return;
      resetActionStates();
      setTabIndexValue();
    },
    [disabled, resetActionStates, setTabIndexValue]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled || loading) return;
      if (isArrowKeyAction(e)) {
        e.preventDefault();
        fireArrowClick();
        setInternalArrowActive(true);
        return;
      }
      if (isDefaultAction(e)) {
        e.preventDefault();
        const target = e.target;
        if (isEnter(e)) {
          if (arrowButtonRef.current && target === arrowButtonRef.current) {
            setInternalArrowActive(true);
            onArrowClick?.();
            return;
          }
          setTextButtonActive(true);
          fireClick();
          return;
        }
        if (isSpace(e)) {
          setTextButtonActive(true);
        }
        if (isTab(e)) {
          resetActionStates();
        }
        return;
      }
      if ((isShift(e) || isEscape(e)) && textButtonActive) {
        e.preventDefault();
        setShiftOrEscapeDuringSpace(true);
      }
      if (isEscape(e) && !textButtonActive) {
        resetActionStates();
      }
      setTabIndex(-1);
    },
    [disabled, loading, fireArrowClick, fireClick, onArrowClick, textButtonActive, resetActionStates]
  );
  const handleKeyUp = useCallback(
    (e) => {
      if (disabled || loading) return;
      if (isArrowKeyAction(e)) {
        e.preventDefault();
        setInternalArrowActive(false);
        return;
      }
      if (isSpace(e)) {
        e.preventDefault();
        e.stopPropagation();
        setTextButtonActive(false);
        if (!shiftOrEscapeDuringSpace) {
          const target = e.target;
          if (arrowButtonRef.current && target === arrowButtonRef.current) {
            onArrowClick?.();
          } else {
            fireClick();
          }
        }
        setShiftOrEscapeDuringSpace(false);
        return;
      }
      if (isEnter(e) || isShift(e) && textButtonActive) {
        setTextButtonActive(false);
      }
    },
    [disabled, loading, shiftOrEscapeDuringSpace, fireClick, onArrowClick, textButtonActive]
  );
  const handleTextMouseDown = useCallback(
    (e) => {
      if (disabled || loading) return;
      e.stopPropagation();
      setTextButtonActive(true);
      setTabIndex(-1);
    },
    [disabled, loading]
  );
  const handleTextMouseUp = useCallback(() => {
    setTextButtonActive(false);
    setTabIndex(-1);
  }, []);
  const handleTextClick = useCallback(
    (e) => {
      if (disabled || loading) return;
      e.stopPropagation();
      fireClick();
    },
    [disabled, loading, fireClick]
  );
  const handleArrowMouseDown = useCallback(
    (e) => {
      if (disabled || loading) return;
      e.stopPropagation();
      setTabIndex(-1);
    },
    [disabled, loading]
  );
  const handleArrowClick = useCallback(
    (e) => {
      if (disabled || loading) return;
      e.preventDefault();
      e.stopPropagation();
      fireArrowClick();
    },
    [disabled, loading, fireArrowClick]
  );
  const handleInnerFocusIn = useCallback(
    (e) => {
      e.stopPropagation();
      setTabIndexValue(true);
      e.target.focus();
    },
    [setTabIndexValue]
  );
  const arrowHasPopup = accessibilityAttributes?.arrowButton?.hasPopup || "menu";
  const arrowExpanded = accessibilityAttributes?.arrowButton?.expanded ?? effectiveActiveArrow;
  const arrowTitle = accessibilityAttributes?.arrowButton?.title || t("SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP");
  const rootHasPopup = accessibilityAttributes?.root?.hasPopup;
  const rootRoleDescription = accessibilityAttributes?.root?.roleDescription || t("SPLIT_BUTTON_DESCRIPTION");
  const rootTitle = accessibilityAttributes?.root?.title;
  const rootAriaKeyShortcuts = accessibilityAttributes?.root?.ariaKeyShortcuts;
  const isDisabled = disabled || loading;
  const isBorderless = borderlessDesigns.has(d);
  const hoverRounding = isBorderless ? hoverRoundingMap[size] : "";
  const activeRounding = isBorderless ? activeRoundingMap[size] : "";
  const isFilled = d === "Primary" /* Primary */ || d === "PrimaryJoule" /* PrimaryJoule */;
  const containerFocusVariant = {
    ["Primary" /* Primary */]: "split-btn-focus split-btn-focus-filled-brand",
    ["PrimaryJoule" /* PrimaryJoule */]: "split-btn-focus split-btn-focus-filled-joule",
    ["Secondary" /* Secondary */]: "split-btn-focus split-btn-focus-brand",
    ["SecondaryJoule" /* SecondaryJoule */]: "split-btn-focus split-btn-focus-joule",
    ["Tertiary" /* Tertiary */]: "split-btn-focus split-btn-focus-brand",
    ["TertiaryJoule" /* TertiaryJoule */]: "split-btn-focus split-btn-focus-joule",
    ["Neutral" /* Neutral */]: "split-btn-focus split-btn-focus-neutral",
    ["SecondaryNeutral" /* SecondaryNeutral */]: "split-btn-focus split-btn-focus-neutral",
    ["TertiaryNeutral" /* TertiaryNeutral */]: "split-btn-focus split-btn-focus-neutral"
  };
  const containerFocusSizeClass = focusSizeClass[size];
  const containerFocusOutlineClass = cn(containerFocusVariant[d], containerFocusSizeClass);
  const innerFocusColorClass = {
    ["Primary" /* Primary */]: "split-btn-inner-white",
    ["PrimaryJoule" /* PrimaryJoule */]: "split-btn-inner-white",
    ["Secondary" /* Secondary */]: "split-btn-inner-brand",
    ["SecondaryJoule" /* SecondaryJoule */]: "split-btn-inner-joule",
    ["Tertiary" /* Tertiary */]: "split-btn-inner-brand",
    ["TertiaryJoule" /* TertiaryJoule */]: "split-btn-inner-joule",
    ["Neutral" /* Neutral */]: "split-btn-inner-neutral",
    ["SecondaryNeutral" /* SecondaryNeutral */]: "split-btn-inner-neutral",
    ["TertiaryNeutral" /* TertiaryNeutral */]: "split-btn-inner-neutral"
  };
  const innerSizeClass = innerSizeClassMap[size];
  const innerFocusClass = cn(
    "split-btn-inner",
    innerSizeClass,
    innerFocusColorClass[d]
  );
  const textFocusRing = innerFocusClass;
  const arrowFocusRing = innerFocusClass;
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: containerRef,
      id,
      role: "group",
      tabIndex: isDisabled ? -1 : tabIndex,
      "aria-labelledby": `${id}-invisibleTextDefault ${id}-invisibleText`,
      "aria-haspopup": rootHasPopup,
      "aria-roledescription": rootRoleDescription,
      "aria-keyshortcuts": rootAriaKeyShortcuts,
      className: cn(
        containerVariants({ design: d }),
        sizeClasses.container,
        "shrink-0 w-fit",
        isDisabled && "pointer-events-none opacity-50",
        "outline-none",
        containerFocusOutlineClass,
        className
      ),
      style,
      onBlur: handleFocusOut,
      onKeyDown: handleKeyDown,
      onKeyUp: handleKeyUp,
      children: [
        /* @__PURE__ */ jsxs(
          "button",
          {
            ref: textButtonRef,
            type: "button",
            tabIndex: -1,
            disabled: isDisabled,
            title: rootTitle,
            onClick: handleTextClick,
            onMouseDown: handleTextMouseDown,
            onMouseUp: handleTextMouseUp,
            onFocus: handleInnerFocusIn,
            className: cn(
              "inline-flex items-center justify-start cursor-pointer relative",
              sizeClasses.text,
              "flex-grow",
              "rounded-s-[inherit] rounded-e-none",
              "bg-transparent",
              isFilled && "border-r border-r-sapphire-neutral-foreground-white",
              textColorClass[d],
              "whitespace-nowrap",
              "outline-none transition-colors",
              "disabled:pointer-events-none",
              textFocusRing,
              !textButtonActive && textBtnHoverClass[d],
              !textButtonActive && hoverRounding,
              textButtonActive && textBtnActiveClass[d],
              textButtonActive && activeRounding
            ),
            children: [
              loading ? /* @__PURE__ */ jsx(Loader2, { className: cn(sizeClasses.icon, "animate-spin") }) : icon && /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses.icon), children: icon }),
              text && /* @__PURE__ */ jsx("span", { children: text }),
              endIcon && !loading && /* @__PURE__ */ jsx("span", { className: "inline-flex items-center shrink-0", children: endIcon })
            ]
          }
        ),
        separatorDesigns.has(d) && /* @__PURE__ */ jsx(
          "span",
          {
            "aria-hidden": "true",
            className: cn(
              "w-px shrink-0 self-stretch z-[1] pointer-events-none",
              separatorColor[d]
            )
          }
        ),
        /* @__PURE__ */ jsx(
          "button",
          {
            ref: arrowButtonRef,
            type: "button",
            tabIndex: -1,
            disabled: isDisabled,
            "aria-haspopup": arrowHasPopup,
            "aria-expanded": arrowExpanded,
            title: arrowTitle,
            onClick: handleArrowClick,
            onMouseDown: handleArrowMouseDown,
            onFocus: handleInnerFocusIn,
            className: cn(
              "inline-flex items-center justify-center cursor-pointer",
              sizeClasses.arrow,
              "shrink-0",
              "rounded-e-[inherit] rounded-s-none",
              "bg-transparent",
              textColorClass[d],
              "outline-none transition-colors overflow-visible relative",
              "disabled:pointer-events-none",
              arrowFocusRing,
              !effectiveActiveArrow && arrowBtnHoverClass[d],
              !effectiveActiveArrow && hoverRounding,
              effectiveActiveArrow && arrowBtnActiveClass[d],
              effectiveActiveArrow && arrowBtnActiveHoverClass[d],
              effectiveActiveArrow && activeRounding
            ),
            children: /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: sizeClasses.chevron })
          }
        ),
        /* @__PURE__ */ jsxs("span", { id: `${id}-invisibleText`, className: "sr-only", children: [
          t("SPLIT_BUTTON_KEYBOARD_HINT"),
          accessibleName ? ` ${accessibleName}` : ""
        ] }),
        /* @__PURE__ */ jsx("span", { id: `${id}-invisibleTextDefault`, className: "sr-only", children: text })
      ]
    }
  );
}

export { SplitButton };
