// @sap-ui/fx-components/dist/chunk-IIYUWYBR.js
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { MessageInformationIcon, MessageWarningIcon, MessageErrorIcon, MessageSuccessIcon } from './chunk-WDT5ZCHE.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useId } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/messagestrip.ts
var MessageStripDesign = /* @__PURE__ */ ((MessageStripDesign2) => {
  MessageStripDesign2["Information"] = "Information";
  MessageStripDesign2["Positive"] = "Positive";
  MessageStripDesign2["Negative"] = "Negative";
  MessageStripDesign2["Warning"] = "Warning";
  return MessageStripDesign2;
})(MessageStripDesign || {});
var messageStripVariants = cva(
  [
    "relative flex items-start gap-sapphire-3xs rounded-[4px] border border-solid min-h-[40px] transition-all duration-200"
  ],
  {
    variants: {
      design: {
        ["Information" /* Information */]: "bg-sapphire-info-bg border-sapphire-info-border",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg border-sapphire-positive-border",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg border-sapphire-negative-border",
        ["Warning" /* Warning */]: "bg-sapphire-warning-bg border-sapphire-warning-border"
      },
      hasTitle: {
        true: "px-sapphire-xs py-sapphire-2xs",
        false: "p-sapphire-2xs"
      }
    },
    defaultVariants: {
      design: "Information" /* Information */,
      hasTitle: false
    }
  }
);
var iconVariants = cva("flex-shrink-0", {
  variants: {
    design: {
      ["Information" /* Information */]: "text-sapphire-info",
      ["Positive" /* Positive */]: "text-sapphire-positive",
      ["Negative" /* Negative */]: "text-sapphire-negative",
      ["Warning" /* Warning */]: "text-sapphire-warning"
    }
  },
  defaultVariants: {
    design: "Information" /* Information */
  }
});
var designAnnouncementKey = {
  ["Information" /* Information */]: "MESSAGE_STRIP_INFORMATION",
  ["Positive" /* Positive */]: "MESSAGE_STRIP_SUCCESS",
  ["Negative" /* Negative */]: "MESSAGE_STRIP_ERROR",
  ["Warning" /* Warning */]: "MESSAGE_STRIP_WARNING"
};
var designCloseButtonKey = {
  ["Information" /* Information */]: "MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION",
  ["Positive" /* Positive */]: "MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE",
  ["Negative" /* Negative */]: "MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE",
  ["Warning" /* Warning */]: "MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL"
};
function MessageStrip({
  design = "Information" /* Information */,
  title,
  children,
  icon,
  hideIcon = false,
  hideCloseButton = false,
  onClose,
  accessibleName,
  accessibleNameRef,
  accessibleRole,
  accessibleDescription,
  accessibleDescriptionRef,
  id: idProp,
  className,
  style,
  "data-testid": dataTestId,
  ref
}) {
  const autoId = useId();
  const descriptionId = useId();
  const { t } = useTranslation("fx");
  const id = idProp || autoId;
  const hiddenTextId = `${id}-hidden-text`;
  const contentTextId = `${id}-content-text`;
  const normalizedDesign = design;
  const hasTitle = !!title;
  const getDefaultIcon = () => {
    switch (normalizedDesign) {
      case "Positive" /* Positive */:
        return /* @__PURE__ */ jsx(MessageSuccessIcon, { className: "block h-4 w-4" });
      case "Negative" /* Negative */:
        return /* @__PURE__ */ jsx(MessageErrorIcon, { className: "block h-4 w-4" });
      case "Warning" /* Warning */:
        return /* @__PURE__ */ jsx(MessageWarningIcon, { className: "block h-4 w-4" });
      case "Information" /* Information */:
      default:
        return /* @__PURE__ */ jsx(MessageInformationIcon, { className: "block h-4 w-4" });
    }
  };
  const announcementKey = designAnnouncementKey[normalizedDesign] || designAnnouncementKey["Information" /* Information */];
  const announcementText = t(announcementKey);
  const closableText = !hideCloseButton ? ` ${t("MESSAGE_STRIP_CLOSABLE")}` : "";
  const hiddenText = `${announcementText}${closableText}`;
  const closeButtonKey = designCloseButtonKey[normalizedDesign] || designCloseButtonKey["Information" /* Information */];
  const closeButtonText = t(closeButtonKey);
  const handleClose = () => {
    onClose?.({ closed: true });
  };
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      id,
      role: accessibleRole || "note",
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef || `${hiddenTextId} ${contentTextId}`,
      "aria-describedby": [accessibleDescriptionRef, accessibleDescription ? descriptionId : null].filter(Boolean).join(" ") || void 0,
      "data-testid": dataTestId,
      className: cn(messageStripVariants({ design: normalizedDesign, hasTitle }), className),
      style,
      children: [
        !hideIcon && /* @__PURE__ */ jsx("div", { className: cn(iconVariants({ design: normalizedDesign }), hasTitle ? "p-2" : "p-1"), "aria-hidden": "true", children: icon || getDefaultIcon() }),
        /* @__PURE__ */ jsx("span", { className: "sr-only", id: hiddenTextId, children: hiddenText }),
        hasTitle ? /* @__PURE__ */ jsxs("div", { className: "flex-1 flex flex-col gap-sapphire-3xs min-w-0 pb-sapphire-2xs", id: contentTextId, children: [
          /* @__PURE__ */ jsx("div", { className: "text-base font-semibold leading-[22px] text-sapphire-text-primary h-8 flex items-center", children: title }),
          /* @__PURE__ */ jsx("div", { className: "text-sm font-normal leading-5 text-sapphire-text-secondary flex items-start gap-1", children })
        ] }) : /* @__PURE__ */ jsx("div", { className: "flex-1 text-sm font-semibold leading-5 text-sapphire-text-primary min-w-0 py-0.5", id: contentTextId, children }),
        !hideCloseButton && /* @__PURE__ */ jsx(
          Button,
          {
            design: "SecondaryNeutral" /* SecondaryNeutral */,
            size: hasTitle ? "Medium" /* Medium */ : "Small" /* Small */,
            icon: /* @__PURE__ */ jsx(DeclineIcon, { className: hasTitle ? "h-4 w-4" : "h-3 w-3" }),
            iconOnly: true,
            onClick: handleClose,
            accessibleName: closeButtonText,
            tooltip: closeButtonText,
            className: "flex-shrink-0"
          }
        ),
        accessibleDescription && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription })
      ]
    }
  );
}

export { MessageStrip, MessageStripDesign };
