// @sap-ui/fx-components/dist/chunk-BJEKLZEH.js
import { Tag } from './chunk-J3MFNYJX.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { OverflowIcon } from './chunk-YHUSMQDT.js';
import { SlimArrowRightIcon } from './chunk-ZQSLIXR7.js';
import { Link } from './chunk-MCJ7D5KG.js';
import { ListItemBase } from './chunk-CTYK4HEK.js';
import { List } from './chunk-CJXZUSUP.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import * as React2 from 'react';
import { useContext, useRef, useCallback, useLayoutEffect, useEffect, useMemo, useImperativeHandle, useState, useId } from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { Loader2 } from 'lucide-react';

var NotificationListContext = React2.createContext(null);
function useNotificationListContext() {
  const context = useContext(NotificationListContext);
  if (!context) {
    throw new Error(
      "Notification item components must be used within a NotificationList"
    );
  }
  return context;
}
function useOptionalNotificationListContext() {
  return useContext(NotificationListContext);
}
function NotificationList(props) {
  const {
    children,
    noDataText = "No notifications",
    // Events
    onItemClick,
    onItemClose,
    onItemToggle,
    // Accessibility
    accessibleName = "Notifications",
    accessibleNameRef,
    // Standard
    className,
    style,
    id,
    "data-testid": dataTestId,
    ref
  } = props;
  const listRef = useRef(null);
  const allNavigationItemsRef = useRef([]);
  const updateNavigationItems = useCallback(() => {
    const container = listRef.current?.scrollContainer;
    if (!container) {
      return;
    }
    const items = [];
    const mainList = container.querySelector(":scope > ul");
    if (!mainList) return;
    const topLevelItems = Array.from(mainList.children);
    topLevelItems.forEach((item) => {
      if (!item.hasAttribute("data-notification-item")) {
        return;
      }
      const isGroup = item.hasAttribute("data-notification-group");
      if (isGroup) {
        const groupButton = item.querySelector('[role="button"]');
        if (groupButton) {
          items.push(groupButton);
        }
        const isCollapsed = groupButton?.getAttribute("aria-expanded") === "false";
        if (!isCollapsed) {
          const childList = item.querySelector("ul[id]");
          if (childList) {
            const childItems = childList.querySelectorAll("li[data-notification-item]");
            childItems.forEach((child) => items.push(child));
          }
          const growingBtn = item.querySelector("[data-growing-button]");
          if (growingBtn) {
            items.push(growingBtn);
          }
        }
      } else {
        items.push(item);
      }
    });
    allNavigationItemsRef.current = items;
    items.forEach((item, index) => {
      const newTabIndex = index === 0 ? "0" : "-1";
      item.setAttribute("tabindex", newTabIndex);
    });
  }, []);
  useLayoutEffect(() => {
    updateNavigationItems();
  }, [children, updateNavigationItems]);
  useEffect(() => {
    const container = listRef.current?.scrollContainer;
    if (!container) return;
    updateNavigationItems();
    const observer = new MutationObserver(() => {
      const items = allNavigationItemsRef.current;
      items.forEach((item, index) => {
        const expectedTabIndex = index === 0 ? "0" : "-1";
        const currentTabIndex = item.getAttribute("tabindex");
        if (currentTabIndex !== expectedTabIndex) {
          item.setAttribute("tabindex", expectedTabIndex);
        }
      });
    });
    observer.observe(container, {
      childList: true,
      subtree: true,
      attributes: true,
      attributeFilter: ["tabindex", "aria-expanded"]
    });
    return () => observer.disconnect();
  }, [updateNavigationItems]);
  const findGroupStart = useCallback((items, currentIndex) => {
    let groupStartIndex = 0;
    for (let i = currentIndex - 1; i >= 0; i--) {
      const item = items[i];
      const isGroupBtn = item.getAttribute("role") === "button" && item.closest("[data-notification-group]");
      if (isGroupBtn) {
        groupStartIndex = i + 1;
        break;
      }
    }
    return groupStartIndex;
  }, []);
  const findGroupEnd = useCallback((items, currentIndex) => {
    let groupEndIndex = items.length - 1;
    for (let i = currentIndex + 1; i < items.length; i++) {
      const item = items[i];
      const isGroupBtn = item.getAttribute("role") === "button" && item.closest("[data-notification-group]");
      if (isGroupBtn) {
        groupEndIndex = i - 1;
        break;
      }
    }
    return groupEndIndex;
  }, []);
  const getNextNavigationIndex = useCallback((key, currentIndex, items, isGroupButton, isExpanded) => {
    if (key === "ArrowDown" || key === "ArrowRight") {
      if (key === "ArrowRight" && isGroupButton && isExpanded) {
        return currentIndex + 1;
      }
      return Math.min(items.length - 1, currentIndex + 1);
    }
    if (key === "ArrowUp" || key === "ArrowLeft") {
      return Math.max(0, currentIndex - 1);
    }
    if (key === "Home") {
      return findGroupStart(items, currentIndex);
    }
    if (key === "End") {
      return findGroupEnd(items, currentIndex);
    }
    return currentIndex;
  }, [findGroupStart, findGroupEnd]);
  const handleKeyDown = useCallback((e) => {
    const target = e.target;
    const isGroupButton = target.getAttribute("role") === "button" && target.closest("[data-notification-group]");
    const isNotificationItem = target.hasAttribute("data-notification-item");
    const isGrowingButton = target.hasAttribute("data-growing-button");
    if (!isGroupButton && !isNotificationItem && !isGrowingButton) {
      return;
    }
    const items = allNavigationItemsRef.current;
    const currentIndex = items.indexOf(target);
    if (currentIndex === -1) return;
    const navigableKeys = ["ArrowDown", "ArrowRight", "ArrowUp", "ArrowLeft", "Home", "End"];
    if (!navigableKeys.includes(e.key)) return;
    e.preventDefault();
    const isExpanded = target.getAttribute?.("aria-expanded") === "true";
    const nextIndex = getNextNavigationIndex(e.key, currentIndex, items, isGroupButton, isExpanded);
    if (nextIndex !== currentIndex) {
      const nextItem = items[nextIndex];
      if (nextItem) {
        items.forEach((item, idx) => {
          item.setAttribute("tabindex", idx === nextIndex ? "0" : "-1");
        });
        nextItem.focus();
      }
    }
  }, [getNextNavigationIndex]);
  const contextValue = useMemo(
    () => ({
      onItemClick,
      onItemClose,
      onItemToggle: (detail) => {
        onItemToggle?.(detail);
        setTimeout(updateNavigationItems, 0);
      },
      updateNavigationItems
    }),
    [onItemClick, onItemClose, onItemToggle, updateNavigationItems]
  );
  useImperativeHandle(
    ref,
    () => ({
      focus: () => {
        listRef.current?.focus();
      },
      blur: () => {
        document.activeElement?.blur();
      },
      isFocused: () => {
        return listRef.current?.scrollContainer?.contains(document.activeElement) ?? false;
      },
      get nativeElement() {
        return listRef.current?.scrollContainer ?? null;
      }
    }),
    []
  );
  return /* @__PURE__ */ jsx(NotificationListContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsx("div", { onKeyDown: handleKeyDown, className: "[&_ul]:flex [&_ul]:flex-col [&_ul]:gap-2", children: /* @__PURE__ */ jsx(
    List,
    {
      ref: listRef,
      noDataText,
      accessibleName,
      accessibleNameRef,
      className,
      style,
      id,
      "data-testid": dataTestId,
      children
    }
  ) }) });
}

// src/types/notification.ts
var NotificationListItemImportance = /* @__PURE__ */ ((NotificationListItemImportance2) => {
  NotificationListItemImportance2["Standard"] = "Standard";
  NotificationListItemImportance2["Important"] = "Important";
  return NotificationListItemImportance2;
})(NotificationListItemImportance || {});
var NotificationListItemState = /* @__PURE__ */ ((NotificationListItemState2) => {
  NotificationListItemState2["None"] = "None";
  NotificationListItemState2["Positive"] = "Positive";
  NotificationListItemState2["Critical"] = "Critical";
  NotificationListItemState2["Negative"] = "Negative";
  NotificationListItemState2["Information"] = "Information";
  return NotificationListItemState2;
})(NotificationListItemState || {});
var titleVariants = cva("text-[13px] leading-4", {
  variants: {
    read: {
      true: "font-normal",
      false: "font-bold"
    },
    wrapping: {
      ["None" /* None */]: "line-clamp-2",
      ["Normal" /* Normal */]: "break-words"
    }
  },
  defaultVariants: {
    read: false,
    wrapping: "None" /* None */
  }
});
var stateIconVariants = cva("h-4 w-4 shrink-0", {
  variants: {
    state: {
      ["None" /* None */]: "hidden",
      ["Positive" /* Positive */]: "text-sapphire-positive",
      ["Critical" /* Critical */]: "text-sapphire-warning",
      ["Negative" /* Negative */]: "text-sapphire-negative",
      ["Information" /* Information */]: "text-sapphire-info"
    }
  },
  defaultVariants: {
    state: "None" /* None */
  }
});
function getStateText(state) {
  const normalized = state;
  switch (normalized) {
    case "Positive" /* Positive */:
      return "Positive";
    case "Critical" /* Critical */:
      return "Critical";
    case "Negative" /* Negative */:
      return "Negative";
    case "Information" /* Information */:
      return "Information";
    default:
      return "";
  }
}
function StateIcon({ state }) {
  const normalized = state;
  const baseClassName = stateIconVariants({ state: normalized });
  switch (normalized) {
    case "Positive" /* Positive */:
      return /* @__PURE__ */ jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsx("circle", { cx: "8", cy: "8", r: "8", fill: "currentColor" }),
        /* @__PURE__ */ jsx("path", { d: "M11.3333 5.5L6.75 10.0833L4.66667 8", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })
      ] });
    case "Critical" /* Critical */:
      return /* @__PURE__ */ jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsx("path", { d: "M6.86078 2.75995L1.21078 12.2499C1.09697 12.4547 1.03736 12.6863 1.03772 12.9218C1.03809 13.1574 1.09843 13.3888 1.21288 13.5933C1.32733 13.7977 1.49214 13.9681 1.69198 14.0881C1.89182 14.2081 2.11982 14.2736 2.35245 14.2783H13.6524C13.8851 14.2736 14.1131 14.2081 14.3129 14.0881C14.5128 13.9681 14.6776 13.7977 14.792 13.5933C14.9065 13.3888 14.9668 13.1574 14.9672 12.9218C14.9675 12.6863 14.9079 12.4547 14.7941 12.2499L9.14412 2.75995C9.02679 2.56103 8.86051 2.39698 8.66099 2.28264C8.46148 2.1683 8.23542 2.10767 8.00495 2.10767C7.77448 2.10767 7.54842 2.1683 7.34891 2.28264C7.14939 2.39698 6.98311 2.56103 6.86578 2.75995H6.86078Z", fill: "currentColor" }),
        /* @__PURE__ */ jsx("path", { d: "M8 6V9", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" }),
        /* @__PURE__ */ jsx("circle", { cx: "8", cy: "11.5", r: "0.75", fill: "white" })
      ] });
    case "Negative" /* Negative */:
      return /* @__PURE__ */ jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsx("circle", { cx: "8", cy: "8", r: "8", fill: "currentColor" }),
        /* @__PURE__ */ jsx("path", { d: "M10 6L6 10M6 6L10 10", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })
      ] });
    case "Information" /* Information */:
      return /* @__PURE__ */ jsxs("svg", { className: baseClassName, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
        /* @__PURE__ */ jsx("circle", { cx: "8", cy: "8", r: "8", fill: "currentColor" }),
        /* @__PURE__ */ jsx("path", { d: "M8 11V8M8 5H8.00667", stroke: "white", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })
      ] });
    default:
      return null;
  }
}
function ImportanceBadge({
  importance
}) {
  if (String(importance) !== "Important") return null;
  return /* @__PURE__ */ jsx(Tag, { design: "Negative" /* Negative */, children: "Important" });
}
function NotificationListItem(props) {
  const {
    // Content
    titleText,
    children,
    avatar,
    menu,
    footnotes,
    // Visual
    wrappingType = "None" /* None */,
    state = "None" /* None */,
    importance = "Standard" /* Standard */,
    read = false,
    // Behavior
    showClose = false,
    loading = false,
    loadingDelay = 1e3,
    // Events
    onClose,
    onClick,
    // Standard
    className,
    style,
    id,
    "data-testid": dataTestId,
    ref
  } = props;
  const baseItemRef = useRef(null);
  const descriptionRef = useRef(null);
  const menuBtnRef = useRef(null);
  const listContext = useOptionalNotificationListContext();
  const itemId = id || baseItemRef.current?.nativeElement?.id;
  const ariaLabelledBy = React2.useMemo(() => {
    if (!itemId) return void 0;
    const ids = [];
    if (String(importance) === "Important") {
      ids.push(`${itemId}-importance`);
    }
    if (String(state) !== "None") {
      ids.push(`${itemId}-state`);
    }
    ids.push(`${itemId}-title`);
    ids.push(`${itemId}-read-status`);
    if (React2.Children.count(children) > 0) {
      ids.push(`${itemId}-description`);
    }
    if (footnotes && footnotes.length > 0) {
      ids.push(`${itemId}-footnotes`);
    }
    return ids.join(" ");
  }, [itemId, importance, state, children, footnotes]);
  const [showLoading, setShowLoading] = useState(false);
  useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  const [expanded, setExpanded] = useState(false);
  const [hasOverflow, setHasOverflow] = useState(false);
  const hasOverflowRef = useRef(false);
  const expandedRef = useRef(false);
  useEffect(() => {
    expandedRef.current = expanded;
  }, [expanded]);
  useEffect(() => {
    const el = descriptionRef.current;
    if (!el || String(wrappingType) === "Normal") return;
    const observer = new ResizeObserver(() => {
      if (!expandedRef.current) {
        const isOverflowing = el.scrollHeight > el.clientHeight;
        hasOverflowRef.current = isOverflowing;
        setHasOverflow(isOverflowing);
      }
    });
    observer.observe(el);
    return () => observer.disconnect();
  }, [wrappingType]);
  const toggleExpanded = useCallback(() => {
    setExpanded((prev) => !prev);
  }, []);
  const [menuOpen, setMenuOpen] = useState(false);
  const handleClick = useCallback(
    (e) => {
      onClick?.(e);
      const el = baseItemRef.current?.nativeElement;
      if (el) {
        listContext?.onItemClick?.({ item: el, originalEvent: e });
      }
    },
    [listContext, onClick]
  );
  const handleClose = useCallback(
    () => {
      onClose?.();
      const el = baseItemRef.current?.nativeElement;
      if (el) {
        listContext?.onItemClose?.({ item: el });
      }
    },
    [onClose, listContext]
  );
  useImperativeHandle(
    ref,
    () => ({
      focus: () => baseItemRef.current?.focus?.(),
      blur: () => baseItemRef.current?.blur?.(),
      isFocused: () => baseItemRef.current?.isFocused?.() ?? false,
      get nativeElement() {
        return baseItemRef.current?.nativeElement ?? null;
      }
    }),
    []
  );
  const normalizedState = state;
  const normalizedWrapping = wrappingType;
  const hasDescription = React2.Children.count(children) > 0;
  const hasFootnotes = footnotes && footnotes.length > 0;
  const showStateIcon = String(state) !== "None";
  return /* @__PURE__ */ jsxs(
    ListItemBase,
    {
      ref: baseItemRef,
      itemKey: id,
      type: "Active" /* Active */,
      onClick: handleClick,
      accessibleNameRef: ariaLabelledBy,
      className: cn(
        "relative border border-border rounded",
        "transition-colors duration-150",
        "hover:bg-[var(--table-row-hover)]",
        "active:bg-[var(--neutral-pressed-background-2)]",
        "bg-[var(--canvas-primary)]",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-notification-item": true,
      children: [
        showLoading && /* @__PURE__ */ jsx("div", { className: "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20 rounded", children: /* @__PURE__ */ jsx(Loader2, { className: "h-5 w-5 animate-spin text-sapphire-brand-foreground" }) }),
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: cn(
              "flex gap-3 w-full min-w-0 p-4",
              showLoading && "opacity-60 pointer-events-none"
            ),
            children: [
              String(importance) === "Important" && itemId && /* @__PURE__ */ jsx("span", { id: `${itemId}-importance`, className: "sr-only", children: "Important" }),
              showStateIcon && itemId && /* @__PURE__ */ jsx("span", { id: `${itemId}-state`, className: "sr-only", children: getStateText(state) }),
              itemId && /* @__PURE__ */ jsx("span", { id: `${itemId}-read-status`, className: "sr-only", children: read ? "Read" : "Unread" }),
              avatar && /* @__PURE__ */ jsx("div", { className: "shrink-0", children: avatar }),
              /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0", children: [
                /* @__PURE__ */ jsxs("div", { className: "flex items-start gap-1 pr-16", children: [
                  /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0 flex items-start gap-1", children: [
                    showStateIcon && /* @__PURE__ */ jsx(StateIcon, { state: normalizedState }),
                    /* @__PURE__ */ jsx(
                      "span",
                      {
                        id: itemId ? `${itemId}-title` : void 0,
                        className: cn(
                          titleVariants({ read, wrapping: normalizedWrapping }),
                          "flex-1 min-w-0"
                        ),
                        children: titleText
                      }
                    )
                  ] }),
                  /* @__PURE__ */ jsxs("div", { className: "absolute top-2 right-2 flex items-center gap-0.5 shrink-0", children: [
                    menu && /* @__PURE__ */ jsxs("span", { onClick: (e) => e.stopPropagation(), children: [
                      /* @__PURE__ */ jsx(
                        Button,
                        {
                          ref: menuBtnRef,
                          design: "Tertiary" /* Tertiary */,
                          size: "Medium" /* Medium */,
                          icon: /* @__PURE__ */ jsx(OverflowIcon, { className: "h-4 w-4" }),
                          onClick: (detail) => {
                            detail.originalEvent.stopPropagation();
                            setMenuOpen(!menuOpen);
                          },
                          accessibleName: "More actions",
                          accessibilityAttributes: { hasPopup: "menu" }
                        }
                      ),
                      React2.isValidElement(menu) && React2.cloneElement(
                        menu,
                        {
                          open: menuOpen,
                          opener: menuBtnRef.current?.nativeElement ?? null,
                          onClose: () => setMenuOpen(false)
                        }
                      )
                    ] }),
                    showClose && /* @__PURE__ */ jsx(
                      Button,
                      {
                        design: "Tertiary" /* Tertiary */,
                        size: "Medium" /* Medium */,
                        icon: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-4 w-4" }),
                        onClick: (detail) => {
                          detail.originalEvent.stopPropagation();
                          handleClose();
                        },
                        accessibleName: "Close notification"
                      }
                    )
                  ] })
                ] }),
                hasDescription && /* @__PURE__ */ jsx("div", { className: "mt-2", children: /* @__PURE__ */ jsx(
                  "div",
                  {
                    id: itemId ? `${itemId}-description` : void 0,
                    ref: descriptionRef,
                    className: cn(
                      "text-sm leading-normal text-[var(--text/text-secondary,#353c4a)]",
                      String(wrappingType) !== "Normal" && !expanded && "line-clamp-2",
                      String(wrappingType) === "Normal" && "break-words"
                    ),
                    children
                  }
                ) }),
                (hasFootnotes || hasOverflow || expanded) && /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 mt-2 text-sm text-[var(--text\\/text-tertiary,#616d85)]", children: [
                  hasFootnotes && /* @__PURE__ */ jsx(Fragment, { children: footnotes.map((footnote, index) => /* @__PURE__ */ jsxs(React2.Fragment, { children: [
                    index > 0 && /* @__PURE__ */ jsx("span", { children: "\u2022" }),
                    /* @__PURE__ */ jsx(
                      "span",
                      {
                        id: index === 0 && itemId ? `${itemId}-footnotes` : void 0,
                        children: footnote
                      }
                    )
                  ] }, index)) }),
                  (hasOverflow || expanded) && /* @__PURE__ */ jsx(
                    Link,
                    {
                      onClick: (detail) => {
                        detail.originalEvent.stopPropagation();
                        toggleExpanded();
                      },
                      children: expanded ? "Less" : "More"
                    }
                  )
                ] }),
                String(importance) === "Important" && /* @__PURE__ */ jsx("div", { className: "mt-2", children: /* @__PURE__ */ jsx(ImportanceBadge, { importance }) })
              ] })
            ]
          }
        )
      ]
    }
  );
}
function NotificationListGroupItem(props) {
  const {
    // Content
    titleText,
    children,
    // Behavior
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    growing = "None" /* None */,
    loading = false,
    loadingDelay = 1e3,
    // Events
    onToggle,
    onLoadMore,
    // Standard
    className,
    style,
    accessibleName,
    "data-testid": dataTestId,
    ref
  } = props;
  const groupRef = useRef(null);
  const listContext = useOptionalNotificationListContext();
  const contentId = useId();
  const [internalCollapsed, setInternalCollapsed] = useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const [showLoading, setShowLoading] = useState(false);
  useEffect(() => {
    if (loading) {
      const timer = setTimeout(() => setShowLoading(true), loadingDelay);
      return () => clearTimeout(timer);
    } else {
      setShowLoading(false);
    }
  }, [loading, loadingDelay]);
  const handleToggle = useCallback(() => {
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.(newCollapsed);
    listContext?.onItemToggle?.({
      item: groupRef.current,
      collapsed: newCollapsed
    });
  }, [isCollapsed, isControlled, onToggle, listContext]);
  const handleHeaderKeyDown = useCallback((e) => {
    const shouldToggle = e.key === "Enter" || e.key === " " || e.key === "+" && isCollapsed || e.key === "-" && !isCollapsed || e.key === "ArrowRight" && isCollapsed || e.key === "ArrowLeft" && !isCollapsed;
    if (!shouldToggle) return;
    e.preventDefault();
    if (e.key === "ArrowRight" || e.key === "ArrowLeft" || e.key === "+" || e.key === "-") {
      e.stopPropagation();
    }
    handleToggle();
  }, [handleToggle, isCollapsed]);
  useImperativeHandle(
    ref,
    () => ({
      focus: () => groupRef.current?.focus(),
      blur: () => groupRef.current?.blur(),
      isFocused: () => document.activeElement === groupRef.current,
      toggle: handleToggle,
      get nativeElement() {
        return groupRef.current;
      }
    }),
    [handleToggle]
  );
  return /* @__PURE__ */ jsx(
    "li",
    {
      ref: groupRef,
      role: "listitem",
      "aria-level": 1,
      "aria-labelledby": accessibleName ? void 0 : `${contentId}-title`,
      "aria-label": accessibleName || void 0,
      "data-notification-item": true,
      "data-notification-group": true,
      className: cn(
        "mb-2",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: /* @__PURE__ */ jsxs("div", { className: "w-full", children: [
        /* @__PURE__ */ jsxs(
          "div",
          {
            role: "button",
            "aria-expanded": !isCollapsed,
            "aria-controls": !isCollapsed ? contentId : void 0,
            "aria-label": accessibleName || titleText,
            tabIndex: 0,
            onClick: handleToggle,
            onKeyDown: handleHeaderKeyDown,
            className: cn(
              "flex items-center gap-1 w-full pl-1 pr-2 py-1 cursor-pointer rounded",
              "hover:bg-sapphire-canvas-tertiary focus:outline-none transition-colors",
              "focus:shadow-[inset_0_0_0_2px_var(--border-focus)]"
            ),
            children: [
              /* @__PURE__ */ jsx("span", { className: "shrink-0 h-8 w-8 flex items-center justify-center text-sapphire-text-primary", children: isCollapsed ? /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-4 w-4" }) }),
              /* @__PURE__ */ jsx(
                "span",
                {
                  id: `${contentId}-title`,
                  className: cn(
                    "flex-1 text-base leading-[22px] truncate font-semibold text-[var(--text/text-primary,#0b0c0f)]"
                  ),
                  children: titleText
                }
              )
            ]
          }
        ),
        !isCollapsed && /* @__PURE__ */ jsxs("div", { className: "relative", children: [
          showLoading && /* @__PURE__ */ jsx("div", { className: "absolute inset-0 flex items-center justify-center bg-background/80 backdrop-blur-sm z-20", children: /* @__PURE__ */ jsx(Loader2, { className: "h-5 w-5 animate-spin text-sapphire-brand-foreground" }) }),
          /* @__PURE__ */ jsx(
            "ul",
            {
              id: contentId,
              className: cn(
                "m-0 p-0 list-none flex flex-col gap-2",
                showLoading && "opacity-60 pointer-events-none"
              ),
              children
            }
          ),
          String(growing) === "Button" /* Button */ && /* @__PURE__ */ jsx("div", { className: "flex items-center justify-center py-2 border-t border-border", children: /* @__PURE__ */ jsxs(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              onClick: (detail) => {
                detail.originalEvent.stopPropagation();
                onLoadMore?.();
              },
              disabled: showLoading,
              className: "w-full mx-4",
              "data-growing-button": "true",
              children: [
                showLoading ? /* @__PURE__ */ jsx(Loader2, { className: "h-4 w-4 animate-spin mr-2" }) : null,
                "More"
              ]
            }
          ) })
        ] })
      ] })
    }
  );
}

export { NotificationList, NotificationListContext, NotificationListGroupItem, NotificationListItem, NotificationListItemImportance, NotificationListItemState, useNotificationListContext, useOptionalNotificationListContext };
