// @sap-ui/fx-components/dist/chunk-45VUVRRJ.js
import { Link } from './chunk-MCJ7D5KG.js';
import { ListItemBase } from './chunk-CTYK4HEK.js';
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { isPhone } from './chunk-YSPWI6FA.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { createContext, useRef, useState, useMemo, useEffect, useImperativeHandle, useCallback, useId, useContext, Children, isValidElement, cloneElement } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/breadcrumbs.ts
var BreadcrumbsDesign = /* @__PURE__ */ ((BreadcrumbsDesign2) => {
  BreadcrumbsDesign2["Standard"] = "Standard";
  BreadcrumbsDesign2["NoCurrentPage"] = "NoCurrentPage";
  return BreadcrumbsDesign2;
})(BreadcrumbsDesign || {});
var BreadcrumbsSeparator = /* @__PURE__ */ ((BreadcrumbsSeparator2) => {
  BreadcrumbsSeparator2["Slash"] = "Slash";
  BreadcrumbsSeparator2["BackSlash"] = "BackSlash";
  BreadcrumbsSeparator2["DoubleBackSlash"] = "DoubleBackSlash";
  BreadcrumbsSeparator2["DoubleSlash"] = "DoubleSlash";
  BreadcrumbsSeparator2["GreaterThan"] = "GreaterThan";
  BreadcrumbsSeparator2["DoubleGreaterThan"] = "DoubleGreaterThan";
  return BreadcrumbsSeparator2;
})(BreadcrumbsSeparator || {});
var SEPARATOR_MAP = {
  ["Slash" /* Slash */]: "/",
  ["BackSlash" /* BackSlash */]: "\\",
  ["DoubleBackSlash" /* DoubleBackSlash */]: "\\\\",
  ["DoubleSlash" /* DoubleSlash */]: "//",
  ["GreaterThan" /* GreaterThan */]: ">",
  ["DoubleGreaterThan" /* DoubleGreaterThan */]: ">>"
};
var BreadcrumbsContext = createContext(null);
var breadcrumbSeparatorVariants = cva([
  "mx-1 text-foreground font-semibold text-base select-none shrink-0"
]);
function isBreadcrumbsItem(child) {
  return child.type?.__isBreadcrumbsItem === true;
}
function extractItems(children) {
  const items = [];
  Children.forEach(children, (child) => {
    if (isValidElement(child) && isBreadcrumbsItem(child)) {
      items.push(child.props);
    }
  });
  return items;
}
function isShowKey(e) {
  if (e.key === "F4" && !e.altKey && !e.ctrlKey && !e.metaKey && !e.shiftKey) return true;
  if (e.altKey && (e.key === "ArrowDown" || e.key === "ArrowUp")) return true;
  return false;
}
function injectIndices(children) {
  let idx = 0;
  return Children.map(children, (child) => {
    if (isValidElement(child) && isBreadcrumbsItem(child)) {
      const injected = cloneElement(child, { _index: idx });
      idx++;
      return injected;
    }
    return child;
  });
}
function Breadcrumbs({
  design = "Standard" /* Standard */,
  separators = "Slash" /* Slash */,
  linkDesign = "Default" /* Default */,
  overflowCount: controlledOverflowCount,
  noPadding = false,
  children,
  accessibleName = "Breadcrumb Trail",
  accessibleDescription,
  accessibleDescriptionRef,
  onItemClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const navRef = useRef(null);
  const olRef = useRef(null);
  const measureRef = useRef(null);
  const overflowBtnRef = useRef(null);
  const itemRefs = useRef(/* @__PURE__ */ new Map());
  const [internalOverflowSize, setInternalOverflowSize] = useState(0);
  const [popoverOpen, setPopoverOpen] = useState(false);
  const [focusedIndex, setFocusedIndex] = useState(-1);
  const [phone] = useState(() => isPhone());
  const allItems = useMemo(() => extractItems(children), [children]);
  const totalCount = allItems.length;
  const rawOverflow = controlledOverflowCount !== void 0 ? controlledOverflowCount : internalOverflowSize;
  const overflowSize = Math.max(0, Math.min(rawOverflow, totalCount));
  const isStandard = design === "Standard" /* Standard */;
  const separatorChar = SEPARATOR_MAP[separators] ?? "/";
  const overflowItems = useMemo(
    () => allItems.slice(0, overflowSize),
    [allItems, overflowSize]
  );
  const visibleItems = useMemo(
    () => allItems.slice(overflowSize),
    [allItems, overflowSize]
  );
  const hasOverflow = overflowItems.length > 0;
  const allOverflowed = overflowSize >= totalCount && totalCount > 0;
  const lastItem = visibleItems[visibleItems.length - 1];
  const endsWithCurrentLocation = isStandard && totalCount > 0;
  const lastItemIsLink = !!lastItem?.href;
  const linkItems = useMemo(() => {
    if (endsWithCurrentLocation && !lastItemIsLink) {
      return visibleItems.slice(0, -1);
    }
    return visibleItems;
  }, [visibleItems, endsWithCurrentLocation, lastItemIsLink]);
  const focusableIds = useMemo(() => {
    const ids = [];
    if (hasOverflow) ids.push("overflow");
    linkItems.forEach((_, i) => ids.push(`link-${overflowSize + i}`));
    return ids;
  }, [hasOverflow, linkItems, overflowSize, endsWithCurrentLocation, lastItemIsLink, totalCount]);
  useEffect(() => {
    setFocusedIndex((prev) => {
      if (prev < 0) return prev;
      return prev >= focusableIds.length ? focusableIds.length - 1 : prev;
    });
  }, [focusableIds]);
  useImperativeHandle(ref, () => ({
    focus() {
      const first = focusableIds[0];
      if (!first) return;
      if (first === "overflow") {
        overflowBtnRef.current?.focus();
      } else {
        const globalIdx = Number(first.split("-")[1]);
        const el = itemRefs.current.get(globalIdx);
        el?.focus();
      }
    },
    get nativeElement() {
      return navRef.current;
    }
  }));
  const widthCacheRef = useRef([]);
  const overflowBtnWidthRef = useRef(0);
  const measureScheduledRef = useRef(false);
  const measureAndUpdate = useCallback(() => {
    const measureOl = measureRef.current;
    const visibleOl = olRef.current;
    if (!measureOl || !visibleOl) return;
    const availableWidth = visibleOl.offsetWidth;
    if (availableWidth === 0) return;
    const wrappers = measureOl.querySelectorAll("[data-bc-measure]");
    const widths = [];
    wrappers.forEach((wrapper) => {
      const idx = Number(wrapper.dataset.bcMeasure);
      widths[idx] = Math.ceil(wrapper.getBoundingClientRect().width);
    });
    widthCacheRef.current = widths;
    const overflowEl = measureOl.querySelector("[data-bc-measure-overflow]");
    overflowBtnWidthRef.current = overflowEl ? Math.ceil(overflowEl.getBoundingClientRect().width) : 0;
    let requiredWidth = widths.reduce((s, w) => s + (w || 0), 0);
    let newOverflow = 0;
    const maxOverflow = Math.max(0, totalCount - 1);
    if (requiredWidth > availableWidth && maxOverflow > 0) {
      requiredWidth += overflowBtnWidthRef.current;
      while (requiredWidth > availableWidth && newOverflow < maxOverflow) {
        requiredWidth -= widths[newOverflow] || 0;
        newOverflow++;
      }
    }
    setInternalOverflowSize((prev) => prev !== newOverflow ? newOverflow : prev);
  }, [totalCount]);
  useEffect(() => {
    if (controlledOverflowCount !== void 0) return;
    const ol = olRef.current;
    if (!ol) return;
    const ro = new ResizeObserver(() => {
      if (!measureScheduledRef.current) {
        measureScheduledRef.current = true;
        requestAnimationFrame(() => {
          measureScheduledRef.current = false;
          measureAndUpdate();
        });
      }
    });
    ro.observe(ol);
    return () => ro.disconnect();
  }, [measureAndUpdate, controlledOverflowCount]);
  useEffect(() => {
    if (controlledOverflowCount !== void 0) return;
    const frameId = requestAnimationFrame(() => measureAndUpdate());
    return () => cancelAnimationFrame(frameId);
  }, [totalCount, children, measureAndUpdate, controlledOverflowCount]);
  const fireItemClick = useCallback(
    (e, itemProps, itemIndex) => {
      const detail = {
        originalEvent: e,
        item: itemProps,
        index: itemIndex,
        altKey: e.altKey,
        ctrlKey: e.ctrlKey,
        metaKey: e.metaKey,
        shiftKey: e.shiftKey
      };
      onItemClick?.(detail);
    },
    [onItemClick]
  );
  const handleLinkClick = useCallback(
    (e, itemProps, itemIndex) => {
      fireItemClick(e, itemProps, itemIndex);
    },
    [fireItemClick]
  );
  const handleCurrentPageClick = useCallback(
    (e, itemProps, itemIndex) => {
      fireItemClick(e, itemProps, itemIndex);
    },
    [fireItemClick]
  );
  const togglePopover = useCallback(() => {
    setPopoverOpen((prev) => !prev);
  }, []);
  const registerItemRef = useCallback(
    (globalIndex, element) => {
      if (element) itemRefs.current.set(globalIndex, element);
      else itemRefs.current.delete(globalIndex);
    },
    []
  );
  const moveFocus = useCallback(
    (newIdx) => {
      if (newIdx < 0 || newIdx >= focusableIds.length) return;
      setFocusedIndex(newIdx);
      const slotId = focusableIds[newIdx];
      if (slotId === "overflow") {
        overflowBtnRef.current?.focus();
      } else {
        const globalIdx = Number(slotId.split("-")[1]);
        const el = itemRefs.current.get(globalIdx);
        el?.focus();
      }
    },
    [focusableIds]
  );
  const handleOlKeyDown = useCallback(
    (e) => {
      const currentIdx = focusedIndex >= 0 ? focusedIndex : 0;
      const isRTL = document.documentElement.dir === "rtl";
      switch (e.key) {
        case "ArrowRight": {
          if (e.altKey) break;
          e.preventDefault();
          const nextIdx = isRTL ? Math.max(currentIdx - 1, 0) : Math.min(currentIdx + 1, focusableIds.length - 1);
          moveFocus(nextIdx);
          break;
        }
        case "ArrowDown": {
          if (e.altKey) break;
          e.preventDefault();
          moveFocus(Math.min(currentIdx + 1, focusableIds.length - 1));
          break;
        }
        case "ArrowLeft": {
          if (e.altKey) break;
          e.preventDefault();
          const prevIdx = isRTL ? Math.min(currentIdx + 1, focusableIds.length - 1) : Math.max(currentIdx - 1, 0);
          moveFocus(prevIdx);
          break;
        }
        case "ArrowUp": {
          if (e.altKey) break;
          e.preventDefault();
          moveFocus(Math.max(currentIdx - 1, 0));
          break;
        }
        case "Home": {
          e.preventDefault();
          moveFocus(0);
          break;
        }
        case "End": {
          e.preventDefault();
          moveFocus(focusableIds.length - 1);
          break;
        }
      }
      if (isShowKey(e) && focusableIds[currentIdx] === "overflow") {
        e.preventDefault();
        togglePopover();
      }
    },
    [focusedIndex, focusableIds, moveFocus, togglePopover]
  );
  const handleOlFocusIn = useCallback(
    (e) => {
      const target = e.target;
      if (target === overflowBtnRef.current?.nativeElement) {
        const idx = focusableIds.indexOf("overflow");
        if (idx >= 0) setFocusedIndex(idx);
        return;
      }
      for (const [globalIdx2, el] of itemRefs.current.entries()) {
        if (el && (el === target || el.contains(target))) {
          const slotId = `link-${globalIdx2}`;
          const idx = focusableIds.indexOf(slotId);
          if (idx >= 0) setFocusedIndex(idx);
          return;
        }
      }
      const globalIdx = target.closest("[data-bc-link-index]")?.dataset.bcLinkIndex;
      if (globalIdx != null) {
        const slotId = `link-${globalIdx}`;
        const idx = focusableIds.indexOf(slotId);
        if (idx >= 0) setFocusedIndex(idx);
      }
    },
    [focusableIds]
  );
  const getTabIndex = useCallback(
    (slotId) => {
      const idx = focusableIds.indexOf(slotId);
      if (focusedIndex < 0) return idx === 0 ? 0 : -1;
      return idx === focusedIndex ? 0 : -1;
    },
    [focusableIds, focusedIndex]
  );
  const visibleCount = visibleItems.length;
  const hasCurrentPageLabel = endsWithCurrentLocation && !lastItemIsLink;
  const contextValue = useMemo(
    () => ({
      separatorChar,
      isStandard,
      totalCount,
      visibleCount,
      overflowSize,
      linkDesign,
      getTabIndex,
      onLinkClick: handleLinkClick,
      onCurrentPageClick: handleCurrentPageClick,
      registerItemRef,
      hasCurrentPageLabel
    }),
    [separatorChar, isStandard, totalCount, visibleCount, overflowSize, linkDesign, getTabIndex, handleLinkClick, handleCurrentPageClick, registerItemRef, hasCurrentPageLabel]
  );
  const indexedChildren = useMemo(() => injectIndices(children), [children]);
  const descriptionId = useId();
  return /* @__PURE__ */ jsxs(
    "nav",
    {
      ref: navRef,
      id,
      "aria-label": accessibleName,
      "aria-describedby": [
        accessibleDescriptionRef,
        accessibleDescription ? descriptionId : null
      ].filter(Boolean).join(" ") || void 0,
      className: cn("block w-full relative", className),
      style,
      "data-testid": dataTestId,
      children: [
        accessibleDescription && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription }),
        /* @__PURE__ */ jsxs(
          "ol",
          {
            ref: measureRef,
            "aria-hidden": "true",
            className: "flex items-center list-none m-0 p-0 whitespace-nowrap absolute top-0 left-0 w-full pointer-events-none",
            style: { visibility: "hidden", height: 0, overflow: "hidden" },
            children: [
              /* @__PURE__ */ jsxs("li", { "data-bc-measure-overflow": true, className: "inline-flex items-center shrink-0", children: [
                /* @__PURE__ */ jsxs("span", { className: "inline-flex items-center gap-0.5 text-base px-0.5 py-0.5", children: [
                  /* @__PURE__ */ jsx("span", { children: "\u2026" }),
                  /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-3 w-3 block" })
                ] }),
                /* @__PURE__ */ jsx("span", { className: "mx-1 text-base font-semibold select-none", children: separatorChar })
              ] }),
              allItems.map((item, idx) => {
                const isLastItem = idx === totalCount - 1;
                const showSep = !(isStandard && isLastItem && !item.href);
                return /* @__PURE__ */ jsxs(
                  "li",
                  {
                    "data-bc-measure": idx,
                    className: "inline-flex items-center shrink-0",
                    children: [
                      /* @__PURE__ */ jsx(
                        "span",
                        {
                          className: "text-base font-semibold whitespace-nowrap px-0.5 py-0.5",
                          children: item.children
                        }
                      ),
                      showSep && /* @__PURE__ */ jsx("span", { className: "mx-1 text-base font-semibold select-none", children: separatorChar })
                    ]
                  },
                  idx
                );
              })
            ]
          }
        ),
        /* @__PURE__ */ jsxs(
          "ol",
          {
            ref: olRef,
            className: cn(
              "flex items-center list-none m-0 p-0 whitespace-nowrap overflow-x-clip overflow-y-visible",
              !noPadding && "px-1 py-0.5"
            ),
            onKeyDown: handleOlKeyDown,
            onFocusCapture: handleOlFocusIn,
            children: [
              /* @__PURE__ */ jsxs(
                "li",
                {
                  className: cn(
                    "inline-flex items-center shrink-0",
                    !hasOverflow && "hidden"
                  ),
                  children: [
                    /* @__PURE__ */ jsx(
                      Link,
                      {
                        ref: overflowBtnRef,
                        accessibleRole: "Button" /* Button */,
                        accessibleName: "More",
                        accessibilityAttributes: {
                          hasPopup: hasOverflow ? "listbox" : void 0,
                          expanded: hasOverflow ? popoverOpen : void 0
                        },
                        design: linkDesign,
                        wrappingType: "None" /* None */,
                        tabIndex: hasOverflow ? getTabIndex("overflow") : -1,
                        onClick: () => togglePopover(),
                        endIcon: /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-3 w-3 block", "aria-hidden": "true" }),
                        className: "text-sm px-0.5 py-0.5 gap-0",
                        children: /* @__PURE__ */ jsx("span", { "aria-hidden": "true", children: "\u2026" })
                      }
                    ),
                    !allOverflowed && /* @__PURE__ */ jsx(
                      "span",
                      {
                        className: breadcrumbSeparatorVariants(),
                        "aria-hidden": "true",
                        children: separatorChar
                      }
                    )
                  ]
                }
              ),
              /* @__PURE__ */ jsx(BreadcrumbsContext.Provider, { value: contextValue, children: indexedChildren })
            ]
          }
        ),
        /* @__PURE__ */ jsx(
          ResponsivePopover,
          {
            opener: overflowBtnRef,
            open: popoverOpen,
            onClose: () => setPopoverOpen(false),
            placement: "Bottom" /* Bottom */,
            horizontalAlign: "Start" /* Start */,
            hideArrow: true,
            noPadding: true,
            accessibleRole: "None" /* None */,
            contentOnlyOnDesktop: true,
            headerText: "Breadcrumb Trail",
            children: /* @__PURE__ */ jsx(
              List,
              {
                accessibleRole: "Menu" /* Menu */,
                accessibleName: "Breadcrumb Trail",
                children: [...phone ? allItems : overflowItems].reverse().map((item, displayIdx) => {
                  const globalIdx = phone ? totalCount - 1 - displayIdx : overflowSize - 1 - displayIdx;
                  return /* @__PURE__ */ jsx(
                    ListItemBase,
                    {
                      itemKey: `overflow-${globalIdx}`,
                      onClick: (e) => {
                        fireItemClick(e, item, globalIdx);
                        setPopoverOpen(false);
                      },
                      className: "px-3 py-2 text-sm",
                      children: item.children
                    },
                    globalIdx
                  );
                })
              }
            )
          }
        )
      ]
    }
  );
}
var breadcrumbCurrentVariants = cva([
  "text-base font-semibold text-foreground truncate",
  "rounded-sm px-0.5 py-0.5",
  "cursor-default"
]);
function BreadcrumbsItem({
  href,
  target,
  accessibleName,
  children,
  className,
  style,
  "data-testid": dataTestId,
  _index,
  ref
}) {
  const ctx = useContext(BreadcrumbsContext);
  const linkRef = useRef(null);
  const spanRef = useRef(null);
  const globalIndex = _index ?? -1;
  const active = ctx != null && _index != null;
  const isStandard = ctx?.isStandard ?? false;
  const totalCount = ctx?.totalCount ?? 0;
  const visibleCount = ctx?.visibleCount ?? 0;
  const overflowSize = ctx?.overflowSize ?? 0;
  const separatorChar = ctx?.separatorChar ?? "/";
  const linkDesign = ctx?.linkDesign ?? "Default" /* Default */;
  const getTabIndex = ctx?.getTabIndex;
  const onLinkClick = ctx?.onLinkClick;
  const registerItemRef = ctx?.registerItemRef;
  const isOverflowed = globalIndex < overflowSize;
  const isLastItem = globalIndex === totalCount - 1;
  const isCurrentPage = isStandard && isLastItem && !href;
  const isCurrentPageLink = isStandard && isLastItem && !!href;
  useEffect(() => {
    if (!active || isOverflowed || !registerItemRef) return;
    const el = isCurrentPage ? spanRef.current : linkRef.current?.nativeElement ?? null;
    registerItemRef(globalIndex, el);
    return () => registerItemRef(globalIndex, null);
  }, [active, isOverflowed, isCurrentPage, globalIndex, registerItemRef]);
  useEffect(() => {
    if (!active || isOverflowed || isCurrentPage) return;
    const el = linkRef.current?.nativeElement;
    if (!el) return;
    el.setAttribute("data-bc-link-index", String(globalIndex));
    if (isCurrentPageLink) {
      el.setAttribute("aria-current", "page");
    } else {
      el.removeAttribute("aria-current");
    }
  }, [active, isOverflowed, isCurrentPage, isCurrentPageLink, globalIndex]);
  if (!active) return null;
  if (isOverflowed) return null;
  const positionInVisible = globalIndex - overflowSize + 1;
  if (isCurrentPage) {
    const textContent2 = typeof children === "string" ? children : "";
    const posLabel2 = `${visibleCount} of ${visibleCount}`;
    const ariaLabel2 = [textContent2, accessibleName, posLabel2].filter(Boolean).join(" ");
    return /* @__PURE__ */ jsx(
      "li",
      {
        ref,
        className: cn("inline-flex items-center min-w-0 flex-1", className),
        style,
        "data-testid": dataTestId,
        children: /* @__PURE__ */ jsx(
          "span",
          {
            ref: spanRef,
            "data-bc-current": true,
            "aria-current": "page",
            "aria-label": ariaLabel2,
            className: breadcrumbCurrentVariants(),
            children
          }
        )
      }
    );
  }
  const textContent = typeof children === "string" ? children : "";
  const posLabel = `${positionInVisible} of ${visibleCount}`;
  const ariaLabel = [textContent, accessibleName, posLabel].filter(Boolean).join(" ");
  const isTerminalItem = isCurrentPageLink || !isStandard && isLastItem;
  const needsSeparator = !isTerminalItem;
  const ownProps = {
    href,
    target,
    accessibleName,
    children,
    className,
    style,
    "data-testid": dataTestId
  };
  return /* @__PURE__ */ jsxs(
    "li",
    {
      ref,
      className: cn("inline-flex items-center min-w-0", isTerminalItem ? "" : "shrink-0", className),
      style,
      "data-testid": dataTestId,
      children: [
        /* @__PURE__ */ jsx(
          Link,
          {
            ref: linkRef,
            href,
            target,
            rel: target === "_blank" ? "noopener noreferrer" : void 0,
            design: isCurrentPageLink ? "Subtle" /* Subtle */ : linkDesign,
            wrappingType: "None" /* None */,
            accessibleName: ariaLabel,
            tabIndex: getTabIndex?.(`link-${globalIndex}`) ?? -1,
            onClick: ({ originalEvent }) => onLinkClick?.(originalEvent, ownProps, globalIndex),
            className: cn(
              "text-base font-semibold px-0.5 py-0.5 mx-0 my-0",
              isCurrentPageLink && "!text-foreground !no-underline",
              isTerminalItem && "truncate max-w-full"
            ),
            children
          }
        ),
        needsSeparator && /* @__PURE__ */ jsx(
          "span",
          {
            className: breadcrumbSeparatorVariants(),
            "aria-hidden": "true",
            children: separatorChar
          }
        )
      ]
    }
  );
}
BreadcrumbsItem.__isBreadcrumbsItem = true;
BreadcrumbsItem.displayName = "BreadcrumbsItem";

export { Breadcrumbs, BreadcrumbsDesign, BreadcrumbsItem, BreadcrumbsSeparator, SEPARATOR_MAP };
