// @sap-ui/fx-components/dist/chunk-MQ3F6SMX.js
import { NavigationRightArrowIcon, ExpandIcon, CollapseIcon } from './chunk-T7Y6YDVS.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useRef, useId, useState, useImperativeHandle, useCallback, useEffect } from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

// src/types/panel.ts
var PanelAccessibleRole = /* @__PURE__ */ ((PanelAccessibleRole2) => {
  PanelAccessibleRole2["Complementary"] = "Complementary";
  PanelAccessibleRole2["Form"] = "Form";
  PanelAccessibleRole2["Region"] = "Region";
  return PanelAccessibleRole2;
})(PanelAccessibleRole || {});
var PanelDesign = /* @__PURE__ */ ((PanelDesign2) => {
  PanelDesign2["Primary"] = "Primary";
  PanelDesign2["Secondary"] = "Secondary";
  PanelDesign2["Child"] = "Child";
  return PanelDesign2;
})(PanelDesign || {});
var roleMap = {
  ["Complementary" /* Complementary */]: "complementary",
  ["Form" /* Form */]: "form",
  ["Region" /* Region */]: "region"
};
var levelToTag = {
  ["H1" /* H1 */]: "h1",
  ["H2" /* H2 */]: "h2",
  ["H3" /* H3 */]: "h3",
  ["H4" /* H4 */]: "h4",
  ["H5" /* H5 */]: "h5",
  ["H6" /* H6 */]: "h6"
};
var rootDesignClasses = {
  ["Primary" /* Primary */]: "bg-sapphire-card-bg-primary border border-sapphire-border-primary",
  ["Secondary" /* Secondary */]: "bg-sapphire-background-secondary border border-sapphire-border-primary",
  ["Child" /* Child */]: ""
};
var headerDesignClasses = {
  ["Primary" /* Primary */]: "pl-3 pr-4 py-3 gap-3",
  ["Secondary" /* Secondary */]: "pl-3 pr-4 py-3 gap-3",
  ["Child" /* Child */]: "pl-1 pr-2 py-1 gap-1"
};
var headerTitleClasses = {
  ["Primary" /* Primary */]: "text-base font-normal",
  ["Secondary" /* Secondary */]: "text-base font-normal",
  ["Child" /* Child */]: "text-sm font-normal"
};
var contentDesignClasses = {
  ["Primary" /* Primary */]: "p-4",
  ["Secondary" /* Secondary */]: "p-4",
  ["Child" /* Child */]: "p-3"
};
var headerBgClasses = {
  ["Primary" /* Primary */]: "bg-sapphire-card-bg-primary",
  ["Secondary" /* Secondary */]: "bg-sapphire-background-secondary",
  ["Child" /* Child */]: "bg-background"
};
var footerDesignClasses = {
  ["Primary" /* Primary */]: "px-4 py-4",
  ["Secondary" /* Secondary */]: "px-4 py-4 bg-sapphire-card-bg-secondary",
  ["Child" /* Child */]: ""
};
function Panel(props) {
  const {
    // Visual
    design = "Primary" /* Primary */,
    // Content
    children,
    headerText,
    header,
    subtitle,
    endSlot,
    footer,
    // Behavior
    fixed = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    noAnimation = false,
    stickyHeader = false,
    noPadding = false,
    // Accessibility
    accessibleRole = "Form" /* Form */,
    headerLevel = "H2" /* H2 */,
    accessibleName,
    // Events
    onToggle,
    // Standard
    className,
    style,
    id: providedId,
    "data-testid": dataTestId,
    ref
  } = props;
  const panelRef = useRef(null);
  const headerRef = useRef(null);
  const contentWrapperRef = useRef(null);
  const contentRef = useRef(null);
  const generatedId = useId();
  const panelId = providedId || generatedId;
  const headerId = `${panelId}-header`;
  const contentId = `${panelId}-content`;
  const [internalCollapsed, setInternalCollapsed] = useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = fixed ? false : isControlled ? controlledCollapsed : internalCollapsed;
  const animationRef = useRef(null);
  const prevCollapsedRef = useRef(isCollapsed);
  const [contentExpanded, setContentExpanded] = useState(!isCollapsed);
  useImperativeHandle(ref, () => ({
    focus() {
      headerRef.current?.focus();
    },
    blur() {
      headerRef.current?.blur();
    },
    isFocused() {
      return document.activeElement === headerRef.current;
    },
    get nativeElement() {
      return panelRef.current;
    }
  }), []);
  const handleToggle = useCallback(() => {
    if (fixed) return;
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.({ collapsed: newCollapsed });
  }, [fixed, isCollapsed, isControlled, onToggle]);
  const pendingToggleRef = useRef(false);
  const handleKeyDown = useCallback((e) => {
    if (fixed) return;
    if (e.key === "Enter") {
      e.preventDefault();
      handleToggle();
    }
    if (e.key === " ") {
      e.preventDefault();
      pendingToggleRef.current = true;
    }
    if (e.key === "Escape" && pendingToggleRef.current) {
      e.preventDefault();
      pendingToggleRef.current = false;
    }
  }, [fixed, handleToggle]);
  const handleKeyUp = useCallback((e) => {
    if (fixed) return;
    if (e.key === " ") {
      if (pendingToggleRef.current) {
        handleToggle();
      }
      pendingToggleRef.current = false;
    }
  }, [fixed, handleToggle]);
  useEffect(() => {
    const wrapper = contentWrapperRef.current;
    if (!wrapper) {
      prevCollapsedRef.current = isCollapsed;
      return;
    }
    const wasCollapsed = prevCollapsedRef.current;
    prevCollapsedRef.current = isCollapsed;
    if (wasCollapsed === isCollapsed) return;
    if (noAnimation || fixed) {
      if (isCollapsed) {
        wrapper.style.display = "none";
        setContentExpanded(false);
      } else {
        wrapper.style.display = "";
        setContentExpanded(true);
      }
      return;
    }
    if (animationRef.current !== null) {
      cancelAnimationFrame(animationRef.current);
      animationRef.current = null;
    }
    const DURATION = 400;
    if (isCollapsed) {
      const computed = getComputedStyle(wrapper);
      const startHeight = parseFloat(computed.height);
      const startPaddingTop = parseFloat(computed.paddingTop);
      const startPaddingBottom = parseFloat(computed.paddingBottom);
      wrapper.style.overflow = "hidden";
      let start = null;
      const step = (timestamp) => {
        if (start === null) start = timestamp;
        const elapsed = timestamp - start;
        const progress = Math.min(elapsed / DURATION, 1);
        wrapper.style.height = `${startHeight * (1 - progress)}px`;
        wrapper.style.paddingTop = `${startPaddingTop * (1 - progress)}px`;
        wrapper.style.paddingBottom = `${startPaddingBottom * (1 - progress)}px`;
        if (progress < 1) {
          animationRef.current = requestAnimationFrame(step);
        } else {
          wrapper.style.display = "none";
          wrapper.style.overflow = "";
          wrapper.style.height = "";
          wrapper.style.paddingTop = "";
          wrapper.style.paddingBottom = "";
          animationRef.current = null;
          setContentExpanded(false);
        }
      };
      animationRef.current = requestAnimationFrame(step);
    } else {
      wrapper.style.display = "block";
      setContentExpanded(true);
      const computed = getComputedStyle(wrapper);
      const targetHeight = parseFloat(computed.height);
      const targetPaddingTop = parseFloat(computed.paddingTop);
      const targetPaddingBottom = parseFloat(computed.paddingBottom);
      wrapper.style.overflow = "hidden";
      wrapper.style.height = "0px";
      wrapper.style.paddingTop = "0px";
      wrapper.style.paddingBottom = "0px";
      let start = null;
      const step = (timestamp) => {
        if (start === null) start = timestamp;
        const elapsed = timestamp - start;
        const progress = Math.min(elapsed / DURATION, 1);
        wrapper.style.display = "block";
        wrapper.style.height = `${targetHeight * progress}px`;
        wrapper.style.paddingTop = `${targetPaddingTop * progress}px`;
        wrapper.style.paddingBottom = `${targetPaddingBottom * progress}px`;
        if (progress < 1) {
          animationRef.current = requestAnimationFrame(step);
        } else {
          wrapper.style.overflow = "";
          wrapper.style.height = "";
          wrapper.style.paddingTop = "";
          wrapper.style.paddingBottom = "";
          animationRef.current = null;
        }
      };
      animationRef.current = requestAnimationFrame(step);
    }
    return () => {
      if (animationRef.current !== null) {
        cancelAnimationFrame(animationRef.current);
        animationRef.current = null;
      }
    };
  }, [isCollapsed, noAnimation, fixed]);
  const resolvedDesign = design in rootDesignClasses ? design : "Primary" /* Primary */;
  const ariaRole = roleMap[accessibleRole] || "form";
  const HeadingTag = levelToTag[headerLevel] || "h2";
  const hasHeader = headerText || header;
  const isInteractive = !fixed;
  const isChild = resolvedDesign === "Child" /* Child */;
  const toggleIcon = isChild ? /* @__PURE__ */ jsx(
    NavigationRightArrowIcon,
    {
      className: cn(
        "h-4 w-4 shrink-0 text-sapphire-text-tertiary",
        !noAnimation && "transition-transform duration-200",
        !isCollapsed && "rotate-90"
      ),
      "aria-hidden": "true"
    }
  ) : isCollapsed ? /* @__PURE__ */ jsx(ExpandIcon, { className: "h-4 w-4 shrink-0 text-sapphire-text-tertiary", "aria-hidden": "true" }) : /* @__PURE__ */ jsx(CollapseIcon, { className: "h-4 w-4 shrink-0 text-sapphire-text-tertiary", "aria-hidden": "true" });
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: panelRef,
      id: panelId,
      role: ariaRole,
      "aria-label": accessibleName,
      "aria-labelledby": !accessibleName && hasHeader ? headerId : void 0,
      className: cn(
        "rounded-lg overflow-clip text-sapphire-text-primary",
        rootDesignClasses[resolvedDesign],
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        (hasHeader || isInteractive) && /* @__PURE__ */ jsxs(
          "div",
          {
            ref: headerRef,
            id: headerId,
            role: isInteractive ? "button" : void 0,
            tabIndex: isInteractive ? 0 : void 0,
            "aria-expanded": isInteractive ? !isCollapsed : void 0,
            "aria-controls": isInteractive ? contentId : void 0,
            onClick: isInteractive ? handleToggle : void 0,
            onKeyDown: isInteractive ? handleKeyDown : void 0,
            onKeyUp: isInteractive ? handleKeyUp : void 0,
            className: cn(
              "flex items-center min-h-11",
              headerDesignClasses[resolvedDesign],
              !stickyHeader && headerBgClasses[resolvedDesign],
              stickyHeader && cn("sticky top-0 z-10", headerBgClasses[resolvedDesign]),
              isInteractive && "cursor-pointer hover:bg-accent transition-colors",
              isInteractive && "focus:outline-none focus:ring-2 focus:ring-ring focus:ring-inset",
              "select-none",
              isChild ? "rounded-lg" : "rounded-t-lg",
              isCollapsed && !isChild && "rounded-b-lg"
            ),
            "data-part": "header",
            children: [
              isInteractive && toggleIcon,
              header ? /* @__PURE__ */ jsx("div", { className: "flex-1 min-w-0", "data-part": "header-content", children: header }) : /* @__PURE__ */ jsxs(Fragment, { children: [
                (headerText || subtitle) && /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0", children: [
                  headerText && /* @__PURE__ */ jsx(
                    HeadingTag,
                    {
                      className: cn(
                        "text-sapphire-text-primary truncate m-0",
                        headerTitleClasses[resolvedDesign]
                      ),
                      "data-part": "header-text",
                      children: headerText
                    }
                  ),
                  subtitle && /* @__PURE__ */ jsx(
                    "p",
                    {
                      className: "text-sm text-sapphire-text-tertiary truncate m-0",
                      "data-part": "header-subtitle",
                      children: subtitle
                    }
                  )
                ] }),
                endSlot && /* @__PURE__ */ jsx(
                  "div",
                  {
                    className: "flex-shrink-0",
                    "data-part": "header-end-slot",
                    onClick: (e) => e.stopPropagation(),
                    onKeyDown: (e) => e.stopPropagation(),
                    onKeyUp: (e) => e.stopPropagation(),
                    children: endSlot
                  }
                )
              ] })
            ]
          }
        ),
        /* @__PURE__ */ jsxs(
          "div",
          {
            ref: contentWrapperRef,
            id: contentId,
            role: "region",
            "aria-labelledby": hasHeader ? headerId : void 0,
            style: !contentExpanded ? { display: "none" } : void 0,
            "data-part": "content",
            children: [
              /* @__PURE__ */ jsx(
                "div",
                {
                  ref: contentRef,
                  className: cn(
                    noPadding ? "" : contentDesignClasses[resolvedDesign]
                  ),
                  children
                }
              ),
              footer && !isChild && /* @__PURE__ */ jsx(
                "div",
                {
                  className: cn(
                    "flex items-center gap-2",
                    footerDesignClasses[resolvedDesign]
                  ),
                  "data-part": "footer",
                  children: footer
                }
              )
            ]
          }
        )
      ]
    }
  );
}

export { Panel, PanelAccessibleRole, PanelDesign };
