// @sap-ui/fx-components/dist/chunk-PXVVADOR.js
import { SlimArrowLeftIcon } from './chunk-6XLAD3TP.js';
import { SlimArrowRightIcon } from './chunk-ZQSLIXR7.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import getCachedLocaleDataInstance from '@ui5/webcomponents-localization/dist/getCachedLocaleDataInstance.js';
import Locale2 from '@ui5/webcomponents-base/dist/locale/Locale.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import i18n from 'i18next';
import { initReactI18next, useTranslation } from 'react-i18next';
import '@ui5/webcomponents-localization/dist/Assets.js';
import '@ui5/webcomponents-localization/dist/sap/ui/core/date/Gregorian.js';
import { fetchCldr } from '@ui5/webcomponents-base/dist/asset-registries/LocaleData.js';
import React, { useState, useEffect, useRef, useCallback, useMemo, useImperativeHandle } from 'react';

// src/types/calendar.ts
var CalendarSelectionMode = /* @__PURE__ */ ((CalendarSelectionMode2) => {
  CalendarSelectionMode2["Single"] = "Single";
  CalendarSelectionMode2["Multiple"] = "Multiple";
  CalendarSelectionMode2["Range"] = "Range";
  return CalendarSelectionMode2;
})(CalendarSelectionMode || {});
var CalendarWeekNumbering = /* @__PURE__ */ ((CalendarWeekNumbering2) => {
  CalendarWeekNumbering2["Default"] = "Default";
  CalendarWeekNumbering2["ISO_8601"] = "ISO_8601";
  CalendarWeekNumbering2["MiddleEastern"] = "MiddleEastern";
  CalendarWeekNumbering2["WesternTraditional"] = "WesternTraditional";
  return CalendarWeekNumbering2;
})(CalendarWeekNumbering || {});
var CalendarPickersMode = /* @__PURE__ */ ((CalendarPickersMode2) => {
  CalendarPickersMode2["DAY_MONTH_YEAR"] = "DAY_MONTH_YEAR";
  CalendarPickersMode2["MONTH_YEAR"] = "MONTH_YEAR";
  CalendarPickersMode2["YEAR"] = "YEAR";
  return CalendarPickersMode2;
})(CalendarPickersMode || {});
var CalendarLegendItemType = /* @__PURE__ */ ((CalendarLegendItemType2) => {
  CalendarLegendItemType2["None"] = "None";
  CalendarLegendItemType2["Today"] = "Today";
  CalendarLegendItemType2["Selected"] = "Selected";
  CalendarLegendItemType2["Working"] = "Working";
  CalendarLegendItemType2["NonWorking"] = "NonWorking";
  CalendarLegendItemType2["Type01"] = "Type01";
  CalendarLegendItemType2["Type02"] = "Type02";
  CalendarLegendItemType2["Type03"] = "Type03";
  CalendarLegendItemType2["Type04"] = "Type04";
  CalendarLegendItemType2["Type05"] = "Type05";
  CalendarLegendItemType2["Type06"] = "Type06";
  CalendarLegendItemType2["Type07"] = "Type07";
  CalendarLegendItemType2["Type08"] = "Type08";
  CalendarLegendItemType2["Type09"] = "Type09";
  CalendarLegendItemType2["Type10"] = "Type10";
  CalendarLegendItemType2["Type11"] = "Type11";
  CalendarLegendItemType2["Type12"] = "Type12";
  CalendarLegendItemType2["Type13"] = "Type13";
  CalendarLegendItemType2["Type14"] = "Type14";
  CalendarLegendItemType2["Type15"] = "Type15";
  CalendarLegendItemType2["Type16"] = "Type16";
  CalendarLegendItemType2["Type17"] = "Type17";
  CalendarLegendItemType2["Type18"] = "Type18";
  CalendarLegendItemType2["Type19"] = "Type19";
  CalendarLegendItemType2["Type20"] = "Type20";
  return CalendarLegendItemType2;
})(CalendarLegendItemType || {});
var CalendarPickerView = /* @__PURE__ */ ((CalendarPickerView2) => {
  CalendarPickerView2["Day"] = "day";
  CalendarPickerView2["Month"] = "month";
  CalendarPickerView2["Year"] = "year";
  return CalendarPickerView2;
})(CalendarPickerView || {});
function getLocale(localeCode) {
  return new Locale2(localeCode || "en-US");
}
function getWeekStartDay(localeCode) {
  const localeData = getCachedLocaleDataInstance(getLocale(localeCode));
  return localeData.getFirstDayOfWeek();
}
function getLocaleDateFormat(localeCode) {
  const localeData = getCachedLocaleDataInstance(getLocale(localeCode));
  return localeData.getDatePattern("medium");
}

// src/lib/date-utils.ts
function format(date, token) {
  switch (token) {
    case "d":
      return String(date.getDate());
    case "dd":
      return String(date.getDate()).padStart(2, "0");
    case "M":
      return String(date.getMonth() + 1);
    case "MM":
      return String(date.getMonth() + 1).padStart(2, "0");
    case "yy":
      return String(date.getFullYear()).slice(-2);
    case "yyyy":
      return String(date.getFullYear());
    default:
      return String(date.getDate());
  }
}
function parseDate(value) {
  if (value instanceof Date) return value;
  if (typeof value === "number") return new Date(value * 1e3);
  const isoMatch = value.match(/^(\d{4})-(\d{2})-(\d{2})$/);
  if (isoMatch) {
    return new Date(+isoMatch[1], +isoMatch[2] - 1, +isoMatch[3]);
  }
  const d = new Date(value);
  if (!isNaN(d.getTime())) return d;
  throw new Error(`Invalid date value: ${value}`);
}
function formatDateToISO(date) {
  const y = date.getFullYear();
  const m = String(date.getMonth() + 1).padStart(2, "0");
  const d = String(date.getDate()).padStart(2, "0");
  return `${y}-${m}-${d}`;
}
function dateToTimestamp(date) {
  return Math.floor(date.getTime() / 1e3);
}
function timestampToDate(timestamp) {
  return new Date(timestamp * 1e3);
}
function getWeekStart(date, weekStartsOn = 1) {
  const d = new Date(date);
  const day = d.getDay();
  const diff = (day - weekStartsOn + 7) % 7;
  d.setDate(d.getDate() - diff);
  d.setHours(0, 0, 0, 0);
  return d;
}
function getWeekEnd(date, weekStartsOn = 1) {
  const start = getWeekStart(date, weekStartsOn);
  const d = new Date(start);
  d.setDate(d.getDate() + 6);
  d.setHours(23, 59, 59, 999);
  return d;
}
function getMonthDays(date, weekStartsOn = 1) {
  const monthStart = new Date(date.getFullYear(), date.getMonth(), 1);
  const monthEnd = new Date(date.getFullYear(), date.getMonth() + 1, 0);
  const gridStart = getWeekStart(monthStart, weekStartsOn);
  const gridEnd = getWeekEnd(monthEnd, weekStartsOn);
  const days = [];
  const cur = new Date(gridStart);
  cur.setHours(0, 0, 0, 0);
  while (cur <= gridEnd) {
    days.push(new Date(cur));
    cur.setDate(cur.getDate() + 1);
  }
  return days;
}
function isoWeekNumber(date) {
  const d = new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()));
  d.setUTCDate(d.getUTCDate() + 4 - (d.getUTCDay() || 7));
  const yearStart = new Date(Date.UTC(d.getUTCFullYear(), 0, 1));
  return Math.ceil(((d.getTime() - yearStart.getTime()) / 864e5 + 1) / 7);
}
function westernWeekNumber(date) {
  const yearStart = new Date(date.getFullYear(), 0, 1);
  const startDow = yearStart.getDay();
  const dayOfYear = Math.floor((date.getTime() - yearStart.getTime()) / 864e5);
  return Math.floor((dayOfYear + startDow) / 7) + 1;
}
function middleEasternWeekNumber(date) {
  const yearStart = new Date(date.getFullYear(), 0, 1);
  const startDow = (yearStart.getDay() + 1) % 7;
  const dayOfYear = Math.floor((date.getTime() - yearStart.getTime()) / 864e5);
  return Math.floor((dayOfYear + startDow) / 7) + 1;
}
function getWeekNumber(date, numbering = "ISO_8601" /* ISO_8601 */) {
  switch (numbering) {
    case "WesternTraditional" /* WesternTraditional */:
      return westernWeekNumber(date);
    case "MiddleEastern" /* MiddleEastern */:
      return middleEasternWeekNumber(date);
    case "ISO_8601" /* ISO_8601 */:
    case "Default" /* Default */:
    default:
      return isoWeekNumber(date);
  }
}
function getWeekYearNumber(date) {
  const d = new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()));
  d.setUTCDate(d.getUTCDate() + 4 - (d.getUTCDay() || 7));
  return d.getUTCFullYear();
}
function isSameDayUtil(date1, date2) {
  return date1.getFullYear() === date2.getFullYear() && date1.getMonth() === date2.getMonth() && date1.getDate() === date2.getDate();
}
function isSameMonthUtil(date, compareDate) {
  return date.getFullYear() === compareDate.getFullYear() && date.getMonth() === compareDate.getMonth();
}
function isDateInRange(date, start, end) {
  if (!start || !end) return false;
  const t = date.getTime();
  return t >= start.getTime() && t <= end.getTime();
}
function isDateBefore(date, compareDate) {
  return date.getTime() < compareDate.getTime();
}
function isDateAfter(date, compareDate) {
  return date.getTime() > compareDate.getTime();
}
function addMonthsToDate(date, amount) {
  const d = new Date(date);
  d.setMonth(d.getMonth() + amount);
  return d;
}
function addYearsToDate(date, amount) {
  const d = new Date(date);
  d.setFullYear(d.getFullYear() + amount);
  return d;
}
function getToday() {
  const today = /* @__PURE__ */ new Date();
  today.setHours(0, 0, 0, 0);
  return today;
}
function isToday(date) {
  return isSameDayUtil(date, getToday());
}
function getMonthName(date, formatStr = "long", localeCode) {
  const localeData = getCachedLocaleDataInstance(getLocale(localeCode));
  const width = formatStr === "long" ? "wide" : "abbreviated";
  return localeData.getMonthsStandAlone(width)[date.getMonth()];
}
function getYear(date) {
  return date.getFullYear();
}
function getMonth(date) {
  return date.getMonth();
}
function createDate(year, month, day = 1) {
  return new Date(year, month, day);
}
function CalendarHeader({
  monthText,
  yearText,
  cldrReady = true,
  isPreviousDisabled,
  isNextDisabled,
  onPrevious,
  onNext,
  onMonthClick,
  onYearClick,
  previousTooltip = "Previous",
  nextTooltip = "Next",
  className,
  ref
}) {
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      className: cn(
        "flex items-center gap-2 px-2 py-[3px]",
        className
      ),
      role: "toolbar",
      "aria-label": "Calendar navigation",
      children: [
        /* @__PURE__ */ jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            size: "Medium" /* Medium */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsx(SlimArrowLeftIcon, { className: "h-4 w-4 rtl:rotate-180" }),
            accessibleName: previousTooltip,
            tooltip: previousTooltip,
            disabled: isPreviousDisabled,
            onClick: onPrevious
          }
        ),
        /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 flex-1 justify-center", role: "group", "aria-label": "Current date", children: [
          monthText && /* @__PURE__ */ jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              size: "Medium" /* Medium */,
              accessibleName: `${monthText}, Select month`,
              onClick: onMonthClick,
              className: "text-sapphire-text-accent",
              children: cldrReady ? monthText : /* @__PURE__ */ jsx("span", { className: "inline-block h-4 w-16 rounded bg-muted animate-pulse" })
            }
          ),
          /* @__PURE__ */ jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              size: "Medium" /* Medium */,
              accessibleName: `${yearText}, Select year`,
              onClick: onYearClick,
              className: "text-sapphire-text-accent",
              children: cldrReady ? yearText : /* @__PURE__ */ jsx("span", { className: "inline-block h-4 w-10 rounded bg-muted animate-pulse" })
            }
          )
        ] }),
        /* @__PURE__ */ jsx(
          Button,
          {
            design: "Tertiary" /* Tertiary */,
            size: "Medium" /* Medium */,
            iconOnly: true,
            icon: /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "h-4 w-4 rtl:rotate-180" }),
            accessibleName: nextTooltip,
            tooltip: nextTooltip,
            disabled: isNextDisabled,
            onClick: onNext
          }
        )
      ]
    }
  );
}

// src/i18n/locales/en.json
var en_default = {
  INPUT_CLEAR: "Clear input",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} suggestions available",
  INPUT_SUGGESTIONS_ONE: "1 suggestion available",
  INPUT_SUGGESTIONS_NONE: "No suggestions",
  INPUT_SHOW_PASSWORD: "Show password",
  INPUT_HIDE_PASSWORD: "Hide password",
  VALUE_STATE_ERROR: "Error",
  VALUE_STATE_WARNING: "Warning",
  VALUE_STATE_SUCCESS: "Success",
  VALUE_STATE_INFORMATION: "Information",
  RADIOBUTTON_VALUE_STATE_ERROR: "Invalid entry",
  RADIOBUTTON_VALUE_STATE_WARNING: "Warning issued",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "Entry successfully validated",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "Informative entry",
  REQUIRED_FIELD: "Required field",
  LOADING: "Loading...",
  NO_DATA: "No data",
  TABLE_NO_DATA: "No data available",
  TABLE_ROW: "Row",
  TABLE_ROW_INDEX: "Row {{index}} of {{total}}",
  TABLE_ROW_SELECTED: "Selected",
  TABLE_ROW_ACTIVE: "Active",
  TABLE_ROW_NAVIGABLE: "Navigable",
  TABLE_ROW_NAVIGATED: "Navigated",
  TABLE_ROW_SELECTOR: "Row selector",
  TABLE_ROW_SINGLE_ACTION: "{{count}} action",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} actions",
  TABLE_ROW_POPIN: "Additional information",
  TABLE_ROW_ACTIONS: "Row actions",
  TABLE_COLUMN_HEADER_ROW: "Column header row",
  TABLE_SELECTION: "Selection",
  TABLE_SELECT_ALL_ROWS: "Select all rows",
  TABLE_DESELECT_ALL_ROWS: "Deselect all rows",
  TABLE_MORE: "More",
  TABLE_MORE_DESCRIPTION: "Load more data",
  TABLE_MULTI_SELECTABLE: "Multi-selectable table",
  TABLE_SINGLE_SELECTABLE: "Single-selectable table",
  TABLE_NAVIGATION: "Navigation",
  TABLE_SORT_ASCENDING: "Sorted ascending",
  TABLE_SORT_DESCENDING: "Sorted descending",
  TABLE_SORT_NONE: "Not sorted",
  TABLE_AI_SORT_PLACEHOLDER: "Describe how to sort...",
  TABLE_AI_FILTER_PLACEHOLDER: "Describe what to filter...",
  TABLE_AI_CHAT_PLACEHOLDER: "Ask about your data...",
  TABLE_AI_INSIGHT_ANOMALY: "Anomaly detected",
  TABLE_AI_INSIGHT_TREND: "Trend identified",
  TABLE_AI_INSIGHT_PATTERN: "Pattern found",
  TABLE_AI_INSIGHT_DISMISS: "Dismiss insight",
  TABLE_AI_GENERATED: "Generated by AI",
  TABLE_LOADING_MORE: "Loading more...",
  TABLE_TOOLBAR: "Table actions",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "Table toolbar",
  TABLE_ROWS_SELECTED: "{{count}} rows selected",
  TOKEN_ARIA_LABEL: "Token",
  TOKEN_ARIA_DELETABLE: "Deletable",
  TOKEN_DELETE: "Remove",
  TOKENIZER_ARIA_LABEL: "Tokenizer",
  TOKENIZER_SHOW_MORE: "{{count}} More",
  TOKENIZER_POPOVER_HEADER: "More Tokens",
  TOKENIZER_CLEAR_ALL: "Clear All",
  TOKENIZER_TOKEN_COUNT: "{{count}} tokens",
  TOKENIZER_TOKENS_SELECTED: "{{count}} tokens selected",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} results available",
  COMBOBOX_ONE_RESULT: "1 result available",
  COMBOBOX_NO_RESULTS: "No results",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "Listbox",
  SELECT_OPTION_SELECTED: "{{text}} selected",
  MULTICOMBOBOX_SELECT_ALL: "Select All",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} items selected",
  MENU_POPOVER_ACCESSIBLE_NAME: "Select an option from the menu",
  DATEPICKER_DATE_DESCRIPTION: "Date Input",
  DATERANGE_DESCRIPTION: "Date Range Input",
  DATEPICKER_OPEN_ICON_TITLE: "Open Picker",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "Close Picker",
  INPUT_CLEAR_ICON_ACC_NAME: "Clear",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "Enter date for {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "Choose Date Range for {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "Segmented button group",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "Segmented button",
  MESSAGE_STRIP_INFORMATION: "Message Strip",
  MESSAGE_STRIP_SUCCESS: "Success Message Strip",
  MESSAGE_STRIP_ERROR: "Error Message Strip",
  MESSAGE_STRIP_WARNING: "Warning Message Strip",
  MESSAGE_STRIP_CLOSABLE: "Closable",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "Information Message Strip Close",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "Positive Message Strip Close",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "Negative Message Strip Close",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "Critical Message Strip Close",
  FILEUPLOADER_PLACEHOLDER: "Browse or drop a file",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "Browse or drop files",
  FILEUPLOADER_BROWSE_TOOLTIP: "Browse files",
  FILEUPLOADER_CLEAR_TOOLTIP: "Clear all",
  FILEUPLOADER_ROLE_DESCRIPTION: "File upload",
  SEARCHFIELD_SEARCH: "Search",
  SEARCHFIELD_CLEAR: "Clear",
  SEARCHFIELD_ARIA_DESCRIPTION: "Type text to start searching",
  SEARCHFIELD_ROLE_DESCRIPTION: "Search Field",
  NAVIGATION_EXPAND: "Expand",
  NAVIGATION_COLLAPSE: "Collapse",
  SIGN_OUT: "Sign Out",
  FX_START_PANE: "Start pane",
  FX_END_PANE: "End pane",
  FX_RESIZE_START_CENTER: "Resize start and center panes",
  FX_RESIZE_CENTER_END: "Resize center and end panes",
  FX_NOTIFICATIONS: "Notifications",
  FX_OPEN_NAVIGATION: "Open navigation",
  FX_BACK_TO_LIST: "Back to list",
  FX_BACK: "Back",
  FX_RELATED: "Related",
  FX_MORE_ACTIONS: "More actions",
  FX_CLOSE: "Close",
  FX_ADD: "Add",
  FX_SEND: "Send",
  FX_DICTATE: "Dictate",
  FX_VOICE: "Voice",
  FX_DOUBLE_CLICK_TO_RENAME: "Double-click to rename",
  FX_CANCEL: "Cancel",
  FX_ACCEPT: "Accept",
  FX_MAIN_NAVIGATION: "Main navigation items",
  FX_USER_ACTIONS: "User actions",
  FX_CURRENT_ACCOUNT: "Current account information",
  FX_EDIT_ACCOUNTS: "Edit Accounts",
  LINK_SUBTLE: "Subtle",
  LINK_EMPHASIZED: "Emphasized",
  SPLIT_BUTTON_DESCRIPTION: "Split Button",
  SPLIT_BUTTON_KEYBOARD_HINT: "Press Space or Enter to trigger default action and Alt + Arrow Down or F4 to trigger arrow action",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "Open Menu"
};

// src/i18n/locales/de.json
var de_default = {
  INPUT_CLEAR: "Eingabe l\xF6schen",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} Vorschl\xE4ge verf\xFCgbar",
  INPUT_SUGGESTIONS_ONE: "1 Vorschlag verf\xFCgbar",
  INPUT_SUGGESTIONS_NONE: "Keine Vorschl\xE4ge",
  INPUT_SHOW_PASSWORD: "Passwort anzeigen",
  INPUT_HIDE_PASSWORD: "Passwort ausblenden",
  VALUE_STATE_ERROR: "Fehler",
  VALUE_STATE_WARNING: "Warnung",
  VALUE_STATE_SUCCESS: "Erfolg",
  VALUE_STATE_INFORMATION: "Information",
  RADIOBUTTON_VALUE_STATE_ERROR: "Ung\xFCltige Eingabe",
  RADIOBUTTON_VALUE_STATE_WARNING: "Warnung ausgegeben",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "Eintrag erfolgreich validiert",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "Informationseintrag",
  REQUIRED_FIELD: "Pflichtfeld",
  LOADING: "Wird geladen...",
  NO_DATA: "Keine Daten",
  TABLE_NO_DATA: "Keine Daten verf\xFCgbar",
  TABLE_ROW: "Zeile",
  TABLE_ROW_INDEX: "Zeile {{index}} von {{total}}",
  TABLE_ROW_SELECTED: "Ausgew\xE4hlt",
  TABLE_ROW_ACTIVE: "Aktiv",
  TABLE_ROW_NAVIGABLE: "Navigierbar",
  TABLE_ROW_NAVIGATED: "Navigiert",
  TABLE_ROW_SELECTOR: "Zeilenauswahl",
  TABLE_ROW_SINGLE_ACTION: "{{count}} Aktion",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} Aktionen",
  TABLE_ROW_POPIN: "Zus\xE4tzliche Informationen",
  TABLE_ROW_ACTIONS: "Zeilenaktionen",
  TABLE_COLUMN_HEADER_ROW: "Spalten\xFCberschriftenzeile",
  TABLE_SELECTION: "Auswahl",
  TABLE_SELECT_ALL_ROWS: "Alle Zeilen ausw\xE4hlen",
  TABLE_DESELECT_ALL_ROWS: "Alle Zeilen abw\xE4hlen",
  TABLE_MORE: "Mehr",
  TABLE_MORE_DESCRIPTION: "Weitere Daten laden",
  TABLE_MULTI_SELECTABLE: "Mehrfachauswahltabelle",
  TABLE_SINGLE_SELECTABLE: "Einzelauswahltabelle",
  TABLE_NAVIGATION: "Navigation",
  TABLE_SORT_ASCENDING: "Aufsteigend sortiert",
  TABLE_SORT_DESCENDING: "Absteigend sortiert",
  TABLE_SORT_NONE: "Nicht sortiert",
  TABLE_AI_SORT_PLACEHOLDER: "Beschreiben Sie die Sortierung...",
  TABLE_AI_FILTER_PLACEHOLDER: "Beschreiben Sie den Filter...",
  TABLE_AI_CHAT_PLACEHOLDER: "Fragen Sie zu Ihren Daten...",
  TABLE_AI_INSIGHT_ANOMALY: "Anomalie erkannt",
  TABLE_AI_INSIGHT_TREND: "Trend identifiziert",
  TABLE_AI_INSIGHT_PATTERN: "Muster gefunden",
  TABLE_AI_INSIGHT_DISMISS: "Erkenntnis ausblenden",
  TABLE_AI_GENERATED: "Von KI generiert",
  TABLE_LOADING_MORE: "Weitere werden geladen...",
  TABLE_TOOLBAR: "Tabellenaktionen",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "Tabellen-Symbolleiste",
  TABLE_ROWS_SELECTED: "{{count}} Zeilen ausgew\xE4hlt",
  TOKEN_ARIA_LABEL: "Token",
  TOKEN_ARIA_DELETABLE: "L\xF6schbar",
  TOKEN_DELETE: "Entfernen",
  TOKENIZER_ARIA_LABEL: "Tokenizer",
  TOKENIZER_SHOW_MORE: "{{count}} Mehr",
  TOKENIZER_POPOVER_HEADER: "Weitere Token",
  TOKENIZER_CLEAR_ALL: "Alle l\xF6schen",
  TOKENIZER_TOKEN_COUNT: "{{count}} Token",
  TOKENIZER_TOKENS_SELECTED: "{{count}} Token ausgew\xE4hlt",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} Ergebnisse verf\xFCgbar",
  COMBOBOX_ONE_RESULT: "1 Ergebnis verf\xFCgbar",
  COMBOBOX_NO_RESULTS: "Keine Ergebnisse",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "Listenfeld",
  SELECT_OPTION_SELECTED: "{{text}} ausgew\xE4hlt",
  MULTICOMBOBOX_SELECT_ALL: "Alle ausw\xE4hlen",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} Elemente ausgew\xE4hlt",
  MENU_POPOVER_ACCESSIBLE_NAME: "W\xE4hlen Sie eine Option aus dem Men\xFC aus",
  DATEPICKER_DATE_DESCRIPTION: "Datumseingabe",
  DATERANGE_DESCRIPTION: "Datumsbereichseingabe",
  DATEPICKER_OPEN_ICON_TITLE: "Auswahl \xF6ffnen",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "Auswahl schlie\xDFen",
  INPUT_CLEAR_ICON_ACC_NAME: "Zur\xFCcksetzen",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "Datum f\xFCr {{0}} ausw\xE4hlen",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "Datumsbereich f\xFCr {{0}} ausw\xE4hlen",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "Drucktastengruppe mit Unterteilung",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "Drucktaste mit Unterteilung",
  MESSAGE_STRIP_INFORMATION: "Meldungsleiste",
  MESSAGE_STRIP_SUCCESS: "Erfolgsmeldungsleiste",
  MESSAGE_STRIP_ERROR: "Fehlermeldungsleiste",
  MESSAGE_STRIP_WARNING: "Warnmeldungsleiste",
  MESSAGE_STRIP_CLOSABLE: "Schlie\xDFbar",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "Informationsmeldungsleiste schlie\xDFen",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "Positive Meldungsleiste schlie\xDFen",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "Negative Meldungsleiste schlie\xDFen",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "Kritische Meldungsleiste schlie\xDFen",
  FILEUPLOADER_PLACEHOLDER: "Datei suchen oder ablegen",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "Dateien suchen oder ablegen",
  FILEUPLOADER_BROWSE_TOOLTIP: "Dateien durchsuchen",
  FILEUPLOADER_CLEAR_TOOLTIP: "Alle entfernen",
  FILEUPLOADER_ROLE_DESCRIPTION: "Dateiupload",
  SEARCHFIELD_SEARCH: "Suchen",
  SEARCHFIELD_CLEAR: "L\xF6schen",
  SEARCHFIELD_ARIA_DESCRIPTION: "Text eingeben, um die Suche zu starten",
  SEARCHFIELD_ROLE_DESCRIPTION: "Suchfeld",
  NAVIGATION_EXPAND: "Erweitern",
  NAVIGATION_COLLAPSE: "Reduzieren",
  SIGN_OUT: "Abmelden",
  FX_START_PANE: "Startbereich",
  FX_END_PANE: "Endbereich",
  FX_RESIZE_START_CENTER: "Start- und Mittelbereich anpassen",
  FX_RESIZE_CENTER_END: "Mittel- und Endbereich anpassen",
  FX_NOTIFICATIONS: "Benachrichtigungen",
  FX_OPEN_NAVIGATION: "Navigation \xF6ffnen",
  FX_BACK_TO_LIST: "Zur\xFCck zur Liste",
  FX_BACK: "Zur\xFCck",
  FX_RELATED: "Verkn\xFCpft",
  FX_MORE_ACTIONS: "Weitere Aktionen",
  FX_CLOSE: "Schlie\xDFen",
  FX_ADD: "Hinzuf\xFCgen",
  FX_SEND: "Senden",
  FX_DICTATE: "Diktieren",
  FX_VOICE: "Sprache",
  FX_DOUBLE_CLICK_TO_RENAME: "Doppelklicken zum Umbenennen",
  FX_CANCEL: "Abbrechen",
  FX_ACCEPT: "\xDCbernehmen",
  FX_MAIN_NAVIGATION: "Hauptnavigation",
  FX_USER_ACTIONS: "Benutzeraktionen",
  FX_CURRENT_ACCOUNT: "Aktuelle Kontoinformationen",
  FX_EDIT_ACCOUNTS: "Konten bearbeiten",
  LINK_SUBTLE: "Dezent",
  LINK_EMPHASIZED: "Hervorgehoben",
  SPLIT_BUTTON_DESCRIPTION: "Zweigeteilte Drucktaste",
  SPLIT_BUTTON_KEYBOARD_HINT: "F\xFCr Standardaktion Leertaste oder Eingabetaste dr\xFCcken; f\xFCr Pfeilaktion Alt + Pfeil nach unten oder F4 dr\xFCcken",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "Men\xFC \xF6ffnen"
};

// src/i18n/locales/ar.json
var ar_default = {
  INPUT_CLEAR: "\u0645\u0633\u062D \u0627\u0644\u0625\u062F\u062E\u0627\u0644",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} \u0627\u0642\u062A\u0631\u0627\u062D\u0627\u062A \u0645\u062A\u0627\u062D\u0629",
  INPUT_SUGGESTIONS_ONE: "\u0627\u0642\u062A\u0631\u0627\u062D \u0648\u0627\u062D\u062F \u0645\u062A\u0627\u062D",
  INPUT_SUGGESTIONS_NONE: "\u0644\u0627 \u062A\u0648\u062C\u062F \u0627\u0642\u062A\u0631\u0627\u062D\u0627\u062A",
  INPUT_SHOW_PASSWORD: "\u0625\u0638\u0647\u0627\u0631 \u0643\u0644\u0645\u0629 \u0627\u0644\u0645\u0631\u0648\u0631",
  INPUT_HIDE_PASSWORD: "\u0625\u062E\u0641\u0627\u0621 \u0643\u0644\u0645\u0629 \u0627\u0644\u0645\u0631\u0648\u0631",
  VALUE_STATE_ERROR: "\u062E\u0637\u0623",
  VALUE_STATE_WARNING: "\u062A\u062D\u0630\u064A\u0631",
  VALUE_STATE_SUCCESS: "\u0646\u062C\u0627\u062D",
  VALUE_STATE_INFORMATION: "\u0645\u0639\u0644\u0648\u0645\u0627\u062A",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u0625\u062F\u062E\u0627\u0644 \u063A\u064A\u0631 \u0635\u0627\u0644\u062D",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u062A\u0645 \u0625\u0635\u062F\u0627\u0631 \u062A\u062D\u0630\u064A\u0631",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u062A\u0645 \u0627\u0644\u062A\u062D\u0642\u0642 \u0645\u0646 \u0635\u062D\u0629 \u0627\u0644\u0625\u062F\u062E\u0627\u0644 \u0628\u0646\u062C\u0627\u062D",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u0625\u062F\u062E\u0627\u0644 \u0645\u0639\u0644\u0648\u0645\u0627\u062A\u064A",
  REQUIRED_FIELD: "\u062D\u0642\u0644 \u0645\u0637\u0644\u0648\u0628",
  LOADING: "\u062C\u0627\u0631\u064A \u0627\u0644\u062A\u062D\u0645\u064A\u0644...",
  NO_DATA: "\u0644\u0627 \u062A\u0648\u062C\u062F \u0628\u064A\u0627\u0646\u0627\u062A",
  TABLE_NO_DATA: "\u0644\u0627 \u062A\u0648\u062C\u062F \u0628\u064A\u0627\u0646\u0627\u062A \u0645\u062A\u0627\u062D\u0629",
  TABLE_ROW: "\u0635\u0641",
  TABLE_ROW_INDEX: "\u0635\u0641 {{index}} \u0645\u0646 {{total}}",
  TABLE_ROW_SELECTED: "\u0645\u062D\u062F\u062F",
  TABLE_ROW_ACTIVE: "\u0646\u0634\u0637",
  TABLE_ROW_NAVIGABLE: "\u0642\u0627\u0628\u0644 \u0644\u0644\u062A\u0646\u0642\u0644",
  TABLE_ROW_NAVIGATED: "\u062A\u0645 \u0627\u0644\u062A\u0646\u0642\u0644",
  TABLE_ROW_SELECTOR: "\u0645\u062D\u062F\u062F \u0627\u0644\u0635\u0641",
  TABLE_ROW_SINGLE_ACTION: "{{count}} \u0625\u062C\u0631\u0627\u0621",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} \u0625\u062C\u0631\u0627\u0621\u0627\u062A",
  TABLE_ROW_POPIN: "\u0645\u0639\u0644\u0648\u0645\u0627\u062A \u0625\u0636\u0627\u0641\u064A\u0629",
  TABLE_ROW_ACTIONS: "\u0625\u062C\u0631\u0627\u0621\u0627\u062A \u0627\u0644\u0635\u0641",
  TABLE_COLUMN_HEADER_ROW: "\u0635\u0641 \u0631\u0623\u0633 \u0627\u0644\u0639\u0645\u0648\u062F",
  TABLE_SELECTION: "\u0627\u0644\u062A\u062D\u062F\u064A\u062F",
  TABLE_SELECT_ALL_ROWS: "\u062A\u062D\u062F\u064A\u062F \u0643\u0644 \u0627\u0644\u0635\u0641\u0648\u0641",
  TABLE_DESELECT_ALL_ROWS: "\u0625\u0644\u063A\u0627\u0621 \u062A\u062D\u062F\u064A\u062F \u0643\u0644 \u0627\u0644\u0635\u0641\u0648\u0641",
  TABLE_MORE: "\u0627\u0644\u0645\u0632\u064A\u062F",
  TABLE_MORE_DESCRIPTION: "\u062A\u062D\u0645\u064A\u0644 \u0627\u0644\u0645\u0632\u064A\u062F \u0645\u0646 \u0627\u0644\u0628\u064A\u0627\u0646\u0627\u062A",
  TABLE_MULTI_SELECTABLE: "\u062C\u062F\u0648\u0644 \u0645\u062A\u0639\u062F\u062F \u0627\u0644\u062A\u062D\u062F\u064A\u062F",
  TABLE_SINGLE_SELECTABLE: "\u062C\u062F\u0648\u0644 \u0623\u062D\u0627\u062F\u064A \u0627\u0644\u062A\u062D\u062F\u064A\u062F",
  TABLE_NAVIGATION: "\u0627\u0644\u062A\u0646\u0642\u0644",
  TABLE_SORT_ASCENDING: "\u0645\u0631\u062A\u0628 \u062A\u0635\u0627\u0639\u062F\u064A\u0627\u064B",
  TABLE_SORT_DESCENDING: "\u0645\u0631\u062A\u0628 \u062A\u0646\u0627\u0632\u0644\u064A\u0627\u064B",
  TABLE_SORT_NONE: "\u063A\u064A\u0631 \u0645\u0631\u062A\u0628",
  TABLE_AI_SORT_PLACEHOLDER: "\u0648\u0635\u0641 \u0637\u0631\u064A\u0642\u0629 \u0627\u0644\u062A\u0631\u062A\u064A\u0628...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u0648\u0635\u0641 \u0645\u0627 \u062A\u0631\u064A\u062F \u062A\u0635\u0641\u064A\u062A\u0647...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u0627\u0633\u0623\u0644 \u0639\u0646 \u0628\u064A\u0627\u0646\u0627\u062A\u0643...",
  TABLE_AI_INSIGHT_ANOMALY: "\u062A\u0645 \u0627\u0643\u062A\u0634\u0627\u0641 \u0634\u0630\u0648\u0630",
  TABLE_AI_INSIGHT_TREND: "\u062A\u0645 \u062A\u062D\u062F\u064A\u062F \u0627\u062A\u062C\u0627\u0647",
  TABLE_AI_INSIGHT_PATTERN: "\u062A\u0645 \u0627\u0644\u0639\u062B\u0648\u0631 \u0639\u0644\u0649 \u0646\u0645\u0637",
  TABLE_AI_INSIGHT_DISMISS: "\u062A\u062C\u0627\u0647\u0644 \u0627\u0644\u0631\u0624\u064A\u0629",
  TABLE_AI_GENERATED: "\u062A\u0645 \u0627\u0644\u0625\u0646\u0634\u0627\u0621 \u0628\u0648\u0627\u0633\u0637\u0629 \u0627\u0644\u0630\u0643\u0627\u0621 \u0627\u0644\u0627\u0635\u0637\u0646\u0627\u0639\u064A",
  TABLE_LOADING_MORE: "\u062C\u0627\u0631\u064A \u062A\u062D\u0645\u064A\u0644 \u0627\u0644\u0645\u0632\u064A\u062F...",
  TABLE_TOOLBAR: "\u0625\u062C\u0631\u0627\u0621\u0627\u062A \u0627\u0644\u062C\u062F\u0648\u0644",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u0634\u0631\u064A\u0637 \u0623\u062F\u0648\u0627\u062A \u0627\u0644\u062C\u062F\u0648\u0644",
  TABLE_ROWS_SELECTED: "{{count}} \u0635\u0641\u0648\u0641 \u0645\u062D\u062F\u062F\u0629",
  TOKEN_ARIA_LABEL: "\u0631\u0645\u0632",
  TOKEN_ARIA_DELETABLE: "\u0642\u0627\u0628\u0644 \u0644\u0644\u062D\u0630\u0641",
  TOKEN_DELETE: "\u0625\u0632\u0627\u0644\u0629",
  TOKENIZER_ARIA_LABEL: "\u0645\u062D\u0631\u0631 \u0627\u0644\u0631\u0645\u0648\u0632",
  TOKENIZER_SHOW_MORE: "{{count}} \u0627\u0644\u0645\u0632\u064A\u062F",
  TOKENIZER_POPOVER_HEADER: "\u0627\u0644\u0645\u0632\u064A\u062F \u0645\u0646 \u0627\u0644\u0631\u0645\u0648\u0632",
  TOKENIZER_CLEAR_ALL: "\u0645\u0633\u062D \u0627\u0644\u0643\u0644",
  TOKENIZER_TOKEN_COUNT: "{{count}} \u0631\u0645\u0648\u0632",
  TOKENIZER_TOKENS_SELECTED: "{{count}} \u0631\u0645\u0648\u0632 \u0645\u062D\u062F\u062F\u0629",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} \u0646\u062A\u0627\u0626\u062C \u0645\u062A\u0627\u062D\u0629",
  COMBOBOX_ONE_RESULT: "\u0646\u062A\u064A\u062C\u0629 \u0648\u0627\u062D\u062F\u0629 \u0645\u062A\u0627\u062D\u0629",
  COMBOBOX_NO_RESULTS: "\u0644\u0627 \u062A\u0648\u062C\u062F \u0646\u062A\u0627\u0626\u062C",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u0645\u0631\u0628\u0639 \u0627\u0644\u0642\u0627\u0626\u0645\u0629",
  SELECT_OPTION_SELECTED: "\u062A\u0645 \u062A\u062D\u062F\u064A\u062F {{text}}",
  MULTICOMBOBOX_SELECT_ALL: "\u062A\u062D\u062F\u064A\u062F \u0627\u0644\u0643\u0644",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} \u0639\u0646\u0627\u0635\u0631 \u0645\u062D\u062F\u062F\u0629",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u062D\u062F\u062F \u062E\u064A\u0627\u0631\u064B\u0627 \u0645\u0646 \u0627\u0644\u0642\u0627\u0626\u0645\u0629",
  DATEPICKER_DATE_DESCRIPTION: "\u0625\u062F\u062E\u0627\u0644 \u0627\u0644\u062A\u0627\u0631\u064A\u062E",
  DATERANGE_DESCRIPTION: "\u0625\u062F\u062E\u0627\u0644 \u0646\u0637\u0627\u0642 \u0627\u0644\u062A\u0648\u0627\u0631\u064A\u062E",
  DATEPICKER_OPEN_ICON_TITLE: "\u0641\u062A\u062D \u0623\u062F\u0627\u0629 \u0627\u0644\u0627\u0646\u062A\u0642\u0627\u0621",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u0625\u063A\u0644\u0627\u0642 \u0623\u062F\u0627\u0629 \u0627\u0644\u0627\u0646\u062A\u0642\u0627\u0621",
  INPUT_CLEAR_ICON_ACC_NAME: "\u0645\u0633\u062D",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0627\u062E\u062A\u064A\u0627\u0631 \u062A\u0627\u0631\u064A\u062E \u0644\u0640 {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0627\u062E\u062A\u064A\u0627\u0631 \u0646\u0637\u0627\u0642 \u062A\u0648\u0627\u0631\u064A\u062E \u0644\u0640 {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u0645\u062C\u0645\u0648\u0639\u0629 \u0627\u0644\u0623\u0632\u0631\u0627\u0631 \u0627\u0644\u0645\u0642\u0633\u064E\u0651\u0645\u0629",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u0627\u0644\u0632\u0631 \u0627\u0644\u0645\u0642\u0633\u064E\u0651\u0645",
  MESSAGE_STRIP_INFORMATION: "\u0634\u0631\u064A\u0637 \u0627\u0644\u0631\u0633\u0627\u0644\u0629",
  MESSAGE_STRIP_SUCCESS: "\u0634\u0631\u064A\u0637 \u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u0646\u062C\u0627\u062D",
  MESSAGE_STRIP_ERROR: "\u0634\u0631\u064A\u0637 \u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u062E\u0637\u0623",
  MESSAGE_STRIP_WARNING: "\u0634\u0631\u064A\u0637 \u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u062A\u062D\u0630\u064A\u0631",
  MESSAGE_STRIP_CLOSABLE: "\u0645\u0645\u0643\u0646 \u0625\u063A\u0644\u0627\u0642\u0647",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u0625\u063A\u0644\u0627\u0642 \u0634\u0631\u064A\u0637 \u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u0645\u0639\u0644\u0648\u0645\u0627\u062A",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u0625\u063A\u0644\u0627\u0642 \u0634\u0631\u064A\u0637 \u0627\u0644\u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u0625\u064A\u062C\u0627\u0628\u064A\u0629",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u0625\u063A\u0644\u0627\u0642 \u0634\u0631\u064A\u0637 \u0627\u0644\u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u0633\u0644\u0628\u064A\u0629",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u0625\u063A\u0644\u0627\u0642 \u0634\u0631\u064A\u0637 \u0627\u0644\u0631\u0633\u0627\u0644\u0629 \u0627\u0644\u062D\u0631\u062C\u0629",
  FILEUPLOADER_PLACEHOLDER: "\u062A\u0635\u0641\u062D \u0623\u0648 \u0623\u0633\u0642\u0637 \u0645\u0644\u0641\u064B\u0627",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u062A\u0635\u0641\u062D \u0623\u0648 \u0623\u0633\u0642\u0637 \u0645\u0644\u0641\u0627\u062A",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u062A\u0635\u0641\u062D \u0627\u0644\u0645\u0644\u0641\u0627\u062A",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u0645\u0633\u062D \u0627\u0644\u0643\u0644",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u062A\u062D\u0645\u064A\u0644 \u0645\u0644\u0641",
  SEARCHFIELD_SEARCH: "\u0628\u062D\u062B",
  SEARCHFIELD_CLEAR: "\u0645\u0633\u062D",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u0627\u0643\u062A\u0628 \u0646\u0635\u064B\u0627 \u0644\u0628\u062F\u0621 \u0627\u0644\u0628\u062D\u062B",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u062D\u0642\u0644 \u0627\u0644\u0628\u062D\u062B",
  NAVIGATION_EXPAND: "\u062A\u0648\u0633\u064A\u0639",
  NAVIGATION_COLLAPSE: "\u0637\u064A",
  SIGN_OUT: "\u062A\u0633\u062C\u064A\u0644 \u0627\u0644\u062E\u0631\u0648\u062C",
  FX_START_PANE: "\u0627\u0644\u0644\u0648\u062D\u0629 \u0627\u0644\u0623\u0648\u0644\u0649",
  FX_END_PANE: "\u0627\u0644\u0644\u0648\u062D\u0629 \u0627\u0644\u0623\u062E\u064A\u0631\u0629",
  FX_RESIZE_START_CENTER: "\u062A\u063A\u064A\u064A\u0631 \u062D\u062C\u0645 \u0627\u0644\u0644\u0648\u062D\u0629 \u0627\u0644\u0623\u0648\u0644\u0649 \u0648\u0627\u0644\u0648\u0633\u0637\u0649",
  FX_RESIZE_CENTER_END: "\u062A\u063A\u064A\u064A\u0631 \u062D\u062C\u0645 \u0627\u0644\u0644\u0648\u062D\u0629 \u0627\u0644\u0648\u0633\u0637\u0649 \u0648\u0627\u0644\u0623\u062E\u064A\u0631\u0629",
  FX_NOTIFICATIONS: "\u0627\u0644\u0625\u0634\u0639\u0627\u0631\u0627\u062A",
  FX_OPEN_NAVIGATION: "\u0641\u062A\u062D \u0627\u0644\u062A\u0646\u0642\u0644",
  FX_BACK_TO_LIST: "\u0627\u0644\u0639\u0648\u062F\u0629 \u0625\u0644\u0649 \u0627\u0644\u0642\u0627\u0626\u0645\u0629",
  FX_BACK: "\u0631\u062C\u0648\u0639",
  FX_RELATED: "\u0630\u0627\u062A \u0635\u0644\u0629",
  FX_MORE_ACTIONS: "\u0627\u0644\u0645\u0632\u064A\u062F \u0645\u0646 \u0627\u0644\u0625\u062C\u0631\u0627\u0621\u0627\u062A",
  FX_CLOSE: "\u0625\u063A\u0644\u0627\u0642",
  FX_ADD: "\u0625\u0636\u0627\u0641\u0629",
  FX_SEND: "\u0625\u0631\u0633\u0627\u0644",
  FX_DICTATE: "\u0625\u0645\u0644\u0627\u0621",
  FX_VOICE: "\u0635\u0648\u062A",
  FX_DOUBLE_CLICK_TO_RENAME: "\u0627\u0646\u0642\u0631 \u0646\u0642\u0631\u064B\u0627 \u0645\u0632\u062F\u0648\u062C\u064B\u0627 \u0644\u0625\u0639\u0627\u062F\u0629 \u0627\u0644\u062A\u0633\u0645\u064A\u0629",
  FX_CANCEL: "\u0625\u0644\u063A\u0627\u0621",
  FX_ACCEPT: "\u0642\u0628\u0648\u0644",
  FX_MAIN_NAVIGATION: "\u0639\u0646\u0627\u0635\u0631 \u0627\u0644\u062A\u0646\u0642\u0644 \u0627\u0644\u0631\u0626\u064A\u0633\u064A\u0629",
  FX_USER_ACTIONS: "\u0625\u062C\u0631\u0627\u0621\u0627\u062A \u0627\u0644\u0645\u0633\u062A\u062E\u062F\u0645",
  FX_CURRENT_ACCOUNT: "\u0645\u0639\u0644\u0648\u0645\u0627\u062A \u0627\u0644\u062D\u0633\u0627\u0628 \u0627\u0644\u062D\u0627\u0644\u064A",
  FX_EDIT_ACCOUNTS: "\u062A\u0639\u062F\u064A\u0644 \u0627\u0644\u062D\u0633\u0627\u0628\u0627\u062A",
  LINK_SUBTLE: "\u063A\u0627\u0645\u0636",
  LINK_EMPHASIZED: "\u0645\u0624\u0643\u062F",
  SPLIT_BUTTON_DESCRIPTION: "\u0632\u0631 \u0627\u0644\u062A\u0642\u0633\u064A\u0645",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u0627\u0636\u063A\u0637 \u0639\u0644\u0649 Space \u0623\u0648 Enter \u0644\u0628\u062F\u0621 \u062A\u0634\u063A\u064A\u0644 \u0627\u0644\u0625\u062C\u0631\u0627\u0621 \u0627\u0644\u0627\u0641\u062A\u0631\u0627\u0636\u064A \u0648Alt + Arrow Down \u0623\u0648 F4 \u0644\u0628\u062F\u0621 \u062A\u0634\u063A\u064A\u0644 \u0625\u062C\u0631\u0627\u0621 \u0627\u0644\u0633\u0647\u0645",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u0641\u062A\u062D \u0627\u0644\u0642\u0627\u0626\u0645\u0629"
};

// src/i18n/locales/fr.json
var fr_default = {
  INPUT_CLEAR: "Effacer",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} suggestions disponibles",
  INPUT_SUGGESTIONS_ONE: "1 suggestion disponible",
  INPUT_SUGGESTIONS_NONE: "Aucune suggestion",
  INPUT_SHOW_PASSWORD: "Afficher le mot de passe",
  INPUT_HIDE_PASSWORD: "Masquer le mot de passe",
  VALUE_STATE_ERROR: "Erreur",
  VALUE_STATE_WARNING: "Avertissement",
  VALUE_STATE_SUCCESS: "Succ\xE8s",
  VALUE_STATE_INFORMATION: "Information",
  RADIOBUTTON_VALUE_STATE_ERROR: "Entr\xE9e non valide",
  RADIOBUTTON_VALUE_STATE_WARNING: "Avertissement \xE9dit\xE9",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "Entr\xE9e valid\xE9e correctement",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "Entr\xE9e pour information",
  REQUIRED_FIELD: "Champ obligatoire",
  LOADING: "Chargement...",
  NO_DATA: "Aucune donn\xE9e",
  TABLE_NO_DATA: "Aucune donn\xE9e disponible",
  TABLE_ROW: "Ligne",
  TABLE_ROW_INDEX: "Ligne {{index}} sur {{total}}",
  TABLE_ROW_SELECTED: "S\xE9lectionn\xE9e",
  TABLE_ROW_ACTIVE: "Active",
  TABLE_ROW_NAVIGABLE: "Navigable",
  TABLE_ROW_NAVIGATED: "Navigu\xE9e",
  TABLE_ROW_SELECTOR: "S\xE9lecteur de ligne",
  TABLE_ROW_SINGLE_ACTION: "{{count}} action",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} actions",
  TABLE_ROW_POPIN: "Informations suppl\xE9mentaires",
  TABLE_ROW_ACTIONS: "Actions de ligne",
  TABLE_COLUMN_HEADER_ROW: "Ligne d'en-t\xEAte de colonne",
  TABLE_SELECTION: "S\xE9lection",
  TABLE_SELECT_ALL_ROWS: "S\xE9lectionner toutes les lignes",
  TABLE_DESELECT_ALL_ROWS: "D\xE9s\xE9lectionner toutes les lignes",
  TABLE_MORE: "Plus",
  TABLE_MORE_DESCRIPTION: "Charger plus de donn\xE9es",
  TABLE_MULTI_SELECTABLE: "Tableau multi-s\xE9lection",
  TABLE_SINGLE_SELECTABLE: "Tableau mono-s\xE9lection",
  TABLE_NAVIGATION: "Navigation",
  TABLE_SORT_ASCENDING: "Tri croissant",
  TABLE_SORT_DESCENDING: "Tri d\xE9croissant",
  TABLE_SORT_NONE: "Non tri\xE9",
  TABLE_AI_SORT_PLACEHOLDER: "D\xE9crivez comment trier...",
  TABLE_AI_FILTER_PLACEHOLDER: "D\xE9crivez ce que vous voulez filtrer...",
  TABLE_AI_CHAT_PLACEHOLDER: "Posez des questions sur vos donn\xE9es...",
  TABLE_AI_INSIGHT_ANOMALY: "Anomalie d\xE9tect\xE9e",
  TABLE_AI_INSIGHT_TREND: "Tendance identifi\xE9e",
  TABLE_AI_INSIGHT_PATTERN: "Motif trouv\xE9",
  TABLE_AI_INSIGHT_DISMISS: "Ignorer l'aper\xE7u",
  TABLE_AI_GENERATED: "G\xE9n\xE9r\xE9 par l'IA",
  TABLE_LOADING_MORE: "Chargement en cours...",
  TABLE_TOOLBAR: "Actions du tableau",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "Barre d'outils du tableau",
  TABLE_ROWS_SELECTED: "{{count}} lignes s\xE9lectionn\xE9es",
  TOKEN_ARIA_LABEL: "Jeton",
  TOKEN_ARIA_DELETABLE: "Supprimable",
  TOKEN_DELETE: "Supprimer",
  TOKENIZER_ARIA_LABEL: "Tokenizer",
  TOKENIZER_SHOW_MORE: "{{count}} De plus",
  TOKENIZER_POPOVER_HEADER: "Plus de jetons",
  TOKENIZER_CLEAR_ALL: "Tout effacer",
  TOKENIZER_TOKEN_COUNT: "{{count}} jetons",
  TOKENIZER_TOKENS_SELECTED: "{{count}} jetons s\xE9lectionn\xE9s",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} r\xE9sultats disponibles",
  COMBOBOX_ONE_RESULT: "1 r\xE9sultat disponible",
  COMBOBOX_NO_RESULTS: "Aucun r\xE9sultat",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "Zone de liste",
  SELECT_OPTION_SELECTED: "{{text}} s\xE9lectionn\xE9",
  MULTICOMBOBOX_SELECT_ALL: "Tout s\xE9lectionner",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} \xE9l\xE9ments s\xE9lectionn\xE9s",
  MENU_POPOVER_ACCESSIBLE_NAME: "S\xE9lectionner une option dans le menu",
  DATEPICKER_DATE_DESCRIPTION: "Saisie de date",
  DATERANGE_DESCRIPTION: "Saisie de plage de dates",
  DATEPICKER_OPEN_ICON_TITLE: "Ouvrir le s\xE9lecteur",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "Fermer le s\xE9lecteur",
  INPUT_CLEAR_ICON_ACC_NAME: "R\xE9initialiser",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "S\xE9lectionner une date pour {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "S\xE9lectionner une plage de dates pour {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "Groupe de boutons segment\xE9s",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "Bouton segment\xE9",
  MESSAGE_STRIP_INFORMATION: "Barre de message",
  MESSAGE_STRIP_SUCCESS: "Barre de message de r\xE9ussite",
  MESSAGE_STRIP_ERROR: "Barre de message d'erreur",
  MESSAGE_STRIP_WARNING: "Barre de message d'avertissement",
  MESSAGE_STRIP_CLOSABLE: "Refermable",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "Fermer la barre de message d'information",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "Fermer la barre de message positif",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "Fermer la barre de message n\xE9gatif",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "Fermer la barre de message critique",
  FILEUPLOADER_PLACEHOLDER: "Parcourir ou d\xE9poser un fichier",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "Parcourir ou d\xE9poser des fichiers",
  FILEUPLOADER_BROWSE_TOOLTIP: "Parcourir les fichiers",
  FILEUPLOADER_CLEAR_TOOLTIP: "Tout effacer",
  FILEUPLOADER_ROLE_DESCRIPTION: "T\xE9l\xE9chargement de fichier",
  SEARCHFIELD_SEARCH: "Rechercher",
  SEARCHFIELD_CLEAR: "Effacer",
  SEARCHFIELD_ARIA_DESCRIPTION: "Saisissez du texte pour lancer la recherche",
  SEARCHFIELD_ROLE_DESCRIPTION: "Champ de recherche",
  NAVIGATION_EXPAND: "D\xE9velopper",
  NAVIGATION_COLLAPSE: "R\xE9duire",
  SIGN_OUT: "Se d\xE9connecter",
  FX_START_PANE: "Volet de d\xE9part",
  FX_END_PANE: "Volet de fin",
  FX_RESIZE_START_CENTER: "Redimensionner les volets de d\xE9part et central",
  FX_RESIZE_CENTER_END: "Redimensionner les volets central et de fin",
  FX_NOTIFICATIONS: "Notifications",
  FX_OPEN_NAVIGATION: "Ouvrir la navigation",
  FX_BACK_TO_LIST: "Retour \xE0 la liste",
  FX_BACK: "Retour",
  FX_RELATED: "Associ\xE9",
  FX_MORE_ACTIONS: "Plus d'actions",
  FX_CLOSE: "Fermer",
  FX_ADD: "Ajouter",
  FX_SEND: "Envoyer",
  FX_DICTATE: "Dicter",
  FX_VOICE: "Voix",
  FX_DOUBLE_CLICK_TO_RENAME: "Double-cliquez pour renommer",
  FX_CANCEL: "Annuler",
  FX_ACCEPT: "Accepter",
  FX_MAIN_NAVIGATION: "\xC9l\xE9ments de navigation principaux",
  FX_USER_ACTIONS: "Actions utilisateur",
  FX_CURRENT_ACCOUNT: "Informations du compte actuel",
  FX_EDIT_ACCOUNTS: "Modifier les comptes",
  LINK_SUBTLE: "Subtil",
  LINK_EMPHASIZED: "Mis en surbrillance",
  SPLIT_BUTTON_DESCRIPTION: "Bouton de fractionnement",
  SPLIT_BUTTON_KEYBOARD_HINT: "Appuyez sur la barre d'espacement ou sur la touche Entr\xE9e pour d\xE9clencher l'action par d\xE9faut, puis sur Alt + touche fl\xE9ch\xE9e vers le bas ou F4 pour d\xE9clencher l'action.",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "Ouvrir le menu"
};

// src/i18n/locales/he.json
var he_default = {
  INPUT_CLEAR: "\u05E0\u05E7\u05D4 \u05E7\u05DC\u05D8",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} \u05D4\u05E6\u05E2\u05D5\u05EA \u05D6\u05DE\u05D9\u05E0\u05D5\u05EA",
  INPUT_SUGGESTIONS_ONE: "\u05D4\u05E6\u05E2\u05D4 \u05D0\u05D7\u05EA \u05D6\u05DE\u05D9\u05E0\u05D4",
  INPUT_SUGGESTIONS_NONE: "\u05D0\u05D9\u05DF \u05D4\u05E6\u05E2\u05D5\u05EA",
  INPUT_SHOW_PASSWORD: "\u05D4\u05E6\u05D2 \u05E1\u05D9\u05E1\u05DE\u05D4",
  INPUT_HIDE_PASSWORD: "\u05D4\u05E1\u05EA\u05E8 \u05E1\u05D9\u05E1\u05DE\u05D4",
  VALUE_STATE_ERROR: "\u05E9\u05D2\u05D9\u05D0\u05D4",
  VALUE_STATE_WARNING: "\u05D0\u05D6\u05D4\u05E8\u05D4",
  VALUE_STATE_SUCCESS: "\u05D4\u05E6\u05DC\u05D7\u05D4",
  VALUE_STATE_INFORMATION: "\u05DE\u05D9\u05D3\u05E2",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u05D4\u05D6\u05E0\u05D4 \u05DC\u05D0 \u05D7\u05D5\u05E7\u05D9\u05EA",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u05D4\u05D5\u05E0\u05E4\u05E7\u05D4 \u05E9\u05D2\u05D9\u05D0\u05D4",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u05D4\u05D6\u05E0\u05D4 \u05D0\u05D5\u05DE\u05EA\u05D4 \u05D1\u05D4\u05E6\u05DC\u05D7\u05D4",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u05D4\u05D6\u05E0\u05D4 \u05D0\u05D9\u05E0\u05E4\u05D5\u05E8\u05DE\u05D8\u05D9\u05D1\u05D9\u05EA",
  REQUIRED_FIELD: "\u05E9\u05D3\u05D4 \u05D7\u05D5\u05D1\u05D4",
  LOADING: "\u05D8\u05D5\u05E2\u05DF...",
  NO_DATA: "\u05D0\u05D9\u05DF \u05E0\u05EA\u05D5\u05E0\u05D9\u05DD",
  TABLE_NO_DATA: "\u05D0\u05D9\u05DF \u05E0\u05EA\u05D5\u05E0\u05D9\u05DD \u05D6\u05DE\u05D9\u05E0\u05D9\u05DD",
  TABLE_ROW: "\u05E9\u05D5\u05E8\u05D4",
  TABLE_ROW_INDEX: "\u05E9\u05D5\u05E8\u05D4 {{index}} \u05DE\u05EA\u05D5\u05DA {{total}}",
  TABLE_ROW_SELECTED: "\u05E0\u05D1\u05D7\u05E8",
  TABLE_ROW_ACTIVE: "\u05E4\u05E2\u05D9\u05DC",
  TABLE_ROW_NAVIGABLE: "\u05E0\u05D9\u05EA\u05DF \u05DC\u05E0\u05D9\u05D5\u05D5\u05D8",
  TABLE_ROW_NAVIGATED: "\u05E0\u05D5\u05D5\u05D8",
  TABLE_ROW_SELECTOR: "\u05D1\u05D5\u05E8\u05E8 \u05E9\u05D5\u05E8\u05D4",
  TABLE_ROW_SINGLE_ACTION: "{{count}} \u05E4\u05E2\u05D5\u05DC\u05D4",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} \u05E4\u05E2\u05D5\u05DC\u05D5\u05EA",
  TABLE_ROW_POPIN: "\u05DE\u05D9\u05D3\u05E2 \u05E0\u05D5\u05E1\u05E3",
  TABLE_ROW_ACTIONS: "\u05E4\u05E2\u05D5\u05DC\u05D5\u05EA \u05E9\u05D5\u05E8\u05D4",
  TABLE_COLUMN_HEADER_ROW: "\u05E9\u05D5\u05E8\u05EA \u05DB\u05D5\u05EA\u05E8\u05EA \u05E2\u05DE\u05D5\u05D3\u05D4",
  TABLE_SELECTION: "\u05D1\u05D7\u05D9\u05E8\u05D4",
  TABLE_SELECT_ALL_ROWS: "\u05D1\u05D7\u05E8 \u05D0\u05EA \u05DB\u05DC \u05D4\u05E9\u05D5\u05E8\u05D5\u05EA",
  TABLE_DESELECT_ALL_ROWS: "\u05D1\u05D8\u05DC \u05D1\u05D7\u05D9\u05E8\u05EA \u05DB\u05DC \u05D4\u05E9\u05D5\u05E8\u05D5\u05EA",
  TABLE_MORE: "\u05E2\u05D5\u05D3",
  TABLE_MORE_DESCRIPTION: "\u05D8\u05E2\u05DF \u05E0\u05EA\u05D5\u05E0\u05D9\u05DD \u05E0\u05D5\u05E1\u05E4\u05D9\u05DD",
  TABLE_MULTI_SELECTABLE: "\u05D8\u05D1\u05DC\u05D4 \u05DE\u05E8\u05D5\u05D1\u05EA \u05D1\u05D7\u05D9\u05E8\u05D4",
  TABLE_SINGLE_SELECTABLE: "\u05D8\u05D1\u05DC\u05D4 \u05D1\u05D7\u05D9\u05E8\u05D4 \u05D9\u05D7\u05D9\u05D3\u05D4",
  TABLE_NAVIGATION: "\u05E0\u05D9\u05D5\u05D5\u05D8",
  TABLE_SORT_ASCENDING: "\u05DE\u05DE\u05D5\u05D9\u05DF \u05D1\u05E1\u05D3\u05E8 \u05E2\u05D5\u05DC\u05D4",
  TABLE_SORT_DESCENDING: "\u05DE\u05DE\u05D5\u05D9\u05DF \u05D1\u05E1\u05D3\u05E8 \u05D9\u05D5\u05E8\u05D3",
  TABLE_SORT_NONE: "\u05DC\u05D0 \u05DE\u05DE\u05D5\u05D9\u05DF",
  TABLE_AI_SORT_PLACEHOLDER: "\u05EA\u05D0\u05E8 \u05D0\u05D9\u05DA \u05DC\u05DE\u05D9\u05D9\u05DF...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u05EA\u05D0\u05E8 \u05DE\u05D4 \u05DC\u05E1\u05E0\u05DF...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u05E9\u05D0\u05DC \u05E2\u05DC \u05D4\u05E0\u05EA\u05D5\u05E0\u05D9\u05DD \u05E9\u05DC\u05DA...",
  TABLE_AI_INSIGHT_ANOMALY: "\u05D6\u05D5\u05D4\u05EA\u05D4 \u05D7\u05E8\u05D9\u05D2\u05D4",
  TABLE_AI_INSIGHT_TREND: "\u05D6\u05D5\u05D4\u05D4 \u05DE\u05D2\u05DE\u05D4",
  TABLE_AI_INSIGHT_PATTERN: "\u05E0\u05DE\u05E6\u05D0 \u05D3\u05E4\u05D5\u05E1",
  TABLE_AI_INSIGHT_DISMISS: "\u05D4\u05EA\u05E2\u05DC\u05DD \u05DE\u05EA\u05D5\u05D1\u05E0\u05D4",
  TABLE_AI_GENERATED: "\u05E0\u05D5\u05E6\u05E8 \u05E2\u05DC \u05D9\u05D3\u05D9 \u05D1\u05D9\u05E0\u05D4 \u05DE\u05DC\u05D0\u05DB\u05D5\u05EA\u05D9\u05EA",
  TABLE_LOADING_MORE: "\u05D8\u05D5\u05E2\u05DF \u05E2\u05D5\u05D3...",
  TABLE_TOOLBAR: "\u05E4\u05E2\u05D5\u05DC\u05D5\u05EA \u05D8\u05D1\u05DC\u05D4",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u05E1\u05E8\u05D2\u05DC \u05DB\u05DC\u05D9\u05DD \u05E9\u05DC \u05D8\u05D1\u05DC\u05D4",
  TABLE_ROWS_SELECTED: "{{count}} \u05E9\u05D5\u05E8\u05D5\u05EA \u05E0\u05D1\u05D7\u05E8\u05D5",
  TOKEN_ARIA_LABEL: "\u05D0\u05E1\u05D9\u05DE\u05D5\u05DF",
  TOKEN_ARIA_DELETABLE: "\u05E0\u05D9\u05EA\u05DF \u05DC\u05DE\u05D7\u05D9\u05E7\u05D4",
  TOKEN_DELETE: "\u05D4\u05E1\u05E8",
  TOKENIZER_ARIA_LABEL: "\u05DE\u05E0\u05D4\u05DC \u05D0\u05E1\u05D9\u05DE\u05D5\u05E0\u05D9\u05DD",
  TOKENIZER_SHOW_MORE: "{{count}} \u05E2\u05D5\u05D3",
  TOKENIZER_POPOVER_HEADER: "\u05D0\u05E1\u05D9\u05DE\u05D5\u05E0\u05D9\u05DD \u05E0\u05D5\u05E1\u05E4\u05D9\u05DD",
  TOKENIZER_CLEAR_ALL: "\u05E0\u05E7\u05D4 \u05D4\u05DB\u05DC",
  TOKENIZER_TOKEN_COUNT: "{{count}} \u05D0\u05E1\u05D9\u05DE\u05D5\u05E0\u05D9\u05DD",
  TOKENIZER_TOKENS_SELECTED: "{{count}} \u05D0\u05E1\u05D9\u05DE\u05D5\u05E0\u05D9\u05DD \u05E0\u05D1\u05D7\u05E8\u05D5",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} \u05EA\u05D5\u05E6\u05D0\u05D5\u05EA \u05D6\u05DE\u05D9\u05E0\u05D5\u05EA",
  COMBOBOX_ONE_RESULT: "\u05EA\u05D5\u05E6\u05D0\u05D4 \u05D0\u05D7\u05EA \u05D6\u05DE\u05D9\u05E0\u05D4",
  COMBOBOX_NO_RESULTS: "\u05D0\u05D9\u05DF \u05EA\u05D5\u05E6\u05D0\u05D5\u05EA",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u05EA\u05D9\u05D1\u05EA \u05E8\u05E9\u05D9\u05DE\u05D4",
  SELECT_OPTION_SELECTED: "{{text}} \u05E0\u05D1\u05D7\u05E8",
  MULTICOMBOBOX_SELECT_ALL: "\u05D1\u05D7\u05E8 \u05D4\u05DB\u05DC",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} \u05E4\u05E8\u05D9\u05D8\u05D9\u05DD \u05E0\u05D1\u05D7\u05E8\u05D5",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u05D1\u05D7\u05E8 \u05D0\u05E4\u05E9\u05E8\u05D5\u05EA \u05DE\u05D4\u05EA\u05E4\u05E8\u05D9\u05D8",
  DATEPICKER_DATE_DESCRIPTION: "\u05E7\u05DC\u05D8 \u05E9\u05DC \u05EA\u05D0\u05E8\u05D9\u05DA",
  DATERANGE_DESCRIPTION: "\u05E7\u05DC\u05D8 \u05E9\u05DC \u05D8\u05D5\u05D5\u05D7 \u05EA\u05D0\u05E8\u05D9\u05DB\u05D9\u05DD",
  DATEPICKER_OPEN_ICON_TITLE: "\u05E4\u05EA\u05D7 \u05D1\u05D5\u05D7\u05E8",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u05E1\u05D2\u05D5\u05E8 \u05D1\u05D5\u05D7\u05E8",
  INPUT_CLEAR_ICON_ACC_NAME: "\u05E0\u05E7\u05D4",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u05D1\u05D7\u05E8 \u05EA\u05D0\u05E8\u05D9\u05DA \u05E2\u05D1\u05D5\u05E8 {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u05D1\u05D7\u05E8 \u05D8\u05D5\u05D5\u05D7 \u05EA\u05D0\u05E8\u05D9\u05DB\u05D9\u05DD \u05E2\u05D1\u05D5\u05E8 {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u05E7\u05D1\u05D5\u05E6\u05EA \u05DB\u05E4\u05EA\u05D5\u05E8\u05D9\u05DD \u05DE\u05E4\u05D5\u05DC\u05D7\u05EA",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u05DB\u05E4\u05EA\u05D5\u05E8 \u05DE\u05E4\u05D5\u05DC\u05D7",
  MESSAGE_STRIP_INFORMATION: "\u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA",
  MESSAGE_STRIP_SUCCESS: "\u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05D4\u05E6\u05DC\u05D7\u05D4",
  MESSAGE_STRIP_ERROR: "\u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05E9\u05D2\u05D9\u05D0\u05D4",
  MESSAGE_STRIP_WARNING: "\u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05D0\u05D6\u05D4\u05E8\u05D4",
  MESSAGE_STRIP_CLOSABLE: "\u05E0\u05D9\u05EA\u05DF \u05DC\u05E1\u05D2\u05D9\u05E8\u05D4",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u05E1\u05D2\u05D5\u05E8 \u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05DE\u05D9\u05D3\u05E2",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u05E1\u05D2\u05D5\u05E8 \u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05D7\u05D9\u05D5\u05D1\u05D9\u05D5\u05EA",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u05E1\u05D2\u05D5\u05E8 \u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05E9\u05DC\u05D9\u05DC\u05D9\u05D5\u05EA",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u05E1\u05D2\u05D5\u05E8 \u05E8\u05E6\u05D5\u05E2\u05EA \u05D4\u05D5\u05D3\u05E2\u05D5\u05EA \u05E7\u05E8\u05D9\u05D8\u05D9\u05D5\u05EA",
  FILEUPLOADER_PLACEHOLDER: "\u05E2\u05D9\u05D9\u05DF \u05D0\u05D5 \u05E9\u05D7\u05E8\u05E8 \u05E7\u05D5\u05D1\u05E5",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u05E2\u05D9\u05D9\u05DF \u05D0\u05D5 \u05E9\u05D7\u05E8\u05E8 \u05E7\u05D1\u05E6\u05D9\u05DD",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u05E2\u05D9\u05D5\u05DF \u05D1\u05E7\u05D1\u05E6\u05D9\u05DD",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u05E0\u05E7\u05D4 \u05D4\u05DB\u05DC",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u05D4\u05E2\u05DC\u05D0\u05EA \u05E7\u05D5\u05D1\u05E5",
  SEARCHFIELD_SEARCH: "\u05D7\u05D9\u05E4\u05D5\u05E9",
  SEARCHFIELD_CLEAR: "\u05E0\u05E7\u05D4",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u05D4\u05E7\u05DC\u05D3 \u05D8\u05E7\u05E1\u05D8 \u05DB\u05D3\u05D9 \u05DC\u05D4\u05EA\u05D7\u05D9\u05DC \u05DC\u05D7\u05E4\u05E9",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u05E9\u05D3\u05D4 \u05D7\u05D9\u05E4\u05D5\u05E9",
  NAVIGATION_EXPAND: "\u05D4\u05E8\u05D7\u05D1",
  NAVIGATION_COLLAPSE: "\u05DB\u05D5\u05D5\u05E5",
  SIGN_OUT: "\u05D4\u05EA\u05E0\u05EA\u05E7",
  FX_START_PANE: "\u05D7\u05DC\u05D5\u05E0\u05D9\u05EA \u05D4\u05EA\u05D7\u05DC\u05D4",
  FX_END_PANE: "\u05D7\u05DC\u05D5\u05E0\u05D9\u05EA \u05E1\u05D9\u05D5\u05DD",
  FX_RESIZE_START_CENTER: "\u05E9\u05E0\u05D4 \u05D2\u05D5\u05D3\u05DC \u05D7\u05DC\u05D5\u05E0\u05D9\u05D5\u05EA \u05D4\u05EA\u05D7\u05DC\u05D4 \u05D5\u05DE\u05E8\u05DB\u05D6",
  FX_RESIZE_CENTER_END: "\u05E9\u05E0\u05D4 \u05D2\u05D5\u05D3\u05DC \u05D7\u05DC\u05D5\u05E0\u05D9\u05D5\u05EA \u05DE\u05E8\u05DB\u05D6 \u05D5\u05E1\u05D9\u05D5\u05DD",
  FX_NOTIFICATIONS: "\u05D4\u05EA\u05E8\u05D0\u05D5\u05EA",
  FX_OPEN_NAVIGATION: "\u05E4\u05EA\u05D7 \u05E0\u05D9\u05D5\u05D5\u05D8",
  FX_BACK_TO_LIST: "\u05D7\u05D6\u05E8\u05D4 \u05DC\u05E8\u05E9\u05D9\u05DE\u05D4",
  FX_BACK: "\u05D7\u05D6\u05E8\u05D4",
  FX_RELATED: "\u05E7\u05E9\u05D5\u05E8",
  FX_MORE_ACTIONS: "\u05E4\u05E2\u05D5\u05DC\u05D5\u05EA \u05E0\u05D5\u05E1\u05E4\u05D5\u05EA",
  FX_CLOSE: "\u05E1\u05D2\u05D5\u05E8",
  FX_ADD: "\u05D4\u05D5\u05E1\u05E3",
  FX_SEND: "\u05E9\u05DC\u05D7",
  FX_DICTATE: "\u05D4\u05DB\u05EA\u05D1",
  FX_VOICE: "\u05E7\u05D5\u05DC",
  FX_DOUBLE_CLICK_TO_RENAME: "\u05DC\u05D7\u05E5 \u05DC\u05D7\u05D9\u05E6\u05D4 \u05DB\u05E4\u05D5\u05DC\u05D4 \u05DC\u05E9\u05D9\u05E0\u05D5\u05D9 \u05E9\u05DD",
  FX_CANCEL: "\u05D1\u05D8\u05DC",
  FX_ACCEPT: "\u05E7\u05D1\u05DC",
  FX_MAIN_NAVIGATION: "\u05E4\u05E8\u05D9\u05D8\u05D9 \u05E0\u05D9\u05D5\u05D5\u05D8 \u05E8\u05D0\u05E9\u05D9\u05D9\u05DD",
  FX_USER_ACTIONS: "\u05E4\u05E2\u05D5\u05DC\u05D5\u05EA \u05DE\u05E9\u05EA\u05DE\u05E9",
  FX_CURRENT_ACCOUNT: "\u05DE\u05D9\u05D3\u05E2 \u05E2\u05DC \u05D4\u05D7\u05E9\u05D1\u05D5\u05DF \u05D4\u05E0\u05D5\u05DB\u05D7\u05D9",
  FX_EDIT_ACCOUNTS: "\u05E2\u05E8\u05D5\u05DA \u05D7\u05E9\u05D1\u05D5\u05E0\u05D5\u05EA",
  LINK_SUBTLE: "\u05E2\u05D3\u05D9\u05DF",
  LINK_EMPHASIZED: "\u05DE\u05D5\u05D3\u05D2\u05E9",
  SPLIT_BUTTON_DESCRIPTION: "\u05DC\u05D7\u05E6\u05DF \u05E4\u05D9\u05E6\u05D5\u05DC",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u05DC\u05D7\u05E5 \u05E8\u05D5\u05D5\u05D7 \u05D0\u05D5 ENTER \u05DB\u05D3\u05D9 \u05DC\u05D4\u05E4\u05E2\u05D9\u05DC \u05E4\u05E2\u05D5\u05DC\u05D4 \u05D1\u05D1\u05E8\u05D9\u05E8\u05EA \u05DE\u05D7\u05D3\u05DC \u05D5- Alt + \u05D7\u05E5 \u05DC\u05DE\u05D8\u05D4 \u05D0\u05D5 F4 \u05DB\u05D3\u05D9 \u05DC\u05D4\u05E4\u05E2\u05D9\u05DC \u05E4\u05E2\u05D5\u05DC\u05EA \u05D7\u05E5",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u05E4\u05EA\u05D7 \u05EA\u05E4\u05E8\u05D9\u05D8"
};

// src/i18n/locales/ja.json
var ja_default = {
  INPUT_CLEAR: "\u5165\u529B\u3092\u30AF\u30EA\u30A2",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}}\u4EF6\u306E\u5019\u88DC\u304C\u3042\u308A\u307E\u3059",
  INPUT_SUGGESTIONS_ONE: "1\u4EF6\u306E\u5019\u88DC\u304C\u3042\u308A\u307E\u3059",
  INPUT_SUGGESTIONS_NONE: "\u5019\u88DC\u304C\u3042\u308A\u307E\u305B\u3093",
  INPUT_SHOW_PASSWORD: "\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8868\u793A",
  INPUT_HIDE_PASSWORD: "\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u96A0\u3059",
  VALUE_STATE_ERROR: "\u30A8\u30E9\u30FC",
  VALUE_STATE_WARNING: "\u8B66\u544A",
  VALUE_STATE_SUCCESS: "\u6210\u529F",
  VALUE_STATE_INFORMATION: "\u60C5\u5831",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u7121\u52B9\u306A\u30A8\u30F3\u30C8\u30EA",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u8B66\u544A\u304C\u767A\u884C\u3055\u308C\u307E\u3057\u305F",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u30A8\u30F3\u30C8\u30EA\u304C\u6B63\u5E38\u306B\u30C1\u30A7\u30C3\u30AF\u3055\u308C\u307E\u3057\u305F",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u60C5\u5831\u30A8\u30F3\u30C8\u30EA",
  REQUIRED_FIELD: "\u5FC5\u9808\u9805\u76EE",
  LOADING: "\u8AAD\u307F\u8FBC\u307F\u4E2D...",
  NO_DATA: "\u30C7\u30FC\u30BF\u304C\u3042\u308A\u307E\u305B\u3093",
  TABLE_NO_DATA: "\u30C7\u30FC\u30BF\u304C\u3042\u308A\u307E\u305B\u3093",
  TABLE_ROW: "\u884C",
  TABLE_ROW_INDEX: "\u884C {{index}}/{{total}}",
  TABLE_ROW_SELECTED: "\u9078\u629E\u6E08\u307F",
  TABLE_ROW_ACTIVE: "\u30A2\u30AF\u30C6\u30A3\u30D6",
  TABLE_ROW_NAVIGABLE: "\u30CA\u30D3\u30B2\u30FC\u30C8\u53EF\u80FD",
  TABLE_ROW_NAVIGATED: "\u30CA\u30D3\u30B2\u30FC\u30C8\u6E08\u307F",
  TABLE_ROW_SELECTOR: "\u884C\u30BB\u30EC\u30AF\u30BF\u30FC",
  TABLE_ROW_SINGLE_ACTION: "{{count}}\u4EF6\u306E\u30A2\u30AF\u30B7\u30E7\u30F3",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}}\u4EF6\u306E\u30A2\u30AF\u30B7\u30E7\u30F3",
  TABLE_ROW_POPIN: "\u8FFD\u52A0\u60C5\u5831",
  TABLE_ROW_ACTIONS: "\u884C\u30A2\u30AF\u30B7\u30E7\u30F3",
  TABLE_COLUMN_HEADER_ROW: "\u5217\u30D8\u30C3\u30C0\u30FC\u884C",
  TABLE_SELECTION: "\u9078\u629E",
  TABLE_SELECT_ALL_ROWS: "\u3059\u3079\u3066\u306E\u884C\u3092\u9078\u629E",
  TABLE_DESELECT_ALL_ROWS: "\u3059\u3079\u3066\u306E\u884C\u306E\u9078\u629E\u3092\u89E3\u9664",
  TABLE_MORE: "\u3082\u3063\u3068\u898B\u308B",
  TABLE_MORE_DESCRIPTION: "\u3055\u3089\u306B\u30C7\u30FC\u30BF\u3092\u8AAD\u307F\u8FBC\u3080",
  TABLE_MULTI_SELECTABLE: "\u8907\u6570\u9078\u629E\u30C6\u30FC\u30D6\u30EB",
  TABLE_SINGLE_SELECTABLE: "\u5358\u4E00\u9078\u629E\u30C6\u30FC\u30D6\u30EB",
  TABLE_NAVIGATION: "\u30CA\u30D3\u30B2\u30FC\u30B7\u30E7\u30F3",
  TABLE_SORT_ASCENDING: "\u6607\u9806\u3067\u30BD\u30FC\u30C8",
  TABLE_SORT_DESCENDING: "\u964D\u9806\u3067\u30BD\u30FC\u30C8",
  TABLE_SORT_NONE: "\u30BD\u30FC\u30C8\u306A\u3057",
  TABLE_AI_SORT_PLACEHOLDER: "\u30BD\u30FC\u30C8\u65B9\u6CD5\u3092\u8AAC\u660E...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u30D5\u30A3\u30EB\u30BF\u30FC\u3092\u8AAC\u660E...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u30C7\u30FC\u30BF\u306B\u3064\u3044\u3066\u8CEA\u554F...",
  TABLE_AI_INSIGHT_ANOMALY: "\u7570\u5E38\u3092\u691C\u51FA",
  TABLE_AI_INSIGHT_TREND: "\u30C8\u30EC\u30F3\u30C9\u3092\u7279\u5B9A",
  TABLE_AI_INSIGHT_PATTERN: "\u30D1\u30BF\u30FC\u30F3\u3092\u767A\u898B",
  TABLE_AI_INSIGHT_DISMISS: "\u30A4\u30F3\u30B5\u30A4\u30C8\u3092\u9589\u3058\u308B",
  TABLE_AI_GENERATED: "AI\u306B\u3088\u3063\u3066\u751F\u6210",
  TABLE_LOADING_MORE: "\u8AAD\u307F\u8FBC\u307F\u4E2D...",
  TABLE_TOOLBAR: "\u30C6\u30FC\u30D6\u30EB\u30A2\u30AF\u30B7\u30E7\u30F3",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u30C6\u30FC\u30D6\u30EB\u30C4\u30FC\u30EB\u30D0\u30FC",
  TABLE_ROWS_SELECTED: "{{count}}\u884C\u304C\u9078\u629E\u3055\u308C\u307E\u3057\u305F",
  TOKEN_ARIA_LABEL: "\u30C8\u30FC\u30AF\u30F3",
  TOKEN_ARIA_DELETABLE: "\u524A\u9664\u53EF\u80FD",
  TOKEN_DELETE: "\u524A\u9664",
  TOKENIZER_ARIA_LABEL: "\u30C8\u30FC\u30AF\u30CA\u30A4\u30B6\u30FC",
  TOKENIZER_SHOW_MORE: "\u3042\u3068{{count}}\u4EF6",
  TOKENIZER_POPOVER_HEADER: "\u305D\u306E\u4ED6\u306E\u30C8\u30FC\u30AF\u30F3",
  TOKENIZER_CLEAR_ALL: "\u3059\u3079\u3066\u30AF\u30EA\u30A2",
  TOKENIZER_TOKEN_COUNT: "{{count}}\u500B\u306E\u30C8\u30FC\u30AF\u30F3",
  TOKENIZER_TOKENS_SELECTED: "{{count}}\u500B\u306E\u30C8\u30FC\u30AF\u30F3\u3092\u9078\u629E\u4E2D",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}}\u4EF6\u306E\u7D50\u679C\u304C\u3042\u308A\u307E\u3059",
  COMBOBOX_ONE_RESULT: "1\u4EF6\u306E\u7D50\u679C\u304C\u3042\u308A\u307E\u3059",
  COMBOBOX_NO_RESULTS: "\u7D50\u679C\u304C\u3042\u308A\u307E\u305B\u3093",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u30EA\u30B9\u30C8\u30DC\u30C3\u30AF\u30B9",
  SELECT_OPTION_SELECTED: "{{text}}\u3092\u9078\u629E",
  MULTICOMBOBOX_SELECT_ALL: "\u3059\u3079\u3066\u9078\u629E",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}}\u4EF6\u9078\u629E\u6E08\u307F",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u30E1\u30CB\u30E5\u30FC\u304B\u3089\u30AA\u30D7\u30B7\u30E7\u30F3\u3092\u9078\u629E",
  DATEPICKER_DATE_DESCRIPTION: "\u65E5\u4ED8\u5165\u529B",
  DATERANGE_DESCRIPTION: "\u65E5\u4ED8\u7BC4\u56F2\u5165\u529B",
  DATEPICKER_OPEN_ICON_TITLE: "\u30D4\u30C3\u30AB\u30FC\u3092\u958B\u304F",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u30D4\u30C3\u30AB\u30FC\u3092\u9589\u3058\u308B",
  INPUT_CLEAR_ICON_ACC_NAME: "\u30AF\u30EA\u30A2",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "{{0}} \u306E\u65E5\u4ED8\u3092\u9078\u629E",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "{{0}} \u306E\u65E5\u4ED8\u7BC4\u56F2\u3092\u9078\u629E",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u30BB\u30B0\u30E1\u30F3\u30C8\u30DC\u30BF\u30F3\u30B0\u30EB\u30FC\u30D7",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u30BB\u30B0\u30E1\u30F3\u30C8\u30DC\u30BF\u30F3",
  MESSAGE_STRIP_INFORMATION: "\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7",
  MESSAGE_STRIP_SUCCESS: "\u6B63\u5E38\u7D42\u4E86\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7",
  MESSAGE_STRIP_ERROR: "\u30A8\u30E9\u30FC\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7",
  MESSAGE_STRIP_WARNING: "\u8B66\u544A\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7",
  MESSAGE_STRIP_CLOSABLE: "\u30AF\u30ED\u30FC\u30BA\u53EF\u80FD",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u60C5\u5831\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7\u3092\u9589\u3058\u308B",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u30DD\u30B8\u30C6\u30A3\u30D6\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7\u3092\u9589\u3058\u308B",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u30CD\u30AC\u30C6\u30A3\u30D6\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7\u3092\u9589\u3058\u308B",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u30AF\u30EA\u30C6\u30A3\u30AB\u30EB\u30E1\u30C3\u30BB\u30FC\u30B8\u30B9\u30C8\u30EA\u30C3\u30D7\u3092\u9589\u3058\u308B",
  FILEUPLOADER_PLACEHOLDER: "\u30D5\u30A1\u30A4\u30EB\u3092\u53C2\u7167\u307E\u305F\u306F\u30C9\u30ED\u30C3\u30D7",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u30D5\u30A1\u30A4\u30EB\u3092\u53C2\u7167\u307E\u305F\u306F\u30C9\u30ED\u30C3\u30D7",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u30D5\u30A1\u30A4\u30EB\u3092\u53C2\u7167",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u3059\u3079\u3066\u30AF\u30EA\u30A2",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u30D5\u30A1\u30A4\u30EB\u30A2\u30C3\u30D7\u30ED\u30FC\u30C9",
  SEARCHFIELD_SEARCH: "\u691C\u7D22",
  SEARCHFIELD_CLEAR: "\u30AF\u30EA\u30A2",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u30C6\u30AD\u30B9\u30C8\u3092\u5165\u529B\u3057\u3066\u691C\u7D22\u3092\u958B\u59CB",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u691C\u7D22\u30D5\u30A3\u30FC\u30EB\u30C9",
  NAVIGATION_EXPAND: "\u5C55\u958B",
  NAVIGATION_COLLAPSE: "\u6298\u308A\u305F\u305F\u3080",
  SIGN_OUT: "\u30B5\u30A4\u30F3\u30A2\u30A6\u30C8",
  FX_START_PANE: "\u958B\u59CB\u30DA\u30A4\u30F3",
  FX_END_PANE: "\u7D42\u4E86\u30DA\u30A4\u30F3",
  FX_RESIZE_START_CENTER: "\u958B\u59CB\u30DA\u30A4\u30F3\u3068\u4E2D\u592E\u30DA\u30A4\u30F3\u306E\u30B5\u30A4\u30BA\u3092\u5909\u66F4",
  FX_RESIZE_CENTER_END: "\u4E2D\u592E\u30DA\u30A4\u30F3\u3068\u7D42\u4E86\u30DA\u30A4\u30F3\u306E\u30B5\u30A4\u30BA\u3092\u5909\u66F4",
  FX_NOTIFICATIONS: "\u901A\u77E5",
  FX_OPEN_NAVIGATION: "\u30CA\u30D3\u30B2\u30FC\u30B7\u30E7\u30F3\u3092\u958B\u304F",
  FX_BACK_TO_LIST: "\u30EA\u30B9\u30C8\u306B\u623B\u308B",
  FX_BACK: "\u623B\u308B",
  FX_RELATED: "\u95A2\u9023",
  FX_MORE_ACTIONS: "\u305D\u306E\u4ED6\u306E\u30A2\u30AF\u30B7\u30E7\u30F3",
  FX_CLOSE: "\u9589\u3058\u308B",
  FX_ADD: "\u8FFD\u52A0",
  FX_SEND: "\u9001\u4FE1",
  FX_DICTATE: "\u97F3\u58F0\u5165\u529B",
  FX_VOICE: "\u97F3\u58F0",
  FX_DOUBLE_CLICK_TO_RENAME: "\u30C0\u30D6\u30EB\u30AF\u30EA\u30C3\u30AF\u3067\u540D\u524D\u3092\u5909\u66F4",
  FX_CANCEL: "\u30AD\u30E3\u30F3\u30BB\u30EB",
  FX_ACCEPT: "\u78BA\u5B9A",
  FX_MAIN_NAVIGATION: "\u30E1\u30A4\u30F3\u30CA\u30D3\u30B2\u30FC\u30B7\u30E7\u30F3\u9805\u76EE",
  FX_USER_ACTIONS: "\u30E6\u30FC\u30B6\u30FC\u30A2\u30AF\u30B7\u30E7\u30F3",
  FX_CURRENT_ACCOUNT: "\u73FE\u5728\u306E\u30A2\u30AB\u30A6\u30F3\u30C8\u60C5\u5831",
  FX_EDIT_ACCOUNTS: "\u30A2\u30AB\u30A6\u30F3\u30C8\u3092\u7DE8\u96C6",
  LINK_SUBTLE: "\u6DE1\u8272",
  LINK_EMPHASIZED: "\u5F37\u8ABF",
  SPLIT_BUTTON_DESCRIPTION: "\u5206\u5272\u30DC\u30BF\u30F3",
  SPLIT_BUTTON_KEYBOARD_HINT: "Space \u307E\u305F\u306F Enter \u3092\u62BC\u3057\u3066\u30C7\u30D5\u30A9\u30EB\u30C8\u30A2\u30AF\u30B7\u30E7\u30F3\u3092\u30C8\u30EA\u30AC\u3057\u3001Alt + \u4E0B\u77E2\u5370\u3001\u307E\u305F\u306F F4 \u3092\u62BC\u3057\u3066\u77E2\u5370\u30A2\u30AF\u30B7\u30E7\u30F3\u3092\u30C8\u30EA\u30AC\u3057\u307E\u3059\u3002",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u30E1\u30CB\u30E5\u30FC\u3092\u958B\u304F"
};

// src/i18n/locales/ko.json
var ko_default = {
  INPUT_CLEAR: "\uC785\uB825 \uC9C0\uC6B0\uAE30",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}}\uAC1C\uC758 \uC81C\uC548 \uC0AC\uC6A9 \uAC00\uB2A5",
  INPUT_SUGGESTIONS_ONE: "1\uAC1C\uC758 \uC81C\uC548 \uC0AC\uC6A9 \uAC00\uB2A5",
  INPUT_SUGGESTIONS_NONE: "\uC81C\uC548 \uC5C6\uC74C",
  INPUT_SHOW_PASSWORD: "\uBE44\uBC00\uBC88\uD638 \uD45C\uC2DC",
  INPUT_HIDE_PASSWORD: "\uBE44\uBC00\uBC88\uD638 \uC228\uAE30\uAE30",
  VALUE_STATE_ERROR: "\uC624\uB958",
  VALUE_STATE_WARNING: "\uACBD\uACE0",
  VALUE_STATE_SUCCESS: "\uC131\uACF5",
  VALUE_STATE_INFORMATION: "\uC815\uBCF4",
  RADIOBUTTON_VALUE_STATE_ERROR: "\uC5D4\uD2B8\uB9AC\uAC00 \uC720\uD6A8\uD558\uC9C0 \uC54A\uC2B5\uB2C8\uB2E4.",
  RADIOBUTTON_VALUE_STATE_WARNING: "\uACBD\uACE0 \uBC1C\uC0DD",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\uC720\uD6A8\uD55C \uC5D4\uD2B8\uB9AC\uC785\uB2C8\uB2E4.",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\uC815\uBCF4 \uC5D4\uD2B8\uB9AC",
  REQUIRED_FIELD: "\uD544\uC218 \uD56D\uBAA9",
  LOADING: "\uB85C\uB529 \uC911...",
  NO_DATA: "\uB370\uC774\uD130 \uC5C6\uC74C",
  TABLE_NO_DATA: "\uC0AC\uC6A9 \uAC00\uB2A5\uD55C \uB370\uC774\uD130 \uC5C6\uC74C",
  TABLE_ROW: "\uD589",
  TABLE_ROW_INDEX: "{{total}}\uAC1C \uC911 {{index}}\uD589",
  TABLE_ROW_SELECTED: "\uC120\uD0DD\uB428",
  TABLE_ROW_ACTIVE: "\uD65C\uC131",
  TABLE_ROW_NAVIGABLE: "\uD0D0\uC0C9 \uAC00\uB2A5",
  TABLE_ROW_NAVIGATED: "\uD0D0\uC0C9\uB428",
  TABLE_ROW_SELECTOR: "\uD589 \uC120\uD0DD\uAE30",
  TABLE_ROW_SINGLE_ACTION: "{{count}}\uAC1C \uC791\uC5C5",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}}\uAC1C \uC791\uC5C5",
  TABLE_ROW_POPIN: "\uCD94\uAC00 \uC815\uBCF4",
  TABLE_ROW_ACTIONS: "\uD589 \uC791\uC5C5",
  TABLE_COLUMN_HEADER_ROW: "\uC5F4 \uD5E4\uB354 \uD589",
  TABLE_SELECTION: "\uC120\uD0DD",
  TABLE_SELECT_ALL_ROWS: "\uBAA8\uB4E0 \uD589 \uC120\uD0DD",
  TABLE_DESELECT_ALL_ROWS: "\uBAA8\uB4E0 \uD589 \uC120\uD0DD \uD574\uC81C",
  TABLE_MORE: "\uB354 \uBCF4\uAE30",
  TABLE_MORE_DESCRIPTION: "\uB354 \uB9CE\uC740 \uB370\uC774\uD130 \uB85C\uB4DC",
  TABLE_MULTI_SELECTABLE: "\uB2E4\uC911 \uC120\uD0DD \uD14C\uC774\uBE14",
  TABLE_SINGLE_SELECTABLE: "\uB2E8\uC77C \uC120\uD0DD \uD14C\uC774\uBE14",
  TABLE_NAVIGATION: "\uD0D0\uC0C9",
  TABLE_SORT_ASCENDING: "\uC624\uB984\uCC28\uC21C \uC815\uB82C\uB428",
  TABLE_SORT_DESCENDING: "\uB0B4\uB9BC\uCC28\uC21C \uC815\uB82C\uB428",
  TABLE_SORT_NONE: "\uC815\uB82C\uB418\uC9C0 \uC54A\uC74C",
  TABLE_AI_SORT_PLACEHOLDER: "\uC815\uB82C \uBC29\uBC95\uC744 \uC124\uBA85\uD558\uC138\uC694...",
  TABLE_AI_FILTER_PLACEHOLDER: "\uD544\uD130\uB9C1\uD560 \uB0B4\uC6A9\uC744 \uC124\uBA85\uD558\uC138\uC694...",
  TABLE_AI_CHAT_PLACEHOLDER: "\uB370\uC774\uD130\uC5D0 \uB300\uD574 \uC9C8\uBB38\uD558\uC138\uC694...",
  TABLE_AI_INSIGHT_ANOMALY: "\uC774\uC0C1 \uAC10\uC9C0\uB428",
  TABLE_AI_INSIGHT_TREND: "\uCD94\uC138 \uD655\uC778\uB428",
  TABLE_AI_INSIGHT_PATTERN: "\uD328\uD134 \uBC1C\uACAC\uB428",
  TABLE_AI_INSIGHT_DISMISS: "\uC778\uC0AC\uC774\uD2B8 \uB2EB\uAE30",
  TABLE_AI_GENERATED: "AI\uAC00 \uC0DD\uC131\uD568",
  TABLE_LOADING_MORE: "\uB354 \uB85C\uB4DC \uC911...",
  TABLE_TOOLBAR: "\uD14C\uC774\uBE14 \uC791\uC5C5",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\uD14C\uC774\uBE14 \uB3C4\uAD6C \uBAA8\uC74C",
  TABLE_ROWS_SELECTED: "{{count}}\uAC1C \uD589 \uC120\uD0DD\uB428",
  TOKEN_ARIA_LABEL: "\uD1A0\uD070",
  TOKEN_ARIA_DELETABLE: "\uC0AD\uC81C \uAC00\uB2A5",
  TOKEN_DELETE: "\uC81C\uAC70",
  TOKENIZER_ARIA_LABEL: "\uD1A0\uD06C\uB098\uC774\uC800",
  TOKENIZER_SHOW_MORE: "{{count}}\uAC1C \uB354",
  TOKENIZER_POPOVER_HEADER: "\uCD94\uAC00 \uD1A0\uD070",
  TOKENIZER_CLEAR_ALL: "\uBAA8\uB450 \uC9C0\uC6B0\uAE30",
  TOKENIZER_TOKEN_COUNT: "{{count}}\uAC1C \uD1A0\uD070",
  TOKENIZER_TOKENS_SELECTED: "{{count}}\uAC1C \uD1A0\uD070 \uC120\uD0DD\uB428",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}}\uAC1C\uC758 \uACB0\uACFC \uC0AC\uC6A9 \uAC00\uB2A5",
  COMBOBOX_ONE_RESULT: "1\uAC1C\uC758 \uACB0\uACFC \uC0AC\uC6A9 \uAC00\uB2A5",
  COMBOBOX_NO_RESULTS: "\uACB0\uACFC \uC5C6\uC74C",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\uB9AC\uC2A4\uD2B8 \uC0C1\uC790",
  SELECT_OPTION_SELECTED: "{{text}} \uC120\uD0DD\uB428",
  MULTICOMBOBOX_SELECT_ALL: "\uBAA8\uB450 \uC120\uD0DD",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}}\uAC1C \uD56D\uBAA9 \uC120\uD0DD\uB428",
  MENU_POPOVER_ACCESSIBLE_NAME: "\uBA54\uB274\uC5D0\uC11C \uC635\uC158\uC744 \uC120\uD0DD\uD558\uC2ED\uC2DC\uC624.",
  DATEPICKER_DATE_DESCRIPTION: "\uC77C\uC790 \uC785\uB825",
  DATERANGE_DESCRIPTION: "\uC77C\uC790 \uBC94\uC704 \uC785\uB825",
  DATEPICKER_OPEN_ICON_TITLE: "\uC120\uD0DD \uB3C4\uAD6C \uC5F4\uAE30",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\uC120\uD0DD \uB3C4\uAD6C \uB2EB\uAE30",
  INPUT_CLEAR_ICON_ACC_NAME: "\uC9C0\uC6B0\uAE30",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "{{0}}\uC758 \uC77C\uC790 \uC120\uD0DD",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "{{0}}\uC758 \uC77C\uC790 \uBC94\uC704 \uC120\uD0DD",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\uC138\uADF8\uBA58\uD14C\uC774\uC158\uB41C \uBC84\uD2BC \uADF8\uB8F9",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\uC138\uADF8\uBA58\uD14C\uC774\uC158\uB41C \uBC84\uD2BC",
  MESSAGE_STRIP_INFORMATION: "\uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD",
  MESSAGE_STRIP_SUCCESS: "\uC131\uACF5 \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD",
  MESSAGE_STRIP_ERROR: "\uC624\uB958 \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD",
  MESSAGE_STRIP_WARNING: "\uACBD\uACE0 \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD",
  MESSAGE_STRIP_CLOSABLE: "\uB2EB\uAE30 \uAC00\uB2A5",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\uC815\uBCF4 \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD \uB2EB\uAE30",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\uAE0D\uC815\uC801 \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD \uB2EB\uAE30",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\uBD80\uC815\uC801 \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD \uB2EB\uAE30",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\uC911\uC694\uD55C \uBA54\uC2DC\uC9C0 \uC2A4\uD2B8\uB9BD \uB2EB\uAE30",
  FILEUPLOADER_PLACEHOLDER: "\uD30C\uC77C \uCC3E\uAE30 \uB610\uB294 \uB193\uAE30",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\uD30C\uC77C \uCC3E\uAE30 \uB610\uB294 \uB193\uAE30",
  FILEUPLOADER_BROWSE_TOOLTIP: "\uD30C\uC77C \uCC3E\uC544\uBCF4\uAE30",
  FILEUPLOADER_CLEAR_TOOLTIP: "\uBAA8\uB450 \uC9C0\uC6B0\uAE30",
  FILEUPLOADER_ROLE_DESCRIPTION: "\uD30C\uC77C \uC5C5\uB85C\uB4DC",
  SEARCHFIELD_SEARCH: "\uAC80\uC0C9",
  SEARCHFIELD_CLEAR: "\uC9C0\uC6B0\uAE30",
  SEARCHFIELD_ARIA_DESCRIPTION: "\uD14D\uC2A4\uD2B8\uB97C \uC785\uB825\uD558\uC5EC \uAC80\uC0C9 \uC2DC\uC791",
  SEARCHFIELD_ROLE_DESCRIPTION: "\uAC80\uC0C9 \uD544\uB4DC",
  NAVIGATION_EXPAND: "\uD3BC\uCE58\uAE30",
  NAVIGATION_COLLAPSE: "\uC811\uAE30",
  SIGN_OUT: "\uB85C\uADF8\uC544\uC6C3",
  FX_START_PANE: "\uC2DC\uC791 \uCC3D",
  FX_END_PANE: "\uB05D \uCC3D",
  FX_RESIZE_START_CENTER: "\uC2DC\uC791 \uBC0F \uC911\uC559 \uCC3D \uD06C\uAE30 \uC870\uC815",
  FX_RESIZE_CENTER_END: "\uC911\uC559 \uBC0F \uB05D \uCC3D \uD06C\uAE30 \uC870\uC815",
  FX_NOTIFICATIONS: "\uC54C\uB9BC",
  FX_OPEN_NAVIGATION: "\uD0D0\uC0C9 \uC5F4\uAE30",
  FX_BACK_TO_LIST: "\uBAA9\uB85D\uC73C\uB85C \uB3CC\uC544\uAC00\uAE30",
  FX_BACK: "\uB4A4\uB85C",
  FX_RELATED: "\uAD00\uB828",
  FX_MORE_ACTIONS: "\uCD94\uAC00 \uC791\uC5C5",
  FX_CLOSE: "\uB2EB\uAE30",
  FX_ADD: "\uCD94\uAC00",
  FX_SEND: "\uBCF4\uB0B4\uAE30",
  FX_DICTATE: "\uBC1B\uC544\uC4F0\uAE30",
  FX_VOICE: "\uC74C\uC131",
  FX_DOUBLE_CLICK_TO_RENAME: "\uC774\uB984\uC744 \uBC14\uAFB8\uB824\uBA74 \uB450 \uBC88 \uD074\uB9AD\uD558\uC138\uC694",
  FX_CANCEL: "\uCDE8\uC18C",
  FX_ACCEPT: "\uC218\uB77D",
  FX_MAIN_NAVIGATION: "\uC8FC \uD0D0\uC0C9 \uD56D\uBAA9",
  FX_USER_ACTIONS: "\uC0AC\uC6A9\uC790 \uC791\uC5C5",
  FX_CURRENT_ACCOUNT: "\uD604\uC7AC \uACC4\uC815 \uC815\uBCF4",
  FX_EDIT_ACCOUNTS: "\uACC4\uC815 \uD3B8\uC9D1",
  LINK_SUBTLE: "\uBBF8\uC57D",
  LINK_EMPHASIZED: "\uAC15\uC870",
  SPLIT_BUTTON_DESCRIPTION: "\uBD84\uD560 \uBC84\uD2BC",
  SPLIT_BUTTON_KEYBOARD_HINT: "\uC2A4\uD398\uC774\uC2A4\uBC14 \uB610\uB294 Enter\uB97C \uB20C\uB7EC \uAE30\uBCF8 \uC561\uC158 \uBC0F Alt + Arrow Down \uB610\uB294 F4\uB97C \uD2B8\uB9AC\uAC70\uD558\uAC70\uB098 \uD654\uC0B4\uD45C \uC561\uC158\uC744 \uD2B8\uB9AC\uAC70\uD569\uB2C8\uB2E4.",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\uBA54\uB274 \uC5F4\uAE30"
};

// src/i18n/locales/zh-CN.json
var zh_CN_default = {
  INPUT_CLEAR: "\u6E05\u9664\u8F93\u5165",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}}\u6761\u5EFA\u8BAE\u53EF\u7528",
  INPUT_SUGGESTIONS_ONE: "1\u6761\u5EFA\u8BAE\u53EF\u7528",
  INPUT_SUGGESTIONS_NONE: "\u65E0\u5EFA\u8BAE",
  INPUT_SHOW_PASSWORD: "\u663E\u793A\u5BC6\u7801",
  INPUT_HIDE_PASSWORD: "\u9690\u85CF\u5BC6\u7801",
  VALUE_STATE_ERROR: "\u9519\u8BEF",
  VALUE_STATE_WARNING: "\u8B66\u544A",
  VALUE_STATE_SUCCESS: "\u6210\u529F",
  VALUE_STATE_INFORMATION: "\u4FE1\u606F",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u65E0\u6548\u8F93\u5165",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u5DF2\u53D1\u51FA\u8B66\u544A",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u5DF2\u6210\u529F\u9A8C\u8BC1\u8F93\u5165",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u4FE1\u606F\u8F93\u5165",
  REQUIRED_FIELD: "\u5FC5\u586B\u5B57\u6BB5",
  LOADING: "\u52A0\u8F7D\u4E2D...",
  NO_DATA: "\u65E0\u6570\u636E",
  TABLE_NO_DATA: "\u6682\u65E0\u6570\u636E",
  TABLE_ROW: "\u884C",
  TABLE_ROW_INDEX: "\u7B2C{{index}}\u884C\uFF0C\u5171{{total}}\u884C",
  TABLE_ROW_SELECTED: "\u5DF2\u9009\u4E2D",
  TABLE_ROW_ACTIVE: "\u6D3B\u52A8",
  TABLE_ROW_NAVIGABLE: "\u53EF\u5BFC\u822A",
  TABLE_ROW_NAVIGATED: "\u5DF2\u5BFC\u822A",
  TABLE_ROW_SELECTOR: "\u884C\u9009\u62E9\u5668",
  TABLE_ROW_SINGLE_ACTION: "{{count}}\u4E2A\u64CD\u4F5C",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}}\u4E2A\u64CD\u4F5C",
  TABLE_ROW_POPIN: "\u9644\u52A0\u4FE1\u606F",
  TABLE_ROW_ACTIONS: "\u884C\u64CD\u4F5C",
  TABLE_COLUMN_HEADER_ROW: "\u5217\u6807\u9898\u884C",
  TABLE_SELECTION: "\u9009\u62E9",
  TABLE_SELECT_ALL_ROWS: "\u9009\u62E9\u6240\u6709\u884C",
  TABLE_DESELECT_ALL_ROWS: "\u53D6\u6D88\u9009\u62E9\u6240\u6709\u884C",
  TABLE_MORE: "\u66F4\u591A",
  TABLE_MORE_DESCRIPTION: "\u52A0\u8F7D\u66F4\u591A\u6570\u636E",
  TABLE_MULTI_SELECTABLE: "\u591A\u9009\u8868\u683C",
  TABLE_SINGLE_SELECTABLE: "\u5355\u9009\u8868\u683C",
  TABLE_NAVIGATION: "\u5BFC\u822A",
  TABLE_SORT_ASCENDING: "\u5347\u5E8F\u6392\u5217",
  TABLE_SORT_DESCENDING: "\u964D\u5E8F\u6392\u5217",
  TABLE_SORT_NONE: "\u672A\u6392\u5E8F",
  TABLE_AI_SORT_PLACEHOLDER: "\u63CF\u8FF0\u5982\u4F55\u6392\u5E8F...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u63CF\u8FF0\u8981\u8FC7\u6EE4\u7684\u5185\u5BB9...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u8BE2\u95EE\u60A8\u7684\u6570\u636E...",
  TABLE_AI_INSIGHT_ANOMALY: "\u68C0\u6D4B\u5230\u5F02\u5E38",
  TABLE_AI_INSIGHT_TREND: "\u8BC6\u522B\u5230\u8D8B\u52BF",
  TABLE_AI_INSIGHT_PATTERN: "\u53D1\u73B0\u6A21\u5F0F",
  TABLE_AI_INSIGHT_DISMISS: "\u5173\u95ED\u6D1E\u5BDF",
  TABLE_AI_GENERATED: "\u7531AI\u751F\u6210",
  TABLE_LOADING_MORE: "\u6B63\u5728\u52A0\u8F7D\u66F4\u591A...",
  TABLE_TOOLBAR: "\u8868\u683C\u64CD\u4F5C",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u8868\u683C\u5DE5\u5177\u680F",
  TABLE_ROWS_SELECTED: "\u5DF2\u9009\u62E9{{count}}\u884C",
  TOKEN_ARIA_LABEL: "\u4EE4\u724C",
  TOKEN_ARIA_DELETABLE: "\u53EF\u5220\u9664",
  TOKEN_DELETE: "\u79FB\u9664",
  TOKENIZER_ARIA_LABEL: "\u4EE4\u724C\u7BA1\u7406\u5668",
  TOKENIZER_SHOW_MORE: "\u8FD8\u6709{{count}}\u4E2A",
  TOKENIZER_POPOVER_HEADER: "\u66F4\u591A\u4EE4\u724C",
  TOKENIZER_CLEAR_ALL: "\u5168\u90E8\u6E05\u9664",
  TOKENIZER_TOKEN_COUNT: "{{count}}\u4E2A\u4EE4\u724C",
  TOKENIZER_TOKENS_SELECTED: "\u5DF2\u9009\u62E9{{count}}\u4E2A\u4EE4\u724C",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}}\u6761\u7ED3\u679C\u53EF\u7528",
  COMBOBOX_ONE_RESULT: "1\u6761\u7ED3\u679C\u53EF\u7528",
  COMBOBOX_NO_RESULTS: "\u65E0\u7ED3\u679C",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u5217\u8868\u6846",
  SELECT_OPTION_SELECTED: "\u5DF2\u9009\u62E9{{text}}",
  MULTICOMBOBOX_SELECT_ALL: "\u5168\u9009",
  MULTICOMBOBOX_ITEMS_SELECTED: "\u5DF2\u9009\u62E9{{count}}\u9879",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u4ECE\u83DC\u5355\u4E2D\u9009\u62E9\u4E00\u4E2A\u9009\u9879",
  DATEPICKER_DATE_DESCRIPTION: "\u65E5\u671F\u8F93\u5165",
  DATERANGE_DESCRIPTION: "\u65E5\u671F\u8303\u56F4\u8F93\u5165",
  DATEPICKER_OPEN_ICON_TITLE: "\u6253\u5F00\u9009\u53D6\u5668",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u5173\u95ED\u9009\u53D6\u5668",
  INPUT_CLEAR_ICON_ACC_NAME: "\u6E05\u9664",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u4E3A {{0}} \u9009\u62E9\u65E5\u671F",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u4E3A {{0}} \u9009\u62E9\u65E5\u671F\u8303\u56F4",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u5206\u6BB5\u5F0F\u6309\u94AE\u7EC4",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u5206\u6BB5\u5F0F\u6309\u94AE",
  MESSAGE_STRIP_INFORMATION: "\u6D88\u606F\u6761",
  MESSAGE_STRIP_SUCCESS: "\u6210\u529F\u6D88\u606F\u6761",
  MESSAGE_STRIP_ERROR: "\u9519\u8BEF\u6D88\u606F\u6761",
  MESSAGE_STRIP_WARNING: "\u8B66\u544A\u6D88\u606F\u6761",
  MESSAGE_STRIP_CLOSABLE: "\u53EF\u5173\u95ED",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u5173\u95ED\u4FE1\u606F\u6D88\u606F\u6761",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u5173\u95ED\u79EF\u6781\u6D88\u606F\u6761",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u5173\u95ED\u6D88\u6781\u6D88\u606F\u6761",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u5173\u95ED\u4E34\u754C\u6D88\u606F\u6761",
  FILEUPLOADER_PLACEHOLDER: "\u6D4F\u89C8\u6216\u62D6\u653E\u6587\u4EF6",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u6D4F\u89C8\u6216\u62D6\u653E\u6587\u4EF6",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u6D4F\u89C8\u6587\u4EF6",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u5168\u90E8\u6E05\u9664",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u6587\u4EF6\u4E0A\u4F20",
  SEARCHFIELD_SEARCH: "\u641C\u7D22",
  SEARCHFIELD_CLEAR: "\u6E05\u9664",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u8F93\u5165\u6587\u672C\u5F00\u59CB\u641C\u7D22",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u641C\u7D22\u5B57\u6BB5",
  NAVIGATION_EXPAND: "\u5C55\u5F00",
  NAVIGATION_COLLAPSE: "\u6298\u53E0",
  SIGN_OUT: "\u6CE8\u9500",
  FX_START_PANE: "\u8D77\u59CB\u7A97\u683C",
  FX_END_PANE: "\u7ED3\u675F\u7A97\u683C",
  FX_RESIZE_START_CENTER: "\u8C03\u6574\u8D77\u59CB\u548C\u4E2D\u592E\u7A97\u683C\u5927\u5C0F",
  FX_RESIZE_CENTER_END: "\u8C03\u6574\u4E2D\u592E\u548C\u7ED3\u675F\u7A97\u683C\u5927\u5C0F",
  FX_NOTIFICATIONS: "\u901A\u77E5",
  FX_OPEN_NAVIGATION: "\u6253\u5F00\u5BFC\u822A",
  FX_BACK_TO_LIST: "\u8FD4\u56DE\u5217\u8868",
  FX_BACK: "\u8FD4\u56DE",
  FX_RELATED: "\u76F8\u5173",
  FX_MORE_ACTIONS: "\u66F4\u591A\u64CD\u4F5C",
  FX_CLOSE: "\u5173\u95ED",
  FX_ADD: "\u6DFB\u52A0",
  FX_SEND: "\u53D1\u9001",
  FX_DICTATE: "\u542C\u5199",
  FX_VOICE: "\u8BED\u97F3",
  FX_DOUBLE_CLICK_TO_RENAME: "\u53CC\u51FB\u4EE5\u91CD\u547D\u540D",
  FX_CANCEL: "\u53D6\u6D88",
  FX_ACCEPT: "\u63A5\u53D7",
  FX_MAIN_NAVIGATION: "\u4E3B\u5BFC\u822A\u9879",
  FX_USER_ACTIONS: "\u7528\u6237\u64CD\u4F5C",
  FX_CURRENT_ACCOUNT: "\u5F53\u524D\u8D26\u6237\u4FE1\u606F",
  FX_EDIT_ACCOUNTS: "\u7F16\u8F91\u8D26\u6237",
  LINK_SUBTLE: "\u9690\u853D",
  LINK_EMPHASIZED: "\u52A0\u91CD",
  SPLIT_BUTTON_DESCRIPTION: "\u62C6\u5206\u6309\u94AE",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u6309\u7A7A\u683C\u952E\u6216 Enter \u89E6\u53D1\u7F3A\u7701\u64CD\u4F5C\uFF0C\u6309 Alt + \u5411\u4E0B\u7BAD\u5934\u6216 F4 \u89E6\u53D1\u7BAD\u5934\u64CD\u4F5C",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u6253\u5F00\u83DC\u5355"
};

// src/i18n/locales/zh-TW.json
var zh_TW_default = {
  INPUT_CLEAR: "\u6E05\u9664\u8F38\u5165",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}}\u689D\u5EFA\u8B70\u53EF\u7528",
  INPUT_SUGGESTIONS_ONE: "1\u689D\u5EFA\u8B70\u53EF\u7528",
  INPUT_SUGGESTIONS_NONE: "\u7121\u5EFA\u8B70",
  INPUT_SHOW_PASSWORD: "\u986F\u793A\u5BC6\u78BC",
  INPUT_HIDE_PASSWORD: "\u96B1\u85CF\u5BC6\u78BC",
  VALUE_STATE_ERROR: "\u932F\u8AA4",
  VALUE_STATE_WARNING: "\u8B66\u544A",
  VALUE_STATE_SUCCESS: "\u6210\u529F",
  VALUE_STATE_INFORMATION: "\u8CC7\u8A0A",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u7121\u6548\u8F38\u5165\u9805",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u5DF2\u767C\u51FA\u8B66\u544A",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u5DF2\u6210\u529F\u9A57\u8B49\u8F38\u5165\u9805",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u8CC7\u8A0A\u8F38\u5165\u9805",
  REQUIRED_FIELD: "\u5FC5\u586B\u6B04\u4F4D",
  LOADING: "\u8F09\u5165\u4E2D...",
  NO_DATA: "\u7121\u8CC7\u6599",
  TABLE_NO_DATA: "\u66AB\u7121\u8CC7\u6599",
  TABLE_ROW: "\u5217",
  TABLE_ROW_INDEX: "\u7B2C{{index}}\u5217\uFF0C\u5171{{total}}\u5217",
  TABLE_ROW_SELECTED: "\u5DF2\u9078\u53D6",
  TABLE_ROW_ACTIVE: "\u6D3B\u52D5",
  TABLE_ROW_NAVIGABLE: "\u53EF\u5C0E\u89BD",
  TABLE_ROW_NAVIGATED: "\u5DF2\u5C0E\u89BD",
  TABLE_ROW_SELECTOR: "\u5217\u9078\u64C7\u5668",
  TABLE_ROW_SINGLE_ACTION: "{{count}}\u500B\u64CD\u4F5C",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}}\u500B\u64CD\u4F5C",
  TABLE_ROW_POPIN: "\u9644\u52A0\u8CC7\u8A0A",
  TABLE_ROW_ACTIONS: "\u5217\u64CD\u4F5C",
  TABLE_COLUMN_HEADER_ROW: "\u6B04\u6A19\u984C\u5217",
  TABLE_SELECTION: "\u9078\u53D6",
  TABLE_SELECT_ALL_ROWS: "\u9078\u53D6\u6240\u6709\u5217",
  TABLE_DESELECT_ALL_ROWS: "\u53D6\u6D88\u9078\u53D6\u6240\u6709\u5217",
  TABLE_MORE: "\u66F4\u591A",
  TABLE_MORE_DESCRIPTION: "\u8F09\u5165\u66F4\u591A\u8CC7\u6599",
  TABLE_MULTI_SELECTABLE: "\u591A\u9078\u8868\u683C",
  TABLE_SINGLE_SELECTABLE: "\u55AE\u9078\u8868\u683C",
  TABLE_NAVIGATION: "\u5C0E\u89BD",
  TABLE_SORT_ASCENDING: "\u5347\u51AA\u6392\u5217",
  TABLE_SORT_DESCENDING: "\u964D\u51AA\u6392\u5217",
  TABLE_SORT_NONE: "\u672A\u6392\u5E8F",
  TABLE_AI_SORT_PLACEHOLDER: "\u63CF\u8FF0\u5982\u4F55\u6392\u5E8F...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u63CF\u8FF0\u8981\u7BE9\u9078\u7684\u5167\u5BB9...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u8A62\u554F\u60A8\u7684\u8CC7\u6599...",
  TABLE_AI_INSIGHT_ANOMALY: "\u5075\u6E2C\u5230\u7570\u5E38",
  TABLE_AI_INSIGHT_TREND: "\u8B58\u5225\u5230\u8DA8\u52E2",
  TABLE_AI_INSIGHT_PATTERN: "\u767C\u73FE\u6A21\u5F0F",
  TABLE_AI_INSIGHT_DISMISS: "\u95DC\u9589\u6D1E\u5BDF",
  TABLE_AI_GENERATED: "\u7531AI\u751F\u6210",
  TABLE_LOADING_MORE: "\u6B63\u5728\u8F09\u5165\u66F4\u591A...",
  TABLE_TOOLBAR: "\u8868\u683C\u64CD\u4F5C",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u8868\u683C\u5DE5\u5177\u5217",
  TABLE_ROWS_SELECTED: "\u5DF2\u9078\u53D6{{count}}\u5217",
  TOKEN_ARIA_LABEL: "\u4EE4\u724C",
  TOKEN_ARIA_DELETABLE: "\u53EF\u522A\u9664",
  TOKEN_DELETE: "\u79FB\u9664",
  TOKENIZER_ARIA_LABEL: "\u4EE4\u724C\u7BA1\u7406\u5668",
  TOKENIZER_SHOW_MORE: "\u9084\u6709{{count}}\u500B",
  TOKENIZER_POPOVER_HEADER: "\u66F4\u591A\u4EE4\u724C",
  TOKENIZER_CLEAR_ALL: "\u5168\u90E8\u6E05\u9664",
  TOKENIZER_TOKEN_COUNT: "{{count}}\u500B\u4EE4\u724C",
  TOKENIZER_TOKENS_SELECTED: "\u5DF2\u9078\u53D6{{count}}\u500B\u4EE4\u724C",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}}\u689D\u7D50\u679C\u53EF\u7528",
  COMBOBOX_ONE_RESULT: "1\u689D\u7D50\u679C\u53EF\u7528",
  COMBOBOX_NO_RESULTS: "\u7121\u7D50\u679C",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u6E05\u55AE\u65B9\u584A",
  SELECT_OPTION_SELECTED: "\u5DF2\u9078\u53D6{{text}}",
  MULTICOMBOBOX_SELECT_ALL: "\u5168\u9078",
  MULTICOMBOBOX_ITEMS_SELECTED: "\u5DF2\u9078\u53D6{{count}}\u9805",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u5F9E\u529F\u80FD\u8868\u9078\u53D6\u9078\u9805",
  DATEPICKER_DATE_DESCRIPTION: "\u65E5\u671F\u8F38\u5165",
  DATERANGE_DESCRIPTION: "\u65E5\u671F\u7BC4\u570D\u8F38\u5165",
  DATEPICKER_OPEN_ICON_TITLE: "\u958B\u555F\u9078\u64C7\u5668",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u95DC\u9589\u9078\u64C7\u5668",
  INPUT_CLEAR_ICON_ACC_NAME: "\u6E05\u9664",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u9078\u64C7 {{0}} \u7684\u65E5\u671F",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u9078\u64C7 {{0}} \u7684\u65E5\u671F\u7BC4\u570D",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u5340\u6BB5\u6309\u9215\u7FA4\u7D44",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u5340\u6BB5\u6309\u9215",
  MESSAGE_STRIP_INFORMATION: "\u8A0A\u606F\u5217",
  MESSAGE_STRIP_SUCCESS: "\u6210\u529F\u8A0A\u606F\u5217",
  MESSAGE_STRIP_ERROR: "\u932F\u8AA4\u8A0A\u606F\u5217",
  MESSAGE_STRIP_WARNING: "\u8B66\u544A\u8A0A\u606F\u5217",
  MESSAGE_STRIP_CLOSABLE: "\u53EF\u95DC\u9589",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u95DC\u9589\u8CC7\u8A0A\u8A0A\u606F\u5217",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u95DC\u9589\u6B63\u5411\u8A0A\u606F\u5217",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u95DC\u9589\u53CD\u5411\u8A0A\u606F\u5217",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u95DC\u9589\u91CD\u8981\u8A0A\u606F\u5217",
  FILEUPLOADER_PLACEHOLDER: "\u700F\u89BD\u6216\u62D6\u653E\u6A94\u6848",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u700F\u89BD\u6216\u62D6\u653E\u6A94\u6848",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u700F\u89BD\u6A94\u6848",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u5168\u90E8\u6E05\u9664",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u6A94\u6848\u4E0A\u50B3",
  SEARCHFIELD_SEARCH: "\u641C\u5C0B",
  SEARCHFIELD_CLEAR: "\u6E05\u9664",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u8F38\u5165\u6587\u5B57\u4EE5\u958B\u59CB\u641C\u5C0B",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u641C\u5C0B\u6B04\u4F4D",
  NAVIGATION_EXPAND: "\u5C55\u958B",
  NAVIGATION_COLLAPSE: "\u647A\u758A",
  SIGN_OUT: "\u767B\u51FA",
  FX_START_PANE: "\u8D77\u59CB\u7A97\u683C",
  FX_END_PANE: "\u7D50\u675F\u7A97\u683C",
  FX_RESIZE_START_CENTER: "\u8ABF\u6574\u8D77\u59CB\u548C\u4E2D\u592E\u7A97\u683C\u5927\u5C0F",
  FX_RESIZE_CENTER_END: "\u8ABF\u6574\u4E2D\u592E\u548C\u7D50\u675F\u7A97\u683C\u5927\u5C0F",
  FX_NOTIFICATIONS: "\u901A\u77E5",
  FX_OPEN_NAVIGATION: "\u958B\u555F\u5C0E\u89BD",
  FX_BACK_TO_LIST: "\u8FD4\u56DE\u5217\u8868",
  FX_BACK: "\u8FD4\u56DE",
  FX_RELATED: "\u76F8\u95DC",
  FX_MORE_ACTIONS: "\u66F4\u591A\u64CD\u4F5C",
  FX_CLOSE: "\u95DC\u9589",
  FX_ADD: "\u65B0\u589E",
  FX_SEND: "\u50B3\u9001",
  FX_DICTATE: "\u807D\u5BEB",
  FX_VOICE: "\u8A9E\u97F3",
  FX_DOUBLE_CLICK_TO_RENAME: "\u6309\u5169\u4E0B\u4EE5\u91CD\u65B0\u547D\u540D",
  FX_CANCEL: "\u53D6\u6D88",
  FX_ACCEPT: "\u63A5\u53D7",
  FX_MAIN_NAVIGATION: "\u4E3B\u5C0E\u89BD\u9805\u76EE",
  FX_USER_ACTIONS: "\u4F7F\u7528\u8005\u64CD\u4F5C",
  FX_CURRENT_ACCOUNT: "\u76EE\u524D\u5E33\u6236\u8CC7\u8A0A",
  FX_EDIT_ACCOUNTS: "\u7DE8\u8F2F\u5E33\u6236",
  LINK_SUBTLE: "\u8F15\u5FAE",
  LINK_EMPHASIZED: "\u5F37\u8ABF",
  SPLIT_BUTTON_DESCRIPTION: "\u5206\u5272\u6309\u9215",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u6309\u4E0B\u7A7A\u683C\u9375\u6216 Enter \u4EE5\u9A45\u52D5\u9810\u8A2D\u52D5\u4F5C\uFF0C\u4EE5\u53CA Alt + \u5411\u4E0B\u7BAD\u982D\u6216 F4 \u4EE5\u9A45\u52D5\u7BAD\u982D\u52D5\u4F5C",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u958B\u555F\u529F\u80FD\u8868"
};

// src/i18n/locales/ru.json
var ru_default = {
  INPUT_CLEAR: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u044C",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} \u043F\u0440\u0435\u0434\u043B\u043E\u0436\u0435\u043D\u0438\u0439 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u043E",
  INPUT_SUGGESTIONS_ONE: "1 \u043F\u0440\u0435\u0434\u043B\u043E\u0436\u0435\u043D\u0438\u0435 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u043E",
  INPUT_SUGGESTIONS_NONE: "\u041D\u0435\u0442 \u043F\u0440\u0435\u0434\u043B\u043E\u0436\u0435\u043D\u0438\u0439",
  INPUT_SHOW_PASSWORD: "\u041F\u043E\u043A\u0430\u0437\u0430\u0442\u044C \u043F\u0430\u0440\u043E\u043B\u044C",
  INPUT_HIDE_PASSWORD: "\u0421\u043A\u0440\u044B\u0442\u044C \u043F\u0430\u0440\u043E\u043B\u044C",
  VALUE_STATE_ERROR: "\u041E\u0448\u0438\u0431\u043A\u0430",
  VALUE_STATE_WARNING: "\u041F\u0440\u0435\u0434\u0443\u043F\u0440\u0435\u0436\u0434\u0435\u043D\u0438\u0435",
  VALUE_STATE_SUCCESS: "\u0423\u0441\u043F\u0435\u0448\u043D\u043E",
  VALUE_STATE_INFORMATION: "\u0418\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u044F",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u041D\u0435\u0434\u0435\u0439\u0441\u0442\u0432\u0438\u0442\u0435\u043B\u044C\u043D\u044B\u0439 \u0432\u0432\u043E\u0434",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u0412\u044B\u0434\u0430\u043D\u043E \u043F\u0440\u0435\u0434\u0443\u043F\u0440\u0435\u0436\u0434\u0435\u043D\u0438\u0435",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u0417\u0430\u043F\u0438\u0441\u044C \u0443\u0441\u043F\u0435\u0448\u043D\u043E \u0432\u0430\u043B\u0438\u0434\u0438\u0440\u043E\u0432\u0430\u043D\u0430",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u0418\u043D\u0444\u043E\u0440\u043C\u0430\u0442\u0438\u0432\u043D\u0430\u044F \u0437\u0430\u043F\u0438\u0441\u044C",
  REQUIRED_FIELD: "\u041E\u0431\u044F\u0437\u0430\u0442\u0435\u043B\u044C\u043D\u043E\u0435 \u043F\u043E\u043B\u0435",
  LOADING: "\u0417\u0430\u0433\u0440\u0443\u0437\u043A\u0430...",
  NO_DATA: "\u041D\u0435\u0442 \u0434\u0430\u043D\u043D\u044B\u0445",
  TABLE_NO_DATA: "\u0414\u0430\u043D\u043D\u044B\u0435 \u043E\u0442\u0441\u0443\u0442\u0441\u0442\u0432\u0443\u044E\u0442",
  TABLE_ROW: "\u0421\u0442\u0440\u043E\u043A\u0430",
  TABLE_ROW_INDEX: "\u0421\u0442\u0440\u043E\u043A\u0430 {{index}} \u0438\u0437 {{total}}",
  TABLE_ROW_SELECTED: "\u0412\u044B\u0431\u0440\u0430\u043D\u0430",
  TABLE_ROW_ACTIVE: "\u0410\u043A\u0442\u0438\u0432\u043D\u0430\u044F",
  TABLE_ROW_NAVIGABLE: "\u0414\u043E\u0441\u0442\u0443\u043F\u043D\u0430 \u043D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u044F",
  TABLE_ROW_NAVIGATED: "\u041F\u043E\u0441\u0435\u0449\u0435\u043D\u0430",
  TABLE_ROW_SELECTOR: "\u0412\u044B\u0431\u043E\u0440 \u0441\u0442\u0440\u043E\u043A\u0438",
  TABLE_ROW_SINGLE_ACTION: "{{count}} \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u0435",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u0439",
  TABLE_ROW_POPIN: "\u0414\u043E\u043F\u043E\u043B\u043D\u0438\u0442\u0435\u043B\u044C\u043D\u0430\u044F \u0438\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u044F",
  TABLE_ROW_ACTIONS: "\u0414\u0435\u0439\u0441\u0442\u0432\u0438\u044F \u0441\u0442\u0440\u043E\u043A\u0438",
  TABLE_COLUMN_HEADER_ROW: "\u0421\u0442\u0440\u043E\u043A\u0430 \u0437\u0430\u0433\u043E\u043B\u043E\u0432\u043A\u043E\u0432 \u0441\u0442\u043E\u043B\u0431\u0446\u043E\u0432",
  TABLE_SELECTION: "\u0412\u044B\u0431\u043E\u0440",
  TABLE_SELECT_ALL_ROWS: "\u0412\u044B\u0431\u0440\u0430\u0442\u044C \u0432\u0441\u0435 \u0441\u0442\u0440\u043E\u043A\u0438",
  TABLE_DESELECT_ALL_ROWS: "\u0421\u043D\u044F\u0442\u044C \u0432\u044B\u0431\u043E\u0440 \u0441\u043E \u0432\u0441\u0435\u0445 \u0441\u0442\u0440\u043E\u043A",
  TABLE_MORE: "\u0415\u0449\u0451",
  TABLE_MORE_DESCRIPTION: "\u0417\u0430\u0433\u0440\u0443\u0437\u0438\u0442\u044C \u0431\u043E\u043B\u044C\u0448\u0435 \u0434\u0430\u043D\u043D\u044B\u0445",
  TABLE_MULTI_SELECTABLE: "\u0422\u0430\u0431\u043B\u0438\u0446\u0430 \u0441 \u043C\u043D\u043E\u0436\u0435\u0441\u0442\u0432\u0435\u043D\u043D\u044B\u043C \u0432\u044B\u0431\u043E\u0440\u043E\u043C",
  TABLE_SINGLE_SELECTABLE: "\u0422\u0430\u0431\u043B\u0438\u0446\u0430 \u0441 \u043E\u0434\u0438\u043D\u043E\u0447\u043D\u044B\u043C \u0432\u044B\u0431\u043E\u0440\u043E\u043C",
  TABLE_NAVIGATION: "\u041D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u044F",
  TABLE_SORT_ASCENDING: "\u0421\u043E\u0440\u0442\u0438\u0440\u043E\u0432\u043A\u0430 \u043F\u043E \u0432\u043E\u0437\u0440\u0430\u0441\u0442\u0430\u043D\u0438\u044E",
  TABLE_SORT_DESCENDING: "\u0421\u043E\u0440\u0442\u0438\u0440\u043E\u0432\u043A\u0430 \u043F\u043E \u0443\u0431\u044B\u0432\u0430\u043D\u0438\u044E",
  TABLE_SORT_NONE: "\u0411\u0435\u0437 \u0441\u043E\u0440\u0442\u0438\u0440\u043E\u0432\u043A\u0438",
  TABLE_AI_SORT_PLACEHOLDER: "\u041E\u043F\u0438\u0448\u0438\u0442\u0435 \u0441\u043F\u043E\u0441\u043E\u0431 \u0441\u043E\u0440\u0442\u0438\u0440\u043E\u0432\u043A\u0438...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u041E\u043F\u0438\u0448\u0438\u0442\u0435, \u0447\u0442\u043E \u0444\u0438\u043B\u044C\u0442\u0440\u043E\u0432\u0430\u0442\u044C...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u0421\u043F\u0440\u043E\u0441\u0438\u0442\u0435 \u043E \u0432\u0430\u0448\u0438\u0445 \u0434\u0430\u043D\u043D\u044B\u0445...",
  TABLE_AI_INSIGHT_ANOMALY: "\u041E\u0431\u043D\u0430\u0440\u0443\u0436\u0435\u043D\u0430 \u0430\u043D\u043E\u043C\u0430\u043B\u0438\u044F",
  TABLE_AI_INSIGHT_TREND: "\u0412\u044B\u044F\u0432\u043B\u0435\u043D \u0442\u0440\u0435\u043D\u0434",
  TABLE_AI_INSIGHT_PATTERN: "\u041D\u0430\u0439\u0434\u0435\u043D \u043F\u0430\u0442\u0442\u0435\u0440\u043D",
  TABLE_AI_INSIGHT_DISMISS: "\u0417\u0430\u043A\u0440\u044B\u0442\u044C \u0438\u043D\u0441\u0430\u0439\u0442",
  TABLE_AI_GENERATED: "\u0421\u0433\u0435\u043D\u0435\u0440\u0438\u0440\u043E\u0432\u0430\u043D\u043E \u0418\u0418",
  TABLE_LOADING_MORE: "\u0417\u0430\u0433\u0440\u0443\u0437\u043A\u0430...",
  TABLE_TOOLBAR: "\u0414\u0435\u0439\u0441\u0442\u0432\u0438\u044F \u0442\u0430\u0431\u043B\u0438\u0446\u044B",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u041F\u0430\u043D\u0435\u043B\u044C \u0438\u043D\u0441\u0442\u0440\u0443\u043C\u0435\u043D\u0442\u043E\u0432 \u0442\u0430\u0431\u043B\u0438\u0446\u044B",
  TABLE_ROWS_SELECTED: "\u0412\u044B\u0431\u0440\u0430\u043D\u043E \u0441\u0442\u0440\u043E\u043A: {{count}}",
  TOKEN_ARIA_LABEL: "\u0422\u043E\u043A\u0435\u043D",
  TOKEN_ARIA_DELETABLE: "\u0423\u0434\u0430\u043B\u044F\u0435\u043C\u044B\u0439",
  TOKEN_DELETE: "\u0423\u0434\u0430\u043B\u0438\u0442\u044C",
  TOKENIZER_ARIA_LABEL: "\u0422\u043E\u043A\u0435\u043D\u0430\u0439\u0437\u0435\u0440",
  TOKENIZER_SHOW_MORE: "\u0415\u0449\u0451 {{count}}",
  TOKENIZER_POPOVER_HEADER: "\u0414\u0440\u0443\u0433\u0438\u0435 \u0442\u043E\u043A\u0435\u043D\u044B",
  TOKENIZER_CLEAR_ALL: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u044C \u0432\u0441\u0435",
  TOKENIZER_TOKEN_COUNT: "{{count}} \u0442\u043E\u043A\u0435\u043D\u043E\u0432",
  TOKENIZER_TOKENS_SELECTED: "{{count}} \u0442\u043E\u043A\u0435\u043D\u043E\u0432 \u0432\u044B\u0431\u0440\u0430\u043D\u043E",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} \u0440\u0435\u0437\u0443\u043B\u044C\u0442\u0430\u0442\u043E\u0432 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u043E",
  COMBOBOX_ONE_RESULT: "1 \u0440\u0435\u0437\u0443\u043B\u044C\u0442\u0430\u0442 \u0434\u043E\u0441\u0442\u0443\u043F\u0435\u043D",
  COMBOBOX_NO_RESULTS: "\u041D\u0435\u0442 \u0440\u0435\u0437\u0443\u043B\u044C\u0442\u0430\u0442\u043E\u0432",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u041F\u043E\u043B\u0435 \u0441\u043E \u0441\u043F\u0438\u0441\u043A\u043E\u043C",
  SELECT_OPTION_SELECTED: "{{text}} \u0432\u044B\u0431\u0440\u0430\u043D\u043E",
  MULTICOMBOBOX_SELECT_ALL: "\u0412\u044B\u0431\u0440\u0430\u0442\u044C \u0432\u0441\u0435",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} \u044D\u043B\u0435\u043C\u0435\u043D\u0442\u043E\u0432 \u0432\u044B\u0431\u0440\u0430\u043D\u043E",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u0412\u044B\u0431\u0435\u0440\u0438\u0442\u0435 \u043E\u043F\u0446\u0438\u044E \u0432 \u043C\u0435\u043D\u044E",
  DATEPICKER_DATE_DESCRIPTION: "\u0412\u0432\u043E\u0434 \u0434\u0430\u0442\u044B",
  DATERANGE_DESCRIPTION: "\u0412\u0432\u043E\u0434 \u0434\u0438\u0430\u043F\u0430\u0437\u043E\u043D\u0430 \u0434\u0430\u0442",
  DATEPICKER_OPEN_ICON_TITLE: "\u041E\u0442\u043A\u0440\u044B\u0442\u044C \u0432\u044B\u0431\u043E\u0440",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u0417\u0430\u043A\u0440\u044B\u0442\u044C \u0432\u044B\u0431\u043E\u0440",
  INPUT_CLEAR_ICON_ACC_NAME: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u044C",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0412\u044B\u0431\u0440\u0430\u0442\u044C \u0434\u0430\u0442\u0443 \u0434\u043B\u044F {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0412\u044B\u0431\u0440\u0430\u0442\u044C \u0434\u0438\u0430\u043F\u0430\u0437\u043E\u043D \u0434\u0430\u0442 \u0434\u043B\u044F {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u0413\u0440\u0443\u043F\u043F\u0430 \u0441\u0435\u0433\u043C\u0435\u043D\u0442\u0438\u0440\u043E\u0432\u0430\u043D\u043D\u044B\u0445 \u043A\u043D\u043E\u043F\u043E\u043A",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u0421\u0435\u0433\u043C\u0435\u043D\u0442\u0438\u0440\u043E\u0432\u0430\u043D\u043D\u0430\u044F \u043A\u043D\u043E\u043F\u043A\u0430",
  MESSAGE_STRIP_INFORMATION: "\u0421\u0442\u0440\u043E\u043A\u0430 \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F",
  MESSAGE_STRIP_SUCCESS: "\u0421\u0442\u0440\u043E\u043A\u0430 \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u043E\u0431 \u0443\u0441\u043F\u0435\u0448\u043D\u043E\u043C \u0432\u044B\u043F\u043E\u043B\u043D\u0435\u043D\u0438\u0438",
  MESSAGE_STRIP_ERROR: "\u0421\u0442\u0440\u043E\u043A\u0430 \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u043E\u0431 \u043E\u0448\u0438\u0431\u043A\u0435",
  MESSAGE_STRIP_WARNING: "\u0421\u0442\u0440\u043E\u043A\u0430 \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u0441 \u043F\u0440\u0435\u0434\u0443\u043F\u0440\u0435\u0436\u0434\u0435\u043D\u0438\u0435\u043C",
  MESSAGE_STRIP_CLOSABLE: "\u041C\u043E\u0436\u043D\u043E \u0437\u0430\u043A\u0440\u044B\u0442\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u0421\u0442\u0440\u043E\u043A\u0430 \u0438\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u043E\u043D\u043D\u043E\u0433\u043E \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u0434\u043B\u044F \u043A\u043D\u043E\u043F\u043A\u0438 \u0417\u0430\u043A\u0440\u044B\u0442\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u0421\u0442\u0440\u043E\u043A\u0430 \u043F\u043E\u0437\u0438\u0442\u0438\u0432\u043D\u043E\u0433\u043E \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u0434\u043B\u044F \u043A\u043D\u043E\u043F\u043A\u0438 \u0417\u0430\u043A\u0440\u044B\u0442\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u0421\u0442\u0440\u043E\u043A\u0430 \u043D\u0435\u0433\u0430\u0442\u0438\u0432\u043D\u043E\u0433\u043E \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u0434\u043B\u044F \u043A\u043D\u043E\u043F\u043A\u0438 \u0417\u0430\u043A\u0440\u044B\u0442\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u0421\u0442\u0440\u043E\u043A\u0430 \u043A\u0440\u0438\u0442\u0438\u0447\u0435\u0441\u043A\u043E\u0433\u043E \u0441\u043E\u043E\u0431\u0449\u0435\u043D\u0438\u044F \u0434\u043B\u044F \u043A\u043D\u043E\u043F\u043A\u0438 \u0417\u0430\u043A\u0440\u044B\u0442\u044C",
  FILEUPLOADER_PLACEHOLDER: "\u0412\u044B\u0431\u0435\u0440\u0438\u0442\u0435 \u0438\u043B\u0438 \u043F\u0435\u0440\u0435\u0442\u0430\u0449\u0438\u0442\u0435 \u0444\u0430\u0439\u043B",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u0412\u044B\u0431\u0435\u0440\u0438\u0442\u0435 \u0438\u043B\u0438 \u043F\u0435\u0440\u0435\u0442\u0430\u0449\u0438\u0442\u0435 \u0444\u0430\u0439\u043B\u044B",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u041E\u0431\u0437\u043E\u0440 \u0444\u0430\u0439\u043B\u043E\u0432",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u044C \u0432\u0441\u0435",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u0417\u0430\u0433\u0440\u0443\u0437\u043A\u0430 \u0444\u0430\u0439\u043B\u0430",
  SEARCHFIELD_SEARCH: "\u041F\u043E\u0438\u0441\u043A",
  SEARCHFIELD_CLEAR: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u044C",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u0442\u0435\u043A\u0441\u0442 \u0434\u043B\u044F \u043D\u0430\u0447\u0430\u043B\u0430 \u043F\u043E\u0438\u0441\u043A\u0430",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u041F\u043E\u043B\u0435 \u043F\u043E\u0438\u0441\u043A\u0430",
  NAVIGATION_EXPAND: "\u0420\u0430\u0437\u0432\u0435\u0440\u043D\u0443\u0442\u044C",
  NAVIGATION_COLLAPSE: "\u0421\u0432\u0435\u0440\u043D\u0443\u0442\u044C",
  SIGN_OUT: "\u0412\u044B\u0439\u0442\u0438",
  FX_START_PANE: "\u041D\u0430\u0447\u0430\u043B\u044C\u043D\u0430\u044F \u043F\u0430\u043D\u0435\u043B\u044C",
  FX_END_PANE: "\u041A\u043E\u043D\u0435\u0447\u043D\u0430\u044F \u043F\u0430\u043D\u0435\u043B\u044C",
  FX_RESIZE_START_CENTER: "\u0418\u0437\u043C\u0435\u043D\u0438\u0442\u044C \u0440\u0430\u0437\u043C\u0435\u0440 \u043D\u0430\u0447\u0430\u043B\u044C\u043D\u043E\u0439 \u0438 \u0446\u0435\u043D\u0442\u0440\u0430\u043B\u044C\u043D\u043E\u0439 \u043F\u0430\u043D\u0435\u043B\u0435\u0439",
  FX_RESIZE_CENTER_END: "\u0418\u0437\u043C\u0435\u043D\u0438\u0442\u044C \u0440\u0430\u0437\u043C\u0435\u0440 \u0446\u0435\u043D\u0442\u0440\u0430\u043B\u044C\u043D\u043E\u0439 \u0438 \u043A\u043E\u043D\u0435\u0447\u043D\u043E\u0439 \u043F\u0430\u043D\u0435\u043B\u0435\u0439",
  FX_NOTIFICATIONS: "\u0423\u0432\u0435\u0434\u043E\u043C\u043B\u0435\u043D\u0438\u044F",
  FX_OPEN_NAVIGATION: "\u041E\u0442\u043A\u0440\u044B\u0442\u044C \u043D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u044E",
  FX_BACK_TO_LIST: "\u0412\u0435\u0440\u043D\u0443\u0442\u044C\u0441\u044F \u043A \u0441\u043F\u0438\u0441\u043A\u0443",
  FX_BACK: "\u041D\u0430\u0437\u0430\u0434",
  FX_RELATED: "\u0421\u0432\u044F\u0437\u0430\u043D\u043D\u044B\u0435",
  FX_MORE_ACTIONS: "\u0414\u043E\u043F\u043E\u043B\u043D\u0438\u0442\u0435\u043B\u044C\u043D\u044B\u0435 \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u044F",
  FX_CLOSE: "\u0417\u0430\u043A\u0440\u044B\u0442\u044C",
  FX_ADD: "\u0414\u043E\u0431\u0430\u0432\u0438\u0442\u044C",
  FX_SEND: "\u041E\u0442\u043F\u0440\u0430\u0432\u0438\u0442\u044C",
  FX_DICTATE: "\u0414\u0438\u043A\u0442\u043E\u0432\u0430\u0442\u044C",
  FX_VOICE: "\u0413\u043E\u043B\u043E\u0441",
  FX_DOUBLE_CLICK_TO_RENAME: "\u0414\u0432\u0430\u0436\u0434\u044B \u0449\u0451\u043B\u043A\u043D\u0438\u0442\u0435 \u0434\u043B\u044F \u043F\u0435\u0440\u0435\u0438\u043C\u0435\u043D\u043E\u0432\u0430\u043D\u0438\u044F",
  FX_CANCEL: "\u041E\u0442\u043C\u0435\u043D\u0430",
  FX_ACCEPT: "\u041F\u0440\u0438\u043D\u044F\u0442\u044C",
  FX_MAIN_NAVIGATION: "\u041E\u0441\u043D\u043E\u0432\u043D\u044B\u0435 \u044D\u043B\u0435\u043C\u0435\u043D\u0442\u044B \u043D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u0438",
  FX_USER_ACTIONS: "\u0414\u0435\u0439\u0441\u0442\u0432\u0438\u044F \u043F\u043E\u043B\u044C\u0437\u043E\u0432\u0430\u0442\u0435\u043B\u044F",
  FX_CURRENT_ACCOUNT: "\u0418\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u044F \u043E \u0442\u0435\u043A\u0443\u0449\u0435\u0439 \u0443\u0447\u0451\u0442\u043D\u043E\u0439 \u0437\u0430\u043F\u0438\u0441\u0438",
  FX_EDIT_ACCOUNTS: "\u0420\u0435\u0434\u0430\u043A\u0442\u0438\u0440\u043E\u0432\u0430\u0442\u044C \u0443\u0447\u0451\u0442\u043D\u044B\u0435 \u0437\u0430\u043F\u0438\u0441\u0438",
  LINK_SUBTLE: "\u041D\u0435\u0437\u0430\u043C\u0435\u0442\u043D\u0430\u044F",
  LINK_EMPHASIZED: "\u0412\u044B\u0434\u0435\u043B\u0435\u043D\u043D\u0430\u044F",
  SPLIT_BUTTON_DESCRIPTION: "\u0420\u0430\u0437\u0434\u0435\u043B\u0435\u043D\u043D\u0430\u044F \u043D\u0430 \u0434\u0432\u0435 \u0447\u0430\u0441\u0442\u0438 \u043A\u043D\u043E\u043F\u043A\u0430",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u041D\u0430\u0436\u043C\u0438\u0442\u0435 \u043F\u0440\u043E\u0431\u0435\u043B \u0438\u043B\u0438 Enter, \u0447\u0442\u043E\u0431\u044B \u0437\u0430\u043F\u0443\u0441\u0442\u0438\u0442\u044C \u043E\u043F\u0435\u0440\u0430\u0446\u0438\u044E \u043F\u043E \u0443\u043C\u043E\u043B\u0447\u0430\u043D\u0438\u044E, \u0438 Alt + \u043A\u043B\u0430\u0432\u0438\u0448\u0443 \u0441\u043E \u0441\u0442\u0440\u0435\u043B\u043A\u043E\u0439 \u0432\u043D\u0438\u0437 \u0438\u043B\u0438 F4, \u0447\u0442\u043E\u0431\u044B \u0437\u0430\u043F\u0443\u0441\u0442\u0438\u0442\u044C \u043E\u043F\u0435\u0440\u0430\u0446\u0438\u044E \u0441\u0442\u0440\u0435\u043B\u043A\u0438",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u041E\u0442\u043A\u0440\u044B\u0442\u044C \u043C\u0435\u043D\u044E"
};

// src/i18n/locales/uk.json
var uk_default = {
  INPUT_CLEAR: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u0438",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} \u043F\u0440\u043E\u043F\u043E\u0437\u0438\u0446\u0456\u0439 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u043E",
  INPUT_SUGGESTIONS_ONE: "1 \u043F\u0440\u043E\u043F\u043E\u0437\u0438\u0446\u0456\u044F \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u0430",
  INPUT_SUGGESTIONS_NONE: "\u041D\u0435\u043C\u0430\u0454 \u043F\u0440\u043E\u043F\u043E\u0437\u0438\u0446\u0456\u0439",
  INPUT_SHOW_PASSWORD: "\u041F\u043E\u043A\u0430\u0437\u0430\u0442\u0438 \u043F\u0430\u0440\u043E\u043B\u044C",
  INPUT_HIDE_PASSWORD: "\u041F\u0440\u0438\u0445\u043E\u0432\u0430\u0442\u0438 \u043F\u0430\u0440\u043E\u043B\u044C",
  VALUE_STATE_ERROR: "\u041F\u043E\u043C\u0438\u043B\u043A\u0430",
  VALUE_STATE_WARNING: "\u041F\u043E\u043F\u0435\u0440\u0435\u0434\u0436\u0435\u043D\u043D\u044F",
  VALUE_STATE_SUCCESS: "\u0423\u0441\u043F\u0456\u0448\u043D\u043E",
  VALUE_STATE_INFORMATION: "\u0406\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0456\u044F",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u041D\u0435\u0434\u0456\u0439\u0441\u043D\u0438\u0439 \u0437\u0430\u043F\u0438\u0441",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u0417\u0430\u0441\u0442\u0435\u0440\u0435\u0436\u0435\u043D\u043D\u044F \u0432\u0438\u0434\u0430\u043D\u043E",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u0417\u0430\u043F\u0438\u0441 \u0443\u0441\u043F\u0456\u0448\u043D\u043E \u043F\u0435\u0440\u0435\u0432\u0456\u0440\u0435\u043D\u043E",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u0406\u043D\u0444\u043E\u0440\u043C\u0430\u0442\u0438\u0432\u043D\u0438\u0439 \u0437\u0430\u043F\u0438\u0441",
  REQUIRED_FIELD: "\u041E\u0431\u043E\u0432'\u044F\u0437\u043A\u043E\u0432\u0435 \u043F\u043E\u043B\u0435",
  LOADING: "\u0417\u0430\u0432\u0430\u043D\u0442\u0430\u0436\u0435\u043D\u043D\u044F...",
  NO_DATA: "\u041D\u0435\u043C\u0430\u0454 \u0434\u0430\u043D\u0438\u0445",
  TABLE_NO_DATA: "\u0414\u0430\u043D\u0456 \u0432\u0456\u0434\u0441\u0443\u0442\u043D\u0456",
  TABLE_ROW: "\u0420\u044F\u0434\u043E\u043A",
  TABLE_ROW_INDEX: "\u0420\u044F\u0434\u043E\u043A {{index}} \u0437 {{total}}",
  TABLE_ROW_SELECTED: "\u0412\u0438\u0431\u0440\u0430\u043D\u043E",
  TABLE_ROW_ACTIVE: "\u0410\u043A\u0442\u0438\u0432\u043D\u0438\u0439",
  TABLE_ROW_NAVIGABLE: "\u0414\u043E\u0441\u0442\u0443\u043F\u043D\u0430 \u043D\u0430\u0432\u0456\u0433\u0430\u0446\u0456\u044F",
  TABLE_ROW_NAVIGATED: "\u0412\u0456\u0434\u0432\u0456\u0434\u0430\u043D\u043E",
  TABLE_ROW_SELECTOR: "\u0412\u0438\u0431\u0456\u0440 \u0440\u044F\u0434\u043A\u0430",
  TABLE_ROW_SINGLE_ACTION: "{{count}} \u0434\u0456\u044F",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} \u0434\u0456\u0439",
  TABLE_ROW_POPIN: "\u0414\u043E\u0434\u0430\u0442\u043A\u043E\u0432\u0430 \u0456\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0456\u044F",
  TABLE_ROW_ACTIONS: "\u0414\u0456\u0457 \u0440\u044F\u0434\u043A\u0430",
  TABLE_COLUMN_HEADER_ROW: "\u0420\u044F\u0434\u043E\u043A \u0437\u0430\u0433\u043E\u043B\u043E\u0432\u043A\u0456\u0432 \u0441\u0442\u043E\u0432\u043F\u0446\u0456\u0432",
  TABLE_SELECTION: "\u0412\u0438\u0431\u0456\u0440",
  TABLE_SELECT_ALL_ROWS: "\u0412\u0438\u0431\u0440\u0430\u0442\u0438 \u0432\u0441\u0456 \u0440\u044F\u0434\u043A\u0438",
  TABLE_DESELECT_ALL_ROWS: "\u0417\u043D\u044F\u0442\u0438 \u0432\u0438\u0431\u0456\u0440 \u0437 \u0443\u0441\u0456\u0445 \u0440\u044F\u0434\u043A\u0456\u0432",
  TABLE_MORE: "\u0429\u0435",
  TABLE_MORE_DESCRIPTION: "\u0417\u0430\u0432\u0430\u043D\u0442\u0430\u0436\u0438\u0442\u0438 \u0431\u0456\u043B\u044C\u0448\u0435 \u0434\u0430\u043D\u0438\u0445",
  TABLE_MULTI_SELECTABLE: "\u0422\u0430\u0431\u043B\u0438\u0446\u044F \u0437 \u043C\u043D\u043E\u0436\u0438\u043D\u043D\u0438\u043C \u0432\u0438\u0431\u043E\u0440\u043E\u043C",
  TABLE_SINGLE_SELECTABLE: "\u0422\u0430\u0431\u043B\u0438\u0446\u044F \u0437 \u043E\u0434\u0438\u043D\u0438\u0447\u043D\u0438\u043C \u0432\u0438\u0431\u043E\u0440\u043E\u043C",
  TABLE_NAVIGATION: "\u041D\u0430\u0432\u0456\u0433\u0430\u0446\u0456\u044F",
  TABLE_SORT_ASCENDING: "\u0421\u043E\u0440\u0442\u0443\u0432\u0430\u043D\u043D\u044F \u0437\u0430 \u0437\u0440\u043E\u0441\u0442\u0430\u043D\u043D\u044F\u043C",
  TABLE_SORT_DESCENDING: "\u0421\u043E\u0440\u0442\u0443\u0432\u0430\u043D\u043D\u044F \u0437\u0430 \u0441\u043F\u0430\u0434\u0430\u043D\u043D\u044F\u043C",
  TABLE_SORT_NONE: "\u0411\u0435\u0437 \u0441\u043E\u0440\u0442\u0443\u0432\u0430\u043D\u043D\u044F",
  TABLE_AI_SORT_PLACEHOLDER: "\u041E\u043F\u0438\u0448\u0456\u0442\u044C \u0441\u043F\u043E\u0441\u0456\u0431 \u0441\u043E\u0440\u0442\u0443\u0432\u0430\u043D\u043D\u044F...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u041E\u043F\u0438\u0448\u0456\u0442\u044C, \u0449\u043E \u0444\u0456\u043B\u044C\u0442\u0440\u0443\u0432\u0430\u0442\u0438...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u0417\u0430\u043F\u0438\u0442\u0430\u0439\u0442\u0435 \u043F\u0440\u043E \u0432\u0430\u0448\u0456 \u0434\u0430\u043D\u0456...",
  TABLE_AI_INSIGHT_ANOMALY: "\u0412\u0438\u044F\u0432\u043B\u0435\u043D\u043E \u0430\u043D\u043E\u043C\u0430\u043B\u0456\u044E",
  TABLE_AI_INSIGHT_TREND: "\u0412\u0438\u044F\u0432\u043B\u0435\u043D\u043E \u0442\u0440\u0435\u043D\u0434",
  TABLE_AI_INSIGHT_PATTERN: "\u0417\u043D\u0430\u0439\u0434\u0435\u043D\u043E \u043F\u0430\u0442\u0435\u0440\u043D",
  TABLE_AI_INSIGHT_DISMISS: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438 \u0456\u043D\u0441\u0430\u0439\u0442",
  TABLE_AI_GENERATED: "\u0417\u0433\u0435\u043D\u0435\u0440\u043E\u0432\u0430\u043D\u043E \u0428\u0406",
  TABLE_LOADING_MORE: "\u0417\u0430\u0432\u0430\u043D\u0442\u0430\u0436\u0435\u043D\u043D\u044F...",
  TABLE_TOOLBAR: "\u0414\u0456\u0457 \u0442\u0430\u0431\u043B\u0438\u0446\u0456",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u041F\u0430\u043D\u0435\u043B\u044C \u0456\u043D\u0441\u0442\u0440\u0443\u043C\u0435\u043D\u0442\u0456\u0432 \u0442\u0430\u0431\u043B\u0438\u0446\u0456",
  TABLE_ROWS_SELECTED: "\u0412\u0438\u0431\u0440\u0430\u043D\u043E \u0440\u044F\u0434\u043A\u0456\u0432: {{count}}",
  TOKEN_ARIA_LABEL: "\u0422\u043E\u043A\u0435\u043D",
  TOKEN_ARIA_DELETABLE: "\u0412\u0438\u0434\u0430\u043B\u044F\u0454\u0442\u044C\u0441\u044F",
  TOKEN_DELETE: "\u0412\u0438\u0434\u0430\u043B\u0438\u0442\u0438",
  TOKENIZER_ARIA_LABEL: "\u0422\u043E\u043A\u0435\u043D\u0430\u0439\u0437\u0435\u0440",
  TOKENIZER_SHOW_MORE: "\u0429\u0435 {{count}}",
  TOKENIZER_POPOVER_HEADER: "\u0406\u043D\u0448\u0456 \u0442\u043E\u043A\u0435\u043D\u0438",
  TOKENIZER_CLEAR_ALL: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u0438 \u0432\u0441\u0435",
  TOKENIZER_TOKEN_COUNT: "{{count}} \u0442\u043E\u043A\u0435\u043D\u0456\u0432",
  TOKENIZER_TOKENS_SELECTED: "{{count}} \u0442\u043E\u043A\u0435\u043D\u0456\u0432 \u0432\u0438\u0431\u0440\u0430\u043D\u043E",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} \u0440\u0435\u0437\u0443\u043B\u044C\u0442\u0430\u0442\u0456\u0432 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u043E",
  COMBOBOX_ONE_RESULT: "1 \u0440\u0435\u0437\u0443\u043B\u044C\u0442\u0430\u0442 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u0438\u0439",
  COMBOBOX_NO_RESULTS: "\u041D\u0435\u043C\u0430\u0454 \u0440\u0435\u0437\u0443\u043B\u044C\u0442\u0430\u0442\u0456\u0432",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u0421\u043F\u0438\u0441\u043E\u043A",
  SELECT_OPTION_SELECTED: "{{text}} \u0432\u0438\u0431\u0440\u0430\u043D\u043E",
  MULTICOMBOBOX_SELECT_ALL: "\u0412\u0438\u0431\u0440\u0430\u0442\u0438 \u0432\u0441\u0435",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} \u0435\u043B\u0435\u043C\u0435\u043D\u0442\u0456\u0432 \u0432\u0438\u0431\u0440\u0430\u043D\u043E",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u0412\u0438\u0431\u0435\u0440\u0456\u0442\u044C \u043E\u043F\u0446\u0456\u044E \u0437 \u043C\u0435\u043D\u044E",
  DATEPICKER_DATE_DESCRIPTION: "\u0412\u0432\u0435\u0434\u0435\u043D\u043D\u044F \u0434\u0430\u0442\u0438",
  DATERANGE_DESCRIPTION: "\u0412\u0432\u0435\u0434\u0435\u043D\u043D\u044F \u0434\u0456\u0430\u043F\u0430\u0437\u043E\u043D\u0443 \u0434\u0430\u0442",
  DATEPICKER_OPEN_ICON_TITLE: "\u0412\u0456\u0434\u043A\u0440\u0438\u0442\u0438 \u0437\u0430\u0441\u0456\u0431 \u0432\u0438\u0431\u043E\u0440\u0443",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438 \u0437\u0430\u0441\u0456\u0431 \u0432\u0438\u0431\u043E\u0440\u0443",
  INPUT_CLEAR_ICON_ACC_NAME: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u0438",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0412\u0438\u0431\u0440\u0430\u0442\u0438 \u0434\u0430\u0442\u0443 \u0434\u043B\u044F {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0412\u0438\u0431\u0440\u0430\u0442\u0438 \u0434\u0456\u0430\u043F\u0430\u0437\u043E\u043D \u0434\u0430\u0442 \u0434\u043B\u044F {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u0413\u0440\u0443\u043F\u0430 \u0441\u0435\u0433\u043C\u0435\u043D\u0442\u043E\u0432\u0430\u043D\u0438\u0445 \u043A\u043D\u043E\u043F\u043E\u043A",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u0421\u0435\u0433\u043C\u0435\u043D\u0442\u043E\u0432\u0430\u043D\u0430 \u043A\u043D\u043E\u043F\u043A\u0430",
  MESSAGE_STRIP_INFORMATION: "\u0420\u044F\u0434\u043E\u043A \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C",
  MESSAGE_STRIP_SUCCESS: "\u0420\u044F\u0434\u043E\u043A \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C \u043F\u0440\u043E \u0443\u0441\u043F\u0456\u0448\u043D\u0435 \u0432\u0438\u043A\u043E\u043D\u0430\u043D\u043D\u044F",
  MESSAGE_STRIP_ERROR: "\u0420\u044F\u0434\u043E\u043A \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C \u043F\u0440\u043E \u043F\u043E\u043C\u0438\u043B\u043A\u0438",
  MESSAGE_STRIP_WARNING: "\u0420\u044F\u0434\u043E\u043A \u043F\u043E\u043F\u0435\u0440\u0435\u0434\u0436\u0443\u0432\u0430\u043B\u044C\u043D\u0438\u0445 \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C",
  MESSAGE_STRIP_CLOSABLE: "\u0417 \u043C\u043E\u0436\u043B\u0438\u0432\u0456\u0441\u0442\u044E \u0437\u0430\u043A\u0440\u0438\u0442\u0442\u044F",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438 \u043F\u0430\u043D\u0435\u043B\u044C \u0456\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0456\u0439\u043D\u0438\u0445 \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438 \u043F\u0430\u043D\u0435\u043B\u044C \u043F\u043E\u0437\u0438\u0442\u0438\u0432\u043D\u0438\u0445 \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438 \u043F\u0430\u043D\u0435\u043B\u044C \u043D\u0435\u0433\u0430\u0442\u0438\u0432\u043D\u0438\u0445 \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438 \u043F\u0430\u043D\u0435\u043B\u044C \u043A\u0440\u0438\u0442\u0438\u0447\u043D\u0438\u0445 \u043F\u043E\u0432\u0456\u0434\u043E\u043C\u043B\u0435\u043D\u044C",
  FILEUPLOADER_PLACEHOLDER: "\u041E\u0431\u0435\u0440\u0456\u0442\u044C \u0430\u0431\u043E \u043F\u0435\u0440\u0435\u0442\u044F\u0433\u043D\u0456\u0442\u044C \u0444\u0430\u0439\u043B",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u041E\u0431\u0435\u0440\u0456\u0442\u044C \u0430\u0431\u043E \u043F\u0435\u0440\u0435\u0442\u044F\u0433\u043D\u0456\u0442\u044C \u0444\u0430\u0439\u043B\u0438",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u041F\u0435\u0440\u0435\u0433\u043B\u044F\u0434 \u0444\u0430\u0439\u043B\u0456\u0432",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u0438 \u0432\u0441\u0435",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u0417\u0430\u0432\u0430\u043D\u0442\u0430\u0436\u0435\u043D\u043D\u044F \u0444\u0430\u0439\u043B\u0443",
  SEARCHFIELD_SEARCH: "\u041F\u043E\u0448\u0443\u043A",
  SEARCHFIELD_CLEAR: "\u041E\u0447\u0438\u0441\u0442\u0438\u0442\u0438",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u0412\u0432\u0435\u0434\u0456\u0442\u044C \u0442\u0435\u043A\u0441\u0442 \u0434\u043B\u044F \u043F\u043E\u0447\u0430\u0442\u043A\u0443 \u043F\u043E\u0448\u0443\u043A\u0443",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u041F\u043E\u043B\u0435 \u043F\u043E\u0448\u0443\u043A\u0443",
  NAVIGATION_EXPAND: "\u0420\u043E\u0437\u0433\u043E\u0440\u043D\u0443\u0442\u0438",
  NAVIGATION_COLLAPSE: "\u0417\u0433\u043E\u0440\u043D\u0443\u0442\u0438",
  SIGN_OUT: "\u0412\u0438\u0439\u0442\u0438",
  FX_START_PANE: "\u041F\u043E\u0447\u0430\u0442\u043A\u043E\u0432\u0430 \u043F\u0430\u043D\u0435\u043B\u044C",
  FX_END_PANE: "\u041A\u0456\u043D\u0446\u0435\u0432\u0430 \u043F\u0430\u043D\u0435\u043B\u044C",
  FX_RESIZE_START_CENTER: "\u0417\u043C\u0456\u043D\u0438\u0442\u0438 \u0440\u043E\u0437\u043C\u0456\u0440 \u043F\u043E\u0447\u0430\u0442\u043A\u043E\u0432\u043E\u0457 \u0442\u0430 \u0446\u0435\u043D\u0442\u0440\u0430\u043B\u044C\u043D\u043E\u0457 \u043F\u0430\u043D\u0435\u043B\u0435\u0439",
  FX_RESIZE_CENTER_END: "\u0417\u043C\u0456\u043D\u0438\u0442\u0438 \u0440\u043E\u0437\u043C\u0456\u0440 \u0446\u0435\u043D\u0442\u0440\u0430\u043B\u044C\u043D\u043E\u0457 \u0442\u0430 \u043A\u0456\u043D\u0446\u0435\u0432\u043E\u0457 \u043F\u0430\u043D\u0435\u043B\u0435\u0439",
  FX_NOTIFICATIONS: "\u0421\u043F\u043E\u0432\u0456\u0449\u0435\u043D\u043D\u044F",
  FX_OPEN_NAVIGATION: "\u0412\u0456\u0434\u043A\u0440\u0438\u0442\u0438 \u043D\u0430\u0432\u0456\u0433\u0430\u0446\u0456\u044E",
  FX_BACK_TO_LIST: "\u041F\u043E\u0432\u0435\u0440\u043D\u0443\u0442\u0438\u0441\u044F \u0434\u043E \u0441\u043F\u0438\u0441\u043A\u0443",
  FX_BACK: "\u041D\u0430\u0437\u0430\u0434",
  FX_RELATED: "\u041F\u043E\u0432'\u044F\u0437\u0430\u043D\u0456",
  FX_MORE_ACTIONS: "\u0414\u043E\u0434\u0430\u0442\u043A\u043E\u0432\u0456 \u0434\u0456\u0457",
  FX_CLOSE: "\u0417\u0430\u043A\u0440\u0438\u0442\u0438",
  FX_ADD: "\u0414\u043E\u0434\u0430\u0442\u0438",
  FX_SEND: "\u041D\u0430\u0434\u0456\u0441\u043B\u0430\u0442\u0438",
  FX_DICTATE: "\u0414\u0438\u043A\u0442\u0443\u0432\u0430\u0442\u0438",
  FX_VOICE: "\u0413\u043E\u043B\u043E\u0441",
  FX_DOUBLE_CLICK_TO_RENAME: "\u0414\u0432\u0456\u0447\u0456 \u043A\u043B\u0430\u0446\u043D\u0456\u0442\u044C \u0434\u043B\u044F \u043F\u0435\u0440\u0435\u0439\u043C\u0435\u043D\u0443\u0432\u0430\u043D\u043D\u044F",
  FX_CANCEL: "\u0421\u043A\u0430\u0441\u0443\u0432\u0430\u0442\u0438",
  FX_ACCEPT: "\u041F\u0440\u0438\u0439\u043D\u044F\u0442\u0438",
  FX_MAIN_NAVIGATION: "\u041E\u0441\u043D\u043E\u0432\u043D\u0456 \u0435\u043B\u0435\u043C\u0435\u043D\u0442\u0438 \u043D\u0430\u0432\u0456\u0433\u0430\u0446\u0456\u0457",
  FX_USER_ACTIONS: "\u0414\u0456\u0457 \u043A\u043E\u0440\u0438\u0441\u0442\u0443\u0432\u0430\u0447\u0430",
  FX_CURRENT_ACCOUNT: "\u0406\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0456\u044F \u043F\u0440\u043E \u043F\u043E\u0442\u043E\u0447\u043D\u0438\u0439 \u043E\u0431\u043B\u0456\u043A\u043E\u0432\u0438\u0439 \u0437\u0430\u043F\u0438\u0441",
  FX_EDIT_ACCOUNTS: "\u0420\u0435\u0434\u0430\u0433\u0443\u0432\u0430\u0442\u0438 \u043E\u0431\u043B\u0456\u043A\u043E\u0432\u0456 \u0437\u0430\u043F\u0438\u0441\u0438",
  LINK_SUBTLE: "\u041F\u0440\u0438\u0445\u043E\u0432\u0430\u043D\u0438\u0439",
  LINK_EMPHASIZED: "\u0412\u0438\u0434\u0456\u043B\u0435\u043D\u043E",
  SPLIT_BUTTON_DESCRIPTION: "\u041A\u043D\u043E\u043F\u043A\u0430 \u0440\u043E\u0437\u0434\u0456\u043B\u0435\u043D\u043D\u044F",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u041D\u0430\u0442\u0438\u0441\u043D\u0456\u0442\u044C \u043A\u043B\u0430\u0432\u0456\u0448\u0443 \u043F\u0440\u043E\u0431\u0456\u043B\u0443 \u0430\u0431\u043E Enter \u0434\u043B\u044F \u0456\u043D\u0456\u0446\u0456\u044E\u0432\u0430\u043D\u043D\u044F \u0443\u0441\u0442\u0430\u043B\u0435\u043D\u043E\u0457 \u043E\u043F\u0435\u0440\u0430\u0446\u0456\u0457, \u0442\u0430 Alt + \u0441\u0442\u0440\u0456\u043B\u043A\u0430 \u0432\u043D\u0438\u0437 \u0430\u0431\u043E F4 \u0434\u043B\u044F \u0456\u043D\u0456\u0446\u0456\u044E\u0432\u0430\u043D\u043D\u044F \u043E\u043F\u0435\u0440\u0430\u0446\u0456\u0457 \u0441\u0442\u0440\u0456\u043B\u043A\u0438",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u0412\u0456\u0434\u043A\u0440\u0438\u0442\u0438 \u043C\u0435\u043D\u044E"
};

// src/i18n/locales/bg.json
var bg_default = {
  INPUT_CLEAR: "\u0418\u0437\u0447\u0438\u0441\u0442\u0438",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} \u043F\u0440\u0435\u0434\u043B\u043E\u0436\u0435\u043D\u0438\u044F \u043D\u0430\u043B\u0438\u0447\u043D\u0438",
  INPUT_SUGGESTIONS_ONE: "1 \u043F\u0440\u0435\u0434\u043B\u043E\u0436\u0435\u043D\u0438\u0435 \u043D\u0430\u043B\u0438\u0447\u043D\u043E",
  INPUT_SUGGESTIONS_NONE: "\u041D\u044F\u043C\u0430 \u043F\u0440\u0435\u0434\u043B\u043E\u0436\u0435\u043D\u0438\u044F",
  INPUT_SHOW_PASSWORD: "\u041F\u043E\u043A\u0430\u0436\u0438 \u043F\u0430\u0440\u043E\u043B\u0430",
  INPUT_HIDE_PASSWORD: "\u0421\u043A\u0440\u0438\u0439 \u043F\u0430\u0440\u043E\u043B\u0430",
  VALUE_STATE_ERROR: "\u0413\u0440\u0435\u0448\u043A\u0430",
  VALUE_STATE_WARNING: "\u041F\u0440\u0435\u0434\u0443\u043F\u0440\u0435\u0436\u0434\u0435\u043D\u0438\u0435",
  VALUE_STATE_SUCCESS: "\u0423\u0441\u043F\u0435\u0445",
  VALUE_STATE_INFORMATION: "\u0418\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u044F",
  RADIOBUTTON_VALUE_STATE_ERROR: "\u041D\u0435\u0432\u0430\u043B\u0438\u0434\u0435\u043D \u0437\u0430\u043F\u0438\u0441",
  RADIOBUTTON_VALUE_STATE_WARNING: "\u0418\u0437\u0432\u0435\u0434\u0435\u043D\u043E \u043F\u0440\u0435\u0434\u0443\u043F\u0440\u0435\u0436\u0434\u0435\u043D\u0438\u0435",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "\u0417\u0430\u043F\u0438\u0441\u044A\u0442 \u0435 \u0443\u0441\u043F\u0435\u0448\u043D\u043E \u043F\u0440\u043E\u0432\u0435\u0440\u0435\u043D",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "\u0418\u043D\u0444\u043E\u0440\u043C\u0430\u0442\u0438\u0432\u0435\u043D \u0437\u0430\u043F\u0438\u0441",
  REQUIRED_FIELD: "\u0417\u0430\u0434\u044A\u043B\u0436\u0438\u0442\u0435\u043B\u043D\u043E \u043F\u043E\u043B\u0435",
  LOADING: "\u0417\u0430\u0440\u0435\u0436\u0434\u0430\u043D\u0435...",
  NO_DATA: "\u041D\u044F\u043C\u0430 \u0434\u0430\u043D\u043D\u0438",
  TABLE_NO_DATA: "\u041D\u044F\u043C\u0430 \u043D\u0430\u043B\u0438\u0447\u043D\u0438 \u0434\u0430\u043D\u043D\u0438",
  TABLE_ROW: "\u0420\u0435\u0434",
  TABLE_ROW_INDEX: "\u0420\u0435\u0434 {{index}} \u043E\u0442 {{total}}",
  TABLE_ROW_SELECTED: "\u0418\u0437\u0431\u0440\u0430\u043D",
  TABLE_ROW_ACTIVE: "\u0410\u043A\u0442\u0438\u0432\u0435\u043D",
  TABLE_ROW_NAVIGABLE: "\u041D\u0430\u0432\u0438\u0433\u0438\u0440\u0443\u0435\u043C",
  TABLE_ROW_NAVIGATED: "\u041D\u0430\u0432\u0438\u0433\u0438\u0440\u0430\u043D",
  TABLE_ROW_SELECTOR: "\u0421\u0435\u043B\u0435\u043A\u0442\u043E\u0440 \u043D\u0430 \u0440\u0435\u0434",
  TABLE_ROW_SINGLE_ACTION: "{{count}} \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u0435",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u044F",
  TABLE_ROW_POPIN: "\u0414\u043E\u043F\u044A\u043B\u043D\u0438\u0442\u0435\u043B\u043D\u0430 \u0438\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u044F",
  TABLE_ROW_ACTIONS: "\u0414\u0435\u0439\u0441\u0442\u0432\u0438\u044F \u0437\u0430 \u0440\u0435\u0434",
  TABLE_COLUMN_HEADER_ROW: "\u0420\u0435\u0434 \u0441\u044A\u0441 \u0437\u0430\u0433\u043B\u0430\u0432\u0438\u044F \u043D\u0430 \u043A\u043E\u043B\u043E\u043D\u0438",
  TABLE_SELECTION: "\u0418\u0437\u0431\u043E\u0440",
  TABLE_SELECT_ALL_ROWS: "\u0418\u0437\u0431\u0435\u0440\u0438 \u0432\u0441\u0438\u0447\u043A\u0438 \u0440\u0435\u0434\u043E\u0432\u0435",
  TABLE_DESELECT_ALL_ROWS: "\u041F\u0440\u0435\u043C\u0430\u0445\u043D\u0438 \u0438\u0437\u0431\u043E\u0440\u0430 \u043E\u0442 \u0432\u0441\u0438\u0447\u043A\u0438 \u0440\u0435\u0434\u043E\u0432\u0435",
  TABLE_MORE: "\u041E\u0449\u0435",
  TABLE_MORE_DESCRIPTION: "\u0417\u0430\u0440\u0435\u0434\u0438 \u043F\u043E\u0432\u0435\u0447\u0435 \u0434\u0430\u043D\u043D\u0438",
  TABLE_MULTI_SELECTABLE: "\u0422\u0430\u0431\u043B\u0438\u0446\u0430 \u0441 \u043C\u043D\u043E\u0436\u0435\u0441\u0442\u0432\u0435\u043D \u0438\u0437\u0431\u043E\u0440",
  TABLE_SINGLE_SELECTABLE: "\u0422\u0430\u0431\u043B\u0438\u0446\u0430 \u0441 \u0435\u0434\u0438\u043D\u0438\u0447\u0435\u043D \u0438\u0437\u0431\u043E\u0440",
  TABLE_NAVIGATION: "\u041D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u044F",
  TABLE_SORT_ASCENDING: "\u0421\u043E\u0440\u0442\u0438\u0440\u0430\u043D\u043E \u0432\u044A\u0437\u0445\u043E\u0434\u044F\u0449\u043E",
  TABLE_SORT_DESCENDING: "\u0421\u043E\u0440\u0442\u0438\u0440\u0430\u043D\u043E \u043D\u0438\u0437\u0445\u043E\u0434\u044F\u0449\u043E",
  TABLE_SORT_NONE: "\u0411\u0435\u0437 \u0441\u043E\u0440\u0442\u0438\u0440\u0430\u043D\u0435",
  TABLE_AI_SORT_PLACEHOLDER: "\u041E\u043F\u0438\u0448\u0435\u0442\u0435 \u043A\u0430\u043A \u0434\u0430 \u0441\u043E\u0440\u0442\u0438\u0440\u0430\u0442\u0435...",
  TABLE_AI_FILTER_PLACEHOLDER: "\u041E\u043F\u0438\u0448\u0435\u0442\u0435 \u043A\u0430\u043A\u0432\u043E \u0434\u0430 \u0444\u0438\u043B\u0442\u0440\u0438\u0440\u0430\u0442\u0435...",
  TABLE_AI_CHAT_PLACEHOLDER: "\u041F\u043E\u043F\u0438\u0442\u0430\u0439\u0442\u0435 \u0437\u0430 \u0432\u0430\u0448\u0438\u0442\u0435 \u0434\u0430\u043D\u043D\u0438...",
  TABLE_AI_INSIGHT_ANOMALY: "\u041E\u0442\u043A\u0440\u0438\u0442\u0430 \u0430\u043D\u043E\u043C\u0430\u043B\u0438\u044F",
  TABLE_AI_INSIGHT_TREND: "\u0418\u0434\u0435\u043D\u0442\u0438\u0444\u0438\u0446\u0438\u0440\u0430\u043D \u0442\u0440\u0435\u043D\u0434",
  TABLE_AI_INSIGHT_PATTERN: "\u041D\u0430\u043C\u0435\u0440\u0435\u043D \u043C\u043E\u0434\u0435\u043B",
  TABLE_AI_INSIGHT_DISMISS: "\u0417\u0430\u0442\u0432\u043E\u0440\u0438 \u043F\u0440\u043E\u0437\u0440\u0435\u043D\u0438\u0435",
  TABLE_AI_GENERATED: "\u0413\u0435\u043D\u0435\u0440\u0438\u0440\u0430\u043D\u043E \u043E\u0442 \u0418\u0418",
  TABLE_LOADING_MORE: "\u0417\u0430\u0440\u0435\u0436\u0434\u0430\u043D\u0435...",
  TABLE_TOOLBAR: "\u0414\u0435\u0439\u0441\u0442\u0432\u0438\u044F \u043D\u0430 \u0442\u0430\u0431\u043B\u0438\u0446\u0430\u0442\u0430",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "\u041B\u0435\u043D\u0442\u0430 \u0441 \u0438\u043D\u0441\u0442\u0440\u0443\u043C\u0435\u043D\u0442\u0438 \u043D\u0430 \u0442\u0430\u0431\u043B\u0438\u0446\u0430\u0442\u0430",
  TABLE_ROWS_SELECTED: "{{count}} \u0438\u0437\u0431\u0440\u0430\u043D\u0438 \u0440\u0435\u0434\u0430",
  TOKEN_ARIA_LABEL: "\u0422\u043E\u043A\u0435\u043D",
  TOKEN_ARIA_DELETABLE: "\u0418\u0437\u0442\u0440\u0438\u0432\u0430\u0435\u043C",
  TOKEN_DELETE: "\u041F\u0440\u0435\u043C\u0430\u0445\u043D\u0438",
  TOKENIZER_ARIA_LABEL: "\u0422\u043E\u043A\u0435\u043D\u0430\u0439\u0437\u0435\u0440",
  TOKENIZER_SHOW_MORE: "\u041E\u0449\u0435 {{count}}",
  TOKENIZER_POPOVER_HEADER: "\u041E\u0449\u0435 \u0442\u043E\u043A\u0435\u043D\u0438",
  TOKENIZER_CLEAR_ALL: "\u0418\u0437\u0447\u0438\u0441\u0442\u0438 \u0432\u0441\u0438\u0447\u043A\u0438",
  TOKENIZER_TOKEN_COUNT: "{{count}} \u0442\u043E\u043A\u0435\u043D\u0430",
  TOKENIZER_TOKENS_SELECTED: "{{count}} \u0442\u043E\u043A\u0435\u043D\u0430 \u0438\u0437\u0431\u0440\u0430\u043D\u0438",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} \u0440\u0435\u0437\u0443\u043B\u0442\u0430\u0442\u0430 \u043D\u0430\u043B\u0438\u0447\u043D\u0438",
  COMBOBOX_ONE_RESULT: "1 \u0440\u0435\u0437\u0443\u043B\u0442\u0430\u0442 \u043D\u0430\u043B\u0438\u0447\u0435\u043D",
  COMBOBOX_NO_RESULTS: "\u041D\u044F\u043C\u0430 \u0440\u0435\u0437\u0443\u043B\u0442\u0430\u0442\u0438",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "\u041A\u043B\u0435\u0442\u043A\u0430 \u043D\u0430 \u0441\u043F\u0438\u0441\u044A\u043A",
  SELECT_OPTION_SELECTED: "{{text}} \u0438\u0437\u0431\u0440\u0430\u043D\u043E",
  MULTICOMBOBOX_SELECT_ALL: "\u0418\u0437\u0431\u0435\u0440\u0438 \u0432\u0441\u0438\u0447\u043A\u0438",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} \u0435\u043B\u0435\u043C\u0435\u043D\u0442\u0430 \u0438\u0437\u0431\u0440\u0430\u043D\u0438",
  MENU_POPOVER_ACCESSIBLE_NAME: "\u0418\u0437\u0431\u0435\u0440\u0435\u0442\u0435 \u043E\u043F\u0446\u0438\u044F \u043E\u0442 \u043C\u0435\u043D\u044E\u0442\u043E",
  DATEPICKER_DATE_DESCRIPTION: "\u0412\u044A\u0432\u0435\u0436\u0434\u0430\u043D\u0435 \u043D\u0430 \u0434\u0430\u0442\u0430",
  DATERANGE_DESCRIPTION: "\u0412\u044A\u0432\u0435\u0436\u0434\u0430\u043D\u0435 \u043D\u0430 \u0434\u0438\u0430\u043F\u0430\u0437\u043E\u043D \u043E\u0442 \u0434\u0430\u0442\u0438",
  DATEPICKER_OPEN_ICON_TITLE: "\u041E\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u0441\u0440\u0435\u0434\u0441\u0442\u0432\u043E \u0437\u0430 \u0438\u0437\u0431\u043E\u0440",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "\u0417\u0430\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u0441\u0440\u0435\u0434\u0441\u0442\u0432\u043E\u0442\u043E \u0437\u0430 \u0438\u0437\u0431\u043E\u0440",
  INPUT_CLEAR_ICON_ACC_NAME: "\u0418\u0437\u0447\u0438\u0441\u0442\u0432\u0430\u043D\u0435",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0418\u0437\u0431\u043E\u0440 \u043D\u0430 \u0434\u0430\u0442\u0430 \u0437\u0430 {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "\u0418\u0437\u0431\u043E\u0440 \u043D\u0430 \u0434\u0438\u0430\u043F\u0430\u0437\u043E\u043D \u043E\u0442 \u0434\u0430\u0442\u0438 \u0437\u0430 {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "\u0413\u0440\u0443\u043F\u0430 \u043D\u0430 \u0441\u0435\u0433\u043C\u0435\u043D\u0442\u0438\u0440\u0430\u043D \u0431\u0443\u0442\u043E\u043D",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "\u0421\u0435\u0433\u043C\u0435\u043D\u0442\u0438\u0440\u0430\u043D \u0431\u0443\u0442\u043E\u043D",
  MESSAGE_STRIP_INFORMATION: "\u041B\u0435\u043D\u0442\u0430 \u0441\u044A\u0441 \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435",
  MESSAGE_STRIP_SUCCESS: "\u041B\u0435\u043D\u0442\u0430 \u0441\u044A\u0441 \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435 \u0437\u0430 \u0443\u0441\u043F\u0435\u0448\u043D\u043E \u0438\u0437\u043F\u044A\u043B\u043D\u0435\u043D\u0438\u0435",
  MESSAGE_STRIP_ERROR: "\u041B\u0435\u043D\u0442\u0430 \u0441\u044A\u0441 \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435 \u0437\u0430 \u0433\u0440\u0435\u0448\u043A\u0430",
  MESSAGE_STRIP_WARNING: "\u041B\u0435\u043D\u0442\u0430 \u0441 \u043F\u0440\u0435\u0434\u0443\u043F\u0440\u0435\u0434\u0438\u0442\u0435\u043B\u043D\u043E \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435",
  MESSAGE_STRIP_CLOSABLE: "\u041C\u043E\u0436\u0435 \u0434\u0430 \u0441\u0435 \u0437\u0430\u0442\u0432\u043E\u0440\u0438",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "\u0417\u0430\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u043B\u0435\u043D\u0442\u0430\u0442\u0430 \u0441 \u0438\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u043E\u043D\u043D\u043E \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "\u0417\u0430\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u043B\u0435\u043D\u0442\u0430\u0442\u0430 \u0441 \u043F\u043E\u043B\u043E\u0436\u0438\u0442\u0435\u043B\u043D\u043E \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "\u0417\u0430\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u043B\u0435\u043D\u0442\u0430\u0442\u0430 \u0441 \u043E\u0442\u0440\u0438\u0446\u0430\u0442\u0435\u043B\u043D\u043E \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "\u0417\u0430\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u043B\u0435\u043D\u0442\u0430\u0442\u0430 \u0441 \u043A\u0440\u0438\u0442\u0438\u0447\u043D\u043E \u0441\u044A\u043E\u0431\u0449\u0435\u043D\u0438\u0435",
  FILEUPLOADER_PLACEHOLDER: "\u041F\u0440\u0435\u0433\u043B\u0435\u0434\u0430\u0439\u0442\u0435 \u0438\u043B\u0438 \u043F\u0443\u0441\u043D\u0435\u0442\u0435 \u0444\u0430\u0439\u043B",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "\u041F\u0440\u0435\u0433\u043B\u0435\u0434\u0430\u0439\u0442\u0435 \u0438\u043B\u0438 \u043F\u0443\u0441\u043D\u0435\u0442\u0435 \u0444\u0430\u0439\u043B\u043E\u0432\u0435",
  FILEUPLOADER_BROWSE_TOOLTIP: "\u041F\u0440\u0435\u0433\u043B\u0435\u0434 \u043D\u0430 \u0444\u0430\u0439\u043B\u043E\u0432\u0435",
  FILEUPLOADER_CLEAR_TOOLTIP: "\u0418\u0437\u0447\u0438\u0441\u0442\u0438 \u0432\u0441\u0438\u0447\u043A\u0438",
  FILEUPLOADER_ROLE_DESCRIPTION: "\u041A\u0430\u0447\u0432\u0430\u043D\u0435 \u043D\u0430 \u0444\u0430\u0439\u043B",
  SEARCHFIELD_SEARCH: "\u0422\u044A\u0440\u0441\u0435\u043D\u0435",
  SEARCHFIELD_CLEAR: "\u0418\u0437\u0447\u0438\u0441\u0442\u0432\u0430\u043D\u0435",
  SEARCHFIELD_ARIA_DESCRIPTION: "\u0412\u044A\u0432\u0435\u0434\u0435\u0442\u0435 \u0442\u0435\u043A\u0441\u0442, \u0437\u0430 \u0434\u0430 \u0437\u0430\u043F\u043E\u0447\u043D\u0435\u0442\u0435 \u0442\u044A\u0440\u0441\u0435\u043D\u0435",
  SEARCHFIELD_ROLE_DESCRIPTION: "\u041F\u043E\u043B\u0435 \u0437\u0430 \u0442\u044A\u0440\u0441\u0435\u043D\u0435",
  NAVIGATION_EXPAND: "\u0420\u0430\u0437\u0433\u044A\u0432\u0430\u043D\u0435",
  NAVIGATION_COLLAPSE: "\u0421\u0432\u0438\u0432\u0430\u043D\u0435",
  SIGN_OUT: "\u0418\u0437\u0445\u043E\u0434",
  FX_START_PANE: "\u041D\u0430\u0447\u0430\u043B\u0435\u043D \u043F\u0430\u043D\u0435\u043B",
  FX_END_PANE: "\u041A\u0440\u0430\u0435\u043D \u043F\u0430\u043D\u0435\u043B",
  FX_RESIZE_START_CENTER: "\u041F\u0440\u0435\u043E\u0440\u0430\u0437\u043C\u0435\u0440\u044F\u0432\u0430\u043D\u0435 \u043D\u0430 \u043D\u0430\u0447\u0430\u043B\u043D\u0438\u044F \u0438 \u0446\u0435\u043D\u0442\u0440\u0430\u043B\u043D\u0438\u044F \u043F\u0430\u043D\u0435\u043B",
  FX_RESIZE_CENTER_END: "\u041F\u0440\u0435\u043E\u0440\u0430\u0437\u043C\u0435\u0440\u044F\u0432\u0430\u043D\u0435 \u043D\u0430 \u0446\u0435\u043D\u0442\u0440\u0430\u043B\u043D\u0438\u044F \u0438 \u043A\u0440\u0430\u0439\u043D\u0438\u044F \u043F\u0430\u043D\u0435\u043B",
  FX_NOTIFICATIONS: "\u0418\u0437\u0432\u0435\u0441\u0442\u0438\u044F",
  FX_OPEN_NAVIGATION: "\u041E\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u043D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u044F\u0442\u0430",
  FX_BACK_TO_LIST: "\u041E\u0431\u0440\u0430\u0442\u043D\u043E \u043A\u044A\u043C \u0441\u043F\u0438\u0441\u044A\u043A\u0430",
  FX_BACK: "\u041D\u0430\u0437\u0430\u0434",
  FX_RELATED: "\u0421\u0432\u044A\u0440\u0437\u0430\u043D\u0438",
  FX_MORE_ACTIONS: "\u041E\u0449\u0435 \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u044F",
  FX_CLOSE: "\u0417\u0430\u0442\u0432\u0430\u0440\u044F\u043D\u0435",
  FX_ADD: "\u0414\u043E\u0431\u0430\u0432\u044F\u043D\u0435",
  FX_SEND: "\u0418\u0437\u043F\u0440\u0430\u0449\u0430\u043D\u0435",
  FX_DICTATE: "\u0414\u0438\u043A\u0442\u0443\u0432\u0430\u043D\u0435",
  FX_VOICE: "\u0413\u043B\u0430\u0441",
  FX_DOUBLE_CLICK_TO_RENAME: "\u0429\u0440\u0430\u043A\u043D\u0435\u0442\u0435 \u0434\u0432\u0430 \u043F\u044A\u0442\u0438 \u0437\u0430 \u043F\u0440\u0435\u0438\u043C\u0435\u043D\u0443\u0432\u0430\u043D\u0435",
  FX_CANCEL: "\u041E\u0442\u043A\u0430\u0437",
  FX_ACCEPT: "\u041F\u0440\u0438\u0435\u043C\u0430\u043D\u0435",
  FX_MAIN_NAVIGATION: "\u041E\u0441\u043D\u043E\u0432\u043D\u0438 \u0435\u043B\u0435\u043C\u0435\u043D\u0442\u0438 \u0437\u0430 \u043D\u0430\u0432\u0438\u0433\u0430\u0446\u0438\u044F",
  FX_USER_ACTIONS: "\u041F\u043E\u0442\u0440\u0435\u0431\u0438\u0442\u0435\u043B\u0441\u043A\u0438 \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u044F",
  FX_CURRENT_ACCOUNT: "\u0418\u043D\u0444\u043E\u0440\u043C\u0430\u0446\u0438\u044F \u0437\u0430 \u0442\u0435\u043A\u0443\u0449\u0438\u044F \u0430\u043A\u0430\u0443\u043D\u0442",
  FX_EDIT_ACCOUNTS: "\u0420\u0435\u0434\u0430\u043A\u0442\u0438\u0440\u0430\u043D\u0435 \u043D\u0430 \u0430\u043A\u0430\u0443\u043D\u0442\u0438",
  LINK_SUBTLE: "\u0411\u043B\u0435\u0434",
  LINK_EMPHASIZED: "\u041F\u043E\u0434\u0447\u0435\u0440\u0442\u0430\u043D\u043E",
  SPLIT_BUTTON_DESCRIPTION: "\u0411\u0443\u0442\u043E\u043D \u0440\u0430\u0437\u0434\u0435\u043B\u044F\u043D\u0435",
  SPLIT_BUTTON_KEYBOARD_HINT: "\u041D\u0430\u0442\u0438\u0441\u043D\u0435\u0442\u0435 \u0438\u043D\u0442\u0435\u0440\u0432\u0430\u043B \u0438\u043B\u0438 Enter, \u0437\u0430 \u0434\u0430 \u0438\u043D\u0438\u0446\u0438\u0438\u0440\u0430\u0442\u0435 \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u0435 \u043F\u043E \u043F\u043E\u0434\u0440\u0430\u0437\u0431\u0438\u0440\u0430\u043D\u0435 \u0438 Alt + \u0441\u0442\u0440\u0435\u043B\u043A\u0430 \u043D\u0430\u0434\u043E\u043B\u0443 \u0438\u043B\u0438 F4, \u0437\u0430 \u0434\u0430 \u0438\u043D\u0438\u0446\u0438\u0438\u0440\u0430\u0442\u0435 \u0434\u0435\u0439\u0441\u0442\u0432\u0438\u0435 \u043D\u0430 \u0441\u0442\u0440\u0435\u043B\u043A\u0430.",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "\u041E\u0442\u0432\u0430\u0440\u044F\u043D\u0435 \u043D\u0430 \u043C\u0435\u043D\u044E\u0442\u043E"
};

// src/i18n/locales/es.json
var es_default = {
  INPUT_CLEAR: "Borrar",
  INPUT_SUGGESTIONS_AVAILABLE: "{{count}} sugerencias disponibles",
  INPUT_SUGGESTIONS_ONE: "1 sugerencia disponible",
  INPUT_SUGGESTIONS_NONE: "Sin sugerencias",
  INPUT_SHOW_PASSWORD: "Mostrar contrase\xF1a",
  INPUT_HIDE_PASSWORD: "Ocultar contrase\xF1a",
  VALUE_STATE_ERROR: "Error",
  VALUE_STATE_WARNING: "Advertencia",
  VALUE_STATE_SUCCESS: "\xC9xito",
  VALUE_STATE_INFORMATION: "Informaci\xF3n",
  RADIOBUTTON_VALUE_STATE_ERROR: "Entrada no v\xE1lida",
  RADIOBUTTON_VALUE_STATE_WARNING: "Advertencia enviada",
  RADIOBUTTON_VALUE_STATE_SUCCESS: "Entrada validada correctamente",
  RADIOBUTTON_VALUE_STATE_INFORMATION: "Entrada informativa",
  REQUIRED_FIELD: "Campo obligatorio",
  LOADING: "Cargando...",
  NO_DATA: "Sin datos",
  TABLE_NO_DATA: "No hay datos disponibles",
  TABLE_ROW: "Fila",
  TABLE_ROW_INDEX: "Fila {{index}} de {{total}}",
  TABLE_ROW_SELECTED: "Seleccionada",
  TABLE_ROW_ACTIVE: "Activa",
  TABLE_ROW_NAVIGABLE: "Navegable",
  TABLE_ROW_NAVIGATED: "Navegada",
  TABLE_ROW_SELECTOR: "Selector de fila",
  TABLE_ROW_SINGLE_ACTION: "{{count}} acci\xF3n",
  TABLE_ROW_MULTIPLE_ACTIONS: "{{count}} acciones",
  TABLE_ROW_POPIN: "Informaci\xF3n adicional",
  TABLE_ROW_ACTIONS: "Acciones de fila",
  TABLE_COLUMN_HEADER_ROW: "Fila de encabezado de columna",
  TABLE_SELECTION: "Selecci\xF3n",
  TABLE_SELECT_ALL_ROWS: "Seleccionar todas las filas",
  TABLE_DESELECT_ALL_ROWS: "Deseleccionar todas las filas",
  TABLE_MORE: "M\xE1s",
  TABLE_MORE_DESCRIPTION: "Cargar m\xE1s datos",
  TABLE_MULTI_SELECTABLE: "Tabla de selecci\xF3n m\xFAltiple",
  TABLE_SINGLE_SELECTABLE: "Tabla de selecci\xF3n \xFAnica",
  TABLE_NAVIGATION: "Navegaci\xF3n",
  TABLE_SORT_ASCENDING: "Ordenado ascendente",
  TABLE_SORT_DESCENDING: "Ordenado descendente",
  TABLE_SORT_NONE: "Sin ordenar",
  TABLE_AI_SORT_PLACEHOLDER: "Describe c\xF3mo ordenar...",
  TABLE_AI_FILTER_PLACEHOLDER: "Describe qu\xE9 filtrar...",
  TABLE_AI_CHAT_PLACEHOLDER: "Pregunta sobre tus datos...",
  TABLE_AI_INSIGHT_ANOMALY: "Anomal\xEDa detectada",
  TABLE_AI_INSIGHT_TREND: "Tendencia identificada",
  TABLE_AI_INSIGHT_PATTERN: "Patr\xF3n encontrado",
  TABLE_AI_INSIGHT_DISMISS: "Cerrar informaci\xF3n",
  TABLE_AI_GENERATED: "Generado por IA",
  TABLE_LOADING_MORE: "Cargando m\xE1s...",
  TABLE_TOOLBAR: "Acciones de tabla",
  TABLE_TOOLBAR_ROLEDESCRIPTION: "Barra de herramientas de tabla",
  TABLE_ROWS_SELECTED: "{{count}} filas seleccionadas",
  TOKEN_ARIA_LABEL: "Token",
  TOKEN_ARIA_DELETABLE: "Eliminable",
  TOKEN_DELETE: "Eliminar",
  TOKENIZER_ARIA_LABEL: "Tokenizador",
  TOKENIZER_SHOW_MORE: "{{count}} M\xE1s",
  TOKENIZER_POPOVER_HEADER: "M\xE1s tokens",
  TOKENIZER_CLEAR_ALL: "Borrar todo",
  TOKENIZER_TOKEN_COUNT: "{{count}} tokens",
  TOKENIZER_TOKENS_SELECTED: "{{count}} tokens seleccionados",
  COMBOBOX_RESULTS_AVAILABLE: "{{count}} resultados disponibles",
  COMBOBOX_ONE_RESULT: "1 resultado disponible",
  COMBOBOX_NO_RESULTS: "Sin resultados",
  COMBOBOX_VALUE_STATE_LINK_SHORTCUT: "To move the focus to the link, press Ctrl+Alt+F8",
  COMBOBOX_VALUE_STATE_LINKS_SHORTCUT: "To go to the first link, press Ctrl+Alt+F8. To move to the next link, use Tab",
  SELECT_ROLE_DESCRIPTION: "Lista desplegable",
  SELECT_OPTION_SELECTED: "{{text}} seleccionado",
  MULTICOMBOBOX_SELECT_ALL: "Seleccionar todo",
  MULTICOMBOBOX_ITEMS_SELECTED: "{{count}} elementos seleccionados",
  MENU_POPOVER_ACCESSIBLE_NAME: "Seleccionar una opci\xF3n del men\xFA",
  DATEPICKER_DATE_DESCRIPTION: "Entrada de fecha",
  DATERANGE_DESCRIPTION: "Entrada de intervalo de fechas",
  DATEPICKER_OPEN_ICON_TITLE: "Abrir selector",
  DATEPICKER_OPEN_ICON_TITLE_OPENED: "Cerrar selector",
  INPUT_CLEAR_ICON_ACC_NAME: "Borrar",
  DATEPICKER_POPOVER_ACCESSIBLE_NAME: "Seleccionar fecha para {{0}}",
  DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME: "Seleccionar intervalo de fechas para {{0}}",
  SEGMENTEDBUTTON_ARIA_DESCRIPTION: "Grupo de botones segmentados",
  SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION: "Bot\xF3n segmentado",
  MESSAGE_STRIP_INFORMATION: "L\xEDnea de mensajes",
  MESSAGE_STRIP_SUCCESS: "L\xEDnea de mensajes de confirmaci\xF3n",
  MESSAGE_STRIP_ERROR: "L\xEDnea de mensajes de error",
  MESSAGE_STRIP_WARNING: "L\xEDnea de mensajes de advertencia",
  MESSAGE_STRIP_CLOSABLE: "Se puede cerrar",
  MESSAGE_STRIP_CLOSE_BUTTON_INFORMATION: "Cerrar l\xEDnea de mensajes informativos",
  MESSAGE_STRIP_CLOSE_BUTTON_POSITIVE: "Cerrar l\xEDnea de mensajes positivos",
  MESSAGE_STRIP_CLOSE_BUTTON_NEGATIVE: "Cerrar l\xEDnea de mensajes negativos",
  MESSAGE_STRIP_CLOSE_BUTTON_CRITICAL: "Cerrar l\xEDnea de mensajes cr\xEDticos",
  FILEUPLOADER_PLACEHOLDER: "Buscar o soltar un archivo",
  FILEUPLOADER_PLACEHOLDER_MULTIPLE: "Buscar o soltar archivos",
  FILEUPLOADER_BROWSE_TOOLTIP: "Buscar archivos",
  FILEUPLOADER_CLEAR_TOOLTIP: "Borrar todo",
  FILEUPLOADER_ROLE_DESCRIPTION: "Carga de archivos",
  SEARCHFIELD_SEARCH: "Buscar",
  SEARCHFIELD_CLEAR: "Borrar",
  SEARCHFIELD_ARIA_DESCRIPTION: "Escriba texto para iniciar la b\xFAsqueda",
  SEARCHFIELD_ROLE_DESCRIPTION: "Campo de b\xFAsqueda",
  NAVIGATION_EXPAND: "Expandir",
  NAVIGATION_COLLAPSE: "Contraer",
  SIGN_OUT: "Cerrar sesi\xF3n",
  FX_START_PANE: "Panel inicial",
  FX_END_PANE: "Panel final",
  FX_RESIZE_START_CENTER: "Cambiar tama\xF1o de paneles inicial y central",
  FX_RESIZE_CENTER_END: "Cambiar tama\xF1o de paneles central y final",
  FX_NOTIFICATIONS: "Notificaciones",
  FX_OPEN_NAVIGATION: "Abrir navegaci\xF3n",
  FX_BACK_TO_LIST: "Volver a la lista",
  FX_BACK: "Atr\xE1s",
  FX_RELATED: "Relacionado",
  FX_MORE_ACTIONS: "M\xE1s acciones",
  FX_CLOSE: "Cerrar",
  FX_ADD: "Agregar",
  FX_SEND: "Enviar",
  FX_DICTATE: "Dictar",
  FX_VOICE: "Voz",
  FX_DOUBLE_CLICK_TO_RENAME: "Haga doble clic para cambiar el nombre",
  FX_CANCEL: "Cancelar",
  FX_ACCEPT: "Aceptar",
  FX_MAIN_NAVIGATION: "Elementos de navegaci\xF3n principal",
  FX_USER_ACTIONS: "Acciones del usuario",
  FX_CURRENT_ACCOUNT: "Informaci\xF3n de la cuenta actual",
  FX_EDIT_ACCOUNTS: "Editar cuentas",
  LINK_SUBTLE: "Discreto",
  LINK_EMPHASIZED: "Resaltado",
  SPLIT_BUTTON_DESCRIPTION: "Bot\xF3n doble",
  SPLIT_BUTTON_KEYBOARD_HINT: "Pulse la barra espaciadora o intro para iniciar la acci\xF3n por defecto y Alt + flecha hacia abajo o F4 para iniciar la acci\xF3n de flecha",
  SPLIT_BUTTON_ARROW_BUTTON_TOOLTIP: "Abrir men\xFA"
};

// src/i18n/resources.ts
var translations = {
  en: en_default,
  de: de_default,
  ar: ar_default,
  fr: fr_default,
  he: he_default,
  ja: ja_default,
  ko: ko_default,
  "zh-CN": zh_CN_default,
  "zh-TW": zh_TW_default,
  ru: ru_default,
  uk: uk_default,
  bg: bg_default,
  es: es_default
};
function loadFxTranslations(lang) {
  return translations[lang] ?? translations[lang.split("-")[0]] ?? translations.en;
}
function buildResources() {
  return Object.fromEntries(
    Object.entries(translations).map(([lang, strings]) => [lang, { fx: strings }])
  );
}
var initPromise = null;
async function initFxI18n(language) {
  if (i18n.isInitialized) {
    const lang = language ?? i18n.language ?? "en";
    const base = lang.split("-")[0];
    for (const code of [lang, base]) {
      if (!i18n.hasResourceBundle(code, "fx")) {
        i18n.addResourceBundle(code, "fx", loadFxTranslations(code), true, true);
      }
    }
    return;
  }
  if (!initPromise) {
    initPromise = i18n.use(initReactI18next).init({
      lng: language ?? "en",
      fallbackLng: "en",
      defaultNS: "fx",
      ns: ["fx"],
      resources: buildResources(),
      interpolation: { escapeValue: false }
    }).then(() => void 0);
  }
  return initPromise;
}

// src/i18n/types.ts
var RTL_LOCALES = /* @__PURE__ */ new Set([
  "ar",
  "ar-SA",
  "ar-EG",
  "ar-AE",
  "he",
  "he-IL",
  "fa",
  "fa-IR",
  "ur",
  "ur-PK"
]);
function isRTLLocale(locale) {
  return RTL_LOCALES.has(locale) || RTL_LOCALES.has(locale.split("-")[0]);
}
function useDirection() {
  const { i18n: i18n2 } = useTranslation();
  return isRTLLocale(i18n2.language) ? "rtl" : "ltr";
}
var cache = /* @__PURE__ */ new Map();
function ensureCldr(localeStr) {
  let p = cache.get(localeStr);
  if (!p) {
    const loc = new Locale2(localeStr);
    p = fetchCldr(loc.getLanguage(), loc.getRegion(), loc.getScript()).catch(
      (err) => {
        console.warn(
          `[fx-components] CLDR load failed for "${localeStr}":`,
          err
        );
        cache.delete(localeStr);
      }
    );
    cache.set(localeStr, p);
  }
  return p;
}
function isCldrLoaded(localeStr) {
  try {
    const localeData = getCachedLocaleDataInstance(new Locale2(localeStr));
    localeData.getDatePattern("medium");
    return true;
  } catch {
    return false;
  }
}
ensureCldr(typeof navigator !== "undefined" ? navigator.language : "en");
function useEnsureCldr(locale = "en-US") {
  const [readyLocale, setReadyLocale] = useState(
    () => isCldrLoaded(locale) ? locale : null
  );
  useEffect(() => {
    if (isCldrLoaded(locale)) {
      setReadyLocale(locale);
      return;
    }
    let cancelled = false;
    setReadyLocale(null);
    ensureCldr(locale).then(() => {
      if (!cancelled) setReadyLocale(locale);
    });
    return () => {
      cancelled = true;
    };
  }, [locale]);
  return readyLocale === locale || isCldrLoaded(locale);
}
var dayCellVariants = cva(
  "relative z-[1] w-8 h-8 rounded text-sm font-normal transition-colors focus:outline-none focus:[box-shadow:inset_0_0_0_2px_var(--border-focus)] flex items-center justify-center",
  {
    variants: {
      variant: {
        default: "text-foreground",
        selected: "",
        rangeStart: "",
        rangeEnd: "",
        rangeBetween: "",
        today: "",
        outside: "text-sapphire-text-tertiary",
        disabled: "text-sapphire-text-tertiary opacity-40 cursor-not-allowed"
      },
      specialType: {
        none: "",
        Type01: "border-b-2 border-b-blue-500",
        Type02: "border-b-2 border-b-green-500",
        Type03: "border-b-2 border-b-yellow-500",
        Type04: "border-b-2 border-b-red-500",
        Type05: "border-b-2 border-b-purple-500",
        Type06: "border-b-2 border-b-pink-500",
        Type07: "border-b-2 border-b-indigo-500",
        Type08: "border-b-2 border-b-cyan-500",
        Type09: "border-b-2 border-b-teal-500",
        Type10: "border-b-2 border-b-orange-500",
        Type11: "border-b-2 border-b-lime-500",
        Type12: "border-b-2 border-b-emerald-500",
        Type13: "border-b-2 border-b-sky-500",
        Type14: "border-b-2 border-b-violet-500",
        Type15: "border-b-2 border-b-fuchsia-500",
        Type16: "border-b-2 border-b-rose-500",
        Type17: "border-b-2 border-b-amber-500",
        Type18: "border-b-2 border-b-slate-500",
        Type19: "border-b-2 border-b-zinc-500",
        Type20: "border-b-2 border-b-neutral-500",
        Today: "font-bold",
        Selected: "",
        Working: "bg-sapphire-positive-bg",
        NonWorking: "bg-sapphire-negative-bg"
      }
    },
    defaultVariants: {
      variant: "default",
      specialType: "none"
    }
  }
);
var dayCellInnerVariants = cva(
  "w-6 h-6 rounded flex items-center justify-center transition-colors",
  {
    variants: {
      variant: {
        default: "hover:bg-sapphire-bg-secondary",
        selected: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background hover:text-sapphire-text-on-surface",
        rangeStart: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background",
        rangeEnd: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background",
        rangeBetween: "border border-sapphire-border-accent bg-sapphire-brand-toggle-background text-sapphire-text hover:bg-sapphire-brand-toggle-background",
        today: "rounded-sm border border-sapphire-border-accent text-sapphire-text-accent font-semibold",
        outside: "",
        disabled: ""
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
var FALLBACK_WIDE_DAYS = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
var FALLBACK_FIRST_DAY_OF_WEEK = 1;
var FALLBACK_WIDE_MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
function DayPicker({
  currentMonth,
  focusedDate,
  autoFocus = false,
  isDateSelected,
  isDateDisabled,
  isRangeStart,
  isRangeEnd,
  isDateBetweenRange,
  isRangeSelectionInProgress = false,
  getSpecialDate,
  onSelectDate,
  onFocusChange,
  onHoverDate,
  showWeekNumbers = false,
  weekNumbering,
  firstDayOfWeek,
  locale,
  className,
  ref
}) {
  const cldrReady = useEnsureCldr(locale);
  const effectiveFirstDay = React.useMemo(() => {
    if (firstDayOfWeek !== void 0) return firstDayOfWeek;
    if (cldrReady) return getWeekStartDay(locale);
    return FALLBACK_FIRST_DAY_OF_WEEK;
  }, [firstDayOfWeek, cldrReady, locale]);
  const days = getMonthDays(currentMonth, effectiveFirstDay);
  const weekDays = React.useMemo(() => {
    if (!cldrReady) return null;
    const localeData = getCachedLocaleDataInstance(getLocale(locale));
    const allDays = localeData.getDaysStandAlone("abbreviated");
    const start = (effectiveFirstDay % 7 + 7) % 7;
    return [...allDays.slice(start), ...allDays.slice(0, start)];
  }, [locale, cldrReady, effectiveFirstDay]);
  const { wideDays, wideMonths } = React.useMemo(() => {
    if (!cldrReady) return {
      wideDays: FALLBACK_WIDE_DAYS,
      wideMonths: FALLBACK_WIDE_MONTHS
    };
    const localeData = getCachedLocaleDataInstance(getLocale(locale));
    return {
      wideDays: localeData.getDaysStandAlone("wide"),
      wideMonths: localeData.getMonthsStandAlone("wide")
    };
  }, [locale, cldrReady]);
  const gridRef = useRef(null);
  const focusedCellRef = useRef(null);
  const isMountedRef = useRef(false);
  const setFocusedCellRef = useCallback((node) => {
    focusedCellRef.current = node;
    if (node && isMountedRef.current) {
      node.focus({ preventScroll: true });
    }
  }, []);
  useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, [focusedDate]);
  useEffect(() => {
    isMountedRef.current = true;
  }, []);
  useEffect(() => {
    if (!autoFocus) return;
    const rafId = requestAnimationFrame(() => {
      focusedCellRef.current?.focus({ preventScroll: true });
    });
    return () => cancelAnimationFrame(rafId);
  }, []);
  const handleKeyDown = useCallback(
    (e, date) => {
      let newDate = null;
      const currentDate = new Date(date);
      switch (e.key) {
        case "ArrowLeft":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() - 1);
          newDate = currentDate;
          break;
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() + 1);
          newDate = currentDate;
          break;
        case "ArrowUp":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() - 7);
          newDate = currentDate;
          break;
        case "ArrowDown":
          e.preventDefault();
          e.stopPropagation();
          currentDate.setDate(currentDate.getDate() + 7);
          newDate = currentDate;
          break;
        case "Home":
          e.preventDefault();
          if (e.ctrlKey) {
            newDate = new Date(
              currentDate.getFullYear(),
              currentDate.getMonth(),
              1
            );
          } else {
            const dayOfWeek = currentDate.getDay();
            const diff = (dayOfWeek - effectiveFirstDay + 7) % 7;
            currentDate.setDate(currentDate.getDate() - diff);
            newDate = currentDate;
          }
          break;
        case "End":
          e.preventDefault();
          if (e.ctrlKey) {
            newDate = new Date(
              currentDate.getFullYear(),
              currentDate.getMonth() + 1,
              0
            );
          } else {
            const dayOfWeek = currentDate.getDay();
            const diff = (effectiveFirstDay + 6 - dayOfWeek + 7) % 7;
            currentDate.setDate(currentDate.getDate() + diff);
            newDate = currentDate;
          }
          break;
        case "PageUp":
          e.preventDefault();
          if (e.shiftKey && e.ctrlKey) {
            currentDate.setFullYear(currentDate.getFullYear() - 10);
          } else if (e.shiftKey) {
            currentDate.setFullYear(currentDate.getFullYear() - 1);
          } else {
            currentDate.setMonth(currentDate.getMonth() - 1);
          }
          newDate = currentDate;
          break;
        case "PageDown":
          e.preventDefault();
          if (e.shiftKey && e.ctrlKey) {
            currentDate.setFullYear(currentDate.getFullYear() + 10);
          } else if (e.shiftKey) {
            currentDate.setFullYear(currentDate.getFullYear() + 1);
          } else {
            currentDate.setMonth(currentDate.getMonth() + 1);
          }
          newDate = currentDate;
          break;
        case "Enter":
        case " ":
          e.preventDefault();
          onSelectDate(date);
          return;
      }
      if (newDate && !isDateDisabled(newDate)) {
        onFocusChange(newDate);
        if (isRangeSelectionInProgress && onHoverDate) {
          onHoverDate(newDate);
        }
      }
    },
    [isDateDisabled, onSelectDate, onFocusChange, isRangeSelectionInProgress, onHoverDate, effectiveFirstDay]
  );
  const handleMouseOver = useCallback(
    (date) => {
      if (isRangeSelectionInProgress && onHoverDate) {
        onHoverDate(date);
      }
    },
    [isRangeSelectionInProgress, onHoverDate]
  );
  const handleMouseLeave = useCallback(() => {
    if (isRangeSelectionInProgress && onHoverDate) {
      onHoverDate(null);
    }
  }, [isRangeSelectionInProgress, onHoverDate]);
  const getDayCellVariant = (date) => {
    if (isDateDisabled(date)) return "disabled";
    const isStart = isRangeStart(date);
    const isEnd = isRangeEnd(date);
    if (isStart && isEnd) return "selected";
    if (isStart) return "rangeStart";
    if (isEnd) return "rangeEnd";
    if (isDateBetweenRange(date)) return "rangeBetween";
    if (isDateSelected(date)) return "selected";
    if (!isSameMonthUtil(date, currentMonth)) return "outside";
    if (isToday(date)) return "today";
    return "default";
  };
  const getDayCellSpecialType = (date) => {
    const special = getSpecialDate(date);
    return special?.type || "None" /* None */;
  };
  const getWeekNumbers = () => {
    if (!showWeekNumbers) return [];
    const weeks = [];
    for (let i = 0; i < days.length; i += 7) {
      weeks.push(getWeekNumber(days[i], weekNumbering));
    }
    return weeks;
  };
  const getRangeStripClass = (date, idx) => {
    const isStart = isRangeStart(date);
    const isEnd = isRangeEnd(date);
    const isBetween = isDateBetweenRange(date);
    if (!isStart && !isEnd && !isBetween) return null;
    const colIdx = idx % 7;
    const isFirstCol = colIdx === 0;
    const isLastCol = colIdx === 6;
    if (isStart && isEnd) {
      return null;
    }
    if (isStart) {
      return isLastCol ? null : "range-strip range-strip-right";
    }
    if (isEnd) {
      return isFirstCol ? null : "range-strip range-strip-left";
    }
    if (isFirstCol) return "range-strip range-strip-right";
    if (isLastCol) return "range-strip range-strip-left";
    return "range-strip range-strip-both";
  };
  const weekNumbers = getWeekNumbers();
  const gridCols = showWeekNumbers ? "grid-cols-[repeat(8,2rem)]" : "grid-cols-[repeat(7,2rem)]";
  return /* @__PURE__ */ jsxs("div", { ref, className: cn("p-2 flex flex-col items-center", className), children: [
    /* @__PURE__ */ jsx("style", { children: `
          .range-strip {
            position: relative;
          }
          .range-strip::before {
            content: "";
            position: absolute;
            top: 0;
            bottom: 0;
            z-index: 0;
            background: transparent;
            border-radius: 0;
            pointer-events: none;
          }
          .range-strip-right::before {
            left: 0;
            right: 0;
            border-radius: 0;
          }
          .range-strip-left::before {
            left: 0;
            right: 0;
            border-radius: 0;
          }
          .range-strip-both::before {
            left: 0;
            right: 0;
            border-radius: 0;
          }
        ` }),
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: gridRef,
        className: "flex flex-col",
        role: "grid",
        "aria-label": "Calendar days",
        onMouseLeave: handleMouseLeave,
        children: [
          /* @__PURE__ */ jsxs("div", { role: "row", className: cn("grid", gridCols), children: [
            showWeekNumbers && /* @__PURE__ */ jsx(
              "div",
              {
                role: "columnheader",
                className: "h-8 w-full flex items-center justify-center text-xs font-medium text-sapphire-text-tertiary"
              }
            ),
            weekDays ? weekDays.map((day) => /* @__PURE__ */ jsx(
              "div",
              {
                role: "columnheader",
                className: "h-8 w-full flex items-center justify-center text-xs font-medium text-sapphire-text-tertiary",
                children: day
              },
              day
            )) : Array.from({ length: 7 }, (_, i) => /* @__PURE__ */ jsx(
              "div",
              {
                role: "columnheader",
                className: "h-8 w-full flex items-center justify-center",
                children: /* @__PURE__ */ jsx("span", { className: "inline-block h-3 w-6 rounded bg-muted animate-pulse" })
              },
              i
            ))
          ] }),
          Array.from({ length: Math.ceil(days.length / 7) }, (_, weekIndex) => {
            const weekStart = weekIndex * 7;
            const weekDates = days.slice(weekStart, weekStart + 7);
            const weekKey = weekDates.length > 0 ? `week-${formatDateToISO(weekDates[0])}` : `week-${weekIndex}`;
            return /* @__PURE__ */ jsxs("div", { role: "row", className: cn("grid items-center", gridCols), children: [
              showWeekNumbers && /* @__PURE__ */ jsx(
                "div",
                {
                  role: "rowheader",
                  className: "h-8 w-full flex items-center justify-center text-xs text-sapphire-text-tertiary",
                  children: cldrReady ? weekNumbers[weekIndex] : /* @__PURE__ */ jsx("span", { className: "inline-block h-3 w-4 rounded bg-muted animate-pulse" })
                }
              ),
              weekDates.map((date, dayIndex) => {
                const idx = weekStart + dayIndex;
                const variant = getDayCellVariant(date);
                const specialType = getDayCellSpecialType(date);
                const special = getSpecialDate(date);
                const isFocused = focusedDate && date.getTime() === focusedDate.getTime();
                const isCurrentMonth = isSameMonthUtil(date, currentMonth);
                const isTodayDate = isToday(date);
                const isDisabled = isDateDisabled(date);
                const rangeStripClass = getRangeStripClass(date, idx);
                const dayName = wideDays[date.getDay()];
                const monthName = wideMonths[date.getMonth()];
                let ariaLabel = `${dayName}, ${monthName} ${date.getDate()}, ${date.getFullYear()}`;
                if (isTodayDate) ariaLabel += ", Today";
                if (!isCurrentMonth) ariaLabel += ", Outside current month";
                if (isDisabled) ariaLabel += ", Not available";
                if (special?.tooltip) ariaLabel += `, ${special.tooltip}`;
                const isWeekend = date.getDay() === 0 || date.getDay() === 6;
                const gridcell = /* @__PURE__ */ jsx(
                  "div",
                  {
                    ref: isFocused ? setFocusedCellRef : void 0,
                    role: "gridcell",
                    "aria-label": ariaLabel,
                    "aria-selected": isDateSelected(date),
                    "aria-current": isTodayDate ? "date" : void 0,
                    "aria-disabled": isDisabled,
                    title: special?.tooltip,
                    tabIndex: isFocused ? 0 : -1,
                    className: cn(
                      dayCellVariants({
                        variant,
                        specialType
                      }),
                      "cursor-pointer",
                      isWeekend && variant === "default" && "bg-sapphire-background-quaternary border border-sapphire-background-primary rounded"
                    ),
                    onClick: () => {
                      if (!isDateDisabled(date)) {
                        onFocusChange(date);
                        onSelectDate(date);
                      }
                    },
                    onMouseOver: () => handleMouseOver(date),
                    onKeyDown: (e) => handleKeyDown(e, date),
                    children: /* @__PURE__ */ jsx("span", { className: dayCellInnerVariants({ variant }), children: cldrReady ? format(date, "d") : /* @__PURE__ */ jsx("span", { className: "inline-block h-3 w-4 rounded bg-muted animate-pulse" }) })
                  }
                );
                return rangeStripClass ? /* @__PURE__ */ jsx("div", { className: rangeStripClass, children: gridcell }, formatDateToISO(date)) : /* @__PURE__ */ jsx(React.Fragment, { children: gridcell }, formatDateToISO(date));
              })
            ] }, weekKey);
          })
        ]
      }
    )
  ] });
}
var monthCellVariants = cva(
  "min-h-[2.75rem] w-full rounded-lg text-sm font-normal transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "hover:bg-sapphire-bg-secondary hover:text-sapphire-text",
        selected: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function MonthPicker({
  currentYear,
  focusedMonth,
  selectedMonth,
  onSelectMonth,
  onFocusChange,
  onPreviousYear,
  onNextYear,
  locale,
  className,
  ref
}) {
  const months = Array.from({ length: 12 }, (_, i) => i);
  const focusedCellRef = useRef(null);
  const cldrReady = useEnsureCldr(locale);
  useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, [focusedMonth]);
  useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, []);
  const handleKeyDown = useCallback(
    (e, month) => {
      let newMonth = null;
      switch (e.key) {
        case "ArrowLeft":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month === 0 ? 11 : month - 1;
          break;
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month === 11 ? 0 : month + 1;
          break;
        case "ArrowUp":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month < 3 ? month + 9 : month - 3;
          break;
        case "ArrowDown":
          e.preventDefault();
          e.stopPropagation();
          newMonth = month > 8 ? month - 9 : month + 3;
          break;
        case "Home":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newMonth = 0;
          } else {
            newMonth = Math.floor(month / 3) * 3;
          }
          break;
        case "End":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newMonth = 11;
          } else {
            newMonth = Math.min(Math.floor(month / 3) * 3 + 2, 11);
          }
          break;
        case "PageUp":
          e.preventDefault();
          e.stopPropagation();
          onPreviousYear?.();
          return;
        case "PageDown":
          e.preventDefault();
          e.stopPropagation();
          onNextYear?.();
          return;
        case "Enter":
        case " ":
          e.preventDefault();
          e.stopPropagation();
          onSelectMonth(month);
          return;
      }
      if (newMonth !== null) {
        onFocusChange(newMonth);
      }
    },
    [onSelectMonth, onFocusChange, onPreviousYear, onNextYear]
  );
  return /* @__PURE__ */ jsx("div", { ref, className: cn("p-3 w-full", className), children: /* @__PURE__ */ jsx("div", { className: "flex flex-col gap-2 w-full", role: "grid", "aria-label": "Select month", children: Array.from({ length: 4 }, (_, rowIndex) => {
    const rowMonths = months.slice(rowIndex * 3, (rowIndex + 1) * 3);
    return /* @__PURE__ */ jsx("div", { role: "row", className: "grid grid-cols-3 gap-2", children: rowMonths.map((month) => {
      const date = createDate(currentYear, month);
      const monthName = cldrReady ? getMonthName(date, "short", locale) : "";
      const isFocused = month === focusedMonth;
      const isSelected = month === selectedMonth;
      return /* @__PURE__ */ jsx(
        "div",
        {
          ref: isFocused ? focusedCellRef : void 0,
          role: "gridcell",
          "aria-label": `${monthName} ${currentYear}`,
          "aria-selected": isSelected,
          tabIndex: isFocused ? 0 : -1,
          className: cn(
            monthCellVariants({
              variant: isSelected ? "selected" : "default"
            }),
            "flex items-center justify-center cursor-pointer"
          ),
          onClick: () => onSelectMonth(month),
          onKeyDown: (e) => handleKeyDown(e, month),
          children: cldrReady ? monthName : /* @__PURE__ */ jsx("span", { className: "inline-block h-3.5 w-8 rounded bg-muted animate-pulse" })
        },
        month
      );
    }) }, rowIndex);
  }) }) });
}
var yearCellVariants = cva(
  "min-h-[2.75rem] w-full rounded-lg text-sm font-normal transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "hover:bg-sapphire-bg-secondary hover:text-sapphire-text",
        selected: "bg-sapphire-brand-hover-background text-sapphire-text-on-surface hover:bg-sapphire-brand-hover-background"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function YearPicker({
  startYear,
  endYear,
  focusedYear,
  selectedYear,
  onSelectYear,
  onFocusChange,
  onPreviousRange,
  onNextRange,
  className,
  ref
}) {
  const years = Array.from(
    { length: endYear - startYear + 1 },
    (_, i) => startYear + i
  );
  const focusedCellRef = useRef(null);
  useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, [focusedYear]);
  useEffect(() => {
    if (focusedCellRef.current) {
      focusedCellRef.current.focus({ preventScroll: true });
    }
  }, []);
  const handleKeyDown = useCallback(
    (e, year) => {
      const currentIndex = years.indexOf(year);
      let newYear = null;
      switch (e.key) {
        case "ArrowLeft":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex > 0) {
            newYear = years[currentIndex - 1];
          } else {
            onPreviousRange();
          }
          break;
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex < years.length - 1) {
            newYear = years[currentIndex + 1];
          } else {
            onNextRange();
          }
          break;
        case "ArrowUp":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex >= 4) {
            newYear = years[currentIndex - 4];
          } else {
            onPreviousRange();
          }
          break;
        case "ArrowDown":
          e.preventDefault();
          e.stopPropagation();
          if (currentIndex < years.length - 4) {
            newYear = years[currentIndex + 4];
          } else {
            onNextRange();
          }
          break;
        case "Home":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newYear = years[0];
          } else {
            newYear = years[Math.floor(currentIndex / 4) * 4];
          }
          break;
        case "End":
          e.preventDefault();
          e.stopPropagation();
          if (e.ctrlKey) {
            newYear = years[years.length - 1];
          } else {
            newYear = years[Math.min(Math.floor(currentIndex / 4) * 4 + 3, years.length - 1)];
          }
          break;
        case "PageUp":
          e.preventDefault();
          e.stopPropagation();
          onPreviousRange();
          return;
        case "PageDown":
          e.preventDefault();
          e.stopPropagation();
          onNextRange();
          return;
        case "Enter":
        case " ":
          e.preventDefault();
          e.stopPropagation();
          onSelectYear(year);
          return;
      }
      if (newYear !== null) {
        onFocusChange(newYear);
      }
    },
    [years, onSelectYear, onFocusChange, onPreviousRange, onNextRange]
  );
  return /* @__PURE__ */ jsx("div", { ref, className: cn("p-3 w-full", className), children: /* @__PURE__ */ jsx("div", { className: "flex flex-col gap-2 w-full", role: "grid", "aria-label": "Select year", children: Array.from({ length: 5 }, (_, rowIndex) => {
    const rowYears = years.slice(rowIndex * 4, (rowIndex + 1) * 4);
    return /* @__PURE__ */ jsx("div", { role: "row", className: "grid grid-cols-4 gap-2", children: rowYears.map((year) => {
      const isFocused = year === focusedYear;
      const isSelected = year === selectedYear;
      return /* @__PURE__ */ jsx(
        "div",
        {
          ref: isFocused ? focusedCellRef : void 0,
          role: "gridcell",
          "aria-label": `Year ${year}`,
          "aria-selected": isSelected,
          tabIndex: isFocused ? 0 : -1,
          className: cn(
            yearCellVariants({
              variant: isSelected ? "selected" : "default"
            }),
            "flex items-center justify-center cursor-pointer"
          ),
          onClick: () => onSelectYear(year),
          onKeyDown: (e) => handleKeyDown(e, year),
          children: year
        },
        year
      );
    }) }, rowIndex);
  }) }) });
}
var legendItemBoxVariants = cva(
  "h-4 w-4 border border-sapphire-border flex items-center justify-center rounded-sm",
  {
    variants: {
      type: {
        ["None" /* None */]: "",
        ["Today" /* Today */]: "border-2 border-sapphire-border-accent bg-transparent",
        ["Selected" /* Selected */]: "bg-sapphire-brand-hover-background",
        ["Working" /* Working */]: "bg-sapphire-positive-bg",
        ["NonWorking" /* NonWorking */]: "bg-sapphire-negative-bg",
        ["Type01" /* Type01 */]: "bg-blue-500",
        ["Type02" /* Type02 */]: "bg-green-500",
        ["Type03" /* Type03 */]: "bg-yellow-500",
        ["Type04" /* Type04 */]: "bg-red-500",
        ["Type05" /* Type05 */]: "bg-purple-500",
        ["Type06" /* Type06 */]: "bg-pink-500",
        ["Type07" /* Type07 */]: "bg-indigo-500",
        ["Type08" /* Type08 */]: "bg-cyan-500",
        ["Type09" /* Type09 */]: "bg-teal-500",
        ["Type10" /* Type10 */]: "bg-orange-500",
        ["Type11" /* Type11 */]: "bg-lime-500",
        ["Type12" /* Type12 */]: "bg-emerald-500",
        ["Type13" /* Type13 */]: "bg-sky-500",
        ["Type14" /* Type14 */]: "bg-violet-500",
        ["Type15" /* Type15 */]: "bg-fuchsia-500",
        ["Type16" /* Type16 */]: "bg-rose-500",
        ["Type17" /* Type17 */]: "bg-amber-500",
        ["Type18" /* Type18 */]: "bg-slate-500",
        ["Type19" /* Type19 */]: "bg-zinc-500",
        ["Type20" /* Type20 */]: "bg-neutral-500"
      }
    },
    defaultVariants: {
      type: "None" /* None */
    }
  }
);
function CalendarLegendItem({ type, text, onClick, className, ref }) {
  const isSelected = type === "Selected" /* Selected */;
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      role: "listitem",
      tabIndex: -1,
      onClick,
      className: cn(
        "flex items-center gap-2 cursor-pointer",
        "hover:bg-sapphire-bg-secondary rounded-sm p-1 transition-colors",
        "focus:outline-none focus:ring-2 focus:ring-sapphire-ring focus:ring-offset-1",
        className
      ),
      children: [
        /* @__PURE__ */ jsx("div", { className: cn(legendItemBoxVariants({ type })), children: isSelected && /* @__PURE__ */ jsx("div", { className: "h-1.5 w-1.5 rounded-full bg-sapphire-text-on-surface" }) }),
        /* @__PURE__ */ jsx("span", { className: "text-sm text-sapphire-text truncate", children: text })
      ]
    }
  );
}
var getDefaultLegendItemText = (type, _locale) => {
  const labels = {
    ["Today" /* Today */]: "Today",
    ["Selected" /* Selected */]: "Selected Day",
    ["Working" /* Working */]: "Working Day",
    ["NonWorking" /* NonWorking */]: "Non-Working Day"
  };
  return labels[type] || type;
};
function CalendarLegend({
  hideToday = false,
  hideSelectedDay = false,
  hideNonWorkingDay = false,
  hideWorkingDay = false,
  items = [],
  onLegendItemSelect,
  locale,
  className,
  ref
}) {
  const defaultItems = [
    {
      type: "Today" /* Today */,
      text: getDefaultLegendItemText("Today" /* Today */),
      hidden: hideToday
    },
    {
      type: "Selected" /* Selected */,
      text: getDefaultLegendItemText("Selected" /* Selected */),
      hidden: hideSelectedDay
    },
    {
      type: "Working" /* Working */,
      text: getDefaultLegendItemText("Working" /* Working */),
      hidden: hideWorkingDay
    },
    {
      type: "NonWorking" /* NonWorking */,
      text: getDefaultLegendItemText("NonWorking" /* NonWorking */),
      hidden: hideNonWorkingDay
    }
  ].filter((item) => !item.hidden);
  const allItems = [...defaultItems, ...items];
  if (allItems.length === 0) {
    return null;
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      role: "list",
      "aria-label": "Calendar legend",
      className: cn(
        "p-3 bg-sapphire-card-bg border-t border-sapphire-border",
        "columns-2 sm:columns-3 gap-2",
        "w-full",
        className
      ),
      children: allItems.map((item, index) => /* @__PURE__ */ jsx(
        CalendarLegendItem,
        {
          type: item.type,
          text: item.text,
          onClick: onLegendItemSelect ? () => onLegendItemSelect({ type: item.type, text: item.text }) : void 0
        },
        `${item.type}-${index}`
      ))
    }
  );
}
function useCalendarSelection(mode, initialDates, rangeStart, rangeEnd, minDate, maxDate, disabledDates) {
  const [selectedDates, setSelectedDates] = useState(() => {
    if (mode === "Range") return [];
    return initialDates?.map((d) => parseDate(d)) || [];
  });
  const [rangeStartDate, setRangeStartDate] = useState(
    () => rangeStart ? parseDate(rangeStart) : null
  );
  const [rangeEndDate, setRangeEndDate] = useState(
    () => rangeEnd ? parseDate(rangeEnd) : null
  );
  const [hoveredDate, setHoveredDate] = useState(null);
  const minDateObj = minDate ? parseDate(minDate) : null;
  const maxDateObj = maxDate ? parseDate(maxDate) : null;
  const isDateDisabled = useCallback(
    (date) => {
      if (minDateObj && isDateBefore(date, minDateObj)) return true;
      if (maxDateObj && isDateAfter(date, maxDateObj)) return true;
      if (disabledDates) {
        for (const range of disabledDates) {
          const start = range.startDate ? parseDate(range.startDate) : null;
          const end = range.endDate ? parseDate(range.endDate) : null;
          if (start && end) {
            if (isDateInRange(date, start, end)) return true;
          } else if (start) {
            if (isSameDayUtil(date, start) || isDateAfter(date, start)) return true;
          } else if (end) {
            if (isSameDayUtil(date, end) || isDateBefore(date, end)) return true;
          }
        }
      }
      return false;
    },
    [minDateObj, maxDateObj, disabledDates]
  );
  const isDateSelected = useCallback(
    (date) => {
      if (mode === "Range") {
        if (!rangeStartDate) return false;
        if (!rangeEndDate) return isSameDayUtil(date, rangeStartDate);
        return isDateInRange(date, rangeStartDate, rangeEndDate);
      }
      return selectedDates.some((d) => isSameDayUtil(d, date));
    },
    [mode, selectedDates, rangeStartDate, rangeEndDate]
  );
  const isRangeStart = useCallback(
    (date) => {
      if (mode !== "Range" || !rangeStartDate) return false;
      if (!rangeEndDate && hoveredDate) {
        if (isDateBefore(hoveredDate, rangeStartDate)) {
          return isSameDayUtil(date, hoveredDate);
        }
      }
      return isSameDayUtil(date, rangeStartDate);
    },
    [mode, rangeStartDate, rangeEndDate, hoveredDate]
  );
  const isRangeEnd = useCallback(
    (date) => {
      if (mode !== "Range") return false;
      if (rangeEndDate) return isSameDayUtil(date, rangeEndDate);
      if (rangeStartDate && hoveredDate) {
        if (isDateBefore(hoveredDate, rangeStartDate)) {
          return isSameDayUtil(date, rangeStartDate);
        }
        return isSameDayUtil(date, hoveredDate);
      }
      return false;
    },
    [mode, rangeStartDate, rangeEndDate, hoveredDate]
  );
  const isDateBetweenRange = useCallback(
    (date) => {
      if (mode !== "Range" || !rangeStartDate) return false;
      if (rangeEndDate) {
        return isDateInRange(date, rangeStartDate, rangeEndDate) && !isSameDayUtil(date, rangeStartDate) && !isSameDayUtil(date, rangeEndDate);
      }
      if (hoveredDate) {
        const previewStart = isDateBefore(hoveredDate, rangeStartDate) ? hoveredDate : rangeStartDate;
        const previewEnd = isDateBefore(hoveredDate, rangeStartDate) ? rangeStartDate : hoveredDate;
        return isDateInRange(date, previewStart, previewEnd) && !isSameDayUtil(date, previewStart) && !isSameDayUtil(date, previewEnd);
      }
      return false;
    },
    [mode, rangeStartDate, rangeEndDate, hoveredDate]
  );
  const selectDate = useCallback(
    (date) => {
      if (isDateDisabled(date)) return null;
      let newSelectedValues = [];
      let newSelectedTimestamps = [];
      if (mode === "Single") {
        setSelectedDates([date]);
        newSelectedValues = [formatDateToISO(date)];
        newSelectedTimestamps = [Math.floor(date.getTime() / 1e3)];
      } else if (mode === "Multiple") {
        let newDates = [];
        setSelectedDates((prev) => {
          const exists2 = prev.some((d) => isSameDayUtil(d, date));
          if (exists2) {
            newDates = prev.filter((d) => !isSameDayUtil(d, date));
          } else {
            newDates = [...prev, date];
          }
          return newDates;
        });
        const exists = selectedDates.some((d) => isSameDayUtil(d, date));
        if (exists) {
          newDates = selectedDates.filter((d) => !isSameDayUtil(d, date));
        } else {
          newDates = [...selectedDates, date];
        }
        newSelectedValues = newDates.map(formatDateToISO);
        newSelectedTimestamps = newDates.map((d) => Math.floor(d.getTime() / 1e3));
      } else if (mode === "Range") {
        if (!rangeStartDate || rangeStartDate && rangeEndDate) {
          setRangeStartDate(date);
          setRangeEndDate(null);
          setHoveredDate(null);
          newSelectedValues = [formatDateToISO(date)];
          newSelectedTimestamps = [Math.floor(date.getTime() / 1e3)];
        } else {
          let newStart;
          let newEnd;
          if (isDateBefore(date, rangeStartDate)) {
            newStart = date;
            newEnd = rangeStartDate;
            setRangeStartDate(date);
            setRangeEndDate(rangeStartDate);
          } else {
            newStart = rangeStartDate;
            newEnd = date;
            setRangeEndDate(date);
          }
          setHoveredDate(null);
          newSelectedValues = [formatDateToISO(newStart), formatDateToISO(newEnd)];
          newSelectedTimestamps = [
            Math.floor(newStart.getTime() / 1e3),
            Math.floor(newEnd.getTime() / 1e3)
          ];
        }
      }
      return { selectedValues: newSelectedValues, selectedTimestamps: newSelectedTimestamps };
    },
    [mode, isDateDisabled, rangeStartDate, rangeEndDate, selectedDates]
  );
  const getSelectedValuesAsISO = useCallback(() => {
    if (mode === "Range") {
      const values = [];
      if (rangeStartDate) values.push(formatDateToISO(rangeStartDate));
      if (rangeEndDate) values.push(formatDateToISO(rangeEndDate));
      return values;
    }
    return selectedDates.map(formatDateToISO);
  }, [mode, selectedDates, rangeStartDate, rangeEndDate]);
  const getSelectedTimestamps = useCallback(() => {
    if (mode === "Range") {
      const timestamps = [];
      if (rangeStartDate) timestamps.push(Math.floor(rangeStartDate.getTime() / 1e3));
      if (rangeEndDate) timestamps.push(Math.floor(rangeEndDate.getTime() / 1e3));
      return timestamps;
    }
    return selectedDates.map((d) => Math.floor(d.getTime() / 1e3));
  }, [mode, selectedDates, rangeStartDate, rangeEndDate]);
  const isRangeSelectionInProgress = mode === "Range" && rangeStartDate !== null && rangeEndDate === null;
  return {
    selectedDates,
    rangeStartDate,
    rangeEndDate,
    hoveredDate,
    setHoveredDate,
    isRangeSelectionInProgress,
    isDateDisabled,
    isDateSelected,
    isRangeStart,
    isRangeEnd,
    isDateBetweenRange,
    selectDate,
    getSelectedValuesAsISO,
    getSelectedTimestamps
  };
}
function useSpecialDates(specialDates) {
  const getSpecialDate = useCallback(
    (date) => {
      if (!specialDates) return null;
      const found = specialDates.find((sd) => {
        const specialDate = parseDate(sd.date);
        return isSameDayUtil(specialDate, date);
      });
      return found ? { type: found.type, tooltip: found.tooltip } : null;
    },
    [specialDates]
  );
  return { getSpecialDate };
}
function getInitialView(pickersMode) {
  switch (pickersMode) {
    case "YEAR" /* YEAR */:
      return "year" /* Year */;
    case "MONTH_YEAR" /* MONTH_YEAR */:
      return "month" /* Month */;
    default:
      return "day" /* Day */;
  }
}
function Calendar({
  selectionMode = "Single" /* Single */,
  hideWeekNumbers = false,
  calendarWeekNumbering,
  pickersMode = "DAY_MONTH_YEAR" /* DAY_MONTH_YEAR */,
  minDate,
  maxDate,
  selectedDates,
  rangeStartDate,
  rangeEndDate,
  specialDates,
  disabledDates,
  focusedDate,
  locale: localeProp,
  firstDayOfWeek,
  showLegend = false,
  hideLegendToday = false,
  hideLegendSelectedDay = false,
  hideLegendWorkingDay = false,
  hideLegendNonWorkingDay = false,
  legendItems = [],
  autoFocus = false,
  onSelectionChange,
  onShowMonthView,
  onShowYearView,
  onLegendItemSelect,
  className,
  ref
}) {
  const { i18n: i18n2 } = useTranslation("fx");
  const locale = localeProp || i18n2.language;
  const cldrReady = useEnsureCldr(locale);
  const [currentView, setCurrentView] = useState(
    () => getInitialView(pickersMode)
  );
  const [displayedDate, setDisplayedDate] = useState(() => {
    if (focusedDate) return parseDate(focusedDate);
    if (selectedDates && selectedDates.length > 0) return parseDate(selectedDates[0]);
    if (rangeStartDate) return parseDate(rangeStartDate);
    return getToday();
  });
  const [focusDate, setFocusDate] = useState(displayedDate);
  const [yearRangeStart, setYearRangeStart] = useState(() => {
    const year = getYear(displayedDate);
    return Math.floor(year / 20) * 20;
  });
  const minDateObj = useMemo(() => minDate ? parseDate(minDate) : void 0, [minDate]);
  const maxDateObj = useMemo(() => maxDate ? parseDate(maxDate) : void 0, [maxDate]);
  const isPreviousDisabled = useMemo(() => {
    if (!minDateObj) return false;
    if (currentView === "day" /* Day */) {
      const prevMonth = addMonthsToDate(displayedDate, -1);
      const lastDayOfPrevMonth = new Date(getYear(prevMonth), getMonth(prevMonth) + 1, 0);
      return lastDayOfPrevMonth < minDateObj;
    } else if (currentView === "month" /* Month */) {
      return getYear(displayedDate) - 1 < getYear(minDateObj);
    } else if (currentView === "year" /* Year */) {
      return yearRangeStart - 1 < getYear(minDateObj);
    }
    return false;
  }, [minDateObj, currentView, displayedDate, yearRangeStart]);
  const isNextDisabled = useMemo(() => {
    if (!maxDateObj) return false;
    if (currentView === "day" /* Day */) {
      const nextMonth = addMonthsToDate(displayedDate, 1);
      const firstDayOfNextMonth = createDate(getYear(nextMonth), getMonth(nextMonth), 1);
      return firstDayOfNextMonth > maxDateObj;
    } else if (currentView === "month" /* Month */) {
      return getYear(displayedDate) + 1 > getYear(maxDateObj);
    } else if (currentView === "year" /* Year */) {
      return yearRangeStart + 20 > getYear(maxDateObj);
    }
    return false;
  }, [maxDateObj, currentView, displayedDate, yearRangeStart]);
  const selection = useCalendarSelection(
    selectionMode,
    selectedDates,
    rangeStartDate,
    rangeEndDate,
    minDate,
    maxDate,
    disabledDates
  );
  const { getSpecialDate } = useSpecialDates(specialDates);
  const handleDateSelect = useCallback(
    (date) => {
      const result = selection.selectDate(date);
      if (onSelectionChange && result) {
        onSelectionChange({
          selectedValues: result.selectedValues,
          selectedDates: result.selectedTimestamps,
          timestamp: Math.floor(date.getTime() / 1e3)
        });
      }
    },
    [selection, onSelectionChange]
  );
  const handleMonthSelect = useCallback((month) => {
    const newDate = createDate(getYear(displayedDate), month);
    setDisplayedDate(newDate);
    setFocusDate(newDate);
    if (pickersMode === "MONTH_YEAR" /* MONTH_YEAR */) {
      if (onSelectionChange) {
        const isoValue = `${getYear(newDate)}-${String(month + 1).padStart(2, "0")}`;
        onSelectionChange({
          selectedValues: [isoValue],
          selectedDates: [Math.floor(newDate.getTime() / 1e3)],
          timestamp: Math.floor(newDate.getTime() / 1e3)
        });
      }
    } else {
      setCurrentView("day" /* Day */);
    }
  }, [displayedDate, pickersMode, onSelectionChange]);
  const handleYearSelect = useCallback((year) => {
    const newDate = createDate(year, getMonth(displayedDate));
    setDisplayedDate(newDate);
    setFocusDate(newDate);
    if (pickersMode === "YEAR" /* YEAR */) {
      if (onSelectionChange) {
        onSelectionChange({
          selectedValues: [String(year)],
          selectedDates: [Math.floor(newDate.getTime() / 1e3)],
          timestamp: Math.floor(newDate.getTime() / 1e3)
        });
      }
    } else {
      setCurrentView("month" /* Month */);
    }
  }, [displayedDate, pickersMode, onSelectionChange]);
  const handlePrevious = useCallback(() => {
    if (currentView === "day" /* Day */) {
      setDisplayedDate((prev) => addMonthsToDate(prev, -1));
    } else if (currentView === "month" /* Month */) {
      setDisplayedDate((prev) => createDate(getYear(prev) - 1, getMonth(prev)));
    } else if (currentView === "year" /* Year */) {
      setYearRangeStart((prev) => prev - 20);
    }
  }, [currentView]);
  const handleNext = useCallback(() => {
    if (currentView === "day" /* Day */) {
      setDisplayedDate((prev) => addMonthsToDate(prev, 1));
    } else if (currentView === "month" /* Month */) {
      setDisplayedDate((prev) => createDate(getYear(prev) + 1, getMonth(prev)));
    } else if (currentView === "year" /* Year */) {
      setYearRangeStart((prev) => prev + 20);
    }
  }, [currentView]);
  const handleShowMonthView = useCallback(() => {
    if (pickersMode === "YEAR" /* YEAR */) return;
    setCurrentView("month" /* Month */);
    onShowMonthView?.();
  }, [onShowMonthView, pickersMode]);
  const handleShowYearView = useCallback(() => {
    setCurrentView("year" /* Year */);
    onShowYearView?.();
  }, [onShowYearView]);
  const getHeaderText = useCallback(() => {
    const year = getYear(displayedDate);
    const FALLBACK_LONG_MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
    const month = cldrReady ? getMonthName(displayedDate, "long", locale) : FALLBACK_LONG_MONTHS[displayedDate.getMonth()];
    if (currentView === "day" /* Day */) {
      return { monthText: month, yearText: String(year) };
    } else if (currentView === "month" /* Month */) {
      return { monthText: "", yearText: String(year) };
    } else if (currentView === "year" /* Year */) {
      return { monthText: "", yearText: `${yearRangeStart} - ${yearRangeStart + 19}` };
    }
    return { monthText: "", yearText: "" };
  }, [currentView, displayedDate, yearRangeStart, locale, cldrReady]);
  const handleHoverDate = useCallback((date) => {
    selection.setHoveredDate(date);
  }, [selection]);
  const handleFocusChange = useCallback((date) => {
    setFocusDate(date);
    if (getMonth(date) !== getMonth(displayedDate)) {
      setDisplayedDate(date);
    }
  }, [displayedDate]);
  const handleContainerKeyDown = useCallback((e) => {
    if (e.key === "F4") {
      e.preventDefault();
      if (e.shiftKey) {
        handleShowYearView();
      } else {
        handleShowMonthView();
      }
    }
  }, [handleShowMonthView, handleShowYearView]);
  const getTooltipText = useCallback(() => {
    if (currentView === "day" /* Day */) {
      return { previous: "Previous month", next: "Next month" };
    } else if (currentView === "month" /* Month */) {
      return { previous: "Previous year", next: "Next year" };
    } else if (currentView === "year" /* Year */) {
      return { previous: "Previous year range", next: "Next year range" };
    }
    return { previous: "Previous", next: "Next" };
  }, [currentView]);
  const headerText = getHeaderText();
  const tooltipText = getTooltipText();
  const showMonthButton = pickersMode !== "YEAR" /* YEAR */ && pickersMode !== "MONTH_YEAR" /* MONTH_YEAR */ && currentView === "day" /* Day */;
  const rootRef = useRef(null);
  useImperativeHandle(ref, () => ({
    switchToDayPicker() {
      if (pickersMode === "YEAR" /* YEAR */ || pickersMode === "MONTH_YEAR" /* MONTH_YEAR */) return;
      setCurrentView("day" /* Day */);
    },
    switchToMonthPicker() {
      if (pickersMode === "YEAR" /* YEAR */) return;
      setCurrentView("month" /* Month */);
    },
    switchToYearPicker() {
      setCurrentView("year" /* Year */);
    },
    get nativeElement() {
      return rootRef.current;
    }
  }), [pickersMode]);
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: rootRef,
      "data-testid": "calendar",
      onKeyDown: handleContainerKeyDown,
      className: cn(
        "flex flex-col rounded-lg border border-sapphire-border-primary bg-sapphire-card-bg-primary text-foreground shadow-[0_4px_10px_-2px_rgba(10,10,10,0.15),0_2px_5px_-2px_rgba(10,10,10,0.05)]",
        "w-fit min-w-[17rem]",
        className
      ),
      children: [
        /* @__PURE__ */ jsx(
          CalendarHeader,
          {
            monthText: showMonthButton ? headerText.monthText : "",
            yearText: headerText.yearText,
            cldrReady,
            isPreviousDisabled,
            isNextDisabled,
            previousTooltip: tooltipText.previous,
            nextTooltip: tooltipText.next,
            onPrevious: handlePrevious,
            onNext: handleNext,
            onMonthClick: handleShowMonthView,
            onYearClick: handleShowYearView
          }
        ),
        /* @__PURE__ */ jsxs("div", { className: "relative w-full", children: [
          currentView === "day" /* Day */ && /* @__PURE__ */ jsx(
            DayPicker,
            {
              currentMonth: displayedDate,
              focusedDate: focusDate,
              autoFocus,
              isDateSelected: selection.isDateSelected,
              isDateDisabled: selection.isDateDisabled,
              isRangeStart: selection.isRangeStart,
              isRangeEnd: selection.isRangeEnd,
              isDateBetweenRange: selection.isDateBetweenRange,
              isRangeSelectionInProgress: selection.isRangeSelectionInProgress,
              getSpecialDate,
              onSelectDate: handleDateSelect,
              onFocusChange: handleFocusChange,
              onHoverDate: handleHoverDate,
              showWeekNumbers: !hideWeekNumbers,
              weekNumbering: calendarWeekNumbering,
              firstDayOfWeek,
              locale
            }
          ),
          currentView === "month" /* Month */ && /* @__PURE__ */ jsx(
            MonthPicker,
            {
              currentYear: getYear(displayedDate),
              focusedMonth: getMonth(focusDate),
              selectedMonth: getMonth(displayedDate),
              locale,
              onSelectMonth: handleMonthSelect,
              onFocusChange: (month) => {
                const newDate = createDate(getYear(displayedDate), month);
                setFocusDate(newDate);
              },
              onPreviousYear: () => {
                setDisplayedDate((prev) => createDate(getYear(prev) - 1, getMonth(prev)));
              },
              onNextYear: () => {
                setDisplayedDate((prev) => createDate(getYear(prev) + 1, getMonth(prev)));
              }
            }
          ),
          currentView === "year" /* Year */ && (cldrReady ? /* @__PURE__ */ jsx(
            YearPicker,
            {
              startYear: yearRangeStart,
              endYear: yearRangeStart + 19,
              focusedYear: getYear(focusDate),
              selectedYear: getYear(displayedDate),
              onSelectYear: handleYearSelect,
              onFocusChange: (year) => {
                const newDate = createDate(year, getMonth(displayedDate));
                setFocusDate(newDate);
              },
              onPreviousRange: () => setYearRangeStart((prev) => prev - 20),
              onNextRange: () => setYearRangeStart((prev) => prev + 20)
            }
          ) : /* @__PURE__ */ jsx("div", { className: "p-3 w-full", children: /* @__PURE__ */ jsx("div", { className: "flex flex-col gap-2 w-full", children: Array.from({ length: 5 }, (_, row) => /* @__PURE__ */ jsx("div", { className: "grid grid-cols-4 gap-2", children: Array.from({ length: 4 }, (_2, col) => /* @__PURE__ */ jsx("div", { className: "min-h-[2.75rem] w-full flex items-center justify-center", children: /* @__PURE__ */ jsx("span", { className: "inline-block h-3.5 w-8 rounded bg-muted animate-pulse" }) }, col)) }, row)) }) }))
        ] }),
        showLegend && /* @__PURE__ */ jsx(
          CalendarLegend,
          {
            hideToday: hideLegendToday,
            hideSelectedDay: hideLegendSelectedDay,
            hideWorkingDay: hideLegendWorkingDay,
            hideNonWorkingDay: hideLegendNonWorkingDay,
            items: legendItems,
            onLegendItemSelect,
            locale
          }
        )
      ]
    }
  );
}

export { Calendar, CalendarHeader, CalendarLegend, CalendarLegendItem, CalendarLegendItemType, CalendarPickerView, CalendarPickersMode, CalendarSelectionMode, CalendarWeekNumbering, DayPicker, MonthPicker, RTL_LOCALES, YearPicker, addMonthsToDate, addYearsToDate, createDate, dateToTimestamp, ensureCldr, format, formatDateToISO, getLocaleDateFormat, getMonth, getMonthDays, getMonthName, getToday, getWeekEnd, getWeekNumber, getWeekStart, getWeekYearNumber, getYear, initFxI18n, isDateAfter, isDateBefore, isDateInRange, isRTLLocale, isSameDayUtil, isSameMonthUtil, isToday, loadFxTranslations, parseDate, timestampToDate, useDirection, useEnsureCldr };
