// @sap-ui/fx-components/dist/chunk-3GZW6BV2.js
import { OverflowIcon } from './chunk-YHUSMQDT.js';
import { Popover } from './chunk-ZDBE5GTH.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { createContext, useContext, useRef, useState, useImperativeHandle, useMemo, useId, useEffect, useCallback, Children, isValidElement } from 'react';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';

var noop = () => {
};
var ToolbarContext = createContext({
  isInOverflow: false,
  closeOverflow: noop
});
var useToolbarContext = () => useContext(ToolbarContext);

// src/types/toolbar.ts
var ToolbarDesign = /* @__PURE__ */ ((ToolbarDesign2) => {
  ToolbarDesign2["Solid"] = "Solid";
  ToolbarDesign2["Transparent"] = "Transparent";
  return ToolbarDesign2;
})(ToolbarDesign || {});
var ToolbarAlign = /* @__PURE__ */ ((ToolbarAlign2) => {
  ToolbarAlign2["Start"] = "Start";
  ToolbarAlign2["End"] = "End";
  return ToolbarAlign2;
})(ToolbarAlign || {});
var ToolbarItemOverflowBehavior = /* @__PURE__ */ ((ToolbarItemOverflowBehavior2) => {
  ToolbarItemOverflowBehavior2["Default"] = "Default";
  ToolbarItemOverflowBehavior2["NeverOverflow"] = "NeverOverflow";
  ToolbarItemOverflowBehavior2["AlwaysOverflow"] = "AlwaysOverflow";
  return ToolbarItemOverflowBehavior2;
})(ToolbarItemOverflowBehavior || {});
var toolbarVariants = cva(
  ["flex items-center w-full min-h-[40px] px-2 gap-2"],
  {
    variants: {
      design: {
        ["Solid" /* Solid */]: "bg-card border-b border-border",
        ["Transparent" /* Transparent */]: "bg-transparent"
      },
      align: {
        ["Start" /* Start */]: "justify-start",
        ["End" /* End */]: "justify-end"
      }
    },
    defaultVariants: {
      design: "Solid" /* Solid */,
      align: "End" /* End */
    }
  }
);
var flattenChildren = (nodes) => {
  const result = [];
  Children.forEach(nodes, (child) => {
    if (!isValidElement(child)) return;
    if (child.type === React.Fragment) {
      result.push(
        ...flattenChildren(
          child.props.children
        )
      );
    } else {
      result.push(child);
    }
  });
  return result;
};
function getOverflowPriority(child) {
  const props = child.props;
  const val = props.overflowPriority;
  if (val === "NeverOverflow" /* NeverOverflow */ || val === "NeverOverflow")
    return "NeverOverflow" /* NeverOverflow */;
  if (val === "AlwaysOverflow" /* AlwaysOverflow */ || val === "AlwaysOverflow")
    return "AlwaysOverflow" /* AlwaysOverflow */;
  return "Default" /* Default */;
}
function isSeparator(child) {
  return child.type?.displayName === "ToolbarSeparator";
}
function startsWithTitle(child) {
  const typeName = child.type?.displayName ?? child.type?.name;
  if (typeName === "Title") return true;
  if (typeName === "ToolbarItem") {
    const inner = Children.toArray(
      child.props.children
    );
    return inner.some(
      (c) => isValidElement(c) && (c.type?.displayName ?? c.type?.name) === "Title"
    );
  }
  return false;
}
var OVERFLOW_BTN_WIDTH = 40;
function Toolbar({
  design = "Solid" /* Solid */,
  alignContent = "End" /* End */,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  children,
  className,
  style,
  id,
  ref
}) {
  const toolbarRef = useRef(null);
  const measureRef = useRef(null);
  const overflowBtnRef = useRef(null);
  const [containerWidth, setContainerWidth] = useState(0);
  const [itemWidths, setItemWidths] = useState([]);
  const [overflowOpen, setOverflowOpen] = useState(false);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        toolbarRef.current?.focus();
      },
      get nativeElement() {
        return toolbarRef.current;
      }
    }),
    []
  );
  const flatItems = useMemo(() => flattenChildren(children), [children]);
  const descriptionId = useId();
  useEffect(() => {
    const el = toolbarRef.current;
    if (!el) return;
    const observer = new ResizeObserver((entries) => {
      const newWidth = Math.floor(entries[0]?.contentRect.width || 0);
      setContainerWidth((prev) => newWidth !== prev ? newWidth : prev);
    });
    observer.observe(el);
    return () => observer.disconnect();
  }, []);
  useEffect(() => {
    if (flatItems.length === 0) return;
    const frame = requestAnimationFrame(() => {
      const container = measureRef.current;
      if (!container) return;
      const widths = [];
      for (let i = 0; i < flatItems.length; i++) {
        const el = container.querySelector(
          `[data-measure="${i}"]`
        );
        widths.push(el ? el.offsetWidth : 0);
      }
      setItemWidths(widths);
    });
    return () => cancelAnimationFrame(frame);
  }, [flatItems]);
  const overflowedSet = useMemo(() => {
    const set = /* @__PURE__ */ new Set();
    if (containerWidth === 0 || itemWidths.length === 0) return set;
    flatItems.forEach((child, i) => {
      if (getOverflowPriority(child) === "AlwaysOverflow" /* AlwaysOverflow */) {
        set.add(i);
      }
    });
    const GAP = 8;
    let totalNeeded = 0;
    flatItems.forEach((_, i) => {
      if (!set.has(i)) {
        totalNeeded += (itemWidths[i] || 0) + GAP;
      }
    });
    const hasAlwaysOverflow = set.size > 0;
    let available = containerWidth;
    if (hasAlwaysOverflow || totalNeeded > available) {
      available -= OVERFLOW_BTN_WIDTH + GAP;
    }
    if (totalNeeded > available) {
      for (let i = flatItems.length - 1; i >= 0; i--) {
        if (set.has(i)) continue;
        const priority = getOverflowPriority(flatItems[i]);
        if (priority === "NeverOverflow" /* NeverOverflow */) continue;
        if (totalNeeded > available) {
          totalNeeded -= (itemWidths[i] || 0) + GAP;
          set.add(i);
        }
      }
    }
    flatItems.forEach((child, i) => {
      if (!isSeparator(child) || set.has(i)) return;
      let hasPrev = false;
      for (let j = i - 1; j >= 0; j--) {
        if (!set.has(j) && !isSeparator(flatItems[j])) {
          hasPrev = true;
          break;
        }
        if (set.has(j)) break;
      }
      let hasNext = false;
      for (let j = i + 1; j < flatItems.length; j++) {
        if (!set.has(j) && !isSeparator(flatItems[j])) {
          hasNext = true;
          break;
        }
        if (set.has(j)) break;
      }
      if (!hasPrev || !hasNext) {
        set.add(i);
      }
    });
    return set;
  }, [containerWidth, itemWidths, flatItems]);
  const visibleItems = flatItems.filter((_, i) => !overflowedSet.has(i));
  const overflowedItems = flatItems.filter((_, i) => overflowedSet.has(i));
  const hasOverflow = overflowedItems.length > 0;
  const closeOverflow = useCallback(() => {
    setOverflowOpen(false);
  }, []);
  const handleOverflowClick = useCallback(() => {
    setOverflowOpen((prev) => !prev);
  }, []);
  const interactiveCount = flatItems.filter((child) => {
    const dn = child.type?.displayName;
    return dn === "ToolbarButton" || dn === "ToolbarSelect" || dn === "ToolbarItem";
  }).length;
  const role = interactiveCount > 1 ? "toolbar" : void 0;
  const hasLeadingTitle = flatItems.length > 0 && startsWithTitle(flatItems[0]);
  const getFocusableItems = useCallback(() => {
    const el = toolbarRef.current;
    if (!el) return [];
    return Array.from(
      el.querySelectorAll(
        'button:not([disabled]), [role="combobox"]:not([disabled]), input:not([disabled]), select:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])'
      )
    ).filter((node) => {
      return !node.closest("[data-measure]") && !node.closest("[aria-hidden='true']");
    });
  }, []);
  const handleKeyDown = useCallback(
    (e) => {
      const items = getFocusableItems();
      if (items.length === 0) return;
      const currentIndex = items.indexOf(
        document.activeElement
      );
      let nextIndex = null;
      switch (e.key) {
        case "ArrowRight":
          e.preventDefault();
          nextIndex = currentIndex < 0 ? 0 : (currentIndex + 1) % items.length;
          break;
        case "ArrowLeft":
          e.preventDefault();
          nextIndex = currentIndex < 0 ? items.length - 1 : (currentIndex - 1 + items.length) % items.length;
          break;
        case "Home":
          e.preventDefault();
          nextIndex = 0;
          break;
        case "End":
          e.preventDefault();
          nextIndex = items.length - 1;
          break;
      }
      if (nextIndex !== null) {
        items[nextIndex].focus();
      }
    },
    [getFocusableItems]
  );
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: toolbarRef,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-describedby": [
          accessibleDescriptionRef,
          accessibleDescription ? descriptionId : null
        ].filter(Boolean).join(" ") || void 0,
        id,
        style,
        onKeyDown: handleKeyDown,
        className: cn(
          toolbarVariants({
            design,
            align: alignContent
          }),
          hasLeadingTitle && "ps-4",
          className
        ),
        children: [
          visibleItems,
          hasOverflow && /* @__PURE__ */ jsx("div", { ref: overflowBtnRef, className: "shrink-0", children: /* @__PURE__ */ jsx(
            Button,
            {
              design: "SecondaryNeutral",
              size: "Medium",
              iconOnly: true,
              icon: /* @__PURE__ */ jsx(OverflowIcon, { className: "h-4 w-4" }),
              accessibleName: "More actions",
              onClick: handleOverflowClick,
              accessibilityAttributes: {
                expanded: overflowOpen,
                hasPopup: "menu"
              }
            }
          ) }),
          accessibleDescription && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription })
        ]
      }
    ),
    hasOverflow && /* @__PURE__ */ jsx(
      Popover,
      {
        open: overflowOpen,
        opener: overflowBtnRef,
        placement: "Bottom",
        horizontalAlign: "End",
        hideArrow: true,
        noPadding: true,
        onClose: closeOverflow,
        accessibleRole: "None",
        children: /* @__PURE__ */ jsx(
          ToolbarContext.Provider,
          {
            value: { isInOverflow: true, closeOverflow },
            children: /* @__PURE__ */ jsx(
              "div",
              {
                className: "flex flex-col w-56 px-sapphire-2xs",
                children: overflowedItems
              }
            )
          }
        )
      }
    ),
    /* @__PURE__ */ jsx(
      "div",
      {
        ref: measureRef,
        "aria-hidden": "true",
        className: "absolute invisible -top-[9999px] -left-[9999px] flex items-center gap-2 w-fit",
        style: { pointerEvents: "none" },
        children: /* @__PURE__ */ jsx(ToolbarContext.Provider, { value: { isInOverflow: false, closeOverflow }, children: flatItems.map((child, i) => /* @__PURE__ */ jsx("div", { "data-measure": i, className: "shrink-0", children: child }, i)) })
      }
    )
  ] });
}

export { Toolbar, ToolbarAlign, ToolbarContext, ToolbarDesign, ToolbarItemOverflowBehavior, useToolbarContext };
