// @sap-ui/fx-components/dist/chunk-E4NK6IRO.js
import { OverflowIcon } from './chunk-YHUSMQDT.js';
import { ListItemBase } from './chunk-CTYK4HEK.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import * as React from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';

function NavArrowIcon({ className }) {
  return /* @__PURE__ */ jsx("svg", { xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", className, children: /* @__PURE__ */ jsx("path", { d: "M5.20493 3.23532C5.48927 2.93424 5.96333 2.92076 6.2645 3.20504L10.7645 7.45211C10.9144 7.59365 10.9997 7.79085 10.9998 7.99704C10.9999 8.20338 10.9145 8.40121 10.7645 8.54293L6.2645 12.7949C5.96343 13.0792 5.48937 13.0656 5.20493 12.7646C4.92058 12.4635 4.93419 11.9895 5.2352 11.705L9.15707 7.99704L5.2352 4.29489C4.93413 4.01054 4.92064 3.53648 5.20493 3.23532Z", fill: "currentColor" }) });
}
var highlightBarVariants = cva(
  "absolute left-0 top-0 bottom-0 w-[3px]",
  {
    variants: {
      highlight: {
        ["None" /* None */]: "bg-transparent",
        ["Positive" /* Positive */]: "bg-sapphire-positive",
        ["Critical" /* Critical */]: "bg-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info",
        ["Negative" /* Negative */]: "bg-sapphire-negative"
      }
    },
    defaultVariants: {
      highlight: "None" /* None */
    }
  }
);
var additionalTextVariants = cva(
  "text-xs shrink-0",
  {
    variants: {
      state: {
        ["None" /* None */]: "text-sapphire-text-tertiary",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Negative" /* Negative */]: "text-sapphire-negative"
      }
    },
    defaultVariants: {
      state: "None" /* None */
    }
  }
);
function ListItem(props) {
  const {
    ref,
    // Content
    text,
    description,
    icon,
    iconEnd = false,
    additionalText,
    additionalTextState = "None" /* None */,
    image,
    action,
    children,
    // Behavior
    type = "Active" /* Active */,
    highlight = "None" /* None */,
    navigated = false,
    wrappingType = "None" /* None */,
    tooltip,
    // Events
    onDetailClick,
    onDelete: _onDelete,
    // Custom elements
    deleteButton,
    // Rest passed to ListItemBase
    ...baseProps
  } = props;
  const shouldWrap = String(wrappingType) === "Normal";
  const displayIcon = icon && !iconEnd;
  const displayIconEnd = icon && iconEnd;
  const handleDetailButtonClick = React.useCallback((detail) => {
    onDetailClick?.(detail.originalEvent);
  }, [onDetailClick]);
  const showNavArrow = String(type) === "Navigation" || navigated;
  return /* @__PURE__ */ jsx(
    ListItemBase,
    {
      ref,
      type,
      deleteButton,
      ...baseProps,
      children: () => /* @__PURE__ */ jsxs(Fragment, { children: [
        String(highlight) !== "None" && /* @__PURE__ */ jsx("span", { className: highlightBarVariants({ highlight }) }),
        /* @__PURE__ */ jsxs("span", { className: "flex-1 flex items-center px-[0.75rem] py-[0.5rem] gap-[0.75rem] min-w-0", children: [
          image && /* @__PURE__ */ jsx("span", { className: "shrink-0 size-8 max-w-8 max-h-8 overflow-hidden flex items-center justify-center", children: image }),
          displayIcon && /* @__PURE__ */ jsx("span", { className: "shrink-0 text-sapphire-text-tertiary", children: icon }),
          /* @__PURE__ */ jsxs("span", { className: "flex-1 flex flex-col justify-center min-w-0 gap-0.5", children: [
            /* @__PURE__ */ jsx("span", { className: "flex items-center gap-2", children: /* @__PURE__ */ jsx(
              "span",
              {
                className: cn(
                  "text-sm font-semibold text-sapphire-text-primary",
                  !shouldWrap && "truncate",
                  shouldWrap && "break-words"
                ),
                title: tooltip,
                children: children || text
              }
            ) }),
            description && /* @__PURE__ */ jsx(
              "span",
              {
                className: cn(
                  "text-xs text-sapphire-text-tertiary",
                  !shouldWrap && "truncate",
                  shouldWrap && "break-words"
                ),
                children: description
              }
            )
          ] }),
          additionalText && /* @__PURE__ */ jsx("span", { className: cn(
            additionalTextVariants({ state: additionalTextState }),
            "shrink-0"
          ), children: additionalText }),
          displayIconEnd && /* @__PURE__ */ jsx("span", { className: "shrink-0 text-sapphire-text-tertiary", children: icon }),
          String(type) === "Detail" && /* @__PURE__ */ jsx("span", { className: "shrink-0", children: /* @__PURE__ */ jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              iconOnly: true,
              icon: /* @__PURE__ */ jsx(OverflowIcon, { className: "h-4 w-4" }),
              onClick: handleDetailButtonClick,
              accessibleName: "Show details",
              tooltip: "Details"
            }
          ) }),
          (action || showNavArrow) && /* @__PURE__ */ jsxs("span", { className: "shrink-0 flex items-center gap-[0.5rem]", children: [
            action && /* @__PURE__ */ jsx(
              "span",
              {
                onClick: (e) => e.stopPropagation(),
                onKeyDown: (e) => e.stopPropagation(),
                children: action
              }
            ),
            showNavArrow && /* @__PURE__ */ jsx(NavArrowIcon, { className: "text-sapphire-text-tertiary" })
          ] })
        ] })
      ] })
    }
  );
}
ListItem.displayName = "ListItem";

export { ListItem };
