// @sap-ui/fx-components/dist/chunk-4C7TQYID.js
import { ArrowLeftIcon } from './chunk-COC3EGPV.js';
import { MessageErrorIcon, MessageSuccessIcon, MessageWarningIcon, MessageInformationIcon } from './chunk-WDT5ZCHE.js';
import { addOpenedPopup, removeOpenedPopup, isPhone } from './chunk-YSPWI6FA.js';
import { ResizeCornerIcon } from './chunk-GCJAV5G5.js';
import { Title } from './chunk-BM2JFIU4.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { __publicField } from './chunk-V6TY7KAL.js';
import * as React from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/dialog.ts
var DialogState = /* @__PURE__ */ ((DialogState2) => {
  DialogState2["None"] = "None";
  DialogState2["Error"] = "Error";
  DialogState2["Warning"] = "Warning";
  DialogState2["Success"] = "Success";
  DialogState2["Information"] = "Information";
  return DialogState2;
})(DialogState || {});

// src/lib/dialogStack.ts
var DialogStackManager = class {
  constructor() {
    __publicField(this, "stack", []);
    __publicField(this, "baseZIndex", 50);
  }
  createBackdrop(onBackdropClick) {
    const el = document.createElement("div");
    el.setAttribute("aria-hidden", "true");
    el.setAttribute("data-dialog-backdrop", "true");
    el.setAttribute("popover", "manual");
    el.style.cssText = "position:fixed;inset:0;pointer-events:all;background:transparent;width:100vw;height:100vh;";
    const stop = (e) => e.stopPropagation();
    el.addEventListener("pointerdown", stop);
    el.addEventListener("pointerup", stop);
    const handleClick = (e) => {
      e.stopPropagation();
      onBackdropClick?.();
    };
    el.addEventListener("click", handleClick);
    document.body.appendChild(el);
    el.showPopover();
    const cleanup = () => {
      el.removeEventListener("pointerdown", stop);
      el.removeEventListener("pointerup", stop);
      el.removeEventListener("click", handleClick);
    };
    return { el, cleanup };
  }
  destroyBackdrop(item) {
    item.cleanupBackdrop();
    try {
      item.backdropEl.hidePopover();
    } catch {
    }
    item.backdropEl.remove();
  }
  /**
   * Register a new dialog on the stack
   */
  push(id, onBackdropClick) {
    const { el: backdropEl, cleanup: cleanupBackdrop } = this.createBackdrop(onBackdropClick);
    this.stack.push({ id, backdropEl, cleanupBackdrop });
    return this.baseZIndex + this.stack.length;
  }
  /**
   * Remove a dialog from the stack
   */
  pop(id) {
    const index = this.stack.findIndex((item2) => item2.id === id);
    if (index === -1) return;
    const item = this.stack[index];
    this.stack.splice(index, 1);
    this.destroyBackdrop(item);
  }
  /**
   * Get z-index for a specific dialog
   */
  getZIndex(id) {
    const index = this.stack.findIndex((item) => item.id === id);
    if (index === -1) return this.baseZIndex;
    return this.baseZIndex + index + 1;
  }
  /**
   * Check if dialog is the topmost
   */
  isTop(id) {
    if (this.stack.length === 0) return false;
    return this.stack[this.stack.length - 1].id === id;
  }
  /**
   * Get the number of open dialogs
   */
  getCount() {
    return this.stack.length;
  }
};
var dialogStack = new DialogStackManager();

// src/lib/focusUtils.ts
function isElementHidden(element) {
  if (element.hidden) {
    return true;
  }
  const style = window.getComputedStyle(element);
  if (style.display === "none" || style.visibility === "hidden") {
    return true;
  }
  return false;
}
function isScrollable(element) {
  const style = window.getComputedStyle(element);
  const overflowX = style.overflowX;
  const overflowY = style.overflowY;
  return (overflowX === "scroll" || overflowX === "auto" || overflowY === "scroll" || overflowY === "auto") && (element.scrollHeight > element.clientHeight || element.scrollWidth > element.clientWidth);
}
function isElementClickable(element) {
  if (element.disabled) {
    return false;
  }
  const tabindex = element.getAttribute("tabindex");
  if (tabindex && parseInt(tabindex, 10) < 0) {
    return false;
  }
  const clickableTags = ["BUTTON", "A", "INPUT", "SELECT", "TEXTAREA"];
  if (clickableTags.includes(element.tagName)) {
    return true;
  }
  if (element.onclick || element.hasAttribute("onclick")) {
    return true;
  }
  if (tabindex && parseInt(tabindex, 10) >= 0) {
    return true;
  }
  return false;
}
function isElemFocusable(element) {
  if (isElementHidden(element)) {
    return false;
  }
  return true;
}
function getChildrenForTraversal(element) {
  const children = [];
  if (element.shadowRoot) {
    children.push(...Array.from(element.shadowRoot.children));
  }
  if (element instanceof HTMLSlotElement) {
    const assigned = element.assignedElements({ flatten: true });
    children.push(...assigned);
  } else {
    children.push(...Array.from(element.children));
  }
  return children.filter((child) => child.nodeType === 1);
}
function findFocusableElementRecursive(container, forward = true, startFromContainer = false) {
  if (startFromContainer && isElemFocusable(container) && isElementClickable(container)) {
    return container;
  }
  const children = getChildrenForTraversal(container);
  const traverseOrder = forward ? children : children.reverse();
  for (const child of traverseOrder) {
    if (isElemFocusable(child) && isElementClickable(child)) {
      const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
      if (isSafari && isScrollable(child) && !isElementClickable(child)) {
        continue;
      }
      return child;
    }
    const found = findFocusableElementRecursive(child, forward, false);
    if (found) {
      return found;
    }
  }
  return null;
}
function findFirstFocusableElement(container, initialFocusId) {
  if (!container) return null;
  if (initialFocusId) {
    const targetElement = document.getElementById(initialFocusId);
    if (targetElement && container.contains(targetElement) && isElemFocusable(targetElement) && isElementClickable(targetElement)) {
      return targetElement;
    }
  }
  return findFocusableElementRecursive(container, true, false);
}
function useIsPhone() {
  const [phone, setPhone] = React.useState(() => isPhone());
  React.useEffect(() => {
    const handler = () => setPhone(isPhone());
    window.addEventListener("resize", handler);
    window.addEventListener("orientationchange", handler);
    return () => {
      window.removeEventListener("resize", handler);
      window.removeEventListener("orientationchange", handler);
    };
  }, []);
  return phone;
}
var dialogVariants = cva(
  [
    "relative flex flex-col",
    "bg-sapphire-card-bg-primary rounded-2xl",
    "shadow-[0_20px_25px_-5px_rgba(10,10,10,0.15),0_10px_15px_-5px_rgba(10,10,10,0.05)]",
    "min-w-[320px] max-w-[554px]",
    "max-h-[90vh]",
    "outline-none",
    "p-0 m-0",
    "border border-border"
  ],
  {
    variants: {
      state: {
        ["None" /* None */]: "",
        ["Error" /* Error */]: "",
        ["Warning" /* Warning */]: "",
        ["Success" /* Success */]: "",
        ["Information" /* Information */]: ""
      },
      stretch: {
        // Desktop stretch: ~90% viewport, no rounding. Phone full-screen handled via JS (isPhone()).
        true: "max-w-[95vw] h-[90vh] w-[90%]",
        false: ""
      },
      resizable: {
        true: "overflow-hidden max-w-[95vw] max-h-[95vh]",
        false: ""
      }
    },
    defaultVariants: {
      state: "None" /* None */,
      stretch: false,
      resizable: false
    }
  }
);
var headerVariants = cva(
  [
    "flex items-center justify-between",
    "px-6 py-5",
    "shrink-0",
    "focus:outline-none focus:rounded-t-2xl focus:[box-shadow:inset_0_0_0_4px_white,inset_0_0_0_6px_var(--ring)]"
  ],
  {
    variants: {
      state: {
        ["None" /* None */]: "border-sapphire-border-primary",
        ["Error" /* Error */]: "bg-sapphire-negative-bg border-sapphire-negative",
        ["Warning" /* Warning */]: "bg-sapphire-warning-bg border-sapphire-warning",
        ["Success" /* Success */]: "bg-sapphire-positive-bg border-sapphire-positive",
        ["Information" /* Information */]: "bg-sapphire-info-bg border-sapphire-info"
      },
      draggable: {
        true: "cursor-move select-none",
        false: ""
      }
    },
    defaultVariants: {
      state: "None" /* None */,
      draggable: false
    }
  }
);
function Dialog({
  className,
  open: controlledOpen,
  defaultOpen = false,
  headerText,
  subHeaderText,
  headerEndContent,
  showBackButton,
  onBackButtonClick,
  hideHeaderBorder,
  header,
  children,
  footer,
  state = "None" /* None */,
  stretch = false,
  draggable = false,
  resizable = false,
  preventInitialFocus = false,
  enableBackdropClick = false,
  noPadding = false,
  initialFocus,
  accessibleName,
  accessibleNameRef,
  accessibleDescribedBy,
  accessibleRole,
  ariaRoleDescription,
  onBeforeOpen,
  onAfterOpen,
  onBeforeClose,
  onAfterClose,
  onEscapePress,
  onOpenChange,
  id,
  ref
}) {
  const [internalOpen, setInternalOpen] = React.useState(defaultOpen);
  const isOpen = controlledOpen ?? internalOpen;
  const onPhone = useIsPhone();
  const [isPopoverActuallyOpen, setIsPopoverActuallyOpen] = React.useState(false);
  const dialogId = React.useId();
  const stableId = id || dialogId;
  const dialogRef = React.useRef(null);
  const headerRef = React.useRef(null);
  const contentRef = React.useRef(null);
  const previousActiveElementRef = React.useRef(null);
  const returnValueRef = React.useRef(void 0);
  const isClosingRef = React.useRef(false);
  const didPopRef = React.useRef(false);
  const [isDragging, setIsDragging] = React.useState(false);
  const [position, setPosition] = React.useState(null);
  const positionRef = React.useRef(position);
  React.useLayoutEffect(() => {
    positionRef.current = position;
  }, [position]);
  const dragStartRef = React.useRef({ x: 0, y: 0, dialogX: 0, dialogY: 0 });
  const effectiveRole = React.useMemo(() => {
    if (state === "Error" /* Error */ || state === "Warning" /* Warning */) {
      return "alertdialog";
    }
    return accessibleRole || "dialog";
  }, [state, accessibleRole]);
  const effectiveAriaDescribedBy = React.useMemo(() => {
    const parts = [];
    if (accessibleDescribedBy) {
      parts.push(accessibleDescribedBy);
    }
    if (draggable || resizable) {
      parts.push(`${stableId}-movement-descr`);
    }
    return parts.length > 0 ? parts.join(" ") : void 0;
  }, [accessibleDescribedBy, draggable, resizable, stableId]);
  const getMovementDescription = () => {
    if (draggable && resizable) {
      return "This dialog can be moved and resized. Use Tab to focus the header, then arrow keys to move. Use Shift+Arrow keys to resize.";
    }
    if (draggable) {
      return "This dialog can be moved. Use Tab to focus the header, then arrow keys to reposition.";
    }
    if (resizable) {
      return "This dialog can be resized. Use Tab to focus the header, then Shift+Arrow keys to resize.";
    }
    return "";
  };
  const ensureDirectPositioning = React.useCallback(() => {
    const dialog = dialogRef.current;
    if (!dialog || positionRef.current) return positionRef.current;
    const rect = dialog.getBoundingClientRect();
    dialog.style.width = `${rect.width}px`;
    const pos = { x: rect.left, y: rect.top };
    setPosition(pos);
    return pos;
  }, []);
  const clampPosition = React.useCallback(
    (x, y) => {
      const dialog = dialogRef.current;
      if (!dialog) return { x, y };
      const w = dialog.offsetWidth;
      const h = dialog.offsetHeight;
      const vw = window.innerWidth;
      const vh = window.innerHeight;
      if (!w || !h || !vw || !vh) return { x, y };
      return {
        x: Math.max(0, Math.min(vw - w, x)),
        y: Math.max(0, Math.min(vh - h, y))
      };
    },
    []
  );
  const handleClose = React.useCallback((escPressed) => {
    const dialog = dialogRef.current;
    if (!dialog) return;
    if (isClosingRef.current) return;
    isClosingRef.current = true;
    const beforeCloseDetail = {
      escPressed,
      returnValue: returnValueRef.current
    };
    const shouldClose = onBeforeClose?.(beforeCloseDetail);
    if (shouldClose === false) {
      isClosingRef.current = false;
      if (controlledOpen === void 0) {
        setInternalOpen(true);
      } else {
        onOpenChange?.(true);
      }
      return;
    }
    if (dialog.matches(":popover-open")) {
      dialog.hidePopover();
      if (!didPopRef.current) {
        didPopRef.current = true;
        dialogStack.pop(stableId);
      }
      setIsPopoverActuallyOpen(false);
    }
    setPosition(null);
    dialog.style.width = "";
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    onOpenChange?.(false);
    const afterCloseDetail = {
      returnValue: returnValueRef.current
    };
    setTimeout(() => {
      onAfterClose?.(afterCloseDetail);
      previousActiveElementRef.current?.focus();
      returnValueRef.current = void 0;
      isClosingRef.current = false;
      didPopRef.current = false;
    }, 200);
  }, [controlledOpen, onBeforeClose, onAfterClose, onOpenChange, stableId]);
  React.useEffect(() => {
    if (!isPopoverActuallyOpen) return;
    const dialog = dialogRef.current;
    if (!dialog) return;
    const resizeObserver = new ResizeObserver(() => {
      if (!dialogRef.current) return;
      if (resizable || draggable) {
        const pos = positionRef.current ?? ensureDirectPositioning();
        if (!pos) return;
        const dialogRect = dialogRef.current.getBoundingClientRect();
        const vw = window.innerWidth;
        const vh = window.innerHeight;
        const newX = Math.max(0, Math.min(vw - dialogRect.width, pos.x));
        const newY = Math.max(0, Math.min(vh - dialogRect.height, pos.y));
        if (newX !== pos.x || newY !== pos.y) {
          setPosition({ x: newX, y: newY });
        }
      }
    });
    resizeObserver.observe(dialog);
    return () => {
      resizeObserver.disconnect();
    };
  }, [isPopoverActuallyOpen, draggable, resizable, ensureDirectPositioning]);
  React.useEffect(() => {
    if (!isPopoverActuallyOpen) return;
    const handleResize = () => {
      const dialog = dialogRef.current;
      if (dialog) {
        dialog.style.width = "";
      }
      setPosition(null);
    };
    window.addEventListener("resize", handleResize);
    return () => window.removeEventListener("resize", handleResize);
  }, [isPopoverActuallyOpen]);
  const isPopoverActuallyOpenRef = React.useRef(isPopoverActuallyOpen);
  React.useEffect(() => {
    isPopoverActuallyOpenRef.current = isPopoverActuallyOpen;
  }, [isPopoverActuallyOpen]);
  const onAfterCloseRef = React.useRef(onAfterClose);
  React.useEffect(() => {
    onAfterCloseRef.current = onAfterClose;
  }, [onAfterClose]);
  const handleCloseRef = React.useRef(handleClose);
  React.useEffect(() => {
    handleCloseRef.current = handleClose;
  }, [handleClose]);
  React.useEffect(() => {
    const dialog = dialogRef.current;
    const id2 = stableId;
    return () => {
      const isOpen2 = dialog?.matches(":popover-open");
      if (isOpen2) {
        try {
          dialog.hidePopover();
        } catch {
        }
      }
      if ((isOpen2 || isPopoverActuallyOpenRef.current) && !didPopRef.current) {
        didPopRef.current = true;
        dialogStack.pop(id2);
        onAfterCloseRef.current?.({ returnValue: returnValueRef.current });
      }
    };
  }, [stableId]);
  React.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog) return;
    const isPopoverOpen = dialog.matches(":popover-open");
    if (isOpen && !isPopoverOpen) {
      isClosingRef.current = false;
      didPopRef.current = false;
      previousActiveElementRef.current = document.activeElement;
      const beforeOpenDetail = {};
      const shouldOpen = onBeforeOpen?.(beforeOpenDetail);
      if (shouldOpen === false) {
        if (controlledOpen === void 0) {
          setInternalOpen(false);
        } else {
          onOpenChange?.(false);
        }
        return;
      }
      dialogStack.push(stableId, enableBackdropClick ? () => handleCloseRef.current(false) : void 0);
      dialog.showPopover();
      setIsPopoverActuallyOpen(true);
      requestAnimationFrame(() => {
        requestAnimationFrame(() => {
          if (!preventInitialFocus) {
            const firstFocusable = findFirstFocusableElement(dialog, initialFocus);
            if (firstFocusable) {
              firstFocusable.focus();
            } else {
              dialog.focus();
            }
          }
        });
      });
      const timer = setTimeout(() => {
        onAfterOpen?.();
      }, 200);
      return () => clearTimeout(timer);
    } else if (!isOpen && isPopoverOpen) {
      handleClose(false);
    }
    if (!isPopoverOpen && isPopoverActuallyOpen) {
      setIsPopoverActuallyOpen(false);
    }
  }, [
    isOpen,
    handleClose,
    initialFocus,
    preventInitialFocus,
    onBeforeOpen,
    onAfterOpen,
    isPopoverActuallyOpen,
    controlledOpen,
    onOpenChange,
    stableId,
    enableBackdropClick
  ]);
  React.useEffect(() => {
    const dialog = dialogRef.current;
    if (!isPopoverActuallyOpen || !dialog) return;
    addOpenedPopup({
      element: dialog,
      close: (escPressed) => {
        if (escPressed) onEscapePress?.();
        handleClose(!!escPressed);
      },
      type: "dialog",
      isModal: true
    });
    return () => removeOpenedPopup(dialog);
  }, [isPopoverActuallyOpen, handleClose, onEscapePress]);
  React.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog || !isPopoverActuallyOpen) return;
    const handleKeyDown = (e) => {
      if (e.key === "Escape") return;
      if (e.key === " " || ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"].includes(e.key)) {
        const activeElement = document.activeElement;
        if (activeElement === dialog || activeElement === document.body) {
          e.preventDefault();
        }
      }
      if (e.key === "Tab") {
        const focusableElements = dialog?.querySelectorAll(
          'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
        );
        if (!focusableElements || focusableElements.length === 0) return;
        const firstElement = focusableElements[0];
        const lastElement = focusableElements[focusableElements.length - 1];
        if (e.shiftKey) {
          if (document.activeElement === firstElement) {
            e.preventDefault();
            lastElement.focus();
          }
        } else {
          if (document.activeElement === lastElement) {
            e.preventDefault();
            firstElement.focus();
          }
        }
      }
      if (["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"].includes(e.key)) {
        const activeElement = document.activeElement;
        const isInput = activeElement instanceof HTMLInputElement || activeElement instanceof HTMLTextAreaElement || activeElement?.getAttribute("contenteditable") === "true";
        if (isInput) return;
        const isHeaderFocused = headerRef.current?.contains(activeElement);
        const shouldResize = e.shiftKey && resizable && isHeaderFocused;
        const shouldDrag = !e.shiftKey && draggable && isHeaderFocused;
        if (!shouldDrag && !shouldResize) {
          return;
        }
        e.preventDefault();
        const step = 10;
        let deltaX = 0;
        let deltaY = 0;
        switch (e.key) {
          case "ArrowUp":
            deltaY = -step;
            break;
          case "ArrowDown":
            deltaY = step;
            break;
          case "ArrowLeft":
            deltaX = -step;
            break;
          case "ArrowRight":
            deltaX = step;
            break;
        }
        if (shouldDrag) {
          const base = positionRef.current ?? ensureDirectPositioning() ?? { x: 0, y: 0 };
          setPosition(clampPosition(base.x + deltaX, base.y + deltaY));
        } else if (shouldResize && dialog) {
          const currentWidth = dialog.offsetWidth;
          const currentHeight = dialog.offsetHeight;
          const isRTL = document.documentElement.dir === "rtl";
          const base = positionRef.current ?? ensureDirectPositioning() ?? { x: 0, y: 0 };
          const effectiveDeltaX = isRTL ? -deltaX : deltaX;
          const newWidth = Math.max(200, currentWidth + effectiveDeltaX * 2);
          const newHeight = Math.max(150, currentHeight + deltaY * 2);
          dialog.style.width = `${newWidth}px`;
          dialog.style.height = `${newHeight}px`;
          const widthDiff = newWidth - currentWidth;
          const newX = isRTL ? base.x - widthDiff : base.x;
          setPosition(clampPosition(newX, base.y));
        }
      }
    };
    dialog.addEventListener("keydown", handleKeyDown);
    return () => {
      dialog.removeEventListener("keydown", handleKeyDown);
    };
  }, [isPopoverActuallyOpen, draggable, resizable, clampPosition, ensureDirectPositioning]);
  React.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog || !isPopoverActuallyOpen) return;
    const handleFocusOut = (e) => {
      const newFocusTarget = e.relatedTarget;
      if (!newFocusTarget || !dialog.contains(newFocusTarget)) {
        requestAnimationFrame(() => {
          const currentFocus = document.activeElement;
          if (!dialog.contains(currentFocus)) {
            dialog.focus();
          }
        });
      }
    };
    dialog.addEventListener("focusout", handleFocusOut);
    return () => {
      dialog.removeEventListener("focusout", handleFocusOut);
    };
  }, [isPopoverActuallyOpen]);
  const handleMouseDown = React.useCallback(
    (e) => {
      if (!draggable) return;
      if (stretch && onPhone) return;
      if (!headerRef.current?.contains(e.target)) return;
      const pos = position ?? ensureDirectPositioning();
      setIsDragging(true);
      dragStartRef.current = {
        x: e.clientX,
        y: e.clientY,
        dialogX: pos?.x ?? 0,
        dialogY: pos?.y ?? 0
      };
    },
    [draggable, stretch, onPhone, position, ensureDirectPositioning]
  );
  const handleMouseMove = React.useCallback(
    (e) => {
      if (!isDragging) return;
      const deltaX = e.clientX - dragStartRef.current.x;
      const deltaY = e.clientY - dragStartRef.current.y;
      const rawX = dragStartRef.current.dialogX + deltaX;
      const rawY = dragStartRef.current.dialogY + deltaY;
      setPosition(clampPosition(rawX, rawY));
    },
    [isDragging, clampPosition]
  );
  const handleMouseUp = React.useCallback(() => {
    setIsDragging(false);
  }, []);
  const resizeState = React.useRef(null);
  const handleResizeMouseDown = React.useCallback(
    (e) => {
      if (!resizable) return;
      if (stretch && onPhone) return;
      e.preventDefault();
      e.stopPropagation();
      const dialog = dialogRef.current;
      if (!dialog) return;
      const pos = positionRef.current ?? ensureDirectPositioning();
      const rect = dialog.getBoundingClientRect();
      const rtl = document.documentElement.dir === "rtl";
      resizeState.current = {
        startX: e.clientX,
        startY: e.clientY,
        startWidth: rect.width,
        startHeight: rect.height,
        startLeft: pos?.x ?? rect.left,
        isRTL: rtl
      };
      const onMove = (ev) => {
        if (!resizeState.current || !dialogRef.current) return;
        const { startX, startY, startWidth, startHeight, startLeft, isRTL } = resizeState.current;
        const dx = ev.clientX - startX;
        const dy = ev.clientY - startY;
        let newWidth;
        let newLeft;
        if (isRTL) {
          newWidth = Math.max(200, Math.min(startWidth - dx, window.innerWidth * 0.95));
          newLeft = startLeft + dx;
          if (newLeft < 0) {
            newWidth = startWidth + startLeft;
            newLeft = 0;
          }
        } else {
          newWidth = Math.max(200, Math.min(startWidth + dx, window.innerWidth * 0.95));
        }
        const newHeight = Math.max(150, Math.min(startHeight + dy, window.innerHeight * 0.95));
        dialogRef.current.style.width = `${newWidth}px`;
        dialogRef.current.style.height = `${newHeight}px`;
        if (isRTL && newLeft !== void 0) {
          setPosition(clampPosition(newLeft, positionRef.current?.y ?? startLeft));
        }
      };
      const onUp = () => {
        resizeState.current = null;
        window.removeEventListener("mousemove", onMove);
        window.removeEventListener("mouseup", onUp);
      };
      window.addEventListener("mousemove", onMove);
      window.addEventListener("mouseup", onUp);
    },
    [resizable, stretch, onPhone, ensureDirectPositioning, clampPosition]
  );
  React.useEffect(() => {
    if (isDragging) {
      document.addEventListener("mousemove", handleMouseMove);
      document.addEventListener("mouseup", handleMouseUp);
      return () => {
        document.removeEventListener("mousemove", handleMouseMove);
        document.removeEventListener("mouseup", handleMouseUp);
      };
    }
  }, [isDragging, handleMouseMove, handleMouseUp]);
  React.useImperativeHandle(
    ref,
    () => ({
      show: () => {
        if (controlledOpen === void 0) {
          setInternalOpen(true);
        }
        onOpenChange?.(true);
      },
      showModal: () => {
        if (controlledOpen === void 0) {
          setInternalOpen(true);
        }
        onOpenChange?.(true);
      },
      close: (returnValue) => {
        returnValueRef.current = returnValue;
        handleClose(false);
      },
      focus: () => {
        dialogRef.current?.focus();
      }
    }),
    [controlledOpen, onOpenChange, handleClose]
  );
  return /* @__PURE__ */ jsxs(
    "dialog",
    {
      ref: dialogRef,
      id: stableId,
      popover: "manual",
      role: effectiveRole,
      "aria-labelledby": accessibleNameRef || (!accessibleName && headerText ? `${stableId}-title` : void 0),
      "aria-describedby": effectiveAriaDescribedBy,
      "aria-label": accessibleName || (!headerText && !accessibleNameRef ? "Dialog" : void 0),
      "aria-roledescription": ariaRoleDescription,
      tabIndex: -1,
      className: cn(
        dialogVariants({ state, stretch, resizable }),
        stretch && onPhone && "w-screen max-w-none h-[100dvh] max-h-none rounded-none min-w-0",
        className
      ),
      style: {
        ...stretch && onPhone ? {
          left: 0,
          top: 0,
          right: "auto",
          transform: "none"
        } : position ? {
          left: `${position.x}px`,
          top: `${position.y}px`,
          right: "auto",
          transform: "none"
        } : {
          left: "50%",
          top: "50%",
          right: "auto",
          transform: "translate(-50%, -50%)"
        }
      },
      onClick: (e) => e.stopPropagation(),
      onMouseDown: handleMouseDown,
      children: [
        (draggable || resizable) && /* @__PURE__ */ jsx(
          "span",
          {
            id: `${stableId}-movement-descr`,
            className: "absolute w-px h-px p-0 -m-px overflow-hidden whitespace-nowrap border-0",
            style: { clip: "rect(0, 0, 0, 0)" },
            children: getMovementDescription()
          }
        ),
        (header || headerText) && /* @__PURE__ */ jsx(
          "div",
          {
            ref: headerRef,
            className: cn(
              headerVariants({ state, draggable }),
              !hideHeaderBorder && state !== "Error" /* Error */ && state !== "Success" /* Success */ && state !== "Information" /* Information */ && state !== "Warning" /* Warning */ && "border-b"
            ),
            tabIndex: draggable || resizable ? 0 : void 0,
            "aria-roledescription": draggable && ariaRoleDescription ? ariaRoleDescription : void 0,
            children: header || /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-3 w-full", children: [
              /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-3 min-w-0", children: [
                state === "Error" && /* @__PURE__ */ jsx(MessageErrorIcon, { className: "h-5 w-5 text-sapphire-negative shrink-0" }),
                state === "Success" && /* @__PURE__ */ jsx(MessageSuccessIcon, { className: "h-5 w-5 text-sapphire-positive shrink-0" }),
                state === "Warning" && /* @__PURE__ */ jsx(MessageWarningIcon, { className: "h-5 w-5 text-sapphire-warning shrink-0" }),
                state === "Information" && /* @__PURE__ */ jsx(MessageInformationIcon, { className: "h-5 w-5 text-sapphire-info shrink-0" }),
                showBackButton && state === "None" && /* @__PURE__ */ jsx(
                  Button,
                  {
                    design: "SecondaryNeutral",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx(ArrowLeftIcon, {}),
                    onClick: onBackButtonClick
                  }
                ),
                /* @__PURE__ */ jsxs("div", { className: "flex flex-col min-w-0", children: [
                  /* @__PURE__ */ jsx(
                    Title,
                    {
                      level: "H1",
                      id: `${stableId}-title`,
                      className: "text-xl font-semibold text-foreground truncate",
                      children: headerText
                    }
                  ),
                  subHeaderText && /* @__PURE__ */ jsx("span", { className: "text-sm font-normal text-sapphire-text-tertiary truncate", children: subHeaderText })
                ] })
              ] }),
              headerEndContent && /* @__PURE__ */ jsx("div", { className: "flex items-center gap-2 shrink-0 max-w-[50%] ml-auto", children: headerEndContent })
            ] })
          }
        ),
        /* @__PURE__ */ jsx(
          "div",
          {
            ref: contentRef,
            className: cn("flex-auto min-h-0 overflow-y-auto", !noPadding && "px-6 py-4"),
            tabIndex: -1,
            children
          }
        ),
        footer && /* @__PURE__ */ jsx("div", { className: "flex items-center justify-end gap-2 px-6 py-5 shrink-0", children: footer }),
        resizable && !onPhone && /* @__PURE__ */ jsx(
          "div",
          {
            className: "dialog-resize-handle absolute bottom-0 w-6 h-6 rounded-full flex items-center justify-center z-[1] end-0",
            onMouseDown: handleResizeMouseDown,
            "data-resize-handle": "true",
            children: /* @__PURE__ */ jsx(
              ResizeCornerIcon,
              {
                className: "h-4 w-4 text-sapphire-text-tertiary rtl:rotate-90"
              }
            )
          }
        )
      ]
    }
  );
}

export { Dialog, DialogState };
