// @sap-ui/fx-components/dist/chunk-ORTPW5DV.js
import { Token } from './chunk-A3BV5G37.js';
import { BrowseFolderIcon } from './chunk-PLKA6UTA.js';
import { WarningIcon } from './chunk-P4WB5QKL.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { InformationIcon, SysEnter2Icon } from './chunk-ZK4METPA.js';
import { ErrorIcon } from './chunk-FL4UPYZT.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useState, useMemo, useEffect, useCallback, useImperativeHandle } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';

// src/types/file-uploader.ts
var FileUploaderSize = /* @__PURE__ */ ((FileUploaderSize2) => {
  FileUploaderSize2["Large"] = "Large";
  FileUploaderSize2["Medium"] = "Medium";
  return FileUploaderSize2;
})(FileUploaderSize || {});
function buildFileList(files) {
  if (typeof DataTransfer !== "undefined") {
    const dt = new DataTransfer();
    files.forEach((f) => dt.items.add(f));
    return dt.files;
  }
  const list = Object.create(FileList.prototype);
  files.forEach((f, i) => {
    list[i] = f;
  });
  Object.defineProperty(list, "length", { value: files.length });
  list.item = (index) => list[index] ?? null;
  list[Symbol.iterator] = function* () {
    for (let i = 0; i < files.length; i++) yield list[i];
  };
  return list;
}
var descIdCounter = 0;
function useStableId(prefix) {
  const ref = useRef(`${prefix}-${++descIdCounter}`);
  return ref.current;
}
var containerVariants = cva(
  [
    "flex items-center w-full border bg-sapphire-canvas-primary",
    "overflow-clip transition-colors relative",
    "pr-1 py-1 gap-1",
    "cursor-pointer outline-none"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: [
          "h-10 rounded-lg pl-1",
          "focus:border-2 focus:pl-[3px] focus:pr-[3px] focus:py-[3px]"
        ],
        ["Medium" /* Medium */]: [
          "h-8 rounded pl-0.5",
          "focus:border-2 focus:pl-px focus:pr-[3px] focus:py-[3px]"
        ]
      },
      valueState: {
        ["None" /* None */]: "border-sapphire-border-active hover:border-sapphire-border-accent focus:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-positive focus:border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-negative focus:border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-warning focus:border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-info focus:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      dragOver: {
        true: "border-sapphire-border-accent border-2 pl-[3px] pr-[3px] py-[3px]",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      dragOver: false
    }
  }
);
var valueStateIcons = {
  ["None" /* None */]: null,
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Critical" /* Critical */]: WarningIcon,
  ["Information" /* Information */]: InformationIcon
};
var valueStateMessageColors = {
  ["None" /* None */]: "",
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Information" /* Information */]: "text-sapphire-info"
};
function FileUploaderBase({
  size = "Large" /* Large */,
  accept,
  multiple = false,
  maxFileSize,
  hideInput = false,
  placeholder,
  value,
  disabled = false,
  required = false,
  valueState = "None" /* None */,
  valueStateMessage,
  name,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onChange,
  onFileSizeExceed,
  children,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const containerRef = useRef(null);
  const fileInputRef = useRef(null);
  const dragCounterRef = useRef(0);
  const fileNamesRef = useRef([]);
  const tokenAreaRef = useRef(null);
  const [fileNames, setFileNames] = useState([]);
  const [isDragOver, setIsDragOver] = useState(false);
  const [isFocused, setIsFocused] = useState(false);
  const [focusedTokenIndex, setFocusedTokenIndex] = useState(-1);
  const { t } = useTranslation("fx");
  const descriptionId = useStableId("fu-desc");
  const resolvedValueState = valueState;
  const isNegativeState = resolvedValueState === "Negative" /* Negative */;
  const isMedium = size === "Medium" /* Medium */;
  const resolvedPlaceholder = placeholder ?? (multiple ? t("FILEUPLOADER_PLACEHOLDER_MULTIPLE") : t("FILEUPLOADER_PLACEHOLDER"));
  const displayNames = useMemo(() => {
    if (value !== void 0 && value !== "") {
      return value.split(",").map((n) => n.trim()).filter(Boolean);
    }
    return fileNames;
  }, [value, fileNames]);
  const hasFiles = displayNames.length > 0;
  const isMultiDisplay = displayNames.length > 1;
  const [visibleCount, setVisibleCount] = useState(displayNames.length);
  const nMoreLabelRef = useRef(null);
  useEffect(() => {
    setVisibleCount(displayNames.length);
  }, [displayNames.length]);
  useEffect(() => {
    if (!isMultiDisplay || isFocused || !tokenAreaRef.current) return;
    const area = tokenAreaRef.current;
    const calculateVisible = () => {
      if (!area) return;
      const areaWidth = area.getBoundingClientRect().width;
      if (areaWidth === 0) return;
      const nMoreWidth = nMoreLabelRef.current ? nMoreLabelRef.current.getBoundingClientRect().width + 8 : 60;
      const tokenEls = Array.from(
        area.querySelectorAll('[data-part="root"]')
      );
      const gap = 4;
      let usedWidth = 0;
      let fitCount = 0;
      for (let i = 0; i < tokenEls.length; i++) {
        const el = tokenEls[i];
        const wasHidden = el.classList.contains("hidden");
        if (wasHidden) el.classList.remove("hidden");
        const w = el.getBoundingClientRect().width;
        if (wasHidden) el.classList.add("hidden");
        const needed = w + (i > 0 ? gap : 0);
        const remaining = tokenEls.length - i - 1;
        const reserve = remaining > 0 ? nMoreWidth : 0;
        if (usedWidth + needed + reserve > areaWidth && fitCount > 0) {
          break;
        }
        usedWidth += needed;
        fitCount++;
      }
      setVisibleCount(fitCount);
    };
    const observer = new ResizeObserver(() => {
      requestAnimationFrame(calculateVisible);
    });
    observer.observe(area);
    calculateVisible();
    return () => observer.disconnect();
  }, [isMultiDisplay, isFocused, displayNames]);
  const overflowCount = isMultiDisplay ? Math.max(0, displayNames.length - visibleCount) : 0;
  const validateFiles = useCallback(
    (files) => {
      if (maxFileSize === void 0 || maxFileSize <= 0) return files;
      const exceededFiles = [];
      const validFiles = [];
      for (let i = 0; i < files.length; i++) {
        const file = files[i];
        const fileSizeMB = file.size / 1024 / 1024;
        if (fileSizeMB > maxFileSize) {
          exceededFiles.push({ fileName: file.name, fileSize: fileSizeMB });
        } else {
          validFiles.push(file);
        }
      }
      if (exceededFiles.length > 0) {
        onFileSizeExceed?.({ filesData: exceededFiles });
      }
      if (validFiles.length === 0 && exceededFiles.length > 0) {
        return null;
      }
      return buildFileList(validFiles);
    },
    [maxFileSize, onFileSizeExceed]
  );
  const filterByAccept = useCallback(
    (files) => {
      if (!accept) return files;
      const acceptTypes = accept.split(",").map((t2) => t2.trim().toLowerCase());
      const matched = [];
      for (let i = 0; i < files.length; i++) {
        const file = files[i];
        const parts = file.name.split(".");
        const fileExt = parts.length > 1 ? "." + parts.pop()?.toLowerCase() : "";
        const fileMime = file.type.toLowerCase();
        const matches = acceptTypes.some((type) => {
          if (type.startsWith(".")) {
            return fileExt === type;
          }
          if (type.endsWith("/*")) {
            return fileMime.startsWith(type.slice(0, -1));
          }
          return fileMime === type;
        });
        if (matches) {
          matched.push(file);
        }
      }
      return buildFileList(matched);
    },
    [accept]
  );
  const openFileBrowser = useCallback(() => {
    if (disabled) return;
    if (fileInputRef.current) {
      fileInputRef.current.value = "";
    }
    fileInputRef.current?.click();
  }, [disabled]);
  const clearFiles = useCallback(() => {
    if (fileInputRef.current) {
      fileInputRef.current.value = "";
    }
    fileNamesRef.current = [];
    setFileNames([]);
    onChange?.({ files: null });
  }, [onChange]);
  const handleNativeChange = useCallback(
    (e) => {
      const files = e.target.files;
      if (!files || files.length === 0) {
        if (fileNamesRef.current.length > 0) {
          clearFiles();
        }
        return;
      }
      const validFiles = validateFiles(files);
      if (!validFiles || validFiles.length === 0) {
        return;
      }
      const names = Array.from(validFiles).map((f) => f.name);
      fileNamesRef.current = names;
      setFileNames(names);
      onChange?.({ files: validFiles });
    },
    [validateFiles, clearFiles, onChange]
  );
  const handleContainerClick = useCallback(() => {
    openFileBrowser();
  }, [openFileBrowser]);
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled) return;
      switch (e.key) {
        case "Enter":
        case " ":
          e.preventDefault();
          openFileBrowser();
          break;
        case "F4":
          e.preventDefault();
          openFileBrowser();
          break;
        case "Delete":
          e.preventDefault();
          if (hasFiles) {
            clearFiles();
          }
          break;
        case "ArrowRight":
          if (displayNames.length > 1 && isFocused) {
            e.preventDefault();
            setFocusedTokenIndex(0);
            const tokens = tokenAreaRef.current?.querySelectorAll('[data-part="root"]');
            tokens?.[0]?.focus();
          }
          break;
      }
      if (e.altKey && (e.key === "ArrowUp" || e.key === "ArrowDown")) {
        e.preventDefault();
        openFileBrowser();
      }
    },
    [disabled, openFileBrowser, hasFiles, clearFiles, displayNames.length, isFocused]
  );
  const handleTokenKeyDown = useCallback(
    (e, index) => {
      if (e.key === "ArrowRight") {
        e.preventDefault();
        e.stopPropagation();
        if (index < displayNames.length - 1) {
          const next = index + 1;
          setFocusedTokenIndex(next);
          const tokens = tokenAreaRef.current?.querySelectorAll('[data-part="root"]');
          const el = tokens?.[next];
          el?.focus();
          el?.scrollIntoView({ block: "nearest", inline: "nearest" });
        }
      } else if (e.key === "ArrowLeft") {
        e.preventDefault();
        e.stopPropagation();
        if (index > 0) {
          const prev = index - 1;
          setFocusedTokenIndex(prev);
          const tokens = tokenAreaRef.current?.querySelectorAll('[data-part="root"]');
          const el = tokens?.[prev];
          el?.focus();
          el?.scrollIntoView({ block: "nearest", inline: "nearest" });
        } else {
          setFocusedTokenIndex(-1);
          containerRef.current?.focus();
        }
      }
    },
    [displayNames.length]
  );
  const handleDragOver = useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
    },
    [disabled]
  );
  const handleDragEnter = useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
      dragCounterRef.current++;
      setIsDragOver(true);
    },
    [disabled]
  );
  const handleDragLeave = useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
      dragCounterRef.current--;
      if (dragCounterRef.current <= 0) {
        dragCounterRef.current = 0;
        setIsDragOver(false);
      }
    },
    [disabled]
  );
  const handleDrop = useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
      dragCounterRef.current = 0;
      setIsDragOver(false);
      let droppedFiles = e.dataTransfer.files;
      if (!droppedFiles || droppedFiles.length === 0) return;
      if (!multiple && droppedFiles.length > 1) {
        droppedFiles = buildFileList([droppedFiles[0]]);
      }
      droppedFiles = filterByAccept(droppedFiles);
      if (droppedFiles.length === 0) return;
      const validFiles = validateFiles(droppedFiles);
      if (!validFiles || validFiles.length === 0) return;
      const names = Array.from(validFiles).map((f) => f.name);
      fileNamesRef.current = names;
      setFileNames(names);
      onChange?.({ files: validFiles });
    },
    [disabled, multiple, filterByAccept, validateFiles, onChange]
  );
  const handleClearClick = useCallback(
    (detail) => {
      detail.originalEvent.stopPropagation();
      clearFiles();
    },
    [clearFiles]
  );
  const handleBrowseClick = useCallback(
    (detail) => {
      detail.originalEvent.stopPropagation();
      openFileBrowser();
    },
    [openFileBrowser]
  );
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        containerRef.current?.focus();
      },
      blur() {
        containerRef.current?.blur();
      },
      openFileBrowser() {
        openFileBrowser();
      },
      clear() {
        clearFiles();
      },
      get files() {
        return fileInputRef.current?.files ?? null;
      },
      get nativeInput() {
        return fileInputRef.current;
      }
    }),
    [openFileBrowser, clearFiles]
  );
  const ValueStateIcon = valueStateIcons[resolvedValueState];
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn("w-full", className),
      style,
      id,
      "data-testid": dataTestId,
      children: [
        /* @__PURE__ */ jsx(
          "input",
          {
            ref: fileInputRef,
            type: "file",
            accept,
            multiple,
            name,
            required,
            disabled,
            tabIndex: -1,
            "aria-hidden": "true",
            className: "sr-only",
            onChange: handleNativeChange,
            "data-testid": dataTestId ? `${dataTestId}-input` : void 0
          }
        ),
        hideInput ? (
          /* Button-only mode: render children as trigger */
          /* @__PURE__ */ jsx(
            "div",
            {
              role: "button",
              tabIndex: disabled ? -1 : 0,
              onClick: disabled ? void 0 : handleContainerClick,
              onKeyDown: disabled ? void 0 : handleKeyDown,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-disabled": disabled || void 0,
              children
            }
          )
        ) : (
          /* Standard mode: styled container */
          /* @__PURE__ */ jsxs(
            "div",
            {
              ref: containerRef,
              tabIndex: disabled ? -1 : 0,
              role: "group",
              "aria-roledescription": t("FILEUPLOADER_ROLE_DESCRIPTION"),
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-describedby": accessibleDescription ? descriptionId : accessibleDescriptionRef,
              "aria-required": required || void 0,
              "aria-invalid": isNegativeState || void 0,
              "aria-haspopup": "dialog",
              className: cn(
                containerVariants({
                  size,
                  valueState: resolvedValueState,
                  disabled,
                  dragOver: isDragOver
                })
              ),
              onClick: handleContainerClick,
              onKeyDown: handleKeyDown,
              onFocus: () => setIsFocused(true),
              onBlur: (e) => {
                if (!containerRef.current?.contains(e.relatedTarget)) {
                  setIsFocused(false);
                  setFocusedTokenIndex(-1);
                }
              },
              onDragOver: handleDragOver,
              onDragEnter: handleDragEnter,
              onDragLeave: handleDragLeave,
              onDrop: handleDrop,
              children: [
                accessibleDescription && /* @__PURE__ */ jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription }),
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    ref: tokenAreaRef,
                    className: cn(
                      "flex-1 min-w-0 flex items-center gap-1",
                      isFocused && isMultiDisplay ? "overflow-x-auto scrollbar-none" : "overflow-hidden"
                    ),
                    children: hasFiles ? displayNames.length === 1 ? /* @__PURE__ */ jsx(Token, { text: displayNames[0], readonly: true, singleToken: true, className: "shrink min-w-0" }) : isFocused ? displayNames.map((fileName, idx) => /* @__PURE__ */ jsx(
                      "span",
                      {
                        onKeyDown: (e) => handleTokenKeyDown(e, idx),
                        className: "shrink-0",
                        children: /* @__PURE__ */ jsx(
                          Token,
                          {
                            text: fileName,
                            readonly: true,
                            tabIndex: idx === focusedTokenIndex ? 0 : -1
                          }
                        )
                      },
                      fileName
                    )) : /* @__PURE__ */ jsxs(Fragment, { children: [
                      displayNames.map((fileName, idx) => /* @__PURE__ */ jsx(
                        Token,
                        {
                          text: fileName,
                          readonly: true,
                          className: idx >= visibleCount ? "hidden" : "shrink-0"
                        },
                        fileName
                      )),
                      overflowCount > 0 && /* @__PURE__ */ jsx(
                        "span",
                        {
                          ref: nMoreLabelRef,
                          className: "inline-block whitespace-nowrap shrink-0 text-sm font-normal text-sapphire-brand-foreground cursor-pointer hover:underline",
                          children: t("TOKENIZER_SHOW_MORE", { count: overflowCount })
                        }
                      )
                    ] }) : /* @__PURE__ */ jsx("span", { className: "text-sapphire-text-tertiary italic text-sm truncate select-none pl-2.5", children: resolvedPlaceholder })
                  }
                ),
                hasFiles && !disabled && /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
                  Button,
                  {
                    design: "SecondaryNeutral" /* SecondaryNeutral */,
                    size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-full w-full" }) }),
                    tabIndex: -1,
                    onClick: handleClearClick,
                    accessibleName: t("FILEUPLOADER_CLEAR_TOOLTIP"),
                    className: "shrink-0",
                    "data-part": "clear"
                  }
                ) }),
                /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
                  Button,
                  {
                    design: "SecondaryNeutral" /* SecondaryNeutral */,
                    size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(BrowseFolderIcon, { className: "h-full w-full" }) }),
                    tabIndex: -1,
                    onClick: handleBrowseClick,
                    accessibleName: t("FILEUPLOADER_BROWSE_TOOLTIP"),
                    className: "shrink-0",
                    "data-part": "browse"
                  }
                ) })
              ]
            }
          )
        ),
        valueStateMessage && resolvedValueState !== "None" /* None */ && /* @__PURE__ */ jsxs(
          "div",
          {
            role: isNegativeState ? "alert" : void 0,
            "aria-live": isNegativeState ? "assertive" : "polite",
            className: cn(
              "mt-1 text-xs flex items-start gap-1",
              valueStateMessageColors[resolvedValueState]
            ),
            children: [
              ValueStateIcon && /* @__PURE__ */ jsx(ValueStateIcon, { className: "h-3 w-3 mt-0.5 shrink-0" }),
              /* @__PURE__ */ jsx("span", { children: valueStateMessage })
            ]
          }
        )
      ]
    }
  );
}
var FileUploader = React.memo(FileUploaderBase);
FileUploader.displayName = "FileUploader";

export { FileUploader, FileUploaderSize };
