// @sap-ui/fx-components/dist/chunk-I56SHA5L.js
import { Select, Option } from './chunk-QDXU3YWA.js';
import { useToolbarContext } from './chunk-3GZW6BV2.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useCallback } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

function ToolbarButton({
  design = "SecondaryNeutral" /* SecondaryNeutral */,
  icon,
  endIcon,
  text,
  tooltip,
  disabled = false,
  width,
  overflowPriority: _overflowPriority,
  preventOverflowClosing = false,
  accessibleName,
  accessibleNameRef,
  onClick,
  className,
  ref
}) {
  const { isInOverflow, closeOverflow } = useToolbarContext();
  const handleOverflowClick = useCallback(
    (e) => {
      onClick?.(e);
      if (!preventOverflowClosing) {
        closeOverflow();
      }
    },
    [onClick, preventOverflowClosing, closeOverflow]
  );
  if (isInOverflow) {
    return /* @__PURE__ */ jsxs(
      "button",
      {
        ref,
        type: "button",
        disabled,
        title: tooltip,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        onClick: handleOverflowClick,
        className: cn(
          "w-full flex items-center gap-2 px-3 py-2 text-sm",
          "hover:bg-muted rounded-md text-left",
          "focus:outline-none focus:ring-2 focus:ring-inset focus:ring-ring focus:z-10",
          "disabled:pointer-events-none disabled:opacity-50",
          className
        ),
        children: [
          icon && /* @__PURE__ */ jsx("span", { className: "inline-flex items-center shrink-0 [&>svg]:h-4 [&>svg]:w-4", children: icon }),
          text && /* @__PURE__ */ jsx("span", { className: "flex-1", children: text }),
          text == null && tooltip && /* @__PURE__ */ jsx("span", { className: "flex-1", children: tooltip }),
          endIcon && /* @__PURE__ */ jsx("span", { className: "inline-flex items-center shrink-0 [&>svg]:h-4 [&>svg]:w-4", children: endIcon })
        ]
      }
    );
  }
  const isIconOnly = !!icon && !text;
  return /* @__PURE__ */ jsx("div", { ref, className: "shrink-0", style: width ? { width } : void 0, children: /* @__PURE__ */ jsx(
    Button,
    {
      design,
      size: "Medium" /* Medium */,
      icon,
      endIcon,
      iconOnly: isIconOnly,
      disabled,
      tooltip,
      accessibleName,
      accessibleNameRef,
      onClick: (detail) => onClick?.(detail.originalEvent),
      className: cn(width && "w-full", className),
      children: text
    }
  ) });
}
ToolbarButton.displayName = "ToolbarButton";
function ToolbarSeparator({ className, ref }) {
  const { isInOverflow } = useToolbarContext();
  if (isInOverflow) {
    return /* @__PURE__ */ jsx(
      "div",
      {
        ref,
        role: "separator",
        className: cn("h-px w-full bg-border my-1", className)
      }
    );
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      role: "separator",
      className: cn("h-[26px] w-px bg-border shrink-0", className)
    }
  );
}
ToolbarSeparator.displayName = "ToolbarSeparator";
function ToolbarSpacer({ width, className, ref }) {
  const { isInOverflow } = useToolbarContext();
  if (isInOverflow) {
    return null;
  }
  const isFixed = width && width !== "auto";
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn(isFixed ? "shrink-0" : "flex-auto", className),
      style: isFixed ? { width } : void 0
    }
  );
}
function ToolbarSelect({
  width,
  value,
  disabled = false,
  overflowPriority: _overflowPriority,
  preventOverflowClosing = false,
  accessibleName,
  accessibleNameRef,
  onChange,
  onOpen,
  onClose,
  children,
  className,
  ref
}) {
  const { isInOverflow, closeOverflow } = useToolbarContext();
  const handleChange = useCallback(
    (detail) => {
      onChange?.(detail);
      if (isInOverflow && !preventOverflowClosing) {
        closeOverflow();
      }
    },
    [onChange, isInOverflow, preventOverflowClosing, closeOverflow]
  );
  if (isInOverflow) {
    return /* @__PURE__ */ jsx("div", { ref, className: cn("w-full px-3 py-1", className), children: /* @__PURE__ */ jsx(
      Select,
      {
        value,
        disabled,
        accessibleName,
        accessibleNameRef,
        onChange: handleChange,
        onOpen,
        onClose,
        className: "w-full",
        children
      }
    ) });
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("shrink-0", className),
      style: width ? { width } : void 0,
      children: /* @__PURE__ */ jsx(
        Select,
        {
          value,
          disabled,
          accessibleName,
          accessibleNameRef,
          onChange: handleChange,
          onOpen,
          onClose,
          className: cn(width && "w-full"),
          children
        }
      )
    }
  );
}
ToolbarSelect.displayName = "ToolbarSelect";
function ToolbarSelectOption({ value, selected, children, ref }) {
  return /* @__PURE__ */ jsx(Option, { ref, value, selected, children });
}
function ToolbarItem({ overflowPriority: _overflowPriority, preventOverflowClosing: _preventOverflowClosing, children, className, ref }) {
  const { isInOverflow } = useToolbarContext();
  if (isInOverflow) {
    return /* @__PURE__ */ jsx("div", { ref, className: cn("w-full px-3 py-2", className), children });
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("shrink-0 inline-flex items-center", className),
      children
    }
  );
}
ToolbarItem.displayName = "ToolbarItem";

export { ToolbarButton, ToolbarItem, ToolbarSelect, ToolbarSelectOption, ToolbarSeparator, ToolbarSpacer };
