// @sap-ui/fx-components/dist/chunk-EWAPHGAS.js
import { inputContainerVariants } from './chunk-GC7VJCBJ.js';
import { CalendarIcon, FobWatchIcon } from './chunk-H4UBXTRR.js';
import { useEnsureCldr, getLocaleDateFormat, Calendar } from './chunk-PXVVADOR.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { isPhone } from './chunk-YSPWI6FA.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { useMemo, useCallback, useState, useRef, useImperativeHandle, useEffect } from 'react';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import DateFormat from '@ui5/webcomponents-localization/dist/DateFormat.js';
import Locale from '@ui5/webcomponents-base/dist/locale/Locale.js';
import { useTranslation } from 'react-i18next';

function rangeWithStep(start, end, step) {
  const result = [];
  for (let i = start; i <= end; i += step) {
    result.push(i);
  }
  return result;
}
var TimeColumn = ({ values, selected, onSelect, label }) => {
  const containerRef = useRef(null);
  const selectedRef = useRef(null);
  useEffect(() => {
    if (selectedRef.current && containerRef.current) {
      const container = containerRef.current;
      const el = selectedRef.current;
      const top = el.offsetTop - container.clientHeight / 2 + el.clientHeight / 2;
      container.scrollTo({ top, behavior: "smooth" });
    }
  }, [selected]);
  const handleKeyDown = useCallback((e, idx) => {
    if (e.key === "ArrowUp" && idx > 0) {
      e.preventDefault();
      onSelect(values[idx - 1].value);
    } else if (e.key === "ArrowDown" && idx < values.length - 1) {
      e.preventDefault();
      onSelect(values[idx + 1].value);
    } else if (e.key === "Enter" || e.key === " ") {
      e.preventDefault();
      onSelect(values[idx].value);
    }
  }, [values, onSelect]);
  return /* @__PURE__ */ jsxs("div", { className: "flex flex-col min-w-[3rem]", children: [
    /* @__PURE__ */ jsx("div", { className: "text-xs font-medium text-sapphire-text-tertiary text-center py-1 border-b border-sapphire-border", children: label }),
    /* @__PURE__ */ jsx(
      "div",
      {
        ref: containerRef,
        className: "overflow-y-auto max-h-48 scrollbar-thin",
        role: "listbox",
        "aria-label": label,
        children: values.map((item, idx) => {
          const isSelected = item.value === selected;
          return /* @__PURE__ */ jsx(
            "button",
            {
              ref: isSelected ? selectedRef : void 0,
              type: "button",
              role: "option",
              "aria-selected": isSelected,
              tabIndex: isSelected ? 0 : -1,
              className: cn(
                "w-full py-1.5 px-2 text-sm text-center transition-colors",
                "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sapphire-ring focus-visible:ring-inset",
                isSelected ? "bg-sapphire-brand-hover-background text-sapphire-text-on-surface font-medium" : "hover:bg-sapphire-bg-secondary text-sapphire-text"
              ),
              onClick: () => onSelect(item.value),
              onKeyDown: (e) => handleKeyDown(e, idx),
              children: item.label
            },
            item.value
          );
        })
      }
    )
  ] });
};
function TimePicker({
  hours,
  minutes,
  seconds,
  showSeconds = false,
  timeFormat = "24h",
  minuteStep = 1,
  secondStep = 1,
  onTimeChange,
  className,
  ref
}) {
  const is12h = timeFormat === "12h";
  const period = hours >= 12 ? "PM" : "AM";
  const hourValues = useMemo(() => {
    if (is12h) {
      return Array.from({ length: 12 }, (_, i) => {
        const h = i === 0 ? 12 : i;
        return { value: i, label: String(h).padStart(2, "0") };
      });
    }
    return Array.from({ length: 24 }, (_, i) => ({
      value: i,
      label: String(i).padStart(2, "0")
    }));
  }, [is12h]);
  const minuteValues = useMemo(() => {
    return rangeWithStep(0, 59, minuteStep).map((m) => ({
      value: m,
      label: String(m).padStart(2, "0")
    }));
  }, [minuteStep]);
  const secondValues = useMemo(() => {
    return rangeWithStep(0, 59, secondStep).map((s) => ({
      value: s,
      label: String(s).padStart(2, "0")
    }));
  }, [secondStep]);
  const periodValues = useMemo(() => [
    { value: 0, label: "AM" },
    { value: 1, label: "PM" }
  ], []);
  const displayHour = is12h ? hours % 12 : hours;
  const handleHourChange = useCallback((h) => {
    if (is12h) {
      const h24 = period === "PM" ? h === 0 ? 12 : h + 12 : h;
      onTimeChange(h24, minutes, seconds);
    } else {
      onTimeChange(h, minutes, seconds);
    }
  }, [is12h, period, minutes, seconds, onTimeChange]);
  const handleMinuteChange = useCallback((m) => {
    onTimeChange(hours, m, seconds);
  }, [hours, seconds, onTimeChange]);
  const handleSecondChange = useCallback((s) => {
    onTimeChange(hours, minutes, s);
  }, [hours, minutes, onTimeChange]);
  const handlePeriodChange = useCallback((p) => {
    const isPM = p === 1;
    let h24 = hours;
    if (isPM && hours < 12) h24 = hours + 12;
    else if (!isPM && hours >= 12) h24 = hours - 12;
    onTimeChange(h24, minutes, seconds);
  }, [hours, minutes, seconds, onTimeChange]);
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      className: cn("flex border-l border-sapphire-border", className),
      role: "group",
      "aria-label": "Time picker",
      children: [
        /* @__PURE__ */ jsx(
          TimeColumn,
          {
            values: hourValues,
            selected: displayHour,
            onSelect: handleHourChange,
            label: "Hr"
          }
        ),
        /* @__PURE__ */ jsx("div", { className: "w-px bg-sapphire-border" }),
        /* @__PURE__ */ jsx(
          TimeColumn,
          {
            values: minuteValues,
            selected: minutes,
            onSelect: handleMinuteChange,
            label: "Min"
          }
        ),
        showSeconds && /* @__PURE__ */ jsxs(Fragment, { children: [
          /* @__PURE__ */ jsx("div", { className: "w-px bg-sapphire-border" }),
          /* @__PURE__ */ jsx(
            TimeColumn,
            {
              values: secondValues,
              selected: seconds,
              onSelect: handleSecondChange,
              label: "Sec"
            }
          )
        ] }),
        is12h && /* @__PURE__ */ jsxs(Fragment, { children: [
          /* @__PURE__ */ jsx("div", { className: "w-px bg-sapphire-border" }),
          /* @__PURE__ */ jsx(
            TimeColumn,
            {
              values: periodValues,
              selected: period === "PM" ? 1 : 0,
              onSelect: handlePeriodChange,
              label: "AP"
            }
          )
        ] })
      ]
    }
  );
}
function isValidDate(d) {
  return d instanceof Date && !isNaN(d.getTime());
}
function getLocaleTimeFormat(locale) {
  try {
    const formatted = new Intl.DateTimeFormat(locale || "en-US", { hour: "numeric" }).format(/* @__PURE__ */ new Date());
    return /AM|PM/i.test(formatted) ? "12h" : "24h";
  } catch {
    return "24h";
  }
}
function DateTimePicker({
  size = "Large" /* Large */,
  value: controlledValue,
  defaultValue = "",
  minDate,
  maxDate,
  disabled = false,
  readonly = false,
  required = false,
  open: controlledOpen,
  showClearIcon = true,
  showSeconds = false,
  timeFormat: timeFormatProp,
  minuteStep = 1,
  secondStep = 1,
  placeholder,
  displayFormat: displayFormatProp,
  valueFormat = "yyyy-MM-dd'T'HH:mm:ss",
  valueState = "None" /* None */,
  valueStateMessage,
  accessibleName,
  accessibleNameRef,
  hideWeekNumbers = false,
  name,
  id,
  onInput,
  onChange,
  onOpen,
  onClose,
  onValueStateChange,
  className,
  style,
  "data-testid": dataTestId,
  ref
}) {
  const { i18n } = useTranslation("fx");
  const locale = i18n.language;
  const cldrReady = useEnsureCldr(locale);
  const isPhoneDevice = useMemo(() => isPhone(), []);
  const [phoneTab, setPhoneTab] = useState("date");
  const isMedium = size === "Medium" /* Medium */;
  const timeFormat = timeFormatProp || getLocaleTimeFormat(locale);
  const dateDisplayFormat = useMemo(() => {
    if (displayFormatProp) return displayFormatProp;
    if (!cldrReady) return "yyyy-MM-dd";
    return getLocaleDateFormat(locale);
  }, [displayFormatProp, cldrReady, locale]);
  const timeDisplayFormat = showSeconds ? timeFormat === "12h" ? "hh:mm:ss a" : "HH:mm:ss" : timeFormat === "12h" ? "hh:mm a" : "HH:mm";
  const effectiveDisplayFormat = `${dateDisplayFormat} ${timeDisplayFormat}`;
  const effectivePlaceholder = placeholder || (!cldrReady && !displayFormatProp ? "" : effectiveDisplayFormat);
  const loc = useMemo(() => new Locale(locale), [locale]);
  const { displayFmt, valueFmt, isoDateFmt } = useMemo(() => {
    if (!cldrReady) return { displayFmt: null, valueFmt: null, isoDateFmt: null };
    return {
      displayFmt: DateFormat.getDateTimeInstance({ pattern: effectiveDisplayFormat }, loc),
      valueFmt: DateFormat.getDateTimeInstance({ pattern: valueFormat }, loc),
      isoDateFmt: DateFormat.getDateInstance({ pattern: "yyyy-MM-dd" }, loc)
    };
  }, [cldrReady, effectiveDisplayFormat, valueFormat, loc]);
  const formatDT = useCallback((d) => displayFmt?.format(d) ?? "", [displayFmt]);
  const formatVal = useCallback((d) => valueFmt?.format(d) ?? "", [valueFmt]);
  const formatISODate = useCallback((d) => isoDateFmt?.format(d) ?? "", [isoDateFmt]);
  const parseDT = useCallback((str, pattern) => {
    if (!cldrReady) return null;
    return DateFormat.getDateTimeInstance({ pattern, strictParsing: true }, loc).parse(str) ?? null;
  }, [cldrReady, loc]);
  const parseISODate = useCallback((str) => {
    if (!cldrReady) return null;
    return DateFormat.getDateInstance({ pattern: "yyyy-MM-dd", strictParsing: true }, loc).parse(str) ?? null;
  }, [cldrReady, loc]);
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [internalOpen, setInternalOpen] = useState(false);
  const [inputText, setInputText] = useState("");
  const [isEditing, setIsEditing] = useState(false);
  const [internalValueState, setInternalValueState] = useState(valueState);
  const isOpen = controlledOpen ?? internalOpen;
  const value = controlledValue ?? internalValue;
  const selectedDate = cldrReady && value ? parseDT(value, valueFormat) ?? void 0 : void 0;
  const [tempHours, setTempHours] = useState(0);
  const [tempMinutes, setTempMinutes] = useState(0);
  const [tempSeconds, setTempSeconds] = useState(0);
  const [tempDate, setTempDate] = useState(void 0);
  const containerRef = useRef(null);
  const inputRef = useRef(null);
  const rootRef = useRef(null);
  const skipFocusSyncRef = useRef(false);
  const minDateObj = useMemo(() => minDate ? parseISODate(minDate) : void 0, [minDate, parseISODate]);
  const maxDateObj = useMemo(() => maxDate ? parseISODate(maxDate) : void 0, [maxDate, parseISODate]);
  const updateValueState = useCallback((newState, valid) => {
    if (onValueStateChange) {
      const result = onValueStateChange({ valueState: newState, valid });
      if (result === false) return;
    }
    setInternalValueState(newState);
  }, [onValueStateChange]);
  const openPopover = useCallback(() => {
    if (disabled || readonly) return;
    setTempDate(selectedDate);
    setTempHours(selectedDate?.getHours() ?? 0);
    setTempMinutes(selectedDate?.getMinutes() ?? 0);
    setTempSeconds(selectedDate?.getSeconds() ?? 0);
    setPhoneTab("date");
    if (controlledOpen === void 0) {
      setInternalOpen(true);
    }
    onOpen?.();
  }, [disabled, readonly, controlledOpen, onOpen, selectedDate]);
  const closePopover = useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    onClose?.();
  }, [controlledOpen, onClose]);
  const getDisplayValue = () => {
    if (!cldrReady) return "";
    if (isEditing) return inputText;
    if (isValidDate(selectedDate)) return formatDT(selectedDate);
    return value;
  };
  const displayValue = getDisplayValue();
  const buildDateTime = useCallback((date, h, m, s) => {
    if (!date) return null;
    const dt = new Date(date);
    dt.setHours(h, m, s, 0);
    return dt;
  }, []);
  const commitDateTime = useCallback((dt) => {
    const formattedValue = formatVal(dt);
    updateValueState(valueState, true);
    if (!controlledValue) {
      setInternalValue(formattedValue);
    }
    onChange?.({
      value: formattedValue,
      dateValue: dt,
      valid: true
    });
  }, [controlledValue, valueState, onChange, updateValueState, formatVal]);
  const handleOk = useCallback(() => {
    const dt = buildDateTime(tempDate, tempHours, tempMinutes, tempSeconds);
    if (dt && isValidDate(dt)) {
      commitDateTime(dt);
    }
    closePopover();
  }, [tempDate, tempHours, tempMinutes, tempSeconds, buildDateTime, commitDateTime, closePopover]);
  const handleCancel = useCallback(() => {
    closePopover();
  }, [closePopover]);
  const handleCalendarSelect = useCallback((detail) => {
    if (detail.selectedValues.length > 0) {
      const parsed = parseISODate(detail.selectedValues[0]);
      if (isValidDate(parsed)) {
        setTempDate(parsed);
      }
    }
  }, [parseISODate]);
  const handleTimeChange = useCallback((h, m, s) => {
    setTempHours(h);
    setTempMinutes(m);
    setTempSeconds(s);
  }, []);
  const handleFocus = useCallback(() => {
    if (skipFocusSyncRef.current) {
      skipFocusSyncRef.current = false;
    } else {
      setInputText(displayValue);
    }
    setIsEditing(true);
  }, [displayValue]);
  const handleInputChange = useCallback((e) => {
    const newText = e.target.value;
    setInputText(newText);
    const parsed = parseDT(newText, effectiveDisplayFormat);
    const valid = isValidDate(parsed);
    onInput?.({
      value: newText,
      dateValue: valid ? parsed : null,
      valid
    });
  }, [effectiveDisplayFormat, onInput, parseDT]);
  const handleBlur = useCallback(() => {
    setIsEditing(false);
    if (!inputText.trim()) {
      updateValueState(valueState, true);
      if (!controlledValue) setInternalValue("");
      onChange?.({ value: "", dateValue: null, valid: true });
      return;
    }
    const parsed = parseDT(inputText.trim(), effectiveDisplayFormat);
    if (isValidDate(parsed)) {
      commitDateTime(parsed);
    } else {
      updateValueState("Negative" /* Negative */, false);
      onChange?.({ value: inputText, dateValue: null, valid: false });
    }
  }, [inputText, effectiveDisplayFormat, commitDateTime, controlledValue, valueState, onChange, updateValueState, parseDT]);
  const handleKeyDown = useCallback((e) => {
    switch (e.key) {
      case "Enter":
        e.preventDefault();
        handleBlur();
        inputRef.current?.blur();
        return;
      case "F4":
        e.preventDefault();
        if (isOpen) closePopover();
        else openPopover();
        return;
      case "ArrowDown":
        if (!e.altKey) return;
        e.preventDefault();
        if (!isOpen) openPopover();
        return;
      case "ArrowUp":
        if (!e.altKey) return;
        e.preventDefault();
        if (isOpen) closePopover();
        return;
    }
  }, [handleBlur, isOpen, openPopover, closePopover]);
  const togglePopover = useCallback(() => {
    if (disabled || readonly) return;
    if (isOpen) closePopover();
    else openPopover();
  }, [disabled, readonly, isOpen, openPopover, closePopover]);
  const handleClear = useCallback(() => {
    setInputText("");
    skipFocusSyncRef.current = true;
    updateValueState(valueState, true);
    if (!controlledValue) setInternalValue("");
    onChange?.({ value: "", dateValue: null, valid: true });
    inputRef.current?.focus();
  }, [controlledValue, valueState, onChange, updateValueState]);
  useImperativeHandle(ref, () => ({
    formatValue(date) {
      return formatDT(date);
    },
    isValidValue(val) {
      return isValidDate(parseDT(val, valueFormat));
    },
    get dateValue() {
      return selectedDate && isValidDate(selectedDate) ? selectedDate : null;
    },
    get nativeElement() {
      return rootRef.current;
    },
    focus() {
      inputRef.current?.focus();
    },
    blur() {
      inputRef.current?.blur();
    }
  }), [valueFormat, selectedDate, formatDT, parseDT]);
  const calendarProps = {
    selectionMode: "Single" /* Single */,
    selectedDates: tempDate ? [formatISODate(tempDate)] : [],
    onSelectionChange: handleCalendarSelect,
    minDate: minDateObj ? formatISODate(minDateObj) : void 0,
    maxDate: maxDateObj ? formatISODate(maxDateObj) : void 0,
    hideWeekNumbers,
    locale
  };
  const timePickerProps = {
    hours: tempHours,
    minutes: tempMinutes,
    seconds: tempSeconds,
    showSeconds,
    timeFormat,
    minuteStep,
    secondStep,
    onTimeChange: handleTimeChange
  };
  const popoverFooter = /* @__PURE__ */ jsxs("div", { className: "flex justify-end gap-2 p-3 border-t border-sapphire-border", children: [
    /* @__PURE__ */ jsx(Button, { design: "Tertiary", onClick: handleCancel, children: "Cancel" }),
    /* @__PURE__ */ jsx(Button, { design: "Primary", onClick: handleOk, children: "OK" })
  ] });
  const getValueStateClass = () => {
    switch (internalValueState) {
      case "Positive" /* Positive */:
        return "text-sapphire-positive";
      case "Negative" /* Negative */:
        return "text-sapphire-negative";
      case "Critical" /* Critical */:
        return "text-sapphire-warning";
      case "Information" /* Information */:
        return "text-sapphire-info";
      default:
        return false;
    }
  };
  return /* @__PURE__ */ jsxs("div", { ref: rootRef, className: cn("relative", className), style, children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: containerRef,
        className: inputContainerVariants({
          size,
          valueState: internalValueState,
          disabled,
          readonly
        }),
        "data-testid": dataTestId,
        children: [
          /* @__PURE__ */ jsx(
            "input",
            {
              ref: inputRef,
              id,
              type: "text",
              value: displayValue,
              onChange: handleInputChange,
              onFocus: handleFocus,
              onBlur: handleBlur,
              onKeyDown: handleKeyDown,
              disabled,
              readOnly: readonly,
              required,
              placeholder: effectivePlaceholder,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              className: cn(
                "flex-1 min-w-0 bg-transparent text-sm outline-none",
                "placeholder:text-sapphire-text-tertiary",
                "disabled:cursor-not-allowed disabled:opacity-50"
              )
            }
          ),
          showClearIcon && displayValue && !disabled && !readonly && /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
            Button,
            {
              design: "SecondaryNeutral" /* SecondaryNeutral */,
              size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
              iconOnly: true,
              icon: /* @__PURE__ */ jsx(DeclineIcon, { className: isMedium ? "h-3 w-3" : "h-4 w-4" }),
              tabIndex: -1,
              onClick: handleClear,
              accessibleName: "Clear datetime",
              className: "shrink-0"
            }
          ) }),
          /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
            Button,
            {
              design: "SecondaryNeutral" /* SecondaryNeutral */,
              size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
              iconOnly: true,
              icon: /* @__PURE__ */ jsx(CalendarIcon, { className: isMedium ? "h-3 w-3" : "h-4 w-4" }),
              tabIndex: -1,
              onClick: togglePopover,
              disabled: disabled || readonly,
              accessibleName: "Open date and time picker",
              accessibilityAttributes: { expanded: isOpen },
              className: "shrink-0"
            }
          ) })
        ]
      }
    ),
    (valueStateMessage || internalValueState === "Negative" /* Negative */) && /* @__PURE__ */ jsx(
      "div",
      {
        className: cn("mt-2 text-sm", getValueStateClass()),
        role: "alert",
        children: valueStateMessage || internalValueState === "Negative" /* Negative */ && "Invalid date/time format"
      }
    ),
    /* @__PURE__ */ jsx(
      ResponsivePopover,
      {
        open: isOpen,
        opener: containerRef,
        placement: "Bottom",
        hideArrow: true,
        noPadding: true,
        headerText: "Select Date & Time",
        footer: popoverFooter,
        onClose: closePopover,
        preventInitialFocus: true,
        children: isPhoneDevice ? (
          // Phone mode: tabs for Date / Time
          /* @__PURE__ */ jsxs("div", { className: "flex flex-col w-full", children: [
            /* @__PURE__ */ jsxs("div", { className: "flex border-b border-sapphire-border", children: [
              /* @__PURE__ */ jsxs(
                "button",
                {
                  type: "button",
                  className: cn(
                    "flex-1 py-2 px-4 text-sm font-medium text-center transition-colors",
                    phoneTab === "date" ? "border-b-2 border-sapphire-border-accent text-sapphire-text-accent" : "text-sapphire-text-tertiary hover:text-sapphire-text"
                  ),
                  onClick: () => setPhoneTab("date"),
                  children: [
                    /* @__PURE__ */ jsx(CalendarIcon, { className: "h-4 w-4 inline-block mr-1" }),
                    "Date"
                  ]
                }
              ),
              /* @__PURE__ */ jsxs(
                "button",
                {
                  type: "button",
                  className: cn(
                    "flex-1 py-2 px-4 text-sm font-medium text-center transition-colors",
                    phoneTab === "time" ? "border-b-2 border-sapphire-border-accent text-sapphire-text-accent" : "text-sapphire-text-tertiary hover:text-sapphire-text"
                  ),
                  onClick: () => setPhoneTab("time"),
                  children: [
                    /* @__PURE__ */ jsx(FobWatchIcon, { className: "h-4 w-4 inline-block mr-1" }),
                    "Time"
                  ]
                }
              )
            ] }),
            phoneTab === "date" ? /* @__PURE__ */ jsx(Calendar, { ...calendarProps }) : /* @__PURE__ */ jsx("div", { className: "p-4 flex justify-center", children: /* @__PURE__ */ jsx(TimePicker, { ...timePickerProps }) })
          ] })
        ) : (
          // Desktop mode: calendar + time picker side-by-side
          /* @__PURE__ */ jsxs("div", { className: "flex", children: [
            /* @__PURE__ */ jsx(Calendar, { ...calendarProps, className: "border-0 shadow-none rounded-none" }),
            /* @__PURE__ */ jsx(TimePicker, { ...timePickerProps })
          ] })
        )
      }
    ),
    name && /* @__PURE__ */ jsx("input", { type: "hidden", name, value })
  ] });
}

export { DateTimePicker, TimePicker };
