// @sap-ui/fx-components/dist/chunk-UXS5LRDO.js
import { TagExploration } from './chunk-NCI4MJ5T.js';
import { SegmentedButton, SegmentedButtonItem } from './chunk-ZH374RLZ.js';
import { SplitButton } from './chunk-ELIHCCZQ.js';
import { Panel } from './chunk-MQ3F6SMX.js';
import { Label } from './chunk-4PDXVETU.js';
import { Menu, MenuItem, MenuHeader, MenuSeparator } from './chunk-EHO24TXD.js';
import { ToggleButton } from './chunk-2DPLFZUY.js';
import { AddIcon, PaperPlaneIcon, MicrophoneIcon, EditIcon, SettingsIcon, UserEditIcon } from './chunk-YYSFHEKJ.js';
import { AcceptIcon } from './chunk-VNDUTC7E.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { ListItemCustom } from './chunk-SZG4UYOP.js';
import { ListItem } from './chunk-E4NK6IRO.js';
import { Avatar } from './chunk-OXMWPC4U.js';
import { PersonPlaceholderIcon } from './chunk-LCV3JX36.js';
import { List as List$1 } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { Title } from './chunk-BM2JFIU4.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import React5, { createContext, useState, useId, useContext, useRef, useEffect, useCallback, useMemo, useImperativeHandle, isValidElement, cloneElement, Children } from 'react';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { useTranslation } from 'react-i18next';
import { Share2, ChevronDown, List, Plus, X, LogOut, AudioWaveform } from 'lucide-react';
import { createPortal } from 'react-dom';

var createIcon = (pathData, fillRule) => {
  const Icon = ({ size = 20, className, ...props }) => /* @__PURE__ */ jsx(
    "svg",
    {
      viewBox: "0 0 20 20",
      width: size,
      height: size,
      fill: "currentColor",
      className,
      ...props,
      children: /* @__PURE__ */ jsx("path", { d: pathData, fillRule })
    }
  );
  return Icon;
};
var createSapIcon = (pathData, fillRule) => {
  const Icon = ({ size = 20, className, ...props }) => /* @__PURE__ */ jsx(
    "svg",
    {
      viewBox: "0 0 512 512",
      width: size,
      height: size,
      fill: "currentColor",
      className,
      ...props,
      children: /* @__PURE__ */ jsx("path", { d: pathData, fillRule })
    }
  );
  return Icon;
};
var ConversationsIcon = createIcon(
  "M0 3.6119C7.53742e-05 1.61714 1.59895 7.6231e-05 3.57129 0H10C10.5523 0 11 0.452806 11 1.01137C11 1.56994 10.5523 2.02274 10 2.02274H3.57129C2.70352 2.02282 2.00008 2.73427 2 3.6119V16.7745L5.43555 13.2999L5.50879 13.2328C5.68672 13.0852 5.91045 13.0037 6.14258 13.0036H16.4287C17.2964 13.0036 17.9998 12.292 18 11.4145V10.0594C18.0003 9.50105 18.4479 9.04802 19 9.04802C19.5521 9.04802 19.9997 9.50105 20 10.0594V11.4145C19.9998 13.4091 18.401 15.0263 16.4287 15.0264H6.55664L1.70703 19.9311C1.42103 20.2204 0.990862 20.3069 0.617188 20.1504C0.243586 19.9938 0 19.6251 0 19.2161V3.6119Z M19.043 4.16217C19.9834 4.45547 19.9834 5.80197 19.043 6.0953L17.2435 6.6568C16.9275 6.75535 16.6802 7.00564 16.5826 7.32516L16.0268 9.14585C15.7367 10.0969 14.4054 10.097 14.1154 9.14585L13.5602 7.32586C13.4628 7.00643 13.2158 6.75551 12.9 6.6568L11.0991 6.0953L10.9334 6.02825C10.162 5.63756 10.2175 4.43713 11.0991 4.16217L12.9 3.60066C13.1764 3.51428 13.3998 3.31157 13.5167 3.04824L13.5609 2.93231L14.1154 1.11162C14.4054 0.160524 15.7368 0.160522 16.0268 1.11162L16.5826 2.93231C16.668 3.21197 16.8677 3.43844 17.1282 3.55667L17.2435 3.60066L19.043 4.16217ZM16.8009 5.04982C16.0109 4.80336 15.3928 4.17823 15.1491 3.37928L15.0711 3.12297L14.993 3.37928C14.7494 4.17826 14.1313 4.8034 13.3413 5.04982L13.0885 5.12804L13.3413 5.20765L13.4877 5.25793C14.1593 5.51431 14.6898 6.05088 14.9433 6.73013L14.993 6.87819L15.0711 7.1331L15.1491 6.87819C15.3927 6.07923 16.0109 5.45411 16.8009 5.20765L17.0536 5.12804L16.8009 5.04982Z M5 11.1251C4.44772 11.1251 4 10.6723 4 10.1137C4 9.55515 4.44772 9.10234 5 9.10234L10 9.10234C10.5523 9.10234 11 9.55515 11 10.1137C11 10.6723 10.5523 11.1251 10 11.1251L5 11.1251Z M5 7.0796C4.44772 7.0796 4 6.62679 4 6.06823C4 5.50966 4.44772 5.05686 5 5.05686L8 5.05686C8.55229 5.05686 9 5.50966 9 6.06823C9 6.62679 8.55229 7.0796 8 7.0796L5 7.0796Z"
);
var DiscoverIcon = createIcon(
  "M0 10.0002C1.7068e-05 5.13124 3.36325 1.05876 7.86743 0.0259049C8.41208 -0.0989723 8.95365 0.24708 9.07712 0.797925C9.20042 1.34867 8.85933 1.89651 8.31477 2.02137C4.71172 2.8476 2.02242 6.10737 2.0224 10.0002C2.0224 13.8935 4.71212 17.1543 8.31576 17.9801C8.86042 18.1049 9.2025 18.6527 9.0791 19.2036C8.95559 19.7541 8.41386 20.0991 7.86941 19.9746C3.36445 18.9423 0 14.8698 0 10.0002ZM18.0009 11.8209C18.1247 11.2704 18.6662 10.9251 19.2106 11.0499C19.7552 11.175 20.0967 11.7236 19.973 12.2744C19.1141 16.098 16.1402 19.1055 12.3593 19.9736C11.8147 20.0986 11.2723 19.7534 11.1487 19.2026C11.0251 18.6519 11.3666 18.1043 11.911 17.9791C14.9329 17.2854 17.3145 14.877 18.0009 11.8209ZM11.1447 0.797425C11.2681 0.246651 11.8098 -0.0991971 12.3544 0.0254056C16.1399 0.892653 19.1179 3.90483 19.975 7.73362C20.0981 8.28443 19.7562 8.83237 19.2116 8.95706C18.667 9.08166 18.1252 8.73587 18.0019 8.18504C17.317 5.12515 14.9334 2.71406 11.908 2.02087C11.3635 1.89611 11.0215 1.3482 11.1447 0.797425ZM10.9143 14.909L11.771 12.1C11.8325 11.8982 11.9887 11.7402 12.1883 11.6779L15.3775 10.6831C15.9714 10.4979 15.9713 9.64778 15.3775 9.46257L12.1883 8.46789C12.1639 8.46027 12.1401 8.45121 12.117 8.44081C11.9762 8.37676 11.8616 8.26184 11.7984 8.11935C11.7878 8.09559 11.7787 8.07105 11.771 8.04582L10.7874 4.82033C10.6043 4.21975 9.76373 4.21975 9.58059 4.82032L8.59702 8.04583C8.5579 8.17415 8.48049 8.28476 8.37801 8.3643C8.36337 8.37566 8.34822 8.38639 8.3326 8.39644C8.31782 8.40595 8.30262 8.41486 8.28703 8.42314C8.253 8.4412 8.21711 8.45623 8.17969 8.46791L4.99034 9.46262C4.68288 9.55861 4.53458 9.83299 4.54554 10.1009C4.55459 10.3191 4.66924 10.5329 4.88949 10.6428L4.99033 10.6832L8.17967 11.6779C8.37924 11.7402 8.53546 11.8982 8.597 12.1L9.45351 14.909L9.5805 15.3256L9.6205 15.4276C9.73742 15.6674 9.97346 15.7834 10.2057 15.7757C10.4542 15.7673 10.6983 15.6172 10.7873 15.3256L10.9143 14.909Z"
);
var JobsIcon = createIcon(
  "M6.77487 6.94464C6.28776 6.2744 6 5.44677 6 4.55116C6 2.3169 7.79086 0.505676 10 0.505676C12.2091 0.505676 14 2.3169 14 4.55116C14 5.54361 13.6466 6.4526 13.0602 7.15644L15.5926 11.6515C15.7265 11.6378 15.8625 11.6308 16 11.6308C18.2091 11.6308 20 13.442 20 15.6762C20 17.9105 18.2091 19.7217 16 19.7217C14.1362 19.7217 12.5701 18.4325 12.126 16.6876H7.87398C7.42994 18.4325 5.86384 19.7217 4 19.7217C1.79086 19.7217 0 17.9105 0 15.6762C0 13.4626 1.75795 11.6642 3.93893 11.6312L6.77487 6.94464ZM5.96837 12.1537L8.34069 8.23323C8.84616 8.46661 9.40806 8.59665 10 8.59665C10.5007 8.59665 10.9798 8.50362 11.4216 8.33372L13.6951 12.3695C12.9318 12.9147 12.3656 13.7235 12.126 14.6649L7.87398 14.6649C7.60042 13.5899 6.90096 12.6878 5.96837 12.1537ZM10 6.5739C8.89543 6.5739 8 5.66829 8 4.55116C8 3.43403 8.89543 2.52842 10 2.52842C11.1046 2.52842 12 3.43403 12 4.55116C12 5.66829 11.1046 6.5739 10 6.5739ZM4 17.699C2.89543 17.699 2 16.7934 2 15.6762C2 14.5591 2.89543 13.6535 4 13.6535C5.10457 13.6535 6 14.5591 6 15.6762C6 16.7934 5.10457 17.699 4 17.699ZM14 15.6762C14 14.5591 14.8954 13.6535 16 13.6535C17.1046 13.6535 18 14.5591 18 15.6762C18 16.7934 17.1046 17.699 16 17.699C14.8954 17.699 14 16.7934 14 15.6762Z",
  "evenodd"
);
var SpacesIcon = createIcon(
  "M12.5722 1.39355C10.989 0.432945 9.01096 0.432946 7.42771 1.39355L2.34372 4.47902C0.40189 5.6575 0.401894 8.50357 2.34372 9.68205L3.0551 10.1138L2.34372 10.5453C0.401701 11.7237 0.401704 14.5708 2.34372 15.7493L7.42771 18.8338C9.01107 19.7945 10.9889 19.7945 12.5722 18.8338L17.6562 15.7493C19.5982 14.5708 19.5982 11.7237 17.6562 10.5453L16.9447 10.1139L17.6562 9.68205C19.5981 8.50357 19.598 5.6575 17.6562 4.47902L12.5722 1.39355ZM3.37302 12.2796L4.9985 11.2932L7.42771 12.7675C9.01096 13.7281 10.989 13.7281 12.5722 12.7675L15.0012 11.2934L16.6269 12.2796C17.2743 12.6724 17.2743 13.6221 16.6269 14.0149L11.5439 17.0994C10.5939 17.6759 9.40607 17.6759 8.45603 17.0994L3.37302 14.0149C2.72568 13.6221 2.72568 12.6724 3.37302 12.2796ZM8.45603 3.12789C9.40607 2.55138 10.5939 2.55138 11.5439 3.12789L16.6269 6.21336C17.2743 6.60618 17.2743 7.55489 16.6269 7.94771L11.5439 11.0332C10.5939 11.6097 9.40607 11.6097 8.45603 11.0332L3.37302 7.94771C2.72568 7.55489 2.72568 6.60618 3.37302 6.21336L8.45603 3.12789Z"
);
var VSCodeIcon = createIcon(
  "M14.5 0.5L18.5 2.5C19.1 2.8 19.5 3.4 19.5 4.1V15.9C19.5 16.6 19.1 17.2 18.5 17.5L14.5 19.5C13.8 19.8 13 19.7 12.4 19.2L6.5 13.5L3.3 16C2.7 16.5 1.8 16.5 1.2 16L0.3 15.2C-0.1 14.7 -0.1 14 0.3 13.5L3 10.5L0.3 7.5C-0.1 7 -0.1 6.3 0.3 5.8L1.2 5C1.8 4.5 2.7 4.5 3.3 5L6.5 7.5L12.4 1.8C13 1.3 13.8 1.2 14.5 0.5ZM14.5 5.5V15.5L9.5 10.5L14.5 5.5Z",
  "evenodd"
);
var CodeIcon = ({ size = 20, className, ...props }) => /* @__PURE__ */ jsx(
  "svg",
  {
    viewBox: "0 0 20 20",
    width: size,
    height: size,
    fill: "currentColor",
    className,
    ...props,
    children: /* @__PURE__ */ jsx("path", { d: "M6.5 4.5L1 10l5.5 5.5 1.4-1.4L3.8 10l4.1-4.1L6.5 4.5zm7 0l-1.4 1.4L16.2 10l-4.1 4.1 1.4 1.4L19 10l-5.5-5.5z" })
  }
);
CodeIcon.displayName = "CodeIcon";
var MenuIcon = ({ size = 20, className, ...props }) => /* @__PURE__ */ jsx(
  "svg",
  {
    viewBox: "0 0 20 16",
    width: size,
    height: size,
    fill: "none",
    className,
    ...props,
    children: /* @__PURE__ */ jsx("path", { d: "M0.777441 0C0.558944 0 0.373476 0.111789 0.221037 0.335366C0.073679 0.566397 0 0.838417 0 1.15142C0 1.46443 0.073679 1.73273 0.221037 1.95631C0.373476 2.17988 0.558944 2.29167 0.777441 2.29167H19.215C19.4335 2.29167 19.6189 2.17988 19.7714 1.95631C19.9238 1.73273 20 1.46443 20 1.15142C20 0.830963 19.9213 0.558944 19.7637 0.335366C19.6113 0.111789 19.4284 0 19.215 0H0.777441ZM0.777441 6.57319C0.558944 6.57319 0.373476 6.68498 0.221037 6.90855C0.073679 7.13958 0 7.4116 0 7.72461C0 8.03762 0.073679 8.30591 0.221037 8.5295C0.373476 8.75307 0.558944 8.86486 0.777441 8.86486H19.215C19.4335 8.86486 19.6189 8.75307 19.7714 8.5295C19.9238 8.30591 20 8.03762 20 7.72461C20 7.40414 19.9213 7.13213 19.7637 6.90855C19.6113 6.68498 19.4284 6.57319 19.215 6.57319H0.777441ZM0.777441 13.1464C0.558944 13.1464 0.373476 13.2582 0.221037 13.4817C0.073679 13.7128 0 13.981 0 14.2866C0 14.6071 0.073679 14.879 0.221037 15.1027C0.373476 15.3263 0.558944 15.4381 0.777441 15.4381H14.6239C14.8423 15.4381 15.0278 15.3263 15.1802 15.1027C15.3327 14.879 15.4089 14.6071 15.4089 14.2866C15.4089 13.9736 15.3302 13.7054 15.1726 13.4817C15.0202 13.2582 14.8373 13.1464 14.6239 13.1464H0.777441Z", fill: "currentColor" })
  }
);
MenuIcon.displayName = "MenuIcon";
var WaveformIcon = createIcon(
  "M3 8 A 1 1 0 0 1 4 9 V 11 A 1 1 0 0 1 3 12 A 1 1 0 0 1 2 11 V 9 A 1 1 0 0 1 3 8 Z M7 5 A 1 1 0 0 1 8 6 V 14 A 1 1 0 0 1 7 15 A 1 1 0 0 1 6 14 V 6 A 1 1 0 0 1 7 5 Z M11 3 A 1 1 0 0 1 12 4 V 16 A 1 1 0 0 1 11 17 A 1 1 0 0 1 10 16 V 4 A 1 1 0 0 1 11 3 Z M15 6 A 1 1 0 0 1 16 7 V 13 A 1 1 0 0 1 15 14 A 1 1 0 0 1 14 13 V 7 A 1 1 0 0 1 15 6 Z"
);
var PanelCloseIcon = createIcon(
  "M18.2002 3.81201C18.2 3.2311 17.6967 2.69971 17 2.69971H6.90039V17.3003H17C17.6968 17.3003 18.2002 16.7681 18.2002 16.187V3.81201ZM13.4473 7.00342C13.7665 6.69814 14.2727 6.70876 14.5781 7.02783C14.8835 7.34716 14.8721 7.85431 14.5527 8.15967L12.1572 10.4497L14.5527 12.7407C14.872 13.0461 14.8835 13.5523 14.5781 13.8716C14.2728 14.1908 13.7666 14.2022 13.4473 13.897L10.4473 11.0278C10.2895 10.8769 10.2002 10.6681 10.2002 10.4497C10.2002 10.2314 10.2895 10.0225 10.4473 9.87158L13.4473 7.00342ZM1.7998 16.187C1.7998 16.7319 2.24242 17.2331 2.87207 17.2935L3 17.3003H5.2998V2.69971H3C2.30331 2.69971 1.80003 3.2311 1.7998 3.81201V16.187ZM19.7998 16.187C19.7998 17.7184 18.5123 18.8999 17 18.8999H3C1.48771 18.8999 0.200195 17.7184 0.200195 16.187V3.81201C0.200414 2.28078 1.48784 1.1001 3 1.1001H17C18.5122 1.1001 19.7996 2.28078 19.7998 3.81201V16.187Z"
);
var PanelOpenIcon = createIcon(
  "M18.2012 3.81207C18.2009 3.23037 17.6974 2.69879 17 2.69879H6.89844V17.3013H17C17.6975 17.3013 18.2012 16.7689 18.2012 16.1871V3.81207ZM10.4229 7.02887C10.7277 6.7103 11.233 6.69968 11.5518 7.00446L14.5518 9.87262C14.7093 10.0232 14.7988 10.2318 14.7988 10.4498C14.7988 10.6678 14.7093 10.8762 14.5518 11.0269L11.5518 13.8961C11.233 14.2007 10.7276 14.1894 10.4229 13.8707C10.118 13.5519 10.1295 13.0466 10.4482 12.7418L12.8438 10.4498L10.4482 8.15875C10.1294 7.85391 10.118 7.34766 10.4229 7.02887ZM1.79883 16.1871C1.79883 16.7324 2.24123 17.2347 2.87109 17.2955L3 17.3013H5.30176V2.69879H3C2.30263 2.69879 1.79905 3.23037 1.79883 3.81207V16.1871ZM19.7988 16.1871C19.7988 17.7177 18.5116 18.899 17 18.899H3C1.48839 18.899 0.201172 17.7177 0.201172 16.1871V3.81207C0.201377 2.3772 1.3328 1.24998 2.71973 1.11481L3 1.10114H17C18.5115 1.10114 19.7986 2.28163 19.7988 3.81207V16.1871Z"
);
var CursorIcon = createIcon("M4 1L4 15L7.5 11.5L10.5 18L13 17L10 10H15L4 1Z");
var BellIcon = createSapIcon(
  "M475 374q5 7 5 16 0 11-7 18.5t-18 7.5H334q-5 27-27 45.5T256 480t-51-18.5-27-45.5H58q-11 0-18.5-7.5T32 390q0-10 6-16 1-1 8-9.5T61 341t14.5-35.5T82 260v-20q0-100 45-154t129-54 129.5 54T431 240v20q0 25 6.5 45.5T452 341t15 23.5 8 9.5zm-69-9q-11-20-19-46.5t-8-58.5v-20q0-32-5.5-60t-20-50-38-34.5T256 83t-59.5 12.5-38 34.5-20 50-5.5 60v20q0 32-8 58.5T106 365h300z"
);
BellIcon.displayName = "BellIcon";
var OpenCommandFieldIcon = createSapIcon(
  "M90 416q-11 0-18.5-7.5T64 390q0-10 8-18l121-116L72 140q-8-8-8-18 0-11 7.5-18.5T90 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L107 409q-7 7-17 7zm192 0q-11 0-18.5-7.5T256 390q0-10 8-18l121-116-121-116q-8-8-8-18 0-11 7.5-18.5T282 96q10 0 17 7l141 134q8 8 8 19 0 12-8 18L299 409q-7 7-17 7z"
);
OpenCommandFieldIcon.displayName = "OpenCommandFieldIcon";
var CloseCommandFieldIcon = createSapIcon(
  "M230 416q-10 0-17-7L72 274q-8-6-8-18 0-11 8-19l141-134q7-7 17-7 11 0 18.5 7.5T256 122q0 10-8 18L127 256l121 116q8 8 8 18 0 11-7.5 18.5T230 416zm192 0q-10 0-17-7L264 274q-8-6-8-18 0-11 8-19l141-134q7-7 17-7 11 0 18.5 7.5T448 122q0 10-8 18L319 256l121 116q8 8 8 18 0 11-7.5 18.5T422 416z"
);
CloseCommandFieldIcon.displayName = "CloseCommandFieldIcon";
var JouleIcon = ({ size = 20, className, isHovered: isHoveredProp, ...props }) => {
  const [selfHovered, setSelfHovered] = useState(false);
  const hovered = isHoveredProp ?? selfHovered;
  const uid = useId();
  const id = (n) => `joule-${uid}-${n}`;
  const t = { transition: "fill 0.3s ease" };
  return /* @__PURE__ */ jsxs(
    "svg",
    {
      viewBox: "0 0 39 35",
      width: size,
      height: size,
      fill: "none",
      className,
      onMouseEnter: isHoveredProp === void 0 ? () => setSelfHovered(true) : void 0,
      onMouseLeave: isHoveredProp === void 0 ? () => setSelfHovered(false) : void 0,
      ...props,
      children: [
        /* @__PURE__ */ jsxs("mask", { id: id("mask"), style: { maskType: "alpha" }, maskUnits: "userSpaceOnUse", x: "2", y: "1", width: "34", height: "33", children: [
          /* @__PURE__ */ jsx("path", { d: "M26.5889 1.00195H11.4111C9.90477 1.00195 8.49812 1.75477 7.66257 3.00809L2.76809 10.3498C1.70855 11.9391 1.76641 14.0238 2.91248 15.5519L15.3959 32.1965C17.1979 34.5992 20.8021 34.5992 22.6041 32.1965L35.0875 15.5519C36.2336 14.0238 36.2915 11.9391 35.2319 10.3498L30.3374 3.0081C29.5019 1.75477 28.0952 1.00195 26.5889 1.00195Z", fill: "white" }),
          /* @__PURE__ */ jsx("path", { d: "M26.5889 1.00195H11.4111C9.90477 1.00195 8.49812 1.75477 7.66257 3.00809L2.76809 10.3498C1.70855 11.9391 1.76641 14.0238 2.91248 15.5519L15.3959 32.1965C17.1979 34.5992 20.8021 34.5992 22.6041 32.1965L35.0875 15.5519C36.2336 14.0238 36.2915 11.9391 35.2319 10.3498L30.3374 3.0081C29.5019 1.75477 28.0952 1.00195 26.5889 1.00195Z", fill: "black", fillOpacity: "0.2" })
        ] }),
        /* @__PURE__ */ jsxs("g", { mask: `url(#${id("mask")})`, children: [
          /* @__PURE__ */ jsx("path", { d: "M25.6691 13.002H12.3359L19.0023 37.002L25.6691 13.002Z", fill: `url(#${id("g0")})`, style: t }),
          /* @__PURE__ */ jsx("path", { d: "M1 13.002L19 37.002L12.3336 13.002H1Z", fill: `url(#${id("g1")})`, style: t }),
          /* @__PURE__ */ jsx("path", { d: "M37 13.002L19 37.002L25.6664 13.002H37Z", fill: `url(#${id("g2")})`, style: t }),
          /* @__PURE__ */ jsx("path", { d: "M9 1.00195L12.3334 13.002H25.6667L29 1.00195H9Z", fill: `url(#${id("g3")})`, style: t }),
          /* @__PURE__ */ jsx("path", { d: "M1 13.002H12.3336L9 1.00195L1 13.002Z", fill: `url(#${id("g4")})`, style: t }),
          /* @__PURE__ */ jsx("path", { d: "M37 13.002H25.6664L29 1.00195L37 13.002Z", fill: `url(#${id("g5")})`, style: t }),
          /* @__PURE__ */ jsx("path", { d: "M26.6284 7.74902C26.5172 7.30423 26.1175 6.99219 25.6591 6.99219C25.2006 6.99219 24.8009 7.30423 24.6897 7.74902L24.563 8.25609C24.1155 10.046 22.7179 11.4436 20.928 11.8911L20.4209 12.0179C19.9761 12.1291 19.6641 12.5287 19.6641 12.9872C19.6641 13.4457 19.9761 13.8453 20.4209 13.9565L20.928 14.0833C22.7179 14.5308 24.1155 15.9284 24.563 17.7183L24.6897 18.2254C24.8009 18.6702 25.2006 18.9822 25.6591 18.9822C26.1175 18.9822 26.5172 18.6702 26.6284 18.2254L26.7552 17.7183C27.2026 15.9284 28.6002 14.5308 30.3902 14.0833L30.8972 13.9565C31.342 13.8453 31.6541 13.4457 31.6541 12.9872C31.6541 12.5287 31.342 12.1291 30.8972 12.0179L30.3902 11.8911C28.6002 11.4436 27.2026 10.046 26.7552 8.25609L26.6284 7.74902Z", fill: "white" })
        ] }),
        /* @__PURE__ */ jsxs("defs", { children: [
          /* @__PURE__ */ jsxs("linearGradient", { id: id("g0"), x1: "15.2204", y1: "28.942", x2: "30.3063", y2: "19.5615", gradientUnits: "userSpaceOnUse", children: [
            /* @__PURE__ */ jsx("stop", { offset: "0.0785293", stopColor: hovered ? "#5a2aff" : "#4013E3" }),
            /* @__PURE__ */ jsx("stop", { offset: "1", stopColor: hovered ? "#ffa8f5" : "#F090EB" })
          ] }),
          /* @__PURE__ */ jsxs("linearGradient", { id: id("g1"), x1: "15.44", y1: "19.962", x2: "7.32", y2: "23.762", gradientUnits: "userSpaceOnUse", children: [
            /* @__PURE__ */ jsx("stop", { stopColor: hovered ? "#6e4dff" : "#5739F5" }),
            /* @__PURE__ */ jsx("stop", { offset: "1", stopColor: hovered ? "#3b20c9" : "#2815A6" })
          ] }),
          /* @__PURE__ */ jsxs("linearGradient", { id: id("g2"), x1: "26.28", y1: "11.462", x2: "32.36", y2: "25.002", gradientUnits: "userSpaceOnUse", children: [
            /* @__PURE__ */ jsx("stop", { stopColor: hovered ? "#ffc8f2" : "#F3B2E9" }),
            /* @__PURE__ */ jsx("stop", { offset: "1", stopColor: hovered ? "#5a2aff" : "#4013E3" })
          ] }),
          /* @__PURE__ */ jsxs("linearGradient", { id: id("g3"), x1: "26.28", y1: "16.6615", x2: "10.34", y2: "5.44149", gradientUnits: "userSpaceOnUse", children: [
            /* @__PURE__ */ jsx("stop", { stopColor: hovered ? "#ffc8f2" : "#F3B2E9" }),
            /* @__PURE__ */ jsx("stop", { offset: "0.995192", stopColor: hovered ? "#5a2aff" : "#4013E3" })
          ] }),
          /* @__PURE__ */ jsxs("linearGradient", { id: id("g4"), x1: "11.68", y1: "11.402", x2: "0.26", y2: "7.88195", gradientUnits: "userSpaceOnUse", children: [
            /* @__PURE__ */ jsx("stop", { stopColor: hovered ? "#6e4dff" : "#5739F5" }),
            /* @__PURE__ */ jsx("stop", { offset: "1", stopColor: hovered ? "#3b20c9" : "#2815A6" })
          ] }),
          /* @__PURE__ */ jsxs("linearGradient", { id: id("g5"), x1: "42.1", y1: "3.44195", x2: "28.7342", y2: "16.5549", gradientUnits: "userSpaceOnUse", children: [
            /* @__PURE__ */ jsx("stop", { stopColor: hovered ? "#8a6fff" : "#735AF6" }),
            /* @__PURE__ */ jsx("stop", { offset: "1", stopColor: hovered ? "#ffa8f5" : "#F090EB" })
          ] })
        ] })
      ]
    }
  );
};
JouleIcon.displayName = "JouleIcon";
var MoreIcon = ({ size = 20, className, ...props }) => /* @__PURE__ */ jsx(
  "svg",
  {
    viewBox: "0 0 20 20",
    width: size,
    height: size,
    fill: "currentColor",
    className,
    ...props,
    children: /* @__PURE__ */ jsx("path", { d: "M2.5 7.5C3.88071 7.5 5 8.61929 5 10C5 11.3807 3.88071 12.5 2.5 12.5C1.11929 12.5 0 11.3807 0 10C0 8.61929 1.11929 7.5 2.5 7.5ZM10 7.5C11.3807 7.5 12.5 8.61929 12.5 10C12.5 11.3807 11.3807 12.5 10 12.5C8.61929 12.5 7.5 11.3807 7.5 10C7.5 8.61929 8.61929 7.5 10 7.5ZM17.5 7.5C18.8807 7.5 20 8.61929 20 10C20 11.3807 18.8807 12.5 17.5 12.5C16.1193 12.5 15 11.3807 15 10C15 8.61929 16.1193 7.5 17.5 7.5Z" })
  }
);
MoreIcon.displayName = "MoreIcon";
var NewConversationIcon = ({ size = 20, className, ...props }) => /* @__PURE__ */ jsxs(
  "svg",
  {
    viewBox: "0 0 20 20",
    width: size,
    height: size,
    fill: "currentColor",
    className,
    ...props,
    children: [
      /* @__PURE__ */ jsx("path", { d: "M0 3.57129C7.53742e-05 1.59896 1.59895 7.5374e-05 3.57129 0H10C10.5523 0 11 0.447716 11 1C11 1.55229 10.5523 2 10 2H3.57129C2.70352 2.00008 2.00008 2.70353 2 3.57129V16.586L5.43555 13.1504L5.50879 13.084C5.68672 12.9381 5.91045 12.8575 6.14258 12.8574H16.4287C17.2964 12.8574 17.9998 12.1538 18 11.2861V9.9463C18.0003 9.39424 18.4479 8.9463 19 8.9463C19.5521 8.9463 19.9997 9.39424 20 9.9463V11.2861C19.9998 13.2584 18.401 14.8574 16.4287 14.8574H6.55664L1.70703 19.7071C1.42103 19.9931 0.990862 20.0786 0.617188 19.9239C0.243586 19.769 0 19.4044 0 19V3.57129Z" }),
      /* @__PURE__ */ jsx("path", { d: "M5 11C4.44772 11 4 10.5523 4 10C4 9.44773 4.44772 9.00001 5 9.00001H10C10.5523 9.00001 11 9.44773 11 10C11 10.5523 10.5523 11 10 11H5Z" }),
      /* @__PURE__ */ jsx("path", { d: "M5 7.00001C4.44772 7.00001 4 6.55229 4 6.00001C4 5.44772 4.44772 5.00001 5 5.00001L8 5.00001C8.55229 5.00001 9 5.44772 9 6.00001C9 6.55229 8.55229 7.00001 8 7.00001H5Z" }),
      /* @__PURE__ */ jsx("path", { d: "M16.8571 0.847507C16.8571 0.379442 16.4734 0 16 0C15.5266 0 15.1429 0.379442 15.1429 0.847507V3.10752H12.8571C12.3838 3.10752 12 3.48697 12 3.95503C12 4.42307 12.3838 4.80254 12.8571 4.80254H15.1429V7.06256C15.1429 7.53059 15.5266 7.91006 16 7.91006C16.4734 7.91006 16.8571 7.53059 16.8571 7.06256V4.80254L19.231 4.79813C19.663 4.75448 20 4.39364 20 3.95503C20 3.5164 19.663 3.15558 19.231 3.11194L16.8571 3.10752V0.847507Z" })
    ]
  }
);
NewConversationIcon.displayName = "NewConversationIcon";
var FxSideNavigationContext = createContext(null);
var useFxSideNavigationContext = () => {
  const context = useContext(FxSideNavigationContext);
  return context;
};
function countNavigationItems(children) {
  let count = 0;
  Children.forEach(children, (child) => {
    if (isValidElement(child)) {
      count++;
    }
  });
  return count;
}
function enrichChildrenWithAria(children, totalCount, startIndex = 1) {
  let currentIndex = startIndex;
  return Children.map(children, (child) => {
    if (isValidElement(child)) {
      const ariaRoleDescription = `navigation item, ${currentIndex} of ${totalCount}`;
      currentIndex++;
      return cloneElement(child, { ariaRoleDescription });
    }
    return child;
  });
}
function FxSideNavigation({
  collapsed = false,
  isTransitioning = false,
  transitionDirection = null,
  forceMerged = false,
  dialogMode = false,
  children,
  fixedItems,
  header,
  onSelectionChange,
  onMergeChange: _onMergeChange,
  // Deprecated - merge is now controlled by parent
  className,
  style,
  ref
}) {
  const { t } = useTranslation("fx");
  const containerRef = useRef(null);
  const flexibleRef = useRef(null);
  const fixedRef = useRef(null);
  const flyoutCloseHandlersRef = useRef(/* @__PURE__ */ new Map());
  const itemRefsRef = useRef(/* @__PURE__ */ new Map());
  const itemOrderRef = useRef([]);
  const registerFlyoutClose = useCallback((name, handler) => {
    flyoutCloseHandlersRef.current.set(name, handler);
  }, []);
  const unregisterFlyoutClose = useCallback((name) => {
    flyoutCloseHandlersRef.current.delete(name);
  }, []);
  const registerItemRef = useCallback((name, ref2) => {
    if (ref2) {
      itemRefsRef.current.set(name, ref2);
      const allItems = Array.from(itemRefsRef.current.entries());
      allItems.sort((a, b) => {
        const aRect = a[1].getBoundingClientRect();
        const bRect = b[1].getBoundingClientRect();
        return aRect.top - bRect.top;
      });
      itemOrderRef.current = allItems.map(([name2]) => name2);
    } else {
      itemRefsRef.current.delete(name);
      itemOrderRef.current = itemOrderRef.current.filter((n) => n !== name);
    }
  }, []);
  const navigateToItem = useCallback((currentName, direction) => {
    const order = itemOrderRef.current;
    const currentIndex = order.indexOf(currentName);
    if (currentIndex === -1) return;
    let targetIndex;
    if (direction === "next") {
      targetIndex = currentIndex + 1;
      if (targetIndex >= order.length) targetIndex = 0;
    } else {
      targetIndex = currentIndex - 1;
      if (targetIndex < 0) targetIndex = order.length - 1;
    }
    const targetName = order[targetIndex];
    const targetRef = itemRefsRef.current.get(targetName);
    targetRef?.focus();
  }, []);
  const closeFlyout = useCallback((itemName) => {
    const handler = flyoutCloseHandlersRef.current.get(itemName);
    handler?.();
  }, []);
  const closeOtherFlyouts = useCallback((exceptName) => {
    flyoutCloseHandlersRef.current.forEach((handler, name) => {
      if (name !== exceptName) {
        handler();
      }
    });
  }, []);
  useImperativeHandle(ref, () => ({
    closeFlyout,
    get isMerged() {
      return forceMerged;
    },
    get nativeElement() {
      return containerRef.current;
    },
    getNativeElement: () => containerRef.current
  }));
  const handleItemClick = useCallback(
    (name) => {
      onSelectionChange?.({ name });
    },
    [onSelectionChange]
  );
  const totalCount = useMemo(() => countNavigationItems(children), [children]);
  const enrichedChildren = useMemo(() => enrichChildrenWithAria(children, totalCount, 1), [children, totalCount]);
  const flexibleContextValue = useMemo(
    () => ({
      collapsed,
      isTransitioning,
      transitionDirection,
      isFixedSection: false,
      onItemClick: handleItemClick,
      registerFlyoutClose,
      unregisterFlyoutClose,
      closeOtherFlyouts,
      registerItemRef,
      navigateToItem
    }),
    [collapsed, isTransitioning, transitionDirection, handleItemClick, registerFlyoutClose, unregisterFlyoutClose, closeOtherFlyouts, registerItemRef, navigateToItem]
  );
  const fixedContextValue = useMemo(
    () => ({
      collapsed,
      isTransitioning,
      transitionDirection,
      isFixedSection: true,
      onItemClick: handleItemClick,
      registerFlyoutClose,
      unregisterFlyoutClose,
      closeOtherFlyouts,
      registerItemRef,
      navigateToItem
    }),
    [collapsed, isTransitioning, transitionDirection, handleItemClick, registerFlyoutClose, unregisterFlyoutClose, closeOtherFlyouts, registerItemRef, navigateToItem]
  );
  return /* @__PURE__ */ jsxs(
    "nav",
    {
      ref: containerRef,
      className: cn(
        "fx-side-navigation flex flex-col bg-sapphire-shell-bg-primary pb-6",
        !dialogMode && "h-full",
        collapsed && "collapsed",
        forceMerged && "merged",
        className
      ),
      style: {
        overflow: "visible",
        // Allow flyouts to extend beyond nav
        paddingLeft: "1.25rem",
        paddingRight: "1.25rem",
        ...style
      },
      children: [
        header && /* @__PURE__ */ jsx(
          "div",
          {
            className: "flex items-center shrink-0 py-6 overflow-hidden",
            style: { paddingLeft: "0.25rem" },
            children: header
          }
        ),
        !dialogMode && /* @__PURE__ */ jsx(
          "div",
          {
            className: forceMerged ? "flex-1" : "shrink-0",
            style: {
              height: forceMerged ? void 0 : "10rem",
              minHeight: 0,
              transition: "height 200ms ease-out"
            }
          }
        ),
        forceMerged ? (
          // Merged: single list with divider (no flex spacer - items flow naturally after top spacer)
          /* @__PURE__ */ jsx(
            "div",
            {
              ref: flexibleRef,
              className: "relative",
              style: { overflow: "visible" },
              children: /* @__PURE__ */ jsxs(
                "ul",
                {
                  role: "list",
                  "aria-label": t("FX_MAIN_NAVIGATION"),
                  className: "list-none p-0 m-0 flex flex-col gap-6",
                  children: [
                    /* @__PURE__ */ jsx(FxSideNavigationContext.Provider, { value: flexibleContextValue, children: enrichedChildren }),
                    fixedItems && /* @__PURE__ */ jsxs(Fragment, { children: [
                      /* @__PURE__ */ jsx("li", { role: "separator", className: "my-1.5 h-px bg-sapphire-border-primary" }),
                      /* @__PURE__ */ jsx(FxSideNavigationContext.Provider, { value: fixedContextValue, children: fixedItems })
                    ] })
                  ]
                }
              )
            }
          )
        ) : (
          // Separate: flexible list + spacer + fixed list
          /* @__PURE__ */ jsxs(Fragment, { children: [
            /* @__PURE__ */ jsx(
              "div",
              {
                ref: flexibleRef,
                className: "shrink-0 relative",
                style: { overflow: "visible" },
                children: /* @__PURE__ */ jsx(
                  "ul",
                  {
                    role: "list",
                    "aria-label": t("FX_MAIN_NAVIGATION"),
                    className: "list-none p-0 m-0 flex flex-col gap-6",
                    children: /* @__PURE__ */ jsx(FxSideNavigationContext.Provider, { value: flexibleContextValue, children: enrichedChildren })
                  }
                )
              }
            ),
            /* @__PURE__ */ jsx("div", { className: "flex-1" }),
            fixedItems && /* @__PURE__ */ jsx(
              "div",
              {
                ref: fixedRef,
                className: "shrink-0 relative",
                style: { overflow: "visible" },
                children: /* @__PURE__ */ jsx("div", { role: "toolbar", "aria-label": t("FX_USER_ACTIONS"), className: "flex flex-col gap-6", children: /* @__PURE__ */ jsx(FxSideNavigationContext.Provider, { value: fixedContextValue, children: fixedItems }) })
              }
            )
          ] })
        )
      ]
    }
  );
}
function useFlyoutState(name, locked, disabled, skipFlyoutOnFocusRef) {
  const [isHovered, setIsHovered] = useState(false);
  const [isFocused, setIsFocused] = useState(false);
  const [isAddButtonFocused, setIsAddButtonFocused] = useState(false);
  const isMounted = typeof window !== "undefined";
  const hoverTimeoutRef = useRef(null);
  const isMouseInsideRef = useRef(false);
  const prevLockedRef = useRef(locked);
  const context = useFxSideNavigationContext();
  useEffect(() => {
    if (context?.registerFlyoutClose) {
      const closeHandler = () => {
        setIsHovered(false);
        setIsFocused(false);
      };
      context.registerFlyoutClose(name, closeHandler);
      return () => context.unregisterFlyoutClose(name);
    }
  }, [context, name]);
  useEffect(() => {
    if (locked && !prevLockedRef.current) {
      setTimeout(() => {
        setIsHovered(false);
        setIsFocused(false);
      }, 0);
    } else if (!locked && prevLockedRef.current) {
      if (isMouseInsideRef.current) {
        setTimeout(() => {
          setIsHovered(true);
        }, 0);
      }
    }
    prevLockedRef.current = locked;
  }, [locked]);
  const clearHoverTimeout = useCallback(() => {
    if (hoverTimeoutRef.current) {
      clearTimeout(hoverTimeoutRef.current);
      hoverTimeoutRef.current = null;
    }
  }, []);
  useEffect(() => {
    return () => {
      clearHoverTimeout();
    };
  }, [clearHoverTimeout]);
  const handleMouseEnter = useCallback(() => {
    isMouseInsideRef.current = true;
    if (locked || disabled) return;
    clearHoverTimeout();
    context?.closeOtherFlyouts(name);
    setIsHovered(true);
  }, [locked, disabled, clearHoverTimeout, context, name]);
  const handleMouseLeave = useCallback(() => {
    isMouseInsideRef.current = false;
    if (!isFocused && !isAddButtonFocused) {
      hoverTimeoutRef.current = setTimeout(() => {
        setIsHovered(false);
      }, 100);
    }
  }, [isFocused, isAddButtonFocused]);
  const handleFocus = useCallback(() => {
    if (locked || disabled) return;
    if (skipFlyoutOnFocusRef?.current) return;
    clearHoverTimeout();
    context?.closeOtherFlyouts(name);
    setIsFocused(true);
    setIsHovered(true);
  }, [locked, disabled, clearHoverTimeout, context, name, skipFlyoutOnFocusRef]);
  const handleBlur = useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    const container = e.currentTarget.closest(".relative");
    if (relatedTarget) {
      if (container?.contains(relatedTarget)) {
        return;
      }
      if (relatedTarget.closest('[data-flyout-child="true"]')) {
        return;
      }
    }
    setIsFocused(false);
    if (!isMouseInsideRef.current) {
      hoverTimeoutRef.current = setTimeout(() => {
        setIsHovered(false);
      }, 100);
    }
  }, []);
  const handleFlyoutMouseEnter = useCallback(() => {
    isMouseInsideRef.current = true;
    clearHoverTimeout();
  }, [clearHoverTimeout]);
  const handleFlyoutMouseLeave = useCallback(() => {
    isMouseInsideRef.current = false;
    if (!isFocused && !isAddButtonFocused) {
      hoverTimeoutRef.current = setTimeout(() => {
        setIsHovered(false);
      }, 100);
    }
  }, [isFocused, isAddButtonFocused]);
  return {
    isHovered,
    isFocused,
    isAddButtonFocused,
    isMounted,
    setIsHovered,
    setIsFocused,
    setIsAddButtonFocused,
    handleMouseEnter,
    handleMouseLeave,
    handleFocus,
    handleBlur,
    handleFlyoutMouseEnter,
    handleFlyoutMouseLeave
  };
}
function useItemInteractions(name, disabled, onClick, setIsHovered, setIsFocused, itemRef) {
  const context = useFxSideNavigationContext();
  const handleClick = useCallback(
    (e) => {
      e.preventDefault();
      if (disabled) return;
      setIsHovered?.(false);
      setIsFocused?.(false);
      if (onClick) {
        onClick(e);
      } else {
        context?.onItemClick(name);
      }
      setTimeout(() => {
        itemRef?.current?.focus();
      }, 0);
    },
    [disabled, onClick, context, name, setIsHovered, setIsFocused, itemRef]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === "ArrowDown") {
        e.preventDefault();
        setIsHovered?.(false);
        setIsFocused?.(false);
        context?.navigateToItem(name, "next");
      } else if (e.key === "ArrowUp") {
        e.preventDefault();
        setIsHovered?.(false);
        setIsFocused?.(false);
        context?.navigateToItem(name, "prev");
      } else if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        if (disabled) return;
        setIsHovered?.(false);
        setIsFocused?.(false);
        if (onClick) {
          onClick(e);
        } else {
          context?.onItemClick(name);
        }
        setTimeout(() => {
          itemRef?.current?.focus();
        }, 0);
      } else if (e.key === "Escape") {
        setIsHovered?.(false);
        setIsFocused?.(false);
        itemRef?.current?.blur();
      }
    },
    [context, name, disabled, onClick, setIsHovered, setIsFocused, itemRef]
  );
  return {
    handleClick,
    handleKeyDown
  };
}
function FxSideNavigationItem({
  name,
  icon,
  text,
  badge,
  showAddButton = false,
  addButtonTooltip,
  selected = false,
  disabled = false,
  locked = false,
  avatar,
  onClick,
  onAddClick,
  className,
  ariaRoleDescription,
  ref
}) {
  const itemRef = useRef(null);
  const flyoutRef = useRef(null);
  const skipFlyoutOnFocusRef = useRef(false);
  const focusTimerRef = useRef(null);
  const rafRef = useRef(null);
  const flagResetTimerRef = useRef(null);
  const [showForcedFocusRing, setShowForcedFocusRing] = useState(false);
  const context = useFxSideNavigationContext();
  const collapsed = context?.collapsed ?? false;
  const isTransitioning = context?.isTransitioning ?? false;
  const isFixedSection = context?.isFixedSection ?? false;
  const flyoutState = useFlyoutState(name, locked, disabled, skipFlyoutOnFocusRef);
  useEffect(() => {
    if (context?.registerItemRef && itemRef.current) {
      context.registerItemRef(name, itemRef.current);
      return () => context.registerItemRef(name, null);
    }
  }, [context, name]);
  useEffect(() => {
    return () => {
      if (focusTimerRef.current !== null) clearTimeout(focusTimerRef.current);
      if (rafRef.current !== null) cancelAnimationFrame(rafRef.current);
      if (flagResetTimerRef.current !== null) clearTimeout(flagResetTimerRef.current);
    };
  }, []);
  const { handleClick, handleKeyDown } = useItemInteractions(
    name,
    disabled,
    onClick,
    flyoutState.setIsHovered,
    flyoutState.setIsFocused,
    itemRef
  );
  const { setIsAddButtonFocused, setIsHovered, setIsFocused } = flyoutState;
  const handleAddClick = useCallback(
    (detail) => {
      detail.originalEvent.stopPropagation();
      onAddClick?.();
      if (collapsed) {
        skipFlyoutOnFocusRef.current = true;
        setIsAddButtonFocused(false);
        setIsHovered(false);
        setIsFocused(false);
        focusTimerRef.current = setTimeout(() => {
          itemRef.current?.blur();
          rafRef.current = requestAnimationFrame(() => {
            if (itemRef.current) {
              setShowForcedFocusRing(true);
              itemRef.current.focus();
            }
            flagResetTimerRef.current = setTimeout(() => {
              skipFlyoutOnFocusRef.current = false;
            }, 50);
          });
        }, 100);
      }
    },
    [onAddClick, setIsAddButtonFocused, setIsHovered, setIsFocused, collapsed]
  );
  useImperativeHandle(ref, () => ({
    closeFlyout: () => {
      flyoutState.setIsHovered(false);
      flyoutState.setIsFocused(false);
    },
    get nativeElement() {
      return itemRef.current;
    },
    getNativeElement: () => itemRef.current
  }));
  const settledCollapsed = collapsed && !isTransitioning;
  const showFlyout = flyoutState.isMounted && settledCollapsed && (flyoutState.isHovered || flyoutState.isFocused || flyoutState.isAddButtonFocused) && !locked;
  const canShowAddButton = showAddButton && !isFixedSection;
  const showBadgeDot = settledCollapsed && !showFlyout && badge;
  const Element = isFixedSection ? "button" : "a";
  const elementProps = isFixedSection ? { type: "button" } : { href: "#" };
  const content = /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        "relative flex items-center",
        !settledCollapsed && canShowAddButton && "gap-2 rounded pr-3",
        !settledCollapsed && canShowAddButton && selected && "bg-sapphire-chrome-button-bg-selected",
        !settledCollapsed && canShowAddButton && !disabled && !selected && "hover:bg-sapphire-chrome-button-bg-selected",
        !settledCollapsed && canShowAddButton && !flyoutState.isAddButtonFocused && "focus-within:ring-2 focus-within:ring-ring focus-within:ring-offset-0"
      ),
      style: { zIndex: showFlyout ? 50 : "auto" },
      onMouseEnter: flyoutState.handleMouseEnter,
      onMouseLeave: flyoutState.handleMouseLeave,
      children: [
        /* @__PURE__ */ jsxs(
          Element,
          {
            ref: itemRef,
            ...elementProps,
            "data-name": name,
            className: cn(
              "fx-side-nav-item relative",
              "flex items-center text-sm rounded cursor-pointer no-underline overflow-hidden min-w-10",
              !settledCollapsed && "gap-3",
              !settledCollapsed && (canShowAddButton ? "flex-1" : "w-full"),
              selected ? "text-sapphire-text-accent" : "text-sapphire-chrome-button-fg-default",
              "focus-visible:outline-none",
              !canShowAddButton && "focus-visible:ring-2 focus-visible:ring-ring",
              showForcedFocusRing && settledCollapsed && !showFlyout && "ring-2 ring-ring",
              settledCollapsed && !showFlyout && !disabled && "hover:bg-sapphire-chrome-button-bg-selected",
              settledCollapsed && selected && !showFlyout && "bg-sapphire-chrome-button-bg-selected",
              !settledCollapsed && !canShowAddButton && selected && "bg-sapphire-chrome-button-bg-selected",
              !settledCollapsed && !canShowAddButton && !disabled && !selected && "hover:bg-sapphire-chrome-button-bg-selected",
              settledCollapsed && showFlyout && "opacity-0",
              disabled && "opacity-50 cursor-not-allowed pointer-events-none",
              isFixedSection && "border-0 bg-transparent text-left",
              className
            ),
            style: {
              padding: "0.625rem"
            },
            onClick: handleClick,
            onKeyDown: handleKeyDown,
            onFocus: flyoutState.handleFocus,
            onBlur: (e) => {
              flyoutState.handleBlur(e);
              setShowForcedFocusRing(false);
            },
            onMouseEnter: !settledCollapsed ? flyoutState.handleMouseEnter : void 0,
            onMouseLeave: !settledCollapsed ? flyoutState.handleMouseLeave : void 0,
            title: settledCollapsed && !showFlyout ? text : void 0,
            "aria-expanded": settledCollapsed ? showFlyout : void 0,
            "aria-current": selected ? "page" : void 0,
            "aria-label": settledCollapsed ? text : badge ? `${text} (${badge})` : text,
            "aria-roledescription": !isFixedSection ? ariaRoleDescription : void 0,
            "data-label": text,
            children: [
              /* @__PURE__ */ jsx("span", { className: "shrink-0 flex items-center justify-center h-5 w-5", "aria-hidden": "true", children: avatar && isFixedSection ? avatar : icon }),
              !settledCollapsed && /* @__PURE__ */ jsxs(Fragment, { children: [
                /* @__PURE__ */ jsx(
                  "span",
                  {
                    className: "flex-1 truncate text-left font-semibold whitespace-nowrap overflow-hidden",
                    style: {
                      opacity: collapsed ? 0 : 1,
                      transition: isTransitioning ? "opacity 150ms ease" : "none"
                    },
                    children: text
                  }
                ),
                badge && /* @__PURE__ */ jsx(
                  "span",
                  {
                    className: "flex items-center justify-center min-w-4 h-4 px-1 text-xs font-semibold rounded-full bg-sapphire-negative text-sapphire-text-on-surface border border-sapphire-text-on-surface shrink-0",
                    style: {
                      fontSize: "0.625rem",
                      opacity: collapsed ? 0 : 1,
                      transition: isTransitioning ? "opacity 150ms ease" : "none"
                    },
                    "aria-hidden": "true",
                    children: badge
                  }
                )
              ] })
            ]
          }
        ),
        canShowAddButton && !collapsed && /* @__PURE__ */ jsx(
          Button,
          {
            design: "Tertiary",
            iconOnly: true,
            icon: /* @__PURE__ */ jsx(AddIcon, { className: "h-4 w-4" }),
            onClick: handleAddClick,
            disabled,
            tooltip: addButtonTooltip,
            "aria-label": addButtonTooltip || `Add new ${text?.toLowerCase()}`,
            tabIndex: 0,
            onFocus: () => setIsAddButtonFocused(true),
            onBlur: () => setIsAddButtonFocused(false),
            className: "h-5 w-5 text-sapphire-chrome-button-fg-default shrink-0",
            style: {
              minWidth: "1.25rem",
              minHeight: "1.25rem",
              borderRadius: "0.25rem"
            }
          }
        ),
        showBadgeDot && /* @__PURE__ */ jsx(
          "span",
          {
            className: "absolute w-2 h-2 rounded-full pointer-events-none bg-sapphire-negative",
            style: { top: "2px", right: collapsed ? "9px" : "2px" },
            "aria-hidden": "true"
          }
        ),
        showFlyout && /* @__PURE__ */ jsxs(
          "div",
          {
            ref: flyoutRef,
            "data-name": name,
            "data-flyout-child": "true",
            onMouseEnter: flyoutState.handleFlyoutMouseEnter,
            onMouseLeave: flyoutState.handleFlyoutMouseLeave,
            onClick: handleClick,
            className: cn(
              "absolute top-0 left-0 flex items-center gap-3 text-sm rounded p-2.5 pr-3 cursor-pointer",
              "shadow-[0_4px_12px_rgba(0,0,0,0.15)]",
              flyoutState.isFocused && !flyoutState.isAddButtonFocused && "ring-2 ring-ring",
              disabled && "opacity-50 cursor-not-allowed",
              selected ? "bg-sapphire-chrome-button-bg-selected text-sapphire-text-accent" : "bg-sapphire-chrome-button-bg-default text-sapphire-chrome-button-fg-default"
            ),
            style: {
              whiteSpace: "nowrap",
              position: "absolute",
              zIndex: 10
            },
            children: [
              /* @__PURE__ */ jsx("span", { className: "shrink-0 flex items-center justify-center h-5 w-5", "aria-hidden": "true", children: avatar && isFixedSection ? avatar : icon }),
              /* @__PURE__ */ jsx("span", { className: "whitespace-nowrap font-semibold", "aria-hidden": "true", children: text }),
              (badge || canShowAddButton) && /* @__PURE__ */ jsxs("span", { className: "flex items-center gap-2 shrink-0", children: [
                badge && /* @__PURE__ */ jsx(
                  "span",
                  {
                    className: "flex items-center justify-center min-w-4 h-4 px-1 text-xs font-semibold rounded-full bg-sapphire-negative text-sapphire-text-on-surface border border-sapphire-text-on-surface",
                    style: { fontSize: "0.625rem" },
                    "aria-hidden": "true",
                    children: badge
                  }
                ),
                canShowAddButton && /* @__PURE__ */ jsx(
                  Button,
                  {
                    design: "Tertiary",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx(AddIcon, { className: "h-4 w-4" }),
                    onClick: handleAddClick,
                    disabled,
                    tooltip: addButtonTooltip,
                    "aria-label": addButtonTooltip || `Add new ${text?.toLowerCase()}`,
                    tabIndex: 0,
                    onFocus: () => setIsAddButtonFocused(true),
                    onBlur: (e) => {
                      setIsAddButtonFocused(false);
                      const relatedTarget = e.relatedTarget;
                      const container = e.currentTarget.closest(".relative");
                      if (!relatedTarget || !container?.contains(relatedTarget)) {
                        setIsHovered(false);
                      }
                    },
                    className: "h-5 w-5 text-sapphire-chrome-button-fg-default",
                    style: {
                      minWidth: "1.25rem",
                      minHeight: "1.25rem",
                      borderRadius: "0.25rem"
                    }
                  }
                )
              ] })
            ]
          }
        )
      ]
    }
  );
  if (isFixedSection) {
    return /* @__PURE__ */ jsx("div", { children: content });
  }
  return /* @__PURE__ */ jsx("li", { className: "list-none", children: content });
}
var DATA_ATTR = "data-sap-ui-fastnavgroup";
var CONTAINER_ATTR = "data-sap-ui-fastnavgroup-container";
var isF6Next = (e) => e.key === "F6" && !e.shiftKey && !e.ctrlKey && !e.altKey && !e.metaKey || e.key === "ArrowDown" && e.ctrlKey && e.altKey && !e.shiftKey && !e.metaKey;
var isF6Previous = (e) => e.key === "F6" && e.shiftKey && !e.ctrlKey && !e.altKey && !e.metaKey || e.key === "ArrowUp" && e.ctrlKey && e.altKey && !e.shiftKey && !e.metaKey;
var isElementVisible = (el) => {
  let current = el;
  while (current && current !== document.documentElement) {
    const style = window.getComputedStyle(current);
    if (style.display === "none" || style.visibility === "hidden" || style.opacity === "0" || style.width === "0px" && style.height === "0px") {
      return false;
    }
    current = current.parentElement;
  }
  return true;
};
var FOCUSABLE_SELECTOR = [
  "a[href]",
  "button:not([disabled])",
  "input:not([disabled])",
  "select:not([disabled])",
  "textarea:not([disabled])",
  '[tabindex]:not([tabindex="-1"])',
  '[contenteditable="true"]'
].join(",");
var getFirstFocusableElement = (container) => {
  if (container.matches(FOCUSABLE_SELECTOR) && isElementVisible(container)) {
    return container;
  }
  const candidates = container.querySelectorAll(FOCUSABLE_SELECTOR);
  for (const el of candidates) {
    if (isElementVisible(el)) return el;
  }
  return null;
};
var collectGroups = (root) => {
  const groups = [];
  const walk = (el) => {
    if (el.getAttribute(DATA_ATTR) === "true") {
      if (isElementVisible(el)) {
        groups.push(el);
      }
    }
    let child = el.firstElementChild;
    while (child) {
      if (child.nodeType === 1) walk(child);
      child = child.nextElementSibling;
    }
  };
  walk(root);
  return groups;
};
var findContainer = () => {
  const el = document.activeElement;
  const scope = el?.closest(`[${CONTAINER_ATTR}="true"]`);
  return scope || document.body;
};
var findSelectedGroup = (groups) => {
  let el = document.activeElement;
  while (el && el !== document.documentElement) {
    if (el.getAttribute(DATA_ATTR) === "true") {
      const idx = groups.indexOf(el);
      if (idx !== -1) return idx;
    }
    el = el.parentElement;
  }
  return -1;
};
var focusNextGroup = (groups, currentIndex) => {
  for (const _group of groups) {
    currentIndex = currentIndex + 1 >= groups.length ? 0 : currentIndex + 1;
    const target = getFirstFocusableElement(groups[currentIndex]);
    if (target) {
      target.focus();
      return;
    }
  }
};
var focusPreviousGroup = (groups, currentIndex) => {
  let startIndex = currentIndex;
  if (startIndex < 0) {
    startIndex = groups.length;
  }
  const currentFocusable = currentIndex >= 0 ? getFirstFocusableElement(groups[currentIndex]) : null;
  let distance = 1;
  if (currentIndex > 0 && currentFocusable) {
    for (let d = 1; d < groups.length; d++) {
      const prevIdx = currentIndex - d;
      if (prevIdx < 0) break;
      const prevFocusable = getFirstFocusableElement(groups[prevIdx]);
      if (prevFocusable === currentFocusable) {
        distance++;
      } else {
        break;
      }
    }
  }
  let targetIndex = startIndex - distance;
  if (targetIndex < 0) targetIndex = groups.length - 1;
  for (const _group of groups) {
    const target = getFirstFocusableElement(groups[targetIndex]);
    if (target) {
      target.focus();
      return;
    }
    targetIndex = targetIndex - 1 < 0 ? groups.length - 1 : targetIndex - 1;
  }
};
var handleKeydown = (event) => {
  const forward = isF6Next(event);
  const backward = isF6Previous(event);
  if (!forward && !backward) return;
  const container = findContainer();
  const groups = collectGroups(container);
  if (groups.length === 0) return;
  event.preventDefault();
  const currentIndex = findSelectedGroup(groups);
  if (forward) {
    focusNextGroup(groups, currentIndex);
  } else {
    focusPreviousGroup(groups, currentIndex);
  }
};
function useF6Navigation() {
  useEffect(() => {
    document.addEventListener("keydown", handleKeydown);
    return () => document.removeEventListener("keydown", handleKeydown);
  }, []);
}
var scrollbarStyle = {
  overflowY: "auto",
  overflowX: "hidden",
  scrollbarWidth: "thin",
  scrollbarColor: "var(--color-neutral-400) transparent"
};
var PANE_MIN_WIDTH = 320;
var NAVIGATION_COMPACT_THRESHOLD = 600;
var SIDEBAR_COLLAPSED_WIDTH = 80;
var SIDEBAR_EXPANDED_WIDTH = 256;
var NAV_ITEM_HEIGHT = 40;
var NAV_ITEM_GAP = 24;
var NAV_HEADER_HEIGHT = 80;
var NAV_NORMAL_TOP_SPACER = 160;
var NAV_MERGED_MIN_SPACER = 0;
var NAV_BOTTOM_PADDING = 24;
var NAV_MERGED_DIVIDER_HEIGHT = 13;
var BREAKPOINT_HYSTERESIS = 24;
function calculateItemsHeight(count) {
  if (count === 0) return 0;
  return count * NAV_ITEM_HEIGHT + (count - 1) * NAV_ITEM_GAP;
}
function calculateVerticalBreakpoints(flexibleItemCount, fixedItemCount) {
  const mergeBreakpoint = NAV_HEADER_HEIGHT + NAV_NORMAL_TOP_SPACER + calculateItemsHeight(flexibleItemCount) + calculateItemsHeight(fixedItemCount) + NAV_BOTTOM_PADDING + 16;
  const totalItems = flexibleItemCount + fixedItemCount;
  const compactBreakpoint = NAV_HEADER_HEIGHT + NAV_MERGED_MIN_SPACER + calculateItemsHeight(totalItems) + NAV_MERGED_DIVIDER_HEIGHT + NAV_BOTTOM_PADDING;
  return { mergeBreakpoint, compactBreakpoint };
}
var FxLayoutContext = createContext(null);
var useFxLayoutContext = () => {
  const context = useContext(FxLayoutContext);
  return context;
};
var KEYBOARD_RESIZE_STEP = 50;
var ResizeHandle = ({
  separator,
  startPaneRef,
  centerPaneRef,
  endPaneRef,
  onResizeEnd,
  onResetToDefault,
  ariaLabel
}) => {
  const handleRef = useRef(null);
  const [isDragging, setIsDragging] = useState(false);
  const [isFocused, setIsFocused] = useState(false);
  const [ariaValueNow, setAriaValueNow] = useState(50);
  const computeAriaValue = useCallback(() => {
    if (separator === "start") {
      const startW = startPaneRef.current?.offsetWidth || 0;
      const centerW = centerPaneRef.current?.offsetWidth || 0;
      const total = startW + centerW;
      if (total > 0) setAriaValueNow(Math.round(startW / total * 100));
    } else {
      const centerW = centerPaneRef.current?.offsetWidth || 0;
      const endW = endPaneRef.current?.offsetWidth || 0;
      const total = centerW + endW;
      if (total > 0) setAriaValueNow(Math.round(centerW / total * 100));
    }
  }, [separator, startPaneRef, centerPaneRef, endPaneRef]);
  useEffect(() => {
    computeAriaValue();
  }, [computeAriaValue]);
  const handleKeyDown = (e) => {
    if (e.key === "Home") {
      e.preventDefault();
      onResetToDefault();
      return;
    }
    if (e.key !== "ArrowLeft" && e.key !== "ArrowRight") return;
    e.preventDefault();
    const startPane = startPaneRef.current;
    const centerPane = centerPaneRef.current;
    const endPane = endPaneRef.current;
    if (!startPane || !centerPane) return;
    const delta = e.key === "ArrowRight" ? KEYBOARD_RESIZE_STEP : -KEYBOARD_RESIZE_STEP;
    if (separator === "start") {
      let newStartWidth = startPane.offsetWidth + delta;
      let newCenterWidth = centerPane.offsetWidth - delta;
      if (newStartWidth < PANE_MIN_WIDTH) {
        newStartWidth = PANE_MIN_WIDTH;
        newCenterWidth = startPane.offsetWidth + centerPane.offsetWidth - PANE_MIN_WIDTH;
      }
      if (newCenterWidth < PANE_MIN_WIDTH) {
        newCenterWidth = PANE_MIN_WIDTH;
        newStartWidth = startPane.offsetWidth + centerPane.offsetWidth - PANE_MIN_WIDTH;
      }
      onResizeEnd(newStartWidth, newCenterWidth, endPane?.offsetWidth || 0);
      requestAnimationFrame(computeAriaValue);
    } else {
      if (!endPane) return;
      let newCenterWidth = centerPane.offsetWidth + delta;
      let newEndWidth = endPane.offsetWidth - delta;
      if (newEndWidth < PANE_MIN_WIDTH) {
        newEndWidth = PANE_MIN_WIDTH;
        newCenterWidth = centerPane.offsetWidth + endPane.offsetWidth - PANE_MIN_WIDTH;
      }
      if (newCenterWidth < PANE_MIN_WIDTH) {
        newCenterWidth = PANE_MIN_WIDTH;
        newEndWidth = centerPane.offsetWidth + endPane.offsetWidth - PANE_MIN_WIDTH;
      }
      onResizeEnd(startPane.offsetWidth, newCenterWidth, newEndWidth);
      requestAnimationFrame(computeAriaValue);
    }
  };
  const handleMouseDown = (e) => {
    e.preventDefault();
    const startPane = startPaneRef.current;
    const centerPane = centerPaneRef.current;
    const endPane = endPaneRef.current;
    if (!startPane || !centerPane) return;
    const initialStartWidth = startPane.offsetWidth;
    const initialCenterWidth = centerPane.offsetWidth;
    const initialEndWidth = endPane?.offsetWidth || 0;
    const startX = e.clientX;
    const originalStartTransition = startPane.style.transition;
    const originalCenterTransition = centerPane.style.transition;
    const originalEndTransition = endPane?.style.transition || "";
    startPane.style.transition = "none";
    centerPane.style.transition = "none";
    if (endPane) endPane.style.transition = "none";
    setIsDragging(true);
    document.body.style.cursor = "col-resize";
    document.body.style.userSelect = "none";
    const handleMouseMove = (ev) => {
      const delta = ev.clientX - startX;
      if (separator === "start") {
        let newStartWidth = initialStartWidth + delta;
        let newCenterWidth = initialCenterWidth - delta;
        if (newStartWidth < PANE_MIN_WIDTH) {
          newStartWidth = PANE_MIN_WIDTH;
          newCenterWidth = initialStartWidth + initialCenterWidth - PANE_MIN_WIDTH;
        }
        if (newCenterWidth < PANE_MIN_WIDTH) {
          newCenterWidth = PANE_MIN_WIDTH;
          newStartWidth = initialStartWidth + initialCenterWidth - PANE_MIN_WIDTH;
        }
        startPane.style.width = `${newStartWidth}px`;
        centerPane.style.width = `${newCenterWidth}px`;
      } else {
        if (!endPane) return;
        let newCenterWidth = initialCenterWidth + delta;
        let newEndWidth = initialEndWidth - delta;
        if (newEndWidth < PANE_MIN_WIDTH) {
          newEndWidth = PANE_MIN_WIDTH;
          newCenterWidth = initialCenterWidth + initialEndWidth - PANE_MIN_WIDTH;
        }
        if (newCenterWidth < PANE_MIN_WIDTH) {
          newCenterWidth = PANE_MIN_WIDTH;
          newEndWidth = initialCenterWidth + initialEndWidth - PANE_MIN_WIDTH;
        }
        centerPane.style.width = `${newCenterWidth}px`;
        endPane.style.width = `${newEndWidth}px`;
      }
    };
    const handleMouseUp = () => {
      document.body.style.cursor = "";
      document.body.style.userSelect = "";
      document.removeEventListener("mousemove", handleMouseMove);
      document.removeEventListener("mouseup", handleMouseUp);
      startPane.style.transition = originalStartTransition;
      centerPane.style.transition = originalCenterTransition;
      if (endPane) endPane.style.transition = originalEndTransition;
      setIsDragging(false);
      handleRef.current?.focus();
      const finalStartWidth = startPane.offsetWidth;
      const finalCenterWidth = centerPane.offsetWidth;
      const finalEndWidth = endPane?.offsetWidth || 0;
      onResizeEnd(finalStartWidth, finalCenterWidth, finalEndWidth);
      requestAnimationFrame(computeAriaValue);
    };
    document.addEventListener("mousemove", handleMouseMove);
    document.addEventListener("mouseup", handleMouseUp);
  };
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref: handleRef,
      role: "separator",
      "aria-orientation": "vertical",
      "aria-label": ariaLabel,
      "aria-valuenow": ariaValueNow,
      "aria-valuemin": 0,
      "aria-valuemax": 100,
      tabIndex: 0,
      className: cn(
        "absolute inset-0 cursor-col-resize z-10",
        "after:content-[''] after:absolute after:top-0 after:bottom-0 after:left-1/2 after:-translate-x-1/2",
        "after:w-px after:bg-border after:transition-all",
        "hover:after:w-0.5 hover:after:bg-sapphire-chrome-button-bg-selected",
        isDragging && "after:w-0.5 after:bg-sapphire-chrome-button-bg-selected",
        isFocused && "after:w-0.5 after:bg-sapphire-border-accent"
      ),
      style: { outline: "none" },
      onMouseDown: handleMouseDown,
      onDoubleClick: onResetToDefault,
      onKeyDown: handleKeyDown,
      onFocus: () => setIsFocused(true),
      onBlur: () => setIsFocused(false)
    }
  );
};
function calculateMaxPanes(width) {
  if (width >= 1965) {
    return 3;
  }
  if (width >= 1020) {
    return 2;
  }
  return 1;
}
var InputContainer = ({
  input,
  inputInEndPane,
  mobileLayout,
  effectiveShowStart,
  effectiveShowEnd,
  finalStartWidth,
  finalEndWidth,
  paneAnimationsEnabled
}) => {
  let inputLeft;
  let inputRight;
  if (mobileLayout) {
    inputLeft = "1rem";
    inputRight = "1rem";
  } else if (inputInEndPane) {
    inputLeft = `calc(100% - ${finalEndWidth} + 1.25rem)`;
    inputRight = "1.25rem";
  } else {
    inputLeft = effectiveShowStart ? `calc(${finalStartWidth} + 1.25rem)` : "1.25rem";
    inputRight = effectiveShowEnd ? `calc(${finalEndWidth} + 1.25rem)` : "1.25rem";
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: "pointer-events-none flex justify-center",
      style: {
        position: "absolute",
        bottom: "1rem",
        left: inputLeft,
        right: inputRight,
        zIndex: 2,
        transition: paneAnimationsEnabled ? "left 350ms cubic-bezier(0.4, 0, 0.2, 1), right 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none",
        padding: "0 0.5rem 0.5rem 0.5rem",
        margin: "0 -0.5rem -0.5rem -0.5rem"
      },
      children: /* @__PURE__ */ jsx(
        "div",
        {
          className: "pointer-events-auto w-full",
          style: { maxWidth: inputInEndPane ? void 0 : "600px" },
          children: input
        }
      )
    }
  );
};
function FxLayout({
  id,
  hideStart: hideStartProp = false,
  hideEnd: hideEndProp = false,
  suppressEnd = false,
  priorityPane: priorityPaneProp = "Center",
  mode = "",
  notificationsBadge,
  notificationsLocked = false,
  navItems = [],
  startHeader,
  centerHeader,
  endHeader,
  startContent,
  centerContent,
  endContent,
  userMenu,
  input,
  onVisibilityChange,
  onModeChange,
  onLayoutChange,
  onAddClick,
  onNotificationsClick,
  className,
  style,
  children,
  ref
}) {
  const layoutRef = useRef(null);
  const sideNavRef = useRef(null);
  const notificationsItemRef = useRef(null);
  const userItemRef = useRef(null);
  const navDialogCloseRef = useRef(null);
  const startPaneRef = useRef(null);
  const centerPaneRef = useRef(null);
  const endPaneRef = useRef(null);
  useF6Navigation();
  const { t } = useTranslation("fx");
  const startContentRef = useRef(null);
  const centerContentRef = useRef(null);
  const endContentRef = useRef(null);
  const [startScrolled, setStartScrolled] = useState(false);
  const [centerScrolled, setCenterScrolled] = useState(false);
  const [endScrolled, setEndScrolled] = useState(false);
  const [navCollapsed, setNavCollapsed] = useState(true);
  const [navTransitioning, setNavTransitioning] = useState(false);
  const [navTransitionDirection, setNavTransitionDirection] = useState(null);
  const navTransitionTimeoutRef = useRef(null);
  const [horizontalCompactMode, setHorizontalCompactMode] = useState(false);
  const [verticalCompactMode, setVerticalCompactMode] = useState(false);
  const [navMerged, setNavMerged] = useState(false);
  const [navigationOpen, setNavigationOpen] = useState(false);
  const [navigationClosing, setNavigationClosing] = useState(false);
  const [maxPanes, setMaxPanes] = useState(3);
  const [totalWidth, setTotalWidth] = useState(0);
  const compactMode = horizontalCompactMode || verticalCompactMode;
  const mobileLayout = horizontalCompactMode;
  const [animationsEnabled, setAnimationsEnabled] = useState(false);
  const [internalHideStart, setInternalHideStart] = useState(hideStartProp);
  const [internalHideEnd, setInternalHideEnd] = useState(hideEndProp);
  const [internalPriorityPane, setInternalPriorityPane] = useState(priorityPaneProp);
  const [userMenuOpen, setUserMenuOpen] = useState(false);
  const [userMenuOpener, setUserMenuOpener] = useState(null);
  const [userMenuFromHamburger, setUserMenuFromHamburger] = useState(false);
  const _userMenuAccounts = userMenu?.props?.accounts;
  const _selectedUserAccount = _userMenuAccounts?.find((a) => a.selected) || _userMenuAccounts?.[0];
  useEffect(() => {
    const timer = setTimeout(() => {
      setAnimationsEnabled(true);
    }, 100);
    return () => clearTimeout(timer);
  }, []);
  const paneAnimationsEnabled = animationsEnabled && maxPanes > 1;
  const toggleNavCollapsed = useCallback(() => {
    const willCollapse = !navCollapsed;
    if (animationsEnabled) {
      setNavTransitioning(true);
      setNavTransitionDirection(willCollapse ? "collapsing" : "expanding");
      if (navTransitionTimeoutRef.current) clearTimeout(navTransitionTimeoutRef.current);
      navTransitionTimeoutRef.current = setTimeout(() => {
        setNavTransitioning(false);
        setNavTransitionDirection(null);
      }, 400);
    }
    setNavCollapsed(willCollapse);
  }, [navCollapsed, animationsEnabled]);
  const handleNavTransitionEnd = useCallback((e) => {
    if (e.propertyName === "width" && e.target === e.currentTarget) {
      setNavTransitioning(false);
      setNavTransitionDirection(null);
      if (navTransitionTimeoutRef.current) {
        clearTimeout(navTransitionTimeoutRef.current);
        navTransitionTimeoutRef.current = null;
      }
    }
  }, []);
  useEffect(() => {
    return () => {
      if (navTransitionTimeoutRef.current) clearTimeout(navTransitionTimeoutRef.current);
    };
  }, []);
  const previousTotalWidthRef = useRef(0);
  const lastMaxPanesRef = useRef(3);
  const isInternalStartChangeRef = useRef(false);
  const isInternalEndChangeRef = useRef(false);
  const isInternalPriorityChangeRef = useRef(false);
  useEffect(() => {
    if (!isInternalStartChangeRef.current) {
      setInternalHideStart(hideStartProp);
    }
    isInternalStartChangeRef.current = false;
  }, [hideStartProp]);
  useEffect(() => {
    if (!isInternalEndChangeRef.current) {
      setInternalHideEnd(hideEndProp);
    }
    isInternalEndChangeRef.current = false;
  }, [hideEndProp]);
  useEffect(() => {
    if (!isInternalPriorityChangeRef.current) {
      setInternalPriorityPane(priorityPaneProp);
    }
    isInternalPriorityChangeRef.current = false;
  }, [priorityPaneProp]);
  const currentNavItem = useMemo(
    () => navItems.find((item) => item.name === mode),
    [navItems, mode]
  );
  const isSettingsMode = mode === "settings";
  const allowStart = isSettingsMode || !currentNavItem?.noStartPane;
  const allowEnd = !suppressEnd && !isSettingsMode;
  const utilityEndPane = currentNavItem?.utilityEndPane ?? false;
  const externalEndPaneInput = currentNavItem?.externalEndPaneInput ?? false;
  const internalHideCenter = false;
  const requestedPaneCount = useMemo(() => {
    return (internalHideStart || !allowStart ? 0 : 1) + 1 + (internalHideEnd || !allowEnd ? 0 : 1);
  }, [internalHideStart, internalHideEnd, allowStart, allowEnd]);
  const effectivePriority = useMemo(() => {
    if (requestedPaneCount <= maxPanes) {
      return "Center";
    }
    return internalPriorityPane;
  }, [requestedPaneCount, maxPanes, internalPriorityPane]);
  useEffect(() => {
    if (requestedPaneCount <= maxPanes && internalPriorityPane !== "Center") {
      setInternalPriorityPane("Center");
      onLayoutChange?.({ priorityPane: "Center" });
    }
  }, [requestedPaneCount, maxPanes, internalPriorityPane, onLayoutChange]);
  const effectiveShowStart = useMemo(() => {
    if (!allowStart) {
      return false;
    }
    if (internalHideStart) {
      return false;
    }
    if (effectivePriority === "Start") {
      return true;
    }
    let centerPriority = 0;
    if (effectivePriority === "Center") {
      centerPriority = 2;
    } else if (effectivePriority === "End") {
      centerPriority = 1;
    }
    const endPriority = effectivePriority === "End" ? 2 : 1;
    let higherPriorityCount = 0;
    if (centerPriority > 0) {
      higherPriorityCount++;
    }
    if (!internalHideEnd && allowEnd && !suppressEnd && !internalHideCenter && endPriority > 0) {
      higherPriorityCount++;
    }
    if (higherPriorityCount >= maxPanes) {
      return false;
    }
    return true;
  }, [allowStart, internalHideStart, effectivePriority, internalHideCenter, internalHideEnd, allowEnd, suppressEnd, maxPanes]);
  const effectiveShowCenter = useMemo(() => {
    if (effectivePriority === "Center") {
      return true;
    }
    let higherPriorityCount = 0;
    if (effectivePriority === "Start" && !internalHideStart && allowStart) {
      higherPriorityCount++;
    }
    if (effectivePriority === "End" && !internalHideEnd && allowEnd) {
      higherPriorityCount++;
    }
    if (higherPriorityCount >= maxPanes) {
      return false;
    }
    return true;
  }, [internalHideCenter, effectivePriority, internalHideStart, allowStart, internalHideEnd, allowEnd, maxPanes]);
  const effectiveShowEnd = useMemo(() => {
    if (!allowEnd) {
      return false;
    }
    if (suppressEnd) {
      return false;
    }
    if (internalHideEnd) {
      return false;
    }
    if (effectivePriority === "End") {
      return true;
    }
    let higherPriorityCount = 0;
    if (effectivePriority === "Start" && !internalHideStart && allowStart) {
      higherPriorityCount++;
    }
    {
      higherPriorityCount++;
    }
    if (higherPriorityCount >= maxPanes) {
      return false;
    }
    return true;
  }, [allowEnd, suppressEnd, internalHideEnd, effectivePriority, internalHideCenter, internalHideStart, allowStart, maxPanes]);
  const hideInputEntirely = externalEndPaneInput && effectiveShowEnd;
  const inputInCenterPane = !hideInputEntirely && (!effectiveShowEnd || utilityEndPane);
  const inputInEndPane = !hideInputEntirely && effectiveShowEnd && !utilityEndPane;
  const inputOverlayPadding = mobileLayout ? "8rem" : "7.5rem";
  const inputFadeMask = `linear-gradient(180deg, #000 0%, #000 calc(100% - ${mobileLayout ? "4.5rem" : "4rem"}), transparent calc(100% - ${mobileLayout ? "2.5rem" : "2rem"}))`;
  const fixedItemCount = 3;
  const { mergeBreakpoint, compactBreakpoint } = useMemo(
    () => calculateVerticalBreakpoints(navItems.length, fixedItemCount),
    [navItems.length]
  );
  const prevVerticalCompactRef = useRef(false);
  useEffect(() => {
    if (!layoutRef.current) return;
    const handleResize = (width, height) => {
      const previousTotalWidth = previousTotalWidthRef.current;
      const browserShrinking = previousTotalWidth > 0 && width < previousTotalWidth;
      previousTotalWidthRef.current = width;
      setTotalWidth(width);
      const isHorizontalCompact = width < NAVIGATION_COMPACT_THRESHOLD;
      setHorizontalCompactMode((prev) => {
        if (prev && !isHorizontalCompact) {
          setUserMenuOpen(false);
        }
        return isHorizontalCompact;
      });
      if (!isHorizontalCompact && height > 0) {
        const shouldMerge = height < mergeBreakpoint;
        setNavMerged(shouldMerge);
        let shouldVerticalCompact;
        if (prevVerticalCompactRef.current) {
          shouldVerticalCompact = height < compactBreakpoint + BREAKPOINT_HYSTERESIS;
        } else {
          shouldVerticalCompact = height < compactBreakpoint;
        }
        prevVerticalCompactRef.current = shouldVerticalCompact;
        setVerticalCompactMode(shouldVerticalCompact);
      } else {
        prevVerticalCompactRef.current = false;
        setNavMerged(false);
        setVerticalCompactMode(false);
      }
      const currentNavWidth = isHorizontalCompact ? 0 : navCollapsed ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_EXPANDED_WIDTH;
      const mainWidth = width - currentNavWidth;
      const newMaxPanes = calculateMaxPanes(mainWidth);
      if (newMaxPanes === lastMaxPanesRef.current) {
        return;
      }
      if (browserShrinking && !isHorizontalCompact && !navCollapsed) {
        const expandedNavMainWidth = width - SIDEBAR_EXPANDED_WIDTH;
        const collapsedNavMainWidth = width - SIDEBAR_COLLAPSED_WIDTH;
        const maxPanesWithExpanded = calculateMaxPanes(expandedNavMainWidth);
        const maxPanesWithCollapsed = calculateMaxPanes(collapsedNavMainWidth);
        if (maxPanesWithCollapsed > maxPanesWithExpanded) {
          setNavCollapsed(true);
          if (animationsEnabled) {
            setNavTransitioning(true);
            setNavTransitionDirection("collapsing");
            if (navTransitionTimeoutRef.current) clearTimeout(navTransitionTimeoutRef.current);
            navTransitionTimeoutRef.current = setTimeout(() => {
              setNavTransitioning(false);
              setNavTransitionDirection(null);
            }, 400);
          }
          const collapsedMaxPanes = calculateMaxPanes(collapsedNavMainWidth);
          setMaxPanes(collapsedMaxPanes);
          lastMaxPanesRef.current = collapsedMaxPanes;
          return;
        }
      }
      setMaxPanes(newMaxPanes);
      lastMaxPanesRef.current = newMaxPanes;
    };
    const observer = new ResizeObserver((entries) => {
      const entry = entries[0];
      if (entry) {
        handleResize(entry.contentRect.width, entry.contentRect.height);
      }
    });
    observer.observe(layoutRef.current);
    return () => {
      observer.disconnect();
    };
  }, [navCollapsed, mergeBreakpoint, compactBreakpoint]);
  const prevStartVisibleRef = useRef(null);
  const prevEndVisibleRef = useRef(null);
  const onVisibilityChangeRef = useRef(onVisibilityChange);
  onVisibilityChangeRef.current = onVisibilityChange;
  useEffect(() => {
    if (prevStartVisibleRef.current === null || prevStartVisibleRef.current !== effectiveShowStart) {
      onVisibilityChangeRef.current?.({
        pane: "start",
        visible: effectiveShowStart
      });
    }
    prevStartVisibleRef.current = effectiveShowStart;
  }, [effectiveShowStart]);
  useEffect(() => {
    if (prevEndVisibleRef.current === null || prevEndVisibleRef.current !== effectiveShowEnd) {
      onVisibilityChangeRef.current?.({
        pane: "end",
        visible: effectiveShowEnd
      });
    }
    prevEndVisibleRef.current = effectiveShowEnd;
  }, [effectiveShowEnd]);
  useEffect(() => {
    if (!mobileLayout) return;
    startContentRef.current?.scrollTo({ top: 0 });
    centerContentRef.current?.scrollTo({ top: 0 });
    endContentRef.current?.scrollTo({ top: 0 });
  }, [mobileLayout, effectiveShowStart, effectiveShowCenter, effectiveShowEnd, internalPriorityPane]);
  useImperativeHandle(ref, () => ({
    focusPane: () => {
    },
    closeProfileFlyout: () => {
      setUserMenuOpen(false);
      setUserMenuFromHamburger(false);
    },
    get nativeElement() {
      return layoutRef.current;
    },
    getNativeElement: () => layoutRef.current
  }));
  const handleNavSelect = useCallback(
    (key) => {
      const previousMode = mode;
      if (key !== previousMode) {
        onModeChange?.({ mode: key, previousMode });
      }
      if (navigationOpen) {
        setNavigationOpen(false);
      }
    },
    [mode, onModeChange, navigationOpen]
  );
  const toggleStartPane = useCallback(() => {
    const shouldHide = effectiveShowStart;
    let newPriority;
    if (shouldHide) {
      newPriority = "Center";
    } else {
      const requestedAfter = (!shouldHide ? 1 : 0) + 1 + (internalHideEnd ? 0 : 1);
      newPriority = requestedAfter <= maxPanes ? "Center" : "Start";
    }
    isInternalStartChangeRef.current = true;
    isInternalPriorityChangeRef.current = true;
    setInternalHideStart(shouldHide);
    setInternalPriorityPane(newPriority);
    onLayoutChange?.({ priorityPane: newPriority, hideStart: shouldHide });
  }, [effectiveShowStart, internalHideEnd, maxPanes, onLayoutChange]);
  const toggleEndPane = useCallback(() => {
    const currentlyVisible = effectiveShowEnd;
    const newHideEnd = currentlyVisible;
    let newPriority;
    if (currentlyVisible) {
      newPriority = "Center";
    } else {
      const requestedPanes = (internalHideStart ? 0 : 1) + 1 + 1;
      newPriority = requestedPanes <= maxPanes ? "Center" : "End";
    }
    isInternalEndChangeRef.current = true;
    isInternalPriorityChangeRef.current = true;
    setInternalHideEnd(newHideEnd);
    setInternalPriorityPane(newPriority);
    onLayoutChange?.({ priorityPane: newPriority, hideEnd: newHideEnd });
  }, [effectiveShowEnd, internalHideStart, maxPanes, onLayoutChange]);
  const openNavigation = useCallback(() => {
    setNavigationOpen(true);
  }, []);
  const closeNavigation = useCallback(() => {
    setNavigationClosing(true);
    setTimeout(() => {
      setNavigationOpen(false);
      setNavigationClosing(false);
    }, 300);
  }, []);
  useEffect(() => {
    if (!compactMode && navigationOpen) {
      setNavigationOpen(false);
      setNavigationClosing(false);
    }
  }, [compactMode, navigationOpen]);
  useEffect(() => {
    if (navigationOpen && navDialogCloseRef.current) {
      navDialogCloseRef.current.focus();
    }
  }, [navigationOpen]);
  const navWidth = compactMode ? 0 : navCollapsed ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_EXPANDED_WIDTH;
  const mainAreaWidth = totalWidth - navWidth;
  const defaultWidths = useMemo(() => {
    const visibleCount = (effectiveShowStart ? 1 : 0) + (effectiveShowCenter ? 1 : 0) + (effectiveShowEnd ? 1 : 0);
    if (visibleCount === 1) {
      if (effectiveShowStart) return { startWidth: "100%", endWidth: "0%" };
      if (effectiveShowEnd) return { startWidth: "0%", endWidth: "100%" };
      return { startWidth: "0%", endWidth: "0%" };
    }
    if (visibleCount === 2) {
      if (effectiveShowStart && effectiveShowCenter && !effectiveShowEnd) {
        return { startWidth: "33%", endWidth: "0%" };
      } else if (!effectiveShowStart && effectiveShowCenter && effectiveShowEnd) {
        return { startWidth: "0%", endWidth: "33%" };
      } else if (effectiveShowStart && !effectiveShowCenter && effectiveShowEnd) {
        return { startWidth: "33%", endWidth: "67%" };
      }
    }
    return { startWidth: "25%", endWidth: "25%" };
  }, [effectiveShowStart, effectiveShowCenter, effectiveShowEnd]);
  const [userStartWidth, setUserStartWidth] = useState(null);
  const [userEndWidth, setUserEndWidth] = useState(null);
  useEffect(() => {
    setUserStartWidth(null);
    setUserEndWidth(null);
  }, [effectiveShowStart, effectiveShowCenter, effectiveShowEnd]);
  const toPercent = useCallback((px) => {
    if (mainAreaWidth <= 0) return "0%";
    return `${(px / mainAreaWidth * 100).toFixed(2)}%`;
  }, [mainAreaWidth]);
  const finalStartWidth = effectiveShowStart ? userStartWidth ?? defaultWidths.startWidth : "0%";
  const finalEndWidth = effectiveShowEnd ? userEndWidth ?? defaultWidths.endWidth : "0%";
  const handleResizeEnd = useCallback((startWidthPx, _centerWidthPx, endWidthPx) => {
    setUserStartWidth(toPercent(startWidthPx));
    setUserEndWidth(toPercent(endWidthPx));
  }, [toPercent]);
  const handleResetToDefault = useCallback(() => {
    setUserStartWidth(null);
    setUserEndWidth(null);
  }, []);
  const inputMode = useMemo(() => {
    const isInEndPane = effectiveShowEnd && !utilityEndPane;
    return isInEndPane ? "oneline" : "multiline";
  }, [effectiveShowEnd, utilityEndPane]);
  const leftmostVisiblePane = useMemo(() => {
    if (effectiveShowStart) return "start";
    if (effectiveShowCenter) return "center";
    return "end";
  }, [effectiveShowStart, effectiveShowCenter]);
  const contextValue = useMemo(
    () => ({
      startVisible: effectiveShowStart,
      endVisible: effectiveShowEnd,
      maxPanes,
      compactMode,
      leftmostVisiblePane,
      allowStart,
      allowEnd,
      suppressEnd,
      utilityEndPane,
      isSettingsMode,
      inputMode,
      toggleStartPane,
      toggleEndPane,
      openNavigation
    }),
    [effectiveShowStart, effectiveShowEnd, maxPanes, compactMode, leftmostVisiblePane, allowStart, allowEnd, suppressEnd, utilityEndPane, isSettingsMode, inputMode, toggleStartPane, toggleEndPane, openNavigation]
  );
  return /* @__PURE__ */ jsx(FxLayoutContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxs(
    "div",
    {
      ref: layoutRef,
      id,
      "data-mode": mode,
      "data-hide-start": !effectiveShowStart ? "" : void 0,
      "data-hide-end": !effectiveShowEnd ? "" : void 0,
      className: cn(
        "flex h-full w-full overflow-hidden relative bg-sapphire-canvas-primary",
        className
      ),
      style: {
        // Use dvh on mobile for accurate viewport height (accounts for browser chrome)
        ...mobileLayout ? { height: "100dvh" } : {},
        ...style
      },
      children: [
        !compactMode && /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "flex flex-col h-full shrink-0 relative",
              effectiveShowStart && "border-r border-border"
            ),
            "data-sap-ui-fastnavgroup": "true",
            style: {
              width: navWidth,
              transition: animationsEnabled ? "width 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none",
              overflow: "visible"
              // Allow flyouts to extend beyond nav column
            },
            onTransitionEnd: handleNavTransitionEnd,
            children: /* @__PURE__ */ jsx(
              FxSideNavigation,
              {
                ref: sideNavRef,
                collapsed: navCollapsed,
                isTransitioning: navTransitioning,
                transitionDirection: navTransitionDirection,
                forceMerged: navMerged,
                onSelectionChange: ({ name }) => handleNavSelect(name),
                header: /* @__PURE__ */ jsxs(
                  "button",
                  {
                    onClick: () => handleNavSelect(navItems[0]?.name || ""),
                    className: "cursor-pointer focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring rounded flex items-center gap-2",
                    children: [
                      /* @__PURE__ */ jsx(JouleIcon, { size: 32 }),
                      !navCollapsed && /* @__PURE__ */ jsxs("span", { className: "text-lg font-semibold whitespace-nowrap", children: [
                        /* @__PURE__ */ jsx("span", { className: "text-purple-600", children: "Joule" }),
                        /* @__PURE__ */ jsx("span", { className: "text-foreground", children: "Work" })
                      ] })
                    ]
                  }
                ),
                fixedItems: /* @__PURE__ */ jsxs(Fragment, { children: [
                  /* @__PURE__ */ jsx(
                    FxSideNavigationItem,
                    {
                      ref: notificationsItemRef,
                      name: "notifications",
                      icon: /* @__PURE__ */ jsx(BellIcon, { className: "h-4 w-4" }),
                      text: t("FX_NOTIFICATIONS"),
                      badge: notificationsBadge?.toString(),
                      selected: notificationsLocked,
                      locked: notificationsLocked,
                      onClick: () => {
                        const opener = notificationsItemRef.current?.getNativeElement();
                        if (opener) onNotificationsClick?.(opener);
                      }
                    }
                  ),
                  /* @__PURE__ */ jsx(
                    FxSideNavigationItem,
                    {
                      ref: userItemRef,
                      name: "profile",
                      text: _selectedUserAccount?.titleText || "User",
                      avatar: /* @__PURE__ */ jsx(
                        Avatar,
                        {
                          size: "XS",
                          shape: "Square",
                          image: _selectedUserAccount?.avatarSrc,
                          initials: _selectedUserAccount?.avatarInitials || "U",
                          colorScheme: _selectedUserAccount?.avatarColorScheme || "Accent1"
                        }
                      ),
                      selected: userMenuOpen,
                      locked: userMenuOpen,
                      onClick: () => {
                        if (userMenu) {
                          if (userMenuOpen) {
                            setUserMenuOpen(false);
                          } else {
                            const opener = userItemRef.current?.getNativeElement();
                            if (opener) setUserMenuOpener(opener);
                            setUserMenuOpen(true);
                          }
                        }
                      }
                    }
                  ),
                  /* @__PURE__ */ jsx(
                    FxSideNavigationItem,
                    {
                      name: "toggle",
                      icon: navCollapsed ? /* @__PURE__ */ jsx(OpenCommandFieldIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsx(CloseCommandFieldIcon, { className: "h-4 w-4" }),
                      text: navCollapsed ? t("NAVIGATION_EXPAND") : t("NAVIGATION_COLLAPSE"),
                      onClick: toggleNavCollapsed
                    }
                  )
                ] }),
                children: navItems.map((item) => /* @__PURE__ */ jsx(
                  FxSideNavigationItem,
                  {
                    name: item.name,
                    icon: item.icon ? /* @__PURE__ */ jsx("span", { className: "h-4 w-4 flex items-center justify-center", children: item.icon }) : void 0,
                    text: item.text,
                    selected: mode === item.name,
                    showAddButton: !!item.createActionTooltip,
                    addButtonTooltip: item.createActionTooltip,
                    onAddClick: () => onAddClick?.({ mode: item.name })
                  },
                  item.name
                ))
              }
            )
          }
        ),
        /* @__PURE__ */ jsxs(
          "main",
          {
            className: "flex flex-1 h-full min-w-0 relative",
            style: {
              // In mobile layout, clip all overflow to prevent any child from making page wider
              overflow: mobileLayout ? "hidden" : "visible",
              overflowX: mobileLayout ? "clip" : void 0
            },
            children: [
              allowStart && /* @__PURE__ */ jsxs(
                "aside",
                {
                  ref: startPaneRef,
                  "aria-label": t("FX_START_PANE"),
                  className: cn(
                    "flex flex-col shrink-0 relative bg-sapphire-shell-bg-primary"
                  ),
                  style: {
                    width: effectiveShowStart ? mobileLayout ? "100%" : finalStartWidth : 0,
                    minWidth: effectiveShowStart ? mobileLayout ? 0 : PANE_MIN_WIDTH : 0,
                    height: "100%",
                    paddingTop: effectiveShowStart ? "1.5rem" : 0,
                    paddingInline: effectiveShowStart ? "0.25rem" : 0,
                    overflow: "hidden",
                    visibility: effectiveShowStart ? "visible" : "hidden",
                    transition: paneAnimationsEnabled ? `width 350ms cubic-bezier(0.4, 0, 0.2, 1), padding 350ms cubic-bezier(0.4, 0, 0.2, 1), visibility 0s ${effectiveShowStart ? "0s" : "350ms"}` : "none"
                  },
                  children: [
                    /* @__PURE__ */ jsx(
                      "div",
                      {
                        className: "shrink-0",
                        "data-sap-ui-fastnavgroup": "true",
                        style: {
                          paddingInline: effectiveShowStart ? mobileLayout ? "0.75rem" : "1rem" : 0,
                          opacity: effectiveShowStart ? 1 : 0,
                          transform: effectiveShowStart ? "translateX(0)" : "translateX(-30%)",
                          transition: paneAnimationsEnabled ? "opacity 350ms cubic-bezier(0.4, 0, 0.2, 1), transform 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none"
                        },
                        children: isValidElement(startHeader) ? cloneElement(startHeader, { showBorder: startScrolled }) : startHeader
                      }
                    ),
                    /* @__PURE__ */ jsx(
                      "div",
                      {
                        ref: startContentRef,
                        className: "flex-1 fx-pane-content min-h-0",
                        "data-sap-ui-fastnavgroup": "true",
                        onScrollCapture: (e) => {
                          const target = e.target;
                          setStartScrolled(target.scrollTop > 0);
                        },
                        style: {
                          ...scrollbarStyle,
                          opacity: effectiveShowStart ? 1 : 0,
                          transform: effectiveShowStart ? "translateX(0)" : "translateX(-30%)",
                          transition: paneAnimationsEnabled ? "opacity 350ms cubic-bezier(0.4, 0, 0.2, 1), transform 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none"
                        },
                        children: startContent
                      }
                    )
                  ]
                }
              ),
              effectiveShowStart && effectiveShowCenter && /* @__PURE__ */ jsx("div", { className: "w-0 shrink-0 relative", style: { marginLeft: "-8px", marginRight: "-8px", width: "16px", zIndex: 10 }, children: /* @__PURE__ */ jsx(
                ResizeHandle,
                {
                  separator: "start",
                  startPaneRef,
                  centerPaneRef,
                  endPaneRef,
                  onResizeEnd: handleResizeEnd,
                  onResetToDefault: handleResetToDefault,
                  ariaLabel: t("FX_RESIZE_START_CENTER")
                }
              ) }),
              /* @__PURE__ */ jsxs(
                "div",
                {
                  ref: centerPaneRef,
                  className: "flex flex-col relative bg-background border-l border-border",
                  style: {
                    boxShadow: "0 22px 14px 0 rgba(0,0,0,0.06), 0 182px 111px 0 rgba(0,0,0,0.05)",
                    flex: effectiveShowCenter ? "1 1 auto" : "0 0 0%",
                    minWidth: effectiveShowCenter ? mobileLayout ? 0 : PANE_MIN_WIDTH : 0,
                    height: "100%",
                    paddingTop: effectiveShowCenter ? "1.5rem" : 0,
                    paddingInline: effectiveShowCenter ? "0.25rem" : 0,
                    overflow: "hidden",
                    visibility: effectiveShowCenter ? "visible" : "hidden",
                    zIndex: 1,
                    transition: paneAnimationsEnabled ? `flex 350ms cubic-bezier(0.4, 0, 0.2, 1), padding 350ms cubic-bezier(0.4, 0, 0.2, 1), visibility 0s ${effectiveShowCenter ? "0s" : "350ms"}` : "none"
                  },
                  children: [
                    /* @__PURE__ */ jsx(
                      "div",
                      {
                        className: "shrink-0",
                        "data-sap-ui-fastnavgroup": "true",
                        style: {
                          paddingInline: effectiveShowCenter ? mobileLayout ? "0.75rem" : "1rem" : 0,
                          opacity: effectiveShowCenter ? 1 : 0,
                          transition: paneAnimationsEnabled ? "opacity 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none"
                        },
                        children: isValidElement(centerHeader) ? cloneElement(centerHeader, { showBorder: centerScrolled }) : centerHeader
                      }
                    ),
                    /* @__PURE__ */ jsxs(
                      "div",
                      {
                        ref: centerContentRef,
                        className: "flex-1 fx-pane-content min-h-0",
                        "data-sap-ui-fastnavgroup": "true",
                        onScrollCapture: (e) => {
                          const target = e.target;
                          setCenterScrolled(target.scrollTop > 0);
                        },
                        style: {
                          ...scrollbarStyle,
                          // Add padding at bottom when input is shown in center pane so content isn't covered
                          paddingBottom: input && inputInCenterPane ? inputOverlayPadding : 0,
                          opacity: effectiveShowCenter ? 1 : 0,
                          transition: paneAnimationsEnabled ? "opacity 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none",
                          // Fade content at bottom edge when input is shown in center pane
                          ...input && inputInCenterPane ? {
                            WebkitMaskImage: inputFadeMask,
                            maskImage: inputFadeMask
                          } : {}
                        },
                        children: [
                          centerContent,
                          children
                        ]
                      }
                    )
                  ]
                }
              ),
              effectiveShowCenter && effectiveShowEnd && /* @__PURE__ */ jsx("div", { className: "w-0 shrink-0 relative", style: { marginLeft: "-8px", marginRight: "-8px", width: "16px", zIndex: 10 }, children: /* @__PURE__ */ jsx(
                ResizeHandle,
                {
                  separator: "end",
                  startPaneRef,
                  centerPaneRef,
                  endPaneRef,
                  onResizeEnd: handleResizeEnd,
                  onResetToDefault: handleResetToDefault,
                  ariaLabel: t("FX_RESIZE_CENTER_END")
                }
              ) }),
              allowEnd && /* @__PURE__ */ jsxs(
                "aside",
                {
                  ref: endPaneRef,
                  "aria-label": t("FX_END_PANE"),
                  className: cn(
                    "flex flex-col shrink-0 relative bg-sapphire-shell-bg-primary",
                    !mobileLayout && "border-l border-border"
                  ),
                  style: {
                    width: effectiveShowEnd ? mobileLayout ? "100%" : finalEndWidth : 0,
                    minWidth: effectiveShowEnd ? mobileLayout ? 0 : PANE_MIN_WIDTH : 0,
                    height: "100%",
                    paddingTop: effectiveShowEnd ? "1.5rem" : 0,
                    paddingInline: effectiveShowEnd ? "0.25rem" : 0,
                    overflow: "hidden",
                    visibility: effectiveShowEnd ? "visible" : "hidden",
                    transition: paneAnimationsEnabled ? `width 350ms cubic-bezier(0.4, 0, 0.2, 1), padding 350ms cubic-bezier(0.4, 0, 0.2, 1), visibility 0s ${effectiveShowEnd ? "0s" : "350ms"}` : "none"
                  },
                  children: [
                    /* @__PURE__ */ jsx(
                      "div",
                      {
                        className: "shrink-0",
                        "data-sap-ui-fastnavgroup": "true",
                        style: {
                          paddingInline: effectiveShowEnd ? mobileLayout ? "0.75rem" : "1rem" : 0,
                          opacity: effectiveShowEnd ? 1 : 0,
                          transform: effectiveShowEnd ? "translateX(0)" : "translateX(30%)",
                          transition: paneAnimationsEnabled ? "opacity 350ms cubic-bezier(0.4, 0, 0.2, 1), transform 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none"
                        },
                        children: isValidElement(endHeader) ? cloneElement(endHeader, { showBorder: endScrolled }) : endHeader
                      }
                    ),
                    /* @__PURE__ */ jsx(
                      "div",
                      {
                        ref: endContentRef,
                        className: "flex-1 fx-pane-content min-h-0",
                        "data-sap-ui-fastnavgroup": "true",
                        onScrollCapture: (e) => {
                          const target = e.target;
                          setEndScrolled(target.scrollTop > 0);
                        },
                        style: {
                          ...scrollbarStyle,
                          // Add padding at bottom when input is shown in end pane so content isn't covered
                          paddingBottom: input && inputInEndPane ? inputOverlayPadding : 0,
                          opacity: effectiveShowEnd ? 1 : 0,
                          transform: effectiveShowEnd ? "translateX(0)" : "translateX(30%)",
                          transition: paneAnimationsEnabled ? "opacity 350ms cubic-bezier(0.4, 0, 0.2, 1), transform 350ms cubic-bezier(0.4, 0, 0.2, 1)" : "none",
                          // Fade content at bottom edge when input is shown in end pane
                          ...input && inputInEndPane ? {
                            WebkitMaskImage: inputFadeMask,
                            maskImage: inputFadeMask
                          } : {}
                        },
                        children: endContent
                      }
                    )
                  ]
                }
              ),
              input && !isSettingsMode && !hideInputEntirely && (effectiveShowCenter || inputInEndPane) && /* @__PURE__ */ jsx(
                InputContainer,
                {
                  input,
                  inputInEndPane,
                  mobileLayout,
                  effectiveShowStart,
                  effectiveShowEnd,
                  finalStartWidth,
                  finalEndWidth,
                  paneAnimationsEnabled
                }
              )
            ]
          }
        ),
        userMenu && React5.cloneElement(userMenu, {
          open: userMenuOpen,
          opener: userMenuOpener,
          useDialog: mobileLayout || userMenuFromHamburger,
          placement: "End",
          verticalAlign: "Top",
          offset: 16,
          onClose: () => {
            setUserMenuOpen(false);
            setUserMenuFromHamburger(false);
          },
          onBackClick: () => {
            setNavigationOpen(true);
          }
        }),
        compactMode && navigationOpen && /* @__PURE__ */ jsxs(
          "div",
          {
            className: "fixed inset-0 z-50 flex",
            onClick: closeNavigation,
            children: [
              /* @__PURE__ */ jsx(
                "div",
                {
                  className: "absolute inset-0",
                  style: {
                    backgroundColor: "color-mix(in srgb, var(--foreground) 50%, transparent)",
                    animation: navigationClosing ? "fadeOut 300ms cubic-bezier(0.4, 0, 0.2, 1) forwards" : "fadeIn 300ms cubic-bezier(0.4, 0, 0.2, 1)"
                  }
                }
              ),
              /* @__PURE__ */ jsxs(
                "div",
                {
                  className: "relative w-64 h-full border-r border-border flex flex-col shadow-xl bg-sapphire-shell-bg-primary",
                  style: {
                    animation: navigationClosing ? "slideOutToLeft 300ms cubic-bezier(0.4, 0, 0.2, 1) forwards" : "slideInFromLeft 300ms cubic-bezier(0.4, 0, 0.2, 1)"
                  },
                  onClick: (e) => e.stopPropagation(),
                  children: [
                    /* @__PURE__ */ jsxs("div", { className: "flex items-center h-14 px-2 border-b border-border shrink-0", children: [
                      /* @__PURE__ */ jsx(
                        Button,
                        {
                          ref: navDialogCloseRef,
                          design: "Tertiary",
                          iconOnly: true,
                          icon: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-4 w-4" }),
                          onClick: closeNavigation,
                          className: "text-sapphire-text-primary"
                        }
                      ),
                      /* @__PURE__ */ jsxs(
                        "button",
                        {
                          onClick: () => handleNavSelect(navItems[0]?.name || ""),
                          className: "cursor-pointer flex items-center gap-2 ml-2 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring rounded",
                          children: [
                            /* @__PURE__ */ jsx(JouleIcon, { size: 28 }),
                            /* @__PURE__ */ jsxs("span", { className: "text-lg font-semibold", children: [
                              /* @__PURE__ */ jsx("span", { className: "text-purple-600", children: "Joule" }),
                              /* @__PURE__ */ jsx("span", { className: "text-foreground", children: "Work" })
                            ] })
                          ]
                        }
                      )
                    ] }),
                    /* @__PURE__ */ jsx("div", { className: "flex-1 overflow-y-auto", children: /* @__PURE__ */ jsx(
                      FxSideNavigation,
                      {
                        collapsed: false,
                        forceMerged: true,
                        dialogMode: true,
                        onSelectionChange: ({ name }) => handleNavSelect(name),
                        fixedItems: /* @__PURE__ */ jsxs(Fragment, { children: [
                          /* @__PURE__ */ jsx(
                            FxSideNavigationItem,
                            {
                              name: "notifications",
                              icon: /* @__PURE__ */ jsx(BellIcon, { className: "h-4 w-4" }),
                              text: t("FX_NOTIFICATIONS"),
                              badge: notificationsBadge?.toString(),
                              onClick: (e) => {
                                const opener = e.target.closest("button");
                                if (opener) onNotificationsClick?.(opener);
                              }
                            }
                          ),
                          /* @__PURE__ */ jsx(
                            FxSideNavigationItem,
                            {
                              name: "profile",
                              text: _selectedUserAccount?.titleText || "User",
                              avatar: /* @__PURE__ */ jsx(
                                Avatar,
                                {
                                  size: "XS",
                                  shape: "Square",
                                  image: _selectedUserAccount?.avatarSrc,
                                  initials: _selectedUserAccount?.avatarInitials || "U",
                                  colorScheme: _selectedUserAccount?.avatarColorScheme || "Accent1"
                                }
                              ),
                              onClick: () => {
                                if (userMenu) {
                                  setNavigationClosing(true);
                                  setTimeout(() => {
                                    setNavigationOpen(false);
                                    setNavigationClosing(false);
                                    setUserMenuFromHamburger(true);
                                    setUserMenuOpen(true);
                                  }, 300);
                                }
                              }
                            }
                          )
                        ] }),
                        children: navItems.map((item) => /* @__PURE__ */ jsx(
                          FxSideNavigationItem,
                          {
                            name: item.name,
                            icon: item.icon ? /* @__PURE__ */ jsx("span", { className: "h-4 w-4 flex items-center justify-center", children: item.icon }) : void 0,
                            text: item.text,
                            selected: mode === item.name,
                            showAddButton: !!item.createActionTooltip,
                            addButtonTooltip: item.createActionTooltip,
                            onAddClick
                          },
                          item.name
                        ))
                      }
                    ) })
                  ]
                }
              )
            ]
          }
        )
      ]
    }
  ) });
}
function FxRenameTitle({
  text,
  editMode: controlledEditMode,
  wrappingType = "Normal",
  onEditAccept,
  onEditCancel,
  onEditModeChange,
  className,
  ref
}) {
  const { t } = useTranslation("fx");
  const inputRef = useRef(null);
  const isCommittingRef = useRef(false);
  const [internalEditMode, setInternalEditMode] = useState(false);
  const [editValue, setEditValue] = useState(text);
  const [originalValue, setOriginalValue] = useState(text);
  const editMode = controlledEditMode ?? internalEditMode;
  useEffect(() => {
    if (!editMode) {
      setEditValue(text);
    }
  }, [text, editMode]);
  useEffect(() => {
    if (editMode && inputRef.current) {
      setTimeout(() => {
        inputRef.current?.focus();
        inputRef.current?.select();
      }, 0);
    }
  }, [editMode]);
  const enterEditMode = useCallback(() => {
    setOriginalValue(text);
    setEditValue(text);
    setInternalEditMode(true);
    onEditModeChange?.({ editMode: true });
  }, [text, onEditModeChange]);
  const exitEditMode = useCallback(() => {
    setInternalEditMode(false);
    onEditModeChange?.({ editMode: false });
  }, [onEditModeChange]);
  const acceptEdit = useCallback(() => {
    const newValue = editValue.trim();
    const previousValue = originalValue;
    exitEditMode();
    onEditAccept?.({ value: newValue, previousValue });
  }, [editValue, originalValue, exitEditMode, onEditAccept]);
  const cancelEdit = useCallback(() => {
    const cancelledValue = editValue;
    exitEditMode();
    onEditCancel?.({ value: cancelledValue });
  }, [editValue, exitEditMode, onEditCancel]);
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === "Enter" && editValue.trim().length > 0) {
        e.preventDefault();
        isCommittingRef.current = true;
        acceptEdit();
      } else if (e.key === "Escape") {
        e.preventDefault();
        isCommittingRef.current = true;
        cancelEdit();
      }
    },
    [editValue, acceptEdit, cancelEdit]
  );
  const handleBlur = useCallback(() => {
    if (isCommittingRef.current) {
      isCommittingRef.current = false;
      return;
    }
    if (editValue.trim().length > 0) {
      acceptEdit();
    } else {
      cancelEdit();
    }
  }, [editValue, acceptEdit, cancelEdit]);
  const handleDoubleClick = useCallback(() => {
    if (!editMode) {
      enterEditMode();
    }
  }, [editMode, enterEditMode]);
  useImperativeHandle(ref, () => ({
    enterEditMode,
    exitEditMode,
    getEditValue: () => editValue
  }));
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        "relative inline-flex items-center min-h-9",
        editMode && "w-full max-w-[600px]",
        className
      ),
      children: [
        !editMode && /* @__PURE__ */ jsx(
          "span",
          {
            className: cn(
              "relative z-10 font-semibold cursor-text flex items-center min-h-9",
              wrappingType === "None" ? "truncate" : "whitespace-pre-line"
            ),
            title: t("FX_DOUBLE_CLICK_TO_RENAME"),
            onDoubleClick: handleDoubleClick,
            children: text
          }
        ),
        editMode && /* @__PURE__ */ jsx("div", { className: "flex items-center w-full bg-sapphire-background-primary border-2 border-sapphire-border-accent rounded-md px-3.5 min-h-9", children: /* @__PURE__ */ jsx(
          "input",
          {
            ref: inputRef,
            type: "text",
            value: editValue,
            onChange: (e) => setEditValue(e.target.value),
            onKeyDown: handleKeyDown,
            onBlur: handleBlur,
            className: "flex-1 min-w-0 bg-transparent border-none outline-none text-base font-semibold text-sapphire-text-primary selection:bg-sapphire-border-accent selection:text-white"
          }
        ) })
      ]
    }
  );
}
var NavBackIcon = ({ className }) => /* @__PURE__ */ jsx("svg", { xmlns: "http://www.w3.org/2000/svg", width: "20", height: "20", viewBox: "0 0 20 20", fill: "none", className, children: /* @__PURE__ */ jsx("path", { d: "M13.4939 4.04418C13.1385 3.66784 12.5459 3.65098 12.1694 4.00634L6.54443 9.31517C6.357 9.4921 6.25047 9.7386 6.25024 9.99633C6.25018 10.2543 6.35696 10.5015 6.54443 10.6787L12.1694 15.9936C12.5458 16.3491 13.1383 16.3321 13.4939 15.9558C13.8493 15.5795 13.8323 14.9869 13.4561 14.6313L8.55371 9.99633L13.4561 5.36864C13.8324 5.01321 13.8492 4.42064 13.4939 4.04418Z", fill: "currentColor" }) });
var BUTTON_GAP = 8;
var BREADCRUMB_TITLE_GAP = 16;
var PRIORITY_ORDER = {
  alwaysOverflow: -1,
  low: 0,
  medium: 1,
  high: 2
};
var SMALL_SCREEN_BREAKPOINT = 600;
var RELATION_TYPE_LABELS = {
  space: "Related Spaces",
  job: "Related Jobs",
  conversation: "Related Conversations"
};
var RELATION_TYPE_ORDER = ["space", "job", "conversation"];
function FxPaneHeaderAction({
  icon,
  text,
  design = "Tertiary",
  showText = false,
  priority = "medium",
  centered = false,
  disabled = false,
  tooltip,
  onClick,
  className,
  ref
}) {
  const showTextLabel = showText && text;
  const buttonRef = useRef(null);
  useImperativeHandle(ref, () => buttonRef.current);
  const handleClick = (_detail) => {
    onClick?.({}, buttonRef.current?.nativeElement || null);
  };
  return /* @__PURE__ */ jsx(
    Button,
    {
      ref: buttonRef,
      design,
      disabled,
      tooltip: tooltip || text,
      icon,
      iconOnly: !showTextLabel,
      onClick: handleClick,
      className: cn("shrink-0 fx-pane-header-action", className),
      "data-priority": priority,
      "data-centered": centered,
      "data-show-text": showText,
      "data-text": text,
      children: showTextLabel && text
    }
  );
}
FxPaneHeaderAction.displayName = "FxPaneHeaderAction";
function FxPaneHeaderSegmentedAction({
  priority = "medium",
  centered = false,
  selectedId,
  showText = true,
  // Can be controlled externally for text collapse
  children,
  onSelectionChange,
  className,
  ref
}) {
  const options = [];
  Children.forEach(children, (child) => {
    if (isValidElement(child) && child.type?.displayName === "FxPaneHeaderSegmentedOption") {
      options.push(child.props);
    }
  });
  return /* @__PURE__ */ jsx(
    SegmentedButton,
    {
      ref,
      selectedId,
      onSelectionChange: (detail) => onSelectionChange?.({ selectedId: detail.selectedId }),
      className: cn("shrink-0", className),
      "data-priority": priority,
      "data-centered": centered,
      "data-show-text": showText,
      children: options.map((option) => /* @__PURE__ */ jsx(
        SegmentedButtonItem,
        {
          id: option.id,
          icon: option.icon,
          text: showText ? option.text : void 0,
          selected: option.selected
        },
        option.id
      ))
    }
  );
}
FxPaneHeaderSegmentedAction.displayName = "FxPaneHeaderSegmentedAction";
var FxPaneHeaderSegmentedOption = () => null;
FxPaneHeaderSegmentedOption.displayName = "FxPaneHeaderSegmentedOption";
function FxPaneHeaderSplitAction({
  icon,
  text,
  showText = false,
  children,
  onClick,
  onOptionSelect,
  className,
  ref
}) {
  const splitRef = useRef(null);
  const [dropdownOpen, setDropdownOpen] = useState(false);
  const showTextLabel = showText && text;
  const options = [];
  Children.forEach(children, (child) => {
    if (isValidElement(child) && child.type?.displayName === "FxPaneHeaderSplitOption") {
      options.push(child.props);
    }
  });
  const mergedRef = useCallback(
    (node) => {
      splitRef.current = node;
      if (typeof ref === "function") {
        ref(node);
      } else if (ref) {
        ref.current = node;
      }
    },
    [ref]
  );
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(
      SplitButton,
      {
        ref: mergedRef,
        icon,
        text: showTextLabel ? text : void 0,
        onClick: () => onClick?.(void 0),
        onArrowClick: () => setDropdownOpen((prev) => !prev),
        activeArrowButton: dropdownOpen,
        className: cn("shrink-0", className)
      }
    ),
    /* @__PURE__ */ jsx(
      Menu,
      {
        opener: splitRef.current?.nativeElement,
        open: dropdownOpen,
        onClose: () => setDropdownOpen(false),
        onItemClick: (detail) => {
          const optionId = detail.item.data;
          if (optionId) onOptionSelect?.({ optionId });
          setDropdownOpen(false);
        },
        children: options.map((option) => /* @__PURE__ */ jsx(
          MenuItem,
          {
            icon: option.icon,
            text: option.text,
            data: option.id
          },
          option.id
        ))
      }
    )
  ] });
}
FxPaneHeaderSplitAction.displayName = "FxPaneHeaderSplitAction";
var FxPaneHeaderSplitOption = () => null;
FxPaneHeaderSplitOption.displayName = "FxPaneHeaderSplitOption";
function FxPaneHeader({
  pane = "center",
  showBack = false,
  title = "",
  showTitleArrow = false,
  showAddButton = false,
  addButtonTooltip,
  related = [],
  titleEditable = false,
  titleEditMode: controlledEditMode,
  headingLevel,
  breadcrumbs,
  subheader,
  children,
  onBackClick,
  onTitleArrowClick,
  onRelationClick,
  onTitleEditAccept,
  onTitleEditCancel,
  onTitleEditModeChange,
  onAddClick,
  showBorder = false,
  className,
  style,
  ref
}) {
  const { t } = useTranslation("fx");
  const headerRef = useRef(null);
  const Heading = headingLevel ?? (pane === "center" ? "h1" : "h2");
  const measureContainerRef = useRef(null);
  const titleArrowRef = useRef(null);
  const relationsButtonRef = useRef(null);
  const overflowButtonRef = useRef(null);
  const renameTitleRef = useRef(null);
  const [internalEditMode, setInternalEditMode] = useState(false);
  const [relationsMenuOpen, setRelationsMenuOpen] = useState(false);
  const [overflowMenuOpen, setOverflowMenuOpen] = useState(false);
  const [headerWidth, setHeaderWidth] = useState(0);
  const [measurements, setMeasurements] = useState(null);
  const editMode = controlledEditMode ?? internalEditMode;
  const layoutContext = useFxLayoutContext();
  const startPaneVisible = layoutContext?.startVisible ?? false;
  const endPaneVisible = layoutContext?.endVisible ?? false;
  const compactMode = layoutContext?.compactMode ?? false;
  const leftmostVisiblePane = layoutContext?.leftmostVisiblePane ?? "start";
  const maxPanes = layoutContext?.maxPanes ?? 3;
  const utilityEndPane = layoutContext?.utilityEndPane ?? false;
  const isSettingsMode = layoutContext?.isSettingsMode ?? false;
  const allowStart = layoutContext?.allowStart ?? true;
  const allowEnd = layoutContext?.allowEnd ?? true;
  const isCenterHeader = pane === "center";
  const isEndHeader = pane === "end";
  const handleToggleStart = layoutContext?.toggleStartPane;
  const handleToggleEnd = layoutContext?.toggleEndPane;
  const handleHamburger = layoutContext?.openNavigation;
  const handleClose = isEndHeader && (maxPanes === 1 || utilityEndPane) ? layoutContext?.toggleEndPane : void 0;
  const showStartToggle = isCenterHeader && allowStart && handleToggleStart && (maxPanes >= 2 || !showBack);
  const toggleStartUsesBackIcon = maxPanes === 1;
  const showEndToggle = isCenterHeader && allowEnd && !utilityEndPane && !isSettingsMode && handleToggleEnd;
  const showHamburger = compactMode && handleHamburger && pane === leftmostVisiblePane;
  const showBackButton = (isCenterHeader || isEndHeader) && showBack;
  const hasRelated = related.length > 0;
  const isSmallScreenEditMode = editMode && headerWidth > 0 && headerWidth < SMALL_SCREEN_BREAKPOINT;
  useImperativeHandle(ref, () => ({
    focus: () => headerRef.current?.focus(),
    get nativeElement() {
      return headerRef.current;
    },
    getNativeElement: () => headerRef.current
  }));
  const actionItems = useMemo(() => {
    const items = [];
    const flattenChildren = (nodes) => {
      const result = [];
      Children.forEach(nodes, (child) => {
        if (!isValidElement(child)) return;
        if (child.type === React5.Fragment) {
          result.push(...flattenChildren(child.props.children));
        } else {
          result.push(child);
        }
      });
      return result;
    };
    const flatChildren = flattenChildren(children);
    flatChildren.forEach((child) => {
      const props = child.props;
      const priorityProp = props.priority || "medium";
      const priorityValue = PRIORITY_ORDER[priorityProp] ?? 1;
      const displayName = child.type?.displayName;
      const isSegmented = displayName === "FxPaneHeaderSegmentedAction";
      items.push({
        element: child,
        priority: priorityValue,
        // For segmented actions, showText defaults to true (always has text initially)
        showText: isSegmented ? props.showText ?? true : props.showText ?? false,
        text: props.text ?? "",
        icon: props.icon,
        centered: props.centered ?? false,
        isSegmented
      });
    });
    return items;
  }, [children]);
  const breadcrumbItemData = useMemo(() => {
    if (!breadcrumbs || !isValidElement(breadcrumbs)) return [];
    const items = [];
    Children.forEach(breadcrumbs.props.children, (child) => {
      if (isValidElement(child) && child.type?.displayName === "BreadcrumbsItem") {
        const childProps = child.props;
        const text = typeof childProps.children === "string" ? childProps.children : "";
        items.push({ text });
      }
    });
    return items;
  }, [breadcrumbs]);
  const overflowState = useMemo(() => {
    const nonOverflowIndices = actionItems.map((item, i) => ({ item, i })).filter(({ item }) => item.priority !== PRIORITY_ORDER.alwaysOverflow).map(({ i }) => i);
    const totalBreadcrumbItems = breadcrumbItemData.length;
    if (!measurements || headerWidth === 0) {
      return {
        visibleIndices: new Set(nonOverflowIndices),
        textVisibility: actionItems.map((item) => item.showText),
        relationsVisible: hasRelated,
        breadcrumbOverflowCount: 0
      };
    }
    const m = measurements;
    const availableWidth = headerWidth;
    let baseFixedWidth = m.startSection + BUTTON_GAP;
    if (showHamburger && m.hamburger > 0) {
      baseFixedWidth += m.hamburger + BUTTON_GAP;
    }
    if (showStartToggle && m.toggleStart > 0) {
      baseFixedWidth += m.toggleStart + BUTTON_GAP;
    }
    if (showBackButton && m.backButton > 0) {
      baseFixedWidth += m.backButton + BUTTON_GAP;
    }
    if (editMode) {
      baseFixedWidth += 200;
    }
    let builtInWidth = 0;
    if (showEndToggle && m.toggleEnd > 0) {
      builtInWidth += m.toggleEnd + BUTTON_GAP;
    }
    const calcBreadcrumbsWidth = (overflowCount) => {
      if (totalBreadcrumbItems === 0) return 0;
      if (overflowCount >= totalBreadcrumbItems) {
        return m.breadcrumbOverflowOnly + BREADCRUMB_TITLE_GAP;
      }
      if (overflowCount === 0) {
        const itemsWidth = m.breadcrumbItems.reduce((sum, w) => sum + w, 0);
        return itemsWidth + BREADCRUMB_TITLE_GAP;
      }
      const visibleItemsWidth = m.breadcrumbItems.slice(overflowCount).reduce((sum, w) => sum + w, 0);
      return m.breadcrumbOverflow + visibleItemsWidth + BREADCRUMB_TITLE_GAP;
    };
    const actionStates = actionItems.map((item, index) => ({
      index,
      showText: item.showText,
      visible: item.priority !== PRIORITY_ORDER.alwaysOverflow,
      priority: item.priority
    }));
    let relationsVisible = hasRelated;
    let breadcrumbOverflowCount = 0;
    const calcActionsWidth = () => {
      let width = 0;
      if (relationsVisible && m.relations > 0) {
        width += m.relations + BUTTON_GAP;
      }
      for (const state of actionStates) {
        if (state.visible && m.actions[state.index]) {
          const aw = m.actions[state.index];
          width += (state.showText ? aw.withText : aw.iconOnly) + BUTTON_GAP;
        }
      }
      const hasHidden = actionStates.some((s) => !s.visible);
      const relationsOverflowed = hasRelated && !relationsVisible;
      if ((hasHidden || relationsOverflowed) && m.overflow > 0) {
        width += m.overflow + BUTTON_GAP;
      }
      return width;
    };
    const fits = (bcOverflow) => {
      const breadcrumbsWidth2 = calcBreadcrumbsWidth(bcOverflow);
      const fixedWidth2 = baseFixedWidth + breadcrumbsWidth2;
      const spaceForButtons2 = availableWidth - fixedWidth2 - builtInWidth;
      return calcActionsWidth() <= spaceForButtons2;
    };
    while (!fits(breadcrumbOverflowCount) && breadcrumbOverflowCount < totalBreadcrumbItems) {
      breadcrumbOverflowCount++;
    }
    const breadcrumbsWidth = calcBreadcrumbsWidth(breadcrumbOverflowCount);
    const fixedWidth = baseFixedWidth + breadcrumbsWidth;
    const spaceForButtons = availableWidth - fixedWidth - builtInWidth;
    const fitsActions = () => calcActionsWidth() <= spaceForButtons;
    while (!fitsActions()) {
      let targetIdx = -1;
      let targetPriority = Infinity;
      let targetOriginalIndex = Infinity;
      for (let i = 0; i < actionStates.length; i++) {
        const state = actionStates[i];
        if (state.visible && state.showText) {
          if (state.priority < targetPriority || state.priority === targetPriority && state.index < targetOriginalIndex) {
            targetIdx = i;
            targetPriority = state.priority;
            targetOriginalIndex = state.index;
          }
        }
      }
      if (targetIdx === -1) break;
      actionStates[targetIdx].showText = false;
    }
    if (!fitsActions() && relationsVisible && !relationsMenuOpen) {
      relationsVisible = false;
    }
    while (!fitsActions()) {
      let targetIdx = -1;
      let targetPriority = Infinity;
      let targetOriginalIndex = Infinity;
      for (let i = 0; i < actionStates.length; i++) {
        const state = actionStates[i];
        if (state.visible) {
          if (state.priority < targetPriority || state.priority === targetPriority && state.index < targetOriginalIndex) {
            targetIdx = i;
            targetPriority = state.priority;
            targetOriginalIndex = state.index;
          }
        }
      }
      if (targetIdx === -1) break;
      actionStates[targetIdx].visible = false;
    }
    return {
      visibleIndices: new Set(actionStates.filter((s) => s.visible).map((s) => s.index)),
      textVisibility: actionStates.map((s) => s.showText),
      relationsVisible,
      breadcrumbOverflowCount
    };
  }, [
    measurements,
    headerWidth,
    actionItems,
    hasRelated,
    showHamburger,
    showStartToggle,
    showBackButton,
    showEndToggle,
    editMode,
    relationsMenuOpen,
    breadcrumbItemData
  ]);
  const centeredActionIndex = useMemo(() => {
    for (let i = 0; i < actionItems.length; i++) {
      if (actionItems[i].centered && overflowState.visibleIndices.has(i)) {
        return i;
      }
    }
    return -1;
  }, [actionItems, overflowState.visibleIndices]);
  const centeringAvailableSpace = useMemo(() => {
    if (!measurements || headerWidth === 0 || centeredActionIndex === -1) {
      return [-1, -1];
    }
    const m = measurements;
    const centeredMeasurement = m.actions[centeredActionIndex];
    if (!centeredMeasurement) {
      return [-1, -1];
    }
    const showText = overflowState.textVisibility[centeredActionIndex];
    const centeredWidth = showText ? centeredMeasurement.withText : centeredMeasurement.iconOnly;
    let leftContentWidth = m.startSection + BUTTON_GAP;
    if (showHamburger && m.hamburger > 0) {
      leftContentWidth += m.hamburger + BUTTON_GAP;
    }
    if (showStartToggle && m.toggleStart > 0) {
      leftContentWidth += m.toggleStart + BUTTON_GAP;
    }
    for (let i = 0; i < centeredActionIndex; i++) {
      if (overflowState.visibleIndices.has(i)) {
        const aw = m.actions[i];
        if (aw) {
          const hasText = overflowState.textVisibility[i];
          leftContentWidth += (hasText ? aw.withText : aw.iconOnly) + BUTTON_GAP;
        }
      }
    }
    let rightContentWidth = 0;
    for (let i = centeredActionIndex + 1; i < actionItems.length; i++) {
      if (overflowState.visibleIndices.has(i)) {
        const aw = m.actions[i];
        if (aw) {
          const hasText = overflowState.textVisibility[i];
          rightContentWidth += (hasText ? aw.withText : aw.iconOnly) + BUTTON_GAP;
        }
      }
    }
    const hasOverflowedActions = actionItems.some((_, i) => !overflowState.visibleIndices.has(i));
    const relationsOverflowed = hasRelated && !overflowState.relationsVisible;
    const needsOverflowButton = hasOverflowedActions || relationsOverflowed;
    if (needsOverflowButton && m.overflow > 0) {
      rightContentWidth += m.overflow + BUTTON_GAP;
    }
    if (overflowState.relationsVisible && hasRelated && m.relations > 0) {
      rightContentWidth += m.relations + BUTTON_GAP;
    }
    if (showEndToggle && m.toggleEnd > 0) {
      rightContentWidth += m.toggleEnd + BUTTON_GAP;
    }
    const halfTotal = headerWidth / 2;
    const halfCentered = centeredWidth / 2;
    const leftSpace = halfTotal - halfCentered - leftContentWidth;
    const rightSpace = halfTotal - halfCentered - rightContentWidth;
    return [Math.floor(leftSpace), Math.floor(rightSpace)];
  }, [
    measurements,
    headerWidth,
    centeredActionIndex,
    actionItems,
    overflowState,
    showHamburger,
    showStartToggle,
    showEndToggle,
    hasRelated
  ]);
  const centeringActive = centeringAvailableSpace[0] >= BUTTON_GAP && centeringAvailableSpace[1] >= BUTTON_GAP;
  const visibleActions = actionItems.filter((_, i) => overflowState.visibleIndices.has(i));
  const overflowedActions = actionItems.filter((_, i) => !overflowState.visibleIndices.has(i));
  const actionsBeforeCentered = centeringActive ? visibleActions.filter((_, i) => {
    const originalIndex = actionItems.indexOf(visibleActions[i]);
    return originalIndex < centeredActionIndex;
  }) : [];
  const actionsAfterCentered = centeringActive ? visibleActions.filter((_, i) => {
    const originalIndex = actionItems.indexOf(visibleActions[i]);
    return originalIndex > centeredActionIndex;
  }) : [];
  const centeredAction = centeringActive && centeredActionIndex !== -1 ? actionItems[centeredActionIndex] : null;
  const showOverflowButton = overflowedActions.length > 0 || hasRelated && !overflowState.relationsVisible;
  useEffect(() => {
    const header = headerRef.current;
    if (!header) return;
    const observer = new ResizeObserver((entries) => {
      const newWidth = Math.floor(entries[0]?.contentRect.width || 0);
      if (newWidth !== headerWidth) {
        setHeaderWidth(newWidth);
      }
    });
    observer.observe(header);
    return () => observer.disconnect();
  }, [headerWidth]);
  const renderBreadcrumbs = useCallback(() => {
    if (!breadcrumbs || !isValidElement(breadcrumbs)) return breadcrumbs;
    return cloneElement(breadcrumbs, {
      overflowCount: overflowState.breadcrumbOverflowCount,
      noPadding: true
    });
  }, [breadcrumbs, overflowState.breadcrumbOverflowCount]);
  useEffect(() => {
    const measureContainer = measureContainerRef.current;
    if (!measureContainer) return;
    requestAnimationFrame(() => {
      const getWidth = (selector) => {
        const el = measureContainer.querySelector(selector);
        return el?.offsetWidth || 0;
      };
      const breadcrumbItems = [];
      for (let i = 0; i < breadcrumbItemData.length; i++) {
        breadcrumbItems.push(getWidth(`[data-measure="bc-item-${i}"]`));
      }
      const actions = [];
      for (let i = 0; i < actionItems.length; i++) {
        actions.push({
          iconOnly: getWidth(`[data-measure="${i}-icon"]`),
          withText: getWidth(`[data-measure="${i}-text"]`)
        });
      }
      setMeasurements({
        breadcrumbItems,
        breadcrumbOverflow: getWidth("[data-measure='bc-overflow']"),
        breadcrumbOverflowOnly: getWidth("[data-measure='bc-overflow-only']"),
        breadcrumbSeparator: getWidth("[data-measure='bc-separator']"),
        startSection: getWidth("[data-measure='start-section']"),
        hamburger: getWidth("[data-measure='hamburger']"),
        toggleStart: getWidth("[data-measure='toggle-start']"),
        backButton: getWidth("[data-measure='back-button']"),
        toggleEnd: getWidth("[data-measure='toggle-end']"),
        overflow: getWidth("[data-measure='overflow']"),
        relations: getWidth("[data-measure='relations']"),
        actions
      });
    });
  }, [actionItems, breadcrumbItemData]);
  const handleTitleEditAccept = useCallback(
    (detail) => {
      setInternalEditMode(false);
      onTitleEditAccept?.(detail);
    },
    [onTitleEditAccept]
  );
  const handleTitleEditCancel = useCallback(
    (detail) => {
      setInternalEditMode(false);
      onTitleEditCancel?.(detail);
    },
    [onTitleEditCancel]
  );
  const handleTitleEditModeChange = useCallback(
    ({ editMode: newEditMode }) => {
      setInternalEditMode(newEditMode);
      onTitleEditModeChange?.({ editMode: newEditMode });
    },
    [onTitleEditModeChange]
  );
  const getRelationIcon = (type) => {
    switch (type) {
      case "space":
        return /* @__PURE__ */ jsx(SpacesIcon, { className: "h-4 w-4" });
      case "job":
        return /* @__PURE__ */ jsx(JobsIcon, { className: "h-4 w-4" });
      case "conversation":
        return /* @__PURE__ */ jsx(ConversationsIcon, { className: "h-4 w-4" });
      default:
        return /* @__PURE__ */ jsx(Share2, { className: "h-4 w-4" });
    }
  };
  const groupedRelations = useMemo(() => {
    const groups = {
      space: [],
      job: [],
      conversation: []
    };
    related.forEach((item) => {
      if (groups[item.type]) {
        groups[item.type].push(item);
      }
    });
    return groups;
  }, [related]);
  const renderRelationMenuItems = useCallback(
    (closeMenu) => {
      const items = [];
      let isFirstGroup = true;
      RELATION_TYPE_ORDER.forEach((type) => {
        const typeRelations = groupedRelations[type];
        if (typeRelations.length === 0) return;
        if (!isFirstGroup) {
          items.push(/* @__PURE__ */ jsx(MenuSeparator, {}, `sep-${type}`));
        }
        isFirstGroup = false;
        items.push(
          /* @__PURE__ */ jsx(MenuHeader, { text: RELATION_TYPE_LABELS[type] }, `header-${type}`)
        );
        typeRelations.forEach((item) => {
          items.push(
            /* @__PURE__ */ jsx(
              MenuItem,
              {
                icon: item.icon || getRelationIcon(item.type),
                text: item.name,
                onClick: () => {
                  onRelationClick?.(item);
                  closeMenu();
                }
              },
              item.id
            )
          );
        });
      });
      return items;
    },
    [groupedRelations, onRelationClick]
  );
  const renderAction = (item, index, showText) => {
    const child = item.element;
    const displayName = child.type?.displayName;
    const props = child.props;
    if (displayName === "FxPaneHeaderAction") {
      return /* @__PURE__ */ jsx(
        FxPaneHeaderAction,
        {
          ...props,
          showText: showText && props.showText
        },
        index
      );
    }
    if (displayName === "FxPaneHeaderSegmentedAction") {
      const segProps = child.props;
      return /* @__PURE__ */ jsx(
        FxPaneHeaderSegmentedAction,
        {
          ...segProps,
          showText
        },
        index
      );
    }
    if (displayName === "FxPaneHeaderSplitAction") {
      const splitProps = child.props;
      return /* @__PURE__ */ jsx(
        FxPaneHeaderSplitAction,
        {
          ...splitProps,
          showText: showText && splitProps.showText
        },
        index
      );
    }
    return React5.cloneElement(child, { key: index });
  };
  const renderOverflowedActions = useCallback(
    (closeMenu, isFirstInMenu) => {
      const items = [];
      overflowedActions.forEach((item, index) => {
        const child = item.element;
        const displayName = child.type?.displayName;
        const isFirst = index === 0 && isFirstInMenu;
        const isLast = index === overflowedActions.length - 1;
        if (displayName === "FxPaneHeaderAction") {
          const props = child.props;
          items.push(
            /* @__PURE__ */ jsx(
              MenuItem,
              {
                icon: props.icon,
                text: props.text || "Action",
                onClick: () => {
                  props.onClick?.({}, overflowButtonRef.current?.nativeElement || null);
                  closeMenu();
                }
              },
              `action-${index}`
            )
          );
        } else if (displayName === "FxPaneHeaderSegmentedAction") {
          const props = child.props;
          const options = [];
          Children.forEach(props.children, (optChild) => {
            if (isValidElement(optChild) && optChild.type?.displayName === "FxPaneHeaderSegmentedOption") {
              options.push(optChild.props);
            }
          });
          if (!isFirst && items.length > 0) {
            items.push(/* @__PURE__ */ jsx(MenuSeparator, {}, `seg-sep-before-${index}`));
          }
          options.forEach((opt) => {
            const isSelected = opt.selected || opt.id === props.selectedId;
            items.push(
              /* @__PURE__ */ jsx(
                MenuItem,
                {
                  icon: opt.icon,
                  text: opt.text,
                  endContent: isSelected ? /* @__PURE__ */ jsx("span", { className: "text-primary", children: "\u2713" }) : void 0,
                  onClick: () => {
                    props.onSelectionChange?.({ selectedId: opt.id });
                    closeMenu();
                  }
                },
                `seg-${index}-${opt.id}`
              )
            );
          });
          if (!isLast) {
            items.push(/* @__PURE__ */ jsx(MenuSeparator, {}, `seg-sep-after-${index}`));
          }
        } else if (displayName === "FxPaneHeaderSplitAction") {
          const props = child.props;
          const options = [];
          Children.forEach(props.children, (optChild) => {
            if (isValidElement(optChild) && optChild.type?.displayName === "FxPaneHeaderSplitOption") {
              options.push(optChild.props);
            }
          });
          items.push(
            /* @__PURE__ */ jsx(
              MenuItem,
              {
                icon: props.icon,
                text: props.text || "Action",
                children: options.map((opt) => /* @__PURE__ */ jsx(
                  MenuItem,
                  {
                    icon: opt.icon,
                    text: opt.text,
                    onClick: () => {
                      props.onOptionSelect?.({ optionId: opt.id });
                      closeMenu();
                    }
                  },
                  `split-opt-${index}-${opt.id}`
                ))
              },
              `split-${index}`
            )
          );
        } else {
          const props = child.props;
          if (props.text || props.icon) {
            items.push(
              /* @__PURE__ */ jsx(
                MenuItem,
                {
                  icon: props.icon,
                  text: props.text || "Action",
                  onClick: () => {
                    props.onClick?.({});
                    closeMenu();
                  }
                },
                `unknown-${index}`
              )
            );
          }
        }
      });
      return items;
    },
    [overflowedActions]
  );
  const renderOverflowRelationsSubmenu = useCallback(
    (closeMenu) => {
      const submenuItems = [];
      let isFirstGroup = true;
      RELATION_TYPE_ORDER.forEach((type) => {
        const typeRelations = groupedRelations[type];
        if (typeRelations.length === 0) return;
        if (!isFirstGroup) {
          submenuItems.push(/* @__PURE__ */ jsx(MenuSeparator, {}, `overflow-sep-${type}`));
        }
        isFirstGroup = false;
        submenuItems.push(
          /* @__PURE__ */ jsx(MenuHeader, { text: RELATION_TYPE_LABELS[type] }, `overflow-header-${type}`)
        );
        typeRelations.forEach((item) => {
          submenuItems.push(
            /* @__PURE__ */ jsx(
              MenuItem,
              {
                icon: item.icon || getRelationIcon(item.type),
                text: item.name,
                onClick: () => {
                  onRelationClick?.(item);
                  closeMenu();
                }
              },
              `overflow-rel-${item.id}`
            )
          );
        });
      });
      return /* @__PURE__ */ jsx(
        MenuItem,
        {
          icon: /* @__PURE__ */ jsx(Share2, { className: "h-4 w-4" }),
          text: t("FX_RELATED"),
          children: submenuItems
        },
        "overflow-relations"
      );
    },
    [groupedRelations, onRelationClick]
  );
  return /* @__PURE__ */ jsxs(
    "header",
    {
      ref: headerRef,
      className: cn(
        "fx-pane-header border-b transition-colors duration-200",
        showBorder ? "border-border" : "border-transparent",
        editMode && "fx-pane-header--title-edit-mode",
        className
      ),
      style: {
        containerType: "inline-size",
        marginInline: "-1.25rem",
        paddingInline: "1.25rem",
        paddingBottom: "1.5rem",
        ...style
      },
      children: [
        /* @__PURE__ */ jsxs(
          "div",
          {
            ref: measureContainerRef,
            "aria-hidden": "true",
            style: {
              position: "absolute",
              visibility: "hidden",
              pointerEvents: "none",
              width: "max-content",
              top: "-9999px",
              left: "-9999px"
            },
            children: [
              breadcrumbItemData.map((item, idx) => /* @__PURE__ */ jsxs("div", { "data-measure": `bc-item-${idx}`, className: "inline-flex items-center", children: [
                /* @__PURE__ */ jsx("span", { className: "inline-flex items-center gap-0.5 text-base font-semibold text-foreground px-0.5 py-0.5", children: item.text }),
                /* @__PURE__ */ jsx("span", { className: "mx-1.5 text-base font-semibold select-none", children: "/" })
              ] }, idx)),
              /* @__PURE__ */ jsxs("div", { "data-measure": "bc-overflow", className: "inline-flex items-center", children: [
                /* @__PURE__ */ jsxs("span", { className: "inline-flex items-center gap-0.5 text-base px-0.5 py-0.5", children: [
                  /* @__PURE__ */ jsx("span", { children: "\u2026" }),
                  /* @__PURE__ */ jsx(ChevronDown, { className: "h-3 w-3" })
                ] }),
                /* @__PURE__ */ jsx("span", { className: "mx-1.5 text-base font-semibold select-none", children: "/" })
              ] }),
              /* @__PURE__ */ jsx("div", { "data-measure": "bc-overflow-only", className: "inline-flex items-center", children: /* @__PURE__ */ jsxs("span", { className: "inline-flex items-center gap-0.5 text-base px-0.5 py-0.5", children: [
                /* @__PURE__ */ jsx("span", { children: "\u2026" }),
                /* @__PURE__ */ jsx(ChevronDown, { className: "h-3 w-3" })
              ] }) }),
              /* @__PURE__ */ jsx("div", { "data-measure": "bc-separator", className: "inline-flex items-center gap-2", children: /* @__PURE__ */ jsx("span", { className: "text-foreground font-semibold", children: "/" }) }),
              /* @__PURE__ */ jsxs("div", { "data-measure": "start-section", className: "inline-flex items-center gap-2", children: [
                /* @__PURE__ */ jsx("span", { className: "text-base font-semibold whitespace-pre-line", children: title }),
                showTitleArrow && /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4" })
              ] }),
              /* @__PURE__ */ jsx("div", { "data-measure": "hamburger", className: "inline-block", children: /* @__PURE__ */ jsx(Button, { design: "Tertiary", iconOnly: true, icon: /* @__PURE__ */ jsx(MenuIcon, { className: "h-4 w-4" }) }) }),
              /* @__PURE__ */ jsx("div", { "data-measure": "toggle-start", className: "inline-block", children: /* @__PURE__ */ jsx(ToggleButton, { design: "SecondaryNeutral", icon: /* @__PURE__ */ jsx(List, { className: "h-4 w-4" }) }) }),
              /* @__PURE__ */ jsx("div", { "data-measure": "back-button", className: "inline-block", children: /* @__PURE__ */ jsx(Button, { design: "SecondaryNeutral", size: "Large", iconOnly: true, icon: /* @__PURE__ */ jsx(NavBackIcon, {}) }) }),
              /* @__PURE__ */ jsx("div", { "data-measure": "toggle-end", className: "inline-block", children: /* @__PURE__ */ jsx(ToggleButton, { design: "SecondaryNeutral", icon: /* @__PURE__ */ jsx(NewConversationIcon, { className: "h-4 w-4" }) }) }),
              /* @__PURE__ */ jsx("div", { "data-measure": "overflow", className: "inline-block", children: /* @__PURE__ */ jsx(Button, { design: "SecondaryNeutral", iconOnly: true, icon: /* @__PURE__ */ jsx(MoreIcon, { className: "h-4 w-4" }) }) }),
              /* @__PURE__ */ jsx("div", { "data-measure": "relations", className: "inline-block", children: /* @__PURE__ */ jsx(Button, { design: "SecondaryNeutral", iconOnly: true, icon: /* @__PURE__ */ jsx(Share2, { className: "h-4 w-4" }) }) }),
              actionItems.map((item, index) => {
                const displayName = item.element.type?.displayName;
                const isSegmented = displayName === "FxPaneHeaderSegmentedAction";
                if (isSegmented) {
                  return /* @__PURE__ */ jsxs(React5.Fragment, { children: [
                    /* @__PURE__ */ jsx("div", { "data-measure": `${index}-icon`, className: "inline-block", children: renderAction(item, index, false) }),
                    /* @__PURE__ */ jsx("div", { "data-measure": `${index}-text`, className: "inline-block", children: renderAction(item, index, true) })
                  ] }, index);
                }
                return /* @__PURE__ */ jsxs(React5.Fragment, { children: [
                  /* @__PURE__ */ jsx("div", { "data-measure": `${index}-icon`, className: "inline-block", children: renderAction(item, index, false) }),
                  /* @__PURE__ */ jsx("div", { "data-measure": `${index}-text`, className: "inline-block", children: renderAction(item, index, true) })
                ] }, index);
              })
            ]
          }
        ),
        /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 min-h-9", children: [
          !centeringActive && /* @__PURE__ */ jsxs(Fragment, { children: [
            /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 min-w-0 flex-1", children: [
              showHamburger && !isSmallScreenEditMode && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "Tertiary",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(MenuIcon, { className: "h-4 w-4" }),
                  onClick: () => handleHamburger(),
                  tooltip: t("FX_OPEN_NAVIGATION"),
                  className: "shrink-0 fx-pane-header-hamburger text-sapphire-text-primary"
                }
              ),
              showStartToggle && !isSmallScreenEditMode && (toggleStartUsesBackIcon ? /* @__PURE__ */ jsx(
                Button,
                {
                  design: "SecondaryNeutral",
                  size: "Large",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(NavBackIcon, {}),
                  onClick: () => handleToggleStart(),
                  tooltip: t("FX_BACK_TO_LIST"),
                  className: "shrink-0 fx-pane-header-toggle-start"
                }
              ) : /* @__PURE__ */ jsx(
                ToggleButton,
                {
                  design: "SecondaryNeutral",
                  icon: /* @__PURE__ */ jsx(List, { className: "h-4 w-4" }),
                  pressed: startPaneVisible,
                  onClick: () => handleToggleStart(),
                  tooltip: startPaneVisible ? "Hide list" : "Show list",
                  className: "fx-pane-header-toggle-start"
                }
              )),
              showBackButton && !isSmallScreenEditMode && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "SecondaryNeutral",
                  size: "Large",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(NavBackIcon, {}),
                  onClick: () => onBackClick?.(),
                  tooltip: t("FX_BACK"),
                  className: "shrink-0 fx-pane-header-back"
                }
              ),
              /* @__PURE__ */ jsx("div", { className: "flex flex-col min-w-0 flex-1", children: /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 min-w-0", children: [
                breadcrumbs && !editMode && /* @__PURE__ */ jsx("div", { className: "min-w-0", children: renderBreadcrumbs() }),
                breadcrumbs && !editMode && title && /* @__PURE__ */ jsx("span", { className: "text-foreground font-semibold text-base select-none shrink-0 -mx-1", children: "/" }),
                titleEditable ? /* @__PURE__ */ jsx(
                  FxRenameTitle,
                  {
                    ref: renameTitleRef,
                    text: title,
                    editMode,
                    wrappingType: "Normal",
                    onEditAccept: handleTitleEditAccept,
                    onEditCancel: handleTitleEditCancel,
                    onEditModeChange: handleTitleEditModeChange,
                    className: "min-w-0"
                  }
                ) : /* @__PURE__ */ jsx(Heading, { className: "text-base font-semibold whitespace-pre-line min-w-0 flex items-center min-h-9", children: title }),
                showTitleArrow && !editMode && /* @__PURE__ */ jsx(
                  "button",
                  {
                    ref: titleArrowRef,
                    type: "button",
                    onClick: () => onTitleArrowClick?.(titleArrowRef.current),
                    className: "shrink-0 p-0.5 hover:bg-muted rounded",
                    children: /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4" })
                  }
                )
              ] }) })
            ] }),
            !isSmallScreenEditMode && /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-1 shrink-0 fx-pane-header-end-section", children: [
              visibleActions.map((item) => {
                const originalIndex = actionItems.indexOf(item);
                return renderAction(item, originalIndex, overflowState.textVisibility[originalIndex]);
              }),
              overflowState.relationsVisible && hasRelated && /* @__PURE__ */ jsxs(Fragment, { children: [
                /* @__PURE__ */ jsx(
                  Button,
                  {
                    ref: relationsButtonRef,
                    design: "SecondaryNeutral",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx(Share2, { className: "h-4 w-4" }),
                    onClick: () => {
                      setOverflowMenuOpen(false);
                      setRelationsMenuOpen(true);
                    },
                    tooltip: t("FX_RELATED"),
                    className: "shrink-0 fx-pane-header-relations"
                  }
                ),
                relationsMenuOpen && /* @__PURE__ */ jsx(
                  Menu,
                  {
                    open: relationsMenuOpen,
                    opener: relationsButtonRef.current?.nativeElement ?? null,
                    onClose: () => setRelationsMenuOpen(false),
                    children: renderRelationMenuItems(() => setRelationsMenuOpen(false))
                  },
                  "relations-menu"
                )
              ] }),
              showOverflowButton && /* @__PURE__ */ jsxs(Fragment, { children: [
                /* @__PURE__ */ jsx(
                  Button,
                  {
                    ref: overflowButtonRef,
                    design: "SecondaryNeutral",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx(MoreIcon, { className: "h-4 w-4" }),
                    onClick: () => {
                      setRelationsMenuOpen(false);
                      setOverflowMenuOpen(true);
                    },
                    tooltip: t("FX_MORE_ACTIONS"),
                    className: "shrink-0 fx-pane-header-overflow"
                  }
                ),
                overflowMenuOpen && /* @__PURE__ */ jsxs(
                  Menu,
                  {
                    open: overflowMenuOpen,
                    opener: overflowButtonRef.current?.nativeElement ?? null,
                    onClose: () => setOverflowMenuOpen(false),
                    children: [
                      !overflowState.relationsVisible && hasRelated && renderOverflowRelationsSubmenu(() => setOverflowMenuOpen(false)),
                      !overflowState.relationsVisible && hasRelated && overflowedActions.length > 0 && /* @__PURE__ */ jsx(MenuSeparator, {}),
                      renderOverflowedActions(
                        () => setOverflowMenuOpen(false),
                        overflowState.relationsVisible || !hasRelated
                      )
                    ]
                  },
                  "overflow-menu"
                )
              ] }),
              showEndToggle && /* @__PURE__ */ jsx(
                ToggleButton,
                {
                  design: "SecondaryNeutral",
                  icon: /* @__PURE__ */ jsx(NewConversationIcon, { className: "h-4 w-4" }),
                  pressed: endPaneVisible,
                  onClick: () => handleToggleEnd(),
                  tooltip: endPaneVisible ? "Hide chat" : "Show chat",
                  className: "fx-pane-header-toggle-end"
                }
              ),
              showAddButton && !editMode && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "Secondary",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(Plus, { className: "h-4 w-4" }),
                  onClick: () => onAddClick?.(),
                  tooltip: addButtonTooltip || "Add",
                  className: "shrink-0 fx-pane-header-add"
                }
              ),
              handleClose && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "Tertiary",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(X, { className: "h-4 w-4" }),
                  onClick: () => handleClose?.(),
                  tooltip: t("FX_CLOSE"),
                  className: "shrink-0 fx-pane-header-close text-sapphire-text-primary"
                }
              )
            ] })
          ] }),
          centeringActive && !isSmallScreenEditMode && /* @__PURE__ */ jsxs(Fragment, { children: [
            /* @__PURE__ */ jsxs("div", { className: "fx-pane-header-left-zone", children: [
              showHamburger && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "Tertiary",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(MenuIcon, { className: "h-4 w-4" }),
                  onClick: () => handleHamburger(),
                  tooltip: t("FX_OPEN_NAVIGATION"),
                  className: "shrink-0 fx-pane-header-hamburger text-sapphire-text-primary"
                }
              ),
              showStartToggle && (toggleStartUsesBackIcon ? /* @__PURE__ */ jsx(
                Button,
                {
                  design: "SecondaryNeutral",
                  size: "Large",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(NavBackIcon, {}),
                  onClick: () => handleToggleStart(),
                  tooltip: t("FX_BACK_TO_LIST"),
                  className: "shrink-0 fx-pane-header-toggle-start"
                }
              ) : /* @__PURE__ */ jsx(
                ToggleButton,
                {
                  design: "SecondaryNeutral",
                  icon: /* @__PURE__ */ jsx(List, { className: "h-4 w-4" }),
                  pressed: startPaneVisible,
                  onClick: () => handleToggleStart(),
                  tooltip: startPaneVisible ? "Hide list" : "Show list",
                  className: "fx-pane-header-toggle-start"
                }
              )),
              showBackButton && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "SecondaryNeutral",
                  size: "Large",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(NavBackIcon, {}),
                  onClick: () => onBackClick?.(),
                  tooltip: t("FX_BACK"),
                  className: "shrink-0 fx-pane-header-back"
                }
              ),
              /* @__PURE__ */ jsx("div", { className: "flex flex-col min-w-0 shrink-0", children: /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 min-w-0", children: [
                breadcrumbs && !editMode && /* @__PURE__ */ jsx("div", { className: "min-w-0", children: renderBreadcrumbs() }),
                breadcrumbs && !editMode && title && /* @__PURE__ */ jsx("span", { className: "text-foreground font-semibold text-base select-none shrink-0 -mx-1", children: "/" }),
                titleEditable ? /* @__PURE__ */ jsx(
                  FxRenameTitle,
                  {
                    ref: renameTitleRef,
                    text: title,
                    editMode,
                    wrappingType: "Normal",
                    onEditAccept: handleTitleEditAccept,
                    onEditCancel: handleTitleEditCancel,
                    onEditModeChange: handleTitleEditModeChange,
                    className: "min-w-0"
                  }
                ) : /* @__PURE__ */ jsx(Heading, { className: "text-base font-semibold whitespace-pre-line min-w-0 flex items-center min-h-9", children: title }),
                showTitleArrow && !editMode && /* @__PURE__ */ jsx(
                  "button",
                  {
                    ref: titleArrowRef,
                    type: "button",
                    onClick: () => onTitleArrowClick?.(titleArrowRef.current),
                    className: "shrink-0 p-0.5 hover:bg-muted rounded",
                    children: /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4" })
                  }
                )
              ] }) }),
              /* @__PURE__ */ jsx("div", { className: "flex-1" }),
              actionsBeforeCentered.map((item) => {
                const originalIndex = actionItems.indexOf(item);
                return renderAction(item, originalIndex, overflowState.textVisibility[originalIndex]);
              })
            ] }),
            /* @__PURE__ */ jsx("div", { className: "fx-pane-header-center-zone", children: centeredAction && renderAction(
              centeredAction,
              centeredActionIndex,
              overflowState.textVisibility[centeredActionIndex]
            ) }),
            /* @__PURE__ */ jsxs("div", { className: "fx-pane-header-right-zone", children: [
              actionsAfterCentered.map((item) => {
                const originalIndex = actionItems.indexOf(item);
                return renderAction(item, originalIndex, overflowState.textVisibility[originalIndex]);
              }),
              overflowState.relationsVisible && hasRelated && /* @__PURE__ */ jsxs(Fragment, { children: [
                /* @__PURE__ */ jsx(
                  Button,
                  {
                    ref: relationsButtonRef,
                    design: "SecondaryNeutral",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx(Share2, { className: "h-4 w-4" }),
                    onClick: () => {
                      setOverflowMenuOpen(false);
                      setRelationsMenuOpen(true);
                    },
                    tooltip: t("FX_RELATED"),
                    className: "shrink-0 fx-pane-header-relations"
                  }
                ),
                relationsMenuOpen && /* @__PURE__ */ jsx(
                  Menu,
                  {
                    open: relationsMenuOpen,
                    opener: relationsButtonRef.current?.nativeElement ?? null,
                    onClose: () => setRelationsMenuOpen(false),
                    children: renderRelationMenuItems(() => setRelationsMenuOpen(false))
                  },
                  "relations-menu"
                )
              ] }),
              showOverflowButton && /* @__PURE__ */ jsxs(Fragment, { children: [
                /* @__PURE__ */ jsx(
                  Button,
                  {
                    ref: overflowButtonRef,
                    design: "SecondaryNeutral",
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsx(MoreIcon, { className: "h-4 w-4" }),
                    onClick: () => {
                      setRelationsMenuOpen(false);
                      setOverflowMenuOpen(true);
                    },
                    tooltip: t("FX_MORE_ACTIONS"),
                    className: "shrink-0 fx-pane-header-overflow"
                  }
                ),
                overflowMenuOpen && /* @__PURE__ */ jsxs(
                  Menu,
                  {
                    open: overflowMenuOpen,
                    opener: overflowButtonRef.current?.nativeElement ?? null,
                    onClose: () => setOverflowMenuOpen(false),
                    children: [
                      !overflowState.relationsVisible && hasRelated && renderOverflowRelationsSubmenu(() => setOverflowMenuOpen(false)),
                      !overflowState.relationsVisible && hasRelated && overflowedActions.length > 0 && /* @__PURE__ */ jsx(MenuSeparator, {}),
                      renderOverflowedActions(
                        () => setOverflowMenuOpen(false),
                        overflowState.relationsVisible || !hasRelated
                      )
                    ]
                  },
                  "overflow-menu"
                )
              ] }),
              showEndToggle && /* @__PURE__ */ jsx(
                ToggleButton,
                {
                  design: "SecondaryNeutral",
                  icon: /* @__PURE__ */ jsx(NewConversationIcon, { className: "h-4 w-4" }),
                  pressed: endPaneVisible,
                  onClick: () => handleToggleEnd(),
                  tooltip: endPaneVisible ? "Hide chat" : "Show chat",
                  className: "fx-pane-header-toggle-end"
                }
              ),
              showAddButton && !editMode && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "SecondaryNeutral",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(Plus, { className: "h-4 w-4" }),
                  onClick: () => onAddClick?.(),
                  tooltip: addButtonTooltip || "Add",
                  className: "shrink-0 fx-pane-header-add"
                }
              ),
              handleClose && /* @__PURE__ */ jsx(
                Button,
                {
                  design: "Tertiary",
                  iconOnly: true,
                  icon: /* @__PURE__ */ jsx(X, { className: "h-4 w-4" }),
                  onClick: () => handleClose?.(),
                  tooltip: t("FX_CLOSE"),
                  className: "shrink-0 fx-pane-header-close text-sapphire-text-primary"
                }
              )
            ] })
          ] })
        ] }),
        subheader && /* @__PURE__ */ jsx("div", { className: "w-full mt-2", children: subheader })
      ]
    }
  );
}
function FxPromptInput({
  value: controlledValue,
  placeholder = "Type a message...",
  disabled = false,
  statusMessage,
  disclaimerMessage,
  showDictate = true,
  showVoice = true,
  mode: explicitMode,
  typeaheadEnabled = true,
  maxSuggestions = 5,
  debounceMs = 150,
  actions,
  contextItems,
  toolbarButtons,
  onSubmit,
  onLiveChange,
  onPlusPress,
  onDictatePress,
  onVoicePress,
  onTypeaheadRequest,
  onActionSelect: _onActionSelect,
  onContextItemRemove,
  className,
  style,
  ref
}) {
  const { t } = useTranslation("fx");
  const inputRef = useRef(null);
  const textareaRef = useRef(null);
  const containerRef = useRef(null);
  const plusButtonRef = useRef(null);
  const [internalValue, setInternalValue] = useState("");
  const [actionsMenuOpen, setActionsMenuOpen] = useState(false);
  const [suggestions, setSuggestions] = useState([]);
  const [suggestionsVisible, setSuggestionsVisible] = useState(false);
  const [selectedIndex, setSelectedIndex] = useState(0);
  const [suggestionsDismissed, setSuggestionsDismissed] = useState(false);
  const [requestIdCounter, setRequestIdCounter] = useState(0);
  const layoutContext = useFxLayoutContext();
  const contextMode = layoutContext?.inputMode;
  const effectiveMode = explicitMode ?? contextMode ?? "oneline";
  const isMultiline = effectiveMode === "multiline";
  const value = controlledValue ?? internalValue;
  const hasValue = value.trim().length > 0;
  const visibleSuggestions = useMemo(
    () => suggestions.slice(0, maxSuggestions),
    [suggestions, maxSuggestions]
  );
  const currentCompletion = useMemo(() => {
    if (!suggestionsVisible || visibleSuggestions.length === 0) return "";
    const selected = visibleSuggestions[selectedIndex];
    return selected?.text ?? "";
  }, [suggestionsVisible, visibleSuggestions, selectedIndex]);
  useImperativeHandle(ref, () => ({
    focus: () => {
      if (isMultiline) {
        textareaRef.current?.focus();
      } else {
        inputRef.current?.focus();
      }
    },
    clear: () => {
      setInternalValue("");
      setSuggestions([]);
      setSuggestionsVisible(false);
      onLiveChange?.({ value: "" });
    },
    getValue: () => value,
    setValue: (v) => {
      setInternalValue(v);
      onLiveChange?.({ value: v });
    },
    get nativeElement() {
      return containerRef.current;
    },
    getNativeElement: () => containerRef.current,
    setSuggestions: (newSuggestions, requestId) => {
      if (requestId < requestIdCounter - 1) return;
      setSuggestions(newSuggestions);
      setSuggestionsVisible(newSuggestions.length > 0 && !suggestionsDismissed);
      setSelectedIndex(0);
    }
  }));
  const handleChange = useCallback(
    (newValue) => {
      setInternalValue(newValue);
      setSuggestionsDismissed(false);
      onLiveChange?.({ value: newValue });
    },
    [onLiveChange]
  );
  const handleSubmit = useCallback(() => {
    if (hasValue && !disabled) {
      onSubmit?.({ value: value.trim() });
      setInternalValue("");
      setSuggestions([]);
      setSuggestionsVisible(false);
    }
  }, [hasValue, value, disabled, onSubmit]);
  const acceptSuggestion = useCallback(() => {
    if (!suggestionsVisible || visibleSuggestions.length === 0) return false;
    const selected = visibleSuggestions[selectedIndex];
    if (!selected) return false;
    const newValue = value + selected.text;
    setInternalValue(newValue);
    onLiveChange?.({ value: newValue });
    setSuggestionsVisible(false);
    setSuggestions([]);
    return true;
  }, [suggestionsVisible, visibleSuggestions, selectedIndex, value, onLiveChange]);
  const handleKeyDown = useCallback(
    (e) => {
      if (e.key === "Tab" && suggestionsVisible && visibleSuggestions.length > 0) {
        e.preventDefault();
        acceptSuggestion();
        return;
      }
      if (suggestionsVisible && visibleSuggestions.length > 0) {
        if (e.key === "ArrowUp") {
          e.preventDefault();
          setSelectedIndex(
            (prev) => prev <= 0 ? visibleSuggestions.length - 1 : prev - 1
          );
          return;
        }
        if (e.key === "ArrowDown") {
          e.preventDefault();
          setSelectedIndex(
            (prev) => prev >= visibleSuggestions.length - 1 ? 0 : prev + 1
          );
          return;
        }
      }
      if (e.key === "Escape") {
        if (suggestionsVisible) {
          e.preventDefault();
          setSuggestionsVisible(false);
          setSuggestionsDismissed(true);
          return;
        }
      }
      if (e.key === "Enter" && (!isMultiline || !e.shiftKey)) {
        e.preventDefault();
        handleSubmit();
      }
    },
    [
      suggestionsVisible,
      visibleSuggestions,
      acceptSuggestion,
      isMultiline,
      handleSubmit
    ]
  );
  const handleSuggestionClick = useCallback(
    (index) => {
      setSelectedIndex(index);
      const selected = visibleSuggestions[index];
      if (!selected) return;
      const newValue = value + selected.text;
      setInternalValue(newValue);
      onLiveChange?.({ value: newValue });
      setSuggestionsVisible(false);
      setSuggestions([]);
    },
    [visibleSuggestions, value, onLiveChange]
  );
  useEffect(() => {
    if (!typeaheadEnabled || !value.trim() || !onTypeaheadRequest || suggestionsDismissed) {
      if (suggestions.length > 0) {
        setSuggestions([]);
      }
      if (suggestionsVisible) {
        setSuggestionsVisible(false);
      }
      return;
    }
    const controller = new AbortController();
    const currentRequestId = requestIdCounter + 1;
    setRequestIdCounter(currentRequestId);
    const timeoutId = setTimeout(async () => {
      try {
        const results = await onTypeaheadRequest({
          value,
          requestId: currentRequestId,
          signal: controller.signal
        });
        if (results && Array.isArray(results)) {
          setSuggestions(results);
          setSuggestionsVisible(results.length > 0);
          setSelectedIndex(0);
        }
      } catch {
      }
    }, debounceMs);
    return () => {
      clearTimeout(timeoutId);
      controller.abort();
    };
  }, [value, typeaheadEnabled, onTypeaheadRequest, debounceMs, suggestionsDismissed]);
  const renderLeftButtons = () => {
    if (!actions && !onPlusPress) return null;
    return /* @__PURE__ */ jsxs("div", { className: "flex shrink-0", children: [
      /* @__PURE__ */ jsx(
        Button,
        {
          ref: plusButtonRef,
          design: "Neutral",
          iconOnly: true,
          icon: /* @__PURE__ */ jsx(AddIcon, { className: "h-4 w-4" }),
          onClick: () => {
            if (actions) {
              setActionsMenuOpen(true);
            }
            onPlusPress?.();
          },
          disabled,
          tooltip: t("FX_ADD"),
          className: "h-7 w-7 bg-muted hover:border-sapphire-prompt-accent text-sapphire-prompt-accent"
        }
      ),
      actions && /* @__PURE__ */ jsx(
        Menu,
        {
          open: actionsMenuOpen,
          opener: plusButtonRef.current?.nativeElement,
          onClose: () => setActionsMenuOpen(false),
          placement: "top-start",
          children: actions
        }
      )
    ] });
  };
  const renderRightButtons = () => {
    return /* @__PURE__ */ jsx("div", { className: "flex shrink-0 gap-1", children: hasValue ? /* @__PURE__ */ jsx(
      Button,
      {
        design: "Neutral",
        iconOnly: true,
        icon: /* @__PURE__ */ jsx(PaperPlaneIcon, { className: "h-4 w-4" }),
        onClick: handleSubmit,
        disabled,
        tooltip: t("FX_SEND"),
        className: "h-7 w-7 bg-muted hover:border-sapphire-prompt-accent text-sapphire-prompt-accent"
      }
    ) : /* @__PURE__ */ jsxs(Fragment, { children: [
      showDictate && /* @__PURE__ */ jsx(
        Button,
        {
          design: "Tertiary",
          iconOnly: true,
          icon: /* @__PURE__ */ jsx(MicrophoneIcon, { className: "h-4 w-4" }),
          onClick: onDictatePress,
          disabled,
          tooltip: t("FX_DICTATE"),
          className: "h-7 w-7 text-sapphire-prompt-accent"
        }
      ),
      showVoice && /* @__PURE__ */ jsx(
        Button,
        {
          design: "Tertiary",
          iconOnly: true,
          icon: /* @__PURE__ */ jsx(AudioWaveform, { className: "h-4 w-4" }),
          onClick: onVoicePress,
          disabled,
          tooltip: t("FX_VOICE"),
          className: "h-7 w-7 text-sapphire-prompt-accent"
        }
      )
    ] }) });
  };
  const renderContextItems = () => {
    if (!contextItems || contextItems.length === 0) return null;
    return /* @__PURE__ */ jsx("div", { className: "flex items-center gap-1.5 shrink-0", children: contextItems.map((item) => /* @__PURE__ */ jsx(
      TagExploration,
      {
        design: "Neutral",
        size: "S",
        hideStateIcon: true,
        className: "text-sapphire-prompt-accent",
        icon: item.icon ? /* @__PURE__ */ jsx("span", { className: "flex items-center justify-center h-3.5 w-3.5", children: item.icon }) : void 0,
        onClose: item.removable !== false && onContextItemRemove ? () => onContextItemRemove({ item }) : void 0,
        children: item.label
      },
      item.id
    )) });
  };
  const renderSuggestions = () => {
    if (!suggestionsVisible || visibleSuggestions.length === 0) return null;
    return /* @__PURE__ */ jsxs("div", { className: "relative flex items-end mb-2 min-h-10", children: [
      !isMultiline && /* @__PURE__ */ jsx(
        "div",
        {
          className: "invisible whitespace-pre text-sm",
          style: { paddingLeft: "3.5rem" },
          children: value
        }
      ),
      /* @__PURE__ */ jsx(
        "div",
        {
          className: "flex flex-col-reverse gap-1.5",
          style: { marginLeft: isMultiline ? 0 : "-1.2rem", maxWidth: "calc(100% - 4rem)" },
          children: visibleSuggestions.map((suggestion, index) => /* @__PURE__ */ jsx(
            "button",
            {
              type: "button",
              className: cn(
                "inline-flex items-center px-3 py-1.5 text-sm rounded-lg cursor-pointer transition-all",
                "whitespace-nowrap max-w-[200px] overflow-hidden text-ellipsis",
                "border bg-background"
              ),
              style: {
                borderColor: index === selectedIndex ? "var(--prompt-accent-muted)" : "color-mix(in srgb, var(--prompt-accent-muted) 50%, transparent)",
                boxShadow: index === selectedIndex ? "0 0 0 2px color-mix(in srgb, var(--prompt-accent-subtle) 50%, transparent)" : void 0
              },
              onMouseDown: (e) => {
                e.preventDefault();
                handleSuggestionClick(index);
              },
              onMouseEnter: () => setSelectedIndex(index),
              children: suggestion.label ?? suggestion.text
            },
            index
          ))
        }
      )
    ] });
  };
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: containerRef,
      className: cn(
        "w-full relative",
        disabled && "opacity-60 pointer-events-none",
        className
      ),
      style,
      children: [
        statusMessage && /* @__PURE__ */ jsx(
          "div",
          {
            className: "px-4 pt-2 pb-5 text-sm rounded-t-lg relative z-[1] -mb-3 animate-in fade-in duration-500 bg-sapphire-info-bg text-sapphire-text-tertiary",
            children: statusMessage
          }
        ),
        renderSuggestions(),
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: "fx-prompt-input relative rounded-lg bg-card",
            style: {
              // Shadow on left, right, bottom - most pronounced at bottom
              boxShadow: `
              -3px 0 8px -3px color-mix(in srgb, var(--prompt-accent-subtle) 20%, transparent),
              3px 0 8px -3px color-mix(in srgb, var(--prompt-accent-subtle) 20%, transparent),
              0 4px 12px -2px color-mix(in srgb, var(--prompt-accent-muted) 30%, transparent)
            `
            },
            children: [
              /* @__PURE__ */ jsx(
                "div",
                {
                  className: "absolute inset-0 rounded-lg pointer-events-none",
                  style: {
                    padding: "1.5px",
                    background: "linear-gradient(to right, color-mix(in srgb, var(--color-brand-blue) 60%, transparent), color-mix(in srgb, var(--color-brand-purple) 60%, transparent))",
                    WebkitMask: "linear-gradient(#fff 0 0) content-box, linear-gradient(#fff 0 0)",
                    WebkitMaskComposite: "xor",
                    maskComposite: "exclude"
                  }
                }
              ),
              isMultiline ? (
                // Multiline mode: textarea with toolbar below
                /* @__PURE__ */ jsxs("div", { className: "flex flex-col p-3", children: [
                  /* @__PURE__ */ jsx("div", { className: "flex-1 min-w-0 mb-2", children: /* @__PURE__ */ jsx(
                    "textarea",
                    {
                      ref: textareaRef,
                      value,
                      onChange: (e) => handleChange(e.target.value),
                      onKeyDown: handleKeyDown,
                      placeholder,
                      disabled,
                      rows: 1,
                      className: cn(
                        "w-full bg-transparent border-none outline-none resize-none text-base min-h-[2rem]",
                        "placeholder:text-sapphire-text-tertiary placeholder:italic placeholder:font-normal focus:ring-0 focus:outline-none"
                      ),
                      style: {
                        boxShadow: "none"
                      }
                    }
                  ) }),
                  /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-1.5", children: [
                    renderLeftButtons(),
                    renderContextItems(),
                    toolbarButtons,
                    /* @__PURE__ */ jsx("div", { className: "flex-1" }),
                    " ",
                    renderRightButtons()
                  ] })
                ] })
              ) : (
                // Oneline mode: two rows - input on top, buttons below
                /* @__PURE__ */ jsxs("div", { className: "flex flex-col p-3", children: [
                  /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0 relative mb-2", children: [
                    currentCompletion && /* @__PURE__ */ jsxs(
                      "div",
                      {
                        className: "absolute top-0 left-0 right-0 bottom-0 pointer-events-none flex items-center overflow-hidden text-sm",
                        children: [
                          /* @__PURE__ */ jsx("span", { className: "invisible whitespace-pre", children: value }),
                          /* @__PURE__ */ jsx("span", { className: "text-sapphire-text-tertiary whitespace-pre overflow-hidden text-ellipsis", children: currentCompletion })
                        ]
                      }
                    ),
                    /* @__PURE__ */ jsx(
                      "input",
                      {
                        ref: inputRef,
                        type: "text",
                        value,
                        onChange: (e) => handleChange(e.target.value),
                        onKeyDown: handleKeyDown,
                        placeholder,
                        disabled,
                        className: cn(
                          "w-full bg-transparent border-none outline-none text-base h-6",
                          "placeholder:text-sapphire-text-tertiary placeholder:italic placeholder:font-normal"
                        )
                      }
                    )
                  ] }),
                  /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-1.5", children: [
                    renderLeftButtons(),
                    renderContextItems(),
                    toolbarButtons,
                    /* @__PURE__ */ jsx("div", { className: "flex-1" }),
                    " ",
                    renderRightButtons()
                  ] })
                ] })
              )
            ]
          }
        ),
        disclaimerMessage && /* @__PURE__ */ jsx(
          "div",
          {
            className: "text-center text-xs leading-4 mt-4 text-secondary-foreground",
            style: { letterSpacing: "0.4px" },
            children: disclaimerMessage
          }
        )
      ]
    }
  );
}
function FxUserMenuContent({
  selectedAccount,
  otherAccounts,
  showManageAccount,
  showOtherAccounts,
  showEditAccounts,
  showEditButton,
  avatarInteractive = true,
  children,
  titleRef,
  onAvatarClick,
  onManageAccountClick,
  onEditAccountsClick,
  onAccountSwitch,
  ref
}) {
  const { t } = useTranslation("fx");
  return /* @__PURE__ */ jsxs("div", { ref, className: "px-2 pb-2", children: [
    selectedAccount && /* @__PURE__ */ jsxs(
      "div",
      {
        className: "flex flex-col items-center mb-2",
        "aria-label": t("FX_CURRENT_ACCOUNT"),
        children: [
          /* @__PURE__ */ jsx("div", { className: "relative mt-1 mb-2", children: /* @__PURE__ */ jsx(
            Avatar,
            {
              size: "L" /* L */,
              image: selectedAccount.avatarSrc,
              initials: selectedAccount.avatarInitials,
              colorScheme: selectedAccount.avatarColorScheme || "Accent6" /* Accent6 */,
              icon: /* @__PURE__ */ jsx(PersonPlaceholderIcon, { className: "h-6 w-6" }),
              interactive: avatarInteractive,
              onClick: avatarInteractive ? () => onAvatarClick?.() : void 0,
              accessibleName: selectedAccount.titleText,
              badge: showEditButton ? /* @__PURE__ */ jsx(
                "span",
                {
                  className: "flex items-center justify-center h-5 w-5 rounded-full",
                  style: { backgroundColor: "var(--brand-background)", color: "var(--neutral-foreground-white)" },
                  children: /* @__PURE__ */ jsx(EditIcon, { className: "h-3 w-3" })
                }
              ) : void 0
            }
          ) }),
          selectedAccount.titleText && /* @__PURE__ */ jsx(
            "span",
            {
              ref: titleRef,
              className: "text-center text-base font-semibold text-foreground mt-1",
              children: selectedAccount.titleText
            }
          ),
          selectedAccount.subtitleText && /* @__PURE__ */ jsx("span", { className: "text-center text-sm text-sapphire-text-tertiary mb-0.5", children: selectedAccount.subtitleText }),
          selectedAccount.description && /* @__PURE__ */ jsx("span", { className: "text-center text-sm text-sapphire-text-tertiary", children: selectedAccount.description }),
          selectedAccount.additionalInfo && /* @__PURE__ */ jsx("span", { className: "text-center text-sm text-sapphire-text-tertiary mt-1", children: selectedAccount.additionalInfo }),
          showManageAccount && /* @__PURE__ */ jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              icon: /* @__PURE__ */ jsx(SettingsIcon, { className: "h-4 w-4" }),
              className: "mt-3 font-semibold",
              onClick: () => onManageAccountClick?.(),
              children: "Manage Account"
            }
          )
        ]
      }
    ),
    showOtherAccounts && otherAccounts.length > 0 && /* @__PURE__ */ jsx(
      Panel,
      {
        headerText: `Other Accounts (${otherAccounts.length})`,
        headerLevel: "H4" /* H4 */,
        defaultCollapsed: true,
        noPadding: true,
        className: "mb-2",
        header: /* @__PURE__ */ jsxs("div", { className: "flex flex-1 items-center justify-between min-w-0", children: [
          /* @__PURE__ */ jsxs("span", { className: "text-sm font-semibold truncate", children: [
            "Other Accounts (",
            otherAccounts.length,
            ")"
          ] }),
          showEditAccounts && /* @__PURE__ */ jsx(
            Button,
            {
              design: "Tertiary" /* Tertiary */,
              icon: /* @__PURE__ */ jsx(UserEditIcon, { className: "h-4 w-4" }),
              iconOnly: true,
              accessibleName: t("FX_EDIT_ACCOUNTS"),
              tooltip: t("FX_EDIT_ACCOUNTS"),
              onClick: (detail) => {
                detail.originalEvent.stopPropagation();
                onEditAccountsClick?.();
              },
              className: "ml-2"
            }
          )
        ] }),
        children: /* @__PURE__ */ jsx(
          List$1,
          {
            accessibleName: `Other Accounts (${otherAccounts.length})`,
            separators: "All" /* All */,
            onItemClick: ({ item }) => {
              const accountId = item.getAttribute("data-item-key");
              const account = otherAccounts.find((a) => a.id === accountId);
              if (account) {
                onAccountSwitch(account);
              }
            },
            children: otherAccounts.map((account) => /* @__PURE__ */ jsx(
              ListItemCustom,
              {
                itemKey: account.id,
                accessibleName: `${account.titleText} ${account.subtitleText || ""} ${account.description || ""}`,
                children: /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-3 w-full", children: [
                  /* @__PURE__ */ jsx(
                    Avatar,
                    {
                      size: "S" /* S */,
                      image: account.avatarSrc,
                      initials: account.avatarInitials,
                      colorScheme: account.avatarColorScheme || "Accent6" /* Accent6 */,
                      icon: /* @__PURE__ */ jsx(PersonPlaceholderIcon, { className: "h-4 w-4" }),
                      accessibleName: account.titleText
                    }
                  ),
                  /* @__PURE__ */ jsxs("div", { className: "flex flex-col flex-1 min-w-0 gap-0.5", children: [
                    account.titleText && /* @__PURE__ */ jsx("span", { className: "text-sm font-semibold text-foreground truncate", children: account.titleText }),
                    account.subtitleText && /* @__PURE__ */ jsx(Label, { className: "text-xs truncate", children: account.subtitleText }),
                    account.description && /* @__PURE__ */ jsx(Label, { className: "text-xs truncate", children: account.description })
                  ] }),
                  account.selected && /* @__PURE__ */ jsx(AcceptIcon, { className: "h-4 w-4 shrink-0 text-sapphire-text-tertiary" }),
                  account.loading && /* @__PURE__ */ jsx("div", { className: "h-4 w-4 shrink-0 animate-spin rounded-full border-2 border-sapphire-text-tertiary border-t-transparent" })
                ] })
              },
              account.id
            ))
          }
        )
      }
    ),
    React5.Children.count(children) > 0 && /* @__PURE__ */ jsx(
      List$1,
      {
        accessibleName: "Actions",
        separators: "None" /* None */,
        children
      }
    )
  ] });
}
var NavBackIcon2 = () => /* @__PURE__ */ jsx("svg", { xmlns: "http://www.w3.org/2000/svg", width: "20", height: "20", viewBox: "0 0 20 20", fill: "none", children: /* @__PURE__ */ jsx("path", { d: "M13.4939 4.04418C13.1385 3.66784 12.5459 3.65098 12.1694 4.00634L6.54443 9.31517C6.357 9.4921 6.25047 9.7386 6.25024 9.99633C6.25018 10.2543 6.35696 10.5015 6.54443 10.6787L12.1694 15.9936C12.5458 16.3491 13.1383 16.3321 13.4939 15.9558C13.8493 15.5795 13.8323 14.9869 13.4561 14.6313L8.55371 9.99633L13.4561 5.36864C13.8324 5.01321 13.8492 4.42064 13.4939 4.04418Z", fill: "currentColor" }) });
var animationsInjected = false;
var injectAnimations = () => {
  if (animationsInjected) return;
  animationsInjected = true;
  const style = document.createElement("style");
  style.id = "fx-user-menu-panel-animations";
  style.textContent = `
    @keyframes fxUmFadeIn {
      from { opacity: 0; }
      to { opacity: 1; }
    }
    @keyframes fxUmFadeOut {
      from { opacity: 1; }
      to { opacity: 0; }
    }
    @keyframes fxUmSlideInFromLeft {
      from { transform: translateX(-100%); }
      to { transform: translateX(0); }
    }
    @keyframes fxUmSlideOutToLeft {
      from { transform: translateX(0); }
      to { transform: translateX(-100%); }
    }
  `;
  document.head.appendChild(style);
};
var ANIMATION_DURATION = 300;
function FxUserMenuPanel({
  open,
  children,
  footer,
  accessibleName,
  onClose,
  onBackClick,
  ref
}) {
  const [visible, setVisible] = useState(false);
  const [closing, setClosing] = useState(false);
  const backButtonRef = useRef(null);
  const previousFocusRef = useRef(null);
  const closeTimerRef = useRef(null);
  const visibleRef = useRef(false);
  useEffect(() => {
    injectAnimations();
  }, []);
  useEffect(() => {
    visibleRef.current = visible;
  }, [visible]);
  useEffect(() => {
    if (closeTimerRef.current) {
      clearTimeout(closeTimerRef.current);
      closeTimerRef.current = null;
    }
    if (open) {
      previousFocusRef.current = document.activeElement;
      setVisible(true);
      setClosing(false);
    } else if (visibleRef.current) {
      setClosing(true);
      closeTimerRef.current = setTimeout(() => {
        setVisible(false);
        setClosing(false);
        closeTimerRef.current = null;
      }, ANIMATION_DURATION);
    }
    return () => {
      if (closeTimerRef.current) {
        clearTimeout(closeTimerRef.current);
        closeTimerRef.current = null;
      }
    };
  }, [open]);
  useEffect(() => {
    if (visible && !closing) {
      requestAnimationFrame(() => {
        backButtonRef.current?.focus();
      });
    }
  }, [visible, closing]);
  useEffect(() => {
    if (!visible && previousFocusRef.current) {
      previousFocusRef.current?.focus();
      previousFocusRef.current = null;
    }
  }, [visible]);
  const startClose = useCallback(
    (callback) => {
      if (closing) return;
      setClosing(true);
      setTimeout(() => {
        setVisible(false);
        setClosing(false);
        onClose();
        callback?.();
      }, ANIMATION_DURATION);
    },
    [closing, onClose]
  );
  const handleBackdropClick = useCallback(() => {
    startClose();
  }, [startClose]);
  const handleBackClick = useCallback(() => {
    startClose(onBackClick);
  }, [startClose, onBackClick]);
  useEffect(() => {
    if (!visible) return;
    const handleKeyDown = (e) => {
      if (e.key === "Escape") {
        e.stopPropagation();
        startClose();
      }
    };
    document.addEventListener("keydown", handleKeyDown);
    return () => document.removeEventListener("keydown", handleKeyDown);
  }, [visible, startClose]);
  useEffect(() => {
    if (visible) {
      const prevOverflow = document.body.style.overflow;
      document.body.style.overflow = "hidden";
      return () => {
        document.body.style.overflow = prevOverflow;
      };
    }
  }, [visible]);
  if (!visible) return null;
  return createPortal(
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref,
        className: "fixed inset-0 z-50 flex",
        role: "dialog",
        "aria-modal": "true",
        "aria-label": accessibleName || "User Menu",
        children: [
          /* @__PURE__ */ jsx(
            "div",
            {
              className: "absolute inset-0 bg-black/50",
              style: {
                animation: closing ? `fxUmFadeOut ${ANIMATION_DURATION}ms cubic-bezier(0.4, 0, 0.2, 1) forwards` : `fxUmFadeIn ${ANIMATION_DURATION}ms cubic-bezier(0.4, 0, 0.2, 1)`
              },
              onClick: handleBackdropClick
            }
          ),
          /* @__PURE__ */ jsxs(
            "div",
            {
              className: "relative h-full flex flex-col shadow-xl rounded-r-2xl",
              style: {
                minWidth: "20rem",
                maxWidth: "calc(100vw - 2rem)",
                backgroundColor: "var(--card-bg-primary)",
                borderRight: "1px solid var(--border-primary)",
                animation: closing ? `fxUmSlideOutToLeft ${ANIMATION_DURATION}ms cubic-bezier(0.4, 0, 0.2, 1) forwards` : `fxUmSlideInFromLeft ${ANIMATION_DURATION}ms cubic-bezier(0.4, 0, 0.2, 1)`
              },
              onClick: (e) => e.stopPropagation(),
              children: [
                /* @__PURE__ */ jsxs(
                  "div",
                  {
                    className: "flex items-center h-12 px-2 shrink-0",
                    style: { borderBottom: "1px solid var(--border-primary)" },
                    children: [
                      /* @__PURE__ */ jsx(
                        Button,
                        {
                          ref: backButtonRef,
                          design: "SecondaryNeutral",
                          size: "Large",
                          iconOnly: true,
                          icon: /* @__PURE__ */ jsx(NavBackIcon2, {}),
                          accessibleName: "Back",
                          onClick: handleBackClick
                        }
                      ),
                      /* @__PURE__ */ jsx(Title, { level: "H4" /* H4 */, className: "flex-1 text-center pr-8", children: "Profile" })
                    ]
                  }
                ),
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    className: "flex-1 overflow-y-auto overflow-x-hidden",
                    style: { overscrollBehavior: "contain" },
                    children
                  }
                ),
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    className: "shrink-0 px-3 py-2",
                    style: { borderTop: "1px solid var(--border-primary)" },
                    children: footer
                  }
                )
              ]
            }
          )
        ]
      }
    ),
    document.body
  );
}
var FxUserMenuContext = createContext({
  closeMenu: () => {
  }
});
var FxUserMenuItemGroupContext = createContext(null);
function FxUserMenu({
  open = false,
  opener,
  useDialog = false,
  showManageAccount = false,
  showOtherAccounts = false,
  showEditAccounts = false,
  showEditButton = false,
  avatarInteractive = true,
  accounts = [],
  children,
  footer,
  onAvatarClick,
  onManageAccountClick,
  onEditAccountsClick,
  onSignOutClick,
  onOpen,
  onClose,
  onBackClick,
  onChangeAccount,
  onItemClick,
  placement = "Bottom",
  horizontalAlign = "End",
  verticalAlign,
  offset,
  className,
  accessibleName,
  ref
}) {
  const popoverRef = useRef(null);
  const { t } = useTranslation("fx");
  const anchorRef = useRef(null);
  const titleRef = useRef(null);
  const [titleMovedToHeader, setTitleMovedToHeader] = useState(false);
  const [isScrolled, setIsScrolled] = useState(false);
  const [popoverReady, setPopoverReady] = useState(false);
  const selectedAccount = accounts.find((a) => a.selected) || accounts[0];
  const otherAccounts = accounts.filter((a) => a !== selectedAccount);
  useImperativeHandle(ref, () => ({
    close: () => onClose?.(),
    isOpen: () => open
  }));
  const closeMenu = useCallback(() => {
    onClose?.();
  }, [onClose]);
  const handleOpen = useCallback(() => {
    setPopoverReady(true);
    onOpen?.();
  }, [onOpen]);
  const handleClose = useCallback(() => {
    setPopoverReady(false);
    setTitleMovedToHeader(false);
    setIsScrolled(false);
    onClose?.();
  }, [onClose]);
  useEffect(() => {
    if (useDialog || !popoverReady) return;
    const scrollContainer = anchorRef.current?.parentElement;
    if (!scrollContainer || !titleRef.current) return;
    const handleScroll = () => {
      setIsScrolled(scrollContainer.scrollTop > 0);
    };
    scrollContainer.addEventListener("scroll", handleScroll);
    const observer = new IntersectionObserver(
      (entries) => {
        entries.forEach((entry) => {
          if (entry.target === titleRef.current) {
            setTitleMovedToHeader(!entry.isIntersecting);
          }
        });
      },
      {
        root: scrollContainer,
        threshold: [0.15]
      }
    );
    observer.observe(titleRef.current);
    return () => {
      scrollContainer.removeEventListener("scroll", handleScroll);
      observer.disconnect();
    };
  }, [useDialog, popoverReady, selectedAccount]);
  const handleAccountSwitch = useCallback(
    (account) => {
      if (!selectedAccount || account.id === selectedAccount.id) return;
      const result = onChangeAccount?.({
        prevSelectedAccount: selectedAccount,
        selectedAccount: account
      });
      if (result === false) return;
    },
    [selectedAccount, onChangeAccount]
  );
  const handleSignOut = useCallback(() => {
    const result = onSignOutClick?.();
    if (result === false) return;
    closeMenu();
  }, [onSignOutClick, closeMenu]);
  const handleItemClick = useCallback(
    (detail) => {
      const result = onItemClick?.(detail);
      if (result === false) return;
      closeMenu();
    },
    [onItemClick, closeMenu]
  );
  const hasCustomFooter = footer !== void 0 && footer !== null;
  const accessibleNameText = accessibleName || (selectedAccount ? `User Menu ${selectedAccount.titleText}` : "User Menu");
  const footerContent = hasCustomFooter ? /* @__PURE__ */ jsx("div", { className: "flex justify-end items-center w-full", children: footer }) : /* @__PURE__ */ jsx("div", { className: "flex justify-end items-center w-full", children: /* @__PURE__ */ jsx(
    Button,
    {
      design: "Tertiary" /* Tertiary */,
      icon: /* @__PURE__ */ jsx(LogOut, { className: "h-4 w-4" }),
      className: "font-semibold",
      onClick: handleSignOut,
      children: t("SIGN_OUT")
    }
  ) });
  const menuContent = /* @__PURE__ */ jsx(
    FxUserMenuContent,
    {
      selectedAccount,
      otherAccounts,
      showManageAccount,
      showOtherAccounts,
      showEditAccounts,
      showEditButton,
      avatarInteractive,
      titleRef,
      onAvatarClick,
      onManageAccountClick,
      onEditAccountsClick,
      onAccountSwitch: handleAccountSwitch,
      children
    }
  );
  if (useDialog) {
    return /* @__PURE__ */ jsx(FxUserMenuContext.Provider, { value: { closeMenu, onItemClick: handleItemClick }, children: /* @__PURE__ */ jsx(
      FxUserMenuPanel,
      {
        open,
        footer: footerContent,
        accessibleName: accessibleNameText,
        onClose: closeMenu,
        onBackClick,
        children: menuContent
      }
    ) });
  }
  const headerContent = /* @__PURE__ */ jsx(
    "div",
    {
      className: cn(
        "flex items-center px-3 min-h-[2.75rem]",
        isScrolled || titleMovedToHeader ? "border-b border-border shadow-sm" : ""
      ),
      children: /* @__PURE__ */ jsx("div", { className: "flex-1 min-w-0", children: titleMovedToHeader && selectedAccount && /* @__PURE__ */ jsx(Title, { level: "H4" /* H4 */, className: "truncate", children: selectedAccount.titleText }) })
    }
  );
  return /* @__PURE__ */ jsx(FxUserMenuContext.Provider, { value: { closeMenu, onItemClick: handleItemClick }, children: /* @__PURE__ */ jsxs(
    ResponsivePopover,
    {
      ref: popoverRef,
      opener,
      open,
      placement,
      horizontalAlign,
      verticalAlign,
      offset,
      noPadding: true,
      accessibleName: accessibleNameText,
      header: headerContent,
      footer: footerContent,
      onOpen: handleOpen,
      onClose: handleClose,
      showCloseButton: true,
      className: cn("w-80", className),
      children: [
        /* @__PURE__ */ jsx("div", { ref: anchorRef, className: "contents" }),
        menuContent
      ]
    }
  ) });
}
function FxUserMenuItem({
  text,
  icon,
  checked,
  disabled = false,
  children: _children,
  onClick,
  className,
  ref
}) {
  const { onItemClick } = useContext(FxUserMenuContext);
  const groupContext = useContext(FxUserMenuItemGroupContext);
  const itemId = useRef(Math.random().toString(36).slice(2)).current;
  const isChecked = groupContext ? groupContext.checkedItems.has(itemId) : checked;
  const handleClick = useCallback(
    (e) => {
      if (disabled) return;
      if (groupContext && groupContext.checkMode !== "None") {
        groupContext.onCheck(itemId);
      }
      if (onClick) {
        onClick(e);
      }
      const result = onItemClick?.({
        item: e.currentTarget,
        text: text || ""
      });
      if (result === false) return;
    },
    [disabled, onClick, onItemClick, text, groupContext, itemId]
  );
  return /* @__PURE__ */ jsx(
    ListItem,
    {
      ref,
      itemKey: itemId,
      text,
      icon,
      disabled,
      additionalText: isChecked ? "\u2713" : void 0,
      onClick: handleClick,
      className: cn(
        "rounded-md",
        className
      )
    }
  );
}
function FxUserMenuItemGroup({
  checkMode = "None",
  children,
  className,
  ref
}) {
  const [checkedItems, setCheckedItems] = useState(/* @__PURE__ */ new Set());
  const handleCheck = useCallback(
    (itemId) => {
      setCheckedItems((prev) => {
        const next = new Set(prev);
        if (checkMode === "Single") {
          next.clear();
          next.add(itemId);
        } else if (checkMode === "Multiple") {
          if (next.has(itemId)) {
            next.delete(itemId);
          } else {
            next.add(itemId);
          }
        }
        return next;
      });
    },
    [checkMode]
  );
  return /* @__PURE__ */ jsx(
    FxUserMenuItemGroupContext.Provider,
    {
      value: {
        checkMode,
        checkedItems,
        onCheck: handleCheck
      },
      children: /* @__PURE__ */ jsx(
        "div",
        {
          ref,
          role: "group",
          className: cn("py-0.5", className),
          children
        }
      )
    }
  );
}

export { BellIcon, CloseCommandFieldIcon, CodeIcon, ConversationsIcon, CursorIcon, DiscoverIcon, FxLayout, FxLayoutContext, FxPaneHeader, FxPaneHeaderAction, FxPaneHeaderSegmentedAction, FxPaneHeaderSegmentedOption, FxPaneHeaderSplitAction, FxPaneHeaderSplitOption, FxPromptInput, FxUserMenu, FxUserMenuItem, FxUserMenuItemGroup, JobsIcon, JouleIcon, MenuIcon, OpenCommandFieldIcon, PanelCloseIcon, PanelOpenIcon, SpacesIcon, VSCodeIcon, WaveformIcon, useFxLayoutContext };
