// @sap-ui/fx-components/dist/chunk-YJEMXAX6.js
import { cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useState, useEffect, useCallback } from 'react';
import { jsx } from 'react/jsx-runtime';

// src/types/toast.ts
var ToastPlacement = /* @__PURE__ */ ((ToastPlacement2) => {
  ToastPlacement2["TopStart"] = "TopStart";
  ToastPlacement2["TopCenter"] = "TopCenter";
  ToastPlacement2["TopEnd"] = "TopEnd";
  ToastPlacement2["MiddleStart"] = "MiddleStart";
  ToastPlacement2["MiddleCenter"] = "MiddleCenter";
  ToastPlacement2["MiddleEnd"] = "MiddleEnd";
  ToastPlacement2["BottomStart"] = "BottomStart";
  ToastPlacement2["BottomCenter"] = "BottomCenter";
  ToastPlacement2["BottomEnd"] = "BottomEnd";
  return ToastPlacement2;
})(ToastPlacement || {});
var MIN_DURATION = 500;
var MAX_TRANSITION = 1e3;
var placementStyles = {
  ["TopStart" /* TopStart */]: { top: "3rem", left: "2rem" },
  ["TopCenter" /* TopCenter */]: { top: "3rem", left: "50%", transform: "translateX(-50%)" },
  ["TopEnd" /* TopEnd */]: { top: "3rem", right: "2rem" },
  ["MiddleStart" /* MiddleStart */]: { top: "50%", left: "2rem", transform: "translateY(-50%)" },
  ["MiddleCenter" /* MiddleCenter */]: { top: "50%", left: "50%", transform: "translate(-50%, -50%)" },
  ["MiddleEnd" /* MiddleEnd */]: { top: "50%", right: "2rem", transform: "translateY(-50%)" },
  ["BottomStart" /* BottomStart */]: { bottom: "3rem", left: "2rem" },
  ["BottomCenter" /* BottomCenter */]: { bottom: "3rem", left: "50%", transform: "translateX(-50%)" },
  ["BottomEnd" /* BottomEnd */]: { bottom: "3rem", right: "2rem" }
};
function Toast({
  open = false,
  duration = 3e3,
  placement = "BottomCenter" /* BottomCenter */,
  children,
  onClose,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const toastRef = useRef(null);
  const [hovered, setHovered] = useState(false);
  const [focused, setFocused] = useState(false);
  const [fading, setFading] = useState(false);
  React.useImperativeHandle(ref, () => toastRef.current, []);
  const effectiveDuration = Math.max(duration, MIN_DURATION);
  const transitionDuration = Math.min(effectiveDuration / 3, MAX_TRANSITION);
  useEffect(() => {
    const el = toastRef.current;
    if (!el) return;
    if (open) {
      if (!el.matches(":popover-open")) {
        el.showPopover();
      }
      requestAnimationFrame(() => {
        setFading(true);
      });
    } else {
      setFading(false);
      if (el.matches(":popover-open")) {
        el.hidePopover();
      }
    }
  }, [open]);
  const handleTransitionEnd = useCallback(() => {
    if (hovered || focused) return;
    setFading(false);
    onClose?.();
  }, [hovered, focused, onClose]);
  const paused = hovered || focused;
  useEffect(() => {
    if (!open) return;
    const handleKeydown = (e) => {
      const isCtrl = e.metaKey || !navigator.userAgent.includes("Mac") && e.ctrlKey;
      if (isCtrl && e.shiftKey && e.key.toLowerCase() === "m") {
        e.preventDefault();
        if (focused) {
          setFocused(false);
        } else {
          toastRef.current?.focus();
        }
      }
    };
    document.addEventListener("keydown", handleKeydown);
    return () => document.removeEventListener("keydown", handleKeydown);
  }, [open, focused]);
  const handleKeyDown = useCallback((e) => {
    if (e.key === "Escape") {
      setFocused(false);
    }
  }, []);
  const posStyles = placementStyles[placement] || placementStyles["BottomCenter" /* BottomCenter */];
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref: toastRef,
      id,
      popover: "manual",
      role: "alert",
      tabIndex: focused ? 0 : -1,
      "data-testid": dataTestId,
      className: cn(
        // Visual style
        "max-w-60 rounded p-4",
        "bg-foreground dark:bg-neutral-300 text-background text-sm text-center",
        "shadow-lg",
        "break-words whitespace-pre-line",
        // Focus ring
        focused && "outline-2 outline-offset-[-3px] outline-[var(--ring)]",
        className
      ),
      style: {
        // Reset popover UA defaults — must be inline to win specificity
        position: "fixed",
        inset: "unset",
        margin: 0,
        padding: void 0,
        border: "none",
        background: void 0,
        // Placement positioning
        ...posStyles,
        // Fade transition
        transitionProperty: "opacity",
        transitionDuration: fading ? `${transitionDuration}ms` : "0ms",
        transitionDelay: fading ? `${effectiveDuration - transitionDuration}ms` : "0ms",
        opacity: fading && !paused ? 0 : 1,
        ...style
      },
      onTransitionEnd: handleTransitionEnd,
      onMouseEnter: () => setHovered(true),
      onMouseLeave: () => setHovered(false),
      onFocusCapture: () => setFocused(true),
      onBlurCapture: () => setFocused(false),
      onKeyDown: handleKeyDown,
      children
    }
  );
}
Toast.displayName = "Toast";

export { Toast, ToastPlacement };
