// @sap-ui/fx-components/dist/chunk-KDP347JV.js
import { Popover } from './chunk-ZDBE5GTH.js';
import { Dialog } from './chunk-4C7TQYID.js';
import { isPhone } from './chunk-YSPWI6FA.js';
import { Title } from './chunk-BM2JFIU4.js';
import { Button } from './chunk-T7M6DGUN.js';
import { useRef, useCallback, useImperativeHandle, useState } from 'react';
import { jsx, jsxs } from 'react/jsx-runtime';

function useIsPhone() {
  const [phone] = useState(() => isPhone());
  return phone;
}
function ResponsivePopover({
  headerText,
  placement,
  horizontalAlign,
  verticalAlign,
  hideArrow,
  offset,
  allowTargetOverlap,
  opener,
  open = false,
  initialFocus,
  preventFocusRestore,
  preventInitialFocus,
  accessibleName,
  accessibleNameRef,
  accessibleRole,
  accessibleDescription,
  header,
  footer,
  children,
  onBeforeOpen,
  onOpen,
  onBeforeClose,
  onClose,
  contentOnlyOnDesktop = false,
  noPadding = false,
  showCloseButton = true,
  className,
  style,
  id,
  ref
}) {
  const isPhone2 = useIsPhone();
  const popoverRef = useRef(null);
  const dialogRef = useRef(null);
  const handleMobileClose = useCallback(() => {
    onBeforeClose?.({ escPressed: false });
    onClose?.();
  }, [onBeforeClose, onClose]);
  useImperativeHandle(
    ref,
    () => ({
      open() {
        if (isPhone2) {
          dialogRef.current?.show();
        } else {
          popoverRef.current?.open();
        }
      },
      close() {
        if (isPhone2) {
          dialogRef.current?.close();
        } else {
          popoverRef.current?.close();
        }
      },
      async applyFocus() {
        if (isPhone2) {
          dialogRef.current?.focus();
        } else {
          await popoverRef.current?.applyFocus();
        }
      },
      isOpen() {
        if (isPhone2) {
          return open;
        }
        return popoverRef.current?.isOpen() ?? false;
      },
      get nativeElement() {
        if (isPhone2) {
          return null;
        }
        return popoverRef.current?.nativeElement ?? null;
      }
    }),
    [isPhone2, open]
  );
  if (isPhone2) {
    const titleId = `${id || "responsive-popover"}-title`;
    const mobileHeader = header || headerText || showCloseButton ? /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-between w-full", children: [
      header || headerText && /* @__PURE__ */ jsx(Title, { level: "H1", id: titleId, className: "text-base font-semibold", children: headerText }),
      showCloseButton && /* @__PURE__ */ jsx(
        Button,
        {
          design: "Tertiary",
          iconOnly: true,
          accessibleName: "Close",
          onClick: handleMobileClose,
          className: "ml-auto",
          icon: /* @__PURE__ */ jsxs(
            "svg",
            {
              xmlns: "http://www.w3.org/2000/svg",
              width: "16",
              height: "16",
              viewBox: "0 0 24 24",
              fill: "none",
              stroke: "currentColor",
              strokeWidth: "2",
              strokeLinecap: "round",
              strokeLinejoin: "round",
              children: [
                /* @__PURE__ */ jsx("line", { x1: "18", y1: "6", x2: "6", y2: "18" }),
                /* @__PURE__ */ jsx("line", { x1: "6", y1: "6", x2: "18", y2: "18" })
              ]
            }
          )
        }
      )
    ] }) : void 0;
    return /* @__PURE__ */ jsx(
      Dialog,
      {
        ref: dialogRef,
        open,
        onOpenChange: (isOpen) => {
          if (!isOpen) onClose?.();
        },
        headerText: !header && !showCloseButton ? headerText : void 0,
        header: mobileHeader,
        footer,
        stretch: true,
        noPadding,
        initialFocus,
        preventInitialFocus,
        accessibleName,
        accessibleNameRef: accessibleNameRef || (!accessibleName && headerText && !header && showCloseButton ? titleId : void 0),
        accessibleRole: accessibleRole === "AlertDialog" ? "alertdialog" : accessibleRole === "None" ? void 0 : "dialog",
        onBeforeOpen: () => onBeforeOpen?.(),
        onAfterOpen: () => onOpen?.(),
        onBeforeClose: (detail) => onBeforeClose?.({ escPressed: !!detail.escPressed }),
        onAfterClose: () => onClose?.(),
        className,
        id,
        children
      }
    );
  }
  return /* @__PURE__ */ jsx(
    Popover,
    {
      ref: popoverRef,
      open,
      opener,
      headerText: contentOnlyOnDesktop ? void 0 : headerText,
      header: contentOnlyOnDesktop ? void 0 : header,
      footer: contentOnlyOnDesktop ? void 0 : footer,
      placement,
      horizontalAlign,
      verticalAlign,
      hideArrow,
      offset,
      allowTargetOverlap,
      noPadding,
      initialFocus,
      preventFocusRestore,
      preventInitialFocus,
      accessibleName,
      accessibleNameRef,
      accessibleRole,
      accessibleDescription,
      onBeforeOpen,
      onOpen,
      onBeforeClose,
      onClose,
      className,
      style,
      id,
      children
    }
  );
}

export { ResponsivePopover };
