// @sap-ui/fx-components/dist/chunk-726M35FW.js
import { StatusPositiveIcon } from './chunk-H6VYFVIE.js';
import { WarningIcon } from './chunk-P4WB5QKL.js';
import { ErrorIcon } from './chunk-FL4UPYZT.js';
import { ListItem } from './chunk-E4NK6IRO.js';
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import * as React from 'react';
import { useContext, useState, useRef, useId, Children, useCallback, useEffect, isValidElement, cloneElement, useMemo, useImperativeHandle } from 'react';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';

// src/types/tabs.ts
var TabLayout = /* @__PURE__ */ ((TabLayout2) => {
  TabLayout2["Inline"] = "Inline";
  TabLayout2["Standard"] = "Standard";
  return TabLayout2;
})(TabLayout || {});
var TabOverflowMode = /* @__PURE__ */ ((TabOverflowMode2) => {
  TabOverflowMode2["End"] = "End";
  TabOverflowMode2["StartAndEnd"] = "StartAndEnd";
  return TabOverflowMode2;
})(TabOverflowMode || {});
var TabContainerBackgroundDesign = /* @__PURE__ */ ((TabContainerBackgroundDesign2) => {
  TabContainerBackgroundDesign2["Solid"] = "Solid";
  TabContainerBackgroundDesign2["Transparent"] = "Transparent";
  TabContainerBackgroundDesign2["Translucent"] = "Translucent";
  return TabContainerBackgroundDesign2;
})(TabContainerBackgroundDesign || {});
var TabsPlacement = /* @__PURE__ */ ((TabsPlacement2) => {
  TabsPlacement2["Top"] = "Top";
  TabsPlacement2["Bottom"] = "Bottom";
  return TabsPlacement2;
})(TabsPlacement || {});
var TabSemanticDesign = /* @__PURE__ */ ((TabSemanticDesign2) => {
  TabSemanticDesign2["Default"] = "Default";
  TabSemanticDesign2["Positive"] = "Positive";
  TabSemanticDesign2["Negative"] = "Negative";
  TabSemanticDesign2["Critical"] = "Critical";
  TabSemanticDesign2["Neutral"] = "Neutral";
  return TabSemanticDesign2;
})(TabSemanticDesign || {});
var tabVariants = cva(
  [
    "relative inline-flex items-center justify-center whitespace-nowrap",
    "px-4 py-2 text-sm font-medium",
    "transition-colors duration-200",
    "cursor-pointer select-none",
    "border-transparent"
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: [
          "text-sapphire-text-tertiary",
          "hover:text-sapphire-text-primary hover:bg-muted/50"
        ],
        ["Positive" /* Positive */]: [
          "text-sapphire-positive",
          "hover:text-sapphire-positive/80 hover:bg-sapphire-positive-bg",
          "data-[selected=true]:text-sapphire-positive data-[selected=true]:border-sapphire-positive"
        ],
        ["Negative" /* Negative */]: [
          "text-sapphire-negative",
          "hover:text-sapphire-negative/80 hover:bg-sapphire-negative/10"
        ],
        ["Critical" /* Critical */]: [
          "text-sapphire-warning",
          "hover:text-sapphire-warning/80 hover:bg-sapphire-warning-bg",
          "data-[selected=true]:text-sapphire-warning data-[selected=true]:border-sapphire-warning"
        ],
        ["Neutral" /* Neutral */]: [
          "text-sapphire-text-tertiary",
          "hover:text-sapphire-text-primary hover:bg-muted/50"
        ]
      },
      disabled: {
        true: "opacity-50 cursor-not-allowed pointer-events-none",
        false: ""
      },
      collapsed: {
        true: "px-3",
        false: ""
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      disabled: false,
      collapsed: false
    }
  }
);
var TabContainerContext = React.createContext(null);
function useTabContainerContext() {
  const context = useContext(TabContainerContext);
  if (!context) {
    throw new Error("Tab must be used within a TabContainer");
  }
  return context;
}
function getSelectedBorderColor(design) {
  switch (design) {
    case "Positive" /* Positive */:
      return "border-sapphire-positive";
    case "Negative" /* Negative */:
      return "border-sapphire-negative";
    case "Critical" /* Critical */:
      return "border-sapphire-warning";
    case "Neutral" /* Neutral */:
      return "border-sapphire-text-tertiary";
    default:
      return "border-sapphire-border-accent";
  }
}
function getSelectedTextColor(design) {
  switch (design) {
    case "Positive" /* Positive */:
      return "text-sapphire-positive";
    case "Negative" /* Negative */:
      return "text-sapphire-negative";
    case "Critical" /* Critical */:
      return "text-sapphire-warning";
    case "Neutral" /* Neutral */:
      return "text-sapphire-text-tertiary";
    default:
      return "text-sapphire-text-primary";
  }
}
function getSemanticIcon(design) {
  switch (design) {
    case "Positive" /* Positive */:
      return { Icon: StatusPositiveIcon, colorClass: "text-sapphire-positive" };
    case "Negative" /* Negative */:
      return { Icon: ErrorIcon, colorClass: "text-sapphire-negative" };
    case "Critical" /* Critical */:
      return { Icon: WarningIcon, colorClass: "text-sapphire-warning" };
    default:
      return null;
  }
}
function NestedTabItem({
  tab,
  level,
  onSelect,
  selectedTabId,
  enableDragDrop,
  onDrop,
  onDragStart,
  onDragEnd,
  draggedTabId
}) {
  const props = tab.props;
  const isSelected = selectedTabId === props.id;
  const isBeingDragged = draggedTabId === props.id;
  const [localDropIndicator, setLocalDropIndicator] = useState(null);
  const subTabElements = [];
  const extractTabs = (children) => {
    Children.forEach(children, (child) => {
      if (!isValidElement(child)) return;
      if (child.type === React.Fragment) {
        extractTabs(child.props.children);
      } else {
        subTabElements.push(child);
      }
    });
  };
  if (props.subTabs) {
    extractTabs(props.subTabs);
  }
  const handleDragOver = useCallback((e) => {
    if (!enableDragDrop || isBeingDragged || draggedTabId === props.id) return;
    e.preventDefault();
    e.stopPropagation();
    e.dataTransfer.dropEffect = "move";
    const rect = e.currentTarget.getBoundingClientRect();
    const relativeY = e.clientY - rect.top;
    const thirdHeight = rect.height / 3;
    let position;
    if (relativeY < thirdHeight) {
      position = "before";
    } else if (relativeY > thirdHeight * 2) {
      position = "after";
    } else {
      position = "on";
    }
    setLocalDropIndicator(position);
  }, [enableDragDrop, isBeingDragged, draggedTabId, props.id]);
  const handleDragEnter = useCallback((e) => {
    if (!enableDragDrop || isBeingDragged || draggedTabId === props.id) return;
    e.preventDefault();
    e.stopPropagation();
  }, [enableDragDrop, isBeingDragged, draggedTabId, props.id]);
  const handleDragLeave = useCallback((e) => {
    if (!enableDragDrop) return;
    const relatedTarget = e.relatedTarget;
    const currentTarget = e.currentTarget;
    if (!relatedTarget || !currentTarget.contains(relatedTarget)) {
      setLocalDropIndicator(null);
    }
  }, [enableDragDrop]);
  const handleDropEvent = useCallback((e) => {
    if (!enableDragDrop || isBeingDragged || draggedTabId === props.id) return;
    e.preventDefault();
    e.stopPropagation();
    const rect = e.currentTarget.getBoundingClientRect();
    const relativeY = e.clientY - rect.top;
    const thirdHeight = rect.height / 3;
    let placement;
    if (relativeY < thirdHeight) {
      placement = "before";
    } else if (relativeY > thirdHeight * 2) {
      placement = "after";
    } else {
      placement = "on";
    }
    setLocalDropIndicator(null);
    onDrop?.(e, props.id, placement);
  }, [enableDragDrop, isBeingDragged, draggedTabId, props.id, onDrop]);
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsxs("div", { className: "relative", children: [
      localDropIndicator === "before" && /* @__PURE__ */ jsx("div", { className: "absolute left-2 right-2 top-0 h-0.5 bg-sapphire-text-accent rounded-full z-10" }),
      localDropIndicator === "on" && /* @__PURE__ */ jsx("div", { className: "absolute inset-1 border-2 border-sapphire-border-accent border-dashed rounded-md z-10 pointer-events-none bg-sapphire-text-accent/10" }),
      /* @__PURE__ */ jsxs(
        "button",
        {
          type: "button",
          draggable: enableDragDrop && !props.disabled,
          onClick: () => {
            if (!props.disabled && props.children) {
              onSelect(props.id);
            }
          },
          disabled: props.disabled,
          onDragStart: enableDragDrop ? (e) => onDragStart?.(e, props.id) : void 0,
          onDragEnd: enableDragDrop ? onDragEnd : void 0,
          onDragOver: enableDragDrop ? handleDragOver : void 0,
          onDragEnter: enableDragDrop ? handleDragEnter : void 0,
          onDragLeave: enableDragDrop ? handleDragLeave : void 0,
          onDrop: enableDragDrop ? handleDropEvent : void 0,
          className: cn(
            "w-full text-left px-3 py-2 text-sm",
            "hover:bg-muted/50 focus:outline-none focus:bg-muted/50",
            "flex items-center gap-2",
            isSelected && "bg-muted font-medium",
            props.disabled && "opacity-50 cursor-not-allowed",
            isBeingDragged && "opacity-50",
            enableDragDrop && !props.disabled && "cursor-grab active:cursor-grabbing"
          ),
          style: { paddingLeft: `${12 + level * 16}px` },
          children: [
            props.design && props.design !== "Default" /* Default */ && props.design !== "Neutral" /* Neutral */ && /* @__PURE__ */ jsx(
              "span",
              {
                className: cn(
                  "w-2 h-2 rounded-full flex-shrink-0",
                  props.design === "Positive" /* Positive */ && "bg-sapphire-positive",
                  props.design === "Negative" /* Negative */ && "bg-sapphire-negative",
                  props.design === "Critical" /* Critical */ && "bg-sapphire-warning"
                )
              }
            ),
            props.icon && /* @__PURE__ */ jsx("span", { className: "flex-shrink-0", children: props.icon }),
            /* @__PURE__ */ jsx("span", { className: "flex-1", children: props.text }),
            props.additionalText && /* @__PURE__ */ jsx("span", { className: "text-xs text-sapphire-text-tertiary", children: props.additionalText })
          ]
        }
      ),
      localDropIndicator === "after" && /* @__PURE__ */ jsx("div", { className: "absolute left-2 right-2 bottom-0 h-0.5 bg-sapphire-text-accent rounded-full z-10" })
    ] }),
    subTabElements.map((subTab) => /* @__PURE__ */ jsx(
      NestedTabItem,
      {
        tab: subTab,
        level: level + 1,
        onSelect,
        selectedTabId,
        enableDragDrop,
        onDrop,
        onDragStart,
        onDragEnd,
        draggedTabId
      },
      subTab.props.id
    ))
  ] });
}
function NestedTabsDropdown({
  subTabs,
  isOpen,
  onClose,
  onSelect,
  selectedTabId,
  anchorRef,
  enableDragDrop,
  onDropTab,
  onDragStart,
  onDragEnd,
  draggedTabId
}) {
  const handleNestedDrop = useCallback((_e, tabId, placement) => {
    onDropTab?.(tabId, placement);
    onClose();
  }, [onDropTab, onClose]);
  const tabElements = [];
  const extractTabs = (children) => {
    Children.forEach(children, (child) => {
      if (!isValidElement(child)) return;
      if (child.type === React.Fragment) {
        extractTabs(child.props.children);
      } else {
        tabElements.push(child);
      }
    });
  };
  extractTabs(subTabs);
  return /* @__PURE__ */ jsx(
    ResponsivePopover,
    {
      open: isOpen,
      opener: anchorRef,
      placement: "Bottom" /* Bottom */,
      horizontalAlign: "Start" /* Start */,
      hideArrow: true,
      showCloseButton: false,
      contentOnlyOnDesktop: true,
      footer: /* @__PURE__ */ jsx("div", { className: "flex justify-end px-4 py-2", children: /* @__PURE__ */ jsx(Button, { design: "Tertiary", onClick: onClose, children: "Cancel" }) }),
      onClose,
      accessibleRole: "None",
      children: /* @__PURE__ */ jsx("div", { className: "-mx-4 -my-3 min-w-[200px] max-h-[300px] overflow-auto py-1", children: tabElements.map((tab) => /* @__PURE__ */ jsx(
        NestedTabItem,
        {
          tab,
          level: 0,
          onSelect: (tabId) => {
            onSelect(tabId);
            onClose();
          },
          selectedTabId,
          enableDragDrop,
          onDrop: handleNestedDrop,
          onDragStart,
          onDragEnd,
          draggedTabId
        },
        tab.props.id
      )) })
    }
  );
}
function Tab(props) {
  const {
    ref,
    id,
    text,
    icon,
    additionalText,
    disabled = false,
    design = "Default" /* Default */,
    movable = true,
    subTabs,
    tooltip,
    accessibleDescription,
    children: _children,
    className,
    style,
    "data-testid": dataTestId,
    _selected,
    _posinset,
    _setsize
  } = props;
  const context = useContext(TabContainerContext);
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
  const internalRef = useRef(null);
  const containerRef = useRef(null);
  const generatedId = useId();
  const tabId = id || generatedId;
  const panelId = `${context?.containerId || "tabs"}-panel`;
  const isSelected = _selected ?? context?.selectedTabId === tabId;
  const isCollapsed = context?.collapsed ?? false;
  const tabLayout = context?.layout ?? "Inline" /* Inline */;
  const isStandard = String(tabLayout) === "Standard" /* Standard */;
  const placement = context?.tabsPlacement ?? "Top" /* Top */;
  const isBottomPlacement = String(placement) === "Bottom" /* Bottom */;
  const tabDisplayMode = context?.tabDisplayMode ?? "text-only";
  const isActiveTab = context?.activeTabId === tabId;
  const hasSubTabs = !!subTabs && Children.count(subTabs) > 0;
  const hasOwnContent = !!_children;
  const isTwoClickArea = hasSubTabs && hasOwnContent;
  const isSingleClickArea = hasSubTabs && !hasOwnContent;
  const isDragDropEnabled = context?.enableDragDrop ?? false;
  const dragProps = isDragDropEnabled && movable ? context?.getDragProps?.(tabId, movable) ?? {} : {};
  const dropIndicator = context?.getDropIndicator?.(tabId);
  const isBeingDragged = context?.isDragging?.(tabId) ?? false;
  const isAnyDragging = context?.isAnyDragging ?? false;
  const dragOpenTimerRef = useRef(null);
  React.useEffect(() => {
    if (hasSubTabs && isAnyDragging && !isBeingDragged && dropIndicator === "on") {
      dragOpenTimerRef.current = setTimeout(() => {
        setIsDropdownOpen(true);
      }, 400);
    } else {
      if (dragOpenTimerRef.current) {
        clearTimeout(dragOpenTimerRef.current);
        dragOpenTimerRef.current = null;
      }
    }
    return () => {
      if (dragOpenTimerRef.current) {
        clearTimeout(dragOpenTimerRef.current);
      }
    };
  }, [hasSubTabs, isAnyDragging, isBeingDragged, dropIndicator]);
  React.useEffect(() => {
    if (!isAnyDragging && dragOpenTimerRef.current) {
      clearTimeout(dragOpenTimerRef.current);
      dragOpenTimerRef.current = null;
    }
  }, [isAnyDragging]);
  React.useLayoutEffect(() => {
    const element = internalRef.current;
    if (context && element) {
      context.registerTab(tabId, element);
      return () => {
        context.unregisterTab(tabId);
      };
    }
  }, [context, tabId]);
  React.useImperativeHandle(ref, () => internalRef.current, []);
  const handleClick = useCallback(() => {
    if (disabled) return;
    if (isSingleClickArea) {
      setIsDropdownOpen((prev) => !prev);
    } else if (context) {
      context.selectTab(tabId);
    }
  }, [disabled, isSingleClickArea, context, tabId]);
  const handleExpandClick = useCallback((e) => {
    e.stopPropagation();
    if (!disabled) {
      setIsDropdownOpen((prev) => !prev);
    }
  }, [disabled]);
  const handleSubTabSelect = useCallback((selectedTabId) => {
    if (context) {
      context.selectTab(selectedTabId);
    }
  }, [context]);
  const handleDropdownDrop = useCallback((destinationTabId, placement2) => {
    if (!context?.draggedTabId || !context?.onTabMove) return;
    const sourceTabId = context.draggedTabId;
    context.onTabMove({
      sourceTabId,
      sourceIndex: -1,
      // Index not relevant for nested drops
      destinationTabId,
      destinationIndex: -1,
      // Index not relevant for nested drops
      placement: placement2
    });
  }, [context]);
  const handleKeyDown = useCallback((e) => {
    if (e.key === "ArrowDown" && hasSubTabs) {
      e.preventDefault();
      setIsDropdownOpen(true);
      return;
    }
    if (context?.onKeyDown) {
      context.onKeyDown(e, tabId);
    }
  }, [context, tabId, hasSubTabs]);
  const handleKeyUp = useCallback((e) => {
    if (isSingleClickArea) return;
    if (context?.onKeyUp) {
      context.onKeyUp(e, tabId);
    }
  }, [context, tabId, isSingleClickArea]);
  const handleFocus = useCallback(() => {
    if (context?.onTabFocus) {
      context.onTabFocus(tabId);
    }
  }, [context, tabId]);
  const handleBlur = useCallback((e) => {
    if (context?.onTabBlur) {
      context.onTabBlur(e);
    }
  }, [context]);
  return /* @__PURE__ */ jsxs("div", { ref: containerRef, className: "fx-tab-item relative inline-flex", children: [
    dropIndicator === "before" && /* @__PURE__ */ jsx("div", { className: "absolute left-0 top-1 bottom-1 w-0.5 bg-sapphire-text-accent rounded-full z-10 -translate-x-1" }),
    dropIndicator === "on" && /* @__PURE__ */ jsx("div", { className: "absolute inset-0 border-2 border-sapphire-border-accent border-dashed rounded-md z-10 pointer-events-none bg-sapphire-text-accent/10" }),
    /* @__PURE__ */ jsxs("div", { className: "inline-flex", children: [
      /* @__PURE__ */ jsxs(
        "button",
        {
          ref: internalRef,
          id: `${context?.containerId || "tabs"}-tab-${tabId}`,
          role: "tab",
          type: "button",
          "aria-selected": isSelected,
          "aria-controls": hasOwnContent ? panelId : void 0,
          "aria-disabled": disabled || void 0,
          "aria-expanded": isSingleClickArea ? isDropdownOpen : void 0,
          "aria-haspopup": isSingleClickArea ? "menu" : void 0,
          "aria-roledescription": isTwoClickArea ? "Split Tab" : void 0,
          "aria-posinset": _posinset,
          "aria-setsize": _setsize,
          "aria-describedby": accessibleDescription ? `${tabId}-desc` : void 0,
          tabIndex: isActiveTab ? 0 : -1,
          title: tooltip,
          disabled,
          "data-selected": isSelected,
          "data-tab-id": tabId,
          "data-movable": movable,
          "data-dragging": isBeingDragged || void 0,
          onClick: handleClick,
          onKeyDown: handleKeyDown,
          onKeyUp: handleKeyUp,
          onFocus: handleFocus,
          onBlur: handleBlur,
          className: cn(
            tabVariants({
              design,
              disabled,
              collapsed: isCollapsed
            }),
            // Border placement: top or bottom depending on tabsPlacement
            isBottomPlacement ? "border-t-2" : "border-b-2",
            // Selected state: border color + text color
            isSelected && getSelectedBorderColor(design),
            isSelected && getSelectedTextColor(design),
            // Standard layout: vertical stacking
            isStandard && !isCollapsed && "flex-col items-center gap-0.5 py-2.5",
            // Text-only mode: compact padding
            tabDisplayMode === "text-only" && !isStandard && "py-1.5",
            isTwoClickArea && "pr-1",
            // Less padding on right for expand button
            isBeingDragged && "opacity-50",
            isDragDropEnabled && movable && !disabled && "cursor-grab active:cursor-grabbing",
            className
          ),
          style,
          "data-testid": dataTestId,
          ...isTwoClickArea ? {} : dragProps,
          children: [
            (() => {
              const semanticInfo = getSemanticIcon(design);
              if (!semanticInfo) return null;
              const { Icon: SemanticIcon, colorClass } = semanticInfo;
              return /* @__PURE__ */ jsx(SemanticIcon, { className: cn("h-3 w-3 flex-shrink-0", colorClass) });
            })(),
            icon && /* @__PURE__ */ jsx("span", { className: cn(
              "flex-shrink-0",
              !isStandard && text && !isCollapsed && "mr-2"
            ), children: icon }),
            !isCollapsed && text && /* @__PURE__ */ jsx("span", { children: text }),
            isSingleClickArea && !isCollapsed && /* @__PURE__ */ jsx(
              SlimArrowDownIcon,
              {
                className: cn(
                  "ml-1 h-4 w-4 transition-transform",
                  isDropdownOpen && "rotate-180"
                )
              }
            ),
            !isCollapsed && additionalText && /* @__PURE__ */ jsx(
              "span",
              {
                className: cn(
                  "inline-flex items-center justify-center",
                  // Mixed mode: tabs without icon get large prominent additionalText
                  tabDisplayMode === "mixed" && !icon ? "text-[1.5rem] font-bold leading-none" : cn(
                    isStandard ? "text-xs" : "ml-2",
                    "px-2 py-0.5 text-xs rounded-full",
                    "bg-muted text-sapphire-text-tertiary",
                    design === "Positive" /* Positive */ && "bg-sapphire-positive-bg text-sapphire-positive",
                    design === "Negative" /* Negative */ && "bg-sapphire-negative/10 text-sapphire-negative",
                    design === "Critical" /* Critical */ && "bg-sapphire-warning-bg text-sapphire-warning"
                  )
                ),
                children: additionalText
              }
            ),
            accessibleDescription && /* @__PURE__ */ jsx("span", { id: `${tabId}-desc`, className: "sr-only", children: accessibleDescription })
          ]
        }
      ),
      isTwoClickArea && /* @__PURE__ */ jsxs(Fragment, { children: [
        /* @__PURE__ */ jsx("div", { className: "w-px h-4 bg-sapphire-border-primary self-center" }),
        /* @__PURE__ */ jsx(
          "button",
          {
            type: "button",
            onClick: handleExpandClick,
            disabled,
            "aria-expanded": isDropdownOpen,
            "aria-haspopup": "menu",
            "aria-label": "Show sub-tabs",
            tabIndex: -1,
            className: cn(
              "inline-flex items-center justify-center",
              "px-1 py-2",
              "text-sapphire-text-tertiary hover:text-sapphire-text-primary",
              "hover:bg-muted/50",
              "focus:outline-none focus:ring-2 focus:ring-ring",
              isBottomPlacement ? "border-t-2 border-transparent" : "border-b-2 border-transparent",
              isSelected && (isBottomPlacement ? "border-t-2" : "border-b-2"),
              isSelected && getSelectedBorderColor(design),
              disabled && "opacity-50 cursor-not-allowed"
            ),
            ...dragProps,
            children: /* @__PURE__ */ jsx(
              SlimArrowDownIcon,
              {
                className: cn(
                  "h-4 w-4 transition-transform",
                  isDropdownOpen && "rotate-180"
                )
              }
            )
          }
        )
      ] })
    ] }),
    dropIndicator === "after" && /* @__PURE__ */ jsx("div", { className: "absolute right-0 top-1 bottom-1 w-0.5 bg-sapphire-text-accent rounded-full z-10 translate-x-1" }),
    hasSubTabs && /* @__PURE__ */ jsx(
      NestedTabsDropdown,
      {
        subTabs,
        isOpen: isDropdownOpen,
        onClose: () => setIsDropdownOpen(false),
        onSelect: handleSubTabSelect,
        selectedTabId: context?.selectedTabId ?? null,
        anchorRef: containerRef,
        enableDragDrop: isDragDropEnabled,
        onDropTab: handleDropdownDrop,
        onDragStart: context?.handleDragStart,
        onDragEnd: context?.handleDragEnd,
        draggedTabId: context?.draggedTabId
      }
    )
  ] });
}
Tab.displayName = "Tab";
function TabSeparator({ className, style, "data-testid": dataTestId, ref }) {
  return /* @__PURE__ */ jsx(
    "span",
    {
      ref,
      role: "separator",
      "aria-orientation": "vertical",
      className: cn(
        "inline-block h-4 w-px mx-2",
        "bg-sapphire-border-primary",
        "self-center",
        className
      ),
      style,
      "data-testid": dataTestId
    }
  );
}
TabSeparator.displayName = "TabSeparator";
function useTabDragDrop(options) {
  const {
    enabled = false,
    tabIds,
    onMove,
    onMoveOver,
    maxNestingLevel,
    getTabNestingLevel,
    getTabChildren
  } = options;
  const [draggedTabId, setDraggedTabId] = useState(null);
  const [dropTargetId, setDropTargetId] = useState(null);
  const [dropPosition, setDropPosition] = useState(null);
  const dragImageRef = useRef(null);
  const wouldCreateCircle = useCallback(
    (sourceId, targetId) => {
      if (!getTabChildren) return false;
      const getDescendants = (id) => {
        const descendants2 = /* @__PURE__ */ new Set();
        const children = getTabChildren(id);
        for (const childId of children) {
          descendants2.add(childId);
          const childDescendants = getDescendants(childId);
          childDescendants.forEach((d) => descendants2.add(d));
        }
        return descendants2;
      };
      const descendants = getDescendants(sourceId);
      return descendants.has(targetId);
    },
    [getTabChildren]
  );
  const canNestAt = useCallback(
    (targetId) => {
      if (maxNestingLevel === void 0) return true;
      if (maxNestingLevel === 0) return false;
      if (getTabNestingLevel) {
        const targetLevel = getTabNestingLevel(targetId);
        return targetLevel < maxNestingLevel;
      }
      return true;
    },
    [maxNestingLevel, getTabNestingLevel]
  );
  const handleDragStart = useCallback(
    (e, tabId) => {
      if (!enabled) return;
      setDraggedTabId(tabId);
      e.dataTransfer.effectAllowed = "move";
      e.dataTransfer.setData("text/plain", tabId);
      e.dataTransfer.setData("application/x-tab-id", tabId);
      const target = e.currentTarget;
      target.classList.add("dragging");
    },
    [enabled]
  );
  const handleDragEnd = useCallback(
    (e) => {
      if (!enabled) return;
      const target = e.currentTarget;
      target.classList.remove("dragging");
      setDraggedTabId(null);
      setDropTargetId(null);
      setDropPosition(null);
      if (dragImageRef.current) {
        dragImageRef.current.remove();
        dragImageRef.current = null;
      }
    },
    [enabled]
  );
  const handleDragOver = useCallback(
    (e, tabId) => {
      if (!enabled || !draggedTabId || draggedTabId === tabId) return;
      if (wouldCreateCircle(draggedTabId, tabId)) {
        e.dataTransfer.dropEffect = "none";
        return;
      }
      e.preventDefault();
      e.dataTransfer.dropEffect = "move";
      const rect = e.currentTarget.getBoundingClientRect();
      const relativeX = e.clientX - rect.left;
      const thirdWidth = rect.width / 3;
      let position;
      if (relativeX < thirdWidth) {
        position = "before";
      } else if (relativeX > thirdWidth * 2) {
        position = "after";
      } else {
        position = canNestAt(tabId) ? "on" : relativeX < rect.width / 2 ? "before" : "after";
      }
      if (onMoveOver) {
        let prevented = false;
        const detail = {
          sourceTabId: draggedTabId,
          destinationTabId: tabId,
          placement: position,
          preventDefault: () => {
            prevented = true;
          },
          get defaultPrevented() {
            return prevented;
          }
        };
        onMoveOver(detail);
        if (prevented) {
          e.dataTransfer.dropEffect = "none";
          setDropTargetId(null);
          setDropPosition(null);
          return;
        }
      }
      setDropTargetId(tabId);
      setDropPosition(position);
    },
    [enabled, draggedTabId, wouldCreateCircle, canNestAt, onMoveOver]
  );
  const handleDragEnter = useCallback(
    (e, tabId) => {
      if (!enabled || !draggedTabId || draggedTabId === tabId) return;
      if (wouldCreateCircle(draggedTabId, tabId)) {
        return;
      }
      e.preventDefault();
      setDropTargetId(tabId);
    },
    [enabled, draggedTabId, wouldCreateCircle]
  );
  const handleDragLeave = useCallback(
    (e, tabId) => {
      if (!enabled) return;
      const relatedTarget = e.relatedTarget;
      const currentTarget = e.currentTarget;
      if (!relatedTarget || !currentTarget.contains(relatedTarget)) {
        if (dropTargetId === tabId) {
          setDropTargetId(null);
          setDropPosition(null);
        }
      }
    },
    [enabled, dropTargetId]
  );
  const handleDrop = useCallback(
    (e, tabId) => {
      if (!enabled || !draggedTabId || draggedTabId === tabId) return;
      if (wouldCreateCircle(draggedTabId, tabId)) {
        return;
      }
      e.preventDefault();
      const rect = e.currentTarget.getBoundingClientRect();
      const relativeX = e.clientX - rect.left;
      const thirdWidth = rect.width / 3;
      let placement;
      if (relativeX < thirdWidth) {
        placement = "before";
      } else if (relativeX > thirdWidth * 2) {
        placement = "after";
      } else {
        placement = canNestAt(tabId) ? "on" : relativeX < rect.width / 2 ? "before" : "after";
      }
      const sourceIndex = tabIds.indexOf(draggedTabId);
      const destinationIndex = tabIds.indexOf(tabId);
      if (destinationIndex === -1) return;
      const detail = {
        sourceTabId: draggedTabId,
        sourceIndex,
        destinationTabId: tabId,
        destinationIndex,
        placement
      };
      onMove?.(detail);
      setDraggedTabId(null);
      setDropTargetId(null);
      setDropPosition(null);
    },
    [enabled, draggedTabId, tabIds, onMove, wouldCreateCircle, canNestAt]
  );
  const getDragProps = useCallback(
    (tabId, isMovable = true) => {
      if (!enabled || !isMovable) {
        return {};
      }
      return {
        draggable: true,
        onDragStart: (e) => handleDragStart(e, tabId),
        onDragEnd: handleDragEnd,
        onDragOver: (e) => handleDragOver(e, tabId),
        onDragEnter: (e) => handleDragEnter(e, tabId),
        onDragLeave: (e) => handleDragLeave(e, tabId),
        onDrop: (e) => handleDrop(e, tabId)
      };
    },
    [
      enabled,
      handleDragStart,
      handleDragEnd,
      handleDragOver,
      handleDragEnter,
      handleDragLeave,
      handleDrop
    ]
  );
  const getDropIndicator = useCallback(
    (tabId) => {
      if (!enabled || !draggedTabId || dropTargetId !== tabId) {
        return null;
      }
      return dropPosition;
    },
    [enabled, draggedTabId, dropTargetId, dropPosition]
  );
  const isDragging = useCallback(
    (tabId) => draggedTabId === tabId,
    [draggedTabId]
  );
  const isAnyDragging = draggedTabId !== null;
  return {
    // State
    draggedTabId,
    dropTargetId,
    dropPosition,
    isAnyDragging,
    // Methods
    getDragProps,
    getDropIndicator,
    isDragging,
    // Raw handlers (for custom implementations)
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDragEnter,
    handleDragLeave,
    handleDrop
  };
}
function useTabOverflow({
  enabled = true,
  overflowMode,
  tabIds,
  selectedTabId,
  tabListRef
}) {
  const [visibleTabIds, setVisibleTabIds] = useState(tabIds);
  const [startOverflowTabIds, setStartOverflowTabIds] = useState([]);
  const [endOverflowTabIds, setEndOverflowTabIds] = useState([]);
  const [isReady, setIsReady] = useState(false);
  const tabWidthsRef = useRef(/* @__PURE__ */ new Map());
  const hasMeasuredRef = useRef(false);
  const rafId = useRef(null);
  const measureTabs = useCallback(() => {
    if (!tabListRef.current) return false;
    const container = tabListRef.current;
    const tabElements = container.querySelectorAll('[role="tab"][data-tab-id]');
    let measuredCount = 0;
    tabElements.forEach((element) => {
      const tabId = element.getAttribute("data-tab-id");
      if (tabId && tabIds.includes(tabId)) {
        const htmlElement = element;
        const rect = htmlElement.getBoundingClientRect();
        if (rect.width > 0) {
          const style = window.getComputedStyle(htmlElement);
          const width = rect.width + parseFloat(style.marginLeft || "0") + parseFloat(style.marginRight || "0");
          tabWidthsRef.current.set(tabId, width);
          measuredCount++;
        } else if (tabWidthsRef.current.has(tabId)) {
          measuredCount++;
        }
      }
    });
    const allMeasured = measuredCount === tabIds.length;
    if (allMeasured) {
      hasMeasuredRef.current = true;
    }
    return allMeasured;
  }, [tabIds, tabListRef]);
  const calculateOverflow = useCallback(() => {
    if (!enabled || !tabListRef.current || tabIds.length === 0) {
      setVisibleTabIds(tabIds);
      setStartOverflowTabIds([]);
      setEndOverflowTabIds([]);
      setIsReady(true);
      return;
    }
    const container = tabListRef.current;
    const containerWidth = container.clientWidth;
    if (containerWidth === 0) {
      return;
    }
    if (!hasMeasuredRef.current || tabWidthsRef.current.size < tabIds.length) {
      if (!measureTabs()) {
        return;
      }
    }
    const tabWidths = tabWidthsRef.current;
    let totalWidth = 0;
    for (const tabId of tabIds) {
      totalWidth += tabWidths.get(tabId) || 0;
    }
    const overflowButtonWidth = 52;
    const isStartAndEnd = String(overflowMode) === "StartAndEnd";
    if (totalWidth <= containerWidth) {
      setVisibleTabIds(tabIds);
      setStartOverflowTabIds([]);
      setEndOverflowTabIds([]);
      setIsReady(true);
      return;
    }
    let availableWidth = containerWidth;
    if (isStartAndEnd) {
      availableWidth -= overflowButtonWidth * 2;
    } else {
      availableWidth -= overflowButtonWidth;
    }
    const selectedIndex = selectedTabId ? tabIds.indexOf(selectedTabId) : 0;
    const safeSelectedIndex = selectedIndex >= 0 ? selectedIndex : 0;
    if (isStartAndEnd) {
      const visible = [];
      const startOverflow = [];
      const endOverflow = [];
      let usedWidth = tabWidths.get(tabIds[safeSelectedIndex]) || 0;
      visible.push(tabIds[safeSelectedIndex]);
      let leftIndex = safeSelectedIndex - 1;
      let rightIndex = safeSelectedIndex + 1;
      let addedSomething = true;
      while (addedSomething && (leftIndex >= 0 || rightIndex < tabIds.length)) {
        addedSomething = false;
        if (rightIndex < tabIds.length) {
          const rightWidth = tabWidths.get(tabIds[rightIndex]) || 0;
          if (usedWidth + rightWidth <= availableWidth) {
            visible.push(tabIds[rightIndex]);
            usedWidth += rightWidth;
            rightIndex++;
            addedSomething = true;
          }
        }
        if (leftIndex >= 0) {
          const leftWidth = tabWidths.get(tabIds[leftIndex]) || 0;
          if (usedWidth + leftWidth <= availableWidth) {
            visible.unshift(tabIds[leftIndex]);
            usedWidth += leftWidth;
            leftIndex--;
            addedSomething = true;
          }
        }
      }
      while (leftIndex >= 0) {
        startOverflow.unshift(tabIds[leftIndex]);
        leftIndex--;
      }
      while (rightIndex < tabIds.length) {
        endOverflow.push(tabIds[rightIndex]);
        rightIndex++;
      }
      setVisibleTabIds(visible);
      setStartOverflowTabIds(startOverflow);
      setEndOverflowTabIds(endOverflow);
    } else {
      const visible = [];
      const endOverflow = [];
      let usedWidth = 0;
      for (let i = 0; i < tabIds.length; i++) {
        const tabId = tabIds[i];
        const width = tabWidths.get(tabId) || 0;
        if (usedWidth + width <= availableWidth) {
          visible.push(tabId);
          usedWidth += width;
        } else {
          endOverflow.push(tabId);
        }
      }
      if (selectedTabId && endOverflow.includes(selectedTabId)) {
        const selectedWidth = tabWidths.get(selectedTabId) || 0;
        while (visible.length > 0) {
          const lastVisible = visible[visible.length - 1];
          if (lastVisible === selectedTabId) break;
          const lastWidth = tabWidths.get(lastVisible) || 0;
          usedWidth -= lastWidth;
          visible.pop();
          const overflowIdx = endOverflow.indexOf(selectedTabId);
          endOverflow.splice(overflowIdx, 0, lastVisible);
          if (usedWidth + selectedWidth <= availableWidth) {
            const selIdx = endOverflow.indexOf(selectedTabId);
            endOverflow.splice(selIdx, 1);
            visible.push(selectedTabId);
            usedWidth += selectedWidth;
            break;
          }
        }
      }
      setVisibleTabIds(visible);
      setStartOverflowTabIds([]);
      setEndOverflowTabIds(endOverflow);
    }
    setIsReady(true);
  }, [enabled, overflowMode, tabIds, selectedTabId, tabListRef, measureTabs]);
  const scheduleCalculation = useCallback(() => {
    if (rafId.current) {
      cancelAnimationFrame(rafId.current);
    }
    rafId.current = requestAnimationFrame(() => {
      calculateOverflow();
      rafId.current = null;
    });
  }, [calculateOverflow]);
  useEffect(() => {
    if (!enabled) {
      setVisibleTabIds(tabIds);
      setIsReady(true);
      return;
    }
    if (tabWidthsRef.current.size !== tabIds.length) {
      hasMeasuredRef.current = false;
    }
    let retryCount = 0;
    const maxRetries = 20;
    let timeoutId = null;
    const tryCalculate = () => {
      if (measureTabs()) {
        calculateOverflow();
      } else if (retryCount < maxRetries) {
        retryCount++;
        timeoutId = setTimeout(tryCalculate, 16);
      }
    };
    tryCalculate();
    return () => {
      if (timeoutId) clearTimeout(timeoutId);
    };
  }, [enabled, tabIds, measureTabs, calculateOverflow]);
  useEffect(() => {
    if (!enabled || !isReady) return;
    calculateOverflow();
  }, [selectedTabId, enabled, isReady, calculateOverflow]);
  useEffect(() => {
    if (!enabled || !tabListRef.current) return;
    let lastWidth = tabListRef.current.clientWidth;
    const observer = new ResizeObserver((entries) => {
      const newWidth = entries[0]?.contentRect?.width ?? 0;
      if (Math.abs(newWidth - lastWidth) > 1) {
        lastWidth = newWidth;
        scheduleCalculation();
      }
    });
    observer.observe(tabListRef.current);
    return () => {
      observer.disconnect();
      if (rafId.current) {
        cancelAnimationFrame(rafId.current);
      }
    };
  }, [enabled, tabListRef, scheduleCalculation]);
  const recalculate = useCallback(() => {
    hasMeasuredRef.current = false;
    tabWidthsRef.current.clear();
    setIsReady(false);
    setTimeout(() => {
      measureTabs();
      calculateOverflow();
    }, 0);
  }, [measureTabs, calculateOverflow]);
  return {
    visibleTabIds,
    startOverflowTabIds,
    endOverflowTabIds,
    hasStartOverflow: startOverflowTabIds.length > 0,
    hasEndOverflow: endOverflowTabIds.length > 0,
    recalculate,
    isReady
  };
}
function TabOverflowButton({ tabIds, tabs, selectedTabId, onSelect, position, overflowMode, onArrowOut, customTrigger, className, ref }) {
  const [isOpen, setIsOpen] = useState(false);
  const buttonRef = useRef(null);
  React.useImperativeHandle(ref, () => buttonRef.current, []);
  const overflowTabs = tabs.filter((tab) => {
    const props = tab.props;
    return tabIds.includes(props.id);
  });
  const hasSelectedInOverflow = selectedTabId && tabIds.includes(selectedTabId);
  const handleClick = useCallback(() => {
    setIsOpen((prev) => !prev);
  }, []);
  const handleKeyDown = useCallback((e) => {
    if (e.key === "ArrowLeft" || e.key === "ArrowRight" || e.key === "ArrowDown" || e.key === "ArrowUp") {
      e.preventDefault();
      if (e.key === "ArrowDown") {
        setIsOpen(true);
      } else {
        const isRTL = document.documentElement.dir === "rtl";
        const towardTabs = position === "end" ? isRTL ? "ArrowRight" : "ArrowLeft" : isRTL ? "ArrowLeft" : "ArrowRight";
        const awayFromTabs = position === "end" ? isRTL ? "ArrowLeft" : "ArrowRight" : isRTL ? "ArrowRight" : "ArrowLeft";
        if (e.key === towardTabs) {
          onArrowOut?.();
        } else if (e.key === awayFromTabs) {
          setIsOpen(true);
        }
      }
    }
  }, [position, onArrowOut]);
  if (tabIds.length === 0) return null;
  const isStartAndEnd = overflowMode === "StartAndEnd" /* StartAndEnd */;
  const label = position === "end" && !isStartAndEnd ? "More" : `+${tabIds.length}`;
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    customTrigger && isValidElement(customTrigger) ? cloneElement(customTrigger, {
      ref: buttonRef,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      "aria-expanded": isOpen,
      "aria-haspopup": "menu",
      "aria-label": position === "start" ? "More tabs at start" : "More tabs"
    }) : /* @__PURE__ */ jsxs(
      "button",
      {
        ref: buttonRef,
        type: "button",
        tabIndex: -1,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        "aria-expanded": isOpen,
        "aria-haspopup": "menu",
        "aria-label": position === "start" ? "More tabs at start" : "More tabs",
        className: cn(
          "inline-flex items-center justify-center gap-1",
          "px-3 py-2 text-sm font-medium",
          "text-sapphire-text-secondary hover:text-sapphire-text-primary",
          "hover:bg-muted/50",
          "border-b-2 border-transparent",
          "relative focus:outline-none fx-tab-item",
          "transition-colors duration-200",
          "flex-shrink-0",
          "fx-tab-overflow-btn",
          hasSelectedInOverflow && "text-sapphire-text-accent border-sapphire-border-accent",
          className
        ),
        children: [
          label,
          position === "end" && /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-3 w-3" })
        ]
      }
    ),
    /* @__PURE__ */ jsx(
      ResponsivePopover,
      {
        open: isOpen,
        opener: buttonRef,
        placement: "Bottom" /* Bottom */,
        horizontalAlign: position === "end" ? "End" /* End */ : "Start" /* Start */,
        hideArrow: true,
        noPadding: true,
        showCloseButton: false,
        contentOnlyOnDesktop: true,
        footer: /* @__PURE__ */ jsx("div", { className: "flex justify-end px-4 py-2", children: /* @__PURE__ */ jsx(Button, { design: "Tertiary", onClick: () => setIsOpen(false), children: "Cancel" }) }),
        onClose: () => setIsOpen(false),
        accessibleRole: "None",
        children: /* @__PURE__ */ jsx("div", { className: "min-w-[200px] max-h-[300px] overflow-auto", children: /* @__PURE__ */ jsx(List, { selectionMode: "None", separators: "None", children: overflowTabs.map((tab) => {
          const tabProps = tab.props;
          const isSelected = selectedTabId === tabProps.id;
          return /* @__PURE__ */ jsx(
            ListItem,
            {
              itemKey: tabProps.id,
              text: tabProps.text ?? "",
              icon: tabProps.icon,
              additionalText: tabProps.additionalText,
              disabled: tabProps.disabled,
              selected: isSelected,
              onClick: () => {
                if (!tabProps.disabled) {
                  onSelect(tabProps.id);
                  setIsOpen(false);
                }
              }
            },
            tabProps.id
          );
        }) }) })
      }
    )
  ] });
}
var containerVariants = cva(
  "w-full",
  {
    variants: {
      backgroundDesign: {
        ["Solid" /* Solid */]: "",
        ["Transparent" /* Transparent */]: "",
        ["Translucent" /* Translucent */]: ""
      }
    },
    defaultVariants: {
      backgroundDesign: "Solid" /* Solid */
    }
  }
);
var headerBackgroundVariants = cva(
  "",
  {
    variants: {
      backgroundDesign: {
        ["Solid" /* Solid */]: "bg-sapphire-background-primary",
        ["Transparent" /* Transparent */]: "bg-transparent",
        ["Translucent" /* Translucent */]: "bg-sapphire-background-primary/80 backdrop-blur-sm"
      }
    },
    defaultVariants: {
      backgroundDesign: "Solid" /* Solid */
    }
  }
);
var tabListVariants = cva(
  [
    "flex items-center",
    "border-b border-sapphire-border-primary"
  ],
  {
    variants: {
      layout: {
        ["Inline" /* Inline */]: "flex-row",
        ["Standard" /* Standard */]: "flex-row"
      },
      hasOverflow: {
        true: "overflow-hidden",
        false: "overflow-x-auto scrollbar-thin scrollbar-thumb-muted"
      }
    },
    defaultVariants: {
      layout: "Inline" /* Inline */,
      hasOverflow: false
    }
  }
);
var tabPanelVariants = cva(
  "p-4",
  {
    variants: {
      backgroundDesign: {
        ["Solid" /* Solid */]: "bg-sapphire-background-primary",
        ["Transparent" /* Transparent */]: "bg-transparent",
        ["Translucent" /* Translucent */]: "bg-sapphire-background-primary/80 backdrop-blur-sm"
      }
    },
    defaultVariants: {
      backgroundDesign: "Solid" /* Solid */
    }
  }
);
function extractNestedTabs(subTabs, tabContents, disabledIds, allTabIds, nestedToParent, parentToChildren, childToDirectParent, parentId, directParentId) {
  Children.forEach(subTabs, (child) => {
    if (!isValidElement(child)) return;
    if (child.type === React.Fragment) {
      extractNestedTabs(child.props.children, tabContents, disabledIds, allTabIds, nestedToParent, parentToChildren, childToDirectParent, parentId, directParentId);
      return;
    }
    const displayName = child.type?.displayName;
    if (displayName === "Tab" || child.type === Tab) {
      const props = child.props;
      const tabId = props.id;
      if (tabId) {
        allTabIds.push(tabId);
        nestedToParent.set(tabId, parentId);
        if (!parentToChildren.has(directParentId)) {
          parentToChildren.set(directParentId, []);
        }
        parentToChildren.get(directParentId).push(tabId);
        childToDirectParent.set(tabId, directParentId);
        if (props.children) {
          tabContents.set(tabId, props.children);
        }
        if (props.disabled) {
          disabledIds.add(tabId);
        }
        if (props.subTabs) {
          extractNestedTabs(props.subTabs, tabContents, disabledIds, allTabIds, nestedToParent, parentToChildren, childToDirectParent, parentId, tabId);
        }
      }
    }
  });
}
function extractTabChildren(children) {
  const tabs = [];
  const tabIds = [];
  const allTabIds = [];
  const tabContents = /* @__PURE__ */ new Map();
  const disabledIds = /* @__PURE__ */ new Set();
  const nestedToParent = /* @__PURE__ */ new Map();
  const parentToChildren = /* @__PURE__ */ new Map();
  const childToDirectParent = /* @__PURE__ */ new Map();
  Children.forEach(children, (child) => {
    if (!isValidElement(child)) return;
    const displayName = child.type?.displayName;
    if (displayName === "Tab" || child.type === Tab) {
      const props = child.props;
      const tabId = props.id;
      if (tabId) {
        tabs.push(child);
        tabIds.push(tabId);
        allTabIds.push(tabId);
        if (props.children) {
          tabContents.set(tabId, props.children);
        }
        if (props.disabled) {
          disabledIds.add(tabId);
        }
        if (props.subTabs) {
          extractNestedTabs(props.subTabs, tabContents, disabledIds, allTabIds, nestedToParent, parentToChildren, childToDirectParent, tabId, tabId);
        }
      }
    } else if (displayName === "TabSeparator" || child.type === TabSeparator) {
      tabs.push(child);
    }
  });
  return { tabs, tabIds, allTabIds, tabContents, disabledIds, nestedToParent, parentToChildren, childToDirectParent };
}
function TabContainer(props) {
  const {
    ref,
    selectedTabId: controlledSelectedTabId,
    defaultSelectedTabId,
    noAutoSelection = false,
    layout = "Inline" /* Inline */,
    backgroundDesign = "Solid" /* Solid */,
    headerBackgroundDesign,
    contentBackgroundDesign,
    collapsed = false,
    tabsPlacement = "Top" /* Top */,
    overflowMode = "End" /* End */,
    tabsOverflowMode = false,
    overflowButton,
    startOverflowButton,
    enableDragDrop = false,
    maxNestingLevel,
    onTabSelect,
    onTabMove,
    onTabMoveOver,
    accessibleName,
    accessibleNameRef,
    children,
    className,
    style,
    "data-testid": dataTestId,
    id: providedId
  } = props;
  const effectiveHeaderBg = headerBackgroundDesign ?? backgroundDesign;
  const effectiveContentBg = contentBackgroundDesign ?? backgroundDesign;
  const isBottom = String(tabsPlacement) === "Bottom" /* Bottom */;
  const generatedId = useId();
  const containerId = providedId || `tabs-${generatedId}`;
  const tabListRef = useRef(null);
  const tabPanelRef = useRef(null);
  const tabRegistrations = useRef(/* @__PURE__ */ new Map());
  const startOverflowButtonRef = useRef(null);
  const endOverflowButtonRef = useRef(null);
  const { tabs, tabIds, allTabIds, tabContents, disabledIds, nestedToParent, parentToChildren, childToDirectParent } = useMemo(
    () => extractTabChildren(children),
    [children]
  );
  const tabDisplayMode = useMemo(() => {
    let hasIcon = false;
    let hasNoIcon = false;
    for (const tab of tabs) {
      if (!isValidElement(tab)) continue;
      const displayName = tab.type?.displayName;
      if (displayName === "Tab" || tab.type === Tab) {
        const tabProps = tab.props;
        if (tabProps.icon) {
          hasIcon = true;
        } else {
          hasNoIcon = true;
        }
      }
    }
    if (hasIcon && hasNoIcon) return "mixed";
    if (hasIcon) return "icon";
    return "text-only";
  }, [tabs]);
  const [internalSelectedTabId, setInternalSelectedTabId] = useState(
    () => {
      if (defaultSelectedTabId) return defaultSelectedTabId;
      if (noAutoSelection) return null;
      return tabIds[0] ?? null;
    }
  );
  const [focusedTabId, setFocusedTabId] = useState(null);
  const selectedTabId = controlledSelectedTabId ?? internalSelectedTabId;
  const selectedTopLevelTabId = useMemo(() => {
    if (!selectedTabId) return null;
    if (tabIds.includes(selectedTabId)) return selectedTabId;
    return nestedToParent.get(selectedTabId) ?? null;
  }, [selectedTabId, tabIds, nestedToParent]);
  const activeTabId = focusedTabId ?? selectedTabId ?? tabIds[0];
  const getTabNestingLevel = useCallback(
    (tabId) => {
      let level = 0;
      let currentId = tabId;
      while (childToDirectParent.has(currentId)) {
        level++;
        currentId = childToDirectParent.get(currentId);
      }
      return level;
    },
    [childToDirectParent]
  );
  const getTabChildren = useCallback(
    (tabId) => {
      return parentToChildren.get(tabId) ?? [];
    },
    [parentToChildren]
  );
  const {
    getDragProps,
    getDropIndicator,
    isDragging,
    isAnyDragging,
    draggedTabId,
    handleDragStart,
    handleDragEnd,
    handleDragOver,
    handleDragEnter,
    handleDragLeave,
    handleDrop
  } = useTabDragDrop({
    enabled: enableDragDrop,
    tabIds,
    onMove: onTabMove,
    onMoveOver: onTabMoveOver,
    maxNestingLevel,
    getTabNestingLevel,
    getTabChildren
  });
  const {
    visibleTabIds,
    startOverflowTabIds,
    endOverflowTabIds,
    hasStartOverflow,
    hasEndOverflow,
    isReady: overflowReady
  } = useTabOverflow({
    enabled: tabsOverflowMode,
    overflowMode,
    tabIds,
    selectedTabId: selectedTopLevelTabId,
    tabListRef
  });
  const focusTabById = useCallback((targetId) => {
    const registration = tabRegistrations.current.get(targetId);
    if (registration?.element) {
      registration.element.focus();
    }
  }, []);
  const selectTab = useCallback(
    (tabId) => {
      if (!allTabIds.includes(tabId)) return;
      if (disabledIds.has(tabId)) return;
      const previousTabId = selectedTabId;
      if (controlledSelectedTabId === void 0) {
        setInternalSelectedTabId(tabId);
      }
      const detail = {
        selectedTabId: tabId,
        previousTabId,
        tabIndex: allTabIds.indexOf(tabId)
      };
      onTabSelect?.(detail);
    },
    [allTabIds, disabledIds, selectedTabId, controlledSelectedTabId, onTabSelect]
  );
  const handleKeyDown = useCallback(
    (e, currentTabId) => {
      const navTabIds = tabsOverflowMode && overflowReady && visibleTabIds.length > 0 ? visibleTabIds : tabIds;
      const currentIndex = navTabIds.indexOf(currentTabId);
      if (currentIndex === -1) return;
      const findNextEnabled = (startIndex, direction) => {
        let index = startIndex;
        const count = navTabIds.length;
        while (true) {
          const next = index + direction;
          if (next < 0 || next >= count) return startIndex;
          index = next;
          if (!disabledIds.has(navTabIds[index])) return index;
        }
      };
      const findFirstEnabled = () => {
        for (let i = 0; i < navTabIds.length; i++) {
          if (!disabledIds.has(navTabIds[i])) return i;
        }
        return 0;
      };
      const findLastEnabled = () => {
        for (let i = navTabIds.length - 1; i >= 0; i--) {
          if (!disabledIds.has(navTabIds[i])) return i;
        }
        return navTabIds.length - 1;
      };
      let targetIndex = -1;
      const isRTL = document.documentElement.dir === "rtl";
      switch (e.key) {
        case "ArrowRight":
          e.preventDefault();
          targetIndex = findNextEnabled(currentIndex, isRTL ? -1 : 1);
          if (targetIndex === currentIndex) {
            const overflowRef = isRTL ? startOverflowButtonRef : endOverflowButtonRef;
            if (overflowRef.current) {
              overflowRef.current.focus();
              return;
            }
          }
          break;
        case "ArrowLeft":
          e.preventDefault();
          targetIndex = findNextEnabled(currentIndex, isRTL ? 1 : -1);
          if (targetIndex === currentIndex) {
            const overflowRef = isRTL ? endOverflowButtonRef : startOverflowButtonRef;
            if (overflowRef.current) {
              overflowRef.current.focus();
              return;
            }
          }
          break;
        case "Home":
          e.preventDefault();
          targetIndex = findFirstEnabled();
          break;
        case "End":
          e.preventDefault();
          targetIndex = findLastEnabled();
          break;
        case "Enter":
          e.preventDefault();
          selectTab(currentTabId);
          return;
        case " ":
          e.preventDefault();
          return;
        default:
          return;
      }
      if (targetIndex >= 0 && targetIndex !== currentIndex) {
        const targetId = navTabIds[targetIndex];
        setFocusedTabId(targetId);
        focusTabById(targetId);
      }
    },
    [tabIds, visibleTabIds, tabsOverflowMode, overflowReady, disabledIds, selectTab, focusTabById]
  );
  const handleKeyUp = useCallback(
    (e, currentTabId) => {
      if (e.key === " ") {
        e.preventDefault();
        selectTab(currentTabId);
      }
    },
    [selectTab]
  );
  const handleTabFocus = useCallback((tabId) => {
    setFocusedTabId(tabId);
  }, []);
  const handleTabBlur = useCallback((e) => {
    const relatedTarget = e.relatedTarget;
    if (!tabListRef.current?.contains(relatedTarget)) {
      setFocusedTabId(null);
    }
  }, []);
  const registerTab = useCallback((id, element) => {
    tabRegistrations.current.set(id, { id, element, disabled: disabledIds.has(id) });
  }, [disabledIds]);
  const unregisterTab = useCallback((id) => {
    tabRegistrations.current.delete(id);
  }, []);
  const getTabIndex = useCallback(
    (id) => tabIds.indexOf(id),
    [tabIds]
  );
  const contextValue = useMemo(
    () => ({
      selectedTabId,
      selectTab,
      registerTab,
      unregisterTab,
      layout,
      collapsed,
      enableDragDrop,
      onTabMove,
      containerId,
      focusTab: focusTabById,
      getTabIndex,
      onKeyDown: handleKeyDown,
      onKeyUp: handleKeyUp,
      onTabFocus: handleTabFocus,
      onTabBlur: handleTabBlur,
      activeTabId,
      getDragProps,
      getDropIndicator,
      isDragging,
      isAnyDragging,
      draggedTabId,
      handleDragStart,
      handleDragEnd,
      handleDragOver,
      handleDragEnter,
      handleDragLeave,
      handleDrop,
      tabsPlacement,
      tabDisplayMode
    }),
    [
      selectedTabId,
      selectTab,
      registerTab,
      unregisterTab,
      layout,
      collapsed,
      enableDragDrop,
      onTabMove,
      containerId,
      focusTabById,
      getTabIndex,
      handleKeyDown,
      handleKeyUp,
      handleTabFocus,
      handleTabBlur,
      activeTabId,
      getDragProps,
      getDropIndicator,
      isDragging,
      isAnyDragging,
      draggedTabId,
      handleDragStart,
      handleDragEnd,
      handleDragOver,
      handleDragEnter,
      handleDragLeave,
      handleDrop,
      tabsPlacement,
      tabDisplayMode
    ]
  );
  useImperativeHandle(ref, () => ({
    selectTab,
    getSelectedTabId: () => selectedTabId,
    getTabIds: () => tabIds,
    scrollToTab: (id) => {
      const registration = tabRegistrations.current.get(id);
      registration?.element?.scrollIntoView({
        behavior: "smooth",
        inline: "center",
        block: "nearest"
      });
    },
    focus: () => {
      const tabToFocus = selectedTabId ?? tabIds[0];
      if (tabToFocus) {
        focusTabById(tabToFocus);
      }
    },
    get tabListElement() {
      return tabListRef.current;
    },
    get tabPanelElement() {
      return tabPanelRef.current;
    }
  }), [selectedTabId, tabIds, selectTab, focusTabById]);
  const renderedTabs = useMemo(() => {
    let tabIndex = 0;
    return Children.map(tabs, (child) => {
      if (!isValidElement(child)) return child;
      const displayName = child.type?.displayName;
      if (displayName === "Tab" || child.type === Tab) {
        const tabProps = child.props;
        const currentIndex = tabIndex;
        tabIndex++;
        const shouldHide = tabsOverflowMode && overflowReady && !visibleTabIds.includes(tabProps.id);
        return cloneElement(child, {
          _selected: selectedTopLevelTabId === tabProps.id,
          _index: currentIndex,
          _posinset: currentIndex + 1,
          _setsize: tabIds.length,
          style: shouldHide ? { ...tabProps.style, display: "none" } : tabProps.style,
          "aria-hidden": shouldHide || void 0
        });
      }
      return child;
    });
  }, [tabs, selectedTopLevelTabId, tabsOverflowMode, visibleTabIds, overflowReady]);
  const selectedTabContent = selectedTabId ? tabContents.get(selectedTabId) : null;
  const panelId = `${containerId}-panel`;
  const tabControlId = selectedTabId ? `${containerId}-tab-${selectedTabId}` : void 0;
  return /* @__PURE__ */ jsx(TabContainerContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        containerVariants({ backgroundDesign }),
        "flex",
        isBottom ? "flex-col-reverse" : "flex-col",
        className
      ),
      style,
      "data-testid": dataTestId,
      id: containerId,
      children: [
        /* @__PURE__ */ jsxs(
          "div",
          {
            ref: tabListRef,
            role: "tablist",
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-orientation": "horizontal",
            className: cn(
              "fx-tab-list",
              tabListVariants({ layout, hasOverflow: tabsOverflowMode && overflowReady }),
              headerBackgroundVariants({ backgroundDesign: effectiveHeaderBg }),
              isBottom && "border-b-0 border-t border-sapphire-border-primary"
            ),
            children: [
              tabsOverflowMode && overflowReady && hasStartOverflow && /* @__PURE__ */ jsx(
                TabOverflowButton,
                {
                  ref: startOverflowButtonRef,
                  tabIds: startOverflowTabIds,
                  tabs,
                  selectedTabId: selectedTopLevelTabId,
                  onSelect: selectTab,
                  position: "start",
                  overflowMode,
                  customTrigger: startOverflowButton,
                  onArrowOut: () => {
                    const firstVisible = visibleTabIds.find((id) => !disabledIds.has(id));
                    if (firstVisible) {
                      setFocusedTabId(firstVisible);
                      focusTabById(firstVisible);
                    }
                  }
                }
              ),
              renderedTabs,
              tabsOverflowMode && overflowReady && hasEndOverflow && /* @__PURE__ */ jsx(
                TabOverflowButton,
                {
                  ref: endOverflowButtonRef,
                  tabIds: endOverflowTabIds,
                  tabs,
                  selectedTabId: selectedTopLevelTabId,
                  onSelect: selectTab,
                  position: "end",
                  overflowMode,
                  customTrigger: overflowButton,
                  onArrowOut: () => {
                    const lastVisible = [...visibleTabIds].reverse().find((id) => !disabledIds.has(id));
                    if (lastVisible) {
                      setFocusedTabId(lastVisible);
                      focusTabById(lastVisible);
                    }
                  }
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "grid transition-[grid-template-rows] duration-300 ease-in-out",
              selectedTabContent ? "grid-rows-[1fr]" : "grid-rows-[0fr]"
            ),
            children: /* @__PURE__ */ jsx("div", { className: "overflow-hidden", children: /* @__PURE__ */ jsx(
              "div",
              {
                ref: tabPanelRef,
                id: panelId,
                role: "tabpanel",
                "aria-labelledby": tabControlId,
                "aria-hidden": !selectedTabContent || void 0,
                className: cn(
                  tabPanelVariants({ backgroundDesign: effectiveContentBg })
                ),
                children: selectedTabContent
              }
            ) })
          }
        )
      ]
    }
  ) });
}
function useTabNavigation(options) {
  const {
    tabCount,
    focusedIndex,
    disabledTabs = /* @__PURE__ */ new Set(),
    onNavigate,
    onSelect,
    rtl = false
  } = options;
  const findNextEnabledTab = useCallback(
    (currentIndex, direction) => {
      if (tabCount === 0) return -1;
      let nextIndex = currentIndex;
      let attempts = 0;
      do {
        nextIndex = nextIndex + direction;
        if (nextIndex >= tabCount) {
          nextIndex = 0;
        } else if (nextIndex < 0) {
          nextIndex = tabCount - 1;
        }
        attempts++;
        if (attempts >= tabCount) {
          return currentIndex;
        }
      } while (disabledTabs.has(nextIndex));
      return nextIndex;
    },
    [tabCount, disabledTabs]
  );
  const findFirstEnabledTab = useCallback(() => {
    for (let i = 0; i < tabCount; i++) {
      if (!disabledTabs.has(i)) {
        return i;
      }
    }
    return 0;
  }, [tabCount, disabledTabs]);
  const findLastEnabledTab = useCallback(() => {
    for (let i = tabCount - 1; i >= 0; i--) {
      if (!disabledTabs.has(i)) {
        return i;
      }
    }
    return tabCount - 1;
  }, [tabCount, disabledTabs]);
  const handleKeyDown = useCallback(
    (e) => {
      const getDirection = (key) => {
        if (key === "ArrowRight") {
          return rtl ? -1 : 1;
        }
        if (key === "ArrowLeft") {
          return rtl ? 1 : -1;
        }
        return null;
      };
      switch (e.key) {
        case "ArrowRight":
        case "ArrowLeft": {
          e.preventDefault();
          const direction = getDirection(e.key);
          if (direction !== null) {
            const nextIndex = findNextEnabledTab(focusedIndex, direction);
            if (nextIndex !== focusedIndex) {
              onNavigate?.(nextIndex);
            }
          }
          break;
        }
        case "Home":
          e.preventDefault();
          {
            const firstIndex = findFirstEnabledTab();
            if (firstIndex !== focusedIndex) {
              onNavigate?.(firstIndex);
            }
          }
          break;
        case "End":
          e.preventDefault();
          {
            const lastIndex = findLastEnabledTab();
            if (lastIndex !== focusedIndex) {
              onNavigate?.(lastIndex);
            }
          }
          break;
        case "Enter":
        case " ":
          e.preventDefault();
          if (focusedIndex >= 0 && !disabledTabs.has(focusedIndex)) {
            onSelect?.(focusedIndex);
          }
          break;
      }
    },
    [
      focusedIndex,
      rtl,
      disabledTabs,
      findNextEnabledTab,
      findFirstEnabledTab,
      findLastEnabledTab,
      onNavigate,
      onSelect
    ]
  );
  return {
    handleKeyDown,
    findNextEnabledTab,
    findFirstEnabledTab,
    findLastEnabledTab
  };
}

export { Tab, TabContainer, TabContainerBackgroundDesign, TabContainerContext, TabLayout, TabOverflowButton, TabOverflowMode, TabSemanticDesign, TabSeparator, TabsPlacement, useTabContainerContext, useTabDragDrop, useTabNavigation, useTabOverflow };
