// @sap-ui/fx-components/dist/chunk-T7M6DGUN.js
import { BusyIndicator } from './chunk-E2WLAC4U.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useRef, useState, useEffect, useImperativeHandle, useCallback } from 'react';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';

var filledBrandStyles = [
  "bg-sapphire-brand-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-brand-hover-background",
  "active:bg-sapphire-brand-pressed-background [&.kb-active]:bg-sapphire-brand-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus"
];
var filledJouleStyles = [
  "bg-sapphire-joule-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-joule-hover-background",
  "active:bg-sapphire-joule-pressed-background [&.kb-active]:bg-sapphire-joule-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus"
];
var outlineBrandStyles = [
  "bg-transparent text-sapphire-brand-foreground border border-sapphire-brand-foreground",
  "hover:bg-sapphire-neutral-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-brand-pressed-foreground [&.kb-active]:text-sapphire-brand-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
];
var outlineJouleStyles = [
  "bg-transparent text-sapphire-joule-foreground border border-sapphire-joule-foreground",
  "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background hover:border-sapphire-joule-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-joule-pressed-foreground [&.kb-active]:text-sapphire-joule-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
];
var buttonVariants = cva(
  [
    // Base styles
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        // --- Primary (filled) ---
        ["Primary" /* Primary */]: filledBrandStyles,
        ["PrimaryJoule" /* PrimaryJoule */]: filledJouleStyles,
        // --- Secondary (outlined) ---
        ["Secondary" /* Secondary */]: outlineBrandStyles,
        ["SecondaryJoule" /* SecondaryJoule */]: outlineJouleStyles,
        // --- Tertiary (ghost, colored text) ---
        ["Tertiary" /* Tertiary */]: [
          "bg-transparent text-sapphire-brand-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-brand-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
        ],
        ["TertiaryJoule" /* TertiaryJoule */]: [
          "bg-transparent text-sapphire-joule-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-joule-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
        ],
        // --- Neutral (ghost, dark text) ---
        ["Neutral" /* Neutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-sapphire-neutral-foreground-muted",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-neutral-foreground-black [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- SecondaryNeutral (ghost, neutral-black text) ---
        ["SecondaryNeutral" /* SecondaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background [&.kb-active]:bg-sapphire-neutral-pressed-background",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- TertiaryNeutral (ghost, muted neutral text, no background) ---
        ["TertiaryNeutral" /* TertiaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-muted border border-transparent",
          "hover:text-sapphire-neutral-foreground-black",
          "active:text-sapphire-neutral-foreground-black [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ]
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var largeSizeClasses = {
  default: "h-10 px-4 py-3 text-base gap-2 rounded-lg",
  iconOnly: "h-10 w-10 p-0 text-base rounded-lg",
  icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
};
var mediumSizeClasses = {
  default: "h-8 px-3 py-2 text-sm gap-1.5 rounded",
  iconOnly: "h-8 w-8 p-0 text-sm rounded",
  icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
};
var smallSizeClasses = {
  default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
  iconOnly: "h-6 w-6 p-0 text-xs rounded",
  icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
};
var sizeClasses = {
  ["Large" /* Large */]: largeSizeClasses,
  ["Medium" /* Medium */]: mediumSizeClasses,
  ["Small" /* Small */]: smallSizeClasses
};
function getButtonType(type) {
  switch (type) {
    case "Submit" /* Submit */:
    case "Submit":
      return "submit";
    case "Reset" /* Reset */:
    case "Reset":
      return "reset";
    default:
      return "button";
  }
}
function Button({
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  type = "Button" /* Button */,
  disabled = false,
  loading = false,
  icon,
  endIcon,
  iconOnly = false,
  children,
  tooltip,
  states,
  state,
  stateTransition = "slide-up",
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Button" /* Button */,
  accessibilityAttributes,
  name,
  value,
  form,
  onClick,
  onFocus,
  onBlur,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  "data-growing-button": dataGrowingButton,
  ref
}) {
  const buttonRef = useRef(null);
  const hiddenButtonRef = useRef(null);
  const [animationPhase, setAnimationPhase] = useState("idle");
  const [displayedState, setDisplayedState] = useState(
    null
  );
  const [buttonWidth, setButtonWidth] = useState(void 0);
  const currentStateObject = states?.find((s) => s.name === state) ?? states?.[0];
  const isMultiState = states && states.length > 0;
  useEffect(() => {
    if (isMultiState && !displayedState) {
      setDisplayedState(currentStateObject ?? null);
    }
  }, [isMultiState, currentStateObject, displayedState]);
  useEffect(() => {
    if (!isMultiState || !currentStateObject || animationPhase !== "idle") {
      return;
    }
    if (displayedState && displayedState.name !== currentStateObject.name) {
      if (buttonRef.current) {
        setButtonWidth(buttonRef.current.offsetWidth);
      }
      setAnimationPhase("fade-out");
    }
  }, [currentStateObject, displayedState, isMultiState, animationPhase]);
  useEffect(() => {
    if (animationPhase === "fade-out") {
      const timer = setTimeout(() => {
        setAnimationPhase("fade-mid");
      }, 180);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-mid") {
      setDisplayedState(currentStateObject ?? null);
      if (hiddenButtonRef.current) {
        setButtonWidth(hiddenButtonRef.current.offsetWidth);
      }
      const timer = setTimeout(() => {
        setAnimationPhase("fade-in");
      }, 20);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-in") {
      const timer = setTimeout(() => {
        setAnimationPhase("idle");
        setButtonWidth(void 0);
      }, 160);
      return () => clearTimeout(timer);
    }
  }, [animationPhase, currentStateObject]);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        buttonRef.current?.focus();
      },
      blur() {
        buttonRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === buttonRef.current;
      },
      get nativeElement() {
        return buttonRef.current;
      }
    }),
    []
  );
  const handleClick = useCallback(
    (e) => {
      if (disabled || loading) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [disabled, loading, onClick]
  );
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled || loading) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " ") {
          e.preventDefault();
        }
        buttonRef.current?.classList.add("kb-active");
        if (e.key === " ") {
          onClick?.({
            originalEvent: e,
            isKeyboard: true
          });
        }
      }
    },
    [disabled, loading, onClick]
  );
  const handleKeyUp = useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        buttonRef.current?.classList.remove("kb-active");
      }
    },
    []
  );
  const ariaExpanded = accessibilityAttributes?.expanded;
  const ariaHasPopup = accessibilityAttributes?.hasPopup;
  const ariaControls = accessibilityAttributes?.controls;
  const ariaPressed = accessibilityAttributes?.pressed;
  const role = accessibleRole === "Link" /* Link */ || accessibleRole === "Link" ? "link" : void 0;
  const isDisabled = disabled;
  const effectiveIcon = isMultiState ? displayedState?.icon : icon;
  const effectiveEndIcon = isMultiState ? displayedState?.endIcon : endIcon;
  const effectiveChildren = isMultiState ? displayedState?.text : children;
  const effectiveIconOnly = isMultiState ? !displayedState?.text && !!displayedState?.icon : iconOnly;
  const effectivePulsing = isMultiState ? displayedState?.pulsing : false;
  const PulsingText = ({ children: text }) => /* @__PURE__ */ jsxs("span", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsx("span", { className: "opacity-50", children: text }),
    /* @__PURE__ */ jsx(
      "span",
      {
        className: "absolute inset-0 overflow-hidden",
        style: {
          background: "linear-gradient(90deg, transparent 0%, currentColor 50%, transparent 100%)",
          backgroundSize: "200% 100%",
          WebkitBackgroundClip: "text",
          backgroundClip: "text",
          WebkitTextFillColor: "transparent",
          animation: "pulse-sweep 1.5s ease-in-out infinite"
        },
        children: text
      }
    ),
    /* @__PURE__ */ jsx("style", { children: `
          @keyframes pulse-sweep {
            0% { background-position: 200% 0; }
            100% { background-position: -200% 0; }
          }
        ` })
  ] });
  const getTransitionStyles = (phase, transition) => {
    if (transition === "none" || phase === "idle") {
      return {};
    }
    const baseTransition = "all 100ms ease-out";
    switch (transition) {
      case "slide-up":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(-8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-down":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(-8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-left":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(-12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "slide-right":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(-12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "fade":
        if (phase === "fade-out") return { opacity: 0, transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0 };
        if (phase === "fade-in") return { opacity: 1, transition: baseTransition };
        break;
      case "scale":
        if (phase === "fade-out") return { opacity: 0, transform: "scale(0.75)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "scale(0.75)" };
        if (phase === "fade-in") return { opacity: 1, transform: "scale(1)", transition: baseTransition };
        break;
      case "flip":
        if (phase === "fade-out") return { opacity: 0, transform: "rotateX(90deg)", transition: baseTransition, transformStyle: "preserve-3d" };
        if (phase === "fade-mid") return { opacity: 0, transform: "rotateX(-90deg)", transformStyle: "preserve-3d" };
        if (phase === "fade-in") return { opacity: 1, transform: "rotateX(0)", transition: baseTransition, transformStyle: "preserve-3d" };
        break;
    }
    return {};
  };
  const contentAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  const iconAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsxs(
      "button",
      {
        ref: buttonRef,
        type: getButtonType(type),
        disabled: isDisabled,
        "aria-disabled": loading || void 0,
        id,
        name,
        value,
        form,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-expanded": ariaExpanded,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-pressed": ariaPressed,
        "aria-busy": loading || void 0,
        title: tooltip,
        tabIndex,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur: (e) => {
          buttonRef.current?.classList.remove("kb-active");
          onBlur?.(e);
        },
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][effectiveIconOnly ? "iconOnly" : "default"],
          isMultiState && !effectiveIconOnly && "transition-[width] duration-300 ease-out overflow-hidden",
          loading && "relative pointer-events-none",
          className
        ),
        style: {
          ...style,
          ...buttonWidth !== void 0 && !effectiveIconOnly ? { width: buttonWidth } : {}
        },
        "data-testid": dataTestId,
        "data-growing-button": dataGrowingButton,
        children: [
          loading && /* @__PURE__ */ jsx("span", { className: "absolute inset-0 flex items-center justify-center z-[1] [&_*]:!text-inherit", inert: true, children: /* @__PURE__ */ jsx(BusyIndicator, { size: "S", active: true, delay: 0 }) }),
          effectiveIcon ? /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? iconAnimationStyles : void 0, children: effectiveIcon }) : null,
          !effectiveIconOnly && effectiveChildren && /* @__PURE__ */ jsx("span", { className: loading ? "opacity-40" : void 0, style: isMultiState ? contentAnimationStyles : void 0, children: effectivePulsing ? /* @__PURE__ */ jsx(PulsingText, { children: effectiveChildren }) : effectiveChildren }),
          effectiveEndIcon && /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? contentAnimationStyles : void 0, children: effectiveEndIcon })
        ]
      }
    ),
    isMultiState && /* @__PURE__ */ jsxs(
      "button",
      {
        ref: hiddenButtonRef,
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][!currentStateObject?.text && currentStateObject?.icon ? "iconOnly" : "default"],
          "absolute -top-[10000px] -left-[10000px] invisible w-fit"
        ),
        tabIndex: -1,
        "aria-hidden": "true",
        children: [
          currentStateObject?.icon && /* @__PURE__ */ jsx("span", { children: currentStateObject.icon }),
          currentStateObject?.text && /* @__PURE__ */ jsx("span", { children: currentStateObject.text }),
          currentStateObject?.endIcon && /* @__PURE__ */ jsx("span", { children: currentStateObject.endIcon })
        ]
      }
    )
  ] });
}

export { Button };
