// @sap-ui/fx-components/dist/chunk-VSZFLW7F.js
import { cva, cn } from './chunk-2F2PZMJ5.js';
import { useRef, useImperativeHandle } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/bar.ts
var BarDesign = /* @__PURE__ */ ((BarDesign2) => {
  BarDesign2["Header"] = "Header";
  BarDesign2["Subheader"] = "Subheader";
  BarDesign2["Footer"] = "Footer";
  BarDesign2["FloatingFooter"] = "FloatingFooter";
  return BarDesign2;
})(BarDesign || {});
var BarAccessibleRole = /* @__PURE__ */ ((BarAccessibleRole2) => {
  BarAccessibleRole2["Toolbar"] = "Toolbar";
  BarAccessibleRole2["None"] = "None";
  return BarAccessibleRole2;
})(BarAccessibleRole || {});
var childStyles = [
  "[&>*]:mx-1 [&>*]:max-w-full [&>*]:box-border",
  "[&>:not(div)]:whitespace-nowrap [&>:not(div)]:overflow-hidden [&>:not(div)]:text-ellipsis"
].join(" ");
var barVariants = cva(
  [
    "flex items-center justify-between w-full overflow-hidden box-border"
  ],
  {
    variants: {
      design: {
        ["Header" /* Header */]: [
          "h-11",
          "bg-[var(--sapPageHeader_Background,var(--card))]",
          "shadow-[var(--sapContent_HeaderShadow,0_0_0.25rem_0_rgba(0,0,0,0.15))]"
        ],
        ["Subheader" /* Subheader */]: [
          "h-12 -mt-px",
          "bg-[var(--sapPageHeader_Background,var(--card))]",
          "shadow-[var(--sapContent_HeaderShadow,0_0_0.25rem_0_rgba(0,0,0,0.15))]"
        ],
        ["Footer" /* Footer */]: [
          "h-11 shadow-none",
          "bg-[var(--sapPageFooter_Background,var(--muted))]",
          "border-t border-[var(--sapPageFooter_BorderColor,var(--border))]"
        ],
        ["FloatingFooter" /* FloatingFooter */]: [
          "h-11 border-none",
          "bg-[var(--sapPageFooter_Background,var(--muted))]",
          "rounded-[var(--sapElement_BorderCornerRadius,var(--radius))]",
          "shadow-[var(--sapContent_Shadow1,0_0_1rem_0_rgba(0,0,0,0.15))]"
        ]
      }
    },
    defaultVariants: {
      design: "Header" /* Header */
    }
  }
);
function Bar({
  design = "Header" /* Header */,
  accessibleRole = "Toolbar" /* Toolbar */,
  accessibleName,
  accessibleNameRef,
  startContent,
  children,
  endContent,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const barRef = useRef(null);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        barRef.current?.focus();
      },
      blur() {
        barRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === barRef.current;
      },
      get nativeElement() {
        return barRef.current;
      }
    }),
    []
  );
  const isToolbar = accessibleRole !== "None";
  const effectiveRole = isToolbar ? "toolbar" : void 0;
  const effectiveAriaLabel = isToolbar ? accessibleName || (accessibleNameRef ? void 0 : design) : accessibleName;
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: barRef,
      id,
      role: effectiveRole,
      "aria-label": accessibleNameRef ? void 0 : effectiveAriaLabel,
      "aria-labelledby": accessibleNameRef,
      className: cn(barVariants({ design }), className),
      style,
      "data-testid": dataTestId,
      "data-design": design,
      children: [
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "flex items-center shrink basis-auto grow-0 ps-4 min-w-0",
              childStyles
            ),
            "data-part": "startContent",
            children: startContent
          }
        ),
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "flex items-center justify-center flex-auto px-2 min-w-0 overflow-hidden",
              childStyles
            ),
            "data-part": "midContent",
            children
          }
        ),
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(
              "flex items-center shrink-0 grow-0 basis-auto pe-4 min-w-0",
              childStyles
            ),
            "data-part": "endContent",
            children: endContent
          }
        )
      ]
    }
  );
}

export { Bar, BarAccessibleRole, BarDesign };
