// @sap-ui/fx-components/dist/chunk-KG7YP5ME.js
import { SynchronizeIcon } from './chunk-WQOS25OP.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useState, useImperativeHandle, useCallback, isValidElement, useEffect, Children } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';

function Card({
  children,
  header,
  footer,
  toolbar,
  toolbarVisible,
  toolbarClassName,
  loading = false,
  loadingDelay = 1e3,
  accessibleName,
  accessibleNameRef,
  accessibleDescriptionRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  onMouseEnter,
  onMouseLeave,
  onMouseDown,
  onMouseUp,
  onClick,
  onFocus,
  onBlur,
  ref
}) {
  const cardRef = useRef(null);
  const [showLoading, setShowLoading] = useState(false);
  const isHeaderInteractive = isValidElement(header) && header.props.interactive === true;
  useEffect(() => {
    if (!loading) {
      setShowLoading(false);
      return;
    }
    if (loadingDelay <= 0) {
      setShowLoading(true);
      return;
    }
    const timer = setTimeout(() => {
      setShowLoading(true);
    }, loadingDelay);
    return () => clearTimeout(timer);
  }, [loading, loadingDelay]);
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        cardRef.current?.focus();
      },
      blur() {
        cardRef.current?.blur();
      },
      get nativeElement() {
        return cardRef.current;
      }
    }),
    []
  );
  const hasContent = children != null && children !== false && children !== "" && Children.count(children) > 0;
  const card = /* @__PURE__ */ jsxs(
    "article",
    {
      ref: cardRef,
      id,
      role: "region",
      "aria-roledescription": "Card",
      "aria-label": accessibleName,
      "aria-labelledby": accessibleNameRef,
      "aria-describedby": accessibleDescriptionRef,
      "aria-busy": loading || void 0,
      className: cn(
        "relative rounded-[8px] border border-sapphire-border-primary bg-sapphire-card-bg-primary text-sapphire-text-primary",
        "flex flex-col",
        isHeaderInteractive && [
          // Only transition border and shadow, not outline (focus should be instant)
          "transition-[border-color,box-shadow] duration-200",
          // Focus ring on the card when header is focused (using outline to avoid layout shift)
          "focus-within:outline focus-within:outline-2 focus-within:outline-sapphire-border-focus focus-within:outline-offset-[-2px]",
          // Hover state
          "hover:border-sapphire-border-active hover:shadow-[0px_1px_2px_0px_rgba(10,10,10,0.25)]"
        ],
        !hasContent && "card--nocontent",
        // When no content/footer, make header corners fully rounded
        !hasContent && !footer && "[&>*:first-child]:rounded-[8px]",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      onMouseEnter,
      onMouseLeave,
      onMouseDown,
      onMouseUp,
      onClick,
      onFocus,
      onBlur,
      children: [
        header,
        hasContent && /* @__PURE__ */ jsxs(
          "div",
          {
            className: "flex-1 relative overflow-hidden",
            role: "group",
            "aria-label": "Card Content",
            "data-part": "content",
            children: [
              children,
              showLoading && /* @__PURE__ */ jsx(
                "div",
                {
                  className: cn(
                    "absolute inset-0",
                    "bg-sapphire-background-primary/80 backdrop-blur-sm",
                    "flex items-center justify-center",
                    "z-10"
                  ),
                  "aria-hidden": "true",
                  children: /* @__PURE__ */ jsx(SynchronizeIcon, { className: "h-8 w-8 animate-spin text-sapphire-text-tertiary" })
                }
              )
            ]
          }
        ),
        footer
      ]
    }
  );
  if (toolbar) {
    return /* @__PURE__ */ jsxs("div", { className: "relative group/card", "data-part": "card-toolbar-wrapper", children: [
      /* @__PURE__ */ jsx(
        "div",
        {
          className: cn(
            "absolute bottom-full left-0 right-0 z-10 translate-y-3",
            !toolbarVisible && "invisible group-hover/card:visible"
          ),
          "data-part": "toolbar",
          children: /* @__PURE__ */ jsx(
            "div",
            {
              className: cn(
                "flex items-center justify-center gap-px p-1",
                "bg-sapphire-card-bg-primary border border-brand-purple rounded-[8px]",
                "shadow-[0px_4px_10px_0px_rgba(10,10,10,0.15),0px_2px_5px_0px_rgba(10,10,10,0.05)]",
                toolbarClassName
              ),
              children: toolbar
            }
          )
        }
      ),
      card
    ] });
  }
  return card;
}

// src/types/card.ts
var CardHeaderStatus = /* @__PURE__ */ ((CardHeaderStatus2) => {
  CardHeaderStatus2["None"] = "None";
  CardHeaderStatus2["Positive"] = "Positive";
  CardHeaderStatus2["Negative"] = "Negative";
  CardHeaderStatus2["Critical"] = "Critical";
  CardHeaderStatus2["Information"] = "Information";
  return CardHeaderStatus2;
})(CardHeaderStatus || {});
var statusVariants = cva("text-sm", {
  variants: {
    status: {
      ["None" /* None */]: "text-sapphire-text-tertiary",
      ["Positive" /* Positive */]: "text-sapphire-positive",
      ["Negative" /* Negative */]: "text-sapphire-negative",
      ["Critical" /* Critical */]: "text-sapphire-warning",
      ["Information" /* Information */]: "text-sapphire-info"
    }
  },
  defaultVariants: {
    status: "None" /* None */
  }
});
var CardHeader = React.forwardRef(
  ({
    titleText,
    subtitleText,
    additionalText,
    interactive = false,
    avatar,
    action,
    status = "None" /* None */,
    ariaLevel = 3,
    accessibleName,
    accessibleNameRef,
    accessibleDescriptionRef,
    onClick,
    className,
    style,
    id,
    "data-testid": dataTestId
  }, ref) => {
    const headerRef = useRef(null);
    const interactiveRef = useRef(null);
    const [isPressed, setIsPressed] = useState(false);
    useImperativeHandle(
      ref,
      () => {
        const target = interactive ? interactiveRef.current : headerRef.current;
        return {
          focus() {
            target?.focus();
          },
          blur() {
            target?.blur();
          },
          isFocused() {
            if (interactive) {
              return document.activeElement === interactiveRef.current || headerRef.current?.contains(document.activeElement) === true;
            }
            return document.activeElement === headerRef.current;
          },
          get nativeElement() {
            return headerRef.current;
          }
        };
      },
      [interactive]
    );
    const handleClick = useCallback(
      (e) => {
        if (!interactive) return;
        onClick?.({ originalEvent: e, isKeyboard: false });
      },
      [interactive, onClick]
    );
    const handleKeyDown = useCallback(
      (e) => {
        if (!interactive) return;
        if (e.key === "Enter" || e.key === " ") {
          e.preventDefault();
          setIsPressed(true);
          onClick?.({ originalEvent: e, isKeyboard: true });
        }
      },
      [interactive, onClick]
    );
    const handleKeyUp = useCallback((e) => {
      if (e.key === "Enter" || e.key === " ") {
        setIsPressed(false);
      }
    }, []);
    const renderTitle = (suppressHeading) => {
      if (!titleText) return null;
      return /* @__PURE__ */ jsx(
        "div",
        {
          role: suppressHeading ? void 0 : "heading",
          "aria-level": suppressHeading ? void 0 : ariaLevel,
          className: "text-base font-semibold text-sapphire-text-primary leading-[22px] line-clamp-3 flex-1 min-w-0",
          "data-part": "title",
          children: titleText
        }
      );
    };
    const renderContent = () => /* @__PURE__ */ jsxs("div", { className: "flex-1 min-w-0", children: [
      (titleText || additionalText) && /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2", children: [
        renderTitle(interactive),
        additionalText && /* @__PURE__ */ jsx(
          "span",
          {
            className: cn("flex-shrink-0", statusVariants({ status })),
            "data-part": "additional-text",
            children: additionalText
          }
        )
      ] }),
      subtitleText && /* @__PURE__ */ jsx("p", { className: "text-sm leading-5 text-sapphire-text-secondary line-clamp-2 mt-1", "data-part": "subtitle", children: subtitleText })
    ] });
    return /* @__PURE__ */ jsxs(
      "div",
      {
        ref: headerRef,
        id,
        className: cn(
          "flex items-center gap-4 p-4 border-b border-sapphire-border-primary rounded-t-[8px]",
          interactive && [
            "hover:bg-sapphire-background-secondary/50 transition-colors",
            isPressed && "bg-sapphire-background-secondary/70"
            // No focus ring on header when interactive - focus shows on card border
          ],
          className
        ),
        style,
        "data-testid": dataTestId,
        "data-part": "root",
        children: [
          /* @__PURE__ */ jsxs(
            "div",
            {
              ref: interactive ? interactiveRef : void 0,
              role: interactive ? "button" : void 0,
              tabIndex: interactive ? 0 : void 0,
              "aria-roledescription": interactive ? "Interactive Card Header" : void 0,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-describedby": accessibleDescriptionRef,
              onClick: interactive ? handleClick : void 0,
              onKeyDown: interactive ? handleKeyDown : void 0,
              onKeyUp: interactive ? handleKeyUp : void 0,
              onMouseDown: interactive ? () => setIsPressed(true) : void 0,
              onMouseUp: interactive ? () => setIsPressed(false) : void 0,
              onMouseLeave: interactive && isPressed ? () => setIsPressed(false) : void 0,
              className: cn(
                "flex items-center gap-3 flex-1 min-w-0",
                interactive && "cursor-pointer focus:outline-none rounded-[4px]"
              ),
              "data-part": interactive ? "interactive-area" : "content-area",
              children: [
                avatar && /* @__PURE__ */ jsx("div", { className: "flex-shrink-0", "data-part": "avatar", children: avatar }),
                renderContent()
              ]
            }
          ),
          action && /* @__PURE__ */ jsx(
            "div",
            {
              className: "flex-shrink-0 z-10",
              "data-part": "action",
              children: action
            }
          )
        ]
      }
    );
  }
);
CardHeader.displayName = "CardHeader";
function CardFooter({ children, className, style, "data-testid": dataTestId, ref }) {
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn(
        "flex items-center justify-end gap-4 p-4 pt-3 border-t border-sapphire-border-primary",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "footer",
      children
    }
  );
}
function CardContent({ children, className, style, "data-testid": dataTestId, ref }) {
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("p-4", className),
      style,
      "data-testid": dataTestId,
      "data-part": "content",
      children
    }
  );
}

export { Card, CardContent, CardFooter, CardHeader, CardHeaderStatus };
