// @sap-ui/fx-components/dist/chunk-GC7VJCBJ.js
import { DateTimeIcon } from './chunk-H4UBXTRR.js';
import { ToggleButton } from './chunk-2DPLFZUY.js';
import { useEnsureCldr, getLocaleDateFormat, Calendar } from './chunk-PXVVADOR.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cva, cn } from './chunk-2F2PZMJ5.js';
import React, { useId, useState, useEffect, useMemo, useCallback, useRef, useImperativeHandle } from 'react';
import DateFormat from '@ui5/webcomponents-localization/dist/DateFormat.js';
import Locale from '@ui5/webcomponents-base/dist/locale/Locale.js';
import { useTranslation } from 'react-i18next';
import * as chrono from 'chrono-node';
import { jsxs, jsx } from 'react/jsx-runtime';

// src/types/datepicker.ts
var DatePickerSize = /* @__PURE__ */ ((DatePickerSize2) => {
  DatePickerSize2["Large"] = "Large";
  DatePickerSize2["Medium"] = "Medium";
  return DatePickerSize2;
})(DatePickerSize || {});
var CalendarType = /* @__PURE__ */ ((CalendarType2) => {
  CalendarType2["Gregorian"] = "Gregorian";
  CalendarType2["Islamic"] = "Islamic";
  CalendarType2["Japanese"] = "Japanese";
  CalendarType2["Buddhist"] = "Buddhist";
  CalendarType2["Persian"] = "Persian";
  return CalendarType2;
})(CalendarType || {});
var DatePickerSelectionMode = /* @__PURE__ */ ((DatePickerSelectionMode2) => {
  DatePickerSelectionMode2["Single"] = "Single";
  DatePickerSelectionMode2["Range"] = "Range";
  return DatePickerSelectionMode2;
})(DatePickerSelectionMode || {});
function isValidDate(d) {
  return d instanceof Date && !isNaN(d.getTime());
}
function getAccessibleLabelText(label) {
  const parts = [];
  const walk = (node) => {
    if (node.nodeType === Node.TEXT_NODE) {
      parts.push(node.textContent ?? "");
      return;
    }
    if (node instanceof HTMLElement) {
      if (node.getAttribute("aria-hidden") === "true") return;
      if (node.classList.contains("sr-only")) return;
    }
    node.childNodes.forEach((child) => walk(child));
  };
  walk(label);
  return parts.join("").trim();
}
var inputContainerVariants = cva(
  [
    "flex items-center border bg-card",
    "overflow-clip transition-colors relative",
    "hover:border-sapphire-border-accent"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: [
          "h-10 rounded-lg",
          "pl-3.5 pr-1 py-1",
          "focus-within:border-2 focus-within:pl-[13px] focus-within:pr-[3px] focus-within:py-[3px]"
        ],
        ["Medium" /* Medium */]: [
          "h-8 rounded",
          "pl-3 pr-1 py-1",
          "focus-within:border-2 focus-within:pl-[11px] focus-within:pr-[3px] focus-within:py-[3px]"
        ]
      },
      valueState: {
        ["None" /* None */]: "border-sapphire-border-active focus-within:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "border-sapphire-positive focus-within:border-sapphire-positive hover:border-sapphire-positive",
        ["Negative" /* Negative */]: "border-sapphire-negative focus-within:border-sapphire-negative hover:border-sapphire-negative",
        ["Critical" /* Critical */]: "border-sapphire-warning focus-within:border-sapphire-warning hover:border-sapphire-warning",
        ["Information" /* Information */]: "border-sapphire-info focus-within:border-sapphire-info hover:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 pointer-events-none",
        false: "hover:border-sapphire-border-accent"
      },
      readonly: {
        true: "bg-sapphire-bg-secondary",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      readonly: false
    }
  }
);
var valueStateTextColorMap = {
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Information" /* Information */]: "text-sapphire-info"
};
function DatePicker({
  size = "Large" /* Large */,
  selectionMode = "Single" /* Single */,
  delimiter = " - ",
  value: controlledValue,
  defaultValue = "",
  minDate,
  maxDate,
  disabled = false,
  readonly = false,
  required = false,
  open: controlledOpen,
  showClearIcon = true,
  placeholder,
  displayFormat,
  valueFormat = "yyyy-MM-dd",
  valueState = "None" /* None */,
  valueStateMessage,
  accessibleName,
  accessibleNameRef,
  hideWeekNumbers = false,
  calendarWeekNumbering,
  firstDayOfWeek,
  name,
  id,
  onInput,
  onChange,
  onOpen,
  onClose,
  onValueStateChange,
  className,
  style,
  "data-testid": dataTestId,
  ref
}) {
  const { t, i18n } = useTranslation("fx");
  const locale = i18n.language;
  const cldrReady = useEnsureCldr(locale);
  const valueStateDescId = useId();
  const [labelText, setLabelText] = useState(void 0);
  useEffect(() => {
    if (accessibleName) {
      setLabelText(void 0);
      return;
    }
    if (!id) {
      setLabelText(void 0);
      return;
    }
    let labelObserver = null;
    const readLabel = () => {
      const input = document.getElementById(id);
      const firstLabel = input?.labels?.[0] ?? null;
      const next = firstLabel ? getAccessibleLabelText(firstLabel) || void 0 : void 0;
      setLabelText((prev) => prev === next ? prev : next);
      if (firstLabel && !labelObserver) {
        labelObserver = new MutationObserver(readLabel);
        labelObserver.observe(firstLabel, { childList: true, characterData: true, subtree: true });
      } else if (!firstLabel && labelObserver) {
        labelObserver.disconnect();
        labelObserver = null;
      }
    };
    readLabel();
    const container = rootRef.current?.parentElement;
    const containerObserver = container ? new MutationObserver(readLabel) : null;
    containerObserver?.observe(container, { childList: true });
    return () => {
      containerObserver?.disconnect();
      labelObserver?.disconnect();
    };
  }, [accessibleName, id]);
  const effectiveAccessibleName = accessibleName ?? labelText ?? "";
  const isRangeMode = selectionMode === "Range" /* Range */;
  const isMedium = size === "Medium" /* Medium */;
  const effectiveDisplayFormat = useMemo(() => {
    if (displayFormat) return displayFormat;
    if (!cldrReady) return "yyyy-MM-dd";
    return getLocaleDateFormat(locale);
  }, [displayFormat, cldrReady, locale]);
  const loc = useMemo(() => new Locale(locale), [locale]);
  const { displayFmt, valueFmt, isoFmt } = useMemo(() => {
    if (!cldrReady) return { displayFmt: null, valueFmt: null, isoFmt: null };
    return {
      displayFmt: DateFormat.getDateInstance({ pattern: effectiveDisplayFormat }, loc),
      valueFmt: DateFormat.getDateInstance({ pattern: valueFormat }, loc),
      isoFmt: DateFormat.getDateInstance({ pattern: "yyyy-MM-dd" }, loc)
    };
  }, [cldrReady, effectiveDisplayFormat, valueFormat, loc]);
  const formatDate = useCallback((d) => displayFmt?.format(d) ?? "", [displayFmt]);
  const formatDateValue = useCallback((d) => d && valueFmt ? valueFmt.format(d) : "", [valueFmt]);
  const formatISO = useCallback((d) => isoFmt?.format(d) ?? "", [isoFmt]);
  const parseDate2 = useCallback((str, fmt) => fmt ? fmt.parse(str) ?? null : null, []);
  const parseDatePattern = useCallback((str, pattern) => {
    if (!cldrReady) return null;
    return parseDate2(str, DateFormat.getDateInstance({ pattern, strictParsing: true }, loc));
  }, [parseDate2, cldrReady, loc]);
  const getPlaceholder = () => {
    if (placeholder) return placeholder;
    if (!cldrReady && !displayFormat) return "";
    if (isRangeMode) return `${effectiveDisplayFormat}${delimiter}${effectiveDisplayFormat}`;
    return effectiveDisplayFormat;
  };
  const effectivePlaceholder = getPlaceholder();
  const [internalValue, setInternalValue] = useState(defaultValue);
  const [internalOpen, setInternalOpen] = useState(false);
  const [inputText, setInputText] = useState(defaultValue);
  const [isEditing, setIsEditing] = useState(false);
  const [internalValueState, setInternalValueState] = useState(valueState);
  const isOpen = controlledOpen ?? internalOpen;
  const value = controlledValue ?? internalValue;
  const selectedDate = !isRangeMode && cldrReady && value ? parseDatePattern(value, valueFormat) ?? void 0 : void 0;
  const rangeParts = isRangeMode && cldrReady && value ? value.split(delimiter) : [];
  const rangeStartDate = rangeParts[0] ? parseDatePattern(rangeParts[0].trim(), valueFormat) ?? void 0 : void 0;
  const rangeEndDate = rangeParts[1] ? parseDatePattern(rangeParts[1].trim(), valueFormat) ?? void 0 : void 0;
  const containerRef = useRef(null);
  const inputRef = useRef(null);
  const rootRef = useRef(null);
  const skipFocusSyncRef = useRef(false);
  const minDateObj = React.useMemo(
    () => minDate ? parseDatePattern(minDate, "yyyy-MM-dd") : void 0,
    [minDate, parseDatePattern]
  );
  const maxDateObj = React.useMemo(
    () => maxDate ? parseDatePattern(maxDate, "yyyy-MM-dd") : void 0,
    [maxDate, parseDatePattern]
  );
  const updateValueState = useCallback((newState, valid) => {
    if (onValueStateChange) {
      const result = onValueStateChange({ valueState: newState, valid });
      if (result === false) return;
    }
    setInternalValueState(newState);
  }, [onValueStateChange]);
  const openPopover = useCallback(() => {
    if (disabled || readonly) return;
    if (controlledOpen === void 0) {
      setInternalOpen(true);
    }
    onOpen?.();
  }, [disabled, readonly, controlledOpen, onOpen]);
  const closePopover = useCallback(() => {
    if (controlledOpen === void 0) {
      setInternalOpen(false);
    }
    onClose?.();
  }, [controlledOpen, onClose]);
  const getDisplayValue = () => {
    if (!cldrReady) return "";
    if (isEditing) return inputText;
    if (isRangeMode && isValidDate(rangeStartDate)) {
      if (isValidDate(rangeEndDate)) {
        return `${formatDate(rangeStartDate)}${delimiter}${formatDate(rangeEndDate)}`;
      }
      return formatDate(rangeStartDate);
    }
    if (!isRangeMode && isValidDate(selectedDate)) {
      return formatDate(selectedDate);
    }
    return value;
  };
  const displayValue = getDisplayValue();
  const parseDateFromInput = useCallback((text) => {
    if (!text || !text.trim()) return null;
    const parsed = parseDatePattern(text.trim(), effectiveDisplayFormat);
    if (isValidDate(parsed)) return parsed;
    try {
      const chronoResult = chrono.parseDate(text.trim());
      if (chronoResult && isValidDate(chronoResult)) return chronoResult;
    } catch {
    }
    const commonFormats = [
      "yyyy-MM-dd",
      "MM/dd/yyyy",
      "dd/MM/yyyy",
      "dd.MM.yyyy",
      "yyyy/MM/dd",
      "MMMM d, yyyy",
      "MMM d, yyyy",
      "d MMMM yyyy",
      "d MMM yyyy",
      "MM-dd-yyyy",
      "dd-MM-yyyy"
    ];
    for (const fmt of commonFormats) {
      const fallback = parseDatePattern(text.trim(), fmt);
      if (isValidDate(fallback)) return fallback;
    }
    return null;
  }, [effectiveDisplayFormat, parseDatePattern]);
  const parseRangeFromInput = useCallback((text) => {
    if (!text || !text.trim()) {
      return { start: null, end: null };
    }
    const trimmedText = text.trim();
    const delimiters = [
      delimiter,
      // User's specified delimiter first
      " - ",
      " to ",
      " \u2013 ",
      // en dash
      " \u2014 ",
      // em dash
      " TO ",
      "-",
      " through ",
      " thru ",
      "..",
      " .. ",
      " until ",
      " till "
    ];
    for (const delim of delimiters) {
      if (trimmedText.includes(delim)) {
        const parts = trimmedText.split(delim).map((p) => p.trim());
        if (parts.length === 2 && parts[0] && parts[1]) {
          const start = parseDateFromInput(parts[0]);
          const end = parseDateFromInput(parts[1]);
          if (start && end) {
            return { start, end };
          } else if (start || end) {
            return { start, end };
          }
        }
      }
    }
    try {
      const chronoResults = chrono.parse(trimmedText);
      if (chronoResults.length >= 2) {
        const start = chronoResults[0].start.date();
        const end = chronoResults[1].start.date();
        if (isValidDate(start) && isValidDate(end)) {
          return { start, end };
        }
      } else if (chronoResults.length === 1 && chronoResults[0].end) {
        const start = chronoResults[0].start.date();
        const end = chronoResults[0].end.date();
        if (isValidDate(start) && isValidDate(end)) {
          return { start, end };
        }
      } else if (chronoResults.length === 1) {
        const date = chronoResults[0].start.date();
        if (isValidDate(date)) {
          return { start: date, end: null };
        }
      }
    } catch {
    }
    const singleDate = parseDateFromInput(trimmedText);
    return { start: singleDate, end: null };
  }, [parseDateFromInput, delimiter]);
  const handleFocus = useCallback(() => {
    if (skipFocusSyncRef.current) {
      skipFocusSyncRef.current = false;
    } else {
      setInputText(displayValue);
    }
    setIsEditing(true);
  }, [displayValue]);
  const handleInputChange = useCallback(
    (e) => {
      const newText = e.target.value;
      setInputText(newText);
      if (isRangeMode) {
        const { start, end } = parseRangeFromInput(newText);
        const valid = start !== null && end !== null;
        onInput?.({
          value: newText,
          startDateValue: start,
          endDateValue: end,
          valid
        });
      } else {
        const dateValue = parseDateFromInput(newText);
        const valid = dateValue !== null;
        onInput?.({
          value: newText,
          dateValue,
          valid
        });
      }
    },
    [parseDateFromInput, parseRangeFromInput, onInput, isRangeMode]
  );
  const handleDateSelect = useCallback(
    (date) => {
      if (isRangeMode) {
        if (!rangeStartDate || rangeStartDate && rangeEndDate) {
          const formattedValue = formatDateValue(date);
          updateValueState(valueState, false);
          if (!controlledValue) {
            setInternalValue(formattedValue);
          }
          onChange?.({
            value: formattedValue,
            startDateValue: date,
            endDateValue: null,
            valid: false
            // Range not complete yet
          });
        } else {
          const start = date < rangeStartDate ? date : rangeStartDate;
          const end = date < rangeStartDate ? rangeStartDate : date;
          const formattedValue = `${formatDateValue(start)}${delimiter}${formatDateValue(end)}`;
          updateValueState(valueState, true);
          if (!controlledValue) {
            setInternalValue(formattedValue);
          }
          onChange?.({
            value: formattedValue,
            startDateValue: start,
            endDateValue: end,
            valid: true
          });
          closePopover();
        }
      } else {
        const formattedValue = formatDateValue(date);
        updateValueState(valueState, true);
        if (!controlledValue) {
          setInternalValue(formattedValue);
        }
        onChange?.({
          value: formattedValue,
          dateValue: date,
          valid: true
        });
        closePopover();
      }
    },
    [
      formatDateValue,
      controlledValue,
      valueState,
      onChange,
      closePopover,
      updateValueState,
      isRangeMode,
      rangeStartDate,
      rangeEndDate,
      delimiter
    ]
  );
  const handleBlur = useCallback(() => {
    setIsEditing(false);
    if (isRangeMode) {
      const { start: startDate, end: endDate } = parseRangeFromInput(inputText);
      if (startDate && isValidDate(startDate) && endDate && isValidDate(endDate)) {
        const formattedValue = `${formatDateValue(startDate)}${delimiter}${formatDateValue(endDate)}`;
        updateValueState(valueState, true);
        if (!controlledValue) {
          setInternalValue(formattedValue);
        }
        onChange?.({
          value: formattedValue,
          startDateValue: startDate,
          endDateValue: endDate,
          valid: true
        });
      } else if (inputText.trim()) {
        updateValueState("Negative" /* Negative */, false);
        onChange?.({
          value: inputText,
          startDateValue: startDate,
          endDateValue: endDate,
          valid: false
        });
      } else {
        updateValueState(valueState, true);
        if (!controlledValue) {
          setInternalValue("");
        }
        onChange?.({
          value: "",
          startDateValue: null,
          endDateValue: null,
          valid: true
        });
      }
    } else {
      const dateValue = parseDateFromInput(inputText);
      if (dateValue && isValidDate(dateValue)) {
        const formattedValue = formatDateValue(dateValue);
        updateValueState(valueState, true);
        if (!controlledValue) {
          setInternalValue(formattedValue);
        }
        onChange?.({
          value: formattedValue,
          dateValue,
          valid: true
        });
      } else if (inputText.trim()) {
        updateValueState("Negative" /* Negative */, false);
        onChange?.({
          value: inputText,
          dateValue: null,
          valid: false
        });
      } else {
        updateValueState(valueState, true);
        if (!controlledValue) {
          setInternalValue("");
        }
        onChange?.({
          value: "",
          dateValue: null,
          valid: true
        });
      }
    }
  }, [
    inputText,
    parseDateFromInput,
    parseRangeFromInput,
    formatDateValue,
    controlledValue,
    valueState,
    onChange,
    updateValueState,
    isRangeMode,
    delimiter
  ]);
  const togglePopover = useCallback(() => {
    if (disabled || readonly) return;
    if (isOpen) {
      closePopover();
    } else {
      openPopover();
    }
  }, [disabled, readonly, isOpen, openPopover, closePopover]);
  const handleClear = useCallback(
    () => {
      setInputText("");
      skipFocusSyncRef.current = true;
      updateValueState(valueState, true);
      if (!controlledValue) {
        setInternalValue("");
      }
      if (isRangeMode) {
        onChange?.({
          value: "",
          startDateValue: null,
          endDateValue: null,
          valid: true
        });
      } else {
        onChange?.({
          value: "",
          dateValue: null,
          valid: true
        });
      }
      inputRef.current?.focus();
    },
    [controlledValue, valueState, onChange, updateValueState, isRangeMode]
  );
  const handleKeyDown = useCallback((e) => {
    if (e.key === "Enter") {
      e.preventDefault();
      handleBlur();
      setIsEditing(true);
      return;
    }
    if (e.key === "F4") {
      e.preventDefault();
      if (!isOpen) {
        openPopover();
      }
      return;
    }
    if (e.altKey && (e.key === "ArrowDown" || e.key === "ArrowUp")) {
      e.preventDefault();
      if (isOpen) {
        closePopover();
      } else {
        openPopover();
      }
      return;
    }
    if (e.key === "Escape") {
      e.preventDefault();
      if (isOpen) {
        closePopover();
      } else if (showClearIcon && displayValue && !disabled && !readonly) {
        handleClear();
      }
      return;
    }
    if (!isOpen && !isRangeMode && (e.key === "PageUp" || e.key === "PageDown")) {
      e.preventDefault();
      const currentDate = selectedDate || /* @__PURE__ */ new Date();
      const direction = e.key === "PageUp" ? 1 : -1;
      let newDate;
      if (e.ctrlKey && e.shiftKey) {
        const d = new Date(currentDate);
        d.setFullYear(d.getFullYear() + direction);
        newDate = d;
      } else if (e.shiftKey) {
        const d = new Date(currentDate);
        d.setMonth(d.getMonth() + direction);
        newDate = d;
      } else {
        const d = new Date(currentDate);
        d.setDate(d.getDate() + direction);
        newDate = d;
      }
      if (minDateObj && newDate < minDateObj) return;
      if (maxDateObj && newDate > maxDateObj) return;
      const formattedValue = formatDateValue(newDate);
      const formattedDisplay = formatDate(newDate);
      updateValueState(valueState, true);
      if (!controlledValue) {
        setInternalValue(formattedValue);
      }
      setInputText(formattedDisplay);
      onChange?.({
        value: formattedValue,
        dateValue: newDate,
        valid: true
      });
      return;
    }
  }, [
    handleBlur,
    isOpen,
    openPopover,
    closePopover,
    isRangeMode,
    selectedDate,
    minDateObj,
    maxDateObj,
    controlledValue,
    valueState,
    onChange,
    updateValueState,
    formatDateValue,
    formatDate,
    showClearIcon,
    displayValue,
    disabled,
    readonly,
    handleClear
  ]);
  const isInValidRange = useCallback((val) => {
    const parsed = parseDatePattern(val, valueFormat);
    if (!isValidDate(parsed)) return false;
    if (minDateObj && parsed < minDateObj) return false;
    if (maxDateObj && parsed > maxDateObj) return false;
    return true;
  }, [valueFormat, minDateObj, maxDateObj, parseDatePattern]);
  const isValidValue = useCallback((val) => {
    return isValidDate(parseDatePattern(val, valueFormat));
  }, [valueFormat, parseDatePattern]);
  useImperativeHandle(ref, () => ({
    formatValue(date) {
      return formatDate(date);
    },
    isValidValue(val) {
      return isValidValue(val);
    },
    isInValidRange(val) {
      return isInValidRange(val);
    },
    isValidDisplayValue(val) {
      return isValidDate(parseDatePattern(val, effectiveDisplayFormat));
    },
    isValidMin() {
      if (!minDate) return true;
      return isValidDate(parseDatePattern(minDate, valueFormat));
    },
    isValidMax() {
      if (!maxDate) return true;
      return isValidDate(parseDatePattern(maxDate, valueFormat));
    },
    get dateValue() {
      return selectedDate && isValidDate(selectedDate) ? selectedDate : null;
    },
    get dateValueUTC() {
      if (!selectedDate || !isValidDate(selectedDate)) return null;
      return new Date(Date.UTC(
        selectedDate.getFullYear(),
        selectedDate.getMonth(),
        selectedDate.getDate()
      ));
    },
    get startDateValue() {
      if (!isRangeMode) return null;
      return rangeStartDate && isValidDate(rangeStartDate) ? rangeStartDate : null;
    },
    get endDateValue() {
      if (!isRangeMode) return null;
      return rangeEndDate && isValidDate(rangeEndDate) ? rangeEndDate : null;
    },
    get startValue() {
      if (!isRangeMode || !rangeStartDate || !isValidDate(rangeStartDate)) return "";
      return formatDate(rangeStartDate);
    },
    get endValue() {
      if (!isRangeMode || !rangeEndDate || !isValidDate(rangeEndDate)) return "";
      return formatDate(rangeEndDate);
    },
    get nativeElement() {
      return rootRef.current;
    },
    get formValidity() {
      const currentValue = controlledValue ?? internalValue;
      const parsed = currentValue ? parseDatePattern(currentValue, valueFormat) : null;
      const isValidDate_ = parsed && isValidDate(parsed);
      const valueMissing = required && !currentValue;
      const patternMismatch = !!currentValue && !isValidDate_;
      const rangeUnderflow = isValidDate_ && minDateObj ? parsed < minDateObj : false;
      const rangeOverflow = isValidDate_ && maxDateObj ? parsed > maxDateObj : false;
      return {
        valueMissing,
        patternMismatch,
        rangeUnderflow,
        rangeOverflow,
        valid: !valueMissing && !patternMismatch && !rangeUnderflow && !rangeOverflow
      };
    },
    focus() {
      inputRef.current?.focus();
    },
    blur() {
      inputRef.current?.blur();
    }
  }), [
    effectiveDisplayFormat,
    selectedDate,
    rangeStartDate,
    rangeEndDate,
    isRangeMode,
    isValidValue,
    isInValidRange,
    controlledValue,
    internalValue,
    valueFormat,
    minDate,
    maxDate,
    required,
    minDateObj,
    maxDateObj,
    formatDate,
    parseDatePattern
  ]);
  return /* @__PURE__ */ jsxs("div", { ref: rootRef, className: cn("relative", className), style, children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        ref: containerRef,
        className: cn(
          inputContainerVariants({
            size,
            valueState: internalValueState,
            disabled,
            readonly
          }),
          isOpen && !disabled && !readonly && internalValueState === "None" /* None */ && "border-2 border-sapphire-border-accent",
          isOpen && !disabled && !readonly && internalValueState === "None" /* None */ && (isMedium ? "pl-[11px] pr-[3px] py-[3px]" : "pl-[13px] pr-[3px] py-[3px]")
        ),
        "data-testid": dataTestId,
        children: [
          /* @__PURE__ */ jsx(
            "input",
            {
              ref: inputRef,
              id,
              type: "text",
              value: displayValue,
              onChange: handleInputChange,
              onFocus: handleFocus,
              onBlur: handleBlur,
              onKeyDown: handleKeyDown,
              disabled,
              readOnly: readonly,
              required,
              placeholder: effectivePlaceholder,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-haspopup": "grid",
              "aria-roledescription": isRangeMode ? t("DATERANGE_DESCRIPTION") : t("DATEPICKER_DATE_DESCRIPTION"),
              "aria-required": required ? "true" : void 0,
              "aria-describedby": valueStateMessage || internalValueState === "Negative" /* Negative */ ? valueStateDescId : void 0,
              className: cn(
                "flex-1 min-w-0 bg-transparent text-sm outline-none",
                "placeholder:text-sapphire-text-tertiary",
                "disabled:cursor-not-allowed disabled:opacity-50"
              )
            }
          ),
          showClearIcon && displayValue && !disabled && !readonly && /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
            Button,
            {
              design: "SecondaryNeutral" /* SecondaryNeutral */,
              size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
              iconOnly: true,
              icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-full w-full" }) }),
              tabIndex: -1,
              onClick: handleClear,
              accessibleName: t("INPUT_CLEAR_ICON_ACC_NAME"),
              className: "shrink-0"
            }
          ) }),
          /* @__PURE__ */ jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsx(
            ToggleButton,
            {
              design: "SecondaryNeutral" /* SecondaryNeutral */,
              size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
              iconOnly: true,
              icon: /* @__PURE__ */ jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsx(DateTimeIcon, { className: "h-full w-full" }) }),
              pressed: isOpen,
              tabIndex: -1,
              onClick: togglePopover,
              disabled: disabled || readonly,
              accessibleName: isOpen ? t("DATEPICKER_OPEN_ICON_TITLE_OPENED") : t("DATEPICKER_OPEN_ICON_TITLE"),
              className: "shrink-0 active:!bg-sapphire-brand-toggle-background"
            }
          ) })
        ]
      }
    ),
    (valueStateMessage || internalValueState === "Negative" /* Negative */) && /* @__PURE__ */ jsx(
      "div",
      {
        id: valueStateDescId,
        className: cn("mt-2 text-sm", valueStateTextColorMap[internalValueState]),
        role: "alert",
        children: valueStateMessage || internalValueState === "Negative" /* Negative */ && "Invalid date format"
      }
    ),
    /* @__PURE__ */ jsx(
      ResponsivePopover,
      {
        open: isOpen,
        opener: containerRef,
        placement: "Bottom",
        horizontalAlign: "Start",
        hideArrow: true,
        noPadding: true,
        headerText: "Select Date",
        preventInitialFocus: true,
        accessibleName: isRangeMode ? t("DATERANGEPICKER_POPOVER_ACCESSIBLE_NAME", { "0": effectiveAccessibleName }) : t("DATEPICKER_POPOVER_ACCESSIBLE_NAME", { "0": effectiveAccessibleName }),
        contentOnlyOnDesktop: true,
        onClose: closePopover,
        children: /* @__PURE__ */ jsx(
          Calendar,
          {
            selectionMode: isRangeMode ? "Range" /* Range */ : "Single" /* Single */,
            selectedDates: !isRangeMode && selectedDate ? [formatISO(selectedDate)] : [],
            rangeStartDate: isRangeMode && rangeStartDate ? formatISO(rangeStartDate) : void 0,
            rangeEndDate: isRangeMode && rangeEndDate ? formatISO(rangeEndDate) : void 0,
            autoFocus: true,
            onSelectionChange: (detail) => {
              if (isRangeMode) {
                if (detail.selectedValues.length === 2) {
                  const start = parseDatePattern(detail.selectedValues[0], "yyyy-MM-dd");
                  const end = parseDatePattern(detail.selectedValues[1], "yyyy-MM-dd");
                  if (isValidDate(start) && isValidDate(end)) {
                    const formattedValue = `${formatDateValue(start)}${delimiter}${formatDateValue(end)}`;
                    updateValueState(valueState, true);
                    if (!controlledValue) {
                      setInternalValue(formattedValue);
                    }
                    onChange?.({
                      value: formattedValue,
                      startDateValue: start,
                      endDateValue: end,
                      valid: true
                    });
                    closePopover();
                  }
                } else if (detail.selectedValues.length === 1) {
                  const date = parseDatePattern(detail.selectedValues[0], "yyyy-MM-dd");
                  if (isValidDate(date)) {
                    handleDateSelect(date);
                  }
                }
              } else {
                if (detail.selectedValues.length > 0) {
                  const parsed = parseDatePattern(detail.selectedValues[0], "yyyy-MM-dd");
                  if (isValidDate(parsed)) handleDateSelect(parsed);
                }
              }
            },
            minDate: minDateObj ? formatISO(minDateObj) : void 0,
            maxDate: maxDateObj ? formatISO(maxDateObj) : void 0,
            hideWeekNumbers,
            calendarWeekNumbering,
            firstDayOfWeek,
            locale,
            className: "border-0 shadow-none rounded-none"
          }
        )
      }
    ),
    name && /* @__PURE__ */ jsx(
      "input",
      {
        type: "hidden",
        name,
        value
      }
    )
  ] });
}

export { CalendarType, DatePicker, DatePickerSelectionMode, DatePickerSize, inputContainerVariants };
