// @sap-ui/fx-components/dist/chunk-IS37ZOQI.js
import { SlimArrowRightIcon } from './chunk-ZQSLIXR7.js';
import { ListItemBase, LIST_ITEM_BASE_CLASSES, LIST_ITEM_FOCUS_CLASSES } from './chunk-CTYK4HEK.js';
import { useOptionalListContext } from './chunk-CJXZUSUP.js';
import { SlimArrowDownIcon } from './chunk-TO7N3UGU.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import * as React from 'react';
import { createContext, useRef, useId, useState, useEffect, useCallback, useContext, useMemo } from 'react';

function ConversationListItem({
  itemKey,
  content,
  selected = false,
  bold = false,
  onClick,
  actions,
  className,
  ref
}) {
  const isStringContent = typeof content === "string";
  return /* @__PURE__ */ jsx(
    ListItemBase,
    {
      ref,
      itemKey,
      type: "Active" /* Active */,
      selected,
      onClick,
      className: cn(
        "group px-4 py-0",
        selected && "!bg-sapphire-brand-background/10",
        className
      ),
      children: ({ isFocused, isFocusedWithin }) => /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2 w-full min-w-0", children: [
        /* @__PURE__ */ jsx("div", { className: "flex-1 min-w-0", children: /* @__PURE__ */ jsx("div", { className: cn(
          "text-sm",
          isStringContent && "truncate",
          selected ? "font-semibold text-sapphire-brand-foreground" : bold ? "font-bold text-sapphire-text-primary group-hover:text-sapphire-brand-foreground" : "font-normal text-sapphire-text-primary group-hover:text-sapphire-brand-foreground"
        ), children: content }) }),
        actions && /* @__PURE__ */ jsx("div", { className: cn(
          "shrink-0 flex items-center gap-1 transition-opacity",
          selected || isFocused || isFocusedWithin ? "visible opacity-100 relative" : "invisible opacity-0 absolute group-hover:visible group-hover:opacity-100 group-hover:relative"
        ), children: actions })
      ] })
    }
  );
}
function ListItemGroup(props) {
  const {
    ref,
    // Content
    headerText,
    headerAccessibleName,
    header,
    children,
    // Behavior
    wrappingType = "None" /* None */,
    stickyHeader = false,
    collapsible = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    onToggle,
    // Standard
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const groupRef = useRef(null);
  const headerRef = useRef(null);
  const generatedId = useId();
  const headerId = `${generatedId}-header`;
  const contentId = `${generatedId}-content`;
  const listContext = useOptionalListContext();
  const [internalCollapsed, setInternalCollapsed] = useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const [isFocused, setIsFocused] = useState(false);
  const headerIndexRef = useRef(-1);
  useEffect(() => {
    if (listContext && headerRef.current) {
      let rootList = groupRef.current?.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(headerRef.current);
        if (index >= 0) {
          headerIndexRef.current = index;
          listContext.registerNavigationItem?.(index, headerRef.current);
        }
      }
    }
    return () => {
      if (listContext && headerIndexRef.current >= 0) {
        listContext.registerNavigationItem?.(headerIndexRef.current, null);
        headerIndexRef.current = -1;
      }
    };
  }, [listContext]);
  const headerIndex = headerIndexRef.current;
  const isFocusedItem = listContext?.focusedIndex === headerIndex;
  const isFirstItem = headerIndex === 0;
  const computedTabIndex = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem ? 0 : -1;
  const handleToggle = useCallback(() => {
    if (!collapsible) return;
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.(newCollapsed);
  }, [collapsible, isCollapsed, isControlled, onToggle]);
  const handleKeyDown = useCallback((e) => {
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      return;
    }
    if (collapsible && (e.key === "Enter" || e.key === " ")) {
      e.preventDefault();
      handleToggle();
    }
    if (collapsible) {
      if (e.key === "ArrowLeft" && !isCollapsed) {
        e.preventDefault();
        handleToggle();
      } else if (e.key === "ArrowRight" && isCollapsed) {
        e.preventDefault();
        handleToggle();
      }
    }
  }, [collapsible, handleToggle, isCollapsed]);
  const shouldWrap = String(wrappingType) === "Normal";
  const hasHeader = headerText || header;
  const separatorClass = listContext?.separators !== "None" ? "border-b border-sapphire-border-primary last:border-b-0" : "";
  return /* @__PURE__ */ jsxs(
    "li",
    {
      ref: (node) => {
        groupRef.current = node;
        if (typeof ref === "function") {
          ref(node);
        } else if (ref) {
          ref.current = node;
        }
      },
      role: "group",
      "aria-labelledby": hasHeader ? headerId : void 0,
      className: cn(
        "relative list-none",
        separatorClass,
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        hasHeader && /* @__PURE__ */ jsxs(
          "div",
          {
            ref: headerRef,
            id: headerId,
            role: collapsible ? "button" : "heading",
            "aria-level": 3,
            "aria-expanded": collapsible ? !isCollapsed : void 0,
            "aria-controls": collapsible ? contentId : void 0,
            "aria-label": headerAccessibleName,
            tabIndex: computedTabIndex,
            onClick: collapsible ? handleToggle : void 0,
            onKeyDown: handleKeyDown,
            onFocus: (e) => {
              const isHeaderItself = e.target === headerRef.current;
              setIsFocused(isHeaderItself);
              if (listContext && headerIndex >= 0) {
                listContext.onFocusItem(headerIndex);
              }
            },
            onBlur: (e) => {
              const isLeavingHeader = !headerRef.current?.contains(e.relatedTarget);
              if (isLeavingHeader) {
                setIsFocused(false);
              }
            },
            "data-group-header": "true",
            className: cn(
              LIST_ITEM_BASE_CLASSES,
              "gap-2 px-4 py-2",
              "bg-muted/50 text-sm font-semibold text-sapphire-text-tertiary",
              "select-none",
              stickyHeader && "sticky top-0 z-10",
              collapsible && "cursor-pointer hover:bg-muted/70",
              isFocused && LIST_ITEM_FOCUS_CLASSES
            ),
            children: [
              collapsible && !header && /* @__PURE__ */ jsx("span", { className: "shrink-0 text-sapphire-text-tertiary", children: isCollapsed ? /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "h-4 w-4" }) : /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-4 w-4" }) }),
              header || /* @__PURE__ */ jsx(
                "span",
                {
                  className: cn(
                    "flex-1",
                    !shouldWrap && "truncate",
                    shouldWrap && "break-words"
                  ),
                  children: headerText
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ jsx(
          "ul",
          {
            id: contentId,
            role: "group",
            className: cn(
              "list-none m-0 p-0 w-full",
              collapsible && isCollapsed && "hidden"
            ),
            children
          }
        )
      ]
    }
  );
}
ListItemGroup.displayName = "ListItemGroup";
function ConversationGroupListItem({
  headerText,
  collapsible = true,
  defaultCollapsed = false,
  collapsed,
  onToggle,
  stickyHeader = false,
  onSeeAll,
  children,
  className,
  "data-testid": dataTestId,
  ref
}) {
  const [internalCollapsed, setInternalCollapsed] = React.useState(defaultCollapsed);
  const isControlled = collapsed !== void 0;
  const isCollapsed = isControlled ? collapsed : internalCollapsed;
  const content = onSeeAll ? /* @__PURE__ */ jsxs(Fragment, { children: [
    children,
    /* @__PURE__ */ jsx("li", { className: "list-none px-4 py-2", children: /* @__PURE__ */ jsx(
      Button,
      {
        design: "Tertiary" /* Tertiary */,
        onClick: onSeeAll,
        className: "text-sm font-semibold text-sapphire-text hover:text-sapphire-brand-foreground",
        children: "See All"
      }
    ) })
  ] }) : children;
  const customHeader = collapsible ? /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-between w-full gap-2", children: [
    /* @__PURE__ */ jsx("span", { className: "flex-1 truncate font-normal text-sm text-sapphire-text-tertiary", children: headerText }),
    /* @__PURE__ */ jsx(
      "span",
      {
        className: "shrink-0 flex items-center justify-center w-8 h-8 text-sapphire-text-tertiary",
        "aria-hidden": "true",
        children: isCollapsed ? /* @__PURE__ */ jsx(SlimArrowRightIcon, { className: "h-4 w-4 pointer-events-none" }) : /* @__PURE__ */ jsx(SlimArrowDownIcon, { className: "h-4 w-4 pointer-events-none" })
      }
    )
  ] }) : /* @__PURE__ */ jsx("span", { className: "font-normal text-sm text-sapphire-text-tertiary", children: headerText });
  return /* @__PURE__ */ jsx(
    ListItemGroup,
    {
      ref,
      header: customHeader,
      collapsible,
      defaultCollapsed,
      collapsed,
      onToggle: (newCollapsed) => {
        if (!isControlled) {
          setInternalCollapsed(newCollapsed);
        }
        onToggle?.(newCollapsed);
      },
      stickyHeader,
      wrappingType: "None" /* None */,
      className: cn(
        "[&>[data-group-header]]:!bg-transparent",
        "[&>[data-group-header]]:pl-2",
        "[&>[data-group-header]]:pr-0",
        "[&>[data-group-header]]:py-1.5",
        "[&>ul>li]:border-b-0",
        className
      ),
      "data-testid": dataTestId,
      children: content
    }
  );
}
function ListItemSeparator(props) {
  const {
    ref,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  return /* @__PURE__ */ jsx(
    "li",
    {
      ref,
      role: "separator",
      "aria-orientation": "horizontal",
      className: cn(
        "relative",
        "w-full",
        "h-px",
        "bg-sapphire-border-primary",
        "my-2",
        "pointer-events-none",
        className
      ),
      style,
      "data-testid": dataTestId
    }
  );
}
var ListItemGroupContext = createContext(null);
function useListItemGroup() {
  return useContext(ListItemGroupContext);
}

// src/components/list/useListGroupHeader.ts
function useListGroupHeader(options = {}) {
  const {
    collapsible = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    onToggle
  } = options;
  const headerRef = useRef(null);
  const generatedId = useId();
  const headerId = `${generatedId}-header`;
  const groupContext = useListItemGroup();
  const contentId = groupContext?.contentId || `${generatedId}-content`;
  const listContext = useOptionalListContext();
  const [internalCollapsed, setInternalCollapsed] = useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const [isFocused, setIsFocused] = useState(false);
  const headerIndexRef = useRef(-1);
  useEffect(() => {
    if (listContext && headerRef.current) {
      let rootList = headerRef.current.parentElement;
      while (rootList && rootList.getAttribute("role") !== "list" && rootList.getAttribute("role") !== "listbox") {
        rootList = rootList.parentElement;
      }
      if (rootList) {
        const allNavigable = Array.from(
          rootList.querySelectorAll(
            'div[data-group-header="true"], li[role="listitem"]'
          )
        );
        const index = allNavigable.indexOf(headerRef.current);
        if (index >= 0) {
          headerIndexRef.current = index;
          listContext.registerNavigationItem?.(index, headerRef.current);
        }
      }
    }
    return () => {
      if (listContext && headerIndexRef.current >= 0) {
        listContext.registerNavigationItem?.(headerIndexRef.current, null);
        headerIndexRef.current = -1;
      }
    };
  }, [listContext]);
  const handleToggle = useCallback(() => {
    if (!collapsible) return;
    const newCollapsed = !isCollapsed;
    if (!isControlled) {
      setInternalCollapsed(newCollapsed);
    }
    onToggle?.(newCollapsed);
  }, [collapsible, isCollapsed, isControlled, onToggle]);
  const handleKeyDown = useCallback((e) => {
    if (["ArrowDown", "ArrowUp", "Home", "End", "PageDown", "PageUp"].includes(e.key)) {
      return;
    }
    if (e.key === "Tab") {
      return;
    }
    if (collapsible && (e.key === "Enter" || e.key === " ")) {
      e.preventDefault();
      handleToggle();
    }
  }, [collapsible, handleToggle]);
  const handleFocus = useCallback((e) => {
    const isHeaderItself = e.target === headerRef.current;
    setIsFocused(isHeaderItself);
    if (listContext && headerIndexRef.current >= 0) {
      listContext.onFocusItem(headerIndexRef.current);
    }
  }, [listContext]);
  const handleBlur = useCallback((e) => {
    const isLeavingHeader = !headerRef.current?.contains(e.relatedTarget);
    if (isLeavingHeader) {
      setIsFocused(false);
    }
  }, []);
  const headerIndex = headerIndexRef.current;
  const isFocusedItem = listContext?.focusedIndex === headerIndex;
  const isFirstItem = headerIndex === 0;
  const tabIndex = isFocusedItem || listContext?.focusedIndex === -1 && isFirstItem ? 0 : -1;
  const focus = useCallback(() => {
    headerRef.current?.focus();
  }, []);
  const blur = useCallback(() => {
    headerRef.current?.blur();
  }, []);
  return {
    // Refs
    headerRef,
    headerId,
    contentId,
    // State
    isCollapsed,
    isFocused,
    // Computed values
    tabIndex,
    // Event handlers
    handleToggle,
    handleKeyDown,
    handleFocus,
    handleBlur,
    // Helper functions
    focus,
    blur
  };
}
function ListGroupHeaderBase(props) {
  const {
    ref,
    collapsible = false,
    collapsed,
    defaultCollapsed = false,
    onToggle,
    accessibleName,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const groupHeader = useListGroupHeader({
    collapsible,
    collapsed,
    defaultCollapsed,
    onToggle
  });
  const renderState = {
    isCollapsed: groupHeader.isCollapsed,
    isFocused: groupHeader.isFocused,
    collapsible
  };
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref: (node) => {
        groupHeader.headerRef.current = node;
        if (typeof ref === "function") {
          ref(node);
        } else if (ref) {
          ref.current = node;
        }
      },
      id: groupHeader.headerId,
      role: collapsible ? "button" : "heading",
      "aria-level": !collapsible ? 3 : void 0,
      "aria-expanded": collapsible ? !groupHeader.isCollapsed : void 0,
      "aria-controls": collapsible ? groupHeader.contentId : void 0,
      "aria-label": accessibleName,
      tabIndex: groupHeader.tabIndex,
      onClick: collapsible ? groupHeader.handleToggle : void 0,
      onKeyDown: groupHeader.handleKeyDown,
      onFocus: groupHeader.handleFocus,
      onBlur: groupHeader.handleBlur,
      "data-group-header": "true",
      className: cn(
        LIST_ITEM_BASE_CLASSES,
        groupHeader.isFocused && LIST_ITEM_FOCUS_CLASSES,
        className
      ),
      style,
      "data-testid": dataTestId,
      children: typeof children === "function" ? children(renderState) : children
    }
  );
}
function ListItemGroupBase(props) {
  const {
    ref,
    header,
    headerAccessibleName,
    collapsible = false,
    collapsed: controlledCollapsed,
    defaultCollapsed = false,
    onToggle,
    children,
    className,
    style,
    "data-testid": dataTestId
  } = props;
  const generatedId = useId();
  const contentId = `${generatedId}-content`;
  const [internalCollapsed, setInternalCollapsed] = useState(defaultCollapsed);
  const isControlled = controlledCollapsed !== void 0;
  const isCollapsed = isControlled ? controlledCollapsed : internalCollapsed;
  const handleToggle = useCallback((collapsed) => {
    if (!isControlled) {
      setInternalCollapsed(collapsed);
    }
    onToggle?.(collapsed);
  }, [isControlled, onToggle]);
  const toggle = useCallback(() => {
    handleToggle(!isCollapsed);
  }, [handleToggle, isCollapsed]);
  const contextValue = useMemo(() => ({
    isCollapsed,
    collapsible,
    toggle,
    contentId
  }), [isCollapsed, collapsible, toggle, contentId]);
  const headerContent = typeof header === "function" ? header({ isCollapsed, isFocused: false }) : header;
  return /* @__PURE__ */ jsxs(ListItemGroupContext.Provider, { value: contextValue, children: [
    /* @__PURE__ */ jsx(
      ListGroupHeaderBase,
      {
        ref,
        collapsible,
        collapsed: isCollapsed,
        onToggle: handleToggle,
        accessibleName: headerAccessibleName,
        className,
        style,
        "data-testid": dataTestId,
        children: headerContent
      }
    ),
    (!collapsible || !isCollapsed) && children
  ] });
}

export { ConversationGroupListItem, ConversationListItem, ListGroupHeaderBase, ListItemGroup, ListItemGroupBase, ListItemSeparator, useListGroupHeader, useListItemGroup };
