// @sap-ui/fx-components/dist/chunk-T77SNGYR.js
import { Token } from './chunk-A3BV5G37.js';
import { DeclineIcon } from './chunk-VPXEN5VV.js';
import { ListItem } from './chunk-E4NK6IRO.js';
import { List } from './chunk-CJXZUSUP.js';
import { ResponsivePopover } from './chunk-KDP347JV.js';
import { Button } from './chunk-T7M6DGUN.js';
import { cn } from './chunk-2F2PZMJ5.js';
import React, { useRef, useState, useImperativeHandle, useEffect, useCallback, useLayoutEffect, cloneElement, Children, isValidElement } from 'react';
import { useTranslation } from 'react-i18next';
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';

var flattenChildren = (nodes) => {
  const result = [];
  Children.forEach(nodes, (child) => {
    if (!isValidElement(child)) return;
    if (child.type === React.Fragment) {
      result.push(...flattenChildren(child.props.children));
    } else {
      result.push(child);
    }
  });
  return result;
};
function Tokenizer({
  children,
  readonly = false,
  disabled = false,
  multiLine: controlledMultiLine,
  showClearAll = false,
  onTokenDelete,
  onSelectionChange,
  onNMoreClick,
  accessibleName,
  accessibleNameRef,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const containerRef = useRef(null);
  const contentRef = useRef(null);
  const nMoreRef = useRef(null);
  const selectionAnchorRef = useRef(-1);
  const skipNextFocusRef = useRef(false);
  const [focusedIndex, setFocusedIndex] = useState(-1);
  const [selectedIndices, setSelectedIndices] = useState(/* @__PURE__ */ new Set());
  const [visibleCount, setVisibleCount] = useState(null);
  const [internalExpanded] = useState(false);
  const [popoverOpen, setPopoverOpen] = useState(false);
  const [liveAnnouncement, setLiveAnnouncement] = useState("");
  const { t } = useTranslation("fx");
  const expanded = controlledMultiLine ?? internalExpanded;
  const tokens = flattenChildren(children).filter(
    (child) => child.type === Token
  );
  const tokenCount = tokens.length;
  const effectiveVisible = expanded ? tokenCount : visibleCount ?? tokenCount;
  const nMoreCount = tokenCount - effectiveVisible;
  useImperativeHandle(
    ref,
    () => ({
      focus() {
        const idx = focusedIndex >= 0 ? focusedIndex : 0;
        setFocusedIndex(idx);
        const el = contentRef.current?.children[idx];
        el?.focus();
      },
      focusLast() {
        const lastIdx = effectiveVisible - 1;
        if (lastIdx >= 0) {
          skipNextFocusRef.current = true;
          setFocusedIndex(lastIdx);
          const el = contentRef.current?.children[lastIdx];
          el?.focus();
        }
      },
      blur() {
        document.activeElement?.blur();
      },
      isFocused() {
        return contentRef.current?.contains(document.activeElement) ?? false;
      },
      get nativeElement() {
        return containerRef.current;
      }
    }),
    [focusedIndex, effectiveVisible]
  );
  const tokenWidthsRef = useRef([]);
  const needsMeasureRef = useRef(true);
  const zeroWidthRetriesRef = useRef(0);
  useEffect(() => {
    needsMeasureRef.current = true;
    zeroWidthRetriesRef.current = 0;
    setVisibleCount(null);
  }, [tokenCount]);
  const measureAndRecalculate = useCallback(() => {
    if (expanded || !contentRef.current || !containerRef.current) return;
    if (needsMeasureRef.current || tokenWidthsRef.current.length !== tokenCount) {
      const content = contentRef.current;
      const tokenEls = Array.from(content.children).filter(
        (el) => el.getAttribute("data-part") === "root"
      );
      const widths2 = tokenEls.map((el) => el.getBoundingClientRect().width);
      if (widths2.length > 0 && widths2.every((w) => w === 0)) {
        if (zeroWidthRetriesRef.current < 10) {
          zeroWidthRetriesRef.current++;
          requestAnimationFrame(measureAndRecalculate);
        }
        return;
      }
      tokenWidthsRef.current = widths2;
      needsMeasureRef.current = false;
    }
    const availableWidth = containerRef.current.getBoundingClientRect().width;
    if (availableWidth === 0) {
      if (zeroWidthRetriesRef.current < 10) {
        zeroWidthRetriesRef.current++;
        requestAnimationFrame(measureAndRecalculate);
      }
      return;
    }
    zeroWidthRetriesRef.current = 0;
    const widths = tokenWidthsRef.current;
    if (widths.length === 0) {
      setVisibleCount(null);
      return;
    }
    const gap = 4;
    let total = 0;
    for (let i = 0; i < widths.length; i++) {
      total += widths[i] + (i > 0 ? gap : 0);
    }
    if (total <= availableWidth) {
      setVisibleCount(null);
      return;
    }
    const nMoreEl = nMoreRef.current;
    const nMoreWidth = nMoreEl ? nMoreEl.getBoundingClientRect().width + gap : 50;
    let used = 0;
    let fits = 0;
    for (let i = 0; i < widths.length; i++) {
      const needed = widths[i] + (i > 0 ? gap : 0);
      if (used + needed + nMoreWidth > availableWidth) break;
      used += needed;
      fits++;
    }
    setVisibleCount(fits);
  }, [expanded, tokenCount]);
  useLayoutEffect(() => {
    if (expanded || !contentRef.current || !containerRef.current) {
      setVisibleCount(null);
      return;
    }
    measureAndRecalculate();
  });
  const lastParentWidthRef = useRef(0);
  useEffect(() => {
    if (expanded || !containerRef.current) return;
    const container = containerRef.current;
    const parent = container.parentElement;
    const observer = new ResizeObserver((entries) => {
      for (const entry of entries) {
        if (entry.target === parent) {
          const newWidth = entry.contentRect.width;
          const grew = newWidth > lastParentWidthRef.current;
          lastParentWidthRef.current = newWidth;
          if (grew && visibleCount !== null) {
            needsMeasureRef.current = true;
            zeroWidthRetriesRef.current = 0;
            setVisibleCount(null);
            return;
          }
        }
      }
      requestAnimationFrame(measureAndRecalculate);
    });
    observer.observe(container);
    if (parent) {
      lastParentWidthRef.current = parent.getBoundingClientRect().width;
      observer.observe(parent);
    }
    return () => observer.disconnect();
  }, [expanded, measureAndRecalculate, visibleCount]);
  const handleNMoreClick = useCallback(() => {
    if (onNMoreClick) {
      onNMoreClick();
    } else {
      setPopoverOpen((prev) => !prev);
    }
  }, [onNMoreClick]);
  const handleTokenSelect = useCallback(
    (index, detail) => {
      if (disabled) return;
      setSelectedIndices((prev) => {
        const next = new Set(prev);
        if (detail.selected) {
          next.add(index);
        } else {
          next.delete(index);
        }
        return next;
      });
      setFocusedIndex(index);
      selectionAnchorRef.current = -1;
    },
    [disabled]
  );
  useEffect(() => {
    onSelectionChange?.({ selectedIndices: Array.from(selectedIndices) });
  }, [selectedIndices]);
  useEffect(() => {
    if (selectedIndices.size > 0) {
      setLiveAnnouncement(
        t("TOKENIZER_TOKENS_SELECTED", { count: selectedIndices.size })
      );
    }
  }, [selectedIndices, t]);
  useEffect(() => {
    if (tokenCount > 0) {
      setLiveAnnouncement(
        t("TOKENIZER_TOKEN_COUNT", { count: tokenCount })
      );
    }
  }, [tokenCount, t]);
  const handleTokenDelete = useCallback(
    (index, _detail) => {
      if (disabled) return;
      onTokenDelete?.({ tokens: [index] });
      const nextIndex = tokenCount <= 1 ? -1 : index >= tokenCount - 1 ? Math.max(0, index - 1) : index;
      setFocusedIndex(nextIndex);
      if (nextIndex >= 0) {
        requestAnimationFrame(() => {
          const el = contentRef.current?.children[nextIndex];
          el?.focus();
        });
      }
    },
    [disabled, onTokenDelete, tokenCount]
  );
  const handlePopoverTokenDelete = useCallback(
    (originalIndex) => {
      if (disabled || readonly) return;
      onTokenDelete?.({ tokens: [originalIndex] });
    },
    [disabled, readonly, onTokenDelete]
  );
  const handleClearAll = useCallback(() => {
    if (disabled || readonly) return;
    const allIndices = Array.from({ length: tokenCount }, (_, i) => i);
    onTokenDelete?.({ tokens: allIndices });
    setSelectedIndices(/* @__PURE__ */ new Set());
  }, [disabled, readonly, onTokenDelete, tokenCount]);
  const scrollTokenIntoView = useCallback((index) => {
    const el = contentRef.current?.children[index];
    el?.scrollIntoView?.({ block: "nearest", inline: "nearest" });
  }, []);
  const handleKeyDown = useCallback(
    (e) => {
      if (disabled || tokenCount === 0) return;
      const visCount = expanded ? tokenCount : effectiveVisible;
      switch (e.key) {
        case "ArrowRight":
        case "ArrowDown": {
          e.preventDefault();
          const next = Math.min((focusedIndex < 0 ? 0 : focusedIndex) + 1, visCount - 1);
          if (e.shiftKey) {
            if (selectionAnchorRef.current < 0) {
              selectionAnchorRef.current = focusedIndex >= 0 ? focusedIndex : 0;
            }
            const anchor = selectionAnchorRef.current;
            const start = Math.min(anchor, next);
            const end = Math.max(anchor, next);
            const range = /* @__PURE__ */ new Set();
            for (let i = start; i <= end; i++) range.add(i);
            setSelectedIndices(range);
          } else {
            selectionAnchorRef.current = -1;
          }
          setFocusedIndex(next);
          const el = contentRef.current?.children[next];
          el?.focus();
          scrollTokenIntoView(next);
          break;
        }
        case "ArrowLeft":
        case "ArrowUp": {
          e.preventDefault();
          if (focusedIndex < 0) break;
          const prev = Math.max(focusedIndex - 1, 0);
          if (e.shiftKey) {
            if (selectionAnchorRef.current < 0) {
              selectionAnchorRef.current = focusedIndex;
            }
            const anchor = selectionAnchorRef.current;
            const start = Math.min(anchor, prev);
            const end = Math.max(anchor, prev);
            const range = /* @__PURE__ */ new Set();
            for (let i = start; i <= end; i++) range.add(i);
            setSelectedIndices(range);
          } else {
            selectionAnchorRef.current = -1;
          }
          setFocusedIndex(prev);
          const el = contentRef.current?.children[prev];
          el?.focus();
          scrollTokenIntoView(prev);
          break;
        }
        case "Home": {
          e.preventDefault();
          selectionAnchorRef.current = -1;
          setFocusedIndex(0);
          const el = contentRef.current?.children[0];
          el?.focus();
          scrollTokenIntoView(0);
          break;
        }
        case "End": {
          e.preventDefault();
          selectionAnchorRef.current = -1;
          const lastIdx = visCount - 1;
          setFocusedIndex(lastIdx);
          const el = contentRef.current?.children[lastIdx];
          el?.focus();
          scrollTokenIntoView(lastIdx);
          break;
        }
        case "a":
        case "A": {
          if (e.ctrlKey || e.metaKey) {
            e.preventDefault();
            if (selectedIndices.size === tokenCount) {
              setSelectedIndices(/* @__PURE__ */ new Set());
            } else {
              setSelectedIndices(new Set(Array.from({ length: tokenCount }, (_, i) => i)));
            }
          }
          break;
        }
        case "c":
        case "C": {
          if (e.ctrlKey || e.metaKey) {
            const selectedTexts = tokens.filter((_, i) => selectedIndices.has(i)).map((t2) => t2.props.text);
            if (selectedTexts.length > 0) {
              navigator.clipboard.writeText(selectedTexts.join("\r\n"));
            }
          }
          break;
        }
        case "x":
        case "X": {
          if ((e.ctrlKey || e.metaKey) && !readonly) {
            const selectedTexts = tokens.filter((_, i) => selectedIndices.has(i)).map((t2) => t2.props.text);
            if (selectedTexts.length > 0) {
              navigator.clipboard.writeText(selectedTexts.join("\r\n"));
              onTokenDelete?.({ tokens: Array.from(selectedIndices).sort((a, b) => a - b) });
              setSelectedIndices(/* @__PURE__ */ new Set());
            }
          }
          break;
        }
        case "Backspace":
        case "Delete": {
          if (readonly || disabled) break;
          if (selectedIndices.size > 1) {
            e.preventDefault();
            e.stopPropagation();
            const sorted = Array.from(selectedIndices).sort((a, b) => a - b);
            onTokenDelete?.({ tokens: sorted });
            setSelectedIndices(/* @__PURE__ */ new Set());
            const nextIdx = Math.min(sorted[0], tokenCount - sorted.length - 1);
            if (nextIdx >= 0) {
              setFocusedIndex(nextIdx);
              requestAnimationFrame(() => {
                const el = contentRef.current?.children[nextIdx];
                el?.focus();
              });
            }
          }
          break;
        }
      }
    },
    [disabled, readonly, tokenCount, effectiveVisible, focusedIndex, expanded, selectedIndices, tokens, onTokenDelete, scrollTokenIntoView]
  );
  const handleFocus = useCallback(
    (e) => {
      if (disabled) return;
      if (skipNextFocusRef.current) {
        skipNextFocusRef.current = false;
        return;
      }
      if (!contentRef.current?.contains(e.relatedTarget)) {
        const idx = focusedIndex >= 0 && focusedIndex < effectiveVisible ? focusedIndex : 0;
        setFocusedIndex(idx);
      }
    },
    [disabled, focusedIndex, effectiveVisible]
  );
  const visibleTokens = tokens.slice(0, effectiveVisible).map((token, index) => {
    return cloneElement(token, {
      key: token.key ?? index,
      tabIndex: index === focusedIndex ? 0 : -1,
      readonly: readonly || token.props.readonly,
      selected: selectedIndices.has(index) || token.props.selected,
      singleToken: tokenCount === 1,
      onSelect: (detail) => {
        token.props.onSelect?.(detail);
        handleTokenSelect(index, detail);
      },
      onDelete: (detail) => {
        token.props.onDelete?.(detail);
        handleTokenDelete(index, detail);
      }
    });
  });
  const showNMore = !expanded && nMoreCount > 0;
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: containerRef,
      id,
      className: cn(
        "box-border",
        expanded ? "inline-flex flex-wrap items-start" : "flex items-center h-9",
        disabled && "opacity-40 pointer-events-none",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "tokenizer-root",
      children: [
        /* @__PURE__ */ jsxs(
          "div",
          {
            ref: contentRef,
            role: "listbox",
            "aria-label": accessibleName,
            "aria-labelledby": accessibleNameRef,
            "aria-disabled": disabled || void 0,
            "aria-readonly": readonly || void 0,
            "aria-orientation": "horizontal",
            "aria-roledescription": "Tokenizer",
            onKeyDown: handleKeyDown,
            onFocus: handleFocus,
            className: cn(
              "flex items-center gap-1 box-border",
              expanded ? "flex-wrap" : "flex-nowrap h-full min-w-0 flex-1",
              tokenCount === 1 && !expanded && "max-w-full flex-1 pr-1"
            ),
            "data-part": "content",
            children: [
              visibleTokens,
              expanded && showClearAll && tokenCount > 0 && !readonly && !disabled && /* @__PURE__ */ jsx(
                "span",
                {
                  role: "button",
                  tabIndex: 0,
                  onClick: handleClearAll,
                  onKeyDown: (e) => {
                    if (e.key === "Enter" || e.key === " ") {
                      e.preventDefault();
                      handleClearAll();
                    }
                  },
                  className: cn(
                    "cursor-pointer whitespace-nowrap outline-none self-center",
                    "text-sm font-normal text-sapphire-brand-foreground",
                    "hover:underline active:no-underline"
                  ),
                  "data-part": "clear-all",
                  children: t("TOKENIZER_CLEAR_ALL")
                }
              )
            ]
          }
        ),
        showNMore && /* @__PURE__ */ jsxs(Fragment, { children: [
          /* @__PURE__ */ jsx(
            "span",
            {
              ref: nMoreRef,
              role: "button",
              tabIndex: 0,
              "aria-haspopup": "dialog",
              onClick: handleNMoreClick,
              onKeyDown: (e) => {
                if (e.key === "Enter" || e.key === " ") {
                  e.preventDefault();
                  handleNMoreClick();
                }
              },
              className: cn(
                "inline-block ml-1 cursor-pointer whitespace-nowrap flex-shrink-0",
                "text-sm font-normal text-sapphire-brand-foreground",
                "hover:underline active:no-underline"
              ),
              "data-part": "n-more",
              children: t("TOKENIZER_SHOW_MORE", { count: nMoreCount })
            }
          ),
          !onNMoreClick && /* @__PURE__ */ jsx(
            ResponsivePopover,
            {
              opener: containerRef,
              open: popoverOpen,
              onClose: () => setPopoverOpen(false),
              placement: "Bottom",
              noPadding: true,
              children: /* @__PURE__ */ jsx(
                List,
                {
                  selectionMode: readonly ? "None" : "Delete",
                  onItemDelete: (detail) => {
                    const originalIndex = Number(detail.item.id);
                    if (!isNaN(originalIndex)) {
                      handlePopoverTokenDelete(originalIndex);
                    }
                  },
                  className: cn(
                    "[&_li]:!min-h-8 [&_li]:!h-8 [&_li]:relative [&_li_>_span]:!py-1",
                    "[&_li]:!ring-0 [&_li]:!shadow-none",
                    "[&_li:focus]:after:absolute [&_li:focus]:after:inset-0.5 [&_li:focus]:after:border-2 [&_li:focus]:after:border-ring [&_li:focus]:after:rounded-sm [&_li:focus]:after:pointer-events-none"
                  ),
                  items: tokens.map((t2, i) => ({
                    id: String(i),
                    text: t2.props.text || ""
                  })),
                  renderItem: (item) => /* @__PURE__ */ jsx(
                    ListItem,
                    {
                      itemKey: item.id,
                      text: item.text,
                      type: readonly ? "Inactive" : "Active",
                      deleteButton: !readonly ? /* @__PURE__ */ jsx(
                        Button,
                        {
                          design: "Tertiary",
                          size: "Small",
                          iconOnly: true,
                          icon: /* @__PURE__ */ jsx(DeclineIcon, { className: "h-3 w-3" }),
                          accessibleName: t("TOKEN_DELETE"),
                          tabIndex: -1
                        }
                      ) : void 0
                    },
                    item.id
                  )
                }
              )
            }
          )
        ] }),
        !expanded && showClearAll && tokenCount > 0 && !readonly && !disabled && /* @__PURE__ */ jsx(
          "span",
          {
            role: "button",
            tabIndex: 0,
            onClick: handleClearAll,
            onKeyDown: (e) => {
              if (e.key === "Enter" || e.key === " ") {
                e.preventDefault();
                handleClearAll();
              }
            },
            className: cn(
              "ml-1 cursor-pointer whitespace-nowrap outline-none flex-shrink-0",
              "text-sm font-normal text-sapphire-brand-foreground",
              "hover:underline active:no-underline"
            ),
            "data-part": "clear-all",
            children: t("TOKENIZER_CLEAR_ALL")
          }
        ),
        /* @__PURE__ */ jsx(
          "div",
          {
            "aria-live": "polite",
            "aria-atomic": "true",
            className: "sr-only",
            "data-part": "live-region",
            children: liveAnnouncement
          }
        )
      ]
    }
  );
}

export { Tokenizer };
