// @sap-ui/fx-components/dist/chunk-UHU7LLWX.js
import { IllustratedMessage, useIllustrationSizeContext } from './chunk-DTDQJNHA.js';
import * as React from 'react';
import { jsx } from 'react/jsx-runtime';

function makeLazySvg(importFn) {
  const Lazy = React.lazy(importFn);
  const Wrapper = ({ className }) => /* @__PURE__ */ jsx(React.Suspense, { fallback: null, children: /* @__PURE__ */ jsx(Lazy, { className }) });
  return Wrapper;
}
function createIllustration(config) {
  const {
    name,
    title,
    subtitle,
    lazyExtraSmall,
    lazySmall,
    lazyMedium,
    lazyLarge
  } = config;
  const lazyVariants = {
    ["ExtraSmall" /* ExtraSmall */]: React.lazy(lazyExtraSmall),
    ["Small" /* Small */]: React.lazy(lazySmall),
    ["Medium" /* Medium */]: React.lazy(lazyMedium),
    ["Large" /* Large */]: React.lazy(lazyLarge)
  };
  const defaultLazy = lazyVariants["Medium" /* Medium */];
  const Illustration = ({
    design,
    titleText,
    subtitleText,
    title: titleSlot,
    subtitle: subtitleSlot,
    children,
    decorative,
    accessibleName,
    className,
    style,
    id,
    "data-testid": dataTestId
  }) => {
    const resolvedTitleText = titleText ?? title;
    const resolvedSubtitleText = subtitleText ?? subtitle;
    return /* @__PURE__ */ jsx(
      IllustratedMessage,
      {
        design,
        illustration: /* @__PURE__ */ jsx(
          LazyIllustrationBridge,
          {
            lazyVariants,
            fallback: defaultLazy
          }
        ),
        titleText: titleSlot ? void 0 : resolvedTitleText,
        title: titleSlot,
        subtitleText: subtitleSlot ? void 0 : resolvedSubtitleText,
        subtitle: subtitleSlot,
        decorative,
        accessibleName: accessibleName || name,
        className,
        style,
        id,
        "data-testid": dataTestId,
        children
      }
    );
  };
  const component = Illustration;
  component.displayName = name;
  component.ExtraSmall = makeLazySvg(lazyExtraSmall);
  component.Small = makeLazySvg(lazySmall);
  component.Medium = makeLazySvg(lazyMedium);
  component.Large = makeLazySvg(lazyLarge);
  component.metadata = { name, title, subtitle };
  return component;
}
function LazyIllustrationBridge({
  lazyVariants,
  fallback
}) {
  const resolvedSize = useIllustrationSizeContext();
  if (resolvedSize === "Base" /* Base */) return null;
  const LazySvg = lazyVariants[resolvedSize] || fallback;
  return /* @__PURE__ */ jsx(React.Suspense, { fallback: null, children: /* @__PURE__ */ jsx(LazySvg, {}) });
}

export { createIllustration };
