// @sap-ui/fx-components/dist/index.js
import './chunk-AQ6GDRXB.js';
export { ToolbarButton, ToolbarItem, ToolbarSelect, ToolbarSelectOption, ToolbarSeparator, ToolbarSpacer } from './chunk-I56SHA5L.js';
import './chunk-52TES2IY.js';
export { Achievement, AddColumn, AddDimensions, AddPeople, AddPeopleToCalendar, AddingColumns, BalloonSky, BeforeSearch, Connection, DragFilesToUpload, EmptyCalendar, EmptyList, EmptyPlanningCalendar, ErrorScreen, FilterTable, FilteringColumns, GroupTable, GroupingColumns, KeyTask, NewMail, NoActivities, NoChartData, NoColumnsSet, NoData, NoDimensionsSet, NoEntries, NoFilterResults, NoMail, NoMail_v1, NoNotifications, NoSavedItems, NoSavedItems_v1, NoSearchResults, NoTasks, NoTasks_v1, PageNotFound, ReceiveAppreciation, ReloadScreen, ResizeColumn, ResizingColumns, SearchEarth, SearchFolder, SignOut, SimpleBalloon, SimpleBell, SimpleCalendar, SimpleCheckMark, SimpleConnection, SimpleEmptyDoc, SimpleEmptyList, SimpleError, SimpleMagnifier, SimpleMail, SimpleNoSavedItems, SimpleNotFoundMagnifier, SimpleReload, SimpleTask, SleepingBell, SortColumn, SortingColumns, SuccessBalloon, SuccessCheckMark, SuccessHighFive, SuccessScreen, Survey, Tent, UnableToLoad, UnableToLoadImage, UnableToUpload, UploadCollection, UploadToCloud, UserHasSignedUp } from './chunk-KIJTYAE6.js';
export { Avatar as TntAvatar, Calculator as TntCalculator, ChartArea as TntChartArea, ChartArea2 as TntChartArea2, ChartBPMNFlow as TntChartBPMNFlow, ChartBar as TntChartBar, ChartBullet as TntChartBullet, ChartDoughnut as TntChartDoughnut, ChartFlow as TntChartFlow, ChartGantt as TntChartGantt, ChartOrg as TntChartOrg, ChartPie as TntChartPie, CodePlaceholder as TntCodePlaceholder, Company as TntCompany, Compass as TntCompass, Components as TntComponents, Dialog as TntDialog, EmptyContentPane as TntEmptyContentPane, ExternalLink as TntExternalLink, FaceID as TntFaceID, Fingerprint as TntFingerprint, Handshake as TntHandshake, Help as TntHelp, Lock as TntLock, Mission as TntMission, MissionFailed as TntMissionFailed, NoApplications as TntNoApplications, NoFlows as TntNoFlows, NoUsers as TntNoUsers, Radar as TntRadar, RoadMap as TntRoadMap, Secrets as TntSecrets, Services as TntServices, SessionExpired as TntSessionExpired, SessionExpiring as TntSessionExpiring, Settings as TntSettings, Success as TntSuccess, SuccessfulAuth as TntSuccessfulAuth, Systems as TntSystems, Teams as TntTeams, Tools as TntTools, Tutorials as TntTutorials, UnableToLoad as TntUnableToLoad, Unlock as TntUnlock, UnsuccessfulAuth as TntUnsuccessfulAuth, User2 as TntUser2 } from './chunk-TQOJRVTN.js';
export { createIllustration } from './chunk-UHU7LLWX.js';
export { Tab, TabContainer, TabContainerBackgroundDesign, TabContainerContext, TabLayout, TabOverflowButton, TabOverflowMode, TabSemanticDesign, TabSeparator, TabsPlacement, useTabContainerContext, useTabDragDrop, useTabNavigation, useTabOverflow } from './chunk-726M35FW.js';
import './chunk-EZTDGHEX.js';
import './chunk-QLZURZFY.js';
export { Text, TextEmptyIndicatorMode } from './chunk-C4DRGF7N.js';
export { Textarea } from './chunk-DJBESHY7.js';
import './chunk-NMNDY5ZS.js';
export { Toast, ToastPlacement } from './chunk-YJEMXAX6.js';
import './chunk-A33H3CV2.js';
export { SearchField, SearchFieldSize } from './chunk-76CYWX53.js';
import './chunk-VI7APGF6.js';
export { OptionCustom } from './chunk-ZDPLMRZF.js';
export { Option, Select, SelectSize, TextSeparator } from './chunk-QDXU3YWA.js';
export { SideNavigation, SideNavigationGroup, SideNavigationItem, SideNavigationSubItem } from './chunk-3OK4EB4T.js';
export { Skeleton } from './chunk-PAAMPCX2.js';
import './chunk-KNEWUE42.js';
export { Switch, SwitchDesign } from './chunk-XU2XVLPF.js';
export { Table, TableCell, TableContext, TableGrowing, TableHeaderCell, TableHeaderCellActionAI, TableHeaderRow, TableRow, TableRowAction, TableRowActionNavigation, TableSelectionMulti, TableSelectionSingle, TableToolbar, useTableContext } from './chunk-MIIXFVME.js';
export { Toolbar, ToolbarAlign, ToolbarContext, ToolbarDesign, ToolbarItemOverflowBehavior, useToolbarContext } from './chunk-3GZW6BV2.js';
export { MessageStrip, MessageStripDesign } from './chunk-IIYUWYBR.js';
export { MultiComboBox, MultiComboBoxItem, MultiComboBoxItemGroup } from './chunk-P6PNCHF2.js';
export { Tokenizer } from './chunk-T77SNGYR.js';
export { NotificationList, NotificationListContext, NotificationListGroupItem, NotificationListItem, NotificationListItemImportance, NotificationListItemState, useNotificationListContext, useOptionalNotificationListContext } from './chunk-BJEKLZEH.js';
export { Tag, TagDesign } from './chunk-J3MFNYJX.js';
import './chunk-IZLDRN5G.js';
import './chunk-6BLRNKFS.js';
export { ProgressIndicator, ProgressIndicatorValueState } from './chunk-S77RYWD7.js';
import './chunk-4EYSI6D3.js';
import './chunk-T4UN2NZ6.js';
export { BellIcon, CloseCommandFieldIcon, CodeIcon, ConversationsIcon, CursorIcon, DiscoverIcon, FxLayout, FxLayoutContext, FxPaneHeader, FxPaneHeaderAction, FxPaneHeaderSegmentedAction, FxPaneHeaderSegmentedOption, FxPaneHeaderSplitAction, FxPaneHeaderSplitOption, FxPromptInput, FxUserMenu, FxUserMenuItem, FxUserMenuItemGroup, JobsIcon, JouleIcon, MenuIcon, OpenCommandFieldIcon, PanelCloseIcon, PanelOpenIcon, SpacesIcon, VSCodeIcon, WaveformIcon, useFxLayoutContext } from './chunk-UXS5LRDO.js';
export { TagExploration, TagExplorationColorScheme, TagExplorationDesign, TagExplorationSize, TagExplorationWrappingType } from './chunk-NCI4MJ5T.js';
export { SegmentedButton, SegmentedButtonItem } from './chunk-ZH374RLZ.js';
export { SplitButton } from './chunk-ELIHCCZQ.js';
export { Panel, PanelAccessibleRole, PanelDesign } from './chunk-MQ3F6SMX.js';
import './chunk-RRNRQSBA.js';
import './chunk-CZSX2QPT.js';
export { IllustratedMessage, IllustrationDesign, IllustrationSize, IllustrationSizeContext, useIllustrationSizeContext } from './chunk-DTDQJNHA.js';
export { Input, InputFilter, InputSize, InputSuggestions, InputType } from './chunk-TNAICDBU.js';
import './chunk-HOF43XR6.js';
export { Label, LabelWrappingType } from './chunk-4PDXVETU.js';
import './chunk-356DL3BJ.js';
import './chunk-PPH3G7AL.js';
export { Menu, MenuHeader, MenuItem, MenuItemGroup, MenuItemGroupCheckMode, MenuSeparator } from './chunk-EHO24TXD.js';
import './chunk-JWTFHLES.js';
export { Card, CardContent, CardFooter, CardHeader, CardHeaderStatus } from './chunk-KG7YP5ME.js';
import './chunk-2T5CHKKY.js';
export { ComboBox, ComboBoxItem, ComboBoxItemGroup } from './chunk-LM6CSHOP.js';
export { ComboBoxPopover } from './chunk-EEGGRFXR.js';
export { ComboBoxFilter, ComboBoxSize, SelectPopover } from './chunk-KQMKCSMK.js';
import './chunk-WWAPKHIM.js';
import './chunk-4SPYXMGC.js';
export { DateTimePicker, TimePicker } from './chunk-EWAPHGAS.js';
export { CalendarType, DatePicker, DatePickerSelectionMode, DatePickerSize } from './chunk-GC7VJCBJ.js';
export { ActionSettingsIcon, ActivityItemsIcon, ExcelAttachmentIcon, FullScreenIcon } from './chunk-H4UBXTRR.js';
import './chunk-H6VYFVIE.js';
import './chunk-Y5TA5SIT.js';
import './chunk-YWNUKDTJ.js';
export { ToggleButton } from './chunk-2DPLFZUY.js';
import './chunk-YYSFHEKJ.js';
import './chunk-4Y6CARK4.js';
import './chunk-T7Y6YDVS.js';
import './chunk-4IU3KSGT.js';
import './chunk-VNDUTC7E.js';
export { Calendar, CalendarHeader, CalendarLegend, CalendarLegendItem, CalendarLegendItemType, CalendarPickerView, CalendarPickersMode, CalendarSelectionMode, CalendarWeekNumbering, DayPicker, MonthPicker, RTL_LOCALES, YearPicker, addMonthsToDate, addYearsToDate, createDate, dateToTimestamp, ensureCldr, format, formatDateToISO, getMonth, getMonthDays, getMonthName, getToday, getWeekEnd, getWeekNumber, getWeekStart, getWeekYearNumber, getYear, initFxI18n, isDateAfter, isDateBefore, isDateInRange, isRTLLocale, isSameDayUtil, isSameMonthUtil, isToday, loadFxTranslations, parseDate, timestampToDate, useDirection, useEnsureCldr } from './chunk-PXVVADOR.js';
import './chunk-6XLAD3TP.js';
import './chunk-WQOS25OP.js';
import './chunk-WYFIVVSE.js';
export { FileUploader, FileUploaderSize } from './chunk-ORTPW5DV.js';
export { Token } from './chunk-A3BV5G37.js';
import './chunk-PLKA6UTA.js';
export { ValueState } from './chunk-GBYPSHX6.js';
import './chunk-P4WB5QKL.js';
import './chunk-VPXEN5VV.js';
import './chunk-ZK4METPA.js';
import './chunk-FL4UPYZT.js';
export { Tabbar, TabbarOverflowButton, useTabbarOverflow } from './chunk-FSSKJCND.js';
export { ConversationGroupListItem, ConversationListItem, ListGroupHeaderBase, ListItemGroup, ListItemGroupBase, ListItemSeparator, useListGroupHeader, useListItemGroup } from './chunk-IS37ZOQI.js';
export { ListItemCustom } from './chunk-SZG4UYOP.js';
export { ListItem } from './chunk-E4NK6IRO.js';
import './chunk-OGRUTROU.js';
import './chunk-YHUSMQDT.js';
import './chunk-ZQSLIXR7.js';
import './chunk-QJDVAT2W.js';
export { Avatar, AvatarColorScheme, AvatarShape, AvatarSize } from './chunk-OXMWPC4U.js';
import './chunk-LCV3JX36.js';
export { Bar, BarAccessibleRole, BarDesign } from './chunk-VSZFLW7F.js';
export { Breadcrumbs, BreadcrumbsDesign, BreadcrumbsItem, BreadcrumbsSeparator, SEPARATOR_MAP } from './chunk-45VUVRRJ.js';
export { Link, LinkAccessibleRole, LinkDesign, LinkInteractiveAreaSize, LinkWrappingType } from './chunk-MCJ7D5KG.js';
export { DropIndicator, LIST_ITEM_BASE_CLASSES, LIST_ITEM_FOCUS_CLASSES, ListItemBase, ListSelectionControl, useListItem, useListItemDragDrop, useListItemFocus, useListItemRegistration, useListItemSelection } from './chunk-CTYK4HEK.js';
import './chunk-O5ILLEDP.js';
export { RadioButton, RadioButtonSize, RadioButtonValueState, RadioButtonWrappingType } from './chunk-2MAGJSMZ.js';
export { CheckBox, CheckBoxSize, WrappingType } from './chunk-TNSICZD6.js';
export { List, ListAccessibleRole, ListContext, ListGrowingMode, ListItemAccessibleRole, ListItemHighlight, ListItemType, ListItemValueState, ListItemWrappingType, ListSelectionMode, ListSeparator, useListContext, useOptionalListContext } from './chunk-CJXZUSUP.js';
import './chunk-MJFYIHUV.js';
import './chunk-56AYC3ZC.js';
export { ResponsivePopover } from './chunk-KDP347JV.js';
import './chunk-TO7N3UGU.js';
export { Popover, PopoverHorizontalAlign, PopoverPlacement, PopoverVerticalAlign, PopupAccessibleRole } from './chunk-ZDBE5GTH.js';
export { Dialog, DialogState } from './chunk-4C7TQYID.js';
import './chunk-COC3EGPV.js';
import './chunk-WDT5ZCHE.js';
import './chunk-YSPWI6FA.js';
import './chunk-GCJAV5G5.js';
export { Title } from './chunk-BM2JFIU4.js';
export { TitleLevel, TitleWrappingType } from './chunk-KHM7ZBIS.js';
export { Icon, IconDesign, IconMode } from './chunk-DFXNKCOL.js';
import './chunk-WUBK4IJL.js';
import './chunk-ORTCK34Y.js';
export { Button } from './chunk-T7M6DGUN.js';
export { ButtonAccessibleRole, ButtonDesign, ButtonSize, ButtonType } from './chunk-CPOUYZCH.js';
export { BusyIndicator, BusyIndicatorSize, BusyIndicatorTextPlacement } from './chunk-E2WLAC4U.js';
export { clsx, cn, cva, findFirstFocusable } from './chunk-2F2PZMJ5.js';
import './chunk-V6TY7KAL.js';
export { useTranslation } from 'react-i18next';
import { createContext, useState, useEffect, useCallback, useMemo, useContext } from 'react';
import { jsx } from 'react/jsx-runtime';

// src/types/table.ts
var TableOverflowMode = /* @__PURE__ */ ((TableOverflowMode2) => {
  TableOverflowMode2["Scroll"] = "Scroll";
  TableOverflowMode2["Popin"] = "Popin";
  return TableOverflowMode2;
})(TableOverflowMode || {});
var TableGrowingMode = /* @__PURE__ */ ((TableGrowingMode2) => {
  TableGrowingMode2["Button"] = "Button";
  TableGrowingMode2["Scroll"] = "Scroll";
  return TableGrowingMode2;
})(TableGrowingMode || {});
var TableCellHorizontalAlign = /* @__PURE__ */ ((TableCellHorizontalAlign2) => {
  TableCellHorizontalAlign2["Left"] = "Left";
  TableCellHorizontalAlign2["Start"] = "Start";
  TableCellHorizontalAlign2["Right"] = "Right";
  TableCellHorizontalAlign2["End"] = "End";
  TableCellHorizontalAlign2["Center"] = "Center";
  return TableCellHorizontalAlign2;
})(TableCellHorizontalAlign || {});
var TableSelectionBehavior = /* @__PURE__ */ ((TableSelectionBehavior2) => {
  TableSelectionBehavior2["RowSelector"] = "RowSelector";
  TableSelectionBehavior2["RowOnly"] = "RowOnly";
  return TableSelectionBehavior2;
})(TableSelectionBehavior || {});
var TableSelectionMultiHeaderSelector = /* @__PURE__ */ ((TableSelectionMultiHeaderSelector2) => {
  TableSelectionMultiHeaderSelector2["SelectAll"] = "SelectAll";
  TableSelectionMultiHeaderSelector2["ClearAll"] = "ClearAll";
  return TableSelectionMultiHeaderSelector2;
})(TableSelectionMultiHeaderSelector || {});
var TableSortOrder = /* @__PURE__ */ ((TableSortOrder2) => {
  TableSortOrder2["None"] = "None";
  TableSortOrder2["Ascending"] = "Ascending";
  TableSortOrder2["Descending"] = "Descending";
  return TableSortOrder2;
})(TableSortOrder || {});

// src/types/segmented-button.ts
var SegmentedButtonSize = /* @__PURE__ */ ((SegmentedButtonSize2) => {
  SegmentedButtonSize2["Large"] = "Large";
  SegmentedButtonSize2["Medium"] = "Medium";
  SegmentedButtonSize2["Small"] = "Small";
  return SegmentedButtonSize2;
})(SegmentedButtonSize || {});

// src/types/fx-user-menu.ts
var FxUserMenuItemCheckMode = /* @__PURE__ */ ((FxUserMenuItemCheckMode2) => {
  FxUserMenuItemCheckMode2["None"] = "None";
  FxUserMenuItemCheckMode2["Single"] = "Single";
  FxUserMenuItemCheckMode2["Multiple"] = "Multiple";
  return FxUserMenuItemCheckMode2;
})(FxUserMenuItemCheckMode || {});
var STORAGE_KEY = "fx-theme";
function getSystemIsDark() {
  if (typeof window === "undefined") return false;
  return window.matchMedia("(prefers-color-scheme: dark)").matches;
}
function getStoredTheme() {
  if (typeof window === "undefined") return null;
  try {
    return localStorage.getItem(STORAGE_KEY);
  } catch {
    return null;
  }
}
function storeTheme(theme) {
  if (typeof window === "undefined") return;
  try {
    localStorage.setItem(STORAGE_KEY, theme);
  } catch {
  }
}
var ThemeContext = createContext(void 0);
function ThemeProvider({
  themes,
  defaultTheme = "light",
  children
}) {
  const [theme, setThemeRaw] = useState(() => getStoredTheme() ?? defaultTheme);
  const [systemIsDark, setSystemIsDark] = useState(getSystemIsDark);
  useEffect(() => {
    const mql = window.matchMedia("(prefers-color-scheme: dark)");
    const handler = (e) => setSystemIsDark(e.matches);
    mql.addEventListener("change", handler);
    return () => mql.removeEventListener("change", handler);
  }, []);
  const resolvedTheme = theme === "system" ? systemIsDark ? "dark" : "light" : theme;
  useEffect(() => {
    const root = document.documentElement;
    root.classList.remove("dark");
    themes.forEach((t) => {
      if (t.id) root.classList.remove(`theme-${t.id}`);
    });
    if (resolvedTheme) {
      root.classList.add(`theme-${resolvedTheme}`);
      if (resolvedTheme === "dark") {
        root.classList.add("dark");
      }
    }
    if (theme === "system") {
      root.style.colorScheme = "";
    } else {
      root.style.colorScheme = resolvedTheme === "dark" ? "dark" : "light";
    }
  }, [theme, resolvedTheme, themes]);
  const setTheme = useCallback((next) => {
    setThemeRaw(next);
    storeTheme(next);
  }, []);
  const value = useMemo(
    () => ({ theme, resolvedTheme, systemIsDark, themes, setTheme }),
    [theme, resolvedTheme, systemIsDark, themes, setTheme]
  );
  return /* @__PURE__ */ jsx(ThemeContext.Provider, { value, children });
}
function useTheme() {
  const ctx = useContext(ThemeContext);
  if (!ctx) {
    throw new Error("useTheme must be used within a <ThemeProvider>");
  }
  return ctx;
}
ThemeProvider.displayName = "ThemeProvider";

// src/lib/fixSafariActiveState.ts
var listenerAttached = false;
function isIOS() {
  if (typeof navigator === "undefined") return false;
  if (/iPhone|iPad|iPod/.test(navigator.platform)) {
    return true;
  }
  if (/Mac/.test(navigator.userAgent) && typeof document !== "undefined" && "ontouchend" in document) {
    return true;
  }
  return false;
}
function isSafari() {
  if (typeof navigator === "undefined") return false;
  const ua = navigator.userAgent;
  return /Safari/.test(ua) && !/Chrome|Chromium|CriOS|EdgiOS/.test(ua);
}
function fixSafariActiveState() {
  if (typeof document !== "undefined" && isSafari() && isIOS() && !listenerAttached) {
    document.body.addEventListener("touchstart", () => {
    }, { passive: true });
    listenerAttached = true;
  }
}
function isSafariActiveStateFixed() {
  return listenerAttached;
}

export { FxUserMenuItemCheckMode, SegmentedButtonSize, TableCellHorizontalAlign, TableGrowingMode, TableOverflowMode, TableSelectionBehavior, TableSelectionMultiHeaderSelector, TableSortOrder, ThemeProvider, fixSafariActiveState, isSafariActiveStateFixed, useTheme };
