// @sap-ui/fx-components/dist/components/file-uploader/index.cjs
'use strict';

var React2 = require('react');
var tailwindMerge = require('tailwind-merge');
var reactI18next = require('react-i18next');
var jsxRuntime = require('react/jsx-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React2__default = /*#__PURE__*/_interopDefault(React2);

// src/components/file-uploader/FileUploader.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var iconVariants = cva(
  [
    "inline-block flex-shrink-0",
    "transition-colors",
    "fill-current"
    // Use fill-current to inherit parent's text color
  ],
  {
    variants: {
      design: {
        ["Default" /* Default */]: "",
        // No color set - inherits from parent
        ["Contrast" /* Contrast */]: "text-foreground-contrast",
        ["Critical" /* Critical */]: "text-sapphire-warning",
        ["Information" /* Information */]: "text-sapphire-info",
        ["Positive" /* Positive */]: "text-sapphire-positive",
        ["Negative" /* Negative */]: "text-sapphire-negative",
        ["Neutral" /* Neutral */]: "text-sapphire-text-tertiary",
        ["NonInteractive" /* NonInteractive */]: "text-sapphire-text-tertiary opacity-60"
      },
      mode: {
        ["Decorative" /* Decorative */]: "",
        ["Image" /* Image */]: "",
        ["Interactive" /* Interactive */]: "cursor-pointer hover:opacity-80 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
      }
    },
    defaultVariants: {
      design: "Default" /* Default */,
      mode: "Decorative" /* Decorative */
    }
  }
);
function Icon({
  pathData: pathData7,
  design = "Default" /* Default */,
  mode = "Decorative" /* Decorative */,
  accessibleName,
  showTooltip = false,
  onClick,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const iconRef = React2.useRef(null);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        iconRef.current?.focus();
      },
      blur() {
        iconRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === iconRef.current;
      },
      get nativeElement() {
        return iconRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [mode, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (mode !== "Interactive" /* Interactive */) return;
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        onClick?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [mode, onClick]
  );
  if (!pathData7 || pathData7.length === 0) {
    return null;
  }
  const isInteractive = mode === "Interactive" /* Interactive */;
  const isImage = mode === "Image" /* Image */;
  const role = isInteractive ? "button" : isImage ? "img" : "presentation";
  const ariaLabel = accessibleName || void 0;
  const ariaHidden = !accessibleName && !isInteractive ? true : void 0;
  const tabIndex = isInteractive ? 0 : void 0;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      ref: iconRef,
      id,
      role,
      "aria-label": ariaLabel,
      "aria-hidden": ariaHidden,
      tabIndex,
      viewBox: "0 0 512 512",
      xmlns: "http://www.w3.org/2000/svg",
      focusable: isInteractive ? "true" : "false",
      preserveAspectRatio: "xMidYMid meet",
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        iconVariants({
          design,
          mode
        }),
        // Default size (can be overridden by className)
        "h-4 w-4",
        className
      ),
      style,
      "data-testid": dataTestId,
      children: [
        showTooltip && accessibleName && /* @__PURE__ */ jsxRuntime.jsx("title", { children: accessibleName }),
        /* @__PURE__ */ jsxRuntime.jsx("g", { role: "presentation", children: pathData7.map((path, index) => /* @__PURE__ */ jsxRuntime.jsx("path", { d: path }, index)) })
      ]
    }
  );
}
var pathData = [
  "M292 256l117 116q7 7 7 18 0 12-7.5 19t-18.5 7q-10 0-18-8L256 292 140 408q-8 8-18 8-11 0-18.5-7.5T96 390q0-10 8-18l116-116-116-116q-8-8-8-18 0-11 7.5-18.5T122 96t18 7l116 117 116-117q7-7 18-7t18.5 7.5T416 122t-7 18z"
];
var DeclineIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData, ...props })
  ),
  { displayName: "DeclineIcon", iconName: "decline" }
);
function Token({
  text,
  closeIcon,
  selected = false,
  readonly = false,
  singleToken = false,
  overflows = false,
  onSelect,
  onDelete,
  className,
  style,
  id,
  "data-testid": dataTestId,
  tabIndex = -1,
  ref
}) {
  const tokenRef = React2.useRef(null);
  const { t } = reactI18next.useTranslation("fx");
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        tokenRef.current?.focus();
      },
      blur() {
        tokenRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === tokenRef.current;
      },
      get nativeElement() {
        return tokenRef.current;
      }
    }),
    [tokenRef]
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (readonly) return;
      onSelect?.({
        originalEvent: e,
        isKeyboard: false,
        selected: !selected
      });
    },
    [readonly, selected, onSelect]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (e.key === " " || e.ctrlKey && e.key === " ") {
        if (readonly) return;
        e.preventDefault();
        onSelect?.({
          originalEvent: e,
          isKeyboard: true,
          selected: !selected
        });
      } else if (e.key === "Backspace") {
        if (readonly) return;
        e.preventDefault();
        onDelete?.({
          originalEvent: e,
          isKeyboard: true,
          backSpace: true
        });
      } else if (e.key === "Delete") {
        if (readonly) return;
        e.preventDefault();
        onDelete?.({
          originalEvent: e,
          isKeyboard: true,
          delete: true
        });
      }
    },
    [readonly, selected, onSelect, onDelete]
  );
  const handleCloseClick = React2.useCallback(
    (e) => {
      e.stopPropagation();
      onDelete?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [onDelete]
  );
  const handleCloseKeyDown = React2.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        e.preventDefault();
        e.stopPropagation();
        onDelete?.({
          originalEvent: e,
          isKeyboard: true
        });
      }
    },
    [onDelete]
  );
  const ariaDescription = readonly ? t("TOKEN_ARIA_LABEL") : `${t("TOKEN_ARIA_LABEL")}, ${t("TOKEN_ARIA_DELETABLE")}`;
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: tokenRef,
      id,
      role: "option",
      tabIndex,
      "aria-selected": selected,
      "aria-description": ariaDescription,
      onClick: handleClick,
      onKeyDown: handleKeyDown,
      className: cn(
        // Base
        "bg-sapphire-canvas-primary inline-flex items-center rounded-sm border gap-1 pl-2 text-sm leading-5 font-normal transition-colors",
        "focus:outline-none focus:border-ring focus:[box-shadow:inset_0_0_0_1px_var(--ring)]",
        !selected && "focus:bg-background",
        // Readonly: symmetric padding, no gap, clip overflow; interactive: flush right for close button
        readonly ? "px-2 py-[2px] gap-0 overflow-clip" : "pr-0",
        // State classes
        selected && !readonly ? "bg-sapphire-chrome-button-bg-selected border-sapphire-chrome-button-fg-selected text-foreground cursor-pointer" : readonly ? "bg-sapphire-canvas-primary border-border text-foreground cursor-default" : "bg-sapphire-background-tertiary border-border text-foreground hover:bg-sapphire-background-quaternary hover:border-sapphire-border-secondary cursor-pointer",
        // Single token — enable truncation
        singleToken && "max-w-full",
        // Overflows — hidden
        overflows && "hidden",
        className
      ),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            className: cn(
              "whitespace-nowrap",
              singleToken && "overflow-hidden text-ellipsis"
            ),
            title: singleToken ? text : void 0,
            "data-part": "text",
            children: text
          }
        ),
        !readonly && /* @__PURE__ */ jsxRuntime.jsx(
          "span",
          {
            tabIndex: -1,
            "aria-hidden": "true",
            title: t("TOKEN_DELETE"),
            onClick: handleCloseClick,
            onKeyDown: handleCloseKeyDown,
            className: cn(
              "flex-shrink-0 inline-flex items-center justify-center size-6 rounded-sm cursor-pointer",
              "text-foreground!"
            ),
            "data-part": "close-icon",
            children: closeIcon || /* @__PURE__ */ jsxRuntime.jsx(DeclineIcon, { className: "h-3 w-3" })
          }
        )
      ]
    }
  );
}
var pathData2 = [
  "M166 429q11 0 18.5 7t7.5 18-7.5 18.5T166 480H90q-38 0-64-26T0 390V122q0-38 26-64t64-26h115q9 0 18 7l44 38h155q38 0 64 26t26 63v96q0 11-7.5 18.5T486 288t-18-7.5-7-18.5v-96q0-16-11-27t-28-11H256q-9 0-18-7l-44-38H90q-17 0-28 11t-11 28v268q0 17 11 28t28 11h76zm339 39q7 7 7 18t-7.5 18.5T486 512t-18-7L363 399q-28 17-59 17-22 0-42.5-9t-36-24.5-24.5-36-9-42.5q0-23 9-43.5t24-35.5 35.5-24 43.5-9 43.5 9 35.5 24 24 35.5 9 43.5q0 31-17 59zM304 365q24 0 42.5-18t18.5-43q0-23-17.5-42T304 243t-43 19-17 42q0 25 18 43t42 18z"
];
var BrowseFolderIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData2, ...props })
  ),
  { displayName: "BrowseFolderIcon", iconName: "browse-folder" }
);
var pathData3 = [
  "M256 0q53 0 100 20t81.5 54.5T492 156t20 100-20 100-54.5 81.5T356 492t-100 20-100-20-81.5-54.5T20 356 0 256t20-100 54.5-81.5T156 20 256 0zm150 183q10-9 10-23 0-13-9.5-22.5T384 128t-22 9L186 308l-68-63q-9-9-22-9t-22.5 9.5T64 268q0 15 10 24l91 83q9 9 21 9 13 0 23-9z"
];
var SysEnter2Icon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData3, ...props })
  ),
  { displayName: "SysEnter2Icon", iconName: "sys-enter-2" }
);
var pathData4 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm45 256l74-73q9-11 9-23 0-13-9.5-22.5T352 128q-12 0-23 9l-73 74-73-74q-10-9-23-9t-22.5 9.5T128 160q0 12 9 23l74 73-74 73q-9 10-9 23t9.5 22.5T160 384t23-9l73-74 73 74q11 9 23 9 13 0 22.5-9.5T384 352t-9-23z"
];
var ErrorIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData4, ...props })
  ),
  { displayName: "ErrorIcon", iconName: "error" }
);
var pathData5 = [
  "M505 399q7 13 7 27 0 21-15.5 37.5T456 480H56q-25 0-40.5-16.5T0 426q0-14 7-27L208 59q17-27 48-27 14 0 27 6.5T304 59zM288 176q0-14-9-23t-23-9-23 9-9 23v96q0 14 9 23t23 9 23-9 9-23v-96zm-32 240q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9z"
];
var WarningIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData5, ...props })
  ),
  { displayName: "WarningIcon", iconName: "warning" }
);
var pathData6 = [
  "M256 0q53 0 99.5 20T437 75t55 81.5 20 99.5-20 99.5-55 81.5-81.5 55-99.5 20-99.5-20T75 437t-55-81.5T0 256t20-99.5T75 75t81.5-55T256 0zm0 160q14 0 23-9t9-23-9-23-23-9-23 9-9 23 9 23 23 9zm32 64q0-14-9-23t-23-9-23 9-9 23v160q0 14 9 23t23 9 23-9 9-23V224z"
];
var InformationIcon = Object.assign(
  React2__default.default.forwardRef(
    (props, ref) => /* @__PURE__ */ jsxRuntime.jsx(Icon, { ref, pathData: pathData6, ...props })
  ),
  { displayName: "InformationIcon", iconName: "information" }
);
var dotVariants = cva("rounded-full bg-current inline-block", {
  variants: {
    size: {
      ["S" /* S */]: "w-2 h-2",
      ["M" /* M */]: "w-4 h-4",
      ["L" /* L */]: "w-8 h-8"
    }
  },
  defaultVariants: { size: "M" /* M */ }
});
var dotGapClasses = {
  ["S" /* S */]: "gap-px",
  ["M" /* M */]: "gap-[0.1875rem]",
  ["L" /* L */]: "gap-1"
};
function BusyIndicator({
  size = "M" /* M */,
  textPlacement = "Bottom" /* Bottom */,
  active = false,
  delay = 1e3,
  text,
  children,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const rootRef = React2.useRef(null);
  const overlayRef = React2.useRef(null);
  const [isBusy, setIsBusy] = React2.useState(false);
  const labelId = React2.useId();
  const hasChildren = React2__default.default.Children.count(children) > 0;
  const normalizedSize = size || "M" /* M */;
  const isTopPlacement = textPlacement === "Top";
  React2.useEffect(() => {
    if (!active) {
      setIsBusy(false);
      return;
    }
    if (delay <= 0) {
      setIsBusy(true);
      return;
    }
    const timer = setTimeout(() => {
      setIsBusy(true);
    }, delay);
    return () => clearTimeout(timer);
  }, [active, delay]);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.focus();
        } else {
          rootRef.current?.focus();
        }
      },
      blur() {
        if (isBusy && overlayRef.current) {
          overlayRef.current.blur();
        } else {
          rootRef.current?.blur();
        }
      },
      isFocused() {
        return document.activeElement === overlayRef.current || document.activeElement === rootRef.current;
      },
      get nativeElement() {
        return rootRef.current;
      }
    }),
    [isBusy]
  );
  const textLabel = text ? /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      id: labelId,
      className: cn(
        "w-full text-center text-sapphire-text-tertiary text-sm font-medium",
        isTopPlacement ? "mb-2" : "mt-2"
      ),
      children: text
    }
  ) : null;
  const dots = /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cn(
        "inline-flex items-center leading-none",
        dotGapClasses[normalizedSize]
      ),
      children: [0, 1, 2].map((i) => /* @__PURE__ */ jsxRuntime.jsx(
        "span",
        {
          className: dotVariants({ size: normalizedSize }),
          style: {
            animation: `busy-grow 1.6s cubic-bezier(0.32, 0.06, 0.85, 1.11) ${i * 200}ms infinite`
          }
        },
        i
      ))
    }
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      ref: rootRef,
      id,
      className: cn(
        "inline-block",
        hasChildren && "relative",
        isBusy && "text-ring",
        className
      ),
      style: { ...style, borderRadius: "inherit", height: "inherit" },
      "data-testid": dataTestId,
      "data-part": "root",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("style", { children: `@keyframes busy-grow{0%,50%,100%{transform:scale(.5)}25%{transform:scale(1)}}` }),
        isBusy && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            ref: overlayRef,
            role: "progressbar",
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            "aria-valuetext": "Busy",
            "aria-labelledby": text ? labelId : void 0,
            "aria-label": !text ? accessibleName ?? "Please wait" : void 0,
            title: !text ? accessibleName ?? "Please wait" : void 0,
            tabIndex: 0,
            "data-sap-focus-ref": true,
            className: cn(
              "flex flex-col items-center justify-center",
              "bg-inherit rounded-[inherit]",
              "focus-visible:outline focus-visible:outline-2 focus-visible:outline-ring focus-visible:outline-offset-[-2px]",
              hasChildren && "absolute inset-0 z-[99]"
            ),
            children: [
              isTopPlacement && textLabel,
              dots,
              !isTopPlacement && textLabel
            ]
          }
        ),
        children && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            className: cn(isBusy && "opacity-40"),
            ...isBusy ? { inert: true } : {},
            children
          }
        )
      ]
    }
  );
}
var filledBrandStyles = [
  "bg-sapphire-brand-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-brand-hover-background",
  "active:bg-sapphire-brand-pressed-background [&.kb-active]:bg-sapphire-brand-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-focus btn-filled-focus"
];
var filledJouleStyles = [
  "bg-sapphire-joule-background text-sapphire-neutral-foreground-white font-semibold",
  "hover:bg-sapphire-joule-hover-background",
  "active:bg-sapphire-joule-pressed-background [&.kb-active]:bg-sapphire-joule-pressed-background",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule btn-filled-focus"
];
var outlineBrandStyles = [
  "bg-transparent text-sapphire-brand-foreground border border-sapphire-brand-foreground",
  "hover:bg-sapphire-neutral-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-brand-pressed-foreground active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-brand-pressed-foreground [&.kb-active]:text-sapphire-brand-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
];
var outlineJouleStyles = [
  "bg-transparent text-sapphire-joule-foreground border border-sapphire-joule-foreground",
  "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background hover:border-sapphire-joule-hover-background",
  "active:bg-sapphire-neutral-pressed-background active:border-sapphire-joule-pressed-foreground active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:border-sapphire-joule-pressed-foreground [&.kb-active]:text-sapphire-joule-pressed-foreground",
  "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
];
var buttonVariants = cva(
  [
    // Base styles
    "inline-flex items-center justify-center gap-2",
    "font-semibold whitespace-nowrap",
    "transition-[color,background-color,border-color,text-decoration-color,fill,stroke] cursor-pointer",
    "outline-none",
    "disabled:pointer-events-none disabled:opacity-40"
  ],
  {
    variants: {
      design: {
        // --- Primary (filled) ---
        ["Primary" /* Primary */]: filledBrandStyles,
        ["PrimaryJoule" /* PrimaryJoule */]: filledJouleStyles,
        // --- Secondary (outlined) ---
        ["Secondary" /* Secondary */]: outlineBrandStyles,
        ["SecondaryJoule" /* SecondaryJoule */]: outlineJouleStyles,
        // --- Tertiary (ghost, colored text) ---
        ["Tertiary" /* Tertiary */]: [
          "bg-transparent text-sapphire-brand-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-brand-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-brand-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-ring"
        ],
        ["TertiaryJoule" /* TertiaryJoule */]: [
          "bg-transparent text-sapphire-joule-foreground border border-transparent",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-joule-hover-background",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-joule-pressed-foreground [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-joule-pressed-foreground",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-sapphire-border-joule"
        ],
        // --- Neutral (ghost, dark text) ---
        ["Neutral" /* Neutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-sapphire-neutral-foreground-muted",
          "hover:bg-sapphire-neutral-hover-background hover:text-sapphire-neutral-foreground-black",
          "active:bg-sapphire-neutral-pressed-background active:text-sapphire-neutral-foreground-black [&.kb-active]:bg-sapphire-neutral-pressed-background [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- SecondaryNeutral (ghost, neutral-black text) ---
        ["SecondaryNeutral" /* SecondaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-black border border-transparent",
          "hover:bg-sapphire-neutral-hover-background",
          "active:bg-sapphire-neutral-pressed-background [&.kb-active]:bg-sapphire-neutral-pressed-background",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ],
        // --- TertiaryNeutral (ghost, muted neutral text, no background) ---
        ["TertiaryNeutral" /* TertiaryNeutral */]: [
          "bg-transparent text-sapphire-neutral-foreground-muted border border-transparent",
          "hover:text-sapphire-neutral-foreground-black",
          "active:text-sapphire-neutral-foreground-black [&.kb-active]:text-sapphire-neutral-foreground-black",
          "focus:[outline-style:solid] focus:outline-2 focus:-outline-offset-2 focus:outline-foreground"
        ]
      }
    },
    defaultVariants: {
      design: "Secondary" /* Secondary */
    }
  }
);
var largeSizeClasses = {
  default: "h-10 px-4 py-3 text-base gap-2 rounded-lg",
  iconOnly: "h-10 w-10 p-0 text-base rounded-lg",
  icon: "h-5 w-5 [&>svg]:h-[17.5px] [&>svg]:w-[17.5px]"
};
var mediumSizeClasses = {
  default: "h-8 px-3 py-2 text-sm gap-1.5 rounded",
  iconOnly: "h-8 w-8 p-0 text-sm rounded",
  icon: "h-4 w-4 [&>svg]:h-3.5 [&>svg]:w-3.5"
};
var smallSizeClasses = {
  default: "h-6 px-2 py-1 text-xs font-normal gap-1 rounded",
  iconOnly: "h-6 w-6 p-0 text-xs rounded",
  icon: "h-3 w-3 [&>svg]:h-[10.5px] [&>svg]:w-[10.5px]"
};
var sizeClasses = {
  ["Large" /* Large */]: largeSizeClasses,
  ["Medium" /* Medium */]: mediumSizeClasses,
  ["Small" /* Small */]: smallSizeClasses
};
function getButtonType(type) {
  switch (type) {
    case "Submit" /* Submit */:
    case "Submit":
      return "submit";
    case "Reset" /* Reset */:
    case "Reset":
      return "reset";
    default:
      return "button";
  }
}
function Button({
  design = "Secondary" /* Secondary */,
  size = "Large" /* Large */,
  type = "Button" /* Button */,
  disabled = false,
  loading = false,
  icon,
  endIcon,
  iconOnly = false,
  children,
  tooltip,
  states,
  state,
  stateTransition = "slide-up",
  accessibleName,
  accessibleNameRef,
  accessibleRole = "Button" /* Button */,
  accessibilityAttributes,
  name,
  value,
  form,
  onClick,
  onFocus,
  onBlur,
  className,
  style,
  id,
  tabIndex,
  "data-testid": dataTestId,
  "data-growing-button": dataGrowingButton,
  ref
}) {
  const buttonRef = React2.useRef(null);
  const hiddenButtonRef = React2.useRef(null);
  const [animationPhase, setAnimationPhase] = React2.useState("idle");
  const [displayedState, setDisplayedState] = React2.useState(
    null
  );
  const [buttonWidth, setButtonWidth] = React2.useState(void 0);
  const currentStateObject = states?.find((s) => s.name === state) ?? states?.[0];
  const isMultiState = states && states.length > 0;
  React2.useEffect(() => {
    if (isMultiState && !displayedState) {
      setDisplayedState(currentStateObject ?? null);
    }
  }, [isMultiState, currentStateObject, displayedState]);
  React2.useEffect(() => {
    if (!isMultiState || !currentStateObject || animationPhase !== "idle") {
      return;
    }
    if (displayedState && displayedState.name !== currentStateObject.name) {
      if (buttonRef.current) {
        setButtonWidth(buttonRef.current.offsetWidth);
      }
      setAnimationPhase("fade-out");
    }
  }, [currentStateObject, displayedState, isMultiState, animationPhase]);
  React2.useEffect(() => {
    if (animationPhase === "fade-out") {
      const timer = setTimeout(() => {
        setAnimationPhase("fade-mid");
      }, 180);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-mid") {
      setDisplayedState(currentStateObject ?? null);
      if (hiddenButtonRef.current) {
        setButtonWidth(hiddenButtonRef.current.offsetWidth);
      }
      const timer = setTimeout(() => {
        setAnimationPhase("fade-in");
      }, 20);
      return () => clearTimeout(timer);
    }
    if (animationPhase === "fade-in") {
      const timer = setTimeout(() => {
        setAnimationPhase("idle");
        setButtonWidth(void 0);
      }, 160);
      return () => clearTimeout(timer);
    }
  }, [animationPhase, currentStateObject]);
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        buttonRef.current?.focus();
      },
      blur() {
        buttonRef.current?.blur();
      },
      isFocused() {
        return document.activeElement === buttonRef.current;
      },
      get nativeElement() {
        return buttonRef.current;
      }
    }),
    []
  );
  const handleClick = React2.useCallback(
    (e) => {
      if (disabled || loading) {
        e.preventDefault();
        return;
      }
      onClick?.({
        originalEvent: e,
        isKeyboard: false
      });
    },
    [disabled, loading, onClick]
  );
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (disabled || loading) return;
      if (e.key === "Enter" || e.key === " ") {
        if (e.key === " ") {
          e.preventDefault();
        }
        buttonRef.current?.classList.add("kb-active");
        if (e.key === " ") {
          onClick?.({
            originalEvent: e,
            isKeyboard: true
          });
        }
      }
    },
    [disabled, loading, onClick]
  );
  const handleKeyUp = React2.useCallback(
    (e) => {
      if (e.key === "Enter" || e.key === " ") {
        buttonRef.current?.classList.remove("kb-active");
      }
    },
    []
  );
  const ariaExpanded = accessibilityAttributes?.expanded;
  const ariaHasPopup = accessibilityAttributes?.hasPopup;
  const ariaControls = accessibilityAttributes?.controls;
  const ariaPressed = accessibilityAttributes?.pressed;
  const role = accessibleRole === "Link" /* Link */ || accessibleRole === "Link" ? "link" : void 0;
  const isDisabled = disabled;
  const effectiveIcon = isMultiState ? displayedState?.icon : icon;
  const effectiveEndIcon = isMultiState ? displayedState?.endIcon : endIcon;
  const effectiveChildren = isMultiState ? displayedState?.text : children;
  const effectiveIconOnly = isMultiState ? !displayedState?.text && !!displayedState?.icon : iconOnly;
  const effectivePulsing = isMultiState ? displayedState?.pulsing : false;
  const PulsingText = ({ children: text }) => /* @__PURE__ */ jsxRuntime.jsxs("span", { className: "relative inline-block", children: [
    /* @__PURE__ */ jsxRuntime.jsx("span", { className: "opacity-50", children: text }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "absolute inset-0 overflow-hidden",
        style: {
          background: "linear-gradient(90deg, transparent 0%, currentColor 50%, transparent 100%)",
          backgroundSize: "200% 100%",
          WebkitBackgroundClip: "text",
          backgroundClip: "text",
          WebkitTextFillColor: "transparent",
          animation: "pulse-sweep 1.5s ease-in-out infinite"
        },
        children: text
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: `
          @keyframes pulse-sweep {
            0% { background-position: 200% 0; }
            100% { background-position: -200% 0; }
          }
        ` })
  ] });
  const getTransitionStyles = (phase, transition) => {
    if (transition === "none" || phase === "idle") {
      return {};
    }
    const baseTransition = "all 100ms ease-out";
    switch (transition) {
      case "slide-up":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(-8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-down":
        if (phase === "fade-out") return { opacity: 0, transform: "translateY(8px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateY(-8px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateY(0)", transition: baseTransition };
        break;
      case "slide-left":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(-12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "slide-right":
        if (phase === "fade-out") return { opacity: 0, transform: "translateX(12px)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "translateX(-12px)" };
        if (phase === "fade-in") return { opacity: 1, transform: "translateX(0)", transition: baseTransition };
        break;
      case "fade":
        if (phase === "fade-out") return { opacity: 0, transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0 };
        if (phase === "fade-in") return { opacity: 1, transition: baseTransition };
        break;
      case "scale":
        if (phase === "fade-out") return { opacity: 0, transform: "scale(0.75)", transition: baseTransition };
        if (phase === "fade-mid") return { opacity: 0, transform: "scale(0.75)" };
        if (phase === "fade-in") return { opacity: 1, transform: "scale(1)", transition: baseTransition };
        break;
      case "flip":
        if (phase === "fade-out") return { opacity: 0, transform: "rotateX(90deg)", transition: baseTransition, transformStyle: "preserve-3d" };
        if (phase === "fade-mid") return { opacity: 0, transform: "rotateX(-90deg)", transformStyle: "preserve-3d" };
        if (phase === "fade-in") return { opacity: 1, transform: "rotateX(0)", transition: baseTransition, transformStyle: "preserve-3d" };
        break;
    }
    return {};
  };
  const contentAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  const iconAnimationStyles = getTransitionStyles(animationPhase, stateTransition);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: buttonRef,
        type: getButtonType(type),
        disabled: isDisabled,
        "aria-disabled": loading || void 0,
        id,
        name,
        value,
        form,
        role,
        "aria-label": accessibleName,
        "aria-labelledby": accessibleNameRef,
        "aria-expanded": ariaExpanded,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-pressed": ariaPressed,
        "aria-busy": loading || void 0,
        title: tooltip,
        tabIndex,
        onClick: handleClick,
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        onFocus,
        onBlur: (e) => {
          buttonRef.current?.classList.remove("kb-active");
          onBlur?.(e);
        },
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][effectiveIconOnly ? "iconOnly" : "default"],
          isMultiState && !effectiveIconOnly && "transition-[width] duration-300 ease-out overflow-hidden",
          loading && "relative pointer-events-none",
          className
        ),
        style: {
          ...style,
          ...buttonWidth !== void 0 && !effectiveIconOnly ? { width: buttonWidth } : {}
        },
        "data-testid": dataTestId,
        "data-growing-button": dataGrowingButton,
        children: [
          loading && /* @__PURE__ */ jsxRuntime.jsx("span", { className: "absolute inset-0 flex items-center justify-center z-[1] [&_*]:!text-inherit", inert: true, children: /* @__PURE__ */ jsxRuntime.jsx(BusyIndicator, { size: "S", active: true, delay: 0 }) }),
          effectiveIcon ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center shrink-0", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? iconAnimationStyles : void 0, children: effectiveIcon }) : null,
          !effectiveIconOnly && effectiveChildren && /* @__PURE__ */ jsxRuntime.jsx("span", { className: loading ? "opacity-40" : void 0, style: isMultiState ? contentAnimationStyles : void 0, children: effectivePulsing ? /* @__PURE__ */ jsxRuntime.jsx(PulsingText, { children: effectiveChildren }) : effectiveChildren }),
          effectiveEndIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center", sizeClasses[size].icon, loading && "opacity-40"), style: isMultiState ? contentAnimationStyles : void 0, children: effectiveEndIcon })
        ]
      }
    ),
    isMultiState && /* @__PURE__ */ jsxRuntime.jsxs(
      "button",
      {
        ref: hiddenButtonRef,
        className: cn(
          buttonVariants({
            design
          }),
          sizeClasses[size][!currentStateObject?.text && currentStateObject?.icon ? "iconOnly" : "default"],
          "absolute -top-[10000px] -left-[10000px] invisible w-fit"
        ),
        tabIndex: -1,
        "aria-hidden": "true",
        children: [
          currentStateObject?.icon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.icon }),
          currentStateObject?.text && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.text }),
          currentStateObject?.endIcon && /* @__PURE__ */ jsxRuntime.jsx("span", { children: currentStateObject.endIcon })
        ]
      }
    )
  ] });
}
function buildFileList(files) {
  if (typeof DataTransfer !== "undefined") {
    const dt = new DataTransfer();
    files.forEach((f) => dt.items.add(f));
    return dt.files;
  }
  const list = Object.create(FileList.prototype);
  files.forEach((f, i) => {
    list[i] = f;
  });
  Object.defineProperty(list, "length", { value: files.length });
  list.item = (index) => list[index] ?? null;
  list[Symbol.iterator] = function* () {
    for (let i = 0; i < files.length; i++) yield list[i];
  };
  return list;
}
var descIdCounter = 0;
function useStableId(prefix) {
  const ref = React2.useRef(`${prefix}-${++descIdCounter}`);
  return ref.current;
}
var containerVariants = cva(
  [
    "flex items-center w-full border bg-sapphire-canvas-primary",
    "overflow-clip transition-colors relative",
    "pr-1 py-1 gap-1",
    "cursor-pointer outline-none"
  ],
  {
    variants: {
      size: {
        ["Large" /* Large */]: [
          "h-10 rounded-lg pl-1",
          "focus:border-2 focus:pl-[3px] focus:pr-[3px] focus:py-[3px]"
        ],
        ["Medium" /* Medium */]: [
          "h-8 rounded pl-0.5",
          "focus:border-2 focus:pl-px focus:pr-[3px] focus:py-[3px]"
        ]
      },
      valueState: {
        ["None" /* None */]: "border-sapphire-border-active hover:border-sapphire-border-accent focus:border-sapphire-border-accent",
        ["Positive" /* Positive */]: "bg-sapphire-positive-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-positive focus:border-sapphire-positive",
        ["Negative" /* Negative */]: "bg-sapphire-negative-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-negative focus:border-sapphire-negative",
        ["Critical" /* Critical */]: "bg-sapphire-warning-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-warning focus:border-sapphire-warning",
        ["Information" /* Information */]: "bg-sapphire-info-bg hover:bg-sapphire-canvas-primary focus:bg-sapphire-canvas-primary border-sapphire-info focus:border-sapphire-info"
      },
      disabled: {
        true: "opacity-40 cursor-not-allowed pointer-events-none",
        false: ""
      },
      dragOver: {
        true: "border-sapphire-border-accent border-2 pl-[3px] pr-[3px] py-[3px]",
        false: ""
      }
    },
    defaultVariants: {
      size: "Large" /* Large */,
      valueState: "None" /* None */,
      disabled: false,
      dragOver: false
    }
  }
);
var valueStateIcons = {
  ["None" /* None */]: null,
  ["Positive" /* Positive */]: SysEnter2Icon,
  ["Negative" /* Negative */]: ErrorIcon,
  ["Critical" /* Critical */]: WarningIcon,
  ["Information" /* Information */]: InformationIcon
};
var valueStateMessageColors = {
  ["None" /* None */]: "",
  ["Positive" /* Positive */]: "text-sapphire-positive",
  ["Negative" /* Negative */]: "text-sapphire-negative",
  ["Critical" /* Critical */]: "text-sapphire-warning",
  ["Information" /* Information */]: "text-sapphire-info"
};
function FileUploaderBase({
  size = "Large" /* Large */,
  accept,
  multiple = false,
  maxFileSize,
  hideInput = false,
  placeholder,
  value,
  disabled = false,
  required = false,
  valueState = "None" /* None */,
  valueStateMessage,
  name,
  accessibleName,
  accessibleNameRef,
  accessibleDescription,
  accessibleDescriptionRef,
  onChange,
  onFileSizeExceed,
  children,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const containerRef = React2.useRef(null);
  const fileInputRef = React2.useRef(null);
  const dragCounterRef = React2.useRef(0);
  const fileNamesRef = React2.useRef([]);
  const tokenAreaRef = React2.useRef(null);
  const [fileNames, setFileNames] = React2.useState([]);
  const [isDragOver, setIsDragOver] = React2.useState(false);
  const [isFocused, setIsFocused] = React2.useState(false);
  const [focusedTokenIndex, setFocusedTokenIndex] = React2.useState(-1);
  const { t } = reactI18next.useTranslation("fx");
  const descriptionId = useStableId("fu-desc");
  const resolvedValueState = valueState;
  const isNegativeState = resolvedValueState === "Negative" /* Negative */;
  const isMedium = size === "Medium" /* Medium */;
  const resolvedPlaceholder = placeholder ?? (multiple ? t("FILEUPLOADER_PLACEHOLDER_MULTIPLE") : t("FILEUPLOADER_PLACEHOLDER"));
  const displayNames = React2.useMemo(() => {
    if (value !== void 0 && value !== "") {
      return value.split(",").map((n) => n.trim()).filter(Boolean);
    }
    return fileNames;
  }, [value, fileNames]);
  const hasFiles = displayNames.length > 0;
  const isMultiDisplay = displayNames.length > 1;
  const [visibleCount, setVisibleCount] = React2.useState(displayNames.length);
  const nMoreLabelRef = React2.useRef(null);
  React2.useEffect(() => {
    setVisibleCount(displayNames.length);
  }, [displayNames.length]);
  React2.useEffect(() => {
    if (!isMultiDisplay || isFocused || !tokenAreaRef.current) return;
    const area = tokenAreaRef.current;
    const calculateVisible = () => {
      if (!area) return;
      const areaWidth = area.getBoundingClientRect().width;
      if (areaWidth === 0) return;
      const nMoreWidth = nMoreLabelRef.current ? nMoreLabelRef.current.getBoundingClientRect().width + 8 : 60;
      const tokenEls = Array.from(
        area.querySelectorAll('[data-part="root"]')
      );
      const gap = 4;
      let usedWidth = 0;
      let fitCount = 0;
      for (let i = 0; i < tokenEls.length; i++) {
        const el = tokenEls[i];
        const wasHidden = el.classList.contains("hidden");
        if (wasHidden) el.classList.remove("hidden");
        const w = el.getBoundingClientRect().width;
        if (wasHidden) el.classList.add("hidden");
        const needed = w + (i > 0 ? gap : 0);
        const remaining = tokenEls.length - i - 1;
        const reserve = remaining > 0 ? nMoreWidth : 0;
        if (usedWidth + needed + reserve > areaWidth && fitCount > 0) {
          break;
        }
        usedWidth += needed;
        fitCount++;
      }
      setVisibleCount(fitCount);
    };
    const observer = new ResizeObserver(() => {
      requestAnimationFrame(calculateVisible);
    });
    observer.observe(area);
    calculateVisible();
    return () => observer.disconnect();
  }, [isMultiDisplay, isFocused, displayNames]);
  const overflowCount = isMultiDisplay ? Math.max(0, displayNames.length - visibleCount) : 0;
  const validateFiles = React2.useCallback(
    (files) => {
      if (maxFileSize === void 0 || maxFileSize <= 0) return files;
      const exceededFiles = [];
      const validFiles = [];
      for (let i = 0; i < files.length; i++) {
        const file = files[i];
        const fileSizeMB = file.size / 1024 / 1024;
        if (fileSizeMB > maxFileSize) {
          exceededFiles.push({ fileName: file.name, fileSize: fileSizeMB });
        } else {
          validFiles.push(file);
        }
      }
      if (exceededFiles.length > 0) {
        onFileSizeExceed?.({ filesData: exceededFiles });
      }
      if (validFiles.length === 0 && exceededFiles.length > 0) {
        return null;
      }
      return buildFileList(validFiles);
    },
    [maxFileSize, onFileSizeExceed]
  );
  const filterByAccept = React2.useCallback(
    (files) => {
      if (!accept) return files;
      const acceptTypes = accept.split(",").map((t2) => t2.trim().toLowerCase());
      const matched = [];
      for (let i = 0; i < files.length; i++) {
        const file = files[i];
        const parts = file.name.split(".");
        const fileExt = parts.length > 1 ? "." + parts.pop()?.toLowerCase() : "";
        const fileMime = file.type.toLowerCase();
        const matches = acceptTypes.some((type) => {
          if (type.startsWith(".")) {
            return fileExt === type;
          }
          if (type.endsWith("/*")) {
            return fileMime.startsWith(type.slice(0, -1));
          }
          return fileMime === type;
        });
        if (matches) {
          matched.push(file);
        }
      }
      return buildFileList(matched);
    },
    [accept]
  );
  const openFileBrowser = React2.useCallback(() => {
    if (disabled) return;
    if (fileInputRef.current) {
      fileInputRef.current.value = "";
    }
    fileInputRef.current?.click();
  }, [disabled]);
  const clearFiles = React2.useCallback(() => {
    if (fileInputRef.current) {
      fileInputRef.current.value = "";
    }
    fileNamesRef.current = [];
    setFileNames([]);
    onChange?.({ files: null });
  }, [onChange]);
  const handleNativeChange = React2.useCallback(
    (e) => {
      const files = e.target.files;
      if (!files || files.length === 0) {
        if (fileNamesRef.current.length > 0) {
          clearFiles();
        }
        return;
      }
      const validFiles = validateFiles(files);
      if (!validFiles || validFiles.length === 0) {
        return;
      }
      const names = Array.from(validFiles).map((f) => f.name);
      fileNamesRef.current = names;
      setFileNames(names);
      onChange?.({ files: validFiles });
    },
    [validateFiles, clearFiles, onChange]
  );
  const handleContainerClick = React2.useCallback(() => {
    openFileBrowser();
  }, [openFileBrowser]);
  const handleKeyDown = React2.useCallback(
    (e) => {
      if (disabled) return;
      switch (e.key) {
        case "Enter":
        case " ":
          e.preventDefault();
          openFileBrowser();
          break;
        case "F4":
          e.preventDefault();
          openFileBrowser();
          break;
        case "Delete":
          e.preventDefault();
          if (hasFiles) {
            clearFiles();
          }
          break;
        case "ArrowRight":
          if (displayNames.length > 1 && isFocused) {
            e.preventDefault();
            setFocusedTokenIndex(0);
            const tokens = tokenAreaRef.current?.querySelectorAll('[data-part="root"]');
            tokens?.[0]?.focus();
          }
          break;
      }
      if (e.altKey && (e.key === "ArrowUp" || e.key === "ArrowDown")) {
        e.preventDefault();
        openFileBrowser();
      }
    },
    [disabled, openFileBrowser, hasFiles, clearFiles, displayNames.length, isFocused]
  );
  const handleTokenKeyDown = React2.useCallback(
    (e, index) => {
      if (e.key === "ArrowRight") {
        e.preventDefault();
        e.stopPropagation();
        if (index < displayNames.length - 1) {
          const next = index + 1;
          setFocusedTokenIndex(next);
          const tokens = tokenAreaRef.current?.querySelectorAll('[data-part="root"]');
          const el = tokens?.[next];
          el?.focus();
          el?.scrollIntoView({ block: "nearest", inline: "nearest" });
        }
      } else if (e.key === "ArrowLeft") {
        e.preventDefault();
        e.stopPropagation();
        if (index > 0) {
          const prev = index - 1;
          setFocusedTokenIndex(prev);
          const tokens = tokenAreaRef.current?.querySelectorAll('[data-part="root"]');
          const el = tokens?.[prev];
          el?.focus();
          el?.scrollIntoView({ block: "nearest", inline: "nearest" });
        } else {
          setFocusedTokenIndex(-1);
          containerRef.current?.focus();
        }
      }
    },
    [displayNames.length]
  );
  const handleDragOver = React2.useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
    },
    [disabled]
  );
  const handleDragEnter = React2.useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
      dragCounterRef.current++;
      setIsDragOver(true);
    },
    [disabled]
  );
  const handleDragLeave = React2.useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
      dragCounterRef.current--;
      if (dragCounterRef.current <= 0) {
        dragCounterRef.current = 0;
        setIsDragOver(false);
      }
    },
    [disabled]
  );
  const handleDrop = React2.useCallback(
    (e) => {
      if (disabled) return;
      e.preventDefault();
      e.stopPropagation();
      dragCounterRef.current = 0;
      setIsDragOver(false);
      let droppedFiles = e.dataTransfer.files;
      if (!droppedFiles || droppedFiles.length === 0) return;
      if (!multiple && droppedFiles.length > 1) {
        droppedFiles = buildFileList([droppedFiles[0]]);
      }
      droppedFiles = filterByAccept(droppedFiles);
      if (droppedFiles.length === 0) return;
      const validFiles = validateFiles(droppedFiles);
      if (!validFiles || validFiles.length === 0) return;
      const names = Array.from(validFiles).map((f) => f.name);
      fileNamesRef.current = names;
      setFileNames(names);
      onChange?.({ files: validFiles });
    },
    [disabled, multiple, filterByAccept, validateFiles, onChange]
  );
  const handleClearClick = React2.useCallback(
    (detail) => {
      detail.originalEvent.stopPropagation();
      clearFiles();
    },
    [clearFiles]
  );
  const handleBrowseClick = React2.useCallback(
    (detail) => {
      detail.originalEvent.stopPropagation();
      openFileBrowser();
    },
    [openFileBrowser]
  );
  React2.useImperativeHandle(
    ref,
    () => ({
      focus() {
        containerRef.current?.focus();
      },
      blur() {
        containerRef.current?.blur();
      },
      openFileBrowser() {
        openFileBrowser();
      },
      clear() {
        clearFiles();
      },
      get files() {
        return fileInputRef.current?.files ?? null;
      },
      get nativeInput() {
        return fileInputRef.current;
      }
    }),
    [openFileBrowser, clearFiles]
  );
  const ValueStateIcon = valueStateIcons[resolvedValueState];
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      className: cn("w-full", className),
      style,
      id,
      "data-testid": dataTestId,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "input",
          {
            ref: fileInputRef,
            type: "file",
            accept,
            multiple,
            name,
            required,
            disabled,
            tabIndex: -1,
            "aria-hidden": "true",
            className: "sr-only",
            onChange: handleNativeChange,
            "data-testid": dataTestId ? `${dataTestId}-input` : void 0
          }
        ),
        hideInput ? (
          /* Button-only mode: render children as trigger */
          /* @__PURE__ */ jsxRuntime.jsx(
            "div",
            {
              role: "button",
              tabIndex: disabled ? -1 : 0,
              onClick: disabled ? void 0 : handleContainerClick,
              onKeyDown: disabled ? void 0 : handleKeyDown,
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-disabled": disabled || void 0,
              children
            }
          )
        ) : (
          /* Standard mode: styled container */
          /* @__PURE__ */ jsxRuntime.jsxs(
            "div",
            {
              ref: containerRef,
              tabIndex: disabled ? -1 : 0,
              role: "group",
              "aria-roledescription": t("FILEUPLOADER_ROLE_DESCRIPTION"),
              "aria-label": accessibleName,
              "aria-labelledby": accessibleNameRef,
              "aria-describedby": accessibleDescription ? descriptionId : accessibleDescriptionRef,
              "aria-required": required || void 0,
              "aria-invalid": isNegativeState || void 0,
              "aria-haspopup": "dialog",
              className: cn(
                containerVariants({
                  size,
                  valueState: resolvedValueState,
                  disabled,
                  dragOver: isDragOver
                })
              ),
              onClick: handleContainerClick,
              onKeyDown: handleKeyDown,
              onFocus: () => setIsFocused(true),
              onBlur: (e) => {
                if (!containerRef.current?.contains(e.relatedTarget)) {
                  setIsFocused(false);
                  setFocusedTokenIndex(-1);
                }
              },
              onDragOver: handleDragOver,
              onDragEnter: handleDragEnter,
              onDragLeave: handleDragLeave,
              onDrop: handleDrop,
              children: [
                accessibleDescription && /* @__PURE__ */ jsxRuntime.jsx("span", { id: descriptionId, className: "sr-only", children: accessibleDescription }),
                /* @__PURE__ */ jsxRuntime.jsx(
                  "div",
                  {
                    ref: tokenAreaRef,
                    className: cn(
                      "flex-1 min-w-0 flex items-center gap-1",
                      isFocused && isMultiDisplay ? "overflow-x-auto scrollbar-none" : "overflow-hidden"
                    ),
                    children: hasFiles ? displayNames.length === 1 ? /* @__PURE__ */ jsxRuntime.jsx(Token, { text: displayNames[0], readonly: true, singleToken: true, className: "shrink min-w-0" }) : isFocused ? displayNames.map((fileName, idx) => /* @__PURE__ */ jsxRuntime.jsx(
                      "span",
                      {
                        onKeyDown: (e) => handleTokenKeyDown(e, idx),
                        className: "shrink-0",
                        children: /* @__PURE__ */ jsxRuntime.jsx(
                          Token,
                          {
                            text: fileName,
                            readonly: true,
                            tabIndex: idx === focusedTokenIndex ? 0 : -1
                          }
                        )
                      },
                      fileName
                    )) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
                      displayNames.map((fileName, idx) => /* @__PURE__ */ jsxRuntime.jsx(
                        Token,
                        {
                          text: fileName,
                          readonly: true,
                          className: idx >= visibleCount ? "hidden" : "shrink-0"
                        },
                        fileName
                      )),
                      overflowCount > 0 && /* @__PURE__ */ jsxRuntime.jsx(
                        "span",
                        {
                          ref: nMoreLabelRef,
                          className: "inline-block whitespace-nowrap shrink-0 text-sm font-normal text-sapphire-brand-foreground cursor-pointer hover:underline",
                          children: t("TOKENIZER_SHOW_MORE", { count: overflowCount })
                        }
                      )
                    ] }) : /* @__PURE__ */ jsxRuntime.jsx("span", { className: "text-sapphire-text-tertiary italic text-sm truncate select-none pl-2.5", children: resolvedPlaceholder })
                  }
                ),
                hasFiles && !disabled && /* @__PURE__ */ jsxRuntime.jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsxRuntime.jsx(
                  Button,
                  {
                    design: "SecondaryNeutral" /* SecondaryNeutral */,
                    size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsxRuntime.jsx(DeclineIcon, { className: "h-full w-full" }) }),
                    tabIndex: -1,
                    onClick: handleClearClick,
                    accessibleName: t("FILEUPLOADER_CLEAR_TOOLTIP"),
                    className: "shrink-0",
                    "data-part": "clear"
                  }
                ) }),
                /* @__PURE__ */ jsxRuntime.jsx("span", { onMouseDown: (e) => e.preventDefault(), children: /* @__PURE__ */ jsxRuntime.jsx(
                  Button,
                  {
                    design: "SecondaryNeutral" /* SecondaryNeutral */,
                    size: isMedium ? "Small" /* Small */ : "Medium" /* Medium */,
                    iconOnly: true,
                    icon: /* @__PURE__ */ jsxRuntime.jsx("span", { className: cn("inline-flex items-center justify-center", isMedium ? "h-3 w-3" : "h-4 w-4"), children: /* @__PURE__ */ jsxRuntime.jsx(BrowseFolderIcon, { className: "h-full w-full" }) }),
                    tabIndex: -1,
                    onClick: handleBrowseClick,
                    accessibleName: t("FILEUPLOADER_BROWSE_TOOLTIP"),
                    className: "shrink-0",
                    "data-part": "browse"
                  }
                ) })
              ]
            }
          )
        ),
        valueStateMessage && resolvedValueState !== "None" /* None */ && /* @__PURE__ */ jsxRuntime.jsxs(
          "div",
          {
            role: isNegativeState ? "alert" : void 0,
            "aria-live": isNegativeState ? "assertive" : "polite",
            className: cn(
              "mt-1 text-xs flex items-start gap-1",
              valueStateMessageColors[resolvedValueState]
            ),
            children: [
              ValueStateIcon && /* @__PURE__ */ jsxRuntime.jsx(ValueStateIcon, { className: "h-3 w-3 mt-0.5 shrink-0" }),
              /* @__PURE__ */ jsxRuntime.jsx("span", { children: valueStateMessage })
            ]
          }
        )
      ]
    }
  );
}
var FileUploader = React2__default.default.memo(FileUploaderBase);
FileUploader.displayName = "FileUploader";

exports.FileUploader = FileUploader;
