// @sap-ui/fx-components/dist/components/text/index.cjs
'use strict';

var react = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

// src/components/text/Text.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
function isEmpty(children) {
  if (children == null) return true;
  if (typeof children === "string" && children.trim() === "") return true;
  if (Array.isArray(children) && children.length === 0) return true;
  return false;
}
function Text({
  maxLines = 0,
  emptyIndicatorMode = "Off" /* Off */,
  children,
  className,
  style,
  ref
}) {
  const innerRef = react.useRef(null);
  react.useImperativeHandle(
    ref,
    () => ({
      focus: () => innerRef.current?.focus(),
      get nativeElement() {
        return innerRef.current;
      }
    }),
    []
  );
  const showEmptyIndicator = emptyIndicatorMode === "On" /* On */ && isEmpty(children);
  const clampStyles = maxLines === 1 ? { maxWidth: "100%" } : maxLines > 1 ? {
    display: "-webkit-inline-box",
    WebkitLineClamp: maxLines,
    lineClamp: maxLines,
    WebkitBoxOrient: "vertical"
  } : {};
  return /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      ref: innerRef,
      className: cn(
        "text-foreground font-normal text-sm",
        maxLines === 1 && "inline-block overflow-hidden text-ellipsis whitespace-nowrap",
        maxLines > 1 && "overflow-hidden",
        className
      ),
      style: { ...clampStyles, ...style },
      children: showEmptyIndicator ? /* @__PURE__ */ jsxRuntime.jsx("span", { className: "text-sapphire-text-tertiary", children: "\u2013" }) : children
    }
  );
}

exports.Text = Text;
