// @sap-ui/fx-components/dist/components/illustrated-message/index.cjs
'use strict';

var React = require('react');
var tailwindMerge = require('tailwind-merge');
var jsxRuntime = require('react/jsx-runtime');

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);

// src/components/illustrated-message/IllustratedMessage.tsx
function toVal(mix) {
  let str = "";
  if (typeof mix === "string" || typeof mix === "number") {
    str += mix;
  } else if (typeof mix === "object") {
    if (Array.isArray(mix)) {
      for (let i = 0; i < mix.length; i++) {
        if (mix[i]) {
          const y = toVal(mix[i]);
          if (y) {
            if (str) str += " ";
            str += y;
          }
        }
      }
    } else {
      for (const k in mix) {
        if (mix[k]) {
          if (str) str += " ";
          str += k;
        }
      }
    }
  }
  return str;
}
function clsx(...inputs) {
  let str = "";
  for (let i = 0; i < inputs.length; i++) {
    if (inputs[i]) {
      const y = toVal(inputs[i]);
      if (y) {
        if (str) str += " ";
        str += y;
      }
    }
  }
  return str;
}
var falsyToString = (value) => typeof value === "boolean" ? `${value}` : value === 0 ? "0" : value;
function cva(base, config) {
  return (props) => {
    if (config?.variants == null) {
      return clsx(base, props?.class, props?.className);
    }
    const { variants, defaultVariants } = config;
    const variantClassNames = Object.keys(variants).map((variant) => {
      const variantProp = props?.[variant];
      const defaultVariantProp = defaultVariants?.[variant];
      if (variantProp === null) return null;
      const variantKey = falsyToString(variantProp) || falsyToString(defaultVariantProp);
      return variants[variant]?.[variantKey];
    });
    const propsWithoutUndefined = props ? Object.entries(props).reduce((acc, [key, value]) => {
      if (value !== void 0) acc[key] = value;
      return acc;
    }, {}) : {};
    const compoundVariantClassNames = config.compoundVariants?.reduce(
      (acc, { class: cvClass, className: cvClassName, ...compoundVariantOptions }) => {
        return Object.entries(compoundVariantOptions).every(([key, value]) => {
          const merged = { ...defaultVariants, ...propsWithoutUndefined };
          return Array.isArray(value) ? value.includes(merged[key]) : merged[key] === value;
        }) ? [...acc, cvClass, cvClassName] : acc;
      },
      []
    );
    return clsx(base, variantClassNames, compoundVariantClassNames, props?.class, props?.className);
  };
}
function cn(...inputs) {
  return tailwindMerge.twMerge(clsx(inputs));
}
var IllustrationSizeContext = React.createContext(
  "Medium" /* Medium */
);
function useIllustrationSizeContext() {
  return React.useContext(IllustrationSizeContext);
}
function useIllustrationSize(containerRef, design) {
  const isAuto = design === "Auto" /* Auto */ || design === "Auto";
  const [resolvedSize, setResolvedSize] = React.useState(
    isAuto ? "Medium" /* Medium */ : design
  );
  React.useEffect(() => {
    if (!isAuto) {
      setResolvedSize(design);
      return;
    }
    const el = containerRef.current;
    if (!el || typeof ResizeObserver === "undefined") return;
    const ro = new ResizeObserver(([entry]) => {
      const width = entry.contentBoxSize?.[0]?.inlineSize ?? entry.contentRect.width;
      let newSize;
      if (width <= 160) {
        newSize = "Base" /* Base */;
      } else if (width <= 260) {
        newSize = "ExtraSmall" /* ExtraSmall */;
      } else if (width <= 360) {
        newSize = "Small" /* Small */;
      } else if (width <= 681) {
        newSize = "Medium" /* Medium */;
      } else {
        newSize = "Large" /* Large */;
      }
      setResolvedSize((prev) => prev !== newSize ? newSize : prev);
    });
    ro.observe(el);
    return () => ro.disconnect();
  }, [design, isAuto, containerRef]);
  return resolvedSize;
}
var rootVariants = cva(
  ["flex w-full h-full box-border"],
  {
    variants: {
      size: {
        ["Large" /* Large */]: "flex-col items-center justify-center text-center gap-4 p-4",
        ["Medium" /* Medium */]: "flex-col items-center justify-center text-center gap-3 p-4",
        ["Small" /* Small */]: "flex-col items-center justify-center text-center gap-2 p-2",
        ["ExtraSmall" /* ExtraSmall */]: "flex-row items-center justify-center gap-3 p-2",
        ["Base" /* Base */]: "flex-col items-center justify-center text-center gap-1 p-1"
      }
    },
    defaultVariants: {
      size: "Large" /* Large */
    }
  }
);
var titleVariants = cva("text-foreground", {
  variants: {
    size: {
      ["Large" /* Large */]: "text-[32px] font-normal leading-[40px] max-w-[61.9375rem]",
      ["Medium" /* Medium */]: "text-xl font-semibold leading-[32px] max-w-[40.5625rem]",
      ["Small" /* Small */]: "text-xl font-semibold leading-[32px] max-w-[21.5rem]",
      ["ExtraSmall" /* ExtraSmall */]: "text-base font-semibold leading-[22px]",
      ["Base" /* Base */]: "text-sm font-semibold leading-[22px] max-w-[10rem]"
    }
  }
});
var subtitleVariants = cva("text-secondary-foreground", {
  variants: {
    size: {
      ["Large" /* Large */]: "text-base leading-[22px] max-w-[61.9375rem]",
      ["Medium" /* Medium */]: "text-sm leading-[20px] max-w-[40.5625rem]",
      ["Small" /* Small */]: "text-sm leading-[20px] max-w-[21.5rem]",
      ["ExtraSmall" /* ExtraSmall */]: "text-xs leading-[20px]",
      ["Base" /* Base */]: "text-xs leading-[20px] max-w-[10rem]"
    }
  }
});
var illustrationSizeClasses = {
  ["Large" /* Large */]: "w-80 h-60",
  ["Medium" /* Medium */]: "w-40 h-40",
  ["Small" /* Small */]: "w-32 h-32",
  ["ExtraSmall" /* ExtraSmall */]: "w-[2.8125rem] h-[2.8125rem] flex-shrink-0",
  ["Base" /* Base */]: "hidden"
};
function IllustratedMessage({
  illustration,
  design = "Auto" /* Auto */,
  titleText,
  subtitleText,
  title,
  subtitle,
  children,
  decorative = false,
  accessibleName,
  className,
  style,
  id,
  "data-testid": dataTestId,
  ref
}) {
  const containerRef = React.useRef(null);
  const resolvedSize = useIllustrationSize(containerRef, design);
  React__namespace.useImperativeHandle(ref, () => ({
    get nativeElement() {
      return containerRef.current;
    }
  }), []);
  const hasTitle = !!(title || titleText);
  const hasSubtitle = !!(subtitle || subtitleText);
  const showIllustration = resolvedSize !== "Base" /* Base */;
  const showActions = resolvedSize !== "Base" /* Base */;
  const effectiveAccessibleName = accessibleName || "Illustration";
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      ref: containerRef,
      id,
      className: cn(rootVariants({ size: resolvedSize }), className),
      style,
      "data-testid": dataTestId,
      "data-part": "root",
      children: /* @__PURE__ */ jsxRuntime.jsxs(IllustrationSizeContext.Provider, { value: resolvedSize, children: [
        showIllustration && illustration && /* @__PURE__ */ jsxRuntime.jsx(
          "div",
          {
            "data-part": "illustration",
            role: decorative ? "presentation" : "img",
            "aria-hidden": decorative || void 0,
            "aria-label": !decorative ? effectiveAccessibleName : void 0,
            className: illustrationSizeClasses[resolvedSize],
            children: illustration
          }
        ),
        resolvedSize === "ExtraSmall" /* ExtraSmall */ ? /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "flex flex-col items-center gap-2", children: [
          (hasTitle || hasSubtitle) && /* @__PURE__ */ jsxRuntime.jsxs("div", { "data-part": "content", className: "text-center", children: [
            hasTitle && /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                "data-part": "title",
                className: titleVariants({ size: resolvedSize }),
                children: title || titleText
              }
            ),
            hasSubtitle && /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                "data-part": "subtitle",
                className: subtitleVariants({ size: resolvedSize }),
                children: subtitle || subtitleText
              }
            )
          ] }),
          showActions && children && /* @__PURE__ */ jsxRuntime.jsx("div", { "data-part": "actions", className: "flex gap-2", children })
        ] }) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
          (hasTitle || hasSubtitle) && /* @__PURE__ */ jsxRuntime.jsxs("div", { "data-part": "content", children: [
            hasTitle && /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                "data-part": "title",
                className: titleVariants({ size: resolvedSize }),
                children: title || titleText
              }
            ),
            hasSubtitle && /* @__PURE__ */ jsxRuntime.jsx(
              "div",
              {
                "data-part": "subtitle",
                className: subtitleVariants({ size: resolvedSize }),
                children: subtitle || subtitleText
              }
            )
          ] }),
          showActions && children && /* @__PURE__ */ jsxRuntime.jsx("div", { "data-part": "actions", className: "flex gap-2 mt-2", children })
        ] })
      ] })
    }
  );
}

exports.IllustratedMessage = IllustratedMessage;
exports.IllustrationSizeContext = IllustrationSizeContext;
exports.useIllustrationSizeContext = useIllustrationSizeContext;
